---
title: "初探mRNA、lncRNA联合分析之下游"
date: 2023-08-09T04:56:11Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Acdemic"]
---
初探mRNA、lncRNA联合分析之下游 by 生信菜鸟团
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><p data-tool="mdnice编辑器">上期转录组周更我们开启了转录组的进阶分析，<a href="https://mp.weixin.qq.com/s?__biz=MzUzMTEwODk0Ng==&amp;mid=2247514989&amp;idx=1&amp;sn=2502564754823455d1833597ebc4cb4b&amp;chksm=fa457e50cd32f746fdde14c55c2ea5faad7ce95b7298813249cedf4a06de4ca4c3be3f3b25d2&amp;scene=21&amp;cur_album_id=2545418429466607617#wechat_redirect" data-linktype="2">初探mRNA、lncRNA联合分析之上游</a>，完成了一个项目的上游分析过程，获得了基因表达矩阵和转录本表达矩阵</p><p data-tool="mdnice编辑器">基本分析流程：</p><ul data-tool="mdnice编辑器"><li><section>上游mRNA、lncRNA定量</section></li><li><section>mRNA、lncRNA差异表达分析</section></li><li><section>差异表达lncRNAs共表达mRNAs的富集分析</section></li><li><section>共表达网络、PPI</section></li></ul><p data-tool="mdnice编辑器">本周我们将完成下游的分析工作</p><hr data-tool="mdnice编辑器"><h3 data-tool="mdnice编辑器"><span></span>基因水平注释分类<span></span></h3><p data-tool="mdnice编辑器">虽然这个项目是在转录本水平上开展的研究，但既然我们拿到了基因表达矩阵，也干脆看一看一些基本情况，这个部分代码此处省略，基本上和后面的转录本水平对应代码，包括使用的封装函数，是一致的</p><p data-tool="mdnice编辑器">DEGs结果</p><p data-tool="mdnice编辑器">mRNA:</p><pre data-tool="mdnice编辑器"><span></span><code>&gt; deg_mrna &lt;- deg(filter_count[mRNA_index,])<br>Warning: some variables <span>in</span> design formula are characters, converting to factorsestimating size factors<br>estimating dispersions<br>gene-wise dispersion estimates<br>mean-dispersion relationship<br>final dispersion estimates<br>fitting model and testing<br><br>  down normal     up <br>   110  15061    380 <br></code></pre><p data-tool="mdnice编辑器">lncRNA:</p><pre data-tool="mdnice编辑器"><span></span><code>&gt; deg_lncrna &lt;- deg(filter_count[lncRNA_index,])<br>Warning: some variables <span>in</span> design formula are characters, converting to factorsestimating size factors<br>estimating dispersions<br>gene-wise dispersion estimates<br>mean-dispersion relationship<br>final dispersion estimates<br>fitting model and testing<br><br>  down normal     up <br>    22   1913     43 <br></code></pre><p data-tool="mdnice编辑器">可以发现基因水平和作者在转录水平得到的结果相差很多</p><p data-tool="mdnice编辑器">联系后面我们做的转录本差异表达结果，可以验证我们前面一篇推文中的结论</p><blockquote data-tool="mdnice编辑器"><p><a href="https://mp.weixin.qq.com/s?__biz=MzUzMTEwODk0Ng==&amp;mid=2247513115&amp;idx=1&amp;sn=dbcbb4ff2963b496d7420c6d9acbbf7e&amp;chksm=fa457126cd32f830efa9497bbe194215833e5e117fc88fb294ec2887c531f64cd69cf1488b2e&amp;scene=21&amp;cur_album_id=2545418429466607617#wechat_redirect" data-linktype="2">明明PCA区分非常好，但是差异基因数量很少？</a></p><p>...</p><p>那么转录本表达量差异分析和我们常见的基因有什么区别呢？</p><p>...</p><p>这篇文章虽然作者用的不是stringtie但其<strong>从同一数据集获得的差异表达结果中转录本确实更少</strong></p><p>...</p><p>与RNA测序转录本差异分析相比，基因水平的差异表达分析更稳健，并且在实验上更可行。然而，使用基因计数进行统计分析可以掩盖转录物水平的动态</p></blockquote><h3 data-tool="mdnice编辑器"><span></span>转录本水平注释分类<span></span></h3><h4 data-tool="mdnice编辑器"><span></span>读取数据<span></span></h4><p data-tool="mdnice编辑器">读取数据</p><pre data-tool="mdnice编辑器"><span></span><code>rm(list = ls())<br>library(edgeR)<br>library(DESeq2)<br>library(FactoMineR)<br>library(factoextra)<br>library(org.Hs.eg.db)<br>library(stringr)<br>library(stringi)<br>library(tidyverse)<br>library(ggplot2)<br>library(patchwork)<br>library(pheatmap)<br>library(VennDiagram)<br>library(RColorBrewer)<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>exp &lt;- read.table("6.exprs_mat/transcript_count_matrix.csv",header = T,sep = ",")<br>exp[1:4,1:4]<br></code></pre><p data-tool="mdnice编辑器">去重，低表达</p><pre data-tool="mdnice编辑器"><span></span><code>exp &lt;- na.omit(exp)<br>range(exp[,2:7])<br>exp$median &lt;- apply(exp[2:7],1,median)<br>exp &lt;- exp[order(exp$transcript_id,exp$median,decreasing = T),]<br><br>keep1 &lt;- !duplicated(exp$transcript_id)<br>exp &lt;- exp[keep1,]<br><br>keep2 &lt;- rowSums(exp[,2:7]&gt;0) &gt;= floor(0.5*ncol(exp[,2:7]))<br>exp &lt;- exp[keep2,]<br><br>table(duplicated(exp$transcript_id))<br>table(rowSums(exp[,2:7]&gt;0))<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>rownames(exp) &lt;- exp$transcript_id<br>exp &lt;- exp[,c(-1,-8)]<br>exp &lt;- exp[grepl("ENST",rownames(exp)),]<br>split_list &lt;- rownames(exp) %&gt;% str_split("\\.",simplify = T)<br>rownames(exp) &lt;- split_list[,1]<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>group_list &lt;-  c(rep("Ctrl",3),rep("ADS",3))<br>group_list<br># table(group_list)<br></code></pre><p data-tool="mdnice编辑器">filter_count</p><pre data-tool="mdnice编辑器"><span></span><code>filter_count &lt;- exp<br>express_cpm &lt;- log2(cpm(filter_count)+1)<br>range(express_cpm)<br></code></pre><p data-tool="mdnice编辑器">exp</p><pre data-tool="mdnice编辑器"><span></span><code>boxplot(express_cpm, las=2)<br>exp &lt;- normalizeBetweenArrays(express_cpm)<br>boxplot(exp, las=2)<br></code></pre><h4 data-tool="mdnice编辑器"><span></span>注释mRNA和lncRNA<span></span></h4><p data-tool="mdnice编辑器">本文所有id的转换都是用biomaRt包进行</p><pre data-tool="mdnice编辑器"><span></span><code>library(biomaRt)<br>mart&lt;-useMart("ensembl")<br>data=listDatasets(mart)<br>data2=useDataset("hsapiens_gene_ensembl",mart=mart)<br>listFilters(data2) %&gt;% View()<br><br>transcript_id &lt;- rownames(exp)<br># write.table(transcript_id,file = "transcript_id.txt",quote = F,sep = ",",col.names = F,row.names = F)<br># magic needed<br># annos &lt;- read.table("annos.txt",sep = ",",header = F)<br><br>annos &lt;- getBM(attributes=c('ensembl_transcript_id','biotype'),<br>       filters = 'ensembl_transcript_id', values = transcript_id,<br>       mart = data2)<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>table(annos$V2) %&gt;% sort() %&gt;% tail()<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>mRNA_index &lt;- rownames(exp) %in% annos$V1[annos$V2=="protein_coding"]<br>table(mRNA_index)<br>lncRNA_index &lt;- rownames(exp) %in% annos$V1[annos$V2=="lncRNA"]<br>table(lncRNA_index)<br></code></pre><h4 data-tool="mdnice编辑器"><span></span>PCA和聚类热图<span></span></h4><pre data-tool="mdnice编辑器"><span></span><code>draw_pca &lt;- function(exp){<br>  exp=t(exp)#画PCA图时要求是行名时样本名，列名时探针名，因此此时需要转换<br>  exp=as.data.frame(exp)#将matrix转换为data.frame <br>  library("FactoMineR")#画主成分分析图需要加载这两个包<br>  library("factoextra")  <br>  exp.pca &lt;- PCA(exp , graph = FALSE)#现在exp最后一列是group_list，需要重新赋值给一个exp.pca,这个矩阵是不含有分组信息的<br>  pca_p&lt;-fviz_pca_ind(exp.pca,<br>                   geom.ind = "point", # show points only (nbut not "text")<br>                   col.ind =  group_list, # color by groups <br>                   addEllipses = T,<br>                   legend.title = "Groups"<br>  )<br>  return(pca_p)<br>}<br><br>p1 &lt;- draw_pca(exp[mRNA_index,])<br>p2 &lt;- draw_pca(exp[lncRNA_index,])<br>p1+p2<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>draw_heatp &lt;- function(exp){<br>  cg=names(tail(sort(apply(exp,1,sd)),1000))#apply按行（'1'是按行取，'2'是按列取）取每一行的方差，从小到大排序，取最大的1000个<br>  library(pheatmap)<br>  pheatmap(exp[cg,],show_colnames =F,show_rownames = F) #对那些提取出来的1000个基因所在的每一行取出，组合起来为一个新的表达矩阵<br>  n=t(scale(t(exp[cg,]))) # 'scale'可以对log-ratio标准化数值进行归一化<br>  n[n&gt;2]=2 <br>  n[n&lt; -2]= -2<br>  n[1:4,1:4]<br>  pheatmap(n,show_colnames =F,show_rownames = F)<br>  <br>  # 绘制这些基因的热图，并将分组信息加入到热图中<br>  ac=data.frame(group=group_list)<br>  rownames(ac)=colnames(n)<br>  pheatmap(n,show_colnames =F,show_rownames = F,<br>           annotation_col=ac)<br>  pheat_p&lt;-pheatmap(n,show_colnames =F,show_rownames = F,<br>               annotation_col=ac,<br>               filename = 'heatmap_top1000_sd.png')<br>  return(pheat_p)<br>}<br><br>library(ggplotify)<br>p1 &lt;- draw_heatp(exp[mRNA_index,]) %&gt;% as.ggplot()<br>p2 &lt;- draw_heatp(exp[lncRNA_index,]) %&gt;% as.ggplot()<br>p1+p2<br></code></pre><p data-tool="mdnice编辑器">mRNA、lncRNA</p><figure data-tool="mdnice编辑器"><img data-ratio="0.6336842105263157" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9v8wiclvsCVMoC1YMNw0cOg77uZROj6Qibk5nqr1fTamNkxJ3cNoZXYIQ/640?wx_fmt=png" data-type="png" data-w="950" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9v8wiclvsCVMoC1YMNw0cOg77uZROj6Qibk5nqr1fTamNkxJ3cNoZXYIQ/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">sd前1000转录本：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.6198347107438017" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9KjVbjc8lLibtX2rVkS2uVcNMzdZnI8q8mIIJntia94aG2iaGmkzuTe0Kg/640?wx_fmt=png" data-type="png" data-w="968" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9KjVbjc8lLibtX2rVkS2uVcNMzdZnI8q8mIIJntia94aG2iaGmkzuTe0Kg/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">其实可以发现，我们的热图和作者一样，都有些奇怪，没有较好地聚在一起</p><figure data-tool="mdnice编辑器"><img data-ratio="1.0241730279898218" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9zMWsTfIZfJiacVY9qWtUEzV7fY75DicACers0sG4Yic9pfhKoEZ0w6tibg/640?wx_fmt=png" data-type="png" data-w="786" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9zMWsTfIZfJiacVY9qWtUEzV7fY75DicACers0sG4Yic9pfhKoEZ0w6tibg/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">但毕竟每组样本只有三个，PCA分组勉勉强强，还是接着往下做了</p><h4 data-tool="mdnice编辑器"><span></span>差异基因分析<span></span></h4><pre data-tool="mdnice编辑器"><span></span><code>deg &lt;- function(filter_count){<br>  # 默认counts所以要求整数 内部有log2<br>  #1 查看分组信息和表达矩阵数据<br>  exprSet &lt;- filter_count<br>  dim(exprSet)<br>  table(group_list)<br>  # 加载包<br>  library(DESeq2)<br>  #2 第一步，构建DESeq2的DESeq对象dds(建立DEseq数据矩阵)<br>  colData &lt;- data.frame(row.names=colnames(exprSet),group_list=group_list)<br>  colData<br>  dds &lt;- DESeqDataSetFromMatrix(countData = exprSet,<br>                                colData = colData,<br>                                design = ~ group_list)<br>  dds<br>  #3 第二步，进行差异表达分析<br>  dds2 &lt;- DESeq(dds)<br>  #4 提取差异分析结果，trt组对untrt组的差异分析结果<br>  tmp &lt;- results(dds2,contrast=c("group_list","ADS","Ctrl"))<br>  DEG_DESeq2 &lt;- as.data.frame(tmp[order(tmp$padj),])<br>  head(DEG_DESeq2)  # 联系之前的内容DEseq2和edge会存在p值为0的情况<br>  # 去除差异分析结果中包含NA值的行<br>  DEG_DESeq2 = na.omit(DEG_DESeq2)<br>  # 筛选上下调差异基因，设定阈值<br>  fc_cutoff &lt;-2.0<br>  fdr &lt;- 0.05#p值<br>  DEG_DESeq2$regulated &lt;- "normal"<br>  loc_up &lt;- intersect(which(DEG_DESeq2$log2FoldChange&gt;log2(fc_cutoff)),<br>                      which(DEG_DESeq2$pvalue&lt;fdr))<br>  loc_down &lt;- intersect(which(DEG_DESeq2$log2FoldChange&lt; (-log2(fc_cutoff))),<br>                        which(DEG_DESeq2$pvalue&lt;fdr))<br>  DEG_DESeq2$regulated[loc_up] &lt;- "up"<br>  DEG_DESeq2$regulated[loc_down] &lt;- "down"<br>  print(table(DEG_DESeq2$regulated))<br>  return(DEG_DESeq2)<br>}<br><br>deg_mrna &lt;- deg(filter_count[mRNA_index,])<br>deg_lncrna &lt;- deg(filter_count[lncRNA_index,])<br><br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.5083333333333333" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9ialalAjGBT1Dnm7nHibjz8aOU1kFianibRPrJiajjwE8TL5g1uohCYgzPTg/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9ialalAjGBT1Dnm7nHibjz8aOU1kFianibRPrJiajjwE8TL5g1uohCYgzPTg/640?wx_fmt=png"></figure><h4 data-tool="mdnice编辑器"><span></span>火山图<span></span></h4><pre data-tool="mdnice编辑器"><span></span><code>library(EnhancedVolcano)<br>draw_volc &lt;- function(deg_res){<br>EnhancedVolcano(deg_res,x='log2FoldChange',y='pvalue',<br>                lab=rownames(deg_res),<br>                pCutoff = 0.05,<br>                FCcutoff = log2(2))<br>}<br><br>p1 &lt;- draw_volc(deg_mrna)<br>p2 &lt;- draw_volc(deg_lncrna)<br>p1+p2<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.45185185185185184" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9Bo9Qa6606ZK2KRicKAfgg5S6CwMlY3HAavSWTkqDoG0NfZbFibqP4yZg/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9Bo9Qa6606ZK2KRicKAfgg5S6CwMlY3HAavSWTkqDoG0NfZbFibqP4yZg/640?wx_fmt=png"></figure><h4 data-tool="mdnice编辑器"><span></span>check<span></span></h4><p data-tool="mdnice编辑器">检查一下作者说的"Top differentially expressed"的转录本:</p><figure data-tool="mdnice编辑器"><img data-ratio="0.7444444444444445" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9r48JicFeZuMQwj4D7TAEy8zhkbhg7r1ARc444WLK06KePoOaTstcTicA/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9r48JicFeZuMQwj4D7TAEy8zhkbhg7r1ARc444WLK06KePoOaTstcTicA/640?wx_fmt=png"></figure><figure data-tool="mdnice编辑器"><img data-ratio="0.3111111111111111" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9UZwnPrtwx02hclzGrz1M4NSTibAZpE3WGjJR2t7EPmoWZDcpCdWwRBg/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9UZwnPrtwx02hclzGrz1M4NSTibAZpE3WGjJR2t7EPmoWZDcpCdWwRBg/640?wx_fmt=png"></figure><figure data-tool="mdnice编辑器"><img data-ratio="0.26944444444444443" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9MM8Ud4tAAIwVFqEEDNZ9icSllG73TdZJvTvR73utwbEJN5qJzP6VqAA/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9MM8Ud4tAAIwVFqEEDNZ9icSllG73TdZJvTvR73utwbEJN5qJzP6VqAA/640?wx_fmt=png"></figure><figure data-tool="mdnice编辑器"><img data-ratio="0.18796296296296297" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9GaTvAtkgsg1ICNfkRjibaPeuGUyibMicNv3UOgn8via1rWgjlmyyuDu9CQ/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9GaTvAtkgsg1ICNfkRjibaPeuGUyibMicNv3UOgn8via1rWgjlmyyuDu9CQ/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">差别很大</p><p data-tool="mdnice编辑器">但可能是由于这是在转录水平上</p><p data-tool="mdnice编辑器">手动查看我们的转录本差异表达结果中的top，发现排第一的ENST00000343067其实也是EPB41</p><figure data-tool="mdnice编辑器"><img data-ratio="0.2037037037037037" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9H4iatGdcFRjuxSLzhzMUD2fCauWjhgdXDQBE6XY5OuMCSDO3pPGxINQ/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9H4iatGdcFRjuxSLzhzMUD2fCauWjhgdXDQBE6XY5OuMCSDO3pPGxINQ/640?wx_fmt=png"></figure><figure data-tool="mdnice编辑器"><img data-ratio="0.31296296296296294" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX97yTEytSAbM5AVKuicjWztf3MpB8OpiaPb5iaS8eficiaQ1jr57o6IWic5o0A/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX97yTEytSAbM5AVKuicjWztf3MpB8OpiaPb5iaS8eficiaQ1jr57o6IWic5o0A/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">于是回到基因水平查看DEGs：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.5472222222222223" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9q7pwFgJluvbVLnWwzbk91thoSXLZz8LIricq1IAmycQUAZ0echZ87EQ/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9q7pwFgJluvbVLnWwzbk91thoSXLZz8LIricq1IAmycQUAZ0echZ87EQ/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">看看和原文差异表达结果logfc的散点图：</p><pre data-tool="mdnice编辑器"><span></span><code>library(data.table)<br>deg_mrna_au &lt;- fread("./author_supply/GSE209825_mRNA_ADS_vs_Normal.DE_ALL.annotaion.xls.gz")<br>deg_lncrna_au &lt;- fread("./author_supply/GSE209825_lncRNA_ADS_vs_Normal.DE_ALL.annotaion.xls.gz")<br><br>range(deg_mrna_au$`log2(foldchange)`)<br># 转换-Inf和Inf<br># mRNA<br>topfc &lt;- sort(deg_mrna_au$`log2(foldchange)`) %&gt;% unique() %&gt;% head()<br>tailfc &lt;- sort(deg_mrna_au$`log2(foldchange)`) %&gt;% unique() %&gt;% tail()<br>deg_mrna_au$`log2(foldchange)`[deg_mrna_au$`log2(foldchange)`==Inf] &lt;- tailfc[5]+1<br>deg_mrna_au$`log2(foldchange)`[deg_mrna_au$`log2(foldchange)`==-Inf] &lt;- topfc[2]-1<br># lncRNA<br>topfc &lt;- sort(deg_lncrna_au$`log2(foldchange)`) %&gt;% unique() %&gt;% head()<br>tailfc &lt;- sort(deg_lncrna_au$`log2(foldchange)`) %&gt;% unique() %&gt;% tail()<br>deg_lncrna_au$`log2(foldchange)`[deg_lncrna_au$`log2(foldchange)`==Inf] &lt;- tailfc[5]+1<br>deg_lncrna_au$`log2(foldchange)`[deg_lncrna_au$`log2(foldchange)`==-Inf] &lt;- topfc[2]-1<br><br>range(deg_mrna_au$`log2(foldchange)`)<br>range(deg_lncrna_au$`log2(foldchange)`)<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>overlap_mrna &lt;- intersect(rownames(deg_mrna), deg_mrna_au$transcript_id)<br>deg_lncrna_au$transcript_id &lt;- str_split(deg_lncrna_au$transcript_id,"\\.",simplify = T)[,1]<br>overlap_lncrna &lt;- intersect(rownames(deg_lncrna), deg_lncrna_au$transcript_id)<br><br>draw_scatter &lt;- function(overlap, deg_my, deg_au){<br>  logfc_my &lt;-  deg_my$log2FoldChange[match(overlap, rownames(deg_my))]<br>  logfc_author &lt;-  deg_au$`log2(foldchange)`[match(overlap, deg_au$transcript_id)]<br>  <br>  overlap_logfc &lt;- data.frame(<br>    genes = overlap,<br>    logfc_my,<br>    logfc_author<br>  )<br>  <br>  # 计算相关系数<br>  cor_1 &lt;- cor(logfc_author[logfc_author&gt;0],logfc_my[logfc_author&gt;0])<br>  cor_2 &lt;- cor(logfc_author[logfc_author&lt;0],logfc_my[logfc_author&lt;0])<br>  <br>  # basic scatterplot<br>  scatter_plot &lt;- ggplot(overlap_logfc, aes(x=logfc_my, y=logfc_author)) + <br>      geom_point()<br>  scatter_plot+<br>    annotate("text",2,2,size=6,label=round(cor_1,3),color="red")+<br>    annotate("text",-2,-2,size=6,label=round(cor_2,3),color="red")<br>}<br><br>p1 &lt;- draw_scatter(overlap_mrna, deg_mrna, deg_mrna_au)<br>p2 &lt;- draw_scatter(overlap_lncrna, deg_lncrna, deg_lncrna_au)<br>p1+p2<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.5566037735849056" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9kRm2xSKGSan9Z3nUickiaDFibVw3UedSlsdbdSnWdXhtldDmicomGJjPWA/640?wx_fmt=png" data-type="png" data-w="848" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9kRm2xSKGSan9Z3nUickiaDFibVw3UedSlsdbdSnWdXhtldDmicomGJjPWA/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">看看我们常规featureCounts定量</p><h2 data-tool="mdnice编辑器"><span></span>NCBI pipeline（check）</h2><p data-tool="mdnice编辑器">NCBI最近使用统一的pipeline：hisat2+featureCounts，对数据库中的bulk转录组数据进行了处理并公开了所有项目的原始计数矩阵+归一化计数矩阵+基因注释，非常方便</p><p data-tool="mdnice编辑器">详情参见：</p><blockquote data-tool="mdnice编辑器"><p>https://www.ncbi.nlm.nih.gov/geo/info/rnaseqcounts.html</p><p><a href="https://mp.weixin.qq.com/s?__biz=Mzk0MDI4MjM4NQ==&amp;mid=2247485068&amp;idx=1&amp;sn=4046392e7b1cd56885f259e411bc1b51&amp;scene=21#wechat_redirect" data-linktype="2">足不出户，GEO能进行RNA-seq差异表达分析啦（测试版）</a></p></blockquote><p data-tool="mdnice编辑器">我们这里获取本数据集的NCBI-generated RNA-seq count data</p><p data-tool="mdnice编辑器">https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE209825</p><p data-tool="mdnice编辑器">也进行我们上述流程看看</p><figure data-tool="mdnice编辑器"><img data-ratio="0.48055555555555557" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9XoBgkLKZkhcTGuQPyFaPFmaLIc1k2Ok7ibNWoFiaJHkdgf6PeS8Pq6Vw/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9XoBgkLKZkhcTGuQPyFaPFmaLIc1k2Ok7ibNWoFiaJHkdgf6PeS8Pq6Vw/640?wx_fmt=png"></figure><figure data-tool="mdnice编辑器"><img data-ratio="0.460762331838565" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9jmm8lmNHAlIxUhDiaibnq7TvIz6TKQ6OAZ1uAz5Kg32nUamIHNV8PRDg/640?wx_fmt=png" data-type="png" data-w="892" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9jmm8lmNHAlIxUhDiaibnq7TvIz6TKQ6OAZ1uAz5Kg32nUamIHNV8PRDg/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">NCBI给的其实是基因水平的定量结果</p><p data-tool="mdnice编辑器">可以发现数量关系和我们前面的结果是颠倒的：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.5509259259259259" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9CCbkg6jjEfVBfVzuGibfpoD0TyCaIoqsUox4v5TOjQcGOn2Tia9gflIw/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9CCbkg6jjEfVBfVzuGibfpoD0TyCaIoqsUox4v5TOjQcGOn2Tia9gflIw/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">于是手动检查EPB41这个基因：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.5712962962962963" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9ibhNicOnB6pibpl9khWd5tMuc9MwEoNgBk6KHa21vrCgDeNt58AzszBJA/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9ibhNicOnB6pibpl9khWd5tMuc9MwEoNgBk6KHa21vrCgDeNt58AzszBJA/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">发现确实是反的</p><p data-tool="mdnice编辑器">我们前面主要是对热图存疑，这里就主要看一下热图：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.7394451145958987" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9cHJCrDsHB6uKuKYKvdwPVMnHqGeG9ibiaibyoBiaLmWt8rWmbjDhvHsHIg/640?wx_fmt=png" data-type="png" data-w="829" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9cHJCrDsHB6uKuKYKvdwPVMnHqGeG9ibiaibyoBiaLmWt8rWmbjDhvHsHIg/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">也不咋地，所以这应该是实验样本或者测序数据本身的问题</p><h4 data-tool="mdnice编辑器"><span></span>寻找lncRNA共表达mRNA<span></span></h4><blockquote data-tool="mdnice编辑器"><p>This study analyzed the correlation between lncRNA and mRNA in the samples using the Pearson correlation coefficient method. The absolute values of correlation coefficients &gt;0.95 and p &lt; 0.001 were specified as screening criteria. Then, the biological functions of lncRNAs were predicted by performing a functional enrichment analysis on mRNAs.</p></blockquote><blockquote data-tool="mdnice编辑器"><p>In brief, the coexpressed mRNAs of each DE lncRNA were identified by the correlation with protein-coding gene expression, which was used for functional enrichment analysis.</p><p>...</p><figure><img data-ratio="0.537109375" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9ma51WMvzAaZQDHH4WmyUgo0d3f8aiaaiaicJXrLVXMZ9IyK4K8zP713NQ/640?wx_fmt=png" data-type="png" data-w="512" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9ma51WMvzAaZQDHH4WmyUgo0d3f8aiaaiaicJXrLVXMZ9IyK4K8zP713NQ/640?wx_fmt=png"></figure></blockquote><pre data-tool="mdnice编辑器"><span></span><code>mat_mrna &lt;- filter_count[mRNA_index,]<br>mat_lncrna &lt;- filter_count[lncRNA_index,]<br><br>up_lncrnas &lt;- rownames(deg_lncrna)[deg_lncrna$regulated=="up"]<br>down_lncrnas &lt;- rownames(deg_lncrna)[deg_lncrna$regulated=="down"]<br><br>get_co_mrna &lt;- function(lncrnas){<br>  <br>  co_mrna &lt;- c()<br>  <br>  for (i in seq(1,length(lncrnas))){<br>    <br>    for (j in seq(1, length(rownames(mat_mrna)))){<br>          res &lt;- cor.test(unlist(mat_lncrna[lncrnas[i],]),<br>                    unlist(mat_mrna[j,]),method = "p")<br>          if (!is.na(res$estimate) &amp; res$estimate&gt;0.95 &amp; res$p.value&lt;0.001){<br>            co_mrna &lt;- c(co_mrna, rownames(mat_mrna)[j])<br>          }<br>    }<br>    <br>  }<br>  <br>  return(co_mrna)<br>  <br>}<br><br>up_lncrnas_co_mrnas &lt;- unique(get_co_mrna(up_lncrnas))<br>down_lncrnas_co_mrnas &lt;- unique(get_co_mrna(down_lncrnas))<br><br></code></pre><p data-tool="mdnice编辑器">获取对应gene symbol</p><pre data-tool="mdnice编辑器"><span></span><code>save(up_lncrnas_co_mrnas,down_lncrnas_co_mrnas,file = "co_mrna.Rdata")<br># biomart magic<br># load("co_genes_ENTREZID.Rdata")<br>up_co_genes &lt;- getBM(attributes=c('ensembl_transcript_id','hgnc_symbol'),<br>                     filters = 'ensembl_transcript_id', up_lncrnas_co_mrnas,<br>                     mart = data2)<br>down_co_genes &lt;- getBM(attributes=c('ensembl_transcript_id','hgnc_symbol'),<br>                       filters = 'ensembl_transcript_id', down_lncrnas_co_mrnas,<br>                       mart = data2)<br><br>up_genes &lt;- unique(up_co_genes$entrezgene_id)<br>down_genes &lt;- unique(down_co_genes$entrezgene_id)<br></code></pre><h4 data-tool="mdnice编辑器"><span></span>富集分析<span></span></h4><p data-tool="mdnice编辑器">我们只看看lncRNA共表达mRNAmap到的基因富集分析情况，作者还对所有差异表达的mRNA进行了分析</p><pre data-tool="mdnice编辑器"><span></span><code>library(clusterProfiler)<br>library(ggthemes)<br>library(org.Hs.eg.db)<br>library(dplyr)<br>library(ggplot2)<br>library(stringr)<br>library(enrichplot)<br></code></pre><p data-tool="mdnice编辑器">1.GO 富集分析</p><pre data-tool="mdnice编辑器"><span></span><code>up_go &lt;- enrichGO(gene = up_genes,<br>                  OrgDb = org.Hs.eg.db,<br>                  ont = "ALL",<br>                  readable = TRUE)<br>down_go &lt;- enrichGO(gene = down_genes,<br>                  OrgDb = org.Hs.eg.db,<br>                  ont = "ALL",<br>                  readable = TRUE)<br>save(up_go,down_go,file="GO.Rdata")<br><br># 可视化<br>#条带图<br>p1&lt;-barplot(up_go)<br>p2 &lt;- barplot(down_go)<br>p1+p2<br><br></code></pre><pre data-tool="mdnice编辑器"><span></span><code># 气泡图<br>p1 &lt;- dotplot(up_go)<br>p2 &lt;- dotplot(down_go)<br>p1+p2<br><br>p1&lt;-dotplot(up_go, split = "ONTOLOGY",<br>            font.size = 10, showCategory = 5) + <br>  facet_grid(ONTOLOGY ~ ., space = "free_y",scales = "free_y") + <br>  scale_y_discrete(labels = function(x) str_wrap(x, width = 45))<br>p2 &lt;- dotplot(down_go, split = "ONTOLOGY",<br>            font.size = 10, showCategory = 5) + <br>  facet_grid(ONTOLOGY ~ ., space = "free_y",scales = "free_y") + <br>  scale_y_discrete(labels = function(x) str_wrap(x, width = 45))<br>p1+p2<br></code></pre><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247510072&amp;idx=1&amp;sn=533b74307a8018b6ad061ef89a6f35e4&amp;scene=21#wechat_redirect" data-linktype="2">一个通路居然在上下调基因集里面都富集到了</a></p><p data-tool="mdnice编辑器">GO:up&amp;down</p><figure data-tool="mdnice编辑器"><img data-ratio="0.4601851851851852" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9dpyHFzsAjTuoPtfl4PAfWfxpDicTqVTLEZruUk5jaDou4HxYrIT4oAw/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9dpyHFzsAjTuoPtfl4PAfWfxpDicTqVTLEZruUk5jaDou4HxYrIT4oAw/640?wx_fmt=png"></figure><figure data-tool="mdnice编辑器"><img data-ratio="0.5305555555555556" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9GkohT7jkwy0LWs4Sichb4cBjXudqtsW5Co1olnFgAUkrYkUdp5UlbzA/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9GkohT7jkwy0LWs4Sichb4cBjXudqtsW5Co1olnFgAUkrYkUdp5UlbzA/640?wx_fmt=png"></figure><figure data-tool="mdnice编辑器"><img data-ratio="0.5157407407407407" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9Up5iaDRK80AXLhztD5fkax9icHxovWDOXfsic2drqrG7XibvukHH2GCtzw/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9Up5iaDRK80AXLhztD5fkax9icHxovWDOXfsic2drqrG7XibvukHH2GCtzw/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">作者：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.5812807881773399" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX99FOyWnwTT4CiccfOG7SN3HuvIOxjAnticO0TQ9m3RVMwnVL5WSwLdW1w/640?wx_fmt=png" data-type="png" data-w="1015" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX99FOyWnwTT4CiccfOG7SN3HuvIOxjAnticO0TQ9m3RVMwnVL5WSwLdW1w/640?wx_fmt=png"></figure><hr data-tool="mdnice编辑器"><p data-tool="mdnice编辑器">2.KEGG 通路富集分析</p><pre data-tool="mdnice编辑器"><span></span><code>up_kegg &lt;- enrichKEGG(up_genes, organism = "hsa")<br>down_kegg &lt;- enrichKEGG(down_genes, organism = "hsa")<br>save(up_kegg,down_kegg,file = "KEGG.Rdata")<br></code></pre><p data-tool="mdnice编辑器">看看富集到了吗</p><pre data-tool="mdnice编辑器"><span></span><code>table(up_kegg@result$p.adjust&lt;0.05)<br>table(down_kegg@result$p.adjust&lt;0.05)<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>p1 &lt;- barplot(up_kegg,showCategory=10,title="KEGG Pathway",<br>              colorBy="p.adjust",font.size = 8,label_format = 15)+<br>  scale_fill_gradient(low="#DC0000B2",high ="#00A087B2" )<br>p2 &lt;- barplot(down_kegg,showCategory=10,title="KEGG Pathway",<br>              colorBy="p.adjust",font.size = 8,label_format = 15)+<br>  scale_fill_gradient(low="#DC0000B2",high ="#00A087B2" )<br>p1+p2<br></code></pre><p data-tool="mdnice编辑器">KEGG:up&amp;down</p><figure data-tool="mdnice编辑器"><img data-ratio="0.6658415841584159" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9sf3LpR5PaLcwoIdtU3yAQGEib4nzSTgMO75AORKGm5o2bWLtz9fkW2Q/640?wx_fmt=png" data-type="png" data-w="808" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9sf3LpR5PaLcwoIdtU3yAQGEib4nzSTgMO75AORKGm5o2bWLtz9fkW2Q/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">作者：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.41388888888888886" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX92iasQP3ibRMmwwcicmpwObWUBejk8uYDPFZY84n2eoA7WvuUr9KHicibtiaw/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX92iasQP3ibRMmwwcicmpwObWUBejk8uYDPFZY84n2eoA7WvuUr9KHicibtiaw/640?wx_fmt=png"></figure><hr data-tool="mdnice编辑器"><h2 data-tool="mdnice编辑器"><span></span>lncRNA–mRNA共表达网络</h2><blockquote data-tool="mdnice编辑器"><p>LncRNAs play a biological role by regulating mRNAs. A coexpression network of lncRNA/mRNA was constructed to investigate the potential interactions between lncRNA and mRNA, which could identify the key lncRNAs involved in ADS and their potential functional. This study analyzed the correlation between lncRNA and mRNA in the samples using the Pearson correlation coefficient method. The absolute values of correlation coefficients &gt;0.95 and p &lt; 0.001 were specified as screening criteria</p></blockquote><blockquote data-tool="mdnice编辑器"><p>Ultimately, eight interested lncRNAs were generated in the analysis, and a lncRNA-mRNA coexpression network was constructed for visualization</p><figure><img data-ratio="1.0607553366174056" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9zMnZZytvaQdlh4pOLRzBRAg3ld1vBLLl55ibLYnpIcRIr6XqdS5LiaNA/640?wx_fmt=png" data-type="png" data-w="609" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9zMnZZytvaQdlh4pOLRzBRAg3ld1vBLLl55ibLYnpIcRIr6XqdS5LiaNA/640?wx_fmt=png"></figure></blockquote><p data-tool="mdnice编辑器">这里作者只根据前面那样计算的相关性确定共表达的mRNA和lncRNA构建网络（所以可以看到mRNA之间没有连线）</p><p data-tool="mdnice编辑器">我们这里尝试用WGCNA寻找共表达模块（mRNA+lncRNA），看看结果能不能有overlap</p><p data-tool="mdnice编辑器">参考：</p><blockquote data-tool="mdnice编辑器"><p><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247503463&amp;idx=2&amp;sn=ddf07fa6d9aeeae0a8e30793ae245dc8&amp;chksm=9b4b8edcac3c07ca6e65aef51598da3c6813accacae28e6f3b4486d80788da180f6118970dee&amp;scene=21&amp;cur_album_id=1322384987060142080#wechat_redirect" data-linktype="2">4个发育时间点的总共12个鸡转录组测序样本的长非编码RNA的鉴定</a></p><p><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247523474&amp;idx=1&amp;sn=d879233dc7182ab9a9635e58ca68e8cf&amp;scene=21#wechat_redirect" data-linktype="2">WGCNA仅仅是划分基因模块，其它都是附加分析</a></p></blockquote><p data-tool="mdnice编辑器">WGCNA</p><pre data-tool="mdnice编辑器"><span></span><code>library(tidyverse)<br>library(edgeR)<br>library(WGCNA)<br>library(FactoMineR)<br>library(factoextra)  <br>library(tidyverse) # ggplot2 stringer dplyr tidyr readr purrr  tibble forcats<br>library(data.table) #多核读取文件<br></code></pre><p data-tool="mdnice编辑器">这时需要把mRNA和lncRNA的标准化后的cpm矩阵放在一起：</p><pre data-tool="mdnice编辑器"><span></span><code>express_cpm[1:4,1:4]<br>range(express_cpm)<br>group_list<br></code></pre><p data-tool="mdnice编辑器">PCA和聚类树参考前面，一组就三个样，懒得看了，还是不拿DEGs做WGCNA了</p><p data-tool="mdnice编辑器">将近12W个转录本太大了，取子集:MAD前2W的转录本，进行</p><pre data-tool="mdnice编辑器"><span></span><code>tmp &lt;- as.data.frame(express_cpm)<br>tmp$mad &lt;- apply(tmp,1,mad)<br>keep &lt;- tmp %&gt;% arrange(-mad) %&gt;% head(20000) %&gt;% rownames()<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code># 挑选最佳阈值power<br>R.sq_cutoff = 0.8  #设置R^2 cut-off<br>datExpr &lt;- t(express_cpm[keep,]) %&gt;% as.data.frame()<br>nGenes &lt;- ncol(datExpr)<br>nSamples &lt;- nrow(datExpr)<br>if(T){<br>  # Call the network topology analysis function<br>  #设置power参数选择范围 1到20，每次增加1，以及22到30，每次增加2<br>  # 为了在选择最佳power时，尽可能涵盖可能的取值范围<br>  powers &lt;- c(seq(1,20,by = 1), seq(22,30,by = 2)) <br>  # 用来选择最佳power参数的，<br>  # 输入参数包括表达数据datExpr、网络类型（这里是无向网络）、power取值的向量、R平方的截断值、以及是否输出过程信息<br>  # 输出结果为一个包含有关于不同power下拟合优度、均值连接度、平均k值等的信息表<br>  sft &lt;- pickSoftThreshold(datExpr, <br>                           networkType = "unsigned",<br>                           powerVector = powers, <br>                           RsquaredCut = R.sq_cutoff,  <br>                           verbose = 5)<br>  #SFT.R.sq &gt; 0.8 , slope ≈ -1<br>  # 将拟合优度图和均值连接度图保存至PDF文件中<br>  pdf("step2_power-value.pdf",width = 16,height = 12)<br>  # Plot the results: 寻找拐点，确认最终power取值<br>  par(mfrow = c(1,2));<br>  cex1 = 0.9;<br>  <br>  # Scale-free topology fit index as a function of the soft-thresholding power<br>  # 画出在不同power下拟合优度的趋势图，x轴为power，y轴为拟合优度（signed R2值）<br>  # 拟合优度要乘以正负1，这是因为负的拟合优度意味着模型拟合得更差<br>  plot(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],<br>       xlab="Soft Threshold (power)",ylab="Scale Free Topology Model Fit,signed R^2",type="n")<br>  # 在每个数据点上添加power的数值标注，颜色为红色<br>  text(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],<br>       labels=powers,cex=cex1,col="red")<br>  # this line corresponds to using an R^2 cut-off of h<br>  # 画一条水平的线，标注R平方的截断点，颜色为红色<br>  abline(h=R.sq_cutoff ,col="red")<br>  <br>  # Mean connectivity as a function of the soft-thresholding power<br>  # 画出在不同power下均值连接度的趋势图，x轴为power，y轴为均值连接度<br>  plot(sft$fitIndices[,1], sft$fitIndices[,5],<br>       xlab="Soft Threshold (power)",ylab="Mean Connectivity", type="n")<br>  # 在每个数据点上添加power的数值标注，颜色为红色<br>  text(sft$fitIndices[,1], sft$fitIndices[,5], labels=powers, cex=cex1,col="red")<br>  # 画一条水平的线，标注均值连接度的截断点，均值连接度大于100的边自动被过滤掉，颜色为红色<br>  abline(h=100,col="red")<br>  dev.off()<br>}  # 可看出拐点大致在power为16时出现，且各项数值基本满足挑选标准，因此设定power=16<br><br># 打印出在不同power下估计的最佳power取值<br>sft$powerEstimate  #查看估计的最佳power [1] NA<br>power = sft$powerEstimate<br><br># 若无向网络在power小于15或有向网络power小于30内，没有一个power值使<br># 无标度网络图谱结构R^2达到0.8且平均连接度在100以下，可能是由于<br># 部分样品与其他样品差别太大。这可能由批次效应、样品异质性或实验条件对<br># 表达影响太大等造成。可以通过绘制样品聚类查看分组信息和有无异常样品。<br># 如果这确实是由有意义的生物变化引起的，也可以使用下面的经验power值。<br><br># 如果最佳power取值为空<br># 根据数据样本数确定经验power取值：<br># 如果样本数小于20，则unsigned类型网格选择power=9，否则选择power=等一系列取值；<br># 如果样本数大于等于20，则unsigned类型网格选择power=6，否则选择power=12。这个默认给出较为通用的power取值方案，但是具体的数据情况下是否适用需再加以考虑。<br>if(is.na(power)){<br>  # 官方推荐 "signed" 或 "signed hybrid"<br>  # 为与原文档一致，故未修改<br>  type = "unsigned"<br>  nSamples=nrow(datExpr)<br>  power = ifelse(nSamples&lt;20, ifelse(type == "unsigned", 9, 18),<br>                 ifelse(nSamples&lt;30, ifelse(type == "unsigned", 8, 16),<br>                        ifelse(nSamples&lt;40, ifelse(type == "unsigned", 7, 14),<br>                               ifelse(type == "unsigned", 6, 12))  <br>                 )<br>  )<br>}<br># 将选择好的最佳power值和soft-thresholding的计算结果保存至RData文件中<br>save(sft, power, file = "step2_power_value.Rdata")<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>######一步法构建加权共表达网络，识别基因模块######<br>if(T){<br>  cor &lt;- WGCNA::cor<br>  # blockwiseModules函数构建加权共表达网络，并识别基因模块<br>  net &lt;- blockwiseModules(<br>    datExpr,  # datExpr是包含基因表达数据的矩阵<br>    power = power,  # power是上一步计算得到的软阈值幂次<br>    maxBlockSize = ncol(datExpr),  # 计算机能处理的最大模块的基因数量<br>    corType = "pearson", #计算相关性的方法默认为"pearson","bicor"则更能考虑离群点的影响<br>    networkType = "unsigned",  # 无向网络 <br>    TOMType = "unsigned",   # 建立邻接矩阵和TOM矩阵时，是否考虑正负相关性，unsigned表示只考虑正相关性<br>    # minModuleSize指每个模块中基因的最少数目<br>    minModuleSize = 10,    ##越大模块越少<br>    # mergeCutHeight指合并模块的阈值<br>    mergeCutHeight = 0.25, ##越大模块越少<br>    numericLabels = TRUE,  # 返回数字作为模块的名称，后面可以再转换为颜色<br>    saveTOMs = F,  # 是否存储TOM矩阵，TOM矩阵计算最耗费时间的步骤之一，如果需要多次分析，可以将其存储起来供后续使用<br>    verbose = 3  # 控制输出信息的详细程度，数值越大输出的信息越多<br>  )<br>  # 统计每个模块中基因的数量<br>  table(net$colors) <br>  # 0  1  2  3  4 <br>  # 18 56 22 16 10<br>  # power: 上一步计算的软阈值<br>  # maxBlockSize:计算机能处理的最大模块的基因数量(默认5000),16G内存可以处理2万个，<br>  # 计算资源允许的情况下最好放在一个block里面。<br>  # corType：计算相关性的方法；可选pearson(默认)，bicor。后者更能考虑离群点的影响。<br>  # networkType:计算邻接矩阵时，是否考虑正负相关性；默认为"unsigned",可选"signed", "signed hybrid"<br>  # TOMType：计算TOM矩阵时，是否考虑正负相关性；默认为"signed",可选"unsigned"。但是根据幂律转换的邻接矩阵(权重)的非负性，所以认为这里选择"signed"也没有太多的意义。<br>  # numericLabels: 返回数字而不是颜色作为模块的名字，后面可以再转换为颜色<br>  # saveTOMs：最耗费时间的计算，可存储起来供后续使用，<br>  # mergeCutHeight: 合并模块的阈值，越大模块越少,一般为0.25<br>  # minModuleSize: 每个模块里最少放多少个基因，设定越大模块越少<br>  # 输出结果根据模块中基因数目的多少，降序排列，依次编号为 `1-最大模块数`。<br>  # **0 (grey)**表示**未**分入任何模块的基因。<br>}<br><br>## 模块可视化，层级聚类树展示各个模块<br># 模块构建完成后，将模块颜色labels转换为可视化所需的颜色，最后用层级聚类树展示各个模块<br>if(T){<br>  # Convert labels to colors for plotting<br>  moduleColors &lt;- labels2colors(net$colors)<br>  table(moduleColors)<br>  # Plot the dendrogram and the module colors underneath<br>  pdf("step3_genes-modules_ClusterDendrogram.pdf",width = 16,height = 12)<br>  plotDendroAndColors(net$dendrograms[[1]], moduleColors[net$blockGenes[[1]]],<br>                      "Module colors",<br>                      dendroLabels = FALSE, hang = 0.03,<br>                      addGuide = TRUE, guideHang = 0.05)<br>  dev.off()<br>}<br># 最后将得到的模块信息和模块颜色保存起来，方便后续的分析和可视化使用<br>save(net, moduleColors, file = "step3_genes_modules.Rdata")<br></code></pre><p data-tool="mdnice编辑器">关联基因模块与表型</p><pre data-tool="mdnice编辑器"><span></span><code># 分组<br>datTraits &lt;- data.frame(<br>  row.names = rownames(datExpr),<br>  exp_group = group_list,<br>  exp_group_No = rep(c(1,2),each=3)<br>)<br>datTraits<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>if(T){<br>  datTraits$exp_group &lt;- as.factor(datTraits$exp_group)<br>  # 表达矩阵对应的表型变量为设计矩阵，并设置组别<br>  design &lt;- model.matrix(~0+datTraits$exp_group)<br>  colnames(design) &lt;- levels(datTraits$exp_group) #获取组别信息<br>  # 计算所有模块的特征基因eigengenes.<br>  MES0 &lt;- moduleEigengenes(datExpr,moduleColors)$eigengenes<br>  # MEblue      MEbrown       MEgrey   MEturquoise     MEyellow<br>  # LH_41_hg19 -0.054224196 -0.113190330  0.017756936 -0.0596382729  0.202767908<br>  # LH_42_hg19 -0.088915385 -0.133718403 -0.078586736 -0.0917636118 -0.026137421<br>  MEs &lt;- orderMEs(MES0)  #将相关的特征基因放在一起<br>  # &gt; dim(MEs)<br>  # [1] 40  5<br>  # &gt; MEs    # 其实就换了下module的顺序<br>  # MEblue     MEyellow      MEbrown   MEturquoise       MEgrey<br>  # LH_41_hg19 -0.054224196  0.202767908 -0.113190330 -0.0596382729  0.017756936<br>  # LH_42_hg19 -0.088915385 -0.026137421 -0.133718403 -0.0917636118 -0.078586736<br>  <br>  # 可操作地方！！！！！<br>  # 计算模块与表型之间的相关性矩阵<br>  moduleTraitCor &lt;- cor(MEs,design,use = "p")<br>  # &gt; moduleTraitCor<br>  # [,1]       [,2]<br>  # MEblue       0.6998699 -0.6998699<br>  # MEyellow    -0.4405819  0.4405819<br>  # MEbrown      0.4076938 -0.4076938<br>  # MEturquoise  0.5932559 -0.5932559<br>  # MEgrey       0.1919625 -0.1919625<br>  moduleTraitPvalue &lt;- corPvalueStudent(moduleTraitCor,nSamples)<br>  # [,1]         [,2]<br>  # MEblue      5.014994e-07 5.014994e-07<br>  # MEyellow    4.437355e-03 4.437355e-03<br>  # MEbrown     9.020902e-03 9.020902e-03<br>  # MEturquoise 5.474019e-05 5.474019e-05<br>  # MEgrey      2.353641e-01 2.353641e-01<br>  <br>  # 设定热图的文本标签为矩阵中每个元素的相关性值及其对应的显著性水平<br>  textMatrix &lt;- paste0(signif(moduleTraitCor,2),"\n(",<br>                       signif(moduleTraitPvalue,1),")")<br>  # &gt; textMatrix<br>  # [1] "0.7\n(5e-07)"   "-0.44\n(0.004)" "0.41\n(0.009)"  "0.59\n(5e-05)" <br>  # [5] "0.19\n(0.2)"    "-0.7\n(5e-07)"  "0.44\n(0.004)"  "-0.41\n(0.009)"<br>  # [9] "-0.59\n(5e-05)" "-0.19\n(0.2)" <br>  # signif 保留位数<br>  # dim函数用于更改或显示矩阵的维数，包括行数和列数。<br>  # 在这里，使用dim(textMatrix)函数来设置textMatrix的维度，<br>  # 其行数和列数与模块与表型相关性的矩阵相同，以确保文本正确地添加到每个网格中。<br>  dim(textMatrix) &lt;- dim(moduleTraitCor)<br>  # [,1]             [,2]            <br>  # [1,] "0.7\n(5e-07)"   "-0.7\n(5e-07)" <br>  # [2,] "-0.44\n(0.004)" "0.44\n(0.004)" <br>  # [3,] "0.41\n(0.009)"  "-0.41\n(0.009)"<br>  # [4,] "0.59\n(5e-05)"  "-0.59\n(5e-05)"<br>  # [5,] "0.19\n(0.2)"    "-0.19\n(0.2)" <br>  <br>  pdf("step4_Module-trait-relationship_heatmap.pdf",<br>      width = 2*length(colnames(design)), <br>      height = 0.6*length(names(MEs)) )<br>  par(mar=c(5, 9, 3, 3)) #留白：下、左、上、右<br>  # 绘制<br>  labeledHeatmap(Matrix = moduleTraitCor,  # 要绘制热图的矩阵<br>                 xLabels = colnames(design),  # X轴上的标签<br>                 yLabels = names(MEs),  # Y轴上的标签<br>                 ySymbols = names(MEs),  # 线条的标记符号<br>                 colorLabels = F,  # 颜色标签 如果为TRUE，则在热图下方添加一个注释区，表示颜色对应的值的范围<br>                 colors = blueWhiteRed(50),  # 颜色网格的颜色图谱<br>                 textMatrix = textMatrix,  # 添加到每个网格中的文本（可以是数字或字符）<br>                 setStdMargins = F,  # 是否将margins调整为标准大小<br>                 cex.text = 0.5,  # 文本大小<br>                 zlim = c(-1,1),   # 颜色图谱的值范围<br>                 main = "Module-trait relationships")  # 热图的标题<br>  dev.off()<br>  save(design, file = "step4_design.Rdata")<br>}<br></code></pre><p data-tool="mdnice编辑器">取MAD前2W转录本进行：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.7324074074074074" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9S87W3Ft1578MVWcnLIENrLibYVnMMlUn1mrvro8icDN0mVGYM5GMicWUA/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9S87W3Ft1578MVWcnLIENrLibYVnMMlUn1mrvro8icDN0mVGYM5GMicWUA/640?wx_fmt=png"></figure><figure data-tool="mdnice编辑器"><img data-ratio="0.2219626168224299" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9ia5YY74fKseqD1WsCLJs9sE5G3smos3z6LI2Xv4I0iaUZZmicRgYuoUpA/640?wx_fmt=png" data-type="png" data-w="856" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9ia5YY74fKseqD1WsCLJs9sE5G3smos3z6LI2Xv4I0iaUZZmicRgYuoUpA/640?wx_fmt=png"></figure><figure data-tool="mdnice编辑器"><img data-ratio="0.7605633802816901" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9mYg4R33fHswNJrMuraP58SUsibiapGKYUY4feCOkj8LG5VBPJmc4mwIQ/640?wx_fmt=png" data-type="png" data-w="781" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9mYg4R33fHswNJrMuraP58SUsibiapGKYUY4feCOkj8LG5VBPJmc4mwIQ/640?wx_fmt=png"></figure><figure data-tool="mdnice编辑器"><img data-ratio="0.6666666666666666" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9yqWpH11HE1iaX76icZuGbLULAqQ2F8wU6dw4RCric6hM57qLqny7rJfrw/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9yqWpH11HE1iaX76icZuGbLULAqQ2F8wU6dw4RCric6hM57qLqny7rJfrw/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">可以发现MRgreen模块与表型关联很强</p><p data-tool="mdnice编辑器">可以进一步找hub，这里就直接看看作者的网络中的34个节点在不在这MEgreen 958个转录本对应的gene symbol中</p><p data-tool="mdnice编辑器"><strong>id转换:</strong></p><blockquote data-tool="mdnice编辑器"><p><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247492395&amp;idx=1&amp;sn=b8228d426ea5bf728c70acebef1b4079&amp;chksm=9b4ba390ac3c2a86edc5d22b31a5fdc9c21a37f069a45916d7f523ce31ca91b684e066e3cb9c&amp;scene=21&amp;cur_album_id=1322384987060142080#wechat_redirect" data-linktype="2">lncRNA的一些基础知识</a></p><figure><img data-ratio="1.3304093567251463" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9F40tVSo3eWHMAibibDZkM81vJK8VTqbW0FyE5y1GTQCe4rZ5kBiauIJhw/640?wx_fmt=png" data-type="png" data-w="684" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9F40tVSo3eWHMAibibDZkM81vJK8VTqbW0FyE5y1GTQCe4rZ5kBiauIJhw/640?wx_fmt=png"></figure><p><img data-ratio="1.0562571756601606" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9CwEuE2FfgvWFvf42jX05rXmReckIDRRc17Fum1AOFIFdHFlNRj6Y6A/640?wx_fmt=png" data-type="png" data-w="871" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9CwEuE2FfgvWFvf42jX05rXmReckIDRRc17Fum1AOFIFdHFlNRj6Y6A/640?wx_fmt=png"><img data-ratio="0.9663032705649157" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9c1BBfHiapQ2R01YfnALEAxEBNUsicKcwRqwsVrOmvQ2zrLhe5N9USXKQ/640?wx_fmt=png" data-type="png" data-w="1009" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9c1BBfHiapQ2R01YfnALEAxEBNUsicKcwRqwsVrOmvQ2zrLhe5N9USXKQ/640?wx_fmt=png"></p><p>LNCipedia进不去</p></blockquote><p data-tool="mdnice编辑器">询问曾老师：这是软件给的编号</p><p data-tool="mdnice编辑器">文章作者也没给出对应的ENSEMBL id</p><p data-tool="mdnice编辑器">看看mRNA算了</p><pre data-tool="mdnice编辑器"><span></span><code>&gt; aut_hub_mrna &lt;- c(<span>"SWT1"</span>,<span>"AXIN2"</span>,<span>"HARS2"</span>,<span>"MMS19"</span>,<span>"HYLS1"</span>,<span>"PTRH1"</span>,<span>"MXRA7"</span>,<span>"PUF60"</span>,<span>"ASAH1"</span>,<span>"CLEC16A"</span>,<span>"NFIX"</span>,<span>"ZNF638"</span>,<span>"FIGNL1"</span>,<span>"BID"</span>,<span>"CNOT2"</span>,<span>"PTPN12"</span>,<span>"CD44"</span>,<span>"NCAM1"</span>,<span>"CANT1"</span>,<span>"DUSP14"</span>,<span>"REPS1"</span>,<span>"MFAP3L"</span>,<span>"EPB41"</span>,<span>"TCF4"</span>,<span>"IFNGR1"</span>,<span>"RNASEH2B"</span>)<br>&gt; table(aut_hub_mrna %<span>in</span>% annos<span>$hgnc_symbol</span>)<br><br>FALSE  TRUE <br>   21     5 <br>&gt; aut_hub_mrna[aut_hub_mrna %<span>in</span>% annos<span>$hgnc_symbol</span>]<br>[1] <span>"NFIX"</span>     <span>"EPB41"</span>    <span>"TCF4"</span>     <span>"IFNGR1"</span>   <span>"RNASEH2B"</span><br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.8217967599410898" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9ibH5DLPo76OyopS5km2nuUHxWZu0FKh77pz6vFsL62vX1rV5I6UsKVA/640?wx_fmt=png" data-type="png" data-w="679" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9ibH5DLPo76OyopS5km2nuUHxWZu0FKh77pz6vFsL62vX1rV5I6UsKVA/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">虽然我们只取MAD前2W转录本进行WGCNA，发现作者的mRNA节点中只有5个在我们WGCNA结果与表型最显著的模块中，但是可以清晰地发现这五个基因在作者共表达网络中的关系</p><p data-tool="mdnice编辑器">顺便看看PPI 的hub gene：</p><pre data-tool="mdnice编辑器"><span></span><code>&gt; PPI_hub &lt;- c(<span>"MYC"</span>,<span>"EGF"</span>,<span>"AKT1"</span>,<span>"SUPT20H"</span>,<span>"UBA52"</span>)<br>&gt; PPI_hub[PPI_hub %<span>in</span>% annos<span>$hgnc_symbol</span>]<br>[1] <span>"SUPT20H"</span> <span>"UBA52"</span> <br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.9748677248677249" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9lyMlMgylK940fg1lOvBhjO1XK0ZIjtfyOOVvOgIxmQldCicFLtBZwaw/640?wx_fmt=png" data-type="png" data-w="756" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicfLtjZboEL3oOiagxggyJX9lyMlMgylK940fg1lOvBhjO1XK0ZIjtfyOOVvOgIxmQldCicFLtBZwaw/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">所以无论是共表达网络、PPI还是WGCNA都是我们寻找hub gene进行narrow down的手段</p><hr data-tool="mdnice编辑器"><p data-tool="mdnice编辑器">以上就是初探mRNA、lncRNA联合分析下游的全部内容，我们部分复现了作者的流程并联系一些参考资料和其它手段进行了学习、解读，但这篇文章只对目前已知lncRNA定量，不涉及鉴定的新lncRNA的流程，毕竟这是人的数据，下周我们将一起看看lncRNA鉴定的全套流程</p></section><p><br></p><p><mp-style-type data-value="10000"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/wV7QDUclhU6gvSybjsE5Lw",target="_blank" rel="noopener noreferrer">原文链接</a>
