---
title: "单细胞pseudobulk分析，一文就够了"
date: 2023-08-24T13:15:23Z
draft: ["false"]
tags: [
  "fetched",
  "生信随笔"
]
categories: ["Acdemic"]
---
单细胞pseudobulk分析，一文就够了 by 生信随笔
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><p data-tool="mdnice编辑器">2021年NC发文《Confronting false discoveries in single-cell differential expression》，评测了当前单细胞转录组数据差异分析的14种方法，例如pseudobulks，Wilcox，DESeq2和MAST等。文章指出pseudobulks方法要优于其他single-cell分析方法，并指出现在的很多发表的差异分析方法是错误的，会有太多的假阳性。</p><p data-tool="mdnice编辑器">根据我的理解来说的话，由于单细胞数量远大于传统Bulk RNA-seq的样本量，因此Limma和DESeq2是非常不适合单细胞数据的，极易造成假阳性问题，具体原理可参考熊写的：<a href="https://mp.weixin.qq.com/s?__biz=MzA5OTM1ODg5NA==&amp;mid=2649679823&amp;idx=1&amp;sn=491fffd5ba95b265bf6f51c1ffce8891&amp;scene=21#wechat_redirect" data-linktype="2">熊读文献｜别再用DEseq2和edgeR进行大样本差异表达基因分析了</a>。所以我是推荐使用FindAllMarkers()函数的Wilcox.test算法。至于MAST算法，本身细胞量多的话FindAllMarkers()函数的Wilcox.test算法已经很花资源和时间了，MAST算法固然是会提升差异基因的准确性，但是会进一步消耗极大的时间和资源。关于pseudobulks的话，非常合适当前单细胞大样本量的数量情况，一是能够加速运算，二是大大减少假阳性问题。难点在于代码层面实现比较困难，一是要得到指定分组和/或指定细胞类型的pseudobulks表达矩阵，二是进行类似Bulk RNA-seq的Limma或DESeq2差异分析。具体来说，pseudobulk差异分析需要足够的两两比较的样本数量，否则结果会出现较强的假阴性现象。<strong>如果用户的样本量不大时，</strong>一是<strong>建议使用传统的FindAllMarkers()函数的Wilcox.test算法</strong>；二是使用<strong>下文的MetaCell结合DESeq2差异分析</strong>。</p><p data-tool="mdnice编辑器"><strong>在此，本文详细介绍如何实现pseudobulks，DESeq2及MetaCell，包装了3个函数，以方便用户使用。本文目录包括：</strong></p><ul data-tool="mdnice编辑器"><li><section>一. 基于单细胞数据得到pseudobulks表达矩阵；</section></li><li><section>二. 基于pseudobulks表达矩阵进行DESeq2差异分析；</section></li><li><section>三. MetaCell结合DESeq2差异分析；</section></li><li><section>四. 附录：测试数据及自编函数custom_seurat_functions_for_pseudobulks.R</section></li></ul><blockquote data-tool="mdnice编辑器"><p>值得介绍的是MetaCell算法，张泽民老师的文章有提到：To gain a concise description of the observed phenotypic diversity, we used our SEACells algorithm (31) (see methods) to calculate metacells separately within each sample. The metacell approach (24) aggregates similar single cells into a set of distinct, highly granular cell states such that any differences between cells within a metacell are likely technical rather than biological. Aggregating counts within each metacell generates a robust , full transcriptomic quantification for each cell state, thus mitigating noise and sparsity in scRNA-seq data. An added benefit of metacells is the enumeration of observed cell states that are more amenable to comparison.</p><p>这个MetaCell的观念就是，把相似的细胞看做一类，以“浓缩”后的矩阵进行下游分析。MetaCell结合DESeq2差异分析找Markers或Signature的方法也非常新颖。此外，MetaCell之后的表达矩阵可以接harmony，CCA等整合分析，张老师的文章证明这种策略可以更好的整合单细胞大样本数据。</p></blockquote><p data-tool="mdnice编辑器"><strong>拓展阅读：</strong></p><ul data-tool="mdnice编辑器"><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247499917&amp;idx=1&amp;sn=0d63f4b51cf593b2e0ac3130275f728c&amp;scene=21#wechat_redirect" data-linktype="2">单细胞差异分析方法评测</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247517945&amp;idx=1&amp;sn=babf5948237875e5b8e8cf005d480b03&amp;scene=21#wechat_redirect" data-linktype="2">单细胞层面的表达量差异分析到底如何做</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzA5OTM1ODg5NA==&amp;mid=2649679823&amp;idx=1&amp;sn=491fffd5ba95b265bf6f51c1ffce8891&amp;scene=21#wechat_redirect" data-linktype="2">熊读文献｜别再用DEseq2和edgeR进行大样本差异表达基因分析了</a></section></li></ul><h3 data-tool="mdnice编辑器">一. 基于单细胞数据得到pseudobulks表达矩阵</h3><h4 data-tool="mdnice编辑器">Step1. 加载R包和读入测试数据</h4><pre data-tool="mdnice编辑器"><span></span><code><span>library</span>(Seurat)<br><span>library</span>(readr)<br><span>library</span>(dplyr)<br><span>library</span>(stringr)<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code><span>#### 1.读入数据</span><br>seurat.data = read_rds(<span>"./pseudoBulk_test.data.rds"</span>)<br>Idents(seurat.data) = <span>"celltype"</span><br>DimPlot(seurat.data)<br><br>table(seurat.data$celltype)<br><span># T_cell       NK  Myeloid   B_cell Platelet      Hpc </span><br><span># 1500     1500     1500     1500      244       20 </span><br><br>table(seurat.data$orig.ident)<br><span># A1  CT3  HC2   P2  SC4 </span><br><span># 1918  949 1328 1048 1021 </span><br></code></pre><p data-tool="mdnice编辑器">seurat.data包5个样本，5种细胞类型。</p><h4 data-tool="mdnice编辑器">Step 2. 计算pseudobulks表达矩阵</h4><p data-tool="mdnice编辑器">这里我针对pseudobulks分析写了2个自编函数，保存在<code>custom_seurat_functions_for_pseudobulks.R</code>文件中：</p><blockquote data-tool="mdnice编辑器"><p><strong>测试数据、自编函数和custom_seurat_functions_for_pseudobulks.R见文末。</strong></p></blockquote><h5 data-tool="mdnice编辑器">例1：计算单样本的pseudobulks表达矩阵（不区分细胞类型）</h5><pre data-tool="mdnice编辑器"><span></span><code><span>source</span>(<span>"custom_seurat_functions_for_pseudobulks.R"</span>)<br><br><span># By samples</span><br>res.pseudo = pseudoBulk(Seurat.data = seurat.data,<br>                        group = <span>"orig.ident"</span>,<br>                        celltype = <span>NA</span> )<br>head(res.pseudo)<br>meta.data = data.frame(sampleID = colnames(res.pseudo),<br>                       row.names = colnames(res.pseudo))<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.45690672963400236" data-src="https://mmbiz.qpic.cn/mmbiz/fTW9zRI3eqWdiaJ4k1KdOw9foiaaydWZDwDRFiaodhd14ABujtq21pclGFZlDNJJYEB76FmZciaQLibsn96goicibemkA/640?wx_fmt=other" data-type="other" data-w="847" src="https://mmbiz.qpic.cn/mmbiz/fTW9zRI3eqWdiaJ4k1KdOw9foiaaydWZDwDRFiaodhd14ABujtq21pclGFZlDNJJYEB76FmZciaQLibsn96goicibemkA/640?wx_fmt=other"><figcaption>image-20230203152049000</figcaption></figure><p data-tool="mdnice编辑器">列名是<strong>样本名称</strong>。</p><h5 data-tool="mdnice编辑器">例2：计算单样本每一个细胞类型的pseudobulks表达矩阵（区分细胞类型）</h5><pre data-tool="mdnice编辑器"><span></span><code>res.pseudo2 = pseudoBulk(Seurat.data = seurat.data,<br>                        group = <span>"orig.ident"</span>,<br>                        celltype = <span>"celltype"</span> )<br>head(res.pseudo2)<br>meta.data2 = data.frame(sampleID = str_split(names(res.pseudo2),<span>"\\."</span>,simplify = <span>T</span>)[,<span>1</span>],<br>                        celltype = str_split(names(res.pseudo2),<span>"\\."</span>,simplify = <span>T</span>)[,<span>2</span>],<br>                        row.names = colnames(res.pseudo2))<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.44960474308300397" data-src="https://mmbiz.qpic.cn/mmbiz/fTW9zRI3eqWdiaJ4k1KdOw9foiaaydWZDwWKR8Acl5ACNaEiaKlJuj0rvxmQNfAMKG7Wk5TEK3NRohib3EoWrdLMvQ/640?wx_fmt=other" data-type="other" data-w="1012" src="https://mmbiz.qpic.cn/mmbiz/fTW9zRI3eqWdiaJ4k1KdOw9foiaaydWZDwWKR8Acl5ACNaEiaKlJuj0rvxmQNfAMKG7Wk5TEK3NRohib3EoWrdLMvQ/640?wx_fmt=other"><figcaption>image-20230203164849116</figcaption></figure><p data-tool="mdnice编辑器">列名是<strong>样本名称+细胞类型</strong></p><pre data-tool="mdnice编辑器"><span></span><code>head(meta.data2)<br><span>#                 sampleID celltype</span><br><span># HC2.NK            HC2       NK</span><br><span># HC2.Myeloid       HC2  Myeloid</span><br><span># HC2.B_cell        HC2   B_cell</span><br><span># HC2.T_cell        HC2   T_cell</span><br><span># HC2.Platelet      HC2 Platelet</span><br><span># HC2.Hpc           HC2      Hpc</span><br></code></pre><h3 data-tool="mdnice编辑器">二. 基于pseudobulks表达矩阵进行DESeq2差异分析</h3><p data-tool="mdnice编辑器">根据单细胞得到的pseudobulks表达矩阵，本质上类似Bulk RNA-seq的表达矩阵，因此推荐使用DESeq2进行差异分析，这里我写了一个DESeq2差异分析函数，便于进行差异分析：</p><pre data-tool="mdnice编辑器"><span></span><code>get.DESeq2 = <span>function</span>(input.data, <span>#输入目标表达矩阵</span><br>                      group_list, <span>#设置分组，因子数据</span><br>                      contrast = <span>NA</span>,<br>                      NA.remove = <span>TRUE</span>, <span>#DESeq2会产生NA值， NA.remove = TRUE移除NA</span><br>                      <span>...</span>){<br>  <span>if</span>(is.na(contrast)){<br>    contrast = rev(levels(group_list))<br>  }<br>  <span># 第一步，构建DESeq2的DESeq对象</span><br>  colData &lt;- data.frame(row.names=colnames(input.data),group_list=group_list)<br>  dds &lt;- DESeqDataSetFromMatrix(countData = input.data,colData = colData,design = ~ group_list)<br>  <br>  <span># 第二步，进行差异表达分析</span><br>  dds2 &lt;- DESeq(dds)<br>  <br>  <span># 提取差异分析结果，trt组对untrt组的差异分析结果</span><br>  table(group_list)<br>  tmp &lt;- results(dds2,contrast=c(<span>"group_list"</span>,contrast))<br>  <span># tmp &lt;- results(dds2)</span><br>  DEG_DESeq2 &lt;- as.data.frame(tmp[order(tmp$padj),])<br>  head(DEG_DESeq2)<br>  <br>  <span>if</span>(NA.remove){<br>    <span># 去除差异分析结果中包含NA值的行</span><br>    DEG_DESeq2 = na.omit(DEG_DESeq2)<br>  }<br>  DEG_DESeq2$Gene = row.names(DEG_DESeq2)<br>  <span>return</span>(DEG_DESeq2)<br>}<br></code></pre><p data-tool="mdnice编辑器">当然，上面的DESeq2函数也适用于Bulk RNA-seq数据。</p><h4 data-tool="mdnice编辑器">例1：基于单样本的pseudobulks表达矩阵（不区分细胞类型）求差异基因</h4><pre data-tool="mdnice编辑器"><span></span><code>&gt; head(res.pseudo)<br>           HC2  P2  A1 SC4 CT3<br>AL627309.1   <span>3</span>   <span>3</span>   <span>2</span>   <span>0</span>   <span>0</span><br>AL669831.5  <span>41</span>  <span>27</span>  <span>56</span>   <span>0</span>   <span>0</span><br>LINC00115   <span>32</span>  <span>18</span>  <span>35</span>   <span>1</span>   <span>2</span><br>FAM41C      <span>15</span>   <span>2</span>  <span>19</span>   <span>7</span>   <span>4</span><br>NOC2L      <span>242</span> <span>139</span> <span>556</span>  <span>63</span>  <span>53</span><br>KLHL17      <span>17</span>   <span>7</span>  <span>19</span>   <span>2</span>   <span>1</span><br></code></pre><p data-tool="mdnice编辑器">假设样本HC2和P2是一组，需要与A1、SC4、CT3进行比较：</p><pre data-tool="mdnice编辑器"><span></span><code>input.data = res.pseudo<br>group_list = ifelse(colnames(input.data) %<span>in</span>% c(<span>"HC2"</span>,<span>"P2"</span>),<span>"case"</span>,<span>"ctrl"</span>) %&gt;%<br>  factor(levels = c(<span>"ctrl"</span>,<span>"case"</span>))<br></code></pre><p data-tool="mdnice编辑器">一行代码完成差异分析：</p><pre data-tool="mdnice编辑器"><span></span><code>example_1 = get.DESeq2(input.data = input.data, <br>                       group_list = group_list,<br>                       NA.remove = <span>T</span>)<br>head(example_1)<br><span>#       baseMean log2FoldChange    lfcSE      stat       pvalue         padj Gene</span><br><span># COX1 11338.188      -30.00000 4.407417 -6.806708 9.985702e-12 8.540771e-08 COX1</span><br><span># COX3  7380.949      -29.49563 4.407420 -6.692265 2.197419e-11 9.397263e-08 COX3</span><br><span># ND4   4886.638      -28.79662 4.407426 -6.533659 6.418179e-11 1.028435e-07  ND4</span><br><span># ND2   4271.731      -28.74874 4.407428 -6.522793 6.901006e-11 1.028435e-07  ND2</span><br><span># ND1   4188.761      -28.71937 4.407429 -6.516128 7.214560e-11 1.028435e-07  ND1</span><br><span># COX2  8580.481      -28.84833 4.407419 -6.545401 5.933587e-11 1.028435e-07 COX2</span><br></code></pre><h4 data-tool="mdnice编辑器">例2：基于单样本每一个细胞类型的pseudobulks表达矩阵（区分细胞类型）求差异基因</h4><pre data-tool="mdnice编辑器"><span></span><code>head(res.pseudo2)<br>head(meta.data2)<br><span>#                 sampleID celltype</span><br><span># HC2.NK            HC2       NK</span><br><span># HC2.Myeloid       HC2  Myeloid</span><br><span># HC2.B_cell        HC2   B_cell</span><br><span># HC2.T_cell        HC2   T_cell</span><br><span># HC2.Platelet      HC2 Platelet</span><br><span># HC2.Hpc           HC2      Hpc</span><br></code></pre><p data-tool="mdnice编辑器">例如计算每个样本中Myeloid的markers:</p><blockquote data-tool="mdnice编辑器"><p>实际上我并不建议这样做，因为1 vs. 多 得到的差异基因不可靠。</p></blockquote><pre data-tool="mdnice编辑器"><span></span><code>input.data.list = split(row.names(meta.data2),<br>                        meta.data2$sampleID)<br><br>Myeloid_DEG_list = lapply(input.data.list, <span>function</span>(sampleID){<br>  input.data = res.pseudo2[,sampleID]<br>  <br>  <span>#髓系 vs. Others</span><br>  group_list = ifelse(grepl(colnames(input.data),pattern = <span>"Myeloid"</span>),<span>"case"</span>,<span>"ctrl"</span>) %&gt;%<br>    factor(levels = c(<span>"ctrl"</span>,<span>"case"</span>))<br>  <span>if</span> (length(unique(group_list))==<span>2</span>) {<br>    tem.DEGs = get.DESeq2(input.data = input.data, <br>                           group_list = group_list,<br>                           NA.remove = <span>T</span>)<br>    <span># head(tem.DEGs)</span><br>    <span>return</span>(tem.DEGs)<br>  }<br>})<br>str(Myeloid_DEG_list)<br>names(Myeloid_DEG_list)<br><span># [1] "A1"  "CT3" "HC2" "P2"  "SC4"</span><br></code></pre><p data-tool="mdnice编辑器">设置Cutoff找差异基因：</p><pre data-tool="mdnice编辑器"><span></span><code>Myeloid.filter = lapply(Myeloid_DEG_list, <span>function</span>(DEG){<br>  DEG$Gene[DEG$log2FoldChange&gt;<span>1</span>&amp;DEG$pvalue&lt;<span>0.05</span>]<br>})<br><br>str(Myeloid.filter)<br><span># List of 5</span><br><span># $ A1 : chr [1:139] "DUSP1" "TSPO" "NUP214" "SDCBP" ...</span><br><span># $ CT3: chr [1:230] "FCN1" "DUSP6" "PSAP" "RETN" ...</span><br><span># $ HC2: chr [1:145] "TYMP" "TNFSF13B" "TSPO" "S100A6" ...</span><br><span># $ P2 : chr [1:341] "S100A11" "AP1S2" "S100A12" "CPVL" ...</span><br><span># $ SC4: chr [1:43] "PPT1" "CTSS" "FCGR2A" "NAGK" ...</span><br></code></pre><p data-tool="mdnice编辑器">可视化交集：</p><pre data-tool="mdnice编辑器"><span></span><code><span>#交集</span><br><span>library</span>(UpSetR)<br>upset(fromList(Myeloid.filter))<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.5893074119076549" data-src="https://mmbiz.qpic.cn/mmbiz/fTW9zRI3eqWdiaJ4k1KdOw9foiaaydWZDwicCLMVGK0Q76K3skdM3jLFKUwEIhULlUibib8zueFGK3rKpWEypdlbqLg/640?wx_fmt=other" data-type="other" data-w="823" src="https://mmbiz.qpic.cn/mmbiz/fTW9zRI3eqWdiaJ4k1KdOw9foiaaydWZDwicCLMVGK0Q76K3skdM3jLFKUwEIhULlUibib8zueFGK3rKpWEypdlbqLg/640?wx_fmt=other"><figcaption>image-20230203163407628</figcaption></figure><p data-tool="mdnice编辑器">可以看到只有仅仅15个基因是所有样本都有的差异基因，我们看一下是哪15个差异基因：</p><pre data-tool="mdnice编辑器"><span></span><code>tem.Myeloid = Myeloid.filter[[<span>1</span>]]<br><span>for</span> (sampleID <span>in</span> names(Myeloid.filter)[-<span>1</span>]) {<br>  tem.Myeloid = intersect(tem.Myeloid,Myeloid.filter[[sampleID]])<br>}<br>tem.Myeloid<br><span># [1] "TSPO"   "FCGRT"  "PYCARD" "BLVRB"  "LGALS1" "PSAP"   "FTL"    "PPT1"   "BRI3"   "NPC2"  </span><br><span># [11] "GRN"    "NEAT1"  "CTSS"   "AP1S2"  "CFD"  </span><br></code></pre><p data-tool="mdnice编辑器"><strong>可以看到这些基因大部分还是髓系细胞的Markers，可以说pseudobulks可以减少假阳性错误，然而代价是引入了假阴性问题。</strong></p><p data-tool="mdnice编辑器">那么如何解决这个“假阴性问题”？</p><p data-tool="mdnice编辑器"><strong>答案是增加样本量，例如：5个样本的Myeloid vs. Others:</strong></p><pre data-tool="mdnice编辑器"><span></span><code>input.data = res.pseudo2<br>group_list = ifelse(grepl(colnames(input.data),pattern = <span>"Myeloid"</span>),<span>"case"</span>,<span>"ctrl"</span>) %&gt;%<br>  factor(levels = c(<span>"ctrl"</span>,<span>"case"</span>))<br><span># [1] ctrl case ctrl ctrl ctrl ctrl case ctrl ctrl ctrl ctrl ctrl case ctrl ctrl ctrl ctrl ctrl ctrl</span><br><span># [20] ctrl case ctrl ctrl ctrl ctrl ctrl ctrl ctrl case ctrl</span><br></code></pre><p data-tool="mdnice编辑器">差异分析：</p><pre data-tool="mdnice编辑器"><span></span><code>Myeloid_DEG = get.DESeq2(input.data = input.data, <br>                         group_list = group_list,<br>                         NA.remove = <span>T</span>)<br>head(Myeloid_DEG)<br></code></pre><p data-tool="mdnice编辑器">设置Cutoff找差异基因：</p><pre data-tool="mdnice编辑器"><span></span><code>Myeloid_cutoff = Myeloid_DEG[Myeloid_DEG$log2FoldChange&gt;<span>1.5</span>&amp;Myeloid_DEG$padj&lt;<span>0.05</span>,] %&gt;% <br>  arrange(desc(log2FoldChange))<br>str(Myeloid_cutoff)<br>length(Myeloid_cutoff$Gene)<br><span>#619</span><br></code></pre><p data-tool="mdnice编辑器">这样就能得到较为稳定的细胞类型差异基因。</p><p data-tool="mdnice编辑器"><strong>总结来说，pseudobulk差异分析需要足够的两两比较的样本数量，否则结果会出现较强的假阴性现象。如果用户的样本量不大时，一是建议使用传统的FindAllMarkers()函数的Wilcox.test算法；二是使用下面的MetaCell结合DESeq2差异分析。</strong></p><h3 data-tool="mdnice编辑器">三. MetaCell结合DESeq2差异分析</h3><p data-tool="mdnice编辑器">对每个样本进行MetaCell分析：</p><blockquote data-tool="mdnice编辑器"><p>MetaCells函数为自编函数，存储于custom_seurat_functions_for_pseudobulks.R。</p></blockquote><p><mp-pay-preview-filter></mp-pay-preview-filter></p></section></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/gWmn6j0iDXHLnSfuzE25eg",target="_blank" rel="noopener noreferrer">原文链接</a>
