---
title: "跟着Nature学数据分析：R语言peer包处理RNAseq分析的批次效应"
date: 2023-08-09T15:11:08Z
draft: ["false"]
tags: [
  "fetched",
  "小明的数据分析笔记本"
]
categories: ["Acdemic"]
---
跟着Nature学数据分析：R语言peer包处理RNAseq分析的批次效应 by 小明的数据分析笔记本
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><h2 data-tool="mdnice编辑器"><span></span><span>论文</span><span> </span></h2><blockquote data-tool="mdnice编辑器"><p>Graph pangenome captures missing heritability and empowers tomato breeding</p></blockquote><p data-tool="mdnice编辑器">https://www.nature.com/articles/s41586-022-04808-9</p><p data-tool="mdnice编辑器"><code>西红柿Nature.pdf</code></p><p data-tool="mdnice编辑器">论文的代码链接</p><p data-tool="mdnice编辑器">https://github.com/YaoZhou89/TGG/tree/main/5.Genetic_analysis</p><p data-tool="mdnice编辑器">论文中做了eQTL的相关分析，转录组数据处理部分的方法写到</p><blockquote data-tool="mdnice编辑器"><p>Genes with TPM values of &gt;0.5 were defined as expressed. Only genes expressed in at least 100 accessions were retained for the downstream analyses. Raw expression levels were normalized with quantile–quantile normalization. To remove potential batch effects and confounding factors affecting gene expression, the probabilistic estimation of expression residuals method was applied with the top four factors as covariates.</p></blockquote><p data-tool="mdnice编辑器">论文里提供了 quantile–quantile normalization 的代码</p><p data-tool="mdnice编辑器">https://github.com/YaoZhou89/TGG/blob/main/5.Genetic_analysis/scripts/prepare_gene_expression.R</p><pre data-tool="mdnice编辑器"><span></span><code>quantile_normalisation &lt;- <span>function</span>(df){<br>  df_rank &lt;- apply(df,2,rank,ties.method=<span>"min"</span>)<br>  df_sorted &lt;- data.frame(apply(df, 2, sort))<br>  df_mean &lt;- apply(df_sorted, 1, mean)<br>  <br>  index_to_mean &lt;- <span>function</span>(my_index, my_mean){<br>    <span>return</span>(my_mean[my_index])<br>  }<br>  <br>  df_final &lt;- apply(df_rank, 2, index_to_mean, my_mean=df_mean)<br>  rownames(df_final) &lt;- rownames(df)<br>  <span>return</span>(df_final)<br>}<br></code></pre><p data-tool="mdnice编辑器">这个代码是什么意思暂时还没看明白，还得查查</p><p data-tool="mdnice编辑器">然后是peer这个R包</p><p data-tool="mdnice编辑器">https://github.com/PMBio/peer 这个R包的主页，但是按照这个主页的安装方法试了一下没有成功，主要是编译的时候需要复制文件到usr目录下，我没有权限，不知道怎么更改编译的目录。主页上提供的R包下载链接好像失效了，</p><p data-tool="mdnice编辑器">找到了一个简书链接 https://www.jianshu.com/p/3b613cafafe8</p><p data-tool="mdnice编辑器">提供了一个编译好的R包的下载链接</p><p data-tool="mdnice编辑器">下载下来我试了一下，只能在linux系统下使用，R语言版本需要用3.几，直接用conda安装一个3.6版本的R，使用是没有问题的</p><p data-tool="mdnice编辑器">加载的时候直接指定R包的路径</p><p data-tool="mdnice编辑器">代码</p><pre data-tool="mdnice编辑器"><span></span><code>library(peer)<br>expr = read.csv(<span>'examples/data/expression.csv'</span>, header=FALSE)<br>model = PEER()<br>PEER_setPhenoMean(model,as.matrix(expr))<br>PEER_setNk(model,20)<br>PEER_getNk(model)<br>PEER_update(model)<br>factors = PEER_getX(model)<br>weights = PEER_getW(model)<br>precision = PEER_getAlpha(model)<br>residuals = PEER_getResiduals(model)<br>PEER_setAdd_mean(model, TRUE)<br>write.table(residuals,<span>"./peer_residuals.tsv"</span>,quote=F,row.names=F,sep=<span>"\t"</span>,col.names=F)<br>write.table(factors,<span>"./peer_covariates.tsv"</span>,quote=F,row.names=F,sep=<span>"\t"</span>,col.names=F)<br></code></pre><p data-tool="mdnice编辑器">PEER_update(model) 这一行的代码会运行比较长的时间</p><p data-tool="mdnice编辑器">整体是啥意思暂时还没看明白，代码能够跑通</p><p data-tool="mdnice编辑器">还有一篇水稻的泛基因组论文也写到了这个方法</p><p data-tool="mdnice编辑器">论文</p><blockquote data-tool="mdnice编辑器"><p>A super pan-genomic landscape of rice</p></blockquote><p data-tool="mdnice编辑器"><code>水稻200rice super-pangenome CR.pdf</code></p><p data-tool="mdnice编辑器">方法部分的内容</p><blockquote data-tool="mdnice编辑器"><p>Genes with a mean FPKM value larger than 0.1 wereused in the downstream analysis and 23,736 genes met this condition. Toobtain a normal distribution of expression values for each gene, FPKMvalues of each gene were further normalized using the quantile-quantilenormalization (qqnorm) function in R (version 3.1.2). The top 20 hiddenand confounding factors in the expression data, the normal quantiletransformed expression values were inferred using the probabilisticestimation of expression residuals (PEER) method</p></blockquote><figure data-tool="mdnice编辑器"><img data-ratio="0.42592592592592593" data-src="https://mmbiz.qpic.cn/sz_mmbiz_jpg/t1wZDoUyFk5dlkuFOpqLvE3s1KlffibcIFWvDMTn3oCfASyFpd4WyNmGdtTUibvPoO7jicFPbNvFv4sx0NtZicC4mA/640?wx_fmt=other" data-type="other" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_jpg/t1wZDoUyFk5dlkuFOpqLvE3s1KlffibcIFWvDMTn3oCfASyFpd4WyNmGdtTUibvPoO7jicFPbNvFv4sx0NtZicC4mA/640?wx_fmt=other"><figcaption>image.png</figcaption></figure><p data-tool="mdnice编辑器">欢迎大家关注我的公众号</p><p data-tool="mdnice编辑器"><strong>小明的数据分析笔记本</strong></p><section><mp-common-profile data-pluginname="mpprofile" data-id="MzI3NzQ3MTcxMg==" data-headimg="http://mmbiz.qpic.cn/mmbiz_png/t1wZDoUyFk5t1sOnM0iabvBhnfIj5YpyqrMib0E1MGCd9ibcYxaOPZd0GWhQBDvK2BPEwsicQxd6y5MHLfphnwHnow/0?wx_fmt=png" data-nickname="小明的数据分析笔记本" data-alias="" data-signature="分享R语言和python在生物信息领域做数据分析和数据可视化的简单小例子；偶尔会分享一些组学数据处理相关的内容" data-from="0" data-is_biz_ban="0"></mp-common-profile></section><p data-tool="mdnice编辑器"><br></p><blockquote data-tool="mdnice编辑器"><p>小明的数据分析笔记本 公众号 主要分享：1、R语言和python做数据分析和数据可视化的简单小例子；2、园艺植物相关转录组学、基因组学、群体遗传学文献阅读笔记；3、生物信息学入门学习资料及自己的学习笔记！</p></blockquote></section><p><br></p><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/HqGm8RFROMfuEBQtRnQBug",target="_blank" rel="noopener noreferrer">原文链接</a>
