---
title: "手把手带你复现NC图表之Figure 2"
date: 2023-06-02T00:23:02Z
draft: ["false"]
tags: [
  "fetched",
  "单细胞天地"
]
categories: ["Acdemic"]
---
手把手带你复现NC图表之Figure 2 by 单细胞天地
------
<div><section data-style-type="5" data-tools="新媒体排版" data-id="2430070"><section data-style-type="5" data-tools="新媒体排版" data-id="2390348"><section><section powered-by="xiumi.us"><section><section><section powered-by="xiumi.us"><section><section><br></section><section><section><section powered-by="xiumi.us"><section><section><p>分享是一种态度</p></section></section></section></section></section></section></section></section><section><section powered-by="xiumi.us"><section><section><img data-ratio="0.8738095238095238" data-src="https://mmbiz.qpic.cn/mmbiz_gif/siaia0BDGJdjSnnb9ibXpFagGPWQ0JjSSTNVBZthROEhicfINuLcxX2Ro3Zb600LJhrzeYIb120tLjSVXEtLn14DwA/640?wx_fmt=gif" data-type="gif" data-w="420" width="100%" src="https://mmbiz.qpic.cn/mmbiz_gif/siaia0BDGJdjSnnb9ibXpFagGPWQ0JjSSTNVBZthROEhicfINuLcxX2Ro3Zb600LJhrzeYIb120tLjSVXEtLn14DwA/640?wx_fmt=gif"></section></section></section></section></section></section></section></section></section><section><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><h2 data-tool="mdnice编辑器"><span></span><span>复现文章信息：</span></h2><blockquote data-tool="mdnice编辑器"><p><span>文章题目</span>：Single-cell analysis reveals prognostic fibroblast subpopulations linked to molecular and immunological subtypes of lung cancer<br><span>期刊</span>：Nature Communications<br><span>日期</span>：2023年1月31日<br><span>DOI</span>: 10.1038/s41467-023-35832-6</p></blockquote><figure data-tool="mdnice编辑器"><img data-ratio="1.3342592592592593" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjR3vMVzwicU8FjVDOSajGf4JpRpJqIdWoDJbLRWOg4QJc5oJqeVnoxQ3yJ29aHO1yM2j73g5cXCvug/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjR3vMVzwicU8FjVDOSajGf4JpRpJqIdWoDJbLRWOg4QJc5oJqeVnoxQ3yJ29aHO1yM2j73g5cXCvug/640?wx_fmt=png"></figure><h2 data-tool="mdnice编辑器"><span></span><span>复现图——Figure 2</span></h2><figure data-tool="mdnice编辑器"><img data-ratio="0.9879227053140096" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjR3vMVzwicU8FjVDOSajGf4J7ibMp53ujxblEwSKgq2mv4mgIsDw67st8PJjibhia0RIptr6c8OOnGPCQ/640?wx_fmt=png" data-type="png" data-w="828" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjR3vMVzwicU8FjVDOSajGf4J7ibMp53ujxblEwSKgq2mv4mgIsDw67st8PJjibhia0RIptr6c8OOnGPCQ/640?wx_fmt=png"><figcaption>通过整合7个scRNA-seq数据集研究NSCLC中的成纤维细胞异质性</figcaption></figure><p data-tool="mdnice编辑器">非小细胞肺癌中存在的成纤维细胞与非癌性肺组织中确定的三个主要亚群一致，并且可能对ECM维持/重构进行差异调节。这些数据还表明，与NSCLC肿瘤的相互作用导致这些亚群中基因表达的显著变化，除了亚群特异性表型变化外，还持续涉及间质胶原的上调。此外，与对照肺成纤维细胞相比，非小细胞肺癌的myoCAF基因特征增加，而对照肺组织的iCAF基因特征增加。</p><h3 data-tool="mdnice编辑器"><span></span><span>R包载入与数据准备</span><span></span></h3><pre data-tool="mdnice编辑器"><span></span><code><span>library</span>(Seurat)<br><span>library</span>(WGCNA)<br><span>library</span>(tidyverse)<br><span>library</span>(ggpubr)<br><span>library</span>(ggsci)<br><span>library</span>(msigdbr)<br><span>library</span>(GSVA)<br><span>library</span>(limma)<br><span>library</span>(readxl)<br><span>library</span>(org.Hs.eg.db)<br><span>library</span>(hpar)<br>data_directory &lt;- <span>"H:\\文献复现\\6\\"</span> <br><span>source</span>(paste0(data_directory, <span>"0_NewFunctions.R"</span>))<br><br>load(paste0(data_directory, <span>"IntegratedFibs_Zenodo.Rdata"</span>))<br>load(paste0(data_directory, <span>"HCLA_annotation.Rdata"</span>))<br>Fibroblast_GeneSets_v2_2021_0623 &lt;- read_excel(paste0(data_directory, <span>"Fibroblast GeneSets v2 2021_0623.xlsx"</span>))<br><span>#https://static-content.springer.com/esm/art%3A10.1038%2Fs41467-023-35832-6/MediaObjects/41467_2023_35832_MOESM21_ESM.xlsx</span><br>Figure2i &lt;- read_excel(paste0(data_directory, <span>"Figure 2i.xlsx"</span>))<br>Figure2j &lt;- read_excel(paste0(data_directory, <span>"Figure 2j.xlsx"</span>))<br></code></pre><h3 data-tool="mdnice编辑器"><span></span><span>成纤维细胞聚类</span><span></span></h3><p data-tool="mdnice编辑器">作者使用来自NSCLC和对照组织样本的另外6个scRNA-seq数据集重复了成纤维细胞分选（不包括壁细胞）的过程。生成了一个包含9673个成纤维细胞的数据集（5183个细胞来自39个对照组样本，3440个来自46个LUAD样本，654个来自16个LUSC样本）。通过无监督聚类识别的三个主要亚群，并且鉴定的三个簇与先前在对照肺组织中描述的成纤维细胞亚群一致（外膜、肺泡和肌成纤维细胞）。</p><pre data-tool="mdnice编辑器"><span></span><code><br>DefaultAssay(Fibs.integrated) &lt;- <span>"integrated"</span><br>Fibs.integrated &lt;- ScaleData(Fibs.integrated)<br>Fibs.integrated &lt;- RunPCA(Fibs.integrated)<br>ElbowPlot(Fibs.integrated, ndims = <span>50</span>)<br>PCs2use &lt;- c(<span>1</span>:<span>30</span>)<br>Fibs.integrated &lt;- RunUMAP(Fibs.integrated, dims = PCs2use)<br>Fibs.integrated &lt;- FindNeighbors(Fibs.integrated, dims = PCs2use, verbose = <span>FALSE</span>)<br>Fibs.integrated &lt;- AddMetaData(Fibs.integrated, as.data.frame(Fibs.integrated@reductions$umap@cell.embeddings))<br>Fibs.integrated &lt;-  FindClusters(Fibs.integrated, resolution = <span>0.1</span>, verbose = <span>FALSE</span>)<br>Fibs.integrated$Fibs_MajorClusters &lt;- factor(Fibs.integrated$seurat_clusters, levels = c(<span>2</span>,<span>1</span>,<span>0</span>), labels = c(<span>"Adventitial"</span>, <span>"Alveolar"</span>, <span>"Myo"</span>))<br><br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="1.0592885375494072" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjR3vMVzwicU8FjVDOSajGf4JIruxwPTLtrGTZicmBCldgsBuOLHzQRNc7diaksJ1Rsnic4iaRqgk40A7uA/640?wx_fmt=png" data-type="png" data-w="253" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjR3vMVzwicU8FjVDOSajGf4JIruxwPTLtrGTZicmBCldgsBuOLHzQRNc7diaksJ1Rsnic4iaRqgk40A7uA/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">在7套数据中作者为了整合和纠正数据集之间的批效应，使用典型相关分析并且使用共享最近邻模块化优化算法执行无监督聚类。为了确定最具生物学信息量的聚类解决方案，改变分辨率参数，迭代增加识别的聚类数量，并检查每个聚类识别的标记基因数量(样本水平平均log foldchange&gt;1,adj.P&lt;0.01，至少有50%的样本表示)。这表明，三个主要的集群被一致地确定，而更高分辨率的聚类导致鉴定集群没有或很少的标记基因符合这些标准</p><h3 data-tool="mdnice编辑器"><span></span><span>Figure 2B</span><span></span></h3><pre data-tool="mdnice编辑器"><span></span><code>Fig_2B &lt;- <br>  DimPlot(Fibs.integrated, group.by = <span>"Fibs_MajorClusters"</span>, pt.size = <span>1.2</span>) &amp;<br>  theme_pubr(base_size = <span>15</span>) &amp;<br>  scale_colour_manual(values = Fibs_col.palette) &amp;<br>  theme(legend.position = c(<span>0</span>,<span>0</span>),<br>        legend.justification = c(<span>0</span>,<span>0</span>),<br>        legend.background = element_blank(),<br>        plot.title = element_blank(),<br>        legend.key.size = unit(<span>7</span>,<span>"pt"</span>)<br>  )<br>Fig_2B<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7083333333333334" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjR3vMVzwicU8FjVDOSajGf4JpC4koOLanyYRw1vNyASU6pGDsibKKvtuydjibBD5rNiaP1lmzB1xkNfEg/640?wx_fmt=png" data-type="png" data-w="912" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjR3vMVzwicU8FjVDOSajGf4JpC4koOLanyYRw1vNyASU6pGDsibKKvtuydjibBD5rNiaP1lmzB1xkNfEg/640?wx_fmt=png"></figure><h3 data-tool="mdnice编辑器"><span></span><span>成纤维细胞标志物分析</span><span></span></h3><p data-tool="mdnice编辑器">首先计算每个亚群(上皮细胞、肺泡细胞和肌成纤维细胞)的样品水平基因表达谱（单个细胞的平均值），然后进行差异基因表达分析</p><pre data-tool="mdnice编辑器"><span></span><code>Fibs.integrated &lt;- SetIdent(Fibs.integrated, value = <span>"Fibs_MajorClusters"</span>)<br>All.Markers &lt;- FindAllMarkers(Fibs.integrated, test.use = <span>"bimod"</span>, assay = <span>"RNA"</span>, only.pos = <span>T</span>)<br>All.Markers$ENSEMBL &lt;- mapIds(org.Hs.eg.db, keys = All.Markers$gene, keytype = <span>"SYMBOL"</span>, column=<span>"ENSEMBL"</span>)<br>test &lt;- hpar::getHpa(All.Markers$ENSEMBL)<br>table(test$Reliability)<br>test &lt;- test[!duplicated(test$Gene.name), ]<br>rownames(test) &lt;- test$Gene.name<br>All.Markers$HPA_reliability &lt;- test[All.Markers$gene, <span>"Reliability"</span>]<br>test &lt;- hpar::getHpa(All.Markers$ENSEMBL, hpadata = <span>"hpaSubcellularLoc"</span>)<br>test &lt;- test[!duplicated(test$Gene.name), ]<br>rownames(test) &lt;- test$Gene.name<br>test$membrane &lt;- <span>1</span>:nrow(test) %<span>in</span>% grep(<span>"membrane"</span>, test$GO.id, fixed = <span>T</span>)<br>test$cytosol &lt;- <span>1</span>:nrow(test) %<span>in</span>% grep(<span>"Cytosol"</span>, test$GO.id, fixed = <span>T</span>)<br>test$filaments &lt;- <span>1</span>:nrow(test) %<span>in</span>% grep(<span>"filaments"</span>, test$GO.id, fixed = <span>T</span>)<br>test$nuclear &lt;- <span>1</span>:nrow(test) %<span>in</span>% grep(<span>"Nucle"</span>, test$GO.id, fixed = <span>T</span>)<br>All.Markers$subcell.loc &lt;- test[All.Markers$gene,<br>                                c(<span>"membrane"</span>,<span>"cytosol"</span>,<span>"filaments"</span>,<span>"nuclear"</span>)]<br>All.Markers$pct.diff &lt;- All.Markers$pct.1 - All.Markers$pct.2<br>All.Markers$Detected.Intracellular &lt;- rowSums(All.Markers$subcell.loc) &gt; <span>0</span><br><span>#样本层面成纤维细胞标记物分析</span><br>Fibs.integrated &lt;- SetIdent(Fibs.integrated, value = <span>"Fibs_MajorClusters"</span>)<br>Fibs.integrated_samples &lt;- AverageExpression(Fibs.integrated, assays = <span>"RNA"</span>, return.seurat = <span>T</span>, <br>                                             group.by = c(<span>"Fibs_MajorClusters"</span>, <span>"SampleID2"</span>),<br>                                             slot = <span>"data"</span>)<br>Fibs.integrated_samples &lt;- NormalizeData(Fibs.integrated_samples)<br>dim(Fibs.integrated_samples)<br><span>#从样本名称中提取元数据</span><br>Fibs.integrated_samples@meta.data &lt;- <br>  Fibs.integrated_samples@meta.data %&gt;%<br>  mutate(Fibs_MajorClusters = str_split_fixed(colnames(Fibs.integrated_samples), <span>"_"</span>, <span>3</span>)[,<span>1</span>],<br>         Dataset = str_split_fixed(colnames(Fibs.integrated_samples), <span>"_"</span>, <span>3</span>)[,<span>2</span>],<br>         SampleID = str_split_fixed(colnames(Fibs.integrated_samples), <span>"_"</span>, <span>3</span>)[,<span>3</span>])<br><span>#整理样本和样本差异分析结果</span><br>Fibs.Sample_MetaData &lt;- Fibs.integrated_samples@meta.data[,<span>"SampleID"</span>, drop = <span>F</span>]<br>matchSamples &lt;- match(Fibs.Sample_MetaData$SampleID, Sample_MetaData$SampleID)<br>table(is.na(matchSamples))<br>Fibs.Sample_MetaData &lt;- cbind(Fibs.Sample_MetaData, Sample_MetaData[matchSamples, ])<br>keepCols_metaData &lt;- c(<span>"SampleID"</span>, <span>"nCount_RNA"</span>, <span>"nFeature_RNA"</span>,<br>                       <span>"Fibs_MajorClusters"</span>, <span>"Dataset"</span>)<br>Fibs.integrated_samples@meta.data &lt;- Fibs.integrated_samples@meta.data[, keepCols_metaData]<br>Fibs.integrated_samples &lt;- AddMetaData(Fibs.integrated_samples, Fibs.Sample_MetaData)<br>Fibs.integrated_samples$Sample.type[Fibs.integrated_samples$Sample.type == <span>"Tumor"</span>] &lt;- <span>"Tumour"</span><br>Fibs.integrated_samples$pN &lt;- droplevels(Fibs.integrated_samples$pN, exclude = c(<span>NA</span>, <span>"NA"</span>))<br>levels(Fibs.integrated_samples$pN)<br>Fibs.integrated_samples &lt;- SetIdent(Fibs.integrated_samples, value = <span>"Fibs_MajorClusters"</span>)<br>Sample.level_Markers &lt;- FindAllMarkers(Fibs.integrated_samples, test.use = <span>"wilcox"</span>, assay = <span>"RNA"</span>,<br>                                       only.pos = <span>F</span>, return.thresh = <span>1</span>, min.pct = <span>0.25</span>, logfc.threshold = <span>0</span>)<br>Marker.match &lt;- match(paste(All.Markers$cluster, All.Markers$gene),<br>                      paste(Sample.level_Markers$cluster, Sample.level_Markers$gene))<br>All.Markers$p_val_adj.Sample &lt;- Sample.level_Markers$p_val_adj[Marker.match]<br>All.Markers$log2FC.Sample &lt;- Sample.level_Markers$avg_log2FC[Marker.match]<br></code></pre><h3 data-tool="mdnice编辑器"><span></span><span>Figure 2C</span><span></span></h3><p data-tool="mdnice编辑器">显示每个亚群10个最显著标志物的样品水平表达（单细胞平均值）的热图</p><pre data-tool="mdnice编辑器"><span></span><code>All.Markers$cluster &lt;- factor(as.character(All.Markers$cluster), levels = c(<span>"Adventitial"</span>, <span>"Alveolar"</span>, <span>"Myo"</span>))<br>All.Markers %&gt;%<br>  group_by(cluster) %&gt;%<br>  filter(avg_log2FC &gt; <span>0</span>) %&gt;%<br>  top_n(w = -p_val_adj.Sample, n = <span>10</span>) -&gt; top10_markers<br>DefaultAssay(Fibs.integrated_samples) &lt;- <span>"RNA"</span><br>Fibs.integrated_samples &lt;- ScaleData(Fibs.integrated_samples, features = top10_markers$gene, assay = <span>"RNA"</span>)<br>levels(Fibs.integrated_samples$Fibs_MajorClusters)<br>Fibs.integrated_samples$Fibs_MajorClusters &lt;- factor(Fibs.integrated_samples$Fibs_MajorClusters,<br>  levels = c(<span>"Adventitial"</span>, <span>"Alveolar"</span>, <span>"Myo"</span>)<br>)<br>Fig_2C &lt;-<br>  DoHeatmap(Fibs.integrated_samples, features = rev(top10_markers$gene),<br>            assay = <span>"RNA"</span>,<br>            angle = <span>45</span>,<br>            group.by = c(<span>"Fibs_MajorClusters"</span>) ,<br>            group.colors = Fibs_col.palette,<br>            label = <span>T</span>, disp.max = <span>2</span>) &amp;<br>  theme(axis.text.y = element_text(face = <span>"italic"</span>, size = <span>10</span>),legend.position = <span>"none"</span>) <br>Fig_2C<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7083333333333334" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjR3vMVzwicU8FjVDOSajGf4JJEmTiaPCFxnsu0QqGhXrnEGDpKLz5zXscu9rrw5DMElJKgIUVQbzIibw/640?wx_fmt=png" data-type="png" data-w="912" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjR3vMVzwicU8FjVDOSajGf4JJEmTiaPCFxnsu0QqGhXrnEGDpKLz5zXscu9rrw5DMElJKgIUVQbzIibw/640?wx_fmt=png"></figure><h3 data-tool="mdnice编辑器"><span></span><span>通路分析</span><span></span></h3><p data-tool="mdnice编辑器">使用REACTOME通路进行GSVA，以及使用先前描述的成纤维细胞亚群的基因标记进行GSVA</p><pre data-tool="mdnice编辑器"><span></span><code>h_gene_sets = msigdbr(species = <span>"human"</span>, category = <span>"C2"</span>, subcategory = <span>"CP:REACTOME"</span>)<br>Reactome_pathwaysH = split(x = h_gene_sets$human_gene_symbol, f = h_gene_sets$gs_name)<br>REACTOME_gsva_res &lt;- gsva(as.matrix(Fibs.integrated_samples@assays$RNA@data), gset.idx.list = Reactome_pathwaysH, method=<span>"gsva"</span>,<br>                          min.sz=<span>10</span>, max.sz=<span>500</span>)<br><span>#通过经验Bayes进行差异分析</span><br>f &lt;- factor(Fibs.integrated_samples$Fibs_MajorClusters, levels = c(<span>"Myo"</span>,<span>"Alveolar"</span>, <span>"Adventitial"</span>))<br>design &lt;- model.matrix(~<span>0</span>+f)<br>colnames(design) &lt;- c(<span>"Myo"</span>, <span>"Alveolar"</span>, <span>"Adventitial"</span>)<br>REACTOME_gsva.fit &lt;- lmFit(REACTOME_gsva_res, design)<br>REACTOME_gsva.fit &lt;- eBayes(REACTOME_gsva.fit)<br>contrast.matrix &lt;- makeContrasts(Myo.v.ALL = Myo-(Alveolar+Adventitial)/<span>2</span>,<br>                                 Alv.v.ALL = Alveolar-(Myo+Adventitial)/<span>2</span>,<br>                                 Adv.v.ALL = Adventitial-(Myo+Alveolar)/<span>2</span>,<br>                                 Myo.v.Alv = Myo - Alveolar,<br>                                 Myo.v.Adv = Myo-Adventitial,<br>                                 Alv.v.Adv = Alveolar-Adventitial,<br>                                 levels=design)<br>REACTOME_gsva.fit2 &lt;- contrasts.fit(REACTOME_gsva.fit, contrast.matrix)<br>REACTOME_gsva.fit2 &lt;- eBayes(REACTOME_gsva.fit2)<br>res2 &lt;- decideTests(REACTOME_gsva.fit2, p.value=<span>0.01</span>, method = <span>"global"</span>)<br>summary(res2)<br>REACTOME_tt.res &lt;- list(<br>  Myo = topTable(REACTOME_gsva.fit2, coef = <span>"Myo.v.ALL"</span>, n=<span>1270</span>),<br>  Alveolar = topTable(REACTOME_gsva.fit2, coef = <span>"Alv.v.ALL"</span>, n=<span>1270</span>),<br>  Adventitial = topTable(REACTOME_gsva.fit2, coef = <span>"Adv.v.ALL"</span>, n=<span>1270</span>)<br>)<br>REACTOME_tt.res_df &lt;- do.call(rbind, REACTOME_tt.res)<br>REACTOME_tt.res_df$Cluster &lt;- do.call(rbind, strsplit(rownames(REACTOME_tt.res_df), <span>"."</span>, <span>2</span>))[,<span>1</span>]<br>REACTOME_tt.res_df$Pathway &lt;- do.call(rbind, strsplit(rownames(REACTOME_tt.res_df), <span>"."</span>, <span>2</span>))[,<span>2</span>]<br>REACTOME_tt.res_df$Pathway.label &lt;- gsub(<span>"REACTOME_"</span>, <span>""</span>, REACTOME_tt.res_df$Pathway)<br>REACTOME_tt.res_df$Pathway.label &lt;- gsub(<span>"_"</span>, <span>" "</span>, REACTOME_tt.res_df$Pathway.label)<br>REACTOME_tt.res_subset &lt;- REACTOME_tt.res_df[REACTOME_tt.res_df$logFC &gt; <span>0</span>, ] %&gt;%<br>  group_by(Cluster) %&gt;%<br>  top_n(wt = -adj.P.Val, n = <span>5</span>)<br>REACTOME_tt.res_subset &lt;- REACTOME_tt.res_subset[-<span>10</span>, ]<br>REACTOME_tt.res_subset &lt;- rbind(REACTOME_tt.res_subset, REACTOME_tt.res_df[<span>"Alveolar.REACTOME_TRP_CHANNELS"</span>, ])<br></code></pre><h3 data-tool="mdnice编辑器"><span></span><span>Figure 2D</span><span></span></h3><p data-tool="mdnice编辑器">条形图显示每个亚群中最显著上调的REACTOME通路的log2倍数变化</p><pre data-tool="mdnice编辑器"><span></span><code>Fig_2D &lt;- <br>  REACTOME_tt.res_subset %&gt;%<br>  ggplot(aes(x = reorder(Pathway.label, logFC), y = logFC, label =  paste0(<span>"adj.P="</span>, signif(adj.P.Val, <span>3</span>)),<br>             fill = Cluster)) +<br>  theme_pubr(base_size = <span>15</span>)+<br>  theme(axis.title.y = element_blank(),legend.title = element_blank(),<br>        axis.text.y = element_text(size = <span>7</span>),<br>        legend.position = <span>"top"</span>, strip.text = element_blank(), legend.key.size = unit(<span>10</span>, <span>"pt"</span>))+<br>  geom_bar(stat = <span>"identity"</span>, alpha = <span>0.75</span>)+<br>  facet_grid(Cluster~., scales = <span>"free"</span>, space = <span>"free"</span>) +<br>  geom_text(hjust = <span>1.1</span>, colour = <span>"white"</span>, size = <span>3</span>) +<br>  scale_fill_manual(values = Fibs_col.palette) +<br>  coord_flip()<br>Fig_2D<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7083333333333334" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjR3vMVzwicU8FjVDOSajGf4JqVcAEib3ibl4gXRUdmwiaibtwlGhvpib9WYVoCIqPXAQ3dzRh15wMGOjHRw/640?wx_fmt=png" data-type="png" data-w="912" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjR3vMVzwicU8FjVDOSajGf4JqVcAEib3ibl4gXRUdmwiaibtwlGhvpib9WYVoCIqPXAQ3dzRh15wMGOjHRw/640?wx_fmt=png"></figure><h3 data-tool="mdnice编辑器"><span></span><span>基质体基因分析</span><span></span></h3><p data-tool="mdnice编辑器">检测每个成纤维细胞亚群是否上调了与特定基质体类别（基底膜、间质胶原、ECM糖蛋白和蛋白聚糖），结果表明肌成纤维细胞上调了每个基质体类别中的多个基因，包括大多数上调的间质胶原</p><pre data-tool="mdnice编辑器"><span></span><code>CGP_gene_sets = msigdbr(species = <span>"human"</span>, category = <span>"C2"</span>)<br>CGP_pathwaysH = split(x = CGP_gene_sets$human_gene_symbol, f = CGP_gene_sets$gs_name)<br>All.Markers$NABA_Matrisome &lt;- All.Markers$gene %<span>in</span>% CGP_pathwaysH$NABA_MATRISOME<br>All.Markers$NABA_BASEMENT_MEMBRANES &lt;- All.Markers$gene %<span>in</span>% CGP_pathwaysH$NABA_BASEMENT_MEMBRANES<br>All.Markers$NABA_COLLAGENS &lt;- All.Markers$gene %<span>in</span>% CGP_pathwaysH$NABA_COLLAGENS<br>All.Markers$NABA_ECM_GLYCOPROTEINS &lt;- All.Markers$gene %<span>in</span>% CGP_pathwaysH$NABA_ECM_GLYCOPROTEINS<br>All.Markers$NABA_PROTEOGLYCANS &lt;- All.Markers$gene %<span>in</span>% CGP_pathwaysH$NABA_PROTEOGLYCANS<br>names(All.Markers)<br>All.Markers$type &lt;- factor(<br>  apply(All.Markers[,c(<span>"NABA_BASEMENT_MEMBRANES"</span>,<span>"NABA_COLLAGENS"</span>,<span>"NABA_ECM_GLYCOPROTEINS"</span>,<span>"NABA_PROTEOGLYCANS"</span>)],<span>1</span>, which.max),<br>                           levels = <span>1</span>:<span>5</span>,<br>                           labels = c(<span>"Basement.Membranes"</span>, <span>"Interstitial.Collagens"</span>,<br>                                      <span>"ECM.Glycoproteins"</span>, <span>"Proteoglycans"</span>, <span>"Other"</span>)<br>)<br>All.Markers$type[All.Markers$type == <span>"Basement.Membranes"</span> &amp;<br>                   All.Markers$NABA_BASEMENT_MEMBRANES == <span>F</span>] &lt;- <span>"Other"</span><br></code></pre><h3 data-tool="mdnice编辑器"><span></span><span>Figure 2E</span><span></span></h3><p data-tool="mdnice编辑器">条形图显示每个亚群差异表达的不同基质体组分的比例</p><pre data-tool="mdnice编辑器"><span></span><code>Fig_2E &lt;- <br>  All.Markers[!All.Markers$type == <span>"Other"</span> &amp;<br>                All.Markers$p_val_adj.Sample &lt; <span>0.01</span>, ] %&gt;%<br>  ggplot(aes(x = type, fill = cluster)) +<br>  theme_pubr(base_size = <span>15</span>) +<br>  theme(legend.position = <span>"top"</span>, legend.key.size = unit(<span>20</span>,<span>"pt"</span>), legend.title = element_blank()) +<br>  geom_bar(position = <span>"fill"</span>) +<br>  scale_fill_manual(values = Fibs_col.palette) +<br>  rotate_x_text(angle = <span>45</span>) +<br>  ylab(<span>"Proportion of DEGs"</span>) +<br>  xlab(<span>"Matrisome Category"</span>)<br>Fig_2E<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7083333333333334" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjR3vMVzwicU8FjVDOSajGf4JbOUhrbc7DpMm6mDaT9xTibYfNPsX3qVhhpOEAzW3nb4F2rvzWr8p79A/640?wx_fmt=png" data-type="png" data-w="912" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjR3vMVzwicU8FjVDOSajGf4JbOUhrbc7DpMm6mDaT9xTibYfNPsX3qVhhpOEAzW3nb4F2rvzWr8p79A/640?wx_fmt=png"></figure><h3 data-tool="mdnice编辑器"><span></span><span>Figure 2F-G</span><span></span></h3><pre data-tool="mdnice编辑器"><span></span><code><span>#计算每个基质体类别的模块评分，并比较了成纤维细胞亚群之间的总体表达）。</span><br>Fibs.integrated_samples &lt;- AddModuleScore(<br>  Fibs.integrated_samples,<br>  features = list(<br>    CGP_pathwaysH$NABA_BASEMENT_MEMBRANES,<br>    CGP_pathwaysH$NABA_COLLAGENS,<br>    CGP_pathwaysH$NABA_ECM_GLYCOPROTEINS,<br>    CGP_pathwaysH$NABA_PROTEOGLYCANS,<br>    CGP_pathwaysH$NABA_COLLAGENS[!CGP_pathwaysH$NABA_COLLAGENS %<span>in</span>% CGP_pathwaysH$NABA_BASEMENT_MEMBRANES]<br>  ),<br>  assay = <span>"RNA"</span><br>)<br>names(Fibs.integrated_samples@meta.data)[grep(<span>"^Cluster"</span>, names(Fibs.integrated_samples@meta.data))] &lt;-<br>  c(<span>"Basement.Membranes"</span>, <span>"Collagens"</span>, <span>"ECM.Glycoproteins"</span>, <span>"Proteoglycans"</span>, <span>"Interstitial.Collagens"</span>)<br>comparisons &lt;- list(c(<span>"Myo"</span>, <span>"Adventitial"</span>), c(<span>"Myo"</span>, <span>"Alveolar"</span>), c(<span>"Alveolar"</span>, <span>"Adventitial"</span>))<br><span>#箱形图显示编码每个亚群基底膜组分的基因的样品水平表达</span><br>pA &lt;- <br>  Fibs.integrated_samples@meta.data %&gt;%<br>  ggplot(aes(x = Fibs_MajorClusters, y = Basement.Membranes,<br>             fill = Fibs_MajorClusters)) +<br>  theme_pubr(base_size = <span>15</span>) +<br>  theme(legend.position = <span>"right"</span>, axis.title.x = element_blank()) +<br>  geom_boxplot(outlier.shape = <span>NA</span>, alpha = <span>0.75</span>) +<br>  geom_jitter(width = <span>0.2</span>, alpha = <span>0.5</span>, size = <span>2</span>) +<br>  stat_compare_means(comparisons = list(c(<span>"Myo"</span>, <span>"Adventitial"</span>), c(<span>"Alveolar"</span>, <span>"Adventitial"</span>)),<br>                     size = <span>5</span>,<br>                     hide.ns = <span>T</span>,label.y = c(<span>0.65</span>,<span>0.8</span>)) +<br>  scale_fill_manual(values = Fibs_col.palette) +<br>  rotate_x_text(angle = <span>45</span>) +<br>  ylab(<span>"Basement Membranes"</span>) +<br>  guides(fill = <span>"none"</span>) +  expand_limits(y = <span>0.9</span>)<br><span>#箱形图显示了每个亚群间质胶原编码基因的样品水平表达</span><br>pB &lt;- <br>  Fibs.integrated_samples@meta.data %&gt;%<br>  ggplot(aes(x = Fibs_MajorClusters, y = Interstitial.Collagens,<br>             fill = Fibs_MajorClusters)) +<br>  theme_pubr(base_size = <span>15</span>) +<br>  theme(legend.position = <span>"right"</span>, axis.title.x = element_blank()) +<br>  geom_boxplot(outlier.shape = <span>NA</span>, alpha = <span>0.75</span>) +<br>  geom_jitter(width = <span>0.2</span>, alpha = <span>0.5</span>, size = <span>2</span>) +<br>  stat_compare_means(comparisons = list(c(<span>"Myo"</span>, <span>"Adventitial"</span>), c(<span>"Myo"</span>, <span>"Alveolar"</span>)),<br>                     size = <span>5</span>,<br>                     hide.ns = <span>T</span>, label.y = c(<span>0.9</span>,<span>1.1</span>)) +<br>  scale_fill_manual(values = Fibs_col.palette) +<br>  ylab(<span>"Interstitial Collagens"</span>) +<br>  rotate_x_text(angle = <span>45</span>) +<br>  guides(fill = <span>"none"</span>) + expand_limits(y = <span>1.25</span>)<br>Fig_2FG &lt;- ggarrange(pA,pB, ncol = <span>2</span>, labels = c(<span>"f"</span>, <span>"g"</span>))<br>Fig_2FG<br><span>#这表明与外膜和肺泡成纤维细胞相比，肌成纤维细胞显著上调了间质胶原;而与外膜成纤维细胞相比，肺泡和肌成纤维细胞的基底膜基因表达均显著增加）。</span><br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7083333333333334" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjR3vMVzwicU8FjVDOSajGf4JMz5urhw2ia1J8S9Ly3x8vvUTBjIGdbvgwJibgnHLUiav8G9rGypxOXjww/640?wx_fmt=png" data-type="png" data-w="912" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjR3vMVzwicU8FjVDOSajGf4JMz5urhw2ia1J8S9Ly3x8vvUTBjIGdbvgwJibgnHLUiav8G9rGypxOXjww/640?wx_fmt=png"></figure><h3 data-tool="mdnice编辑器"><span></span><span>Figure 2H</span><span></span></h3><p data-tool="mdnice编辑器">箱形图显示了按组织类型划分的每个亚群间质胶原编码基因的样本水平表达</p><pre data-tool="mdnice编辑器"><span></span><code>comparisons &lt;- list(c(<span>"Control"</span>, <span>"Tumour"</span>))<br>Fig_2H &lt;- <br>  Fibs.integrated_samples@meta.data %&gt;%<br>  drop_na(Sample.type) %&gt;%<br>  ggplot(aes(x = Sample.type, y = Interstitial.Collagens)) +<br>  theme_pubr(base_size = <span>15</span>) +<br>  facet_wrap(~Fibs_MajorClusters) +<br>  geom_boxplot(aes(fill = Fibs_MajorClusters), outlier.shape = <span>NA</span>, alpha = <span>0.75</span>) +<br>  geom_jitter(width = <span>0.2</span>, alpha = <span>0.5</span>, size = <span>2</span>) +<br>  stat_compare_means(comparisons = comparisons,<br>                     size = <span>5</span>,<br>                     label.y = <span>0.9</span>) +<br>  scale_fill_manual(values = Fibs_col.palette) +<br>  rotate_x_text(angle = <span>45</span>) +<br>  ylab(<span>"Interstitial Collagens"</span>) +<br>  guides(fill = <span>"none"</span>) +<br>  theme(axis.title.x = element_blank(),<br>        legend.position = <span>"none"</span>) + expand_limits(y = <span>1.1</span>)<br>Fig_2H<br><br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7083333333333334" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjR3vMVzwicU8FjVDOSajGf4JnOhWlESMicMzYtgC7mtHbZqxRmSlC99GBYMlF9zUlwSpa2n6ibPKtuEQ/640?wx_fmt=png" data-type="png" data-w="912" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjR3vMVzwicU8FjVDOSajGf4JnOhWlESMicMzYtgC7mtHbZqxRmSlC99GBYMlF9zUlwSpa2n6ibPKtuEQ/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">在三个亚群的肿瘤样品中间质胶原表达显著增加</p><h3 data-tool="mdnice编辑器"><span></span><span>Figure 2I</span><span></span></h3><p data-tool="mdnice编辑器">箱形图显示按组织类型划分的每个亚群编码myoCAF标志物的基因的样本水平表达</p><pre data-tool="mdnice编辑器"><span></span><code>name&lt;-colnames(Figure2i)<br>name[<span>3</span>]&lt;-<span>"myoCAF"</span><br>colnames(Figure2i)&lt;-name<br>Figure2i&lt;-cbind(Fibs.integrated_samples@meta.data,Figure2i[,c(<span>1</span>,<span>3</span>)])<br>comparisons &lt;- list(c(<span>"Control"</span>, <span>"Tumour"</span>))<br>Fig_2I &lt;- <br>  Figure2i %&gt;%<br>  drop_na(Sample.type) %&gt;%<br>  ggplot(aes(x = Sample.type, y = myoCAF signature (relative expression))) +<br>  theme_pubr(base_size = <span>5</span>) +<br>  facet_wrap(~Fibs_MajorClusters) +<br>  geom_boxplot(aes(fill = Fibs_MajorClusters), outlier.shape = <span>NA</span>, alpha = <span>0.75</span>) +<br>  geom_jitter(width = <span>0.2</span>, alpha = <span>0.5</span>, size = <span>1</span>) +<br>  stat_compare_means(comparisons = comparisons,<br>                     size = <span>2.5</span>,<br>                     label.y = <span>0.9</span>) +<br>  scale_fill_manual(values = Fibs_col.palette) +<br>  rotate_x_text(angle = <span>45</span>) +<br>  ylab(<span>"myoCAF Signature"</span>) +<br>  guides(fill = <span>"none"</span>) +<br>  theme(axis.title.x = element_blank(),<br>        legend.position = <span>"none"</span>) + expand_limits(y = <span>1.1</span>)<br>Fig_2I<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.75" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjR3vMVzwicU8FjVDOSajGf4JteUcYTxJCUd3PMmEjSLaswyDfEictA8QpH9I4rdCibwibleCj7x1fhoIA/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjR3vMVzwicU8FjVDOSajGf4JteUcYTxJCUd3PMmEjSLaswyDfEictA8QpH9I4rdCibwibleCj7x1fhoIA/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">myoCAF基因标签在肿瘤样品中显著上调</p><h3 data-tool="mdnice编辑器"><span></span><span>Figure 2J</span><span></span></h3><p data-tool="mdnice编辑器">箱形图显示了按组织类型划分的各亚群中编码iCAF标志物的基因的样本水平表达</p><pre data-tool="mdnice编辑器"><span></span><code>name&lt;-colnames(Figure2j)<br>name[<span>3</span>]&lt;-<span>"iCAF"</span><br>colnames(Figure2j)&lt;-name<br>Figure2j&lt;-cbind(Fibs.integrated_samples@meta.data,Figure2j[,c(<span>1</span>,<span>3</span>)])<br>comparisons &lt;- list(c(<span>"Control"</span>, <span>"Tumour"</span>))<br>Fig_2J &lt;- <br>  Figure2j %&gt;%<br>  drop_na(Sample.type) %&gt;%<br>  ggplot(aes(x = Sample.type, y = iCAF)) +<br>  theme_pubr(base_size = <span>15</span>) +<br>  facet_wrap(~Fibs_MajorClusters) +<br>  geom_boxplot(aes(fill = Fibs_MajorClusters), outlier.shape = <span>NA</span>, alpha = <span>0.75</span>) +<br>  geom_jitter(width = <span>0.2</span>, alpha = <span>0.5</span>, size = <span>2</span>) +<br>  stat_compare_means(comparisons = comparisons,<br>                     size = <span>5</span>,<br>                     label.y = <span>0.9</span>) +<br>  scale_fill_manual(values = Fibs_col.palette) +<br>  rotate_x_text(angle = <span>45</span>) +<br>  ylab(<span>"Interstitial Collagens"</span>) +<br>  guides(fill = <span>"none"</span>) +<br>  theme(axis.title.x = element_blank(),<br>        legend.position = <span>"none"</span>) + expand_limits(y = <span>1.1</span>)<br>Fig_2J<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.75" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjR3vMVzwicU8FjVDOSajGf4Ja3s8nbFgeSJTeBSOeicnnezC3C9JvWW4vUjaOhDzpInu6FHoIBlvVnA/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjR3vMVzwicU8FjVDOSajGf4Ja3s8nbFgeSJTeBSOeicnnezC3C9JvWW4vUjaOhDzpInu6FHoIBlvVnA/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">与对照组织相比，从肿瘤样品分离的成纤维细胞中iCAF基因标记显著下调</p></section><p><br></p><section data-style-type="5" data-tools="新媒体排版" data-id="2440476"><section><section><section><section><img data-ratio="0.9495798319327731" data-src="https://mmbiz.qpic.cn/mmbiz_gif/09gp6SvPE04j3m2v7Hr889icHUyibTOHs8YuUibicl7ibRD0ZwG5pDTjBluRreZvuib1o3BibvLkicYhnA4YW7dQsjn0cA/640?wx_fmt=gif" data-type="gif" data-w="119" data-width="100%" src="https://mmbiz.qpic.cn/mmbiz_gif/09gp6SvPE04j3m2v7Hr889icHUyibTOHs8YuUibicl7ibRD0ZwG5pDTjBluRreZvuib1o3BibvLkicYhnA4YW7dQsjn0cA/640?wx_fmt=gif"></section><section data-brushtype="text">往期回顾</section><section><br></section></section></section></section><section><section data-autoskip="1"><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247513057&amp;idx=1&amp;sn=6f737f0a5ae7795645a42894897f7b87&amp;chksm=ea1cb563dd6b3c7567db78011efb52823c1f6bc94a760cc11ef8a07d3d5d6c779e4813c4d523&amp;scene=21#wechat_redirect" textvalue="手把手带你复现NC图表之Figure 1" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2"><span>手把手带你复现NC图表之Figure 1</span></a></p><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247513143&amp;idx=1&amp;sn=ac15652a0d046d28330676994ff7b1d3&amp;chksm=ea1cb2b5dd6b3ba3c5ea3836fa531a78f40ac0983dcb3f20e9b7027850a1664f536662b94b3f&amp;scene=21#wechat_redirect" textvalue="单细胞基因集合打分方法-08-UCell包-算法" linktype="text" imgurl="" imgdata="null" data-itemshowtype="11" tab="innerlink" data-linktype="2"><span>单细胞基因集合打分方法-08-UCell包-算法</span></a><br></p><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247513127&amp;idx=1&amp;sn=1cde97aaf329c5146f3cd34ced1bd216&amp;chksm=ea1cb2a5dd6b3bb3124f5e5a828edb1808d002ea4849ed877cc4d715d4d5c616f633bc10b2d6&amp;scene=21#wechat_redirect" textvalue="单细胞转录组揭示局灶节段性肾小球硬化症的炎症特征" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2"><span>单细胞转录组揭示局灶节段性肾小球硬化症的炎症特征</span></a><br></p><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247513107&amp;idx=1&amp;sn=7b623fd2697efca6e0c67545d5f4c763&amp;chksm=ea1cb291dd6b3b8727f6ceac30d0bcf62397c696665be861e526c45d91cef478b6a0d49e851b&amp;scene=21#wechat_redirect" textvalue="细胞互作中的“avoidance ”是什么？" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2"><span>细胞互作中的“avoidance ”是什么？</span></a><br></p><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247512995&amp;idx=1&amp;sn=edbebc945e1de8d0e8e09e9aa4d1ced2&amp;chksm=ea1cb521dd6b3c370ad791e86dbbad69c4199279b555bcf70028d552f48188162a9cb229e5b9&amp;scene=21#wechat_redirect" textvalue="bollito—单细胞转录组的流程化工具" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2"><span>bollito—单细胞转录组的流程化工具</span></a><br></p></section></section><hr><p><br></p></section><section data-style-type="5" data-tools="新媒体排版" data-id="2440475"><hr><p><br></p><hr><section><p>如果你对单细胞转录组研究感兴趣，但又不知道如何入门，也许你可以关注一下下面的课程<span></span></p><p><br></p><ul><li><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247522216&amp;idx=1&amp;sn=7e8a61383489f4a2c68172c4c436fa83&amp;chksm=9b4bd713ac3c5e0525da8dc1d038854366c4ad643396d3ee67f1e8a117c3657530043dda1b8d&amp;scene=21#wechat_redirect" textvalue="生信入门&amp;数据挖掘线上直播课6月班" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2" hasload="1">生信入门&amp;数据挖掘线上直播课6月班</a><br></p></li><li><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247519765&amp;idx=2&amp;sn=6cb33654c7751f4c3df0f84743f77aaf&amp;chksm=9b4bceaeac3c47b8899afc00077b96357b87a4ed6b75e7c434ba14071fd6c8448e4c218de5e0&amp;scene=21#wechat_redirect" textvalue="144线程640Gb内存服务器共享一年仍然是仅需800" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2" hasload="1">144线程640Gb内存服务器共享一年仍然是仅需800</a></p></li><li><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247519765&amp;idx=1&amp;sn=ce5a8c8182f854c88043059f8c2cb9ff&amp;chksm=9b4bceaeac3c47b88c19941d43dbb1401f3a92206481a0afc41159927868199643f795d62a7e&amp;scene=21#wechat_redirect" textvalue="千呼万唤始出来的独享生物信息学云服务器" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2" hasload="1">千呼万唤始出来的独享生物信息学云服务器</a></p></li></ul><p><img data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz_gif/4TKeL1ZejtlKxOib5kmKX6ic6eX0w0WK5jvhtz9yBRsO3OI4yr6S5iaLNM7AbAeuPDHXMvDdur2DRz9wyiax4lEviag/640?wx_fmt=gif" data-type="gif" data-w="240" src="https://mmbiz.qpic.cn/mmbiz_gif/4TKeL1ZejtlKxOib5kmKX6ic6eX0w0WK5jvhtz9yBRsO3OI4yr6S5iaLNM7AbAeuPDHXMvDdur2DRz9wyiax4lEviag/640?wx_fmt=gif"><br></p><p><img data-ratio="0.05278592375366569" data-src="https://mmbiz.qpic.cn/mmbiz/4TKeL1Zejtlq03ZOSZiaTlic1MxgdKiaxTbOZ7ZSe0Xx1Ca8xF3L6Nyj1FYUajtYrSmRIHyZVSsAve0EAvEicZONpg/640?wx_fmt=jpeg" data-type="other" data-w="341" src="https://mmbiz.qpic.cn/mmbiz/4TKeL1Zejtlq03ZOSZiaTlic1MxgdKiaxTbOZ7ZSe0Xx1Ca8xF3L6Nyj1FYUajtYrSmRIHyZVSsAve0EAvEicZONpg/640?wx_fmt=jpeg"></p><p><strong><span>看完记得顺手点个</span></strong><span><strong><span>“在看”</span></strong></span><strong><span>哦！</span></strong></p></section><section><section data-id="93668"><section><section data-width="95%"><section><section><section data-width="38%"><section><section data-tools="135编辑器" data-id="93668"><section><section data-width="95%"><section><section><section data-width="61.8%"><section><section><section><p><br></p><span><strong data-burshtype="text"><img data-copyright="0" data-cropselx1="0" data-cropselx2="109" data-cropsely1="0" data-cropsely2="109" data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz/siaia0BDGJdjRMGrkqo64BGKecYk4akuHpGHVQs7FeOpY7eWbIPGC1tRw5Tw0oEPmx053mR9FTVerWvhuZchIpZw/640?wx_fmt=jpeg" data-type="other" data-w="258" title="qrcode_for_gh_5a3c482ed99f_1280.jpg" src="https://mmbiz.qpic.cn/mmbiz/siaia0BDGJdjRMGrkqo64BGKecYk4akuHpGHVQs7FeOpY7eWbIPGC1tRw5Tw0oEPmx053mR9FTVerWvhuZchIpZw/640?wx_fmt=jpeg"><strong data-burshtype="text">生物</strong><strong data-burshtype="text"> | 单细胞 | 转录组丨资料</strong></strong></span></section><section><span><strong data-burshtype="text">每天都精彩</strong></span></section></section></section><section><section><section><section><section><section><p><span>长按扫码可关注</span></p></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/UIEXAecZTBCjoVOqgri0GA",target="_blank" rel="noopener noreferrer">原文链接</a>
