---
title: "手把手带你复现NC图表之Figure 4"
date: 2023-06-09T14:19:19Z
draft: ["false"]
tags: [
  "fetched",
  "单细胞天地"
]
categories: ["Acdemic"]
---
手把手带你复现NC图表之Figure 4 by 单细胞天地
------
<div><section data-style-type="5" data-tools="新媒体排版" data-id="2430070"><section data-style-type="5" data-tools="新媒体排版" data-id="2390348"><section><section powered-by="xiumi.us"><section><section><section powered-by="xiumi.us"><section><section><br></section><section><section><section powered-by="xiumi.us"><section><section><p>分享是一种态度</p></section></section></section></section></section></section></section></section><section><section powered-by="xiumi.us"><section><section><img data-ratio="0.8738095238095238" data-src="https://mmbiz.qpic.cn/mmbiz_gif/siaia0BDGJdjSnnb9ibXpFagGPWQ0JjSSTNVBZthROEhicfINuLcxX2Ro3Zb600LJhrzeYIb120tLjSVXEtLn14DwA/640?wx_fmt=gif" data-type="gif" data-w="420" width="100%" src="https://mmbiz.qpic.cn/mmbiz_gif/siaia0BDGJdjSnnb9ibXpFagGPWQ0JjSSTNVBZthROEhicfINuLcxX2Ro3Zb600LJhrzeYIb120tLjSVXEtLn14DwA/640?wx_fmt=gif"></section></section></section></section></section></section></section></section></section><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><h2 data-tool="mdnice编辑器"><span></span><span>复现文章信息：</span></h2><blockquote data-tool="mdnice编辑器"><p><span>文章题目</span>：<strong>Single-cell analysis reveals prognostic fibroblast subpopulations linked to molecular and immunological subtypes of lung cancer</strong><br><span>期刊</span>：Nature Communications<br><span>日期</span>：2023年1月31日<br><span>DOI</span>: 10.1038/s41467-023-35832-6</p></blockquote><figure data-tool="mdnice编辑器"><img data-ratio="1.3342592592592593" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjTcic8xtJEV6v6NPncQhdQ9g1y8q0ojh43nRIavAHOkeHNDWzw2P0ibHtvZrXdwVmutIvW1saYIQclA/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjTcic8xtJEV6v6NPncQhdQ9g1y8q0ojh43nRIavAHOkeHNDWzw2P0ibHtvZrXdwVmutIvW1saYIQclA/640?wx_fmt=png"></figure><h2 data-tool="mdnice编辑器"><span></span><span>复现图——Figure 4</span></h2><figure data-tool="mdnice编辑器"><img data-ratio="1.0037926675094817" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjTcic8xtJEV6v6NPncQhdQ9ghYWboBEdxJakHnGgcAjP53XEeKC0SLVbvOvhwk9IMicfD8Xkzp3HU4g/640?wx_fmt=png" data-type="png" data-w="791" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjTcic8xtJEV6v6NPncQhdQ9ghYWboBEdxJakHnGgcAjP53XEeKC0SLVbvOvhwk9IMicfD8Xkzp3HU4g/640?wx_fmt=png"><figcaption>轨迹推断识别与肺泡或外膜成纤维细胞向肌成纤维细胞转分化相关的共有基因模块</figcaption></figure><h3 data-tool="mdnice编辑器"><span></span><span>R包载入与数据准备</span><span></span></h3><p data-tool="mdnice编辑器">代码如下：</p><pre data-tool="mdnice编辑器"><span></span><code><span>library</span>(Seurat)<br><span>library</span>(ggplot2)<br><span>library</span>(WGCNA)<br><span>library</span>(tidyverse)<br><span>library</span>(ggpubr)<br><span>library</span>(ggsci)<br><span>library</span>(destiny)<br><span>library</span>(metap)<br><span>library</span>(pheatmap)<br><span>library</span>(xlsx)<br><span>library</span>(ggplot2)<br><span>library</span>(ggfittext)<br>data_directory &lt;- <span>"H:\\文献复现\\6"</span> <br><span>source</span>(paste0(data_directory, <span>"0_NewFunctions.R"</span>))<br><br>load(paste0(data_directory, <span>"Trajectory Analysis results.Rdata"</span>))<br>load(paste0(data_directory, <span>"IntegratedFibs_Zenodo.Rdata"</span>))<br>load(paste0(data_directory, <span>"MxIHC_Zenodo.Rdata"</span>))<br>REACTOME&lt;-read.xlsx(<span>"41467_2023_35832_MOESM16_ESM.xlsx"</span>,<br>                   as.data.frame=<span>T</span>,<br>                   header=<span>T</span>)<br><span>#在文章中的Supplementary Data 13可以下载</span><br>Fibs.integrated &lt;- AddMetaData(Fibs.integrated, DPT_output)<br>A.M_pT_res_stats2$Gene &lt;- rownames(A.M_pT_res_stats2)<br>T.M_pT_res_stats2$Gene &lt;- rownames(T.M_pT_res_stats2)<br></code></pre><h3 data-tool="mdnice编辑器"><span></span><span>Figure 4A-C</span><span></span></h3><p data-tool="mdnice编辑器">探究从肺泡和外膜成纤维细胞亚群（在对照组织中富集）到肌成纤维细胞（在肿瘤组织中富集）的分化过程，对scRNA-seq数据集进行轨迹推断。结果表明外膜和肺泡成纤维细胞可以作为独立的祖细胞，肌成纤维细胞可以从这些祖细胞转分化</p><pre data-tool="mdnice编辑器"><span></span><code>pA &lt;- ggplot(DM_dim_data, aes(x = DC1, y = DC2, colour = Fibs_subpopulation)) +<br>  theme_pubr(base_size = <span>10</span>) +<br>  theme(axis.text = element_blank(), legend.title = element_blank(), legend.position = <span>"right"</span>) +<br>  geom_point(size = <span>0.1</span>) +<br>  scale_color_manual(values = Fibs_col.palette) +<br>  guides(colour = guide_legend(override.aes = list(size = <span>2</span>)))<br><span>#突出显示分配给每个扩散拟时序（DPT）分支的细胞</span><br>pB &lt;- ggplot(DM_dim_data, aes(x = DC1, y = DC2, colour = factor(dpt_branch))) +<br>  theme_pubr(base_size = <span>10</span>) +<br>  theme(axis.text = element_blank(), legend.position = <span>"right"</span>) +<br>  geom_point(size = <span>0.1</span>) +<br>  scale_color_brewer(name = <span>"DPT branch"</span>) +<br>  guides(colour = guide_legend(override.aes = list(size = <span>2</span>), title.position = <span>"top"</span>, title.hjust =  <span>0.5</span>))<br><span>#对细胞进行拟时序排序，代表它们向肌成纤维细胞表型的相对进展</span><br>pC &lt;- ggplot(DM_dim_data, aes(x = DC1, y = DC2, colour = All_pT)) +<br>  theme_pubr(base_size = <span>10</span>) +<br>  theme(axis.text = element_blank(), legend.key.width = unit(<span>5</span>, <span>"pt"</span>),<br>        legend.position = <span>"right"</span>, legend.key.height = unit(<span>10</span>, <span>"pt"</span>)) +<br>  geom_point(size = <span>0.1</span>) +<br>  scale_color_viridis_c(name = <span>"Pseudotime"</span>) +<br>  guides(colour = guide_colourbar(title.position = <span>"top"</span>, title.hjust =  <span>0.5</span>))<br><br>Fig_4ABC &lt;- ggarrange(pA,<span>NULL</span>,pB,<span>NULL</span>,pC, ncol = <span>5</span>, widths = c(<span>1</span>,<span>0.1</span>,<span>1</span>,<span>0.1</span>), align = <span>"h"</span>,<br>                  labels = c(<span>"a"</span>, <span>""</span>, <span>"b"</span>, <span>""</span>, <span>"c"</span>))<br><br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.5498960498960499" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjTcic8xtJEV6v6NPncQhdQ9g70qqj0n0xdO0LQgtyfA9IBB32qWKfVpUDC2Me2NGnr6pABL7VPLYqg/640?wx_fmt=png" data-type="png" data-w="962" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjTcic8xtJEV6v6NPncQhdQ9g70qqj0n0xdO0LQgtyfA9IBB32qWKfVpUDC2Me2NGnr6pABL7VPLYqg/640?wx_fmt=png"><figcaption>Fig_4ABC</figcaption></figure><h3 data-tool="mdnice编辑器"><span></span><span>Alveolar到Myo</span><span></span></h3><pre data-tool="mdnice编辑器"><span></span><code>names(A.M_pT_res)<br>datExpr_Full &lt;- list()<br><span>for</span>(i <span>in</span> names(A.M_pT_res)){<br>  datExpr_Full[[i]]&lt;- t(A.M_pT_res[[i]]$pT_fit)<br>}<br>sig_genes &lt;- as.character(na.omit(rownames(A.M_pT_res_stats2)[A.M_pT_res_stats2$meta_p_adj.BH &lt; <span>1e-10</span>]))<br>length(sig_genes)<br>datExpr_CTR.TUMOUR &lt;- list()<br><span>for</span>(i <span>in</span> names(datExpr_Full)[c(<span>1</span>:<span>4</span>,<span>7</span>,<span>8</span>)]){<br>  datExpr_CTR.TUMOUR[[i]] &lt;- datExpr_Full[[i]][, sig_genes]<br>}<br>datExpr_CTR.TUMOUR_median &lt;- <br>  Reduce(pmedian, datExpr_CTR.TUMOUR)<br>fit &lt;- hclust(as.dist(<span>1</span>-cor(datExpr_CTR.TUMOUR_median)), method=<span>"ward.D2"</span>)<br>plot(fit) <br>groups &lt;- cutree(fit, k=<span>4</span>)<br>pheatmap(t(datExpr_CTR.TUMOUR_median[, order(groups)]), scale = <span>"row"</span>,<br>                   cluster_cols = <span>F</span>,<br>                   cluster_rows = <span>F</span>,<br>                   clustering_distance_rows = <span>"correlation"</span>,<br>                   clustering_method = <span>"ward.D2"</span>,<br>                   show_rownames = <span>F</span>,<br>                   annotation_row = data.frame(<br>                     row.names = sig_genes,<br>                     cluster = as.factor(groups)<br>                   ))<br>rowMatch &lt;- match(rownames(A.M_pT_res_stats2) ,sig_genes)<br>A.M_pT_res_stats2$Cluster &lt;- as.factor(groups)[rowMatch]<br>A.M_pT_res_stats2$Module &lt;- factor(<br>  A.M_pT_res_stats2$Cluster,<br>  levels = c(<span>1</span>:<span>4</span>),<br>  labels = c(<span>"Progenitor"</span>,<span>"Differentiation"</span>,<span>"Transient"</span>,<span>"Proto.Differentiation"</span>)<br>)<br>levels(A.M_pT_res_stats2$Module)<br>A.M_pT_res_stats2$Module &lt;- factor(A.M_pT_res_stats2$Module, levels = c(<span>"Progenitor"</span>, <span>"Transient"</span>, <span>"Proto.Differentiation"</span>, <span>"Differentiation"</span>))<br>gene_order &lt;- factor(<br>  groups,<br>  levels = c(<span>1</span>:<span>4</span>),<br>  labels = c(<span>"Progenitor"</span>,<span>"Differentiation"</span>,<span>"Transient"</span>,<span>"Proto.Differentiation"</span>)<br>)<br>gene_order &lt;- factor(gene_order, levels = c(<span>"Progenitor"</span>, <span>"Transient"</span>, <span>"Proto.Differentiation"</span>, <span>"Differentiation"</span>))<br>Mod_cols &lt;- RColorBrewer::brewer.pal(<span>5</span>, <span>"Set1"</span>)<br>names(Mod_cols) &lt;- levels(A.M_pT_res_stats2$Module)<br></code></pre><h3 data-tool="mdnice编辑器"><span></span><span>Figure 4D</span><span></span></h3><p data-tool="mdnice编辑器">热图显示DPT中肺泡成纤维细胞进展为肌成纤维细胞时差异表达的基因。用层次聚类法将这些基因分组到DPT表达谱定义的模块中</p><pre data-tool="mdnice编辑器"><span></span><code>AM.PT_TUMOUR_Heatmap &lt;- <br>  pheatmap(t(datExpr_CTR.TUMOUR_median)[order(gene_order), ], scale = <span>"row"</span>,<br>                     cluster_cols = <span>F</span>, cluster_rows = <span>F</span>,<br>                     breaks = seq(-<span>3</span>,<span>3</span>,length=<span>101</span>),<br>                     show_rownames = <span>F</span>, show_colnames = <span>F</span>,<br>                     legend = <span>F</span>,<br>                     annotation_legend = <span>F</span>,<br>                     fontsize = <span>7</span>,<br>                     annotation_names_row = <span>F</span>,<br>                     annotation_names_col = <span>F</span>,<br>                     gaps_row = c(<br>                       sum(as.numeric(gene_order) == <span>1</span>, na.rm = <span>T</span>),<br>                       sum(as.numeric(gene_order) %<span>in</span>% c(<span>1</span>:<span>2</span>), na.rm = <span>T</span>),<br>                       sum(as.numeric(gene_order) %<span>in</span>% c(<span>1</span>:<span>3</span>), na.rm = <span>T</span>),<br>                       sum(as.numeric(gene_order) %<span>in</span>% c(<span>1</span>:<span>4</span>), na.rm = <span>T</span>)<br>                     ),<br>                     annotation_row = data.frame(<br>                       row.names = A.M_pT_res_stats2$Gene[A.M_pT_res_stats2$Gene %<span>in</span>% sig_genes],<br>                       Module = A.M_pT_res_stats2$Module[A.M_pT_res_stats2$Gene %<span>in</span>% sig_genes]<br>                     ),<br>                     annotation_col = data.frame(<br>                       row.names = rownames(datExpr_CTR.TUMOUR_median),<br>                       Pseudotime = as.numeric(rownames(datExpr_CTR.TUMOUR_median))<br>                     ),<br>                     annotation_colors = list(<br>                       Pseudotime = viridis::viridis(<span>3</span>),<br>                       Module = Mod_cols<br>                     )<br>  )<br><br>AM.PT_TUMOUR_Heatmap<br><br>DC_plotting.df &lt;-   data.frame(<br>  DM_dim_data,<br>  Fibs.integrated@meta.data<br>)<br>names(DC_plotting.df)<br>AM_pT_plot &lt;-  <br>  DC_plotting.df %&gt;%<br>  ggplot(aes(x = DC1, y = DC2, colour = A.M_pT)) +<br>  geom_point(size = <span>0.1</span>) + <br>  theme_void(base_size = <span>10</span>) +<br>  theme(legend.position = c(<span>0.1</span>,<span>1</span>), legend.justification = c(<span>"left"</span>, <span>"top"</span>),<br>        legend.direction = <span>"horizontal"</span>,<br>        legend.key.width = unit(<span>8</span>, <span>"pt"</span>), legend.key.height = unit(<span>2</span>, <span>"pt"</span>)) +<br>  guides(colour = guide_colourbar(title.position=<span>"top"</span>, title.hjust = <span>0.5</span>)) +<br>  scale_color_viridis_c(name = <span>"DPT"</span>, na.value = <span>"grey80"</span>) <br>AM_pT_plot<br><br>Fig_4D &lt;- ggarrange(<br>  ggarrange(<span>NULL</span>, AM_pT_plot, ncol = <span>2</span>),<br>  AM.PT_TUMOUR_Heatmap$gtable,<br>  nrow = <span>2</span>, heights = c(<span>2.5</span>,<span>6.5</span>)<br>)<br><br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="1.0884773662551441" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjTcic8xtJEV6v6NPncQhdQ9gFCJPcpo45vEAb6Bh3VicOuLsbc3aXkEYB5N3TQyGAg0FHuuw5lkgvKQ/640?wx_fmt=png" data-type="png" data-w="486" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjTcic8xtJEV6v6NPncQhdQ9gFCJPcpo45vEAb6Bh3VicOuLsbc3aXkEYB5N3TQyGAg0FHuuw5lkgvKQ/640?wx_fmt=png"><figcaption>Fig_4D</figcaption></figure><h3 data-tool="mdnice编辑器"><span></span><span>Adventitial到Myo</span><span></span></h3><pre data-tool="mdnice编辑器"><span></span><code>names(T.M_pT_res)<br>datExpr_Full &lt;- list()<br><span>for</span>(i <span>in</span> names(T.M_pT_res)){<br>  datExpr_Full[[i]]&lt;- t(T.M_pT_res[[i]]<span>$pT_fit</span>)<br>}<br>sig_genes &lt;- as.character(na.omit(rownames(T.M_pT_res_stats2)[T.M_pT_res_stats2<span>$meta_p_adj</span>.BH &lt; 1e-10]))<br>length(sig_genes)<br>datExpr_CTR.TUMOUR &lt;- list()<br><span>for</span>(i <span>in</span> names(datExpr_Full)[c(1:4,7,8)]){<br>  datExpr_CTR.TUMOUR[[i]] &lt;- datExpr_Full[[i]][, sig_genes]<br>}<br>datExpr_CTR.TUMOUR_median &lt;- <br>  Reduce(pmedian, datExpr_CTR.TUMOUR)<br>fit &lt;- hclust(as.dist(1-cor(datExpr_CTR.TUMOUR_median)), method=<span>"ward.D2"</span>)<br>plot(fit) <span># display dendogram</span><br>groups &lt;- cutree(fit, k=4)<br>pheatmap(t(datExpr_CTR.TUMOUR_median[, order(groups)]), scale = <span>"row"</span>,<br>                   cluster_cols = F,<br>                   cluster_rows = F,<br>                   clustering_distance_rows = <span>"correlation"</span>,<br>                   clustering_method = <span>"ward.D2"</span>,<br>                   <span>#cutree_rows = 4,</span><br>                   show_rownames = F,<br>                   annotation_row = data.frame(<br>                     row.names = sig_genes,<br>                     cluster = as.factor(groups)<br>                   ))<br>rowMatch &lt;- match(rownames(T.M_pT_res_stats2) ,sig_genes)<br>T.M_pT_res_stats2<span>$Cluster</span> &lt;- as.factor(groups)[rowMatch]<br>T.M_pT_res_stats2<span>$Module</span> &lt;- factor(<br>  T.M_pT_res_stats2<span>$Cluster</span>,<br>  levels = c(1:4),<br>  labels = c(<span>"Proto.Differentiation"</span>,<span>"Progenitor"</span>,<span>"Transient"</span>,<span>"Differentiation"</span>)<br>)<br>levels(T.M_pT_res_stats2<span>$Module</span>)<br>T.M_pT_res_stats2<span>$Module</span> &lt;- factor(T.M_pT_res_stats2<span>$Module</span>, levels = c(<span>"Progenitor"</span>, <span>"Transient"</span>, <span>"Proto.Differentiation"</span>, <span>"Differentiation"</span>))<br>gene_order &lt;- factor(<br>  groups, levels = c(1:4),<br>  labels = c(<span>"Proto.Differentiation"</span>,<span>"Progenitor"</span>,<span>"Transient"</span>,<span>"Differentiation"</span>)<br>)<br>gene_order &lt;- factor(gene_order, levels = c(<span>"Progenitor"</span>, <span>"Transient"</span>, <span>"Proto.Differentiation"</span>, <span>"Differentiation"</span>))<br>Mod_cols &lt;- RColorBrewer::brewer.pal(5, <span>"Set1"</span>)<br>names(Mod_cols) &lt;- levels(T.M_pT_res_stats2<span>$Module</span>)<br>Heatmap_df &lt;- t(datExpr_CTR.TUMOUR_median)[order(gene_order), ]<br>Heatmap_df &lt;- Heatmap_df[!is.na(gene_order[order(gene_order)]), ]<br>dim(Heatmap_df)<br></code></pre><h3 data-tool="mdnice编辑器"><span></span><span>Figure 4E</span><span></span></h3><p data-tool="mdnice编辑器">热图显示DPT中外膜成纤维细胞进展为肌成纤维细胞时差异表达的基因。用层次聚类法将这些基因分组到DPT表达谱定义的模块中</p><pre data-tool="mdnice编辑器"><span></span><code>TM.PT_TUMOUR_Heatmap &lt;- <br>                     pheatmap(Heatmap_df, scale = <span>"row"</span>,<br>                     cluster_cols = <span>F</span>, cluster_rows = <span>F</span>,<br>                     breaks = seq(-<span>3</span>,<span>3</span>,length=<span>101</span>),<br>                     show_rownames = <span>F</span>, show_colnames = <span>F</span>,<br>                     legend = <span>F</span>,<br>                     annotation_legend = <span>F</span>,<br>                     fontsize = <span>7</span>,<br>                     annotation_names_row = <span>F</span>,<br>                     annotation_names_col = <span>F</span>,<br>                     gaps_row = c(<br>                       sum(as.numeric(gene_order) == <span>1</span>, na.rm = <span>T</span>),<br>                       sum(as.numeric(gene_order) %<span>in</span>% c(<span>1</span>:<span>2</span>), na.rm = <span>T</span>),<br>                       sum(as.numeric(gene_order) %<span>in</span>% c(<span>1</span>:<span>3</span>), na.rm = <span>T</span>),<br>                       sum(as.numeric(gene_order) %<span>in</span>% c(<span>1</span>:<span>4</span>), na.rm = <span>T</span>)<br>                     ),<br>                     annotation_row = data.frame(<br>                       row.names = T.M_pT_res_stats2$Gene[T.M_pT_res_stats2$Gene %<span>in</span>% sig_genes],<br>                       Module = T.M_pT_res_stats2$Module[T.M_pT_res_stats2$Gene %<span>in</span>% sig_genes]<br>                     ),<br>                     annotation_col = data.frame(<br>                       row.names = rownames(datExpr_CTR.TUMOUR_median),<br>                       Pseudotime = as.numeric(rownames(datExpr_CTR.TUMOUR_median))<br>                     ),<br>                     annotation_colors = list(<br>                       Pseudotime = viridis::viridis(<span>3</span>),<br>                       Module = Mod_cols<br>                     )<br>  )<br>TM_pT_plot &lt;-  <br>  DC_plotting.df %&gt;%<br>  ggplot(aes(x = DC1, y = DC2, colour = T.M_pT)) +<br>  geom_point(size = <span>0.1</span>) + <br>  theme_void(base_size = <span>10</span>) +<br>  theme(legend.position = c(<span>0.1</span>,<span>1</span>), legend.justification = c(<span>"left"</span>, <span>"top"</span>),<br>        legend.direction = <span>"horizontal"</span>,<br>        legend.key.width = unit(<span>8</span>, <span>"pt"</span>), legend.key.height = unit(<span>2</span>, <span>"pt"</span>)) +<br>  guides(colour = guide_colourbar(title.position=<span>"top"</span>, title.hjust = <span>0.5</span>)) +<br>  scale_color_viridis_c(name = <span>"DPT"</span>, na.value = <span>"grey80"</span>) <br>TM_pT_plot<br><br>Fig_4E &lt;- ggarrange(<br>  ggarrange(<span>NULL</span>, TM_pT_plot, ncol = <span>2</span>),<br>  TM.PT_TUMOUR_Heatmap$gtable,<br>  nrow = <span>2</span>, heights = c(<span>2.5</span>,<span>6.5</span>)<br>)<br><br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="1.0884773662551441" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjTcic8xtJEV6v6NPncQhdQ9gias3iaHu4RNMKyhHXJmlL1tSmEl30065SavSvIC2wNQW84vRLeB2e3lQ/640?wx_fmt=png" data-type="png" data-w="486" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjTcic8xtJEV6v6NPncQhdQ9gias3iaHu4RNMKyhHXJmlL1tSmEl30065SavSvIC2wNQW84vRLeB2e3lQ/640?wx_fmt=png"><figcaption>Fig_4E</figcaption></figure><p data-tool="mdnice编辑器">这鉴定了两条轨迹上的四个模块，代表转分化的不同阶段：祖细胞、早期激活、原分化和分化</p><h3 data-tool="mdnice编辑器"><span></span><span>Figure 4F</span><span></span></h3><p data-tool="mdnice编辑器">损失曲线图显示了由对照和肿瘤样本或仅对照组织样本组成的数据集中每个共有DPT模块的表达谱。共有模块由在肺泡成纤维细胞至肌成纤维细胞和外膜成纤维细胞至肌成纤维细胞轨迹中分配到相同簇的基因组成，列出了每个模块的10个代表性基因</p><pre data-tool="mdnice编辑器"><span></span><code>Mod.Overlap_df &lt;-  cbind(A.M_pT_res_stats2[,c(<span>"Gene"</span>, <span>"Module"</span>, <span>"meta_p_adj.BH"</span>)],<br>                         T.M_pT_res_stats2[,c(<span>"Gene"</span>, <span>"Module"</span>, <span>"meta_p_adj.BH"</span>)])<br>names(Mod.Overlap_df) &lt;- c(<span>"GeneA"</span>, <span>"Alv2M_Module"</span>, <span>"Alv2M_adjP"</span>, <span>"GeneB"</span>, <span>"Adv2M_Module"</span>, <span>"Adv2M_adjP"</span>)<br>Core.Mods &lt;- Mod.Overlap_df[Mod.Overlap_df$Alv2M_Module == Mod.Overlap_df$Adv2M_Module, ]<br>Module.Genes_list &lt;- list()<br><span>for</span>(i <span>in</span> levels(Core.Mods$Alv2M_Module)) {<br>  Module.Genes_list[[i]] &lt;- as.character(na.omit(Core.Mods$GeneA[Core.Mods$Alv2M_Module == i]))<br>}<br>names(Fibs.integrated@meta.data)<br>DefaultAssay(Fibs.integrated) &lt;- <span>"RNA"</span><br>Fibs.integrated &lt;- <br>  AddModuleScore(Fibs.integrated,<br>                 features = Module.Genes_list)<br>names(Fibs.integrated@meta.data)<br>names(Fibs.integrated@meta.data)[grep(<span>"^Cluster"</span>, names(Fibs.integrated@meta.data))] &lt;-<br>  paste0(<span>"Core_"</span>, levels(T.M_pT_res_stats2$Module))<br>pT_df &lt;- data.frame(<br>  Fibs.integrated@meta.data[, c(<span>"Sample.type"</span>, <span>"All_pT"</span>, <span>"dpt_branch"</span>, <span>"Fibs_MajorClusters"</span>, <span>"Dataset"</span>, <span>"Sample.Subtype"</span>,<br>                                paste(<span>"Core"</span>, levels(T.M_pT_res_stats2$Module), sep = <span>"_"</span>))]<br>  <br>)<br>variables = paste(<span>"Core"</span>, levels(T.M_pT_res_stats2$Module), sep = <span>"_"</span>)<br>pT_df_long &lt;- reshape2::melt(pT_df[, c(<span>"All_pT"</span>, <span>"Sample.type"</span>, <span>"Dataset"</span>,variables)],<br>                             id.vars = c(<span>"All_pT"</span>,<span>"Sample.type"</span>, <span>"Dataset"</span>), na.rm = <span>T</span>)<br>levels(pT_df_long$variable)<br>pT_df_long$Dataset_ordered &lt;- <br>  factor(pT_df_long$Dataset, levels = levels(pT_df_long$Dataset) )<br>pT_df_long$Dataset_type &lt;- <br>  factor(pT_df_long$Dataset_ordered, levels = levels(pT_df_long$Dataset_ordered),<br>         labels = c(<span>"CTR &amp; Tumour"</span>, <span>"CTR Only"</span>, <span>"CTR Only"</span>, rep(<span>"CTR &amp; Tumour"</span>, <span>6</span>)))<br>names(pT_df_long)<br>levels(pT_df_long$variable)<br>Mod_cols2 &lt;- Mod_cols<br>names(Mod_cols2) &lt;- levels(pT_df_long$variable)<br>pT_df_long$Mod.label &lt;- factor(pT_df_long$variable,<br>                               labels = c(<span>"Progenitor"</span>, <span>"Early-activation"</span>,<br>                                          <span>"Proto-differentiation"</span>, <span>"Differentiation"</span>))<br>Fig_4F &lt;- <br>  pT_df_long[!is.na(pT_df_long$variable), ] %&gt;%<br>  ggplot(aes(x = All_pT, y = value, colour = variable)) +<br>  facet_grid(Mod.label~Dataset_type, scales = <span>"free_y"</span>) + <br>  geom_smooth(method = <span>"loess"</span>) +<br>  guides(colour = guide_legend(override.aes = list(size = <span>2</span>))) +<br>  theme_bw(base_size = <span>15</span>) +<br>  theme(legend.position = <span>"none"</span>, panel.grid = element_blank(),<br>        strip.text = element_text(size = <span>10</span>)) +<br>  xlab(<span>"Pseudotime"</span>) + ylab(<span>"Module Score"</span>) +<br>  scale_color_manual(values = Mod_cols2)<br><br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.9268635724331927" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjTcic8xtJEV6v6NPncQhdQ9ghIqNuauKZUFibPVq82CFKyVxPGW41n6G9IaiaCss3bNolGUWDWoUfP8g/640?wx_fmt=png" data-type="png" data-w="711" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjTcic8xtJEV6v6NPncQhdQ9ghIqNuauKZUFibPVq82CFKyVxPGW41n6G9IaiaCss3bNolGUWDWoUfP8g/640?wx_fmt=png"><figcaption>Fig_4F</figcaption></figure><h3 data-tool="mdnice编辑器"><span></span><span>Figure 4G</span><span></span></h3><p data-tool="mdnice编辑器">箱形图显示按组织类型和拟时序五分位数分组的细胞中共有DPT模块的表达</p><pre data-tool="mdnice编辑器"><span></span><code>Fibs.integrated$All_pT_5cat &lt;- factor(Hmisc::cut2(Fibs.integrated$All_pT, g = <span>5</span>),<br>                                      labels = paste0(<span>"pT q"</span>, <span>1</span>:<span>5</span>))<br><br><span>#Early−activation模块</span><br>test &lt;- aggregate(x =Fibs.integrated@meta.data,<br>                  by =Core_Transient ~ All_pT_5cat + SampleID + Sample.type,<br>                  FUN = median)<br>Transient.boxplot &lt;- <br>  test %&gt;%<br>  drop_na(All_pT_5cat) %&gt;%<br>  ggplot(aes(x = Sample.type, y = Core_Transient, fill = Sample.type)) +<br>  theme_pubr(base_size = <span>15</span>) +<br>  theme(axis.text.x = element_blank()) +<br>  geom_boxplot(outlier.shape = <span>NA</span>) +<br>  geom_jitter(width = <span>0.2</span>, alpha = <span>0.5</span>, size = <span>0.1</span>) +<br>  facet_grid(~All_pT_5cat) +<br>  ylab(<span>"Early-activation Module\n(Normalised exprs)"</span>) +<br>  stat_compare_means(comparisons = list(c(<span>"Control"</span>, <span>"Tumour"</span>)),<br>                     size = <span>2.5</span>,<br>                     bracket.size = <span>0.1</span>, tip.length = <span>0.01</span>,<br>                     label.y = max(test$Core_Transient)+<span>0.1</span>,<br>                     <span>#vjust = 0.5,</span><br>                     hide.ns = <span>T</span>)+<br>  expand_limits(y = <span>2</span>)<br><span>#Proto−Differentation模块</span><br>test &lt;- aggregate(x =Fibs.integrated@meta.data,<br>                  by =Core_Proto.Differentiation ~ All_pT_5cat + SampleID + Sample.type,<br>                  FUN = median)<br><br>PD.boxplot &lt;- <br>  test %&gt;%<br>  drop_na(All_pT_5cat) %&gt;%<br>  ggplot(aes(x = Sample.type, y = Core_Proto.Differentiation, fill = Sample.type)) +<br>  theme_pubr(base_size = <span>15</span>) +<br>  theme(axis.text.x = element_blank()) +<br>  geom_boxplot(outlier.shape = <span>NA</span>) +<br>  geom_jitter(width = <span>0.2</span>, alpha = <span>0.5</span>, size = <span>0.1</span>) +<br>  facet_grid(~All_pT_5cat) +<br>  ylab(<span>"Proto-Differentation Module\n(Normalised exprs)"</span>) +<br>  stat_compare_means(comparisons = list(c(<span>"Control"</span>, <span>"Tumour"</span>)),<br>                     size = <span>2.5</span>,<br>                     bracket.size = <span>0.1</span>, tip.length = <span>0.01</span>,<br>                     label.y = max(test$Core_Proto.Differentiation)+<span>0.1</span>,<br>                     <span>#vjust = 0.5,</span><br>                     hide.ns = <span>T</span>)+<br>  expand_limits(y = <span>2</span>)<br><br><span>#Differentation模块</span><br>test &lt;- aggregate(x =Fibs.integrated@meta.data,<br>                  by =Core_Differentiation ~ All_pT_5cat + SampleID + Sample.type,<br>                  FUN = median)<br>Differentiation.boxplot &lt;- <br>  test %&gt;%<br>  drop_na(All_pT_5cat) %&gt;%<br>  ggplot(aes(x = Sample.type, y = Core_Differentiation, fill = Sample.type)) +<br>  theme_pubr(base_size = <span>15</span>) +<br>  theme(axis.text.x = element_blank()) +<br>  geom_boxplot(outlier.shape = <span>NA</span>) +<br>  geom_jitter(width = <span>0.2</span>, alpha = <span>0.5</span>, size = <span>0.1</span>) +<br>  facet_grid(~All_pT_5cat) +<br>  ylab(<span>"Differentation Module\n(Normalised exprs)"</span>) +<br>  stat_compare_means(comparisons = list(c(<span>"Control"</span>, <span>"Tumour"</span>)),<br>                     size = <span>2.5</span>,<br>                     bracket.size = <span>0.1</span>, tip.length = <span>0.01</span>,<br>                     label.y = max(test$Core_Differentiation)+<span>0.1</span>,<br>                     hide.ns = <span>T</span>)+<br>  expand_limits(y = <span>1.5</span>)<br><br>Fig_4G &lt;- <br>  ggarrange(<br>    Transient.boxplot,<br>    PD.boxplot,<br>    Differentiation.boxplot,<br>    nrow = <span>3</span>, common.legend = <span>T</span>, legend = <span>"right"</span>)<br><br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.9118303571428571" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjTcic8xtJEV6v6NPncQhdQ9gug8UVWpAHkjibxyaqSLTv8fZvjlJUjNq5UHJRHLFqWbFh8Qjou5a9Pg/640?wx_fmt=png" data-type="png" data-w="896" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjTcic8xtJEV6v6NPncQhdQ9gug8UVWpAHkjibxyaqSLTv8fZvjlJUjNq5UHJRHLFqWbFh8Qjou5a9Pg/640?wx_fmt=png"><figcaption>Fig_4G</figcaption></figure><h3 data-tool="mdnice编辑器"><span></span><span>Figure 4H</span><span></span></h3><p data-tool="mdnice编辑器">条形图显示每个共识DPT模块的REACTOME途径富集结果</p><pre data-tool="mdnice编辑器"><span></span><code>REACTOME&lt;-REACTOME[c(<span>5</span>,<span>1</span>,<span>18</span>,<span>49</span>,<span>59</span>,<span>58</span>),]<br>REACTOME$<span>"Pathway genes in Consensus Module (%)"</span>&lt;-(REACTOME$Coverage.pct)*<span>100</span><br>REACTOME&lt;-REACTOME[rev(rownames(REACTOME)),]<br>col&lt;-rep(c(<span>"#984EA3"</span>,<span>"#4DAF4A"</span>,<span>"#377EB8"</span>),each = <span>2</span>)<br>names(col)&lt;-REACTOME$Name<br>ggplot(REACTOME, aes(x=factor(Name,levels=REACTOME$Name), y=`Pathway genes in Consensus Module (%)`, <br>                     fill = Name,label=paste0(REACTOME$Name,<span>" (adj.P="</span>, signif(REACTOME$q.value.FDR.B.H, <span>3</span>),<span>") "</span>)))+<br>  geom_bar(stat = <span>"identity"</span>, position = <span>"dodge"</span>,color = <span>"black"</span>)+<br>  labs(x=<span>"REACTOME pathway"</span>, y=<span>"Pathway genes in Consensus Module (%)"</span>, fill=<span>"Module"</span>) +<br>  theme_bw() +<br>  theme(axis.text.y = element_blank())+<br>  coord_flip()+<br>  scale_fill_manual(values = col)+theme(legend.position = <span>"none"</span>)+<br>  geom_bar_text(position=<span>"stack"</span>,<br>                place=<span>"left"</span>,<br>                color=<span>"white"</span>,<br>                min.size = <span>10</span>,<br>                reflow = <span>TRUE</span>)+<br>  scale_y_continuous(expand = c(<span>0</span>,<span>0</span>),<br>                     limits = c(<span>0</span>,<span>55</span>))+ <br>  scale_x_discrete(expand=c(<span>0</span>,<span>0</span>))<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.6138888888888889" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjTcic8xtJEV6v6NPncQhdQ9gYk30JJXT1d7svaROhU4j6C9sPlGSuIFBC8JS3fAJNzkgngtyCjBrMQ/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjTcic8xtJEV6v6NPncQhdQ9gYk30JJXT1d7svaROhU4j6C9sPlGSuIFBC8JS3fAJNzkgngtyCjBrMQ/640?wx_fmt=png"></figure><h3 data-tool="mdnice编辑器"><span></span><span>Figure 4I</span><span></span></h3><p data-tool="mdnice编辑器">线图显示了通过mxIHC的每个成纤维细胞亚群的配对肿瘤和相邻正常组织之间的平均HSPA1A表达水平</p><pre data-tool="mdnice编辑器"><span></span><code>Fibs_HSPA1A.agg &lt;- aggregate(<br>  MxIHC_Fibroblasts$HSPA1A,<br>  by = list(Sample = MxIHC_Fibroblasts$Slide.name,<br>            Subpop = MxIHC_Fibroblasts$Cell.type,<br>            TvN = MxIHC_Fibroblasts$Sample.Region.TvN,<br>            Subtype = MxIHC_Fibroblasts$Subtype,<br>            Inflam.Fibro = MxIHC_Fibroblasts$Control...Fibrosis.Inflammation,<br>            exclusion = MxIHC_Fibroblasts$Include.Normal.95pct),<br>  median)<br><br>Fig_4I &lt;- <br>  Fibs_HSPA1A.agg %&gt;%<br>  drop_na(TvN) %&gt;%<br>  filter(!Sample == <span>"Drop-Seq Cohort 1"</span>) %&gt;%<br>  ggplot(aes(x = TvN, y = x)) +<br>  theme_pubr(base_size = <span>15</span>) +<br>  theme(legend.position = <span>"none"</span>, axis.title.x = element_blank()) +<br>  facet_grid(~Subpop) +<br>  geom_point(size = <span>0.1</span>, aes(colour = Subpop)) +<br>  geom_line(aes(group = Sample, colour = Subpop)) +<br>  scale_colour_manual(values = Fibs_col.palette) +<br>  rotate_x_text(angle = <span>45</span>) +<br>  ylab(<span>"HSPA1A\n(MxIHC Median per cell exprs)"</span>) +<br>  stat_compare_means(comparisons = list(c(<span>"Control"</span>, <span>"Tumour"</span>)), paired = <span>T</span>,<br>                     size = <span>4</span>,<br>                     method = <span>"wilcox"</span>, label.y = <span>0.18</span>, tip.length = <span>0.01</span>)+<br>  expand_limits(y = <span>0.2</span>)<br><br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.6775" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjTcic8xtJEV6v6NPncQhdQ9g9XhCsUKBfEYTTRWr9GE8lUg4VCEXiaczJQ6ibmwQcFrFbGvsf5b4kQBw/640?wx_fmt=png" data-type="png" data-w="800" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjTcic8xtJEV6v6NPncQhdQ9g9XhCsUKBfEYTTRWr9GE8lUg4VCEXiaczJQ6ibmwQcFrFbGvsf5b4kQBw/640?wx_fmt=png"><figcaption>Fig_4I</figcaption></figure><h2 data-tool="mdnice编辑器"><span></span><span>Figure 4</span></h2><p data-tool="mdnice编辑器">这些结果表明了两种成纤维细胞亚群(外膜和肺泡)都可以作为肌成纤维细胞的组织驻留祖细胞。这些数据还表明，无论祖细胞亚群如何，转分化过程都重要:其涉及炎症基因上调的短暂阶段，独立于肿瘤的相互作用关系;随后是涉及热休克反应信号的原始分化，通过与肿瘤的相互作用而增加;最终导致完全分化的肌成纤维细胞表型，其ECM组织和胶原形成能力增加，这在肿瘤相关刺激下显著增强。</p></section><section><section data-style-type="5" data-tools="新媒体排版" data-id="2440476"><section><section><section><section><img data-ratio="0.9495798319327731" data-src="https://mmbiz.qpic.cn/mmbiz_gif/09gp6SvPE04j3m2v7Hr889icHUyibTOHs8YuUibicl7ibRD0ZwG5pDTjBluRreZvuib1o3BibvLkicYhnA4YW7dQsjn0cA/640?wx_fmt=gif" data-type="gif" data-w="119" data-width="100%" src="https://mmbiz.qpic.cn/mmbiz_gif/09gp6SvPE04j3m2v7Hr889icHUyibTOHs8YuUibicl7ibRD0ZwG5pDTjBluRreZvuib1o3BibvLkicYhnA4YW7dQsjn0cA/640?wx_fmt=gif"></section><section data-brushtype="text">往期回顾</section><section><br></section></section></section></section><section><section data-autoskip="1"><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247513351&amp;idx=1&amp;sn=9d0c9d69470a1aa148d2746ede7822b1&amp;chksm=ea1cb385dd6b3a93235bc954878a6ecbb787661f301efe16a8744ad0edd4e9e676efbb7cf931&amp;scene=21#wechat_redirect" textvalue="单细胞转录组联合脂质代谢提供肺癌的早筛新方法" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2"><span>单细胞转录组联合脂质代谢提供肺癌的早筛新方法</span></a><br></p><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247513321&amp;idx=1&amp;sn=3def521dfeaf54a3b364833b7a591747&amp;chksm=ea1cb26bdd6b3b7da08d86a9468ee85a05d1c1380dd571c37decf60aca440d94cd70a3080892&amp;scene=21#wechat_redirect" textvalue="单细胞基因集合打分方法-T01-横向测评" linktype="text" imgurl="" imgdata="null" data-itemshowtype="11" tab="innerlink" data-linktype="2"><span>单细胞基因集合打分方法-T01-横向测评</span></a><br></p><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247513310&amp;idx=1&amp;sn=5aa8db2bdc7c6ca1b1bdb6d1a9863130&amp;chksm=ea1cb25cdd6b3b4a2edf5a6c816f9e5ac46f6667dabe4adc2995a3db7a98e065c044004487d0&amp;scene=21#wechat_redirect" textvalue="手把手带你复现NC图表之Figure 3" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2"><span>手把手带你复现NC图表之Figure 3</span></a><br></p><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247513202&amp;idx=1&amp;sn=3f8a4947cce8596f454fb82090f599e0&amp;chksm=ea1cb2f0dd6b3be68237d48393e535a8399b9e0e8a091dcc225b3bce2100c6fc4d31a0fd027b&amp;scene=21#wechat_redirect" textvalue="手把手带你复现NC图表之Figure 2" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2"><span>手把手带你复现NC图表之Figure 2</span></a><br></p><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247513057&amp;idx=1&amp;sn=6f737f0a5ae7795645a42894897f7b87&amp;chksm=ea1cb563dd6b3c7567db78011efb52823c1f6bc94a760cc11ef8a07d3d5d6c779e4813c4d523&amp;scene=21#wechat_redirect" textvalue="手把手带你复现NC图表之Figure 1" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2"><span>手把手带你复现NC图表之Figure 1</span></a><br></p></section></section><hr><p><br></p></section><section data-style-type="5" data-tools="新媒体排版" data-id="2440475"><hr><p><br></p><hr><section><p>如果你对单细胞转录组研究感兴趣，但又不知道如何入门，也许你可以关注一下下面的课程<span></span></p><p><br></p><ul><li><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247522216&amp;idx=1&amp;sn=7e8a61383489f4a2c68172c4c436fa83&amp;chksm=9b4bd713ac3c5e0525da8dc1d038854366c4ad643396d3ee67f1e8a117c3657530043dda1b8d&amp;scene=21#wechat_redirect" textvalue="生信入门&amp;数据挖掘线上直播课6月班" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2" hasload="1">生信入门&amp;数据挖掘线上直播课6月班</a><br></p></li><li><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247519765&amp;idx=2&amp;sn=6cb33654c7751f4c3df0f84743f77aaf&amp;chksm=9b4bceaeac3c47b8899afc00077b96357b87a4ed6b75e7c434ba14071fd6c8448e4c218de5e0&amp;scene=21#wechat_redirect" textvalue="144线程640Gb内存服务器共享一年仍然是仅需800" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2" hasload="1">144线程640Gb内存服务器共享一年仍然是仅需800</a></p></li><li><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247519765&amp;idx=1&amp;sn=ce5a8c8182f854c88043059f8c2cb9ff&amp;chksm=9b4bceaeac3c47b88c19941d43dbb1401f3a92206481a0afc41159927868199643f795d62a7e&amp;scene=21#wechat_redirect" textvalue="千呼万唤始出来的独享生物信息学云服务器" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2" hasload="1">千呼万唤始出来的独享生物信息学云服务器</a></p></li></ul><p><img data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz_gif/4TKeL1ZejtlKxOib5kmKX6ic6eX0w0WK5jvhtz9yBRsO3OI4yr6S5iaLNM7AbAeuPDHXMvDdur2DRz9wyiax4lEviag/640?wx_fmt=gif" data-type="gif" data-w="240" src="https://mmbiz.qpic.cn/mmbiz_gif/4TKeL1ZejtlKxOib5kmKX6ic6eX0w0WK5jvhtz9yBRsO3OI4yr6S5iaLNM7AbAeuPDHXMvDdur2DRz9wyiax4lEviag/640?wx_fmt=gif"><br></p><p><img data-ratio="0.05278592375366569" data-src="https://mmbiz.qpic.cn/mmbiz/4TKeL1Zejtlq03ZOSZiaTlic1MxgdKiaxTbOZ7ZSe0Xx1Ca8xF3L6Nyj1FYUajtYrSmRIHyZVSsAve0EAvEicZONpg/640?wx_fmt=jpeg" data-type="other" data-w="341" src="https://mmbiz.qpic.cn/mmbiz/4TKeL1Zejtlq03ZOSZiaTlic1MxgdKiaxTbOZ7ZSe0Xx1Ca8xF3L6Nyj1FYUajtYrSmRIHyZVSsAve0EAvEicZONpg/640?wx_fmt=jpeg"></p><p><strong><span>看完记得顺手点个</span></strong><span><strong><span>“在看”</span></strong></span><strong><span>哦！</span></strong></p></section><section><section data-id="93668"><section><section data-width="95%"><section><section><section data-width="38%"><section><section data-tools="135编辑器" data-id="93668"><section><section data-width="95%"><section><section><section data-width="61.8%"><section><section><section><p><br></p><span><strong data-burshtype="text"><img data-copyright="0" data-cropselx1="0" data-cropselx2="109" data-cropsely1="0" data-cropsely2="109" data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz/siaia0BDGJdjRMGrkqo64BGKecYk4akuHpGHVQs7FeOpY7eWbIPGC1tRw5Tw0oEPmx053mR9FTVerWvhuZchIpZw/640?wx_fmt=jpeg" data-type="other" data-w="258" title="qrcode_for_gh_5a3c482ed99f_1280.jpg" src="https://mmbiz.qpic.cn/mmbiz/siaia0BDGJdjRMGrkqo64BGKecYk4akuHpGHVQs7FeOpY7eWbIPGC1tRw5Tw0oEPmx053mR9FTVerWvhuZchIpZw/640?wx_fmt=jpeg"><strong data-burshtype="text">生物</strong><strong data-burshtype="text"> | 单细胞 | 转录组丨资料</strong></strong></span></section><section><span><strong data-burshtype="text">每天都精彩</strong></span></section></section></section><section><section><section><section><section><section><p><span>长按扫码可关注</span></p></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/1ILrbhdfY44WrVH9ZxOciQ",target="_blank" rel="noopener noreferrer">原文链接</a>
