---
title: "拟时序| monocle3+velo.R+scvelo呈现"
date: 2023-06-16T00:43:12Z
draft: ["false"]
tags: [
  "fetched",
  "朴素的科研打工仔"
]
categories: ["Acdemic"]
---
拟时序| monocle3+velo.R+scvelo呈现 by 朴素的科研打工仔
------
<div><p><span><strong> </strong></span></p><p><span><strong>写在前面的话</strong></span></p><p><span><strong>    </strong> </span><span>拟时序分析已经是单细胞和空间转录组中的应用文章中的必需分析，干货记录分享，简单易懂！</span></p><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><h2 data-tool="mdnice编辑器"><span></span><span>提取亚群再细分</span></h2><pre data-tool="mdnice编辑器"><span></span><code>library(Seurat)<br>library(monocle3)<br>library(tidyverse)<br>library(patchwork)<br>## 提取只含T细胞的子集<br>scRNA.T &lt;- subset(scRNA,idents=<span>"Tc"</span>)<br>## 重新降维聚类<br>scRNA.T &lt;- NormalizeData(scRNA.T, normalization.method = <span>"LogNormalize"</span>, scale.factor = <span>1e4</span>) <br>scRNA.T &lt;- FindVariableFeatures(scRNA.T, selection.method = <span>'vst'</span>, nfeatures = <span>3000</span>)<br>scRNA.T &lt;- ScaleData(scRNA.T,rownames(scRNA.T))<br>scRNA.T &lt;- RunPCA(scRNA.T, features = VariableFeatures(object = scRNA.T)) <br>library(harmony)<br>scRNA.T &lt;- RunHarmony(scRNA.T, <span>"orig.ident"</span>)<br>scRNA.T &lt;- RunUMAP(scRNA.T,dims = <span>1</span>:<span>30</span>, <br>              reduction = <span>"harmony"</span>)<br>scRNA.T &lt;- FindNeighbors(scRNA.T, reduction = <span>"harmony"</span>,<br>                    dims = <span>1</span>:<span>30</span>) <br><span>for</span> (<span>res in <span>c</span><span>(<span>0.01</span>, <span>0.05</span>, <span>0.1</span>, <span>0.2</span>, <span>0.3</span>, <span>0.5</span>,<span>0.8</span>,<span>1</span>)</span>) </span>{<br>  scRNA.T=FindClusters(scRNA.T, #graph.name = <span>"CCA_snn"</span>, <br>                  resolution = res, algorithm = <span>1</span>)<br>}<br>apply(scRNA.T<span>@meta</span>.data[,grep(<span>"RNA_snn"</span>,colnames(scRNA.T<span>@meta</span>.data))],<span>2</span>,table)<br>Idents(scRNA.T)&lt;- <span>"RNA_snn_res.0.1"</span><br>p1 &lt;- DimPlot(scRNA.T, reduction = <span>"umap"</span>,<br>        label = T,label.box = T)+xlim(-<span>15</span>,<span>15</span>) <br></code></pre><h2 data-tool="mdnice编辑器"><span></span><span>monocle3</span></h2><pre data-tool="mdnice编辑器"><span></span><code>## monocle3 轨迹分析<br># ##创建CDS对象并预处理数据<br>data &lt;- GetAssayData(scRNA.T, assay = <span>'RNA'</span>, slot = <span>'counts'</span>)<br>cell_metadata &lt;- scRNA.T<span>@meta</span>.data<br>gene_annotation &lt;- data.frame(gene_short_name = rownames(data))<br>rownames(gene_annotation) &lt;- rownames(data)<br>cds &lt;- new_cell_data_set(data,<br>                         cell_metadata = cell_metadata,<br>                         gene_metadata = gene_annotation)<br>#preprocess_cds函数相当于seurat中NormalizeData+ScaleData+RunPCA<br>cds &lt;- preprocess_cds(cds, num_dim = <span>50</span>)<br>##从seurat导入整合过的umap坐标<br>cds.embed &lt;- cds<span>@int</span>_colData$reducedDims$UMAP<br><span>int</span>.embed &lt;- Embeddings(scRNA.T, reduction = <span>"umap"</span>)<br><span>int</span>.embed &lt;- <span>int</span>.embed[rownames(cds.embed),]<br>cds<span>@int</span>_colData$reducedDims$UMAP &lt;- <span>int</span>.embed<br>p2 &lt;- plot_cells(cds, reduction_method=<span>"UMAP"</span>, color_cells_by=<span>"celltype1"</span>) + ggtitle(<span>'int.umap'</span>)<br>## Monocle3聚类分区<br>cds &lt;- cluster_cells(cds)<br>p1 &lt;- plot_cells(cds, show_trajectory_graph = FALSE) + ggtitle(<span>"label by clusterID"</span>)<br>p2 &lt;- plot_cells(cds, color_cells_by = <span>"partition"</span>, show_trajectory_graph = FALSE) + <br>  ggtitle(<span>"label by partitionID"</span>)<br>p = wrap_plots(p1, p2)<br>## 识别轨迹<br>cds &lt;- learn_graph(cds)<br>p = plot_cells(cds, label_groups_by_cluster = FALSE, label_leaves = FALSE, <br>               label_branch_points = FALSE,color_cells_by = <span>"celltype"</span>, group_label_size=<span>4</span>,cell_size=<span>1.5</span>)<br># 确定root<br>cds = order_cells(cds)<br><br>plot_cells(cds,<br>           color_cells_by = <span>"pseudotime"</span>,<br>           label_cell_groups=FALSE,<br>           label_leaves=TRUE,<br>           label_branch_points=TRUE,<br>           graph_label_size=<span>1.5</span>,<br>           group_label_size=<span>4</span>,cell_size=<span>1.5</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.90625" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/Ze2f5C6nc67v8NLTMS9nmJ8Wn6Y2icgrqkNK9P6uxuYbooeZy43Sd1yRKUmW6FablTdicXtO8JKxSWE35C9wvIhg/640?wx_fmt=jpeg" data-type="jpeg" data-w="992" src="https://mmbiz.qpic.cn/mmbiz_jpg/Ze2f5C6nc67v8NLTMS9nmJ8Wn6Y2icgrqkNK9P6uxuYbooeZy43Sd1yRKUmW6FablTdicXtO8JKxSWE35C9wvIhg/640?wx_fmt=jpeg"></figure><h2 data-tool="mdnice编辑器"><span></span><span>velocyto.R</span></h2><pre data-tool="mdnice编辑器"><span></span><code>#loom 文件生成<br>cellranger_gtf=$HOME/reference/scgenome.gtf<br>rmsk_gtf=$HOME/reference/Sus_repeat_rmsk.gtf  ## UCSC 版本确定<br>cellranger_outDir=/ifs/home/yusenwei/data/project/N1<br>velocyto run10x -<span>m $rmsk_gtf  $cellranger_outDir $cellranger_gtf<br>R<br><span>library</span><span>(velocyto.R)</span><br><span>library</span><span>(Seurat)</span><br><span>library</span><span>(SeuratWrappers)</span><br>scRNA.T &lt;- <span>readRDS</span><span>(<span>"./scRNA.T.rds"</span>)</span><br>#载入 loom文件<br>N1</span>= read.loom.matrices(<span>"./N1/velocyto/N2.loom"</span>)<br>N2= read.loom.matrices(<span>"./N2/velocyto/N2.loom"</span>)<br>N3= read.loom.matrices(<span>"./N3/velocyto/N3.loom"</span>)<br>R1= read.loom.matrices(<span>"./R1/velocyto/R1.loom"</span>)<br>R2= read.loom.matrices(<span>"./R2/velocyto/R2.loom"</span>)<br>R3= read.loom.matrices(<span>"./R3/velocyto/R3.loom"</span>)<br>#修改列名<br>N1$spliced[<span>1</span>:<span>5</span>, <span>1</span>:<span>5</span>]<br>N1 &lt;- lapply(N1, function(x) {<br>  colnames(x) &lt;- sub(<span>"x"</span>, <span>"-1"</span>, sub(<span>"N1"</span>, <span>"N1_"</span>, colnames(x)))<br>  x<br>})<br>N2&lt;- lapply(N2, function(x) {<br>  colnames(x) &lt;- sub(<span>"x"</span>, <span>"-1"</span>, sub(<span>"N2"</span>, <span>"N2_"</span>, colnames(x)))<br>  x<br>})<br>N3&lt;- lapply(N3, function(x) {<br>  colnames(x) &lt;- sub(<span>"x"</span>, <span>"-1"</span>, sub(<span>"N3"</span>, <span>"N3_"</span>, colnames(x)))<br>  x<br>})<br>R1 &lt;- lapply(R1, function(x) {<br>  colnames(x) &lt;- sub(<span>"x"</span>, <span>"-1"</span>, sub(<span>"R1"</span>, <span>"R1_"</span>, colnames(x)))<br>  x<br>})<br>R2 &lt;- lapply(IUGR2, function(x) {<br>  colnames(x) &lt;- sub(<span>"x"</span>, <span>"-1"</span>, sub(<span>"R2"</span>, <span>"R2_"</span>, colnames(x)))<br>  x<br>})<br>R3 &lt;- lapply(R3, function(x) {<br>  colnames(x) &lt;- sub(<span>"x"</span>, <span>"-1"</span>, sub(<span>"R3"</span>, <span>"R3_"</span>, colnames(x)))<br>  x<br>})<br>ldat &lt;- list(<br>  N1=N1,<br>  N2=N2,<br>  N3=N3,<br>  R1=R1,<br>  R2=R2,<br>  R3=R3<br>)<br>matrix.name &lt;- names(ldat$NBW1)<br>ldat &lt;- lapply(matrix.name, function(x){<br>  dat.list &lt;- lapply(ldat, function(y){<br>    y[[x]]<br>  })<br>  dat.merged &lt;- <span>do</span>.call(cbind, dat.list)<br>  dat.merged<br>})<br>names(ldat) &lt;- matrix.name<br>cells.id &lt;- colnames(scRNA.T)<br>cells.id[<span>1</span>:<span>5</span>]<br>ldat &lt;- lapply(ldat, function(x) {<br>  x[, cells.id]<br>})<br>lapply(ldat, dim)<br>emat &lt;- ldat$spliced<br>nmat &lt;- ldat$unspliced<br>cell_cluster &lt;- as.character(scRNA.T<span>@meta</span>.data$celltype1)<br>names(cell_cluster) &lt;- cells.id<br>emat &lt;- filter.genes.by.cluster.expression(emat, cell_cluster, min.max.cluster.average = .<span>1</span>)<br>nmat &lt;- filter.genes.by.cluster.expression(nmat, cell_cluster, min.max.cluster.average = .<span>1</span>)<br>length(intersect(rownames(emat), rownames(nmat)))<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>#RNA Velocity分析<br>### 参数设置<br>fit.quantile = <span>0.05</span> # 官方教程设定为 <span>0.05</span><br>deltaT = <span>1</span> # <span>default</span>: <span>1</span><br>kCells = <span>10</span> # <span>default</span>: <span>10</span><br>### RNA velocity分析<br>rvel.qf &lt;- gene.relative.velocity.estimates(emat, nmat, deltaT = deltaT, kCells = kCells, fit.quantile = fit.quantile)<br><br><br>### 参数设定<br>emb = t(reducedDimS(monocle_cds)) # DDRTree坐标 monocle2<br>emb = Embeddings(scRNA.T, reduction = <span>"umap"</span>) # monocle3 seurat<br>vel = rvel.qf # velocity estimates<br>n = <span>100</span> # 最邻近细胞的数量<br>scale = <span>"sqrt"</span> # scale方法<br># 散点的颜色（用以区分不同的细胞状态）<br>cell.colors = plyr::mapvalues(cell_cluster, names(colors), colors)<br>cell.alpha = <span>0.2</span> # 散点颜色的透明度<br>cell.cex = <span>1</span> # 散点的尺寸<br>arrow.scale = <span>1</span> # 箭头的长度<br>arrow.lwd = <span>1.5</span> # 箭头的粗细<br>grid.n = <span>50</span> # grids的数量<br><br>### <span>plot<br><span>pdf</span><span>(file=<span>"velo.pdf"</span>)</span><br>show.velocity.on.embedding.<span>cor</span><span>(emb, vel, n, scale=scale, <br>                               cell.colors = ac(cell.colors, alpha = cell.alpha)</span>,<br>                               cex </span>= cell.cex, arrow.scale = arrow.scale, <br>                               show.grid.flow = TRUE, min.grid.cell.mass = <span>1</span>,<br>                               grid.n = grid.n, arrow.lwd = arrow.lwd,cell.border.alpha = <span>0.1</span>)<br>dev.off()<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="1.0148026315789473" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/Ze2f5C6nc67v8NLTMS9nmJ8Wn6Y2icgrqaKFycdoibQBCKIJY7mM1iar3dr9FR8X6kdVduOicVI0Q8SRpx8U7BEzIg/640?wx_fmt=jpeg" data-type="jpeg" data-w="608" src="https://mmbiz.qpic.cn/mmbiz_jpg/Ze2f5C6nc67v8NLTMS9nmJ8Wn6Y2icgrqaKFycdoibQBCKIJY7mM1iar3dr9FR8X6kdVduOicVI0Q8SRpx8U7BEzIg/640?wx_fmt=jpeg"></figure><h2 data-tool="mdnice编辑器"><span></span><span>scvelo</span></h2><pre data-tool="mdnice编辑器"><span></span><code><span>import</span> anndata<br><span>import</span> scvelo as scv<br><span>import</span> pandas as pd<br><span>import</span> numpy as np<br><span>import</span> matplotlib as plt<br><span>import</span> scanpy as sc<br><span>import</span> os<br>#载入 loom文件<br>N1= anndata.read_loom(<span>"./N1/velocyto/N2.loom"</span>)<br>N2= anndata.read_loom(<span>"./N2/velocyto/N2.loom"</span>)<br>N3= anndata.read_loom(<span>"./N3/velocyto/N3.loom"</span>)<br>R1= anndata.read_loom(<span>"./R1/velocyto/R1.loom"</span>)<br>R2= anndata.read_loom(<span>"./R2/velocyto/R2.loom"</span>)<br>R3= anndata.read_loom(<span>"./R3/velocyto/R3.loom"</span>)<br>#修改列名<br>barcodes = [bc.split(<span>':'</span>)[<span>1</span>] <span>for</span> bc in N1.obs.index.tolist()] <br>barcodes=[bc[<span>0</span>:len(bc)-<span>1</span>]+<span>'-1'</span> <span>for</span> bc in barcodes]<br>barcodes=[<span>'N1_'</span>+ bc[<span>0</span>:len(bc)] <span>for</span> bc in barcodes]<br>N1.obs.index = barcodes<br>N1.var_names_make_unique()<br>barcodes = [bc.split(<span>':'</span>)[<span>1</span>] <span>for</span> bc in N2.obs.index.tolist()] <br>barcodes=[bc[<span>0</span>:len(bc)-<span>1</span>]+<span>'-1'</span> <span>for</span> bc in barcodes]<br>barcodes=[<span>'N2_'</span>+ bc[<span>0</span>:len(bc)] <span>for</span> bc in barcodes]<br>N2.obs.index = barcodes<br>N2.var_names_make_unique()<br>barcodes = [bc.split(<span>':'</span>)[<span>1</span>] <span>for</span> bc in N3.obs.index.tolist()] <br>barcodes=[bc[<span>0</span>:len(bc)-<span>1</span>]+<span>'-1'</span> <span>for</span> bc in barcodes]<br>barcodes=[<span>'N3_'</span>+ bc[<span>0</span>:len(bc)] <span>for</span> bc in barcodes]<br>N3.obs.index = barcodes<br>N3.var_names_make_unique()<br>barcodes = [bc.split(<span>':'</span>)[<span>1</span>] <span>for</span> bc in R1.obs.index.tolist()] <br>barcodes=[bc[<span>0</span>:len(bc)-<span>1</span>]+<span>'-1'</span> <span>for</span> bc in barcodes]<br>barcodes=[<span>'R1_'</span>+ bc[<span>0</span>:len(bc)] <span>for</span> bc in barcodes]<br>R1.obs.index = barcodes<br>R1.var_names_make_unique()<br>barcodes = [bc.split(<span>':'</span>)[<span>1</span>] <span>for</span> bc in R2.obs.index.tolist()] <br>barcodes=[bc[<span>0</span>:len(bc)-<span>1</span>]+<span>'-1'</span> <span>for</span> bc in barcodes]<br>barcodes=[<span>'R2_'</span>+ bc[<span>0</span>:len(bc)] <span>for</span> bc in barcodes]<br>R2.obs.index = barcodes<br>R2.var_names_make_unique()<br>barcodes = [bc.split(<span>':'</span>)[<span>1</span>] <span>for</span> bc in R3.obs.index.tolist()] <br>barcodes=[bc[<span>0</span>:len(bc)-<span>1</span>]+<span>'-1'</span> <span>for</span> bc in barcodes]<br>barcodes=[<span>'R3_'</span>+ bc[<span>0</span>:len(bc)] <span>for</span> bc in barcodes]<br>R3.obs.index = barcodes<br>R3.var_names_make_unique()<br>ldata= N1.concatenate([N2,N3,R1,R2,R3])<br>adata=sc.read_h5ad(<span>'./data/project1/scRNA.T.h5ad'</span>)<br>adata= scv.utils.merge(adata,ldata)<br>scv.pp.filter_and_normalize(adata,min_shared_counts=<span>20</span>,n_top_genes=<span>2000</span>)<br>scv.pp.moments(adata,n_pcs=<span>30</span>,n_neighbors=<span>30</span>)<br>scv.tl.recover_dynamics(adata)<br>scv.tl.velocity(adata,mode=<span>'dynamical'</span>)<br>scv.tl.velocity_graph(adata)<br>scv.pl.velocity_embedding_stream(adata,basis=<span>'umap'</span>,color=<span>'celltype'</span>, title=<span>''</span>, smooth=.<span>8</span>, min_mass=<span>4</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.6632996632996633" data-src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc67v8NLTMS9nmJ8Wn6Y2icgrq1oR62e3W1n4O8QwLhEcrMxKwRGDJT1yZiaCzXxg3tiaFxabd7c1S24KQ/640?wx_fmt=png" data-type="png" data-w="891" src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc67v8NLTMS9nmJ8Wn6Y2icgrq1oR62e3W1n4O8QwLhEcrMxKwRGDJT1yZiaCzXxg3tiaFxabd7c1S24KQ/640?wx_fmt=png"></figure></section><p><br></p><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/AWkdOOFub0uWSjoW_WtYOw",target="_blank" rel="noopener noreferrer">原文链接</a>
