---
title: "关键单细胞亚群辅助判定之生存分析"
date: 2023-10-29T13:42:44Z
draft: ["false"]
tags: [
  "fetched",
  "生信技能树"
]
categories: ["Acdemic"]
---
关键单细胞亚群辅助判定之生存分析 by 生信技能树
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><blockquote data-tool="mdnice编辑器"><p>做过单细胞转录组数据分析的小伙伴都知道，多分组多样品的单细胞项目很容易就十几万个单细胞了所以第一层次降维聚类分群就有十个左右亚群，然后每个亚群又可以细分十几个，算起来就是近百的单细胞亚群。</p></blockquote><p data-tool="mdnice编辑器">这样的话图表会非常多，但是数据分析的终极目标是发文章，而文章就必须要有故事性有落脚点。接下来我们就系统性梳理一下使用生存分析来辅助判定关键单细胞亚群。这里我们采用的仍然是2022的肺腺癌相关的文章，标题是：《Delineating the dynamic evolution from preneoplasia to invasive lung adenocarcinoma by integrating single-cell RNA sequencing and spatial transcriptomics》，肺癌单细胞数据集也有好几十个了，拿到表达量矩阵后的第一层次降维聚类分群通常是：</p><ul data-tool="mdnice编辑器"><li><section>immune (CD45+,PTPRC),</section></li><li><section>epithelial/cancer (EpCAM+,EPCAM),</section></li><li><section>stromal (CD10+,MME,fibo or CD31+,PECAM1,endo)</section></li></ul><p data-tool="mdnice编辑器">参考我前面介绍过 <a href="https://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247488940&amp;idx=1&amp;sn=1cc8a8a74715087939b9721c0881775d&amp;scene=21#wechat_redirect" data-linktype="2">CNS图表复现08—肿瘤单细胞数据第一次分群通用规则</a>，这3大单细胞亚群构成了肿瘤免疫微环境的复杂。其实这个文章的第一层次降维聚类分群详见：<a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247525924&amp;idx=1&amp;sn=aaa305d856a835adda2f842840b96ec2&amp;scene=21#wechat_redirect" data-linktype="2">换一个分析策略会导致文章的全部论点都得推倒重来吗</a>，然后文章针对上皮细胞进行细分后，使用inferCNV判断了恶性的亚群，接下来就使用生存分析从恶性亚群里面的挑选了增殖亚群（UBE2C+），因为它这个亚群的基因可以在tcga数据库的luad数据集里面的有统计学显著的生存分析意义。如下所示：</p><p><img data-galleryid="" data-ratio="1.3706896551724137" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wzmY2v04DGKDiaTnO1vM1kMVSaiaibOtvoDpnhhFfGqBb1uQyjFfeeK4x6GnvwqINkleZjCj08WD3gnw/640?wx_fmt=png" data-type="png" data-w="928" src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wzmY2v04DGKDiaTnO1vM1kMVSaiaibOtvoDpnhhFfGqBb1uQyjFfeeK4x6GnvwqINkleZjCj08WD3gnw/640?wx_fmt=png"></p><figure data-tool="mdnice编辑器"><figcaption>增殖亚群（UBE2C+）</figcaption></figure><h2 data-tool="mdnice编辑器"><span></span>生存分析三步走</h2><p data-tool="mdnice编辑器">生存分析主要是需要一个病人队列而且病人的生存时间信息以及其最后的状态信息是齐全的，然后我们需要病人的表达量矩阵信息，这样的话就可以从矩阵里面挑选感兴趣的基因或者说基因列表去矩阵里面打分后，把病人高低分组后做生存分析来判断其重要性。</p><p data-tool="mdnice编辑器">这里我们采用ucsc的xena浏览器下载tcga数据库的luad数据集表达量矩阵，每个癌症都有固定的数据文件格式：</p><p data-tool="mdnice编辑器">比如TCGA-LAML，如下所示：</p><pre data-tool="mdnice编辑器"><span></span><code>https://gdc-hub.s3.us-east-1.amazonaws.com/download/TCGA-LAML.htseq_counts.tsv.gz<br>https://gdc-hub.s3.us-east-1.amazonaws.com/download/TCGA-LAML.GDC_phenotype.tsv.gz<br>https://gdc-hub.s3.us-east-1.amazonaws.com/download/TCGA-LAML.survival.tsv<br>https://gdc-hub.s3.us-east-1.amazonaws.com/download/TCGA-LAML.mutect2_snv.tsv.gz<br></code></pre><p data-tool="mdnice编辑器">很容易批量下载，也很容易根据感兴趣的癌症来下载对应的文件。</p><h3 data-tool="mdnice编辑器"><span></span>第1步：格式化表达量矩阵文件<span></span></h3><p data-tool="mdnice编辑器">下面的代码很容易理解，需要自己熟悉这个<code>TCGA-LUAD.htseq_counts.tsv.gz</code>文件的规则即可：</p><pre data-tool="mdnice编辑器"><span></span><code>rm(list=ls())<br><br><span>library</span>(data.table) <br>dat &lt;- data.table::fread(<span>'TCGA-LUAD.htseq_counts.tsv.gz'</span>,<br>                          data.table = <span>F</span>)  <br>head(dat[,<span>1</span>:<span>4</span>])<br>tail(dat[,<span>1</span>:<span>4</span>]) <br>dat = dat[<span>1</span>:(nrow(dat)-<span>5</span>),]<br>rownames(dat) = dat$Ensembl_ID<br>a = dat<br>a = a[,-<span>1</span>]<br><span>##逆转 log</span><br>a = as.matrix(<span>2</span>^a - <span>1</span>)<br><span># 用apply转换为整数矩阵</span><br>head(a[,<span>1</span>:<span>4</span>])<br>tail(a[,<span>1</span>:<span>4</span>]) <br>colSums(a)/<span>1e6</span><br><span># 普通转录组定量 20m</span><br><span># 想看融合基因，可变剪切，100M</span><br>exp = apply(a, <span>2</span>, as.integer)<br>rownames(exp) = rownames(dat)<br>exp= log(edgeR::cpm(exp)+<span>1</span>)<br><span>library</span>(stringr)<br>head(rownames(exp))<br><span>library</span>(AnnoProbe)<br><span>library</span>(tinyarray)<br>rownames(exp) = substr(rownames(exp), <span>1</span>, <span>15</span>) <br>re = annoGene(rownames(exp),ID_type = <span>"ENSEMBL"</span>);head(re)<br>exp = trans_array(exp,ids = re,from = <span>"ENSEMBL"</span>,to = <span>"SYMBOL"</span>)<br>head(exp[,<span>1</span>:<span>4</span>])<br>tail(exp[,<span>1</span>:<span>4</span>]) <br>proj=<span>'tcga-luad'</span><br>save(exp,file = paste0(proj,<span>".htseq_counts.rdata"</span>) ) <br></code></pre><h3 data-tool="mdnice编辑器"><span></span>第2步：整理临床信息<span></span></h3><p data-tool="mdnice编辑器">做生存分析的时候，其实只需要表达量矩阵里面的肿瘤样品即可，所以简单的过滤一下。然后临床信息里面我们也是简单的取 OS 即可：</p><pre data-tool="mdnice编辑器"><span></span><code>rm(list=ls())<br><br>proj=<span>'tcga-luad'</span><br>load(file = paste0(proj,<span>".htseq_counts.rdata"</span>) )<br>Group = ifelse(as.numeric(str_sub(colnames(exp),<span>14</span>,<span>15</span>)) &lt; <span>10</span>,<span>'tumor'</span>,<span>'normal'</span>)  <br>Group = factor(Group,levels = c(<span>"normal"</span>,<span>"tumor"</span>))<br>print(table(Group))<br><span># 生存分析只需要tumor样品即可</span><br>exprSet = exp[,Group==<span>'tumor'</span>]<br><br><br>clinical = read.delim(<span>'TCGA-LUAD.GDC_phenotype.tsv.gz'</span>,<br>                      fill = <span>T</span>,header = <span>T</span>,sep = <span>"\t"</span>)<br>surv = read.delim(<span>'TCGA-LUAD.survival.tsv'</span>,header = <span>T</span>) <br><span>library</span>(tidyverse)<br>meta = left_join(surv,clinical,by = c(<span>"sample"</span>= <span>"submitter_id.samples"</span>))<br>head(meta[,<span>1</span>:<span>4</span>])<br>tail(meta[,<span>1</span>:<span>4</span>]) <br>print(dim(meta))<br> <br><span>#去掉生存信息不全或者生存时间小于30天的样本，样本纳排标准不唯一，且差别很大.</span><br>k1 = meta$OS.time &gt;= <span>30</span><br>k2 = !(is.na(meta$OS.time)|is.na(meta$OS))<br>meta = meta[k1&amp;k2,]<br>meta = meta[,c(<br>  <span>'sample'</span>,<br>  <span>'OS'</span>,<br>  <span>'OS.time'</span><br>)]<br>colnames(meta)=c(<span>'ID'</span>,<span>'event'</span>,<span>'time'</span>)<br>meta$time = meta$time/<span>30</span><br>rownames(meta) &lt;- meta$ID<br>s = intersect(rownames(meta),colnames(exprSet))<br>exprSet = exprSet[,s]<br>meta = meta[s,]<br>identical(rownames(meta),colnames(exprSet))<br>save(exprSet,meta,file = paste0(proj,<span>".for_survival.rdata"</span>) ) <br></code></pre><p data-tool="mdnice编辑器">前面的表达量矩阵跟临床生存分析已经是对应好了，就可以进行后续的生存分析啦</p><h3 data-tool="mdnice编辑器"><span></span>第3步：批量cox分析<span></span></h3><p data-tool="mdnice编辑器">这个代码也是七年前的了，蛮简单的的一个循环而已</p><pre data-tool="mdnice编辑器"><span></span><code>rm(list=ls())<br><span>library</span>(survival)<br><span>library</span>(survminer) <br><span>library</span>(ggstatsplot) <br>proj=<span>'tcga-luad'</span><br>load(file = paste0(proj,<span>".for_survival.rdata"</span>) )<br><br><span># 1. prepare data for coxph----</span><br><span>## 批量生存分析使用coxph回归方法 </span><br>exprSet[<span>1</span>:<span>4</span>,<span>1</span>:<span>4</span>]<br>phe=meta<br>head(phe)<br>mySurv &lt;- with(phe, Surv(time, event))<br>survival_dat=phe <br><br>cox_results &lt;-apply(exprSet , <span>1</span> , <span>function</span>(gene){<br>  <span># gene= as.numeric(exprSet[1,]) </span><br>  group=ifelse(gene&gt;median(gene),<span>'high'</span>,<span>'low'</span>) <br>  group=factor(  group,levels = c(<span>'low'</span>,<span>'high'</span>))<br>  <span>if</span>( length(table(group))&lt;<span>2</span>)<br>    <span>return</span>(<span>NULL</span>)<br>  survival_dat &lt;- data.frame(group=group,<span># stage=phe$stage,</span><br>                             stringsAsFactors = <span>F</span>)<br>  <br>  m=coxph(mySurv ~ group, <br>          <span># mySurv ~  stage+ group,  # 如果有交叉变量</span><br>          data =  survival_dat)<br>  <br>  beta &lt;- coef(m)<br>  se &lt;- sqrt(diag(vcov(m)))<br>  HR &lt;- exp(beta)<br>  HRse &lt;- HR * se<br>  <br>  <span>#summary(m)</span><br>  tmp &lt;- round(cbind(coef = beta, se = se, z = beta/se, <br>                     p = <span>1</span> - pchisq((beta/se)^<span>2</span>, <span>1</span>),<br>                     HR = HR, HRse = HRse,<br>                     HRz = (HR - <span>1</span>) / HRse, HRp = <span>1</span> - pchisq(((HR - <span>1</span>)/HRse)^<span>2</span>, <span>1</span>),<br>                     HRCILL = exp(beta - qnorm(<span>.975</span>, <span>0</span>, <span>1</span>) * se),<br>                     HRCIUL = exp(beta + qnorm(<span>.975</span>, <span>0</span>, <span>1</span>) * se)), <span>3</span>)<br>  <span>return</span>(tmp[<span>'grouphigh'</span>,])<br>  <br>})<br><br><span># 2. specify the value----</span><br>cox_results=t(cox_results)<br>head(cox_results)<br>cox_results=cox_results[order(cox_results[,<span>'HR'</span>],decreasing = <span>T</span>),]<br><br>table(cox_results[,<span>4</span>]&lt;<span>0.01</span>)<br>table(cox_results[,<span>4</span>]&lt;<span>0.05</span>) <br>save(cox_results, <br>     file = <span>'batch_cox_results.Rdata'</span>)<br>load(     file = <span>'batch_cox_results.Rdata'</span>) <br></code></pre><p data-tool="mdnice编辑器">值得注意的是结果矩阵里面很多基因的HR值非常离谱，其实是因为这个基因在样品队列里面的表达量失衡，比如它仅仅是在少量的几个预后非常差的病人里面有表达量在所有的其它病人里面都是0值，这样的生存分析结果大概率是不可靠的！</p><h2 data-tool="mdnice编辑器"><span></span>基于单细胞转录组的生存分析</h2><h3 data-tool="mdnice编辑器"><span></span>第1步：根据单细胞亚群基因集在肿瘤病人表达量矩阵里面进行gsva打分<span></span></h3><p data-tool="mdnice编辑器">如果是想看单个基因是否有统计学显著的生存分析意义其实前面的批量cox分析就可以提取结果了，但是我们的单细胞亚群基因集结果里面的基本上每个亚群都是几十个高表达量特异性基因，详见：<a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247525924&amp;idx=1&amp;sn=aaa305d856a835adda2f842840b96ec2&amp;scene=21#wechat_redirect" data-linktype="2">换一个分析策略会导致文章的全部论点都得推倒重来吗</a>，就知道如何输出如下所示的<code>cosg_celltype__marker_cosg.Rdata</code>文件啦：</p><pre data-tool="mdnice编辑器"><span></span><code>rm(list=ls())<br><span>library</span>(survival)<br><span>library</span>(survminer) <br><span>library</span>(ggstatsplot) <br><span>library</span>(gplots)<br><span>library</span>(ggplot2) <br><span>library</span>(pheatmap)<br><span>library</span>(clusterProfiler) <br><span>library</span>(org.Hs.eg.db)<br><span>library</span>(GSVA) <br><span>library</span>(GSEABase)<br> <br><span># 1. 载入表达量矩阵和临床信息 ----</span><br>proj=<span>'tcga-luad'</span><br>load(file = paste0(proj,<span>".for_survival.rdata"</span>) ) <br>exprSet[<span>1</span>:<span>4</span>,<span>1</span>:<span>4</span>]<br>phe=meta<br>head(phe)<br>mySurv &lt;- with(phe, Surv(time, event))<br>survival_dat=phe<br><br><span># 2. creat geneset----</span><br><span># load('../../paper-figures/cosg_celltype__marker_cosg.Rdata')</span><br>load(<span>'cosg_celltype__marker_cosg.Rdata'</span>)<br>head(marker_cosg$names)<br>deg_list =  as.list(marker_cosg$names)<br>names(deg_list)<br>deg_list<br>gs = lapply(deg_list, toupper) <br>geneset &lt;- GeneSetCollection(mapply(<span>function</span>(geneIds, keggId) {<br>  GeneSet(geneIds, geneIdType=EntrezIdentifier(),<br>          collectionType=KEGGCollection(keggId),<br>          setName=keggId)<br>}, gs, names(gs)))<br>geneset<br><br><span># 3. run gsva----</span><br>X=as.matrix(exprSet)<br>es.max &lt;- gsva(X, geneset, <br>               mx.diff=<span>FALSE</span>, verbose=<span>FALSE</span>, <br>               parallel.sz=<span>4</span>)<br>es.max[<span>1</span>:<span>4</span>, <span>1</span>:<span>4</span>] <br>pheatmap(es.max) <br></code></pre><h3 data-tool="mdnice编辑器"><span></span>第2步：根据gsva打分值高低分组进行生存分析<span></span></h3><p data-tool="mdnice编辑器">一般来说， 就是根据gsva打分的中位值高低分组进行生存分析，代码如下所示：</p><pre data-tool="mdnice编辑器"><span></span><code><span># 4. 根据gsva结果高低分组后批量生存分析 ----</span><br>es.max[<span>1</span>:<span>4</span>, <span>1</span>:<span>4</span>]<br>splots &lt;- list()<br>g = <span>1</span><br><span>for</span> (i <span>in</span>  names(deg_list) ) {<br>  <span># i =  names(deg_list) [1]</span><br>  subset = paste0(<span>'cluster_'</span>,i)<br>  print(subset)<br>  v = as.numeric(es.max[i,])   <span>#每一个亚群表达量。</span><br>  sub_group &lt;- ifelse( v &lt; <span>0</span>,<span>"low"</span>,<span>"high"</span>)   <span>#如果表达量小于0的话，就定义为low。gsva处理过表达量。0.几左右</span><br>  table(sub_group) <br>  phe$sub_group=sub_group<br>  <span># Fit survival curves</span><br>  <span>require</span>(<span>"survival"</span>)<br>  fit &lt;- survfit(Surv(time, event) ~ sub_group, data = phe)<br>  <span>library</span>(<span>"survminer"</span>)<br>  survp &lt;- ggsurvplot(fit, data = phe,<br>                      surv.median.line = <span>"hv"</span>, <span># Add medians survival</span><br>                      pval = <span>TRUE</span>,             <span># Add p-value and tervals </span><br>                      conf.int = <span>TRUE</span>,        <span># Add the 95% confidence band</span><br>                      risk.table = <span>TRUE</span>,      <span># Add risk table</span><br>                      tables.height = <span>0.2</span>,<br>                      tables.theme = theme_cleantable(),<br>                      palette = <span>"jco"</span>,<br>                      ggtheme = theme_bw(),<br>                      title = subset)<br>  print(survp)<br>  splots[[g]] &lt;-  survp<br>  g = g + <span>1</span><br>}<br><br>length(splots)<br>x1 = ceiling(sqrt(length(splots)))<br>y1 = x1<br><br>all_plot &lt;- arrange_ggsurvplots(splots,<br>                                print = <span>F</span>,<br>                                ncol = x1, nrow = y1,<br>                                risk.table.height = <span>0.3</span>,<br>                                surv.plot.height = <span>0.7</span>)<br><span># all_plot </span><br>x2=<span>5</span>*x1<br>y2=<span>5</span>*y1<br>prefix=<span>''</span><br>pro=<span>''</span><br>ggsave(all_plot, <span>#path = prefix,</span><br>       filename = paste0(pro, <span>'all_survival_plot.pdf'</span>),<br>       width = x2,height = y2)<br></code></pre><p data-tool="mdnice编辑器">就可以看到如下所示的每个单细胞亚群的生存分析结果，很明显跟文章类似的，也是增殖亚群（UBE2C+） 它这个亚群的基因可以在tcga数据库的luad数据集里面的有统计学显著的生存分析意义。</p><p><img data-galleryid="" data-ratio="1.0361111111111112" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wzmY2v04DGKDiaTnO1vM1kMVEctYMcbNQ6XSnWicXbQx4I3qO9Cca6tNgHEcL7KiaLQvqicnIARkuFLbQ/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wzmY2v04DGKDiaTnO1vM1kMVEctYMcbNQ6XSnWicXbQx4I3qO9Cca6tNgHEcL7KiaLQvqicnIARkuFLbQ/640?wx_fmt=png"></p><figure data-tool="mdnice编辑器"><figcaption>生存分析意义</figcaption></figure><h3 data-tool="mdnice编辑器"><span></span>第3步：根据gsva打分值进行取巧分组进行生存分析<span></span></h3><p data-tool="mdnice编辑器">如果是上面的根据gsva打分的中位值高低分组进行生存分析都没有生存分析统计学显著意义，但是又想看看每个亚群的具体的到底是保护因子还是风险因子，也可以使用<code>surv_cutpoint</code>函数哦：</p><pre data-tool="mdnice编辑器"><span></span><code><span>## cut point</span><br><span>## cutpoint</span><br>head(phe)<br>csplots &lt;- list()<br>cg = <span>1</span><br><span>for</span> (i <span>in</span>  names(deg_list) ) {<br>  <span># i =  names(deg_list) [1]</span><br>  subset = paste0(<span>'cluster_'</span>,i)<br>  print(subset)<br>  v = as.numeric(es.max[i,])   <span>#每一个亚群表达量。</span><br>  phe$v &lt;- v<br>  head(phe)<br>  sur.cut &lt;- surv_cutpoint(phe,<br>                           time= <span>'time'</span>,<br>                           event = <span>'event'</span> ,<br>                           variables = <span>'v'</span> )<br>  sur.cat &lt;- surv_categorize(sur.cut)<br>  head(sur.cat)<br>  sfit &lt;- survfit(Surv(time, event) ~ v, data = sur.cat)<br>  p_surv_cut &lt;- ggsurvplot(sfit, data = phe,<br>                           surv.median.line = <span>"hv"</span>, <span># Add medians survival</span><br>                           pval = <span>TRUE</span>,             <span># Add p-value and tervals </span><br>                           conf.int = <span>TRUE</span>,        <span># Add the 95% confidence band</span><br>                           risk.table = <span>TRUE</span>,      <span># Add risk table</span><br>                           tables.height = <span>0.2</span>,<br>                           tables.theme = theme_cleantable(),<br>                           palette = <span>"jco"</span>,<br>                           ggtheme = theme_bw(),<br>                           title = subset)<br>  print(p_surv_cut)<br>  csplots[[cg]] &lt;-  p_surv_cut<br>  cg = cg + <span>1</span><br>}<br><br>length(csplots)<br>x1 = ceiling(sqrt(length(csplots)))<br>y1 = x1<br><br>all_plot &lt;- arrange_ggsurvplots(csplots,<br>                                print = <span>F</span>,<br>                                ncol = x1, nrow = y1,<br>                                risk.table.height = <span>0.3</span>,<br>                                surv.plot.height = <span>0.7</span>)<br><span># all_plot </span><br>x2=<span>5</span>*x1<br>y2=<span>5</span>*y1<br>ggsave(all_plot, <span>#path = prefix,</span><br>       filename = paste0(pro, <span>'all_cut_point_survival_plot.pdf'</span>),width = x2,height = y2)<br><br><br><br></code></pre><p data-tool="mdnice编辑器">因为它并不是中位值高低分组，所以两个分组的病人数量是不平衡的， 如下所示：</p><p><img data-galleryid="" data-ratio="1.0361111111111112" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wzmY2v04DGKDiaTnO1vM1kMVXmJicRu866LlpiaoecC4kFOndXmcL2FkgyayLibBmlRNIrMCOytHS47gg/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wzmY2v04DGKDiaTnO1vM1kMVXmJicRu866LlpiaoecC4kFOndXmcL2FkgyayLibBmlRNIrMCOytHS47gg/640?wx_fmt=png"></p><figure data-tool="mdnice编辑器"><figcaption>数量是不平衡的</figcaption></figure><p data-tool="mdnice编辑器">很明显可以看到，它取巧了，所以基本上可以让每个单细胞亚群的特异性基因列表，都有统计学显著的生存分析结果哦！</p><h2 data-tool="mdnice编辑器"><span></span>后记</h2><p data-tool="mdnice编辑器">我们按照文章的思路， 定位到了特殊的也就是具有统计学生存分析意义的亚群，详见：<a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247525924&amp;idx=1&amp;sn=aaa305d856a835adda2f842840b96ec2&amp;scene=21#wechat_redirect" data-linktype="2">换一个分析策略会导致文章的全部论点都得推倒重来吗</a>，但是也有其它思路可以拿到同样的结果。但是上面的代码已经是足够复杂啦，我们还是等待大家运行后我们先视频号答疑后，再分享后面的思路哈！代码会持续更新在这个链接哈，大家一定要抽空测试：</p><pre data-tool="mdnice编辑器"><span></span><code>链接: https://pan.baidu.com/s/1geW1MTLRizcJWEESdjMN7g?<span>pwd</span>=96t6 提取码: 96t6<br></code></pre><p data-tool="mdnice编辑器"><strong>我们会在视频号直播互动答疑，欢迎预约我们的直播链接！</strong></p><section><mp-common-videosnap data-pluginname="mpvideosnap" data-headimgurl="https://wx.qlogo.cn/finderhead/PiajxSqBRaEI7scvWIPdECSfnUpSjTib9Y7RI14r1VVzxaA57PjcCERw/0" data-username="v2_060000231003b20faec8c7e1881bcad2ca06ec35b07788412aec898c89eb1e34f9a354475e8c@finder" data-nickname="生信技能树" data-desc="将在11月01日 18:00 直播" data-livewording="预约" data-intro="单细胞+生存分析" data-type="live" data-status="0" data-noticeid="finderlivenotice-v2_060000231003b20faec8c7e1881bcad2ca06ec35b07788412aec898c89eb1e34f9a354475e8c@finder-1698583822270546-1422988549" data-isdisabled="0" data-errortips=""></mp-common-videosnap></section></section><p><br></p><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/zxDRM73Cj8hzlQWYNkWyZg",target="_blank" rel="noopener noreferrer">原文链接</a>
