---
title: "给学徒的ATAC-seq数据实战（附上收费视频）"
date: 2023-10-31T01:01:03Z
draft: ["false"]
tags: [
  "fetched",
  "生信技能树"
]
categories: ["Acdemic"]
---
给学徒的ATAC-seq数据实战（附上收费视频） by 生信技能树
------
<div><blockquote><p>本次给学徒讲解的文章是 :The landscape of accessible chromatin in mammalian preimplantation embryos. Nature 2016 Jun 30;534(7609):652-7. PMID: 27309802</p></blockquote><p>查看文章发现数据上传到了GEO，是：https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE66581</p><p>在SRA数据库可以下载原始测序数据 , 从文章找到数据的ID： https://www.ncbi.nlm.nih.gov/sra?term=SRP055881 把下面的内容保存到文件，命名为 <code>srr.list</code> 就可以使用prefetch这个函数来下载。</p><h3><span>linux环境及软件安装</span></h3><p>这里首推conda</p><pre><code><span>#</span><span> https://mirrors.tuna.tsinghua.edu.cn/<span>help</span>/anaconda/</span><br><span>#</span><span> https://mirrors.tuna.tsinghua.edu.cn/anaconda/miniconda/ </span><br>wget https://mirrors.tuna.tsinghua.edu.cn/anaconda/miniconda/Miniconda3-latest-Linux-x86_64.sh<br>bash Miniconda3-latest-Linux-x86_64.sh<br>source ~/.bashrc <br><span>#</span><span><span># 安装好conda后需要设置镜像。</span></span><br>conda config --add channels https://mirrors.tuna.tsinghua.edu.cn/anaconda/pkgs/free<br>conda config --add channels https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/conda-forge<br>conda config --add channels https://mirrors.tuna.tsinghua.edu.cn/anaconda/cloud/bioconda<br>conda config --set show_channel_urls yes<br><br>conda  create -n atac -y   python=2 bwa<br>conda info --envs<br>source activate atac<br><span>#</span><span> 可以用search先进行检索</span><br>conda search trim_galore<br><span>#</span><span><span># 保证所有的软件都是安装在 wes 这个环境下面</span></span><br>conda install -y sra-tools  <br>conda install -y trim-galore  samtools bedtools<br>conda install -y deeptools homer  meme<br>conda install -y macs2 bowtie bowtie2 <br>conda install -y  multiqc <br>conda install -y  sambamba<br></code></pre><p>代码里面提到的软件，都是根据我们对ATAC-seq搜索学习总结的。</p><p>值得一提的是自己为了ATAC-seq建立的软件环境可以很方便移植到另外一台电脑!</p><p>首先通过<code>activate target_env</code>要分享的环境target_env，然后输入下面的命令会在当前工作目录下生成一个<code>environment.yml文件</code>，<br><code>conda env export &gt; environment.yml</code><br>小伙伴拿到<code>environment.yml文件</code>后，将该文件放在工作目录下，可以通过以下命令从该文件创建环境<br><code>conda env create -f environment.yml</code></p><h3><span>下载作者的数据</span></h3><p>前面提到的SRA数据库，该文章配套数据太多，我们节选部分作为练习，文件<code>config.sra</code> 如下：</p><pre><code>2-cell-1 SRR2927015<br>2-cell-2 SRR2927016<br>2-cell-5 SRR3545580<br>2-cell-4 SRR2927018<br></code></pre><p>因为conda安装好了sra-toolkit，所以prefetch函数可以直接使用</p><pre><code><span>#</span><span><span># 下载数据 </span></span><br><span>#</span><span> cat srr.list |<span>while</span> <span>read</span> id;<span>do</span> (nohup <span>$prefetch</span> <span>$id</span> -X 100G  &amp; );<span>done</span></span><br><span>#</span><span><span># 注意组织好自己的项目</span></span><br>mkdir -p  ~/project/atac/<br>cd ~/project/atac/<br>mkdir {sra,raw,clean,align,peaks,motif,qc}<br>cd sra <br><span>#</span><span><span># vim 或者cat命令创建 srr.list 文件, 里面保存着作为练习使用的4个数据ID </span></span><br>source activate atac <br>cat srr.list |while read id;do ( nohup  prefetch $id &amp; );done<br><span>#</span><span><span># 默认下载目录：~/ncbi/public/sra/ </span></span><br>ls -lh ~/ncbi/public/sra/<br><span>#</span><span><span># 下载耗时，自行解决，学员使用现成数据：/public/project/epi/atac/sra </span></span><br><span><br>#</span><span><span># 假如提前下载好了数据。</span></span><br>cd ~/project/atac/ <br>ln -s /public/project/epi/atac/sra  sra<br></code></pre><p>总之数据如下：</p><pre><code><span>-rw-r--r--</span> 1 <span>stu</span> <span>stu</span> 4<span>.2G</span> <span>Aug</span> 25 11<span>:10</span> <span>SRR2927015</span><span>.sra</span><br><span>-rw-r--r--</span> 1 <span>stu</span> <span>stu</span> 5<span>.5G</span> <span>Aug</span> 25 11<span>:13</span> <span>SRR2927016</span><span>.sra</span><br><span>-rw-r--r--</span> 1 <span>stu</span> <span>stu</span> 2<span>.0G</span> <span>Aug</span> 25 11<span>:12</span> <span>SRR2927018</span><span>.sra</span><br><span>-rw-r--r--</span> 1 <span>stu</span> <span>stu</span> 7<span>.0G</span> <span>Aug</span> 25 11<span>:13</span> <span>SRR3545580</span><span>.sra</span><br></code></pre><h3><span>第一步，得到fastq测序数据</span></h3><p>通常我们应该是自己的实验数据，自己找公司测序后拿到原始数据，本次讲解使用的是公共数据，所以需要把下载的sra数据转换为fq格式。</p><pre><code><span>#</span><span><span># 下面需要用循环</span></span><br>cd ~/project/atac/<br>source activate atac<br>dump=fastq-dump<br>analysis_dir=raw<br>mkdir -p $analysis_dir<br><span>#</span><span><span># 下面用到的 config.sra 文件，就是上面自行制作的。</span></span><br><span><br>#</span><span> <span>$fastq</span>-dump sra/SRR2927015.sra  --gzip --split-3  -A 2-cell-1 -O clean/</span><br>cat config.sra |while read id;<br>do echo $id<br>arr=($id)<br>srr=${arr[1]}<br>sample=${arr[0]}<br><span>#</span><span>  测序数据的sra转fasq</span><br>nohup $dump -A  $sample -O $analysis_dir  --gzip --split-3  sra/$srr.sra &amp; <br>done <br><span><br>#</span><span><span>## 如果不只是4个文件，需要使用shell脚本批处理。</span></span><br>cut -f  10,13 SRP055881/SraRunTable.txt|\<br>sed 's/Embryonic stem cell/ESC/'|sed 's/early 2-cell/e2-cell/' |\<br>perl -alne '{$h{$F[1]}++;print "$_-$h{$F[1]}"}' |tail -n+2|awk '{print $2"\t"$1}'&gt; config.sra<br></code></pre><p>得到的原始fq数据如下:</p><pre><code><span>-rw-rw-r--</span> 1 <span>jmzeng</span> <span>jmzeng</span> 2<span>.6G</span> <span>Aug</span> 24 23<span>:10</span> 2<span>-cell-1_1</span><span>.fastq</span><span>.gz</span><br><span>-rw-rw-r--</span> 1 <span>jmzeng</span> <span>jmzeng</span> 2<span>.6G</span> <span>Aug</span> 24 23<span>:10</span> 2<span>-cell-1_2</span><span>.fastq</span><span>.gz</span><br><span>-rw-rw-r--</span> 1 <span>jmzeng</span> <span>jmzeng</span> 3<span>.4G</span> <span>Aug</span> 24 23<span>:31</span> 2<span>-cell-2_1</span><span>.fastq</span><span>.gz</span><br><span>-rw-rw-r--</span> 1 <span>jmzeng</span> <span>jmzeng</span> 3<span>.7G</span> <span>Aug</span> 24 23<span>:31</span> 2<span>-cell-2_2</span><span>.fastq</span><span>.gz</span><br><span>-rw-rw-r--</span> 1 <span>jmzeng</span> <span>jmzeng</span> 1<span>.2G</span> <span>Aug</span> 24 22<span>:46</span> 2<span>-cell-4_1</span><span>.fastq</span><span>.gz</span><br><span>-rw-rw-r--</span> 1 <span>jmzeng</span> <span>jmzeng</span> 1<span>.2G</span> <span>Aug</span> 24 22<span>:46</span> 2<span>-cell-4_2</span><span>.fastq</span><span>.gz</span><br><span>-rw-rw-r--</span> 1 <span>jmzeng</span> <span>jmzeng</span> 4<span>.4G</span> <span>Aug</span> 24 23<span>:52</span> 2<span>-cell-5_1</span><span>.fastq</span><span>.gz</span><br><span>-rw-rw-r--</span> 1 <span>jmzeng</span> <span>jmzeng</span> 4<span>.9G</span> <span>Aug</span> 24 23<span>:52</span> 2<span>-cell-5_2</span><span>.fastq</span><span>.gz</span><br></code></pre><h3><span>第二步，测序数据的质量控制</span></h3><p>这个时候选择trim_galore软件进行过滤，双端测序数据的代码如下:<br>需要自行制作 config.raw 文件， 是3列，第一列占位用，没有意义，第二列是fq1的地址，第3列是fq2的地址。</p><pre><code>cd ~/project/atac/<br>mkdir -p clean <br>source activate atac  <br><span>#</span><span> trim_galore -q 25 --phred33 --length 35 -e 0.1 --stringency 4 --paired -o clean/ raw/2-cell-1_1.fastq.gz raw/2-cell-1_2.fastq.gz</span><br>cat config.raw  |while read id;<br>do echo $id<br>arr=($id)<br>fq2=${arr[2]}<br>fq1=${arr[1]}<br>sample=${arr[0]}<br>nohup  trim_galore -q 25 --phred33 --length 35 -e 0.1 --stringency 4 --paired -o  clean  $fq1   $fq2  &amp; <br>done <br>ps -ef |grep trim<br></code></pre><p>得到过滤后的fq文件如下：</p><pre><code><span>-rw-rw-r--</span> 1 <span>jmzeng</span> <span>jmzeng</span> 2<span>.4G</span> <span>Aug</span> 25 09<span>:35</span> 2<span>-cell-1_1_val_1</span><span>.fq</span><span>.gz</span><br><span>-rw-rw-r--</span> 1 <span>jmzeng</span> <span>jmzeng</span> 2<span>.3G</span> <span>Aug</span> 25 09<span>:35</span> 2<span>-cell-1_2_val_2</span><span>.fq</span><span>.gz</span><br><span>-rw-rw-r--</span> 1 <span>jmzeng</span> <span>jmzeng</span> 3<span>.1G</span> <span>Aug</span> 25 10<span>:10</span> 2<span>-cell-2_1_val_1</span><span>.fq</span><span>.gz</span><br><span>-rw-rw-r--</span> 1 <span>jmzeng</span> <span>jmzeng</span> 3<span>.3G</span> <span>Aug</span> 25 10<span>:10</span> 2<span>-cell-2_2_val_2</span><span>.fq</span><span>.gz</span><br><span>-rw-rw-r--</span> 1 <span>jmzeng</span> <span>jmzeng</span> 1<span>.1G</span> <span>Aug</span> 25 08<span>:52</span> 2<span>-cell-4_1_val_1</span><span>.fq</span><span>.gz</span><br><span>-rw-rw-r--</span> 1 <span>jmzeng</span> <span>jmzeng</span> 1<span>.1G</span> <span>Aug</span> 25 08<span>:52</span> 2<span>-cell-4_2_val_2</span><span>.fq</span><span>.gz</span><br><span>-rw-rw-r--</span> 1 <span>jmzeng</span> <span>jmzeng</span> 3<span>.7G</span> <span>Aug</span> 25 10<span>:27</span> 2<span>-cell-5_1_val_1</span><span>.fq</span><span>.gz</span><br><span>-rw-rw-r--</span> 1 <span>jmzeng</span> <span>jmzeng</span> 3<span>.9G</span> <span>Aug</span> 25 10<span>:27</span> 2<span>-cell-5_2_val_2</span><span>.fq</span><span>.gz</span><br></code></pre><p>质量控制前后都需要可视化，肯定是fastqc+multiqc，代码如下；</p><pre><code>cd ~/project/atac/qc <br>mkdir -p clean <br>fastqc -t 5  ../clean/2-cell-*gz -o clean <br>mkdir -p raw <br>fastqc -t 5  ../raw/2-cell-*gz -o clean <br><span>#</span><span> https://zh.wikipedia.org/wiki/ASCII</span><br><span>#</span><span><span># 还有很多其它工具，比如：</span></span><br>qualimap='/home/jianmingzeng/biosoft/Qualimap/qualimap_v2.2.1/qualimap'<br><span>$</span><span>qualimap bamqc --java-mem-size=20G  -bam <span>$id</span>  -outdir ./ </span><br></code></pre><h3><span>第三步，比对</span></h3><p>比对需要的index，看清楚物种，根据对应的软件来构建，这里直接<strong>用bowtie2进行比对和统计比对率,</strong> 需要提前下载参考基因组然后使用命令构建索引，或者直接就下载索引文件：下载小鼠参考基因组的索引和注释文件, 这里用常用的mm10</p><pre><code><span>#</span><span> 索引大小为3.2GB， 不建议自己下载基因组构建，可以直接下载索引文件，代码如下：</span><br>mkdir referece &amp;&amp; cd reference<br>wget -4 -q ftp://ftp.ccb.jhu.edu/pub/data/bowtie2_indexes/mm10.zip<br>unzip mm10.zip<br></code></pre><p>解压后的索引如下：</p><pre><code><span>848</span>M Jul  <span>5</span> <span>05</span>:<span>03</span> /<span>public</span>/reference/<span>index</span>/bowtie/mm10.<span>1</span>.bt2<br><span>633</span>M Jul  <span>5</span> <span>05</span>:<span>00</span> /<span>public</span>/reference/<span>index</span>/bowtie/mm10.<span>2</span>.bt2<br><span>6.0</span>K Jul  <span>5</span> <span>05</span>:<span>05</span> /<span>public</span>/reference/<span>index</span>/bowtie/mm10.<span>3</span>.bt2<br><span>633</span>M Jul  <span>5</span> <span>05</span>:<span>05</span> /<span>public</span>/reference/<span>index</span>/bowtie/mm10.<span>4</span>.bt2 <br><span>848</span>M Jul  <span>5</span> <span>04</span>:<span>52</span> /<span>public</span>/reference/<span>index</span>/bowtie/mm10.rev.<span>1</span>.bt2<br><span>633</span>M Jul  <span>5</span> <span>04</span>:<span>49</span> /<span>public</span>/reference/<span>index</span>/bowtie/mm10.rev.<span>2</span>.bt2<br></code></pre><p>这些索引文件一个都不能少，而且文件名的<strong>前缀</strong>很重要，保证一致。</p><p>单端测序数据的比对代码如下：</p><p>首先可以对测试样本走流程，完善代码:</p><pre><code>zcat ../clean/2-cell-1_1_val_1.fq.gz |head -10000 &gt; test1.fq <br>zcat ../clean/2-cell-1_2_val_2.fq.gz  |head -10000 &gt; test2.fq<br>bowtie2 -x /public/reference/index/bowtie/mm10 -1 test1.fq  -2 test2.fq<br>bowtie2 -x /public/reference/index/bowtie/mm10 -1 test1.fq  -2 test2.fq  -S test.sam<br>bowtie2 -x /public/reference/index/bowtie/mm10 -1 test1.fq  -2 test2.fq  |samtools sort -@ 5 -O bam -o test.bam -<br><span>#</span><span><span># 建议抛弃 samtools markdup功能，避免麻烦。</span></span><br><span>#</span><span><span># https://www.jianshu.com/p/1e6189f641db</span></span><br>samtools markdup -r  test.bam test.samtools.rmdup.bam <br><span>#</span><span><span># 把报错信息在谷歌搜索后，在两个网页上找到了答案。</span></span><br>https://github.com/samtools/samtools/issues/765<br>https://www.biostars.org/p/288496/<br><span><br>#</span><span><span># gatk 可以在GitHub下载</span></span><br>/public/biosoft/GATK/gatk-4.0.3.0/gatk  MarkDuplicates \<br>-I test.bam -O test.picard.rmdup.bam  --REMOVE_SEQUENCING_DUPLICATES true -M test.log <br><span><br>#</span><span><span>## picards 被包装在GATK里面：</span></span><br><span>#</span><span><span>## sambamba 文档： http://lomereiter.github.io/sambamba/docs/sambamba-markdup.html</span></span><br>conda install -y  sambamba<br>sambamba --help<br>sambamba markdup --help<br>sambamba markdup -r test.bam  test.sambamba.rmdup.bam<br>samtools flagstat test.sambamba.rmdup.bam<br>samtools flagstat test.bam<br><span>#</span><span><span># 接下来只保留两条reads要比对到同一条染色体(Proper paired) ，还有高质量的比对结果(Mapping quality&gt;=30)</span></span><br><span>#</span><span><span># 顺便过滤 线粒体reads</span></span><br>samtools view -f 2 -q 30  test.sambamba.rmdup.bam |grep -v chrM|wc<br>samtools view -f 2 -q 30  test.sambamba.rmdup.bam |wc<br>samtools view -h -f 2 -q 30  test.sambamba.rmdup.bam |grep -v chrM| samtools sort  -O bam  -@ 5 -o - &gt; test.last.bam<br>bedtools bamtobed -i test.last.bam  &gt; test.bed <br>ls *.bam  |xargs -i samtools index {}<br></code></pre><p>探索好了整个流程，就可以直接写批处理，代码如下：</p><pre><code>ls /home/jmzeng/project/atac/clean/*_1.fq.gz &gt; 1<br>ls /home/jmzeng/project/atac/clean/*_2.fq.gz &gt; 2<br>ls /home/jmzeng/project/atac/clean/*_2.fq.gz |cut -d"/" -f 7|cut -d"_" -f 1  &gt; 0<br>paste 0 1 2  &gt; config.clean ## 供mapping使用的配置文件<br><br>cd ~/project/epi/align<br><span>#</span><span><span># 相对目录需要理解</span></span><br>bowtie2_index=/public/reference/index/bowtie/mm10<br><span>#</span><span><span># 一定要搞清楚自己的bowtie2软件安装在哪里，以及自己的索引文件在什么地方！！！</span></span><br><span>#</span><span><span>source</span> activate atac </span><br>cat config.clean |while read id;<br>do echo $id<br>arr=($id)<br>fq2=${arr[2]}<br>fq1=${arr[1]}<br>sample=${arr[0]}<br><span>#</span><span><span># 比对过程15分钟一个样本</span></span><br>bowtie2  -p 5  --very-sensitive -X 2000 -x  $bowtie2_index -1 $fq1 -2 $fq2 |samtools sort  -O bam  -@ 5 -o - &gt; ${sample}.raw.bam <br>samtools index ${sample}.raw.bam <br>bedtools bamtobed -i ${sample}.raw.bam  &gt; ${sample}.raw.bed<br>samtools flagstat ${sample}.raw.bam  &gt; ${sample}.raw.stat<br><span>#</span><span> https://github.com/biod/sambamba/issues/177</span><br>sambamba markdup --overflow-list-size 600000  --tmpdir='./'  -r ${sample}.raw.bam  ${sample}.rmdup.bam<br>samtools index   ${sample}.rmdup.bam <br><span><br>#</span><span><span># ref:https://www.biostars.org/p/170294/ </span></span><br><span>#</span><span><span># Calculate %mtDNA:</span></span><br>mtReads=$(samtools idxstats  ${sample}.rmdup.bam | grep 'chrM' | cut -f 3)<br>totalReads=$(samtools idxstats  ${sample}.rmdup.bam | awk '{SUM += $3} END {print SUM}')<br>echo '==&gt; mtDNA Content:' $(bc &lt;&lt;&lt; "scale=2;100*$mtReads/$totalReads")'%'<br><br>samtools flagstat  ${sample}.rmdup.bam &gt; ${sample}.rmdup.stat<br>samtools view  -h  -f 2 -q 30    ${sample}.rmdup.bam   |grep -v chrM |samtools sort  -O bam  -@ 5 -o - &gt; ${sample}.last.bam<br>samtools index   ${sample}.last.bam <br>samtools flagstat  ${sample}.last.bam &gt; ${sample}.last.stat <br>bedtools bamtobed -i ${sample}.last.bam  &gt; ${sample}.bed<br>done <br></code></pre><p>其中bowtie2比对加入了<code>-X 2000</code> 参数，是最大插入片段，宽泛的插入片段范围(10-1000bp)</p><p>第一步得到的bam文件如下：</p><pre><code><span>-rw-rw-r--</span> 1 <span>stu</span> <span>stu</span> 3<span>.7G</span> <span>Aug</span> 25 14<span>:17</span> 2<span>-cell-1</span><span>.bam</span><br><span>-rw-rw-r--</span> 1 <span>stu</span> <span>stu</span> 4<span>.6G</span> <span>Aug</span> 25 15<span>:32</span> 2<span>-cell-2</span><span>.bam</span><br><span>-rw-rw-r--</span> 1 <span>stu</span> <span>stu</span> 1<span>.8G</span> <span>Aug</span> 25 15<span>:47</span> 2<span>-cell-4</span><span>.bam</span><br><span>-rw-rw-r--</span> 1 <span>stu</span> <span>stu</span> 5<span>.5G</span> <span>Aug</span> 25 16<span>:49</span> 2<span>-cell-5</span><span>.bam</span><br></code></pre><p>过滤后的bam文件是：</p><pre><code>3<span>.7G</span> <span>Aug</span> 25 21<span>:08</span> 2<span>-cell-1</span><span>.bam</span><br>490<span>M</span> <span>Aug</span> 25 21<span>:14</span> 2<span>-cell-1</span><span>.last</span><span>.bam</span><br>776<span>M</span> <span>Aug</span> 25 21<span>:13</span> 2<span>-cell-1</span><span>.rmdup</span><span>.bam</span><br>4<span>.6G</span> <span>Aug</span> 25 23<span>:51</span> 2<span>-cell-2</span><span>.bam</span><br>678<span>M</span> <span>Aug</span> 25 23<span>:58</span> 2<span>-cell-2</span><span>.last</span><span>.bam</span><br>1<span>.1G</span> <span>Aug</span> 25 23<span>:57</span> 2<span>-cell-2</span><span>.rmdup</span><span>.bam</span><br>1<span>.8G</span> <span>Aug</span> 26 00<span>:41</span> 2<span>-cell-4</span><span>.bam</span><br>427<span>M</span> <span>Aug</span> 26 00<span>:43</span> 2<span>-cell-4</span><span>.last</span><span>.bam</span><br>586<span>M</span> <span>Aug</span> 26 00<span>:43</span> 2<span>-cell-4</span><span>.rmdup</span><span>.bam</span><br>5<span>.5G</span> <span>Aug</span> 26 03<span>:26</span> 2<span>-cell-5</span><span>.bam</span><br>523<span>M</span> <span>Aug</span> 26 03<span>:33</span> 2<span>-cell-5</span><span>.last</span><span>.bam</span><br>899<span>M</span> <span>Aug</span> 26 03<span>:32</span> 2<span>-cell-5</span><span>.rmdup</span><span>.bam</span><br></code></pre><p>上述脚本的步骤都可以<strong>拆分运行</strong>，比如bam文件构建index或者转为bed的:</p><pre><code>ls *.last.bam|xargs -i samtools index {} <br>ls *.last.bam|while read id;do (bedtools bamtobed -i $id &gt;${id%%.*}.bed) ;done<br>ls *.raw.bam|while read id;do (nohup bedtools bamtobed -i $id &gt;${id%%.*}.raw.bed &amp; ) ;done<br></code></pre><p>最后得到的bed文件是：</p><pre><code>237<span>M</span> <span>Aug</span> 26 08<span>:00</span> 2<span>-cell-1</span><span>.bed</span><br>338<span>M</span> <span>Aug</span> 26 08<span>:01</span> 2<span>-cell-2</span><span>.bed</span><br>203<span>M</span> <span>Aug</span> 26 08<span>:01</span> 2<span>-cell-4</span><span>.bed</span><br>254<span>M</span> <span>Aug</span> 26 08<span>:01</span> 2<span>-cell-5</span><span>.bed</span><br></code></pre><h3><span>第4步，使用macs2找peaks</span></h3><pre><code><span>#</span><span> macs2 callpeak -t 2-cell-1.bed  -g mm --nomodel --<span>shift</span> -100 --extsize 200  -n 2-cell-1 --outdir ../peaks/</span><br>ls *.bed | while read id ;do (macs2 callpeak -t $id  -g mm --nomodel --sHit  -100 --extsize 200  -n ${id%%.*} --outdir ../peaks/) ;done <br><span>#</span><span><span># shell 13问</span></span><br></code></pre><p>macs2软件说明书详见：https://www.jianshu.com/p/21e8c51fca23</p><h3><span>第5步，计算插入片段长度，FRiP值，IDR计算重复情况</span></h3><p>非冗余非线粒体能够比对的fragment、比对率、NRF、PBC1、PBC2、peak数、无核小体区NFR、TSS富集、FRiP 、IDR重复的一致性！</p><p>名词解释：https://www.encodeproject.org/data-standards/terms/</p><p>参考：https://www.encodeproject.org/atac-seq/</p><p>看 <strong>sam文件第9列，</strong>在R里面统计绘图</p><pre><code>cmd=commandArgs(trailingOnly=<span>TRUE</span>); <br>input=cmd[<span>1</span>]; output=cmd[<span>2</span>]; <br>a=abs(as.numeric(read.table(input)[,<span>1</span>])); <br>png(file=output);<br>hist(a,<br>main=<span>"Insertion Size distribution"</span>,<br>ylab=<span>"Read Count"</span>,xlab=<span>"Insert Size"</span>,<br>xaxt=<span>"n"</span>,<br>breaks=seq(<span>0</span>,max(a),by=<span>10</span>)<br>); <br><br>axis(side=<span>1</span>,<br>at=seq(<span>0</span>,max(a),by=<span>100</span>),<br>labels=seq(<span>0</span>,max(a),by=<span>100</span>)<br>);<br><br>dev.off()  <br></code></pre><p>有了上面的绘图R脚本就可以在批量检验bam文件进行出图。</p><p>还有NFR:https://github.com/GreenleafLab/NucleoATAC/issues/18</p><p><strong>FRiP值</strong>的计算：fraction of reads in called peak regions</p><pre><code>bedtools intersect -a ../new/2-cell-1.bed -b 2-cell-1_peaks.narrowPeak |wc -l<br>148928<br>wc ../new/2-cell-1.bed<br>5105844<br>wc ../new/2-cell-1.raw.bed<br>5105844<br><span>#</span><span><span>## 搞清楚 FRiP值具体定义：</span></span><br><br><br>ls *narrowPeak|while  read id;<br>do <br>echo $id<br>bed=../new/$(basename $id "_peaks.narrowPeak").raw.bed<br><span>#</span><span>ls  -lh <span>$bed</span> </span><br>Reads=$(bedtools intersect -a $bed -b $id |wc -l|awk '{print $1}')<br>totalReads=$(wc -l $bed|awk '{print $1}')<br>echo $Reads  $totalReads <br>echo '==&gt; FRiP value:' $(bc &lt;&lt;&lt; "scale=2;100*$Reads/$totalReads")'%'<br>done <br></code></pre><p><strong>Fraction of reads in peaks (FRiP)</strong> - Fraction of all mapped reads that fall into the called peak regions, i.e. usable reads in significantly enriched peaks divided by all usable reads. In general, FRiP scores correlate positively with the number of regions. (Landt et al, Genome Research Sept. 2012, 22(9): 1813–1831)</p><p>文章其它指标：https://www.nature.com/articles/sdata2016109/tables/4</p><p>可以使用R包看不同peaks文件的overlap情况。</p><pre><code><span>if</span>(<span>F</span>){<br>  options(BioC_mirror=<span>"https://mirrors.ustc.edu.cn/bioc/"</span>) <br>  options(<span>"repos"</span> = c(CRAN=<span>"https://mirrors.tuna.tsinghua.edu.cn/CRAN/"</span>))<br>  <span>source</span>(<span>"http://bioconductor.org/biocLite.R"</span>) <br><br>  <span>library</span>(<span>'BiocInstaller'</span>)<br>  biocLite(<span>"ChIPpeakAnno"</span>)<br>  biocLite(<span>"ChIPseeker"</span>)<br><br>}<br><br><span>library</span>(ChIPseeker)<br><span>library</span>(ChIPpeakAnno)<br>list.files(<span>'project/atac/peaks/'</span>,<span>"*.narrowPeak"</span>)<br>tmp=lapply(list.files(<span>'project/atac/peaks/'</span>,<span>"*.narrowPeak"</span>),<span>function</span>(x){<br>  <span>return</span>(readPeakFile(file.path(<span>'project/atac/peaks/'</span>, x))) <br>})<br><br>ol &lt;- findOverlapsOfPeaks(tmp[[<span>1</span>]],tmp[[<span>2</span>]])<br>png(<span>'overlapVenn.png'</span>)<br>makeVennDiagram(ol)<br>dev.off()<br></code></pre><p>也可以使用专业软件，<strong>IDR</strong> 来进行计算出来，同时考虑peaks间的overlap，和富集倍数的一致性 。</p><pre><code>source activate atac<br><span>#</span><span> 可以用search先进行检索</span><br>conda search idr<br>source  deactivate<br><span>#</span><span><span># 保证所有的软件都是安装在 wes 这个环境下面</span></span><br>conda  create -n py3 -y   python=3 idr<br>conda activate py3<br>idr -h <br>idr --samples  2-cell-1_peaks.narrowPeak 2-cell-2_peaks.narrowPeak  --plot<br></code></pre><p>结果如下：</p><pre><code>Initial parameter values: [<span>0.10</span> <span>1.00</span> <span>0.20</span> <span>0.50</span>]<br>Final parameter values: [<span>0.00</span> <span>1.06</span> <span>0.64</span> <span>0.87</span>]<br><span>Number</span> <span>of</span> reported peaks - <span>5893</span>/<span>5893</span> (<span>100.0</span>%)<br><br><span>Number</span> <span>of</span> peaks passing IDR cutoff <span>of</span> <span>0.05</span> - <span>674</span>/<span>5893</span> (<span>11.4</span>%)<br></code></pre><p>参考：https://www.biostat.wisc.edu/~kendzior/STAT877/SLIDES/keles3.pdf</p><h3><span>第6步，deeptools的可视化</span></h3><p>具体仍然是见：https://mp.weixin.qq.com/s/a4qAcKE1DoukpLVV_ybobA 在ChiP-seq 讲解。</p><p>首先把bam文件转为bw文件，详情：http://www.bio-info-trainee.com/1815.html</p><pre><code>cd  ~/project/atac/new<br>source activate atac<br><span>#</span><span>ls  *.bam  |xargs -i samtools index {} </span><br>ls *last.bam |while read id;do<br>nohup bamCoverage -p 5 --normalizeUsing CPM -b $id -o ${id%%.*}.last.bw &amp; <br>done <br><br>cd dup <br>ls  *.bam  |xargs -i samtools index {} <br>ls *.bam |while read id;do<br>nohup bamCoverage --normalizeUsing CPM -b $id -o ${id%%.*}.rm.bw &amp; <br>done <br></code></pre><p>查看<strong>TSS附件信号</strong>强度：</p><pre><code><span>#</span><span><span># both -R and -S can accept multiple files </span></span><br>mkdir -p  ~/project/atac/tss<br>cd   ~/project/atac/tss <br>source activate atac<br>computeMatrix reference-point  --referencePoint TSS  -p 15  \<br>-b 10000 -a 10000    \<br>-R /public/annotation/CHIPseq/mm10/ucsc.refseq.bed  \<br>-S ~/project/atac/new/*.bw  \<br>--skipZeros  -o matrix1_test_TSS.gz  \<br>--outFileSortedRegions regions1_test_genes.bed<br><span><br>#</span><span><span>#     both plotHeatmap and plotProfile will use the output from   computeMatrix</span></span><br>plotHeatmap -m matrix1_test_TSS.gz  -out test_Heatmap.png<br>plotHeatmap -m matrix1_test_TSS.gz  -out test_Heatmap.pdf --plotFileFormat pdf  --dpi 720  <br>plotProfile -m matrix1_test_TSS.gz  -out test_Profile.png<br>plotProfile -m matrix1_test_TSS.gz  -out test_Profile.pdf --plotFileFormat pdf --perGroup --dpi 720 <br><span><br>#</span><span><span>## 如果要批处理 ，需要学习好linux命令。</span></span><br></code></pre><p>下载 bed文件：https://genome.ucsc.edu/cgi-bin/hgTables 只需要3列坐标格式文件。</p><p><strong>查看基因body的信号强度</strong></p><pre><code>source activate atac<br>computeMatrix scale-regions  -p 15  \<br>-R /public/annotation/CHIPseq/mm10/ucsc.refseq.bed  \<br>-S ~/project/atac/new/*.bw  \<br>-b 10000 -a 10000  \<br>--skipZeros -o matrix1_test_body.gz<br>plotHeatmap -m matrix1_test_body.gz  -out ExampleHeatmap1.png <br><br>plotHeatmap -m matrix1_test_body.gz  -out test_body_Heatmap.png<br>plotProfile -m matrix1_test_body.gz  -out test_body_Profile.png<br></code></pre><p><strong>ngsplot</strong>也是可以的。</p><h3><span>第7步，peaks注释</span></h3><p>统计peak在<strong>promoter，exon，intron和intergenic区域</strong>的分布</p><pre><code><span>if</span>(<span>F</span>){<br>options(BioC_mirror=<span>"https://mirrors.ustc.edu.cn/bioc/"</span>) <br>options(<span>"repos"</span> = c(CRAN=<span>"https://mirrors.tuna.tsinghua.edu.cn/CRAN/"</span>))<br><span>source</span>(<span>"http://bioconductor.org/biocLite.R"</span>) <br><br><span>library</span>(<span>'BiocInstaller'</span>)<br>biocLite(<span>"TxDb.Mmusculus.UCSC.mm10.knownGene"</span>)<br>biocLite(<span>"org.Mm.eg.db"</span>)<br>}<br><br>bedPeaksFile         = <span>'8WG16_summits.bed'</span>; <br>bedPeaksFile<br><span>## loading packages</span><br><span>require</span>(ChIPseeker)<br><span>require</span>(TxDb.Mmusculus.UCSC.mm10.knownGene)<br>txdb &lt;- TxDb.Mmusculus.UCSC.mm10.knownGene<br><span>require</span>(clusterProfiler) <br>peak &lt;- readPeakFile( bedPeaksFile )  <br>keepChr= !grepl(<span>'_'</span>,seqlevels(peak))<br>seqlevels(peak, pruning.mode=<span>"coarse"</span>) &lt;- seqlevels(peak)[keepChr]<br>peakAnno &lt;- annotatePeak(peak, tssRegion=c(-<span>3000</span>, <span>3000</span>), <br>                         TxDb=txdb, annoDb=<span>"org.Mm.eg.db"</span>) <br>peakAnno_df &lt;- as.data.frame(peakAnno)<br><br><br>promoter &lt;- getPromoters(TxDb=txdb, upstream=<span>3000</span>, downstream=<span>3000</span>)<br>tagMatrix &lt;- getTagMatrix(peak, windows=promoter) <br><span># 然后查看这些peaks在所有基因的启动子附近的分布情况，热图模式</span><br>tagHeatmap(tagMatrix, xlim=c(-<span>3000</span>, <span>3000</span>), color=<span>"red"</span>)<br><span># 然后查看这些peaks在所有基因的启动子附近的分布情况，信号强度曲线图</span><br>plotAvgProf(tagMatrix, xlim=c(-<span>3000</span>, <span>3000</span>), <br>            xlab=<span>"Genomic Region (5'-&gt;3')"</span>, ylab = <span>"Read Count Frequency"</span>)<br>plotAnnoPie(peakAnno)<br></code></pre><p>可以载入IGV看看效果，检测软件找到的peaks是否真的合理，还可以配合rmarkdown来出<strong>自动化报告。</strong> https://ke.qq.com/course/274681</p><p>也可以使用其它代码进行下游分析； https://github.com/jmzeng1314/NGS-pipeline/tree/master/CHIPseq</p><p>Homer 可以做</p><pre><code><span># perl ~/miniconda3/envs/atac/share/homer-4.9.1-5/configureHomer.pl  -install mm10 </span><br><span># ln -s /home/jmzeng/miniconda3/envs/chipseq/share/homer-4.9.1-5/data/genomes/ genomes</span><br><span># cp  /home/jmzeng/miniconda3/envs/chipseq/share/homer-4.9.1-5/config.txt  /home/stu/miniconda3/envs/atac/share/homer-4.9.1-5/config.txt</span><br><span>## 保证数据库下载是OK</span><br>ls -lh  ~/miniconda3/envs/atac/share/homer-<span>4.9</span><span>.1</span>-<span>5</span>/data/genomes  <br>mkdir -p  ~/project/atac/peaks<br><span>source</span> activate atac<br>cd   ~/project/atac/peaks  <br>ls *.narrowPeak |<span>while</span> read id;<br>do <br>echo $id<br>awk <span>'{print $4"\t"$1"\t"$2"\t"$3"\t+"}'</span> $id &gt;{id%%.*}.homer_peaks.tmp<br>annotatePeaks.pl  {id%%.*}.homer_peaks.tmp mm10  <span>1</span>&gt;${id%%.*}.peakAnn.xls<br>  <span>2</span>&gt;${id%%.*}.annLog.txt<br>done <br></code></pre><p>Bedtools 也可以做 ：https://bedtools.readthedocs.io/en/latest/content/tools/annotate.html</p><h3><span>第8步，motif寻找及注释</span></h3><p>Homer 可以做</p><pre><code>ls -lh  ~/miniconda3/envs/atac/share/homer-<span>4.9</span><span>.1</span>-<span>5</span>/data/genomes  <br>mkdir -p  ~/project/atac/motif<br>cd   ~/project/atac/motif  <br><span>source</span> activate atac<br>ls ../peaks/*.narrowPeak |<span>while</span> read id;<br>do <br>file=$(basename $id )<br>sample=${file%%.*} <br>echo $sample <br>awk <span>'{print $4"\t"$1"\t"$2"\t"$3"\t+"}'</span> $id &gt; ${sample}.homer_peaks.tmp<br>nohup findMotifsGenome.pl ${sample}.homer_peaks.tmp  mm10 ${sample}_motifDir -len <span>8</span>,<span>10</span>,<span>12</span>  &amp; <br>done <br></code></pre><p><strong>meme</strong> 也可以做 ， 获取 序列：https://github.com/jmzeng1314/NGS-pipeline/blob/master/CHIPseq/step7-peaks2sequence.R</p><p>R包，比如 <code>motifmatchr</code>包 也可以做。 https://bioconductor.org/packages/release/bioc/html/motifmatchr.html</p><h3><span>第9步，差异peaks分析</span></h3><p>diffbind 使用R包DiffBind进行chip-seq差异峰的分析-表观组-生信技能树</p><p>自行摸索R包用法</p><h3><span>第10步，为什么我不用 esATAC</span></h3><p>新鲜出炉的一篇文章，esATAC: an easy-to-use systematic pipeline for ATAC-seq data analysis 发表于 Bioinformatics.</p><pre><code><span># https://stackoverflow.com/questions/30738974/rjava-load-error-in-rstudio-r-after-upgrading-to-osx-yosemite</span><br>options(BioC_mirror=<span>"https://mirrors.ustc.edu.cn/bioc/"</span>) <br>options(<span>"repos"</span> = c(CRAN=<span>"https://mirrors.tuna.tsinghua.edu.cn/CRAN/"</span>))<br><span>source</span>(<span>"http://bioconductor.org/biocLite.R"</span>) <br><br><span>library</span>(<span>'BiocInstaller'</span>)<br>biocLite(<span>"TxDb.Mmusculus.UCSC.mm10.knownGene"</span>)<br>biocLite(<span>"org.Mm.eg.db"</span>)<br>biocLite(<span>"esATAC"</span>)<br>install.packages(<span>'idr'</span>)<br><span>library</span>(esATAC)<br>printMap()<br></code></pre><p>因为这个R包就是包装了前面我们讲解的多个分析步骤，而我比较擅长命令行处理数据，所以不喜欢用这样的封装。</p><p>其实你会了我讲解的整个流程，很容易看懂很多新的文章，比如：</p><p><img data-ratio="0.6185567010309279" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wwo2dn1JUBh3VOX7Bpiatm8AZiazK2spEyvDxh1n0R7AtQgtvhlGLT1956eGtWysR71WRZBVXnyydPQ/640?wx_fmt=png" data-type="png" data-w="970" src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wwo2dn1JUBh3VOX7Bpiatm8AZiazK2spEyvDxh1n0R7AtQgtvhlGLT1956eGtWysR71WRZBVXnyydPQ/640?wx_fmt=png"></p><h3><span>第11步，多组学整合分析</span></h3><p>待续，高级课程制作中。</p><p>然后，如果你付费参加了我的这个课程，就应该是需要教学视频配套PPT，综述以及文献，及各大公司的结题报告，思维导图和教学示例数据的，现在就<strong><span>扫码加入</span></strong>我们的答疑微信聊天群吧！（附上你的购买截图，ID，小助手会核实，谢谢合作！）</p><p><br></p><section data-style-type="1" data-tools="新媒体排版" data-id="13013"><section><section><section><section><p><span><span>扫描下面二维码 </span>添加<strong><span>小助手公众号</span></strong></span><br></p></section></section></section></section></section><p><br></p><p><br></p><p><span><strong>投稿、合作等事宜请发送到邮箱：jmzeng1314@163.com</strong></span></p><p><br></p><p><br></p><section><section><section><section><p><span><strong><span>    欢迎添加入群   </span></strong></span></p><section><section><p><img data-copyright="0" data-cropselx1="0" data-cropselx2="156" data-cropsely1="0" data-cropsely2="156" data-ratio="1" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/cZNhZQ6j4wy0mMDISw79z65DbO2FD62RpFOTAQzJSP4jM1zmQ3emHgQG93cmzH6aGTV4JLArRSZNAbOJ5lknGw/640?wx_fmt=jpeg" data-type="jpeg" data-w="430" src="https://mmbiz.qpic.cn/mmbiz_jpg/cZNhZQ6j4wy0mMDISw79z65DbO2FD62RpFOTAQzJSP4jM1zmQ3emHgQG93cmzH6aGTV4JLArRSZNAbOJ5lknGw/640?wx_fmt=jpeg"></p></section><p><span>（长按图片扫描二维码可添加微信）</span></p></section></section></section></section></section><p><br></p><p><br></p><p>点击阅读原文直达我们的网易云课堂购买链接！（请务必思考再三哦，不要瞎买）</p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/XQITfdnS6Zn6iTBtuladPw",target="_blank" rel="noopener noreferrer">原文链接</a>
