---
title: "R语言做基因表达量和变异位点的关联分析eQTL"
date: 2024-05-02T08:23:01Z
draft: ["false"]
tags: [
  "fetched",
  "小明的数据分析笔记本"
]
categories: ["Acdemic"]
---
R语言做基因表达量和变异位点的关联分析eQTL by 小明的数据分析笔记本
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com" data-mpa-powered-by="yiban.io"><h2 data-tool="mdnice编辑器"><span></span><span>参考链接</span><span></span><span> </span></h2><p data-tool="mdnice编辑器">http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/runit.html</p><h2 data-tool="mdnice编辑器"><span></span><span>表达数据来源于论文</span><span></span><span> </span></h2><blockquote data-tool="mdnice编辑器"><span></span><p>Population level gene expression can repeatedly link genes to functions in maize</p></blockquote><p data-tool="mdnice编辑器">https://www.biorxiv.org/content/10.1101/2023.10.31.565032v1</p><p data-tool="mdnice编辑器">数据下载链接 https://doi.org/10.6084/m9.figshare.24470758.v1</p><h2 data-tool="mdnice编辑器"><span></span><span>变异数据来源于论文</span><span></span><span> </span></h2><blockquote data-tool="mdnice编辑器"><span></span><p>A common resequencing-based genetic marker data set for global maize diversity</p></blockquote><p data-tool="mdnice编辑器">https://onlinelibrary.wiley.com/doi/10.1111/tpj.16123</p><p data-tool="mdnice编辑器">数据下载链接 https://datadryad.org/stash/dataset/doi:10.5061/dryad.bnzs7h4f1</p><p data-tool="mdnice编辑器">这篇论文对应的代码 https://github.com/mgrzy/Maize_Genetic_Variants_v5/tree/main</p><h2 data-tool="mdnice编辑器"><span></span><span>论文用到的参考基因组</span><span></span><span> </span></h2><p data-tool="mdnice编辑器">B73_RefGen_V5 maize reference genome</p><p data-tool="mdnice编辑器">De novo assembly, annotation, and comparative analysis of 26 diverse maize genomes</p><p data-tool="mdnice编辑器">https://www.science.org/doi/full/10.1126/science.abg5289</p><p data-tool="mdnice编辑器">参考基因组下载链接https://download.maizegdb.org/Zm-B73-REFERENCE-NAM-5.0/</p><h2 data-tool="mdnice编辑器"><span> </span></h2><h2 data-tool="mdnice编辑器"><span></span><span>变异数据的处理</span><span></span><span> </span></h2><p data-tool="mdnice编辑器">只下载了8 9 10 号染色体数据，只保留插入缺失变异，只保留了100个样本，最小等位基因频率0.05</p><p data-tool="mdnice编辑器">使用 VCF2PCACluster 这个软件计算PCA , 这个软件只计算snp ，需要自己写脚本<code>editRefAlt.py</code>修改vcf文件里的ref和alt列</p><p data-tool="mdnice编辑器">自己写脚本<code>convertVcfTo012Matrix.py</code>把vcf文件转换成 0 1 2 矩阵</p><h2 data-tool="mdnice编辑器"><span></span><span>表达量数据处理</span><span></span><span> </span></h2><p data-tool="mdnice编辑器">8 9 10 号染色体的基因，只保留100个样本。在&gt;=80个样本中 TPM &gt; 0.05 的基因保留，最后只保留了4000多个基因，标准化，然后peer 计算隐藏因子 <code>run_peer.R</code></p><h2 data-tool="mdnice编辑器"><span></span><span>最终的输入数据</span><span></span><span> </span></h2><figure data-tool="mdnice编辑器"><img data-imgfileid="100013604" data-ratio="0.44124168514412415" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/t1wZDoUyFk7icZnmE67wTs3KcviaTSnZ5aEW6MDzBkIDWp6neibianUB83gAoq5ePQTc071Q4goPPiaknUm0peyucQg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="451" src="https://mmbiz.qpic.cn/sz_mmbiz_png/t1wZDoUyFk7icZnmE67wTs3KcviaTSnZ5aEW6MDzBkIDWp6neibianUB83gAoq5ePQTc071Q4goPPiaknUm0peyucQg/640?wx_fmt=png&amp;from=appmsg"><figcaption>image.png</figcaption></figure><h2 data-tool="mdnice编辑器"><span></span><span>R语言里的代码</span><span></span><span> </span></h2><pre data-tool="mdnice编辑器"><span></span><code>library(MatrixEQTL)<br>SNP_file_name&lt;-<span>"eQTL_maize/chr.onlyInDels.impute.maf.recode.012.matrix"</span><br>snps = SlicedData<span>$new</span>();<br>snps<span>$fileDelimiter</span> = <span>"\t"</span>;      <span># the TAB character</span><br>snps<span>$fileOmitCharacters</span> = <span>"NA"</span>; <span># denote missing values;</span><br>snps<span>$fileSkipRows</span> = 1;          <span># one row of column labels</span><br>snps<span>$fileSkipColumns</span> = 1;       <span># one column of row labels</span><br>snps<span>$fileSliceSize</span> = 2000;      <span># read file in slices of 2,000 rows</span><br>snps<span>$LoadFile</span>(SNP_file_name);<br><br>expression_file_name&lt;-<span>"eQTL_maize/quant.norm.tpm.tsv"</span><br><br>gene = SlicedData<span>$new</span>();<br>gene<span>$fileDelimiter</span> = <span>"\t"</span>;      <span># the TAB character</span><br>gene<span>$fileOmitCharacters</span> = <span>"NA"</span>; <span># denote missing values;</span><br>gene<span>$fileSkipRows</span> = 1;          <span># one row of column labels</span><br>gene<span>$fileSkipColumns</span> = 1;       <span># one column of row labels</span><br>gene<span>$fileSliceSize</span> = 2000;      <span># read file in slices of 2,000 rows</span><br>gene<span>$LoadFile</span>(expression_file_name);<br><br>covariates_file_name&lt;-<span>"eQTL_maize/covariates.tsv"</span><br><br>cvrt = SlicedData<span>$new</span>();<br>cvrt<span>$fileDelimiter</span> = <span>"\t"</span>;      <span># the TAB character</span><br>cvrt<span>$fileOmitCharacters</span> = <span>"NA"</span>; <span># denote missing values;</span><br>cvrt<span>$fileSkipRows</span> = 1;          <span># one row of column labels</span><br>cvrt<span>$fileSkipColumns</span> = 1;       <span># one column of row labels</span><br>cvrt<span>$LoadFile</span>(covariates_file_name);<br><br>snps_location_file_name&lt;-<span>"eQTL_maize/snpPos.tsv"</span><br>gene_location_file_name&lt;-<span>"eQTL_maize/genePos.tsv"</span><br><br>snpspos = read.table(snps_location_file_name, header = TRUE, stringsAsFactors = FALSE);<br>genepos = read.table(gene_location_file_name, header = TRUE, stringsAsFactors = FALSE);<br><br>genepos %&gt;% head()<br><br>cisDist&lt;-5000<br>pvOutputThreshold_cis&lt;-0.1<br>pvOutputThreshold_tra&lt;-0.00000000001<br><br>errorCovariance&lt;-numeric()<br><br>useModel&lt;-modelLINEAR<br><br>output_file_name_cis = tempfile();<br>output_file_name_tra = tempfile();<br><br>me = Matrix_eQTL_main(<br>  snps = snps,<br>  gene = gene,<br>  cvrt = cvrt,<br>  output_file_name     = output_file_name_tra,<br>  pvOutputThreshold     = pvOutputThreshold_tra,<br>  useModel = useModel,<br>  errorCovariance = errorCovariance,<br>  verbose = TRUE,<br>  output_file_name.cis = output_file_name_cis,<br>  pvOutputThreshold.cis = pvOutputThreshold_cis,<br>  snpspos = snpspos,<br>  genepos = genepos,<br>  cisDist = cisDist,<br>  pvalue.hist = <span>"qqplot"</span>,<br>  min.pv.by.genesnp = FALSE,<br>  noFDRsaveMemory = FALSE);<br><br><br>unlink(output_file_name_tra);<br>unlink(output_file_name_cis);<br><br>me<span>$cis</span><span>$eqtls</span> %&gt;% <br>  left_join(snpspos,by=c(<span>"snps"</span>=<span>"svid"</span>)) %&gt;% <br>  ggplot(aes(x=pos,y=-log10(FDR)))+<br>  geom_point(aes(color=chr))+<br>  facet_wrap(~chr,nrow = 1)+<br>  theme_bw()+<br>  theme(panel.grid = element_blank(),<br>        panel.border = element_blank())<br><br>me<span>$cis</span><span>$eqtls</span> %&gt;% head()<br>me<span>$cis</span><span>$eqtls</span> %&gt;% pull(gene) %&gt;% unique() %&gt;% length()<br>genepos %&gt;% dim()<br>read_tsv(<span>"eQTL_maize/quant.norm.tpm.tsv"</span>) %&gt;% dim()<br><br>me<span>$cis</span><span>$eqtls</span> %&gt;% <br>  filter(-log10(FDR)&gt;=25)<br>genepos %&gt;% filter(ID==<span>"Zm00001eb347950"</span>)<br><br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100013607" data-ratio="0.4268224819143016" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/t1wZDoUyFk7icZnmE67wTs3KcviaTSnZ5aV3PHqiauKtm2biajwJ13cHxmicIvC3IW7aPRseQnCthJI8HIB3dM66k1w/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1797" src="https://mmbiz.qpic.cn/sz_mmbiz_png/t1wZDoUyFk7icZnmE67wTs3KcviaTSnZ5aV3PHqiauKtm2biajwJ13cHxmicIvC3IW7aPRseQnCthJI8HIB3dM66k1w/640?wx_fmt=png&amp;from=appmsg"><figcaption>image.png</figcaption></figure><p data-tool="mdnice编辑器"><strong>欢迎大家关注我的公众号</strong></p><p data-tool="mdnice编辑器"><strong>小明的数据分析笔记本</strong></p><section><mp-common-profile data-pluginname="mpprofile" data-id="MzI3NzQ3MTcxMg==" data-headimg="http://mmbiz.qpic.cn/mmbiz_png/t1wZDoUyFk5t1sOnM0iabvBhnfIj5YpyqrMib0E1MGCd9ibcYxaOPZd0GWhQBDvK2BPEwsicQxd6y5MHLfphnwHnow/0?wx_fmt=png" data-nickname="小明的数据分析笔记本" data-alias="" data-signature="分享R语言和python在生物信息领域做数据分析和数据可视化的简单小例子；偶尔会分享一些组学数据处理相关的内容" data-from="0" data-is_biz_ban="0"></mp-common-profile></section><p data-tool="mdnice编辑器"><strong></strong></p><blockquote data-tool="mdnice编辑器"><span></span><p>小明的数据分析笔记本 公众号 主要分享：1、R语言和python做数据分析和数据可视化的简单小例子；2、园艺植物相关转录组学、基因组学、群体遗传学文献阅读笔记；3、生物信息学入门学习资料及自己的学习笔记！</p></blockquote><p data-tool="mdnice编辑器">微信公众号好像又有改动，如果没有将这个公众号设为星标的话，会经常错过公众号的推文，个人建议将 小明的数据分析笔记本 公众号添加星标，添加方法是</p><p data-tool="mdnice编辑器"><strong>点开公众号的页面，右上角有三个点</strong><img data-imgfileid="100013606" data-ratio="0.9693333333333334" data-src="https://mmbiz.qpic.cn/sz_mmbiz_jpg/t1wZDoUyFk7icZnmE67wTs3KcviaTSnZ5aXeTAZp2iavHIY0pAibIe0ln2xBAhv26UltsopauNDjibIjkzTj4yibaanQ/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="750" src="https://mmbiz.qpic.cn/sz_mmbiz_jpg/t1wZDoUyFk7icZnmE67wTs3KcviaTSnZ5aXeTAZp2iavHIY0pAibIe0ln2xBAhv26UltsopauNDjibIjkzTj4yibaanQ/640?wx_fmt=other&amp;from=appmsg"></p><p data-tool="mdnice编辑器"><strong>点击三个点，会跳出界面</strong></p><figure data-tool="mdnice编辑器"><img data-imgfileid="100013605" data-ratio="2.120879120879121" data-src="https://mmbiz.qpic.cn/sz_mmbiz_jpg/t1wZDoUyFk7icZnmE67wTs3KcviaTSnZ5anWnRl4h61bPQicEUdT5jia4SEQfITSTRJXEnS9IMHoRVOdlkDH5BQNwA/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="364" src="https://mmbiz.qpic.cn/sz_mmbiz_jpg/t1wZDoUyFk7icZnmE67wTs3KcviaTSnZ5anWnRl4h61bPQicEUdT5jia4SEQfITSTRJXEnS9IMHoRVOdlkDH5BQNwA/640?wx_fmt=other&amp;from=appmsg"></figure><p data-tool="mdnice编辑器"><strong>直接点击</strong></p></section><p><br></p><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/gcdP8d4CRdG-LfX1NsznAA",target="_blank" rel="noopener noreferrer">原文链接</a>
