---
title: "SCS【36】单细胞转录组之k-近邻图差异丰度测试(miloR)"
date: 2024-05-16T14:08:49Z
draft: ["false"]
tags: [
  "fetched",
  "桓峰基因"
]
categories: ["Acdemic"]
---
SCS【36】单细胞转录组之k-近邻图差异丰度测试(miloR) by 桓峰基因
------
<div><p><img data-ratio="0.4249471458773784" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMx7OKa5qBGY4pgKCLKezynZRDsJPKlHCRu4DPrPiaD2kbh9pByUtUFkIRvwH4XRJDyf9KuuZianqlHg/640?wx_fmt=pngwxfrom=5wx_lazy=1wx_co=1" data-type="pngwxfrom=5wx_lazy=1wx_co=1" data-w="946" src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMx7OKa5qBGY4pgKCLKezynZRDsJPKlHCRu4DPrPiaD2kbh9pByUtUFkIRvwH4XRJDyf9KuuZianqlHg/640?wx_fmt=pngwxfrom=5wx_lazy=1wx_co=1"></p><section><mp-common-profile data-pluginname="mpprofile" data-id="MzU0NDY2NTUwOA==" data-headimg="http://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMzQ2iaDaME4SMFFyIc3WCrVf5loZv2JiahbJiaozT9fNzGPkphu6icmGg8FlIsOicJ6F7OJicH2qA0r3k5g/0?wx_fmt=png" data-nickname="桓峰基因" data-alias="KyohoGene" data-signature="桓峰基因推出全新生信分析文字版和视频版免费教程，欢迎大家来围观，在公众号搜索即可获得相关教程！有测序，生信分析，文章合作，实验室各种耗材试剂等业务欢迎来咨询，关注桓峰基因官网：http://www.kyohogene.com/" data-from="0" data-is_biz_ban="0"></mp-common-profile></section><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><hr data-tool="mdnice编辑器"><h2 data-tool="mdnice编辑器">简 介</h2><p data-tool="mdnice编辑器">当前用于比较分析单细胞数据集的计算工作流程在测试不同实验条件下的差异丰度时，通常使用离散 cluster 作为输入。然而，聚类并不总是提供适当的分辨率，也不能捕获连续的轨迹。在这里，我们提出了Milo，一个可扩展的统计框架，通过在k近邻图上将单细胞分配到部分重叠的邻域来执行差异丰度测试。使用模拟和单细胞RNA测序(scRNA-seq)数据，表明 Milo 可以识别因离散细胞成 cluster 而被掩盖的扰动，它在批效应中保持错误发现率控制，并且它优于其他差异丰度测试策略。Milo  确定了衰老小鼠胸腺中命运偏向性上皮前体的衰退，并确定了人类肝硬化中多个谱系的扰动。由于Milo是基于细胞-细胞相似性结构，它可能也适用于scRNA-seq以外的单细胞数据。</p><p><img data-imgfileid="100009084" data-ratio="0.3479188900747065" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMyeTMEkfvjPHZ5gDuKNkoZdVDd8oTtNicjE49IToDkVZa4SmX63Wz7L7KhVpvdianB3UQbeqUKe2SoQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="937" src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMyeTMEkfvjPHZ5gDuKNkoZdVDd8oTtNicjE49IToDkVZa4SmX63Wz7L7KhVpvdianB3UQbeqUKe2SoQ/640?wx_fmt=png&amp;from=appmsg"></p><p data-tool="mdnice编辑器">作为差分富度图邻域的扰动细胞状态检测。a，Milo工作流程示意图。邻域定义在索引细胞上，使用图采样算法选择。根据实验设计对细胞进行定量，生成计数表。每个社区细胞计数使用负二项GLM建模，并进行假设检验以确定差异丰富的社区。b， KNN图的力定向布局，表示从两个实验条件下采样的细胞的模拟连续轨迹(上图:条件A，紫色;条件B，白色;下图:条件B中细胞的核密度。c，使用Milo进行假设检验，准确且特异性地检测到差异丰富的邻域(FDR为1%)。红点表示不同数量的邻域。d，米洛差分丰度检验结果的图示。节点是邻域，通过它们的对数折叠变化来着色。非差异丰富的邻域(FDR 1%)是白色的，大小对应于邻域中的细胞数量。图边描述了相邻邻域之间共享的单细胞数量。节点的布局由邻域索引单细胞在单个细胞的力定向嵌入中的位置决定。</p></section><p><img data-imgfileid="100009085" data-ratio="0.407149950347567" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMyeTMEkfvjPHZ5gDuKNkoZdEQwGib4CmcsYeu5zWSx4UsLuW4l9GicqyaibsgWElDsr3MOB30yf1nh7w/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1007" src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMyeTMEkfvjPHZ5gDuKNkoZdEQwGib4CmcsYeu5zWSx4UsLuW4l9GicqyaibsgWElDsr3MOB30yf1nh7w/640?wx_fmt=png&amp;from=appmsg"></p><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><h2 data-tool="mdnice编辑器"><span></span>软件包安装</h2><p data-tool="mdnice编辑器">安装较为简单，直接安装即可：</p><pre data-tool="mdnice编辑器"><span></span><code><span>if</span> (!requireNamespace(<span>"BiocManager"</span>, quietly = <span>TRUE</span>))<br>  install.packages(<span>"BiocManager"</span>)<br><br>BiocManager::install(<span>"miloR"</span>)<br><br><span>## Install development version</span><br>devtools::install_github(<span>"MarioniLab/miloR"</span>, ref=<span>"devel"</span>)<br></code></pre><h2 data-tool="mdnice编辑器"><span></span>数据读取</h2><p data-tool="mdnice编辑器">读入一个典型的单细胞数据，包括meta和表达矩阵，这里的矩阵是 SingleCellExperiment：</p><pre data-tool="mdnice编辑器"><span></span><code><span>library</span>(miloR)<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>## 载入需要的程辑包：edgeR<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>## 载入需要的程辑包：limma<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>data(sim_trajectory)<br>milo.meta &lt;- sim_trajectory$meta<br>head(milo.meta)<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>##   cell_id group_id Condition Replicate Sample<br>## 1      C1       M2         B        R1   B_R1<br>## 2      C2       M2         B        R1   B_R1<br>## 3      C3       M1         A        R1   A_R1<br>## 4      C4       M2         B        R1   B_R1<br>## 5      C5       M1         B        R1   B_R1<br>## 6      C6       M1         B        R1   B_R1<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>milo.obj &lt;- Milo(sim_trajectory$SCE)<br>milo.obj<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>## class: Milo <br>## dim: 500 500 <br>## metadata(0):<br>## assays(2): counts logcounts<br>## rownames(500): G1 G2 ... G499 G500<br>## rowData names(0):<br>## colnames(500): C1 C2 ... C499 C500<br>## colData names(2): cell_id group_id<br>## reducedDimNames(1): PCA<br>## mainExpName: NULL<br>## altExpNames(0):<br>## nhoods dimensions(2): 1 1<br>## nhoodCounts dimensions(2): 1 1<br>## nhoodDistances dimension(1): 0<br>## graph names(0):<br>## nhoodIndex names(1): 0<br>## nhoodExpression dimension(2): 1 1<br>## nhoodReducedDim names(0):<br>## nhoodGraph names(0):<br>## nhoodAdjacency dimension(2): 1 1<br></code></pre><h2 data-tool="mdnice编辑器"><span></span>实例操作</h2><h3 data-tool="mdnice编辑器"><span></span>差异分析<span></span></h3><p data-tool="mdnice编辑器">对workflow的基本描述：</p><p data-tool="mdnice编辑器">1，数据预处理与降维</p><p data-tool="mdnice编辑器">2，minimizing batch effects.如harmony等整合多sample的算法，但不要使用conos</p><p data-tool="mdnice编辑器">3，building the KNN graph</p><p data-tool="mdnice编辑器">4，definition of cell neighbourhoods and index cell samplling</p><p data-tool="mdnice编辑器">5，testing for differencial abundance in neighbourhoods</p><pre data-tool="mdnice编辑器"><span></span><code>milo.obj &lt;- buildGraph(milo.obj, k=<span>20</span>, d=<span>30</span>)<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>## Constructing kNN graph with k:20<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>milo.obj &lt;- makeNhoods(milo.obj, k=<span>20</span>, d=<span>30</span>, refined=<span>TRUE</span>, prop=<span>0.2</span>)<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>## Checking valid object<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>## Running refined sampling with reduced_dim<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>milo.obj &lt;- calcNhoodDistance(milo.obj, d=<span>30</span>)<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>## as(&lt;dgTMatrix&gt;, "dgCMatrix") is deprecated since Matrix 1.5-0; do as(., "CsparseMatrix") instead<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>milo.obj &lt;- countCells(milo.obj, samples=<span>"Sample"</span>, meta.data=milo.meta)<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>## Checking meta.data validity<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>## Counting cells in neighbourhoods<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>milo.design &lt;- as.data.frame(xtabs(~ Condition + Sample, data=milo.meta))<br>milo.design &lt;- milo.design[milo.design$Freq &gt; <span>0</span>, ]<br>rownames(milo.design) &lt;- milo.design$Sample<br>milo.design &lt;- milo.design[colnames(nhoodCounts(milo.obj)),]<br><br>milo.res &lt;- testNhoods(milo.obj, design=~Condition, design.df=milo.design)<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>## Using TMM normalisation<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>## Performing spatial FDR correction with k-distance weighting<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>head(milo.res)<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>##        logFC   logCPM           F      PValue         FDR Nhood  SpatialFDR<br>## 1 -1.2323695 15.20142 14.77221137 0.001157153 0.003085743     1 0.003009444<br>## 2 -0.2608166 15.18022  0.53107695 0.506271278 0.558644169     2 0.560950918<br>## 3  0.1058813 14.92559  0.07601276 0.812617751 0.813801465     3 0.813801465<br>## 4 -0.4161003 15.51904  1.75022259 0.220752573 0.261632679     4 0.259524289<br>## 5  0.5080137 15.50203  2.29454278 0.164196289 0.202087741     5 0.199463470<br>## 6  0.6824722 15.18011  3.12914902 0.106498870 0.136318554     6 0.135804176<br></code></pre><h3 data-tool="mdnice编辑器"><span></span>绘制火山图<span></span></h3><p data-tool="mdnice编辑器">首先设置pvalue和logFC的阈值，然后根据阈值分别为上调基因设置‘up’，下调基因设置‘Down’，无差异设置‘None’，保存到change列。这里的change列用来设置火山图点的颜色</p><pre data-tool="mdnice编辑器"><span></span><code>cut_off_pvalue = <span>0.05</span><br>cut_off_logFC = <span>1</span><br><span># </span><br>milo.res$Sig = ifelse(milo.res$PValue &lt; cut_off_pvalue &amp; <br>                   abs(milo.res$logFC) &gt;= cut_off_logFC, <br>                 ifelse(milo.res$logFC&gt; cut_off_logFC ,<span>'Up'</span>,<span>'Down'</span>),<span>'None'</span>)<br>table(milo.res$Sig)<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>## <br>## Down None   Up <br>##    4   19    9<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code><span>library</span>(ggplot2)<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>## Warning: 程辑包'ggplot2'是用R版本4.2.3 来建造的<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>ggplot(milo.res, aes(x = logFC, y = -log10(PValue), colour=Sig)) +<br>  geom_point(alpha=<span>0.4</span>, size=<span>3.5</span>) +<br>  scale_color_manual(values=c(<span>"#546de5"</span>, <span>"#d2dae2"</span>,<span>"#ff4757"</span>))+<br>  <span># 辅助线</span><br>  geom_vline(xintercept=c(-<span>1</span>,<span>1</span>),lty=<span>4</span>,col=<span>"black"</span>,lwd=<span>0.8</span>) +<br>  geom_hline(yintercept = -log10(cut_off_pvalue),<br>             lty=<span>4</span>,col=<span>"black"</span>,lwd=<span>0.8</span>) +<br>  <span># 坐标轴</span><br>  labs(x=<span>"log2(Fold Change)"</span>,<br>       y=<span>"-log10 (P-value)"</span>)+<br>  theme_bw()+<br>  ggtitle(<span>"Volcano Plot"</span>)+<br>  <span># 图例</span><br>  theme(plot.title = element_text(hjust = <span>0.5</span>), <br>        legend.position=<span>"right"</span>, <br>        legend.title = element_blank()<br>  )<br></code></pre><p><img data-galleryid="" data-imgfileid="100009086" data-ratio="0.7756756756756756" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMyeTMEkfvjPHZ5gDuKNkoZdEr4rf0PKZqsPjjeeSChzMRzYjsMWam9dnrENwGRGQTbDKXY6PElA0Q/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="740" src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMyeTMEkfvjPHZ5gDuKNkoZdEr4rf0PKZqsPjjeeSChzMRzYjsMWam9dnrENwGRGQTbDKXY6PElA0Q/640?wx_fmt=png&amp;from=appmsg"></p><h4 data-tool="mdnice编辑器"><span></span>Reference<span></span></h4><p data-tool="mdnice编辑器">Dann, E., Henderson, N.C., Teichmann, S.A. et al. Differential abundance testing on single-cell data using k-nearest neighbor graphs. Nat Biotechnol (2021).</p><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><h2 data-tool="mdnice编辑器">单细胞生信分析教程</h2><p data-tool="mdnice编辑器"><span><strong>桓峰基因公众号推出单细胞生信分析教程并配有视频在线教程，目前整理出来的相关教程目录如下：</strong></span></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247483888&amp;idx=1&amp;sn=823811c2126938cf65fd94f5049ac58f&amp;scene=21#wechat_redirect" data-linktype="2"><span>Topic 6. 克隆进化之 Canopy</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247483897&amp;idx=1&amp;sn=6368eb8c4c554b2f9a94f53af7f3da37&amp;scene=21#wechat_redirect" data-linktype="2"><span>Topic 7. 克隆进化之 Cardelino</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247483907&amp;idx=1&amp;sn=b3019237edc67037dead18087e6feb9d&amp;scene=21#wechat_redirect" data-linktype="2"><span>Topic 8. 克隆进化之 RobustClone</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247487851&amp;idx=1&amp;sn=731575d593bae66aa91b29f49b6e5336&amp;scene=21#wechat_redirect" data-linktype="2"><span>SCS【1】今天开启单细胞之旅，述说单细胞测序的前世今生</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247488337&amp;idx=1&amp;sn=5142015a512f03a29b7160fb5b9412d1&amp;scene=21#wechat_redirect" data-linktype="2"><span>SCS【2】单细胞转录组 之 cellranger</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247488386&amp;idx=1&amp;sn=64907558954c43f19e9e41c59508fe94&amp;scene=21#wechat_redirect" data-linktype="2"><span>SCS【3】单细胞转录组数据 GEO 下载及读取</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247488454&amp;idx=1&amp;sn=eada05aadb85098fad116bd0211f0f76&amp;scene=21#wechat_redirect" data-linktype="2"><span>SCS【4】单细胞转录组数据可视化分析 (Seurat 4.0)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247488706&amp;idx=1&amp;sn=042eb3b7cd15844eb40db9cc32ccc63d&amp;scene=21#wechat_redirect" data-linktype="2"><span>SCS【5】单细胞转录组数据可视化分析 (scater)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247488842&amp;idx=1&amp;sn=1819425966c31d320f12062344687c44&amp;scene=21#wechat_redirect" data-linktype="2"><span>SCS【6】单细胞转录组之细胞类型自动注释 (SingleR)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247488893&amp;idx=1&amp;sn=bc0e4eb66eb8ee42f9cc62d22dbfd726&amp;scene=21#wechat_redirect" data-linktype="2"><span>SCS【7】单细胞转录组之轨迹分析 (Monocle 3) 聚类、分类和计数细胞</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247488963&amp;idx=1&amp;sn=ab7dd51c79234a1349d24ed4a8492d2a&amp;scene=21#wechat_redirect" data-linktype="2"><span>SCS【8】单细胞转录组之筛选标记基因 (Monocle 3)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247488996&amp;idx=1&amp;sn=281c4b3d1d1790cb44f0fec6bbdd9140&amp;scene=21#wechat_redirect" data-linktype="2"><span>SCS【9】单细胞转录组之构建细胞轨迹 (Monocle 3)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247489048&amp;idx=1&amp;sn=4332e67084d304d6c894cfbcc9daf6e5&amp;scene=21#wechat_redirect" data-linktype="2"><span>SCS【10】单细胞转录组之差异表达分析 (Monocle 3)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247489097&amp;idx=1&amp;sn=51585f2d1c7debf7c1597fde7bad7fdb&amp;scene=21#wechat_redirect" data-linktype="2"><span>SCS【11】单细胞ATAC-seq 可视化分析 (Cicero)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247489127&amp;idx=1&amp;sn=af3fadbad021280eeeac624e554ac44a&amp;scene=21#wechat_redirect" data-linktype="2"><span>SCS【12】单细胞转录组之评估不同单细胞亚群的分化潜能 (Cytotrace)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247489160&amp;idx=1&amp;sn=4b81c64d758a060703ed61abc636d9f2&amp;scene=21#wechat_redirect" data-linktype="2"><span>SCS【13】单细胞转录组之识别细胞对“基因集”的响应 (AUCell)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247489909&amp;idx=1&amp;sn=9f81ee7b73dd3b69497a65d88ca61bca&amp;scene=21#wechat_redirect" data-linktype="2"><span>SCS【14】单细胞调节网络推理和聚类 (SCENIC)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247489944&amp;idx=1&amp;sn=b501c85a5cd7a15d17d42119c69e4bed&amp;scene=21#wechat_redirect" data-linktype="2"><span>SCS【15】细胞交互：受体-配体及其相互作用的细胞通讯数据库 (CellPhoneDB)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247489961&amp;idx=1&amp;sn=83b66b85fb3e99dc7e5bd4c74fd22708&amp;scene=21#wechat_redirect" data-linktype="2"><span>SCS【16】从肿瘤单细胞RNA-Seq数据中推断拷贝数变化 (inferCNV)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247489995&amp;idx=1&amp;sn=2d261847f0faa2ea669b5225fa8f7460&amp;scene=21#wechat_redirect" data-linktype="2"><span>SCS【17】从单细胞转录组推断肿瘤的CNV和亚克隆 (copyKAT)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247490488&amp;idx=1&amp;sn=152a4d2330cfb485eabb4df9f42eba48&amp;scene=21#wechat_redirect" data-linktype="2"><span>SCS【18】细胞交互：受体-配体及其相互作用的细胞通讯数据库 (iTALK)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247490892&amp;idx=1&amp;sn=3a4f7258ae29a2e3dadb45f24029e141&amp;scene=21#wechat_redirect" data-linktype="2"><span>SCS【19】单细胞自动注释细胞类型 (Symphony)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247490966&amp;idx=1&amp;sn=7724d5791bbdf0a0afc645c9783c050b&amp;scene=21#wechat_redirect" data-linktype="2"><span>SCS【20】单细胞数据估计组织中细胞类型(Music)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247491050&amp;idx=1&amp;sn=2d72daf5677f4d3df7e80f23042c69af&amp;scene=21#wechat_redirect" data-linktype="2"><span>SCS【21】单细胞空间转录组可视化 (Seurat V5)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247491108&amp;idx=1&amp;sn=862ef5bbbe44df508a3cb0ebb42824f6&amp;scene=21#wechat_redirect" data-linktype="2"><span>SCS【22】单细胞转录组之 RNA 速度估计 (Velocyto.R)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247491180&amp;idx=1&amp;sn=8617df0886c987261f4e76d7892082b8&amp;scene=21#wechat_redirect" data-linktype="2"><span>SCS【23】单细胞转录组之数据整合 (Harmony)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247491231&amp;idx=1&amp;sn=39097eecc510ea8ef3de3ac9206f0265&amp;scene=21#wechat_redirect" data-linktype="2"><span>SCS【24】单细胞数据量化代谢的计算方法 (scMetabolism)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247491342&amp;idx=1&amp;sn=7888bf2aa41e2e71a440488df16ccb74&amp;scene=21#wechat_redirect" data-linktype="2"><span>SCS【25】单细胞细胞间通信第一部分细胞通讯可视化（CellChat）</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247491379&amp;idx=1&amp;sn=e65ec8f4283432b3fcf537d63895be75&amp;scene=21#wechat_redirect" data-linktype="2"><span>SCS【26】单细胞细胞间通信第二部分通信网络的系统分析（CellChat）</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247491451&amp;idx=1&amp;sn=a3ad6c2d65e518cad2e9083776396f06&amp;scene=21#wechat_redirect" data-linktype="2"><span>SCS【27】单细胞转录组之识别标记基因 （scran）</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247491482&amp;idx=1&amp;sn=202c8d10779b06c336d1b4efb129ce09&amp;scene=21#wechat_redirect" data-linktype="2"><span>SCS【28】单细胞转录组加权基因共表达网络分析(hdWGCNA)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247491547&amp;idx=1&amp;sn=6754f7cda83581233f10be200d1862ac&amp;scene=21#wechat_redirect" data-linktype="2"><span>SCS【29】单细胞基因富集分析 (singleseqgset)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247491836&amp;idx=1&amp;sn=130f6f5ff6c85b260e0d2157acec2431&amp;scene=21#wechat_redirect" data-linktype="2"><span>SCS【30】单细胞空间转录组学数据库(STOmics DB)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247491897&amp;idx=1&amp;sn=a628ebe9838c7d46f6de670ad7e4d5da&amp;scene=21#wechat_redirect" data-linktype="2"><span>SCS【31】减少障碍，加速单细胞研究数据库(Single Cell PORTAL)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247492021&amp;idx=1&amp;sn=ead10ddf6c9874f08acd0821f84a4091&amp;scene=21#wechat_redirect" data-linktype="2"><span>SCS【32】基于scRNA-seq数据中推断单细胞的eQTLs (eQTLsingle)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247492050&amp;idx=1&amp;sn=0b8e408bbdfd0cdd3fa0c75a5d1e0fd2&amp;scene=21#wechat_redirect" data-linktype="2"><span>SCS【33】单细胞转录之全自动超快速的细胞类型鉴定 (ScType)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247492121&amp;idx=1&amp;sn=8db073cf61ec9825bac5fa1e2f13ad2a&amp;scene=21#wechat_redirect" data-linktype="2"><span>SCS【34】单细胞/T细胞/抗体免疫库数据分析(immunarch)</span></a></p><h1><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247492284&amp;idx=1&amp;sn=f276c3aa04775134cf7fd4a36f1ed71e&amp;chksm=fb7a1877cc0d9161afdd7e7c233e773e1e5b0179e3af146538c7648d4d4ae9d334401c702365&amp;scene=21#wechat_redirect" textvalue="SCS【35】单细胞转录组之去除双细胞 (DoubletFinder)" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2"><span>SCS【35】单细胞转录组之去除双细胞 (DoubletFinder)</span></a></h1><p data-tool="mdnice编辑器"><br></p></section><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><p><span><span>利用这个软件包实现了快速计算单细胞差异丰度</span></span><span><span>，有需求的老师可以联系桓峰基因，关注桓峰基因公众号，轻松学生</span></span><span>信，高效发文章！</span></p><p><span></span><br></p><p><span>号外号外，桓峰基因单细胞生信分析免费培训课程即将开始，快来报名吧！</span></p><p><span>桓峰基因，铸造成功的您！</span></p><p><span>未来桓峰基因公众号将不间断的推出单细胞系列生信分析教程，</span></p><p><span>敬请期待！！</span></p></section><p><br></p><p><span><span>桓峰基因官网正式上线，请大家多多关注，还有很多不足之处，大家多多指正！</span></span></p><p><span><span>http://www.kyohogene.com/</span></span></p><p><span><span><span>桓峰基因和投必得合作，文章润色优惠85折，需要文章润色的老师可以直接到网站输入领取桓峰基因专属优惠券码：KYOHOGENE，然后上传，付款时选择桓峰基因优惠券即可享受85折优惠哦！https://www.topeditsci.com/</span></span></span></p><p><span><img data-galleryid="" data-imgfileid="100009087" data-ratio="0.5657407407407408" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwDj3ich1uOSggdibWHw5pgkMx26IaaiaQoKJbU2EaQI2CYh9n2stiaMicbs1RyInFfqO6qWwpDxsFfl8w/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwDj3ich1uOSggdibWHw5pgkMx26IaaiaQoKJbU2EaQI2CYh9n2stiaMicbs1RyInFfqO6qWwpDxsFfl8w/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></span></p></section><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/SzUd8fWftoVVmmEnN8xfhQ",target="_blank" rel="noopener noreferrer">原文链接</a>
