---
title: "全网最全！哈佛生信女神刘小乐团队研究成果汇总"
date: 2024-05-22T01:17:43Z
draft: ["false"]
tags: [
  "fetched",
  "生信人"
]
categories: ["Acdemic"]
---
全网最全！哈佛生信女神刘小乐团队研究成果汇总 by 生信人
------
<div><p><img data-galleryid="" data-imgfileid="503745034" data-ratio="0.3274582560296846" data-src="https://mmbiz.qpic.cn/mmbiz_gif/N3X4LBoaQjU9abk37Q6pRd9QUqmbgwBYMLh9SImibmH2QJhYav5XWHhgxw9GuyLIXqOmyasZ5njzHdvEreJazxg/640?wx_fmt=gif&amp;from=appmsg" data-type="gif" data-w="1078" src="https://mmbiz.qpic.cn/mmbiz_gif/N3X4LBoaQjU9abk37Q6pRd9QUqmbgwBYMLh9SImibmH2QJhYav5XWHhgxw9GuyLIXqOmyasZ5njzHdvEreJazxg/640?wx_fmt=gif&amp;from=appmsg"></p><section><span><strong><span>1.背景介绍</span></strong></span><span><strong><span><p></p></span></strong></span></section><p><span><span>说到</span><span><span>刘小乐（</span><span>X. Shirley Liu</span><span>）教授</span></span></span><span>，相信做生信的小伙伴们没有不认识的，刘小乐教授团队不仅开发出了多种经典的生信分析算法，还在自己的</span><span><span><span>YouTube</span><span>主页分享了广受许多学习生信的小伙伴好评的经典课程（</span></span><span>https://www.youtube.com/channel/UCiqOwwdgfak4GclHh5_CSpQ/playlists</span><span><span>）。作为一名计算生物学家，刘小乐教授运用她在癌症表观遗传学和癌症免疫学方面的专业知识，在转化癌症研究的算法开发和数据集成建模领域作出了杰出的贡献。更令人敬佩的是，她在</span><span>2022</span><span>年离开</span><span>Dana-Farber</span><span>癌症研究所，成为</span><span>GV20 Therapeutics </span><span>的首席执行官，但她并没有彻底离开学术界，此后仍然以第一或共同通讯的身份发表了多篇高水平论文，甚至在前不久</span></span></span><span><strong><span><span>4</span><span>月</span><span>2</span></span><span>3</span><span>日</span></strong></span><span><span><span>还以共同通讯的身份在</span><span><strong><span>Cell</span></strong></span><span>上发表了一篇<span>关于先天免疫检查点和肿瘤免疫治疗靶点的研究论文</span>。</span></span><span><p></p></span></span></p><p><span><span><span></span></span></span></p><p><span><span><span>刘小乐教授团队的研究方向大致可以分为四类：癌症表观遗传学、</span><span>CRISPR</span><span>筛选、癌症免疫学和单细胞技术。今天小编就借此机会从这四类来系统总结一下截至目前刘小乐教授团队的重要研究发现。在这需要说明的一点是，由于刘小乐教授挂名的研究论文实在是太多了，其中很多都是合作课题的研究成果。为了方便大家快速了解刘小乐教授研究的精华，本次小编只重点介绍刘小乐教授作为末尾通讯，以开发算法或网站为核心的</span></span><span>20</span><span>项研究成果。</span><span><p></p></span></span></p><p><br></p><p><span><strong><span>2.癌症表观遗传学</span></strong></span><span><strong><span><p></p></span></strong></span></p><p><span><span><span>在这个研究方向下，刘小乐教授团队一直在开发算法来促进表观基因组数据的分析，并使用综合建模方法来研究肿瘤发生和进展的基因组转录和表观遗传基因调控机制。同时，团队也正在开发新方法，利用丰富的公共</span> <span>ChIP-seq </span><span>数据来推断调节表达的功能增强子，并了解表观遗传药物在癌症中的靶点和特异性。</span></span><span><p></p></span></span></p><p><br></p><p><span><span><span>刘小乐教授在这个领域的探索最早可以追溯到</span><span>2</span></span><span>0</span><span><span>多年前她还是个学生的时候，当时染色质免疫沉淀</span><span>-cDNA</span><span>微阵列杂交 </span><span>(ChIP-array) </span><span>还是研究全基因组蛋白质</span><span>-DNA</span><span>相互作用和转录调控的流行方法，她开发了一种名为</span><span>MDscan</span><span>的算法，该方法结合了两种广泛采用的主题搜索策略（单词枚举和位置特定权重矩阵更新）的优点，不仅可用于在 </span><span>ChIP </span><span>阵列实验中查找</span><span>DNA</span><span>基序，还可在其他实验中查找</span><span>DNA</span><span>基序。此后，刘小乐教授不断开发和优化算法，和哈佛大学</span><span>Dana-Farber</span><span>癌症研究所的</span></span><span>Clifford A. Meyer, Myles Brown</span><span>，李蔚等大牛开展了深入的长期合作，在这个领域内产出不断。</span><span><p></p></span></span></p><p><span><span><span><br></span></span></span></p><p><span><span><span>最著名的</span><span>MACS</span><span>（</span></span><span>Model-based Analysis of ChIP-Seq data</span><span><span>）算法就是刘小乐教授和李蔚教授在</span><span>2</span></span><span>008</span><span><span>年共同开发的，</span><span>MACS</span><span>根据经验对</span><span>ChIP-Seq</span><span>标签的位移大小进行建模，并使用它来提高预测结合位点的空间分辨率。</span><span>MACS</span><span>还使用动态泊松分布来有效捕获基因组中的局部偏差，从而实现更稳健的预测。该算法自提出以来引用量已超过</span><span>1</span><span>万，是目前最主流的用于分析</span><span>ChIP</span></span><span>-</span><span><span>seq</span><span>、</span><span>CUT</span></span><span>&amp;</span><span><span>Tag</span><span>等数据的</span><span>peak</span></span><span> </span><span><span>calling</span><span>的工具。该算法也一直在更新维护，从</span><span>2</span></span><span>008</span><span><span>年的</span><span>1</span></span><span>.0</span><span><span>版本，到</span><span>2</span></span><span>011</span><span><span>年的</span><span>2</span></span><span>.0</span><span><span>版本，截至目前已更新至最新版本</span><span>3</span></span><span>.0</span><span>。</span><span><p></p></span></span></p><p><span><p></p></span></p><section><img data-galleryid="" data-imgfileid="503746058" data-ratio="1.4861367837338262" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/N3X4LBoaQjU8UrlCag7mJxhtImPmUaJbY20T3xxDicUSBEttibDiaRXCgnHYP0OciacVBvGmQMynfNzLoImAYy0vrQ/640?wx_fmt=jpeg&amp;from=appmsg" data-type="jpeg" data-w="541" src="https://mmbiz.qpic.cn/mmbiz_jpg/N3X4LBoaQjU8UrlCag7mJxhtImPmUaJbY20T3xxDicUSBEttibDiaRXCgnHYP0OciacVBvGmQMynfNzLoImAYy0vrQ/640?wx_fmt=jpeg&amp;from=appmsg"></section><section><span>图<span>1</span></span><span>. </span><span><span>MACS</span>工作流程</span></section><p><br></p><p><span><span><span>另一个从</span><span>ChIP</span><span>芯片和</span><span>ChIP-Seq</span><span>数据中表征全基因组蛋白质</span><span>-DNA</span><span>相互作用模式的名为</span><span>CEAS</span><span>（</span></span><span>cis-regulatory element annotation system</span><span><span>）的算法随后被提出，该算法能提供重要基因组区域（例如单个染色体、启动子、基因体或外显子）的</span><span>ChIP </span><span>富集汇总统计数据，并推断最有可能受到调控的基因正在研究的结合因子。</span><span>CEAS</span><span>还能够可视化特定基因组区域的平均</span><span>ChIP</span><span>富集信号，特别是允许观察连续和广泛的</span><span>ChIP</span><span>富集，这种富集可能过于微弱而无法从</span><span>ChIP</span><span>峰中检测到。</span></span><span><p></p></span></span></p><p><span> </span></p><section><img data-galleryid="" data-imgfileid="503746059" data-ratio="1.0388349514563107" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/N3X4LBoaQjU8UrlCag7mJxhtImPmUaJbuE99seDjG07iaOaWnDmkvk9ibMKouzmyElz5tUrSZhhv0GK1ZgP9rGGA/640?wx_fmt=jpeg&amp;from=appmsg" data-type="jpeg" data-w="206" src="https://mmbiz.qpic.cn/mmbiz_jpg/N3X4LBoaQjU8UrlCag7mJxhtImPmUaJbuE99seDjG07iaOaWnDmkvk9ibMKouzmyElz5tUrSZhhv0GK1ZgP9rGGA/640?wx_fmt=jpeg&amp;from=appmsg"></section><section><span>图2.</span><span> C</span><span>EAS的输入、模块和输出的流程图</span></section><p><span><span><span><br></span></span></span></p><p><span><span><span>随着来自</span><span>ChIP</span><span>芯片和</span><span>ChIP-seq</span><span>的数据量不断增加，对标准、集成和可重复的生物信息学数据分析平台提出了挑战。刘小乐团队开发了一个名为</span><span>Cistrome</span><span>的网页版应用程序，</span></span><span><span>目标是提供一个基因组顺式作用元件分析的综合性数据库，通过收集来自</span><span>GEO,ENCODE</span><span>等公共数据库中的</span><span>ChIP-seq, DNase</span></span><span>-</span><span>seq, ATAC</span><span>-</span><span><span>seq</span><span>等原始数据，采用统一的分析方法，用</span><span>bwa</span><span>比对参考基因组，然后采用</span><span>MACS2</span><span>进行</span><span>peak calling, </span><span>将分析的结果加以整合，做成了在线数据库。</span><span>Cistrome</span><span>可在 </span><span>http://cistrom.org/ap/ </span><span>上获取。</span></span></span><img data-galleryid="" data-imgfileid="503746060" data-ratio="0.5787037037037037" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/N3X4LBoaQjU8UrlCag7mJxhtImPmUaJbkam9BhKgbG7jkFP9TNvf7w5xb5ICRohST7jmpFDRpJO6ao3UydowkQ/640?wx_fmt=png&amp;from=appmsg" data-type="jpeg" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/N3X4LBoaQjU8UrlCag7mJxhtImPmUaJbkam9BhKgbG7jkFP9TNvf7w5xb5ICRohST7jmpFDRpJO6ao3UydowkQ/640?wx_fmt=png&amp;from=appmsg"></p><p><span><span>图3. Cistrome分析平台内的工作流程</span></span><span><span><p></p></span></span></p><p><span><span><br></span></span></p><p><span><span>此后，刘小乐团队又紧接着开发了</span><span>Cistrome Data Browser</span><span>，</span><span>Cistrome-GO</span><span>，</span><span>CistromeDB Toolkit</span><span>等网页版分析工具</span><span>，做表观遗传的小伙伴绝对不容错过。简单介绍一下，</span><span>Cistrome Data Browser</span><span><span>主要用于展示所收集和整理的公共可获取的人和小鼠</span><span>ChIP-seq</span><span>、</span><span>DNase-seq</span><span>和</span><span>ATAC-seq</span><span>数据；</span><span>Cistrome-GO</span><span>可以做</span><span>ChIP</span><span>数据集靶基因的功能富集分析；</span></span><span>CistromeDB Toolkit</span><span>主要基于所收集数据的分析结果为用户提供搜索功能，以便用户快速使用这些数据来回答他们的研究问题。</span><span><p></p></span></span></p><p><span><span><span>Cistrome DB</span><span>数据库的作者们</span><span>2</span></span><span>020</span><span><span>年还在</span></span></span><span><span><strong><span>Quantitative Biology</span></strong></span></span><span><span><span>期刊上发表了题为“</span></span></span><span><span><strong><span>Cistrome Data Browser and Toolkit: analyzing human and mouse genomic data using compendia of ChIP-seq and chromatin accessibility data</span></strong></span></span><span><span><span>”的文章，文中介绍了</span><span>Cistrome DB</span><span>和</span><span>Toolkit</span><span>的构建方法和详细的使用教程，并具体地描述了一些特定的使用场景，譬如该如何筛选有意义的数据等，文中也表达了</span><span>Cistrome DB</span><span>将被不断地维护和更新，希望能为生物医学领域提供有用的数据资源和分析工具的愿景。</span></span><span><p></p></span></span></p><p><span><span><span>Cistrome DB</span><span>是人类和小鼠中统一处理的转录因子</span><span>TF</span></span><span> </span><span><span>ChIP-seq</span><span>（约</span><span>11,000</span><span>个）和染色质图谱（约</span><span>12,000</span><span>个组蛋白标记的</span><span>ChIP-seq</span><span>和</span><span>DNase-seq</span><span>）的集合，但是如何有效利用这些数据来推断调控从人类或小鼠差异或相关基因表达分析中得出的基因集的转录调节因子这一问题仍然有待解决。因此刘小乐团队开发了</span></span><span><span>Lisa (http://lisa.cistrome.org/) </span><span>来预测差异表达或共表达基因集的转录调节因子 </span><span>(TR)</span></span><span>。这项题为“</span><span><strong><span>Lisa: inferring transcriptional regulators through integrative modeling of public chromatin accessibility and ChIP-seq data</span></strong></span><span>”的研究成果于</span><span><strong><span>2020</span></strong></span><span>年发表在</span><span><strong><span>Genome Biology</span></strong></span></span><span>期刊上。</span><span><p></p></span></p><p><br></p><p><span><span><span>基于输入的基因集，</span><span>Lisa</span><span>使用组蛋白标记</span><span>ChIP-seq</span><span>和</span><span>ATAC-seq</span><span>构建与这些基因调控相关的染色质模型。</span><span>Lisa </span><span>使用</span><span>TR ChIP-seq </span><span>峰或估算的</span><span>TR</span><span>结合位点，利用计算机删除技术探测染色质模型，以找到最相关的</span><span>TR</span><span>。</span></span><span><p></p></span></span></p><p><span><p></p></span></p><p><img data-galleryid="" data-imgfileid="503746061" data-ratio="1.437956204379562" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/N3X4LBoaQjU8UrlCag7mJxhtImPmUaJbpNibzHaXdZu1XFMVwThZTURd55F1ibkRjFrmmmvs1WKcjUZIv9GgAbsA/640?wx_fmt=png&amp;from=appmsg" data-type="jpeg" data-w="685" src="https://mmbiz.qpic.cn/mmbiz_png/N3X4LBoaQjU8UrlCag7mJxhtImPmUaJbpNibzHaXdZu1XFMVwThZTURd55F1ibkRjFrmmmvs1WKcjUZIv9GgAbsA/640?wx_fmt=png&amp;from=appmsg"><span>图4. Lisa算法的示意图</span></p><p><span><span><span>为了进一步揭示基因表达调控的分子机制，刘小乐团队于</span></span></span><span><span><strong><span>2013</span></strong></span></span><span><span>年在</span></span><span><span><strong><span>Nature Protocols</span></strong></span></span><span><span>期刊上发表了题为“</span></span><span><span><strong><span>Target analysis by integration of transcriptome and ChIP-seq data with BETA</span></strong></span></span><span><span><span>”的研究论文，开发了名为</span><span>BETA</span><span>（</span></span><span>Binding and expression target analysis</span><span><span>）的算法，用于整合</span><span>ChIP-seq</span><span>和</span><span>RNA-seq</span><span>数据。</span></span><span>BETA</span><span>具有三个功能：（</span><span>i</span><span>）预测该因子是否具有激活或抑制功能</span><span>;</span><span>（</span><span>ii</span><span>）推断因子的靶基因</span><span>;</span><span>（</span><span>iii</span><span>）确定因子及其合作者的主题，这可能调节因子的激活或抑制功能。</span><span>BETA</span><span>也可以在</span><span> http://cistrome.org/</span><span>上免费使用。</span></span><img data-galleryid="" data-imgfileid="503746062" data-ratio="0.6817518248175183" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/N3X4LBoaQjU8UrlCag7mJxhtImPmUaJbXcc7gVeF7jiaiaH40XmUlOn05vZsh8ZyBYXbbJ2f8CNQMZvbGucwTR7g/640?wx_fmt=png&amp;from=appmsg" data-type="jpeg" data-w="685" src="https://mmbiz.qpic.cn/mmbiz_png/N3X4LBoaQjU8UrlCag7mJxhtImPmUaJbXcc7gVeF7jiaiaH40XmUlOn05vZsh8ZyBYXbbJ2f8CNQMZvbGucwTR7g/640?wx_fmt=png&amp;from=appmsg"></p><p><span><span>图5. BETA工作流程图</span></span><span><span><p></p></span></span></p><p><span><span><br></span></span></p><p><span><strong><span>3.CRISPR筛选</span></strong></span><span><span><p></p></span></span></p><p><span><span><span>在这个研究方向下，刘小乐团队开发了用于全基因组</span> <span>CRISPR </span><span>筛选的设计</span><span>(SSC)</span><span>、分析 </span><span>(MAGeCK)</span><span>、可视化 </span><span>(VISPR) </span><span>和全面分析</span><span>pipeline</span><span>（</span></span><span>MAGeCKFlute</span><span><span>）的计算方法。他们利用这写些算法成功识别了乳腺癌和前列腺肿瘤进展和耐药性的关键基因。此外，他们还开发</span> <span>CRISPR </span><span>筛选平台，以了解增强子和长链非编码 </span><span>RNA </span><span>的功能，并识别癌症中的合成致死基因对，从而优化癌症精准医疗。</span></span><span><p></p></span></span></p><p><span><span><span><br></span></span></span></p><p><span><span><span>众所周知，使用</span> <span>CRISPR/Cas9 </span><span>介导的敲除或 </span><span>dCas9 </span><span>融合介导的抑制</span><span>/</span><span>激活 （</span><span>CRISPRi/a</span><span>） 进行全基因组功能筛选是发现表型相关基因功能的强大技术。刘小乐团队系统地评估了有助于基于</span><span>CRISPR</span><span>的筛选中单向导</span><span>RNA</span><span>（</span><span>sgRNA</span><span>）效率的</span><span>DNA</span><span>序列特征，开发了一种名为</span></span></span><span><span><strong><span>SSC（Spacer Scoring for CRISPR）</span></strong></span></span><span><span><span>的序列模型，用于预测</span><span>CRISPR/Cas9</span><span>敲除实验中的</span><span>sgRNA</span><span>效率，研究成果于</span></span><span><strong><span>2</span><span>015</span><span>年</span></strong></span><span>发表在</span></span><span><span><strong><span>Genome Research</span></strong></span></span><span><span><span>期刊上，题为“</span></span></span><span><span><strong><span>Sequence determinants of improved CRISPR sgRNA design</span></strong></span></span><span><span>”。</span></span><img data-galleryid="" data-imgfileid="503746063" data-ratio="0.76625" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/N3X4LBoaQjU8UrlCag7mJxhtImPmUaJbEeOeRzRicF2bEPIicNpIV9F1cgAnmQ3PDVmDR1NJDUE3sjqlYxNlDONA/640?wx_fmt=png&amp;from=appmsg" data-type="jpeg" data-w="800" src="https://mmbiz.qpic.cn/mmbiz_png/N3X4LBoaQjU8UrlCag7mJxhtImPmUaJbEeOeRzRicF2bEPIicNpIV9F1cgAnmQ3PDVmDR1NJDUE3sjqlYxNlDONA/640?wx_fmt=png&amp;from=appmsg"></p><p><span><span>图6. sgRNA选择和分类程序的示意图</span></span><span><span><p></p></span></span></p><p><br></p><p><span><span><span>在分析</span><span>CRISPR </span><span>筛选结果方面，刘小乐团队提出了名为</span></span></span><span><span><strong><span>MAGeCK(Model-based Analysis of Genome-wide CRISPR/Cas9 Knockout)</span></strong></span></span><span><span><span>的分析算法，于</span><span>2</span></span><span>014</span><span><span>年发表在</span><span>Genome</span></span><span> </span><span><span>Biology</span><span>期刊上，用于在基因组规模的</span><span>CRISPR/Cas9</span><span>敲除筛选中优先考虑单向导</span><span>RNA</span><span>、基因和通路。与当时其他方法相比，</span><span>MAGeCK</span><span>可同时识别阳性和阴性选择的基因，并在不同的实验条件下报告了可靠的结果。</span></span></span><img data-galleryid="" data-imgfileid="503746064" data-ratio="1.3897707231040564" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/N3X4LBoaQjU8UrlCag7mJxhtImPmUaJbSf8HBF8PzqHGMZALuTPVLhXhX3EiaFb5675q8jqMPJGQ5c9fsdZqgpw/640?wx_fmt=png&amp;from=appmsg" data-type="jpeg" data-w="567" src="https://mmbiz.qpic.cn/mmbiz_png/N3X4LBoaQjU8UrlCag7mJxhtImPmUaJbSf8HBF8PzqHGMZALuTPVLhXhX3EiaFb5675q8jqMPJGQ5c9fsdZqgpw/640?wx_fmt=png&amp;from=appmsg"></p><p><span><span>图7. MAGeCK 算法概述</span></span><span><span><p></p></span></span></p><p><br></p><p><span><span>随后，刘小乐团队又提出了</span></span><span><span><strong><span>MAGeCK-VISPR算法</span></strong></span></span><span><span><span>，这是一种用于</span><span>CRISPR </span><span>筛选的综合质量控制（</span><span>QC</span><span>）、分析和可视化工作流程，核心目的是收集不同级别的质量控制（</span><span>QC</span><span>）测量值。</span><span>MAGeCK-VISPR</span><span>定义了一组</span><span>QC</span><span>措施来评估实验的质量，并包括一个最大似然算法，用于在多种条件下同时调用必需基因。该算法使用广义线性模型对不同的效应进行反卷积，并采用期望最大化来迭代估计</span><span>sgRNA</span><span>敲除效率和基因本质性。</span><span>MAGeCK-VISPR</span><span>还包括</span><span>VISPR</span><span>，这是一个用于交互式可视化和探索</span><span>QC</span><span>和分析结果的框架。这项工作于</span></span><span><strong><span>2015</span><span>年</span></strong></span><span>发表在</span><span><strong><span>Genome</span><span> </span><span>Biology</span></strong></span><span>期刊上。</span></span><img data-galleryid="" data-imgfileid="503746065" data-ratio="0.8227848101265823" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/N3X4LBoaQjU8UrlCag7mJxhtImPmUaJbPBxEOvSQf5UR2kpYicibH319y08LAYOhHAQsgpp9ibeoBXDWzxJIFf1Ew/640?wx_fmt=png&amp;from=appmsg" data-type="jpeg" data-w="474" src="https://mmbiz.qpic.cn/mmbiz_png/N3X4LBoaQjU8UrlCag7mJxhtImPmUaJbPBxEOvSQf5UR2kpYicibH319y08LAYOhHAQsgpp9ibeoBXDWzxJIFf1Ew/640?wx_fmt=png&amp;from=appmsg"></p><p><span><span>图8. MAGeCK-VISPR工作流程概述</span></span><span><span><p></p></span></span></p><p><span><span><br></span></span></p><p><span><span>2019</span><span>年，刘小乐团队提出了</span></span><span><span><span>MAGeCKFlute算法</span></span></span><span><span><span>，结合了</span><span>MAGeCK</span><span>和</span><span>MAGeCK-VISPR</span><span>算法，并整合了额外的下游分析功能。</span><span>MAGeCKFlute</span><span>与其他工具的区别在于其全面的</span><span>pipeline</span><span>，其中包含一系列用于分析</span><span>CRISPR</span><span>筛选数据的功能。这项题为“</span></span><span><strong><span>Integrative analysis of pooled CRISPR genetic screens using MAGeCKFlute</span></strong></span><span>”的研究成果发表在</span><span><strong><span>Nature Protocols</span></strong></span><span><span>，描述了如何使用</span><span>MAGeCKFlute</span><span>对</span><span>CRISPR</span><span>筛选进行质量控制（</span><span>QC</span><span>），归一化，批次效应去除，拷贝数偏倚校正，基因命中鉴定和下游功能富集分析。在运行</span><span>Linux</span><span>或</span><span>Mac OS</span><span>且支持</span><span>R</span><span>的台式计算机上完成整个</span><span>MAGeCKFlute pipeline</span><span>需要大约</span><span>3</span><span>小时。</span></span><span><p></p></span></span></p><p><span><p></p></span></p><p><img data-galleryid="" data-imgfileid="503746066" data-ratio="1.1328467153284671" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/N3X4LBoaQjU8UrlCag7mJxhtImPmUaJb3wAhdmGcttttxmfZIjtibAtDh8dlvWzibZ6cRQj6pLRkJQ1lbUK3Cq8A/640?wx_fmt=png&amp;from=appmsg" data-type="jpeg" data-w="685" src="https://mmbiz.qpic.cn/mmbiz_png/N3X4LBoaQjU8UrlCag7mJxhtImPmUaJb3wAhdmGcttttxmfZIjtibAtDh8dlvWzibZ6cRQj6pLRkJQ1lbUK3Cq8A/640?wx_fmt=png&amp;from=appmsg"><span>图</span><span>9. 使用MAGeCKFlute进行CRISPR-Cas9筛选分析的示意图</span></p><p><span><span><br></span></span></p><p><span><strong><span>4.癌症免疫学</span></strong></span><span><strong><span><p></p></span></strong></span></p><p><span><span><span>在这个研究方向下，刘小乐团队致力于开发新颖的计算算法，以系统地对不同免疫细胞类型的丰度进行解卷积。对来自</span><span>TCGA</span><span>的</span><span>1</span><span>万多个肿瘤样本应用反卷积方法，发现了与肿瘤浸润免疫细胞的广泛临床关联。</span></span></span></p><p><br></p><p><span><span><span>在癌症免疫这个领域内，刘小乐团队最早的研究可以追溯到</span></span></span><span><span><strong><span>2016年</span></strong></span></span><span><span><span>发表于</span></span></span><span><span><strong><span>Nature Genetics</span></strong></span></span><span><span><span>上的一篇题“</span></span></span><span><span><strong><span>Landscape of tumor-infiltrating T cell repertoire of human cancers</span></strong></span></span><span><span>”的研究论文，在这项工作中，刘小乐团队开发了一种名为</span></span><span><span><strong><span>TRUST（TCR repertoire utilities for solid tissue）</span></strong></span></span><span><span><span>的使用双端</span><span>RNA-seq</span><span>数据对</span><span>CDR3</span><span>区域进行从头组装的新计算方法，并将其应用于癌症基因组图谱（</span><span>TCGA</span><span>）的</span><span>9142</span><span>个样本。与之前基于</span><span>RNA-seq</span><span>的分析相比，作者组装了更清晰的</span><span>CDR3</span><span>序列，从而能够对肿瘤微环境的</span><span>TCR</span><span>库进行更深入的分析。研究观察到肿瘤和宿主免疫系统之间有趣的相互作用，并确定了可能对多种免疫疗法有用的潜在治疗靶点。</span></span></span><img data-galleryid="" data-imgfileid="503746067" data-ratio="0.9586776859504132" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/N3X4LBoaQjU8UrlCag7mJxhtImPmUaJbsYzpZBgQH8mBM41zIAXFbPic5iahsAickBicuCRGm8nicDDXiaZ8SjWDFo4Q/640?wx_fmt=jpeg&amp;from=appmsg" data-type="jpeg" data-w="726" src="https://mmbiz.qpic.cn/mmbiz_jpg/N3X4LBoaQjU8UrlCag7mJxhtImPmUaJbsYzpZBgQH8mBM41zIAXFbPic5iahsAickBicuCRGm8nicDDXiaZ8SjWDFo4Q/640?wx_fmt=jpeg&amp;from=appmsg"></p><p><span><span>图10. 利用TRUST算法从RNA-seq数据组装CDR3序列的工作流程</span></span><span><span><p></p></span></span></p><p><br></p><p><span><span><span>TRUST</span><span>算法用于直接从组织或血液</span><span>RNA-seq</span><span>数据中从头组装免疫受体库，随着</span><span>scRNA-seq</span><span>技术的进步，刘小乐团队将</span><span>TRUST</span><span>算法重新设计为</span></span></span><span><span><span>TRUST4</span></span></span><span><span><span>，</span></span></span><span><span><span>其功能显着增强</span></span></span><span><span>，除了在组装更长（甚至全长）受体库时更快、更灵敏，</span><span>TRUST4</span><span>还可以从</span><span>scRNA-seq</span><span>数据调用库序列，无需</span><span>V(D)J</span><span>富集，并且与</span><span> SMART-seq</span><span>和</span><span> 5' 10x Genomics</span><span>平台兼容。这项题为“</span></span><span><span><strong><span>TRUST4: immune repertoire reconstruction from bulk and single-cell RNA-seq data</span></strong></span></span><span><span><span>”的工作于</span></span></span><span><span><strong><span>2021年</span></strong></span></span><span><span><span>发表在</span></span></span><span><span><strong><span>Nature Methods</span></strong></span></span><span><span><span>期刊上。</span></span><span>TRUST4</span><span><span>主要是用来分析实体组织，包括肿瘤组织的</span><span>RNA</span><span>测序的数据的</span><span>TCR</span><span>和</span><span>BCR</span><span>序列。</span></span></span><img data-galleryid="" data-imgfileid="503746068" data-ratio="0.8189781021897811" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/N3X4LBoaQjU8UrlCag7mJxhtImPmUaJbFe0sLpYUibXhwbIias3ghq4zHLeMgjdUQibJ5jUoOvSmW3jO7raDqQibiag/640?wx_fmt=png&amp;from=appmsg" data-type="jpeg" data-w="685" src="https://mmbiz.qpic.cn/mmbiz_png/N3X4LBoaQjU8UrlCag7mJxhtImPmUaJbFe0sLpYUibXhwbIias3ghq4zHLeMgjdUQibJ5jUoOvSmW3jO7raDqQibiag/640?wx_fmt=png&amp;from=appmsg"></p><p><span><span>图11. TRUST4在scRNA-seq数据上的性能</span></span><span><span><p></p></span></span></p><p><span><span><span><br></span></span></span></p><p><span><span><span>除了</span><span>TRUST</span><span>算法，刘小乐团队在</span></span></span><span><span><strong><span>2016年</span></strong></span></span><span><span><span>提出了一种肿瘤免疫细胞解卷积的新方法，来研究肿瘤浸润免疫细胞及其与癌细胞的相互作用，并分析了</span><span>TCGA</span><span>中</span><span>23</span><span>种癌症类型中六种免疫浸润细胞的丰度。研究成果题为“</span></span></span><span><span><strong><span>Comprehensive analyses of tumor immunity: implications for cancer immunotherapy</span></strong></span></span><span><span><span>”发表在</span></span></span><span><span><strong><span>Genome Biology</span></strong></span></span><span><span><span>上，作者通过蒙特卡罗模拟、基于</span><span>DNA</span><span>甲基化的推论的正交估计以及病理评估对通过计算推断的进行了验证，</span>并构建了一个<span>用户交互式网站</span><span>TIMER</span><span>（</span></span><span>Tumor Immune Estimation Resource</span><span>），将估计的免疫细胞丰度以及相关发现作为公共资源，供生物医学研究人员解决癌症免疫学中更有趣的问题。</span></span><img data-galleryid="" data-imgfileid="503746069" data-ratio="1.3686067019400352" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/N3X4LBoaQjU8UrlCag7mJxhtImPmUaJbkKvLZn077lSchcOfOqfdn4HgLAgvzOXMr6zVVC8ib9s9aIW3fxuCC5A/640?wx_fmt=png&amp;from=appmsg" data-type="jpeg" data-w="567" src="https://mmbiz.qpic.cn/mmbiz_png/N3X4LBoaQjU8UrlCag7mJxhtImPmUaJbkKvLZn077lSchcOfOqfdn4HgLAgvzOXMr6zVVC8ib9s9aIW3fxuCC5A/640?wx_fmt=png&amp;from=appmsg"></p><p><span><span>图12. 估计肿瘤浸润免疫细胞丰度的计算方法</span></span><span><span><p></p></span></span></p><p><span><span><span><br></span></span></span></p><p><span><span>随后，刘小乐团队进一步将</span><span>TIMER</span><span>优化，于</span></span><span><strong><span>2017</span><span>年</span></strong></span><span><span>在</span><span><strong><span>Cancer Research</span></strong></span><span>期刊上发表了一篇题为“</span><span><strong><span>TIMER: A Web Server for Comprehensive Analysis of Tumor-Infiltrating Immune Cells</span></strong></span><span><span>”的论文，全面地研究了肿瘤免疫相互作用的分子特征。通过反卷积算法，作者针对</span><span>32</span><span>种癌症类型的</span><span>10897</span><span>个肿瘤预先计算了</span><span>6</span><span>个肿瘤浸润免疫子集（</span><span>B</span><span>细胞、</span><span>CD4</span></span><sup><span>+</span></sup><span> <span>T</span><span>细胞、</span><span>CD8</span></span><sup><span>+</span></sup><span> <span>T</span><span>细胞、巨噬细胞、中性粒细胞、和树突状细胞）的水平。</span><span>TIMER</span><span>提供</span><span>6</span><span>个主要分析模块，允许用户交互式探索免疫浸润与多种因素之间的关联，包括基因表达、临床结果、体细胞突变和体细胞拷贝数改变。</span></span></span><img data-galleryid="" data-imgfileid="503746070" data-ratio="0.9442307692307692" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/N3X4LBoaQjU8UrlCag7mJxhtImPmUaJbtgZN9iaTvh42Pllh6dB3zpcCbzA0lvftGliaATFJ0y0RK8OPFOtWlunQ/640?wx_fmt=png&amp;from=appmsg" data-type="jpeg" data-w="520" src="https://mmbiz.qpic.cn/mmbiz_png/N3X4LBoaQjU8UrlCag7mJxhtImPmUaJbtgZN9iaTvh42Pllh6dB3zpcCbzA0lvftGliaATFJ0y0RK8OPFOtWlunQ/640?wx_fmt=png&amp;from=appmsg"></p><p><span><span>图13. 网站上TIMER模块的概述</span></span><span><span><p></p></span></span></p><p><span><span><span><br></span></span></span></p><p><span><span><span>TIMER</span><span>是当时第一种允许用户对肿瘤免疫学、临床和基因组数据进行综合分析的方法。作者承诺对</span><span>TIMER</span><span>网络服务器的开发和维护进行至少</span><span>5</span><span>年，实际上，在</span><span>2</span></span><span>020</span><span>年，为了降低分析复杂的肿瘤免疫相互作用的障碍，刘小乐团队进一步开发出了</span></span><span><span><span>更强大的TIMER 2.0，不只使用一种算法，而是使用六种状态为癌症基因组图谱(TCGA)或用户提供的肿瘤概况提供更可靠的免疫浸润水平估计。</span></span></span><span><span>这项题为“</span></span><span><span><strong><span>TIMER 2.0 for analysis of tumor-infiltrating immune cells</span></strong></span></span><span><span>”的工作发表在</span></span><span><span><strong><span>Nucleic Acids Research</span></strong></span></span><span><span>期刊上。</span><span><p></p></span></span></p><p><span><span><br></span></span></p><p><span><span>TIMER </span><span><span>2.0</span><span>提供了四个模块用于研究免疫浸润与遗传或临床特征之间的关联，以及四个模块用于探索 </span><span>TCGA </span><span>队列中与癌症相关的关联。每个模块都可以生成功能热图表，使用户能够轻松地同时识别多种癌症类型的显着关联。总体而言，</span><span>TIMER</span></span><span> </span><span><span>2.0</span><span>网络服务器提供了肿瘤浸润免疫细胞的全面分析和可视化功能，受到了更多癌症免疫学研究界的喜爱和赞赏，也被被大家广泛引用。</span></span><span><p></p></span></span></p><p><br></p><p><span><span><span>除了研究免疫浸润，刘小乐团队在</span><span>2</span></span><span>018</span><span>年开发出了一种名为</span></span><span><span><span>TIDE（Tumor Immune Dysfunction and Exclusion）的计算框架，以识别肿瘤免疫逃逸背后的因素，并对肿瘤样本基因表达谱中肿瘤免疫逃逸的可能性进行可靠的评估。</span></span></span><span><span>这项题为“</span></span><span><span><strong><span>Signatures of T cell dysfunction and exclusion predict cancer immunotherapy response</span></strong></span></span><span><span><span>”的工作发表于</span></span></span><span><span><strong><span>Nature Medicine</span></strong></span></span><span><span><span>期刊上。</span><span>TIDE</span><span>对</span><span>189</span><span>项人类癌症研究的数据进行了整合和建模，总共包含</span><span>33,197</span><span>个样本，作者假设并验证了模拟肿瘤免疫逃逸的准确基因特征可以作为预测</span><span>ICB</span><span>反应的可靠替代生物标志物。</span></span></span><img data-galleryid="" data-imgfileid="503746071" data-ratio="0.727007299270073" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/N3X4LBoaQjU8UrlCag7mJxhtImPmUaJb6bLh9oHMb4TAZudYVsRmwzcZYQviaUCgia8hZJ2QvGHrAFlSN1L89KbQ/640?wx_fmt=png&amp;from=appmsg" data-type="jpeg" data-w="685" src="https://mmbiz.qpic.cn/mmbiz_png/N3X4LBoaQjU8UrlCag7mJxhtImPmUaJb6bLh9oHMb4TAZudYVsRmwzcZYQviaUCgia8hZJ2QvGHrAFlSN1L89KbQ/640?wx_fmt=png&amp;from=appmsg"></p><p><span>图14. TIDE 特征预测 ICB 免疫治疗反应</span><br></p><p><br></p><p><span><span><span>此外，识别可靠的药物反应生物标志物是癌症研究中的一项重大挑战。刘小乐团队于</span></span></span><span><span><strong><span>2018年</span></strong></span></span><span><span>提出了</span></span><span><span><span>耐药性计算分析CARE（computational analysis of resistance）</span></span></span><span><span>，这是一种专注于靶向治疗的计算方法，用于从细胞系化合物筛选中推断药物功效的全基因组转录组特征。这项题为“</span></span><span><span><strong><span>Genome-Scale Signatures of Gene Interaction from Compound Screens Predict Clinical Efficacy of Targeted Cancer Therapies</span></strong></span></span><span><span><span>”的研究成果发表在</span></span></span><span><span><strong><span>Cell System</span></strong></span></span><span><span><span>期刊上。</span></span><span><p></p></span></span></p><p><span><span><br></span></span></p><p><span><span>CARE</span><span>输出基因组规模分数，以测量药物靶基因如何与其他基因相互作用，从而影响化合物筛选中的抑制剂功效。当使用临床研究的转录组数据进行评估时，</span><span>CARE </span><span><span>可以比其他计算方法和基因组学实验的特征更好地预测治疗结果。因此，</span><span>CARE </span><span>应该能够使用化合物筛选数据大规模推断靶向治疗的反应生物标志物和药物组合。</span></span></span><img data-galleryid="" data-imgfileid="503746072" data-ratio="1.0025510204081634" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/N3X4LBoaQjU8UrlCag7mJxhtImPmUaJbAqiadmRgA2mxyN6CHyncYmmLUT8T0cA5QuNXLkNxveePdJTkQSsWJqQ/640?wx_fmt=jpeg&amp;from=appmsg" data-type="jpeg" data-w="784" src="https://mmbiz.qpic.cn/mmbiz_jpg/N3X4LBoaQjU8UrlCag7mJxhtImPmUaJbAqiadmRgA2mxyN6CHyncYmmLUT8T0cA5QuNXLkNxveePdJTkQSsWJqQ/640?wx_fmt=jpeg&amp;from=appmsg"></p><p><span><span>图15. CARE用于预测靶向癌症治疗的临床疗效</span></span><span><span><p></p></span></span></p><p><br></p><p><span><strong><span>5.单细胞技术</span></strong></span><span><strong><span><p></p></span></strong></span></p><p><span><span><span>在这个研究方向下，刘小乐教授团队正在开发用于单细胞</span> <span>RNA-seq (scRNA-seq) </span><span>和</span><span>ATAC-seq (scATAC-seq) </span><span>数据综合分析的计算方法 </span><span>(MAESTRO)</span><span>。通过将公共转录因子（</span><span>TF</span><span>）</span><span>ChIP-seq </span><span>数据集集成到 </span><span>Cistrome DB </span><span>中，他们正在阐明肿瘤微环境的调控网络，并确定不同免疫细胞之间的关键调控因子。同时他们也在开发用于空间转录组数据分析的新颖计算算法。</span></span><span><p></p></span></span></p><p><br></p><p><span><span><span>刘小乐团队在</span><span>2</span></span><span>020</span><span>年提出了</span></span><span><span><span>MAESTRO（Model-based Analyses of Transcriptome and RegulOme），这是一个全面的开源计算工作流程 (http://github.com/liulab-dfci/MAESTRO)，用于处理来自多个平台的scRNA-seq和scATAC-seq数据。</span></span></span><span><span>这项题为“</span></span><span><span><strong><span>Integrative analyses of single-cell transcriptome and regulome using MAESTRO</span></strong></span></span><span><span><span>”的研究成果发表在</span></span></span><span><span><strong><span>Genome Biology</span></strong></span></span><span><span><span>期刊上。</span></span><span>MAESTRO</span><span><span>提供预处理、比对、质量控制、表达和染色质可及性定量、聚类、差异分析和注释等功能，在单细胞水平上对染色质可及性的基因调控潜力进行建模。此外，</span><span>MAESTRO</span><span>还支持使用预定义的细胞类型标记基因进行自动细胞类型注释，并从</span><span>scRNA-seq</span><span>差异基因和</span><span>scATAC-seq</span><span>差异</span><span>peak</span><span>中识别驱动调节因子。</span></span></span><img data-galleryid="" data-imgfileid="503746073" data-ratio="0.41605839416058393" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/N3X4LBoaQjU8UrlCag7mJxhtImPmUaJbWOR8TwxhO7K4uIpVgoVzZg9s8hlF39TpyWuZjmOCOUkk5CEiaWxu74g/640?wx_fmt=png&amp;from=appmsg" data-type="jpeg" data-w="685" src="https://mmbiz.qpic.cn/mmbiz_png/N3X4LBoaQjU8UrlCag7mJxhtImPmUaJbWOR8TwxhO7K4uIpVgoVzZg9s8hlF39TpyWuZjmOCOUkk5CEiaWxu74g/640?wx_fmt=png&amp;from=appmsg"></p><p><span><span>图16. MAESTRO工作流程概述</span></span><span><span><p></p></span></span></p><p><br></p><p><span><strong><span>6.说在最后</span></strong></span><span><strong><span><p></p></span></strong></span></p><p><span><span>看到这，我们已经把刘小乐教授近些年来的以开发算法为核心的主要研究成果都学习完了。本篇推文可以说是全网截止到目前对刘小乐教授科研成果进行的最系统、最全面的总结了。</span><span><p></p></span></span></p><p><br></p><p><span>其实除了科研成果，刘小乐教授也经常跟大家分享她对生物信息学研究的思考，她写的一篇题目为</span><span>《</span><span><span>生物信息学研究的几个阶段(Levels of Bioinformatics Research)</span><span>》</span></span><span>的博文也非常值得大家去学习。她在这篇博文中提到，一个生物信息学者的成长，从最初级的“为了建模而建模”，经历“分析自己实验室或合作者的未发表的数据来做出新的生物发现”，到“面向生物医学领域的新的高通量技术来开发计算方法和数据库”以及“有效的整合公共数据做出好的生物学发现”三个中间级别，最后进化至“对大联合体产生的海量数据提供关键的整合和建模”这一最高“<span>X</span>”级别</span><span><span><span>。刘小乐教授毫无疑问是最高“</span><span>X</span><span>”大神级别的人物，作为生物信息领域杰出的女科学家，她的故事不断激励着一众生信小白。同时，她带领团队开发出的一系列的用户交互式网站显示出了绝对的专业性，极大地促进了生物医学领域的发展。相信刘小乐教授进军</span><span>AI</span><span>制药领域也能取得卓越的成就，让我们一起期待未来她给我们带来更多的惊喜！</span></span><span><p></p></span></span></p><p><br></p><p><span><span><br></span></span></p><p><span><span>参考文献</span><span><p></p></span></span></p><p><span>1.Liu, X. Shirley, Douglas L. Brutlag, and Jun S. Liu. "An algorithm for finding protein–DNA binding sites with applications to chromatin-immunoprecipitation microarray experiments." Nature biotechnology 20.8 (2002): 835-839.<p></p></span></p><p><span>2.Zhang, Yong, et al. "Model-based analysis of ChIP-Seq (MACS)." Genome biology 9 (2008): 1-9.<p></p></span></p><p><span>3.Shin, Hyunjin, et al. "CEAS: cis-regulatory element annotation system." Bioinformatics 25.19 (2009): 2605-2606.<p></p></span></p><p><span>4.Liu, Tao, et al. "Cistrome: an integrative platform for transcriptional regulation studies." Genome biology 12 (2011): 1-10.<p></p></span></p><p><span>5.Zheng, Rongbin, et al. "Cistrome Data Browser: expanded datasets and new tools for gene regulatory analysis." Nucleic acids research 47.D1 (2019): D729-D735.<p></p></span></p><p><span>6.Li, Shaojuan, et al. "Cistrome-GO: a web server for functional enrichment analysis of transcription factor ChIP-seq peaks." Nucleic acids research 47.W1 (2019): W206-W211.<p></p></span></p><p><span>7.Wang, Su, et al. "Target analysis by integration of transcriptome and ChIP-seq data with BETA." Nature protocols 8.12 (2013): 2502-2515.<p></p></span></p><p><span>8.Xu, Han, et al. "Sequence determinants of improved CRISPR sgRNA design." Genome research 25.8 (2015): 1147-1157.<p></p></span></p><p><span>9.Li, Wei, et al. "MAGeCK enables robust identification of essential genes from genome-scale CRISPR/Cas9 knockout screens." Genome biology 15 (2014): 1-12.<p></p></span></p><p><span>10.Li, Wei, et al. "Quality control, modeling, and visualization of CRISPR screens with MAGeCK-VISPR." Genome biology 16 (2015): 1-13.<p></p></span></p><p><span>11.Wang, Binbin, et al. "Integrative analysis of pooled CRISPR genetic screens using MAGeCKFlute." Nature protocols 14.3 (2019): 756-780.<p></p></span></p><p><span>12.Li, Bo, et al. "Landscape of tumor-infiltrating T cell repertoire of human cancers." Nature genetics 48.7 (2016): 725-732.<p></p></span></p><p><span>13.Song, Li, et al. "TRUST4: immune repertoire reconstruction from bulk and single-cell RNA-seq data." Nature methods 18.6 (2021): 627-630.<p></p></span></p><p><span>14.Li, Bo, et al. "Comprehensive analyses of tumor immunity: implications for cancer immunotherapy." Genome biology 17 (2016): 1-16.<p></p></span></p><p><span>15.Li, Taiwen, et al. "TIMER: a web server for comprehensive analysis of tumor-infiltrating immune cells." Cancer research 77.21 (2017): e108-e110.<p></p></span></p><p><span>16.Li, Taiwen, et al. "TIMER2. 0 for analysis of tumor-infiltrating immune cells." Nucleic acids research 48.W1 (2020): W509-W514.<p></p></span></p><p><span>17.Jiang, Peng, et al. "Signatures of T cell dysfunction and exclusion predict cancer immunotherapy response." Nature medicine 24.10 (2018): 1550-1558.<p></p></span></p><p><span>18.Jiang, Peng, et al. "Genome-scale signatures of gene interaction from compound screens predict clinical efficacy of targeted cancer therapies." Cell systems 6.3 (2018): 343-354.<p></p></span></p><p><span>19.Wang, Chenfei, et al. "Integrative analyses of single-cell transcriptome and regulome using MAESTRO." Genome biology 21 (2020): 1-28.<p></p></span></p><p><span>20.Lynch, Allen W., et al. "MIRA: joint regulatory modeling of multimodal expression and chromatin accessibility in single cells." Nature methods 19.9 (2022): 1097-1108.<p></p></span></p><p><span><span>21.Zhang, Yi, et al. "MetaTiME integrates single-cell gene expression to characterize the meta-components of the tumor immune microenvironment." Nature communications 14.1 (2023): 2634.</span><span><p></p></span></span></p><p><br></p><section data-tools="135编辑器" data-id="125617" data-color="#00259f"><section><section><section><section data-role="paragraph"><p><span><strong>更多个性化方案设计</strong></span></p><p><span><strong>扫码领取</strong></span></p><p><img data-cropselx1="2" data-cropselx2="152" data-cropsely1="0" data-cropsely2="150" data-imgfileid="503746074" data-ratio="0.9728813559322034" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/N3X4LBoaQjXGEqnFwdrCTFhibYEjDbvpjjjl8m8yn2VBPG3u2s9RY41yZ4oFviavd3gp1VZOtvicygZxDCPUx2E2g/640?wx_fmt=jpeg&amp;from=appmsg" data-type="png" data-w="885" data-width="150px" src="https://mmbiz.qpic.cn/mmbiz_jpg/N3X4LBoaQjXGEqnFwdrCTFhibYEjDbvpjjjl8m8yn2VBPG3u2s9RY41yZ4oFviavd3gp1VZOtvicygZxDCPUx2E2g/640?wx_fmt=jpeg&amp;from=appmsg"></p><p><strong>适合临床医生的科研方案</strong></p></section></section></section></section></section><section data-role="paragraph"><p><br></p><p><br></p><section data-tools="135编辑器" data-id="136209" data-color="#00259f"><section><section><section><section><section><section><br></section><section><br></section></section><section><section><br></section><section><br></section></section></section><section><strong>最新文章汇总</strong><span>（持续更新ing）</span></section></section><section data-width="100%"><br></section></section><section><section><section><section><br></section></section><section data-width="100%"><section><span><strong data-brushtype="text"><strong><span>最新热点方向</span></strong></strong></span></section></section></section><section><section data-autoskip="1"><p><span>1、<a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzA5NjU5NjQ4MA==&amp;mid=2651229281&amp;idx=2&amp;sn=7f80405e8d2b40aa69f2658e9da9183d&amp;chksm=8b5f8b0ebc28021849b172672cd4fc89309f4997537ad283f9a47dff119e2af19c3c4f2e4f92&amp;scene=21#wechat_redirect" textvalue="今天开心，安利一个超好发版干湿结合思路" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">今天开心，安利一个超好发版干湿结合思路</a></span></p><p><span>2、</span><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzA5NjU5NjQ4MA==&amp;mid=2651229282&amp;idx=4&amp;sn=60bbb2c5ae38b362ff1a7e5a8e80bf3c&amp;chksm=8b5f8b0dbc28021bb15e7b948b1ec5d06a2c60388b00fd50368948e8d0ce5ffca2335ab75ff8&amp;scene=21#wechat_redirect" textvalue="掌握孟德尔随机化发20分不是梦" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">掌握孟德尔随机化发20分不是梦</a></p><p><span>3、</span><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzA5NjU5NjQ4MA==&amp;mid=2651229280&amp;idx=1&amp;sn=eca04f7905098b44749e53c5cebe0076&amp;chksm=8b5f8b0fbc280219f1ebcdde8ce58d0993965baaf01d05461882d08bbf07ae2eef8242d1f242&amp;scene=21#wechat_redirect" textvalue="研究中草药难发SCI？" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">研究中草药难发SCI？</a></p><p><span>4、</span><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzA5NjU5NjQ4MA==&amp;mid=2651229264&amp;idx=6&amp;sn=fe295c172ae11bff98867fb5e83264aa&amp;chksm=8b5f8b3fbc2802290a72d4c5d3bfe2c82e7d53d90c781b42cc0c02ea3c8acecc2de2e6b6551d&amp;scene=21#wechat_redirect" textvalue="公开数据纯分析发7+，内容不卷赢在选题" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">公开数据纯分析发7+，内容不卷赢在选题</a></p><p><span>5、</span><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzA5NjU5NjQ4MA==&amp;mid=2651229264&amp;idx=2&amp;sn=9154686fece39cb6a3d1072a06aac041&amp;chksm=8b5f8b3fbc280229fba49f3ef98e79ec84a770d14fdd2ac716c5ffd6077fab6bf5d6a8456ae5&amp;scene=21#wechat_redirect" textvalue="人手一篇的“淋巴转移”，摇身一变50+" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">人手一篇的“淋巴转移”，摇身一变50+</a></p></section></section></section><section><section><section><section><br></section></section><section data-width="100%"><section><span><strong data-brushtype="text"><strong><span>生信人课堂</span></strong></strong></span></section></section></section><section><section data-autoskip="1"><p><span>1、</span><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzA5NjU5NjQ4MA==&amp;mid=2651228970&amp;idx=7&amp;sn=1310dc5fcd181d48020b801f55e97863&amp;chksm=8b5f8c45bc28055319cd1bbd3a400fdc4385d4ad469f62b530356a24c711b44a61a4cfd918c0&amp;scene=21#wechat_redirect" textvalue="临床医生科研规划" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">临床医生科研规划</a></p><p><span>2、</span><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzA5NjU5NjQ4MA==&amp;mid=2651229044&amp;idx=3&amp;sn=91fc8d9347393a7bfce9c1a030728334&amp;chksm=8b5f8c1bbc28050da00f8ed43168999dd8791d13e699641cd4d75c23a2a809db0c98bb2bc5f3&amp;scene=21#wechat_redirect" textvalue="热点基因集预后实操课程" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">热点基因集预后实操课程</a></p><p><span>3、</span><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzA5NjU5NjQ4MA==&amp;mid=2651229206&amp;idx=3&amp;sn=d2900b86cd4bc6f452702b444fc278ae&amp;chksm=8b5f8b79bc28026faed6a6ed29566641aca7fc8fe4c6291bfe41901ec49dfc362939886ca6a0&amp;scene=21#wechat_redirect" textvalue="孟德尔随机与基因组" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">孟德尔随机与基因组</a></p><p><span>4、</span><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzA5NjU5NjQ4MA==&amp;mid=2651215654&amp;idx=3&amp;sn=02efc087cdc07e2da0526468a21bac9d&amp;chksm=8b5f5049bc28d95f77b9d128be607bfc76884483247f5f844827965f0b416aaa44fd6ef56533&amp;scene=21#wechat_redirect" textvalue="R语言入门" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">R语言入门</a></p><p><br></p><p><a href="http://mp.weixin.qq.com/s?__biz=MzA5NjU5NjQ4MA==&amp;mid=2651228874&amp;idx=6&amp;sn=e2c9030b187c654d7faf43f9ae16fa0e&amp;chksm=8b5f8da5bc2804b3a02f80ebe48b9d26cc95b395cc6a955b53b0fd97a996653dd4ed4cc37e08&amp;scene=21#wechat_redirect" data-linktype="2">课题设计</a><span> | <a href="http://mp.weixin.qq.com/s?__biz=MzA5NjU5NjQ4MA==&amp;mid=2651223612&amp;idx=6&amp;sn=6c1d6704031c8bf69ecc76742b34518e&amp;chksm=8b5f7153bc28f8456c411f5b04b012c4596c2d39e3719a2c9fdd554e8adbe734fb552106fafb&amp;scene=21#wechat_redirect" data-linktype="2">生信分析</a> | <a href="http://mp.weixin.qq.com/s?__biz=MzA5NjU5NjQ4MA==&amp;mid=2651228415&amp;idx=2&amp;sn=2c65d74e35632ce52094d25e0d2fcf33&amp;chksm=8b5f8f90bc2806866eb21d5e78bc339cdec20a0a08cff9a8e9bd7829bc0cea9616bccd421f1b&amp;scene=21#wechat_redirect" data-linktype="2">数字产品</a></span></p><p><span>概普生物 让科研丰富</span></p><section data-tools="135编辑器" data-id="125822" data-color="#00259f"><section data-bgw="266" data-ratio="0.8421052631578947"><section data-width="40%"><section><img data-imgfileid="503746075" data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/N3X4LBoaQjU8UrlCag7mJxhtImPmUaJbqd8waEEXicyE8MX2LvbajEibMWU9ulpoeqqcwcjg66icLQVlu5SVXzKmQ/640?wx_fmt=jpeg&amp;from=appmsg" data-type="jpeg" data-w="258" data-width="100%" src="https://mmbiz.qpic.cn/mmbiz_jpg/N3X4LBoaQjU8UrlCag7mJxhtImPmUaJbqd8waEEXicyE8MX2LvbajEibMWU9ulpoeqqcwcjg66icLQVlu5SVXzKmQ/640?wx_fmt=jpeg&amp;from=appmsg"></section></section><section data-width="60%"><section><section><section><strong data-brushtype="code">生信人</strong></section><section><section><br></section></section></section></section><section>专注于基因技术相关知识分享</section><section>扫码关注 获取更多</section></section></section></section><p><br></p><p><br></p><section data-tools="135编辑器" data-id="118316" data-color="#00259f"><section><section><section><p><br></p><section><br></section><p><br></p></section><section><strong data-brushtype="text">END</strong></section><section><section><br></section></section></section></section></section><p><br></p></section></section></section></section></section></section><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/QXZgq-kefkiX1GwtKxvIUw",target="_blank" rel="noopener noreferrer">原文链接</a>
