---
title: "拷贝数变异（CNV）分析全家桶"
date: 2024-05-02T08:17:27Z
draft: ["false"]
tags: [
  "fetched",
  "生信星球"
]
categories: ["Acdemic"]
---
拷贝数变异（CNV）分析全家桶 by 生信星球
------
<div><section data-mpa-powered-by="yiban.io"><br></section><section><span>‍</span></section><section><img data-imgfileid="100010876" data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz_png/8oKPbJgbBHrDic8XGmJ0b7oibVJajb0emLBHSvuibGG49ooBgtaAibE3TNJ00iaHviaMtdIKQJfCwtUfuHicDImtSfIxg/640?wx_fmt=png" data-type="png" data-w="64" width="20px" src="https://mmbiz.qpic.cn/mmbiz_png/8oKPbJgbBHrDic8XGmJ0b7oibVJajb0emLBHSvuibGG49ooBgtaAibE3TNJ00iaHviaMtdIKQJfCwtUfuHicDImtSfIxg/640?wx_fmt=png"><img data-imgfileid="100010877" data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz_png/8oKPbJgbBHrDic8XGmJ0b7oibVJajb0emLPukRHCbicy4pNKeEv9qd7aWSfsx7roib2od3xPrRPicw3a0kbn0uQ6JmQ/640?wx_fmt=png" data-type="png" data-w="64" width="20px" src="https://mmbiz.qpic.cn/mmbiz_png/8oKPbJgbBHrDic8XGmJ0b7oibVJajb0emLPukRHCbicy4pNKeEv9qd7aWSfsx7roib2od3xPrRPicw3a0kbn0uQ6JmQ/640?wx_fmt=png"><span> 今天是生信星球陪你的<span><strong>第948天</strong></span></span><img data-imgfileid="100010875" data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz_png/8oKPbJgbBHrDic8XGmJ0b7oibVJajb0emLBHSvuibGG49ooBgtaAibE3TNJ00iaHviaMtdIKQJfCwtUfuHicDImtSfIxg/640?wx_fmt=png" data-type="png" data-w="64" width="20px" src="https://mmbiz.qpic.cn/mmbiz_png/8oKPbJgbBHrDic8XGmJ0b7oibVJajb0emLBHSvuibGG49ooBgtaAibE3TNJ00iaHviaMtdIKQJfCwtUfuHicDImtSfIxg/640?wx_fmt=png"></section><hr><section><span><span>   </span><span>大神一句话，菜鸟跑半年。我不是大神，但我可以缩短你走弯路的半年~</span></span></section><section><span>   就像歌儿唱的那样，如果你不知道该往哪儿走，就留在这学点生信好不好~</span></section><p><span>   这里有豆豆和花花的学习历程，从新手到进阶，生信路上有你有我！</span></p><p><img data-galleryid="" data-imgfileid="100010914" data-ratio="0.4515209125475285" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/8oKPbJgbBHqVicoRyu0XmVy3BiawZR8Y1oon6YMC9a9nQdZzKIVmqObqZSdAIDGgozbBLuuAKfYFSkh3icj4Y9R8w/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1052" src="https://mmbiz.qpic.cn/mmbiz_png/8oKPbJgbBHqVicoRyu0XmVy3BiawZR8Y1oon6YMC9a9nQdZzKIVmqObqZSdAIDGgozbBLuuAKfYFSkh3icj4Y9R8w/640?wx_fmt=png&amp;from=appmsg"></p><p><span></span></p><section><p>本文比较长，长到需要个目录<br>1.定义<br>2.TCGA拷贝数变异数据处理流程<br>  2.1 数据下载<br>  2.2 数据整理<br>3.GISTIC2.0<br>4.maftools可视化<br>  找出感兴趣的区域里面的基因<br>5.基因层面的分析<br>  5.1 文件和数值大小意义<br>  5.2 差异CNV基因鉴定<br>  5.3 棒棒糖图<br>6.与基因表达量联合分析<br>  6.1 与表达量画箱线图<br>  6.2 与表达量的相关性<br>7.生存分析</p><h3><span>1.定义</span></h3><p>拷贝数变异(Copy number variation,CNV)是由基因组发生重排而导致的,一般指长度为1kb以上的基因组大片段的拷贝数增加或者减少,主要表现为亚显微水平的缺失和重复。CNV是基因组结构变异(Structuralvariation,SV)的重要组成部分。CNV位点的突变率远高于SNP(Single nucleotide polymorphism), 是人类疾病的重要致病因素之一。</p><p>–来自百度百科</p><h3><span>2.TCGA拷贝数变异数据处理流程</span></h3><p>TCGA 的拷贝数变异主要是Affymetrix SNP6.0 array，处理流程在： https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/CNV_Pipeline/</p><p>有三个workflow：ASCAT,ABSOLUTE和DNAcopy，其中ASCAT流程能够生成具有整数拷贝数值的等位基因特异性拷贝数段数据，以及派生的整数基因级拷贝数。参考基因组是hg19</p><p>ABSOLUTE 参考基因组是hg19，在GDC出版物页面有论文发表。</p><p>DNAcopy ：进行循环二进制分割(CBS)将嘈杂的强度测量值转换为拷贝数相等的染色体区域。最终输出文件被分割成基因组区域，每个区域都有估计的拷贝数。GDC 进一步将这些拷贝数值转换为segment mean(段平均值)，该值等于 log2（copy-number/ 2）。二倍体区域的段均值为零，扩增区域的区段均值为正值，缺失区的区段均值为负值。参考基因组是hg38，产生Copy Number Segment or Masked Copy Number Segment，其中Masked是指去掉了生殖相关SNV的数据。</p><h4><span><span> </span>2.1 数据下载</span></h4><p>我一般是喜欢用tcgabiolinks来下载，可以拿到最新的数据。</p><p>首先是下载Masked Copy Number Segment的数据，也就是上说的DNAcopy流程的结果。</p><pre><code>rm(list = ls())<br>library(TCGAbiolinks)<br>f = "masked.Rdata"<br>if(!file.exists(f)){<br>  query &lt;- GDCquery(project = "TCGA-CHOL",<br>                  data.category = "Copy Number Variation",<br>                  data.type = "Masked Copy Number Segment")<br><br>  GDCdownload(query)<br>  seg &lt;- GDCprepare(query) <br>  colnames(seg)<br>  seg = seg[,c(7,2:6)]<br>  save(seg,file = f)<br>  write.table(seg,file = "seg.txt",row.names = F,quote = F,sep = "\t")<br>}<br>load(f)<br>head(seg)<br><span><br>#</span><span># # A tibble: 6 × 6</span><br><span>#</span><span>#   Sample                       Chromosome   Start    End Num_Probes Segment_Mean</span><br><span>#</span><span>#   &lt;chr&gt;                        &lt;chr&gt;        &lt;dbl&gt;  &lt;dbl&gt;      &lt;dbl&gt;        &lt;dbl&gt;</span><br><span>#</span><span># 1 TCGA-W5-AA2X-11A-11D-A416-01 1           3.30e6 1.04e8      58516      -0.001 </span><br><span>#</span><span># 2 TCGA-W5-AA2X-11A-11D-A416-01 1           1.04e8 1.04e8          5      -0.972 </span><br><span>#</span><span># 3 TCGA-W5-AA2X-11A-11D-A416-01 1           1.04e8 1.79e8      27859       0.0016</span><br><span>#</span><span># 4 TCGA-W5-AA2X-11A-11D-A416-01 1           1.79e8 1.79e8          2      -1.26  </span><br><span>#</span><span># 5 TCGA-W5-AA2X-11A-11D-A416-01 1           1.79e8 2.48e8      43396       0.0014</span><br><span>#</span><span># 6 TCGA-W5-AA2X-11A-11D-A416-01 2           4.81e5 1.71e7      10742       0.008</span><br><span><br>length(unique(seg$</span><span>Sample)) <span>#人数</span></span><br><span><br>#</span><span># [1] 85</span><br><span>#</span><span>如果只需要tumor样本那么就：</span><br><span>#</span><span>k = as.numeric(str_sub(seg<span>$Sample</span>,14,15))&lt;10</span><br><span>#</span><span>seg = seg[k,]</span><br></code></pre><p>然后是下载markers file，在gdc网站的参考文件中能够找到它,下载后放在工作目录下。</p><p>https://gdc.cancer.gov/about-data/gdc-data-processing/gdc-reference-files</p><h4><span><span> </span>2.2 数据整理</span></h4><p>此处有一句说明:If you are using Masked Copy Number Segment for GISTIC analysis, please only keep probesets with freqcnv = FALSE</p><pre><code>marker = data.table::fread("snp6.na35.remap.hg38.subset.txt.gz",data.table = F)<br>head(marker)<br><span><br>#</span><span>#         probeid chr       pos strand type freqcnv   par</span><br><span>#</span><span># 1 SNP_A-1780270   7  78970267      +  SNP    TRUE FALSE</span><br><span>#</span><span># 2 SNP_A-1780271  15  33103578      +  SNP   FALSE FALSE</span><br><span>#</span><span># 3 SNP_A-1780272   1 189838554      -  SNP    TRUE FALSE</span><br><span>#</span><span># 4 SNP_A-1780274  20  35320106      -  SNP   FALSE FALSE</span><br><span>#</span><span># 5 SNP_A-1780277  12  75270266      +  SNP   FALSE FALSE</span><br><span>#</span><span># 6 SNP_A-1780278   1 218717316      +  SNP   FALSE FALSE</span><br></code></pre><p>可以看到数据里缺失有freqcnv 列，内容是TRUE和FALSE</p><pre><code><span>table(marker$</span><span>freqcnv)</span><br><span><br>#</span><span># </span><br><span>#</span><span>#   FALSE    TRUE </span><br><span>#</span><span># 1595984  241523</span><br><br>marker = marker[!marker$freqcnv,] #只保留FALSE<br>marker =marker[,1:3]<br>colnames(marker) = c("Marker Name","Chromosome","Marker Position")<br>head(marker)<br><span><br>#</span><span>#     Marker Name Chromosome Marker Position</span><br><span>#</span><span># 2 SNP_A-1780271         15        33103578</span><br><span>#</span><span># 4 SNP_A-1780274         20        35320106</span><br><span>#</span><span># 5 SNP_A-1780277         12        75270266</span><br><span>#</span><span># 6 SNP_A-1780278          1       218717316</span><br><span>#</span><span># 7 SNP_A-1780283          4       126709121</span><br><span>#</span><span># 8 SNP_A-1780285          6        90209746</span><br><br>write.table(marker,file = "marker.txt",row.names = F,quote = F,sep = "\t")<br></code></pre><p>colnames(marker)…那一句是把marker文件的列名修改为了GISTIC要求的名字。</p><p>至此，GISTIC2.0需要的输入文件就准备好了。</p><h3><span>3.GISTIC2.0</span></h3><p>https://broadinstitute.github.io/gistic2/</p><p>全称是Genomic Identification of Significant Targets in Cancer(癌症中重要靶点的基因组鉴定)，这是一个linux软件，GenePattern 网站上面有GISTIC模块，允许在线运行。</p><p>GISTIC可以识别基因组中在一组样本中被显著扩增或删除的区域，得到G-score和q值用于一组样本整体的可视化。</p><figure><img data-imgfileid="100010904" data-ratio="0.4166666666666667" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/8oKPbJgbBHqVicoRyu0XmVy3BiawZR8Y1ovjVyicAmmHJkX8Jq0ZpTjuLN8IUy4Y9gqVkBZiclGvj3yQedPriaajLcA/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="1080" title="" src="https://mmbiz.qpic.cn/mmbiz_jpg/8oKPbJgbBHqVicoRyu0XmVy3BiawZR8Y1ovjVyicAmmHJkX8Jq0ZpTjuLN8IUy4Y9gqVkBZiclGvj3yQedPriaajLcA/640?wx_fmt=other&amp;from=appmsg"><figcaption></figcaption></figure><figure><img data-imgfileid="100010901" data-ratio="0.5894117647058823" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/8oKPbJgbBHqVicoRyu0XmVy3BiawZR8Y1o8bAYdH3HARicVOwaV83qQL8YIBu8BBac3j3HzbaZTFUDNQdITkL1jWQ/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="850" title="" src="https://mmbiz.qpic.cn/mmbiz_jpg/8oKPbJgbBHqVicoRyu0XmVy3BiawZR8Y1o8bAYdH3HARicVOwaV83qQL8YIBu8BBac3j3HzbaZTFUDNQdITkL1jWQ/640?wx_fmt=other&amp;from=appmsg"><figcaption></figcaption></figure><br>置信区间填0.99<br><figure><img data-imgfileid="100010902" data-ratio="0.527536231884058" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/8oKPbJgbBHqVicoRyu0XmVy3BiawZR8Y1o3SkpzGCjTgZiarxKzt3tYhdjHicrH2UI7uHHiaX6V5R0NDhrQfCrtZpnA/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="1035" title="" src="https://mmbiz.qpic.cn/mmbiz_jpg/8oKPbJgbBHqVicoRyu0XmVy3BiawZR8Y1o3SkpzGCjTgZiarxKzt3tYhdjHicrH2UI7uHHiaX6V5R0NDhrQfCrtZpnA/640?wx_fmt=other&amp;from=appmsg"><figcaption></figcaption></figure><br>然后拉到页面最后点run<p><br></p><p>正常的运行结果是有很多文件的，如果看到只有这三个文件那就是运行失败了。</p><figure><img data-imgfileid="100010903" data-ratio="0.24390243902439024" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/8oKPbJgbBHqVicoRyu0XmVy3BiawZR8Y1oczZ6KAqpIdAkiauTIibIfPW2YuKKLCAtQg2ibaibDgQzkxEhiahAfYyhnMA/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="410" title="" src="https://mmbiz.qpic.cn/mmbiz_jpg/8oKPbJgbBHqVicoRyu0XmVy3BiawZR8Y1oczZ6KAqpIdAkiauTIibIfPW2YuKKLCAtQg2ibaibDgQzkxEhiahAfYyhnMA/640?wx_fmt=other&amp;from=appmsg"><figcaption></figcaption></figure><p>出现这个结果的常见问题有：</p><blockquote><p>1.markers文件要看清楚subset.txt.gz，不要下载错了<br>2.文件需要以tab为分隔符，所以上面的代码里sep = "\t"是不能少的。<br>3.markers文件的列名没改，所以上面的代码里colnames。。。那句代码是不能少的。</p></blockquote><p>如果都排查完了还是三个文件，就试试不要上传markers。</p><p>打包下载并解压，放在工作目录下。<br></p><figure><img data-imgfileid="100010905" data-ratio="0.6459276018099548" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/8oKPbJgbBHqVicoRyu0XmVy3BiawZR8Y1o45WaAxdhpTIO4co4Ze8gDcXr7v4zULQKXAVwdF99dJzRwnqOVMgBkA/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="884" title="" src="https://mmbiz.qpic.cn/mmbiz_jpg/8oKPbJgbBHqVicoRyu0XmVy3BiawZR8Y1o45WaAxdhpTIO4co4Ze8gDcXr7v4zULQKXAVwdF99dJzRwnqOVMgBkA/640?wx_fmt=other&amp;from=appmsg"><figcaption></figcaption></figure><h3><span>4.maftools可视化</span></h3><p>这个是区域层面的分析，单个基因层面的在下一部分。</p><pre><code>library(maftools)<br>library(tidyverse)<br>laml.gistic = readGistic(gisticAllLesionsFile = "579436/all_lesions.conf_99.txt",<br>                         gisticAmpGenesFile = "579436/amp_genes.conf_99.txt", <br>                         gisticDelGenesFile = "579436/del_genes.conf_99.txt", <br>                         gisticScoresFile = "579436/scores.gistic", <br>                         isTCGA = TRUE)<br><span><br>#</span><span># -Processing Gistic files..</span><br><span>#</span><span># --Processing amp_genes.conf_99.txt</span><br><span>#</span><span># --Processing del_genes.conf_99.txt</span><br><span>#</span><span># --Processing scores.gistic</span><br><span>#</span><span># --Summarizing by samples</span><br><br>head(laml.gistic@data)<br><span><br>#</span><span>#    Hugo_Symbol Tumor_Sample_Barcode Variant_Classification Variant_Type</span><br><span>#</span><span># 1:       CCND1         TCGA-W5-AA30                    Amp          CNV</span><br><span>#</span><span># 2:        FGF4         TCGA-W5-AA30                    Amp          CNV</span><br><span>#</span><span># 3:       FGF19         TCGA-W5-AA30                    Amp          CNV</span><br><span>#</span><span># 4:       MYEOV         TCGA-W5-AA30                    Amp          CNV</span><br><span>#</span><span># 5:      MRGPRD         TCGA-W5-AA30                    Amp          CNV</span><br><span>#</span><span># 6:      MRGPRF         TCGA-W5-AA30                    Amp          CNV</span><br><span>#</span><span>#        Cytoband</span><br><span>#</span><span># 1: AP_2:11q13.3</span><br><span>#</span><span># 2: AP_2:11q13.3</span><br><span>#</span><span># 3: AP_2:11q13.3</span><br><span>#</span><span># 4: AP_2:11q13.3</span><br><span>#</span><span># 5: AP_2:11q13.3</span><br><span>#</span><span># 6: AP_2:11q13.3</span><br></code></pre><p>ChromPlot，从左向右是全部的染色体。有颜色的部分突出了明显的扩增和删失区域,图上的文字例如”17q12”， “17” 指的是染色体编号，“q” 指的是长臂， “12” 是该臂上的具体区带或条带编号。</p><pre><code>gisticChromPlot(gistic = laml.gistic, <span>ref</span>.build = <span>'hg38'</span>,markBands = <span>"all"</span>)<br></code></pre><figure><img data-imgfileid="100010907" data-ratio="0.7138888888888889" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/8oKPbJgbBHqVicoRyu0XmVy3BiawZR8Y1oJVOdU744FmsSS9x1kXPJEKzHdshp2IGj7hbeEu3xRicgN9bKA7hyhsg/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="1080" title="" src="https://mmbiz.qpic.cn/mmbiz_jpg/8oKPbJgbBHqVicoRyu0XmVy3BiawZR8Y1oJVOdU744FmsSS9x1kXPJEKzHdshp2IGj7hbeEu3xRicgN9bKA7hyhsg/640?wx_fmt=other&amp;from=appmsg"><figcaption></figcaption></figure><pre><code>head(laml.gistic@gis.scores)<br><span><br>#</span><span>#    Variant_Classification Chromosome Start_Position End_Position fdr  G_Score</span><br><span>#</span><span># 1:                    Amp          1        3301765      3530249   0 0.003032</span><br><span>#</span><span># 2:                    Amp          1        3558079      4117838   0 0.000000</span><br><span>#</span><span># 3:                    Amp          1        4118243      4402844   0 0.003870</span><br><span>#</span><span># 4:                    Amp          1        4403866     21803810   0 0.000000</span><br><span>#</span><span># 5:                    Amp          1       21804530     22275547   0 0.004227</span><br><span>#</span><span># 6:                    Amp          1       22276873     32428780   0 0.000000</span><br><span>#</span><span>#    Avg_amplitude frequency</span><br><span>#</span><span># 1:      0.406272  0.011765</span><br><span>#</span><span># 2:      0.000000  0.000000</span><br><span>#</span><span># 3:      0.258664  0.023529</span><br><span>#</span><span># 4:      0.363623  0.011765</span><br><span>#</span><span># 5:      0.142905  0.011765</span><br><span>#</span><span># 6:      0.000000  0.000000</span><br></code></pre><p>这个图的美化版本：https://blog.csdn.net/Ayue0616/article/details/127824394</p><p>Bubbleplot，横坐标是每个区域里发生拷贝数变异的样本数量，纵坐标是每个区域包含的基因数量，大小是按照每个区域的q值。</p><pre><code>gisticBubblePlot(gistic = laml.gistic)<br></code></pre><figure><img data-imgfileid="100010910" data-ratio="0.7138888888888889" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/8oKPbJgbBHqVicoRyu0XmVy3BiawZR8Y1oVOmpKKC13f6F18Bxh99Ia0vtMOEFbsbqFCOqkOUr8hibic9h9dKMg4IQ/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="1080" title="" src="https://mmbiz.qpic.cn/mmbiz_jpg/8oKPbJgbBHqVicoRyu0XmVy3BiawZR8Y1oVOmpKKC13f6F18Bxh99Ia0vtMOEFbsbqFCOqkOUr8hibic9h9dKMg4IQ/640?wx_fmt=other&amp;from=appmsg"><figcaption></figcaption></figure><pre><code>head(laml.gistic@cytoband.summary)<br><span><br>#</span><span>#      Unique_Name nGenes nSamples Variant_Classification Cytoband</span><br><span>#</span><span># 1:  DP_1:1p36.21    110       31                    Del  1p36.21</span><br><span>#</span><span># 2:  DP_13:9p21.3      2       23                    Del   9p21.3</span><br><span>#</span><span># 3:   DP_5:4q34.3     69       23                    Del   4q34.3</span><br><span>#</span><span># 4:  AP_2:11q13.3     13       10                    Amp  11q13.3</span><br><span>#</span><span># 5: DP_23:16q23.1      1        7                    Del  16q23.1</span><br><span>#</span><span># 6:   DP_17:11q25   1635       14                    Del    11q25</span><br><span>#</span><span>#            Wide_Peak_Limits    qvalues</span><br><span>#</span><span># 1:    chr1:8816334-14600926 4.9692e-26</span><br><span>#</span><span># 2:   chr9:21865499-21997723 2.7424e-07</span><br><span>#</span><span># 3: chr4:176330306-190214555 4.6580e-07</span><br><span>#</span><span># 4:  chr11:68952543-69788268 2.4238e-04</span><br><span>#</span><span># 5:  chr16:78095161-79593873 2.0617e-03</span><br><span>#</span><span># 6:        chr11:1-135086622 3.5922e-03</span><br></code></pre><p>OncoPlot,颜色太辣眼了必须得改改。。。</p><p>这个图一行是一个区域，一列是一个样本，展示每个区域在多少样本里发生了拷贝数变异。</p><pre><code>mycolor = c(<span>"#f87669"</span>,<span>"#2fa1dd"</span>)<br>names(mycolor) = c(<span>"Amp"</span>,<span>"Del"</span>)<br>gisticOncoPlot(laml.gistic,top = 20,colors = mycolor)<br></code></pre><figure><img data-imgfileid="100010908" data-ratio="0.7138888888888889" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/8oKPbJgbBHqVicoRyu0XmVy3BiawZR8Y1ozgzurJo1XepTYMFkrzFMG8SOohnqygRib7CCXibKjjtlxwEeDdmHuPicw/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="1080" title="" src="https://mmbiz.qpic.cn/mmbiz_jpg/8oKPbJgbBHqVicoRyu0XmVy3BiawZR8Y1ozgzurJo1XepTYMFkrzFMG8SOohnqygRib7CCXibKjjtlxwEeDdmHuPicw/640?wx_fmt=other&amp;from=appmsg"><figcaption></figcaption></figure><h5><span>找出感兴趣的区域里面的基因</span></h5><pre><code>tmp = as.data.frame(laml.gistic@data[,c(1,5)])<br>library(tidyverse)<br>tmp = separate(tmp,col = Cytoband,into = c("cnv","cyto"),sep = ":")<br>head(tmp)<br><span><br>#</span><span>#   Hugo_Symbol  cnv    cyto</span><br><span>#</span><span># 1       CCND1 AP_2 11q13.3</span><br><span>#</span><span># 2        FGF4 AP_2 11q13.3</span><br><span>#</span><span># 3       FGF19 AP_2 11q13.3</span><br><span>#</span><span># 4       MYEOV AP_2 11q13.3</span><br><span>#</span><span># 5      MRGPRD AP_2 11q13.3</span><br><span>#</span><span># 6      MRGPRF AP_2 11q13.3</span><br><span><br>table(tmp$</span><span>cnv)</span><br><span><br>#</span><span># </span><br><span>#</span><span>#  AP_1  AP_2  DP_1 DP_10 DP_11 DP_12 DP_13 DP_14 DP_15 DP_16 DP_17 DP_18 DP_19 </span><br><span>#</span><span># 12825   130  3410  5460  1944 20615    46  4452    18  5421 22890 12069  2227 </span><br><span>#</span><span>#  DP_2 DP_20 DP_21 DP_22 DP_23  DP_3  DP_4  DP_5  DP_6  DP_7  DP_8  DP_9 </span><br><span>#</span><span>#  4760    28    28  6680     7 10136    17  1587  2168   150 26600  2222</span><br><br>k = tmp$Hugo_Symbol[tmp$cyto %in% c("1p36.21","11q25")] %&gt;% unique()<br>length(k)<br><span><br>#</span><span># [1] 1745</span><br></code></pre><h3><span>5.基因层面的分析</span></h3><h4><span><span> </span>5.1 文件和数值大小意义</span></h4><p>在GISTIC的输出结果里有4个genes.txt为结尾的文件</p><pre><code>dir(<span>"579436/"</span>,pattern = <span>"genes.txt$"</span>)<br><br><span>## [1] <span>"all_data_by_genes.txt"</span>        <span>"all_thresholded.by_genes.txt"</span></span><br><span>## [3] <span>"broad_data_by_genes.txt"</span>      <span>"focal_data_by_genes.txt"</span></span><br></code></pre><p>其中：</p><section><blockquote><p>all_data_by_genes.txt：包含基因在不同样本中具体的拷贝数数值。 <br>all_thresholded.by_genes.txt：基因在不同样本中的拷贝数数值离散化后的结果，GISTIC2进一步将估计值阈值定为-2，-1,0,1,2，代表纯合子缺失、单拷贝缺失、二倍体正常拷贝、低水平拷贝数扩增或高拷贝数扩增。（ homozygous deletion,single copy deletion,diploid normal copy,low-level copy number amplification,or high-level copy number amplification） <br>broad_data_by_genes.txt：只考虑臂级（arm-level）事件的基因在不同样本中具体的拷贝数数值。 <br>focal_data_by_genes.txt：只考虑局灶性（focal）事件的基因在不同样本中具体的拷贝数数值。</p></blockquote></section><pre><code>all_data = read.delim("579436/all_data_by_genes.txt",row.names = 1,check.names = F)<br>all_data[1:4,1:5]<br><span><br>#</span><span>#         Gene ID Cytoband TCGA-W5-AA2X-11A-11D-A416-01</span><br><span>#</span><span># ACAP3    116983  1p36.33                       -0.003</span><br><span>#</span><span># ACTRT2   140625  1p36.32                       -0.003</span><br><span>#</span><span># AGRN     375790  1p36.33                       -0.003</span><br><span>#</span><span># ANKRD65  441869  1p36.33                       -0.003</span><br><span>#</span><span>#         TCGA-W5-AA2X-10A-01D-A419-01 TCGA-W5-AA30-01A-31D-A416-01</span><br><span>#</span><span># ACAP3                         -0.001                       -0.526</span><br><span>#</span><span># ACTRT2                        -0.001                       -0.526</span><br><span>#</span><span># AGRN                          -0.001                       -0.526</span><br><span>#</span><span># ANKRD65                       -0.001                       -0.526</span><br><br>range(all_data[,-(1:2)])<br><span><br>#</span><span># [1] -1.293  3.657</span><br><br>all_thr = read.delim("579436/all_thresholded.by_genes.txt",row.names = 1,check.names = F)<br>all_thr[1:4,1:5]<br><span><br>#</span><span>#         Locus ID Cytoband TCGA-W5-AA2X-11A-11D-A416-01</span><br><span>#</span><span># ACAP3     116983  1p36.33                            0</span><br><span>#</span><span># ACTRT2    140625  1p36.32                            0</span><br><span>#</span><span># AGRN      375790  1p36.33                            0</span><br><span>#</span><span># ANKRD65   441869  1p36.33                            0</span><br><span>#</span><span>#         TCGA-W5-AA2X-10A-01D-A419-01 TCGA-W5-AA30-01A-31D-A416-01</span><br><span>#</span><span># ACAP3                              0                           -1</span><br><span>#</span><span># ACTRT2                             0                           -1</span><br><span>#</span><span># AGRN                               0                           -1</span><br><span>#</span><span># ANKRD65                            0                           -1</span><br><br>range(all_thr[,-(1:2)])<br><span><br>#</span><span># [1] -2  2</span><br><br>tmp = all_thr[,-(1:2)]<br>table(tmp[tmp!=0])<br><span><br>#</span><span># </span><br><span>#</span><span>#     -2     -1      1      2 </span><br><span>#</span><span>#   2021 218561 166880   8555</span><br></code></pre><p>主要就看这两个文件啦。</p><p>对于这些数值大小的有好多种衍生的翻译</p><p>在timer2.0上面： “deep deletion”, “arm-level deletion”, “diploid/normal”, “arm-level gain”, and “high amplification”</p><p>在GSCA上面：Homozygous Deletion,Heterozygous Deletion,Heterozygous Amplification,Homozygous Deletion</p><p>Hete.(Heterozygous，杂合); Homo.(Homozygous，纯合)</p><h5><span>5.2 差异CNV基因鉴定</span></h5><p>翻了一些文献和github，可以根据是否有发生CNV将样本分成CNV（!=0）和Wild(==0)，或者是分成扩增(&gt;0)、缺失(&lt;0)和正常(==0)三个组，看tumor和normal样本中每个基因的CNV是否有差别,检验方法主要是卡方检验。</p><p>Y叔团队开发的GeoTcgaData包有现成的函数differential_CNV,但我用的时候发现有个问题，源代码里面有一句<code>gene &lt;- gsub("\\..*", "", rownames(cnvData))</code>,代表把基因名中的点号后面的部分给去掉，这其实是给ensembel矩阵准备的。但如果是symbol为行名的矩阵，就要出错。</p><pre><code><span>if(!require(GeoTcgaData))BiocManager::install("GeoTcgaData")</span><br>library(GeoTcgaData)<br>cnv = all_thr[,-(1:2)]<br>rownames(cnv) = 1:nrow(cnv)<br>cnv[cnv &gt;= 1] = 1<br>cnv[cnv &lt;= -1] = -1<br>library(tinyarray)<br>group = make_tcga_group(cnv)<br>diffcnv = differential_CNV(cnv,group)<br></code></pre><p>对这个源代码进行修改就比较麻烦，不如把行名改成数字，然后再替换回symbol就是了。</p><pre><code><span>diffcnv$</span><span>gene = rownames(all_thr)</span><br>rownames(cnv) = rownames(all_thr)<br>head(diffcnv)<br><span><br>#</span><span>#      gene      P.Value    adj.P.Val estimate</span><br><span>#</span><span># 1   ACAP3 1.980127e-14 1.072074e-12        0</span><br><span>#</span><span># 2  ACTRT2 1.980127e-14 1.072074e-12        0</span><br><span>#</span><span># 3    AGRN 1.980127e-14 1.072074e-12        0</span><br><span>#</span><span># 4 ANKRD65 1.980127e-14 1.072074e-12        0</span><br><span>#</span><span># 5  ATAD3A 1.980127e-14 1.072074e-12        0</span><br><span>#</span><span># 6  ATAD3B 1.980127e-14 1.072074e-12        0</span><br><span><br>table(diffcnv$</span><span>P.Value&lt;0.05)</span><br><span><br>#</span><span># </span><br><span>#</span><span># FALSE  TRUE </span><br><span>#</span><span>#  1452 24536</span><br><span><br>table(diffcnv$</span><span>adj.P.Val&lt;0.05)</span><br><span><br>#</span><span># </span><br><span>#</span><span># FALSE  TRUE </span><br><span>#</span><span>#  1452 24536</span><br></code></pre><h5><span>5.3 棒棒糖图</span></h5><p>一组感兴趣的基因</p><pre><code>my_gene = c(<span>"CXCL8"</span>, <span>"FN1"</span>, <span>"COL3A1"</span>, <span>"ISG15"</span>, <span>"COL1A2"</span>, <span>"CXCL10"</span>, <span>"ICAM1"</span>, <span>"MMP9"</span>)<br>cnv = cnv[my_gene,group==<span>"tumor"</span>]<br><br>library(tidyverse)<br>cnv_df = as.data.frame(t(cnv)) %&gt;% rownames_to_column(<span>"Sample"</span>)<br>cnv_long &lt;- gather(cnv_df, key = <span>"Gene"</span>, value = <span>"CNV"</span>, 2:9)<br><br><span># 统计每个基因的拷贝数变化样本比例</span><br>cnv_summary &lt;- cnv_long %&gt;%<br>  filter(CNV != 0) %&gt;%  <span># 排除正常的样本</span><br>  group_by(Gene, CNV) %&gt;%<br>  summarise(Freq = n()/ncol(cnv) *100) %&gt;%<br>  ungroup() %&gt;%<br>  mutate(CNV = ifelse(CNV&gt;0,<span>"Amp"</span>,<span>"Del"</span>))<br><span># 控制基因顺序</span><br>tmp = cnv_summary %&gt;%<br>  group_by(Gene) %&gt;%<br>  summarise(max = max(Freq)) %&gt;%<br>  arrange(desc(max))<br>cnv_summary$Gene = factor(cnv_summary$Gene,levels = tmp$Gene)<br><span># 使用 ggplot2 绘制点图</span><br>library(ggplot2)<br>ggplot(cnv_summary, aes(x = Gene, y = Freq)) +<br>  geom_segment( aes(x=Gene, xend=Gene, y=0, yend=Freq),color = <span>"grey"</span>,linewidth = 3) +<br>  geom_point(aes(color = CNV),size = 5) +<br>  scale_color_manual(values = c(<span>"Del"</span> = <span>"#2fa1dd"</span>, <span>"Amp"</span> = <span>"#f87669"</span>)) +<br>  theme_minimal()<br></code></pre><figure><img data-imgfileid="100010906" data-ratio="0.6186186186186187" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/8oKPbJgbBHqVicoRyu0XmVy3BiawZR8Y1ojJlXnia54zTZCsaSh8cU0PTlEiaExibEOJLC7XBcpQJGQhJXJBZw6XFiaw/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="666" title="" src="https://mmbiz.qpic.cn/mmbiz_jpg/8oKPbJgbBHqVicoRyu0XmVy3BiawZR8Y1ojJlXnia54zTZCsaSh8cU0PTlEiaExibEOJLC7XBcpQJGQhJXJBZw6XFiaw/640?wx_fmt=other&amp;from=appmsg"><figcaption></figcaption></figure><h3><span>6.与基因表达量联合分析</span></h3><pre><code>cnv_thr = as.matrix(all_thr[,colnames(cnv)])<br>cnvall = as.matrix(all_data[,colnames(cnv)])<br>library(tinyarray)<br><span>load("D:/TCGA_RNA_seq/tpm/chol_tpm.Rdata")</span><br>exp = trans_exp_new(chol_tpm)<br>exp = log2(exp+1)<br>colnames(exp) = str_sub(colnames(exp),1,16)<br>colnames(cnv_thr) = str_sub(colnames(cnv_thr),1,16)<br>colnames(cnvall) = str_sub(colnames(cnvall),1,16)<br><br>s = intersect(colnames(cnv_thr),colnames(exp))<br>exp = exp[my_gene,s]<br>cnv_thr = cnv_thr[,s]<br>cnvall = cnvall[,s]<br></code></pre><h4><span><span> </span>6.1 与表达量画箱线图</span></h4><pre><code>g = "ISG15"<br>dat = data.frame(mrna = exp[g,],<br>                 cnv = as.factor(cnv_thr[g,]),<br>                 cnvscore = cnvall[g,])<br>head(dat)<br><span><br>#</span><span>#                      mrna cnv cnvscore</span><br><span>#</span><span># TCGA-W5-AA30-01A 6.839125  -1   -0.526</span><br><span>#</span><span># TCGA-W5-AA2W-01A 7.535962  -1   -0.889</span><br><span>#</span><span># TCGA-ZH-A8Y4-01A 5.859495  -1   -0.802</span><br><span>#</span><span># TCGA-W5-AA2Z-01A 8.296128   0    0.027</span><br><span>#</span><span># TCGA-W6-AA0S-01A 7.077692  -1   -0.765</span><br><span>#</span><span># TCGA-W5-AA33-01A 6.458971  -1   -0.409</span><br><span><br>table(dat$</span><span>cnv)</span><br><span><br>#</span><span># </span><br><span>#</span><span># -2 -1  0  1 </span><br><span>#</span><span>#  1 27  6  1</span><br><br>library(ggplot2)<br>ggplot(dat,aes(cnv,mrna,fill = cnv))+<br>  geom_boxplot()+<br>  scale_fill_brewer(palette = "Set1")+<br>  theme_minimal()<br></code></pre><figure><img data-imgfileid="100010909" data-ratio="0.7138888888888889" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/8oKPbJgbBHqVicoRyu0XmVy3BiawZR8Y1oyOtZCQtO2vLRsfOCibibdl8ibzytaGvD2AwnmibrpsZF60gcXvePAztgSw/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="1080" title="" src="https://mmbiz.qpic.cn/mmbiz_jpg/8oKPbJgbBHqVicoRyu0XmVy3BiawZR8Y1oyOtZCQtO2vLRsfOCibibdl8ibzytaGvD2AwnmibrpsZF60gcXvePAztgSw/640?wx_fmt=other&amp;from=appmsg"><figcaption></figcaption></figure><pre><code><span>library</span>(ggstatsplot)<br>ggbetweenstats(dat,x = <span>"cnv"</span>,y = <span>"mrna"</span>,drop = <span>F</span>)<br></code></pre><figure><img data-imgfileid="100010913" data-ratio="0.7138888888888889" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/8oKPbJgbBHqVicoRyu0XmVy3BiawZR8Y1oewMiaYcTp8qYyMaCyUJLlvFLXBjLtI722xUt1RDtcSSLia9ogMUOIhNw/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="1080" title="" src="https://mmbiz.qpic.cn/mmbiz_jpg/8oKPbJgbBHqVicoRyu0XmVy3BiawZR8Y1oewMiaYcTp8qYyMaCyUJLlvFLXBjLtI722xUt1RDtcSSLia9ogMUOIhNw/640?wx_fmt=other&amp;from=appmsg"><figcaption></figcaption></figure><h4><span><span> </span>6.2 与表达量的相关性</span></h4><p>还是上面的数据</p><pre><code><span>corscatterplot</span>(dat,x = <span>"cnvscore"</span>,y = <span>"mrna"</span>)<br></code></pre><figure><img data-imgfileid="100010912" data-ratio="0.7138888888888889" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/8oKPbJgbBHqVicoRyu0XmVy3BiawZR8Y1oY6h0xickibhwXibtd2FfSFtUgJiaQIMEE04picicNHKz7vOR50QdSAEfiaW3A/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="1080" title="" src="https://mmbiz.qpic.cn/mmbiz_jpg/8oKPbJgbBHqVicoRyu0XmVy3BiawZR8Y1oY6h0xickibhwXibtd2FfSFtUgJiaQIMEE04picicNHKz7vOR50QdSAEfiaW3A/640?wx_fmt=other&amp;from=appmsg"><figcaption></figcaption></figure><h3><span>7.生存分析</span></h3><pre><code>surv = read.delim(<span>"D:/TCGA_RNA_seq/survival/TCGA-CHOL.survival.tsv"</span>,row.names = 1)<br>head(surv)<br><br><span>##                  OS    X_PATIENT OS.time</span><br><span>## TCGA-W5-AA31-01A  0 TCGA-W5-AA31      10</span><br><span>## TCGA-W5-AA31-11A  0 TCGA-W5-AA31      10</span><br><span>## TCGA-WD-A7RX-01A  1 TCGA-WD-A7RX      21</span><br><span>## TCGA-YR-A95A-01A  1 TCGA-YR-A95A      26</span><br><span>## TCGA-4G-AAZR-11A  1 TCGA-4G-AAZR      28</span><br><span>## TCGA-4G-AAZR-01A  1 TCGA-4G-AAZR      28</span><br><br>colnames(surv)[c(1,3)] = c(<span>"event"</span>,<span>"time"</span>) <br>s = intersect(colnames(cnv_thr),rownames(surv))<br>surv = surv[s,]<br>cnv_thr = cnv_thr[,s]<br>g = <span>"ISG15"</span><br>cnv_group = ifelse(cnv_thr[g,]==0,<span>"Wild"</span>,<span>"CNV"</span>)<br>cnv_group = factor(cnv_group,levels = c(<span>"Wild"</span>,<span>"CNV"</span>))<br>draw_KM(surv,cnv_group)<br></code></pre><figure><img data-imgfileid="100010911" data-ratio="0.7138888888888889" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/8oKPbJgbBHqVicoRyu0XmVy3BiawZR8Y1oFcautwwSibX0Hd1yiafZsbg1HztF2diaQao1cWGDRic5vN9uhLnL3sYAnA/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="1080" title="" src="https://mmbiz.qpic.cn/mmbiz_jpg/8oKPbJgbBHqVicoRyu0XmVy3BiawZR8Y1oFcautwwSibX0Hd1yiafZsbg1HztF2diaQao1cWGDRic5vN9uhLnL3sYAnA/640?wx_fmt=other&amp;from=appmsg"><figcaption></figcaption></figure></section><p><span></span></p><p><span><br></span></p><section><section><section><figure><figcaption></figcaption></figure></section><figure><figcaption></figcaption></figure></section><figure><span></span><span></span></figure></section><section><figure><section><figure><figcaption></figcaption></figure></section></figure></section><section><figure><figcaption></figcaption></figure></section><section><figure><figcaption></figcaption></figure></section><section><figure><figcaption></figcaption></figure></section><section><blockquote><section><span>如果因为代码看不懂，而跟不上正文的节奏，可以来找我，相当于来一个新手保护期。</span><span>我的课程都是循环开课。</span><span>下一期的时间，点进去咨询微信咯</span><br></section><section><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzU4NjU4ODQ2MQ==&amp;mid=2247494547&amp;idx=1&amp;sn=90165024f9fe55db558502c3ab5dde6e&amp;chksm=fdfba3d1ca8c2ac7d81212b631784044b635a7f11bb921d41eaaa7677623fb9c07aebbfdb994&amp;scene=21#wechat_redirect" textvalue="生信分析直播课‍程" linktype="text" imgurl="" imgdata="null" data-itemshowtype="11" tab="innerlink" data-linktype="2">生信分析直播课程</a></section><section><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzU4NjU4ODQ2MQ==&amp;mid=2247494009&amp;idx=1&amp;sn=035d253f973f65c0d1069e515eec2ce8&amp;chksm=fdfba13bca8c282de5a9222a2ab8bc0caa2ddf60cab543c2181e6d4a90b553f4dcde5f154adf&amp;scene=21#wechat_redirect" textvalue="生信新手保护学习小组‍" linktype="text" imgurl="" imgdata="null" data-itemshowtype="11" tab="innerlink" data-linktype="2">生信新手保护学习小组</a><br></section></blockquote></section><section><section><br></section></section><section><br></section><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/z-8K95DJ-aIOdQMubqpMqQ",target="_blank" rel="noopener noreferrer">原文链接</a>
