---
title: "BayesPrism(贝叶斯棱镜法)可提取单细胞数据去卷积后将信息映射至bulkRNA数据对其进行细胞分群"
date: 2024-07-18T15:10:02Z
draft: ["false"]
tags: [
  "fetched",
  "生信方舟"
]
categories: ["Acdemic"]
---
BayesPrism(贝叶斯棱镜法)可提取单细胞数据去卷积后将信息映射至bulkRNA数据对其进行细胞分群 by 生信方舟
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><p data-tool="mdnice编辑器">贝叶斯棱镜法作为一种工具可以<strong>根据scRNA数据</strong>(作为先验模型)去<strong>推断bulkRNA数据</strong>中<strong>肿瘤微环境组成</strong>(不同免疫细胞组分/不同细胞群)和<strong>基因表达情况</strong>。</p><p data-tool="mdnice编辑器">开发者展示的图片就很形象了，左边图展示了把标注了不同细胞类型的单细胞数据作为先验信息(prior info)的基因信息和bulkRNA测序数据输入进贝叶斯棱镜算法，右图就是通过了棱镜算法之后慢慢地把bulkRNA测序数据中每个样本里所占的标注细胞类型的细胞百分比展示出来。</p><figure data-tool="mdnice编辑器"><img data-imgfileid="100000757" data-ratio="0.7196356275303644" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyFaw9371NSVBYgaqwC9uRudia5QFDIhRvjt5WNoHUvwX5ziadCiaotJnjNxkYiat9qFvlpEDEYQ9GjTpg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="988" src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyFaw9371NSVBYgaqwC9uRudia5QFDIhRvjt5WNoHUvwX5ziadCiaotJnjNxkYiat9qFvlpEDEYQ9GjTpg/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">那么再通俗的来说，贝叶斯棱镜法就是将<strong>单细胞数据</strong>中的<strong>不同细胞亚群信息提取出来</strong>，并<strong>按照提取出来的信息</strong>将<strong>bulkRNA数据</strong>的<strong>每一个样本</strong>也<strong>分成不同细胞亚群</strong>(不够专业但方便理解)。</p><figure data-tool="mdnice编辑器"><img data-imgfileid="100000754" data-ratio="0.4691119691119691" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyFaw9371NSVBYgaqwC9uRudVHibQ8meVGV3InqibxHsw7eiaH3Q4ic91fmR98SRpO7xTzCMPkbbqnINow/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="518" src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyFaw9371NSVBYgaqwC9uRudVHibQ8meVGV3InqibxHsw7eiaH3Q4ic91fmR98SRpO7xTzCMPkbbqnINow/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">但其实类似的算法也有很多，另外一个比较热门的算法是CIBERSOTx，但相比于CIBERSOTx, 贝叶斯棱镜法利用了贝叶斯定理(Bayes’theorem)及吉布斯采样(Gibbs sampling)减少技术性批次效应和生物学差异。</p><p data-tool="mdnice编辑器">从研究者发布的文章数据来看，贝叶斯棱镜法在估计实际细胞类型百分比得出的皮尔逊相关系数和误差均优于其他的算法。所以目前在选择算法时当然首选BayesPrism(贝叶斯棱镜法)~</p><figure data-tool="mdnice编辑器"><img data-imgfileid="100000755" data-ratio="0.31296296296296294" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyFaw9371NSVBYgaqwC9uRud5Q8TGic6EyNfbolC2S3k3IxrgWBYuCy3VGIdlxE4ibb3wUXJqwww0dew/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyFaw9371NSVBYgaqwC9uRud5Q8TGic6EyNfbolC2S3k3IxrgWBYuCy3VGIdlxE4ibb3wUXJqwww0dew/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">最终分析结果部分会得到每个细胞亚群在样本中的百分比<strong>theta值</strong>、<strong>theta.cv值</strong>(theta值的变异信息), 不同细胞类型的<strong>特异基因的count矩阵值</strong>。</p><p data-tool="mdnice编辑器">这跟之前写的Scissor算法有异曲同工之妙，一种是<strong>把bulkRNA数据的表型信息映射到单细胞数据</strong>上去，贝叶斯棱镜法是<strong>把单细胞数据的表型信息映射到bulkRNA数据</strong>上(不够专业但方便理解)。</p><p data-tool="mdnice编辑器">Scissor算法：<a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzkwMjYyMDA1OA==&amp;mid=2247484390&amp;idx=1&amp;sn=02d9ea73c2d639f0b56f67bf0ebe4fe9&amp;chksm=c0a3f513f7d47c05d90e2d84997627d0366544a518f554f3353914d3914c0b0d8f683a0b379c&amp;scene=21#wechat_redirect" textvalue="Scissor算法-从含有表型的bulkRNA数据中提取信息进而鉴别单细胞数据亚群" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">Scissor算法-从含有表型的bulkRNA数据中提取信息进而鉴别单细胞数据亚群</a></p><h3 data-tool="mdnice编辑器"><span></span>分析步骤-from github<span></span></h3><h4 data-tool="mdnice编辑器"><span></span>1、加载BayesPrism R包<span></span></h4><pre data-tool="mdnice编辑器"><span></span><code><span>library</span>(<span>"devtools"</span>);<br>install_github(<span>"Danko-Lab/BayesPrism/BayesPrism"</span>)<br>suppressWarnings(<span>library</span>(BayesPrism))<br><span>#&gt; Loading required package: snowfall</span><br><span>#&gt; Loading required package: snow</span><br><span>#&gt; Loading required package: NMF</span><br><span>#&gt; Loading required package: pkgmaker</span><br><span>#&gt; Loading required package: registry</span><br><span>#&gt; Loading required package: rngtools</span><br><span>#&gt; Loading required package: cluster</span><br><span>#&gt; NMF - BioConductor layer [OK] | Shared memory capabilities [NO: bigmemory] | Cores 71/72</span><br><span>#&gt;   To enable shared memory capabilities, try: install.extras('</span><br><span>#&gt; NMF</span><br><span>#&gt; ')</span><br></code></pre><h4 data-tool="mdnice编辑器"><span></span>2、加载示例数据<span></span></h4><pre data-tool="mdnice编辑器"><span></span><code>load(<span>"../tutorial.dat/tutorial.gbm.rdata"</span>)<br>ls()<br><span>#&gt; [1] "bk.dat"            "cell.state.labels" "cell.type.labels"  "sc.dat"</span><br></code></pre><p data-tool="mdnice编辑器">sc.dat: 单细胞数据</p><p data-tool="mdnice编辑器">cell.state.labels: 细胞状态标签</p><p data-tool="mdnice编辑器">cell.type.labels: 细胞类型标签</p><p data-tool="mdnice编辑器">bk.dat: bulk RNA数据</p><pre data-tool="mdnice编辑器"><span></span><code>dim(bk.dat)<br><span>#&gt; [1]   169 60483</span><br>head(rownames(bk.dat))<br><span>#&gt; [1] "TCGA-06-2563-01A-01R-1849-01" "TCGA-06-0749-01A-01R-1849-01" "TCGA-06-5418-01A-01R-1849-01" "TCGA-06-0211-01B-01R-1849-01" "TCGA-19-2625-01A-01R-1850-01" "TCGA-19-4065-02A-11R-2005-01"</span><br>head(colnames(bk.dat))<br><span>#&gt; [1] "ENSG00000000003.13" "ENSG00000000005.5"  "ENSG00000000419.11" "ENSG00000000457.12" "ENSG00000000460.15" "ENSG00000000938.11"</span><br><br>dim(sc.dat)<br><span>#&gt; [1] 23793 60294</span><br>head(rownames(sc.dat))<br><span>#&gt; [1] "PJ016.V3" "PJ016.V4" "PJ016.V5" "PJ016.V6" "PJ016.V7" "PJ016.V8"</span><br>head(colnames(sc.dat))<br></code></pre><p data-tool="mdnice编辑器">bulk RNA 和 scRNA 数据格式均行是样本名，基因是列名。</p><p data-tool="mdnice编辑器">bulk RNA和scRNA 要采用一样的基因信息，代码块中均为ensembl ID。</p><p data-tool="mdnice编辑器">此外，bulkRNA数据需采用count数据。若count数据不可用时，TPM，RPM，RPKM，FPKM等也是可接受的，但不要进行log转换。</p><pre data-tool="mdnice编辑器"><span></span><code>sort(table(cell.type.labels))<br><span>#&gt; cell.type.labels</span><br><span>#&gt;       tcell       oligo    pericyte endothelial     myeloid       tumor </span><br><span>#&gt;          67         160         489         492        2505       20080</span><br><br>sort(table(cell.state.labels))<br><span>#&gt; cell.state.labels</span><br><span>#&gt; PJ017-tumor-6 PJ032-tumor-5     myeloid_8 PJ032-tumor-4 PJ032-tumor-3 PJ017-tumor-5         tcell PJ032-tumor-2 PJ030-tumor-5     myeloid_7 PJ035-tumor-8 PJ017-tumor-4 PJ017-tumor-3 PJ017-tumor-2 PJ017-tumor-0 PJ017-tumor-1 PJ018-tumor-5     myeloid_6     myeloid_5 PJ035-tumor-7         oligo PJ048-tumor-8 PJ032-tumor-1 PJ032-tumor-0 PJ048-tumor-7 PJ025-tumor-9 PJ035-tumor-6 PJ048-tumor-6 PJ016-tumor-6 PJ018-tumor-4     myeloid_4 PJ048-tumor-5 PJ048-tumor-4 PJ016-tumor-5 PJ025-tumor-8 PJ048-tumor-3 PJ035-tumor-5 PJ030-tumor-4 PJ018-tumor-3 PJ016-tumor-4 PJ025-tumor-7     myeloid_3 PJ035-tumor-4     myeloid_2 PJ025-tumor-6 PJ018-tumor-2 PJ030-tumor-3 PJ016-tumor-3 PJ025-tumor-5 PJ030-tumor-2 PJ018-tumor-1 PJ035-tumor-3 PJ048-tumor-2 PJ018-tumor-0 PJ048-tumor-1 PJ035-tumor-2 PJ035-tumor-1 PJ016-tumor-2 PJ030-tumor-1      pericyte   endothelial PJ035-tumor-0 PJ025-tumor-4     myeloid_1 PJ048-tumor-0     myeloid_0 PJ030-tumor-0 PJ025-tumor-3 PJ016-tumor-1 PJ016-tumor-0 PJ025-tumor-2 PJ025-tumor-1 PJ025-tumor-0 </span><br><span>#&gt;            22            41            49            57            62            64            67            72            73            75            81            83            89           101           107           107           113           130           141           150           160           169           171           195           228           236           241           244           261           262           266           277           303           308           319           333           334           348           361           375           381           382           385           386           397           403           419           420           421           425           429           435           437           444           463           471           474           481           482           489           492           512           523           526           545           550           563           601           619           621           630           941           971</span><br></code></pre><p data-tool="mdnice编辑器"><strong>cell.type.labels代表着纳入分析的scRNA数据中不同细胞类型的数量</strong>。</p><p data-tool="mdnice编辑器"><strong>如何进行cell type分类</strong>，开发团队提供的依据/建议是：Define cell types as the cluster of cells having a sufficient number of significantly differentially expressed genes than other cell types, e.g., greater than 50 or even 100.</p><p data-tool="mdnice编辑器">简单来说就是<strong>划分出来的不同细胞类型是存在着显著的表观等差别的</strong>。</p><pre data-tool="mdnice编辑器"><span></span><code>sort(table(cell.state.labels))<br><span>#&gt; cell.state.labels</span><br><span>#&gt; PJ017-tumor-6 PJ032-tumor-5     myeloid_8 PJ032-tumor-4 PJ032-tumor-3 PJ017-tumor-5         tcell PJ032-tumor-2 PJ030-tumor-5     myeloid_7 PJ035-tumor-8 PJ017-tumor-4 PJ017-tumor-3 PJ017-tumor-2 PJ017-tumor-0 PJ017-tumor-1 PJ018-tumor-5     myeloid_6     myeloid_5 PJ035-tumor-7         oligo PJ048-tumor-8 PJ032-tumor-1 PJ032-tumor-0 PJ048-tumor-7 PJ025-tumor-9 PJ035-tumor-6 PJ048-tumor-6 PJ016-tumor-6 PJ018-tumor-4     myeloid_4 PJ048-tumor-5 PJ048-tumor-4 PJ016-tumor-5 PJ025-tumor-8 PJ048-tumor-3 PJ035-tumor-5 PJ030-tumor-4 PJ018-tumor-3 PJ016-tumor-4 PJ025-tumor-7     myeloid_3 PJ035-tumor-4     myeloid_2 PJ025-tumor-6 PJ018-tumor-2 PJ030-tumor-3 PJ016-tumor-3 PJ025-tumor-5 PJ030-tumor-2 PJ018-tumor-1 PJ035-tumor-3 PJ048-tumor-2 PJ018-tumor-0 PJ048-tumor-1 PJ035-tumor-2 PJ035-tumor-1 PJ016-tumor-2 PJ030-tumor-1      pericyte   endothelial PJ035-tumor-0 PJ025-tumor-4     myeloid_1 PJ048-tumor-0     myeloid_0 PJ030-tumor-0 PJ025-tumor-3 PJ016-tumor-1 PJ016-tumor-0 PJ025-tumor-2 PJ025-tumor-1 PJ025-tumor-0 </span><br><span>#&gt;            22            41            49            57            62            64            67            72            73            75            81            83            89           101           107           107           113           130           141           150           160           169           171           195           228           236           241           244           261           262           266           277           303           308           319           333           334           348           361           375           381           382           385           386           397           403           419           420           421           425           429           435           437           444           463           471           474           481           482           489           492           512           523           526           545           550           563           601           619           621           630           941           971</span><br></code></pre><p data-tool="mdnice编辑器"><strong>cell.state.labels代表着纳入分析的scRNA数据中不同细胞类型的细分状态</strong>。</p><p data-tool="mdnice编辑器">开发者给出的例子中对tumor和myeloid细胞进行了细分，原因是开发团队觉得肿瘤和髓系细胞中内部的不同细胞亚型具有不同的生物学作用而且很重要。</p><p data-tool="mdnice编辑器">对于使用者而言，选择什么细胞进行细分“state”亚群这就根据自身的研究而定。若不想细分也可以，后续可以把相关参数设置NULL。</p><p data-tool="mdnice编辑器">细分后定义的细胞状态亚群群的细胞数量至少需要&gt;20，严格一点需要&gt;50.</p><p data-tool="mdnice编辑器"><strong>如何进行cell state分类</strong>，开发团队提供的依据/建议是：For clusters that are too similar in transcription, we recommend treating them as cell states, which will be summed up before the final Gibbs sampling. Therefore, cell states are often suitable for cells that form a continuum on the phenotypic manifold rather than distinct clusters.</p><p data-tool="mdnice编辑器">简单来说就是<strong>对于转录特征很相似的细胞簇(表观上很相似)，但可能在功能上/生物学意义上存在一定的差别，建议设置不同的细胞状态，比如肿瘤细胞存在很强的异质性/很有探究价值等</strong>。</p><h4 data-tool="mdnice编辑器"><span></span>3、对细胞类型和细胞状态进行质量控制<span></span></h4><pre data-tool="mdnice编辑器"><span></span><code>plot.cor.phi (input=sc.dat,<br>                         input.labels=cell.state.labels,<br>                         title=<span>"cell state correlation"</span>,<br>                         <span>#specify pdf.prefix if need to output to pdf</span><br>                         <span>#pdf.prefix="gbm.cor.cs", </span><br>                         cexRow=<span>0.2</span>, cexCol=<span>0.2</span>,<br>                         margins=c(<span>2</span>,<span>2</span>)<br>                         )<br><br>plot.cor.phi (input=sc.dat, <br>                         input.labels=cell.type.labels, <br>                         title=<span>"cell type correlation"</span>,<br>                         <span>#specify pdf.prefix if need to output to pdf</span><br>                         <span>#pdf.prefix="gbm.cor.ct",</span><br>                         cexRow=<span>0.5</span>, cexCol=<span>0.5</span>,<br>                         )<br></code></pre><p data-tool="mdnice编辑器"><img data-imgfileid="100000756" data-ratio="0.5111111111111111" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyFaw9371NSVBYgaqwC9uRudtu3vG8kjfvRqHEqnMwZsiavHFBImF3fykb7kicvuQQYm88Sp3H16gcKg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyFaw9371NSVBYgaqwC9uRudtu3vG8kjfvRqHEqnMwZsiavHFBImF3fykb7kicvuQQYm88Sp3H16gcKg/640?wx_fmt=png&amp;from=appmsg">分别对细胞类型和细胞状态进行相关性分析，这个步骤是识别相似的簇，若观察到相似的簇，应考虑合并。</p><h4 data-tool="mdnice编辑器"><span></span>4、可视化scRNA数据中的异常基因<span></span></h4><pre data-tool="mdnice编辑器"><span></span><code>sc.stat &lt;- plot.scRNA.outlier(<br>  input=sc.dat, <span>#make sure the colnames are gene symbol or ENSMEBL ID </span><br>  cell.type.labels=cell.type.labels,<br>  species=<span>"hs"</span>, <span>#currently only human(hs) and mouse(mm) annotations are supported</span><br>  return.raw=<span>TRUE</span> <span>#return the data used for plotting. </span><br>  <span>#pdf.prefix="gbm.sc.stat" specify pdf.prefix if need to output to pdf</span><br>)<br><span>#&gt; EMSEMBLE IDs detected.</span><br></code></pre><p data-tool="mdnice编辑器">有很多大量表达基因，比如核糖体蛋白质基因和线粒体基因，它们在分簇的时候往往给不了有用的信息，还会干扰分簇并使结果产生偏倚，因此需要去除。</p><p data-tool="mdnice编辑器"><img data-imgfileid="100000758" data-ratio="0.625" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyFaw9371NSVBYgaqwC9uRudSaXRsfvIZanKNZ6xDKO1A4a23z2stl8RBOibhB4lmvwvHYfUBCI525w/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyFaw9371NSVBYgaqwC9uRudSaXRsfvIZanKNZ6xDKO1A4a23z2stl8RBOibhB4lmvwvHYfUBCI525w/640?wx_fmt=png&amp;from=appmsg">如图所示：核糖体蛋白质基因通常表现出高平均表达(X轴)和低细胞类型特异性得分(Y轴)。</p><h4 data-tool="mdnice编辑器"><span></span>5、可视化bulkRNA数据中的异常基因<span></span></h4><pre data-tool="mdnice编辑器"><span></span><code>bk.stat &lt;- plot.bulk.outlier(<br>  bulk.input=bk.dat,<span>#make sure the colnames are gene symbol or ENSMEBL ID </span><br>    sc.input=sc.dat, <span>#make sure the colnames are gene symbol or ENSMEBL ID </span><br>  cell.type.labels=cell.type.labels,<br>  species=<span>"hs"</span>, <span>#currently only human(hs) and mouse(mm) annotations are supported</span><br>  return.raw=<span>TRUE</span><br>  <span>#pdf.prefix="gbm.bk.stat" specify pdf.prefix if need to output to pdf</span><br>)<br><span>#&gt; EMSEMBLE IDs detected.</span><br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100000764" data-ratio="0.625" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyFaw9371NSVBYgaqwC9uRudKFtuD7xbKIsFJ5Pnib3nRu3C1JXItGP5sxZYoY6uBKM6JBKTia2KPQdg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyFaw9371NSVBYgaqwC9uRudKFtuD7xbKIsFJ5Pnib3nRu3C1JXItGP5sxZYoY6uBKM6JBKTia2KPQdg/640?wx_fmt=png&amp;from=appmsg"></figure><h4 data-tool="mdnice编辑器"><span></span>6、过滤scRNA数据的异常基因<span></span></h4><pre data-tool="mdnice编辑器"><span></span><code>sc.dat.filtered &lt;- cleanup.genes (input=sc.dat,<br>                                  input.type=<span>"count.matrix"</span>,<br>                                  species=<span>"hs"</span>, <br>                                  gene.group=c( <span>"Rb"</span>,<span>"Mrp"</span>,<span>"other_Rb"</span>,<span>"chrM"</span>,<span>"MALAT1"</span>,<span>"chrX"</span>,<span>"chrY"</span>) ,<br>                                  exp.cells=<span>5</span>)<br><span>#&gt; EMSEMBLE IDs detected.</span><br><span>#&gt; number of genes filtered in each category: </span><br><span>#&gt;       Rb      Mrp other_Rb     chrM   MALAT1     chrX     chrY </span><br><span>#&gt;       89       78     1011       37        1     2464      594 </span><br><span>#&gt; A total of  4214  genes from Rb Mrp other_Rb chrM MALAT1 chrX chrY  have been excluded </span><br><span>#&gt; A total of  24343  gene expressed in fewer than  5  cells have been excluded</span><br></code></pre><p data-tool="mdnice编辑器">去除设置的细胞群基因。</p><h4 data-tool="mdnice编辑器"><span></span>7、检查基因与不同基因集之间的相关性<span></span></h4><pre data-tool="mdnice编辑器"><span></span><code><span>#note this function only works for human data. For other species, you are advised to make plots by yourself.</span><br>plot.bulk.vs.sc (sc.input = sc.dat.filtered,<br>                            bulk.input = bk.dat<br>                            <span>#pdf.prefix="gbm.bk.vs.sc" specify pdf.prefix if need to output to pdf</span><br>)<br><span>#&gt; EMSEMBLE IDs detected.</span><br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100000761" data-ratio="0.625" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyFaw9371NSVBYgaqwC9uRudsToqlniabmbImY1bBMEvdcSvH1ibVHfyWcaLq88vnFs4Qns04eGqfJ6Q/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyFaw9371NSVBYgaqwC9uRudsToqlniabmbImY1bBMEvdcSvH1ibVHfyWcaLq88vnFs4Qns04eGqfJ6Q/640?wx_fmt=png&amp;from=appmsg"></figure><h4 data-tool="mdnice编辑器"><span></span>8、提取蛋白编码基因<span></span></h4><pre data-tool="mdnice编辑器"><span></span><code>sc.dat.filtered.pc &lt;-  select.gene.type (sc.dat.filtered,<br>                                        gene.type = <span>"protein_coding"</span>)<br><span>#&gt; EMSEMBLE IDs detected.</span><br><span>#&gt; number of genes retained in each category: </span><br><span>#&gt; </span><br><span>#&gt; protein_coding </span><br><span>#&gt;          16148</span><br></code></pre><h4 data-tool="mdnice编辑器"><span></span>9、构建BayesPrism模型<span></span></h4><pre data-tool="mdnice编辑器"><span></span><code>myPrism &lt;- new.prism(<br>  reference=sc.dat.filtered.pc, <br>  mixture=bk.dat,<br>  input.type=<span>"count.matrix"</span>, <br>  cell.type.labels = cell.type.labels, <br>  cell.state.labels = cell.state.labels,<br>  key=<span>"tumor"</span>, <br>  outlier.cut=<span>0.01</span>,<br>  outlier.fraction=<span>0.1</span>,<br>)<br><span>#&gt; number of cells in each cell state </span><br><span>#&gt; cell.state.labels</span><br><span>#&gt; PJ017-tumor-6 PJ032-tumor-5     myeloid_8 PJ032-tumor-4 PJ032-tumor-3 PJ017-tumor-5         tcell PJ032-tumor-2 PJ030-tumor-5     myeloid_7 PJ035-tumor-8 PJ017-tumor-4 PJ017-tumor-3 PJ017-tumor-2 PJ017-tumor-0 PJ017-tumor-1 PJ018-tumor-5     myeloid_6     myeloid_5 PJ035-tumor-7         oligo PJ048-tumor-8 PJ032-tumor-1 PJ032-tumor-0 PJ048-tumor-7 PJ025-tumor-9 PJ035-tumor-6 PJ048-tumor-6 PJ016-tumor-6 PJ018-tumor-4     myeloid_4 PJ048-tumor-5 PJ048-tumor-4 PJ016-tumor-5 PJ025-tumor-8 PJ048-tumor-3 PJ035-tumor-5 PJ030-tumor-4 PJ018-tumor-3 PJ016-tumor-4 PJ025-tumor-7     myeloid_3 PJ035-tumor-4     myeloid_2 PJ025-tumor-6 PJ018-tumor-2 PJ030-tumor-3 PJ016-tumor-3 PJ025-tumor-5 PJ030-tumor-2 PJ018-tumor-1 PJ035-tumor-3 PJ048-tumor-2 PJ018-tumor-0 PJ048-tumor-1 PJ035-tumor-2 PJ035-tumor-1 PJ016-tumor-2 PJ030-tumor-1      pericyte   endothelial PJ035-tumor-0 PJ025-tumor-4     myeloid_1 PJ048-tumor-0     myeloid_0 PJ030-tumor-0 PJ025-tumor-3 PJ016-tumor-1 PJ016-tumor-0 PJ025-tumor-2 PJ025-tumor-1 PJ025-tumor-0 </span><br><span>#&gt;            22            41            49            57            62            64            67            72            73            75            81            83            89           101           107           107           113           130           141           150           160           169           171           195           228           236           241           244           261           262           266           277           303           308           319           333           334           348           361           375           381           382           385           386           397           403           419           420           421           425           429           435           437           444           463           471           474           481           482           489           492           512           523           526           545           550           563           601           619           621           630           941           971 </span><br><span>#&gt; Number of outlier genes filtered from mixture = 6 </span><br><span>#&gt; Aligning reference and mixture... </span><br><span>#&gt; Nornalizing reference...</span><br><br><span>#Note that outlier.cut and outlier.fraction=0.1 filter genes in X whose expression fraction is greater than outlier.cut (Default=0.01) in more than outlier.fraction (Default=0.1) of bulk data. </span><br><span>#Typically for dataset with reasonable quality control, very few genes will be filtered. </span><br><span>#Removal of outlier genes will ensure that the inference will not be dominated by outliers, which sometimes may be resulted from poor QC in mapping.</span><br></code></pre><p data-tool="mdnice编辑器">input.type 除了常规的设置"count.matrix"以外，还有另一种设置类型是“GEP”，开发者给出的解释是：the other option for input.type is "GEP" (gene expression profile) which is a cell state by gene matrix. This option is used when using reference derived from other assays, such as sorted bulk data. 这个GEP暂时没有理解，求高手指点~</p><p data-tool="mdnice编辑器">参数key是cell.type.label 中对应于使用者自行命名的恶性细胞的名称，比如在cell.type.label中命名为tumor，那么key参数后面也可以写tumor。如果没有恶性细胞，也可以设置为NULL，在这种情况下，所有类型的细胞将被会被算法平等对待。</p><h4 data-tool="mdnice编辑器"><span></span>10、运行BayesPrism模型<span></span></h4><pre data-tool="mdnice编辑器"><span></span><code>bp.res &lt;- run.prism(prism = myPrism, n.cores=<span>50</span>)<br></code></pre><p data-tool="mdnice编辑器">如果电脑配置比较弱，建议把n.cores降低一点。笔者就对自己的电脑过于自信而不降低n.cores值，导致运行这个算法的时候电脑被干死机过...</p><h4 data-tool="mdnice编辑器"><span></span>11、提取细胞比例theta值<span></span></h4><pre data-tool="mdnice编辑器"><span></span><code><span># extract posterior mean of cell type fraction theta</span><br>theta &lt;- get.fraction (bp=bp.res,<br>            which.theta=<span>"final"</span>,<br>            state.or.type=<span>"type"</span>)<br><br>head(theta)<br><span>#&gt;                                  tumor    myeloid     pericyte endothelial        tcell        oligo</span><br><span>#&gt; TCGA-06-2563-01A-01R-1849-01 0.8392297 0.04329259 2.999022e-02  0.07528272 6.474488e-04 0.0115573149</span><br><span>#&gt; TCGA-06-0749-01A-01R-1849-01 0.7090654 0.17001073 8.995526e-07  0.01275526 1.179331e-06 0.1081665709</span><br><span>#&gt; TCGA-06-5418-01A-01R-1849-01 0.8625322 0.09839143 9.729268e-03  0.02416954 7.039913e-07 0.0051768589</span><br><span>#&gt; TCGA-06-0211-01B-01R-1849-01 0.8893449 0.04482991 1.131622e-02  0.05435490 2.508238e-06 0.0001515524</span><br><span>#&gt; TCGA-19-2625-01A-01R-1850-01 0.9406438 0.03546026 1.932740e-03  0.01309753 4.535897e-06 0.0088610997</span><br><span>#&gt; TCGA-19-4065-02A-11R-2005-01 0.6763166 0.08374439 1.849921e-01  0.01918132 3.541126e-04 0.0354114398</span><br></code></pre><p data-tool="mdnice编辑器">这一步就是提取每种细胞亚群占每一个样本的比例。可以自行计算，同一个样本的不同细胞类型之和为1。</p><h4 data-tool="mdnice编辑器"><span></span>12、提取细胞比例theta值的变异分数<span></span></h4><pre data-tool="mdnice编辑器"><span></span><code><span># extract coefficient of variation (CV) of cell type fraction</span><br>theta.cv &lt;- bp.res@posterior.theta_f@theta.cv<br><br>head(theta.cv)<br><span>#&gt;                                     tumor      myeloid     pericyte endothelial      tcell       oligo</span><br><span>#&gt; TCGA-06-2563-01A-01R-1849-01 0.0001722829 0.0016025331 0.0026568433 0.001853843 0.05452368 0.005606057</span><br><span>#&gt; TCGA-06-0749-01A-01R-1849-01 0.0002333853 0.0006859332 0.8109050326 0.005683516 0.74729658 0.001276784</span><br><span>#&gt; TCGA-06-5418-01A-01R-1849-01 0.0001601128 0.0009389782 0.0070872942 0.004131925 0.83670176 0.011362855</span><br><span>#&gt; TCGA-06-0211-01B-01R-1849-01 0.0001175529 0.0014412303 0.0055064706 0.001673105 0.60905434 0.280609474</span><br><span>#&gt; TCGA-19-2625-01A-01R-1850-01 0.0001327408 0.0023661566 0.0335184780 0.006286823 0.73453327 0.006138184</span><br><span>#&gt; TCGA-19-4065-02A-11R-2005-01 0.0002862600 0.0012227981 0.0009100677 0.005660069 0.06371147 0.002243368</span><br></code></pre><p data-tool="mdnice编辑器">这里的变异系数(CV)反应了后验分布的变异程度，theta值越高CV值就越低。</p><p data-tool="mdnice编辑器">在执行排名相关的统计分析时，例如斯皮尔曼等级相关系数，研究者可以选择在变异系数低于某个阈值（例如小于0.1）的情况下将 theta 设置为零(更高的排序)。这种做法可以帮助屏蔽那些在后验分布中变异性较小、较为确定的测量值，以便更清晰地聚焦于那些变异性更高的数据点。</p><h4 data-tool="mdnice编辑器"><span></span>13、提取特定细胞类型基因count矩阵的后验平均值<span></span></h4><pre data-tool="mdnice编辑器"><span></span><code><span># extract posterior mean of cell type-specific gene expression count matrix Z  </span><br>Z.tumor &lt;- get.exp (bp=bp.res,<br>                          state.or.type=<span>"type"</span>,<br>                          cell.name=<span>"tumor"</span>)<br><br>head(t(Z.tumor[<span>1</span>:<span>5</span>,]))<br><span>#&gt;                    TCGA-06-2563-01A-01R-1849-01 TCGA-06-0749-01A-01R-1849-01 TCGA-06-5418-01A-01R-1849-01 TCGA-06-0211-01B-01R-1849-01 TCGA-19-2625-01A-01R-1850-01</span><br><span>#&gt; ENSG00000130876.10                       55.980                      444.980                        9.000                       56.996                       15.996</span><br><span>#&gt; ENSG00000165244.6                       206.344                       38.096                      291.872                      530.732                      507.788</span><br><span>#&gt; ENSG00000173597.7                        17.100                        6.588                       21.804                       28.744                       10.800</span><br><span>#&gt; ENSG00000158022.6                         0.000                        0.476                        2.616                        0.960                        7.824</span><br><span>#&gt; ENSG00000167220.10                     2273.824                     1060.976                     2775.180                     2368.016                     2084.620</span><br><span>#&gt; ENSG00000126106.12                      678.468                      685.948                      481.000                      698.888                     1296.080</span><br></code></pre><p data-tool="mdnice编辑器">14、保存数据</p><pre data-tool="mdnice编辑器"><span></span><code><span>#save the result</span><br>save(bp.res, file=<span>"bp.res.rdata"</span>)<br></code></pre><p data-tool="mdnice编辑器">后续就可以根据不同细胞比例的theta值，CV值，或者特异基因数据进行不同细胞比率的图形绘制，或者跟生存数据联合分析等</p><figure data-tool="mdnice编辑器"><img data-imgfileid="100000759" data-ratio="0.575925925925926" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyFaw9371NSVBYgaqwC9uRudFDQhmtRBhJkGeNBXSDXvaTlzA9A33u0sR6vQRCib5b4SZE7BqCUwV6w/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyFaw9371NSVBYgaqwC9uRudFDQhmtRBhJkGeNBXSDXvaTlzA9A33u0sR6vQRCib5b4SZE7BqCUwV6w/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">除了上述的个性化分析以外，开发者内置了基于BayesPrism的恶性细胞表达嵌入学习模块(<strong>对恶性细胞进行非负矩阵分解分簇</strong>)</p><p data-tool="mdnice编辑器">这块内容在分析前<strong>需</strong>对BayesPrism模型<strong>设置参数key</strong>！</p><h4 data-tool="mdnice编辑器"><span></span>15、提取信息<span></span></h4><pre data-tool="mdnice编辑器"><span></span><code>Z.tum.norm &lt;- t(bp.res@reference.update@psi)<br>estim.Z.tum.norm &lt;- nmf(Z.tum.norm, rank=<span>2</span>:<span>12</span>, seed=<span>123456</span>)<br>plot(estim.Z.tum.norm)<br>consensusmap(estim.Z.tum.norm, labCol=<span>NA</span>, labRow=<span>NA</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100000760" data-ratio="0.6884353741496598" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyFaw9371NSVBYgaqwC9uRudfJ0ia5ODYmHUxwmEq1s8Ft5rCwibucmxo0X7WAjXlsWySXWf3QV8q8Gg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="735" src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyFaw9371NSVBYgaqwC9uRudfJ0ia5ODYmHUxwmEq1s8Ft5rCwibucmxo0X7WAjXlsWySXWf3QV8q8Gg/640?wx_fmt=png&amp;from=appmsg"></figure><h4 data-tool="mdnice编辑器"><span></span>16、运行NMF<span></span></h4><pre data-tool="mdnice编辑器"><span></span><code>ebd.res &lt;- learn.embedding.nmf (bp = bp.res, <br>                                <span>#bp.res needs to be run under the tumor mode.</span><br>                                K = <span>2</span>,<br>                                cycle = <span>20</span>, <span>#EM typically converges in 50 cycles. set to 20 for demo</span><br>                                compute.elbo = <span>T</span>)<br>str(ebd.res)<br><span>#&gt; List of 4</span><br><span>#&gt; $ eta_prior: num [1:5, 1:16145] 6.91e-07 2.16e-06 1.72e-05 4.24e-07 2.61e-06 ...</span><br> <span>#&gt;   ..- attr(*, "dimnames")=List of 2</span><br> <span>#&gt;   .. ..$ : chr [1:5] "program-1" "program-2" "program-3" "program-4" ...</span><br> <span>#&gt;   .. ..$ : chr [1:16145] "ENSG00000130876.10" "ENSG00000165244.6" "ENSG00000173597.7" "ENSG00000158022.6" ...</span><br> <span>#&gt;  $ eta_post : num [1:5, 1:16145] 7.88e-07 2.37e-06 1.34e-05 1.11e-07 1.67e-06 ...</span><br> <span>#&gt;   ..- attr(*, "dimnames")=List of 2</span><br> <span>#&gt;   .. ..$ : chr [1:5] "program-1" "program-2" "program-3" "program-4" ...</span><br> <span>#&gt;   .. ..$ : chr [1:16145] "ENSG00000130876.10" "ENSG00000165244.6" "ENSG00000173597.7" "ENSG00000158022.6" ...</span><br> <span>#&gt;  $ omega    : num [1:169, 1:5] 0.701906 0.000555 0.324483 0.627062 0.172039 ...</span><br> <span>#&gt;   ..- attr(*, "dimnames")=List of 2</span><br> <span>#&gt;   .. ..$ : chr [1:169] "TCGA-06-2563-01A-01R-1849-01" "TCGA-06-0749-01A-01R-1849-01" "TCGA-06-5418-01A-01R-1849-01" "TCGA-06-0211-01B-01R-184</span><br> <span>#&gt;   .. ..$ : chr [1:5] "program-1" "program-2" "program-3" "program-4" ...</span><br> <span>#&gt;  $ elbo     : num [1:21] Inf 1.06e+11 1.06e+11 1.06e+11 1.06e+11 ...</span><br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100000763" data-ratio="0.31666666666666665" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyFaw9371NSVBYgaqwC9uRudOE7SmlPyjM3ote3oiafYpV593htdSClQTaY9UXRbRx8BlBX9E9Z3GSQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyFaw9371NSVBYgaqwC9uRudOE7SmlPyjM3ote3oiafYpV593htdSClQTaY9UXRbRx8BlBX9E9Z3GSQ/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">大概的对应解释是：</p><p data-tool="mdnice编辑器">● eta_prior（先验均值矩阵）：这是一个 K × G 的矩阵，用来表示 eta 的先验均值输入。在调用 learn.embedding.nmf 函数时，可以从非负矩阵分解（NMF）得到 eta_prior。简单来说，这个矩阵包含了在开始模型估计之前，我们对每个基因表达程序的预期水平的先验知识。</p><p data-tool="mdnice编辑器">● eta_post（后验估计矩阵）：这也是一个 K × G 的矩阵，表示通过期望最大化（EM）算法计算得到的 eta 的最大后验（MAP）估计值。这个矩阵更新了我们对每个基因表达程序强度的估计，基于实际观测数据进行调整。</p><p data-tool="mdnice编辑器">● omega（后验均值矩阵）：这是一个 N × K 的矩阵，用于表示在 eta_post（即 eta 的后验估计值）下 omega 的后验均值，也就是每个批量样本中恶性基因程序的权重。这个矩阵反映了在具体样本中，不同恶性基因表达程序的相对表达强度。</p><p data-tool="mdnice编辑器">● elbo（证据下界向量）：如果设置 compute.elbo = TRUE，则这是一个数值向量，表示每次迭代计算的证据下界（ELBO）。ELBO是一个在统计模型中用来衡量模型拟合优度的指标，通常用于贝叶斯框架中评估模型的表现。</p><p data-tool="mdnice编辑器">得到了不同簇的数据之后，后续还可以进行富集分析观察每个簇中的通路情况~</p><p data-tool="mdnice编辑器"><img data-imgfileid="100000786" data-ratio="1.1086443079829948" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyFaw9371NSVBYgaqwC9uRud3hBpTAZIFNJHZPLDD3WlsguB56K7TMlxf38XXIjMrVZtwHbvpuiaB4A/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="2117" src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyFaw9371NSVBYgaqwC9uRud3hBpTAZIFNJHZPLDD3WlsguB56K7TMlxf38XXIjMrVZtwHbvpuiaB4A/640?wx_fmt=png&amp;from=appmsg"></p><p data-tool="mdnice编辑器"><br></p><p data-tool="mdnice编辑器"><strong>参考资料：</strong></p><p data-tool="mdnice编辑器">1、https://www.bayesprism.org/pages/tutorial_deconvolution (网站上还可以对数据进行在线分析哦~）</p><p data-tool="mdnice编辑器">2、https://github.com/Danko-Lab/BayesPrism/tree/main (github)</p><p data-tool="mdnice编辑器">3、Cell type and gene expression deconvolution with BayesPrism enables Bayesian integrative analysis across bulk and single-cell RNA sequencing in oncology. Nat Cancer. 2022 Apr;3(4):505-517.</p><p data-tool="mdnice编辑器">4、生信技能树推文：</p><p data-tool="mdnice编辑器"><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzUzMTEwODk0Ng==&amp;mid=2247517328&amp;idx=1&amp;sn=57e4389c49c008f73db70701b2cba442&amp;chksm=fa4541adcd32c8bb40c5fee5ecea97b9ec5e84ed8b504dd1521abfd89176f2044980d10465b8&amp;scene=21#wechat_redirect" textvalue="利用单细胞数据对bulk进行反卷积" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">利用单细胞数据对bulk进行反卷积</a> </p><p data-tool="mdnice编辑器"><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzUzMTEwODk0Ng==&amp;mid=2247509903&amp;idx=1&amp;sn=33b077e4d235cb512597860fedc451e9&amp;chksm=fa4562b2cd32eba4e63a9cc04adef874bfbc24709377ddd33b4c981650cd629d012c785968e3&amp;scene=21#wechat_redirect" textvalue="这个bulk RNA-seq反卷积工具，你可能还不知道" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">这个bulk RNA-seq反卷积工具，你可能还不知道</a><br></p><p data-tool="mdnice编辑器"><br></p><p data-tool="mdnice编辑器"><strong>致谢</strong>：感谢曾老师，小洁老师以及生信技能树团队全体成员。</p><p data-tool="mdnice编辑器"><strong>注</strong>：若对内容有疑惑或者有发现明确错误的朋友，请联系后台(欢迎交流)。</p><span>- END -</span></section><p><br></p><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/Rj8PL6wJqwWhsmLG8tx89w",target="_blank" rel="noopener noreferrer">原文链接</a>
