---
title: "CellRanger ARC—单细胞RNAseq和ATAC联合分析套件"
date: 2024-07-16T15:18:49Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Acdemic"]
---
CellRanger ARC—单细胞RNAseq和ATAC联合分析套件 by 生信菜鸟团
------
<div><section data-md5="d93a6" act="xmy"><section data-md5="d93a6"><section data-md5="d93a6"><section data-md5="d93a6"><section data-md5="d93a6"><section data-md5="d93a6"><section data-md5="d93a6"><p data-md5="d93a6"><strong data-md5="d93a6">工欲善其事必先利其器</strong></p></section></section></section></section></section></section></section><p><br></p><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><h2 data-tool="mdnice编辑器"><span></span><span>Cell Ranger ARC</span><span></span></h2><p data-tool="mdnice编辑器">CellRanger ARC 是10x Genomics 专为单细胞多组学数据分析设计，同时分析单细胞RNA测序（scRNA-seq）和单细胞ATAC测序（scATAC-seq）数据的一组分析pipline 。</p><p data-tool="mdnice编辑器">主要包含以下几个方面：</p><ol data-tool="mdnice编辑器"><li><section><strong>数据预处理</strong></section></li></ol><ul><li><section>将原始的测序数据转换为可分析的数据格式。</section></li><li><section>包括对FASTQ文件的读取、质量控制、去重复和对齐等步骤。</section></li></ul><li><section><strong>细胞检测和定量</strong></section></li><ul><li><section>自动检测单个细胞，并对每个细胞进行转录组和染色质可及性数据的定量分析。</section></li><li><section>生成包含基因表达和染色质状态的矩阵文件。</section></li></ul><li><section><strong>多组学整合</strong></section></li><ul><li><section>将scRNA-seq和scATAC-seq数据整合在一起，进行联合分析。</section></li><li><section>可以揭示基因表达和染色质可及性之间的关系。</section></li></ul><li><section><strong>可视化和下游分析</strong></section></li><ul><li><section>提供多种数据可视化选项，如UMAP或t-SNE图，用于展示细胞的聚类和特征。</section></li><li><section>支持下游的生物信息学分析，如差异表达分析、轨迹分析等。</section></li></ul><p data-tool="mdnice编辑器">官网：</p><ul data-tool="mdnice编辑器"><li><section>https://support.10xgenomics.com/single-cell-multiome-atac-gex/software/overview/welcome</section></li></ul><h2 data-tool="mdnice编辑器"><span></span><span>如何安装</span><span></span></h2><p data-tool="mdnice编辑器">10X的软件安装都很简单，官网下载解压即可使用。</p><p data-tool="mdnice编辑器">官网下载链接：</p><ul data-tool="mdnice编辑器"><li><section>https://support.10xgenomics.com/single-cell-multiome-atac-gex/software/downloads/latest</section></li></ul><p data-tool="mdnice编辑器">首先是需要注册一下信息</p><figure data-tool="mdnice编辑器"><img data-imgfileid="100041377" data-ratio="0.8268518518518518" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losib1SRPYAut2ETxJe19eYndA4Dy614PoibvibGc3haKhLiaRvRW0K365r7Jnou88Tkibqfy6nSjUvEumvA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losib1SRPYAut2ETxJe19eYndA4Dy614PoibvibGc3haKhLiaRvRW0K365r7Jnou88Tkibqfy6nSjUvEumvA/640?wx_fmt=png&amp;from=appmsg"><figcaption>申请</figcaption></figure><p data-tool="mdnice编辑器">填写信息后，可跳转到下载界面</p><figure data-tool="mdnice编辑器"><img data-imgfileid="100041376" data-ratio="0.7833333333333333" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losib1SRPYAut2ETxJe19eYndAmwEK8VpjURtN3DY6bg2HhAOROX5ymzf8q20EsOW3QFhXibqyG48MVfg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losib1SRPYAut2ETxJe19eYndAmwEK8VpjURtN3DY6bg2HhAOROX5ymzf8q20EsOW3QFhXibqyG48MVfg/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">下载软件及所需的参考文件</p><pre data-tool="mdnice编辑器"><span></span><code><span>#</span><span><span># 软件下载</span></span><br>wget -O cellranger-arc-2.0.2.tar.gz "https://cf.10xgenomics.com/releases/cell-arc/cellranger-arc-2.0.2.tar.gz?Expires=1721143948&amp;Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiaHR0cHM6Ly9jZi4xMHhnZW5vbWljcy5jb20vcmVsZWFzZXMvY2VsbC1hcmMvY2VsbHJhbmdlci1hcmMtMi4wLjIudGFyLmd6IiwiQ29uZGl0aW9uIjp7IkRhdGVMZXNzVGhhbiI6eyJBV1M6RXBvY2hUaW1lIjoxNzIxMTQzOTQ4fX19XX0_&amp;Signature=Lx8Sg8mwwlYvUtxVwNl0LmxwtSEVpZy~W6XO5Y~1c~4O7P8zyzNcijN0H~FzE4P10CTc~btSMa5RpaNQN~AqLed~CxDikYWvuMoWb9ffMm3dxu4i1M31aUQtI4DzbGpeOSKl4--VL6qpDNyrSGwiS-T-72C14pdEKEur8m1RPNq8pfxHGc-5eRvvyliJrDN49zJRR1Dnr0LH5valgNTCqzCio5-UApkdl9JLNa07-kvKkh4T7hUecoEpoDGzugOuejJGU1kJzmZoOlQVqwoGUepeQUG6wrqD0~CYj19-tyPXxWCUKPjVZBlk8ERY2trWcr2Z6QB4qqAui~EJguWtwg__&amp;Key-Pair-Id=APKAI7S6A5RYOXBWRPDA"<br><span>#</span><span><span># 软件大小：</span></span><br>699M 7月  10 23:26 cellranger-arc-2.0.2.tar.gz<br><span>#</span><span><span># 解压即可使用</span></span><br>tar -xf cellranger-arc-2.0.2.tar.gz<br><span><br><br>#</span><span><span>#人参考基因组下载</span></span><br>wget -c https://cf.10xgenomics.com/supp/cell-arc/refdata-cellranger-arc-GRCh38-2020-A-2.0.0.tar.gz<br><span><br>#</span><span><span>#文件大小</span></span><br>14G 5月   3  2021 refdata-cellranger-arc-GRCh38-2020-A-2.0.0.tar.gz<br><span>#</span><span><span>#解压后21G</span></span><br>21G     refdata-cellranger-arc-GRCh38-2020-A-2.0.0<br><span><br>#</span><span><span>#小鼠参考基因组下载</span></span><br>wget -c https://cf.10xgenomics.com/supp/cell-arc/refdata-cellranger-arc-mm10-2020-A-2.0.0.tar.gz<br><span>#</span><span><span>#文件大小</span></span><br>13G 6月  21 01:04 refdata-cellranger-arc-mm10-2020-A-2.0.0.tar.gz<br><span>#</span><span><span>#解压后19G</span></span><br>19G     refdata-cellranger-arc-mm10-2020-A-2.0.0<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100041380" data-ratio="0.40370370370370373" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losib1SRPYAut2ETxJe19eYndAjcjOTx96PGwpibzwIDFClpLKl1SEgJEialj2uSxrTVGrT9mZ2GerjGsg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losib1SRPYAut2ETxJe19eYndAjcjOTx96PGwpibzwIDFClpLKl1SEgJEialj2uSxrTVGrT9mZ2GerjGsg/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">如果需要自定义参考基因组文件，可参考：</p><ul data-tool="mdnice编辑器"><li><section>https://support.10xgenomics.com/single-cell-multiome-atac-gex/software/pipelines/latest/advanced/references</section></li></ul><h2 data-tool="mdnice编辑器"><span></span><span>功能简述</span><span></span></h2><figure data-tool="mdnice编辑器"><img data-imgfileid="100041379" data-ratio="0.3287037037037037" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losib1SRPYAut2ETxJe19eYndAWTqDVia9uSofeMNric7OOD3HBZ6NcBTWmIUyD8MrTWRQvrPh5Q5iaIiajg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losib1SRPYAut2ETxJe19eYndAWTqDVia9uSofeMNric7OOD3HBZ6NcBTWmIUyD8MrTWRQvrPh5Q5iaIiajg/640?wx_fmt=png&amp;from=appmsg"><figcaption>功能模块</figcaption></figure><h2 data-tool="mdnice编辑器"><span></span><span>最小化使用</span><span></span></h2><p data-tool="mdnice编辑器">通常我们用到的是count定量</p><pre data-tool="mdnice编辑器"><span></span><code>cellranger-arc count --id=sampleid \<br>                    --reference=/opt/refdata-cellranger-arc-GRCh38-2020-A-2.0.0 \<br>                    --libraries=/home/jdoe/runs/libraries.csv \<br>                    --localcores=16 \<br>                    --localmem=64<br><br>--id #运行ID符，同时也是pipline的输出结果目录。可以使用字母、数字、下划线和连字符任意定义，保证唯一即可且不超64个字符即可，一般使用样本名来作为ID<br>--reference #参考基因组文件<br>--libraries # 一个3列的CSV文件，文件声明了输入ATAC和GEX FASTQ的fastq文件路径、样本名和库类型<br>--localcores #指定调用线程数，默认调用系统所有可用内核<br>--localmem #指定调用的内存大小 ，默认调用系统90%的可用内存<br></code></pre><p data-tool="mdnice编辑器">需要特别注意的是CSV输入文件的格式，如果格式有误，程序则无法运行</p><p data-tool="mdnice编辑器">文件包含3列</p><ul data-tool="mdnice编辑器"><li><section>fastq文件路径</section></li><li><section>样本名：fastq文件的前缀</section></li><li><section>数据类型：其中，RNAseq数据是<code>Gene Expression</code>   ATAC数据是 <code>Chromatin Accessibility</code></section></li></ul><pre data-tool="mdnice编辑器"><span></span><code>fastqs,sample,library_type #固定格式，不能修改<br>/home/jdoe/runs/HNGEXSQXXX/outs/fastq_path,example,Gene Expression #按需修改<br>/home/jdoe/runs/HNATACSQXX/outs/fastq_path,example,Chromatin Accessibility #按需修改<br></code></pre><h2 data-tool="mdnice编辑器"><span></span><span>一个实例</span><span></span></h2><p data-tool="mdnice编辑器">了解了基本用法，我们来运行一个实例数据看看。<span>使用的数据来自于一项关于人类免疫缺陷病毒（HIV）潜伏期逆转的单细胞多组学分析研究，研究者采用了scRNA-seq和scATAC-seq方法，同时分析了约125,000个经过三种不同潜伏期逆转剂（LRAs）激活的潜伏感染的初级CD4+ T细胞的转录组和表观基因组特征。研究题目是 “<strong>Integrated Single-cell Multiomic Analysis of HIV Latency Reversal Reveals Novel Regulators of Viral Reactivation</strong>” 。数据已上传GEO数据库，编号 <strong>GSE210146</strong> （https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE210146）。</span></p><h3 data-tool="mdnice编辑器"><span></span><span>下载数据和拆分</span><span></span></h3><p data-tool="mdnice编辑器">数据集我们从ENA数据库下载：</p><ul data-tool="mdnice编辑器"><li><section><span>PRJNA864047: </span>https://www.ebi.ac.uk/ena/browser/view/PRJNA864047</section></li></ul><pre data-tool="mdnice编辑器"><span></span><code>cat filereport_read_run_PRJNA864047_tsv.txt |awk -F "\t" -v OFS="\t" 'NR&gt;1{print $3,$8,$17,$19}' |grep "DMSO"|cut -f 3 &gt; down_ascp.txt<br><span>#</span><span><span># ascp 下载</span></span><br>cat ../a_spinfo/down_ascp.txt |while read id;do(ascp -QT -l 100m -P33001 -i ~/miniconda3/envs/kingfisher/etc/asperaweb_id_dsa.openssh era-fasp@${id} ./);done 1&gt;ascp.log 2&gt;&amp;1<br></code></pre><p data-tool="mdnice编辑器">下载文件大小：</p><pre data-tool="mdnice编辑器"><span></span><code>$cat down_ascp.txt<br>fasp.sra.ebi.ac.uk:/vol1/srr/SRR207/010/SRR20724710/SRR20724710.lite<br>fasp.sra.ebi.ac.uk:/vol1/srr/SRR207/002/SRR20724702/SRR20724702.lite<br>fasp.sra.ebi.ac.uk:/vol1/srr/SRR207/001/SRR20724701/SRR20724701.lite<br>fasp.sra.ebi.ac.uk:/vol1/srr/SRR207/009/SRR20724709/SRR20724709.lite<br><br>3.8G 7月  11 00:29 SRR20724710.lite<br>3.4G 7月  11 01:22 SRR20724702.lite<br>3.4G 7月  11 01:45 SRR20724701.lite<br>3.5G 7月  11 12:01 SRR20724709.lite<br></code></pre><p data-tool="mdnice编辑器">拆分数据</p><pre data-tool="mdnice编辑器"><span></span><code>ls SRR*.lite | while read id;do(fasterq-dump -O ~/scRNA/PRJNA692722_atac/b_rawdata/fq_data/ --split-files -e 10 --include-technical ${id} &amp;&amp; pigz -p 10 ~/scRNA/PRJNA692722_atac/b_rawdata/fq_data/${id}_*.fastq);done 1&gt;split.log 2&gt;&amp;1<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100041378" data-ratio="0.07962962962962963" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losib1SRPYAut2ETxJe19eYndA4EaVBcI5AkTPAf4lHC9PMichiaPvJ0BuYCkxDvdpVJJ7ibiaRCNEPPVo7A/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losib1SRPYAut2ETxJe19eYndA4EaVBcI5AkTPAf4lHC9PMichiaPvJ0BuYCkxDvdpVJJ7ibiaRCNEPPVo7A/640?wx_fmt=png&amp;from=appmsg"><figcaption>拆分后文件</figcaption></figure><p data-tool="mdnice编辑器">可以看到每个SRA文件都拆分出来4个fastq文件</p><h3 data-tool="mdnice编辑器"><span></span><span>文件重命名使其符合cellranger-arc 输入要求</span><span></span></h3><p data-tool="mdnice编辑器">这一步也需要特别注意一下，文件命名不要搞错。</p><p data-tool="mdnice编辑器">首先我们要知道的是cellranger-arc的文件命名规则</p><p data-tool="mdnice编辑器">fastq文件命名同10X cellranger的格式要求即：<code>[Sample Name]_S[Number]_L00[Lane Number]_[Read Type]_001.fastq.gz</code></p><p data-tool="mdnice编辑器"><strong>GEX FASTQs</strong></p><p data-tool="mdnice编辑器">对应RNAseq数据，我们已经熟知，对于 I1、I2、R1、R2 四个文件，我们只需要确定R1 ,R2 作为输入即可正常运行</p><ul data-tool="mdnice编辑器"><li><section><code>I1</code>: Dual index i7 read (optional)</section></li><li><section><code>I2</code>: Dual index i5 read (optional)</section></li><li><section><code>R1</code>: Read 1</section></li><li><section><code>R2</code>: Read 2</section></li></ul><p data-tool="mdnice编辑器"><strong>ATAC FASTQs</strong></p><p data-tool="mdnice编辑器">有区别的一点是ATAC的数据，其需要输入的4个文件对应的命名为:</p><ul data-tool="mdnice编辑器"><li><section><code>I1</code>: Dual index i7 read (optional)</section></li><li><section><code>R1</code>: Read 1</section></li><li><section><code>R2</code>: Dual index i5 read</section></li><li><section><code>R3</code>: Read 2</section></li></ul><p data-tool="mdnice编辑器">或者</p><ul data-tool="mdnice编辑器"><li><section><code>I1</code>: Dual index i7 read (optional)</section></li><li><section><code>R1</code>: Read 1</section></li><li><section><code>I2</code>: Dual index i5 read</section></li><li><section><code>R2</code>: Read 2</section></li></ul><p data-tool="mdnice编辑器">这时候在文件名重名的时候就要特别区分，我们拆分出来的1，2，3，4四个文件，分别对应的是哪个。</p><h4 data-tool="mdnice编辑器"><span></span><span>重命名</span><span></span></h4><p data-tool="mdnice编辑器">这里贴一下重命名的脚本。由于示例所用的RNAseq的数据和ATAC数据命名规则还不一致，所以我们需要修改这个脚本分别来命名。</p><p data-tool="mdnice编辑器"><strong>对RNAseq数据重命名</strong></p><pre data-tool="mdnice编辑器"><span></span><code><span>$</span><span>cat rename_rna.sh </span><br><span>#</span><span>!/bin/bash</span><br><span>#</span><br>rn_file=${1}<br>out_dir=${2}<br><br>declare -A sample_map<br><br>declare -A sample_count<br><br>declare -A srr_sample_number_map<br><br>while read -r key value; do<br>    sample_map["$key"]="$value"<br>done &lt; ${rn_file}<br><br>for file in *.fastq.gz; do<br>    if [[ $file =~ (SRR[0-9]+)\.lite_([0-9]+)\.fastq\.gz ]]; then<br>        original_number="${BASH_REMATCH[1]}"<br>        read_number="${BASH_REMATCH[2]}"<br>        <br>        sample_name="${sample_map[$original_number]}"<br>        <br>        if [[ -n "$sample_name" ]]; then<br>            if [[ -z "${srr_sample_number_map[$original_number]}" ]]; then<br>                if [[ -z "${sample_count[$sample_name]}" ]]; then<br>                    sample_count["$sample_name"]=0<br>                fi<br>                sample_count["$sample_name"]=$((sample_count["$sample_name"] + 1))<br>                srr_sample_number_map["$original_number"]=${sample_count["$sample_name"]}<br>            fi<br>            <br>            sample_number=${srr_sample_number_map["$original_number"]}<br>            <br>            case $read_number in<br>                1)<br>                    new_read_number="R1"<br>                    ;;<br>                2)<br>                    new_read_number="R2"<br>                    ;;<br>                *)<br>                    echo "Unexpected read number: $read_number"<br>                    continue<br>                    ;;<br>            esac<br>            <br>            new_file_name="${sample_name}_S${sample_number}_L001_${new_read_number}_001.fastq.gz"<br>            <br>            ln -s ${PWD}/"$file" ${out_dir}/"$new_file_name"<br>            echo "Renamed $file to $new_file_name"<br>        fi<br>    fi<br>done<br></code></pre><p data-tool="mdnice编辑器"><strong>对ATAC数据重命名</strong></p><pre data-tool="mdnice编辑器"><span></span><code><span>#</span><span>!/bin/bash</span><br><span>#</span><br>rn_file=${1}<br>out_dir=${2}<br><br>declare -A sample_map<br><br>declare -A sample_count<br><br>declare -A srr_sample_number_map<br><br>while read -r key value; do<br>    sample_map["$key"]="$value"<br>done &lt; ${rn_file}<br><br>for file in *.fastq.gz; do<br>    if [[ $file =~ (SRR[0-9]+)\.lite_([0-9]+)\.fastq\.gz ]]; then<br>        original_number="${BASH_REMATCH[1]}"<br>        read_number="${BASH_REMATCH[2]}"<br>        <br>        sample_name="${sample_map[$original_number]}"<br>        <br>        if [[ -n "$sample_name" ]]; then<br>            if [[ -z "${srr_sample_number_map[$original_number]}" ]]; then<br>                if [[ -z "${sample_count[$sample_name]}" ]]; then<br>                    sample_count["$sample_name"]=0<br>                fi<br>                sample_count["$sample_name"]=$((sample_count["$sample_name"] + 1))<br>                srr_sample_number_map["$original_number"]=${sample_count["$sample_name"]}<br>            fi<br>            <br>            sample_number=${srr_sample_number_map["$original_number"]}<br>            <br>            case $read_number in<br>                1)<br>                    new_read_number="R1"<br>                    ;;<br>                2)<br>                    new_read_number="R2"<br>                    ;;<br>                3)<br>                    new_read_number="R3"<br>                    ;;<br>                4)<br>                    new_read_number="I1"<br>                    ;;<br>                *)<br>                    echo "Unexpected read number: $read_number"<br>                    continue<br>                    ;;<br>            esac<br>            <br>            new_file_name="${sample_name}_S${sample_number}_L001_${new_read_number}_001.fastq.gz"<br>            <br>            ln -s ${PWD}/"$file" ${out_dir}/"$new_file_name"<br>            echo "Renamed $file to $new_file_name"<br>        fi<br>    fi<br>done<br></code></pre><blockquote data-tool="mdnice编辑器"><span></span><p>注意：该重命名脚本使用需完全理解，然后按自己数据实际情况修改，以免改错</p></blockquote><figure data-tool="mdnice编辑器"><img data-imgfileid="100041385" data-ratio="0.6970387243735763" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losib1SRPYAut2ETxJe19eYndAcKcDzgfdDBoAf9j7ibY9YUSLJem2GxagibicrE8Ia4XNesZhPicm4SJK8Q/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="878" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losib1SRPYAut2ETxJe19eYndAcKcDzgfdDBoAf9j7ibY9YUSLJem2GxagibicrE8Ia4XNesZhPicm4SJK8Q/640?wx_fmt=png&amp;from=appmsg"><figcaption>重命名后的文件</figcaption></figure><p data-tool="mdnice编辑器">构建如下所示的CSV文件</p><pre data-tool="mdnice编辑器"><span></span><code>$cat DMSO.csv <br>fastqs,sample,library_type<br>/home/data/t020559/scRNA/PRJNA692722_atac/b_rawdata/fq_data/,DMSO-nuclei,Gene Expression<br>/home/data/t020559/scRNA/PRJNA692722_atac/b_rawdata/fq_data/,DMSO-ATACseq,Chromatin Accessibility<br></code></pre><h3 data-tool="mdnice编辑器"><span></span><span>cellranger arc 定量</span><span></span></h3><p data-tool="mdnice编辑器">准备好以上工作就可以提交任务，运行定量流程了</p><pre data-tool="mdnice编辑器"><span></span><code><span>#</span><span><span>#定量脚本 arc_homo.sh</span></span><br><span><br>#</span><span>! /bin/bash -xe</span><br><span>#</span><br><br>bin=/home/data/t020559/biosoft/cellranger-arc-2.0.2/cellranger-arc<br>db=/home/data/t020559/ref/homo/refdata-cellranger-arc-GRCh38-2020-A-2.0.0<br><br>id=${1}<br>csv=${2}<br><br>/usr/bin/time -v ${bin} count --id=${1} \<br>  --reference=${db} \<br>  --libraries=${2} \<br>  --localcores=8 \<br>  --localmem=100<br></code></pre><p data-tool="mdnice编辑器"><strong>提交任务</strong></p><pre data-tool="mdnice编辑器"><span></span><code>nohup bash /home/data/t020559/scRNA/arc_homo.sh DMSO DMSO.csv 1&gt;log_DMSO.txt 2&gt;&amp;1 &amp;<br></code></pre><h3 data-tool="mdnice编辑器"><span></span><span>结果文件</span><span></span></h3><p data-tool="mdnice编辑器">定量后，我们需要的结果文件主要在<code>/path/DMSO/outs</code>  目录下</p><figure data-tool="mdnice编辑器"><img data-imgfileid="100041384" data-ratio="1.4759259259259259" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losib1SRPYAut2ETxJe19eYndA0lrhOiblQJzm8K8jzrzxdqA4uRpI3SsrLS3libwI47OYn6ujzR2eJd7w/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losib1SRPYAut2ETxJe19eYndA0lrhOiblQJzm8K8jzrzxdqA4uRpI3SsrLS3libwI47OYn6ujzR2eJd7w/640?wx_fmt=png&amp;from=appmsg"><figcaption>结果文件</figcaption></figure><ul data-tool="mdnice编辑器"><li><section><code>web_summary.html</code>  结果报告的html文件。</section></li><li><section><code>raw_feature_bc_matrix.h5</code>  原始的cell × features矩阵的h5格式结果文件。行包括了peaks和基因在内，列为每个细胞的barcodes。</section></li><li><section><code>raw_feature_bc_matrix</code>   原始的cell × features矩阵的结果目录，里面就是三个我们熟知的10X标准结果文件（barcodes、features、matrix），其中features包括了peaks和基因在内，且有5列。</section></li><li><section><code>atac_peaks.bed</code>  Peak Calling的结果。</section></li><li><section><code>atac_peak_annotation.tsv</code>  Peak临近基因注释结果</section></li></ul><figure data-tool="mdnice编辑器"><img data-imgfileid="100041383" data-ratio="0.4740740740740741" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losib1SRPYAut2ETxJe19eYndAFWF5oicZcDUDpIqdbRa8alsLg6aCTsZgtujPCUibzYibgsUpANWyrTDUA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losib1SRPYAut2ETxJe19eYndAFWF5oicZcDUDpIqdbRa8alsLg6aCTsZgtujPCUibzYibgsUpANWyrTDUA/640?wx_fmt=png&amp;from=appmsg"><figcaption>web_summary.html</figcaption></figure><hr data-tool="mdnice编辑器"><p data-tool="mdnice编辑器">参考：</p><ul data-tool="mdnice编辑器"><li><section>https://support.10xgenomics.com/single-cell-multiome-atac-gex/software/pipelines/latest/using/count#libraries</section></li><li><section>https://mp.weixin.qq.com/s/M9E-tGPKlSH0CUPlS_LE2A</section></li></ul></section><section><section act="xmy"><section data-id="123409" data-type="lspecial02,lspecial04" powered-by="xmyeditor.com" data-md5="8b46d"><section powered-by="xmyeditor.com" data-md5="8b46d"><section powered-by="xmyeditor.com" data-md5="8b46d"><section powered-by="xmyeditor.com" data-md5="8b46d"><br></section><section powered-by="xmyeditor.com" data-md5="8b46d"><img data-imgfileid="100041395" data-ratio="0.6940298507462687" data-type="gif" data-w="134" data-src="https://mmbiz.qpic.cn/mmbiz_gif/iaRJcrq2LosibrvQnXTV3PMktpwwmDsjia2JG4e3ibpLQ1G6M4nktW4PiaiaKfnpiaRmNbTT6St7QLfzAMAomKgWxQ4kA/640?wx_fmt=gif&amp;wxfrom=5&amp;wx_lazy=1&amp;tp=webp" src="https://mmbiz.qpic.cn/mmbiz_gif/iaRJcrq2LosibrvQnXTV3PMktpwwmDsjia2JG4e3ibpLQ1G6M4nktW4PiaiaKfnpiaRmNbTT6St7QLfzAMAomKgWxQ4kA/640?wx_fmt=gif&amp;wxfrom=5&amp;wx_lazy=1&amp;tp=webp"></section><section powered-by="xmyeditor.com" data-md5="8b46d"><br></section></section></section></section></section></section><p><span></span><br></p><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><h4 data-tool="mdnice编辑器">文末友情宣传</h4><p data-tool="mdnice编辑器">强烈建议你推荐给身边的<strong>博士后以及年轻生物学PI</strong>，多一点数据认知，让他们的科研上一个台阶：</p><ul data-tool="mdnice编辑器"><li><section><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247531929&amp;idx=1&amp;sn=f6f16b7bf6b907360d6d0052e3d10cf6&amp;chksm=9b4b3d22ac3cb434b6aa7753a4cf0f266578147ccf10b49cc834e46af578ee6de99be0accb30&amp;scene=21#wechat_redirect" textvalue="生信入门&amp;数据挖掘线上直播课8月班" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">生信入门&amp;数据挖掘线上直播课8月班</a>，你的生物信息学入门课</section></li><li><section><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247524148&amp;idx=1&amp;sn=7806da6feb41a36493c519c1cfc1d3ac&amp;chksm=9b4bdf8fac3c569960369602f1ef26639cb366b250f233b2297d1f059471c0458335bfc0b829&amp;scene=21#wechat_redirect" textvalue="时隔5年，我们的生信技能树VIP学徒继续招生啦" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2" hasload="1">时隔5年，我们的生信技能树VIP学徒继续招生啦</a><br></section></li><li><section><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247528363&amp;idx=1&amp;sn=5e02f3e9b2e148191e23ebc2c0d780e7&amp;chksm=9b4b2f10ac3ca606c1c4bac8cf112bb9b0f18e3c4262f5f2b8c0dba3bfedf2ba201507247005&amp;scene=21#wechat_redirect" textvalue="2024的共享服务器交个朋友福利价仍然是800" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2" hasload="1">2024的共享服务器交个朋友福利价仍然是800</a></section></li></ul></section><p><br></p><p><mp-style-type data-value="10000"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/J8RbqgJtNlnc7bsLJJUPGA",target="_blank" rel="noopener noreferrer">原文链接</a>
