---
title: "DecoupleR/CollecTRI network-单细胞转录因子活性分析"
date: 2024-07-18T15:09:52Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Acdemic"]
---
DecoupleR/CollecTRI network-单细胞转录因子活性分析 by 生信菜鸟团
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><p data-tool="mdnice编辑器">DecoupleR包是一个可以从组学数据中分析内部生物学活性的计算方法集合。该R包内部收纳了11种分析方法，包括GSVA,GSEA,univariate linear model (ULM),VIPER等。</p><p data-tool="mdnice编辑器"><img data-imgfileid="100000559" data-ratio="0.5293333333333333" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyH6KxPfYYjuZXuIYKwezcTTGs4Wibib03CTkovu2yjl6c33FkuLS9naHiaNrCISptUBb6AuI7jNCibrBA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="750" src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyH6KxPfYYjuZXuIYKwezcTTGs4Wibib03CTkovu2yjl6c33FkuLS9naHiaNrCISptUBb6AuI7jNCibrBA/640?wx_fmt=png&amp;from=appmsg">PMID：36699385</p><p data-tool="mdnice编辑器">根据包含的计算方法，我们也能大概的知道decoupleR包能够做哪些分析了，包括基本分析(AUCell，Fast GSEA, GSVA和viper)，bulk/scRNA-seq的通路活性、转录因子分析，转录因子和激酶活性分析。这次我们主要学习一下如何使用该R包进行scRNA-seq的转录因子活性分析。</p><p data-tool="mdnice编辑器"><img data-imgfileid="100000558" data-ratio="0.42155525238744884" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyH6KxPfYYjuZXuIYKwezcTTORnn2JKzxt98aLJHhtOicszUWpnXADEm4R3nY3lWOUU22Xv1RNehJzQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="733" src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyH6KxPfYYjuZXuIYKwezcTTORnn2JKzxt98aLJHhtOicszUWpnXADEm4R3nY3lWOUU22Xv1RNehJzQ/640?wx_fmt=png&amp;from=appmsg">github链接：https://saezlab.github.io/decoupleR/index.html</p><p data-tool="mdnice编辑器">那么在进行分析之前，首先需要理解一下"转录因子在特定细胞中的活性水平"这个概念。粗略来说可以理解为转录因子(TFs)对其靶基因发挥调控潜能的程度，所以这个调控潜能的程度高低并不一定意味着跟转录因子的表达量有密切的正相关，而是TFs跟这个细胞中所表达的基因中的TFs靶基因的表达情况的“相关性打分”。</p><p data-tool="mdnice编辑器"><img data-imgfileid="100000557" data-ratio="0.42533333333333334" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyH6KxPfYYjuZXuIYKwezcTTUsMZvb66ia7k74eQzYRz32OhfStqyX0ACd214YMjNGaS5HxZzF9p9Lw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="750" src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyH6KxPfYYjuZXuIYKwezcTTUsMZvb66ia7k74eQzYRz32OhfStqyX0ACd214YMjNGaS5HxZzF9p9Lw/640?wx_fmt=png&amp;from=appmsg">PMID：33135076</p><p data-tool="mdnice编辑器">从代码分析的角度来看，为了推断TFs的活性分数，需要采用单变量线性模型(ulm)方法。对于数据集中的每个样本(每一个单细胞)和TFs数据集中的每个TFs拟合一个线性模型，该模型是基于TFs的TFs-基因的相互作用权重来预测观察到的基因表达。拟合后，得到的斜率t值就是得分。如果它是阳性的，就解释为TFs是活跃的，如果它是阴性的，就认为它是非活跃的。</p><p data-tool="mdnice编辑器">此外，我们还需要获得TFs的调控子(可以认为是TFs-靶基因集群)的数据集。目前推荐使用CollecTRI(比DoRothEA更全面),这个资源整合了多个来源的数据集，目前纳入了1186个TFs和43175个有一定证据的TF-基因相互作用数据。</p><p data-tool="mdnice编辑器"><img data-imgfileid="100000556" data-ratio="0.75" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyH6KxPfYYjuZXuIYKwezcTTXEJBTjOdhzlt5TWaydGDHXQVdDBYDEZiaNegWxq5k8Ntn0YaxNCYv1Q/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyH6KxPfYYjuZXuIYKwezcTTXEJBTjOdhzlt5TWaydGDHXQVdDBYDEZiaNegWxq5k8Ntn0YaxNCYv1Q/640?wx_fmt=png&amp;from=appmsg">PMID：37843125</p><h4 data-tool="mdnice编辑器"><span></span>具体分析流程<span></span></h4><p data-tool="mdnice编辑器">1.数据读取和加载R包</p><p data-tool="mdnice编辑器">我这里采用的GSE148190-SM4455931-10X样本</p><p data-tool="mdnice编辑器">Step1.final.Rdata已经是经过了降维、聚类、分群后的结果。</p><pre data-tool="mdnice编辑器"><span data-remoteid="c1717037546073" data-cacheurl="https://files.mdnice.com/user/3441/876cad08-0422-409d-bb5a-08afec5da8ee.svg"></span><code>rm(list=ls())<br><span>library</span>(Seurat)<br><span>library</span>(decoupleR)<br><span>library</span>(dplyr)<br><span>library</span>(tibble)<br><span>library</span>(tidyr)<br><span>library</span>(patchwork)<br><span>library</span>(ggplot2)<br><span>library</span>(pheatmap)<br><br>load(<span>"step1.final.Rdata"</span>)<br>sce &lt;- step1.final<br>table(Idents(sce))<br></code></pre><p data-tool="mdnice编辑器">2.先看一下聚类图</p><pre data-tool="mdnice编辑器"><span data-remoteid="c1717037546074" data-cacheurl="https://files.mdnice.com/user/3441/876cad08-0422-409d-bb5a-08afec5da8ee.svg"></span><code>DimPlot(sce,label = <span>T</span>,repel = <span>T</span>) <br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100000555" data-ratio="0.8650088809946714" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyH6KxPfYYjuZXuIYKwezcTTRNVoBI1Mj18icIZmsC4K40cl6hJFtRP8EW38HuSJyYfzsWy628p38fw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="563" src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyH6KxPfYYjuZXuIYKwezcTTRNVoBI1Mj18icIZmsC4K40cl6hJFtRP8EW38HuSJyYfzsWy628p38fw/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">3.转录因子分析</p><pre data-tool="mdnice编辑器"><span data-remoteid="c1717037546075" data-cacheurl="https://files.mdnice.com/user/3441/876cad08-0422-409d-bb5a-08afec5da8ee.svg"></span><code><span>#多线程模式</span><br>plan(<span>"multisession"</span>, workers = <span>4</span>)<br><span>#获取转录因子数据</span><br>net &lt;- get_collectri(organism=<span>'human'</span>, split_complexes=<span>FALSE</span>)<br>net<br><span># A tibble: 43,178 × 3</span><br><span>#   source target   mor</span><br><span>#   &lt;chr&gt;  &lt;chr&gt;  &lt;dbl&gt;</span><br><span># 1 MYC    TERT       1</span><br><span># 2 SPI1   BGLAP      1</span><br><span># 3 SMAD3  JUN        1</span><br><span># 4 SMAD4  JUN        1</span><br><span># 5 STAT5A IL2        1</span><br><span># 6 STAT5B IL2        1</span><br><span># 7 RELA   FAS        1</span><br><span># 8 WT1    NR0B1      1</span><br><span># 9 NR0B2  CASP1      1</span><br><span># 10 SP1    ALDOA      1</span><br><span># ℹ 43,168 more rows</span><br><span># ℹ Use `print(n = ...)` to see more rows</span><br><br><span>#提取数据</span><br>mat &lt;- as.matrix(sce@assays$RNA@data)<br><span>#下面的运算会耗费一点时间，取决于线程数量和CPU算力</span><br>acts &lt;- run_ulm(mat, net,minsize = <span>5</span>,<br>                .source=<span>'source'</span>, .target=<span>'target'</span>,.mor=<span>'mor'</span>)<br>acts<br><span># A tibble: 1,933,792 × 5</span><br><span>#   statistic source condition             score p_value</span><br><span>#   &lt;chr&gt;     &lt;chr&gt;  &lt;chr&gt;                 &lt;dbl&gt;   &lt;dbl&gt;</span><br><span># 1 ulm       ABL1   AAACCTGAGGACGAAA-1  0.274   0.784  </span><br><span># 2 ulm       ABL1   AAACCTGAGGGTGTTG-1  0.724   0.469  </span><br><span># 3 ulm       ABL1   AAACCTGCAAGGACTG-1  2.29    0.0221 </span><br><span># 4 ulm       ABL1   AAACCTGCAATCTGCA-1 -0.423   0.673  </span><br><span># 5 ulm       ABL1   AAACCTGCACTGTTAG-1 -0.00305 0.998  </span><br><span># 6 ulm       ABL1   AAACCTGGTCTAGTGT-1  3.01    0.00262</span><br><span># 7 ulm       ABL1   AAACCTGTCGTACCGG-1  0.550   0.582  </span><br><span># 8 ulm       ABL1   AAACGGGAGGCAATTA-1  1.51    0.131  </span><br><span># 9 ulm       ABL1   AAACGGGAGGTGCTTT-1  0.312   0.755  </span><br><span>#10 ulm       ABL1   AAACGGGCAGCCTATA-1  0.896   0.370  </span><br><span># ℹ 1,933,782 more rows</span><br><span># ℹ Use `print(n = ...)` to see more rows</span><br><br><span>#提取获得的数据,修改格式并保存</span><br>sce[[<span>'tfsulm'</span>]] &lt;- acts %&gt;%<br>  pivot_wider(id_cols = <span>'source'</span>, <br>              names_from = <span>'condition'</span>,<br>              values_from = <span>'score'</span>) %&gt;%<br>  column_to_rownames(<span>'source'</span>) %&gt;%<br>  Seurat::CreateAssayObject(.)<br><span>#更改默认的分析对象</span><br>DefaultAssay(object = sce) &lt;- <span>"tfsulm"</span><br><span>#缩放数据并重新保存</span><br>sce &lt;- ScaleData(sce)<br>sce@assays$tfsulm@data &lt;- sce@assays$tfsulm@scale.data<br></code></pre><p data-tool="mdnice编辑器">4.可视化</p><pre data-tool="mdnice编辑器"><span data-remoteid="c1717037546076" data-cacheurl="https://files.mdnice.com/user/3441/876cad08-0422-409d-bb5a-08afec5da8ee.svg"></span><code>p1 &lt;- DimPlot(sce, reduction = <span>"umap"</span>, <br>              label = <span>TRUE</span>, pt.size = <span>0.5</span>) + <br>      NoLegend() + ggtitle(<span>'Cell types'</span>)<br>p1<br><br>p2 &lt;- (FeaturePlot(sce, features = c(<span>"E2F1"</span>)) &amp; <br>  scale_colour_gradient2(low = <span>'blue'</span>, mid = <span>'white'</span>, high = <span>'red'</span>)) +<br>  ggtitle(<span>'E2F1 activity'</span>)<br>p2<br><br>DefaultAssay(object = sce) &lt;- <span>"RNA"</span><br>p3 &lt;- FeaturePlot(sce, features = c(<span>"E2F1"</span>)) + <br>          ggtitle(<span>'PAX5 expression'</span>)<br>p3<br><br>DefaultAssay(object = sce) &lt;- <span>"tfsulm"</span><br>p1 + p2 + p3<br></code></pre><p data-tool="mdnice编辑器">挑选了一个很常见的转录因子E2F1，从表达量分析看它似乎”价值不大“，但从TFs活性分析来看其在CD4 T,CD8 T和B细胞中都显示出了较大的活性。<img data-imgfileid="100000554" data-ratio="0.4" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyH6KxPfYYjuZXuIYKwezcTT5HA6MnxvOicGoBS5wbYxPicSqjOIPyH7hLNKy6xHxWOeV8chOVzFAwWw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyH6KxPfYYjuZXuIYKwezcTT5HA6MnxvOicGoBS5wbYxPicSqjOIPyH7hLNKy6xHxWOeV8chOVzFAwWw/640?wx_fmt=png&amp;from=appmsg"></p><p data-tool="mdnice编辑器">5.top25的TFs</p><pre data-tool="mdnice编辑器"><span data-remoteid="c1717037546077" data-cacheurl="https://files.mdnice.com/user/3441/876cad08-0422-409d-bb5a-08afec5da8ee.svg"></span><code>n_tfs &lt;- <span>25</span><br><span>#提取活性数据并修改数据格式</span><br>df &lt;- t(as.matrix(sce@assays$tfsulm@data)) %&gt;%<br>  as.data.frame() %&gt;%<br>  mutate(cluster = Idents(sce)) %&gt;%<br>  pivot_longer(cols = -cluster, <span>#指定除cluster列以外的都需要转换</span><br>               names_to = <span>"source"</span>, <span>#列名放到source列中</span><br>               values_to = <span>"score"</span>) %&gt;% <span>#值放到score列中</span><br>  group_by(cluster, <span>source</span>) %&gt;% <span>#将数据按cluster和source两列进行分组</span><br>  summarise(mean = mean(score))<br><br><span>#获得不同簇中活性差异较大的TFs</span><br>tfs &lt;- df %&gt;%<br>  group_by(<span>source</span>) %&gt;%<br>  summarise(std = sd(mean)) %&gt;%<br>  arrange(-abs(std)) %&gt;%<br>  head(n_tfs) %&gt;%<br>  pull(<span>source</span>)<br><br><span>#修改数据格式</span><br>top_acts_mat &lt;- df %&gt;%<br>  filter(<span>source</span> %<span>in</span>% tfs) %&gt;%<br>  pivot_wider(id_cols = <span>'cluster'</span>, <br>              names_from = <span>'source'</span>,<br>              values_from = <span>'mean'</span>) %&gt;%<br>  column_to_rownames(<span>'cluster'</span>) %&gt;%<br>  as.matrix()<br><br><span>#绘图</span><br>palette_length = <span>100</span><br>my_color = colorRampPalette(c(<span>"Darkblue"</span>,<span>"white"</span>,<span>"red"</span>))(palette_length)<br><br>my_breaks &lt;- c(seq(-<span>3</span>, <span>0</span>, length.out=ceiling(palette_length/<span>2</span>) + <span>1</span>),<br>               seq(<span>0.05</span>, <span>3</span>, length.out=floor(palette_length/<span>2</span>)))<br><br><span>#显示</span><br>pheatmap(top_acts_mat, <br>         border_color = <span>NA</span>,<br>         color=my_color, <br>         breaks = my_breaks) <br></code></pre><p data-tool="mdnice编辑器"><img data-imgfileid="100000553" data-ratio="0.5" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyH6KxPfYYjuZXuIYKwezcTTDep1VoMZAZfo9zHaJeUVOWLK37H1WibyLIhIj3ibB5oDHib2QwYa1kofw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/0SOG4MpDAyH6KxPfYYjuZXuIYKwezcTTDep1VoMZAZfo9zHaJeUVOWLK37H1WibyLIhIj3ibB5oDHib2QwYa1kofw/640?wx_fmt=png&amp;from=appmsg"><br><strong>参考文献：</strong></p><p data-tool="mdnice编辑器">1.Pau Badia-I-Mompel, Jesús Vélez Santiago, Jana Braunger et al. decoupleR: ensemble of computational methods to infer biological activities from omics data. Bioinform Adv. 2022 Mar 8;2(1):vbac016. doi: 10.1093/bioadv/vbac016.</p><p data-tool="mdnice编辑器">2.Cynthia Z Ma, Michael R Brent et al. Inferring TF activities and activity regulators from gene expression data with constraints from TF perturbation data. Bioinformatics. 2021 Jun 9;37(9):1234-1245.</p><p data-tool="mdnice编辑器">3.Sophia Müller-Dott, Eirini Tsirvouli, Miguel Vazquez et al. Expanding the coverage of regulons from high-confidence prior knowledge for accurate estimation of transcription factor activities. Nucleic Acids Res. 2023 Nov 10;51(20):10934-10949.</p><p data-tool="mdnice编辑器"><strong>注</strong>：若对内容有疑惑或者发现有明确错误的朋友，请联系后台(希望多多交流)。</p><span>- END -<span></span></span></section></section><p><br></p><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/-ljyLbXIbU1BU9yqdCD3Kw",target="_blank" rel="noopener noreferrer">原文链接</a>
