---
title: "RNA 42. 基于转录组及转录因子库构建转录因子调控网络 (NetAct)"
date: 2024-07-19T05:43:44Z
draft: ["false"]
tags: [
  "fetched",
  "桓峰基因"
]
categories: ["Acdemic"]
---
RNA 42. 基于转录组及转录因子库构建转录因子调控网络 (NetAct) by 桓峰基因
------
<div><p><img data-imgfileid="100010592" data-ratio="0.4267326732673267" data-src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMxdjEsWs1Shxy54alETSRmMmPUlia8ajqVjDDSOicd4BTWT2lUAFqz3P660MFWt5tqx8TrH4Yy67GtQ/640?wx_fmt=other&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1&amp;tp=webp" data-w="1010" src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMxdjEsWs1Shxy54alETSRmMmPUlia8ajqVjDDSOicd4BTWT2lUAFqz3P660MFWt5tqx8TrH4Yy67GtQ/640?wx_fmt=other&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1&amp;tp=webp"></p><section><mp-common-profile data-pluginname="mpprofile" data-id="MzU0NDY2NTUwOA==" data-headimg="http://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMzQ2iaDaME4SMFFyIc3WCrVf5loZv2JiahbJiaozT9fNzGPkphu6icmGg8FlIsOicJ6F7OJicH2qA0r3k5g/0?wx_fmt=png" data-nickname="桓峰基因" data-alias="KyohoGene" data-signature="桓峰基因推出全新生信分析文字版和视频版免费教程，欢迎大家来围观，在公众号搜索即可获得相关教程！有测序，生信分析，文章合作，实验室各种耗材试剂等业务欢迎来咨询，关注桓峰基因官网：http://www.kyohogene.com/" data-from="0" data-is_biz_ban="0"></mp-common-profile></section><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><h2 data-tool="mdnice编辑器"><span>简     介</span></h2><p data-tool="mdnice编辑器">系统生物学的一个主要问题是如何识别控制生物过程决策的核心基因调控回路。一个名为NetAct的计算平台，用于使用转录组学数据和基于文献的转录因子目标数据库构建核心转录因子调控网络。NetAct使用目标表达稳健地推断调控因子的活性，基于转录活性构建网络，并集成数学模型进行验证。我们的计算机基准测试表明，NetAct在推断转录活性和基因网络方面优于现有算法。我们演示了应用NetAct来模拟驱动TGF-β诱导的上皮-间质转化和巨噬细胞极化的网络。</p></section><p><img data-imgfileid="100010584" data-ratio="0.43950995405819293" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwn6hF11K88m3tviatIJ2PIdIIdtyKQ5I1TmHQtCibI6IZWU5vdQZJy4A45FN9CxumQYkKhiasYpX8Fw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="653" src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwn6hF11K88m3tviatIJ2PIdIIdtyKQ5I1TmHQtCibI6IZWU5vdQZJy4A45FN9CxumQYkKhiasYpX8Fw/640?wx_fmt=png&amp;from=appmsg"></p><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><p data-tool="mdnice编辑器">NetAct是一个计算平台，使用转录组数据和基于文献的转录因子靶点数据库来构建核心转录因子调控网络。本教程旨在演示NetAct的核心功能组件，以及如何使用它来构建和建模转录因子调节网络。本工作流程中使用的实验RNA-seq数据来自Sheridan等人(2015)，包括从雌性处女小鼠的乳腺中分选的三个细胞群(基底、腔祖细胞(LP)和成熟腔细胞(ML))。使用GEO:GSE63310，这些样品的count数据可以从Gene Expression Omnibus (GEO)下载。</p></section><p><img data-imgfileid="100010585" data-ratio="1.037037037037037" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwn6hF11K88m3tviatIJ2PIdjMYa5icw1lPZs2wrD0NZcxcNLVmE4AyNCdrYFOfwT2OR151DGLMwZRQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="648" src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwn6hF11K88m3tviatIJ2PIdjMYa5icw1lPZs2wrD0NZcxcNLVmE4AyNCdrYFOfwT2OR151DGLMwZRQ/640?wx_fmt=png&amp;from=appmsg"></p><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><h2 data-tool="mdnice编辑器"><span></span><span>软件包安装</span></h2><p data-tool="mdnice编辑器">软件包安装非常简单，但是需要注意一下Rtools，R，BiocManager版本号，这种问题经常出现，更新匹配一直就没问题了。</p><pre data-tool="mdnice编辑器"><span></span><code><span>if</span>(!<span>require</span>(Scissor))<br>  devtools::install_github(<span>"lusystemsbio/NetAct"</span>)<br><br>BiocManager::install(<span>"org.Mm.eg.db"</span>)<br></code></pre><h2 data-tool="mdnice编辑器"><span></span><span>数据读取</span></h2><p data-tool="mdnice编辑器">数据是从GEO上下载，但是这里给了一段代码用于下载数据，但是如果网速很慢还是直接从网站上下载吧：</p><pre data-tool="mdnice编辑器"><span></span><code>url &lt;- <span>"https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE63310&amp;format=file"</span><br>utils::download.file(url, destfile=<span>"GSE63310_RAW.tar"</span>, mode=<span>"wb"</span>) <br>utils::untar(<span>"GSE63310_RAW.tar"</span>, exdir = <span>"."</span>)<br>files &lt;- c(<span>"GSM1545535_10_6_5_11.txt"</span>, <span>"GSM1545536_9_6_5_11.txt"</span>, <span>"GSM1545538_purep53.txt"</span>,<br>           <span>"GSM1545539_JMS8-2.txt"</span>, <span>"GSM1545540_JMS8-3.txt"</span>, <span>"GSM1545541_JMS8-4.txt"</span>,<br>           <span>"GSM1545542_JMS8-5.txt"</span>, <span>"GSM1545544_JMS9-P7c.txt"</span>, <span>"GSM1545545_JMS9-P8c.txt"</span>)<br><span>for</span>(i <span>in</span> paste(files, <span>".gz"</span>, sep=<span>""</span>)){<br>  R.utils::gunzip(i, overwrite=<span>TRUE</span>)<br>}<br></code></pre><p data-tool="mdnice编辑器">对数据进行整理，预处理等操作：</p><pre data-tool="mdnice编辑器"><span></span><code>files &lt;- c(<span>"GSM1545535_10_6_5_11.txt"</span>, <span>"GSM1545536_9_6_5_11.txt"</span>, <span>"GSM1545538_purep53.txt"</span>,<br>    <span>"GSM1545539_JMS8-2.txt"</span>, <span>"GSM1545540_JMS8-3.txt"</span>, <span>"GSM1545541_JMS8-4.txt"</span>, <span>"GSM1545542_JMS8-5.txt"</span>,<br>    <span>"GSM1545544_JMS9-P7c.txt"</span>, <span>"GSM1545545_JMS9-P8c.txt"</span>)<br><br>x &lt;- edgeR::readDGE(files, columns = c(<span>1</span>, <span>3</span>))<br>group &lt;- as.factor(c(<span>"LP"</span>, <span>"ML"</span>, <span>"Basal"</span>, <span>"Basal"</span>, <span>"ML"</span>, <span>"LP"</span>, <span>"Basal"</span>, <span>"ML"</span>, <span>"LP"</span>))<br>x$samples$group &lt;- group<br>samplenames &lt;- c(<span>"LP1"</span>, <span>"ML1"</span>, <span>"Basal1"</span>, <span>"Basal2"</span>, <span>"ML2"</span>, <span>"LP2"</span>, <span>"Basal3"</span>, <span>"ML3"</span>,<br>    <span>"LP3"</span>)<br>colnames(x) &lt;- samplenames<br>head(x)<br><span>## An object of class "DGEList"</span><br><span>## $samples</span><br><span>##                           files group lib.size norm.factors</span><br><span>## LP1    GSM1545535_10_6_5_11.txt    LP 32863052            1</span><br><span>## ML1     GSM1545536_9_6_5_11.txt    ML 35335491            1</span><br><span>## Basal1   GSM1545538_purep53.txt Basal 57160817            1</span><br><span>## Basal2    GSM1545539_JMS8-2.txt Basal 51368625            1</span><br><span>## ML2       GSM1545540_JMS8-3.txt    ML 75795034            1</span><br><span>## LP2       GSM1545541_JMS8-4.txt    LP 60517657            1</span><br><span>## Basal3    GSM1545542_JMS8-5.txt Basal 55086324            1</span><br><span>## ML3     GSM1545544_JMS9-P7c.txt    ML 21311068            1</span><br><span>## LP3     GSM1545545_JMS9-P8c.txt    LP 19958838            1</span><br><span>## </span><br><span>## $counts</span><br><span>##            Samples</span><br><span>## Tags        LP1 ML1 Basal1 Basal2  ML2 LP2 Basal3 ML3 LP3</span><br><span>##   497097      1   2    342    526    3   3    535   2   0</span><br><span>##   100503874   0   0      5      6    0   0      5   0   0</span><br><span>##   100038431   0   0      0      0    0   0      1   0   0</span><br><span>##   19888       0   1      0      0   17   2      0   1   0</span><br><span>##   20671       1   1     76     40   33  14     98  18   8</span><br><span>##   27395     431 771   1368   1268 1564 769    818 468 342</span><br></code></pre><h2 data-tool="mdnice编辑器"><span></span><span>实例操作</span></h2><h3 data-tool="mdnice编辑器"><span></span>1. 建立比较和表型数据<span></span></h3><pre data-tool="mdnice编辑器"><span></span><code><span>library</span>(org.Mm.eg.db)<br><span>library</span>(NetAct)<br>compList &lt;- c(<span>"Basal-LP"</span>, <span>"Basal-ML"</span>, <span>"LP-ML"</span>)<br>phenoData = new(<span>"AnnotatedDataFrame"</span>, data = data.frame(celltype = group))<br>rownames(phenoData) = colnames(x$counts)<br>head(x)<br><span>## An object of class "DGEList"</span><br><span>## $samples</span><br><span>##                           files group lib.size norm.factors</span><br><span>## LP1    GSM1545535_10_6_5_11.txt    LP 32863052            1</span><br><span>## ML1     GSM1545536_9_6_5_11.txt    ML 35335491            1</span><br><span>## Basal1   GSM1545538_purep53.txt Basal 57160817            1</span><br><span>## Basal2    GSM1545539_JMS8-2.txt Basal 51368625            1</span><br><span>## ML2       GSM1545540_JMS8-3.txt    ML 75795034            1</span><br><span>## LP2       GSM1545541_JMS8-4.txt    LP 60517657            1</span><br><span>## Basal3    GSM1545542_JMS8-5.txt Basal 55086324            1</span><br><span>## ML3     GSM1545544_JMS9-P7c.txt    ML 21311068            1</span><br><span>## LP3     GSM1545545_JMS9-P8c.txt    LP 19958838            1</span><br><span>## </span><br><span>## $counts</span><br><span>##            Samples</span><br><span>## Tags        LP1 ML1 Basal1 Basal2  ML2 LP2 Basal3 ML3 LP3</span><br><span>##   497097      1   2    342    526    3   3    535   2   0</span><br><span>##   100503874   0   0      5      6    0   0      5   0   0</span><br><span>##   100038431   0   0      0      0    0   0      1   0   0</span><br><span>##   19888       0   1      0      0   17   2      0   1   0</span><br><span>##   20671       1   1     76     40   33  14     98  18   8</span><br><span>##   27395     431 771   1368   1268 1564 769    818 468 342</span><br></code></pre><h3 data-tool="mdnice编辑器"><span></span>2. 预处理和DEG分析<span></span></h3><p data-tool="mdnice编辑器">对于RNA-seq数据，NetAct提供了一个有用的预处理函数，称为preprocess_counts()，可以过滤掉低表达的基因，并为原始计数数据检索相关的基因。然后使用limma/DESeq2包鉴定差异表达基因(DEG)。生成的规范化表达式数据与phenoData元数据一起保存到标准ExpressionSet对象中，以便进行下游分析。</p><h4 data-tool="mdnice编辑器"><span></span>limma<span></span></h4><pre data-tool="mdnice编辑器"><span></span><code>counts &lt;- Preprocess_counts(counts = x$counts, groups = group, mouse = <span>TRUE</span>)<br>DErslt = RNAseqDegs_limma(counts = counts, phenodata = phenoData, complist = compList,<br>    qval = <span>0.05</span>)<br><span>##        Basal-LP Basal-ML LP-ML</span><br><span>## Down       4308     4568  2570</span><br><span>## NotSig     7071     6728 11506</span><br><span>## Up         4882     4965  2185</span><br>neweset = Biobase::ExpressionSet(assayData = as.matrix(DErslt$Overall$e), phenoData = phenoData)<br></code></pre><h4 data-tool="mdnice编辑器"><span></span>DESeq2<span></span></h4><pre data-tool="mdnice编辑器"><span></span><code>DErslt2 = RNAseqDegs_DESeq(counts = counts, phenodata = phenoData, complist = compList,<br>    qval = <span>0.05</span>)<br>neweset2 = Biobase::ExpressionSet(assayData = as.matrix(DErslt2$Overall$e), phenoData = phenoData)<br></code></pre><h3 data-tool="mdnice编辑器"><span></span>3. 鉴定显著富集的TF<span></span></h3><p data-tool="mdnice编辑器">采用置换的方法，通过GSEA算法选择显著富集的TF。通过考虑q值截止值，从多个比较中汇总TF。此步骤通常耗时最长，因此建议通过提供文件名来保存输出。</p><pre data-tool="mdnice编辑器"><span></span><code>data(<span>"mDB"</span>)<br>calc &lt;- <span>FALSE</span><br><br><span>if</span> (calc) {<br>    gsearslts &lt;- TF_Selection(GSDB = mDB, DErslt = DErslt, minSize = <span>5</span>, nperm = <span>10000</span>,<br>        qval = <span>0.05</span>, compList = compList, nameFile = <span>"gsearslts_tutorial"</span>)<br>} <span>else</span> {<br>    gsearslts &lt;- readRDS(file = <span>"gsearslts_tutorial.RDS"</span>)<br>}<br><br>tfs &lt;- gsearslts$tfs<br>tfs<br><span>##  [1] "Ar"     "Cebpa"  "Chd7"   "Ctnnb1" "E2f1"   "Egr1"   "Esr1"   "Ets1"  </span><br><span>##  [9] "Foxo1"  "Foxo3"  "Gli1"   "Gli2"   "Hnf1a"  "Hoxc8"  "Lef1"   "Myocd" </span><br><span>## [17] "Myod1"  "Nfatc1" "Nfkb1"  "Npas2"  "Pgr"    "Pou2f1" "Ppara"  "Pparg" </span><br><span>## [25] "Sfn"    "Smad3"  "Smad4"  "Sox2"   "Sp1"    "Sp3"    "Srf"    "Tfap2a"</span><br><span>## [33] "Trp53"</span><br><br>Reselect_TFs(GSEArslt = gsearslts$GSEArslt, qval = <span>0.01</span>)<br><span>##  [1] "Ar"     "Cebpa"  "Ctnnb1" "Foxo1"  "Foxo3"  "Gli1"   "Gli2"   "Myocd" </span><br><span>##  [9] "Myod1"  "Pou2f1" "Ppara"  "Pparg"  "Smad4"  "Sp1"    "Tfap2a"</span><br></code></pre><h3 data-tool="mdnice编辑器"><span></span>4. 计算所选TF的活性<span></span></h3><p data-tool="mdnice编辑器">NetAct根据其目标的表达来推断所选监管机构的活动。结果显示与继承基因表达相比，TF活性模式更清晰。</p><pre data-tool="mdnice编辑器"><span></span><code>act.me &lt;- TF_Activity(tfs, mDB, neweset, DErslt$Overall)<br>acts_mat = act.me$all_activities<br>Activity_heatmap(acts_mat, neweset)<br></code></pre><p><img data-galleryid="" data-imgfileid="100010586" data-ratio="0.8351783517835178" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwn6hF11K88m3tviatIJ2PIdLLiblzfoE4WAbCBRwKPV1onticHytRicXibcs25LqBnVACFDMadmF2Vz4A/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="813" src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwn6hF11K88m3tviatIJ2PIdLLiblzfoE4WAbCBRwKPV1onticHytRicXibcs25LqBnVACFDMadmF2Vz4A/640?wx_fmt=png&amp;from=appmsg"></p><h3 data-tool="mdnice编辑器"><span></span>5. 构建基因调控网络<span></span></h3><p data-tool="mdnice编辑器">在这一步中，通过基于互信息和熵的链路关系过滤，输出基因调控网络拓扑结构。plot_network()函数提供网络拓扑的交互式视图。</p><pre data-tool="mdnice编辑器"><span></span><code>tf_links = TF_Filter(acts_mat, mDB, miTh = <span>0.05</span>, nbins = <span>8</span>, corMethod = <span>"spearman"</span>,<br>    DPI = <span>T</span>)<br><span>## [1] 0</span><br><span>## [1] 0.5</span><br><br>plot_network(tf_links)</code></pre><p><img data-galleryid="" data-imgfileid="100010588" data-ratio="0.9629629629629629" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwn6hF11K88m3tviatIJ2PId8alHMag07BAEpfYmUpDlXMokhXIAp6ibwxV1HTp5Ilg9Pb7NHR3m7ew/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="918" src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwn6hF11K88m3tviatIJ2PId8alHMag07BAEpfYmUpDlXMokhXIAp6ibwxV1HTp5Ilg9Pb7NHR3m7ew/640?wx_fmt=png&amp;from=appmsg"></p><h3 data-tool="mdnice编辑器"></h3><h3 data-tool="mdnice编辑器"><span></span>6. RACIPE模拟<span></span></h3><p data-tool="mdnice编辑器">然后，使用RACIPE算法对构建的网络的动态行为进行建模。建议模拟10000个模型。</p><pre data-tool="mdnice编辑器"><span></span><code>racipe_results &lt;- sRACIPE::sracipeSimulate(circuit = tf_links, numModels = <span>200</span>, plots = <span>TRUE</span>)<br></code></pre><p><img data-galleryid="" data-imgfileid="100010589" data-ratio="0.7136204889406287" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwn6hF11K88m3tviatIJ2PIdwsw6REkZmjY7kr9ov12dVEuVEwloulSYMibOhZC7z9aWoSPeIf76ibrw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="859" src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwn6hF11K88m3tviatIJ2PIdwsw6REkZmjY7kr9ov12dVEuVEwloulSYMibOhZC7z9aWoSPeIf76ibrw/640?wx_fmt=png&amp;from=appmsg"></p><h4 data-tool="mdnice编辑器"><span></span></h4><p><img data-imgfileid="100010590" data-ratio="0.6445366528354081" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwn6hF11K88m3tviatIJ2PId8gXIHe001iaYtaSjyZsENwANOGqs6e35GGxRrOlafDwGRoNKsqQMDqw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="723" src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwn6hF11K88m3tviatIJ2PId8gXIHe001iaYtaSjyZsENwANOGqs6e35GGxRrOlafDwGRoNKsqQMDqw/640?wx_fmt=png&amp;from=appmsg"></p><p><img data-imgfileid="100010591" data-ratio="0.6445366528354081" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwn6hF11K88m3tviatIJ2PIdqDnls32Q6PrxicAuW4QbaVjkuzoQTtD7jLCQicibu27kSqOeGmLqhicBtw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="723" src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwn6hF11K88m3tviatIJ2PIdqDnls32Q6PrxicAuW4QbaVjkuzoQTtD7jLCQicibu27kSqOeGmLqhicBtw/640?wx_fmt=png&amp;from=appmsg"></p><h4 data-tool="mdnice编辑器"><span></span></h4><h4 data-tool="mdnice编辑器"><span></span><span>Reference</span><span></span></h4><p data-tool="mdnice编辑器">Su K, et al (2022) NetAct: a computational platform to construct core transcription factor regulatory networks using gene activity, Genome Biology, 23:270.</p></section><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><p><span>桓峰基因，铸造成功的您！</span></p><p><span>未来桓峰基因公众号将不间断的推出单细胞系列生信分析教程，</span></p><p><span>敬请期待！！</span></p></section><p><br></p><p><span>桓峰基因官网正式上线，请大家多多关注，还有很多不足之处，大家多多指正！</span><span>http://www.kyohogene.com/</span></p><p><span>桓峰基因和投必得合作，文章润色优惠85折，需要文章润色的老师可以直接到网站输入领取桓峰基因专属优惠券码：KYOHOGENE，然后上传，付款时选择桓峰基因优惠券即可享受85折优惠哦！https://www.topeditsci.com/</span></p><p><span><img data-galleryid="" data-imgfileid="100010593" data-ratio="0.5657407407407408" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwDj3ich1uOSggdibWHw5pgkMx26IaaiaQoKJbU2EaQI2CYh9n2stiaMicbs1RyInFfqO6qWwpDxsFfl8w/640?wx_fmt=other&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1&amp;tp=webp" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwDj3ich1uOSggdibWHw5pgkMx26IaaiaQoKJbU2EaQI2CYh9n2stiaMicbs1RyInFfqO6qWwpDxsFfl8w/640?wx_fmt=other&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1&amp;tp=webp"></span></p></section><section><mp-common-videosnap data-pluginname="mpvideosnap" data-url="https://findermp.video.qq.com/251/20304/stodownload?encfilekey=rjD5jyTuFrIpZ2ibE8T7YmwgiahniaXswqzGPlk0HGa25xhE1gh0V0D99bK6ibFJxSoAwvRbBPw36Y18R6xohxPuVaaDXricE1f1t0Eyr2Ic3JicxEU02OCVWWzQ&amp;token=6xykWLEnztJXNVKjumLQcib04xyUK5zqTicwuQGoEKveicnqPKFjUvqK9zcjTwyMpvMynGlNXnepu2usgMNg5gp4eJtlMGIgO25opb7HODzOzLTyBoGODKeyIFJZWcz0NIM&amp;idx=1&amp;dotrans=0&amp;hy=SH&amp;m=&amp;scene=2&amp;uzid=2" data-headimgurl="http://wx.qlogo.cn/finderhead/Q3auHgzwzM6NFe6qfZ98pccnStLOBzJIic5564gE1oW2m34RtSicQHIQ/0" data-username="v2_060000231003b20faec8c6eb891cc2ddca0ce43cb07767727967ddf958ff98912bf15d898b0d@finder" data-nickname="桓峰基因" data-desc="RNA 42. 基于转录组及转录因子库构建转录因子调控网络 (NetAct)#转录组#转录因子#基因网络@桓峰基因" data-nonceid="17759689879944331768" data-type="video" data-mediatype="undefined" data-authiconurl="" data-from="new" data-width="1920" data-height="1080" data-id="export/UzFfAgtgekIEAQAAAAAA6lkGLSteLwAAAAstQy6ubaLX4KHWvLEZgBPE6KFYXW44ddeLzNPgMIvpyA-PpzHw0GTEuc7EHWvz" data-isdisabled="0" data-errortips=""></mp-common-videosnap></section><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/o6Ihgjk_Ewdj_qHto0bYUw",target="_blank" rel="noopener noreferrer">原文链接</a>
