---
title: "R语言和数据分析 |11.R语言和数据分析之GEO差异分析"
date: 2024-07-18T13:57:57Z
draft: ["false"]
tags: [
  "fetched",
  "医学生之学习生信"
]
categories: ["Acdemic"]
---
R语言和数据分析 |11.R语言和数据分析之GEO差异分析 by 医学生之学习生信
------
<div><blockquote><p>R语言和数据分析系列是以果子老师发布在b站的课程为基础记录的学习笔记，希望可以帮助刚学习R语言的同学。在课程里可以体会到老师付出时间与精力，在此衷心感谢果子老师的无私分享！感兴趣的同学可以去B站一起加入学习R语言的队伍~今天推出果子老师R语言和数据分析学习笔记<strong>第11期</strong>。本期分享内容为：<strong>R语言和数据分析之GEO差异分析</strong>。对应——果子学生信-R语言入坑必看的<strong>第29-31节</strong>视频内容。<br>【果子学生信-R语言入坑必看-01_introduction_and_warm_up】https://www.bilibili.com/video/BV118411d7JB?vd_source=20ee9ef1fa406aa6039ca07a30c4fdd1<br>对生信感兴趣的同学还可以点击关注果子老师公众号~</p></blockquote><section><mp-common-profile data-pluginname="mpprofile" data-id="MzIyMzA2MTcwMg==" data-headimg="http://mmbiz.qpic.cn/mmbiz_png/NDy5aEnReX12QhHoIrQQ5OKnIAmrdAxVtlT7WzJWKic4OC5sX7pkylLFG5u7lAws2tfUfMia6icu6CuCavSsJNzUw/0?wx_fmt=png" data-nickname="果子学生信" data-alias="guozixueshengxin" data-signature="记忆力开始下降，需要动笔记录一下。" data-from="0" data-is_biz_ban="0"></mp-common-profile></section><p>从第29节课开始课程就正式进入GEO分析了，这节的内容是使用GSE42872这篇文章的数据进行差异分析。<br><ne-clipboard source="https%3A%2F%2Fwww.yuque.com%2Fyuqueyonghupxlbjg%2Fmikh8s%2Fuualu4yzlk7l22vl%3Flanguage%3Dzh-CN%23Mq7di"></ne-clipboard></p><p><img data-imgfileid="100008805" data-ratio="0.9305555555555556" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icZbRAw90iagO5o506YfODEx98iaG1C1HAIf7M6mVrWFVXXecPNZ6ezpJKlHFOoDNafUlaBTO7slRhgw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" width="738" src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icZbRAw90iagO5o506YfODEx98iaG1C1HAIf7M6mVrWFVXXecPNZ6ezpJKlHFOoDNafUlaBTO7slRhgw/640?wx_fmt=png&amp;from=appmsg"></p><blockquote><blockquote><p>文献的目标是鉴定可能的生物标志物，并探索Vemurafenib对黑色素瘤抑制作用的机制。研究中下载了<strong>GSE42872</strong>的Affymetrix微阵列数据，进行了差异表达基因（Differentially Expressed Genes, DEGs）的识别。通过<strong>基因本体（Gene Ontology, GO）和通路富集分析</strong>，以及<strong>蛋白质-蛋白质相互作用（Protein-Protein Interaction, PPI）</strong>网络的构建，进一步筛选了PPI网络的功能模块，并进行了<strong>KEGG通路富集分析</strong>。<br>研究结果显示，共筛选出794个上调转录本对应214个基因和977个下调转录本对应325个基因。下调的DEGs在细胞周期、DNA复制和p53信号通路中显著富集。上调的DEGs在磷脂酰肌醇信号系统和肌醇磷酸代谢中显著富集。上调DEGs主要富集在单一多细胞生物过程和多细胞生物过程。此外，模块中的细胞周期蛋白6（Cell division cycle 6）、检查点激酶1（Checkpoint kinase 1, CHEK1）、E2F转录因子1（E2F transcription factor 1, E2F1）、表皮生长因子受体（Epidermal growth factor receptor, EGFR）和磷脂酰肌醇-3-激酶调节亚基1-α（Phosphoinositide-3-kinase, regulatory subunit 1-α, PIK3R1）显著富集在细胞周期、凋亡、粘着斑和DNA复制等通路。<br>研究的结论是，<strong>细胞周期蛋白6、CHEK1、E2F1、EGFR和PIK3R1及其相关的通路、细胞周期和粘着斑可能在Vemurafenib抑制黑色素瘤进展中发挥重要作用</strong> 。</p></blockquote></blockquote><h3><span>读入数据</span></h3><pre><code><span> 1</span>exprSet &lt;- read.table(<span>"data/GSE42872_series_matrix.txt"</span>,<br><span> 2</span>                      comment.char=<span>"!"</span>,<br><span> 3</span>                      stringsAsFactors=<span>F</span>,<br><span> 4</span>                      header=<span>T</span>)<br><span> 5</span>head(exprSet)<br><span> 6</span>   ID_REF GSM1052615 GSM1052616 GSM1052617 GSM1052618 GSM1052619 GSM1052620<br><span> 7</span><span>1</span> <span>7892501</span>    <span>7.24559</span>    <span>6.80686</span>    <span>7.73301</span>    <span>6.18961</span>    <span>7.05335</span>    <span>7.20371</span><br><span> 8</span><span>2</span> <span>7892502</span>    <span>6.82711</span>    <span>6.70157</span>    <span>7.02471</span>    <span>6.20493</span>    <span>6.76554</span>    <span>6.24252</span><br><span> 9</span><span>3</span> <span>7892503</span>    <span>4.39977</span>    <span>4.50781</span>    <span>4.88250</span>    <span>4.36295</span>    <span>4.18137</span>    <span>4.73492</span><br><span>10</span><span>4</span> <span>7892504</span>    <span>9.48025</span>    <span>9.67952</span>    <span>9.63074</span>    <span>9.69200</span>    <span>9.91324</span>    <span>9.65897</span><br><span>11</span><span>5</span> <span>7892505</span>    <span>4.54734</span>    <span>4.45247</span>    <span>5.11753</span>    <span>4.87307</span>    <span>5.15505</span>    <span>3.99340</span><br><span>12</span><span>6</span> <span>7892506</span>    <span>6.80701</span>    <span>6.90597</span>    <span>6.72472</span>    <span>6.77028</span>    <span>6.77058</span>    <span>6.77685</span><br><span>13</span><span>### 高频操作来了：第一列变成行名</span><br><span>14</span><span>### 分两步操作</span><br><span>15</span>rownames(exprSet) &lt;- exprSet[,<span>1</span>]<br><span>16</span>exprSet &lt;- exprSet[,-<span>1</span>]<br></code></pre><p>结果：<br><ne-clipboard source="https%3A%2F%2Fwww.yuque.com%2Fyuqueyonghupxlbjg%2Fmikh8s%2Fuualu4yzlk7l22vl%3Flanguage%3Dzh-CN%23Mq7di"></ne-clipboard></p><p><img data-imgfileid="100008806" data-ratio="0.5305555555555556" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icZbRAw90iagO5o506YfODEx9vGaPfRTZQaicp22TUsfn79uPHiazI5ENAP5qkCicDr0duWic4bdGicicbNZQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" width="687" src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icZbRAw90iagO5o506YfODEx9vGaPfRTZQaicp22TUsfn79uPHiazI5ENAP5qkCicDr0duWic4bdGicicbNZQ/640?wx_fmt=png&amp;from=appmsg"></p><h3><span>数据预处理、探针ID转换、去重</span></h3><p><strong>对数据进行对数变换（log transformation）</strong>是一种常见的方法，用于改善数据的正态性和等方差性，或减少数据的偏态。</p><pre><code><span> 1</span><span>## 自动log化</span><br><span> 2</span><span>## 注意此时数据的格式是矩阵</span><br><span> 3</span>ex &lt;- exprSet<br><span> 4</span>qx &lt;- as.numeric(quantile(ex, c(<span>0.</span>, <span>0.25</span>, <span>0.5</span>, <span>0.75</span>, <span>0.99</span>, <span>1.0</span>), na.rm=<span>T</span>))<span>#na.rm=T参数表示在计算分位数时忽略NA值。</span><br><span> 5</span>LogC &lt;- (qx[<span>5</span>] &gt; <span>100</span>) ||<br><span> 6</span>  (qx[<span>6</span>]-qx[<span>1</span>] &gt; <span>50</span> &amp;&amp; qx[<span>2</span>] &gt; <span>0</span>) ||<br><span> 7</span>  (qx[<span>2</span>] &gt; <span>0</span> &amp;&amp; qx[<span>2</span>] &lt; <span>1</span> &amp;&amp; qx[<span>4</span>] &gt; <span>1</span> &amp;&amp; qx[<span>4</span>] &lt; <span>2</span>)<br><span> 8</span><span>## 开始判断有没有0的，防止报错</span><br><span> 9</span><span>if</span> (LogC) { <br><span>10</span>  ex[which(ex &lt;= <span>0</span>)] &lt;- <span>NaN</span><br><span>11</span>  <span>## 取log2</span><br><span>12</span>  exprSet &lt;- log2(ex)<br><span>13</span>  print(<span>"log2 transform finished"</span>)<br><span>14</span>}<span>else</span>{<br><span>15</span>  print(<span>"log2 transform not needed"</span>)  <br><span>16</span>}<br><span>17</span><span>#[1] "log2 transform not needed"</span><br><span>18</span><span>library</span>(limma) <br><span>19</span>boxplot(exprSet,outline=<span>FALSE</span>, notch=<span>T</span>, las=<span>2</span>)<br><span>20</span><span>### 该函数默认使用quntile 矫正差异 </span><br><span>21</span>exprSet=normalizeBetweenArrays(exprSet)<br><span>22</span>boxplot(exprSet,outline=<span>FALSE</span>, notch=<span>T</span>, las=<span>2</span>)<br><span>23</span><span>## 这步把矩阵转换为数据框很重要</span><br><span>24</span>exprSet &lt;- as.data.frame(exprSet)<br></code></pre><p><strong><img data-imgfileid="100008808" data-ratio="0.8465553235908142" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icZbRAw90iagO5o506YfODEx9YkNeozxU0N6hBuQVOBhjQibwSOWB3Y28RbIdwGlGWU6kiaVq2zibD5x0w/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="958" width="479" src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icZbRAw90iagO5o506YfODEx9YkNeozxU0N6hBuQVOBhjQibwSOWB3Y28RbIdwGlGWU6kiaVq2zibD5x0w/640?wx_fmt=png&amp;from=appmsg"></strong></p><p><strong>||</strong> 连接了三个条件，意味着如果这三个条件中任意一个为TRUE，LogC 将被赋值为TRUE。这种逻辑运算在决定是否需要对数据进行对数变换的上下文中很有用。<br>（）<br>获取平台信息：<strong>"GPL6244"</strong><br><ne-clipboard source="https%3A%2F%2Fwww.yuque.com%2Fyuqueyonghupxlbjg%2Fmikh8s%2Fuualu4yzlk7l22vl%3Flanguage%3Dzh-CN%23Mq7di"></ne-clipboard></p><p><ne-clipboard source="https%3A%2F%2Fwww.yuque.com%2Fyuqueyonghupxlbjg%2Fmikh8s%2Fuualu4yzlk7l22vl%3Flanguage%3Dzh-CN%23Mq7di"></ne-clipboard></p><p><img data-imgfileid="100008809" data-ratio="0.774074074074074" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icZbRAw90iagO5o506YfODEx9jvgiaicJKSq0ynOUeM7zCPgD4XGgQsliaC0ib0pmibssTCcjFP80wb4zM8Q/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" width="762" src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icZbRAw90iagO5o506YfODEx9jvgiaicJKSq0ynOUeM7zCPgD4XGgQsliaC0ib0pmibssTCcjFP80wb4zM8Q/640?wx_fmt=png&amp;from=appmsg"></p><pre><code><span> 1</span><span>## 探针基因名转换（需要数据框）</span><br><span> 2</span><span>##platformMap 中有常见的平台个R注释包的对应关系，这是我整理的。</span><br><span> 3</span><span>## 读取，这都是我们已经讲过的</span><br><span> 4</span>platformMap &lt;- data.table::fread(<span>"resource/platformMap.txt"</span>,data.table = <span>F</span>)<br><span> 5</span><br><span> 6</span>head(platformMap)<br><span> 7</span>                                                                        title<br><span> 8</span><span>1</span> Illumina Sentrix Array Matrix (SAM) - GoldenGate Methylation Cancer Panel I<br><span> 9</span><span>2</span>      Illumina HumanMethylation27 BeadChip (HumanMethylation27_270596_v.1.2)<br><span>10</span><span>3</span>        Illumina HumanMethylation450 BeadChip (HumanMethylation450_15017482)<br><span>11</span><span>4</span>   GE Healthcare/Amersham Biosciences CodeLink™   ADME Rat <span>16</span>-Assay Bioarray<br><span>12</span><span>5</span>                                    [AG] Affymetrix Arabidopsis Genome Array<br><span>13</span><span>6</span>                      [ATH1-<span>121501</span>] Affymetrix Arabidopsis ATH1 Genome Array<br><span>14</span>       gpl                 bioc_package   manufacturer             organism<br><span>15</span><span>1</span> GPL15380     GGHumanMethCancerPanelv1       Illumina         Homo sapiens<br><span>16</span><span>2</span>  GPL8490  IlluminaHumanMethylation27k Illumina, Inc.         Homo sapiens<br><span>17</span><span>3</span> GPL13534 IlluminaHumanMethylation450k Illumina, Inc.         Homo sapiens<br><span>18</span><span>4</span>  GPL2898                    adme16cod  GE Healthcare    Rattus norvegicus<br><span>19</span><span>5</span>    GPL71                           ag     Affymetrix Arabidopsis thaliana<br><span>20</span><span>6</span>   GPL198                   ath1121501     Affymetrix Arabidopsis thaliana<br><span>21</span>  data_row_count<br><span>22</span><span>1</span>           <span>1536</span><br><span>23</span><span>2</span>          <span>27578</span><br><span>24</span><span>3</span>         <span>485577</span><br><span>25</span><span>4</span>           <span>1280</span><br><span>26</span><span>5</span>           <span>8297</span><br><span>27</span><span>6</span>          <span>22810</span><br><span>28</span><br><span>29</span><span>## 从文献获得平台的名称</span><br><span>30</span>index &lt;- <span>"GPL6244"</span><br><span>31</span>platformMap$bioc_package[grep(index,platformMap$gpl)]<br><span>32</span>[<span>1</span>] <span>"hugene10sttranscriptcluster"</span><br><span>33</span><br><span>34</span><span>## 数据储存在bioc_package这一列中</span><br><span>35</span>paste0(platformMap$bioc_package[grep(index,platformMap$gpl)],<span>".db"</span>)<br><span>36</span>[<span>1</span>] <span>"hugene10sttranscriptcluster.db"</span><br><span>37</span><br><span>38</span><span>## 安装R包,可以直接安装，这里用了判断</span><br><span>39</span><span>if</span>(!requireNamespace(<span>"hugene10sttranscriptcluster.db"</span>)){<br><span>40</span>  options(BioC_mirror=<span>"https://mirrors.ustc.edu.cn/bioc/"</span>)<br><span>41</span>  BiocManager::install(<span>"hugene10sttranscriptcluster.db"</span>,update = <span>F</span>,ask = <span>F</span>)<br><span>42</span>} <br><span>43</span><br><span>44</span><span>## 加载R包</span><br><span>45</span><span>library</span>(hugene10sttranscriptcluster.db)<br><span>46</span><br><span>47</span><span>## 获取探针和基因的对应关系：这是探针注释的关键步骤。将探针集的表达值转换为基因集的表达值。</span><br><span>48</span>probe2symbol_df &lt;- toTable(get(<span>"hugene10sttranscriptclusterSYMBOL"</span>))<br><span>49</span>head(probe2symbol_df)<br><span>50</span>  probe_id     symbol<br><span>51</span><span>1</span>  <span>7896746</span>   MTND1P23<br><span>52</span><span>2</span>  <span>7896754</span> SEPTIN7P13<br><span>53</span><span>3</span>  <span>7896759</span>  LINC01128<br><span>54</span><span>4</span>  <span>7896761</span>     SAMD11<br><span>55</span><span>5</span>  <span>7896779</span>     KLHL17<br><span>56</span><span>6</span>  <span>7896798</span>    PLEKHN1<br><span>57</span><br><span>58</span><span>## 探针有多少个？</span><br><span>59</span>length(unique(probe2symbol_df$probe_id))<br><span>60</span>[<span>1</span>] <span>19868</span><br><span>61</span><span>## 这么多行中，基因名称有重复的么？</span><br><span>62</span>length(unique(probe2symbol_df$symbol))<br><span>63</span>[<span>1</span>] <span>18857</span><br></code></pre><p><strong>toTable()</strong>: 这个函数将获取的对象转换成表格格式。toTable 是Bioconductor中AnnotationDbi包提供的函数之一，用于将注释信息转换为数据框（data frame）。<br>结果有1011个<strong>基因名字重复</strong>的，下一步进行探针转换以及去重</p><pre><code><span> 1</span><span>### 探针转换以及去重，获得最终的表达矩阵</span><br><span> 2</span><span>library</span>(dplyr)<br><span> 3</span><span>library</span>(tibble)<br><span> 4</span>exprSet &lt;- exprSet %&gt;% <br><span> 5</span>  <span>## 行名转列名,因为只有变成数据框的列,才可以用inner_join</span><br><span> 6</span>  rownames_to_column(<span>"probe_id"</span>) %&gt;% <br><span> 7</span>  <span>## 合并探针的信息</span><br><span> 8</span>  inner_join(probe2symbol_df,by=<span>"probe_id"</span>) %&gt;% <br><span> 9</span>  <span>## 去掉多余信息</span><br><span>10</span>  select(-probe_id) %&gt;%  <br><span>11</span>  <span>## 重新排列</span><br><span>12</span>  select(symbol,everything()) %&gt;%  <br><span>13</span>  <span>## rowMeans求出行的平均数(这边的.代表上面传入的数据)</span><br><span>14</span>  <span>## .[,-1]表示去掉出入数据的第一列，然后求行的平均值</span><br><span>15</span>  mutate(rowMean =rowMeans(.[,-<span>1</span>])) %&gt;% <br><span>16</span>  <span>## 把表达量的平均值按从大到小排序</span><br><span>17</span>  arrange(desc(rowMean)) %&gt;% <br><span>18</span>  <span>## 去重，symbol留下第一个</span><br><span>19</span>  distinct(symbol,.keep_all = <span>T</span>) %&gt;% <br><span>20</span>  <span>## 反向选择去除rowMean这一列</span><br><span>21</span>  select(-rowMean) %&gt;% <br><span>22</span>  <span>## 列名转行名</span><br><span>23</span>  column_to_rownames(<span>"symbol"</span>)<br><span>24</span><br><span>25</span><span>### 保存数据</span><br><span>26</span>save(exprSet,file = <span>"output/exprSet_rmdup.Rdata"</span>)<br></code></pre><p>处理结果：可以进行下一步差异分析了<br><ne-clipboard source="https%3A%2F%2Fwww.yuque.com%2Fyuqueyonghupxlbjg%2Fmikh8s%2Fuualu4yzlk7l22vl%3Flanguage%3Dzh-CN%23Mq7di"></ne-clipboard></p><p><img data-imgfileid="100008810" data-ratio="0.6194444444444445" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icZbRAw90iagO5o506YfODEx9FPMqUbeO74sIib3aobQAGCmicLYpWsYyPh6xl6ruHmBDjhaEQIEgeluA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" width="614.5" src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icZbRAw90iagO5o506YfODEx9FPMqUbeO74sIib3aobQAGCmicLYpWsYyPh6xl6ruHmBDjhaEQIEgeluA/640?wx_fmt=png&amp;from=appmsg"></p><blockquote><p>此处我们借用R包进行注释。那么没有R包，还有三种情况进行注释：<br>1、我们可以自己从平台文件中获取探针和基因的对应关系。<br><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzIyMzA2MTcwMg==&amp;mid=2650732920&amp;idx=1&amp;sn=265f76c1d22240b55fc9b0bdfd74f3f1&amp;chksm=f029aad1c75e23c7abdc42734a2e78cbf83f11f89344db9c97aa922584cf5b37334b040b182a&amp;scene=21#wechat_redirect" textvalue="skr！GEO芯片数据的探针ID转换" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">skr！GEO芯片数据的探针ID转换</a><br>2、NM_，NR_开头的识别号转换成基因名称<br><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzIyMzA2MTcwMg==&amp;mid=2650733383&amp;idx=1&amp;sn=7b27ff1abac2d1cd9f4bf73cc7b0e3ec&amp;chksm=f029a8eec75e21f8017ff9ed8dc230e66e2dec592822f7d374323f58fb69bdb25c033f4178c7&amp;scene=21#wechat_redirect" textvalue="GEO芯片中的NM_，NR_开头的识别号如何转换成基因名称？" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">GEO芯片中的NM_，NR_开头的识别号如何转换成基因名称？</a><br>3、非编码芯片，只有序列没有其他可用信息的芯片<br><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzIyMzA2MTcwMg==&amp;mid=2650733629&amp;idx=1&amp;sn=4ca3c558014c4fb22c5997b7766c9b36&amp;chksm=f029af94c75e268248c7e81a317a9ea6a54d9f082f25a0f063ff0e12086576edbd6d2843a1c0&amp;scene=21#wechat_redirect" textvalue="GEO芯片分析的倒数第2个关卡被没有了。。" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">GEO芯片分析的倒数第2个关卡被没有了。。</a><br></p></blockquote><h3><span>差异分析</span></h3><h4><span><strong>1.创建分组</strong></span></h4><pre><code><span>1</span><span>### 1.创建分组</span><br><span>2</span><span>### 这一步根据样本来就行，原则就是: 跟样本匹配，取决于样本的排序。</span><br><span>3</span>group &lt;- c(rep(<span>"con"</span>,<span>3</span>),rep(<span>"treat"</span>,<span>3</span>)) <br><span>4</span><span>##group &lt;- c("con","con","treat","con","treat","treat") </span><br><span>5</span><span>### 分组变成向量，并且限定leves的顺序</span><br><span>6</span><span>### levels里面，注意要把对照组放在前面</span><br><span>7</span>group &lt;- factor(group,levels = c(<span>"con"</span>,<span>"treat"</span>))<br></code></pre><p><span>注意：group的顺序取决于样本的排序！</span><br><ne-clipboard source="https%3A%2F%2Fwww.yuque.com%2Fyuqueyonghupxlbjg%2Fmikh8s%2Fuualu4yzlk7l22vl%3Flanguage%3Dzh-CN%23Mq7di"></ne-clipboard></p><p><img data-imgfileid="100008811" data-ratio="0.34085778781038373" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icZbRAw90iagO5o506YfODEx9wmGrZ1azjV1ywicERw4b5T4cYQrcZ30knEaXLLoLrzQyLvlr1O76ibPA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="886" width="443" src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icZbRAw90iagO5o506YfODEx9wmGrZ1azjV1ywicERw4b5T4cYQrcZ30knEaXLLoLrzQyLvlr1O76ibPA/640?wx_fmt=png&amp;from=appmsg"></p><blockquote><p>若样本名称太多而且排序不规则，分组太复杂怎么办呢，可以参考这篇文章：</p><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzIyMzA2MTcwMg==&amp;mid=2650733505&amp;idx=1&amp;sn=8a1e4daf5ed5ea7e78f944fb25645adb&amp;chksm=f029a868c75e217e63e839f81184a6abaf7436d45911cac6d51cdd3b128e278db06265e83b5f&amp;scene=21#wechat_redirect" textvalue="GEO的样本名称太多而且排序不规则，你们都是手动分组的么？" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">GEO的样本名称太多而且排序不规则，你们都是手动分组的么？</a><br></p></blockquote><h4><span><strong>2、主成分分析PCA</strong></span></h4><blockquote><p><span></span><strong>主成分分析（PCA，Principal Component Analysis）</strong><span>是一种统计方法，用于数据降维和数据可视化。它</span><strong>通过线性变换将数据转换到新的坐标系统中，使得新坐标轴上的方差最大化</strong><span>。这种方法可以</span><strong>减少数据的维度，同时尽可能保留原始数据的信息。</strong><br><span>PCA 的应用：</span><br><strong>数据降维</strong><span>：减少数据的维度，同时尽可能保留原始数据的信息。</span><br><strong>数据可视化</strong><span>：将高维数据投影到二维或三维空间，便于观察和分析。</span><br><strong>特征提取</strong><span>：在机器学习中，PCA可以作为特征提取的一种方法，帮助减少模型的复杂度和计算量。</span></p></blockquote><pre><code><span> 1</span><span>### 主成分分析PCA：提前预测结果</span><br><span> 2</span><span>### 行是样本，列是基因</span><br><span> 3</span><br><span> 4</span>res.pca &lt;- prcomp(t(exprSet), scale = <span>TRUE</span>)<br><span> 5</span><br><span> 6</span><span>### 有表达值恒定的基因，会产生报错</span><br><span> 7</span><span>if</span>(<span>F</span>){<br><span> 8</span>  <span># 1.转置</span><br><span> 9</span>  data &lt;- t(exprSet)<br><span>10</span>  <span>## 2.NAN值变成0</span><br><span>11</span>  data[is.nan(data)] &lt;- <span>0</span> <br><span>12</span>  <span>## 3.删除没有变异的基因，使用函数var</span><br><span>13</span>  data &lt;-  data[,which(apply(data,<span>2</span>,var) != <span>0</span>)]<br><span>14</span>  res.pca &lt;- prcomp(data, scale = <span>TRUE</span>)<br><span>15</span>}<br><span>16</span><br><span>17</span><span>library</span>(factoextra)<br><span>18</span>fviz_pca_ind(res.pca,col.ind = group)<br></code></pre><p>分开明显，说明差异基因很多<br><ne-clipboard source="https%3A%2F%2Fwww.yuque.com%2Fyuqueyonghupxlbjg%2Fmikh8s%2Fuualu4yzlk7l22vl%3Flanguage%3Dzh-CN%23Mq7di"></ne-clipboard></p><p><img data-imgfileid="100008812" data-ratio="0.7166666666666667" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icZbRAw90iagO5o506YfODEx9Dc82ib9Bz5nDr9JWoxy63ht0j3m9eubkGd5lE2D9hl1s7l6Tia0nplxQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" width="660" src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icZbRAw90iagO5o506YfODEx9Dc82ib9Bz5nDr9JWoxy63ht0j3m9eubkGd5lE2D9hl1s7l6Tia0nplxQ/640?wx_fmt=png&amp;from=appmsg"></p><h4><span><strong>3、构建比较矩阵，进行差异分析</strong></span></h4><p><img data-imgfileid="100008813" data-ratio="0.9476987447698745" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icZbRAw90iagO5o506YfODEx9AlS0xC5d1U8oWHbjl9CXciajCdFnLZyY8fSKFlicZtNgm0W4bOlVuGiag/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="478" width="239" src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icZbRAw90iagO5o506YfODEx9AlS0xC5d1U8oWHbjl9CXciajCdFnLZyY8fSKFlicZtNgm0W4bOlVuGiag/640?wx_fmt=png&amp;from=appmsg"></p><pre><code><span> 1</span><span>### 1.构建比较矩阵。注意顺序不能错。</span><br><span> 2</span>design &lt;- model.matrix(~group)<br><span> 3</span><span>### 比较矩阵命名</span><br><span> 4</span>colnames(design) &lt;- levels(group)<br><span> 5</span>design<br><span> 6</span><br><span> 7</span><span>### 2.线性模型拟合</span><br><span> 8</span>fit &lt;- lmFit(exprSet,design)<br><span> 9</span><span>### 3.贝叶斯检验</span><br><span>10</span>fit2 &lt;- eBayes(fit)<br><span>11</span><span>### 4.输出差异分析结果,其中coef的数目不能操过design的列数</span><br><span>12</span><span>### 此处的2代表的是design中第二列和第一列的比较</span><br><span>13</span>allDiff=topTable(fit2,adjust=<span>'fdr'</span>,coef=<span>2</span>,number=<span>Inf</span>) <span>#不用限定</span><br><span>14</span><span>### 这个数据很重要需要保存一下</span><br><span>15</span>save(allDiff,file = <span>"output/allDiff.Rdata"</span>)<br></code></pre><p>保存结果:所有基因的数据，下一步<strong>定义差异基因</strong><br><ne-clipboard source="https%3A%2F%2Fwww.yuque.com%2Fyuqueyonghupxlbjg%2Fmikh8s%2Fuualu4yzlk7l22vl%3Flanguage%3Dzh-CN%23Mq7di"></ne-clipboard></p><blockquote><p><strong>差异基因（Differential Gene Expression）</strong><span>通常指的是在不同条件下（如不同组织、不同时间点、不同疾病状态等）</span><strong>表达水平发生显著变化</strong><span>的基因。这种变化可能与生物学过程、疾病发展或对治疗的反应有关。</span></p><p><span><span>为了减少假阳性结果，通常会设定一个显著性阈值。这可能是</span><strong>p值（通常小于0.05或0.01）、FDR（假发现率，通常小于0.1或0.05）</strong><span>或其他统计指标。</span><br><span>除了显著性阈值，还需要设定一个表达变化的</span><strong>阈值</strong><span>。这通常是通过计算基因表达的倍数变化（如2倍变化）来确定的。</span><br><span>在进行大量基因的比较时，需要使用</span><strong>多重比较校正方法</strong><span>（如Bonferroni校正、Benjamini-Hochberg校正）来控制假发现率。</span></span></p></blockquote><p><img data-imgfileid="100008814" data-ratio="0.7689243027888446" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icZbRAw90iagO5o506YfODEx9QsYpzFxP9tq90LPAoGd9Noc1hZoY3eRZlpY8fsutORZmxERHOaQwPQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1004" width="502" src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icZbRAw90iagO5o506YfODEx9QsYpzFxP9tq90LPAoGd9Noc1hZoY3eRZlpY8fsutORZmxERHOaQwPQ/640?wx_fmt=png&amp;from=appmsg"></p><h3><span>定义差异基因</span></h3><blockquote><p><strong>定义差异基因：差异倍数2倍，矫正后的p值小于0.05</strong></p></blockquote><p><strong>1、差异倍数（Fold Change, FC）：</strong><br>差异倍数是指在不同条件下，基因表达水平的相对变化。通常用倍数变化来衡量，比如2倍变化意味着在一种条件下的表达水平是另一种条件下的两倍。<br><strong>2、矫正后的p值（Adjusted p-value）：</strong><br>在统计分析中，p值用于衡量观察到的数据与零假设（即没有差异）之间的一致性。p值越小，表示观察到的差异越不可能是随机发生的。<br>由于在基因表达分析中通常需要比较成千上万的基因，因此需要进行多重比较校正，以控制假发现率（FDR）。常用的校正方法包括Bonferroni校正、Benjamini-Hochberg校正等。<br>矫正后的p值小于0.05通常被认为具有统计显著性，表示基因表达的差异不太可能是偶然发生的。</p><pre><code><span> 1</span><span>### 定义差异基因：差异倍数2倍，矫正后的p值小于0.05</span><br><span> 2</span><span>###方案1：</span><br><span> 3</span><span>library</span>(dplyr)<br><span> 4</span>diffgene &lt;- allDiff %&gt;% <br><span> 5</span>  filter(adj.P.Val &lt; <span>0.05</span>) %&gt;% <br><span> 6</span>  filter(abs(logFC) &gt;<span>1</span>)<br><span> 7</span><br><span> 8</span><span>###方案2： 如果出现行名丢失的情况，需要先把行名变成列，处理完毕后再把列变成行名</span><br><span> 9</span><span>### 这个工作是由tibble这个包里面的rownames_to_column()和column_to_rownames()完成的</span><br><span>10</span><span>library</span>(tibble)<br><span>11</span>diffgene &lt;- allDiff %&gt;% <br><span>12</span>  rownames_to_column() %&gt;%  <span>#保护行名</span><br><span>13</span>  filter(adj.P.Val &lt; <span>0.05</span>) %&gt;% <br><span>14</span>  filter(abs(logFC) &gt;<span>1</span>) %&gt;% <br><span>15</span>  column_to_rownames() <span>#保护行名</span><br><span>16</span><br><span>17</span><span>###方案3:使用subset直接获取,&amp;是and的意思</span><br><span>18</span>diffgene &lt;- subset(allDiff,abs(logFC) &gt;<span>1</span> &amp; adj.P.Val &lt; <span>0.05</span>)<br><span>19</span>test &lt;- allDiff[allDiff$adj.P.Val &lt; <span>0.05</span> &amp; abs(allDiff$logFC)&gt;<span>1</span>,]<br><span>20</span><span>### 该数据也需要保存，此处一次性保存两个数据，如果是多个，一次写入变量名称即可。</span><br><span>21</span>save(diffgene,group,file = <span>"output/diffgene.Rdata"</span>)<br></code></pre><p><img data-imgfileid="100008815" data-ratio="0.7410628019323672" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icZbRAw90iagO5o506YfODEx9nv9a8ibPia2vhbKibwibOBQzXOHUs66DgFROE4rS9GS21YJKeibuG28zicxw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1035" width="517.5" src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icZbRAw90iagO5o506YfODEx9nv9a8ibPia2vhbKibwibOBQzXOHUs66DgFROE4rS9GS21YJKeibuG28zicxw/640?wx_fmt=png&amp;from=appmsg"><ne-clipboard source="https%3A%2F%2Fwww.yuque.com%2Fyuqueyonghupxlbjg%2Fmikh8s%2Fuualu4yzlk7l22vl%3Flanguage%3Dzh-CN%23Mq7di"></ne-clipboard></p><h3><span>差异分析之作图</span></h3><h4><span><strong>1、清洁数据</strong></span></h4><pre><code><span>1</span><span>### 1.探针ID转换，2.行列转置，3，添加分组信息。最终获得的是数据框</span><br><span>2</span><br><span>3</span><span>### 行列转置</span><br><span>4</span>exprSet &lt;- as.data.frame(t(exprSet))<br><span>5</span><span>### 添加分组信息</span><br><span>6</span>dd &lt;- cbind(group=group,exprSet)<br><span>7</span><span>### 截取部分展示,这就是清洁数据</span><br><span>8</span>test = dd[,<span>1</span>:<span>10</span>]<br></code></pre><p><ne-clipboard source="https%3A%2F%2Fwww.yuque.com%2Fyuqueyonghupxlbjg%2Fmikh8s%2Fuualu4yzlk7l22vl%3Flanguage%3Dzh-CN%23Mq7di"></ne-clipboard></p><p><img data-imgfileid="100008816" data-ratio="0.2675925925925926" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icZbRAw90iagO5o506YfODEx9zIUzfWiaYS7NVkYSYqk3OqTN42vXo6PgVmZcLVcsDltNnq7JG5Jic77Q/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" width="792" src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icZbRAw90iagO5o506YfODEx9zIUzfWiaYS7NVkYSYqk3OqTN42vXo6PgVmZcLVcsDltNnq7JG5Jic77Q/640?wx_fmt=png&amp;from=appmsg"></p><h4><span><strong>2、单基因作图</strong></span></h4><p>可以对任意基因画图（以CD36基因为例子）</p><pre><code><span>1</span><span>library</span>(ggplot2)<br><span>2</span>ggplot(data = dd,aes(x=group,y=CD36,fill=group))+<br><span>3</span>  geom_boxplot()+<br><span>4</span>  geom_point()+<br><span>5</span>  theme_bw()<br></code></pre><p><ne-clipboard data="%7B%22type%22%3A%22fragment%22%2C%22name%22%3A%22%23fragment%22%2C%22children%22%3A%5B%7B%22type%22%3A%22card%22%2C%22id%22%3A%22u7b364191%22%2C%22name%22%3A%22image%22%2C%22attrs%22%3A%7B%22value%22%3A%7B%22src%22%3A%22https%3A%2F%2Fcdn.nlark.com%2Fyuque%2F0%2F2024%2Fpng%2F33644105%2F1721145014701-5fe96ade-0a79-4b6e-8daf-86b23f1fd400.png%22%2C%22original%22%3A%7B%22type%22%3A%22binary%22%2C%22from%22%3A%22paste%22%2C%22ratio%22%3A2%2C%22width%22%3A1247%2C%22height%22%3A776%7D%2C%22name%22%3A%22image.png%22%2C%22size%22%3A15941%2C%22width%22%3A623.5%2C%22status%22%3A%22done%22%2C%22style%22%3A%22none%22%2C%22taskId%22%3A%22ube21efa3-07dd-4913-8667-584c160483c%22%2C%22clientId%22%3A%22u8ff0b699-ddb4-4%22%2C%22linkExternal%22%3Atrue%2C%22ocr%22%3A%5B%7B%22x%22%3A50%2C%22y%22%3A75%2C%22width%22%3A30%2C%22height%22%3A21%2C%22text%22%3A%2210%22%7D%2C%7B%22x%22%3A1121%2C%22y%22%3A294%2C%22width%22%3A74%2C%22height%22%3A23%2C%22text%22%3A%22GROUP%22%7D%2C%7B%22x%22%3A11%2C%22y%22%3A313%2C%22width%22%3A32%2C%22height%22%3A78%2C%22text%22%3A%22CD36%22%7D%2C%7B%22x%22%3A1178%2C%22y%22%3A344%2C%22width%22%3A41%2C%22height%22%3A18%2C%22text%22%3A%22CON%22%7D%2C%7B%22x%22%3A1178%2C%22y%22%3A390%2C%22width%22%3A52%2C%22height%22%3A18%2C%22text%22%3A%22TREAT%22%7D%2C%7B%22x%22%3A783%2C%22y%22%3A701%2C%22width%22%3A48%2C%22height%22%3A19%2C%22text%22%3A%22TREAT%22%7D%2C%7B%22x%22%3A339%2C%22y%22%3A703%2C%22width%22%3A39%2C%22height%22%3A16%2C%22text%22%3A%22CON%22%7D%2C%7B%22x%22%3A545%2C%22y%22%3A737%2C%22width%22%3A77%2C%22height%22%3A24%2C%22text%22%3A%22GROUP%22%7D%5D%2C%22search%22%3A%2210%20GROUP%20CD36%20CON%20TREAT%20TREAT%20CON%20GROUP%22%2C%22crop%22%3A%5B0%2C0%2C1%2C1%5D%2C%22showTitle%22%3Afalse%2C%22title%22%3A%22%22%2C%22rotation%22%3A0%2C%22averageHue%22%3A%22%23fbfbfb%22%2C%22__spacing%22%3A%22both%22%7D%2C%22cardType%22%3A%22inline%22%7D%2C%22cardType%22%3A%22inline%22%7D%5D%2C%22attrs%22%3A%7B%7D%7D" source="https%3A%2F%2Fwww.yuque.com%2Fyuqueyonghupxlbjg%2Fmikh8s%2Fuualu4yzlk7l22vl%3Flanguage%3Dzh-CN%23Mq7di"></ne-clipboard></p><p><img data-imgfileid="100008817" data-ratio="0.6222222222222222" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icZbRAw90iagO5o506YfODEx9qiaqB2HM5T91kMqYqYFacaL6TNgutZIwr1nxkyoicjtGCWpvaT17ibm5Q/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" width="623.5" src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icZbRAw90iagO5o506YfODEx9qiaqB2HM5T91kMqYqYFacaL6TNgutZIwr1nxkyoicjtGCWpvaT17ibm5Q/640?wx_fmt=png&amp;from=appmsg"></p><pre><code><span> 1</span>my_comparisons &lt;- list(<br><span> 2</span>  c(<span>"treat"</span>, <span>"con"</span>)<br><span> 3</span>)<br><span> 4</span><span>library</span>(ggpubr)<br><span> 5</span>ggboxplot(<br><span> 6</span>  dd, x = <span>"group"</span>, y = <span>"CD36"</span>,<br><span> 7</span>  color = <span>"group"</span>, palette = c(<span>"#00AFBB"</span>, <span>"#E7B800"</span>),<br><span> 8</span>  add = <span>"jitter"</span><br><span> 9</span>)+<br><span>10</span>  stat_compare_means(comparisons = my_comparisons, method = <span>"t.test"</span>)<br></code></pre><p><ne-clipboard source="https%3A%2F%2Fwww.yuque.com%2Fyuqueyonghupxlbjg%2Fmikh8s%2Fuualu4yzlk7l22vl%3Flanguage%3Dzh-CN%23Mq7di"></ne-clipboard></p><p><img data-imgfileid="100008818" data-ratio="0.6083333333333333" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icZbRAw90iagO5o506YfODEx9Xx9unj9uMVSTJAib8mFUch47Yyn6wcHxB92JWBTUcpnpDlXExgKOkXQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" width="632.5" src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icZbRAw90iagO5o506YfODEx9Xx9unj9uMVSTJAib8mFUch47Yyn6wcHxB92JWBTUcpnpDlXExgKOkXQ/640?wx_fmt=png&amp;from=appmsg"></p><p>还可以改写成<strong>函数</strong></p><pre><code><span> 1</span>diffplot &lt;- <span>function</span>(gene){<br><span> 2</span>  my_comparisons &lt;- list(<br><span> 3</span>    c(<span>"treat"</span>, <span>"con"</span>)<br><span> 4</span>  )<br><span> 5</span>  <span>library</span>(ggpubr)<br><span> 6</span>  ggboxplot(<br><span> 7</span>    dd, x = <span>"group"</span>, y = gene,<br><span> 8</span>    color = <span>"group"</span>, palette = c(<span>"#00AFBB"</span>, <span>"#E7B800"</span>),<br><span> 9</span>    add = <span>"jitter"</span><br><span>10</span>  )+<br><span>11</span>    stat_compare_means(comparisons = my_comparisons, method = <span>"t.test"</span>)<br><span>12</span>}<br><span>13</span>diffplot(<span>"MOXD1"</span>)<br></code></pre><p><ne-clipboard source="https%3A%2F%2Fwww.yuque.com%2Fyuqueyonghupxlbjg%2Fmikh8s%2Fuualu4yzlk7l22vl%3Flanguage%3Dzh-CN%23Mq7di"></ne-clipboard></p><p><img data-imgfileid="100008819" data-ratio="0.612037037037037" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icZbRAw90iagO5o506YfODEx92uWELLqUqNTOdAUvmz7ntmr7msAD5BeRT2zaSVg7R2fVYN7ZFFgCjg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" width="628" src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icZbRAw90iagO5o506YfODEx92uWELLqUqNTOdAUvmz7ntmr7msAD5BeRT2zaSVg7R2fVYN7ZFFgCjg/640?wx_fmt=png&amp;from=appmsg"></p><h4><span><strong>3、多基因作图</strong></span></h4><pre><code><span> 1</span><span>## 先把基因提取出来</span><br><span> 2</span>genelist &lt;- rownames(diffgene)[<span>1</span>:<span>6</span>]<br><span> 3</span><span>## 再提取表达量，使用名称选取行</span><br><span> 4</span>data &lt;- dd[,c(<span>"group"</span>,genelist)]<br><span> 5</span><span>## 用pivot_longer调整数据，数据变长，增加的是行</span><br><span> 6</span><span>library</span>(tidyr)<br><span> 7</span>data &lt;- data %&gt;% <br><span> 8</span>  pivot_longer(cols=-<span>1</span>,<br><span> 9</span>               names_to= <span>"gene"</span>,<br><span>10</span>               values_to = <span>"expression"</span>)<br><span>11</span><span>## 多基因作图</span><br><span>12</span><span>## 作图</span><br><span>13</span>ggplot(data = data,aes(x=gene,y=expression,fill=group))+<br><span>14</span>  geom_boxplot()+<br><span>15</span>  geom_jitter()+<br><span>16</span>  theme_bw()+<br><span>17</span>  stat_compare_means(aes(group=group), label = <span>"p.signif"</span>, method = <span>"t.test"</span>)<br></code></pre><p><ne-clipboard source="https%3A%2F%2Fwww.yuque.com%2Fyuqueyonghupxlbjg%2Fmikh8s%2Fuualu4yzlk7l22vl%3Flanguage%3Dzh-CN%23Mq7di"></ne-clipboard></p><p><img data-imgfileid="100008820" data-ratio="0.6157407407407407" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icZbRAw90iagO5o506YfODEx9lmD9C2IrPJiafBbOZC5opnYIVANdzybqmlchb5HjqSekj0VHzptK9QQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" width="626" src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icZbRAw90iagO5o506YfODEx9lmD9C2IrPJiafBbOZC5opnYIVANdzybqmlchb5HjqSekj0VHzptK9QQ/640?wx_fmt=png&amp;from=appmsg"></p><pre><code><span>1</span><span>## 尝试更清晰的展示</span><br><span>2</span>ggplot(data = data,aes(x=group,y=expression,fill=group))+<br><span>3</span>  geom_boxplot()+<br><span>4</span>  geom_jitter()+<br><span>5</span>  theme_bw()+<br><span>6</span>  facet_grid(.~gene)+<br><span>7</span>  stat_compare_means(comparisons = my_comparisons, label = <span>"p.signif"</span>, method = <span>"t.test"</span>)<br></code></pre><p><ne-clipboard source="https%3A%2F%2Fwww.yuque.com%2Fyuqueyonghupxlbjg%2Fmikh8s%2Fuualu4yzlk7l22vl%3Flanguage%3Dzh-CN%23Mq7di"></ne-clipboard></p><p><img data-imgfileid="100008821" data-ratio="0.6157407407407407" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icZbRAw90iagO5o506YfODEx9t8lTbnM1VQKOEze6bLFBRTAD5td4tMrEeWePf9aAXOHic4hmzWrR45A/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" width="627" src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icZbRAw90iagO5o506YfODEx9t8lTbnM1VQKOEze6bLFBRTAD5td4tMrEeWePf9aAXOHic4hmzWrR45A/640?wx_fmt=png&amp;from=appmsg"></p><p><span><strong>4、图片导出</strong></span></p><pre><code><span>1</span><span>library</span>(export)<br><span>2</span><span>## 导成PPT可编辑的格式</span><br><span>3</span>graph2ppt(file=<span>"output/diffgenboxplot.pptx"</span>)<br><span>4</span><span>## 其他自己想要的格式</span><br><span>5</span>graph2pdf(file=<span>"output/diffgenboxplot.pdf"</span>)<br><span>6</span>graph2tif(file=<span>"output/diffgenboxplot.tif"</span>)<br><span>7</span><span>## 导成AI可以编辑的状态</span><br><span>8</span>graph2eps(file=<span>"output/diffgenboxplot.eps"</span>)<br></code></pre><p>以上就是差异分析（对照组和实验组）的过程啦<br>如果GEO芯片<strong>超过了（con,treat）两组</strong>或是<strong>配对样本</strong>的情况时，如何进行差异分析呢。可以参考果子老师的这两篇文章~<br><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzIyMzA2MTcwMg==&amp;mid=2650733583&amp;idx=2&amp;sn=e68e00a3aa945ce029482a5b2b445b6b&amp;chksm=f029afa6c75e26b07e681f5540a3a32d43cb639ea5e1b04e98316030f99b1b4eba3cbd5e2aea&amp;scene=21#wechat_redirect" textvalue="GEO芯片如果超过了两组，也可以一次搞定差异分析" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">GEO芯片如果超过了两组，也可以一次搞定差异分析</a><br><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzIyMzA2MTcwMg==&amp;mid=2650733588&amp;idx=1&amp;sn=6ecfaa2b1cc74416ce56819ad8ed6ada&amp;chksm=f029afbdc75e26abcf9b63540534b8418bf3c677df529f295b165c58c9a5b4843ee318def478&amp;scene=21#wechat_redirect" textvalue="GEO芯片中配对样本如何做差异分析" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">GEO芯片中配对样本如何做差异分析</a><br></p><h3><span>差异分析之热图</span></h3><blockquote><p><strong>热图</strong>是一种数据可视化技术，通常用于展示数据矩阵中的数值大小。它通过颜色的深浅来表示数值的大小，使得观察者可以直观地识别数据中的模式、趋势和异常值。以下是解读热图的一些基本步骤和技巧：</p></blockquote><p><strong>1. 颜色编码：</strong></p><ul><li><p><span>热图中的颜色通常从一种极端（如蓝色或绿色）渐变到另一种极端（如红色或黄色），表示数值的从小到大的变化。</span></p></li><li><p><span>通常，颜色条（或图例）会显示颜色和数值的对应关系。</span></p></li></ul><p><strong>2. 矩阵布局：</strong></p><ul><li><p><span>热图通常以矩阵的形式展示，行和列分别代表数据的两个维度。例如，行代表不同的样本，列代表不同基因的表达变量。</span></p></li></ul><p><strong>3. 数值大小：</strong></p><ul><li><p><span>颜色的深浅直接反映了数值的大小。颜色越深，数值越大；颜色越浅，数值越小。</span></p></li></ul><p><strong>4. 模式识别：</strong></p><ul><li><p><span>观察热图中的色块分布，识别数据中的模式。某些行或列的颜色可能特别深或特别浅，这可能表示某些样本或特征在数据集中特别突出。</span></p></li></ul><p><strong>5. 趋势分析：</strong></p><ul><li><p><span>观察颜色的变化趋势，分析数据随时间或其他变量的变化。例如，如果热图中的某些行或列的颜色逐渐变深，可能表示某些变量或样本的数值在增加。</span></p></li></ul><pre><code><span> 1</span><span>### 用行名提取数据</span><br><span> 2</span>rm(list = ls())<br><span> 3</span><span>## 1.加载表达数据</span><br><span> 4</span>load(file = <span>"output/exprSet_rmdup.Rdata"</span>)<br><span> 5</span><span>## 2.加载差异列表</span><br><span> 6</span>load(file = <span>"output/allDiff.Rdata"</span>)<br><span> 7</span><span>library</span>(dplyr)<br><span> 8</span><span>library</span>(tibble)<br><span> 9</span>diffgene &lt;- allDiff %&gt;% <br><span>10</span>  rownames_to_column() %&gt;% <br><span>11</span>  filter(adj.P.Val &lt; <span>0.05</span>) %&gt;% <br><span>12</span>  filter(abs(logFC) &gt;<span>1</span>) %&gt;% <br><span>13</span>  column_to_rownames()<br><span>14</span><br><span>15</span><span>## 3.用名称提取部分数据用作热图绘制</span><br><span>16</span>heatdata &lt;- exprSet[rownames(diffgene),]<br><span>17</span><span>## 4.制作一个分组信息annotation_col用于注释</span><br><span>18</span>group &lt;- c(rep(<span>"con"</span>,<span>3</span>),rep(<span>"treat"</span>,<span>3</span>)) <br><span>19</span>annotation_col &lt;- data.frame(group)<br><span>20</span>rownames(annotation_col) &lt;- colnames(heatdata)<br><span>21</span><br><span>22</span><span>## 加载热图的R包</span><br><span>23</span><span>library</span>(pheatmap)<br><span>24</span><span>## 颜色包 viridisLite</span><br><span>25</span><span>library</span>(viridisLite)<br><span>26</span><span>## 直接作图</span><br><span>27</span>pheatmap(heatdata)</code></pre><p><img data-imgfileid="100008826" data-ratio="0.6037037037037037" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icblpTTS0L69b3ctKoEIibgpzEEibBExFV9peE7eZvbykEohFHcTDXFfONdLiaxJtMhn823lHXGuqyVhA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" width="632.5" src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icblpTTS0L69b3ctKoEIibgpzEEibBExFV9peE7eZvbykEohFHcTDXFfONdLiaxJtMhn823lHXGuqyVhA/640?wx_fmt=png&amp;from=appmsg"></p><p>太粗糙，需要修饰一下~~</p><pre><code><span> 1</span><span>### 经过修饰的图</span><br><span> 2</span>pheatmap(heatdata, <span>#热图的数据</span><br><span> 3</span>         cluster_rows = <span>TRUE</span>,<span>#行聚类</span><br><span> 4</span>         cluster_cols = <span>TRUE</span>,<span>#列聚类，可以看出样本之间的区分度</span><br><span> 5</span>         annotation_col =annotation_col, <span>#标注样本分类</span><br><span> 6</span>         annotation_legend=<span>TRUE</span>, <span># 显示注释</span><br><span> 7</span>         show_rownames = <span>F</span>,<span># 显示行名</span><br><span> 8</span>         scale = <span>"row"</span>, <span>#以行来标准化，这个功能很不错</span><br><span> 9</span>         color = viridis(<span>10</span>, alpha = <span>1</span>, begin = <span>0.5</span>, end = <span>1</span>, direction = <span>1</span>),<span>#颜色</span><br><span>10</span>         cellwidth = <span>60</span>, <span># 格子宽度</span><br><span>11</span>         cellheight = <span>0.2</span>,<span># 格子高度</span><br><span>12</span>         fontsize = <span>10</span> <span># 字体大小</span><br><span>13</span>         )</code></pre><p><img data-imgfileid="100008825" data-ratio="0.8657407407407407" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icblpTTS0L69b3ctKoEIibgpzsGg0UDuYH4raa4vtSeibWicuIxicOPjI740GSLWxbIibmA7dJgI7vU8PEQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" width="551.5" src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icblpTTS0L69b3ctKoEIibgpzsGg0UDuYH4raa4vtSeibWicuIxicOPjI740GSLWxbIibmA7dJgI7vU8PEQ/640?wx_fmt=png&amp;from=appmsg"></p><p>处理组和对照组分的很开，聚类明显，实验操作没问题。<br>但是差异基因较多，我们可以重新筛选，阈值设大一点</p><pre><code><span> 1</span>diffgene &lt;- allDiff %&gt;% <br><span> 2</span>  rownames_to_column() %&gt;% <br><span> 3</span>  filter(adj.P.Val &lt; <span>0.05</span>) %&gt;% <br><span> 4</span>  filter(abs(logFC) &gt;<span>3</span>) %&gt;% <br><span> 5</span>  column_to_rownames()<br><span> 6</span><br><span> 7</span><span>## 用名称提取差异基因用作热图绘制</span><br><span> 8</span>heatdata &lt;- exprSet[rownames(diffgene),]<br><span> 9</span><span>## 制作一个分组信息用于注释</span><br><span>10</span>group &lt;- c(rep(<span>"con"</span>,<span>3</span>),rep(<span>"treat"</span>,<span>3</span>)) <br><span>11</span>annotation_col &lt;- data.frame(group)<br><span>12</span>rownames(annotation_col) &lt;- colnames(heatdata)<br><span>13</span><br><span>14</span><span>#如果注释出界，可以通过调整格子比例和字体修正</span><br><span>15</span>pheatmap(heatdata, <span>#热图的数据</span><br><span>16</span>         cluster_rows = <span>TRUE</span>,<span>#行聚类</span><br><span>17</span>         cluster_cols = <span>TRUE</span>,<span>#列聚类，可以看出样本之间的区分度</span><br><span>18</span>         annotation_col =annotation_col, <span>#标注样本分类</span><br><span>19</span>         annotation_legend=<span>TRUE</span>, <span># 显示注释</span><br><span>20</span>         show_rownames = <span>T</span>,<span># 显示行名（注意）</span><br><span>21</span>         scale = <span>"row"</span>, <span>#以行来标准化，这个功能很不错</span><br><span>22</span>         color = viridis(<span>10</span>, alpha = <span>1</span>, begin = <span>0.5</span>, end = <span>1</span>, direction = <span>1</span>),<span>#调色</span><br><span>23</span>         <span>#filename = "heatmap_F.pdf",#是否保存</span><br><span>24</span>         cellwidth = <span>40</span>, <span># 格子宽度</span><br><span>25</span>         cellheight = <span>14</span>,<span># 格子高度</span><br><span>26</span>         fontsize = <span>10</span> <span># 字体大小</span><br><span>27</span>         )</code></pre><p><img data-imgfileid="100008824" data-ratio="1.0333333333333334" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icblpTTS0L69b3ctKoEIibgpzNEQRHuuKV0c8XibCRuT1gGnYPIlDAyXqbicg4srDyp1o9IEot7Bg1kXw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" width="612.5" src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icblpTTS0L69b3ctKoEIibgpzNEQRHuuKV0c8XibCRuT1gGnYPIlDAyXqbicg4srDyp1o9IEot7Bg1kXw/640?wx_fmt=png&amp;from=appmsg"></p><p>可看得出相关性基因（同一个聚类）</p><h3><span>差异分析之火山图</span></h3><blockquote><p><strong>火山图</strong>是一种常用于生物信息学和基因组学中的图形表示方法，主要用于展示基因表达数据或其他类型的数据。它通常用于展示基因表达的显著性差异，比如在不同条件下的基因表达变化。</p></blockquote><p><strong>火山图的基本结构如下：</strong></p><p><strong>x轴：</strong>通常表示基因表达的对数比值（如log2 fold change），表示<strong>基因表达的变化倍数。</strong></p><p><strong>y轴</strong>通常表示统计显著性（如负对数p值，-log10 p-value），表示<strong>基因表达变化的显著性。</strong></p><p><strong>点</strong>：每个点代表一个基因，点的位置表示该基因在x轴和y轴上的值。</p><p><strong>颜色：</strong>通常用来表示基因表达的变化方向，比如红色表示上调，蓝色表示下调。</p><p><strong>阈值线</strong>通常在图中会画出两条线，表示显著性阈值和表达变化倍数阈值，帮助识别显著差异的基因。</p><p>输入：<ne-clipboard source="https%3A%2F%2Fwww.yuque.com%2Fyuqueyonghupxlbjg%2Fmikh8s%2Fftegbp0pey2n47ux"></ne-clipboard></p><p><img data-imgfileid="100008830" data-ratio="0.6822244289970208" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icblpTTS0L69b3ctKoEIibgpzOOvpxwGiclY6picTiaFuYib0lKV73B7fLOAHrGwNrwZGXEVd5GVIs3Grpw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1007" width="503.5" src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icblpTTS0L69b3ctKoEIibgpzOOvpxwGiclY6picTiaFuYib0lKV73B7fLOAHrGwNrwZGXEVd5GVIs3Grpw/640?wx_fmt=png&amp;from=appmsg"></p><p>要添加gene列（将行名变列）<br><ne-clipboard source="https%3A%2F%2Fwww.yuque.com%2Fyuqueyonghupxlbjg%2Fmikh8s%2Fftegbp0pey2n47ux"></ne-clipboard></p><p><img data-imgfileid="100008829" data-ratio="0.662962962962963" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icblpTTS0L69b3ctKoEIibgpzsRkYrsXCOEZJnaSqOalibcb6OfxiaXJZqRgQMwo2NqvnuW6aTdhCvl9g/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" width="609.5" src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icblpTTS0L69b3ctKoEIibgpzsRkYrsXCOEZJnaSqOalibcb6OfxiaXJZqRgQMwo2NqvnuW6aTdhCvl9g/640?wx_fmt=png&amp;from=appmsg"></p><pre><code><span> 1</span>rm(list = ls())<br><span> 2</span><span>##用ggplot2</span><br><span> 3</span><span>library</span>(ggplot2)<br><span> 4</span><span>library</span>(ggrepel)<br><span> 5</span><span>library</span>(dplyr)<br><span> 6</span>load(file = <span>"output/allDiff.Rdata"</span>)<br><span> 7</span><span>### 无论是什么名称，都改为data</span><br><span> 8</span>data &lt;- allDiff<br><span> 9</span><span>### 如果没有gene这一列就需要添加</span><br><span>10</span>data$gene &lt;- rownames(data)<br><span>11</span><span>## 仔细观察data数据</span><br><span>12</span><span>## logFC，P.Value，gene</span><br><span>13</span><span>### 火山图</span><br><span>14</span>logFCfilter = <span>1.5</span><span>#控制竖线，可以改变 </span><br><span>15</span>logFCcolor = <span>3.5</span><span>#在3.5化别的颜色</span><br><span>16</span><span>### 标记上下调</span><br><span>17</span>index = data$adj.P.Val &lt;<span>0.05</span> &amp; abs(data$logFC) &gt; logFCfilter<br><span>18</span>data$group &lt;- <span>0</span><br><span>19</span>data$group[index &amp; data$logFC&gt;<span>0</span>] = <span>1</span><br><span>20</span>data$group[index &amp; data$logFC&lt;<span>0</span>] = -<span>1</span><br><span>21</span>data$group &lt;- factor(data$group,levels = c(<span>1</span>,<span>0</span>,-<span>1</span>),labels =c(<span>"Up"</span>,<span>"NS"</span>,<span>"Down"</span>) )<br><span>22</span><span>### 正式画图</span><br><span>23</span>ggplot(data=data, aes(x=logFC, y =-log10(adj.P.Val),color=group)) +<br><span>24</span>  geom_point(alpha=<span>0.8</span>, size=<span>1.2</span>)+<br><span>25</span>  scale_color_manual(values = c(<span>"red"</span>, <span>"grey50"</span>, <span>"blue4"</span>))+<br><span>26</span>  labs(x=<span>"log2 (fold change)"</span>,y=<span>"-log10 (adj.P.Val)"</span>)+<br><span>27</span>  theme(plot.title = element_text(hjust = <span>0.4</span>))+<br><span>28</span>  geom_hline(yintercept = -log10(<span>0.05</span>),lty=<span>4</span>,lwd=<span>0.6</span>,alpha=<span>0.8</span>)+<br><span>29</span>  geom_vline(xintercept = c(-logFCfilter,logFCfilter),lty=<span>4</span>,lwd=<span>0.6</span>,alpha=<span>0.8</span>)+<br><span>30</span>  theme_bw()+<br><span>31</span>  theme(panel.border = element_blank(),<br><span>32</span>        panel.grid.major = element_blank(),<br><span>33</span>        panel.grid.minor = element_blank(),<br><span>34</span>        axis.line = element_line(colour = <span>"black"</span>)) +<br><span>35</span>  theme(legend.position=<span>"top"</span>)+<br><span>36</span>  geom_point(data=subset(data, abs(logFC) &gt;= logFCcolor &amp; adj.P.Val &lt;<span>0.05</span>),alpha=<span>0.8</span>, size=<span>3</span>,col=<span>"green4"</span>)+<br><span>37</span>  geom_text_repel(data=subset(data, abs(logFC) &gt;= logFCcolor &amp; adj.P.Val &lt;<span>0.05</span>),<br><span>38</span>                  aes(label=gene),col=<span>"black"</span>,alpha = <span>0.8</span>)<br></code></pre><p><img data-imgfileid="100008828" data-ratio="0.6074074074074074" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icblpTTS0L69b3ctKoEIibgpzF2QMX5Ce10Ccazm2M4icmvicRwUIQ9JuBQ31CpvmMfuskTriaWjtwOAlQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" width="624.5" src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icblpTTS0L69b3ctKoEIibgpzF2QMX5Ce10Ccazm2M4icmvicRwUIQ9JuBQ31CpvmMfuskTriaWjtwOAlQ/640?wx_fmt=png&amp;from=appmsg"><ne-clipboard source="https%3A%2F%2Fwww.yuque.com%2Fyuqueyonghupxlbjg%2Fmikh8s%2Fftegbp0pey2n47ux"></ne-clipboard></p><h3><span>差异分析之MAplot图</span></h3><blockquote><p><strong>MA图（Mean-Averaging Plot）</strong>是一种在基因表达分析中常用的图形，用于展示基因表达数据的显著性差异。MA图主要用于微阵列数据的分析，帮助识别那些在不同条件下表达水平发生变化的基因。</p></blockquote><p><strong>MA图的基本结构：</strong><br><strong>x轴：</strong>通常表示基因表达的对数比值（log2 fold change），即平均表达值的对数比率。它显示了基因在不同条件下表达水平的变化倍数。<strong>正值表示在比较条件下基因表达上调，负值表示下调。</strong></p><p><strong>y轴：</strong>通常表示基因表达的对数平均值（log2 average expression），即基因在所有条件下表达水平的平均值的对数。它显示了<strong>基因在所有条件下的相对表达水平。</strong></p><p><strong>点：</strong>每个点代表一个基因，点的位置表示该基因在x轴和y轴上的值。</p><p><strong>颜色：</strong>通常用来表示基因表达的变化方向，比如红色表示上调，蓝色表示下调。</p><p><strong>阈值线：</strong>在图中可能会画出一些线，表示显著性阈值，帮助识别显著差异的基因。</p><pre><code><span> 1</span><span>##MAplot，更加实用！！！</span><br><span> 2</span><span>library</span>(ggplot2)<br><span> 3</span><span>library</span>(ggrepel)<br><span> 4</span>logFCfilter = <span>1.5</span><br><span> 5</span>logFCcolor = <span>3</span><br><span> 6</span><span>### 标记上下调</span><br><span> 7</span>index = data$adj.P.Val &lt;<span>0.05</span> &amp; abs(data$logFC) &gt; logFCfilter<br><span> 8</span>data$group &lt;- <span>0</span><br><span> 9</span>data$group[index &amp; data$logFC&gt;<span>0</span>] = <span>1</span><br><span>10</span>data$group[index &amp; data$logFC&lt;<span>0</span>] = -<span>1</span><br><span>11</span>data$group &lt;- factor(data$group,levels = c(<span>1</span>,<span>0</span>,-<span>1</span>),labels =c(<span>"Up"</span>,<span>"NS"</span>,<span>"Down"</span>) )<br><span>12</span>&gt; head(data)<br><span>13</span>          logFC   AveExpr         t      P.Value    adj.P.Val<br><span>14</span>CD36   <span>5.804404</span>  <span>7.376917</span>  <span>84.62377</span> <span>3.019535e-15</span> <span>5.693938e-11</span><br><span>15</span>DUSP6 -<span>4.175519</span>  <span>9.099007</span> -<span>66.72144</span> <span>3.019872e-14</span> <span>2.243317e-10</span><br><span>16</span>DCT    <span>5.681340</span>  <span>8.766397</span>  <span>65.58028</span> <span>3.568941e-14</span> <span>2.243317e-10</span><br><span>17</span>SPRY2 -<span>3.776101</span>  <span>9.704355</span> -<span>56.10048</span> <span>1.617683e-13</span> <span>7.626164e-10</span><br><span>18</span>MOXD1  <span>3.278876</span> <span>10.196318</span>  <span>52.74322</span> <span>2.938746e-13</span> <span>1.108319e-09</span><br><span>19</span>ETV4  -<span>3.818034</span>  <span>9.648439</span> -<span>48.37271</span> <span>6.782826e-13</span> <span>2.034560e-09</span><br><span>20</span>             B  gene group<br><span>21</span>CD36  <span>23.68973</span>  CD36    Up<br><span>22</span>DUSP6 <span>22.28742</span> DUSP6  Down<br><span>23</span>DCT   <span>22.17427</span>   DCT    Up<br><span>24</span>SPRY2 <span>21.08429</span> SPRY2  Down<br><span>25</span>MOXD1 <span>20.62242</span> MOXD1    Up<br><span>26</span>ETV4  <span>19.94776</span>  ETV4  Down<br><span>27</span><span>### 正式画图</span><br><span>28</span>ggplot(data=data, aes(x=log2(AveExpr), y =logFC,color=group)) +<br><span>29</span>  geom_point(alpha=<span>0.8</span>, size=<span>1.2</span>)+<br><span>30</span>  scale_color_manual(values = c(<span>"darkred"</span>, <span>"grey50"</span>, <span>"royalblue4"</span>))+<br><span>31</span>  labs(y=<span>"log2 (Fold Change)"</span>,x=<span>"log2 (Base Mean)"</span>)+<br><span>32</span>  theme(plot.title = element_text(hjust = <span>0.4</span>))+<br><span>33</span>  geom_hline(yintercept = c(logFCfilter,-logFCfilter),lty=<span>2</span>,lwd=<span>1</span>)+<br><span>34</span>  geom_hline(yintercept = <span>0</span>,lwd=<span>1.2</span>)+<br><span>35</span>  theme_bw()+<br><span>36</span>  theme(panel.border = element_blank(),<br><span>37</span>        panel.grid.major = element_blank(),<br><span>38</span>        panel.grid.minor = element_blank(),<br><span>39</span>        axis.line = element_line(colour = <span>"black"</span>)) +<br><span>40</span>  theme(legend.position=<span>"top"</span>)+<br><span>41</span>  geom_point(data=subset(data, abs(logFC) &gt;= logFCcolor &amp; adj.P.Val &lt;<span>0.05</span>),alpha=<span>0.8</span>, size=<span>3</span>,col=<span>"green4"</span>)+<br><span>42</span>  geom_text_repel(data=subset(data, abs(logFC) &gt;= logFCcolor &amp; adj.P.Val &lt;<span>0.05</span>),<br><span>43</span>                  aes(label=gene),col=<span>"black"</span>,alpha = <span>0.8</span>)<br></code></pre><p>倍数大，表达量小说明误差大！<br>应该尽量选择靠右边表达量较大的。<br><ne-clipboard source="https%3A%2F%2Fwww.yuque.com%2Fyuqueyonghupxlbjg%2Fmikh8s%2Fftegbp0pey2n47ux"></ne-clipboard></p><p><img data-imgfileid="100008827" data-ratio="0.612037037037037" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icblpTTS0L69b3ctKoEIibgpzKzNbaTPJhd6Et0cLbGHp99lznkLrV7RDtWFbcHcmzc2vqT6NsWMZWg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" width="623" src="https://mmbiz.qpic.cn/sz_mmbiz_png/dgYTF46r3icblpTTS0L69b3ctKoEIibgpzKzNbaTPJhd6Et0cLbGHp99lznkLrV7RDtWFbcHcmzc2vqT6NsWMZWg/640?wx_fmt=png&amp;from=appmsg"></p><p>注意不要导出成ppt可编辑格式（数据量大会卡顿！）</p><pre><code><span>1</span><span>## 加载R包</span><br><span>2</span><span>library</span>(export)<br><span>3</span><span>## 导出</span><br><span>4</span>graph2ppt(file=<span>"output/volcano.pptx"</span>)<span>##注意不要这样做！电脑会卡</span><br><span>5</span>graph2pdf(file=<span>"output/volcano.pdf"</span>)</code></pre><p><br></p><blockquote><p>感谢果子老师课程中的循循教导和付出，认真学习这个课程，收获很大！<br>对于生信分析感兴趣的同学，还可以看看果子老师其它的生信课程：<br>SC00果子单细胞大课<br>https://shop449275017.v.weidian.com/itemhtml?itemlD=5948666785<br>肿瘤数据挖掘大师课TM00<br>https://shop449275017.v.weidian.com/item.html?itemID=6509772590<br>GZ00果子R数据挖掘大课<br>https://shop449275017.v.weidian.com/item.html?itemID=4410312838</p></blockquote><p><span>作者：黎扬</span></p><p msttexthash="23311756" msthash="54"><span>编辑：</span><span>黎扬</span></p><p msttexthash="24160396" msthash="55"><span msttexthash="4865315" msthash="125">校审</span><span>：林炳昱</span></p><p><br></p><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/nkX-BS_sxlg0JwhBfaCVEw",target="_blank" rel="noopener noreferrer">原文链接</a>
