---
title: "代码分享|| 同一套代码和思路已经发了三篇，提供完整代码和数据的泛癌分析文献分享！"
date: 2024-07-18T13:51:29Z
draft: ["false"]
tags: [
  "fetched",
  "新叶生物"
]
categories: ["Acdemic"]
---
代码分享|| 同一套代码和思路已经发了三篇，提供完整代码和数据的泛癌分析文献分享！ by 新叶生物
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><blockquote data-tool="mdnice编辑器"><span></span><p>最近大家做代码资源分享的公众号非常多，感觉这个赛道已经白热化，<span>你的收藏夹里面是不是已经收藏了N多代码，但是都在吃灰？有个朋友告诉我他已经收藏了100多套了！！！</span>后面我打算做视频讲解复现分析好了，虽然很多文献有代码有数据，但是对大部分没有生信基础的人来说，复现起来应该还是有很大的门槛！</p></blockquote><p data-tool="mdnice编辑器"><span>今天分享的文献有一套代码，使用这套代码 类似的思路差不多发了三篇文献啦！来看看吧！</span></p><p data-tool="mdnice编辑器">提供代码的文献：</p><figure data-tool="mdnice编辑器"><img data-imgfileid="100000715" data-ratio="0.4462962962962963" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/by7zpDFL3LIO8UGBnPDmqnYqFWDcfxXLCWVs7b4xYOdVxbFHh0YaZr85hYPLstnrJ68Gp3xzSErxXV6rTchv8A/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/by7zpDFL3LIO8UGBnPDmqnYqFWDcfxXLCWVs7b4xYOdVxbFHh0YaZr85hYPLstnrJ68Gp3xzSErxXV6rTchv8A/640?wx_fmt=png&amp;from=appmsg"><figcaption>image-20240718203530389</figcaption></figure><p data-tool="mdnice编辑器">其他两篇文献如下：</p><p data-tool="mdnice编辑器">H.-R. Feng, X.-N. Shen, X.-M. Zhu, W.-T. Zhong, D.-X. Zhu, J. Zhao, Y.-J. Chen, F. Shen, K. Liu, L. Liang, <span>Unveiling Major Histocompatibility Complex-mediated Pan-Cancer Immune Features by Integrated Single-Cell and Bulk RNA Sequencing</span>, Cancer Letters, https://doi.org/10.1016/j.canlet.2024.217062</p><p data-tool="mdnice编辑器">Wu, Zhengquan et al. “<span>A transcriptomic pan-cancer signature for survival prognostication and prediction of immunotherapy response based on endothelial senescence.</span>” Journal of biomedical science vol. 30,1 21. 28 Mar. 2023, doi:10.1186/s12929-023-00915-5</p><h3 data-tool="mdnice编辑器"><span></span><span>代码和数据</span><span></span></h3><p data-tool="mdnice编辑器"><span>https://figshare.com/articles/online_resource/Integrated_analysis_of_single-cell_and_bulk_RNA_sequencing_data_reveals_a_pan-cancer_stemness_signature_predicting_immunotherapy_response/17654633</span></p><p data-tool="mdnice编辑器">每一幅都有对应的代码：</p><figure data-tool="mdnice编辑器"><img data-imgfileid="100000714" data-ratio="0.5916542473919523" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/by7zpDFL3LIO8UGBnPDmqnYqFWDcfxXLxKH9hvicW4IoIp3TxCricBOj4l3cxYhGZ1Brgr7IU4ibuDHcG69KEHCpg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="671" src="https://mmbiz.qpic.cn/sz_mmbiz_png/by7zpDFL3LIO8UGBnPDmqnYqFWDcfxXLxKH9hvicW4IoIp3TxCricBOj4l3cxYhGZ1Brgr7IU4ibuDHcG69KEHCpg/640?wx_fmt=png&amp;from=appmsg"><figcaption>image-20240718210855924</figcaption></figure><h3 data-tool="mdnice编辑器"><span></span><span>文献主要内容</span><span></span></h3><p data-tool="mdnice编辑器"><span>看看思路，你是不是也能做一篇出来~</span></p><p data-tool="mdnice编辑器">文章做的癌细胞干性泛癌分析，基本假设：<span>免疫检查点抑制剂（ICI）是癌症治疗的一个突破，但只有有限的一部分患者从中受益。癌症干细胞性可能是ICI耐药性的潜在罪魁祸首，但目前还缺乏直接的临床证据。</span>作者收集了来自ICI治疗的患者的公开scRNA-Seq数据集，以阐明癌症干性和ICI反应之间的关系。</p><p data-tool="mdnice编辑器">所以本文还为大家提供了ICI治疗的单细胞数据队列，可以利用起来，为自己的科学研究课题所用！</p><p data-tool="mdnice编辑器">作者使用大规模泛癌数据：包括34个scRNA-Seq数据集、TCGA数据集、10 ICI转录组队列，开发和验证了 一个新的 stemness signature (Stem.Sig) 。此外，Stem.Sig中的基因（EMC3, BECN1, VPS35, PCBP2, VPS29, PSMF1, GCLC, KXD1, SPRR1B） 使用 17个筛选潜在免疫治疗靶点的CRISPR数据集进行了治疗方面的探索。</p><h3 data-tool="mdnice编辑器"><span></span><span><span>ICI治疗的单细胞数据队列</span></span><span></span></h3><figure data-tool="mdnice编辑器"><img data-imgfileid="100000713" data-ratio="0.35348837209302325" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/by7zpDFL3LIO8UGBnPDmqnYqFWDcfxXLPPSn1hnjpsIo2nW9tpdoroWBNhMPkFCF4kVFSJGjjGZFgkD6KINdnA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="860" src="https://mmbiz.qpic.cn/sz_mmbiz_png/by7zpDFL3LIO8UGBnPDmqnYqFWDcfxXLPPSn1hnjpsIo2nW9tpdoroWBNhMPkFCF4kVFSJGjjGZFgkD6KINdnA/640?wx_fmt=png&amp;from=appmsg"><figcaption>image-20240718205118716</figcaption></figure><h3 data-tool="mdnice编辑器"><span></span><span><span>用于开发Stem.Sig的scRNA数据集</span></span><span></span></h3><figure data-tool="mdnice编辑器"><img data-imgfileid="100000717" data-ratio="1.1803751803751803" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/by7zpDFL3LIO8UGBnPDmqnYqFWDcfxXL90ricHibHqrocUlr58z49GZM1Fn18sIb1SoPfszmrYSr1OfvdRKXVnsQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="693" src="https://mmbiz.qpic.cn/sz_mmbiz_png/by7zpDFL3LIO8UGBnPDmqnYqFWDcfxXL90ricHibHqrocUlr58z49GZM1Fn18sIb1SoPfszmrYSr1OfvdRKXVnsQ/640?wx_fmt=png&amp;from=appmsg"><figcaption>image-20240718205232453</figcaption></figure><h3 data-tool="mdnice编辑器"><span></span><span><span>免疫治疗队列</span></span><span></span></h3><figure data-tool="mdnice编辑器"><img data-imgfileid="100000716" data-ratio="0.38055555555555554" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/by7zpDFL3LIO8UGBnPDmqnYqFWDcfxXLJfebl6rc9PkY5gfXXvU76ELkF6gibbqZtPuFLSYVOibSHy9tCqgt9w1Q/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/by7zpDFL3LIO8UGBnPDmqnYqFWDcfxXLJfebl6rc9PkY5gfXXvU76ELkF6gibbqZtPuFLSYVOibSHy9tCqgt9w1Q/640?wx_fmt=png&amp;from=appmsg"><figcaption>image-20240718205316959</figcaption></figure><h3 data-tool="mdnice编辑器"><span></span><span><span>CRISPR数据集</span></span><span></span></h3><figure data-tool="mdnice编辑器"><img data-imgfileid="100000718" data-ratio="0.8575268817204301" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/by7zpDFL3LIO8UGBnPDmqnYqFWDcfxXLnZXlTvacuTib6k9mtq7356EUhC4bzDKF9iakk4bCVbk4J3x7xibeRpqUA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="744" src="https://mmbiz.qpic.cn/sz_mmbiz_png/by7zpDFL3LIO8UGBnPDmqnYqFWDcfxXLnZXlTvacuTib6k9mtq7356EUhC4bzDKF9iakk4bCVbk4J3x7xibeRpqUA/640?wx_fmt=png&amp;from=appmsg"><figcaption>image-20240718205426805</figcaption></figure><h3 data-tool="mdnice编辑器"><span></span><span><span>免疫治疗的预测基因表达特征列表</span></span><span></span></h3><figure data-tool="mdnice编辑器"><img data-imgfileid="100000721" data-ratio="0.375" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/by7zpDFL3LIO8UGBnPDmqnYqFWDcfxXLHjbhjrNKeSgk1U5EuniacxfNwZtsHsRKt3lj2hP9u8HIJgjWeTmiamsA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/by7zpDFL3LIO8UGBnPDmqnYqFWDcfxXLHjbhjrNKeSgk1U5EuniacxfNwZtsHsRKt3lj2hP9u8HIJgjWeTmiamsA/640?wx_fmt=png&amp;from=appmsg"><figcaption>image-20240718205500924</figcaption></figure><h3 data-tool="mdnice编辑器"><span></span><span>图1：识别和验证癌细胞干细胞性和ICI结果之间的负相关</span><span></span></h3><figure data-tool="mdnice编辑器"><img data-imgfileid="100000719" data-ratio="1.2080924855491328" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/by7zpDFL3LIO8UGBnPDmqnYqFWDcfxXLK7Ys3IRlLPTOtHYbn6M3e9wYag5FNzPv9wZL0ETAxPAUjRicVmrtcdg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="692" src="https://mmbiz.qpic.cn/sz_mmbiz_png/by7zpDFL3LIO8UGBnPDmqnYqFWDcfxXLK7Ys3IRlLPTOtHYbn6M3e9wYag5FNzPv9wZL0ETAxPAUjRicVmrtcdg/640?wx_fmt=png&amp;from=appmsg"><figcaption>image-20240718205547923</figcaption></figure><p data-tool="mdnice编辑器"><span>图1的代码：</span></p><pre data-tool="mdnice编辑器"><span></span><code><span>###==Figure 1===###</span><br><br>rm (list=ls())<br><br>gc()<br><br><span>#====loading data===####</span><br>library(CytoTRACE)<br>library(Seurat)<br>library(dplyr)<br>library(ggpubr)<br><span>source</span>(<span>'Code/plotCytoTRACE.v2.0.R'</span>) <span>#modified plotCytoTRACE function implemented in CytoTRACE</span><br><span>source</span>(<span>'Code/R_rainclouds.R'</span>)<br><br><br><span>####====Fig. 1A and 1B===####</span><br><span>#GSE115978</span><br>cohort &lt;- <span>"SKCM_GSE115978_aPD1"</span> <br>load(paste0(<span>"data/"</span>,cohort,<span>'/scRNA.Rdata'</span>))<br>load(paste0(<span>'data/'</span>,cohort,<span>'/phenotype.Rdata'</span>))<br><br>cell &lt;- phenotype[phenotype<span>$cellType</span> == <span>'Malignant cells'</span> ,<span>'Cell'</span>] <br>response &lt;- phenotype[phenotype<span>$cellType</span> == <span>'Malignant cells'</span> ,<span>'Response'</span>]<br>names(response) &lt;- phenotype[phenotype<span>$cellType</span> == <span>'Malignant cells'</span> ,<span>'Cell'</span>]<br><br>scRNA &lt;- scRNA[,colnames(scRNA) %<span>in</span>% cell]<br>scRNA &lt;- as.data.frame(scRNA)<br>results &lt;- CytoTRACE(scRNA)<br>  <br>Fig.1A &lt;- plotCytoTRACE.v2.0(results,response)<br><br><br><br>boxdata &lt;- data.frame(Cell = names(results<span>$CytoTRACE</span>),cytoTRACE=results<span>$CytoTRACE</span>)<br>boxdata &lt;- left_join(boxdata,phenotype[,c(<span>'Cell'</span>,<span>'Response'</span>)],by=<span>'Cell'</span>)<br>boxdata<span>$Response</span> &lt;- factor(boxdata<span>$Response</span>,levels = c(<span>"NR"</span>,<span>"TN"</span>))<br><br><br>Fig.1B &lt;- ggplot(boxdata, aes(x = Response, y = cytoTRACE, fill = Response)) +<br>  geom_flat_violin(aes(fill = Response),position = position_nudge(x = 0.1, y = 0),  trim = TRUE, alpha = .5, colour = NA)+<br>  geom_point(aes(x = .55, y = cytoTRACE, colour = Response),position = position_jitter(width = .05), size = 1, shape = 20)+<br>  geom_boxplot(aes(x = Response, y = cytoTRACE, fill = Response),outlier.shape = NA, alpha = .5, width = .1, colour = <span>"black"</span>)+<br>  scale_colour_manual(values =  c(<span>"#E7B800"</span>,<span>"#2E9FDF"</span>))+<br>  scale_fill_manual(values = c(<span>"#E7B800"</span>,<span>"#2E9FDF"</span>))+<br>  theme_classic()+<br>  geom_signif(comparisons = list(c(<span>"NR"</span>,<span>"TN"</span>)),map_signif_level = TRUE,<span>test</span> = <span>"wilcox.test"</span>)+<br>  theme(axis.title = element_text(size=15),<br>        axis.title.x = element_blank(),<br>        axis.text = element_text(size=12))+<br>  <span># ylim(0,1)+</span><br>  ylab(<span>'CytoTRACE scores'</span>)<br><br>Fig.1B<br><br><span>####====Fig. 1C and 1D===####</span><br><br>cohort &lt;- <span>"BCC_GSE123813_aPD1"</span><br>load(paste0(<span>"data/"</span>,cohort,<span>'/scRNA.Rdata'</span>)) <br>load(paste0(<span>"data/"</span>,cohort,<span>'/phenotype.Rdata'</span>))<br><br>cell &lt;- phenotype[phenotype<span>$cellType</span> == <span>'Malignant cells'</span> ,<span>'Cell'</span>] <br>response &lt;- phenotype[phenotype<span>$cellType</span> == <span>'Malignant cells'</span> ,<span>'Response'</span>]<br>names(response) &lt;- phenotype[phenotype<span>$cellType</span> == <span>'Malignant cells'</span> ,<span>'Cell'</span>]<br><br>scRNA &lt;- scRNA[,colnames(scRNA) %<span>in</span>% cell]<br>scRNA &lt;- as.data.frame(scRNA)<br>results &lt;- CytoTRACE(scRNA)<br><br><br>Fig.1C &lt;- plotCytoTRACE.v2.0(results,response)<br><br>boxdata &lt;- data.frame(Cell = names(results<span>$CytoTRACE</span>),cytoTRACE=results<span>$CytoTRACE</span>)<br>boxdata &lt;- left_join(boxdata,phenotype[,c(<span>'Cell'</span>,<span>'Response'</span>)],by=<span>'Cell'</span>)<br>boxdata<span>$Response</span> &lt;- factor(boxdata<span>$Response</span>,levels = c(<span>'NR'</span>,<span>'R'</span>))<br><br><br>Fig.1D &lt;- ggplot(boxdata, aes(x = Response, y = cytoTRACE, fill = Response)) +<br>  geom_flat_violin(aes(fill = Response),position = position_nudge(x = 0.1, y = 0),  trim = TRUE, alpha = .5, colour = NA)+<br>  geom_point(aes(x = .55, y = cytoTRACE, colour = Response),position = position_jitter(width = .05), size = 1, shape = 20)+<br>  geom_boxplot(aes(x = Response, y = cytoTRACE, fill = Response),outlier.shape = NA, alpha = .5, width = .1, colour = <span>"black"</span>)+<br>  scale_colour_manual(values =  c(<span>"#E7B800"</span>,<span>"#2E9FDF"</span>))+<br>  scale_fill_manual(values = c(<span>"#E7B800"</span>,<span>"#2E9FDF"</span>))+<br>  theme_classic()+<br>  geom_signif(comparisons = list(c(<span>"NR"</span>,<span>"R"</span>)),map_signif_level = TRUE,<span>test</span> = <span>"wilcox.test"</span>)+<br>  theme(axis.title = element_text(size=15),<br>        axis.title.x = element_blank(),<br>        axis.text = element_text(size=12))+<br>  ylab(<span>'CytoTRACE scores'</span>)<br><br>Fig.1D<br></code></pre><h3 data-tool="mdnice编辑器"><span></span><span>图2：stemness signature的开发和描述</span><span></span></h3><figure data-tool="mdnice编辑器"><img data-imgfileid="100000722" data-ratio="0.9757575757575757" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/by7zpDFL3LIO8UGBnPDmqnYqFWDcfxXLiaapvBeGqOKoabliahFkOQBwLcYcNhmibsNFFhYHOLiceloMDljtRSNfdg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="825" src="https://mmbiz.qpic.cn/sz_mmbiz_png/by7zpDFL3LIO8UGBnPDmqnYqFWDcfxXLiaapvBeGqOKoabliahFkOQBwLcYcNhmibsNFFhYHOLiceloMDljtRSNfdg/640?wx_fmt=png&amp;from=appmsg"><figcaption>image-20240718205628038</figcaption></figure><p data-tool="mdnice编辑器"><span>图2的代码：</span></p><pre data-tool="mdnice编辑器"><span></span><code><span>####===Figure 2===####</span><br>rm (list=ls())<br>gc()<br><br><br><span>####===Fig. 2A===####</span><br>library(circlize)<br>library(RColorBrewer)<br>library(png)<br>library(graphics)<br>library(circlize)<br>library(RColorBrewer)<br>library(stringr)<br>library(ComplexHeatmap)<br>library(grImport2)<br>library(gridBase)<br>library(ggplot2)<br><br>dir &lt;- data.table::fread(<span>'data/dir/dir.csv'</span>)<br>circos.clear()<br>Set3 &lt;- brewer.pal(12,<span>"Set3"</span>)<br>Set2 &lt;- brewer.pal(8,<span>"Set2"</span>)<br>col &lt;- c(Set3,Set2)<br><br>sectors = dir<span>$dataset</span><br>names(sectors) &lt;- sectors<br><br>dataset &lt;- data.frame(dataset = sectors)<br>rownames(dataset) &lt;- sectors<br><br>col_cancer &lt;- c(col[1], rep(col[2],2), col[3],col[4],rep(col[5],11),col[6],col[7],col[8],rep(col[9],2),col[10],col[11],rep(col[12],4),col[13],rep(col[14],2),<br>                rep(col[15],2),col[16],col[19])<br>names(col_cancer) &lt;- sectors<br><br><br><br>image = <span>'data/png/venn.png'</span><br>image = as.raster(readPNG(image))<br><br>circos.clear()<br>circos.par(gap.degree = 2, cell.padding = c(0, 0, 0, 0),<br>           track.margin = c(0.01, 0.01))<br>circos.heatmap(dataset, split = dataset<span>$dataset</span>, col = col_cancer, track.height = 0.02,rownames.side = <span>'outside'</span>)<br><br>circos.track(ylim = c(0, 1),track.height=0.15,bg.border = <span>"#EEEEEE"</span>, panel.fun = <span>function</span>(x, y) {<br>  circos.raster(image, CELL_META<span>$xcenter</span>, CELL_META<span>$ycenter</span>, <br>                width = <span>"0.7cm"</span>,<br>                facing = <span>"inside"</span>)<br>  <br>})<br><br>circos.track(ylim = c(0, 1), sectors = sectors,<br>             bg.col = <span>"#8190A5"</span>, bg.border = <span>"#EEEEEE"</span> , track.height = 0.25)<br><br>circos.trackText(x = rep(0.5, 34), y = rep(0.5, 34),<br>                 labels = paste0(rep(<span>'G'</span>,34), 1:34),<br>                 cex = 0.5, sectors = sectors, col = <span>"white"</span>, font = 2, facing = <span>"clockwise"</span>,<br>                 niceFacing=T)<br><br>draw.sector(center = c(0, 0), start.degree = 0, end.degree = 360,<br>            rou1 = 0.25, col = <span>"#4D6381"</span>, border = <span>"#EEEEEE"</span>)<br><br>text(0,0,<span>"Stem.sig"</span>,col = <span>'white'</span>,cex = 2,font = 2)<br><br><br>cancer &lt;- unique(col_cancer)<br>names(cancer) &lt;- unique(dir<span>$cancer</span>)<br>lgd_cancer = Legend(title = <span>"Cancer"</span>, at = names(cancer),<br>                 legend_gp = gpar(fill = cancer))<br>draw(lgd_cancer, x = unit(1.2, <span>"snpc"</span>), just = <span>"left"</span>)<br><br><br><br><span>####===Fig. 2B===####</span><br>library(clusterProfiler)<br>library(org.Hs.eg.db)<br>library(ReactomePA)<br><br>load(<span>'data/sig/Stem.Sig.Rdata'</span>)<br><br>sig &lt;- Stem.Sig<br><br>sig_ID &lt;- bitr(sig,fromType = <span>'SYMBOL'</span>,toType = <span>'ENTREZID'</span>,OrgDb = <span>"org.Hs.eg.db"</span>)<br><br>eReac &lt;- enrichPathway(gene = sig_ID<span>$ENTREZID</span>,<br>                       organism = <span>'human'</span>,<br>                       pvalueCutoff = 0.05)<br><br>bardata &lt;- eReac@result[1:20,]<br>bardata &lt;- bardata[order(bardata<span>$qvalue</span>),]<br>bardata &lt;- bardata[order(bardata<span>$Description</span>),]<br>bardata<span>$Description</span> &lt;- factor(bardata<span>$Description</span>,levels=rev(bardata<span>$Description</span>))<br>barplot &lt;- ggplot(bardata,aes(y=Description,x=Count))+geom_bar(<span>stat</span> = <span>"identity"</span>,aes(fill=qvalue))+scale_fill_gradientn(colours =  c(<span>"#4D6381"</span>, <span>"#BAC2CC"</span>))+theme_bw()<br><br>eReacx &lt;- setReadable(eReac, <span>'org.Hs.eg.db'</span>, <span>'ENTREZID'</span>)<br><br>cnetplot &lt;-  cnetplot(eReacx, circular = TRUE, colorEdge = TRUE,categorySize=<span>"pvalue"</span>,showCategory = 20,layout = <span>'kk'</span>) <br><br>barplot <br>cnetplot<br></code></pre><h3 data-tool="mdnice编辑器"><span></span><span>图3：使用泛癌TCGA队列分析Stem.Sig与免疫耐药之间的潜在联系</span><span></span></h3><figure data-tool="mdnice编辑器"><img data-imgfileid="100000720" data-ratio="1.271341463414634" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/by7zpDFL3LIO8UGBnPDmqnYqFWDcfxXL17pE7cC5o0ibDPDCbgReDzgtyadPySoocQOBq2PeqUQThFhDm6WzdrA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="656" src="https://mmbiz.qpic.cn/sz_mmbiz_png/by7zpDFL3LIO8UGBnPDmqnYqFWDcfxXL17pE7cC5o0ibDPDCbgReDzgtyadPySoocQOBq2PeqUQThFhDm6WzdrA/640?wx_fmt=png&amp;from=appmsg"><figcaption>image-20240718205713834</figcaption></figure><h3 data-tool="mdnice编辑器"><span></span><span>图4：使用Stem.Sig预测ICI结果</span><span></span></h3><figure data-tool="mdnice编辑器"><img data-imgfileid="100000727" data-ratio="1.0304568527918783" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/by7zpDFL3LIO8UGBnPDmqnYqFWDcfxXLo0hFHbhGFqhWl7BrGuRtIP9LbyNYMIWEJwjEUvM48cHBSTIQlqGfaQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="788" src="https://mmbiz.qpic.cn/sz_mmbiz_png/by7zpDFL3LIO8UGBnPDmqnYqFWDcfxXLo0hFHbhGFqhWl7BrGuRtIP9LbyNYMIWEJwjEUvM48cHBSTIQlqGfaQ/640?wx_fmt=png&amp;from=appmsg"><figcaption>image-20240718205817954</figcaption></figure><h3 data-tool="mdnice编辑器"><span></span><span>图5：Stem.Sig的AUC与其他预测基因特征的比较</span><span></span></h3><figure data-tool="mdnice编辑器"><img data-imgfileid="100000726" data-ratio="1.0276381909547738" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/by7zpDFL3LIO8UGBnPDmqnYqFWDcfxXLxlHJU4SRyNLlqsvEpb4HbUQ4IDZOdgVg7ojibowdXcjhjtK4pDZlaTw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="796" src="https://mmbiz.qpic.cn/sz_mmbiz_png/by7zpDFL3LIO8UGBnPDmqnYqFWDcfxXLxlHJU4SRyNLlqsvEpb4HbUQ4IDZOdgVg7ojibowdXcjhjtK4pDZlaTw/640?wx_fmt=png&amp;from=appmsg"><figcaption>image-20240718205854111</figcaption></figure><h3 data-tool="mdnice编辑器"><span></span><span>图6：利用CRISPR筛选数据探索来自Stem.Sig的潜在治疗靶点</span><span></span></h3><figure data-tool="mdnice编辑器"><img data-imgfileid="100000728" data-ratio="0.8735777496839444" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/by7zpDFL3LIO8UGBnPDmqnYqFWDcfxXLO9xkWzZb12jmHtFLLqg9iabRd8p9PA15HUQ3lUZSp3YB6gQj2Kux6NA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="791" src="https://mmbiz.qpic.cn/sz_mmbiz_png/by7zpDFL3LIO8UGBnPDmqnYqFWDcfxXLO9xkWzZb12jmHtFLLqg9iabRd8p9PA15HUQ3lUZSp3YB6gQj2Kux6NA/640?wx_fmt=png&amp;from=appmsg"><figcaption>image-20240718205940800</figcaption></figure><p data-tool="mdnice编辑器">Cite：</p><p data-tool="mdnice编辑器">Zhang, Zhen et al. “Integrated analysis of single-cell and bulk RNA sequencing data reveals a pan-cancer stemness signature predicting immunotherapy response.” <em>Genome medicine</em> vol. 14,1 45. 29 Apr. 2022, doi:10.1186/s13073-022-01050-w</p><p data-tool="mdnice编辑器"><span><br></span></p><p data-tool="mdnice编辑器"><span>往期精彩</span></p></section><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzkzMzYzMzk0Ng==&amp;mid=2247484333&amp;idx=1&amp;sn=bd43e6a23e6b7991b50c467c6849471c&amp;chksm=c248c958f53f404ef2fdd1cf10bfa78f5d4be59516542f461d4227273a6afe41e4b8a40fbc8f&amp;scene=21#wechat_redirect" textvalue="代码分享|| IF10+的 python版本的单细胞数据分析，图的颜值超高！" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">代码分享|| IF10+的 python版本的单细胞数据分析，图的颜值超高！</a><br></p><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzkzMzYzMzk0Ng==&amp;mid=2247484289&amp;idx=1&amp;sn=5fa021c28bad65529659b7d5551756cc&amp;chksm=c248c974f53f406217af1eb4d4fb6b881f31b3c523c246ba0328ca35b61e0fb29220edd8357a&amp;scene=21#wechat_redirect" textvalue="代码分享|| 学习一个单细胞与空转数据库 ssREAD（阿尔兹海默症）网站的构建：提供前端+后端代码，文章图表分析代码" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">代码分享|| 学习一个单细胞与空转数据库 ssREAD（阿尔兹海默症）网站的构建：提供前端+后端代码，文章图表分析代码</a><br></p><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzkzMzYzMzk0Ng==&amp;mid=2247484274&amp;idx=1&amp;sn=4547f1d6e45a0506880ebfd65382d0b0&amp;chksm=c248c987f53f4091afd5f19bc25f9d3f87c2ea1ac21a59fea2fa8e7f3099342fd0b65cbe2414&amp;scene=21#wechat_redirect" textvalue="代码分享|| 单细胞亚群注释软件你还有不知道的吗？快来看看这个发表在Nature Communications的工具！" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">代码分享|| 单细胞亚群注释软件你还有不知道的吗？快来看看这个发表在Nature Communications的工具！</a><br></p><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzkzMzYzMzk0Ng==&amp;mid=2247484183&amp;idx=1&amp;sn=119014087d0fee7c49fe6dc10bbea00e&amp;chksm=c248c9e2f53f40f4472e9d170a904e69c3f1b5a6dc102291f99ee85beeae80075bbf8eae07e0&amp;scene=21#wechat_redirect" textvalue="代码分享|| Nature肝脏疾病进展单细胞分析超详细代码：上游分析+下游，详细到fq文件重命名，数据预处理等！快上车！" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">代码分享|| Nature肝脏疾病进展单细胞分析超详细代码：上游分析+下游，详细到fq文件重命名，数据预处理等！快上车！</a><br></p><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzkzMzYzMzk0Ng==&amp;mid=2247484170&amp;idx=1&amp;sn=f0fa846262c160d392e06a5cfb9f66cb&amp;chksm=c248c9fff53f40e90a0079dd223aa266f4e26d6248e7e34bb48a0def94ed98005c8b7007c26c&amp;scene=21#wechat_redirect" textvalue="代码分享|| 单细胞上车尤未晚！囊括了单细胞分析方方面面包括大类注释、亚群分析、细胞通讯、轨迹分析、生存、多因素cox分析等" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">代码分享|| 单细胞上车尤未晚！囊括了单细胞分析方方面面包括大类注释、亚群分析、细胞通讯、轨迹分析、生存、多因素cox分析等</a><br></p><p><br></p><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/F60uTx3mLlX-FpcGMoOhrw",target="_blank" rel="noopener noreferrer">原文链接</a>
