---
title: "你还缺scRNA-seq的workflow吗？"
date: 2024-07-25T22:48:52Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Acdemic"]
---
你还缺scRNA-seq的workflow吗？ by 生信菜鸟团
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><h2 data-tool="mdnice编辑器"><span></span><span>前言</span><span></span></h2><p data-tool="mdnice编辑器">之前曾老师给我看了一位在pipebio工作的生信工程师Roman Hillje的scRNA-seq的workflow，今天整理一下分享给大家。</p><p data-tool="mdnice编辑器">Roman Hillje的github主页：https://github.com/romanhaa</p><p data-tool="mdnice编辑器">workflow：https://romanhaa.github.io/projects/scrnaseq_workflow/</p><figure data-tool="mdnice编辑器"><img data-imgfileid="100042350" data-ratio="0.31666666666666665" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0htq0nO5ODMJamo7OeXFJ42m54vmEedGSa5UtFYr58Dkia8eWkgR6XCyqA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0htq0nO5ODMJamo7OeXFJ42m54vmEedGSa5UtFYr58Dkia8eWkgR6XCyqA/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">示例数据来源：<strong>GSE120221</strong> (https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE120221) 下的前三个样品:</p><ul data-tool="mdnice编辑器"><li><section>GSM3396161Bone marrow A</section></li><li><section>GSM3396162Bone marrow B</section></li><li><section>GSM3396163Bone marrow C1</section></li></ul><h2 data-tool="mdnice编辑器"><span></span><span>1.准备工作</span><span></span></h2><pre data-tool="mdnice编辑器"><span></span><code><span>#1.1 加载R包</span><br>rm(list = ls())<br><span>library</span>(tidyverse)<br><span>library</span>(Seurat)<br><span>library</span>(scran)<br><span>library</span>(patchwork)<br><span>library</span>(viridis)<br><span>library</span>(ggforce)<br><span>library</span>(gghalves)<br><span>library</span>(ggridges)<br><span>library</span>(scDblFinder)<br><span>library</span>(SingleR)<br><span>library</span>(intrinsicDimension)<br><br><span>#1.2 设置可视化配色</span><br>custom_colors &lt;- list()<br><br>colors_dutch &lt;- c(<br>  <span>'#FFC312'</span>,<span>'#C4E538'</span>,<span>'#12CBC4'</span>,<span>'#FDA7DF'</span>,<span>'#ED4C67'</span>,<br>  <span>'#F79F1F'</span>,<span>'#A3CB38'</span>,<span>'#1289A7'</span>,<span>'#D980FA'</span>,<span>'#B53471'</span>,<br>  <span>'#EE5A24'</span>,<span>'#009432'</span>,<span>'#0652DD'</span>,<span>'#9980FA'</span>,<span>'#833471'</span>,<br>  <span>'#EA2027'</span>,<span>'#006266'</span>,<span>'#1B1464'</span>,<span>'#5758BB'</span>,<span>'#6F1E51'</span><br>)<br><br>colors_spanish &lt;- c(<br>  <span>'#40407a'</span>,<span>'#706fd3'</span>,<span>'#f7f1e3'</span>,<span>'#34ace0'</span>,<span>'#33d9b2'</span>,<br>  <span>'#2c2c54'</span>,<span>'#474787'</span>,<span>'#aaa69d'</span>,<span>'#227093'</span>,<span>'#218c74'</span>,<br>  <span>'#ff5252'</span>,<span>'#ff793f'</span>,<span>'#d1ccc0'</span>,<span>'#ffb142'</span>,<span>'#ffda79'</span>,<br>  <span>'#b33939'</span>,<span>'#cd6133'</span>,<span>'#84817a'</span>,<span>'#cc8e35'</span>,<span>'#ccae62'</span><br>)<br><br>custom_colors$discrete &lt;- c(colors_dutch, colors_spanish)<br><br>custom_colors$cell_cycle &lt;- setNames(<br>  c(<span>'#45aaf2'</span>, <span>'#f1c40f'</span>, <span>'#e74c3c'</span>, <span>'#7f8c8d'</span>),<br>  c(<span>'G1'</span>,      <span>'S'</span>,       <span>'G2M'</span>,     <span>'-'</span>)<br>)<br><br><span>#1.3 定义一些函数</span><br>load10xData &lt;- <span>function</span>(<br>    sample_name,<br>    path,<br>    max_number_of_cells = <span>NULL</span><br>) {<br>  <br>  <span>## read transcript count matrix</span><br>  transcript_counts &lt;- list.files(<br>    path,<br>    recursive = <span>TRUE</span>,<br>    full.names = <span>TRUE</span><br>  ) %&gt;%<br>    .[grep(., pattern = <span>'mtx'</span>)] %&gt;%<br>    Matrix::readMM()<br>  <br>  <span>## read cell barcodes and assign as column names of transcript count matrix</span><br>  colnames(transcript_counts) &lt;- list.files(<br>    path,<br>    recursive = <span>TRUE</span>,<br>    full.names = <span>TRUE</span><br>  ) %&gt;%<br>    .[grep(., pattern = <span>'barcodes'</span>)] %&gt;%<br>    .[grep(., pattern = <span>'tsv'</span>)] %&gt;%<br>    read_tsv(col_names = <span>FALSE</span>, col_types = cols()) %&gt;%<br>    pull(<span>1</span>) %&gt;%<br>    gsub(., pattern = <span>'-[0-9]{1,2}'</span>, replacement = <span>''</span>) %&gt;%<br>    paste0(., <span>'-'</span>, sample_name)<br>  <br>  <span>## read gene names</span><br>  gene_names &lt;- list.files(<br>    path,<br>    recursive = <span>TRUE</span>,<br>    full.names = <span>TRUE</span><br>  ) %&gt;%<br>    .[grep(., pattern = <span>'genes|features'</span>)] %&gt;%<br>    read_tsv(col_names = <span>FALSE</span>, col_types = cols()) %&gt;%<br>    pull(ifelse(ncol(.) &gt; <span>1</span>, <span>2</span>, <span>1</span>))<br>  <br>  <span>## if necessary, merge counts from different transcripts of the same gene</span><br>  <span>if</span> ( any(duplicated(gene_names)) == <span>TRUE</span> ) {<br>    <br>    <span>## get names of genes with multiple entries</span><br>    duplicated_gene_names &lt;- unique(gene_names[which(duplicated(gene_names))])<br>    <br>    <span>## log message</span><br>    message(<br>      glue::glue(<br>        <span>"{format(Sys.time(), '[%Y-%m-%d %H:%M:%S]')} Summing up counts for "</span>,<br>        <span>"{length(duplicated_gene_names)} genes with multiple entries."</span><br>      )<br>    )<br>    <br>    <span>## go through genes with multiple entries</span><br>    <span>for</span> ( i <span>in</span> duplicated_gene_names ) {<br>      <br>      <span>## extract transcripts counts for current gene</span><br>      tmp_counts &lt;- transcript_counts[which(gene_names == i),]<br>      <br>      <span>## make sure at least 2 rows are present</span><br>      <span>if</span> ( nrow(tmp_counts) &gt;= <span>2</span> ) {<br>        <br>        <span>## sum up counts in each cell</span><br>        tmp_counts_sum &lt;- Matrix::colSums(tmp_counts)<br>        <br>        <span>## remove original counts from transcript matrix and the gene name from</span><br>        <span>## the list of gene names</span><br>        transcript_counts &lt;- transcript_counts[-c(which(gene_names == i)),]<br>        gene_names &lt;- gene_names[-c(which(gene_names == i))]<br>        <br>        <span>## add summed counts to end of transcript count matrix and current gene</span><br>        <span>## name to end of gene names</span><br>        transcript_counts &lt;- rbind(transcript_counts, tmp_counts_sum)<br>        gene_names &lt;- c(gene_names, i)<br>      }<br>    }<br>  }<br>  <br>  <span>## assign gene names as row names</span><br>  rownames(transcript_counts) &lt;- gene_names<br>  <br>  <span>## down-sample cells if more cells than `max_number_of_cells` are present in</span><br>  <span>## count matrix</span><br>  <span>if</span> (<br>    !is.null(max_number_of_cells) &amp;&amp;<br>    is.numeric(max_number_of_cells) &amp;&amp;<br>    max_number_of_cells &lt; ncol(transcript_counts)<br>  ) {<br>    temp_cells_to_keep &lt;- base::sample(<span>1</span>:ncol(transcript_counts), max_number_of_cells)<br>    transcript_counts &lt;- transcript_counts[,temp_cells_to_keep]<br>  }<br>  <br>  <span>##</span><br>  message(<br>    glue::glue(<br>      <span>"{format(Sys.time(), '[%Y-%m-%d %H:%M:%S]')} Loaded counts for sample "</span><br>    )<br>  )<br>  <br>  <span>##</span><br>  <span>return</span>(transcript_counts)<br>}<br><br><span>#When provided with a list of transcript count matrices, </span><br><span>#this function checks whether the gene names are the same in all the count matrices, </span><br><span>#which is an important premise for merging them.</span><br><br><span>#当提供许多待合并的转录组count矩阵后，这个函数检查所有count矩阵中基因名是否相同，</span><br><span>#这是合并他们的重要前提</span><br><br>sameGeneNames &lt;- <span>function</span>(counts) {<br>  <br>  <span>## create place holder for gene names from each count matrix</span><br>  gene_names &lt;- list()<br>  <br>  <span>## go through count matrices</span><br>  <span>for</span> ( i <span>in</span> names(counts) ) {<br>    <br>    <span>## extract gene names</span><br>    gene_names[[i]] &lt;- rownames(counts[[i]])<br>  }<br>  <br>  <span>## check if all gene names are the same (using the first matrix as a reference</span><br>  <span>## for the others)</span><br>  <span>return</span>(all(sapply(gene_names, FUN = identical, gene_names[[<span>1</span>]])))<br>}<br><br><span>#This function outputs a table of mean and median transcripts and expressed genes for every sample in our data set.</span><br><span>#这个函数输出数据集中每个样本产生转录表达的平均数和中位数</span><br><br>averagenCountnFeature &lt;- <span>function</span>(cells) {<br>  output &lt;- tibble(<br>    <span>'sample'</span> = character(),<br>    <span>'mean(nCount)'</span> = numeric(),<br>    <span>'median(nCount)'</span> = numeric(),<br>    <span>'mean(nFeature)'</span> = numeric(),<br>    <span>'median(nFeature)'</span> = numeric()<br>  )<br>  <span>for</span> ( i <span>in</span> levels(cells$sample) ) {<br>    tmp &lt;- tibble(<br>      <span>'sample'</span> = i,<br>      <span>'mean(nCount)'</span> = cells %&gt;% filter(sample == i) %&gt;% pull(nCount) %&gt;% mean(),<br>      <span>'median(nCount)'</span> = cells %&gt;% filter(sample == i) %&gt;% pull(nCount) %&gt;% median(),<br>      <span>'mean(nFeature)'</span> = cells %&gt;% filter(sample == i) %&gt;% pull(nFeature) %&gt;% mean(),<br>      <span>'median(nFeature)'</span> = cells %&gt;% filter(sample == i) %&gt;% pull(nFeature) %&gt;% median()<br>    )<br>    output &lt;- bind_rows(output, tmp)<br>  }<br>  <span>return</span>(output)<br>}<br><br><span>#1.4 创建文件夹</span><br>dir.create(<span>'data'</span>)<br>dir.create(<span>'plots'</span>)<br></code></pre><h2 data-tool="mdnice编辑器"><span></span><span>2.载入数据</span><span></span></h2><p data-tool="mdnice编辑器">首先用之前定义的load10xData函数读取10x样本。我们这里使用参数<code>max_number_of_cells</code>随机取2000个细胞为了快速演示这个流程</p><pre data-tool="mdnice编辑器"><span></span><code><span>#2.1 读入10x数据</span><br>sample_names &lt;- list.dirs(<span>'RAW'</span>, recursive = <span>FALSE</span>) %&gt;% basename()<br><br>transcripts &lt;- list(<br>  raw = list()<br>)<br><br><span>for</span> ( i <span>in</span> sample_names ) {<br>  transcripts$raw[[i]] &lt;- load10xData(<br>    i, paste0(<span>'RAW/'</span>, i, <span>'/'</span>),<br>    max_number_of_cells = <span>2000</span><br>  )<br>}<br><br><span>#2.2 合并样本</span><br><span>#在合并count矩阵之前，需要检查他们的基因名是否相同</span><br>sameGeneNames(transcripts$raw)<br><span># TRUE</span><br><span>#本文数据来源的三个样本经过同样的上游处理</span><br><span>#他们之间的基因名相同所以我们可以直接合并</span><br>transcripts$raw$merged &lt;- do.call(cbind, transcripts$raw)<br><br>dim(transcripts$raw$merged)<br>save(transcripts,file = <span>"./data/raw_merged_count.RData"</span>)<br><span># 33660  6000</span><br><br><span>#注意:如果基因名称不相同，有两种策略。</span><br><span>#您可以将转录组count矩阵转换为数据框格式，然后使用full_join()函数合并它们。</span><br><span>#根据你正在处理的数据集的大小，这可能会导致内存问题。</span><br><span>#或者，你将缺失的、充满0计数的基因添加到缺失基因的矩阵中。</span><br><span>#在合并之前，必须确保基因的顺序相同。</span><br><span>#你可以在下面找到第一种方法的大纲:</span><br><span># for ( i in sample_names ) {</span><br><span>#   gene_names &lt;- rownames(transcripts$raw[[i]])</span><br><span>#   transcripts$raw[[i]] &lt;- as.data.frame(as.matrix(transcripts$raw[[i]]))</span><br><span>#   transcripts$raw[[i]] &lt;- transcripts$raw[[i]] %&gt;%</span><br><span>#     mutate(gene = gene_names) %&gt;%</span><br><span>#     select(gene, everything())</span><br><span># }</span><br><span># </span><br><span># transcripts$raw$merged &lt;- full_join(transcripts$raw$A, transcripts$raw$B, by = "gene") %&gt;%</span><br><span>#   full_join(., transcripts$raw$E, by = "gene")</span><br><span># </span><br><span># transcripts$raw$merged[ is.na(transcripts$raw$merged) ] &lt;- 0</span><br></code></pre><h2 data-tool="mdnice编辑器"><span></span><span>3.质控</span><span></span></h2><p data-tool="mdnice编辑器">在我们进行分析之前，我们将从合并的count矩阵中移除低质量的细胞，以及低表达的基因</p><pre data-tool="mdnice编辑器"><span></span><code><span>## 3.1 Filter cells过滤细胞</span><br><span>### 3.1.1 Prepare data准备数据</span><br>我们先获取转录组count，每个细胞表达的基因数量，以及每个细胞线粒体转录本占总的百分比<br>load(<span>"./data/raw_merged_count.RData"</span>)<br>cells &lt;- data.frame(<br>  cell = colnames(transcripts$raw$merged),<br>  sample = colnames(transcripts$raw$merged) %&gt;%<br>    strsplit(<span>'-'</span>) %&gt;%<br>    vapply(FUN.VALUE = character(<span>1</span>), `[`, <span>2</span>) %&gt;%<br>    factor(levels = sample_names),<br>  nCount = colSums(transcripts$raw$merged),<br>  nFeature = colSums(transcripts$raw$merged != <span>0</span>)<br>)<br><br><span>#这里他给的代码载入了一个含有研究物种对应线粒体基因名的配置文件，</span><br><span>#可我们没有啊，我们只能粗略根据"^MT-"找了</span><br><br><span>#内存不够的话不要这样写</span><br>transcripts$raw$merged -&gt; merged_count<br><span>library</span>(stringr)<br>cells$percent_MT &lt;- apply(merged_count[str_detect(rownames(merged_count),<span>"^MT-"</span>),],<span>2</span>,sum)/apply(merged_count,<span>2</span>,sum)<br>rm(merged_count)<br><br><span>### 3.1.2 Doublets去除双细胞</span><br><span>#使用scDblFinder找doublets</span><br>sce &lt;- SingleCellExperiment(assays = list(counts = transcripts$raw$merged))<br><span>#这里笔者遇到一个matrix报错：</span><br><span>#程序包'as_cholmod_sparse'不提供'Matrix'这样的函数</span><br><span>#太玄学了，重装irlba包和Matrix包就ok了</span><br><span># remove.packages("Matrix")</span><br><span># remove.packages("irlba")</span><br><span># BiocManager::install("irlba",force = T)</span><br><span># BiocManager::install("Matrix",force = T)</span><br><span>#remotes::install_version("Matrix", version = "1.6-1.1")</span><br>sce &lt;- scDblFinder(sce)<br><span>#"SingleCellExperiment"对象</span><br>class(sce)<br><span>#在multiplet_class列标注出单细胞还是双细胞</span><br>cells$multiplet_class &lt;- colData(sce)$scDblFinder.class<br><span>#可视化,各个样本中双细胞数目</span><br>cells[cells$multiplet_class != <span>'singlet'</span>,<span>"sample"</span>] %&gt;% table %&gt;%<br>  as.data.frame() -&gt;plotdf<br>colnames(plotdf)&lt;-c(<span>"sample"</span>,<span>"count"</span>)<br><br>p&lt;-ggplot(plotdf ,aes(x = sample, y = count, fill = sample)) +<br>  geom_col(color = <span>'black'</span>) +<br>  theme_bw() +<br>  scale_fill_manual(values = custom_colors$discrete) +<br>  scale_x_discrete(limits = rev(levels(cells$sample))) +<br>  scale_y_continuous(name = <span>'Number of doublets'</span>, labels = scales::comma) +<br>  theme(<br>    axis.title.y = element_blank(),<br>    panel.grid.major.y = element_blank(),<br>    panel.grid.minor.y = element_blank(),<br>    legend.position = <span>'none'</span><br>  ) +<br>  coord_flip()<br><br>p<br><br>ggsave(<br>  <span>'plots/qc_number_of_doublets_by_sample.png'</span>,<br>  p, height = <span>3</span>, width = <span>6</span><br>)<br><span>#接下来我们可视化单细胞和双细胞中nCount和nFeature的分布</span><br>p1 &lt;- ggplot(cells, aes(x = multiplet_class, y = nCount, fill = multiplet_class)) +<br>  geom_violin(draw_quantiles = c(<span>0.5</span>), scale = <span>'area'</span>, trim = <span>FALSE</span>) +<br>  geom_jitter(aes(color=multiplet_class),width = <span>0.4</span>,size=<span>0.3</span>,alpha=<span>0.7</span>)+<br>  theme_bw() +<br>  scale_fill_manual(values = custom_colors$discrete) +<br>  scale_x_discrete(limits = rev(levels(cells$multiplet_class))) +<br>  scale_y_continuous(labels = scales::comma) +<br>  labs(title = <span>'Number of transcripts'</span>, subtitle = <span>'linear scale'</span>) +<br>  theme(<br>    axis.title = element_blank(),<br>    panel.grid.major.y = element_blank(),<br>    panel.grid.minor.y = element_blank(),<br>    legend.position = <span>'none'</span><br>  ) +<br>  coord_flip()<br>p2 &lt;- ggplot(cells, aes(x = multiplet_class, y = nCount, fill = multiplet_class)) +<br>  geom_violin(draw_quantiles = c(<span>0.5</span>), scale = <span>'area'</span>, trim = <span>FALSE</span>) +<br>  geom_jitter(aes(color=multiplet_class),width = <span>0.4</span>,size=<span>0.3</span>,alpha=<span>0.7</span>)+<br>  theme_bw() +<br>  scale_fill_manual(values = custom_colors$discrete) +<br>  scale_x_discrete(limits = rev(levels(cells$multiplet_class))) +<br>  scale_y_log10(labels = scales::comma) +<br>  labs(title = <span>'Number of transcripts'</span>, subtitle = <span>'log-scale'</span>) +<br>  theme(<br>    axis.title = element_blank(),<br>    panel.grid.major.y = element_blank(),<br>    panel.grid.minor.y = element_blank(),<br>    legend.position = <span>'none'</span><br>  ) +<br>  coord_flip()<br>p3 &lt;- ggplot(cells, aes(x = multiplet_class, y = nFeature, fill = multiplet_class)) +<br>  geom_violin(draw_quantiles = c(<span>0.5</span>), scale = <span>'area'</span>, trim = <span>FALSE</span>) +<br>  geom_jitter(aes(color=multiplet_class),width = <span>0.4</span>,size=<span>0.3</span>,alpha=<span>0.7</span>)+<br>  theme_bw() +<br>  scale_fill_manual(values = custom_colors$discrete) +<br>  scale_x_discrete(limits = rev(levels(cells$multiplet_class))) +<br>  scale_y_continuous(labels = scales::comma) +<br>  labs(title = <span>'Number of expressed genes'</span>, subtitle = <span>'linear scale'</span>) +<br>  theme(<br>    axis.title = element_blank(),<br>    panel.grid.major.y = element_blank(),<br>    panel.grid.minor.y = element_blank(),<br>    legend.position = <span>'none'</span><br>  ) +<br>  coord_flip()<br>p4 &lt;- ggplot(cells, aes(x = multiplet_class, y = nFeature, fill = multiplet_class)) +<br>  geom_violin(draw_quantiles = c(<span>0.5</span>), scale = <span>'area'</span>, trim = <span>FALSE</span>) +<br>  geom_jitter(aes(color=multiplet_class),width = <span>0.4</span>,size=<span>0.3</span>,alpha=<span>0.7</span>)+<br>  theme_bw() +<br>  scale_fill_manual(values = custom_colors$discrete) +<br>  scale_x_discrete(limits = rev(levels(cells$multiplet_class))) +<br>  scale_y_log10(labels = scales::comma) +<br>  labs(title = <span>'Number of expressed genes'</span>, subtitle = <span>'log-scale'</span>) +<br>  theme(<br>    axis.title = element_blank(),<br>    panel.grid.major.y = element_blank(),<br>    panel.grid.minor.y = element_blank(),<br>    legend.position = <span>'none'</span><br>  ) +<br>  coord_flip()<br><br><br>p1 + p3 +<br>    p2 + p4 + plot_layout(ncol = <span>2</span>)<br><br>ggsave(<br>  <span>'plots/qc_ncount_nfeature_by_multiplet_class.png'</span>,<br>  p1 + p3 +<br>    p2 + p4 + plot_layout(ncol = <span>2</span>),<br>  height = <span>14</span>, width = <span>20</span><br>)<br><span>#去除双细胞</span><br>cells &lt;- cells[cells$multiplet_class == <span>'singlet'</span>,]<br><br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100042351" data-ratio="0.5" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0htgoOA6u4wUq0zBgPqRCQ0hFcXTgpttKm9NcjSxt73NeyXL3GFIdz4Sw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0htgoOA6u4wUq0zBgPqRCQ0hFcXTgpttKm9NcjSxt73NeyXL3GFIdz4Sw/640?wx_fmt=png&amp;from=appmsg"></figure><figure data-tool="mdnice编辑器"><img data-imgfileid="100042354" data-ratio="0.6981481481481482" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0htwT0VElojP0HqcoCISSu9zLPpPxAxUJE32ByNicyGnrOW2zKRywIOSBA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0htwT0VElojP0HqcoCISSu9zLPpPxAxUJE32ByNicyGnrOW2zKRywIOSBA/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">在去除任何细胞之前，我们先看看转录本count的分布，每个细胞表达的基因数，以及每个细胞线粒体转录本的百分比。这些分布可以用不同的可视化方式表示。哪一个看起来最好取决于个人习惯和数据集。原文有四种可视化方法，展现出大佬雄厚的ggplot2绘图功力，我选我最喜欢的一种放在下面。</p><pre data-tool="mdnice编辑器"><span></span><code><span>### 3.1.3 Before filtering</span><br><br>p1 &lt;- ggplot(cells, aes(nCount, fill = sample)) +<br>  geom_histogram(bins = <span>200</span>) +<br>  theme_bw() +<br>  scale_fill_manual(values = custom_colors$discrete) +<br>  scale_x_continuous(labels = scales::comma) +<br>  labs(title = <span>'Number of transcripts'</span>, subtitle = <span>'linear scale'</span>, y = <span>'Number of cells'</span>) +<br>  theme(legend.position = <span>'none'</span>, axis.title.x = element_blank())<br><br>p2 &lt;- ggplot(cells, aes(nCount, fill = sample)) +<br>  geom_histogram(bins = <span>200</span>) +<br>  theme_bw() +<br>  scale_fill_manual(values = custom_colors$discrete) +<br>  scale_x_log10(labels = scales::comma) +<br>  labs(title = <span>'Number of transcripts'</span>, subtitle = <span>'log-scale'</span>, y = <span>'Number of cells'</span>) +<br>  theme(legend.position = <span>'none'</span>, axis.title.x = element_blank())<br><br>p3 &lt;- ggplot(cells, aes(nFeature, fill = sample)) +<br>  geom_histogram(bins = <span>200</span>) +<br>  theme_bw() +<br>  scale_fill_manual(values = custom_colors$discrete) +<br>  scale_x_continuous(labels = scales::comma) +<br>  labs(title = <span>'Number of expressed genes'</span>, subtitle = <span>'linear scale'</span>, y = <span>'Number of cells'</span>) +<br>  theme(legend.position = <span>'none'</span>,axis.title.x = element_blank())<br><br>p4 &lt;- ggplot(cells, aes(nFeature, fill = sample)) +<br>  geom_histogram(bins = <span>200</span>) +<br>  theme_bw() +<br>  scale_fill_manual(values = custom_colors$discrete) +<br>  scale_x_log10(labels = scales::comma) +<br>  labs(title = <span>'Number of expressed genes'</span>, subtitle = <span>'log-scale'</span>, y = <span>'Number of cells'</span>, fill = <span>'Sample'</span>) +<br>  theme(legend.position = <span>'none'</span>, axis.title.x = element_blank())<br><br>p5 &lt;- ggplot(cells, aes(percent_MT, fill = sample)) +<br>  geom_histogram(bins = <span>200</span>) +<br>  theme_bw() +<br>  scale_x_continuous(labels = scales::percent) +<br>  scale_fill_manual(name = <span>'Sample'</span>, values = custom_colors$discrete) +<br>  labs(title = <span>'Percent MT transcripts'</span>, subtitle = <span>'linear scale'</span>, y = <span>'Number of cells'</span>) +<br>  theme(legend.position = <span>'right'</span>, axis.title.x = element_blank(), legend.text = element_text(size = <span>8</span>))<br><br>p1 + p3 + p5 +<br>    p2 + p4 + plot_layout(ncol = <span>3</span>)<br><br>ggsave(<br>  <span>'plots/qc_histogram_ncount_nfeature_percentmt_before_filtering_3.png'</span>,<br>  p1 + p3 + p5 +<br>    p2 + p4 + plot_layout(ncol = <span>3</span>),<br>  height = <span>7</span>, width = <span>10</span><br>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100042353" data-ratio="0.7" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0hteZdtSdZ6mtALvIgKg25IsGOxsZWpTGr1hvibpyt0StaZjgKtFQahMVg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0hteZdtSdZ6mtALvIgKg25IsGOxsZWpTGr1hvibpyt0StaZjgKtFQahMVg/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">另一个有趣的关系是每个细胞中表达基因的数量比转录本的数量。这通常会产生如下所示的曲线。有时有许多细胞具有特定的模式，即表达很多基因和高转录本数量的细胞可能是垂死的细胞。</p><pre data-tool="mdnice编辑器"><span></span><code>p &lt;- ggplot(cells, aes(nCount, nFeature, color = percent_MT)) +<br>  geom_point(size = <span>0.5</span>) +<br>  scale_x_continuous(name = <span>'Number of transcripts'</span>, labels = scales::comma) +<br>  scale_y_continuous(name = <span>'Number of expressed genes'</span>, labels = scales::comma) +<br>  theme_bw() +<br>  scale_color_viridis(<br>    name = <span>'Percent MT\ntranscripts'</span>,<br>    limits = c(<span>0</span>,<span>1</span>),<br>    labels = scales::percent,<br>    guide = guide_colorbar(frame.colour = <span>'black'</span>, ticks.colour = <span>'black'</span>)<br>  )<br><br>p<br><br>ggsave(<span>'plots/qc_nfeature_over_ncount_before_filtering.png'</span>, p, height = <span>4</span>, width = <span>6</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100042352" data-ratio="0.6666666666666666" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0ht0gJFqcAoEH7zRaicZ6T5v2N5L3sibVzXbYhJODEQGo35wb0NsNrewnwQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0ht0gJFqcAoEH7zRaicZ6T5v2N5L3sibVzXbYhJODEQGo35wb0NsNrewnwQ/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">现在我们需要确定我们正在处理的矩阵的上下限。有很多种办法来做，一种就是肉眼看，另一种则是估计中位数和MAD(中位数绝对偏差)。如果你决定使用后者处理，一般建议保持在中位数附近的3-5MADs范围内。在本文中，我们将慷慨地接受任何低于中位数加上5倍的MAD。我们不需要下限，因为矩阵显然已经预先用每个细胞转录本和表达基因数量的可接受下限进行了过滤。</p><pre data-tool="mdnice编辑器"><span></span><code><span>### 3.1.4 Define thresholds确定阈值</span><br>median_nCount &lt;- median(cells$nCount)<br><span># 4255</span><br>mad_nCount &lt;- mad(cells$nCount)<br><span># 2630.874</span><br><br>median_nFeature &lt;- median(cells$nFeature)<br><span># 803</span><br>mad_nFeature &lt;- mad(cells$nFeature)<br><span># 544.1142</span><br><br>median_percent_MT &lt;- median(cells$percent_MT)<br><span># 0.02760973</span><br>mad_percent_MT &lt;- mad(cells$percent_MT)<br><span># 0.02103674</span><br><br>thresholds_nCount &lt;- c(<span>0</span>, median_nCount + <span>5</span>*mad_nCount)<br><span># 0.00 17409.37</span><br>thresholds_nFeature &lt;- c(<span>0</span>, median_nFeature + <span>5</span>*mad_nFeature)<br><span># 0.000 3523.571</span><br>thresholds_percent_MT &lt;- c(<span>0</span>, median_percent_MT + <span>5</span>*mad_percent_MT)<br><span># 0.0000000 0.1327934</span><br>save(thresholds_nCount,thresholds_nFeature,thresholds_percent_MT,file = <span>"thresholds.RData"</span>)<br><br>p1 &lt;- ggplot(cells, aes(nCount, fill = sample)) +<br>  geom_histogram(bins = <span>200</span>) +<br>  geom_vline(xintercept = median_nCount, color = <span>'black'</span>) +<br>  geom_vline(xintercept = thresholds_nCount, color = <span>'red'</span>) +<br>  theme_bw() +<br>  scale_fill_manual(values = custom_colors$discrete) +<br>  scale_x_continuous(labels = scales::comma) +<br>  labs(title = <span>'Number of transcripts'</span>, subtitle = <span>'linear scale'</span>, y = <span>'Number of cells'</span>) +<br>  theme(legend.position = <span>'none'</span>, axis.title.x = element_blank())<br><br>p2 &lt;- ggplot(cells, aes(nCount, fill = sample)) +<br>  geom_histogram(bins = <span>200</span>) +<br>  geom_vline(xintercept = median_nCount, color = <span>'black'</span>) +<br>  geom_vline(xintercept = thresholds_nCount, color = <span>'red'</span>) +<br>  theme_bw() +<br>  scale_fill_manual(values = custom_colors$discrete) +<br>  scale_x_log10(labels = scales::comma) +<br>  labs(title = <span>'Number of transcripts'</span>, subtitle = <span>'log-scale'</span>, y = <span>'Number of cells'</span>) +<br>  theme(legend.position = <span>'none'</span>, axis.title.x = element_blank())<br><br>p3 &lt;- ggplot(cells, aes(nFeature, fill = sample)) +<br>  geom_histogram(bins = <span>200</span>) +<br>  geom_vline(xintercept = median_nFeature, color = <span>'black'</span>) +<br>  geom_vline(xintercept = thresholds_nFeature, color = <span>'red'</span>) +<br>  theme_bw() +<br>  scale_fill_manual(values = custom_colors$discrete) +<br>  scale_x_continuous(labels = scales::comma) +<br>  labs(title = <span>'Number of expressed genes'</span>, subtitle = <span>'linear scale'</span>, y = <span>'Number of cells'</span>) +<br>  theme(legend.position = <span>'none'</span>,axis.title.x = element_blank())<br><br>p4 &lt;- ggplot(cells, aes(nFeature, fill = sample)) +<br>  geom_histogram(bins = <span>200</span>) +<br>  geom_vline(xintercept = median_nFeature, color = <span>'black'</span>) +<br>  geom_vline(xintercept = thresholds_nFeature, color = <span>'red'</span>) +<br>  theme_bw() +<br>  scale_fill_manual(values = custom_colors$discrete) +<br>  scale_x_log10(labels = scales::comma) +<br>  labs(title = <span>'Number of expressed genes'</span>, subtitle = <span>'log-scale'</span>, y = <span>'Number of cells'</span>, fill = <span>'Sample'</span>) +<br>  theme(legend.position = <span>'none'</span>, axis.title.x = element_blank())<br><br>p5 &lt;- ggplot(cells, aes(percent_MT, fill = sample)) +<br>  geom_histogram(bins = <span>200</span>) +<br>  geom_vline(xintercept = median_percent_MT, color = <span>'black'</span>) +<br>  geom_vline(xintercept = thresholds_percent_MT, color = <span>'red'</span>) +<br>  theme_bw() +<br>  scale_x_continuous(labels = scales::percent) +<br>  scale_fill_manual(name = <span>'Sample'</span>, values = custom_colors$discrete) +<br>  labs(title = <span>'Percent MT transcripts'</span>, subtitle = <span>'linear scale'</span>, y = <span>'Number of cells'</span>) +<br>  theme(legend.position = <span>'right'</span>, axis.title.x = element_blank(), legend.text = element_text(size = <span>8</span>))<br><br>p1 + p3 + p5 +<br>    p2 + p4 + plot_layout(ncol = <span>3</span>)<br><br>ggsave(<br>  <span>'plots/qc_histogram_ncount_nfeature_percentmt_thresholds.png'</span>,<br>  p1 + p3 + p5 +<br>    p2 + p4 + plot_layout(ncol = <span>3</span>),<br>  height = <span>7</span>, width = <span>10</span><br>)<br><br><span>#对另一种表现方法的图加入阈值</span><br><br>p &lt;- ggplot(cells, aes(nCount, nFeature, color = percent_MT)) +<br>  geom_point(size = <span>0.5</span>) +<br>  geom_hline(yintercept = thresholds_nFeature, color = <span>'red'</span>) +<br>  geom_vline(xintercept = thresholds_nCount, color = <span>'red'</span>) +<br>  scale_x_continuous(name = <span>'Number of transcripts'</span>, labels = scales::comma) +<br>  scale_y_continuous(name = <span>'Number of expressed genes'</span>, labels = scales::comma) +<br>  theme_bw() +<br>  scale_color_viridis(<br>    name = <span>'Percent MT\ntranscripts'</span>,<br>    limits = c(<span>0</span>,<span>1</span>),<br>    labels = scales::percent,<br>    guide = guide_colorbar(frame.colour = <span>'black'</span>, ticks.colour = <span>'black'</span>)<br>  )<br><br>p<br><br>ggsave(<span>'plots/qc_nfeature_over_ncount_thresholds.png'</span>, p, height = <span>4</span>, width = <span>6</span>)<br>```<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100042360" data-ratio="0.7" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0ht6zZzd6RIEbqkGfH67wR26AY1dJtHI5PSd5xLr29SplUSg0dMeOk8eQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0ht6zZzd6RIEbqkGfH67wR26AY1dJtHI5PSd5xLr29SplUSg0dMeOk8eQ/640?wx_fmt=png&amp;from=appmsg"></figure><figure data-tool="mdnice编辑器"><img data-imgfileid="100042356" data-ratio="0.6666666666666666" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0htm4nqZVibGznlojlhx1vicUgSVkpJlvcVUicppAaZ1VBE8rKPUGia0UWrGw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0htm4nqZVibGznlojlhx1vicUgSVkpJlvcVUicppAaZ1VBE8rKPUGia0UWrGw/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">我们识别出低质量的细胞，现在进行过滤</p><pre data-tool="mdnice编辑器"><span></span><code><span>### 3.1.5 After filtering过滤</span><br>cells_filtered &lt;- cells[cells$nCount &gt;= thresholds_nCount[<span>1</span>]&amp;<br>    cells$nCount &lt;= thresholds_nCount[<span>2</span>]&amp;<br>    cells$nFeature &gt;= thresholds_nFeature[<span>1</span>]&amp;<br>    cells$nFeature &lt;= thresholds_nFeature[<span>2</span>]&amp;<br>    cells$percent_MT &gt;= thresholds_percent_MT[<span>1</span>]&amp;<br>    cells$percent_MT &lt;= thresholds_percent_MT[<span>2</span>],]<br><br>p1 &lt;- ggplot(cells_filtered, aes(nCount, fill = sample)) +<br>  geom_histogram(bins = <span>200</span>) +<br>  geom_vline(xintercept = median_nCount, color = <span>'black'</span>) +<br>  geom_vline(xintercept = thresholds_nCount, color = <span>'red'</span>) +<br>  theme_bw() +<br>  scale_fill_manual(values = custom_colors$discrete) +<br>  scale_x_continuous(labels = scales::comma) +<br>  labs(title = <span>'Number of transcripts'</span>, subtitle = <span>'linear scale'</span>, y = <span>'Number of cells'</span>) +<br>  theme(legend.position = <span>'none'</span>, axis.title.x = element_blank())<br><br>p2 &lt;- ggplot(cells_filtered, aes(nCount, fill = sample)) +<br>  geom_histogram(bins = <span>200</span>) +<br>  geom_vline(xintercept = median_nCount, color = <span>'black'</span>) +<br>  geom_vline(xintercept = thresholds_nCount, color = <span>'red'</span>) +<br>  theme_bw() +<br>  scale_fill_manual(values = custom_colors$discrete) +<br>  scale_x_log10(labels = scales::comma) +<br>  labs(title = <span>'Number of transcripts'</span>, subtitle = <span>'log-scale'</span>, y = <span>'Number of cells'</span>) +<br>  theme(legend.position = <span>'none'</span>, axis.title.x = element_blank())<br><br>p3 &lt;- ggplot(cells_filtered, aes(nFeature, fill = sample)) +<br>  geom_histogram(bins = <span>200</span>) +<br>  geom_vline(xintercept = median_nFeature, color = <span>'black'</span>) +<br>  geom_vline(xintercept = thresholds_nFeature, color = <span>'red'</span>) +<br>  theme_bw() +<br>  scale_fill_manual(values = custom_colors$discrete) +<br>  scale_x_continuous(labels = scales::comma) +<br>  labs(title = <span>'Number of expressed genes'</span>, subtitle = <span>'linear scale'</span>, y = <span>'Number of cells'</span>) +<br>  theme(legend.position = <span>'none'</span>,axis.title.x = element_blank())<br><br>p4 &lt;- ggplot(cells_filtered, aes(nFeature, fill = sample)) +<br>  geom_histogram(bins = <span>200</span>) +<br>  geom_vline(xintercept = median_nFeature, color = <span>'black'</span>) +<br>  geom_vline(xintercept = thresholds_nFeature, color = <span>'red'</span>) +<br>  theme_bw() +<br>  scale_fill_manual(values = custom_colors$discrete) +<br>  scale_x_log10(labels = scales::comma) +<br>  labs(title = <span>'Number of expressed genes'</span>, subtitle = <span>'log-scale'</span>, y = <span>'Number of cells'</span>, fill = <span>'Sample'</span>) +<br>  theme(legend.position = <span>'none'</span>, axis.title.x = element_blank())<br><br>p5 &lt;- ggplot(cells_filtered, aes(percent_MT, fill = sample)) +<br>  geom_histogram(bins = <span>200</span>) +<br>  geom_vline(xintercept = median_percent_MT, color = <span>'black'</span>) +<br>  geom_vline(xintercept = thresholds_percent_MT, color = <span>'red'</span>) +<br>  theme_bw() +<br>  scale_x_continuous(labels = scales::percent) +<br>  scale_fill_manual(name = <span>'Sample'</span>, values = custom_colors$discrete) +<br>  labs(title = <span>'Percent MT transcripts'</span>, subtitle = <span>'linear scale'</span>, y = <span>'Number of cells'</span>) +<br>  theme(legend.position = <span>'right'</span>, axis.title.x = element_blank(), legend.text = element_text(size = <span>8</span>))<br><br>p1 + p3 + p5 +<br>    p2 + p4 + plot_layout(ncol = <span>3</span>)<br><br>ggsave(<br>  <span>'plots/qc_histogram_ncount_nfeature_percentmt_filtered.png'</span>,<br>  p1 + p3 + p5 +<br>    p2 + p4 + plot_layout(ncol = <span>3</span>),<br>  height = <span>7</span>, width = <span>10</span><br>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100042358" data-ratio="0.7" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0htacJrRRdrq9Y5cHqGJ1UmYQ9SjrOPiaHgWF6y7IJyNDkrxZS9h7vgacg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0htacJrRRdrq9Y5cHqGJ1UmYQ9SjrOPiaHgWF6y7IJyNDkrxZS9h7vgacg/640?wx_fmt=png&amp;from=appmsg"></figure><figure data-tool="mdnice编辑器"><img data-imgfileid="100042357" data-ratio="0.6666666666666666" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0htORDor9MWSkyLsc5FY0vdJH7XibFhgWSWiauO4H1QRsYwXGjbzQF6jj3A/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0htORDor9MWSkyLsc5FY0vdJH7XibFhgWSWiauO4H1QRsYwXGjbzQF6jj3A/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">如前文提到，我们应当过滤掉在小于五个细胞中表达的基因</p><pre data-tool="mdnice编辑器"><span></span><code><span>## 3.2 Filter genes过滤基因</span><br>genes &lt;- data.frame(<br>  gene = rownames(transcripts$raw$merged),<br>  count = Matrix::rowSums(transcripts$raw$merged),<br>  cells = Matrix::rowSums(transcripts$raw$merged != <span>0</span>)<br>)<br><br>genes_to_keep &lt;- genes[genes$cells &gt;= <span>5</span>,<span>"gene"</span>]<br>length(genes_to_keep)<br><span>#33000个过滤到16000个</span><br><br><span>## 3.3 Generate filtered transcript matrix生成过滤后的转录本矩阵</span><br>transcripts$raw$filtered &lt;- transcripts$raw$merged[genes_to_keep,cells_to_keep]<br>save(transcripts,file = <span>"./data/filtered_transcripts_list.RData"</span>)<br><br>cells_per_sample_after_filtering &lt;- data.frame(<br>  sample = character(),<br>  before = numeric(),<br>  after = numeric()<br>)<br><br><span>for</span> ( i <span>in</span> sample_names ) {<br>  tmp &lt;- data.frame(<br>    sample = i,<br>    before = grep(colnames(transcripts$raw$merged), pattern = paste0(<span>'-'</span>, i)) %&gt;% length(),<br>    after = grep(colnames(transcripts$raw$filtered), pattern = paste0(<span>'-'</span>, i)) %&gt;% length()<br>  )<br>  cells_per_sample_after_filtering &lt;- rbind(cells_per_sample_after_filtering, tmp)<br>}<br><br>knitr::kable(cells_per_sample_after_filtering)<br><span>#在下表中，我们可以看到过滤前后每个样品中有多少细胞:</span><br></code></pre><h2 data-tool="mdnice编辑器"><span></span><span>4.生成Seurat对象</span><span></span></h2><pre data-tool="mdnice编辑器"><span></span><code><span>#准备好转录组count矩阵后，我们现在可以初始化我们的Seurat对象。之后，我们在meta data中存储每个样本赋值。</span><br>seurat &lt;- CreateSeuratObject(<br>  counts = transcripts$raw$filtered,<br>  min.cells = <span>0</span>,<br>  min.features = <span>0</span><br>)<br><br>seurat@meta.data$sample &lt;- Cells(seurat) %&gt;%<br>  strsplit(<span>'-'</span>) %&gt;%<br>  vapply(FUN.VALUE = character(<span>1</span>), `[`, <span>2</span>) %&gt;%<br>  factor(levels = sample_names)<br><br><span>#为了确保万无一失，让我们再看一次样本中每个细胞转录本和表达基因的分布。</span><br>temp_labels &lt;- seurat@meta.data$sample %&gt;%<br>  table %&gt;% as.data.frame()<br>colnames(temp_labels) &lt;- c(<span>"sample"</span>,<span>"n"</span>)<br><br>p1 &lt;- ggplot() +<br>  geom_half_violin(<br>    data = seurat@meta.data, aes(sample, nCount_RNA, fill = sample),<br>    side = <span>'l'</span>, show.legend = <span>FALSE</span>, trim = <span>FALSE</span><br>  ) +<br>  geom_half_boxplot(<br>    data = seurat@meta.data, aes(sample, nCount_RNA, fill = sample),<br>    side = <span>'r'</span>, outlier.color = <span>NA</span>, width = <span>0.4</span>, show.legend = <span>FALSE</span><br>  ) +<br>  geom_text(<br>    data = temp_labels,<br>    aes(x = sample, y = -<span>Inf</span>, label = paste0(<span>'n = '</span>, format(n, big.mark = <span>','</span>, trim = <span>TRUE</span>)), vjust = -<span>1</span>),<br>    color = <span>'black'</span>, size = <span>2.8</span><br>  ) +<br>  scale_color_manual(values = custom_colors$discrete) +<br>  scale_fill_manual(values = custom_colors$discrete) +<br>  scale_y_continuous(labels = scales::comma, expand = c(<span>0.08</span>,<span>0</span>)) +<br>  theme_bw() +<br>  labs(x = <span>''</span>, y = <span>'Number of transcripts'</span>) +<br>  theme(<br>    panel.grid.major.x = element_blank(),<br>    axis.title.x = element_blank()<br>  )<br><br>p2 &lt;- ggplot() +<br>  geom_half_violin(<br>    data = seurat@meta.data, aes(sample, nFeature_RNA, fill = sample),<br>    side = <span>'l'</span>, show.legend = <span>FALSE</span>, trim = <span>FALSE</span><br>  ) +<br>  geom_half_boxplot(<br>    data = seurat@meta.data, aes(sample, nFeature_RNA, fill = sample),<br>    side = <span>'r'</span>, outlier.color = <span>NA</span>, width = <span>0.4</span>, show.legend = <span>FALSE</span><br>  ) +<br>  geom_text(<br>    data = temp_labels,<br>    aes(x = sample, y = -<span>Inf</span>, label = paste0(<span>'n = '</span>, format(n, big.mark = <span>','</span>, trim = <span>TRUE</span>)), vjust = -<span>1</span>),<br>    color = <span>'black'</span>, size = <span>2.8</span><br>  ) +<br>  scale_color_manual(values = custom_colors$discrete) +<br>  scale_fill_manual(values = custom_colors$discrete) +<br>  scale_y_continuous(<br>    name = <span>'Number of expressed genes'</span>,<br>    labels = scales::comma, expand = c(<span>0.08</span>,<span>0</span>)<br>  ) +<br>  theme_bw() +<br>  theme(<br>    panel.grid.major.x = element_blank(),<br>    axis.title.x = element_blank()<br>  )<br><br>p1 + p2 + plot_layout(ncol = <span>2</span>)<br><br>ggsave(<br>  <span>'plots/ncount_nfeature_by_sample.png'</span>,<br>  p1 + p2 + plot_layout(ncol = <span>2</span>), height = <span>4</span>, width = <span>10</span><br>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100042355" data-ratio="0.4" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0htlHQhd7ic7YwZ1ytoZG6MsVAkI4f0gBMerkSKACo7uNsDvn8Oz0n1WEg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0htlHQhd7ic7YwZ1ytoZG6MsVAkI4f0gBMerkSKACo7uNsDvn8Oz0n1WEg/640?wx_fmt=png&amp;from=appmsg"></figure><h2 data-tool="mdnice编辑器"><span></span><span>5.Normalize expression data归一化</span><span></span></h2><p data-tool="mdnice编辑器">接下来，我们将原始转录本count归一化(以便每个细胞具有相同数量的转录本)并使用对数刻度。这可以用不同的方法和不同的工具来完成。Seurat包中的<code>SCTransform()</code>和<code>NormalizeData()</code>方法以及scran中的<code>logNormCounts()</code>函数有着很好的运行效果。</p><p data-tool="mdnice编辑器">这个函数调用<code>sctransform::vst</code>。sctransform包可从 https://github.com/satijalab/sctransform 获得。可使用此函数作为NormalizeData、FindVariableFeatures、ScaleData workflow的替代方法。结果保存在新的assay中(默认命名为SCT)，counts为校正后counts，data为log1p(counts)，scale.data为Pearson残差。<code>sctransform::vst</code>的中间结果保存在新的assay的misc槽中。</p><pre data-tool="mdnice编辑器"><span></span><code><span>if</span>(!<span>require</span>(glmGamPoi)) {<br>  BiocManager::install(<span>'glmGamPoi'</span>)<br>}<br><span>library</span>(glmGamPoi)<br>seurat &lt;- SCTransform(seurat, assay = <span>'RNA'</span>)<br>save(seurat,file = <span>"SCT_seurat.RData"</span>)<br></code></pre><h2 data-tool="mdnice编辑器"><span></span><span>6. Data integration 数据整合</span><span></span></h2><p data-tool="mdnice编辑器">在一些情况下，在不同预处理之后显示有明显批次效应，整合数据集则是有意义的。同样，有许多方法可以执行此任务。Seurat包中内置的数据整合过程已经被证明有很好的表现，我个人也有过很好的使用经验。对于这里分析的样本，我们不需要它，因此将跳过这一部分。</p><h2 data-tool="mdnice编辑器"><span></span><span>7. Principal component analysis 主成分分析</span><span></span></h2><p data-tool="mdnice编辑器">主成分分析(PCA)经常被用作数据集降维的第一步。这里，我们计算前50个主成分。然后，我们必须选择用于继续分析的主成分数量。在最近的一篇论文中Germain等人建议使用intrinsicDimension包中的<code>maxLikGlobalDimEst()</code>函数来测试数据集的维数。为此，还必须为参数k选择一个值。根据作者的经验，10-20范围内的值给出了合理且没有太大差异的结果。在下面的PCA结果图中，作者突出显示了<code>maxLikGlobalDimEst()</code>(蓝线)的输出值，它在这里接近8。在作者看来，这个估计偏低了，所以将继续使用前15的主成分(红线)。此外，通常情况下，与使用太少的主成分相比，使用更多的主成分要好得多。</p><pre data-tool="mdnice编辑器"><span></span><code>seurat &lt;- RunPCA(seurat, assay = <span>'SCT'</span>, npcs = <span>50</span>)<br><br>intrinsicDimension::maxLikGlobalDimEst(seurat@reductions$pca@cell.embeddings, k = <span>10</span>)<br><span># 8.202795 </span><br><br>p &lt;- data.frame(<br>    PC = <span>1</span>:<span>50</span>,<br>    stdev = seurat@reductions$pca@stdev<br>  ) %&gt;%<br>  ggplot(aes(PC, stdev)) +<br>  geom_point() +<br>  geom_vline(xintercept = <span>8</span>, color = <span>'blue'</span>) +<br>  geom_vline(xintercept = <span>15</span>, color = <span>'red'</span>) +<br>  theme_bw() +<br>  labs(x = <span>'Principal components'</span>, y = <span>'Standard deviation'</span>)<br><br>p<br><br>ggsave(<span>'plots/principal_components.png'</span>, p, height = <span>4</span>, width = <span>5</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100042365" data-ratio="0.8" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0htMAYlBiazCIDiarfVAbe3aY9L9RDhBOdj6eKkCP6OvH3M2BVB6lCR9URw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0htMAYlBiazCIDiarfVAbe3aY9L9RDhBOdj6eKkCP6OvH3M2BVB6lCR9URw/640?wx_fmt=png&amp;from=appmsg"></figure></section><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><h2 data-tool="mdnice编辑器"><span></span><span>8 Clustering聚类</span><span></span></h2><p data-tool="mdnice编辑器">鉴定具有相似转录谱的细胞亚群</p><pre data-tool="mdnice编辑器"><span></span><code><span>## 8.1 Define cluster聚类</span><br><span>#我们使用Seurat的FindNeighbors和FindClusters聚类我们做的数据集中的细胞</span><br>seurat &lt;- FindNeighbors(seurat, reduction = <span>'pca'</span>, dims = <span>1</span>:<span>15</span>)<br>seurat &lt;- FindClusters(seurat, resolution = <span>0.8</span>)<br><br><span>## 8.2 Number of transcripts and expressed genes</span><br>如前文我们对样本执行过的操作，我们检查每一个聚类中的平均转录本和表达的基因数量。<br>temp_labels &lt;- seurat@meta.data[,<span>"seurat_clusters"</span>] %&gt;% <br>  table %&gt;% as.data.frame<br>colnames(temp_labels)&lt;-c(<span>"seurat_clusters"</span>,<span>"n"</span>)<br><br>p1 &lt;- ggplot() +<br>  geom_half_violin(<br>    data = seurat@meta.data, aes(seurat_clusters, nCount_RNA, fill = seurat_clusters),<br>    side = <span>'l'</span>, show.legend = <span>FALSE</span>, trim = <span>FALSE</span><br>  ) +<br>  geom_half_boxplot(<br>    data = seurat@meta.data, aes(seurat_clusters, nCount_RNA, fill = seurat_clusters),<br>    side = <span>'r'</span>, outlier.color = <span>NA</span>, width = <span>0.4</span>, show.legend = <span>FALSE</span><br>  ) +<br>  geom_text(<br>    data = temp_labels,<br>    aes(x = seurat_clusters, y = -<span>Inf</span>, label = paste0(<span>'n = '</span>, format(n, big.mark = <span>','</span>, trim = <span>TRUE</span>)), vjust = -<span>1</span>),<br>    color = <span>'black'</span>, size = <span>2.8</span><br>  ) +<br>  scale_color_manual(values = custom_colors$discrete) +<br>  scale_fill_manual(values = custom_colors$discrete) +<br>  scale_y_continuous(name = <span>'Number of transcripts'</span>, labels = scales::comma, expand = c(<span>0.08</span>, <span>0</span>)) +<br>  theme_bw() +<br>  theme(<br>    panel.grid.major.x = element_blank(),<br>    axis.title.x = element_blank()<br>  )<br><br>p2 &lt;- ggplot() +<br>  geom_half_violin(<br>    data = seurat@meta.data, aes(seurat_clusters, nFeature_RNA, fill = seurat_clusters),<br>    side = <span>'l'</span>, show.legend = <span>FALSE</span>, trim = <span>FALSE</span><br>  ) +<br>  geom_half_boxplot(<br>    data = seurat@meta.data, aes(seurat_clusters, nFeature_RNA, fill = seurat_clusters),<br>    side = <span>'r'</span>, outlier.color = <span>NA</span>, width = <span>0.4</span>, show.legend = <span>FALSE</span><br>  ) +<br>  geom_text(<br>    data = temp_labels,<br>    aes(x = seurat_clusters, y = -<span>Inf</span>, label = paste0(<span>'n = '</span>, format(n, big.mark = <span>','</span>, trim = <span>TRUE</span>)), vjust = -<span>1</span>),<br>    color = <span>'black'</span>, size = <span>2.8</span><br>  ) +<br>  scale_color_manual(values = custom_colors$discrete) +<br>  scale_fill_manual(values = custom_colors$discrete) +<br>  scale_y_continuous(name = <span>'Number of expressed genes'</span>, labels = scales::comma, expand = c(<span>0.08</span>, <span>0</span>)) +<br>  theme_bw() +<br>  theme(<br>    panel.grid.major.x = element_blank(),<br>    axis.title.x = element_blank()<br>  )<br><br>p1 + p2 + plot_layout(ncol = <span>1</span>)<br><br>ggsave(<br>  <span>'plots/ncount_nfeature_by_cluster.png'</span>,<br>  p1 + p2 + plot_layout(ncol = <span>1</span>),<br>  height = <span>7</span>, width = <span>14</span><br>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100042375" data-ratio="0.5" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0htORZYrgtX39ThPW5xebGLZ86Cu7IicDL56Jj6ASLFJOt5B1UZSXQNeyg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0htORZYrgtX39ThPW5xebGLZ86Cu7IicDL56Jj6ASLFJOt5B1UZSXQNeyg/640?wx_fmt=png&amp;from=appmsg"></figure><h2 data-tool="mdnice编辑器"><span></span><span>9.Cell cycle analysis细胞周期分析</span><span></span></h2><p data-tool="mdnice编辑器">使用Seurat包的<code>CellCycleScoring()</code>函数，我们可以为每个细胞分配细胞周期：G1, S, G2M</p><pre data-tool="mdnice编辑器"><span></span><code><span>## 9.1 Infer cell cycle status推断细胞周期状态</span><br>seurat &lt;- CellCycleScoring(<br>  seurat,<br>  assay = <span>'SCT'</span>,<br>  s.features = cc.genes$s.genes,<br>  g2m.features = cc.genes$g2m.genes<br>)<br><br>seurat@meta.data$cell_cycle_seurat &lt;- seurat@meta.data$Phase<br>seurat@meta.data$Phase &lt;- <span>NULL</span><br><span>#字符串转因子</span><br>seurat@meta.data$cell_cycle_seurat &lt;- factor(<br>  seurat@meta.data$cell_cycle_seurat, levels = c(<span>'G1'</span>, <span>'S'</span>, <span>'G2M'</span>)<br>)<br><br><span>## 9.2 Composition of samples and clusters by cell cycle样本和聚类中细胞的细胞周期组成</span><br><span>#现在检查每个样本和聚类中的细胞周期阶段分布</span><br><span>#样本中：</span><br>table_samples_by_cell_cycle&lt;-data.frame()<br><span>for</span>(i <span>in</span> unique(seurat@meta.data$sample)){<br>  seurat@meta.data[seurat@meta.data$sample==i,<span>"cell_cycle_seurat"</span>] %&gt;% table -&gt; nCycle<br>  c(i,sum(nCycle),nCycle) -&gt;nCycle<br>  names(nCycle)[<span>1</span>:<span>2</span>]&lt;-c(<span>"sample"</span>,<span>"total_cell_count"</span>)<br>  rbind(table_samples_by_cell_cycle,nCycle)-&gt;table_samples_by_cell_cycle<br>}<br>colnames(table_samples_by_cell_cycle)&lt;-names(nCycle)<br><span>for</span> (i <span>in</span> <span>2</span>:<span>5</span>) {<br>  as.numeric(table_samples_by_cell_cycle[,i])-&gt;table_samples_by_cell_cycle[,i]<br>}<br><br>knitr::kable(table_samples_by_cell_cycle)<br><br><span>#聚类中：</span><br>table_clusters_by_cell_cycle&lt;-data.frame()<br><span>for</span>(i <span>in</span> unique(seurat@meta.data$seurat_clusters)){<br>  seurat@meta.data[seurat@meta.data$seurat_clusters==i,<span>"cell_cycle_seurat"</span>] %&gt;% table -&gt; nCycle<br>  c(i,sum(nCycle),nCycle) -&gt;nCycle<br>  names(nCycle)[<span>1</span>:<span>2</span>]&lt;-c(<span>"cluster"</span>,<span>"total_cell_count"</span>)<br>  rbind(table_clusters_by_cell_cycle,nCycle)-&gt;table_clusters_by_cell_cycle<br>}<br>colnames(table_clusters_by_cell_cycle)&lt;-names(nCycle)<br><br><span>for</span> (i <span>in</span> <span>1</span>:<span>5</span>) {<br>  as.numeric(table_clusters_by_cell_cycle[,i])-&gt;table_clusters_by_cell_cycle[,i]<br>}<br><br>table_clusters_by_cell_cycle[order(table_clusters_by_cell_cycle$cluster,decreasing = <span>F</span>),]-&gt;table_clusters_by_cell_cycle<br>rownames(table_clusters_by_cell_cycle)&lt;-<span>NULL</span><br><br>knitr::kable(table_clusters_by_cell_cycle)<br><br><span>#不同阶段细胞所占比例</span><br>temp_labels &lt;- seurat@meta.data$sample %&gt;% table %&gt;% as.data.frame<br>colnames(temp_labels)&lt;-c(<span>"sample"</span>,<span>"n"</span>)<br><br>p1 &lt;- table_samples_by_cell_cycle %&gt;%<br>  select(-c(<span>'total_cell_count'</span>)) %&gt;%<br>  reshape2::melt(id.vars = <span>'sample'</span>) %&gt;%<br>  mutate(sample = factor(sample, levels = levels(seurat@meta.data$sample))) %&gt;%<br>  ggplot(aes(sample, value)) +<br>  geom_bar(aes(fill = variable), position = <span>'fill'</span>, stat = <span>'identity'</span>) +<br>  geom_text(<br>    data = temp_labels,<br>    aes(x = sample, y = <span>Inf</span>, label = paste0(<span>'n = '</span>, format(n, big.mark = <span>','</span>, trim = <span>TRUE</span>)), vjust = -<span>1</span>),<br>    color = <span>'black'</span>, size = <span>2.8</span><br>  ) +<br>  scale_fill_manual(name = <span>'Cell cycle'</span>, values = custom_colors$cell_cycle) +<br>  scale_y_continuous(name = <span>'Percentage [%]'</span>, labels = scales::percent_format(), expand = c(<span>0.01</span>,<span>0</span>)) +<br>  coord_cartesian(clip = <span>'off'</span>) +<br>  theme_bw() +<br>  theme(<br>    legend.position = <span>'none'</span>,<br>    plot.title = element_text(hjust = <span>0.5</span>),<br>    text = element_text(size = <span>16</span>),<br>    panel.grid.major = element_blank(),<br>    panel.grid.minor = element_blank(),<br>    axis.title.x = element_blank(),<br>    axis.text.x = element_text(angle = <span>45</span>, hjust = <span>1</span>, vjust = <span>1</span>),<br>    plot.margin = margin(t = <span>20</span>, r = <span>0</span>, b = <span>0</span>, l = <span>0</span>, unit = <span>'pt'</span>)<br>  )<br><br>p1<br><br>ggsave(<br>  <span>'plots/composition_samples_clusters_by_cell_cycle_by_percent.png'</span>,<br>  p1 + p2 +<br>  plot_layout(ncol = <span>2</span>, widths = c(<br>    seurat@meta.data$sample %&gt;% unique() %&gt;% length(),<br>    seurat@meta.data$seurat_clusters %&gt;% unique() %&gt;% length()<br>  )),<br>  width = <span>18</span>, height = <span>8</span><br>)<br><br>save(seurat,file = <span>"Cyclin_seurat.RData"</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100042376" data-ratio="0.4444444444444444" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0hte5Sumtv8mjxZk65LaSzyjxhU5icMicUcc1MQ5wRWjVvPTWxgnkdyoz1Q/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0hte5Sumtv8mjxZk65LaSzyjxhU5icMicUcc1MQ5wRWjVvPTWxgnkdyoz1Q/640?wx_fmt=png&amp;from=appmsg"></figure><h2 data-tool="mdnice编辑器"><span></span><span>10.Dimensional reduction降维</span><span></span></h2><p data-tool="mdnice编辑器">在下文中我们讲数据集中细胞的表达谱投影到二维或三维中展示，通常使用t-SNE或UMAP的方法</p><pre data-tool="mdnice编辑器"><span></span><code><span>##10.1 t-SNE图</span><br><span>library</span>(ggnetwork)<br>load(<span>"Cyclin_seurat.RData"</span>)<br>SCT_snn &lt;- seurat@graphs$SCT_snn %&gt;%<br>  as.matrix() %&gt;%<br>  ggnetwork() %&gt;%<br>  left_join(seurat@meta.data %&gt;% mutate(vertex.names = rownames(.)), by = <span>'vertex.names'</span>)<br><br>p1 &lt;- ggplot(SCT_snn, aes(x = x, y = y, xend = xend, yend = yend)) +<br>  geom_edges(color = <span>'grey50'</span>, alpha = <span>0.05</span>) +<br>  geom_nodes(aes(color = sample), size = <span>0.5</span>) +<br>  scale_color_manual(<br>    name = <span>'Sample'</span>, values = custom_colors$discrete,<br>    guide = guide_legend(ncol = <span>1</span>, override.aes = list(size = <span>2</span>))<br>  ) +<br>  theme_blank() +<br>  theme(legend.position = <span>'left'</span>) +<br>  annotate(<br>    geom = <span>'text'</span>, x = <span>Inf</span>, y = -<span>Inf</span>,<br>    label = paste0(<span>'n = '</span>, format(nrow(seurat@meta.data), big.mark = <span>','</span>, trim = <span>TRUE</span>)),<br>    vjust = -<span>1.5</span>, hjust = <span>1.25</span>, color = <span>'black'</span>, size = <span>2.5</span><br>  )<br><br>p2 &lt;- ggplot(SCT_snn, aes(x = x, y = y, xend = xend, yend = yend)) +<br>  geom_edges(color = <span>'grey50'</span>, alpha = <span>0.05</span>) +<br>  geom_nodes(aes(color = seurat_clusters), size = <span>0.5</span>) +<br>  scale_color_manual(<br>    name = <span>'Cluster'</span>, values = custom_colors$discrete,<br>    guide = guide_legend(ncol = <span>1</span>, override.aes = list(size = <span>2</span>))<br>  ) +<br>  theme_blank() +<br>  theme(legend.position = <span>'right'</span>) +<br>  annotate(<br>    geom = <span>'text'</span>, x = <span>Inf</span>, y = -<span>Inf</span>,<br>    label = paste0(<span>'n = '</span>, format(nrow(seurat@meta.data), big.mark = <span>','</span>, trim = <span>TRUE</span>)),<br>    vjust = -<span>1.5</span>, hjust = <span>1.25</span>, color = <span>'black'</span>, size = <span>2.5</span><br>  )<br><br>ggsave(<br>  <span>'plots/snn_graph_by_sample_cluster.png'</span>,<br>  p1 + p2 + plot_layout(ncol = <span>2</span>),<br>  height = <span>5</span>, width = <span>11</span><br>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100042377" data-ratio="0.4546296296296296" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0htQS1YdNFk74awLZx1tJNJR1hRZsxBic2EGRajyJ1lbVWOnkPusHicDJYQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0htQS1YdNFk74awLZx1tJNJR1hRZsxBic2EGRajyJ1lbVWOnkPusHicDJYQ/640?wx_fmt=png&amp;from=appmsg"></figure><pre data-tool="mdnice编辑器"><span></span><code><span>## 10.2 UMAP</span><br><span>### 10.2.1 Calculate UMAP计算UMAP</span><br><span>#我们使用与前文用于聚类一致的15个主成分进行UMAP计算</span><br>seurat &lt;- RunUMAP(<br>  seurat,<br>  reduction.name = <span>'UMAP'</span>,<br>  reduction = <span>'pca'</span>,<br>  dims = <span>1</span>:<span>15</span>,<br>  n.components = <span>2</span>,<br>  seed.use = <span>100</span><br>)<br><br><span>### 10.2.2 Overview预览</span><br><span>#在计算完UMAP之后，我们分别根据转录本数量，样本来源，聚类结果，和细胞周期对其进行上色</span><br><br>plot_umap_by_nCount &lt;- bind_cols(seurat@meta.data, as.data.frame(seurat@reductions$UMAP@cell.embeddings)) %&gt;%<br>  ggplot(aes(UMAP_1, UMAP_2, color = nCount_RNA)) +<br>  geom_point(size = <span>0.2</span>) +<br>  theme_bw() +<br>  scale_color_viridis(<br>    guide = guide_colorbar(frame.colour = <span>'black'</span>, ticks.colour = <span>'black'</span>),<br>    labels = scales::comma,<br>  ) +<br>  labs(color = <span>'Number of\ntranscripts'</span>) +<br>  theme(legend.position = <span>'left'</span>) +<br>  coord_fixed() +<br>  annotate(<br>    geom = <span>'text'</span>, x = <span>Inf</span>, y = -<span>Inf</span>,<br>    label = paste0(<span>'n = '</span>, format(nrow(seurat@meta.data), big.mark = <span>','</span>, trim = <span>TRUE</span>)),<br>    vjust = -<span>1.5</span>, hjust = <span>1.25</span>, color = <span>'black'</span>, size = <span>2.5</span><br>  )<br><br>plot_umap_by_sample &lt;- bind_cols(seurat@meta.data, as.data.frame(seurat@reductions$UMAP@cell.embeddings)) %&gt;%<br>  ggplot(aes(UMAP_1, UMAP_2, color = sample)) +<br>  geom_point(size = <span>0.2</span>) +<br>  theme_bw() +<br>  scale_color_manual(values = custom_colors$discrete) +<br>  labs(color = <span>'Sample'</span>) +<br>  guides(colour = guide_legend(override.aes = list(size = <span>2</span>))) +<br>  theme(legend.position = <span>'right'</span>) +<br>  coord_fixed() +<br>  annotate(<br>    geom = <span>'text'</span>, x = <span>Inf</span>, y = -<span>Inf</span>,<br>    label = paste0(<span>'n = '</span>, format(nrow(seurat@meta.data), big.mark = <span>','</span>, trim = <span>TRUE</span>)),<br>    vjust = -<span>1.5</span>, hjust = <span>1.25</span>, color = <span>'black'</span>, size = <span>2.5</span><br>  )<br><br>plot_umap_by_cluster &lt;- bind_cols(seurat@meta.data, as.data.frame(seurat@reductions$UMAP@cell.embeddings)) %&gt;%<br>  ggplot(aes(UMAP_1, UMAP_2, color = seurat_clusters)) +<br>  geom_point(size = <span>0.2</span>) +<br>  theme_bw() +<br>  scale_color_manual(<br>    name = <span>'Cluster'</span>, values = custom_colors$discrete,<br>    guide = guide_legend(ncol = <span>2</span>, override.aes = list(size = <span>2</span>))<br>  ) +<br>  theme(legend.position = <span>'left'</span>) +<br>  coord_fixed() +<br>  annotate(<br>    geom = <span>'text'</span>, x = <span>Inf</span>, y = -<span>Inf</span>,<br>    label = paste0(<span>'n = '</span>, format(nrow(seurat@meta.data), big.mark = <span>','</span>, trim = <span>TRUE</span>)),<br>    vjust = -<span>1.5</span>, hjust = <span>1.25</span>, color = <span>'black'</span>, size = <span>2.5</span><br>  )<br><br>plot_umap_by_cell_cycle &lt;- bind_cols(seurat@meta.data, as.data.frame(seurat@reductions$UMAP@cell.embeddings)) %&gt;%<br>  ggplot(aes(UMAP_1, UMAP_2, color = cell_cycle_seurat)) +<br>  geom_point(size = <span>0.2</span>) +<br>  theme_bw() +<br>  scale_color_manual(values = custom_colors$cell_cycle) +<br>  labs(color = <span>'Cell cycle'</span>) +<br>  guides(colour = guide_legend(override.aes = list(size = <span>2</span>))) +<br>  theme(legend.position = <span>'right'</span>) +<br>  coord_fixed() +<br>  annotate(<br>    geom = <span>'text'</span>, x = <span>Inf</span>, y = -<span>Inf</span>,<br>    label = paste0(<span>'n = '</span>, format(nrow(seurat@meta.data), big.mark = <span>','</span>, trim = <span>TRUE</span>)),<br>    vjust = -<span>1.5</span>, hjust = <span>1.25</span>, color = <span>'black'</span>, size = <span>2.5</span><br>  )<br><br>ggsave(<br>  <span>'plots/umap.png'</span>,<br>  plot_umap_by_nCount + plot_umap_by_sample +<br>  plot_umap_by_cluster + plot_umap_by_cell_cycle +<br>  plot_layout(ncol = <span>2</span>),<br>  height = <span>6</span>,<br>  width = <span>8.5</span><br>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100042374" data-ratio="0.7055555555555556" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0hte80ZDlNEN4bW23F42f5yVDLg23aw1Qg8FkqBuyg1oY5zeoU4ax9Y0Q/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0hte80ZDlNEN4bW23F42f5yVDLg23aw1Qg8FkqBuyg1oY5zeoU4ax9Y0Q/640?wx_fmt=png&amp;from=appmsg"></figure><pre data-tool="mdnice编辑器"><span></span><code><span>### 10.2.3 Samples样本间UMAP</span><br><span>#为了探究不同样本的结果是否能较好重合，我们根据样本划分板块再次做UMAP。</span><br>temp_labels &lt;- seurat@meta.data %&gt;%<br>  group_by(sample) %&gt;%<br>  tally()<br><br>p &lt;- bind_cols(seurat@meta.data, as.data.frame(seurat@reductions$UMAP@cell.embeddings)) %&gt;%<br>  ggplot(aes(UMAP_1, UMAP_2, color = sample)) +<br>  geom_point(size = <span>0.2</span>, show.legend = <span>FALSE</span>) +<br>  geom_text(<br>    data = temp_labels,<br>    aes(x = <span>Inf</span>, y = -<span>Inf</span>, label = paste0(<span>'n = '</span>, format(n, big.mark = <span>','</span>, trim = <span>TRUE</span>)), vjust = -<span>1.5</span>, hjust = <span>1.25</span>),<br>    color = <span>'black'</span>, size = <span>2.8</span><br>  ) +<br>  theme_bw() +<br>  scale_color_manual(values = custom_colors$discrete) +<br>  coord_fixed() +<br>  theme(<br>    legend.position = <span>'none'</span>,<br>    strip.text = element_text(face = <span>'bold'</span>, size = <span>10</span>)<br>  ) +<br>  facet_wrap(~sample, ncol = <span>5</span>)<br><br>ggsave(<br>  <span>'plots/umap_by_sample_split.png'</span>,<br>  p,<br>  height = <span>4</span>,<br>  width = <span>10</span><br>)<br><br><span>### 10.2.4 聚类</span><br><span>#对于聚类，我们在每个聚类的几何中心添个标签或自己的注释。</span><br>UMAP_centers_cluster &lt;- tibble(<br>    UMAP_1 = as.data.frame(seurat@reductions$UMAP@cell.embeddings)$UMAP_1,<br>    UMAP_2 = as.data.frame(seurat@reductions$UMAP@cell.embeddings)$UMAP_2,<br>    cluster = seurat@meta.data$seurat_clusters<br>  ) %&gt;%<br>  group_by(cluster) %&gt;%<br>  summarize(x = median(UMAP_1), y = median(UMAP_2))<br><br>p &lt;- bind_cols(seurat@meta.data, as.data.frame(seurat@reductions$UMAP@cell.embeddings)) %&gt;%<br>  ggplot(aes(UMAP_1, UMAP_2, color = seurat_clusters)) +<br>  geom_point(size = <span>0.2</span>) +<br>  geom_label(<br>    data = UMAP_centers_cluster,<br>    mapping = aes(x, y, label = cluster),<br>    size = <span>4.5</span>,<br>    fill = <span>'white'</span>,<br>    color = <span>'black'</span>,<br>    fontface = <span>'bold'</span>,<br>    alpha = <span>0.5</span>,<br>    label.size = <span>0</span>,<br>    show.legend = <span>FALSE</span><br>  ) +<br>  theme_bw() +<br>  scale_color_manual(<br>    name = <span>'Cluster'</span>, values = custom_colors$discrete,<br>    guide = guide_legend(override.aes = list(size = <span>2</span>))<br>  ) +<br>  theme(legend.position = <span>'right'</span>) +<br>  coord_fixed() +<br>  annotate(<br>    geom = <span>'text'</span>, x = <span>Inf</span>, y = -<span>Inf</span>,<br>    label = paste0(<span>'n = '</span>, format(nrow(seurat@meta.data), big.mark = <span>','</span>, trim = <span>TRUE</span>)),<br>    vjust = -<span>1.5</span>, hjust = <span>1.25</span>, color = <span>'black'</span>, size = <span>2.5</span><br>  )<br><br>ggsave(<br>  <span>'plots/umap_by_clusters.png'</span>,<br>  p, height = <span>5.5</span>, width = <span>5.5</span><br>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100042373" data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0htwJButwcYl1MzdFWhNphC1F8bX0P5NevrdQiaU07P8TsS0zpicDb6x9lw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0htwJButwcYl1MzdFWhNphC1F8bX0P5NevrdQiaU07P8TsS0zpicDb6x9lw/640?wx_fmt=png&amp;from=appmsg"></figure><figure data-tool="mdnice编辑器"><img data-imgfileid="100042378" data-ratio="0.4" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0htgl24ZQ2MvmEhiallvSl1rViauMEdAm1ESsd0W9iaGRMrxcc17wsZwMFrA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0htgl24ZQ2MvmEhiallvSl1rViauMEdAm1ESsd0W9iaGRMrxcc17wsZwMFrA/640?wx_fmt=png&amp;from=appmsg"></figure><h2 data-tool="mdnice编辑器"><span></span><span>11 Cell type annotation with SingleR使用SingleR进行细胞类型注释</span><span></span></h2><p data-tool="mdnice编辑器">细胞类型的推断有多种软件包可以实现，或者根据自己生物学背景进行手动注释。作者使用SingleR包进行了自动注释</p><pre data-tool="mdnice编辑器"><span></span><code><span>##### 11.1 Assign labels分配标签</span><br><span>#首先我们给每个细胞分配了标签作为参考。然后提取标签和分配分数，并将它们添加到Seurat对象中。</span><br><span>library</span>(SingleR)<br><br>singler_ref &lt;- celldex::BlueprintEncodeData()<br><br>singler_results_blueprintencode_main &lt;- SingleR::SingleR(<br>  test = GetAssayData(seurat, assay = <span>'SCT'</span>, layer = <span>'data'</span>),<br>  ref = singler_ref,<br>  labels = singler_ref@colData@listData$label.main<br>)<br><br>saveRDS(singler_results_blueprintencode_main, <span>"data/singler_results_blueprintencode_main.rds"</span>)<br><br>p &lt;- plotScoreHeatmap(<br>    singler_results_blueprintencode_main,<br>    show.labels = <span>TRUE</span>,<br>    annotation_col = data.frame(<br>      donor = seurat@meta.data$sample,<br>      row.names = rownames(singler_results_blueprintencode_main)<br>    )<br>  )<br><br>p<br><br>ggsave(<br>  <span>'plots/singler_score_heatmap_blueprintencode_main.png'</span>, p,<br>  height = <span>11</span>, width = <span>14</span><br>)<br><br>seurat@meta.data$cell_type_singler_blueprintencode_main &lt;- singler_results_blueprintencode_main@listData$labels<br><br>singler_scores &lt;- singler_results_blueprintencode_main@listData$scores %&gt;%<br>  as_tibble() %&gt;%<br>  dplyr::mutate(assigned_score = <span>NA</span>)<br><br><span>for</span> ( i <span>in</span> seq_len(nrow(singler_scores)) ) {<br>  singler_scores$assigned_score[i] &lt;- singler_scores[[singler_results_blueprintencode_main@listData$labels[i]]][i]<br>}<br><br>seurat@meta.data$cell_type_singler_blueprintencode_main_score &lt;- singler_scores$assigned_score<br><br>save(seurat,file=<span>"labels_assinged_seurat.Rdata"</span>)<br><br><span>## 11.2 Assignment score by sample, cluster and cell type</span><br><span>#在这里，我们绘制了每个样本、聚类和细胞类型的分配打分分布。</span><br>load(<span>"labels_assinged_seurat.Rdata"</span>)<br>temp_labels &lt;- seurat@meta.data %&gt;%<br>  group_by(sample) %&gt;%<br>  tally()<br>as.data.frame(temp_labels) -&gt; temp_labels<br><br>p1 &lt;- ggplot() +<br>  geom_half_violin(<br>    data = seurat@meta.data,<br>    aes(<br>      x = sample,<br>      y = cell_type_singler_blueprintencode_main_score,<br>      fill = sample<br>    ),<br>    side = <span>'l'</span>, show.legend = <span>FALSE</span>, trim = <span>FALSE</span><br>  ) +<br>  geom_half_boxplot(<br>    data = seurat@meta.data,<br>    aes(<br>      x = sample,<br>      y = cell_type_singler_blueprintencode_main_score,<br>      fill = sample<br>    ),<br>    side = <span>'r'</span>, outlier.color = <span>NA</span>, width = <span>0.4</span>, show.legend = <span>FALSE</span><br>  ) +<br>  geom_text(<br>    data = temp_labels,<br>    aes(<br>      x = sample,<br>      y = -<span>Inf</span>,<br>      label = paste0(<span>'n = '</span>, format(n, big.mark = <span>','</span>, trim = <span>TRUE</span>)),<br>      vjust = -<span>1</span><br>    ),<br>    color = <span>'black'</span>, size = <span>2.8</span><br>  ) +<br>  scale_color_manual(values = custom_colors$discrete) +<br>  scale_fill_manual(values = custom_colors$discrete) +<br>  scale_y_continuous(<br>    name = <span>'Assignment score'</span>,<br>    labels = scales::comma,<br>    limits = c(<span>0</span>,<span>1</span>)<br>  ) +<br>  theme_bw() +<br>  labs(title = <span>'Samples'</span>) +<br>  theme(<br>    axis.text.x = element_text(angle = <span>45</span>, hjust = <span>1</span>, vjust = <span>1</span>),<br>    axis.title.x = element_blank(),<br>    panel.grid.major.x = element_blank()<br>  )<br><br>temp_labels &lt;- seurat@meta.data %&gt;%<br>  group_by(seurat_clusters) %&gt;%<br>  tally()<br>as.data.frame(temp_labels) -&gt; temp_labels<br><br>p2 &lt;- ggplot() +<br>  geom_half_violin(<br>    data = seurat@meta.data,<br>    aes(<br>      x = seurat_clusters,<br>      y = cell_type_singler_blueprintencode_main_score,<br>      fill = seurat_clusters<br>    ),<br>    side = <span>'l'</span>, show.legend = <span>FALSE</span>, trim = <span>FALSE</span><br>  ) +<br>  geom_half_boxplot(<br>    data = seurat@meta.data,<br>    aes(<br>      x = seurat_clusters,<br>      y = cell_type_singler_blueprintencode_main_score,<br>      fill = seurat_clusters<br>    ),<br>    side = <span>'r'</span>, outlier.color = <span>NA</span>, width = <span>0.4</span>, show.legend = <span>FALSE</span><br>  ) +<br>  geom_text(<br>    data = temp_labels,<br>    aes(<br>      x = seurat_clusters,<br>      y = -<span>Inf</span>,<br>      label = paste0(<span>'n = '</span>, format(n, big.mark = <span>','</span>, trim = <span>TRUE</span>)),<br>      vjust = -<span>1</span><br>    ),<br>    color = <span>'black'</span>, size = <span>2.8</span><br>  ) +<br>  scale_color_manual(values = custom_colors$discrete) +<br>  scale_fill_manual(values = custom_colors$discrete) +<br>  scale_y_continuous(<br>    name = <span>'Assignment score'</span>,<br>    labels = scales::comma,<br>    limits = c(<span>0</span>,<span>1</span>)<br>  ) +<br>  labs(title = <span>'Clusters'</span>) +<br>  theme_bw() +<br>  theme(<br>    axis.title.x = element_blank(),<br>    panel.grid.major.x = element_blank()<br>  )<br><br>temp_labels &lt;- seurat@meta.data %&gt;%<br>  group_by(cell_type_singler_blueprintencode_main) %&gt;%<br>  tally()<br>as.data.frame(temp_labels) -&gt; temp_labels<br><br>temp_labels[temp_labels[,<span>1</span>]!=<span>"Fibroblasts"</span>,] -&gt; temp_labels<br><br><span>#table(seurat@meta.data$cell_type_singler_blueprintencode_main)</span><br><span>#DC类细胞只有一个画不了图，得删掉</span><br>data.frame(seurat@meta.data$cell_type_singler_blueprintencode_main,<br>seurat@meta.data$cell_type_singler_blueprintencode_main_score) -&gt; tmpPlotDf<br>tmpPlotDf[tmpPlotDf[,<span>1</span>]!=<span>"DC"</span>,] -&gt; tmpPlotDf<br>table(tmpPlotDf[,<span>1</span>])<br>as.character(tmpPlotDf[,<span>1</span>]) -&gt; tmpPlotDf[,<span>1</span>]<br><br>p3 &lt;- ggplot() +<br>  geom_half_violin(<br>    data = tmpPlotDf, <span>#seurat@meta.data</span><br>    aes(<br>      x = seurat.meta.data.cell_type_singler_blueprintencode_main,<br>      y = seurat.meta.data.cell_type_singler_blueprintencode_main_score,<br>      fill = seurat.meta.data.cell_type_singler_blueprintencode_main<br>    ),<br>    side = <span>'l'</span>, show.legend = <span>FALSE</span>, trim = <span>FALSE</span>,<br>  ) +<br>  geom_half_boxplot(<br>    data = tmpPlotDf,<br>    aes(<br>      x = seurat.meta.data.cell_type_singler_blueprintencode_main,<br>      y = seurat.meta.data.cell_type_singler_blueprintencode_main_score,<br>      fill = seurat.meta.data.cell_type_singler_blueprintencode_main<br>    ),<br>    side = <span>'r'</span>, outlier.color = <span>NA</span>, width = <span>0.4</span>, show.legend = <span>FALSE</span><br>  ) +<br>  geom_text(<br>    data = temp_labels,<br>    aes(<br>      x = cell_type_singler_blueprintencode_main,<br>      y = -<span>Inf</span>,<br>      label = paste0(<span>'n = '</span>, format(n, big.mark = <span>','</span>, trim = <span>TRUE</span>)),<br>      vjust = -<span>1</span><br>    ),<br>    color = <span>'black'</span>, size = <span>2.8</span><br>  ) +<br>  scale_color_manual(values = custom_colors$discrete) +<br>  scale_fill_manual(values = custom_colors$discrete) +<br>  scale_y_continuous(<br>    name = <span>'Assignment score'</span>,<br>    labels = scales::comma,<br>    limits = c(<span>0</span>,<span>1</span>)<br>  ) +<br>  labs(title = <span>'Cell types'</span>) +<br>  theme_bw() +<br>  theme(<br>    axis.title.x = element_blank(),<br>    panel.grid.major.x = element_blank()<br>  )<br><br>(p1 | p3) / p2 + plot_layout(ncol = <span>1</span>)<br><br><span>#若出现以下报错，说明数据不足以画小提琴图，可能是某个种类细胞只有一个，剔除后就正常了</span><br><span># #Error in `geom_half_violin()`:</span><br><span># ! Problem while converting geom to grob.</span><br><span># ℹ Error occurred in the 1st layer.</span><br><span># Caused by error in `if ((is_panel &amp; (side[1] == "l")) | is_group) ...`:</span><br><span># ! 需要TRUE/FALSE值的地方不可以用缺少值</span><br><span># Run `rlang::last_trace()` to see where the error occurred.</span><br><span># Warning message:</span><br><span># Groups with fewer than two datapoints have been dropped.</span><br><span># ℹ Set `drop = FALSE` to consider such groups for position adjustment purposes. </span><br><br>ggsave(<br>  <span>'plots/singler_blueprintencode_main_assignment_scores_by_sample_cluster_cell_type.png'</span>,<br>  (p1 | p3) / p2 + plot_layout(ncol = <span>1</span>), height = <span>13</span>, width = <span>16</span><br>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100042381" data-ratio="0.7861111111111111" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0htWpymsnI65Dro6T7wibUicwEYEnYO9TN4a69S6kezsTwHwP0df9MzZ3Cg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0htWpymsnI65Dro6T7wibUicwEYEnYO9TN4a69S6kezsTwHwP0df9MzZ3Cg/640?wx_fmt=png&amp;from=appmsg"></figure><figure data-tool="mdnice编辑器"><img data-imgfileid="100042382" data-ratio="0.812962962962963" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0htS1X3z4z0mxZhDZvQNlFN2OHUarUC6bzdqYjQibDYndCtMhia2bPpnaCQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0htS1X3z4z0mxZhDZvQNlFN2OHUarUC6bzdqYjQibDYndCtMhia2bPpnaCQ/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">我们然后检查样本和聚类在细胞类型方面是怎样构成的</p><pre data-tool="mdnice编辑器"><span></span><code><span>## 11.3 Composition of samples and clusters by cell</span><br>table_samples_by_cell_type &lt;- seurat@meta.data %&gt;%<br>  dplyr::group_by(sample, cell_type_singler_blueprintencode_main) %&gt;%<br>  dplyr::summarize(count = n()) %&gt;%<br>  tidyr::spread(cell_type_singler_blueprintencode_main, count, fill = <span>0</span>) %&gt;%<br>  dplyr::ungroup() %&gt;%<br>  dplyr::mutate(total_cell_count = rowSums(.[c(<span>2</span>:ncol(.))])) %&gt;%<br>  dplyr::select(c(<span>"sample"</span>, <span>"total_cell_count"</span>, dplyr::everything()))<br><br>table_samples_by_cell_type %&gt;% knitr::kable()<br><br>table_clusters_by_cell_type &lt;- seurat@meta.data %&gt;%<br>  dplyr::group_by(seurat_clusters, cell_type_singler_blueprintencode_main) %&gt;%<br>  dplyr::rename(cluster = seurat_clusters) %&gt;%<br>  dplyr::summarize(count = n()) %&gt;%<br>  tidyr::spread(cell_type_singler_blueprintencode_main, count, fill = <span>0</span>) %&gt;%<br>  dplyr::ungroup() %&gt;%<br>  dplyr::mutate(total_cell_count = rowSums(.[c(<span>2</span>:ncol(.))])) %&gt;%<br>  dplyr::select(c(<span>"cluster"</span>, <span>"total_cell_count"</span>, dplyr::everything()))<br><br>table_clusters_by_cell_type %&gt;% knitr::kable()<br><br><span>#下图是按样本/聚类的细胞百分比组成。</span><br>temp_labels &lt;- seurat@meta.data %&gt;%<br>  group_by(sample) %&gt;%<br>  tally()<br><br>p1 &lt;- table_samples_by_cell_type %&gt;%<br>  select(-c(<span>'total_cell_count'</span>)) %&gt;%<br>  reshape2::melt(id.vars = <span>'sample'</span>) %&gt;%<br>  mutate(sample = factor(sample, levels = levels(seurat@meta.data$sample))) %&gt;%<br>  ggplot(aes(sample, value)) +<br>  geom_bar(aes(fill = variable), position = <span>'fill'</span>, stat = <span>'identity'</span>, show.legend = <span>FALSE</span>) +<br>  geom_text(<br>    data = temp_labels,<br>    aes(<br>      x = sample,<br>      y = <span>Inf</span>,<br>      label = paste0(<span>'n = '</span>, format(n, big.mark = <span>','</span>, trim = <span>TRUE</span>)),<br>      vjust = -<span>1</span><br>    ),<br>    color = <span>'black'</span>, size = <span>2.8</span><br>  ) +<br>  scale_fill_manual(name = <span>'Cell type'</span>, values = custom_colors$discrete) +<br>  scale_y_continuous(<br>    name = <span>'Percentage [%]'</span>,<br>    labels = scales::percent_format(),<br>    expand = c(<span>0.01</span>,<span>0</span>)<br>  ) +<br>  coord_cartesian(clip = <span>'off'</span>) +<br>  theme_bw() +<br>  theme(<br>    legend.position = <span>'none'</span>,<br>    plot.title = element_text(hjust = <span>0.5</span>),<br>    text = element_text(size = <span>16</span>),<br>    panel.grid.major = element_blank(),<br>    panel.grid.minor = element_blank(),<br>    axis.title.x = element_blank(),<br>    axis.text.x = element_text(angle = <span>45</span>, hjust = <span>1</span>, vjust = <span>1</span>),<br>    plot.margin = margin(t = <span>20</span>, r = <span>0</span>, b = <span>0</span>, l = <span>0</span>, unit = <span>'pt'</span>)<br>  )<br><br>temp_labels &lt;- seurat@meta.data %&gt;%<br>  group_by(seurat_clusters) %&gt;%<br>  tally() %&gt;%<br>  dplyr::rename(<span>'cluster'</span> = seurat_clusters)<br><br>p2 &lt;- table_clusters_by_cell_type %&gt;%<br>  select(-c(<span>'total_cell_count'</span>)) %&gt;%<br>  reshape2::melt(id.vars = <span>'cluster'</span>) %&gt;%<br>  mutate(cluster = factor(cluster, levels = levels(seurat@meta.data$seurat_clusters))) %&gt;%<br>  ggplot(aes(cluster, value)) +<br>  geom_bar(aes(fill = variable), position = <span>'fill'</span>, stat = <span>'identity'</span>) +<br>  geom_text(<br>    data = temp_labels,<br>    aes(<br>      x = cluster,<br>      y = <span>Inf</span>,<br>      label = paste0(<span>'n = '</span>, format(n, big.mark = <span>','</span>, trim = <span>TRUE</span>)),<br>      vjust = -<span>1</span><br>    ),<br>    color = <span>'black'</span>, size = <span>2.8</span><br>  ) +<br>  scale_fill_manual(name = <span>'Cell type'</span>, values = custom_colors$discrete) +<br>  scale_y_continuous(<br>    name = <span>'Percentage [%]'</span>,<br>    labels = scales::percent_format(),<br>    expand = c(<span>0.01</span>,<span>0</span>)<br>  ) +<br>  coord_cartesian(clip = <span>'off'</span>) +<br>  theme_bw() +<br>  theme(<br>    legend.position = <span>'right'</span>,<br>    plot.title = element_text(hjust = <span>0.5</span>),<br>    text = element_text(size = <span>16</span>),<br>    panel.grid.major = element_blank(),<br>    panel.grid.minor = element_blank(),<br>    axis.title = element_blank(),<br>    plot.margin = margin(t = <span>20</span>, r = <span>0</span>, b = <span>0</span>, l = <span>10</span>, unit = <span>'pt'</span>)<br>  )<br><br>p1 + p2 +<br>  plot_layout(ncol = <span>2</span>, widths = c(<br>    seurat@meta.data$sample %&gt;% unique() %&gt;% length(),<br>    seurat@meta.data$seurat_clusters %&gt;% unique() %&gt;% length()<br>  ))<br><br>ggsave(<br>  <span>'plots/composition_samples_clusters_by_cell_type_singler_blueprintencode_main_by_percent.png'</span>,<br>  p1 + p2 +<br>  plot_layout(ncol = <span>2</span>, widths = c(<br>    seurat@meta.data$sample %&gt;% unique() %&gt;% length(),<br>    seurat@meta.data$seurat_clusters %&gt;% unique() %&gt;% length()<br>  )),<br>  width = <span>18</span>, height = <span>8</span><br>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100042380" data-ratio="0.4444444444444444" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0htuuAC8icL6XosFicuVFJjeuJWoRxeiccT6Lq5KX3K7oLSQcC3wWKzn96wA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0htuuAC8icL6XosFicuVFJjeuJWoRxeiccT6Lq5KX3K7oLSQcC3wWKzn96wA/640?wx_fmt=png&amp;from=appmsg"></figure><h3 data-tool="mdnice编辑器"><span></span><span>冲积图</span><span></span></h3><p data-tool="mdnice编辑器">看细胞类型在样本中的分布</p><pre data-tool="mdnice编辑器"><span></span><code><span>### 11.3.1 Alluvial plots冲积图</span><br><span>## get sample and cluster names</span><br>samples &lt;- levels(seurat@meta.data$sample)<br>clusters &lt;- levels(seurat@meta.data$seurat_clusters)<br><br><span>## create named vector holding the color assignments for both samples and</span><br><span>## clusters</span><br>color_assignments &lt;- setNames(<br>  c(custom_colors$discrete[<span>1</span>:length(samples)], custom_colors$discrete[<span>1</span>:length(clusters)]),<br>  c(samples,clusters)<br>)<br><br><span>## prepare data for the plot; factor() calls are necessary for the right order</span><br><span>## of columns (first samples then clusters) and boxes within each column (</span><br><span>## cluster 1, 2, 3, ..., not 1, 10, 11, ...)</span><br>data &lt;- seurat@meta.data %&gt;%<br>  group_by(sample,seurat_clusters) %&gt;%<br>  tally() %&gt;%<br>  ungroup() %&gt;%<br>  gather_set_data(<span>1</span>:<span>2</span>) %&gt;%<br>  dplyr::mutate(<br>    x = factor(x, levels = unique(x)),<br>    y = factor(y, levels = unique(y))<br>  )<br><br>DataFrame(data)<br><br>data_labels &lt;- tibble(<br>    group = c(<br>      rep(<span>'sample'</span>, length(samples)),<br>      rep(<span>'seurat_clusters'</span>, length(clusters))<br>    )<br> ) %&gt;%<br>  mutate(<br>    hjust = ifelse(group == <span>'sample'</span>, <span>1</span>, <span>0</span>),<br>    nudge_x = ifelse(group == <span>'sample'</span>, -<span>0.1</span>, <span>0.1</span>)<br>  )<br><br>DataFrame(data_labels)<br><br><span>## create plot</span><br>p1 &lt;- ggplot(data, aes(x, id = id, split = y, value = n)) +<br>  geom_parallel_sets(aes(fill = seurat_clusters), alpha = <span>0.75</span>, axis.width = <span>0.15</span>) +<br>  geom_parallel_sets_axes(aes(fill = y), color = <span>'black'</span>, axis.width = <span>0.1</span>) +<br>  geom_text(<br>    aes(y = n, split = y), stat = <span>'parallel_sets_axes'</span>, fontface = <span>'bold'</span>,<br>    hjust = data_labels$hjust, nudge_x = data_labels$nudge_x<br>  ) +<br>  scale_x_discrete(labels = c(<span>'Sample'</span>,<span>'Cluster'</span>)) +<br>  scale_fill_manual(values = color_assignments) +<br>  theme_bw() +<br>  theme(<br>    legend.position = <span>'none'</span>,<br>    axis.title = element_blank(),<br>    axis.text.x = element_text(face = <span>'bold'</span>, colour = <span>'black'</span>, size = <span>15</span>),<br>    axis.text.y = element_blank(),<br>    axis.ticks = element_blank(),<br>    panel.grid.major = element_blank(),<br>    panel.grid.minor = element_blank(),<br>    panel.border = element_blank()<br>  )<br>```<br><br>聚类结果对细胞类型<br>```{r}<br>clusters &lt;- levels(seurat@meta.data$seurat_clusters)<br>cell_types &lt;- sort(unique(seurat@meta.data$cell_type_singler_blueprintencode_main))<br><br>color_assignments &lt;- setNames(<br>  c(custom_colors$discrete[<span>1</span>:length(clusters)], custom_colors$discrete[<span>1</span>:length(cell_types)]),<br>  c(clusters,cell_types)<br>)<br><br>data &lt;- seurat@meta.data %&gt;%<br>  dplyr::rename(cell_type = cell_type_singler_blueprintencode_main) %&gt;%<br>  dplyr::mutate(cell_type = factor(cell_type, levels = cell_types)) %&gt;%<br>  group_by(seurat_clusters, cell_type) %&gt;%<br>  tally() %&gt;%<br>  ungroup() %&gt;%<br>  gather_set_data(<span>1</span>:<span>2</span>) %&gt;%<br>  dplyr::mutate(<br>    x = factor(x, levels = unique(x)),<br>    y = factor(y, levels = c(clusters,cell_types))<br>  )<br><br>data_labels &lt;- tibble(<br>    group = c(<br>      rep(<span>'seurat_clusters'</span>, length(clusters)),<br>      rep(<span>'cell_type'</span>, length(cell_types))<br>    )<br> ) %&gt;%<br>  mutate(<br>    hjust = ifelse(group == <span>'seurat_clusters'</span>, <span>1</span>, <span>0</span>),<br>    nudge_x = ifelse(group == <span>'seurat_clusters'</span>, -<span>0.1</span>, <span>0.1</span>)<br>  )<br><br>p2 &lt;- ggplot(data, aes(x, id = id, split = y, value = n)) +<br>  geom_parallel_sets(aes(fill = seurat_clusters), alpha = <span>0.75</span>, axis.width = <span>0.15</span>) +<br>  geom_parallel_sets_axes(aes(fill = y), color = <span>'black'</span>, axis.width = <span>0.1</span>) +<br>  geom_text(<br>    aes(y = n, split = y), stat = <span>'parallel_sets_axes'</span>, fontface = <span>'bold'</span>,<br>    hjust = data_labels$hjust, nudge_x = data_labels$nudge_x<br>  ) +<br>  scale_x_discrete(labels = c(<span>'Cluster'</span>,<span>'Cell type'</span>)) +<br>  scale_fill_manual(values = color_assignments) +<br>  theme_bw() +<br>  theme(<br>    legend.position = <span>'none'</span>,<br>    axis.title = element_blank(),<br>    axis.text.x = element_text(face = <span>'bold'</span>, colour = <span>'black'</span>, size = <span>15</span>),<br>    axis.text.y = element_blank(),<br>    axis.ticks = element_blank(),<br>    panel.grid.major = element_blank(),<br>    panel.grid.minor = element_blank(),<br>    panel.border = element_blank()<br>  )<br>p1 + p2 + plot_layout(ncol = <span>2</span>)<br><br>ggsave(<br>  <span>'plots/samples_clusters_cell_types_alluvial.png'</span>,<br>  p1 + p2 + plot_layout(ncol = <span>2</span>),<br>  height = <span>6</span>, width = <span>8</span><br>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100042379" data-ratio="0.75" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0htBw4tMdqCs9fzIr7jWlNYm3t4PDyOJ72Pdicl22n7pRn7qh8asOKEA6Q/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0htBw4tMdqCs9fzIr7jWlNYm3t4PDyOJ72Pdicl22n7pRn7qh8asOKEA6Q/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">类似于根据聚类结果做UMAP，我们在UMAP中加入细胞类型标签</p><pre data-tool="mdnice编辑器"><span></span><code><span>## 11.4 UMAP by cell type根据细胞类型画UMAP</span><br>UMAP_centers_cell_type &lt;- tibble(<br>    UMAP_1 = as.data.frame(seurat@reductions$UMAP@cell.embeddings)$UMAP_1,<br>    UMAP_2 = as.data.frame(seurat@reductions$UMAP@cell.embeddings)$UMAP_2,<br>    cell_type = seurat@meta.data$cell_type_singler_blueprintencode_main<br>  ) %&gt;%<br>  group_by(cell_type) %&gt;%<br>  summarize(x = median(UMAP_1), y = median(UMAP_2))<br><br>p &lt;- bind_cols(seurat@meta.data, as.data.frame(seurat@reductions$UMAP@cell.embeddings)) %&gt;%<br>  ggplot(aes(UMAP_1, UMAP_2, color = cell_type_singler_blueprintencode_main)) +<br>  geom_point(size = <span>0.2</span>) +<br>  geom_label(<br>    data = UMAP_centers_cell_type,<br>    mapping = aes(x, y, label = cell_type),<br>    size = <span>3.5</span>,<br>    fill = <span>'white'</span>,<br>    color = <span>'black'</span>,<br>    fontface = <span>'bold'</span>,<br>    alpha = <span>0.5</span>,<br>    label.size = <span>0</span>,<br>    show.legend = <span>FALSE</span><br>  ) +<br>  theme_bw() +<br>  expand_limits(x = c(-<span>22</span>,<span>15</span>)) +<br>  scale_color_manual(values = custom_colors$discrete) +<br>  labs(color = <span>'Cell type'</span>) +<br>  guides(colour = guide_legend(override.aes = list(size = <span>2</span>))) +<br>  theme(legend.position = <span>'right'</span>) +<br>  coord_fixed() +<br>  annotate(<br>    geom = <span>'text'</span>, x = <span>Inf</span>, y = -<span>Inf</span>,<br>    label = paste0(<span>'n = '</span>, format(nrow(seurat@meta.data), big.mark = <span>','</span>, trim = <span>TRUE</span>)),<br>    vjust = -<span>1.5</span>, hjust = <span>1.25</span>, color = <span>'black'</span>, size = <span>2.5</span><br>  )<br><br>p<br><br>ggsave(<br>  <span>'plots/umap_by_cell_type_singler_blueprintencode_main.png'</span>,<br>  p,<br>  height = <span>4</span>,<br>  width = <span>6</span><br>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100042387" data-ratio="0.6666666666666666" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0htFIvPjGUKvVFF7qYVEFwOl7J8CN5qHAvuRa3MoKM11FF4dLibGHHZ5zA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0htFIvPjGUKvVFF7qYVEFwOl7J8CN5qHAvuRa3MoKM11FF4dLibGHHZ5zA/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">为了了解UMAP中的细胞类型是否重叠(这在之前的图中很难看到)，我们并按细胞类型拆分画板绘制了UMAP图。</p><pre data-tool="mdnice编辑器"><span></span><code>temp_labels &lt;- seurat@meta.data %&gt;%<br>  group_by(cell_type_singler_blueprintencode_main) %&gt;%<br>  tally()<br><br>p &lt;- bind_cols(seurat@meta.data, as.data.frame(seurat@reductions$UMAP@cell.embeddings)) %&gt;%<br>  ggplot(aes(UMAP_1, UMAP_2, color = cell_type_singler_blueprintencode_main)) +<br>  geom_point(size = <span>0.2</span>, show.legend = <span>FALSE</span>) +<br>  geom_text(<br>    data = temp_labels,<br>    aes(<br>      x = <span>Inf</span>,<br>      y = -<span>Inf</span>,<br>      label = paste0(<span>'n = '</span>, format(n, big.mark = <span>','</span>, trim = <span>TRUE</span>)),<br>      vjust = -<span>1.5</span>,<br>      hjust = <span>1.25</span><br>    ),<br>    color = <span>'black'</span>, size = <span>2.8</span><br>  ) +<br>  theme_bw() +<br>  scale_color_manual(values = custom_colors$discrete) +<br>  labs(color = <span>'Cell type'</span>) +<br>  guides(colour = guide_legend(override.aes = list(size = <span>2</span>))) +<br>  coord_fixed() +<br>  facet_wrap(~cell_type_singler_blueprintencode_main, ncol = <span>4</span>) +<br>  theme(<br>    legend.position = <span>'right'</span>,<br>    strip.text = element_text(face = <span>'bold'</span>, size = <span>10</span>)<br>  )<br><br>p<br><br>ggsave(<br>  <span>'plots/umap_by_cell_type_singler_blueprintencode_main_split.png'</span>,<br>  p, height = <span>5</span>, width = <span>8</span><br>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100042385" data-ratio="0.625" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0ht6mJNHRqpBIvNmtYS8aibMgibeYCMxVQYhQpaTblGFD8lSboALOdND9HQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0ht6mJNHRqpBIvNmtYS8aibMgibeYCMxVQYhQpaTblGFD8lSboALOdND9HQ/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">我们最后做一次UMAP，根据细胞注释时的分配分数</p><pre data-tool="mdnice编辑器"><span></span><code><span>## 11.5 UMAP by assignment score根据分配分数进行UMAP</span><br>p &lt;- bind_cols(seurat@meta.data, as.data.frame(seurat@reductions$UMAP@cell.embeddings)) %&gt;%<br>  ggplot(aes(UMAP_1, UMAP_2, color = cell_type_singler_blueprintencode_main_score)) +<br>  geom_point(size = <span>0.2</span>) +<br>  theme_bw() +<br>  scale_color_viridis(<br>    name = <span>'Score'</span>, limits = c(<span>0</span>,<span>1</span>), oob = scales::squish,<br>    guide = guide_colorbar(frame.colour = <span>'black'</span>, ticks.colour = <span>'black'</span>)<br>  ) +<br>  labs(color = <span>'Cell type'</span>) +<br>  theme(legend.position = <span>'right'</span>) +<br>  coord_fixed() +<br>  annotate(<br>    geom = <span>'text'</span>, x = <span>Inf</span>, y = -<span>Inf</span>,<br>    label = paste0(<span>'n = '</span>, format(nrow(seurat@meta.data), big.mark = <span>','</span>, trim = <span>TRUE</span>)),<br>    vjust = -<span>1.5</span>, hjust = <span>1.25</span>, color = <span>'black'</span>, size = <span>2.5</span><br>  )<br><br>p<br><br>ggsave(<br>  <span>'plots/umap_by_cell_type_singler_blueprintencode_main_score.png'</span>,<br>  p,<br>  height = <span>5</span>,<br>  width = <span>5.5</span><br>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100042386" data-ratio="0.9092592592592592" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0ht2vAxm4lAdsGGwHH98ia7aEKsAY8qu6iazOgVeubdC3E504uE8Whfcfsg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8lxl2xSzJ6VULEIPiaZu0ht2vAxm4lAdsGGwHH98ia7aEKsAY8qu6iazOgVeubdC3E504uE8Whfcfsg/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">后续的进阶分析可以在seurat对象上进行，这里篇幅有限就不做赘述了，感兴趣的可可以跟着原github的workflow做一做。但是因为他是基于seurat4的workflow作者还没有更新，所以需要进行修改才能跑通。下期会更新后续的marker基因的点图、热图，以及GSVA分析等。</p></section><p><br></p><p><mp-style-type data-value="10000"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/P9h_YTMsSgDPniPZqgn-yg",target="_blank" rel="noopener noreferrer">原文链接</a>
