---
title: "如何突破单细胞数据获取的门槛：从GEO到Cell Ranger"
date: 2024-07-17T04:37:40Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Acdemic"]
---
如何突破单细胞数据获取的门槛：从GEO到Cell Ranger by 生信菜鸟团
------
<div><section><span></span><p>书接上回，一步步尝试代码复现，然后，我们就来到了Figure 2.I，乍看只是平平无奇的堆叠图嘛，殊不知这是多个外部数据集整理后的对比~</p><p><img data-backh="214" data-backw="562" data-imgfileid="100041311" data-ratio="0.37962962962962965" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losib1SRPYAut2ETxJe19eYndAWGlESG1t8CM1ANsvib6dAubmTAWaycNDQWyuhHe7dPibdGteP812iafaA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losib1SRPYAut2ETxJe19eYndAWGlESG1t8CM1ANsvib6dAubmTAWaycNDQWyuhHe7dPibdGteP812iafaA/640?wx_fmt=png&amp;from=appmsg"></p><p>在文章的<code>External dataset mapping</code>部分，作者给出了这几个数据集的来源：</p><blockquote><ol><li><p>Samples from De Jong et al. were downloaded from EBI (E-MTAB-9139), and myeloma samples were excluded (CBM1_1, CBM1_2, CBM2_1, CBM2_2, S9, S10, S11, S12, and S13). Count matrices were generated using Cell Ranger 4.0.0 with default parameters. Samples were integrated and CXCL12+ clusters were exported.</p></li><li><p>Ennis et al. data from GSE227903 and used the SeuratDisk package to convert from the provided aggregated h5ad data to a Seurat object and reprocessed the data.</p></li><li><p>Jardine et al.原文给的是EMBL-EBI的ID，但是，数据格式如下：</p><p><img data-galleryid="" data-imgfileid="100041313" data-ratio="0.13240740740740742" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losib1SRPYAut2ETxJe19eYndAx5JZvgOic9If5JE2FknsyTS0ibuswhgibf4XjwoNFibdelAcRruqjqSqUg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losib1SRPYAut2ETxJe19eYndAx5JZvgOic9If5JE2FknsyTS0ibuswhgibf4XjwoNFibdelAcRruqjqSqUg/640?wx_fmt=png&amp;from=appmsg"><br></p><p>https://github.com/haniffalab/FCA_bone_marrow</p><p>于是在这里找到了：. The following data are also available to download as Scanpy h5ad objects with transformed counts via our interactive webportal: https://fbm.cellatlas.io/: i) DS FBM scRNA-seq, ii) non-DS FBM scRNA-seq, iii) CD34+ FBM, FL and CB CITE-seq, iv) FBM total CITE-seq. 获取到lH5AD 格式的文件，处理起来更有头绪~</p></li><li><p>Data from Li et al. were downloaded from NCBI GEO (GSE190965) and directly read into Seurat and reprocessed as in our dataset described above.</p></li><li><p>For Wang et al. data, we downloaded feature barcode matrices from NCBI GSE147287 and read into Seurat after concatenating the spliced and unspliced matrix files and reprocessed as above.</p></li><li><p>For Triana et al. data, we retrieved the Seurat object directly from Figshare (https:// figshare.com/articles/dataset/Expression_of_97_surface_markers_and_462_mRNAs_in_70017_cells_from_healthy_young_healthy_ old_and_leukemic_human_bone_marrow/13397651).</p></li></ol></blockquote><p>哇噢！六个数据集，又可以get六个经验值，那就赶紧学习起来~</p><p>先从第一个数据集开始，上来就是fastq文件，需要cellranger加工一下，那就开始吧——</p><h2>获取数据</h2><blockquote><p>E-MTAB-9139 &lt; ArrayExpress &lt;https://www.ebi.ac.uk/biostudies/arrayexpress/studies/E-MTAB-9139</p><p><img data-backh="255" data-backw="529" data-galleryid="" data-imgfileid="100041314" data-ratio="0.4824074074074074" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losib1SRPYAut2ETxJe19eYndAqNjjOb1EdENtas4aXraCJ4J8C19GN8ehPmnLpMCvGPIOXjYwF5nFwQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losib1SRPYAut2ETxJe19eYndAqNjjOb1EdENtas4aXraCJ4J8C19GN8ehPmnLpMCvGPIOXjYwF5nFwQ/640?wx_fmt=png&amp;from=appmsg"></p><p><br></p></blockquote><p>这么大的数据，肯定是按需下载，只下载非疾病组的样本即可。我们应该如何对应上样本信息呢？</p><p>Detailed sample information and links to data [view table https://www.ebi.ac.uk/biostudies/arrayexpress/studies/E-MTAB-9139/sdrf ]根据这里来。➡幸运的是作者提供了sh代码，直接下载对应的fastq文件即可~</p><h2>下载cellranger和参考基因组</h2><blockquote><p>[Download Cell Ranger - Official 10x Genomics Support https://www.10xgenomics.com/support/software/cell-ranger/downloads#download-links)</p></blockquote><blockquote><p>[Reference Release Notes - Official 10x Genomics Support https://www.10xgenomics.com/support/software/cell-ranger/latest/release-notes/cr-reference-release-notes#2020-a)</p></blockquote><p>然后根据安装教程一步步来完成后续工作：</p><blockquote><p>[Installing Cell Ranger - Official 10x Genomics Support https://www.10xgenomics.com/support/software/cell-ranger/latest/tutorials/cr-tutorial-in#tutorial)</p></blockquote><p>最重要的是别忘了添加Cellranger到环境路径中：</p><pre><span></span><code> export PATH=/home/data/t140334/Single_cellranger/cellranger-8.0.1:$PATH<br> which cellranger<br></code></pre><h2>cellranger count 重要参数</h2><table><thead><tr><th>参数</th><th width="392.3333333333333">作用</th></tr></thead><tbody><tr><td>--id</td><td width="392.3333333333333">【<span>必需</span>】唯一的运行 ID 字符串（如 sample345）。该名称是任意的，将用于命名包含所有管道生成的文件和输出的目录。只允许使用字母、数字、下划线和连字符（最多 64 个字符）。</td></tr><tr><td>--output-dir</td><td width="372.3333333333333">【非必要】用于存储运行结果的自定义输出目录的路径。如果不使用该参数，输出结果将被导入默认路径：/path/to/ID/outs/。</td></tr><tr><td>--transcriptome</td><td width="372.3333333333333">【<span>必需</span>】包含 10x 兼容转录组参考文献的文件夹路径。</td></tr><tr><td>--fastqs</td><td width="372.3333333333333">fastq_path 文件夹的路径</td></tr><tr><td>--sample</td><td width="372.3333333333333">【<span>必需</span>】提供给 FASTQ 生成软件的样本表中指定的样本名称。</td></tr><tr><td>--create-bam</td><td width="372.3333333333333">【<span>必需</span>】启用或禁用 BAM 文件生成。设置--create-bam=false可减少总计算时间和输出目录的大小（不生成 BAM 文件）。</td></tr><tr><td>--feature-ref</td><td width="372.3333333333333">【<span>必需</span>】特征条形码分析所必需。特征参考 CSV 文件的路径，该文件声明了实验中使用的特征条形码试剂。</td></tr></tbody></table><h2>运行</h2><blockquote><p>结合原文给的代码</p></blockquote><pre><span></span><code>/mnt/isilon/tan_lab/xuj5/software/cellranger4-0/cellranger-4.0.0/cellranger count --fastqs=/mnt/isilon/tan_lab/bandyopads/SB15_normal_huMSC_scRNA_deJong/data/CBM1_1/ --<span>id</span>=CBM1_1 --transcriptome=/mnt/isilon/tan_lab/chenc6/Tools/SingleCellAnnotation/GRCh38<br></code></pre><p>还有官方示例：</p><blockquote><p>[Running Cell Ranger count - Official 10x Genomics Support https://www.10xgenomics.com/support/software/cell-ranger/latest/analysis/running-pipelines/cr-gex-count#cr-count-gex-fb)</p></blockquote><pre><span></span><code>cd /home/jdoe/runs<br>cellranger count --id=sample345 \<br>           --transcriptome=/opt/refdata-gex-GRCh38-2020-A \<br>           --fastqs=/home/jdoe/runs/HAWT7ADXX/outs/fastq_path \<br>           --sample=mysample \<br>           --create-bam=true \<br>           --localcores=8 \<br>           --localmem=64<br></code></pre><p>再根据自己的文件夹路径修改一下：</p><pre><span></span><code>/home/data/t140334/Single_cellranger/cellranger-8.0.1/cellranger <br></code></pre><p>如果是多样本呢</p><blockquote><blockquote><p>多样本分析 [cellranger安装以及使用- https://blog.csdn.net/m0_59974475/article/details/138108362)</p></blockquote><pre><span></span><code>x=("24h_CK" "Ac_NPV_0h" "CK_0h")<br>for sample_id in "${x[@]}"; do<br>  fastq_path=".~/jwy/snrna-seq/$sample_id"  <br>  cellranger count --id $sample_id --fastqs ~/jwy/snrna-seq/$sample_id --transcriptome AcMNPV_genome --create-bam false --nosecondary<br>done<br></code></pre></blockquote><p>灵活变通一下：</p><pre><span></span><code>x=("S10"  "S11"  "S12"  "S13"  "S9")<br> for sample_id in "${x[@]}"; do   fastq_path=/home/data/t140334/DeJong_Preprocessing/$sample_id;   /home/data/t140334/Single_cellranger/cellranger-8.0.1/cellranger count --id=$sample_id --fastqs=$fastq_path --transcriptome=/home/data/t140334/References_CellRanger/refdata-gex-GRCh38-2020-A --create-bam false --nosecondary; done<br></code></pre><p><span>绝对路径更保险</span>！！！</p><p>还有个问题，这里其实应该规定一个output-dir的，这样文件输出会比较规整~</p><h2>输出</h2><p>看看自己的：</p><p><img data-imgfileid="100041315" data-ratio="0.944794952681388" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losib1SRPYAut2ETxJe19eYndAcWXOdaDPjJcjxrrmaVAayWZqadiaI6iaaBwFxpu1Vu0jHrgl9HKb1PKw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="634" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losib1SRPYAut2ETxJe19eYndAcWXOdaDPjJcjxrrmaVAayWZqadiaI6iaaBwFxpu1Vu0jHrgl9HKb1PKw/640?wx_fmt=png&amp;from=appmsg"></p><p>和官方的对比一下：</p><p><img data-imgfileid="100041316" data-ratio="0.6652806652806653" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losib1SRPYAut2ETxJe19eYndAKnGoic3z1O3mY1oj4LuujkeCubVz1QIviaHicicey2sxribv2ovbYpvHia0g/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="962" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losib1SRPYAut2ETxJe19eYndAKnGoic3z1O3mY1oj4LuujkeCubVz1QIviaHicicey2sxribv2ovbYpvHia0g/640?wx_fmt=png&amp;from=appmsg"></p><p>现在我有多个<code>filtered_feature_bc_matrix.h5</code>文件放在不同样本对应的文件夹下，我们的目的是把这些文件取出来放到一个新的文件夹下便于下游分析，该怎么办呢？</p><p>是否能通过sh脚本的方式来达到这样的目的呢？</p><p>首先创建一个jio本</p><pre><span></span><code>nano extract_h5_files.sh<br></code></pre><p>然后在编辑界面输入：</p><pre><span></span><code><span>#!/bin/bash</span><br><br><span># 目标文件夹</span><br>destination=<span>"/all_filtered_files"</span> <span>##这里最好用绝对路径</span><br><br><span># 创建目标文件夹</span><br><span>mkdir</span> -p <span>"<span>$destination</span>"</span><br><br><span># 查找以 S 开头的文件夹</span><br><span>for</span> s_dir <span>in</span> S*/; <span>do</span><br>  <span># 查找 outs 文件夹中的 filtered_feature_bc_matrix.h5 文件</span><br>  h5_file=<span>"<span>${s_dir}</span>outs/filtered_feature_bc_matrix.h5"</span><br>  <span>if</span> [ -f <span>"<span>$h5_file</span>"</span> ]; <span>then</span><br>    <span># 提取 S 文件夹的名称作为前缀</span><br>    prefix=$(<span>basename</span> <span>"<span>$s_dir</span>"</span>)<br>    <span># 复制文件并添加前缀</span><br>    <span>cp</span> <span>"<span>$h5_file</span>"</span> <span>"<span>$destination</span>/<span>${prefix}</span>_filtered_feature_bc_matrix.h5"</span><br>  <span>fi</span><br><span>done</span><br><br><span>echo</span> <span>"所有文件已复制到 <span>$destination</span>"</span><br></code></pre><p>感谢万能的chatgpt~</p><pre><span></span><code><span>chmod</span> +x extract_h5_files.sh<br>./extract_h5_files.sh<br></code></pre><p>看看文件夹的内容是否与预期一致——</p><p><img data-imgfileid="100041317" data-ratio="0.20610687022900764" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losib1SRPYAut2ETxJe19eYndAPoRyrODXGicVbiaibdg6Zpz76VGhS1RgOQRvOr1IuUJZxI9buUpYwGGIw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="786" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losib1SRPYAut2ETxJe19eYndAPoRyrODXGicVbiaibdg6Zpz76VGhS1RgOQRvOr1IuUJZxI9buUpYwGGIw/640?wx_fmt=png&amp;from=appmsg"></p><h2>cellranger到seurat对象</h2><p>这回我们已经拿到了五个样本的h5文件，常规流程走起来——</p><pre><span></span><code><span>if</span>(<span>T</span>){<br>  dir<span>=</span><span>'ExternalDatasets/DeJong/'</span> <br>  samples<span>=</span>list.files( dir )<br>  samples <br>  sceList <span>=</span> lapply(samples,<span>function</span>(pro){ <br>    <span># pro=samples[5] </span><br>    print(pro)  <br>   <br>    sce <span>=</span>CreateSeuratObject(counts <span>=</span>  Read10X_h5(file.path(dir,pro)) ,<br>                            project <span>=</span>  pro  ,<br>                            min.cells <span>=</span> <span>5</span>,<br>                            min.features <span>=</span> <span>300</span>  )<br>    <span>return</span>(sce)<br>  }) <br>  do.call(rbind,lapply(sceList, <span>dim</span>)) <br>  sce.all<span>=</span>merge(x<span>=</span>sceList[[<span>1</span>]],<br>                y<span>=</span>sceList[ <span>-</span><span>1</span> ],<br>                add.cell.ids <span>=</span> str_split(samples,<span>'_'</span>,simplify <span>=</span> <span>T</span>)[,<span>1</span>]  ) <br>}<br><br><span>names</span>(sce.all<span>@</span>assays<span>$</span>RNA<span>@</span>layers)<br>sce.all[[<span>"RNA"</span>]]<span>$</span>counts <br><span># Alternate accessor function with the same result</span><br>LayerData(sce.all, assay <span>=</span> <span>"RNA"</span>, layer <span>=</span> <span>"counts"</span>)<br>sce.all <span>&lt;-</span> JoinLayers(sce.all)<br><span>dim</span>(sce.all[[<span>"RNA"</span>]]<span>$</span>counts )<br><span># 24250 27633</span><br><br>as.data.frame(sce.all<span>@</span>assays<span>$</span>RNA<span>$</span>counts[<span>1</span><span>:</span><span>10</span>, <span>1</span><span>:</span><span>2</span>])<br>head(sce.all<span>@</span>meta.data, <span>10</span>)<br>table(sce.all<span>$</span>orig.ident) <br><br>sce.all<span>$</span>orig.ident<span>=</span>str_split(sce.all<span>$</span>orig.ident,<span>'[_]'</span>,simplify <span>=</span> <span>T</span>)[,<span>1</span>]<br>table(sce.all<span>$</span>orig.ident)<br>sce.all<br></code></pre><p>完成！</p><span></span></section><p><br></p><p><mp-style-type data-value="10000"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/DL6Hv4XPmpsOnTi1KQ9WJA",target="_blank" rel="noopener noreferrer">原文链接</a>
