---
title: "普通的Bulk转录组也能做“拟时序”分析吗？"
date: 2024-07-07T15:56:58Z
draft: ["false"]
tags: [
  "fetched",
  "BioInfo"
]
categories: ["Acdemic"]
---
普通的Bulk转录组也能做“拟时序”分析吗？ by BioInfo
------
<div><section><p><strong>写在前面：Debug不易，码字也不易，如果你采用了本文的代码，请引用一下下面的几篇文章，拜托了</strong><img data-ratio="1" data-src="https://res.wx.qq.com/t/wx_fed/we-emoji/res/v1.3.10/assets/newemoji/smiley_83b.png" data-w="128" src="https://res.wx.qq.com/t/wx_fed/we-emoji/res/v1.3.10/assets/newemoji/smiley_83b.png">。</p><p>那么，正文开始<img data-ratio="1" data-src="https://res.wx.qq.com/t/wx_fed/we-emoji/res/v1.3.10/assets/newemoji/Party.png" data-w="128" src="https://res.wx.qq.com/t/wx_fed/we-emoji/res/v1.3.10/assets/newemoji/Party.png"></p><p>一般来说，<strong>拟时序分析（Pseudotime analysis）</strong>通常出现在单细胞转录组数据中，通过模拟细胞在发育或者分化过程中的时间顺序来揭示不同细胞状态之间的动态变化。在单细胞转录组数据中常用Monocle或Slingshot来进行拟时序分析。</p><p><strong>但是普通的Bulk转录组是否也能进行拟时序分析呢？</strong></p><p>例如在这篇两文章中，作者都是利用Bulk转录组数据做了拟时序分析，对基因表达的先后进行排序，从而使得基因拥有了“时序性”。</p><figure><figcaption><p><img data-galleryid="" data-imgfileid="100000646" data-ratio="0.9923664122137404" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaJWVwpZdHpSrKa1cbfxpBuu2PScIUBLaqskcCEvBtRBeRp72aglVqVfWicxAsEIYI1aeK2O0XYKuf5HEf2pgb6w/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="786" src="https://mmbiz.qpic.cn/mmbiz_png/iaJWVwpZdHpSrKa1cbfxpBuu2PScIUBLaqskcCEvBtRBeRp72aglVqVfWicxAsEIYI1aeK2O0XYKuf5HEf2pgb6w/640?wx_fmt=png&amp;from=appmsg"></p><br></figcaption></figure><figure><figcaption><p><img data-galleryid="" data-imgfileid="100000647" data-ratio="0.8811188811188811" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaJWVwpZdHpSrKa1cbfxpBuu2PScIUBLaiaMMbciaXW9dSlCRpicSnPY76TOSr3RS333KaEXKicnW7ntKeEgx9ABrCw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="572" src="https://mmbiz.qpic.cn/mmbiz_png/iaJWVwpZdHpSrKa1cbfxpBuu2PScIUBLaiaMMbciaXW9dSlCRpicSnPY76TOSr3RS333KaEXKicnW7ntKeEgx9ABrCw/640?wx_fmt=png&amp;from=appmsg"></p><br></figcaption></figure><p>该文章阐述了六倍体小麦胚胎发育过程中的动态染色质调控程序。在Fig5中，作者利用Bulk的RNA-seq和ATAC-seq数据进行了“拟时序分析”为基因赋予了“时序性”，从而搭建了具有“时序性”的转录调控网络。</p><figure><figcaption><p><img data-galleryid="" data-imgfileid="100000648" data-ratio="0.926829268292683" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaJWVwpZdHpSrKa1cbfxpBuu2PScIUBLa7CEUEliaxSl9GoSpIrmHQaQWBFxKjykIySCIFZI8licBicWopkHRxXSaA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="861" src="https://mmbiz.qpic.cn/mmbiz_png/iaJWVwpZdHpSrKa1cbfxpBuu2PScIUBLa7CEUEliaxSl9GoSpIrmHQaQWBFxKjykIySCIFZI8licBicWopkHRxXSaA/640?wx_fmt=png&amp;from=appmsg"></p><br></figcaption></figure><figure><figcaption><p><img data-galleryid="" data-imgfileid="100000649" data-ratio="0.6821106821106822" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaJWVwpZdHpSrKa1cbfxpBuu2PScIUBLaZZribHfnsxcJASibvqSQrczKvRw2YiakIibUMMFnMPjZ94sbbv0ylSlW8w/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="777" src="https://mmbiz.qpic.cn/mmbiz_png/iaJWVwpZdHpSrKa1cbfxpBuu2PScIUBLaZZribHfnsxcJASibvqSQrczKvRw2YiakIibUMMFnMPjZ94sbbv0ylSlW8w/640?wx_fmt=png&amp;from=appmsg"></p></figcaption></figure><p>同样地，在第二篇文章中，作者则通过类似的方式搭建了小麦再生过程中的“时序性”的转录调控网络。</p><p>两篇文章中使用的方法均自之前的一篇文章，利用拟时序分析来对玉米和大麦花序发育过程中的转录组进行比较。</p><figure><figcaption><p><img data-galleryid="" data-imgfileid="100000650" data-ratio="1.18639798488665" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaJWVwpZdHpSrKa1cbfxpBuu2PScIUBLa2Febts4bicFXxBOe9DxBxLoFUNaC1lDGfwD9I9tRGfhr6Tc7yrQcF6Q/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="794" src="https://mmbiz.qpic.cn/mmbiz_png/iaJWVwpZdHpSrKa1cbfxpBuu2PScIUBLa2Febts4bicFXxBOe9DxBxLoFUNaC1lDGfwD9I9tRGfhr6Tc7yrQcF6Q/640?wx_fmt=png&amp;from=appmsg"></p></figcaption></figure><p><strong>但其实，这种利用bulk转录组做的拟时序分析与单细胞转录组中的拟时序分析是完全不同的。</strong></p><p><strong>单细胞转录组中的拟时序分析是对细胞进行分析，根据每个细胞的基因表达谱数据，为每个细胞分配发育时间点的信息。</strong>例如一个样本中同时存在祖先细胞和分化后的细胞，如果将发育最早的祖先细胞的时间设置为0，将发育最晚的分化后的细胞的时间设置为10，那么拟时序分析的作用就是为这个样本中所有细胞都分配一个在0-10之间的发育时间点，越接近0就说明这个细胞与祖先细胞越相似，而越接近于10则说明这个细胞的分化程度越高。</p><p><strong>但是在bulk转录组中，拟时序的对象不再是细胞，而是基因（严格意义上来说，不清楚这种分析是否也能被称为拟时序分析，在本文中暂且将其称为拟时序分析）。</strong>在bulk转录组中，一般只有少数样本，最多可能20-30个样本点，其分辨率远远达不到单细胞转录组中的几千甚至上万个细胞的分辨率（在单细胞转录组中，1个细胞即可被认为是一个样本点）。因此用bulk转录组进行拟时序分析时，首先需要将少数几个样本点变成500或者更多个样本点，这样才能更好地根据基因表达的峰值的时间点来对基因表达的先后顺序进行排序，进而得到“拟时序”的基因。</p><p><strong>具体做法如下：</strong></p><p>假设我们有5个样本点（A, B, C, D, E）的1000个基因表达量的数据，我们希望对这1000个基因进行所谓的“拟时序”分析，以获得这1000个基因表达的先后顺序。</p><p><strong>Step1</strong>：对5个样本点进行PCA分析，并通过PC1和PC2计算样本点之间的欧式距离获得时间线，最终将时间线缩放到0-10之间。</p><pre><code><span>library</span>(splines)<br><span>library</span>(factoextra)<br><span>library</span>(FactoMineR)<br><span>library</span>(scales)<br><span>#读取基因表达的TPM矩阵</span><br>tpm=read.table(<span>"gene.TPM.txt"</span>,header=<span>T</span>,row.names=<span>1</span>)<br><span>#对5个样本进行PCA分析</span><br>pca=PCA(t(tpm),scale.unit=<span>T</span>,ncp=<span>5</span>,graph=<span>F</span>)<br><span>#提取PC1和PC2的结果，计算5个样本点的欧式距离</span><br>res=pca$ind$coord<br>res=res[,<span>1</span>:<span>2</span>]<br>dis=dist(res)<br>dis=as.matrix(dis)<br><span>#利用相邻样本点的欧式距离计算5个样本点原始的时间线</span><br>point1=<span>0</span><br>point2=point1+dis[<span>1</span>,<span>2</span>]<br>point3=point2+dis[<span>2</span>,<span>3</span>]<br>point4=point3+dis[<span>3</span>,<span>4</span>]<br>point5=point4+dis[<span>4</span>,<span>5</span>]<br>raw.timeline=c(point1,point2,point3,point4,point5)<br><span>#通过scales包中的rescale函数将原始的时间线经过线性变换为0-10之间的新的时间线</span><br>new.timeline=rescale(raw.timeline,to=c(<span>0</span>,<span>10</span>))<br></code></pre><p><strong>Step2</strong>：利用modelr中的loess函数，根据新时间线，产生500个时间点的表达值。</p><pre><code><span>library</span>(tidyverse)<br><span>library</span>(modelr)<br><span>#对tpm值进行zscale标准化</span><br>tpm.z=as.data.frame(t(apply(tpm,<span>1</span>,scale)))<br>names(tpm.z)=names(tpm)<br><span>#将5个时间点的zcale之后的表达值变成500个点的表达值</span><br>grid=data.frame(time=seq(<span>0</span>,<span>10</span>,length.out=<span>500</span>))<br>pseudotime.model.fun=<span>function</span>(value){time=new.timeline;data = tibble(value=value,time =time);model=loess(value ~ time,data);predict=grid %&gt;% add_predictions(model);<span>return</span>(predict)}<br>res=apply(tpm.z,<span>1</span>,pseudotime.model.fun)<br>results=res %&gt;% reduce(bind_cols)<br>exp500=as.data.frame(t(results[,seq(<span>0</span>,ncol(results),<span>2</span>)]))<br>names(exp500)=results$time...1<br>row.names(exp500)=row.names(tpm)<br></code></pre><p><strong>Step3</strong>: 利用atan2函数对基因表达进行排序，定义基因表达的先后顺序</p><pre><code><span>library</span>(ComplexHeatmap)<br><span>library</span>(factoextra)<br><span>library</span>(FactoMineR)<br><span>#对基因进行PCA分析，得到每个基因的Dim1和Dim2</span><br>gene.pca=PCA(exp500,scale.unit=<span>T</span>,ncp=<span>5</span>,graph=<span>F</span>)<br>res=gene.pca$ind$coord<br>res=res[,<span>1</span>:<span>2</span>]<br>exp500=cbind(exp500,res)<br><span>#用基因的Dim1和Dim2，进行排列组合，计算四种atan2值</span><br>exp500$atan2.1=atan2(exp500$Dim.1,exp500$Dim.2)<br>exp500$atan2.2=atan2(exp500$Dim.1,-exp500$Dim.2)<br>exp500$atan2.3=atan2(-exp500$Dim.1,exp500$Dim.2)<br>exp500$atan2.4=atan2(-exp500$Dim.1,-exp500$Dim.2)<br><span>#分别根据四种atan2值从低到高进行排序，然后用ComplexHeatmap可视化结果选定排序方式</span><br>order1=arrange(exp500,exp500$atan2.1)<br>order2=arrange(exp500,exp500$atan2.2)<br>order3=arrange(exp500,exp500$atan2.3)<br>order4=arrange(exp500,exp500$atan2.4)<br>p1=Heatmap(order1[,<span>1</span>:<span>500</span>],cluster_rows = <span>F</span>,cluster_columns = <span>F</span>,show_row_names = <span>F</span>,show_column_names = <span>F</span>,column_title = <span>"Order1",heatmap_legend_param = list(title="Order1",legend_height=unit(2,"cm")),col=colorRampPalette(rev(brewer.pal(n = 11, name="RdYlBu")))(100)</span>)<br>p2=Heatmap(order2[,<span>1</span>:<span>500</span>],cluster_rows = <span>F</span>,cluster_columns = <span>F</span>,show_row_names = <span>F</span>,show_column_names = <span>F</span>,column_title = <span>"Order2",heatmap_legend_param = list(title="Order2",legend_height=unit(2,"cm")),col=colorRampPalette(rev(brewer.pal(n = 11, name="RdYlBu")))(100)</span>)<br>p3=Heatmap(order3[,<span>1</span>:<span>500</span>],cluster_rows = <span>F</span>,cluster_columns = <span>F</span>,show_row_names = <span>F</span>,show_column_names = <span>F</span>,column_title = <span>"Order3",heatmap_legend_param = list(title="Order3",legend_height=unit(2,"cm")),col=colorRampPalette(rev(brewer.pal(n = 11, name="RdYlBu")))(100)</span>)<br>p4=Heatmap(order4[,<span>1</span>:<span>500</span>],cluster_rows = <span>F</span>,cluster_columns = <span>F</span>,show_row_names = <span>F</span>,show_column_names = <span>F</span>,column_title = <span>"Order4",heatmap_legend_param = list(title="Order4",legend_height=unit(2,"cm")),col=colorRampPalette(rev(brewer.pal(n = 11, name="RdYlBu")))(100)</span>)<br>p1+p2+p3+p4<br></code></pre><p>从结果中可以看到，Order3是符合预期的，因此保留Order3中的结果。<span></span></p><figure><figcaption><p><img data-galleryid="" data-imgfileid="100000654" data-ratio="0.568870523415978" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaJWVwpZdHpSrKa1cbfxpBuu2PScIUBLaIoSk5xJ1X5CXictXE9wmm1IPthtQgjYJzqr3dFaxHKX3QfmUdjsmlDw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="726" src="https://mmbiz.qpic.cn/mmbiz_png/iaJWVwpZdHpSrKa1cbfxpBuu2PScIUBLaIoSk5xJ1X5CXictXE9wmm1IPthtQgjYJzqr3dFaxHKX3QfmUdjsmlDw/640?wx_fmt=png&amp;from=appmsg"></p><br></figcaption></figure><p><strong>PS：Debug不易，码字也不易，如果你采用了本文的代码，请引用一下上述的几篇文章。</strong></p><p>Zhao L, Yang Y, Chen J, et al. Dynamic chromatin regulatory programs during embryogenesis of hexaploid wheat. <em>Genome Biol</em>. 2023;24(1):7. Published 2023 Jan 13. doi:10.1186/s13059-022-02844-2</p><p>Liu X, Bie XM, Lin X, et al. Uncovering the transcriptional regulatory network involved in boosting wheat regeneration and transformation [published correction appears in Nat Plants. 2024 Jan;10(1):194. doi: 10.1038/s41477-024-01619-w]. <em>Nat Plants</em>. 2023;9(6):908-925. doi:10.1038/s41477-023-01406-z</p><p>Leiboff S, Hake S. Reconstructing the Transcriptional Ontogeny of Maize and Sorghum Supports an Inverse Hourglass Model of Inflorescence Development. <em>Curr Biol</em>. 2019;29(20):3410-3419.e3. doi:10.1016/j.cub.2019.08.044</p></section><p><br></p><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/Y203ep6cpR6DyTjoMKgflA",target="_blank" rel="noopener noreferrer">原文链接</a>
