---
title: "课后补充----单细胞突变分析的高分文章运用"
date: 2024-07-13T23:18:11Z
draft: ["false"]
tags: [
  "fetched",
  "单细胞空间交响乐"
]
categories: ["Acdemic"]
---
课后补充----单细胞突变分析的高分文章运用 by 单细胞空间交响乐
------
<div><h4>作者，Evil Genius</h4><h4>我们的单细胞6课已经讲完了，但是大家一定要保持持续学习的心态，才能不断的进步，固步自封不可取。</h4><h4>今天的参考文章在Spatially restricted drivers and transitional cell populations cooperate with the microenvironment in untreated and chemo-resistant pancreatic cancer(nature)</h4><h4>这篇文章之前分享过，但是侧重于空间转录组的部分，但是我们今天要补充的是单细胞突变信息分析的部分。</h4><h4>其中，关键是<strong><span>Mapping mutations and copy number events distinguished tumor populations from normal and transitional cells, including acinar-to-ductal metaplasia and pancreatic intraepithelial neoplasia.</span></strong></h4><p>肿瘤细胞被分成患者特异性cluster，与每个患者的基因组landscope一致。<span><span>将突变和拷贝数改变映射到单个细胞上</span>，<span>通过比较肿瘤细胞的每个亚群与给定的 KRAS 突变的基因表达谱来测试 KRAS 热点变异的影响</span>。<span>含有KRAS G12V的肿瘤细胞会上调与更具侵袭性或转移性肿瘤相关的几个基因，包括COL1A1、VIM和 MUC5B</span>。</span>分析鉴定了5例具有多个KRAS 热点驱动的病例。<span><span>在同一患者中携带不同 KRAS 驱动突变的两个不同的克隆在空间上分离，具有不同的基因表达谱</span>。</span><br></p><p><img data-imgfileid="100008410" data-ratio="1.2733333333333334" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmnNO2uzyFRjKjvV9LEMKrzT3Z0CUtCmDuvvTohPq8pXiaUyVTiaqIf1F00YFbicG89CFtz79mH2IicSqQ/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="1200" src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmnNO2uzyFRjKjvV9LEMKrzT3Z0CUtCmDuvvTohPq8pXiaUyVTiaqIf1F00YFbicG89CFtz79mH2IicSqQ/640?wx_fmt=other&amp;from=appmsg"></p><p><br></p><h4>这个地方用到了单细胞突变、单细胞空间联合分析，<span>不同的突变细胞群在空间位置和微环境上大有不同。</span></h4><p><img data-imgfileid="100008408" data-ratio="0.26666666666666666" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmnNO2uzyFRjKjvV9LEMKrzTiaYO739QXxCWBe4hw3Ydm8wnTCHOU4gPoLzPviaMjyoFsRNaXqTUxQrQ/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="1200" src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmnNO2uzyFRjKjvV9LEMKrzTiaYO739QXxCWBe4hw3Ydm8wnTCHOU4gPoLzPviaMjyoFsRNaXqTUxQrQ/640?wx_fmt=other&amp;from=appmsg"></p><h5>By mapping both KRAS and CDKN2A mutations, we identified several cells in the ADM_tumor population with either a KRAS mutation (n = 1) or CDKN2A mutation (n = 7), although this was not as widespread as the predicted PanIN populations (KRAS: 23 cells; CDKN2A: 163 cells)</h5><p><img data-imgfileid="100008409" data-ratio="0.3983333333333333" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmnNO2uzyFRjKjvV9LEMKrzT3WXBYZQP0tDGf8ShM4yGxHPtJFlibtV3gm0icKrxE1IbhiaSeRIGsnjJQ/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="1200" src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmnNO2uzyFRjKjvV9LEMKrzT3WXBYZQP0tDGf8ShM4yGxHPtJFlibtV3gm0icKrxE1IbhiaSeRIGsnjJQ/640?wx_fmt=other&amp;from=appmsg"></p><p><br></p><p><img data-imgfileid="100008407" data-ratio="0.5848527349228612" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmnNO2uzyFRjKjvV9LEMKrzTm8j99MCyv6dvCiapenToUibhB6Qf7kJvXLHEiasHERGgJ2AkObLHziad0A/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="713" src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmnNO2uzyFRjKjvV9LEMKrzTm8j99MCyv6dvCiapenToUibhB6Qf7kJvXLHEiasHERGgJ2AkObLHziad0A/640?wx_fmt=other&amp;from=appmsg"></p><h4>单细胞基因突变的检测方法</h4><h5>KRAS hotspot and within-case genotyping</h5><blockquote><p>To verify manually and/or determine the KRAS mutation status at KRAS hotspots G12, G13 and Q61, we used bam-readcount. For each case, we first applied bam-readcount to generate readcounts for each of the nine bases in these loci and then calculated VAF values of all the KRAS hotspots based on reference and alternative base read counts at each position. Additionally, we manually verified every variant present in a sample in a pairwise fashion against other samples within the same case.</p></blockquote><h5><strong><span>scVarScan</span></strong> mutation mapping</h5><blockquote><p>We applied our in-house tool <strong><span>scVarScan</span></strong> that can identify reads supporting the reference and variant alleles covering the variant site in each individual cell by tracing cell and molecular barcode information in an<strong><span> scRNA bam</span></strong> file. For mapping, we used high-confidence somatic mutations from WES data. Additionally, we use cancerhotspots.org to obtain the most common KRAS hotspot mutations at <strong><span>G12, G13 and Q61, and use scVarScan to detect potential minority KRAS mutations in each sample</span></strong>.</p></blockquote><h5>Single-cell mutation and CNV plotting</h5><blockquote><p>For clarity, we assigned each cell, represented by a single dot in a UMAP plot, with only one genetic alteration, in a hierarchical fashion. For clarity and to not overcomplicate plotting due to too many comparison groups, if a mutation and a copy number event are detected in the same cell, the cell is labeled with the mutation. Additionally, when multiple mutations or copy number events are detected in the same cell, we plot them hierarchically as follows: KRAS &gt; CDKN2A &gt; SMAD4 &gt; TP53.</p></blockquote><h4>生活很好，有你更好</h4><p><br></p><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/98YfHX45uB-ScukdLnlzRQ",target="_blank" rel="noopener noreferrer">原文链接</a>
