---
title: "MachineLearning 38. 机器学习之基于最近收缩质心分类法的肿瘤亚型分类器 (pamr)"
date: 2024-06-07T00:49:44Z
draft: ["false"]
tags: [
  "fetched",
  "桓峰基因"
]
categories: ["Acdemic"]
---
MachineLearning 38. 机器学习之基于最近收缩质心分类法的肿瘤亚型分类器 (pamr) by 桓峰基因
------
<div><p><img data-imgfileid="100010151" data-ratio="0.4267326732673267" data-src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMxdjEsWs1Shxy54alETSRmMmPUlia8ajqVjDDSOicd4BTWT2lUAFqz3P660MFWt5tqx8TrH4Yy67GtQ/640?wx_fmt=other&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1&amp;tp=webp" data-w="1010" src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMxdjEsWs1Shxy54alETSRmMmPUlia8ajqVjDDSOicd4BTWT2lUAFqz3P660MFWt5tqx8TrH4Yy67GtQ/640?wx_fmt=other&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1&amp;tp=webp"></p><section><mp-common-profile data-pluginname="mpprofile" data-id="MzU0NDY2NTUwOA==" data-headimg="http://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMzQ2iaDaME4SMFFyIc3WCrVf5loZv2JiahbJiaozT9fNzGPkphu6icmGg8FlIsOicJ6F7OJicH2qA0r3k5g/0?wx_fmt=png" data-nickname="桓峰基因" data-alias="KyohoGene" data-signature="桓峰基因推出全新生信分析文字版和视频版免费教程，欢迎大家来围观，在公众号搜索即可获得相关教程！有测序，生信分析，文章合作，实验室各种耗材试剂等业务欢迎来咨询，关注桓峰基因官网：http://www.kyohogene.com/" data-from="0" data-is_biz_ban="0"></mp-common-profile></section><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><hr data-tool="mdnice编辑器"><h2 data-tool="mdnice编辑器"><span>简     介</span></h2><p data-tool="mdnice编辑器">基于最近收缩质心分类法(nearest shrunken centroids)的基因表达谱预测癌症类别的方法。缩小了原型，从而得到了一个通常比竞争方法更准确的分类器。“最近的收缩质心”方法确定了最能表征每个类别的基因子集。该技术是通用的，可用于许多其他分类问题。为了证明其有效性，表明该方法在寻找用于分类小圆蓝细胞肿瘤和白血病的基因方面非常有效。</p></section><p><img data-imgfileid="100010180" data-ratio="0.4297719087635054" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwHB29ibp0PLTn0j3EAmGOqnt6Q5BeluuFUEDqjCgxodnKiaV7sMxg9L23o5vhMq3Xia7BNA64fqrQLA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="833" src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwHB29ibp0PLTn0j3EAmGOqnt6Q5BeluuFUEDqjCgxodnKiaV7sMxg9L23o5vhMq3Xia7BNA64fqrQLA/640?wx_fmt=png&amp;from=appmsg"></p><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><h2 data-tool="mdnice编辑器"><span></span><span>软件包安装</span></h2><p data-tool="mdnice编辑器">软件包安装：</p><pre data-tool="mdnice编辑器"><span></span><code>install.packages(<span>'pamr'</span>)<br></code></pre><h2 data-tool="mdnice编辑器"><span></span><span>数据读取</span></h2><p data-tool="mdnice编辑器">khan 有2309行和65列。这些是Tibshirani等人在PNAS上发表的关于最近的收缩质心的论文中使用的数据集之一。第一行包含示例标签。前两列是基因id和名称。</p><p data-tool="mdnice编辑器">读取数据并构造输入列表，包括</p><ol data-tool="mdnice编辑器"><li><section><p>x：表达矩阵</p></section></li><li><section><p>y：类型</p></section></li><li><section><p>geneid：基因名</p></section></li><li><section><p>genename：基因名称</p></section></li></ol><pre data-tool="mdnice编辑器"><span></span><code><span>library</span>(pamr)<br>data(khan)<br>dim(khan)<br><span>## [1] 2309   65</span><br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>khan[<span>1</span>:<span>5</span>, <span>1</span>:<span>5</span>]<br><span>##       X                         X1      sample1      sample2      sample3</span><br><span>## 1                                           EWS          EWS          EWS</span><br><span>## 2 GENE1  "\\"catenin (cadherin-a"   0.773343723 -0.078177781 -0.084469157</span><br><span>## 3 GENE2   "farnesyl-diphosphate"   -2.438404816 -2.415753791 -1.649739209</span><br><span>## 4 GENE3  "\\"phosphofructokinase"  -0.482562158  0.412771683 -0.241307522</span><br><span>## 5 GENE4   "cytochrome c-1"         -2.721135441 -2.825145973  -2.87528612</span><br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>colnames(khan)<br><span>##  [1] "X"        "X1"       "sample1"  "sample2"  "sample3"  "sample4" </span><br><span>##  [7] "sample5"  "sample6"  "sample7"  "sample8"  "sample9"  "sample10"</span><br><span>## [13] "sample11" "sample12" "sample13" "sample14" "sample15" "sample16"</span><br><span>## [19] "sample17" "sample18" "sample19" "sample20" "sample21" "sample22"</span><br><span>## [25] "sample23" "sample24" "sample25" "sample26" "sample27" "sample28"</span><br><span>## [31] "sample29" "sample30" "sample31" "sample32" "sample33" "sample34"</span><br><span>## [37] "sample35" "sample36" "sample37" "sample38" "sample39" "sample40"</span><br><span>## [43] "sample41" "sample42" "sample43" "sample44" "sample45" "sample46"</span><br><span>## [49] "sample47" "sample48" "sample49" "sample50" "sample51" "sample52"</span><br><span>## [55] "sample53" "sample54" "sample55" "sample56" "sample57" "sample58"</span><br><span>## [61] "sample59" "sample60" "sample61" "sample62" "sample63"</span><br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>x &lt;- data.matrix(khan[-<span>1</span>, c(-<span>1</span>, -<span>2</span>)])<br>x[<span>1</span>:<span>5</span>, <span>1</span>:<span>5</span>]<br><span>##   sample1 sample2 sample3 sample4 sample5</span><br><span>## 2    1873      57      53    2059    1537</span><br><span>## 3    1251    1350    1140    1385    1261</span><br><span>## 4     314    1758     162    1857    1939</span><br><span>## 5    1324    1428    1468    1250    1666</span><br><span>## 6     776     476     679     772    1307</span><br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>rownames(x) = khan[-<span>1</span>, <span>1</span>]<br>y &lt;- as.vector(t(khan[<span>1</span>, c(-<span>1</span>, -<span>2</span>)]))<br>is.vector(y)<br><span>## [1] TRUE</span><br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>length(y)<br><span>## [1] 63</span><br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>mydata &lt;- list(x = x, y = factor(y), geneid = rownames(x), genenames = paste(khan[-<span>1</span>,<br>    <span>1</span>], khan[-<span>1</span>, <span>2</span>], <span>" "</span>))<br></code></pre><h2 data-tool="mdnice编辑器"><span></span><span>实例操作</span></h2><h3 data-tool="mdnice编辑器"><span></span>参数说明<span></span></h3><p data-tool="mdnice编辑器"><strong>Arguments</strong></p><ul data-tool="mdnice编辑器"><li><section><p>data：	The input data. A list with components: x- an expression genes in the rows, samples in the columns), and y- a vector of the class labels for each sample. Optional components- genenames, a vector of gene names, and geneid- a vector of gene identifiers.</p></section></li><li><section><p>gene.subsetSubset of genes to be used. Can be either a logical vector of length total number of genes, or a list of integers of the row numbers of the genes to be used</p></section></li><li><section><p>sample.subsetSubset of samples to be used. Can be either a logical vector of length total number of samples, or a list of integers of the column numbers of the samples to be used.</p></section></li><li><section><p>thresholdA vector of threshold values for the centroid shrinkage.Default is a set of 30 values chosen by the software</p></section></li><li><section><p>n.thresholdNumber of threshold values desired (default 30)</p></section></li><li><section><p>scale.sdScale each threshold by the wthin class standard deviations? Default: true</p></section></li><li><section><p>threshold.scaleAdditional scaling factors to be applied to the thresholds. Vector of length equal to the number of classes. Default- a vectors of ones.</p></section></li><li><section><p>se.scaleVector of scaling factors for the within class standard errors. Default is sqrt(1/n.class-1/n), where n is the overall sample size and n.class is the sample sizes in each class. This default adjusts for different class sizes.</p></section></li><li><section><p>offset.percentFudge factor added to the denominator of each t-statistic, expressed as a percentile of the gene standard deviation values. This is a small positive quantity to penalize genes with expression values near zero, which can result in very large ratios. This factor is expecially impotant for Affy data. Default is the median of the standard deviations of each gene.</p></section></li><li><section><p>heteroShould a heterogeneity transformation be done? If yes, hetero must be set to one of the class labels (see Details below). Default is no (hetero=NULL)</p></section></li><li><section><p>priorVector of length the number of classes, representing prior probabilities for each of the classes. The prior is used in Bayes rule for making class prediction. Default is NULL, and prior is then taken to be n.class/n, where n is the overall sample size and n.class is the sample sizes in each class.</p></section></li><li><section><p>remove.zerosRemove threshold values yielding zero genes? Default TRUE</p></section></li><li><section><p>sign.contrastDirections of allowed deviations of class-wise average gene expression from the overall average gene expression. Default is “both” (positive or negative). Can also be set to “positive” or “negative”.</p></section></li><li><section><p>ngroup.survivalNumber of groups formed for survival data. Default 2</p></section></li></ul><h3 data-tool="mdnice编辑器"><span></span><span>构建模型</span><span></span></h3><h4 data-tool="mdnice编辑器"><span></span>训练分类器（非生存数据）<span></span></h4><h3 data-tool="mdnice编辑器"><span></span><span>训练分类器（非生存数据）</span><span></span></h3><pre data-tool="mdnice编辑器"><span></span><code><span># train classifier</span><br>mytrain &lt;- pamr.train(mydata, gene.subset = <span>NULL</span>, sample.subset = <span>NULL</span>, threshold = <span>NULL</span>,<br>    n.threshold = <span>30</span>, scale.sd = <span>TRUE</span>, threshold.scale = <span>NULL</span>, se.scale = <span>NULL</span>, offset.percent = <span>50</span>,<br>    hetero = <span>NULL</span>, prior = <span>NULL</span>, remove.zeros = <span>TRUE</span>, sign.contrast = <span>"both"</span>, ngroup.survival = <span>4</span>)<br><span>## 123456789101112131415161718192021222324252627282930</span><br></code></pre><h3 data-tool="mdnice编辑器"><span></span><span>交叉验证</span><span></span></h3><p data-tool="mdnice编辑器">通过交叉验证我们可以根据errors最低，筛选基因threshold，确定基因个数：当errors=0,threshold=1.821,筛选基因数为243个。</p><pre data-tool="mdnice编辑器"><span></span><code>mycv &lt;- pamr.cv(mytrain, mydata)<br><span>## 1234Fold 1 :123456789101112131415161718192021222324252627282930</span><br><span>## Fold 2 :123456789101112131415161718192021222324252627282930</span><br><span>## Fold 3 :123456789101112131415161718192021222324252627282930</span><br><span>## Fold 4 :123456789101112131415161718192021222324252627282930</span><br><span>## Fold 5 :123456789101112131415161718192021222324252627282930</span><br><span>## Fold 6 :123456789101112131415161718192021222324252627282930</span><br><span>## Fold 7 :123456789101112131415161718192021222324252627282930</span><br><span>## Fold 8 :123456789101112131415161718192021222324252627282930</span><br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>mycv<br><span>## Call:</span><br><span>## pamr.cv(fit = mytrain, data = mydata)</span><br><span>##    threshold nonzero errors</span><br><span>## 1  0.000     2308    3     </span><br><span>## 2  0.202     2264    4     </span><br><span>## 3  0.405     2133    4     </span><br><span>## 4  0.607     1834    5     </span><br><span>## 5  0.809     1469    4     </span><br><span>## 6  1.012     1115    5     </span><br><span>## 7  1.214      790    5     </span><br><span>## 8  1.416      537    4     </span><br><span>## 9  1.619      362    3     </span><br><span>## 10 1.821      243    1     </span><br><span>## 11 2.023      159    0     </span><br><span>## 12 2.226      102    2     </span><br><span>## 13 2.428       72    3     </span><br><span>## 14 2.630       56    3     </span><br><span>## 15 2.832       38    5     </span><br><span>## 16 3.035       24    11    </span><br><span>## 17 3.237       17    17    </span><br><span>## 18 3.439       15    22    </span><br><span>## 19 3.642        9    23    </span><br><span>## 20 3.844        5    23    </span><br><span>## 21 4.046        4    23    </span><br><span>## 22 4.249        4    24    </span><br><span>## 23 4.451        4    25    </span><br><span>## 24 4.653        3    25    </span><br><span>## 25 4.856        3    28    </span><br><span>## 26 5.058        3    33    </span><br><span>## 27 5.260        1    34    </span><br><span>## 28 5.463        1    40    </span><br><span>## 29 5.665        1    40    </span><br><span>## 30 5.867        0    40</span><br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>pamr.plotcv(mycv)<br></code></pre><p data-tool="mdnice编辑器"><br></p></section><p><img data-galleryid="" data-imgfileid="100010181" data-ratio="2.023866348448687" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwHB29ibp0PLTn0j3EAmGOqn7H9xxSGIVeeoVaDHFZMEb2tvpu3j2dVpWpQZY9tQ89Ria7HXWHUorKA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="419" src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwHB29ibp0PLTn0j3EAmGOqn7H9xxSGIVeeoVaDHFZMEb2tvpu3j2dVpWpQZY9tQ89Ria7HXWHUorKA/640?wx_fmt=png&amp;from=appmsg"></p><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><p><br></p><pre data-tool="mdnice编辑器"><span></span><code>pamr.plotcvprob(mycv, mydata, threshold = <span>2</span>)<br></code></pre><p data-tool="mdnice编辑器"><br></p></section><p><img data-galleryid="" data-imgfileid="100010182" data-ratio="1.0439952437574316" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwHB29ibp0PLTn0j3EAmGOqneV5DaMRINq7mKbicqHmicwSG6sXkH2HialibHy8BIh80RTQp7HW1ib0iaibgg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="841" src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwHB29ibp0PLTn0j3EAmGOqneV5DaMRINq7mKbicqHmicwSG6sXkH2HialibHy8BIh80RTQp7HW1ib0iaibgg/640?wx_fmt=png&amp;from=appmsg"></p><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><p><br></p><pre data-tool="mdnice编辑器"><span></span><code>pamr.listgenes(mytrain, mydata, threshold = <span>3.5</span>)[<span>1</span>:<span>10</span>, ]<br><span>##       id       BL-score EWS-score NB-score RMS-score</span><br><span>##  [1,] GENE1954 0        0.3933    0        0        </span><br><span>##  [2,] GENE1955 0        0         0        0.2979   </span><br><span>##  [3,] GENE1389 0        0.2687    0        0        </span><br><span>##  [4,] GENE246  0        0.1916    0        0        </span><br><span>##  [5,] GENE842  0        0         -0.0784  0        </span><br><span>##  [6,] GENE545  0        0.061     0        0        </span><br><span>##  [7,] GENE742  0        0         0.0549   0        </span><br><span>##  [8,] GENE951  -0.0507  0         0        0        </span><br><span>##  [9,] GENE84   -0.0416  0         0        0        </span><br><span>## [10,] GENE910  0        0         0        0.0243   </span><br><span>## [11,] GENE1645 0        0.0239    0        0        </span><br><span>## [12,] GENE77   -0.0149  0         0        0        </span><br><span>## [13,] GENE509  0        0         0        0.0089   </span><br><span>## [14,] GENE1772 0        0.0067    0        0</span><br><span>##       id         BL-score  EWS-score NB-score  RMS-score</span><br><span>##  [1,] "GENE1954" "0"       "0.3933"  "0"       "0"      </span><br><span>##  [2,] "GENE1955" "0"       "0"       "0"       "0.2979" </span><br><span>##  [3,] "GENE1389" "0"       "0.2687"  "0"       "0"      </span><br><span>##  [4,] "GENE246"  "0"       "0.1916"  "0"       "0"      </span><br><span>##  [5,] "GENE842"  "0"       "0"       "-0.0784" "0"      </span><br><span>##  [6,] "GENE545"  "0"       "0.061"   "0"       "0"      </span><br><span>##  [7,] "GENE742"  "0"       "0"       "0.0549"  "0"      </span><br><span>##  [8,] "GENE951"  "-0.0507" "0"       "0"       "0"      </span><br><span>##  [9,] "GENE84"   "-0.0416" "0"       "0"       "0"      </span><br><span>## [10,] "GENE910"  "0"       "0"       "0"       "0.0243"</span><br></code></pre><h3 data-tool="mdnice编辑器"><span></span>基因筛选<span></span></h3><p data-tool="mdnice编辑器">绘制筛选结果时发现，数量太多，我们可以减少点基因数据，控制threshold作图：</p><pre data-tool="mdnice编辑器"><span></span><code>pamr.geneplot(mytrain, mydata, threshold = <span>3.5</span>)<br></code></pre><p><img data-galleryid="" data-imgfileid="100010184" data-ratio="0.6694444444444444" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwHB29ibp0PLTn0j3EAmGOqnMicQOyp8AVKXicTJzeDZa7m0O90zibLhic8bugykTcr6TsUmib4rXkfjKicw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwHB29ibp0PLTn0j3EAmGOqnMicQOyp8AVKXicTJzeDZa7m0O90zibLhic8bugykTcr6TsUmib4rXkfjKicw/640?wx_fmt=png&amp;from=appmsg"></p><pre data-tool="mdnice编辑器"><span></span><code>pamr.plotcen(mytrain, mydata, threshold = <span>3.5</span>)<br></code></pre><p data-tool="mdnice编辑器"><img data-galleryid="" data-imgfileid="100010183" data-ratio="0.804642166344294" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwHB29ibp0PLTn0j3EAmGOqnYgOibvo2Mk14J7U8C8icS14gNZwWtJxeuWyMkuryVcbkHcoXM1Oy4XoA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1034" src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwHB29ibp0PLTn0j3EAmGOqnYgOibvo2Mk14J7U8C8icS14gNZwWtJxeuWyMkuryVcbkHcoXM1Oy4XoA/640?wx_fmt=png&amp;from=appmsg"><br></p></section><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><p><br></p><h3 data-tool="mdnice编辑器"><span></span><span>一致性分析</span><span></span></h3><p data-tool="mdnice编辑器">pamr软件包提供交叉验证的函数 pamr.confusion，我们可以看下当threshold=3.5时的准确性：</p><pre data-tool="mdnice编辑器"><span></span><code>pamr.confusion(mytrain, threshold = <span>3.5</span>)<br><span>##     BL EWS NB RMS Class Error rate</span><br><span>## BL   0   4  0   4             1.00</span><br><span>## EWS  0  23  0   0             0.00</span><br><span>## NB   0   1  0  11             1.00</span><br><span>## RMS  0   1  0  19             0.05</span><br><span>## Overall error rate= 0.323</span><br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>pamr.confusion(mycv, threshold = <span>3.5</span>)<br><span>##     BL EWS NB RMS Class Error rate</span><br><span>## BL   0   4  0   4       1.00000000</span><br><span>## EWS  0  22  0   1       0.04347826</span><br><span>## NB   0   1  0  11       1.00000000</span><br><span>## RMS  0   2  0  18       0.10000000</span><br><span>## Overall error rate= 0.353</span><br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>prob &lt;- pamr.predict(mytrain, mydata$x, threshold = <span>1</span>, type = <span>"posterior"</span>)<br>pamr.indeterminate(prob, mingap = <span>0.75</span>)<br><span>##  [1] EWS  EWS  EWS  EWS  EWS  &lt;NA&gt; EWS  EWS  EWS  EWS  EWS  EWS  EWS  EWS  EWS  EWS  EWS  EWS  EWS  EWS  EWS  EWS  EWS  RMS  RMS  RMS  RMS  RMS  RMS  RMS  RMS  RMS  RMS  RMS  RMS  RMS  RMS  RMS  RMS  RMS  RMS  RMS  RMS  NB   NB   NB   NB   NB   NB   NB   NB   NB   NB   NB   NB   BL   BL   BL   BL   BL   BL   BL   BL  </span><br><span>## Levels: BL EWS NB RMS</span><br></code></pre><h3 data-tool="mdnice编辑器"><span></span><span>训练分类器（生存数据）</span><span></span></h3><p data-tool="mdnice编辑器">我们仍然使用khan数据，通过模型生存数据，将添加time 和 status,进行生存分析：</p><pre data-tool="mdnice编辑器"><span></span><code>gendata &lt;- <span>function</span>(n = <span>100</span>, p = <span>2000</span>) {<br>    tim &lt;- <span>3</span> * abs(rnorm(n))<br>    u &lt;- runif(n, min(tim), max(tim))<br>    y &lt;- pmin(tim, u)<br>    ic &lt;- <span>1</span> * (tim &lt; u)<br>    m &lt;- median(tim)<br>    x &lt;- matrix(rnorm(p * n), ncol = n)<br>    x[<span>1</span>:<span>100</span>, tim &gt; m] &lt;- x[<span>1</span>:<span>100</span>, tim &gt; m] + <span>3</span><br>    <span>return</span>(list(x = x, y = y, ic = ic))<br>}<br>junk &lt;- gendata(n = <span>63</span>)<br>mydata$time = junk$y<br>mydata$status = junk$ic<br>a3 &lt;- pamr.train(mydata, ngroup.survival = <span>4</span>)<br><span>## 123456789101112131415161718192021222324252627282930</span><br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>a3<br><span>## Call:</span><br><span>## pamr.train(data = mydata, ngroup.survival = 4)</span><br><span>##    threshold nonzero errors</span><br><span>## 1  0.000     2308    0     </span><br><span>## 2  0.202     2264    0     </span><br><span>## 3  0.405     2133    0     </span><br><span>## 4  0.607     1834    0     </span><br><span>## 5  0.809     1469    0     </span><br><span>## 6  1.012     1115    0     </span><br><span>## 7  1.214      790    0     </span><br><span>## 8  1.416      537    0     </span><br><span>## 9  1.619      362    0     </span><br><span>## 10 1.821      243    0     </span><br><span>## 11 2.023      159    0     </span><br><span>## 12 2.226      102    0     </span><br><span>## 13 2.428       72    0     </span><br><span>## 14 2.630       56    0     </span><br><span>## 15 2.832       38    2     </span><br><span>## 16 3.035       24    6     </span><br><span>## 17 3.237       17    16    </span><br><span>## 18 3.439       15    21    </span><br><span>## 19 3.642        9    21    </span><br><span>## 20 3.844        5    21    </span><br><span>## 21 4.046        4    21    </span><br><span>## 22 4.249        4    21    </span><br><span>## 23 4.451        4    22    </span><br><span>## 24 4.653        3    22    </span><br><span>## 25 4.856        3    25    </span><br><span>## 26 5.058        3    34    </span><br><span>## 27 5.260        1    40    </span><br><span>## 28 5.463        1    40    </span><br><span>## 29 5.665        1    40    </span><br><span>## 30 5.867        0    40</span><br></code></pre><p data-tool="mdnice编辑器">根据计算预测结果，并进行绘制生存曲线：</p><pre data-tool="mdnice编辑器"><span></span><code>yhat &lt;- pamr.predict(a3, mydata$x, threshold = <span>1</span>)<br>proby &lt;- pamr.surv.to.class2(mydata$time, mydata$status, n.class = a3$ngroup.survival)$prob<br>pamr.test.errors.surv.compute(proby, yhat)<br><span>## $confusion</span><br><span>##           1         2        3        4 Class Error rate</span><br><span>## 1 2.0000000  3.338030 2.040816 4.100078        0.8257677</span><br><span>## 2 3.1766702  2.394302 2.210077 3.502929        0.7878140</span><br><span>## 3 0.6362316  4.371070 2.566928 6.466078        0.8171744</span><br><span>## 4 2.1870982 12.896598 5.182178 5.930915        0.7736014</span><br><span>## </span><br><span>## $error</span><br><span>## [1] 0.7568664</span><br></code></pre><pre data-tool="mdnice编辑器"><span></span><code><br>pamr.plotsurvival(yhat, mydata$time, mydata$status)</code></pre><p><img data-galleryid="" data-imgfileid="100010185" data-ratio="0.6698113207547169" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwHB29ibp0PLTn0j3EAmGOqnRsN0mAxrkc8MibmU3IgRmDh6QoP2ofOJibfiaTqibic22kYMuhTJl3P8a8w/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="848" src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwHB29ibp0PLTn0j3EAmGOqnRsN0mAxrkc8MibmU3IgRmDh6QoP2ofOJibfiaTqibic22kYMuhTJl3P8a8w/640?wx_fmt=png&amp;from=appmsg"></p><h3 data-tool="mdnice编辑器"><span></span></h3><h3 data-tool="mdnice编辑器"><span></span><span>绘制ROC曲线</span><span></span></h3><p data-tool="mdnice编辑器">由于我们所作的模型根时间密切相关因此我们选择timeROC,可以快速的技术出来不同时期的ROC，进一步作图：</p><pre data-tool="mdnice编辑器"><span></span><code><span>library</span>(timeROC)<br>res &lt;- timeROC(<span>T</span> = mydata$time, delta = mydata$status, marker = proby[, <span>1</span>], cause = <span>1</span>, weighting = <span>"marginal"</span>, times = c(<span>1</span>, <span>2</span>, <span>3</span>), ROC = <span>TRUE</span>, iid = <span>TRUE</span>)<br>res$AUC<br><span>##       t=1       t=2       t=3 </span><br><span>## 0.8617985 0.6870657 0.6220219</span><br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>confint(res, level = <span>0.95</span>)$CI_AUC<br><span>##      2.5% 97.5%</span><br><span>## t=1 74.65 97.70</span><br><span>## t=2 59.69 77.72</span><br><span>## t=3 55.75 68.65</span><br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>plot(res, time = <span>1</span>, col = <span>"red"</span>, title = <span>FALSE</span>, lwd = <span>2</span>)<br>plot(res, time = <span>2</span>, add = <span>TRUE</span>, col = <span>"blue"</span>, lwd = <span>2</span>)<br>plot(res, time = <span>3</span>, add = <span>TRUE</span>, col = <span>"green"</span>, lwd = <span>2</span>)<br>legend(<span>"bottomright"</span>, c(paste(<span>"1 Years "</span>, round(res$AUC[<span>1</span>], <span>2</span>)), paste(<span>"2 Years "</span>, round(res$AUC[<span>2</span>], <span>2</span>)), paste(<span>"3 Years "</span>, round(res$AUC[<span>3</span>], <span>2</span>))), col = c(<span>"red"</span>, <span>"blue"</span>, <span>"green"</span>), lty = <span>1</span>, lwd = <span>2</span>)<br></code></pre><p data-tool="mdnice编辑器"><br></p></section><p><img data-galleryid="" data-imgfileid="100010186" data-ratio="0.8517915309446255" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwHB29ibp0PLTn0j3EAmGOqnFtPGb0PEfIlV13I85licMnk4895ICg9T5DjicZSJS9YkcAEOJhyhZzTQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="614" src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwHB29ibp0PLTn0j3EAmGOqnFtPGb0PEfIlV13I85licMnk4895ICg9T5DjicZSJS9YkcAEOJhyhZzTQ/640?wx_fmt=png&amp;from=appmsg"></p><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><p><br></p><h3 data-tool="mdnice编辑器"><span></span><span>不同时间节点的AUC曲线及其置信区间</span><span></span></h3><p data-tool="mdnice编辑器">再分析不同时间节点的AUC曲线及其置信区间，由于数据量非常小，此图并不明显。</p><pre data-tool="mdnice编辑器"><span></span><code>plotAUCcurve(res, conf.int = <span>TRUE</span>, col = <span>"red"</span>)<br></code></pre><p data-tool="mdnice编辑器"><br></p></section><p><img data-galleryid="" data-imgfileid="100010187" data-ratio="0.7821138211382114" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwHB29ibp0PLTn0j3EAmGOqnn7fxqoqHjqo5hX3k13gJSK5xRTfXX7ePL5WPuQTM9pMibrrFZmnTlpQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="615" src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwHB29ibp0PLTn0j3EAmGOqnn7fxqoqHjqo5hX3k13gJSK5xRTfXX7ePL5WPuQTM9pMibrrFZmnTlpQ/640?wx_fmt=png&amp;from=appmsg"></p><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><h4 data-tool="mdnice编辑器"><span></span><span>Reference</span><span></span></h4><ol data-tool="mdnice编辑器"><li><section><p>Robert Tibshirani, Trevor Hastie, Balasubramanian Narasimhan, and Gilbert Chu Diagnosis of multiple cancer types by shrunken centroids of gene expression PNAS 99: 6567-6572. Available at www.pnas.org</p></section></li><li><section><p>Khan, J., Wei, J., Ringnér, M. et al. Classification and diagnostic prediction of cancers using gene expression profiling and artificial neural networks. Nat Med 7, 673–679 (2001). https://doi.org/10.1038/89044</p></section></li></ol><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><hr data-tool="mdnice编辑器"><h2 data-tool="mdnice编辑器">基于机器学习构建临床预测模型<br></h2><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247486022&amp;idx=1&amp;sn=90ed16e9cfbfb28a24f0a7629f1c617d&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 1. 主成分分析（PCA）</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247486105&amp;idx=1&amp;sn=9fbb662cbd6d81ed7ebf40aa556e0110&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 2. 因子分析（Factor Analysis）</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247486413&amp;idx=1&amp;sn=12a465fbc1cb526f27e773569e54293b&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 3. 聚类分析（Cluster Analysis）</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247486727&amp;idx=1&amp;sn=2194f775144f7c6cd2857c75748bec33&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 4. 癌症诊断方法之 K-邻近算法（KNN）</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247486754&amp;idx=1&amp;sn=a44bd69f151a9d61131e42bf6c056266&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 5. 癌症诊断和分子分型方法之支持向量机（SVM)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247486804&amp;idx=1&amp;sn=7263d7b1161fdda8cbe558d18e656994&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 6. 癌症诊断机器学习之分类树（Classification Trees)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247486845&amp;idx=1&amp;sn=51610c0c0ef3f942d3db3e9b3994fc2d&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 7. 癌症诊断机器学习之回归树（Regression Trees)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247486875&amp;idx=1&amp;sn=388862bb0bbe06aa4c780ddec4c629d0&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 8. 癌症诊断机器学习之随机森林（Random Forest)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247486922&amp;idx=1&amp;sn=1053cd449e142e28e5895412dac0e105&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 9. 癌症诊断机器学习之梯度提升算法（Gradient Boosting)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247486950&amp;idx=1&amp;sn=6fae0b47ca1b5cb363c402633b557a5a&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 10. 癌症诊断机器学习之神经网络（Neural network)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247488315&amp;idx=1&amp;sn=7dd73671eed123468cb8794280373e9f&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 11. 机器学习之随机森林生存分析（randomForestSRC）</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247488557&amp;idx=1&amp;sn=d4e5ea75e40a28e740c1acae8d3a251f&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 12. 机器学习之降维方法t-SNE及可视化(Rtsne)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247488608&amp;idx=1&amp;sn=683ac11c4d7574d2054f33bf3be9e3cc&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 13. 机器学习之降维方法UMAP及可视化 (umap)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247492372&amp;idx=1&amp;sn=33a5eeb643149f458f34a1cf7038d348&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 14. 机器学习之集成分类器(AdaBoost)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247492403&amp;idx=1&amp;sn=67d7b3c23e645427c8c022f5a783180e&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 15. 机器学习之集成分类器(LogitBoost)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247492432&amp;idx=1&amp;sn=cb98520cef0d7c9fa8d0f6dbdad2da41&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 16. 机器学习之梯度提升机(GBM)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247492489&amp;idx=1&amp;sn=8b3b665f91a53349feda90e187becfd2&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 17. 机器学习之围绕中心点划分算法(PAM)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247492525&amp;idx=1&amp;sn=3f51135459b21e84d2c472f1870ae25a&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 18. 机器学习之贝叶斯分析类器(Naive Bayes)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247492596&amp;idx=1&amp;sn=deba1ae9163056461365d87f09c60f8d&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 19. 机器学习之神经网络分类器(NNET)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247492599&amp;idx=1&amp;sn=3c27c04dc11653e8362abbf265bfb321&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 20. 机器学习之袋装分类回归树(Bagged CART)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247493064&amp;idx=1&amp;sn=435ec2d12715313c6b1cf1d35fd0243b&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 21. 机器学习之临床医学上的生存分析(xgboost)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247493112&amp;idx=1&amp;sn=eefadf45454c642699fcea7336344496&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 22. 机器学习之有监督主成分分析筛选基因(SuperPC)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247493145&amp;idx=1&amp;sn=1de46a9ac9ca43ab170fa58a0ed089f4&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 23. 机器学习之岭回归预测基因型和表型(Ridge)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247493176&amp;idx=1&amp;sn=447dc4c9e5adeb209c54ab89edeaa8ab&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 24. 机器学习之似然增强Cox比例风险模型筛选变量及预后估计(CoxBoost)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247493195&amp;idx=1&amp;sn=2cf43e48459a7d6a72bda8e00636c9dd&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 25. 机器学习之支持向量机应用于生存分析(survivalsvm)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247493226&amp;idx=1&amp;sn=35488e174d710de0674fa65ea700e1ea&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 26. 机器学习之弹性网络算法应用于生存分析(Enet)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247493244&amp;idx=1&amp;sn=5bd32773b9a22cbd905c4205e597f7ee&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 27. 机器学习之逐步Cox回归筛选变量(StepCox)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247493276&amp;idx=1&amp;sn=f1cd0a44c7d0091aba28c39047a8a1a8&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 28. 机器学习之偏最小二乘回归应用于生存分析(plsRcox)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247493321&amp;idx=1&amp;sn=99d80ce61792d02596c6aff25be15d1a&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 29. 机器学习之嵌套交叉验证(Nested CV)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247493372&amp;idx=1&amp;sn=1e1b9c9de5e592a4f0cceb101de51d3c&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 30. 机器学习之特征选择森林之神(Boruta)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247493426&amp;idx=1&amp;sn=2896436a5d1eb3a47e11ab1450cdc225&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 31. 机器学习之基于RNA-seq的基因特征筛选 (GeneSelectR)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247493449&amp;idx=1&amp;sn=8cef8281e01d0a330c72c2b44b7f2c68&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 32. 机器学习之支持向量机递归特征消除的特征筛选 (mSVM-RFE)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247493626&amp;idx=1&amp;sn=f74bdca61523822753c29668bbc878a1&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 33. 机器学习之时间-事件预测与神经网络和Cox回归</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247493648&amp;idx=1&amp;sn=5922654d222c2ea005e653198237bf37&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 34. 机器学习之竞争风险生存分析的深度学习方法(DeepHit)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247493703&amp;idx=1&amp;sn=60da7352205eec98bbb2dc698e53ed20&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 35. 机器学习之Lasso+Cox回归筛选变量 (LassoCox)</span></a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzU0NDY2NTUwOA==&amp;mid=2247493723&amp;idx=1&amp;sn=875a09690162706127a70693a2b321dc&amp;scene=21#wechat_redirect" data-linktype="2"><span>MachineLearning 36. 机器学习之基于神经网络的Cox比例风险模型 (Deepsurv)</span></a></p></section><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><p><span>桓峰基因，铸造成功的您！</span></p><p><span>未来桓峰基因公众号将不间断的推出单细胞系列生信分析教程，</span></p><p><span>敬请期待！！</span></p></section><p><span>桓峰基因官网正式上线，请大家多多关注，还有很多不足之处，大家多多指正！</span><span>http://www.kyohogene.com/</span></p><p><span>桓峰基因和投必得合作，文章润色优惠85折，需要文章润色的老师可以直接到网站输入领取桓峰基因专属优惠券码：KYOHOGENE，然后上传，付款时选择桓峰基因优惠券即可享受85折优惠哦！https://www.topeditsci.com/</span></p><p><span><img data-galleryid="" data-imgfileid="100010188" data-ratio="0.5657407407407408" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwDj3ich1uOSggdibWHw5pgkMx26IaaiaQoKJbU2EaQI2CYh9n2stiaMicbs1RyInFfqO6qWwpDxsFfl8w/640?wx_fmt=other&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1&amp;tp=webp" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/5ia62S5YZaMwDj3ich1uOSggdibWHw5pgkMx26IaaiaQoKJbU2EaQI2CYh9n2stiaMicbs1RyInFfqO6qWwpDxsFfl8w/640?wx_fmt=other&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1&amp;tp=webp"></span></p></section><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/DeUGcRSX63GePhOxYs-PEA",target="_blank" rel="noopener noreferrer">原文链接</a>
