---
title: "单细胞全自动注释篇(一)——clustifyr"
date: 2024-06-04T01:21:06Z
draft: ["false"]
tags: [
  "fetched",
  "生信益站"
]
categories: ["Acdemic"]
---
单细胞全自动注释篇(一)——clustifyr by 生信益站
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><figure data-tool="mdnice编辑器"><img data-imgfileid="100027535" data-ratio="0.278125" data-src="https://mmbiz.qpic.cn/sz_mmbiz_gif/mhoJzVKWSib9fZ15fROicIBV0ia3mRMSrtMPAYzVuuu7kWWdwm11CkFqCNnvH1tZrLRM9sQUZDXUAcv5tgavLnFsQ/640?wx_fmt=gif&amp;from=appmsg" data-type="gif" data-w="640" src="https://mmbiz.qpic.cn/sz_mmbiz_gif/mhoJzVKWSib9fZ15fROicIBV0ia3mRMSrtMPAYzVuuu7kWWdwm11CkFqCNnvH1tZrLRM9sQUZDXUAcv5tgavLnFsQ/640?wx_fmt=gif&amp;from=appmsg"></figure><p data-tool="mdnice编辑器"><code>生信益站，一点就有益</code>！<strong>祝友友们天天开心，月月发 CNS~</strong></p><section><mp-common-profile data-pluginname="mpprofile" data-id="MzU1NTk0MTUxMg==" data-headimg="http://mmbiz.qpic.cn/sz_mmbiz_png/mhoJzVKWSib8n88v0Cj6koLMO8nOuGacicngmJdb0zgmMpxaXYnrB7sjV5iar1cTzTDLsicoyNKrjic1Hgric41HObSQ/0?wx_fmt=png" data-nickname="生信益站" data-alias="" data-signature="生物信息或基因测序数据分析、软件算法、科研绘图、Python/Perl/R代码分享。" data-from="0" data-is_biz_ban="0"></mp-common-profile></section><hr data-tool="mdnice编辑器"><figure data-tool="mdnice编辑器"><img data-imgfileid="100027531" data-ratio="0.3768518518518518" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/mhoJzVKWSib9fZ15fROicIBV0ia3mRMSrtMPhjmC1aXGbxWs2xGKyibgBTa36IbKnibO5ONDUVSdo7vImRF5IWuVdMA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/mhoJzVKWSib9fZ15fROicIBV0ia3mRMSrtMPhjmC1aXGbxWs2xGKyibgBTa36IbKnibO5ONDUVSdo7vImRF5IWuVdMA/640?wx_fmt=png&amp;from=appmsg"><figcaption>设我为🌟星标，每天就能第一时间看到推送啦~</figcaption></figure><h1 data-tool="mdnice编辑器"><span></span><span>1 为什么要使用clustifyr？</span><span></span></h1><p data-tool="mdnice编辑器">如果<strong>不了解相关系统的基础生物学知识，单细胞转录组很难进行注释</strong>。即使具备这些知识，由于<strong>缺乏由批量 RNA 测序、流式细胞术、其他单细胞 RNA 测序平台等定义的常见标记基因</strong>的可检测表达，准确识别仍具有挑战性。</p><p data-tool="mdnice编辑器">clustifyr通过提供使用批量 RNA 测序数据或标记基因列表（排序或未排序）<strong>自动注释</strong>单个细胞或簇的功能解决了这个问题。clustifyr允许对单细胞 RNA 测序数据集和参考数据之间的<strong>计算相似性进行探索性分析</strong>。</p><h1 data-tool="mdnice编辑器"><span></span><span>2 安装</span><span></span></h1><p data-tool="mdnice编辑器">要安装clustifyr，必须安装BiocManager。</p><pre data-tool="mdnice编辑器"><span></span><code>install.packages(<span>"BiocManager"</span>)<br><br>BiocManager::install(<span>"clustifyr"</span>)<br></code></pre><h1 data-tool="mdnice编辑器"><span></span><span>3 一个简单的例子：10x Genomics PBMCs</span><span></span></h1><p data-tool="mdnice编辑器">在此示例中，从外周血单核细胞 (PBMC) 中获取 10x Genomics 3’ scRNA-seq 数据集，并使用从 CITE-seq 实验分配的 scRNA-seq 细胞簇注释细胞簇（使用 Seurat 识别）。</p><pre data-tool="mdnice编辑器"><span></span><code><span>library</span>(clustifyr)<br><span>library</span>(ggplot2)<br><span>library</span>(cowplot)<br><br><span># Matrix of normalized single-cell RNA-seq counts</span><br>pbmc_matrix &lt;- clustifyr::pbmc_matrix_small<br><br><span># meta.data table containing cluster assignments for each cell</span><br><span># The table that we are using also contains the known cell identities in the "classified" column</span><br>pbmc_meta &lt;- clustifyr::pbmc_meta<br></code></pre><h1 data-tool="mdnice编辑器"><span></span><span>4 计算相关系数</span><span></span></h1><p data-tool="mdnice编辑器">为了识别细胞类型，clustifyr()函数需要几个输入：</p><ul data-tool="mdnice编辑器"><li><section>input：SingleCellExperiment 或 Seurat 对象或标准化单细胞 RNA 序列计数矩阵</section></li><li><section>metadata：包含每个细胞的聚类分配的 meta.data 表（如果给出了 Seurat 对象，则不需要）</section></li><li><section>ref_mat：包含每种感兴趣的细胞类型的 RNA-seq 表达数据的参考矩阵</section></li><li><section>query_genes：用于比较的基因列表（可选但推荐）</section></li></ul><p data-tool="mdnice编辑器">当使用 scRNA-seq 计数矩阵时，clustifyr()将返回每个细胞类型和簇的相关系数矩阵，其中行名与簇编号相对应。</p><pre data-tool="mdnice编辑器"><span></span><code><span># Calculate correlation coefficients for each cluster (spearman by default)</span><br>vargenes &lt;- pbmc_vargenes[<span>1</span>:<span>500</span>]<br><br>res &lt;- clustify(<br>  input = pbmc_matrix, <span># matrix of normalized scRNA-seq counts (or SCE/Seurat object)</span><br>  metadata = pbmc_meta, <span># meta.data table containing cell clusters</span><br>  cluster_col = <span>"seurat_clusters"</span>, <span># name of column in meta.data containing cell clusters</span><br>  ref_mat = cbmc_ref, <span># matrix of RNA-seq expression data for each cell type</span><br>  query_genes = vargenes <span># list of highly varible genes identified with Seurat</span><br>)<br><br><span># Peek at correlation matrix</span><br>res[<span>1</span>:<span>5</span>, <span>1</span>:<span>5</span>]<br><span>#&gt;           B CD14+ Mono CD16+ Mono     CD34+     CD4 T</span><br><span>#&gt; 0 0.6563466  0.6454029  0.6485863 0.7089861 0.8804508</span><br><span>#&gt; 1 0.6394363  0.6388404  0.6569401 0.7027430 0.8488750</span><br><span>#&gt; 2 0.5524081  0.9372089  0.8930158 0.5879264 0.5347312</span><br><span>#&gt; 3 0.8945380  0.5801453  0.6146857 0.6955897 0.6566739</span><br><span>#&gt; 4 0.5711643  0.5623870  0.5826233 0.6280913 0.7467347</span><br><br><span># Call cell types</span><br>res2 &lt;- cor_to_call(<br>  cor_mat = res,                  <span># matrix correlation coefficients</span><br>  cluster_col = <span>"seurat_clusters"</span> <span># name of column in meta.data containing cell clusters</span><br>)<br>res2[<span>1</span>:<span>5</span>, ]<br><span>#&gt; # A tibble: 5 × 3</span><br><span>#&gt; # Groups:   seurat_clusters [5]</span><br><span>#&gt;   seurat_clusters type           r</span><br><span>#&gt;   &lt;chr&gt;           &lt;chr&gt;      &lt;dbl&gt;</span><br><span>#&gt; 1 3               B          0.895</span><br><span>#&gt; 2 2               CD14+ Mono 0.937</span><br><span>#&gt; 3 5               CD16+ Mono 0.929</span><br><span>#&gt; 4 0               CD4 T      0.880</span><br><span>#&gt; 5 1               CD4 T      0.849</span><br><br><span># Insert into original metadata as "type" column</span><br>pbmc_meta2 &lt;- call_to_metadata(<br>  res = res2,                     <span># data.frame of called cell type for each cluster</span><br>  metadata = pbmc_meta,           <span># original meta.data table containing cell clusters</span><br>  cluster_col = <span>"seurat_clusters"</span> <span># name of column in meta.data containing cell clusters</span><br>)<br></code></pre><p data-tool="mdnice编辑器">为了可视化clustifyr()结果，我们可以使用该 plot_cor_heatmap()函数绘制每个簇和每种细胞类型的相关系数。</p><pre data-tool="mdnice编辑器"><span></span><code><span># Create heatmap of correlation coefficients using clustifyr() output</span><br>plot_cor_heatmap(cor_mat = res)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100027533" data-ratio="0.7138888888888889" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/mhoJzVKWSib9fZ15fROicIBV0ia3mRMSrtMnibEzMN7INZcDG6wMqFfWhfCb3M6UyU7RY20kgiblZmNWA9LRcnxoXNA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/mhoJzVKWSib9fZ15fROicIBV0ia3mRMSrtMnibEzMN7INZcDG6wMqFfWhfCb3M6UyU7RY20kgiblZmNWA9LRcnxoXNA/640?wx_fmt=png&amp;from=appmsg"></figure><h1 data-tool="mdnice编辑器"><span></span><span>5 绘制聚类身份和相关系数</span><span></span></h1><p data-tool="mdnice编辑器">clustifyr还提供了将相关系数叠加在预先计算的 tSNE 嵌入（或任何其他降维方法的嵌入）上的函数。</p><pre data-tool="mdnice编辑器"><span></span><code><span># Overlay correlation coefficients on UMAPs for the first two cell types</span><br>corr_umaps &lt;- plot_cor(<br>  cor_mat = res,                     <span># matrix of correlation coefficients from clustifyr()</span><br>  metadata = pbmc_meta,              <span># meta.data table containing UMAP or tSNE data</span><br>  data_to_plot = colnames(res)[<span>1</span>:<span>2</span>], <span># name of cell type(s) to plot correlation coefficients</span><br>  cluster_col = <span>"seurat_clusters"</span>    <span># name of column in meta.data containing cell clusters</span><br>)<br><br>plot_grid(<br>  plotlist = corr_umaps,<br>  rel_widths = c(<span>0.47</span>, <span>0.53</span>)<br>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100027532" data-ratio="0.3888888888888889" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/mhoJzVKWSib9fZ15fROicIBV0ia3mRMSrtMB8QHDhuTYuMsgdzx0uF3SjUpjyiaFRmTqSjbkdXPWA3Vz9b3j1remuw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/mhoJzVKWSib9fZ15fROicIBV0ia3mRMSrtMB8QHDhuTYuMsgdzx0uF3SjUpjyiaFRmTqSjbkdXPWA3Vz9b3j1remuw/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">plot_best_call()函数可用于将每个簇标记为具有最高相关系数的细胞类型。使用该pot_dims()函数，我们还可以绘制每个簇的已知身份，这些身份存储在 meta.data 表的“classified”列中。下图显示，参考 RNA-seq 数据和 10x Genomics scRNA-seq 数据集之间的最高相关性仅限于正确的细胞簇。</p><pre data-tool="mdnice编辑器"><span></span><code><span># Label clusters with clustifyr cell identities</span><br>clustifyr_types &lt;- plot_best_call(<br>  cor_mat = res,          <span># matrix of correlation coefficients from clustifyr()</span><br>  metadata = pbmc_meta,   <span># meta.data table containing UMAP or tSNE data</span><br>  do_label = <span>TRUE</span>,        <span># should the feature label be shown on each cluster?</span><br>  do_legend = <span>FALSE</span>,      <span># should the legend be shown?</span><br>  do_repel = <span>FALSE</span>,       <span># use ggrepel to avoid overlapping labels</span><br>  cluster_col = <span>"seurat_clusters"</span><br>) +<br>  ggtitle(<span>"clustifyr cell types"</span>)<br><br><span># Compare clustifyr results with known cell identities</span><br>known_types &lt;- plot_dims(<br>  data = pbmc_meta,       <span># meta.data table containing UMAP or tSNE data</span><br>  feature = <span>"classified"</span>, <span># name of column in meta.data to color clusters by</span><br>  do_label = <span>TRUE</span>,        <span># should the feature label be shown on each cluster?</span><br>  do_legend = <span>FALSE</span>,      <span># should the legend be shown?</span><br>  do_repel = <span>FALSE</span><br>) +<br>  ggtitle(<span>"Known cell types"</span>)<br><br>plot_grid(known_types, clustifyr_types)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100027534" data-ratio="0.4583333333333333" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/mhoJzVKWSib9fZ15fROicIBV0ia3mRMSrtMZDATJcE5S0ueSKCd1DFrGvTYJOLgDLGH1Kg6opbAs0NDL4swvbdj8Q/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/mhoJzVKWSib9fZ15fROicIBV0ia3mRMSrtMZDATJcE5S0ueSKCd1DFrGvTYJOLgDLGH1Kg6opbAs0NDL4swvbdj8Q/640?wx_fmt=png&amp;from=appmsg"></figure><h1 data-tool="mdnice编辑器"><span></span><span>6 使用已知标记基因对细胞进行分类</span><span></span></h1><p data-tool="mdnice编辑器">clustify_lists()函数允许根据已知标记基因分配细胞类型。该函数需要一个包含每种目标细胞类型标记的表格。可以使用多种统计测试来分配细胞类型，包括超几何、Jaccard、Spearman 和 GSEA。</p><pre data-tool="mdnice编辑器"><span></span><code><span># Take a peek at marker gene table</span><br>cbmc_m<br><span>#&gt;   CD4 T CD8 T Memory CD4 T CD14+ Mono Naive CD4 T   NK     B CD16+ Mono</span><br><span>#&gt; 1 ITM2A  CD8B        ADCY2     S100A8       CDHR3 GNLY  IGHM     CDKN1C</span><br><span>#&gt; 2 TXNIP  CD8A       PTGDR2     S100A9  DICER1-AS1 NKG7 CD79A       HES4</span><br><span>#&gt; 3   AES S100B      CD200R1        LYZ       RAD9A CST7 MS4A1        CKB</span><br><span>#&gt;      CD34+ Eryth    Mk    DC   pDCs</span><br><span>#&gt; 1   SPINK2   HBM   PF4  ENHO LILRA4</span><br><span>#&gt; 2  C1QTNF4  AHSP  SDPR  CD1E   TPM2</span><br><span>#&gt; 3 KIAA0125   CA1 TUBB1 NDRG2    SCT</span><br><br><span># Available metrics include: "hyper", "jaccard", "spearman", "gsea"</span><br>list_res &lt;- clustify_lists(<br>  input = pbmc_matrix,             <span># matrix of normalized single-cell RNA-seq counts</span><br>  metadata = pbmc_meta,            <span># meta.data table containing cell clusters</span><br>  cluster_col = <span>"seurat_clusters"</span>, <span># name of column in meta.data containing cell clusters</span><br>  marker = cbmc_m,                 <span># list of known marker genes</span><br>  metric = <span>"pct"</span>                   <span># test to use for assigning cell types</span><br>)<br><br><span># View as heatmap, or plot_best_call</span><br>plot_cor_heatmap(<br>  cor_mat = list_res,              <span># matrix of correlation coefficients from clustify_lists()</span><br>  cluster_rows = <span>FALSE</span>,            <span># cluster by row?</span><br>  cluster_columns = <span>FALSE</span>,         <span># cluster by column?</span><br>  legend_title = <span>"% expressed"</span>     <span># title of heatmap legend</span><br>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100027538" data-ratio="0.6666666666666666" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/mhoJzVKWSib9fZ15fROicIBV0ia3mRMSrtMNqqZkrRwz4x9HkQV6QbE6x0wZyfXfKCgwSmNp2tXrej8o1B6FeZ6yg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/mhoJzVKWSib9fZ15fROicIBV0ia3mRMSrtMNqqZkrRwz4x9HkQV6QbE6x0wZyfXfKCgwSmNp2tXrej8o1B6FeZ6yg/640?wx_fmt=png&amp;from=appmsg"></figure><pre data-tool="mdnice编辑器"><span></span><code><span># Downstream functions same as clustify()</span><br><span># Call cell types</span><br>list_res2 &lt;- cor_to_call(<br>  cor_mat = list_res,              <span># matrix correlation coefficients</span><br>  cluster_col = <span>"seurat_clusters"</span>  <span># name of column in meta.data containing cell clusters</span><br>)<br><br><span># Insert into original metadata as "list_type" column</span><br>pbmc_meta3 &lt;- call_to_metadata(<br>  res = list_res2,                 <span># data.frame of called cell type for each cluster</span><br>  metadata = pbmc_meta,            <span># original meta.data table containing cell clusters</span><br>  cluster_col = <span>"seurat_clusters"</span>, <span># name of column in meta.data containing cell clusters</span><br>  rename_prefix = <span>"list_"</span>          <span># set a prefix for the new column</span><br>)<br></code></pre><h1 data-tool="mdnice编辑器"><span></span><span>7 直接处理SingleCellExperiment对象</span><span></span></h1><p data-tool="mdnice编辑器">clustifyr还可以使用 SingleCellExperiment对象作为输入并返回一个新 SingleCellExperiment对象，其中将细胞类型作为 colData 中的一列添加。</p><pre data-tool="mdnice编辑器"><span></span><code><span>library</span>(SingleCellExperiment)<br>sce &lt;- sce_pbmc()<br>res &lt;- clustify(<br>  input = sce,                <span># an SCE object</span><br>  ref_mat = cbmc_ref,         <span># matrix of RNA-seq expression data for each cell type</span><br>  cluster_col = <span>"clusters"</span>,   <span># name of column in meta.data containing cell clusters</span><br>  obj_out = <span>TRUE</span>              <span># output SCE object with cell type inserted as "type" column</span><br>)<br><br>colData(res)[<span>1</span>:<span>10</span>, c(<span>"type"</span>, <span>"r"</span>)]<br><span>#&gt; DataFrame with 10 rows and 2 columns</span><br><span>#&gt;                       type         r</span><br><span>#&gt;                &lt;character&gt; &lt;numeric&gt;</span><br><span>#&gt; AAACATACAACCAC       CD4 T  0.861083</span><br><span>#&gt; AAACATTGAGCTAC           B  0.909358</span><br><span>#&gt; AAACATTGATCAGC       CD4 T  0.861083</span><br><span>#&gt; AAACCGTGCTTCCG  CD14+ Mono  0.914543</span><br><span>#&gt; AAACCGTGTATGCG          NK  0.894090</span><br><span>#&gt; AAACGCACTGGTAC       CD4 T  0.861083</span><br><span>#&gt; AAACGCTGACCAGT          NK  0.825784</span><br><span>#&gt; AAACGCTGGTTCTT          NK  0.825784</span><br><span>#&gt; AAACGCTGTAGCCA       CD4 T  0.889149</span><br><span>#&gt; AAACGCTGTTTCTG  CD16+ Mono  0.929491</span><br></code></pre><h1 data-tool="mdnice编辑器"><span></span><span>8 直接处理Seurat对象</span><span></span></h1><p data-tool="mdnice编辑器">clustifyr还可以使用Seurat对象作为输入并返回一个新Seurat对象，其中将细胞类型添加为meta数据中的一列。</p><pre data-tool="mdnice编辑器"><span></span><code>so &lt;- so_pbmc()<br>res &lt;- clustify(<br>  input = so,      <span># a Seurat object</span><br>  ref_mat = cbmc_ref,   <span># matrix of RNA-seq expression data for each cell type</span><br>  cluster_col = <span>"seurat_clusters"</span>, <span># name of column in meta.data containing cell clusters</span><br>  obj_out = <span>TRUE</span>  <span># output Seurat object with cell type inserted as "type" column</span><br>)<br><br>res@meta.data[<span>1</span>:<span>10</span>, c(<span>"type"</span>, <span>"r"</span>)]<br><span>#&gt;                      type         r</span><br><span>#&gt; AAACATACAACCAC      CD4 T 0.8424452</span><br><span>#&gt; AAACATTGAGCTAC          B 0.8984684</span><br><span>#&gt; AAACATTGATCAGC      CD4 T 0.8424452</span><br><span>#&gt; AAACCGTGCTTCCG CD14+ Mono 0.9319558</span><br><span>#&gt; AAACCGTGTATGCG         NK 0.8816119</span><br><span>#&gt; AAACGCACTGGTAC      CD4 T 0.8424452</span><br><span>#&gt; AAACGCTGACCAGT         NK 0.8147040</span><br><span>#&gt; AAACGCTGGTTCTT         NK 0.8147040</span><br><span>#&gt; AAACGCTGTAGCCA      CD4 T 0.8736163</span><br><span>#&gt; AAACGCTGTTTCTG CD16+ Mono 0.9321784</span><br></code></pre><h1 data-tool="mdnice编辑器"><span></span><span>9 从单细胞表达矩阵构建参考矩阵</span><span></span></h1><p data-tool="mdnice编辑器">最简单的形式是，通过按簇对单细胞 RNA 序列表达矩阵的表达进行平均（还包括取中位数的选项）来构建参考矩阵。支持对数变换矩阵或原始计数矩阵。</p><pre data-tool="mdnice编辑器"><span></span><code>new_ref_matrix &lt;- average_clusters(<br>  mat = pbmc_matrix,<br>  metadata = pbmc_meta<span>$classified</span>, <span># or use metadata = pbmc_meta, cluster_col = "classified"</span><br>  if_log = TRUE                    <span># whether the expression matrix is already log transformed</span><br>)<br><br>head(new_ref_matrix)<br><span>#&gt;                 B CD14+ Mono      CD8 T         DC FCGR3A+ Mono</span><br><span>#&gt; PPBP   0.09375021 0.28763857 0.35662599 0.06527347    0.2442300</span><br><span>#&gt; LYZ    1.42699419 5.21550849 1.35146753 4.84714962    3.4034309</span><br><span>#&gt; S100A9 0.62123058 4.91453355 0.58823794 2.53310734    2.6277996</span><br><span>#&gt; IGLL5  2.44576997 0.02434753 0.03284986 0.10986617    0.2581198</span><br><span>#&gt; GNLY   0.37877736 0.53592906 2.53161887 0.46959958    0.2903092</span><br><span>#&gt; FTL    3.66698837 5.86217774 3.37056910 4.21848878    5.9518479</span><br><span>#&gt;        Memory CD4 T Naive CD4 T         NK  Platelet</span><br><span>#&gt; PPBP     0.06494743  0.04883837 0.00000000 6.0941782</span><br><span>#&gt; LYZ      1.39466552  1.40165143 1.32701580 2.5303912</span><br><span>#&gt; S100A9   0.58080250  0.55679700 0.52098541 1.6775692</span><br><span>#&gt; IGLL5    0.04826212  0.03116080 0.05247669 0.2501642</span><br><span>#&gt; GNLY     0.41001072  0.46041901 4.70481754 0.3845813</span><br><span>#&gt; FTL      3.31062958  3.35611600 3.38471536 4.5508242</span><br><br><span># For further convenience, a shortcut function for generating reference matrix from `SingleCellExperiment` or `seurat` object is used.</span><br>new_ref_matrix_sce &lt;- object_ref(<br>  input = sce,                     <span># SCE object</span><br>  cluster_col = <span>"clusters"</span>       <span># name of column in colData containing cell identities</span><br>)<br><br>new_ref_matrix_so &lt;- seurat_ref(<br>  seurat_object = so,        <span># Seurat object</span><br>  cluster_col = <span>"seurat_clusters"</span>    <span># name of column in meta.data containing cell identities</span><br>)<br><br>tail(new_ref_matrix_so)<br><span>#&gt;                  0          1           2          3          4</span><br><span>#&gt; RHOC   0.245754269 0.40431050 0.590053057 0.37702525 0.86466156</span><br><span>#&gt; CISH   0.492444272 0.54773003 0.079843557 0.08962348 0.66024943</span><br><span>#&gt; CD27   1.195370020 1.28719850 0.100562312 0.54487892 1.28322681</span><br><span>#&gt; LILRA3 0.004576215 0.03686387 0.544743180 0.00000000 0.03409087</span><br><span>#&gt; CLIC2  0.007570624 0.00000000 0.021200958 0.00000000 0.00000000</span><br><span>#&gt; HEMGN  0.034099324 0.04619359 0.006467157 0.00000000 0.00000000</span><br><span>#&gt;                 5         6          7        8</span><br><span>#&gt; RHOC   2.20162898 1.6518448 0.72661911 1.465067</span><br><span>#&gt; CISH   0.10588034 0.1339815 0.18231747 0.000000</span><br><span>#&gt; CD27   0.09640885 0.1600911 0.04639912 0.000000</span><br><span>#&gt; LILRA3 1.35418074 0.0000000 0.00000000 0.000000</span><br><span>#&gt; CLIC2  0.00000000 0.0000000 0.46542832 0.000000</span><br><span>#&gt; HEMGN  0.00000000 0.0000000 0.00000000 1.079083</span><br></code></pre><p data-tool="mdnice编辑器">更多教程请访问：https://rnabioco.github.io/clustifyrdata/articles/otherformats.html</p><h2 data-tool="mdnice编辑器"><span></span><span>10 sessionInfo()</span><span></span></h2><pre data-tool="mdnice编辑器"><span></span><code><span>#&gt; R version 4.3.3 (2024-02-29)</span><br><span>#&gt; Platform: x86_64-pc-linux-gnu (64-bit)</span><br><span>#&gt; Running under: Ubuntu 22.04.4 LTS</span><br><span>#&gt; </span><br><span>#&gt; Matrix products: default</span><br><span>#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span><br><span>#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span><br><span>#&gt; </span><br><span>#&gt; locale:</span><br><span>#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span><br><span>#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span><br><span>#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span><br><span>#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span><br><span>#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span><br><span>#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span><br><span>#&gt; </span><br><span>#&gt; time zone: UTC</span><br><span>#&gt; tzcode source: system (glibc)</span><br><span>#&gt; </span><br><span>#&gt; attached base packages:</span><br><span>#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span><br><span>#&gt; [8] base     </span><br><span>#&gt; </span><br><span>#&gt; other attached packages:</span><br><span>#&gt;  [1] SingleCellExperiment_1.24.0 SummarizedExperiment_1.32.0</span><br><span>#&gt;  [3] Biobase_2.62.0              GenomicRanges_1.54.1       </span><br><span>#&gt;  [5] GenomeInfoDb_1.38.8         IRanges_2.36.0             </span><br><span>#&gt;  [7] S4Vectors_0.40.2            BiocGenerics_0.48.1        </span><br><span>#&gt;  [9] MatrixGenerics_1.14.0       matrixStats_1.3.0          </span><br><span>#&gt; [11] cowplot_1.1.3               ggplot2_3.5.0              </span><br><span>#&gt; [13] clustifyr_1.15.4            BiocStyle_2.30.0           </span><br><span>#&gt; </span><br><span>#&gt; loaded via a namespace (and not attached):</span><br><span>#&gt;  [1] bitops_1.0-7            rlang_1.1.3            </span><br><span>#&gt;  [3] magrittr_2.0.3          clue_0.3-65            </span><br><span>#&gt;  [5] GetoptLong_1.0.5        compiler_4.3.3         </span><br><span>#&gt;  [7] png_0.1-8               systemfonts_1.0.6      </span><br><span>#&gt;  [9] vctrs_0.6.5             shape_1.4.6.1          </span><br><span>#&gt; [11] pkgconfig_2.0.3         crayon_1.5.2           </span><br><span>#&gt; [13] fastmap_1.1.1           XVector_0.42.0         </span><br><span>#&gt; [15] labeling_0.4.3          utf8_1.2.4             </span><br><span>#&gt; [17] rmarkdown_2.26          ragg_1.3.0             </span><br><span>#&gt; [19] purrr_1.0.2             xfun_0.43              </span><br><span>#&gt; [21] zlibbioc_1.48.2         cachem_1.0.8           </span><br><span>#&gt; [23] jsonlite_1.8.8          highr_0.10             </span><br><span>#&gt; [25] DelayedArray_0.28.0     BiocParallel_1.36.0    </span><br><span>#&gt; [27] cluster_2.1.6           parallel_4.3.3         </span><br><span>#&gt; [29] R6_2.5.1                bslib_0.7.0            </span><br><span>#&gt; [31] RColorBrewer_1.1-3      parallelly_1.37.1      </span><br><span>#&gt; [33] jquerylib_0.1.4         Rcpp_1.0.12            </span><br><span>#&gt; [35] bookdown_0.39           iterators_1.0.14       </span><br><span>#&gt; [37] knitr_1.46              future.apply_1.11.2    </span><br><span>#&gt; [39] Matrix_1.6-5            tidyselect_1.2.1       </span><br><span>#&gt; [41] abind_1.4-5             yaml_2.3.8             </span><br><span>#&gt; [43] doParallel_1.0.17       codetools_0.2-20       </span><br><span>#&gt; [45] listenv_0.9.1           lattice_0.22-6         </span><br><span>#&gt; [47] tibble_3.2.1            withr_3.0.0            </span><br><span>#&gt; [49] evaluate_0.23           future_1.33.2          </span><br><span>#&gt; [51] desc_1.4.3              circlize_0.4.16        </span><br><span>#&gt; [53] pillar_1.9.0            BiocManager_1.30.22    </span><br><span>#&gt; [55] foreach_1.5.2           generics_0.1.3         </span><br><span>#&gt; [57] sp_2.1-3                RCurl_1.98-1.14        </span><br><span>#&gt; [59] munsell_0.5.1           scales_1.3.0           </span><br><span>#&gt; [61] globals_0.16.3          glue_1.7.0             </span><br><span>#&gt; [63] tools_4.3.3             data.table_1.15.4      </span><br><span>#&gt; [65] fgsea_1.28.0            fs_1.6.3               </span><br><span>#&gt; [67] dotCall64_1.1-1         fastmatch_1.1-4        </span><br><span>#&gt; [69] grid_4.3.3              tidyr_1.3.1            </span><br><span>#&gt; [71] colorspace_2.1-0        GenomeInfoDbData_1.2.11</span><br><span>#&gt; [73] cli_3.6.2               textshaping_0.3.7      </span><br><span>#&gt; [75] spam_2.10-0             fansi_1.0.6            </span><br><span>#&gt; [77] S4Arrays_1.2.1          ComplexHeatmap_2.18.0  </span><br><span>#&gt; [79] dplyr_1.1.4             gtable_0.3.4           </span><br><span>#&gt; [81] sass_0.4.9              digest_0.6.35          </span><br><span>#&gt; [83] progressr_0.14.0        SparseArray_1.2.4      </span><br><span>#&gt; [85] farver_2.1.1            rjson_0.2.21           </span><br><span>#&gt; [87] htmlwidgets_1.6.4       SeuratObject_5.0.1     </span><br><span>#&gt; [89] memoise_2.0.1           entropy_1.3.1          </span><br><span>#&gt; [91] htmltools_0.5.8.1       pkgdown_2.0.9          </span><br><span>#&gt; [93] lifecycle_1.0.4         httr_1.4.7             </span><br><span>#&gt; [95] GlobalOptions_0.1.2</span><br></code></pre><h1 data-tool="mdnice编辑器"><span></span><span>益站可以评论、留言啦。赶紧发表您的看法吧~</span><span></span></h1><hr data-tool="mdnice编辑器"><p data-tool="mdnice编辑器">OK，今天的分享到此为止。咱们明天见~</p><h2 data-tool="mdnice编辑器"><span></span><span>联系站长</span><span></span></h2><blockquote data-tool="mdnice编辑器"><span>❝</span><p>对<code>本篇文章有疑问</code>，或者有<code>科研服务需求</code>的友友可以在益站<code>发消息留言</code>，也欢迎各位童鞋<code>扫下面的二维码</code>加入我们的 <code>QQ</code> 交流群。</p></blockquote><figure data-tool="mdnice编辑器"><img data-imgfileid="100027537" data-ratio="1.5" data-src="https://mmbiz.qpic.cn/sz_mmbiz_jpg/mhoJzVKWSib9fZ15fROicIBV0ia3mRMSrtMcv8NBgTO5Td5Ig7VBic4AER8zu3xj9eRu9q77YTGxJjeYLkibX8ibDUEQ/640?wx_fmt=jpeg&amp;from=appmsg" data-type="jpeg" data-w="512" src="https://mmbiz.qpic.cn/sz_mmbiz_jpg/mhoJzVKWSib9fZ15fROicIBV0ia3mRMSrtMcv8NBgTO5Td5Ig7VBic4AER8zu3xj9eRu9q77YTGxJjeYLkibX8ibDUEQ/640?wx_fmt=jpeg&amp;from=appmsg"><figcaption>最新QQ群</figcaption></figure><hr data-tool="mdnice编辑器"><figure data-tool="mdnice编辑器"><img data-imgfileid="100027539" data-ratio="0.7277777777777777" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/mhoJzVKWSib9fZ15fROicIBV0ia3mRMSrtMibyPRoiblwuuV0Dmuvp3RE99DzdwvYoial8AdDJJSiaeRtiaQNzKG8jm2kA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/mhoJzVKWSib9fZ15fROicIBV0ia3mRMSrtMibyPRoiblwuuV0Dmuvp3RE99DzdwvYoial8AdDJJSiaeRtiaQNzKG8jm2kA/640?wx_fmt=png&amp;from=appmsg"><figcaption>科研服务</figcaption></figure><hr data-tool="mdnice编辑器"><figure data-tool="mdnice编辑器"><img data-imgfileid="100027541" data-ratio="1" data-src="https://mmbiz.qpic.cn/sz_mmbiz_gif/mhoJzVKWSib9fZ15fROicIBV0ia3mRMSrtMMHaNbyj6Rd6XrcKfuVcnVYicVfqooCbnUDfQfYsNYq8qrQBBVbAgtoQ/640?wx_fmt=gif&amp;from=appmsg" data-type="gif" data-w="240" src="https://mmbiz.qpic.cn/sz_mmbiz_gif/mhoJzVKWSib9fZ15fROicIBV0ia3mRMSrtMMHaNbyj6Rd6XrcKfuVcnVYicVfqooCbnUDfQfYsNYq8qrQBBVbAgtoQ/640?wx_fmt=gif&amp;from=appmsg"><figcaption>您的<code>关注、点赞、在看、转发</code>是对益站最大的<code>鼓励和支持</code>哈</figcaption></figure><hr data-tool="mdnice编辑器"></section><section><mp-common-profile data-pluginname="mpprofile" data-id="MzU1NTk0MTUxMg==" data-headimg="http://mmbiz.qpic.cn/sz_mmbiz_png/mhoJzVKWSib8n88v0Cj6koLMO8nOuGacicngmJdb0zgmMpxaXYnrB7sjV5iar1cTzTDLsicoyNKrjic1Hgric41HObSQ/0?wx_fmt=png" data-nickname="生信益站" data-alias="" data-signature="生物信息或基因测序数据分析、软件算法、科研绘图、Python/Perl/R代码分享。" data-from="0" data-is_biz_ban="0"></mp-common-profile></section><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/GQiRQIbr1rx5xKYzC1My3Q",target="_blank" rel="noopener noreferrer">原文链接</a>
