---
title: "国家基因组科学数据中心 GSA数据下载、Linux上游分析实战（二）"
date: 2024-06-28T01:54:49Z
draft: ["false"]
tags: [
  "fetched",
  "生信小博士"
]
categories: ["Acdemic"]
---
国家基因组科学数据中心 GSA数据下载、Linux上游分析实战（二） by 生信小博士
------
<div><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=Mzg2NDcxMzYwNg==&amp;mid=2247487272&amp;idx=1&amp;sn=f25af6531697331162873df3b82d81b9&amp;chksm=ce6461c1f913e8d7e5f78bba590deac6e62c95a12f13f0c5cae4b47996c17f0cbc5d7ae01248&amp;scene=21#wechat_redirect" textvalue="上次从GSA下载的结果如下" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2"><strong>上次从GSA下载的结果如下</strong></a>:<strong></strong></p><p><img data-galleryid="" data-imgfileid="100003557" data-ratio="0.5634615384615385" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SkulMWobjj7YR8PF1juAlIfopk57I2BBicGd73BRCnt3dskcGaianibsvC6xGzzmJovEBEm0vS0tXmKzg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1040" src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SkulMWobjj7YR8PF1juAlIfopk57I2BBicGd73BRCnt3dskcGaianibsvC6xGzzmJovEBEm0vS0tXmKzg/640?wx_fmt=png&amp;from=appmsg"></p><p><br></p><p><strong>1.初步质控看看结果如何</strong></p><section><ul><li><li><li><li></ul><pre data-lang="nginx"><code><span><span>conda</span> activate rna2</span></code><code><span>mkdir -p fastqc_output</span></code><code><span>nohup fastqc -t <span>6</span> -o fastqc_output CRR*/<span>*.fq.gz</span> &amp;</span></code><code><span><br></span></code></pre></section><p><img data-galleryid="" data-imgfileid="100003561" data-ratio="1.4426605504587156" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345Sks7osyILdoYFR497XU7DNDsTYZfgcyl3VspfichERFZQY60vdUo2Aic7b7bTGGA6lSNEbPamrV8nerQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="436" src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345Sks7osyILdoYFR497XU7DNDsTYZfgcyl3VspfichERFZQY60vdUo2Aic7b7bTGGA6lSNEbPamrV8nerQ/640?wx_fmt=png&amp;from=appmsg"></p><p><br></p><p><span><strong>2.使用multiqc合并质控结果</strong></span></p><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="sql"><code><span>multiqc ./</span></code><code><span><br></span></code><code><span>multiqc的主要参数</span></code><code><span><span>--force            -f  Overwrite any existing reports                                                 │</span></span></code><code><span>│ <span>--config           -c  Specific config file to load, after those in MultiQC dir / home dir / working  │</span></span></code><code><span>│                        dir.                                                                           │</span></code><code><span>│                        (PATH)                                                                         │</span></code><code><span>│ <span>--cl-config            Specify MultiQC config YAML on the command line (TEXT)                         │</span></span></code><code><span>│ <span>--filename         -n  Report filename. Use 'stdout' to print to standard out. (TEXT)                 │</span></span></code><code><span>│ <span>--outdir           -o  Create report in the specified output directory. (TEXT)                        │</span></span></code><code><span>│ <span>--ignore           -x  Ignore analysis files (GLOB EXPRESSION)                                        │</span></span></code><code><span>│ <span>--ignore-samples       Ignore sample names (GLOB EXPRESSION)                                          │</span></span></code><code><span>│ <span>--ignore-symlinks      Ignore symlinked directories and files                                         │</span></span></code><code><span>│ <span>--file-list        -l  Supply a file containing a list of file paths to be searched, one per row</span></span></code></pre></section><section>结果如下：</section><p><img data-galleryid="" data-imgfileid="100003562" data-ratio="0.42314814814814816" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345Sks7osyILdoYFR497XU7DNDsYIYXr3qO1ficazic6Bvuu5aRVEvr8B6qvmmdMKyuDgFeBZraEPRB5Hdw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345Sks7osyILdoYFR497XU7DNDsYIYXr3qO1ficazic6Bvuu5aRVEvr8B6qvmmdMKyuDgFeBZraEPRB5Hdw/640?wx_fmt=png&amp;from=appmsg"></p><section><strong>使用浏览器打开看看</strong></section><section><strong>从质控图上看，测序质量挺好，接头含量也没超标<br></strong></section><p><img data-galleryid="" data-imgfileid="100003564" data-ratio="0.4888888888888889" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345Sks7osyILdoYFR497XU7DNDsyLa0xeg8da0VjFQmffj7tW8stdDZqvEQTuE60C5hX41pibWoacpKoKw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345Sks7osyILdoYFR497XU7DNDsyLa0xeg8da0VjFQmffj7tW8stdDZqvEQTuE60C5hX41pibWoacpKoKw/640?wx_fmt=png&amp;from=appmsg"></p><p><img data-galleryid="" data-imgfileid="100003563" data-ratio="0.4787037037037037" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345Sks7osyILdoYFR497XU7DNDs4Pwhroa2Fn9kY9VnKxTpNtrGqSia8TzSw7hFNpV0ibcJMnuAL3VaeyRg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345Sks7osyILdoYFR497XU7DNDs4Pwhroa2Fn9kY9VnKxTpNtrGqSia8TzSw7hFNpV0ibcJMnuAL3VaeyRg/640?wx_fmt=png&amp;from=appmsg"></p><section><br></section><section><strong>3.使用trimgalory进行质量控制(对于本数据可以忽略这一步)</strong></section><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="nginx"><code><span><span>ls</span> | grep <span>'^CRR'</span> &gt;sample.id</span></code><code><span> <span>#批量获取样本名称</span></span></code><code><span>find ../CRA005925/ -mindepth <span>1</span> -type d|cut -d<span>"/"</span> -f3 &gt;sample.id</span></code><code><span> </span></code><code><span><br></span></code><code><span>conda activate rna2</span></code><code><span><br></span></code><code><span><span>#!/bin/bash</span></span></code><code><span>cat sample.id | while read id</span></code><code><span>do</span></code><code><span><br></span></code><code><span>mkdir ~/silicosis_wangchen/data/02_trimgalory/<span>${id}</span></span></code><code><span>echo <span>"<span>${id}</span>"</span></span></code><code><span><br></span></code><code><span>echo <span>"++++++++===========================+++++++++++"</span></span></code><code><span><br></span></code><code><span><br></span></code><code><span>trim_galore --phred33 -q <span>20</span> --length <span>36</span>    \</span></code><code><span>  --stringency <span>3</span> --fastqc --paired --max_n <span>3</span> \</span></code><code><span>  -o  ~/silicosis_wangchen/data/02_trimgalory/<span>${id}</span>                          \</span></code><code><span>  ~/silicosis_wangchen/data/CRA005925/<span>${id}</span>/<span>${id}</span>_f1.fq.gz \</span></code><code><span>  ~/silicosis_wangchen/data/CRA005925/<span>${id}</span>/<span>${id}</span>_f2.fq.gz</span></code><code><span>done</span></code><code><span><br></span></code><code><span>chmod +x batch_trimgalore.sh</span></code><code><span>nohup  bash  batch_trimgalore.sh   &gt;trim_galore.log  <span>2</span>&gt;&amp;<span>1</span>  &amp;</span></code><code><span><br></span></code></pre></section><section><strong></strong></section><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="python"><code><span>trimgalory 详细参数</span></code><code><span><br></span></code><code><span>-q/--quality &lt;INT&gt;      Trim low-quality ends <span>from</span> reads <span>in</span> addition to adapter removal. For</span></code><code><span>                        RRBS samples, quality trimming will be performed first, <span>and</span> adapter</span></code><code><span>                        trimming <span>is</span> carried <span>in</span> a second round. Other files are quality <span>and</span> adapter</span></code><code><span>                        trimmed <span>in</span> a single <span>pass</span>. The algorithm <span>is</span> the same <span>as</span> the one used by BWA</span></code><code><span>                        (Subtract INT <span>from</span> all qualities; compute partial sums <span>from</span> all indices</span></code><code><span>                        to the end of the sequence; cut sequence at the index at which the sum <span>is</span></span></code><code><span>                        minimal). Default Phred score: <span>20.</span></span></code><code><span><br></span></code><code><span>--phred33               Instructs Cutadapt to use ASCII+<span>33</span> quality scores <span>as</span> Phred scores</span></code><code><span>                        (Sanger/Illumina <span>1.9</span>+ encoding) <span>for</span> quality trimming. Default: ON.</span></code><code><span><br></span></code><code><span>--phred64               Instructs Cutadapt to use ASCII+<span>64</span> quality scores <span>as</span> Phred scores</span></code><code><span>                        (Illumina <span>1.5</span> encoding) <span>for</span> quality trimming.</span></code><code><span><br></span></code><code><span>--fastqc                Run FastQC <span>in</span> the default mode on the FastQ file once trimming <span>is</span> complete.</span></code><code><span><br></span></code><code><span>--fastqc_args <span>"&lt;ARGS&gt;"</span>  Passes extra arguments to FastQC. If more than one argument <span>is</span> to be passed</span></code><code><span>                        to FastQC they must be <span>in</span> the form <span>"arg1 arg2 etc."</span>. An example would be:</span></code><code><span>                        --fastqc_args <span>"--nogroup --outdir /home/"</span>. Passing extra arguments will</span></code><code><span>                        automatically invoke FastQC, so --fastqc does <span>not</span> have to be specified</span></code><code><span>                        separately.</span></code><code><span><br></span></code><code><span>-a/--adapter &lt;STRING&gt;   Adapter sequence to be trimmed. If <span>not</span> specified explicitly, Trim Galore will</span></code><code><span>                        <span>try</span> to auto-detect whether the Illumina universal, Nextera transposase <span>or</span> Illumina</span></code><code><span>                        small RNA adapter sequence was used. Also see <span>'--illumina'</span>, <span>'--nextera'</span> <span>and</span></span></code><code><span>                        <span>'--small_rna'</span>. If no adapter can be detected within the first <span>1</span> million sequences</span></code><code><span>                        of the first file specified <span>or</span> <span>if</span> there <span>is</span> a tie between several adapter sequences,</span></code><code><span>                        Trim Galore defaults to <span>'--illumina'</span> (<span>as</span> long <span>as</span> the Illumina adapter was one of the</span></code><code><span>                        options, <span>else</span> <span>'--nextera'</span> <span>is</span> the default). A single base</span></code><code><span>                        may also be given <span>as</span> e.g. -a A{<span>10</span>}, to be expanded to -a AAAAAAAAAA.</span></code><code><span><br></span></code><code><span>                        At a special request, multiple adapters can also be specified like so:</span></code><code><span>                        -a  <span>" AGCTCCCG -a TTTCATTATAT -a TTTATTCGGATTTAT"</span></span></code><code><span>                        -a2 <span>" AGCTAGCG -a TCTCTTATAT -a TTTCGGATTTAT"</span>, <span>or</span> so:</span></code><code><span>                        -a <span>"file:../multiple_adapters.fa"</span></span></code><code><span>                        -a2 <span>"file:../different_adapters.fa"</span></span></code><code><span>                        Potentially <span>in</span> conjucntion <span>with</span> the parameter <span>"-n 3"</span> to trim all adapters. Please note</span></code><code><span>                        that this <span>is</span> NOT needed <span>for</span> standard trimming! </span></code><code><span>                        More Information here: https://github.com/FelixKrueger/TrimGalore/issues/<span>86</span></span></code><code><span><br></span></code><code><span>--max_length &lt;INT&gt;      Discard reads that are longer than &lt;INT&gt; bp after trimming. This <span>is</span> only advised <span>for</span></span></code><code><span>                        smallRNA sequencing to remove non-small RNA sequences.</span></code><code><span><br></span></code><code><span><br></span></code><code><span>--stringency &lt;INT&gt;      Overlap <span>with</span> adapter sequence required to trim a sequence. Defaults to a</span></code><code><span>                        very stringent setting of <span>1</span>, i.e. even a single bp of overlapping sequence</span></code><code><span>                        will be trimmed off <span>from</span> the <span>3</span><span>' end of any read.</span></span></code><code><span><br></span></code><code><span><br></span></code><code><span>--length &lt;INT&gt;          Discard reads that became shorter than length INT because of either</span></code><code><span>                        quality <span>or</span> adapter trimming. A value of <span>'0'</span> effectively disables</span></code><code><span>                        this behaviour. Default: <span>20</span> bp.</span></code><code><span><br></span></code><code><span>                        For paired-end files, both reads of a read-pair need to be longer than</span></code><code><span>                        &lt;INT&gt; bp to be printed out to validated paired-end files (see option --paired).</span></code><code><span>                        If only one read became too short there <span>is</span> the possibility of keeping such</span></code><code><span>                        unpaired single-end reads (see --retain_unpaired). Default pair-cutoff: <span>20</span> bp.</span></code><code><span><br></span></code><code><span>--max_n COUNT           The total number of Ns a read may contain before it will be removed altogether.</span></code><code><span>                        In a paired-end setting, either read exceeding this limit will result <span>in</span> the entire</span></code><code><span>                        pair being removed <span>from</span> the trimmed output files. If COUNT <span>is</span> a number between <span>0</span> <span>and</span> <span>1</span>,</span></code><code><span>                        it <span>is</span> interpreted <span>as</span> a fraction of the read length.</span></code><code><span><br></span></code><code><span><br></span></code></pre></section><section><strong></strong></section><p><img data-galleryid="" data-imgfileid="100003566" data-ratio="0.19074074074074074" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345Sks7osyILdoYFR497XU7DNDspZQP5DJVib0HRvriaYf0BcJO3cXY0aaicib7Tib70AlrhT55ttibKlAV3iasw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345Sks7osyILdoYFR497XU7DNDspZQP5DJVib0HRvriaYf0BcJO3cXY0aaicib7Tib70AlrhT55ttibKlAV3iasw/640?wx_fmt=png&amp;from=appmsg"></p><p><br></p><p>在Linux中，你可以使用<code>ls</code>命令与通配符来<strong>获取当前目录下所有以"CRR"开头的文件名称。</strong>具体命令如下：<span></span></p><pre><div><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="http"><code><span><span>ls CRR*</span></span></code><code><span><br></span></code><code><span>此命令会列出当前目录中所有以"CRR"开头的文件和目录。</span></code><code><span>如果你只想列出文件名，可以结合grep和ls命令来过滤：</span></code><code><span><br></span></code><code><span><br></span></code><code><span>ls | grep '^CRR'</span></code><code><span><br></span></code><code><span>此命令会列出当前目录中所有以"CRR"开头的文件和目录名称。</span></code><code><span>如果你只想要文件而不包含目录，可以使用find命令：</span></code><code><span>find . -maxdepth 1 -type f -name 'CRR*'</span></code><code><span><br></span></code><code><span>此命令会列出当前目录（不包括子目录）中所有以"CRR"开头的文件。</span></code><code><span><br></span></code></pre></section><p><img data-galleryid="" data-imgfileid="100003565" data-ratio="0.2702439024390244" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345Sks7osyILdoYFR497XU7DNDs89WrBiaekJrGUAriazwxwtgfAoPuMtA5kIibib3B5YpZYpI2y9iav6rvoqA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1025" src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345Sks7osyILdoYFR497XU7DNDs89WrBiaekJrGUAriazwxwtgfAoPuMtA5kIibib3B5YpZYpI2y9iav6rvoqA/640?wx_fmt=png&amp;from=appmsg"></p><p><br></p><p><br></p><p><br></p><p><strong>4.Hista2比对</strong></p><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="bash"><code><span>cat &gt;map.sh</span></code><code><span><br></span></code><code><span><span>#!/bin/bash</span></span></code><code><span><br></span></code><code><span><span>#获取样本名称，其实就是每个fastq文件的前缀</span></span></code><code><span>ls ../CRA005925/ |grep <span>'^CRR'</span> |<span>while</span> <span>read</span> i</span></code><code><span><br></span></code><code><span><span>do</span></span></code><code><span>  <span>echo</span> <span>${i}</span></span></code><code><span>  <span>echo</span> <span>"==========*****************======="</span></span></code><code><span>  hisat2 -x ~/20240125_rnaseq/data/4086673388_YoungLeeyyL/_/index/grcm38/genome -p 20 --summary-file ./<span>${i}</span>.mapinfo.txt \</span></code><code><span>            -1 ~/silicosis_wangchen/data/CRA005925/<span>${i}</span>/<span>${i}</span>_f1.fq.gz \</span></code><code><span>            -2 ~/silicosis_wangchen/data/CRA005925/<span>${i}</span>/<span>${i}</span>_r2.fq.gz \</span></code><code><span>            |samtools sort -@ 20 -o <span>${i}</span>.sorted.bam</span></code><code><span><span>done</span></span></code><code><span><br></span></code><code><span>chmod +x map.sh</span></code><code><span>nohup  bash  map.sh  &gt;map.log  2&gt;&amp;1  &amp;</span></code><code><span><br></span></code></pre></section><p><strong></strong></p><p><img data-galleryid="" data-imgfileid="100003567" data-ratio="0.049074074074074076" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345Sks7osyILdoYFR497XU7DNDsPERzUD5vzLPDnjqmIq0ACqySxlYmzYnjcFLtwC8zGG8Q4DkbS7QOCA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345Sks7osyILdoYFR497XU7DNDsPERzUD5vzLPDnjqmIq0ACqySxlYmzYnjcFLtwC8zGG8Q4DkbS7QOCA/640?wx_fmt=png&amp;from=appmsg"></p><p><strong></strong></p></div></pre><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="sql"><code><span>重要参数</span></code><code><span><br></span></code><code><span> <span>--phred33          qualities are Phred+33 (default)</span></span></code><code><span> <span>--phred64          qualities are Phred+64</span></span></code><code><span><br></span></code><code><span> -p/<span>--threads &lt;int&gt; number of alignment threads to launch (1)</span></span></code><code><span><br></span></code><code><span>Usage: </span></code><code><span>  hisat2 [options]* -x &lt;ht2-idx&gt; {-1 &lt;m1&gt; -2 &lt;m2&gt; | -U &lt;r&gt;} [-S &lt;sam&gt;]</span></code><code><span><br></span></code><code><span>  &lt;ht2-idx&gt;  Index filename prefix (minus trailing .X.ht2).</span></code><code><span>  &lt;m1&gt;       Files <span>with</span> <span>#1 mates, paired with files in &lt;m2&gt;.</span></span></code><code><span>             Could be gzip<span>'ed (extension: .gz) or bzip2'</span>ed (extension: .bz2).</span></code><code><span>  &lt;m2&gt;       Files <span>with</span> <span>#2 mates, paired with files in &lt;m1&gt;.</span></span></code><code><span>             Could be gzip<span>'ed (extension: .gz) or bzip2'</span>ed (extension: .bz2).</span></code><code><span>  &lt;r&gt;        Files <span>with</span> unpaired reads.</span></code><code><span>             Could be gzip<span>'ed (extension: .gz) or bzip2'</span>ed (extension: .bz2).</span></code><code><span>  &lt;sam&gt;      <span>File</span> <span>for</span> SAM <span>output</span> (<span>default</span>: stdout)</span></code><code><span><br></span></code><code><span>  &lt;m1&gt;, &lt;m2&gt;, &lt;r&gt; can be comma-separated lists (<span>no</span> <span>whitespace</span>) <span>and</span> can be</span></code><code><span>  specified many times.  E.g. <span>'-U file1.fq,file2.fq -U file3.fq'</span>.</span></code><code><span><br></span></code><code><span><br></span></code></pre></section><section><br></section><section><strong>5.featurecounts定量</strong></section><section><ul><li><li><li><li><li><li><li><li></ul><pre data-lang="bash"><code><span>cat  &gt;count.sh</span></code><code><span> </span></code><code><span><span>#!/bin/bash</span></span></code><code><span>featureCounts -a ~/20240125_rnaseq/data/4086673388_YoungLeeyyL/_/index/Mus_musculus.GRCm38.102.gtf  \</span></code><code><span>--extraAttributes gene_name,gene_biotype -T 60 \</span></code><code><span>-t exon -p -o ./all_samples_counts.txt  ~/silicosis_wangchen/data/03map-data/*.bam</span></code><code><span><br></span></code><code><span><br></span></code></pre></section><section><strong></strong></section><section><br></section><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="http"><code><span>重要参数：</span></code><code><span><span>Usage</span>: featureCounts [options] -a &lt;annotation_file&gt; -o &lt;output_file&gt; input_file1 [input_file2] ... </span></code><code><span><br></span></code><code><span><span>## Mandatory arguments:</span></span></code><code><span><br></span></code><code><span>  -a &lt;<span>string</span>&gt;         Name of an annotation file. GTF/GFF format <span>by</span> <span>default</span>. See</span></code><code><span>                      -F option <span>for</span> more format information. <span>Inbuilt <span>annotations</span></span></span></code><code><span>                      (<span>SAF format</span>) <span>is</span> available <span>in</span> 'annotation' directory of the</span></code><code><span>                      package. Gzipped file <span>is</span> also accepted.</span></code><code><span><br></span></code><code><span>  -o &lt;<span>string</span>&gt;         Name of output file including read counts. A separate file</span></code><code><span>                      including summary statistics of counting results <span>is</span> also</span></code><code><span>                      included <span>in</span> the <span>output</span> (<span><span>'&lt;string&gt;.summary'</span></span>). Both files</span></code><code><span>                      are <span>in</span> tab delimited format.</span></code><code><span><br></span></code><code><span>  input_file1 [input_file2] ...   A list of SAM or BAM format files. They can be</span></code><code><span>                      either name or location sorted. If no files provided,</span></code><code><span>                      &lt;stdin&gt; input <span>is</span> expected. Location-sorted paired-end reads are automatically sorted <span>by</span> read names.</span></code><code><span><br></span></code><code><span>-T &lt;<span>int</span>&gt;            Number of the threads. 1 <span>by</span> <span>default</span>.</span></code><code><span><br></span></code><code><span># Parameters specific to paired end reads</span></code><code><span><br></span></code><code><span>  -p                  If specified, libraries are assumed to contain paired-end</span></code><code><span>                      reads. For any library that contains paired-end reads, the</span></code><code><span>                      'countReadPairs' parameter controls <span>if</span> read pairs or reads</span></code><code><span>                      should be counted.</span></code><code><span><br></span></code></pre></section><section><strong></strong></section><p><br></p><section><ul><li><li><li><li><li><li><li></ul><pre data-lang="javascript"><code><span>不可查看链接：</span></code><code><span>马拉松课程</span></code><code><span>https:<span>//mp.csdn.net/mp_blog/creation/editor/135890652</span></span></code><code><span><br></span></code><code><span>jimmy2022</span></code><code><span>https:<span>//mp.csdn.net/mp_blog/creation/editor/128177352</span></span></code><code><span><br></span></code></pre></section><section><mp-common-profile data-pluginname="mpprofile" data-id="Mzg2NDcxMzYwNg==" data-headimg="http://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345Sks7JhVxUX46ZKxPSob6ptqIZgnIEnHOn5VMwCX8sN6MQy1Pq4XXFEOJ6grAmsoQugyCDKOZictDBHA/0?wx_fmt=png" data-nickname="生信小博士" data-alias="bioinformatics_Dr" data-signature="【生物信息学】R语言开始，学习生信。Seurat，单细胞测序，空间转录组。 Python，scanpy，cell2location。资料分享" data-from="0" data-is_biz_ban="0"></mp-common-profile></section><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/uVMKL1Us4YWDVen2JB4kGQ",target="_blank" rel="noopener noreferrer">原文链接</a>
