---
title: "QuSAGE，单基因集多基因集分析两手抓"
date: 2023-02-27T05:37:40Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Acdemic"]
---
QuSAGE，单基因集多基因集分析两手抓 by 生信菜鸟团
------
<div><h1 data-tool="mdnice编辑器">引入</h1><p data-tool="mdnice编辑器">今日命题作文：</p><section><figure data-tool="mdnice编辑器"><img data-ratio="0.5895522388059702" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8C9ZGT1SXuLh80N11D1aTRtYZc7TLJactaHg8HNwZkFClJD6XKrp7bwr96nCKaogzMg3eepLQCvQ/640?wx_fmt=png" data-type="png" data-w="268" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8C9ZGT1SXuLh80N11D1aTRtYZc7TLJactaHg8HNwZkFClJD6XKrp7bwr96nCKaogzMg3eepLQCvQ/640?wx_fmt=png"><section><span>既然咱可以用ChatGPT，那就试试ChatGPT的学习效果，先不管这是啥，先丢给ChatGPT 看看，</span><strong>问：quantita- tive set analysis for gene expression (QuSAGE) R包怎么使用。</strong></section></figure></section><p data-tool="mdnice编辑器"><span>答</span><strong>：</strong><strong></strong></p><blockquote data-tool="mdnice编辑器"><p>QuSAGE是一个用于基因表达定量集分析的R包，可以用来识别基因集在特定生物学上下文中的表达变化。以下是使用QuSAGE进行基因表达定量集分析的一些基本步骤：</p></blockquote><blockquote data-tool="mdnice编辑器"><section><span>1. 安装和加载QuSAGE包 首先，您需要安装QuSAGE包。在R中，您可以使用以下代码安装QuSAGE包：</span></section></blockquote><pre data-tool="mdnice编辑器"><code>install.packages(<span>"QuSAGE"</span>)<br></code></pre><blockquote data-tool="mdnice编辑器"><p>ps: ChatGPT 显示的是python代码</p></blockquote><blockquote data-tool="mdnice编辑器"><p>安装完成后，使用以下代码加载QuSAGE包：</p></blockquote><pre data-tool="mdnice编辑器"><code><span>library</span>(QuSAGE)<br></code></pre><blockquote data-tool="mdnice编辑器"><section><span>2. 准备数据 您需要准备一个包含基因表达数据的表格。数据表格应该包含基因的名称和每个样本的表达值。您还需要准备一个包含您感兴趣的基因集的列表。</span></section></blockquote><blockquote data-tool="mdnice编辑器"><section><span>3. 运行QuSAGE 使用QuSAGE()函数运行QuSAGE。该函数需要四个参数：</span></section></blockquote><blockquote data-tool="mdnice编辑器"><ul><li><section><span>gene_expression_data: 包含基因表达数据的数据框。</span></section></li><li><section><span>gene_set_list: 包含感兴趣基因集的列表。</span></section></li><li><section><span>adjust_method: P值校正方法，可以选择'fdr'或'bonferroni'。</span></section></li><li><section><span>n_permutation: 用于计算P值的置换次数。</span></section></li></ul><p><span></span>以下是一个使用QuSAGE的示例代码：</p></blockquote><pre data-tool="mdnice编辑器"><code><span># 导入基因表达数据</span><br>data(<span>"exampleExpressionSet"</span>)<br>gene_expression_data &lt;- exprs(exampleExpressionSet)<br><br><span># 导入基因集列表</span><br>data(<span>"exampleGeneSetList"</span>)<br>gene_set_list &lt;- geneSetList(exampleGeneSetList)<br><br><span># 运行QuSAGE</span><br>results &lt;- QuSAGE(gene_expression_data, gene_set_list, adjust_method = <span>"fdr"</span>, n_permutation = <span>1000</span>)<br></code></pre><blockquote data-tool="mdnice编辑器"><section><span>4. 解释结果 运行QuSAGE后，您将获得一个包含每个基因集的结果的表格。每个基因集的结果包括基因集的名称、基因集中基因的数量、基因集的富集度得分、基因集中的显著差异基因数、显著性水平和经过校正的显著性水平。</span></section><p>使用QuSAGE进行基因表达定量集分析需要更多的细节和参数调整，具体取决于您的数据和研究问题。因此，我们建议您查看QuSAGE包的文档和示例，以深入了解如何使用QuSAGE进行基因表达定量集分析。</p></blockquote><p data-tool="mdnice编辑器">下面，我们按他说的试试。首先第一步，安装R包：</p><pre data-tool="mdnice编辑器"><code>install.packages(<span>"QuSAGE"</span>)<br><span>#&gt; Installing package into ‘/home/data/t120328/R/x86_64-pc-linux-gnu-library/4.2’</span><br><span>#&gt; (as ‘lib’ is unspecified)</span><br><span>#&gt; Warning in install.packages :</span><br><span>#&gt;   package ‘QuSAGE’ is not available for this version of R</span><br><span>#&gt; </span><br><span>#&gt; A version of this package for your version of R might be available elsewhere,</span><br><span>#&gt; see the ideas at</span><br><span>#&gt; https://cran.r-project.org/doc/manuals/r-patched/R-admin.html#Installing-packages</span><br></code></pre><p data-tool="mdnice编辑器">出现报错，咦怎么回事呢？我们来必应搜索一下，<code>QuSAGE</code>是个啥。<img data-ratio="0.5194300518134715" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8C9ZGT1SXuLh80N11D1aTRTQt2hPCZCCf1kf1oqvXicKIaPE0ZtTlqKxiaTw5ybtBCLtic0e0EaU0SA/640?wx_fmt=png" data-type="png" data-w="772" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8C9ZGT1SXuLh80N11D1aTRTQt2hPCZCCf1kf1oqvXicKIaPE0ZtTlqKxiaTw5ybtBCLtic0e0EaU0SA/640?wx_fmt=png"></p><p data-tool="mdnice编辑器">哦，原来他是个<code>bioconductor</code> R包，所以，他的安装方式是：</p><pre data-tool="mdnice编辑器"><code><span>if</span> (!<span>require</span>(<span>"BiocManager"</span>, quietly = <span>TRUE</span>))<br>  install.packages(<span>"BiocManager"</span>)<br><br>BiocManager::install(<span>"qusage"</span>)<br></code></pre><p data-tool="mdnice编辑器">ChatGPT 果然又欺骗了我。假设他已经正常安装了，里面是不是存在他说的示例data呢？</p><pre data-tool="mdnice编辑器"><code><span>library</span>(qusage)<br><span>#&gt; Loading required package: limma</span><br>data(<span>"exampleExpressionSet"</span>)<br><span>#&gt; Warning message:</span><br><span>#&gt; In data("exampleExpressionSet") : data set ‘exampleExpressionSet’ not found</span><br></code></pre><p data-tool="mdnice编辑器">后续还试着告诉ChatGPT 它是bioconductor的包，虽然ChatGPT承认了错误，但输出的代码依旧是错的。果然，啥也不是。那……还是好好看看学习文档吧。</p><h1 data-tool="mdnice编辑器">介绍</h1><p data-tool="mdnice编辑器">R包 qusage 是发布在Bioconductor的R包，目前已更新至v2.32.0。按Bioconductor的描述：(翻译by彩云小译)</p><blockquote data-tool="mdnice编辑器"><p>该软件包是基因表达定量集分析(QuSAGE)方法的实现(Yaari G. 等，Nucl Acids Res，2013)。这是一个新颖的基因集富集型测试，旨在为基因表达研究提供一个更快，更准确，更容易理解的测试。Quseage Wu 等人提出的方差膨胀因子技术解释了基因间的相关性(Nucleic Acids Res，2012)。此外，q 使用完全概率密度函数量化基因组活性，而不是简单地用一个数字(一个 P 值)来评估与零假设的偏差。从这个 PDF，P 值和置信区间可以很容易地提取。保存 PDF 还可以进行事后比较(例如，基因集活性的成对比较) ，同时保持统计可追溯性。最后，虽然 quse 与来自现有方法(例如 LIMMA)的个体基因统计相容，但是实现了一种基于 Welch 的方法，该方法显示可以提高特异性。如(Turner JA 等，BMC Bioinformation，2015)所述，QuSAGE 包还包括混合效应模型实现，以及如(Meng H，et al。计算机生物学。2019年)。如有问题，请联系 Chris Bolen (cbolen1@gmail.com)或 Steven Kleinstein (Steven.Kleinstein@yale.edu)</p></blockquote><p data-tool="mdnice编辑器">原文见：</p><blockquote data-tool="mdnice编辑器"><p>Yaari G, Bolen CR, Thakar J, Kleinstein SH. Quantitative set analysis for gene expression: a method to quantify gene set differential expression including gene-gene correlations. <em>Nucleic Acids Res</em>. 2013;41(18):e170. doi:10.1093/nar/gkt660</p></blockquote><p data-tool="mdnice编辑器">总的来说，qusage 是一类基因集评估工具。ChatGPT“QuSAGE是一个用于基因表达定量集分析的R包，可以用来识别基因集在特定生物学上下文中的表达变化”这句描述，勉强合理。</p><h1 data-tool="mdnice编辑器">以单一基因集为演示使用qusage</h1><p data-tool="mdnice编辑器">安装见上<code>qusage</code> 实际上是对包中三个函数（<code>makeComparison</code>，<code>aggregateGeneSet</code>, and <code>calcVIF</code>,）的一个再包装；如果深入理解了<code>qusage</code>的过程，也可以考虑根据需求单独运行各个函数。</p><h2 data-tool="mdnice编辑器"><span>input</span><span> </span></h2><p data-tool="mdnice编辑器">在使用默认设置时，<code>qusage</code>需要四个输入：<code>eset</code>, <code>labels</code>, <code>contrast</code>, &amp; <code>geneSets</code>以下为演示：</p><pre data-tool="mdnice编辑器"><code><span>library</span>(qusage)<br><span>#&gt; 载入需要的程辑包：limma</span><br><span>library</span>(nlme)<br>data(<span>"fluExample"</span>)<br>data(<span>"GeneSets"</span>)<br></code></pre><p><img data-ratio="0.2530562347188264" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8C9ZGT1SXuLh80N11D1aTRIQv8fOz0icYxZByS05zcaoqz5y8RoUabPFz1v3qbicvYcAneg6nFyTfA/640?wx_fmt=png" data-type="png" data-w="818" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8C9ZGT1SXuLh80N11D1aTRIQv8fOz0icYxZByS05zcaoqz5y8RoUabPFz1v3qbicvYcAneg6nFyTfA/640?wx_fmt=png"></p><section><span>适当减少数据量，加快演示分析速度。</span></section><pre data-tool="mdnice编辑器"><code>i = c(which(flu.meta$Hours==<span>0</span>), which(flu.meta$Hours==<span>77</span>))<br>eset = eset.full[,i]<br>resp = flu.meta$Condition[i]<br></code></pre><h3 data-tool="mdnice编辑器"><code>eset</code></h3><p data-tool="mdnice编辑器">（<code>eset</code> 既可以是log2转化过的<code>matrix</code>，也可以是含log2标准化表达矩阵的<code>ExpressionSet</code>格式的对象）</p><pre data-tool="mdnice编辑器"><code>dim(eset)<br>eset[<span>1</span>:<span>5</span>,<span>1</span>:<span>5</span>]<br><span>#&gt;        GSM757899 GSM757915 GSM757931 GSM757947 GSM757962</span><br><span>#&gt; AKT3    4.600442  4.804845  4.454166  4.459647  4.589498</span><br><span>#&gt; ACOT8   8.235681  7.667416  8.483951  8.471545  8.343489</span><br><span>#&gt; GNPDA1  8.074555  8.732101  8.308964  8.499822  8.285027</span><br><span>#&gt; CDH2    4.673311  5.088127  4.764211  4.630295  4.517296</span><br><span>#&gt; TANK    9.039308  8.435497  8.835157  9.031125  8.951254</span><br></code></pre><p data-tool="mdnice编辑器">这里给出的eset是作者提供的演示数据（Huang Y等人（GEO ID：GSE30550），这里选择了该研究观察了17名患者流感病毒暴露前和暴露后77小时的基因表达。（该数据集已经被修整以去除极低表达的基因）</p><h3 data-tool="mdnice编辑器"><code>lables</code></h3><p data-tool="mdnice编辑器">字符向量，数据通常由两组组成，使用“mock”和“treatment”或“healthy”和“disease”来描述  .这里用t0,t1分别表示暴露前和暴露后。</p><pre data-tool="mdnice编辑器"><code>labels = c(rep(<span>"t0"</span>,<span>17</span>),rep(<span>"t1"</span>,<span>17</span>))<br>labels<br><span>#&gt; [1] "t0" "t0" "t0" "t0" "t0" "t0" "t0" "t0" "t0" "t0" "t0" "t0" "t0" "t0" "t0" "t0" "t0" "t1" "t1" "t1"</span><br><span>#&gt; [21] "t1" "t1" "t1" "t1" "t1" "t1" "t1" "t1" "t1" "t1" "t1" "t1" "t1" "t1"</span><br></code></pre><h3 data-tool="mdnice编辑器"><code>contrast</code></h3><p data-tool="mdnice编辑器">这个<code>labels</code>和<code>contrast</code> 设置让我想起了limma。（qusage 依赖limma 包，可能有关）</p><pre data-tool="mdnice编辑器"><code>contrast = <span>"t1-t0"</span><br></code></pre><h3 data-tool="mdnice编辑器"><code>geneSets</code></h3><p data-tool="mdnice编辑器">这里作者提供了一个观察变量，<code>ISG.geneSet</code>：干扰素基因集。</p><pre data-tool="mdnice编辑器"><code>ISG.geneSet[<span>1</span>:<span>10</span>]<br><span>#&gt; [1] "ADAR"    "ADM"     "ALDH1A1" "ANKFY1"  "APOL1"   "APOL2"   "APOL6"   "ARG2"    "ARHGEF3" "ATF3"</span><br></code></pre><h2 data-tool="mdnice编辑器"><span>run qusage</span><span> </span></h2><p data-tool="mdnice编辑器">当存在这些内容时，我们可以按以下代码运行qusage：</p><pre data-tool="mdnice编辑器"><code>qs.results = qusage(eset, labels, contrast, ISG.geneSet)<br><span>#&gt; Calculating gene-by-gene comparisons...Done.</span><br><span>#&gt; Aggregating gene data for gene sets.Done.</span><br><span>#&gt; Calculating variance inflation factors...Done.</span><br>head(qs.results)<br><span>### 计算p值  ###</span><br>pdf.pVal(qs.results)<br><span>#&gt; [1] 0.000181850</span><br></code></pre><p data-tool="mdnice编辑器">qusage方法将返回一个QSarray对象，其中包含单个基因分布和<code>pathway</code>本身的统计数据。<code>pdf.pVal</code>函数，它计算一个p值，可以评估的基因组的卷积分布是否signifcantly enriched（即基因组的平均倍数变化不明显不等于0）。这里的p值&lt;&lt;0.05, 可以说很显著。</p><h3 data-tool="mdnice编辑器">概率分布函数密度可视化</h3><pre data-tool="mdnice编辑器"><code><span>### 简要对该概率密度函数可视化###</span><br>plot(qs.results, xlab=<span>"Gene Set Activation"</span>)<br></code></pre><p><img data-ratio="0.6822660098522167" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8C9ZGT1SXuLh80N11D1aTRMR0ibFAu388eqynibLibqHZCGEH8BrHYic32nCdneiaYGXK5gAbN2AzV8IA/640?wx_fmt=png" data-type="png" data-w="812" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8C9ZGT1SXuLh80N11D1aTRMR0ibFAu388eqynibLibqHZCGEH8BrHYic32nCdneiaYGXK5gAbN2AzV8IA/640?wx_fmt=png"></p><section><span>这个图应该怎么解读呢？（这东西听着很熟，但是真的不大会了，求指教）</span></section><h3 data-tool="mdnice编辑器">单个基因可视化</h3><p data-tool="mdnice编辑器">plotCIsGenes以查看哪些基因最活跃。此函数生成的图与plotCIs函数非常相似，但在本例中，它将绘制均值和置信区间pathway中每个单独基因的密度区间。</p><pre data-tool="mdnice编辑器"><code>plotCIsGenes(qs.results)<br></code></pre><p><img data-ratio="0.32171428571428573" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8C9ZGT1SXuLh80N11D1aTRmCicZKITr0YDjJ6yH4eS1Nib9NbwJrjadP77cjicq4uTa8ialQUaDeTCLA/640?wx_fmt=png" data-type="png" data-w="1750" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8C9ZGT1SXuLh80N11D1aTRmCicZKITr0YDjJ6yH4eS1Nib9NbwJrjadP77cjicq4uTa8ialQUaDeTCLA/640?wx_fmt=png"></p><section>这个图个人感觉是要比GSEA 的那个图看起来舒服的，至少我能知道哪些基因排在前面。（此处行名是基因）</section><h1 data-tool="mdnice编辑器">多个基因集qusage 如何处理</h1><p data-tool="mdnice编辑器">在示例文件中，我们清晰看到有 <code>MSIG.geneSets</code>。即是多个基因集的集合，来自于msigdb数据库“c2.cp.kegg.v4.0.symbols.gmt。如感兴趣，可自行下载。</p><h2 data-tool="mdnice编辑器"><span>msig.genesets</span><span> </span></h2><pre data-tool="mdnice编辑器"><code>summary(MSIG.geneSets[<span>1</span>:<span>5</span>])<br><span>#&gt;                                               Length Class  Mode     </span><br><span>#&gt; KEGG_GLYCOLYSIS_GLUCONEOGENESIS               62     -none- character</span><br><span>#&gt; KEGG_CITRATE_CYCLE_TCA_CYCLE                  32     -none- character</span><br><span>#&gt; KEGG_PENTOSE_PHOSPHATE_PATHWAY                27     -none- character</span><br><span>#&gt; KEGG_PENTOSE_AND_GLUCURONATE_INTERCONVERSIONS 28     -none- character</span><br><span>#&gt; KEGG_FRUCTOSE_AND_MANNOSE_METABOLISM          34     -none- character</span><br>MSIG.geneSets[<span>2</span>]<br><span>#&gt; $KEGG_CITRATE_CYCLE_TCA_CYCLE</span><br><span>#&gt;  [1] "LOC642502" "OGDHL"     "OGDH"      "PDHB"      "IDH3G"     "LOC283398" "IDH2"      "IDH1"     </span><br><span>#&gt;  [9] "PDHA2"     "PDHA1"     "SUCLA2"    "FH"        "DLST"      "ACO2"      "SUCLG2"    "ACO1"     </span><br><span>#&gt; [17] "SUCLG1"    "CS"        "IDH3B"     "ACLY"      "DLAT"      "PCK2"      "IDH3A"     "PCK1"     </span><br><span>#&gt; [25] "SDHA"      "SDHB"      "SDHC"      "SDHD"      "DLD"       "MDH2"      "PC"        "MDH1"</span><br></code></pre><h2 data-tool="mdnice编辑器"><span>run qusage</span><span> </span></h2><pre data-tool="mdnice编辑器"><code>qs.results.msig = qusage(eset, labels, contrast, MSIG.geneSets)<br><span>#&gt; Calculating gene-by-gene comparisons...Done.</span><br><span>#&gt; Aggregating gene data for gene sets......................................Done.</span><br><span>#&gt; Calculating variance inflation factors...Done.</span><br>numPathways(qs.results.msig)<br><span>#&gt; [1] 186</span><br>p.vals = pdf.pVal(qs.results.msig)<br>head(p.vals)<br><span>#&gt; [1] 0.288368617 0.970277527 0.009600519 0.528030282 0.112549381 0.106747491</span><br>q.vals = p.adjust(p.vals, method=<span>"fdr"</span>)<br>head(q.vals)<br><span>#&gt; [1] 0.54178346 0.98776294 0.09398403 0.75548948 0.36093422 0.36093422</span><br></code></pre><h2 data-tool="mdnice编辑器"><span>批量查看结果</span><span> </span></h2><pre data-tool="mdnice编辑器"><code> qsTable(qs.results.msig, number=<span>10</span>)<br><span>#&gt;                                        pathway.name log.fold.change      p.Value         FDR</span><br><span>#&gt; 135                    KEGG_CIRCADIAN_RHYTHM_MAMMAL     -0.16994258 1.359124e-05 0.002244544</span><br><span>#&gt; 22                        KEGG_HISTIDINE_METABOLISM     -0.06765522 3.205426e-05 0.002244544</span><br><span>#&gt; 127                 KEGG_HEMATOPOIETIC_CELL_LINEAGE     -0.10276783 3.620233e-05 0.002244544</span><br><span>#&gt; 124      KEGG_RIG_I_LIKE_RECEPTOR_SIGNALING_PATHWAY      0.13163298 3.774956e-04 0.017553544</span><br><span>#&gt; 15  KEGG_ALANINE_ASPARTATE_AND_GLUTAMATE_METABOLISM     -0.07912495 4.744648e-04 0.017650089</span><br><span>#&gt; 26                     KEGG_BETA_ALANINE_METABOLISM     -0.08318178 7.196554e-04 0.020600285</span><br><span>#&gt; 25                       KEGG_TRYPTOPHAN_METABOLISM      0.07844123 7.752796e-04 0.020600285</span><br><span>#&gt; 102                     KEGG_MTOR_SIGNALING_PATHWAY     -0.06927698 1.023345e-03 0.022348264</span><br><span>#&gt; 33                       KEGG_O_GLYCAN_BIOSYNTHESIS      0.13299596 1.085326e-03 0.022348264</span><br><span>#&gt; 125              KEGG_CYTOSOLIC_DNA_SENSING_PATHWAY      0.13611612 1.201520e-03 0.022348264</span><br></code></pre><h2 data-tool="mdnice编辑器"><span>可视化</span><span> </span></h2><pre data-tool="mdnice编辑器"><code>plot(qs.results.msig)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.4806428049671293" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8C9ZGT1SXuLh80N11D1aTRK785gMclayKufsOywvB2FMpcA8wSzYhso2sgAjwXSv8wz955huSqIA/640?wx_fmt=png" data-type="png" data-w="1369" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8C9ZGT1SXuLh80N11D1aTRK785gMclayKufsOywvB2FMpcA8wSzYhso2sgAjwXSv8wz955huSqIA/640?wx_fmt=png"><figcaption><span><br></span></figcaption><p><span>这默认情况下，当QSarray对象包含10个以上路径的数据时，</span><code>plot</code><span>将自动使用函数</span><code>plotCI</code><span>。此函数计算95%置信度使用</span><code>calcBayesCI</code><span>函数计算每个基因集的置信区间，并将其与每个基因集的平均倍数变化沿着绘图。当存在多个基因集时，这样展示相对更有效，但是仍然可以使用对</span><code>plotDensityCurves</code><span>的调用来生成PDF的图（此处行名是 pathway）。</span><span>概率密度分布函数可视化：</span></p><figcaption></figcaption></figure><pre data-tool="mdnice编辑器"><code><span>##select the best 5 gene sets</span><br>p.vals = pdf.pVal(qs.results.msig)<br>topSets = order(p.vals)[<span>1</span>:<span>5</span>]<br><span>##plot</span><br>plotDensityCurves(qs.results.msig, path.index=topSets, col=<span>1</span>:<span>5</span>, <br>                        lwd=<span>2</span>, xlab=expression(log[<span>2</span>](Activity)))<br></code></pre><p><img data-ratio="0.7754820936639119" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8C9ZGT1SXuLh80N11D1aTRkw4RiaX0YO5RD10RTHDynK0Kqq1Qjn7wYia0YL4qskfibOW9lJKOuY1vA/640?wx_fmt=png" data-type="png" data-w="726" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8C9ZGT1SXuLh80N11D1aTRkw4RiaX0YO5RD10RTHDynK0Kqq1Qjn7wYia0YL4qskfibOW9lJKOuY1vA/640?wx_fmt=png"></p><section>这里各个颜色和pathway 好像没有指定对应关系。读者在使用时可能需要探索自行设置的方法。</section><h1 data-tool="mdnice编辑器">结语</h1><p data-tool="mdnice编辑器">关于<code>qusage</code>的更多内容，在<code>bioconductor</code>的文档中有更多详细的解读以及使用操作描述。这里简要介绍qusage在单个和多个基因集中的应用，满足大部分分析场景。在<code>qusage</code>的文档中，还有更多例如配对样本分析的使用教程，在2015年的更新版本中，还给<code>qusage</code> 增加了“广义线性混合模型”的内容，函数命名为<code>qgen</code>。如有使用需求，下期再述。</p><h1 data-tool="mdnice编辑器">附：QSarray 对象内容</h1><blockquote data-tool="mdnice编辑器"><ul><li><section>mean, SD, dof - Vectors containing the means, standard deviations, and degrees</section><p>of freedom for each gene as output by makeComparison (see section 4.1). Togetherthey describe the t-distribution of each gene in the dataset.</p></li><li><section>n.samples - The number of samples used in the comparison. If the QSarray object </section><p>is the result of a call to combinePDF, this will be a numeric vector detailing thenumber of samples in each input QSarray.</p></li><li><section>path.PDF - This is a matrix which contains the density values for the PDF of </section><p>each gene set provided to aggregateGeneSet. The number of rows of path.PDFis determined by the n.points parameter, which by default is 4096.</p></li><li><section>path.mean - A numeric vector containing the mean fold change of each gene set </section><p>provided to aggregateGeneSet.</p></li><li><section>ranges - The (uncorrected) range that each PDF was calculated over. If the VIF </section><p>is not used to correct the range, the x-coordinates of the PDF are the sequence ofn.points from path.mean-ranges to path.mean+ranges</p></li><li><section>path.size - A vector containing the number of features in each pathway that </section><p>mapped to the input data.</p></li><li><section>vif - A vector containing the Variance In	ation Factor for each pathway, as cal-</section><p>culated by calcVIF.</p></li><li><section>pathways - A version of the input geneSets, represented as a list of numeric </section><p>vectors which map to the rows of eset which contain features in that gene set.</p></li><li><section>labels, contrast, pairVector, var.method, n.points - Copies of the input </section><p>parameters which are used in future calculations.</p></li><li><section>design - A design matrix for the linear model 	tting in LIMMA. Only created if </section><p>limma is used in makeComparison</p></li><li><section>QSlist - If the QSarray is the result of a call to combinePDF , the </section><p>QSlist will contain the list of QSarray objects used during the meta-analysis.</p></li></ul></blockquote><p><span>- END -</span></p><p><br></p><p><mp-style-type data-value="10000"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/fmvt0Y7jcCb5PMV-0cjsIA",target="_blank" rel="noopener noreferrer">原文链接</a>
