---
title: "整合单细胞数据和Bulk数据的多种方法（二）：R包Scissor"
date: 2023-02-24T03:12:03Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Acdemic"]
---
整合单细胞数据和Bulk数据的多种方法（二）：R包Scissor by 生信菜鸟团
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><p data-tool="mdnice编辑器">上个帖子介绍了整合单细胞数据和Bulk数据的最新方法scAB，<a href="https://mp.weixin.qq.com/s?__biz=MzUzMTEwODk0Ng==&amp;mid=2247510115&amp;idx=1&amp;sn=5549cab7c977410047d846df56196607&amp;scene=21#wechat_redirect" data-linktype="2">整合单细胞数据和Bulk数据的多种方法（一）：R包scAB</a>。除了scAB以外，衔接单细胞数据和Bulk测序数据的方法还有Scissor， scPrognosis 和DEGAS 等工具。本系列文章计划依次介绍这些算法，提供给大家更多的选择。<strong>本文将继续介绍Scissor算法</strong>。</p><p data-tool="mdnice编辑器">Scissor发表在2021年Nature Biotechnology上，题为《Identifying phenotype-associated subpopulations by integrating bulk and single-cell sequencing data》。和scAB算法一样，Scissor算法可利用单细胞数据和bulk RNA-seq数据及表型信息识别与疾病高度相关的细胞亚群。</p><figure data-tool="mdnice编辑器"><img data-ratio="0.6233885819521179" data-src="https://mmbiz.qpic.cn/mmbiz/iaRJcrq2Los837JMPkzt0jIMqnbh4rlxv0NQjRRSeZfOmQ8a0ichXUTFcy7RsmAIsoD61d4XLLb0uC1MIUV6DG9g/640?wx_fmt=other" data-type="other" data-w="1086" src="https://mmbiz.qpic.cn/mmbiz/iaRJcrq2Los837JMPkzt0jIMqnbh4rlxv0NQjRRSeZfOmQ8a0ichXUTFcy7RsmAIsoD61d4XLLb0uC1MIUV6DG9g/640?wx_fmt=other"><figcaption>image-20230125165155292</figcaption></figure><p data-tool="mdnice编辑器"><strong>本文将介绍：</strong></p><ul data-tool="mdnice编辑器"><li><section>Scissor算法的工作流程；</section></li><li><section>使用R实现方法（github: https://github.com/sunduanchen/Scissor）：为了横向比较Scissor和scAB算法，本文采用scAB包的示例数据，最后将Scissor的结果和scAB算法的结果进行对比。</section></li></ul><h3 data-tool="mdnice编辑器">一. Scissor算法的工作流程</h3><figure data-tool="mdnice编辑器"><img data-ratio="1.0949098621420996" data-src="https://mmbiz.qpic.cn/mmbiz/iaRJcrq2Los837JMPkzt0jIMqnbh4rlxvOAO6SgjhQyqBcicBibyttABpdiav1Hjq2rgWayYyyV4k9UhNFWhFSiazHw/640?wx_fmt=other" data-type="other" data-w="1886" src="https://mmbiz.qpic.cn/mmbiz/iaRJcrq2Los837JMPkzt0jIMqnbh4rlxvOAO6SgjhQyqBcicBibyttABpdiav1Hjq2rgWayYyyV4k9UhNFWhFSiazHw/640?wx_fmt=other"><figcaption>Figure_Method</figcaption></figure><p data-tool="mdnice编辑器">（a）输入数据为</p><ul data-tool="mdnice编辑器"><li><section>scRNA-seq单细胞数据；</section></li><li><section>其次是对应组织的Bulk RNA-seq数据，以及表型数据。</section></li></ul><p data-tool="mdnice编辑器">（b）然后计算每对细胞和bulk样本的Pearson相关系数<strong>构建相关系数矩阵</strong> ，Scissor会通过优化样本表型Y与相关矩阵S的回归模型。</p><p data-tool="mdnice编辑器">（c）由上述优化模型求解的<strong>非零系数β</strong>用于选择与目标表型相关的细胞亚群。其中Scissor+ 表示所选择的细胞与感兴趣的表型呈正相关，Scissor-为负相关。<strong>表型信息可以是连续变量、二分向量或临床生存数据</strong>，会分别对应不同的回归方法。</p><p data-tool="mdnice编辑器">（d and e） 为控制假阳性后续会有一个可靠性显著性检验以及差异表达基因分析、功能富集分析和motif分析等下游分析。</p><blockquote data-tool="mdnice编辑器"><p>参考来源：生信补给站 <a href="https://mp.weixin.qq.com/s?__biz=MzIyNDI1MzgzOQ==&amp;mid=2650398889&amp;idx=1&amp;sn=ffa4dd92adc81b9113a38e7e87fff7da&amp;scene=21#wechat_redirect" data-linktype="2">跟NBT学Scissor| bulk RNA + scRNA鉴定与目标表型相关的细胞亚群</a></p></blockquote><h3 data-tool="mdnice编辑器">二. Scissor的代码实现</h3><p data-tool="mdnice编辑器">Scissor R包github在 https://github.com/sunduanchen/Scissor.</p><p data-tool="mdnice编辑器"><strong>安装：</strong></p><pre data-tool="mdnice编辑器"><span></span><code>devtools::install_github(<span>"jinworks/scAB"</span>)<br></code></pre><h4 data-tool="mdnice编辑器">Step1. 加载R包</h4><pre data-tool="mdnice编辑器"><span></span><code><span>library</span>(Scissor)<br></code></pre><h4 data-tool="mdnice编辑器">Step2. 加载示例数据</h4><p data-tool="mdnice编辑器">这里采用scAB包的示例数据进行演示：</p><ul data-tool="mdnice编辑器"><li><section>可参考：<a href="https://mp.weixin.qq.com/s?__biz=MzUzMTEwODk0Ng==&amp;mid=2247510115&amp;idx=1&amp;sn=5549cab7c977410047d846df56196607&amp;scene=21#wechat_redirect" data-linktype="2">整合单细胞数据和Bulk数据的多种方法（一）：R包scAB</a></section></li></ul><pre data-tool="mdnice编辑器"><span></span><code><span>library</span>(Seurat)<br><span>library</span>(preprocessCore)<br><span>library</span>(scAB)<br><span>#设置可用的内存</span><br><span># options(future.globals.maxSize = 10 * 1024^3)</span><br><br><span>#Load data</span><br>data(<span>"data_survival"</span>)<br>dim(sc_dataset)<br><span>#&gt; [1] 21287  8853</span><br>sc_dataset<br><span># An object of class Seurat </span><br><span># 21287 features across 8853 samples within 1 assay </span><br><span># Active assay: RNA (21287 features, 0 variable features)</span><br>head(bulk_dataset[,<span>1</span>:<span>10</span>])<br>head(phenotype)<br><span>#                 time status</span><br><span># TCGA-2Y-A9GS-01  724      1</span><br><span># TCGA-2Y-A9GT-01 1624      1</span><br><span># TCGA-2Y-A9GU-01 1939      0</span><br><span># TCGA-2Y-A9GV-01 2532      1</span><br><span># TCGA-2Y-A9GW-01 1271      1</span><br><span># TCGA-2Y-A9GX-01 2442      0</span><br>colnames(bulk_dataset) == row.names(phenotype)<br><br>sc_dataset &lt;- run_seurat(sc_dataset,verbose = <span>FALSE</span>)<br>sc_dataset<br><br>UMAP_celltype &lt;- DimPlot(sc_dataset, reduction =<span>"umap"</span>,<br>                         group.by=<span>"celltype"</span>,label = <span>T</span>)<br>UMAP_celltype<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7058823529411765" data-src="https://mmbiz.qpic.cn/mmbiz/iaRJcrq2Los837JMPkzt0jIMqnbh4rlxvsicAtgmSvCXLF2J0Q3eZDLqOQ6WKovPySx4ne9WEpmZ6PwWRpkFgqwg/640?wx_fmt=other" data-type="other" data-w="765" src="https://mmbiz.qpic.cn/mmbiz/iaRJcrq2Los837JMPkzt0jIMqnbh4rlxvsicAtgmSvCXLF2J0Q3eZDLqOQ6WKovPySx4ne9WEpmZ6PwWRpkFgqwg/640?wx_fmt=other"><figcaption>image-20230223174400371</figcaption></figure><h4 data-tool="mdnice编辑器">Step3. 运行Scissor</h4><p data-tool="mdnice编辑器">与scAB一样，在获取到了单细胞表达矩阵、Bulk RNA-seq表达矩阵及其临床信息之后，运行Scissor：</p><pre data-tool="mdnice编辑器"><span></span><code>infos1 &lt;- Scissor(bulk_dataset, sc_dataset, phenotype, alpha = <span>0.05</span>, <br>                  family = <span>"cox"</span>, Save_file = <span>'./Scissor_survival.RData'</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.19474313022700118" data-src="https://mmbiz.qpic.cn/mmbiz/iaRJcrq2Los837JMPkzt0jIMqnbh4rlxv226T37TNR8xEMXGm8EhkkQLdYqOJZo7Box6sNf2BibsoWL2GoAGVsjg/640?wx_fmt=other" data-type="other" data-w="837" src="https://mmbiz.qpic.cn/mmbiz/iaRJcrq2Los837JMPkzt0jIMqnbh4rlxv226T37TNR8xEMXGm8EhkkQLdYqOJZo7Box6sNf2BibsoWL2GoAGVsjg/640?wx_fmt=other"><figcaption>image-20230223174116647</figcaption></figure><p data-tool="mdnice编辑器">这里我在Linux服务器上遇到了这个报错，而在Windows下运行成功，没搞明白。。所以我检查了源代码，修复了这个bug，只要运行下面的函数，再跑一次就行了：</p><pre data-tool="mdnice编辑器"><span></span><code>Scissor &lt;- <span>function</span> (bulk_dataset, sc_dataset, phenotype, tag = <span>NULL</span>, alpha = <span>NULL</span>, <br>                     cutoff = <span>0.2</span>, family = c(<span>"gaussian"</span>, <span>"binomial"</span>, <span>"cox"</span>), <br>                     Save_file = <span>"Scissor_inputs.RData"</span>, Load_file = <span>NULL</span>) <br>{<br>  <span>library</span>(Seurat)<br>  <span>library</span>(Matrix)<br>  <span>library</span>(preprocessCore)<br>  <span>if</span> (is.null(Load_file)) {<br>    common &lt;- intersect(rownames(bulk_dataset), rownames(sc_dataset))<br>    <span>if</span> (length(common) == <span>0</span>) {<br>      <span>stop</span>(<span>"There is no common genes between the given single-cell and bulk samples."</span>)<br>    }<br>    <span>if</span> (class(sc_dataset) == <span>"Seurat"</span>) {<br>      sc_exprs &lt;- as.matrix(sc_dataset@assays$RNA@data)<br>      network &lt;- as.matrix(sc_dataset@graphs$RNA_snn)<br>    }<br>    <span>else</span> {<br>      sc_exprs &lt;- as.matrix(sc_dataset)<br>      Seurat_tmp &lt;- CreateSeuratObject(sc_dataset)<br>      Seurat_tmp &lt;- FindVariableFeatures(Seurat_tmp, selection.method = <span>"vst"</span>, <br>                                         verbose = <span>F</span>)<br>      Seurat_tmp &lt;- ScaleData(Seurat_tmp, verbose = <span>F</span>)<br>      Seurat_tmp &lt;- RunPCA(Seurat_tmp, features = VariableFeatures(Seurat_tmp), <br>                           verbose = <span>F</span>)<br>      Seurat_tmp &lt;- FindNeighbors(Seurat_tmp, dims = <span>1</span>:<span>10</span>, <br>                                  verbose = <span>F</span>)<br>      network &lt;- as.matrix(Seurat_tmp@graphs$RNA_snn)<br>    }<br>    diag(network) &lt;- <span>0</span><br>    network[which(network != <span>0</span>)] &lt;- <span>1</span><br>    dataset0 &lt;- cbind(bulk_dataset[common, ], sc_exprs[common, <br>    ])<br>    dataset1 &lt;- normalize.quantiles(as.matrix(dataset0))<br>    rownames(dataset1) &lt;- rownames(dataset0)<br>    colnames(dataset1) &lt;- colnames(dataset0)<br>    Expression_bulk &lt;- dataset1[, <span>1</span>:ncol(bulk_dataset)]<br>    Expression_cell &lt;- dataset1[, (ncol(bulk_dataset) + <br>                                     <span>1</span>):ncol(dataset1)]<br>    X &lt;- cor(Expression_bulk, Expression_cell)<br>    quality_check &lt;- quantile(X)<br>    print(<span>"|**************************************************|"</span>)<br>    print(<span>"Performing quality-check for the correlations"</span>)<br>    print(<span>"The five-number summary of correlations:"</span>)<br>    print(quality_check)<br>    print(<span>"|**************************************************|"</span>)<br>    <span>if</span> (quality_check[<span>3</span>] &lt; <span>0.01</span>) {<br>      <span>warning</span>(<span>"The median correlation between the single-cell and bulk samples is relatively low."</span>)<br>    }<br>    <span>if</span> (family == <span>"binomial"</span>) {<br>      Y &lt;- as.numeric(phenotype)<br>      z &lt;- table(Y)<br>      <span>if</span> (length(z) != length(tag)) {<br>        <span>stop</span>(<span>"The length differs between tags and phenotypes. Please check Scissor inputs and selected regression type."</span>)<br>      }<br>      <span>else</span> {<br>        print(sprintf(<span>"Current phenotype contains %d %s and %d %s samples."</span>, <br>                      z[<span>1</span>], tag[<span>1</span>], z[<span>2</span>], tag[<span>2</span>]))<br>        print(<span>"Perform logistic regression on the given phenotypes:"</span>)<br>      }<br>    }<br>    <span>if</span> (family == <span>"gaussian"</span>) {<br>      Y &lt;- as.numeric(phenotype)<br>      z &lt;- table(Y)<br>      <span>if</span> (length(z) != length(tag)) {<br>        <span>stop</span>(<span>"The length differs between tags and phenotypes. Please check Scissor inputs and selected regression type."</span>)<br>      }<br>      <span>else</span> {<br>        tmp &lt;- paste(z, tag)<br>        print(paste0(<span>"Current phenotype contains "</span>, <br>                     paste(tmp[<span>1</span>:(length(z) - <span>1</span>)], collapse = <span>", "</span>), <br>                     <span>", and "</span>, tmp[length(z)], <span>" samples."</span>))<br>        print(<span>"Perform linear regression on the given phenotypes:"</span>)<br>      }<br>    }<br>    <span>if</span> (family == <span>"cox"</span>) {<br>      Y &lt;- as.matrix(phenotype)<br>      <span>if</span> (ncol(Y) != <span>2</span>) {<br>        <span>stop</span>(<span>"The size of survival data is wrong. Please check Scissor inputs and selected regression type."</span>)<br>      }<br>      <span>else</span> {<br>        print(<span>"Perform cox regression on the given clinical outcomes:"</span>)<br>      }<br>    }<br>    save(X, Y, network, Expression_bulk, Expression_cell, <br>         file = Save_file)<br>  }<br>  <span>else</span> {<br>    load(Load_file)<br>  }<br>  <span>if</span> (is.null(alpha)) {<br>    alpha &lt;- c(<span>0.005</span>, <span>0.01</span>, <span>0.05</span>, <span>0.1</span>, <span>0.2</span>, <span>0.3</span>, <span>0.4</span>, <span>0.5</span>, <br>               <span>0.6</span>, <span>0.7</span>, <span>0.8</span>, <span>0.9</span>)<br>  }<br>  <span>for</span> (i <span>in</span> <span>1</span>:length(alpha)) {<br>    set.seed(<span>123</span>)<br>    fit0 &lt;- APML1(X, Y, family = family, penalty = <span>"Net"</span>, <br>                  alpha = alpha[i], Omega = network, nlambda = <span>100</span>, <br>                  nfolds = min(<span>10</span>, nrow(X)))<br>    fit1 &lt;- APML1(X, Y, family = family, penalty = <span>"Net"</span>, <br>                  alpha = alpha[i], Omega = network, lambda = fit0$lambda.min)<br>    <span>if</span> (family == <span>"binomial"</span>) {<br>      Coefs &lt;- as.numeric(fit1$Beta[<span>2</span>:(ncol(X) + <span>1</span>)])<br>    }<br>    <span>else</span> {<br>      Coefs &lt;- as.numeric(fit1$Beta)<br>    }<br>    Cell1 &lt;- colnames(X)[which(Coefs &gt; <span>0</span>)]<br>    Cell2 &lt;- colnames(X)[which(Coefs &lt; <span>0</span>)]<br>    percentage &lt;- (length(Cell1) + length(Cell2))/ncol(X)<br>    print(sprintf(<span>"alpha = %s"</span>, alpha[i]))<br>    print(sprintf(<span>"Scissor identified %d Scissor+ cells and %d Scissor- cells."</span>, <br>                  length(Cell1), length(Cell2)))<br>    print(sprintf(<span>"The percentage of selected cell is: %s%%"</span>, <br>                  formatC(percentage * <span>100</span>, format = <span>"f"</span>, digits = <span>3</span>)))<br>    <span>if</span> (percentage &lt; cutoff) {<br>      <span>break</span><br>    }<br>    cat(<span>"\n"</span>)<br>  }<br>  print(<span>"|**************************************************|"</span>)<br>  <span>return</span>(list(para = list(alpha = alpha[i], lambda = fit0$lambda.min, <br>                          family = family), Coefs = Coefs, Scissor_pos = Cell1, <br>              Scissor_neg = Cell2))<br>}<br></code></pre><p data-tool="mdnice编辑器">再次运行：</p><pre data-tool="mdnice编辑器"><span></span><code>infos1 &lt;- Scissor(bulk_dataset, sc_dataset, phenotype, alpha = <span>0.05</span>, <br>                  family = <span>"cox"</span>, Save_file = <span>'./Scissor_survival.RData'</span>)<br><br>names(infos1)<br><span># [1] "para"        "Coefs"       "Scissor_pos" "Scissor_neg"</span><br>length(infos1$Scissor_pos)<br><span># [1] 429</span><br>infos1$Scissor_pos[<span>1</span>:<span>4</span>]<br>length(infos1$Scissor_neg)<br><span># 1001</span><br>infos1$Scissor_neg<br></code></pre><p data-tool="mdnice编辑器">运行大概三五分钟，在本地文件夹下即可生成结果Rdata，检查结果可见共识别429个Scissor+细胞（即与表型呈<strong>正相关</strong>的细胞）和1001个Scissor阴性细胞（即与表型呈<strong>负相关</strong>的细胞）。<strong>这里要强调一下Scissor R包的运行速度是真的快。而scAB运行这个8000个细胞的示例数据，需要花费好几个小时，效率上相差甚远。</strong></p><h4 data-tool="mdnice编辑器">Step4. 可视化结果</h4><pre data-tool="mdnice编辑器"><span></span><code>Scissor_select &lt;- rep(<span>0</span>, ncol(sc_dataset))<br>names(Scissor_select) &lt;- colnames(sc_dataset)<br>Scissor_select[infos1$Scissor_pos] &lt;- <span>"Scissor+"</span><br>Scissor_select[infos1$Scissor_neg] &lt;- <span>"Scissor-"</span><br>sc_dataset &lt;- AddMetaData(sc_dataset, metadata = Scissor_select, col.name = <span>"scissor"</span>)<br>UMAP_scissor &lt;- DimPlot(sc_dataset, reduction = <span>'umap'</span>, <br>        group.by = <span>'scissor'</span>,<br>        cols = c(<span>'grey'</span>,<span>'royalblue'</span>,<span>'indianred1'</span>), <br>        pt.size = <span>0.001</span>, order = c(<span>"Scissor+"</span>,<span>"Scissor-"</span>))<br>UMAP_scissor<br>patchwork::wrap_plots(plots = list(UMAP_celltype,UMAP_scissor), ncol = <span>2</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.439453125" data-src="https://mmbiz.qpic.cn/mmbiz/iaRJcrq2Los837JMPkzt0jIMqnbh4rlxvTV2s6DsHLDzMtpLxhzaO7gU8IvotzkmTtORVwoPCpgm8IzZqRWyHWg/640?wx_fmt=other" data-type="other" data-w="1024" src="https://mmbiz.qpic.cn/mmbiz/iaRJcrq2Los837JMPkzt0jIMqnbh4rlxvTV2s6DsHLDzMtpLxhzaO7gU8IvotzkmTtORVwoPCpgm8IzZqRWyHWg/640?wx_fmt=other"><figcaption>image-20230223180526286</figcaption></figure><p data-tool="mdnice编辑器">检查结果可知Scissor+ 细胞主要为肿瘤细胞，Scissor阴性细胞主要为T细胞和CAFs：</p><pre data-tool="mdnice编辑器"><span></span><code>table(sc_dataset$scissor,sc_dataset$celltype)<br><span>#           B cell  CAF HPC-like Malignant cell T cell  TAM  TEC</span><br><span># 0           907  718      728           1066   2405  652  947</span><br><span># Scissor-     85  225       78            101    358   52  102</span><br><span># Scissor+      1    4       35            337     29   20    3</span><br><br><span>library</span>(gplots)<br>balloonplot(table(sc_dataset$scissor,sc_dataset$celltype))<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.49614197530864196" data-src="https://mmbiz.qpic.cn/mmbiz/iaRJcrq2Los837JMPkzt0jIMqnbh4rlxviciaFLlmDpI5y2vNHbVlicW0oIQ8lYHxLEswPNReT6s7jicOOTO97fnP2A/640?wx_fmt=other" data-type="other" data-w="1296" src="https://mmbiz.qpic.cn/mmbiz/iaRJcrq2Los837JMPkzt0jIMqnbh4rlxviciaFLlmDpI5y2vNHbVlicW0oIQ8lYHxLEswPNReT6s7jicOOTO97fnP2A/640?wx_fmt=other"><figcaption>image-20230223180753824</figcaption></figure><h4 data-tool="mdnice编辑器">Step5. 对比scAB的结果</h4><p data-tool="mdnice编辑器">加载上期scAB的结果：</p><pre data-tool="mdnice编辑器"><span></span><code>scAB.res = read_rds(<span>"scAB.res.rds"</span>)<br>table(row.names(sc_dataset@meta.data) == row.names(scAB.res@meta.data))<br><span>#TRUE </span><br><span>#8853 </span><br>sc_dataset$scAB = scAB.res$scAB_select<br></code></pre><p data-tool="mdnice编辑器">可视化：</p><pre data-tool="mdnice编辑器"><span></span><code>UMAP_scAB &lt;- DimPlot(sc_dataset,group.by=<span>"scAB"</span>,<br>                     cols=c(<span>"#80b1d3"</span>,<span>"red"</span>),<br>                     pt.size=<span>0.001</span>,<br>                     order=c(<span>"scAB+ cells"</span>,<span>"Other cells"</span>))<br><br>patchwork::wrap_plots(plots = list(UMAP_celltype,UMAP_scissor,UMAP_scAB), ncol = <span>3</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.303583977512298" data-src="https://mmbiz.qpic.cn/mmbiz/iaRJcrq2Los837JMPkzt0jIMqnbh4rlxvPyJibtjF1ZHLAe46duDHjgplfAZ0kibqoSQCLO7EnYK4uwl9WLdrO7rQ/640?wx_fmt=other" data-type="other" data-w="1423" src="https://mmbiz.qpic.cn/mmbiz/iaRJcrq2Los837JMPkzt0jIMqnbh4rlxvPyJibtjF1ZHLAe46duDHjgplfAZ0kibqoSQCLO7EnYK4uwl9WLdrO7rQ/640?wx_fmt=other"><figcaption>image-20230223180558402</figcaption></figure><p data-tool="mdnice编辑器">取交集：</p><pre data-tool="mdnice编辑器"><span></span><code>balloonplot(table(sc_dataset$scissor,sc_dataset$scAB))<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.4278688524590164" data-src="https://mmbiz.qpic.cn/mmbiz/iaRJcrq2Los837JMPkzt0jIMqnbh4rlxvt7pKksXiazKF9ONRpgskoaqpMwh3ZSkiamjCdYPa0skgU0nGicqf4274g/640?wx_fmt=other" data-type="other" data-w="1220" src="https://mmbiz.qpic.cn/mmbiz/iaRJcrq2Los837JMPkzt0jIMqnbh4rlxvt7pKksXiazKF9ONRpgskoaqpMwh3ZSkiamjCdYPa0skgU0nGicqf4274g/640?wx_fmt=other"><figcaption>image-20230223180919836</figcaption></figure><p data-tool="mdnice编辑器">只有143个细胞是scAB+ Scissor+细胞。个人感觉结果相差还是有点大的。</p><p data-tool="mdnice编辑器"><strong>总结来说：</strong></p><ul data-tool="mdnice编辑器"><li><section>scAB运行较慢（8853个细胞需2小时左右），但是似乎会得到更多的表型相关阳性细胞（915/8853 vs. 429/8853），<strong>另外scAB可以整合scATAC-seq数据</strong>；</section></li><li><section>而<strong>Scissor运行很快</strong>（8853个细胞需5分钟左右），但是似乎会得到更少的表型相关阳性细胞，另外Scissor还提供了表型相关阴性细胞。</section></li></ul><span>- END -</span></section><p><br></p><p><mp-style-type data-value="10000"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/bObgu2Z90_qd5GYaVn4oJw",target="_blank" rel="noopener noreferrer">原文链接</a>
