---
title: "scRNA | CSOmap-R版 利用单细胞转录组预测细胞类型的空间通讯"
date: 2024-08-20T01:12:35Z
draft: ["false"]
tags: [
  "fetched",
  "生信补给站"
]
categories: ["Acdemic"]
---
scRNA | CSOmap-R版 利用单细胞转录组预测细胞类型的空间通讯 by 生信补给站
------
<div><p><span>前面介绍了<a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzIyNDI1MzgzOQ==&amp;mid=2650400060&amp;idx=1&amp;sn=3710673aa3708721b54fb36e0bf85a67&amp;chksm=f01c87dcc76b0eca4c906e8186c500a4b8f47f9d5d5bef1b730d33b5031ae76a0897332708da&amp;scene=21#wechat_redirect" textvalue="scRNA分析 | 解决可能的报错，从0开始教你完成细胞通讯分析-cellphoneDB" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">scRNA分析 | 解决可能的报错，从0开始教你完成细胞通讯分析-cellphoneDB</a>cellphoneDB 和 <a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzIyNDI1MzgzOQ==&amp;mid=2650400513&amp;idx=1&amp;sn=49119e2d60aa7de239e9bd06ffe0aec3&amp;chksm=f01c85e1c76b0cf7a4c961902475c9209add74ff13f170a299bdd63975f6199c7c5911b698f2&amp;scene=21#wechat_redirect" textvalue="scRNA分析|使用CellChat完成细胞通讯分析-简单且可视化出众，代码自取" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">scRNA分析|使用CellChat完成细胞通讯分析-简单且可视化出众，代码自取</a>进行细胞间通讯的分析，这里介绍<span>张泽民/任仙文课题组在《</span><strong><span>Cell Research</span></strong><span>》杂志上发表题为＂</span><strong><span>Reconstruction of cell spatial organization from single-cell RNA sequencing data based on ligand-receptor mediated self-assembly</span></strong><span>＂的文章，提出了利用</span></span><span>单细胞转录组数据</span><span>预测</span><span>细胞空间关系</span><span>的全新生物信息学方法。</span></p><p><img data-imgfileid="502918787" data-ratio="0.8611111111111112" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/Y21ubvXhTjCY7ibnQIOJu8fWiarBibvcFM2YNZOFpfpHaVaY1MRmmP9upic7yIvCbjMAWygrX7TqWcAHAIrGMFjGeQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/Y21ubvXhTjCY7ibnQIOJu8fWiarBibvcFM2YNZOFpfpHaVaY1MRmmP9upic7yIvCbjMAWygrX7TqWcAHAIrGMFjGeQ/640?wx_fmt=png&amp;from=appmsg"></p><p><span>CSOmap的创新性体现在其可以回答细胞空间关系是由哪些关键的配体-受体相互作用决定的。</span><span>官方的github路径中给出了R版本以及matlab版本，本推文</span><span>只介绍R版本</span><span>的使用方法。</span></p><section data-mpa-template="t" mpa-from-tpl="t"><section mpa-from-tpl="t"><section><section mpa-from-tpl="t"><p><span><span>一 载入R包，</span>数据</span></p></section></section></section></section><p><br></p><p><span>注意根据作者说明当前windows下可能不适用 ，不确定是否已优化。</span></p><blockquote data-type="2" data-url="" data-author-name="" data-content-utf8-length="125" data-source-title=""><section><p>R package for CSOmap(developing)Right now, CSOmapR is only available on the linux-based machines. Installation on windows may encounter errors.</p></section></blockquote><section><ul><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="cpp"><code><span># install.packages("devtools")</span></code><code><span>devtools::install_github("lijxug/CSOmapR")</span></code><code><span># install CSOmapR.demo package for easy-loading demo data</span></code><code><span># devtools::install_github("lijxug/CSOmapR.demo")</span></code><code><span>library(CSOmapR)</span></code><code><span>library(CSOmapR.demo)</span></code><code><span>##可以查看数据形式</span></code><code><span>invisible(TPM) </span></code><code><span>invisible(LR)</span></code><code><span>invisible(labelData)</span></code></pre></section><p><span>安装R包的过程中出现报错就根据缺少的包进行安装，网络问题的可以下载下来然后本地安装。</span></p><h3></h3><h4><span><strong>1. 数据处理-单细胞矩阵（TPM）</strong></span></h4><p><span>载入单细胞对象，这里示例数据抽取5%比例的数据，根据资源情况自行决定</span></p><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="sql"><code><span>load("sc_cellchat.Rdata")</span></code><code><span><br></span></code><code><span>set.seed(1234)</span></code><code><span>labelData_in &lt;- sc_cellchat@meta.data %&gt;%</span></code><code><span>  dplyr::select(celltype) %&gt;%</span></code><code><span>  sample_frac(0.05)</span></code><code><span>#抽取细胞</span></code><code><span>sc_cellchat_in &lt;- subset(sc_cellchat, barcode %in% rownames(labelData_in) )</span></code><code><span>#TPM数据</span></code><code><span>TPM_in=as.data.frame(sc_cellchat_in[["RNA"]]@counts) %&gt;% </span></code><code><span>  apply(2,function(x){x/sum(x) * 10000})</span></code><code><span>TPM_in[1:4,1:4]</span></code><code><span>#           Li1_AACCATGTCATTCGTT-1 Li1_AAGCATCGTTTCACAG-1 Li1_AAGGAATGTGGTCTAT-1 Li1_ACAACCAGTCCTACGG-1</span></code><code><span>#MIR1302-2HG                      0                      0                      0                      0</span></code><code><span>#FAM138A                          0                      0                      0                      0</span></code><code><span>#OR4F5                            0                      0                      0                      0</span></code><code><span>#AL627309.1                       0                      0                      0                      0</span></code></pre></section><p><img data-galleryid="" data-imgfileid="502918796" data-ratio="0.08148148148148149" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/Y21ubvXhTjCY7ibnQIOJu8fWiarBibvcFM2gKe7r8Oiar6y9HbadcxVKlVCgDIR53bmwxISfIbTMU0TcypjwoGVSCw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/Y21ubvXhTjCY7ibnQIOJu8fWiarBibvcFM2gKe7r8Oiar6y9HbadcxVKlVCgDIR53bmwxISfIbTMU0TcypjwoGVSCw/640?wx_fmt=png&amp;from=appmsg"></p><h4><span></span></h4><h4><span><strong>2. 数据处理-</strong></span><strong>标签信息</strong></h4><p><span>共两列信息，分别是与TPM信息对应的barcode 以及 标签信息（celltype 或者 cluster均可）</span></p><section><ul><li><li><li><li><li><li><li></ul><pre data-lang="css"><code><span><span>labelData_in2</span> &lt;<span>-</span> <span>labelData_in</span> %&gt;%</span></code><code><span>  <span>rownames_to_column</span>("<span>cells</span>") </span></code><code><span><span>head</span>(<span>labelData_in2</span>,2)</span></code><code><span><br></span></code><code><span># <span>cells</span> <span>celltype</span></span></code><code><span><span>#Li2_TCCGATCTCGCTAATG-1</span> <span>CD8</span><span>.GZMK</span></span></code><code><span><span>#Li2_TGCGGCAGTTCTCGTC-1</span>   <span>CD4</span><span>.Tn</span></span></code></pre></section><h4><span></span></h4><h4><span><strong>3. 数据处理-</strong></span><strong>配体受体LR文件</strong><span>（使用内置的1961个配体受体对）</span></h4><section><ul><li><li><li><li><li><li><li><li></ul><pre data-lang="properties"><code><span><span>head(LR)</span></span></code><code><span>      <span>V1</span>     <span>V2 V3</span></span></code><code><span><span>1</span>    <span>A2M   LRP1  1</span></span></code><code><span><span>2</span>  <span>AANAT MTNR1A  1</span></span></code><code><span><span>3</span>  <span>AANAT MTNR1B  1</span></span></code><code><span><span>4</span>    <span>ACE  AGTR2  1</span></span></code><code><span><span>5</span>    <span>ACE BDKRB2  1</span></span></code><code><span><span>6</span> <span>ADAM10    AXL  1</span></span></code></pre></section><p><br></p><h3></h3><section data-mpa-template="t" mpa-from-tpl="t"><section mpa-from-tpl="t"><section><section mpa-from-tpl="t"><p><span>二 CSOmapR分析</span></p></section></section></section></section><p><span><br></span></p><p><strong><span>1. CSOmap分析</span></strong><br></p><p><span>分析函数按照github，使用runExactTSNE_R函数得到解析结果后，进行数据的处理</span></p><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="makefile"><code><span>affinityMat = getAffinityMat(as.matrix(TPM_in), LR, verbose = T)</span></code><code><span><br></span></code><code><span>coords_res = runExactTSNE_R(</span></code><code><span>  X = affinityMat,</span></code><code><span>  no_dims = 3,</span></code><code><span>  max_iter = 1000,</span></code><code><span>  verbose = T</span></code><code><span>)</span></code><code><span><br></span></code><code><span><span>#数据处理</span></span></code><code><span>coords = coords_res$Y</span></code><code><span>rownames(coords) &lt;- colnames(TPM)</span></code><code><span>colnames(coords) &lt;- c('x', 'y', 'z')</span></code><code><span><br></span></code><code><span><span># arrange data</span></span></code><code><span>coords_tbl = bind_cols(cellName = rownames(coords), as.data.frame(coords))</span></code><code><span><br></span></code><code><span>join_vec = setNames(colnames(labelData)[1], nm = colnames(coords_tbl)[1])</span></code><code><span>cellinfo_tbl = left_join(coords_tbl, labelData, by = join_vec)</span></code><code><span><br></span></code><code><span>density_obj = getDensity3D(cellinfo_tbl$x, cellinfo_tbl$y, cellinfo_tbl$z)</span></code><code><span>cellinfo_tbl = cellinfo_tbl %&gt;% mutate(density = density_obj)</span></code><code><span><br></span></code></pre></section><p><img data-galleryid="" data-imgfileid="502918798" data-ratio="0.24074074074074073" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/Y21ubvXhTjCY7ibnQIOJu8fWiarBibvcFM2V9mKGuYrR6cS8L6Q4H9hbDHxb0gNtQDxPkCoA3A6FHDYjBcfiaPo3Tg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/Y21ubvXhTjCY7ibnQIOJu8fWiarBibvcFM2V9mKGuYrR6cS8L6Q4H9hbDHxb0gNtQDxPkCoA3A6FHDYjBcfiaPo3Tg/640?wx_fmt=png&amp;from=appmsg"></p><h4><strong><span>2，可视化</span></strong></h4><p><span>（1）按照github中的density 填充颜色</span></p><section><ul><li><li></ul><pre data-lang="makefile"><code><span>p_3Ddensity = plot3D(cellinfo_tbl, color_by = "density", title = "3D density")</span></code><code><span>p_3Ddensity</span></code></pre></section><p><img data-imgfileid="502918788" data-ratio="0.8546296296296296" data-src="https://mmbiz.qpic.cn/sz_mmbiz_jpg/Y21ubvXhTjCY7ibnQIOJu8fWiarBibvcFM2r0T8pdGQJvJIQvygIlQOMsx4YRINBlYex5GqOgwYDAQtrP4nuxVObg/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_jpg/Y21ubvXhTjCY7ibnQIOJu8fWiarBibvcFM2r0T8pdGQJvJIQvygIlQOMsx4YRINBlYex5GqOgwYDAQtrP4nuxVObg/640?wx_fmt=other&amp;from=appmsg"></p><p><span>（2）按照注释的celltype 填充颜色</span></p><section><ul><li><li><li><li><li><li></ul><pre data-lang="makefile"><code><span>p_3Ddensity2 = plot3D(cellinfo_tbl, color_by = "celltype", title = "3D density",alpha = 0.8)</span></code><code><span>p_3Ddensity2</span></code><code><span>#Warning messages:</span></code><code><span>#1: In RColorBrewer::brewer.pal(N, "Set2") :</span></code><code><span>#  n too large, allowed maximum for palette Set2 is 8</span></code><code><span>#Returning the palette you asked for with that many colors</span></code></pre></section><p><img data-galleryid="" data-imgfileid="502918799" data-ratio="0.8759259259259259" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/Y21ubvXhTjCY7ibnQIOJu8fWiarBibvcFM2qzW6XRaBdMo03v3rpOsCMhasm8whF6SQxoBLagoBWkXhOqQG0duZvQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/Y21ubvXhTjCY7ibnQIOJu8fWiarBibvcFM2qzW6XRaBdMo03v3rpOsCMhasm8whF6SQxoBLagoBWkXhOqQG0duZvQ/640?wx_fmt=png&amp;from=appmsg"></p><p><span>（3）自定义颜色</span></p><p><span>直接使用plot3D中的colors 指定颜色无效，于是<span>通过View(plot3D)查看源函数，可以直接使用plotly函数自定义为与umap图相同的颜色。</span></span></p><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="php"><code><span>zzm60colors &lt;- c(<span>'#4b6aa8'</span>,<span>'#3ca0cf'</span>,<span>'#c376a7'</span>,<span>'#ad98c3'</span>,<span>'#cea5c7'</span>,</span></code><code><span>                 <span>'#d69971'</span>,<span>'#df5734'</span>,<span>'#6c408e'</span>,<span>'#ac6894'</span>,<span>'#d4c2db'</span>,</span></code><code><span>                 <span>'#537eb7'</span>,<span>'#83ab8e'</span>,<span>'#ece399'</span>,<span>'#405993'</span>,<span>'#cc7f73'</span>,</span></code><code><span>                 <span>'#b95055'</span>,<span>'#d5bb72'</span>,<span>'#bc9a7f'</span>,<span>'#e0cfda'</span>,<span>'#d8a0c0'</span>,</span></code><code><span>                 <span>'#e6b884'</span>,<span>'#b05545'</span>,<span>'#d69a55'</span>,<span>'#64a776'</span>,<span>'#cbdaa9'</span>,</span></code><code><span>                 <span>'#efd2c9'</span>,<span>'#da6f6d'</span>,<span>'#ebb1a4'</span>,<span>'#a44e89'</span>,<span>'#a9c2cb'</span>,</span></code><code><span>                 <span>'#b85292'</span>,<span>'#6d6fa0'</span>,<span>'#8d689d'</span>,<span>'#c8c7e1'</span>,<span>'#d25774'</span>,</span></code><code><span>                 <span>'#c49abc'</span>,<span>'#927c9a'</span>,<span>'#3674a2'</span>,<span>'#9f8d89'</span>,<span>'#72567a'</span>,</span></code><code><span>                 <span>'#63a3b8'</span>,<span>'#c4daec'</span>,<span>'#61bada'</span>,<span>'#b7deea'</span>,<span>'#e29eaf'</span>,</span></code><code><span>                 <span>'#4490c4'</span>,<span>'#e6e2a3'</span>,<span>'#de8b36'</span>,<span>'#c4612f'</span>,<span>'#9a70a8'</span>,</span></code><code><span>                 <span>'#76a2be'</span>,<span>'#408444'</span>,<span>'#c6adb0'</span>,<span>'#9d3b62'</span>,<span>'#2d3462'</span>)</span></code><code><span><br></span></code><code><span>fig_density = plotly::plot_ly(cellinfo_tbl, x = ~x, y = ~y, z = ~z, </span></code><code><span>                              alpha = <span>0.8</span>)</span></code><code><span><br></span></code><code><span><span># 使用add_markers函数，指定自定义颜色</span></span></code><code><span>fig_density &lt;- plotly::add_markers(fig_density, </span></code><code><span>                                   color = <span>eval</span>(parse(text = sprintf(<span>"~%s"</span>, </span></code><code><span>                                                                     <span>"celltype"</span>))),</span></code><code><span>                                   colors = zzm60colors)</span></code><code><span><br></span></code><code><span>fig_density</span></code></pre></section><p><img data-galleryid="" data-imgfileid="502918800" data-ratio="0.9120370370370371" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/Y21ubvXhTjCY7ibnQIOJu8fWiarBibvcFM2M2QeDqpHJcAjCtqu1qNicnWEtbKpDSqdE5uBbicVt7JuXgjBJwxd5jUg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/Y21ubvXhTjCY7ibnQIOJu8fWiarBibvcFM2M2QeDqpHJcAjCtqu1qNicnWEtbKpDSqdE5uBbicVt7JuXgjBJwxd5jUg/640?wx_fmt=png&amp;from=appmsg"></p><p><strong><span>3，细胞交互结果</span></strong><br></p><p><span>除了上述3D的plot图之外，还可以通过getSignificance函数得到显著通讯的细胞类型，可以看结果的qvalue，低于0.05的可以认为有可能有空间的接触。</span></p><pre data-language="r"></pre><section><ul><li><li><li><li><li></ul><pre data-lang="makefile"><code><span>signif_results = getSignificance(coords,</span></code><code><span>                                 labels = cellinfo_tbl$celltype,</span></code><code><span>                                 verbose = T)</span></code><code><span>contribution_list = getContribution(TPM_in, LR, signif_results$detailed_connections)</span></code><code><span><br></span></code></pre></section><p><img data-galleryid="" data-imgfileid="502918802" data-ratio="0.21666666666666667" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/Y21ubvXhTjCY7ibnQIOJu8fWiarBibvcFM2tiaiaibHV3gePhAdFy5UuEGbqfVic1bPAtvpJh1vbUBQtJOUkYvuv0veZw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/Y21ubvXhTjCY7ibnQIOJu8fWiarBibvcFM2tiaiaibHV3gePhAdFy5UuEGbqfVic1bPAtvpJh1vbUBQtJOUkYvuv0veZw/640?wx_fmt=png&amp;from=appmsg"></p><p><span>简单的绘制热图，当然可以更多的优化</span><br></p><section><ul><li></ul><pre data-lang="php"><code><span>pheatmap::pheatmap(signif_results$qvalue,display_numbers = T,number_format = "%.1e")</span></code></pre></section><p><img data-galleryid="" data-imgfileid="502918801" data-ratio="0.49537037037037035" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/Y21ubvXhTjCY7ibnQIOJu8fWiarBibvcFM2SbeQOMd2lzblsfqAdYMEj57cweGkECKskx2vh3HI9ahvK5RjrH8nIA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/Y21ubvXhTjCY7ibnQIOJu8fWiarBibvcFM2SbeQOMd2lzblsfqAdYMEj57cweGkECKskx2vh3HI9ahvK5RjrH8nIA/640?wx_fmt=png&amp;from=appmsg"></p><p><span><span>OK，R版本的CSOmap 目前就开发到这里了。</span></span></p><p><span><strong><span>文献中还提到的关键的配体-受体相互作用 以及 扰动后的结果暂未找到</span></strong><strong><span>。</span></strong></span></p><p><span><strong><span>参考资料：</span></strong></span></p><p><span>[1]Reconstruction of cell spatial organization from single-cell RNA sequencing data based on ligand-receptor mediated self-assembly</span></p><p><span>GitHub - lijxug/CSOmapR: R package of CSOmap(developing</span></p><p><br></p><section data-style-type="7" data-tools="新媒体排版" data-id="9190"><p><span><span data-txtless="spin" data-txtlessp="120">◆ </span><span>◆ </span><span data-txtless="spin" data-txtlessp="-120">◆  <span data-txtless="spin" data-txtlessp="120">◆ </span><span>◆</span></span></span></p><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzIyNDI1MzgzOQ==&amp;mid=2650398215&amp;idx=1&amp;sn=730225fb5ba3a51ceb8df0a531632515&amp;chksm=f01cbce7c76b35f1bebbf327d78a51d859770a46ae1329d362e3b705c4f8765ee2dc461d24dc&amp;scene=21#wechat_redirect" data-itemshowtype="0" tab="innerlink" data-linktype="2" wah-hotarea="click" hasload="1">精心整理（含图PLUS版）|R语言生信分析，可视化</a><span>（R统计，ggplot2绘图，生信图形可视化汇总）</span></p></section><p><span><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzIyNDI1MzgzOQ==&amp;mid=2650400742&amp;idx=1&amp;sn=dca078db8e6c0c742264dcb0bf56a8e2&amp;chksm=f01c8506c76b0c107782efb89c5f5c0e244a9d664c15ea2b9d16dc70067b7249bd0e7ea9f9ae&amp;scene=21#wechat_redirect" textvalue="RNAseq纯生信挖掘思路分享？不，主要是送你代码！（建议收藏）" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2" hasload="1">RNAseq纯生信挖掘思路分享？不，主要是送你代码！（建议收藏）</a></span><span></span><br></p><section><mp-common-profile data-pluginname="mpprofile" data-id="MzIyNDI1MzgzOQ==" data-headimg="http://mmbiz.qpic.cn/mmbiz_png/Y21ubvXhTjCh1HyEjoPcBiaKy86NdKGZtju9o0oDBPicVK0lcoovRfticNuwPSj0Kib8cxqaVfBcTVCDAJ5icptybsg/300?wx_fmt=png&amp;wxfrom=19" data-nickname="生信补给站" data-alias="Bioinfo_R_Python" data-signature="生信，R语言， Python，数据处理、统计检验、模型构建、数据可视化，我输出您输入！" data-from="2" data-origin_num="181" data-isban="0" data-biz_account_status="0" data-index="0" data-is_biz_ban="0"></mp-common-profile></section><section><br></section><section><span>觉得对您有点帮助的希望可以点赞，在看，转发！</span></section><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/wy3A-S-HYikVZX89-qfnMA",target="_blank" rel="noopener noreferrer">原文链接</a>
