---
title: "全网最全最准确的肿瘤样本药物敏感性预测教程"
date: 2024-08-09T02:27:54Z
draft: ["false"]
tags: [
  "fetched",
  "R语言学徒"
]
categories: ["Acdemic"]
---
全网最全最准确的肿瘤样本药物敏感性预测教程 by R语言学徒
------
<div><p data-mpa-powered-by="yiban.io"><img data-cropselx1="0" data-cropselx2="578" data-cropsely1="0" data-cropsely2="135" data-galleryid="" data-ratio="0.2222222222222222" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEUckerRictfTicSHoNskkHZjoa7nSeGI9ticg8pf3FOEkewHQTNLbNhdKK36fxBPpNLzVm0W8LjZYKEg/640?wx_fmt=png&amp;from=appmsg" data-type="jpeg" data-w="900" src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEUckerRictfTicSHoNskkHZjoa7nSeGI9ticg8pf3FOEkewHQTNLbNhdKK36fxBPpNLzVm0W8LjZYKEg/640?wx_fmt=png&amp;from=appmsg"></p><p><span>今天师妹给大家带来的分享也非常实用，利用pRRophetic包基于两个药物敏感性数据库PRISM和CTRP进行肿瘤样本药物敏感性预测，通过AUC值作为药物敏感性的衡量标准，较低的 AUC 值表明对治疗的敏感性增加，这样预测的结果会更加准确和全面，有需要的小伙伴赶紧行动起来吧，跟着师妹一起开始今天的实操学习吧！</span></p><p><strong><span>1.加载需要的R包</span></strong><span></span></p><section><ul><li><li><li><li><li><li></ul><pre data-lang="bash"><code><span>library(tidyverse)</span></code><code><span>library(impute) <span># 用于KNN填补药敏数据</span></span></code><code><span>library(pRRophetic) <span># 用于药敏预测</span></span></code><code><span>library(SimDesign) <span># 用于禁止药敏预测过程输出的信息</span></span></code><code><span>library(ggplot2) <span># 绘图</span></span></code><code><span>library(cowplot) <span>#拼图</span></span></code></pre></section><p><strong><span>2.数据读取与处理</span></strong><span></span></p><section><ul><li><li></ul><pre data-lang="xml"><code><span># 读取肝癌TPM表达谱，行名为基因名，列名为样本名。</span></code><code><span>expr <span>&lt;<span>-</span> <span>read.table</span>("<span>LIHC.TPM.txt</span>",<span>sep</span> = <span>"\t"</span>,<span>row.names</span> = <span>1,check.names</span> = <span>F,stringsAsFactors</span> = <span>F,header</span> = <span>T)</span></span></span></code></pre></section><p><img data-imgfileid="100005988" data-ratio="0.24814814814814815" data-src="https://mmbiz.qpic.cn/sz_mmbiz_jpg/oCSOtP61aEWNEG0fFLrTUKtCS0fZKXcCpml5Wuz2Sn6ZudHSoZwSicwJQcEaGWsO3PKB8eiaoiaJukfUToAX6dia7w/640?wx_fmt=jpeg" data-type="jpeg" data-w="1080" height="137.5999755859375" width="553.6099853515625" src="https://mmbiz.qpic.cn/sz_mmbiz_jpg/oCSOtP61aEWNEG0fFLrTUKtCS0fZKXcCpml5Wuz2Sn6ZudHSoZwSicwJQcEaGWsO3PKB8eiaoiaJukfUToAX6dia7w/640?wx_fmt=jpeg"><span></span></p><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="bash"><code><span>normsam &lt;- colnames(expr[,<span>which</span>(substr(colnames(expr),11,12) == <span>"11"</span>)])</span></code><code><span>tumosam &lt;- colnames(expr[,<span>which</span>(substr(colnames(expr),11,12) == <span>"01"</span>)])</span></code><code><span><span># 读取maf突变文件(于cBioPortal下载)</span></span></code><code><span>maf &lt;- read_tsv(<span>"data_mutations_mskcc.txt"</span>, comment = <span>"#"</span>)</span></code><code><span>maf<span>$Tumor_Sample_Barcode</span> &lt;- paste0(<span>"LIHC"</span>,substr(maf<span>$Tumor_Sample_Barcode</span>,8,15))</span></code><code><span><span># 提取既有表达数据又有突变数据的肿瘤样本    </span></span></code><code><span>tumosam &lt;- intersect(tumosam,unique(maf<span>$Tumor_Sample_Barcode</span>))</span></code><code><span>maf &lt;- maf[<span>which</span>(maf<span>$Tumor_Sample_Barcode</span> %<span>in</span>% tumosam),]</span></code><code><span>expr &lt;- expr[,c(tumosam,normsam)]</span></code><code><span><span># 提取TP53突变信息，并创建样本注释</span></span></code><code><span>tp53 &lt;- c()</span></code><code><span><span>for</span> (i <span>in</span> tumosam) {</span></code><code><span>  tmp &lt;- maf[<span>which</span>(maf<span>$Tumor_Sample_Barcode</span> == i),]</span></code><code><span>  <span>if</span>(is.element(<span>"TP53"</span>, tmp<span>$Hugo_Symbol</span>)) { <span># 如果存在TP53</span></span></code><code><span>    tp53 &lt;- c(tp53,1) <span># 记录1</span></span></code><code><span>  } <span>else</span> {</span></code><code><span>    tp53 &lt;- c(tp53,0) <span># 否则记录0</span></span></code><code><span>  }</span></code><code><span>}</span></code><code><span>names(tp53) &lt;- tumosam</span></code><code><span><span># 取出有TP53突变的患者</span></span></code><code><span>tp53.mutsam &lt;- names(tp53[<span>which</span>(tp53 == 1)])</span></code><code><span><span># 1.读取17-gene signature，计算PPS得分(原文Table S3)</span></span></code><code><span>signature &lt;- read.table(<span>"17gene.txt"</span>,sep = <span>"\t"</span>,row.names = 1,check.names = F,stringsAsFactors = F,header = T)</span></code><code><span>dim(signature)</span></code><code><span><span># 计算pps，用到TP53突变患者的17个基因的表达矩阵</span></span></code><code><span>pps &lt;- as.numeric(apply(t(log2(expr[rownames(signature),tp53.mutsam] + 1)), 1, <span>function</span>(x) {x %*% signature<span>$Coefficient</span>}))</span></code><code><span><span># 标准化，把pps处理到0-1之间</span></span></code><code><span>range01 &lt;- <span>function</span>(x){(x-min(x))/(max(x)-min(x))}</span></code><code><span>npps &lt;- range01(pps)</span></code><code><span><span># 创建样本信息</span></span></code><code><span>Sinfo &lt;- data.frame(PPS = npps,</span></code><code><span>                    TP53 = tp53[tp53.mutsam],</span></code><code><span>                    row.names = tp53.mutsam,</span></code><code><span>                    stringsAsFactors = F)</span></code><code><span><span># 把pps保存到文件</span></span></code><code><span>write.csv(Sinfo, <span>"output_PPS.csv"</span>, quote = F)</span></code><code><span>normexpr &lt;- as.matrix(expr[,normsam])    </span></code><code><span>tumoexpr &lt;- as.matrix(expr[,tp53.mutsam])</span></code><code><span>           </span></code><code><span>runpure &lt;- F <span># 如果想运行就把这个改为T</span></span></code><code><span><span>if</span>(runpure) {</span></code><code><span>  set.seed(123)</span></code><code><span>  <span># Run ISOpureR Step 1 - Cancer Profile Estimation</span></span></code><code><span>  ISOpureS1model &lt;- ISOpure.step1.CPE(tumoexpr, normexpr)</span></code><code><span>  <span># For reproducible results, set the random seed</span></span></code><code><span>  set.seed(456);</span></code><code><span>  <span># Run ISOpureR Step 2 - Patient Profile Estimation</span></span></code><code><span>  ISOpureS2model &lt;- ISOpure.step2.PPE(tumoexpr,normexpr,ISOpureS1model)</span></code><code><span>  pure.tumoexpr &lt;- ISOpureS2model<span>$cc_cancerprofiles</span></span></code><code><span>}</span></code><code><span>           </span></code><code><span><span>if</span>(!runpure) {</span></code><code><span>  pure.tumoexpr &lt;- tumoexpr</span></code><code><span>}</span></code><code><span><span># 制作CTRP AUC矩阵，保存到CTRP_AUC.txt文件</span></span></code><code><span>auc &lt;- read.table(<span>"CTRP_AUC_raw.txt"</span>,sep = <span>"\t"</span>,row.names = NULL,check.names = F,stringsAsFactors = F,header = T) <span># Supplementary Data Set 3</span></span></code><code><span>auc<span>$comb</span> &lt;- paste(auc<span>$master_cpd_id</span>,auc<span>$master_ccl_id</span>,sep = <span>"-"</span>)</span></code><code><span>auc &lt;- apply(auc[,<span>"area_under_curve"</span>,drop = F], 2, <span>function</span>(x) tapply(x, INDEX=factor(auc<span>$comb</span>), FUN=max, na.rm=TRUE)) <span># 重复项取最大AUC</span></span></code><code><span>auc &lt;- as.data.frame(auc)</span></code><code><span>auc<span>$master_cpd_id</span> &lt;- sapply(strsplit(rownames(auc),<span>"-"</span>,fixed = T),<span>"["</span>,1)</span></code><code><span>auc<span>$master_ccl_id</span> &lt;- sapply(strsplit(rownames(auc),<span>"-"</span>,fixed = T),<span>"["</span>,2)</span></code><code><span>auc &lt;- reshape(auc,</span></code><code><span>               direction = <span>"wide"</span>,</span></code><code><span>               timevar = <span>"master_cpd_id"</span>,</span></code><code><span>               idvar = <span>"master_ccl_id"</span>)</span></code><code><span>colnames(auc) &lt;- gsub(<span>"area_under_curve."</span>,<span>""</span>,colnames(auc),fixed = T)</span></code><code><span>ctrp.ccl.anno &lt;- read.table(<span>"CTRP_ccl_anno.txt"</span>,sep = <span>"\t"</span>,row.names = NULL,check.names = F,stringsAsFactors = F,header = T) <span># Supplementary Data Set 1</span></span></code><code><span>ctrp.cpd.anno &lt;- read.delim(<span>"CTRP_cpd_anno.txt"</span>,sep = <span>"\t"</span>,row.names = NULL,check.names = F,stringsAsFactors = F,header = T) <span># Supplementary Data Set 2</span></span></code><code><span><span># 保存到文件    </span></span></code><code><span>write.table(auc,<span>"CTRP_AUC.txt"</span>,sep = <span>"\t"</span>,row.names = F,col.names = T,quote = F)</span></code><code><span><span># 加载药敏AUC矩阵并进行数据处理</span></span></code><code><span>ctrp.auc &lt;- read.table(<span>"CTRP_AUC.txt"</span>,sep = <span>"\t"</span>,row.names = 1,check.names = F,stringsAsFactors = F,header = T)</span></code><code><span>prism.auc &lt;- read.delim(<span>"PRISM_AUC.txt"</span>,sep = <span>"\t"</span>,row.names = 1,check.names = F,stringsAsFactors = F,header = T) <span># 数据来自https://depmap.org/portal/download/ Drug sensitivity AUC (PRISM Repurposing Secondary Screen) 19Q4</span></span></code><code><span>prism.ccl.anno &lt;- prism.auc[,1:5] <span># 前5列为细胞系注释信息</span></span></code><code><span>prism.auc &lt;- prism.auc[,-c(1:5)]</span></code><code><span><span>## a. 移除缺失值大于20%的药物</span></span></code><code><span>ctrp.auc &lt;- ctrp.auc[,apply(ctrp.auc,2,<span>function</span>(x) sum(is.na(x))) &lt; 0.2*nrow(ctrp.auc)]</span></code><code><span>prism.auc &lt;- prism.auc[,apply(prism.auc,2,<span>function</span>(x) sum(is.na(x))) &lt; 0.2*nrow(prism.auc)]</span></code><code><span><span>## b. 移除CTRP数据里源自haematopoietic_and_lymphoid_tissue的细胞系</span></span></code><code><span>rmccl &lt;- paste0(<span>"CCL"</span>,na.omit(ctrp.ccl.anno[<span>which</span>(ctrp.ccl.anno<span>$ccle_primary_site</span> == <span>"haematopoietic_and_lymphoid_tissue"</span>),<span>"master_ccl_id"</span>]))</span></code><code><span>rownames(ctrp.auc) &lt;- paste0(<span>"CCL"</span>,rownames(ctrp.auc))</span></code><code><span>ctrp.auc &lt;- ctrp.auc[setdiff(rownames(ctrp.auc),rmccl),]</span></code><code><span><span>## c. KNN填补缺失值</span></span></code><code><span>ctrp.auc.knn &lt;- impute.knn(as.matrix(ctrp.auc))<span>$data</span></span></code><code><span>prism.auc.knn &lt;- impute.knn(as.matrix(prism.auc))<span>$data</span></span></code><code><span><span>## d. 数据量级修正</span></span></code><code><span>ctrp.auc.knn &lt;- ctrp.auc.knn/ceiling(max(ctrp.auc.knn)) <span># 参考Expression Levels of Therapeutic Targets as Indicators of Sensitivity to Targeted Therapeutics (2019, Molecular Cancer Therapeutics)</span></span></code><code><span>prism.auc.knn &lt;- prism.auc.knn/ceiling(max(prism.auc.knn))</span></code><code><span><span># 加载CCLE细胞系的表达谱，作为训练集</span></span></code><code><span>ccl.expr &lt;- read.table(<span>"CCLE_RNAseq_rsem_genes_tpm_20180929.txt"</span>,sep = <span>"\t"</span>,row.names = 1,check.names = F,stringsAsFactors = F,header = T)</span></code></pre></section><p><span> </span><img data-imgfileid="100005990" data-ratio="0.2611111111111111" data-src="https://mmbiz.qpic.cn/sz_mmbiz_jpg/oCSOtP61aEWNEG0fFLrTUKtCS0fZKXcCZy4ctL3I045I01Olf18vxfdpZ81EeicEK3yibLJYQRaXrDI9cCVSJ7tA/640?wx_fmt=jpeg" data-type="jpeg" data-w="1080" height="144.27000427246094" width="553.6799926757812" src="https://mmbiz.qpic.cn/sz_mmbiz_jpg/oCSOtP61aEWNEG0fFLrTUKtCS0fZKXcCZy4ctL3I045I01Olf18vxfdpZ81EeicEK3yibLJYQRaXrDI9cCVSJ7tA/640?wx_fmt=jpeg"><span> </span><span></span></p><section><ul><li><li></ul><pre data-lang="xml"><code><span># 加载基因注释文件，用于基因ID转换</span></code><code><span>Ginfo <span>&lt;<span>-</span> <span>read.table</span>("<span>overlapTable27.txt</span>",<span>sep</span> = <span>"\t"</span>,<span>row.names</span> = <span>1,check.names</span> = <span>F,stringsAsFactors</span> = <span>F,header</span> = <span>T)</span></span></span></code></pre></section><p><img data-imgfileid="100005989" data-ratio="0.29814814814814816" data-src="https://mmbiz.qpic.cn/sz_mmbiz_jpg/oCSOtP61aEWNEG0fFLrTUKtCS0fZKXcC6IpCULuV7tBaPquMdKjOmibAwxYiajn4FKXfbUrvOxAJDEHLWl5VIzibA/640?wx_fmt=jpeg" data-type="jpeg" data-w="1080" height="165.1400146484375" width="553.3400268554688" src="https://mmbiz.qpic.cn/sz_mmbiz_jpg/oCSOtP61aEWNEG0fFLrTUKtCS0fZKXcC6IpCULuV7tBaPquMdKjOmibAwxYiajn4FKXfbUrvOxAJDEHLWl5VIzibA/640?wx_fmt=jpeg"><span></span></p><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="bash"><code><span><span># 把基因的ensembl ID转换为gene symbol</span></span></code><code><span>ccl.expr &lt;- ccl.expr[,-1]; rownames(ccl.expr) &lt;- sapply(strsplit(rownames(ccl.expr),<span>"."</span>,fixed = T),<span>"["</span>,1)</span></code><code><span>comgene &lt;- intersect(rownames(ccl.expr),rownames(Ginfo))</span></code><code><span>ccl.expr &lt;- ccl.expr[comgene,]</span></code><code><span>ccl.expr<span>$gene</span> &lt;- Ginfo[comgene,<span>"genename"</span>]; ccl.expr &lt;- ccl.expr[!duplicated(ccl.expr<span>$gene</span>),]; rownames(ccl.expr) &lt;- ccl.expr<span>$gene</span>; ccl.expr &lt;- ccl.expr[,-ncol(ccl.expr)]</span></code><code><span>keepgene &lt;- apply(ccl.expr, 1, mad) &gt; 0.5 <span># 保留表达值有效的基因</span></span></code><code><span>trainExpr &lt;- log2(ccl.expr[keepgene,] + 1)</span></code><code><span>colnames(trainExpr) &lt;- sapply(strsplit(colnames(trainExpr),<span>"_"</span>,fixed = T),<span>"["</span>,1) <span># 重置细胞系名</span></span></code><code><span>trainPtype &lt;- as.data.frame(ctrp.auc.knn)</span></code><code><span>ccl.name &lt;- ccl.miss &lt;- c() <span># 替换细胞系名</span></span></code><code><span><span>for</span> (i <span>in</span> rownames(trainPtype)) {</span></code><code><span>  <span>if</span>(!is.element(gsub(<span>"CCL"</span>,<span>""</span>,i),ctrp.ccl.anno<span>$master_ccl_id</span>)) {</span></code><code><span>    cat(i,<span>"\n"</span>)</span></code><code><span>    ccl.miss &lt;- c(ccl.miss, i) <span># 没有匹配到的细胞系</span></span></code><code><span>    ccl.name &lt;- c(ccl.name, i) <span># 插入未匹配的细胞系    </span></span></code><code><span>  } <span>else</span> {</span></code><code><span>    ccl.name &lt;- c(ccl.name,  ctrp.ccl.anno[<span>which</span>(ctrp.ccl.anno<span>$master_ccl_id</span> == gsub(<span>"CCL"</span>,<span>""</span>,i)),<span>"ccl_name"</span>]) <span># 插入匹配的细胞系</span></span></code><code><span>  }</span></code><code><span>}</span></code><code><span>cpd.name &lt;- cpd.miss &lt;- c() <span># 替换药物名</span></span></code><code><span><span>for</span> (i <span>in</span> colnames(trainPtype)) {</span></code><code><span>  <span>if</span>(!is.element(i,ctrp.cpd.anno<span>$master_cpd_id</span>)) {</span></code><code><span>    cat(i,<span>"\n"</span>)</span></code><code><span>    cpd.miss &lt;- c(cpd.miss, i) <span># 没有匹配到的药物</span></span></code><code><span>    cpd.name &lt;- c(cpd.name, i) <span># 插入未匹配的药物</span></span></code><code><span>  } <span>else</span> {</span></code><code><span>    cpd.name &lt;- c(cpd.name,  ctrp.cpd.anno[<span>which</span>(ctrp.cpd.anno<span>$master_cpd_id</span> == i),<span>"cpd_name"</span>]) <span># 插入匹配的药物</span></span></code><code><span>  }</span></code><code><span>}</span></code><code><span>rownames(trainPtype) &lt;- ccl.name</span></code><code><span>trainPtype &lt;- trainPtype[setdiff(rownames(trainPtype),ccl.miss),] <span># 去除未匹配的细胞系</span></span></code><code><span>colnames(trainPtype) &lt;- cpd.name</span></code><code><span>trainPtype &lt;- trainPtype[,setdiff(colnames(trainPtype),cpd.miss)] <span># 去除未匹配的药物</span></span></code><code><span>comccl &lt;- intersect(rownames(trainPtype),colnames(trainExpr)) <span># 提取有表达且有药敏的细胞系</span></span></code><code><span>trainExpr &lt;- trainExpr[,comccl]</span></code><code><span>trainPtype &lt;- trainPtype[comccl,]</span></code><code><span><span># 测试集</span></span></code><code><span>keepgene &lt;- apply(pure.tumoexpr, 1, mad) &gt; 0.5 <span># 纯化的测试集取表达稳定的基因    </span></span></code><code><span>testExpr &lt;- log2(pure.tumoexpr[keepgene,] + 1) <span># 表达谱对数化</span></span></code><code><span><span># 取训练集和测试集共有的基因</span></span></code><code><span>comgene &lt;- intersect(rownames(trainExpr),rownames(testExpr))</span></code><code><span>trainExpr &lt;- as.matrix(trainExpr[comgene,])</span></code><code><span>testExpr &lt;- testExpr[comgene,]</span></code></pre></section><p><img data-galleryid="" data-ratio="0.3888888888888889" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_jpg/oCSOtP61aEUFpwgWbDAbKCv4EY895qWIvPxQsM4wDkGT3dSLjEfEYReYOgEj3iakD4AibCSicRHbjmERLKpIlyj8w/640?wx_fmt=jpeg" data-type="jpeg" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_jpg/oCSOtP61aEUFpwgWbDAbKCv4EY895qWIvPxQsM4wDkGT3dSLjEfEYReYOgEj3iakD4AibCSicRHbjmERLKpIlyj8w/640?wx_fmt=jpeg"></p><p><strong><span>3.预测药物敏感性</span></strong><span></span></p><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="php"><code><span>outTab &lt;- <span>NULL</span></span></code><code><span><span># 循环很慢，请耐心</span></span></code><code><span><span>for</span> (i in <span>1</span>:ncol(trainPtype)) {</span></code><code><span>  d &lt;- colnames(trainPtype)[i]</span></code><code><span>  tmp &lt;- log2(<span>as</span>.vector(trainPtype[,d]) + <span>0.00001</span>) <span># 由于CTRP的AUC可能有0值，因此加一个较小的数值防止报错</span></span></code><code><span>  <span># 岭回归预测药物敏感性</span></span></code><code><span>  ptypeOut &lt;- quiet(calcPhenotype(trainingExprData = trainExpr,</span></code><code><span>                                  trainingPtype = tmp,</span></code><code><span>                                  testExprData = testExpr,</span></code><code><span>                                  powerTransformPhenotype = F,</span></code><code><span>                                  selection = <span>1</span>))</span></code><code><span>  ptypeOut &lt;- <span>2</span>^ptypeOut - <span>0.00001</span> <span># 反对数</span></span></code><code><span>  outTab &lt;- rbind.data.frame(outTab,ptypeOut)</span></code><code><span>}</span></code><code><span>dimnames(outTab) &lt;- <span>list</span>(colnames(trainPtype),colnames(testExpr))</span></code><code><span>prism.pred.auc &lt;- outTab</span></code><code><span>top.pps &lt;- Sinfo[Sinfo$PPS &gt;= quantile(Sinfo$PPS,probs = seq(<span>0</span>,<span>1</span>,<span>0.1</span>))[<span>10</span>],] <span># 定义上十分位的样本</span></span></code><code><span>bot.pps &lt;- Sinfo[Sinfo$PPS &lt;= quantile(Sinfo$PPS,probs = seq(<span>0</span>,<span>1</span>,<span>0.1</span>))[<span>2</span>],] <span># 定义下十分位的样本</span></span></code><code><span>ctrp.log2fc &lt;- c()</span></code><code><span><span>for</span> (i in <span>1</span>:nrow(ctrp.pred.auc)) {</span></code><code><span>  d &lt;- rownames(ctrp.pred.auc)[i]</span></code><code><span>  a &lt;- mean(<span>as</span>.numeric(ctrp.pred.auc[d,rownames(top.pps)])) <span># 上十分位数的AUC均值    </span></span></code><code><span>  b &lt;- mean(<span>as</span>.numeric(ctrp.pred.auc[d,rownames(bot.pps)])) <span># 下十分位数的AUC均值</span></span></code><code><span>  fc &lt;- b/a</span></code><code><span>  log2fc &lt;- log2(fc); names(log2fc) &lt;- d</span></code><code><span>  ctrp.log2fc &lt;- c(ctrp.log2fc,log2fc)</span></code><code><span>}</span></code><code><span>candidate.ctrp &lt;- ctrp.log2fc[ctrp.log2fc &gt; <span>0.2</span>] <span># 这里我调整了阈值，控制结果数目</span></span></code><code><span>prism.log2fc &lt;- c()</span></code><code><span><span>for</span> (i in <span>1</span>:nrow(prism.pred.auc)) {</span></code><code><span>  display.progress(index = i,totalN = nrow(prism.pred.auc))</span></code><code><span>  d &lt;- rownames(prism.pred.auc)[i]</span></code><code><span>  a &lt;- mean(<span>as</span>.numeric(prism.pred.auc[d,rownames(top.pps)])) <span># 上十分位数的AUC均值</span></span></code><code><span>  b &lt;- mean(<span>as</span>.numeric(prism.pred.auc[d,rownames(bot.pps)])) <span># 下十分位数的AUC均值</span></span></code><code><span>  fc &lt;- b/a</span></code><code><span>  log2fc &lt;- log2(fc); names(log2fc) &lt;- d</span></code><code><span>  prism.log2fc &lt;- c(prism.log2fc,log2fc)</span></code><code><span>}</span></code><code><span>candidate.prism &lt;- prism.log2fc[prism.log2fc &gt; <span>0.2</span>] <span># 这里我调整了阈值，控制结果数目</span></span></code><code><span>ctrp.cor &lt;- ctrp.cor.p &lt;- c()</span></code><code><span><span>for</span> (i in <span>1</span>:nrow(ctrp.pred.auc)) {</span></code><code><span>  d &lt;- rownames(ctrp.pred.auc)[i]</span></code><code><span>  a &lt;- <span>as</span>.numeric(ctrp.pred.auc[d,rownames(Sinfo)])</span></code><code><span>  b &lt;- <span>as</span>.numeric(Sinfo$PPS)</span></code><code><span>  r &lt;- cor.test(a,b,method = <span>"spearman"</span>)$estimate; names(r) &lt;- d</span></code><code><span>  p &lt;- cor.test(a,b,method = <span>"spearman"</span>)$p.value; names(p) &lt;- d</span></code><code><span>  ctrp.cor &lt;- c(ctrp.cor,r)</span></code><code><span>  ctrp.cor.p &lt;- c(ctrp.cor.p,p)</span></code><code><span>}</span></code><code><span>candidate.ctrp2 &lt;- ctrp.cor[ctrp.cor &lt; <span>-0.4</span>]  <span># 这里我调整了阈值，控制结果数目    </span></span></code><code><span>ctrp.candidate &lt;- intersect(names(candidate.ctrp),names(candidate.ctrp2))</span></code><code><span>prism.cor &lt;- prism.cor.p &lt;- c()</span></code><code><span><span>for</span> (i in <span>1</span>:nrow(prism.pred.auc)) {</span></code><code><span>  d &lt;- rownames(prism.pred.auc)[i]</span></code><code><span>  a &lt;- <span>as</span>.numeric(prism.pred.auc[d,rownames(Sinfo)])</span></code><code><span>  b &lt;- <span>as</span>.numeric(Sinfo$PPS)</span></code><code><span>  r &lt;- cor.test(a,b,method = <span>"spearman"</span>)$estimate; names(r) &lt;- d</span></code><code><span>  p &lt;- cor.test(a,b,method = <span>"spearman"</span>)$p.value; names(p) &lt;- d</span></code><code><span>  prism.cor &lt;- c(prism.cor,r)</span></code><code><span>  prism.cor.p &lt;- c(prism.cor.p,p)</span></code><code><span>}</span></code><code><span>candidate.prism2 &lt;- prism.cor[prism.cor &lt; <span>-0.35</span>] </span></code><code><span>prism.candidate &lt;- intersect(names(candidate.prism),names(candidate.prism2))</span></code></pre></section><p><span></span></p><p><strong><span>4.绘制棒棒糖图和显著差异分析箱线图</span></strong><span></span></p><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="swift"><code><span>#绘制相关性棒棒糖图</span></code><code><span># 设置颜色</span></code><code><span>darkblue &lt;- <span>"#0772B9"</span></span></code><code><span>lightblue &lt;- <span>"#48C8EF"</span></span></code><code><span>cor.data &lt;- data.frame(drug = ctrp.candidate,</span></code><code><span>                       r = ctrp.cor[ctrp.candidate],</span></code><code><span>                       p = -log10(ctrp.cor.p[ctrp.candidate]))</span></code><code><span>p1 &lt;- ggplot(data = cor.data,aes(r,forcats::fct_reorder(drug,r,.desc = <span>T</span>))) +</span></code><code><span>  geom_segment(aes(xend=<span>0</span>,yend=drug),linetype = <span>2</span>) +</span></code><code><span>  geom_point(aes(size=p),col = darkblue) +</span></code><code><span>  scale_size_continuous(range =<span>c</span>(<span>2</span>,<span>8</span>)) +</span></code><code><span>  scale_x_reverse(breaks = <span>c</span>(<span>0</span>, -<span>0.3</span>, -<span>0.5</span>),</span></code><code><span>                  expand = expansion(mult = <span>c</span>(<span>0.01</span>,.<span>1</span>))) + #左右留空</span></code><code><span>  theme_classic() +</span></code><code><span>  labs(x = <span>"Correlation coefficient"</span>, y = <span>""</span>, size = bquote(<span>"-log"</span>[<span>10</span>]~<span>"("</span>~italic(<span>P</span>)~<span>"-value)"</span>)) +</span></code><code><span>  theme(legend.position = <span>"bottom"</span>,</span></code><code><span>        axis.line.y = element_blank())</span></code><code><span>cor.data &lt;- data.frame(drug = prism.candidate,</span></code><code><span>                       r = prism.cor[prism.candidate],</span></code><code><span>                       p = -log10(prism.cor.p[prism.candidate]))</span></code><code><span>cor.data$drug &lt;- sapply(strsplit(cor.data$drug,<span>" ("</span>,fixed = <span>T</span>), <span>"["</span>,<span>1</span>)</span></code><code><span>p2 &lt;- ggplot(data = cor.data,aes(r,forcats::fct_reorder(drug,r,.desc = <span>T</span>))) +</span></code><code><span>  geom_segment(aes(xend=<span>0</span>,yend=drug),linetype = <span>2</span>) +    </span></code><code><span>  geom_point(aes(size=p),col = darkblue) +</span></code><code><span>  scale_size_continuous(range =<span>c</span>(<span>2</span>,<span>8</span>)) +</span></code><code><span>  scale_x_reverse(breaks = <span>c</span>(<span>0</span>, -<span>0.3</span>, -<span>0.5</span>),</span></code><code><span>                  expand = expansion(mult = <span>c</span>(<span>0.01</span>,.<span>1</span>))) + #左右留空</span></code><code><span>  theme_classic() +</span></code><code><span>  labs(x = <span>"Correlation coefficient"</span>, y = <span>""</span>, size = bquote(<span>"-log"</span>[<span>10</span>]~<span>"("</span>~italic(<span>P</span>)~<span>"-value)"</span>)) +</span></code><code><span>  theme(legend.position = <span>"bottom"</span>,</span></code><code><span>        axis.line.y = element_blank())</span></code><code><span>ctrp.boxdata &lt;- <span>NULL</span></span></code><code><span><span>for</span> (d <span>in</span> ctrp.candidate) {</span></code><code><span>  a &lt;- <span>as</span>.numeric(ctrp.pred.auc[d,rownames(top.pps)])</span></code><code><span>  b &lt;- <span>as</span>.numeric(ctrp.pred.auc[d,rownames(bot.pps)])</span></code><code><span>  p &lt;- wilcox.test(a,b)$p.value</span></code><code><span>  s &lt;- <span>as</span>.character(cut(p,<span>c</span>(<span>0</span>,<span>0.001</span>,<span>0.01</span>,<span>0.05</span>,<span>1</span>),labels = <span>c</span>(<span>"***"</span>,<span>"**"</span>,<span>"*"</span>,<span>""</span>)))</span></code><code><span>  ctrp.boxdata &lt;- rbind.data.frame(ctrp.boxdata,</span></code><code><span>                                   data.frame(drug = d,</span></code><code><span>                                              auc = <span>c</span>(a,b),</span></code><code><span>                                              p = p,</span></code><code><span>                                              s = s,</span></code><code><span>                                              group = rep(<span>c</span>(<span>"High PPS"</span>,<span>"Low PPS"</span>),<span>c</span>(nrow(top.pps),nrow(bot.pps))),</span></code><code><span>                                              stringsAsFactors = <span>F</span>),</span></code><code><span>                                   stringsAsFactors = <span>F</span>)</span></code><code><span>}</span></code><code><span>#绘制显著差异箱线图</span></code><code><span>p3 &lt;- ggplot(ctrp.boxdata, aes(drug, auc, fill=group)) +</span></code><code><span>  geom_boxplot(aes(col = group),outlier.shape = <span>NA</span>) +</span></code><code><span>  geom_text(aes(drug, y=<span>max</span>(auc)),</span></code><code><span>                label=ctrp.boxdata$s,</span></code><code><span>            data=ctrp.boxdata, </span></code><code><span>            inherit.aes=<span>F</span>) +</span></code><code><span>  scale_fill_manual(values = <span>c</span>(darkblue, lightblue)) +</span></code><code><span>  scale_color_manual(values = <span>c</span>(darkblue, lightblue)) +</span></code><code><span>  xlab(<span>NULL</span>) + ylab(<span>"Estimated AUC value"</span>) +</span></code><code><span>  theme_classic() +</span></code><code><span>  theme(axis.text.x = element_text(angle = <span>45</span>, hjust = <span>0.5</span>,vjust = <span>0.5</span>,size = <span>10</span>),</span></code><code><span>        legend.position = <span>"bottom"</span>,</span></code><code><span>        legend.title = element_blank())</span></code><code><span>dat &lt;- ggplot_build(p3)$data[[<span>1</span>]]</span></code><code><span>p3 &lt;- p3 + geom_segment(data=dat, aes(x=xmin, xend=xmax, y=middle, yend=middle), color=<span>"white"</span>, inherit.aes = <span>F</span>)    </span></code><code><span>prism.boxdata &lt;- <span>NULL</span></span></code><code><span><span>for</span> (d <span>in</span> prism.candidate) {</span></code><code><span>  a &lt;- <span>as</span>.numeric(prism.pred.auc[d,rownames(top.pps)])</span></code><code><span>  b &lt;- <span>as</span>.numeric(prism.pred.auc[d,rownames(bot.pps)])</span></code><code><span>  p &lt;- wilcox.test(a,b)$p.value</span></code><code><span>  s &lt;- <span>as</span>.character(cut(p,<span>c</span>(<span>0</span>,<span>0.001</span>,<span>0.01</span>,<span>0.05</span>,<span>1</span>),labels = <span>c</span>(<span>"***"</span>,<span>"**"</span>,<span>"*"</span>,<span>""</span>)))</span></code><code><span>  prism.boxdata &lt;- rbind.data.frame(prism.boxdata,</span></code><code><span>                                   data.frame(drug = d,</span></code><code><span>                                              auc = <span>c</span>(a,b),</span></code><code><span>                                              p = p,</span></code><code><span>                                              s = s,</span></code><code><span>                                              group = rep(<span>c</span>(<span>"High PPS"</span>,<span>"Low PPS"</span>),<span>c</span>(nrow(top.pps),nrow(bot.pps))),</span></code><code><span>                                              stringsAsFactors = <span>F</span>),</span></code><code><span>                                   stringsAsFactors = <span>F</span>)</span></code><code><span>}</span></code><code><span>prism.boxdata$drug &lt;- sapply(strsplit(prism.boxdata$drug,<span>" ("</span>,fixed = <span>T</span>), <span>"["</span>,<span>1</span>)</span></code><code><span>p4 &lt;- ggplot(prism.boxdata, aes(drug, auc, fill=group)) +</span></code><code><span>  geom_boxplot(aes(col = group),outlier.shape = <span>NA</span>) +</span></code><code><span>  geom_text(aes(drug, y=<span>max</span>(auc)),</span></code><code><span>            label=prism.boxdata$s,</span></code><code><span>            data=prism.boxdata, </span></code><code><span>            inherit.aes=<span>F</span>) +</span></code><code><span>  scale_fill_manual(values = <span>c</span>(darkblue, lightblue)) +</span></code><code><span>  scale_color_manual(values = <span>c</span>(darkblue, lightblue)) +</span></code><code><span>  xlab(<span>NULL</span>) + ylab(<span>"Estimated AUC value"</span>) +</span></code><code><span>  theme_classic() +</span></code><code><span>  theme(axis.text.x = element_text(angle = <span>45</span>, hjust = <span>0.5</span>,vjust = <span>0.5</span>,size = <span>10</span>),</span></code><code><span>      legend.position = <span>"bottom"</span>,</span></code><code><span>      legend.title = element_blank())</span></code><code><span>dat &lt;- ggplot_build(p4)$data[[<span>1</span>]]</span></code><code><span>p4 &lt;- p4 + geom_segment(data=dat, aes(x=xmin, xend=xmax, y=middle, yend=middle), color=<span>"white"</span>, inherit.aes = <span>F</span>)</span></code><code><span>#进行拼图</span></code><code><span>plot_grid(p1, p3, p2, p4, labels=<span>c</span>(<span>"A"</span>, <span>""</span>, <span>"B"</span>, <span>""</span>),</span></code><code><span>          ncol=<span>2</span>, </span></code><code><span>          rel_widths = <span>c</span>(<span>2</span>, <span>2</span>)) #左右两列的宽度比例</span></code><code><span>#保存图片    </span></code><code><span>ggsave(filename = <span>"drug target.pdf"</span>,width = <span>8</span>,height = <span>8</span>)</span></code></pre></section><p><span></span></p><p><strong><span>5.结果展示</span></strong><span></span></p><p><span>5.1drug target.pdf</span></p><p><img data-imgfileid="100005987" data-ratio="0.9982817869415808" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEWNEG0fFLrTUKtCS0fZKXcCgrbtwFia8XS3Qq8hcbyoVfyDIh1CibIEeGzIaGpElYp54YgSnkZz0PSw/640?wx_fmt=png" data-type="png" data-w="582" height="387.3399658203125" width="388.010009765625" src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEWNEG0fFLrTUKtCS0fZKXcCgrbtwFia8XS3Qq8hcbyoVfyDIh1CibIEeGzIaGpElYp54YgSnkZz0PSw/640?wx_fmt=png"></p><p><span>最终师妹利用pRRophetic包成功的预测了肿瘤样本药物敏感性，这里利用了两个药物敏感性数据库PRISM和CTRP进行预测，结果更加全面，肿瘤药物敏感性和肿瘤药物预测相关分析内容，比如GDSC细胞系药物敏感性分析(http://www.biocloudservice.com/757/757.php),cmap方法进行药物预测(http://www.biocloudservice.com/787/787.php)等都可以用本公司新开发的零代码云平台生信分析小工具，一键完成该分析奥，感兴趣的小伙伴欢迎来尝试奥，文末阅读原文查看，网址：http://www.biocloudservice.com/home.html。<span>欢迎大家和师妹</span><span>一起讨论学习呀！</span></span></p><p><span> </span><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=Mzg5MDk3Mzg4OA==&amp;mid=2247488767&amp;idx=2&amp;sn=f4de9eddb643d07901e8026f6a809030&amp;chksm=cfd52f50f8a2a6464ed97abaae48727355571d8c1b14ba989f58d9e8125b937e7c2f1310aa2d&amp;scene=21#wechat_redirect" textvalue="你已选中了添加链接的内容" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2" hasload="1"><span></span></a><span><span><p> </p></span><span> </span></span><img data-galleryid="" data-imgfileid="100005996" data-ratio="0.4255555555555556" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEXIKicwUpMyKu4leia6g9hicEcR0pHEtianl9Z041w5j3xsEzQqBBlpWP4MdibsBbbdSYpOfP2zNEwo0eg/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="900" src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEXIKicwUpMyKu4leia6g9hicEcR0pHEtianl9Z041w5j3xsEzQqBBlpWP4MdibsBbbdSYpOfP2zNEwo0eg/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></p><p><br></p><section><section powered-by="xiumi.us"><section><section powered-by="xiumi.us"><section><svg viewbox="0 0 1 1"></svg></section></section></section><section><section powered-by="xiumi.us"><p>往期回顾</p></section></section><section><section powered-by="xiumi.us"><section><img data-imgfileid="100005995" data-ratio="0.34869240348692404" data-s="300,640" data-type="png" data-w="803" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEUib55HJnYldhvDYfwRibcD0mvibYkv232FbpSq6w8GU3UpUO7hju2w7e4bnK5exm72kZLULGiaKBLgXg/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEUib55HJnYldhvDYfwRibcD0mvibYkv232FbpSq6w8GU3UpUO7hju2w7e4bnK5exm72kZLULGiaKBLgXg/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></section></section></section></section><section powered-by="xiumi.us"><section><section powered-by="xiumi.us"><section><p><span>·</span> <a target="_blank" href="http://mp.weixin.qq.com/s?__biz=Mzg5MDk3Mzg4OA==&amp;mid=2247484765&amp;idx=1&amp;sn=8265eebe2c60d89687a23ed208e72ffb&amp;chksm=cfd53ef2f8a2b7e45f09a4053d393ea932a58f119dffe26b8599d80e94f11aabd9ebac9f3be5&amp;scene=21#wechat_redirect" textvalue="用R进行GWAS分析，原来如此简单" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2" hasload="1">用R进行GWAS分析，原来如此简单</a></p></section></section></section></section><section powered-by="xiumi.us"><section><section powered-by="xiumi.us"><section><p><span>· </span><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=Mzg5MDk3Mzg4OA==&amp;mid=2247487464&amp;idx=1&amp;sn=a482466b09dbb452c48af3efcfae73c3&amp;chksm=cfd53447f8a2bd512ab81087174f1ae07d19fcbc31efd07f6f4d6bcda16ab5b6b8e2004b89d0&amp;scene=21#wechat_redirect" textvalue="三分钟解决R语言中的切片" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2" hasload="1">三分钟解决R语言中的切片</a></p></section></section></section></section><section powered-by="xiumi.us"><section><section powered-by="xiumi.us"><section><p><span>·</span><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=Mzg5MDk3Mzg4OA==&amp;mid=2247486842&amp;idx=1&amp;sn=48b175e9f1c957fa53aa226a07670533&amp;chksm=cfd536d5f8a2bfc32383e3091d54568c13f7f7fa745bc8a732e51b2fa4db07f2fc53b76c9d97&amp;scene=21#wechat_redirect" textvalue=" R包EnhancedVolcano" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2" hasload="1"><span> </span>R包EnhancedVolcano</a></p></section></section></section></section><section powered-by="xiumi.us"><section><section powered-by="xiumi.us"><section><p><span>· </span><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=Mzg5MDk3Mzg4OA==&amp;mid=2247488767&amp;idx=2&amp;sn=f4de9eddb643d07901e8026f6a809030&amp;chksm=cfd52f50f8a2a6464ed97abaae48727355571d8c1b14ba989f58d9e8125b937e7c2f1310aa2d&amp;scene=21#wechat_redirect" textvalue="1024G存储‍生信服务器           " linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2" hasload="1">1024G存储生信服务器<span>     </span><span>      </span></a><span> </span></p></section></section></section></section></section><p><br></p><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/nsXeiYjoaMM7aXZEAwar7Q",target="_blank" rel="noopener noreferrer">原文链接</a>
