---
title: "复现单细胞结合常规转录组的Nat Med文章数据挖掘部分"
date: 2024-08-14T10:02:33Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Acdemic"]
---
复现单细胞结合常规转录组的Nat Med文章数据挖掘部分 by 生信菜鸟团
------
<div><p><strong>英文标题：</strong><span>High systemic and tumor-associated IL-8 correlates with reduced clinical benefit of PD-L1 blockade</span><br></p><p><span><strong>期刊：</strong>Nat Med. 2020 May;26(5):693-698.<br><strong>影响因子：</strong>1区，82.9<br><strong>DOI:</strong> 10.1038/s41591-020-0860-1</span><br><span><strong>研究领域：</strong>PD-L1抑制剂；IL-8</span></p><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><blockquote data-tool="mdnice编辑器"><p>IMvigor210CoreBiologies这个R包中有一些免疫治疗相关的数据。看看这个试验的设计和结果。</p></blockquote><figure data-tool="mdnice编辑器"><img data-ratio="1.1305555555555555" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIBe2w1CF6tNQExicLKq4iapSeOEuaZQcc0N0xt9Boh83TLAyRYrib5Cj7Q/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIBe2w1CF6tNQExicLKq4iapSeOEuaZQcc0N0xt9Boh83TLAyRYrib5Cj7Q/640?wx_fmt=png"></figure><h3 data-tool="mdnice编辑器"><span></span><span>首先安装这个包</span><span></span></h3><pre data-tool="mdnice编辑器"><span></span><code><span>#IMvigor210CoreBiologies安装，先下载到本地，使用本地安装方式</span><br><span>#http://research-pub.gene.com/IMvigor210CoreBiologies/packageVersions/</span><br>install.packages(<span>"./IMvigor210CoreBiologies_1.0.0.tar.gz"</span>, repos = NULL, <span>type</span> = <span>"source"</span>)<br></code></pre><p data-tool="mdnice编辑器">会发现装不上，不要着急，先装一点依赖包</p><pre data-tool="mdnice编辑器"><span></span><code><span>### 设置镜像</span><br>options(<span>"repos"</span>=c(CRAN=<span>"https://mirrors.tuna.tsinghua.edu.cn/CRAN/"</span>))<br>options(BioC_mirror=<span>"https://mirrors.tuna.tsinghua.edu.cn/bioconductor"</span>)<br><span>### 安装缺失的R包-CRAN</span><br><span>if</span>(!requireNamespace(<span>"dplyr"</span>,quietly = TRUE)) install.packages(<span>"dplyr"</span>)<br><span>if</span>(!requireNamespace(<span>"DT"</span>,quietly = TRUE)) install.packages(<span>"DT"</span>)<br><span>if</span>(!requireNamespace(<span>"ggplot2"</span>,quietly = TRUE)) install.packages(<span>"ggplot2"</span>)<br><span>if</span>(!requireNamespace(<span>"reshape2"</span>,quietly = TRUE)) install.packages(<span>"reshape2"</span>)<br><span>if</span>(!requireNamespace(<span>"plyr"</span>,quietly = TRUE)) install.packages(<span>"plyr"</span>)<br><span>if</span>(!requireNamespace(<span>"survival"</span>,quietly = TRUE)) install.packages(<span>"survival"</span>)<br><span>if</span>(!requireNamespace(<span>"circlize"</span>,quietly = TRUE)) install.packages(<span>"circlize"</span>)<br><span>if</span>(!requireNamespace(<span>"lsmeans"</span>,quietly = TRUE)) install.packages(<span>"lsmeans"</span>)<br><span>if</span>(!requireNamespace(<span>"spatstat"</span>,quietly = TRUE)) install.packages(<span>"spatstat"</span>)<br><span>if</span>(!requireNamespace(<span>"corrplot"</span>,quietly = TRUE)) install.packages(<span>"corrplot"</span>)<br><span>### 安装缺失的R包-Bioconductor</span><br><span>if</span>(!requireNamespace(<span>"ComplexHeatmap"</span>,quietly = TRUE)) BiocManager::install(<span>"ComplexHeatmap"</span>)<br><span>if</span>(!requireNamespace(<span>"biomaRt"</span>,quietly = TRUE)) BiocManager::install(<span>"biomaRt"</span>)<br><span>if</span>(!requireNamespace(<span>"DESeq2"</span>,quietly = TRUE)) BiocManager::install(<span>"DESeq2"</span>)<br><span>if</span>(!requireNamespace(<span>"edgeR"</span>,quietly = TRUE)) BiocManager::install(<span>"edgeR"</span>)<br><span>if</span>(!requireNamespace(<span>"limma"</span>,quietly = TRUE)) BiocManager::install(<span>"limma"</span>)<br></code></pre><p data-tool="mdnice编辑器">再装一下Deseq包，直接装装不上，在这个链接中下载老版本。https://www.bioconductor.org/packages/3.11/bioc/html/DESeq.html</p><p data-tool="mdnice编辑器">下载到本地</p><figure data-tool="mdnice编辑器"><img data-ratio="0.3474842767295597" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIMTV4q2qxTibxoGCJefZKKtgFqOnXw7E6tfjq3JkiaiaU65IXUic0ibD4PTQ/640?wx_fmt=png" data-type="png" data-w="636" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIMTV4q2qxTibxoGCJefZKKtgFqOnXw7E6tfjq3JkiaiaU65IXUic0ibD4PTQ/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code><span>if</span>(!requireNamespace(<span>"genefilter"</span>,quietly = TRUE)) BiocManager::install(<span>"genefilter"</span>)<br>library(genefilter)<br>install.packages(<span>"DESeq_1.39.0.tar.gz"</span>, repos = NULL, <span>type</span> = <span>"source"</span>)<br></code></pre><p data-tool="mdnice编辑器">安装完成后再来一遍</p><pre data-tool="mdnice编辑器"><span></span><code>install.packages(<span>"IMvigor210CoreBiologies_1.0.0.tar.gz"</span>, repos = NULL, <span>type</span> = <span>"source"</span>)<br></code></pre><p data-tool="mdnice编辑器">这样就好了。</p><pre data-tool="mdnice编辑器"><span></span><code>library(IMvigor210CoreBiologies)<br><br>data(cds)<br>expreSet &lt;- as.data.frame(counts(cds))<br>annoData &lt;- fData(cds)<br>phenoData &lt;- pData(cds)<br></code></pre><h2 data-tool="mdnice编辑器"><span><span>1</span></span><span>首先是Plasma IL-8 and clinical outcomes in muC and mRCC</span></h2><p data-tool="mdnice编辑器">首先看看原图，想要在这个数据集中根据pIL-8的表达划分高低组，方法中写的是根据median，可能还做了什么处理，导致我的数据和文中有一些偏差。不过没关系，大概的趋势是正确的，IL-8表达高带来生存获益较差，这里的获益是在患者接受atezo治疗后。</p><figure data-tool="mdnice编辑器"><img data-ratio="0.4203703703703704" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIPE4RgYEzPOR1HFdb5oRjJ8hEp4b5uquTcicmOya2VWre8cOuQt4nd9g/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIPE4RgYEzPOR1HFdb5oRjJ8hEp4b5uquTcicmOya2VWre8cOuQt4nd9g/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">先来第一个图，普通的生存分析</p><pre data-tool="mdnice编辑器"><span></span><code><span>#IL-8的基因为CXCL8</span><br>rm(list = ls())<br><span>library</span>(pacman)<br><span>library</span>(IMvigor210CoreBiologies)<br><span>library</span>(dplyr)<br><br><span>#2.数据预处理</span><br><span>##2.1表达矩阵</span><br>data(cds)<br>expMatrix &lt;- counts(cds)<br>eff_length2 &lt;- fData(cds)[,c(<span>"entrez_id"</span>,<span>"length"</span>,<span>"symbol"</span>)]<br>rownames(eff_length2) &lt;- eff_length2$entrez_id<br>head(eff_length2)<br>feature_ids &lt;- rownames(expMatrix)<br>expMatrix &lt;- expMatrix[feature_ids %<span>in</span>% rownames(eff_length2),]<br>nrow (expMatrix)<br>ncol (expMatrix)<br>mm &lt;- match(rownames(expMatrix),rownames(eff_length2))<br>eff_length2 &lt;- eff_length2[mm,]<br><br>x &lt;- expMatrix/eff_length2$length<br>eset &lt;- t(t(x)/colSums(x))*<span>1e6</span><br>summary(duplicated(rownames(eset)))<br><br><span>if</span>(!requireNamespace(<span>"remotes"</span>, quietly =<span>TRUE</span>)) install(<span>"remotes"</span>)<br><span>if</span>(!requireNamespace(<span>"IOBR"</span>, quietly =<span>TRUE</span>))remotes::install_github(<span>"IOBR/IOBR"</span>,ref=<span>"master"</span>)<br><br>eset &lt;- IOBR::anno_eset(eset = eset,<br>                        annotation = eff_length2,<br>                        symbol = <span>"symbol"</span>,<br>                        probe = <span>"entrez_id"</span>,<br>                        method = <span>"mean"</span>)<br><span>#BLCA Bladder Urothelial Carcinoma 膀胱尿路上皮癌</span><br>tumor_type &lt;- <span>"BLCA"</span><br><span>if</span>(max(eset)&gt;<span>100</span>) eset &lt;- log2(eset+<span>1</span>)<br><br><br><span>##2.2 表型文件</span><br><span>library</span>(tibble)<br>pdata &lt;- pData(cds)<br>colnames(pdata) &lt;- gsub(colnames(pdata),pattern = <span>" "</span>,replacement = <span>"_"</span>)<br>pdata&lt;-pdata[!is.na(pdata$binaryResponse),]<br>table(pdata$binaryResponse)<br>pdata$group&lt;-ifelse(pdata$binaryResponse==<span>"CR/PR"</span>,<span>"R"</span>,<span>"NR"</span>)<br>table(pdata$group)<br>table(pdata$Best_Confirmed_Overall_Response)<br>save(expMatrix,eset,pdata,file = <span>"expcli_IMvigor210.Rdata"</span>)<br><br><br>rm(list = ls())<br>load(<span>'expcli_IMvigor210.Rdata'</span>)<br>exp = eset[,colnames(eset) %<span>in</span>% rownames(pdata)]<br><br>p = identical(colnames(exp),rownames(pdata));p<br><br>CXCL8_exp = exp[<span>'CXCL8'</span>,]<br>a = as.data.frame(t(CXCL8_exp)) <br>p = identical(rownames(a),rownames(pdata));p<br>pdata$CXCL8 = a$CXCL8<br>pdata$CXCL8_group &lt;- ifelse(pdata$CXCL8 &gt;= median(pdata$CXCL8) ,<span>"pIL-8 High"</span>,<span>"pIL-8 Low"</span>)<br>table(pdata$CXCL8_group)<br><br><span>library</span>(survival)<br><span>library</span>(ggplot2)<br><span>library</span>(survminer)<br>colnames(pdata)<br>fit &lt;- survfit(Surv(os, censOS) ~ CXCL8_group, data = pdata)<br><br><span># 拟合Cox比例风险模型</span><br>pdata$CXCL8_group = factor(pdata$CXCL8_group,levels = c(<span>"pIL-8 Low"</span>,<span>"pIL-8 High"</span>))<br>pdata$CXCL8_group<br>cox_fit &lt;- coxph(Surv(os, censOS) ~ CXCL8_group, data = pdata)<br>hr &lt;- exp(coef(cox_fit))<br>hr<br><br>g &lt;- ggsurvplot(fit, data = pdata, risk.table = <span>TRUE</span>, pval = <span>TRUE</span>, surv.median.line = <span>"hv"</span>,<br>                legend.title = <span>" "</span>, <br>                legend.labs = c(<span>"pIL-8 High"</span>, <span>"pIL-8 Low"</span>),<br>                title = <span>"pIL-8 cohort2 OS IMvigor210"</span>)<br>g<br><br>p_value &lt;- summary(cox_fit)$coefficients[, <span>"Pr(&gt;|z|)"</span>]<br>g$table &lt;- g$table +<br>  theme(legend.position = <span>"none"</span>) +<br>  labs(title = <span>"OS(months)"</span>, x = <span>"OS(months)"</span>, y = <span>"风险比"</span>) +<br>  theme(plot.title = element_text(hjust = <span>0.5</span>))<br><br>g$plot &lt;- g$plot + theme(plot.title = element_text(hjust = <span>0.5</span>))+<br>        annotate(<span>"text"</span>, x = <span>1.57</span>, y = <span>0.12</span>, label = paste(<span>"HR ="</span>, round(hr, <span>2</span>)), size = <span>4.75</span>)<br>print(g)<br></code></pre><p data-tool="mdnice编辑器">这里有一个点需要注意到，因为中间有一步计算了HR值，按照正常情况来讲，低pIL-8表达组比高pIL-8表达组降低了风险，最开始得到的HR是0.6，降低了40%。但是文中很明显是反过来比的，说高pIL-8表达组比低pIL-8表达组增加了风险，所以我们这里也反过来。</p><p data-tool="mdnice编辑器"><em>注意的就是生存曲线中的线，如果在得到fit前就设置factor，后面命名图例，很可能颜色会反。</em></p><figure data-tool="mdnice编辑器"><img data-ratio="0.32075471698113206" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIgf1Z194xCmKWA7Tymez4PFYenyzJ5m3UhJ3nouG8DdZsszKfGjGLSA/640?wx_fmt=png" data-type="png" data-w="424" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIgf1Z194xCmKWA7Tymez4PFYenyzJ5m3UhJ3nouG8DdZsszKfGjGLSA/640?wx_fmt=png"></figure><figure data-tool="mdnice编辑器"><img data-ratio="0.9712962962962963" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DImvaxib7kTqnxYhyIxp2WIefJaQRr6wiarJY3KcZZITdpPXJTT8aUF7mg/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DImvaxib7kTqnxYhyIxp2WIefJaQRr6wiarJY3KcZZITdpPXJTT8aUF7mg/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code>rm(list = ls())<br>load(<span>'expcli_IMvigor210.Rdata'</span>)<br>exp = eset[,colnames(eset) %<span>in</span>% rownames(pdata)]<br><br>p = identical(colnames(exp),rownames(pdata));p<br>voomD &lt;- filterNvoom(exp,<br>                     minSamples=ncol(exp)/<span>10</span>, <span># 最小样本数</span><br>                     minCpm=<span>0.25</span>, <span># 最小cpm值</span><br>                     DESeq=<span>FALSE</span>) <span>#  edgeR normFactors will used for normalization(默认),</span><br><br>names(voomD)<br>m &lt;- voomD$E<br>dim(m) <br>m[<span>1</span>:<span>3</span>,<span>1</span>:<span>3</span>]<br><br><span>#归一化：scale</span><br><span>#对数据进行处理，可以将大范围变化数据大范围变化落入一个小的特定区间，如：[0, 1]或[-1, 1]</span><br><span># 所谓数据的归一化是指中心化之后的数据在除以数据集的标准差，即数据集中的各项数据减去数据集的均值再除以数据集的标准差。</span><br>m &lt;- t(scale( t( m ),<br>              center=<span>TRUE</span>, <span># 中心化</span><br>              scale=<span>TRUE</span>) <span># 标准化</span><br>)<br>dim(m)<br>m[<span>1</span>:<span>3</span>,<span>1</span>:<span>3</span>]<br>m = as.data.frame(m)<br>CXCL8_exp = m[<span>'CXCL8'</span>,]<br>a = as.data.frame(t(CXCL8_exp)) <br>p = identical(rownames(a),rownames(pdata));p<br><br>pdata$CXCL8 = a$CXCL8<br>pdata$CXCL8_group &lt;- ifelse(pdata$CXCL8 &gt;= median(pdata$CXCL8) ,<span>"pIL-8 High"</span>,<span>"pIL-8 Low"</span>)<br><br>table(pdata$CXCL8_group)<br><br>clinical=pdata<br>colnames(clinical)<br><br><span>#基因集计算</span><br>data(human_gene_signatures)<br>ind_genes &lt;- human_gene_signatures<br>goi &lt;- names(ind_genes)<br>goi <span># 20个基因集</span><br><br><span>for</span> (sig <span>in</span> goi) {<br>  clinical[, sig] &lt;- <span>NA</span> <span># 在pData新增一列列名为sig，内容NA</span><br>  genes &lt;- ind_genes[[sig]] <span># 在基因集的基因</span><br>  genes &lt;- genes[genes %<span>in</span>% rownames(m)] <span># 与原矩阵基因求交集</span><br>  tmp &lt;- m[genes, , drop=<span>FALSE</span>] <span># 交集基因的表达量</span><br>  clinical[, sig] &lt;- gsScore(tmp) <span># gsScore计算分数，并写入该对应列中</span><br>}<br><br>dim(clinical)<br>colnames(clinical)<br><br><br>BOR &lt;- <span>"Best_Confirmed_Overall_Response"</span><br>table(clinical[, BOR])<br>tmpDat &lt;- clinical[ clinical[, BOR] != <span>"NE"</span>, ]<br>table(tmpDat[, BOR])<br>tmpDat[, BOR] &lt;- droplevels(tmpDat[, BOR]) <span># 将level=NE删除掉</span><br>table(tmpDat[, BOR])<br><br><span>#Fisher's exact test( 费希尔精确检验)</span><br>colnames(clinical)<br>IL8 &lt;- table(tmpDat$CXCL8_group, tmpDat[, <span>"binaryResponse"</span>])<br>IL8<br><br>pval &lt;- signif(fisher.test(IL8)$p.value, <span>2</span>) <span># fisher.test检验，p值取两位有效数字</span><br>print(paste(<span>"Fisher P for IL8 by binary response:"</span>, pval))<br><span>#0.21,ORR之前没啥差异</span><br><br><span>##堆积柱状图</span><br><span>library</span>(tidyr)<br><span>library</span>(reshape2)<br><br>colnames(tmpDat)<br>tb=table(tmpDat$CXCL8_group,<br>         tmpDat[, BOR])<br>head(tb)<br><span>library</span> (gplots) <br>balloonplot(tb)<br>bar_data &lt;- as.data.frame(tb)<br><br>bar_per &lt;- bar_data %&gt;% <br>  group_by(Var1) %&gt;%<br>  mutate(sum(Freq)) %&gt;%<br>  mutate(percent = Freq / `sum(Freq)`)<br>head(bar_per) <br><span>#write.csv(bar_per,file = "celltype_by_group_percent.csv")</span><br>col =c(<span>"#3176B7"</span>,<span>"#F78000"</span>,<span>"#3FA116"</span>,<span>"#CE2820"</span>,<span>"#9265C1"</span>,<br>                <span>"#885649"</span>,<span>"#DD76C5"</span>,<span>"#7F7F7F"</span>,<span>"#BBBE00"</span>,<span>"#41BED1"</span>)<br>                <br>ggplot(bar_per, aes(x = percent, y = Var1)) +<br>  geom_bar(aes(fill = Var2) , stat = <span>"identity"</span>) + coord_flip() +<br>  theme(axis.ticks = element_line(linetype = <span>"blank"</span>),<br>        legend.position = <span>"top"</span>,<br>        panel.grid.minor = element_line(colour = <span>NA</span>,linetype = <span>"blank"</span>), <br>        panel.background = element_rect(fill = <span>NA</span>),<br>        plot.background = element_rect(colour = <span>NA</span>)) +<br>  labs(y = <span>" "</span>, fill = <span>NULL</span>)+labs(x = <span>'Fraction of patients'</span>)+<br>  scale_fill_manual(values=col)+<br>  labs(title = <span>"pIL-8 cohort2 ORR IMvigor210"</span>)+<br>  theme_classic()+<br>  theme(plot.title = element_text(size=<span>12</span>,hjust=<span>0.5</span>))<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.08055555555555556" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DI1oqsnsTHKDtKk1Ehicia20fy2U66VEJTZttRhnqb1e5ZHjEjuH72kP3w/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DI1oqsnsTHKDtKk1Ehicia20fy2U66VEJTZttRhnqb1e5ZHjEjuH72kP3w/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">虽然ORR之间的p值是没有差异的，但是有一些趋势。</p><figure data-tool="mdnice编辑器"><img data-ratio="1.0342592592592592" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIxEMd24iaR4wgibDpE58kFDEeSXzPNNve7ibmslHQm4tcqvmo0vU3PHdicA/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIxEMd24iaR4wgibDpE58kFDEeSXzPNNve7ibmslHQm4tcqvmo0vU3PHdicA/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">pIL-8组更多的CR患者，PR患者也多了3个（文章中明显多了，开始怀疑这个数据集到底是不是cohort2，刚读取进来的时候是348例，但cohort2应该只有310例患者，cohort1有119例患者）。但是ORR竟然拉不开差异，没有得到文章中的结果</p><p data-tool="mdnice编辑器"><strong><em>好了这个IL-8竟然都不是plasma IL-8，做之前一定要仔细检查数据。</em></strong></p><figure data-tool="mdnice编辑器"><img data-ratio="0.1175925925925926" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIicic50iaNdhLD8Tt2e63iaSLh2cVyqB0UoQVLbfVK1DOqPia2FibZpJh2Y7Q/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIicic50iaNdhLD8Tt2e63iaSLh2cVyqB0UoQVLbfVK1DOqPia2FibZpJh2Y7Q/640?wx_fmt=png"></figure><figure data-tool="mdnice编辑器"><img data-ratio="0.9537037037037037" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIUrhhlAIQx5RoMT4rclSdtvXMTFhMrYBKCSxbv7ufISNw6Z68wLNZjw/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIUrhhlAIQx5RoMT4rclSdtvXMTFhMrYBKCSxbv7ufISNw6Z68wLNZjw/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">接下来是同时Teff和IL-8，看看高高，高低，低高，低低对生存的影响。Teff主要存在于CD8+ T细胞中，那么就可以直接使用gsScore 函数计算20个基因集的分数中的CD8 T effector评分。当然，也可以自己计算，通过文中给的基因，计算四个基因的平均值。</p><figure data-tool="mdnice编辑器"><img data-ratio="0.04537037037037037" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIGAtzGdReiahZCiaRxs8GMibRq1bNMdjhUPX1lIqLuTl9mmJn86LS597tQ/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIGAtzGdReiahZCiaRxs8GMibRq1bNMdjhUPX1lIqLuTl9mmJn86LS597tQ/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code><span>###上一步计算了Teff</span><br>tmpDat$Teff_group &lt;- ifelse(tmpDat$`CD 8 T effector` &gt;= median(tmpDat$`CD 8 T effector`) ,<span>"Teff High"</span>,<span>"Teff Low"</span>)<br>table(tmpDat$Teff_group)<br>table(tmpDat$CXCL8_group)<br>tmpDat$group &lt;- ifelse(tmpDat$Teff_group %<span>in</span>% c(<span>"Teff High"</span>) &amp; tmpDat$CXCL8_group %<span>in</span>% c(<span>"pIL-8 High"</span>),<span>"Teff High,IL8 High"</span>,<br>                ifelse(tmpDat$Teff_group %<span>in</span>% c(<span>"Teff High"</span>) &amp; tmpDat$CXCL8_group %<span>in</span>% c(<span>"pIL-8 Low"</span>),<span>"Teff High,IL8 Low"</span>,<br>                       ifelse(tmpDat$Teff_group %<span>in</span>% c(<span>"Teff Low"</span>) &amp; tmpDat$CXCL8_group %<span>in</span>% c(<span>"pIL-8 High"</span>),<span>"Teff Low,IL8 High"</span>, <span>"Teff Low,IL8 Low"</span>)))<br><br>table(tmpDat$group)<br><br><span>library</span>(survival)<br><span>library</span>(ggplot2)<br><span>library</span>(survminer)<br>colnames(tmpDat)<br>fit &lt;- survfit(Surv(os, censOS) ~ group, data = tmpDat)<br><br><span># 拟合Cox比例风险模型</span><br>tmpDat$group = factor(tmpDat$group,levels = c(<span>"pIL-8 Low"</span>,<span>"pIL-8 High"</span>))<br>tmpDat$group<br>cox_fit &lt;- coxph(Surv(os, censOS) ~ group, data = tmpDat)<br><br><span># 提取风险比的估计值</span><br>hr &lt;- exp(coef(cox_fit))<br>hr<br><br>g &lt;- ggsurvplot(fit, data = pdata, risk.table = <span>TRUE</span>, pval = <span>TRUE</span>, surv.median.line = <span>"hv"</span>,<br>                legend.title = <span>" "</span>, <br>                xlab = <span>"OS(months)"</span>, <br>                legend = c(<span>0.8</span>,<span>0.75</span>), <br>                legend.labs = c(<span>"Teff High,IL8 High"</span>, <span>"Teff High,IL8 Low"</span>,<span>"Teff Low,IL8 High"</span>, <span>"Teff Low,IL8 Low"</span>),<br>                title = <span>"pIL-8 Teff high/low cohort2 OS IMvigor210"</span>)<br>g<br>p_value &lt;- summary(cox_fit)$coefficients[, <span>"Pr(&gt;|z|)"</span>]<br><br>g$table &lt;- g$table +<br>  theme(legend.position = <span>"none"</span>) +<br>  labs(title = <span>" "</span>, x = <span>"OS(months)"</span>, y = <span>" "</span>) +<br>  theme(plot.title = element_text(hjust = <span>0.5</span>))<br><br>g$plot &lt;- g$plot + theme(plot.title = element_text(hjust = <span>0.5</span>))+<br>  annotate(<span>"text"</span>, x = <span>11.4</span>, y = <span>0</span>, label = <span>'11.4'</span>, size = <span>3.6</span>,colour = <span>"#E87B70"</span>)+<br>  annotate(<span>"text"</span>, x = <span>7.2</span>, y = <span>0</span>, label = <span>'7.2'</span>, size = <span>3.6</span>,colour = <span>"#53BBC1"</span>)+<br>  annotate(<span>"text"</span>, x = <span>10.8</span>, y = <span>0</span>, label = <span>'10.8'</span>, size = <span>3.6</span>,colour = <span>"#BC7FF8"</span>)<br><br>print(g)<br></code></pre><p data-tool="mdnice编辑器">还是按照我们自己的HR顺序吧，可以很明显的看出，Teff高IL8低降低风险最多，这一组生存较长。</p><figure data-tool="mdnice编辑器"><img data-ratio="0.09262759924385633" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIKw7NvYqrYNL2rrJRicW2El4E3BibIIpOk8sLZghYlQham5j0ILpFjdbg/640?wx_fmt=png" data-type="png" data-w="1058" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIKw7NvYqrYNL2rrJRicW2El4E3BibIIpOk8sLZghYlQham5j0ILpFjdbg/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">看一下中位生存时间，标上去</p><figure data-tool="mdnice编辑器"><img data-ratio="0.2052401746724891" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIPhDp6c8lYG66V5U0uklzcNCUsIWzRRfdibKWzIHOQYibriaRj1AXnZib2A/640?wx_fmt=png" data-type="png" data-w="916" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIPhDp6c8lYG66V5U0uklzcNCUsIWzRRfdibKWzIHOQYibriaRj1AXnZib2A/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">趋势符合逻辑，Teff低IL8高表达，相对而言，生存最差。</p><figure data-tool="mdnice编辑器"><img data-ratio="0.9675925925925926" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIcJsHAXjRmSU6z5aHfPUDWCbKSquTxfchgXYRqtQ68lkF5icqTIicPJkg/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIcJsHAXjRmSU6z5aHfPUDWCbKSquTxfchgXYRqtQ68lkF5icqTIicPJkg/640?wx_fmt=png"></figure><h2 data-tool="mdnice编辑器"><span><span>2</span></span><span>Poor clinical outcome and lower expression of antigen-presentation genes associated with IL8-high myeloid subsets in PBMCs</span></h2><p data-tool="mdnice编辑器">首先我们看看这个原图，不管是髓系还是淋巴细胞中，Nonresponders表达IL-8更多。对髓系细胞的IL8高和低表达进行差异分析后发现，IL8高的髓系细胞中髓系炎症反应基因(橙色)表达丰富，而IL8低的髓系细胞中抗原呈递基因(蓝色)表达较高。PBMCs中IL8基因的高表达与mUC IMvigor210中较差的OS显著相关。</p><figure data-tool="mdnice编辑器"><img data-ratio="0.45555555555555555" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIQaHkO0NBDj8ScbrIgdSuVxDadXPaCicGk94L1Wv5OSZXMJDkMicBtwTA/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIQaHkO0NBDj8ScbrIgdSuVxDadXPaCicGk94L1Wv5OSZXMJDkMicBtwTA/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器"><img data-ratio="1.1194444444444445" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIXXRjTp1S5mZibMibOibGMx6YGp2adicdahnEPwQJ1GAgw6xVoEZVS7FT6w/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIXXRjTp1S5mZibMibOibGMx6YGp2adicdahnEPwQJ1GAgw6xVoEZVS7FT6w/640?wx_fmt=png"><img data-ratio="0.5416666666666666" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DI3Gq5uciaXZoCQ6GeoibMaB2ibE9Bric0x9qwYrkqGapOpdMORSarg53fSA/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DI3Gq5uciaXZoCQ6GeoibMaB2ibE9Bric0x9qwYrkqGapOpdMORSarg53fSA/640?wx_fmt=png"></p><pre data-tool="mdnice编辑器"><span></span><code>rm(list=ls())<br><span>library</span>(Seurat)<br><span>library</span>(ggplot2)<br><span>library</span>(clustree)<br><span>library</span>(cowplot)<br><span>library</span>(dplyr)<br>getwd()<br><br>dir.create(<span>"2-harmony"</span>)<br>getwd()<br>setwd(<span>"2-harmony"</span>)<br>sce=readRDS(<span>"../1-QC/sce.all_qc.rds"</span>)<br>sce<br>sce &lt;- NormalizeData(sce, <br>                   normalization.method = <span>"LogNormalize"</span>,<br>                    scale.factor = <span>1e4</span>) <br>sce &lt;- FindVariableFeatures(sce)<br>sce &lt;- ScaleData(sce)<br>sce &lt;- RunPCA(sce, features = VariableFeatures(object = sce))<br><br><span>library</span>(harmony)<br>table(sce$orig.ident)<br>seuratObj &lt;- RunHarmony(sce, <span>"orig.ident"</span>)<br>names(seuratObj@reductions)<br>seuratObj &lt;- RunUMAP(seuratObj,  dims = <span>1</span>:<span>15</span>, <br>                     reduction = <span>"harmony"</span>)<br>DimPlot(seuratObj,reduction = <span>"umap"</span>,label=<span>T</span> ) <br><br><br>sce=seuratObj<br><span>#奇怪的报错，下面是解决方式</span><br><span># remove.packages("Matrix")</span><br><span># install.packages("https://cran.r-project.org/src/contrib/Archive/Matrix/Matrix_1.5-1.tar.gz")</span><br><span># #之后重启</span><br><span># #先library(Matrix)，再调用其他包</span><br><br>sce &lt;- FindNeighbors(sce, reduction = <span>"harmony"</span>,<br>                     dims = <span>1</span>:<span>15</span>) <br>sce.all=sce<br><span>#设置不同的分辨率，观察分群效果(选择哪一个？)</span><br><span>for</span> (res <span>in</span> c(<span>0.01</span>, <span>0.05</span>, <span>0.1</span>, <span>0.2</span>, <span>0.3</span>, <span>0.5</span>,<span>0.8</span>,<span>1</span>)) {<br>  sce.all=FindClusters(sce.all, <span>#graph.name = "CCA_snn", </span><br>                       resolution = res, algorithm = <span>1</span>)<br>}<br>colnames(sce.all@meta.data)<br>apply(sce.all@meta.data[,grep(<span>"RNA_snn"</span>,colnames(sce.all@meta.data))],<span>2</span>,table)<br><br>p2_tree=clustree(sce.all@meta.data, prefix = <span>"RNA_snn_res."</span>)<br><span>#ggsave(plot=p2_tree, filename="Tree_diff_resolution.pdf")</span><br><br><span>#按照分辨率为1进行 </span><br>sel.clust = <span>"RNA_snn_res.1"</span><br>sce.all &lt;- SetIdent(sce.all, value = sel.clust)<br>table(sce.all@active.ident) <br>saveRDS(sce.all, <span>"sce.all_int.rds"</span>)<br></code></pre><p data-tool="mdnice编辑器">依旧是常规流程。</p><pre data-tool="mdnice编辑器"><span></span><code>rm(list=ls())<br>setwd(<span>'../'</span>)<br>dir.create(<span>"3-cell"</span>)<br>setwd(<span>"3-cell"</span>) <br>sce.all=readRDS(<span>"../2-harmony/sce.all_int.rds"</span>)<br>colnames(sce.all@meta.data)<br>p1 = DimPlot(sce.all, reduction = <span>"umap"</span>, group.by = <span>"sample"</span>,label = <span>T</span>,raster=<span>FALSE</span>,cols = c(<span>'red'</span>,<span>'green'</span>)) <br><span>#ggsave('umap_by_patient.pdf',width = 9,height = 7)</span><br>col = c(<span>'#DF4298'</span>, <span>'#7C388C'</span>, <span>'#D58EBE'</span> ,  <span>'#746FB4'</span>,<span>'#3A53A4'</span>,<span>'#FEC026'</span> ,<span>'#F57F20'</span> ,<span>'#CB952B'</span> , <span>'#A1CC59'</span>,<span>'#186533'</span>)<br>p2 = DimPlot(sce.all, reduction = <span>"umap"</span>, group.by = <span>"orig.ident"</span>,raster=<span>FALSE</span>,cols = col) <br><span>#ggsave('umap_by_RNA_snn_res.0.3.pdf',width = 9,height = 7)</span><br><span>library</span>(patchwork)<br>p1+p2<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.5" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIl3Z0ULh02QWmynGYLZYBTzqfI7pQSuD9vPY2BBibjXcIYfaoa0rtHPw/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIl3Z0ULh02QWmynGYLZYBTzqfI7pQSuD9vPY2BBibjXcIYfaoa0rtHPw/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code><span>library</span>(ggplot2) <br>genes_to_check = c(<span>'ITGAX'</span>, <span>'CD11b'</span>, <span>'CD86'</span>, <span>'CD1C'</span>, <span>'CD209'</span>, <span>'CLEC9A'</span>,<span>'CD1E'</span>,  <span>#DC</span><br>                    <span>'CD11a'</span>,  <span>'IRF8'</span>, <span>'CCR5'</span>, <span>'HLA-DRB1'</span>, <span>#DC-like</span><br>                   <span>'CD14'</span>,  <span>'CD64'</span>,<span>'CCR2'</span>, <span>'S100A8'</span>, <span>'S100A9'</span>, <span>#mono</span><br>                   <span>'FCGR3A'</span>, <span>#CD16 Monocytes</span><br>                   <span>'PTPRC'</span>, <span>'CD3D'</span>, <span>'CD3E'</span>, <span>'CD4'</span>,<span>'CD8A'</span>,<span>'CD8B'</span>, <br>                   <span>'CD8'</span>,  <span>'CD45RA'</span>, <span>'CD197'</span>, <span>'CCR7'</span>, <span>#CD8T</span><br>                   <span>'IL7R'</span>, <span>'BCL2'</span>, <span>#CD8Tcm</span><br>                   <span>'GZMB'</span>, <span>'PRF1'</span>, <span>'KLRG1'</span>,<span>'TBX21'</span>,<span>#CD8Tem</span><br>                    <span>'CD27'</span>, <span>#CD4T</span><br>                   <span>'FOXP3'</span>, <span>'IL2RA'</span>, <span>'CTLA4'</span>, <span>'IKZF2'</span>, <span>'LRRC32'</span>, <span>#Treg</span><br>                   <span>'CD45RO'</span>,<span>'CD62L'</span>, <span>'CD28'</span>, <span>#CD4Tcm</span><br>                   <span>'CD57'</span>,<span>'CD127'</span>, <span>#CD4Tem</span><br>                   <span>'CD19'</span>, <span>'CD79A'</span>, <span>'MS4A1'</span> ,<span>'BLK'</span>, <span>#B</span><br>                   <span>'KLRB1'</span>,<span>'NCR1'</span>, <span>'NKG7'</span>, <span>'KLRD1'</span>,<span>'CD56'</span>,<span>'NKG2D'</span>, <span># NK </span><br>                   <span>'FGF7'</span>,<span>'MME'</span>, <span>'ACTA2'</span>, <span>## fibo </span><br>                   <span>'CD41'</span>, <span>'CD61'</span>, <span>'CD42b'</span>, <span>'CD34'</span>, <span>'CD235a'</span>,<span>'MPL'</span>, <span>'PF4'</span>, <span>'ITGA2B'</span>, <span>'ITGB3'</span>, <span>'GP1BB'</span>,<span>#Megakaryocytes</span><br>                   <span>'EPCAM'</span> , <span>'KRT19'</span>, <span>'PROM1'</span>, <span>'ALDH1A1'</span> )<br><span>library</span>(stringr)  <br>p_all_markers &lt;- DotPlot(sce.all, features = genes_to_check,<br>                         assay=<span>'RNA'</span>  )  + coord_flip()<br><br>p_all_markers<br><br>p_umap=DimPlot(sce.all, reduction = <span>"umap"</span>,<br>               group.by = <span>"RNA_snn_res.1"</span>,label = <span>T</span>,raster=<span>FALSE</span>) <br>p_umap<br><span>library</span>(patchwork)<br>p_all_markers+p_umap<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.6185185185185185" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIHm9iaFuS7gb1dmpkWcPcOJ2UyIicspIuy2sKWwV1ufPAj9WOet8Q73cg/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIHm9iaFuS7gb1dmpkWcPcOJ2UyIicspIuy2sKWwV1ufPAj9WOet8Q73cg/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code>rm(list=ls())<br>options(stringsAsFactors = <span>F</span>)<br><span>library</span>(Seurat)<br><span>library</span>(ggplot2)<br><span>library</span>(clustree)<br><span>library</span>(cowplot)<br><span>library</span>(dplyr)<br>getwd()<br>setwd(<span>'3-cell/'</span>)<br>sce.all=readRDS( <span>"../2-harmony/sce.all_int.rds"</span>)<br>sce.all<br><br>celltype=data.frame(ClusterID=<span>0</span>:<span>16</span>,<br>                    celltype= <span>0</span>:<span>16</span>) <br><br>celltype[celltype$ClusterID %<span>in</span>% c( <span>6</span>),<span>2</span>]=<span>'DC'</span> <br>celltype[celltype$ClusterID %<span>in</span>% c(<span>1</span>,<span>3</span>,<span>9</span>,<span>15</span> ),<span>2</span>]=<span>'Monocytes'</span> <br>celltype[celltype$ClusterID %<span>in</span>% c( <span>8</span> ),<span>2</span>]=<span>'CD16 monoccytes'</span><br>celltype[celltype$ClusterID %<span>in</span>% c(<span>12</span>),<span>2</span>]=<span>'DC-like'</span> <br>celltype[celltype$ClusterID %<span>in</span>% c( <span>9</span>),<span>2</span>]=<span>'Megakaryocytes'</span>  <br>celltype[celltype$ClusterID %<span>in</span>% c(<span>11</span>  ),<span>2</span>]=<span>'CD8 Tcm'</span> <br>celltype[celltype$ClusterID %<span>in</span>% c( <span>16</span> ),<span>2</span>]=<span>'CD8 Tem'</span> <br>celltype[celltype$ClusterID %<span>in</span>% c( <span>0</span> ),<span>2</span>]=<span>'CD4 T'</span> <br>celltype[celltype$ClusterID %<span>in</span>% c(<span>2</span> ),<span>2</span>]=<span>'CD4 Tcm'</span> <br>celltype[celltype$ClusterID %<span>in</span>% c( <span>14</span>),<span>2</span>]=<span>'CD4 Tem'</span> <br>celltype[celltype$ClusterID %<span>in</span>% c( <span>13</span> ),<span>2</span>]=<span>'Treg'</span><br>celltype[celltype$ClusterID %<span>in</span>% c(<span>4</span>,<span>5</span>,<span>10</span>),<span>2</span>]=<span>'NK'</span> <br>celltype[celltype$ClusterID %<span>in</span>% c( <span>7</span>),<span>2</span>]=<span>'B'</span>  <br><br>head(celltype)<br>celltype<br>table(celltype$celltype)<br>sce.all@meta.data$celltype = <span>"NA"</span><br><span>for</span>(i <span>in</span> <span>1</span>:nrow(celltype)){<br>  sce.all@meta.data[which(sce.all@meta.data$RNA_snn_res.1 == celltype$ClusterID[i]),<span>'celltype'</span>] &lt;- celltype$celltype[i]}<br>table(sce.all@meta.data$celltype)<br><br>table(sce.all@meta.data$celltype,sce.all@meta.data$RNA_snn_res.1) <br><br>mycolors &lt;-c(<span>'#E5D2DD'</span>, <span>'#53A85F'</span>, <span>'#F1BB72'</span>, <span>'#F3B1A0'</span>, <span>'#D6E7A3'</span>, <span>'#57C3F3'</span>,<br>                      <span>'#E95C59'</span>, <span>'#E59CC4'</span>, <span>'#AB3282'</span>,  <span>'#BD956A'</span>, <br>                      <span>'#9FA3A8'</span>, <span>'#E0D4CA'</span>,  <span>'#C5DEBA'</span>, <span>'#F7F398'</span>,<br>                      <span>'#C1E6F3'</span>, <span>'#6778AE'</span>, <span>'#91D0BE'</span>, <span>'#B53E2B'</span>,<br>                      <span>'#712820'</span>, <span>'#DCC1DD'</span>, <span>'#CCE0F5'</span>,  <span>'#CCC9E6'</span>, <span>'#625D9E'</span>, <span>'#68A180'</span>, <span>'#3A6963'</span>,<br>                      <span>'#968175'</span>)<br>                                      <br><span>library</span>(patchwork)<br>p_umap=DimPlot(sce.all, reduction = <span>"umap"</span>, group.by = <span>"celltype"</span>,label = <span>T</span>,<br>                     label.box=<span>T</span>,raster=<span>FALSE</span>,cols = mycolors)<br>p_all_markers+p_umap<br><br><span>library</span>(patchwork)<br>p_all_markers=DotPlot(sce.all, features = genes_to_check,<br>                    assay=<span>'RNA'</span> ,group.by = <span>'celltype'</span> )  + coord_flip()+th<br>p_umap=DimPlot(sce.all, reduction = <span>"umap"</span>, group.by = <span>"celltype"</span>,label = <span>T</span>,<br>                    label.box=<span>T</span>,raster=<span>FALSE</span>,cols = mycolors)<br>p_all_markers+p_umap<br><br>p_sample=DimPlot(sce.all, reduction = <span>"umap"</span>, group.by = <span>"celltype"</span>,label = <span>T</span>,raster=<span>FALSE</span>,cols = mycolors,split.by = <span>'sample'</span>)<br>p_sample<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.8675925925925926" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIBicjheJVQFuZrvYzSY8lyGN6Z1tkCeksibuNEn1QcH1icnQUYDqaZU8iaA/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIBicjheJVQFuZrvYzSY8lyGN6Z1tkCeksibuNEn1QcH1icnQUYDqaZU8iaA/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">明显看出在Responder中单核细胞和巨核细胞小了一圈，其他细胞亚群看起来或多或少有所增加。</p><figure data-tool="mdnice编辑器"><img data-ratio="0.6175925925925926" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIkRDyicktw51Gm1wyF3VB0yqUpcFRAc64oriccjjrQ6W1qpWl3vjaWuew/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIkRDyicktw51Gm1wyF3VB0yqUpcFRAc64oriccjjrQ6W1qpWl3vjaWuew/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code><span>####Featureplot</span><br>sce = sce.all<br>cell_marker &lt;- c(<span>"CD3D"</span>,<span>"NKG7"</span>,<span>"CD79A"</span>,<span>"CD14"</span>,<span>"IL1B"</span>,<span>"CXCL8"</span>)<br>FeaturePlot(sce,features = cell_marker,cols = c(<span>"lightgrey"</span> ,<span>"#DE1F1F"</span>),ncol=<span>3</span>)<br><br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.5805555555555556" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DINXqMmOULb3ZiayE8RSQ0vThkhIk4udCLIqFcCI3qUJ3GPf2icia3kwciag/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DINXqMmOULb3ZiayE8RSQ0vThkhIk4udCLIqFcCI3qUJ3GPf2icia3kwciag/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code>FeaturePlot(sce,features = <span>'CXCL8'</span>,cols = c(<span>"lightgrey"</span> ,<span>"#DE1F1F"</span>),split.by = <span>'sample'</span>)<br></code></pre><p data-tool="mdnice编辑器">可以明显看到CXCL8在不响应的人群中表达更高。</p><figure data-tool="mdnice编辑器"><img data-ratio="0.5787037037037037" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIgIHDoUvkK25ebFicoUHAnIpq0ZpZBOibiamthWeNYibGaXYqSDxac7OzVQ/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIgIHDoUvkK25ebFicoUHAnIpq0ZpZBOibiamthWeNYibGaXYqSDxac7OzVQ/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code><span>##小提琴图</span><br><span>library</span>(ggplot2)<br><span>library</span>(gghalves)<br><span>library</span>(tidyverse)<br><br><span>### 提取某一基因的表达值到metadata中</span><br>expr &lt;- sce@assays$RNA@scale.data<br>gene_name &lt;- c(<span>"CXCL8"</span>)<br>gene_expression &lt;- expr %&gt;% <br>  .[gene_name,] %&gt;% <br>  <span>#t() %&gt;% </span><br>  as.data.frame()<br>colnames(gene_expression) &lt;- paste0(gene_name)<br>identical(colnames(sce),row.names(gene_expression))<br>sce$CXCL8 &lt;- gene_expression[,paste0(gene_name)]<br>identical(sce@meta.data[,paste0(gene_name)],gene_expression[,paste0(gene_name)])<br>meta &lt;- sce@meta.data<br>colnames(meta)<br>table(meta$sample)<br>R &lt;- meta %&gt;% filter(sample==<span>"Responder"</span>)<br><br>NR &lt;- meta %&gt;% filter(sample==<span>"NonResponder"</span>)<br><br><span>library</span>(ggsignif)<br><span>library</span>(ggpubr)<br>ggplot() +<br>  geom_half_violin(data = R, aes(x = celltype, y = CXCL8),<br>                   position = position_dodge(width = <span>1</span>),<br>                   scale = <span>'width'</span>,<br>                   fill = <span>"#47BB0C"</span>,<br>                   side = <span>"l"</span>) +<br>  geom_half_violin(data = NR, aes(x = celltype, y = CXCL8),<br>                   position = position_dodge(width = <span>1</span>),<br>                   scale = <span>'width'</span>,<br>                   fill = <span>"#DC291A"</span>,<br>                   side = <span>"r"</span>) +<br>  theme_classic() +<br>  xlab(<span>""</span>) +<br>  ylab(<span>"CXCL8 expression"</span>) +<br>  stat_compare_means(data = meta, aes(x = celltype, y = CXCL8, group = sample),<br>                     symnum.args = list(cutpoints = c(<span>0</span>, <span>0.001</span>, <span>0.01</span>, <span>0.05</span>, <span>1</span>),<br>                                        symbols = c(<span>"***"</span>, <span>"**"</span>, <span>"*"</span>, <span>"ns"</span>)),<br>                     label = <span>"p.signif"</span>)<br><br></code></pre><p data-tool="mdnice编辑器">和原文趋势一样，无响应者的CXCL8几乎在所有细胞亚型中分布都较高。</p><figure data-tool="mdnice编辑器"><img data-ratio="0.5694444444444444" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DI7SubaEnuW5uxwc3r3kyPxGhZYYQicfgkXeicWm9oMOqQtQl9iaiaTDicreA/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DI7SubaEnuW5uxwc3r3kyPxGhZYYQicfgkXeicWm9oMOqQtQl9iaiaTDicreA/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code><span>source</span>(<span>'VolcanoPlot.R'</span>)<br>table(sce$celltype)<br>myeloid &lt;- subset(sce, celltype %<span>in</span>% c(<span>"CD16 monoccytes"</span>,<span>"DC"</span>,<span>"DC-like"</span>,<span>"Megakaryocytes"</span>,<span>"Monocytes"</span>))<br>myeloid$CXCL8_group &lt;- ifelse(myeloid$CXCL8 &gt;= median(myeloid$CXCL8) ,<span>"IL-8 High"</span>,<span>"IL-8 Low"</span>)<br>diff_myeloid &lt;- FindMarkers(myeloid, min.pct = <span>0.25</span>, <br>                           logfc.threshold = <span>0.25</span>,<br>                           group.by = <span>"CXCL8_group"</span>,<br>                           ident.1 =<span>"IL-8 High"</span>,<br>                           ident.2=<span>"IL-8 Low"</span>)       <br><span>#avg_logFC：两组之间平均表达的对数折叠通道。正值表示该特征在第一组中的表达更高。</span><br><br>dif=data.frame(<br>  symbol=rownames(diff_myeloid),<br>  log2FoldChange=diff_myeloid$avg_log2FC,<br>  padj=diff_myeloid$p_val_adj<br>)<br><br><span># 可以指定要标记的DEG数量，选出FC最大和最小的基因标记</span><br>VolcanoPlot(dif, padj=<span>0.05</span>, title=<span>"IL-8 High vs IL-8 Low"</span>, label.max = <span>50</span>, cols=c(<span>"blue"</span>, <span>"red"</span>))<br><br></code></pre><p data-tool="mdnice编辑器">可以看到CXCL8在IL-8 High中高表达</p><figure data-tool="mdnice编辑器"><img data-ratio="0.8166666666666667" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIwgHHvgOX9A6Urwib3N364ZJopicnG9vJic8fkV1vZrDV0lZeujg9zicAZA/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIwgHHvgOX9A6Urwib3N364ZJopicnG9vJic8fkV1vZrDV0lZeujg9zicAZA/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code><span>###细胞亚群比例</span><br><span>##堆积柱状图</span><br><span>library</span>(tidyr)<br><span>library</span>(reshape2)<br><br>colnames(meta)<br>tb=table(meta$celltype,<br>         meta$sample)<br>head(tb)<br><span>library</span> (gplots) <br>balloonplot(tb)<br>bar_data &lt;- as.data.frame(tb)<br><br>bar_per &lt;- bar_data %&gt;% <br>  group_by(Var1) %&gt;%<br>  mutate(sum(Freq)) %&gt;%<br>  mutate(percent = Freq / `sum(Freq)`)<br>head(bar_per) <br>col =c(<span>"#3176B7"</span>,<span>"#F78000"</span>,<span>"#3FA116"</span>,<span>"#CE2820"</span>,<span>"#9265C1"</span>,<br>                <span>"#885649"</span>,<span>"#DD76C5"</span>,<span>"#7F7F7F"</span>,<span>"#BBBE00"</span>,<span>"#41BED1"</span>)<br>                <br><span>library</span>(ggthemes)<br>g1 = ggplot(bar_per, aes(x = percent, y = Var1)) +<br>  geom_bar(aes(fill = Var2) , stat = <span>"identity"</span>) + <br>  theme(axis.ticks = element_line(linetype = <span>"blank"</span>),<br>        legend.position = <span>"top"</span>,<br>        panel.grid.minor = element_line(colour = <span>NA</span>,linetype = <span>"blank"</span>), <br>        panel.background = element_rect(fill = <span>NA</span>),<br>        plot.background = element_rect(colour = <span>NA</span>)) +<br>  labs(y = <span>"cell type"</span>, fill = <span>NULL</span>)+labs(x = <span>'Fraction of cells'</span>)+<br>  scale_fill_manual(values=col)+<br>  labs(title = <span>" "</span>)+<br>  theme_tufte()+<br>  theme(plot.title = element_text(size=<span>12</span>,hjust=<span>0.5</span>))<br><br><br><span>###2</span><br><span>##堆积柱状图</span><br><span>library</span>(tidyr)<br><span>library</span>(reshape2)<br><br>colnames(meta)<br>tb=table(meta$celltype,<br>         meta$orig.ident)<br>head(tb)<br><span>library</span> (gplots) <br>balloonplot(tb)<br>bar_data &lt;- as.data.frame(tb)<br><br>bar_per &lt;- bar_data %&gt;% <br>  group_by(Var1) %&gt;%<br>  mutate(sum(Freq)) %&gt;%<br>  mutate(percent = Freq / `sum(Freq)`)<br>head(bar_per) <br><span>#write.csv(bar_per,file = "celltype_by_group_percent.csv")</span><br>col =c(<span>"#3176B7"</span>,<span>"#F78000"</span>,<span>"#3FA116"</span>,<span>"#CE2820"</span>,<span>"#9265C1"</span>,<br>                <span>"#885649"</span>,<span>"#DD76C5"</span>,<span>"#7F7F7F"</span>,<span>"#BBBE00"</span>,<span>"#41BED1"</span>)<br>                <br><span>library</span>(ggthemes)<br>g2 = ggplot(bar_per, aes(x = percent, y = Var1)) +<br>  geom_bar(aes(fill = Var2) , stat = <span>"identity"</span>) + <br>  theme(axis.ticks = element_line(linetype = <span>"blank"</span>),<br>        legend.position = <span>"top"</span>,<br>        panel.grid.minor = element_line(colour = <span>NA</span>,linetype = <span>"blank"</span>), <br>        panel.background = element_rect(fill = <span>NA</span>),<br>        plot.background = element_rect(colour = <span>NA</span>)) +<br>  labs(y = <span>"cell type"</span>, fill = <span>NULL</span>)+labs(x = <span>'Fraction of cells'</span>)+<br>  scale_fill_manual(values=col)+<br>  labs(title = <span>" "</span>)+<br>  theme_tufte()+<br>  theme(plot.title = element_text(size=<span>12</span>,hjust=<span>0.5</span>))<br><br><br><span>library</span>(patchwork)<br>g1+g2<br></code></pre><p data-tool="mdnice编辑器">这里比上面的dotplot图更能清晰的看出，单核细胞和聚合细胞以及DC-like在NonResponder中更多一点。</p><figure data-tool="mdnice编辑器"><img data-ratio="0.6046296296296296" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIHWCn5ibh5FT6UHwyGb37pJMQlANspMiaE2pHFTK9WnbfaEKm6bImHTEQ/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIHWCn5ibh5FT6UHwyGb37pJMQlANspMiaE2pHFTK9WnbfaEKm6bImHTEQ/640?wx_fmt=png"></figure><h2 data-tool="mdnice编辑器"><span><span>3</span></span><span>scRNA-seq analysis of IL8 gene expression in immune subsets from matched intratumoral and peripheral blood leukocytes from patients with RCC and association of tumor IL8 gene expression with clinical outcomes in muC and mRCC</span></h2><p data-tool="mdnice编辑器">接下来是另一个数据集RCC对于上面结论的进一步佐证，当然，方法和上面一样。连图形都是再来一遍。<img data-ratio="0.7092592592592593" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIkVUdc1ns7EL0wxLg9SRdKftrib8u8633dMdlePLvPgHuVC0bHROEZjg/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIkVUdc1ns7EL0wxLg9SRdKftrib8u8633dMdlePLvPgHuVC0bHROEZjg/640?wx_fmt=png"></p><figure data-tool="mdnice编辑器"><img data-ratio="0.5879629629629629" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIKPdNOGibBYo0ZWLHbRgLjhqQj31KPnWv8cUgtEZOR5RDy5VzdIUQh2A/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIKPdNOGibBYo0ZWLHbRgLjhqQj31KPnWv8cUgtEZOR5RDy5VzdIUQh2A/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code><span>###### step5:检查常见分群情况  ######</span><br>rm(list=ls())<br>setwd(<span>'../'</span>)<br>dir.create(<span>"3-cell"</span>)<br>setwd(<span>"3-cell"</span>) <br>sce.all=readRDS(<span>"../2-harmony/sce.all_int.rds"</span>)<br>colnames(sce.all@meta.data)<br>table(sce.all$orig.ident)<br>sce.all$orig.ident=str_split(sce.all$orig.ident ,<span>'[_]'</span>,simplify = <span>T</span>)[,<span>2</span>]<br>table(sce.all$orig.ident)<br>p1 = DimPlot(sce.all, reduction = <span>"umap"</span>, group.by = <span>"sample"</span>,raster=<span>FALSE</span>,cols = c(<span>'red'</span>,<span>'green'</span>)) <br><span>#ggsave('umap_by_patient.pdf',width = 9,height = 7)</span><br>col = c(<span>'#DF4298'</span>, <span>'#7C388C'</span>, <span>'#D58EBE'</span> ,  <span>'#746FB4'</span>,<span>'#3A53A4'</span>,<span>'#FEC026'</span> ,<span>'#F57F20'</span> ,<span>'#CB952B'</span> , <span>'#A1CC59'</span>,<span>'#186533'</span>)<br>p2 = DimPlot(sce.all, reduction = <span>"umap"</span>, group.by = <span>"orig.ident"</span>,raster=<span>FALSE</span>,cols = col) <br><span>#ggsave('umap_by_RNA_snn_res.0.3.pdf',width = 9,height = 7)</span><br><span>library</span>(patchwork)<br>p1+p2<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.6157407407407407" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIg27hXyP9bcGXg8zcrJSp6JwTibt98uWPpIJEr6HzBmKU1g0vH8RhnQA/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIg27hXyP9bcGXg8zcrJSp6JwTibt98uWPpIJEr6HzBmKU1g0vH8RhnQA/640?wx_fmt=png"></figure><ul data-tool="mdnice编辑器"><li><section>Human Primary Cell Atlas (HPCA): 基因表达集合(GEO数据集)，包含713个微阵列样本，分类为38种主要细胞类型，进一步注释为169个子类型 (Mabbott et al. 2013)。大多数标签指的是血液亚群，但细胞类型从其他组织也可用。</section></li><li><section>Blueprint/ENCODE 参考由 Blueprint (Martens and Stunnenberg 2013)和 ENCODE 项目(ENCODE Project Consortium 2012)产生的纯基质和免疫细胞的大量 RNA-seq 数据组成。</section></li><li><section>DICE参考由来自同名项目(Schmiedel等人，2018年)的排序细胞群的大量RNA-seq样本组成。</section></li><li><section>Novershtern参考资料(以前称为分化图)由GSE24759中排序的造血细胞群的微阵列数据集组成(Novershtern et al. 2011)。</section></li><li><section>Monaco参考由来自GSE107011的免疫细胞群的大量RNA-seq样本组成(Monaco et al. 2019)。</section></li></ul><pre data-tool="mdnice编辑器"><span></span><code>rm(list=ls())<br>options(stringsAsFactors = <span>F</span>)<br><span>library</span>(Seurat)<br><span>library</span>(ggplot2)<br><span>library</span>(clustree)<br><span>library</span>(cowplot)<br><span>library</span>(dplyr)<br><span>library</span>(SingleR)<br><span>library</span>(celldex)<br><span>library</span>(singleseqgset)<br><span>library</span>(devtools)<br>getwd()<br>setwd(<span>'3-cell/'</span>)<br>sce=readRDS( <span>"../2-harmony/sce.all_int.rds"</span>)<br>sce<br><br><span># singleR注释</span><br><span>#hpca.se &lt;- HumanPrimaryCellAtlasData()</span><br><span>#save(hpca.se,file = 'hpca.RData')</span><br>load(<span>'../UC_IMvigor210/2-harmony/hpca.RData'</span>)<br><span>#bpe.se &lt;- BlueprintEncodeData()</span><br><span>#save(bpe.se,file = 'bpe.RData')</span><br>load(<span>'../UC_IMvigor210/2-harmony/bpe.RData'</span>)<br>unique(hpca.se$label.main)<br>unique(hpca.se$label.fine)<br>unique(bpe.se$label.main)<br>unique(bpe.se$label.fine)<br><span>##这个运行太太太慢了</span><br>DICE &lt;- DatabaseImmuneCellExpressionData() <br><span>#save(DICE,file = 'DICE.RData')</span><br>NHD &lt;- NovershternHematopoieticData() <br><span>#save(NHD,file = 'NHD.RData')</span><br>MID &lt;- MonacoImmuneData()<br><span>#save(MID,file = 'MID.RData')</span><br>unique(NHD$label.main)<br>unique(NHD$label.fine)<br>unique(MID$label.main)<br>unique(MID$label.fine)<br>anno &lt;- SingleR(sce@assays$RNA@data,<br>                ref = list(BP=bpe.se,HPCA=hpca.se),<br>                labels = list(bpe.se$label.fine,hpca.se$label.fine),<br>                clusters = sce@meta.data$seurat_clusters<br>)<br><span>##我开始怀疑了，每次更新什么东西，matrix这个包就会更新版本，然后，需要就会持续报错</span><br><span>#上一次遇到这个情况还是在FindNeighbors</span><br><br>plotScoreHeatmap(anno,clusters = anno@rownames,show_colnames = <span>T</span>)<br>table(anno$labels)<br><br>celltype = data.frame(ClusterID=rownames(anno), <br>                      celltype=anno$labels, <br>                      stringsAsFactors = <span>F</span>) <br><br>sce@meta.data$singleR = celltype[match(sce@meta.data$seurat_clusters,celltype$ClusterID),<span>'celltype'</span>]<br>P &lt;- DimPlot(sce, reduction = <span>"umap"</span>, group.by = <span>"singleR"</span>)<br>P<br></code></pre><p data-tool="mdnice编辑器">和文中的细胞亚群相比，相差并不大。</p><figure data-tool="mdnice编辑器"><img data-ratio="0.8685185185185185" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIHkwyUpEZKtLBucSZuMRGjX7iauFdfC7suCACFcz1I69zawrUdSwEDkA/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los98Wlq7do1rOYclgO0yo4DIHkwyUpEZKtLBucSZuMRGjX7iauFdfC7suCACFcz1I69zawrUdSwEDkA/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">后面的分析流程也是一样，差异分析火山图，细胞比例，只是复制粘贴再来一遍，这里就不再赘述了。</p><p data-tool="mdnice编辑器"><strong><em>提问：作者是怎么挑到这个基因的？</em></strong></p><p data-tool="mdnice编辑器"><strong>凭借文中的只言片语，较难理解为什么IL-8可能成为一个预测性的生物标志物，还得结合文中的一些生存分析曲线，但是由于临床试验的数据很多并不公开，我也没找到另外两个数据集，所以并没有进行复现，</strong></p><p data-tool="mdnice编辑器">文章的结论是在mUC和mRCC患者中，血浆、pbmc和肿瘤中高水平的IL8与atezolizumab疗效降低相关，即使在典型的CD8+ T细胞炎症的肿瘤中也是如此。mUC患者低基线pIL8与atezolizumab和化疗反应增加相关。<em>在接受atezolizumab治疗而非化疗的情况下，经历了治疗期pIL8下降的mUC患者表现出了改善的总生存期</em>。scRNAseq显示，IL8主要表达于循环细胞和瘤内髓系细胞中，IL8的高表达与抗原递呈机制的下调有关。能够逆转IL8介导的髓系炎症影响的疗法对于改善接受免疫检查点抑制剂治疗的患者的预后至关重要。</p></section><p><br></p><p><mp-style-type data-value="10000"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/Qw0ylKfrkxtOVIkxNITuEA",target="_blank" rel="noopener noreferrer">原文链接</a>
