---
title: "拒绝千篇一律！R包Ragas带你打开单细胞聚类分析及可视化的新大门"
date: 2024-08-20T14:07:35Z
draft: ["false"]
tags: [
  "fetched",
  "R语言学徒"
]
categories: ["Acdemic"]
---
拒绝千篇一律！R包Ragas带你打开单细胞聚类分析及可视化的新大门 by R语言学徒
------
<div><p><img data-galleryid="" data-imgfileid="100007793" data-ratio="0.2222222222222222" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEUckerRictfTicSHoNskkHZjoa7nSeGI9ticg8pf3FOEkewHQTNLbNhdKK36fxBPpNLzVm0W8LjZYKEg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="900" src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEUckerRictfTicSHoNskkHZjoa7nSeGI9ticg8pf3FOEkewHQTNLbNhdKK36fxBPpNLzVm0W8LjZYKEg/640?wx_fmt=png&amp;from=appmsg"><span>         </span><p> </p></p><p><span>前言</span></p><section><span>今天小师妹又和大家见面了，跟随师妹的步伐来给自己空瘪的皮囊注入点知识的能量吧！接触单细胞分析的小伙伴们都知道亚簇分析是改善单细胞RNA-Seq数据聚类和表征的有力手段。然而，目前还没有工具可以系统地整合来自多个子集群的结果，这为下游分析中的准确数据量化、可视化和解释带来了障碍。所以今天小师妹给你介绍的R包Ragas一定要睁大眼睛好好学一下哦！它集成了多级子聚类对象，用于简化分析和可视化。实施了一种新的数据结构，以无缝连接和组装来自不同子聚类级别的各种单细胞分析，以及几个新的或增强的可视化功能。此外，还开发了一种重新投影算法来整合来自多个亚簇的最近邻图，以最大限度地提高它们在组合细胞嵌入上的可分离性，从而显着改善了稀有和均质亚群的呈现。此外，Ragas包有与Seurat对象兼容的统一数据结构，以协调单个单元数据的分析和可视化，是不是用起来更方便了呢？</span><span>练了十年生信分析的师妹对于单细胞的知识已经如鱼得水，相信有师妹的加持一定能让大家学会！没有时间学习且有单细胞分析的小伙伴们快来滴滴师妹噢！从分析到可视化直到你满意为止！</span><page></page></section><p><span><img data-imgfileid="100010251" data-ratio="0.5923076923076923" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEWENU6cW4mhGiblcQ2EQjQic0AnKgvdCNwwQxh8FibDfghF0ib57UxfibDwibL4wtTf72qPG8QgdojRxswA/640?wx_fmt=png" data-type="png" data-w="520" src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEWENU6cW4mhGiblcQ2EQjQic0AnKgvdCNwwQxh8FibDfghF0ib57UxfibDwibL4wtTf72qPG8QgdojRxswA/640?wx_fmt=png"></span></p><section><span>单细胞的数据实在是太大了，一个样本要吃掉你2-4G的内存，电脑吃不消的小伙伴们建议使用服务器，欢迎大家联系小师妹，保证服务器的性价比最高哦！</span></section><p><img data-galleryid="" data-imgfileid="100009666" data-ratio="0.4444444444444444" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_jpg/oCSOtP61aEWsBEr9hm2ibYhjkWqcyBSJw4aiaaoaLTeEmryREtHqskAFUol6ysOoEXWYpDYCntLhTv89QNYGFSdg/640?wx_fmt=jpeg&amp;from=appmsg" data-type="jpeg" data-w="900" src="https://mmbiz.qpic.cn/sz_mmbiz_jpg/oCSOtP61aEWsBEr9hm2ibYhjkWqcyBSJw4aiaaoaLTeEmryREtHqskAFUol6ysOoEXWYpDYCntLhTv89QNYGFSdg/640?wx_fmt=jpeg&amp;from=appmsg"><span>     </span><p> </p></p><p><span>代码实现</span></p><p><span>加载R包并构建Pi对象</span></p><p><span>公众号回复666获取代码、示例数据等哦~~编号：240820<br></span></p><p><span>##本实验所使用的数据集已经集成在R包Ragas里面了，小伙伴们直接加载R包读取数据就可以啦。</span></p><section><ul><li><li><li><li><li><li><li></ul><pre data-lang="shell"><code><span><span>#</span><span><span>#首先，加载 Ragas 包</span></span></span></code><code><span>library(Ragas)</span></code><code><span>csle.pbmc.small</span></code><code><span><span>#</span><span><span># An object of class Seurat</span></span></span></code><code><span><span>#</span><span><span># 1503 features across 3000 samples within 1 assay</span></span></span></code><code><span><span>#</span><span><span># Active assay: RNA (1503 features, 1501 variable features)</span></span></span></code><code><span><span>#</span><span><span>#  3 dimensional reductions calculated: pca, harmony, umap            </span></span></span></code></pre></section><p><span>上面的 Seurat 对象是 Ragas 中的示例对象之一 包，这是 scRNA-Seq 数据分析的一部分，PBMC 来自 健康 （cHD） 和系统性红斑狼疮 （cSLE） 儿童 （https://www.nature.com/articles/s41590-020-0743-0）。执行 Ragas 分析，我们需要创建一个基于Seurat对象。集成后对象是单细胞数据对象已规范化、批量集成并执行 UMAP 和 聚类分析。注意：在我们的分析中，我们使用 Harmony 进行批量集成。</span></p><p><span>出于演示目的，我们预先构建了一个名为 Pi 的对象 “csle.pbmc.pi”基于示例 Seurat 对象 （csle.pbmc.small）， 它与 Ragas 包一起发布，作为示例数据的一部分。</span></p><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="shell"><code><span>csle.pbmc.pi</span></code><code><span><span>#</span><span><span># An object of class Pi</span></span></span></code><code><span><span>#</span><span><span># 6 fields in the object: seurat.obj, exp.freq, markers, ds, cell.prop, parent.meta.data.</span></span></span></code><code><span><span>#</span><span><span># The following fields have been processed:</span></span></span></code><code><span><span>#</span><span><span>#  seurat.obj: A Seurat object of 1503 features and 2901 cells.</span></span></span></code><code><span><span>#</span><span><span>#      1 assay: RNA, and 4 reductions: pca, harmony, umap, rp</span></span></span></code><code><span><span>#</span><span><span>#  exp.freq: A list of numeric matrices containing per gene expression frequencies</span></span></span></code><code><span><span>#</span><span><span>#      1 analysis run:</span></span></span></code><code><span><span>#</span><span><span>#        ExpFreq|subcluster_idents|cutoff=0</span></span></span></code><code><span><span>#</span><span><span>#  markers: A list of data frames containing marker results.</span></span></span></code><code><span><span>#</span><span><span>#      1 analysis run:</span></span></span></code><code><span><span>#</span><span><span>#        Markers|subcluster_idents|AllMarkers|test.use=wilcox    </span></span></span></code><code><span><span>#</span><span><span>#  ds: A list of lists and data frames containing pseudobulk analysis results.</span></span></span></code><code><span><span>#</span><span><span>#      1 analysis run:</span></span></span></code><code><span><span>#</span><span><span>#        DS|subcluster_idents|edgeR|group=Groups;sample=Names;gp1=cSLE;gp2=cHD;contrast=cSLE-cHD</span></span></span></code><code><span><span>#</span><span><span>#  cell.prop: A list of data frames containing cell proportion analysis results</span></span></span></code><code><span><span>#</span><span><span>#      1 analysis run:</span></span></span></code><code><span><span>#</span><span><span>#        CellProp|subcluster_idents|unpooled|unpool_by_Names|group=Groups</span></span></span></code><code><span><span>#</span><span><span># Metadata from the parent object provided? No</span></span></span></code><code><span><span>#</span><span><span># Subclusters integrated? Yes</span></span></span></code><code><span>           </span></code><code><span>my.pbmc.pi &lt;- CreatePostIntegrationObject(object = csle.pbmc.small)</span></code><code><span><span>#</span><span><span># Post-integration object created</span></span></span></code><code><span>my.pbmc.pi</span></code><code><span><span>#</span><span><span># An object of class Pi</span></span></span></code><code><span><span>#</span><span><span># 6 fields in the object: seurat.obj, exp.freq, markers, ds, cell.prop, parent.meta.data.</span></span></span></code><code><span><span>#</span><span><span># The following field has been processed:</span></span></span></code><code><span><span>#</span><span><span>#  seurat.obj: A Seurat object of 1503 features and 3000 cells.</span></span></span></code><code><span><span>#</span><span><span>#      1 assay: RNA, and 3 reductions: pca, harmony, umap</span></span></span></code><code><span><span>#</span><span><span># Metadata from the parent object provided? No    </span></span></span></code><code><span><span>#</span><span><span># Subclusters integrated? No</span></span></span></code></pre></section><p><span>上面的代码创建了一个仅包含 Seurat 的“最小”Pi 对象 对象。如果用户具有子集群中的其他对象 分析，可以遵循子聚类分析 用于将优化的子集群标识重新投影到 PBMC 的 Vignette。创建 Pi 对象后，可以先运行 UMAP 并检查 对象中所有单元格的簇。</span></p><section><ul><li></ul><pre data-lang="cs"><code><span>RunDimPlot(<span>object</span> = my.pbmc.pi)</span></code></pre></section><p><span></span></p><p><span><img data-imgfileid="100010252" data-ratio="0.875" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEWENU6cW4mhGiblcQ2EQjQic0F0Mia84GaApgogMErr5QjKSiamEtO5qXS1htTa56hhH1uiblZtKhxhlKg/640?wx_fmt=png" data-type="png" data-w="384" src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEWENU6cW4mhGiblcQ2EQjQic0F0Mia84GaApgogMErr5QjKSiamEtO5qXS1htTa56hhH1uiblZtKhxhlKg/640?wx_fmt=png"></span></p><section><ul><li><li></ul><pre data-lang="cs"><code><span>RunDimPlot(<span>object</span> = my.pbmc.pi,</span></code><code><span>           <span>group</span>.<span>by</span> = <span>"cluster.annotation"</span>)</span></code></pre></section><p><page></page></p><p><span><img data-imgfileid="100010249" data-ratio="0.8" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEWENU6cW4mhGiblcQ2EQjQic0v3ksQx0aDCBwfVYicDkiaJqPC878SFzzvJ7GQplm7k84AKEptbdfvxNA/640?wx_fmt=png" data-type="png" data-w="480" src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEWENU6cW4mhGiblcQ2EQjQic0v3ksQx0aDCBwfVYicDkiaJqPC878SFzzvJ7GQplm7k84AKEptbdfvxNA/640?wx_fmt=png"></span></p><p><span>分析数据</span></p><p><span>Ragas 提供了一个用户友好的界面来集成和可视化 多种单细胞分析功能，包括 Marker 分析，使用伪散体法进行差分状态分析，以及差分细胞比例分析。</span><span>    </span><p> </p></p><p><span>标记分析和矩阵图：用户可以通过运行 包装函数 RunFindAllMarkers，用于存储标记结果 在 Pi 对象中，以便以后可视化。</span></p><section><ul><li><li><li></ul><pre data-lang="javascript"><code><span>my.pbmc.pi &lt;- RunFindAllMarkers(my.pbmc.pi,</span></code><code><span>                                ident = <span>"cluster.annotation"</span>)</span></code><code><span>my.pbmc.pi</span></code></pre></section><p><span></span></p><p><span>运行 RunFindAllMarkers 后，一个名为“markers”的新数据字段将 被添加到当前 Pi 对象中，该对象包含具有长 唯一标识符称为 “标记“cluster.annotation|AllMarkers|test.use=wilcox“。这个独一无二的 标识符可用于检索标记结果。</span><page></page></p><section><ul><li></ul><pre data-lang="css"><code><span><span>my</span><span>.pbmc</span><span>.pi</span><span>[["markers"]</span>]          </span></code></pre></section><p><span>矩阵图：标记分析完成后，每个聚类的顶级标记可以使用矩阵图绘制</span></p><section><ul><li><li><li><li></ul><pre data-lang="javascript"><code><span>RunMatrixPlot(my.pbmc.pi,</span></code><code><span>              markers.key = <span>"Markers|cluster.annotation|AllMarkers|test.use=wilcox"</span>,</span></code><code><span>              column.anno.name.rot = <span>45</span>,</span></code><code><span>              heatmap.height = <span>6</span>)</span></code></pre></section><p><span><img data-imgfileid="100010250" data-ratio="0.4" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEWENU6cW4mhGiblcQ2EQjQic0j3Qia5ZBJyibkia6IYQMwo9UehSpNOUHVhC6TEs8QicIKcPU97FFOCldbg/640?wx_fmt=png" data-type="png" data-w="960" src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEWENU6cW4mhGiblcQ2EQjQic0j3Qia5ZBJyibkia6IYQMwo9UehSpNOUHVhC6TEs8QicIKcPU97FFOCldbg/640?wx_fmt=png"></span></p><p><span>伪体积分析</span></p><p><span>单细胞研究的另一项重要任务是识别 各簇间差异表达基因，也称为差分状态 （DS） 分析。我们使用 函数从 Muscat 包到对多样本多组scRNA-Seq进行伪批量分析设计。</span></p><p><span>在运行伪批量分析之前，我们需要先进行表达 频率分析。对给定簇的低表达基因 将在最终的伪批量输出中被过滤掉。</span></p><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="cs"><code><span>my.pbmc.pi &lt;- CalculateExpFreqs(my.pbmc.pi,</span></code><code><span>                                ident = <span>"cluster.annotation"</span>,</span></code><code><span>                                verbose = FALSE)    </span></code><code><span>my.pbmc.pi</span></code><code><span><span>#批量分析</span></span></code><code><span>my.pbmc.pi &lt;- RunPseudobulkAnalysis(<span>object</span> = my.pbmc.pi,</span></code><code><span>                                    ident.<span>var</span> = <span>"cluster.annotation"</span>,</span></code><code><span>                                    <span>group</span>.<span>var</span> = <span>"Groups"</span>,</span></code><code><span>                                    sample.<span>var</span> = <span>"Names"</span>,</span></code><code><span>                                    <span>group</span><span>.1</span> = <span>"cSLE"</span>,</span></code><code><span>                                    <span>group</span><span>.2</span> = <span>"cHD"</span>)</span></code><code><span>my.pbmc.pi</span></code><code><span><span>#使用 DSDotPlot 可视化伪批量结果</span></span></code><code><span>RunDSDotPlot(<span>object</span> = my.pbmc.pi,</span></code><code><span>             exp.freq.key = <span>"ExpFreq|cluster.annotation|cutoff=0"</span>,</span></code><code><span>             ds.key = <span>"DS|cluster.annotation|edgeR|group=Groups;sample=Names;gp1=cSLE;gp2=cHD;contrast=cSLE-cHD"</span>,</span></code><code><span>             p.filter = <span>0.05</span>,</span></code><code><span>             FC.filter = <span>4</span>,</span></code><code><span>             to.adjust = TRUE</span></code><code><span>                    )    </span></code></pre></section><p><page></page></p><p><span><img data-imgfileid="100010253" data-ratio="1" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEWENU6cW4mhGiblcQ2EQjQic0QtEujLdEBHiaEtXuhPcdv6AnI5zHQTdz3wfTN45rqTf8L79K93axKVQ/640?wx_fmt=png" data-type="png" data-w="768" src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEWENU6cW4mhGiblcQ2EQjQic0QtEujLdEBHiaEtXuhPcdv6AnI5zHQTdz3wfTN45rqTf8L79K93axKVQ/640?wx_fmt=png"></span></p><p><span>比例分析</span></p><p><span>Ragas还提供了一组丰富的函数来可视化单元格比例数据。比例分析有两种方法:池化和非池化。混合比例分析在单细胞研究中得到了广泛的应用，但它的主要缺点是由于混合了所有样本的所有细胞而失去了对生物复制的跟踪，这使得很难评估细胞比例的可变性。因此，对于多样本多组设计，非池化是首选。</span></p><section><ul><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="cs"><code><span><span>#使用“池化”图时，用户可以进一步选择其中之一 以下两种方法计算比例：按“聚类”或按 “组”。</span></span></code><code><span>RunProportionPlot(csle.pbmc.small, <span>## input can be a Seurat object</span></span></code><code><span>                  ident = <span>"cluster.annotation"</span>,    </span></code><code><span>                  <span>group</span>.<span>by</span> = <span>"Groups"</span>,</span></code><code><span>                  method = <span>"pooled"</span>,</span></code><code><span>                  pooled.prop.<span>by</span> = <span>"cluster"</span>, <span>## default</span></span></code><code><span>                  axis.text.angle = <span>45</span>,</span></code><code><span>                  axis.text.size = <span>8</span>,</span></code><code><span>                  <span>return</span>.<span>value</span> = <span>'ggplot'</span></span></code><code><span>                  )</span></code></pre></section><p><span></span></p><p><span><img data-imgfileid="100010254" data-ratio="0.4166666666666667" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEWENU6cW4mhGiblcQ2EQjQic0L81fyulYhmbRqrR1JAiat0ZugruVLhowHGriaDHq8UZKf8B9K6nHpbfA/640?wx_fmt=png" data-type="png" data-w="576" src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEWENU6cW4mhGiblcQ2EQjQic0L81fyulYhmbRqrR1JAiat0ZugruVLhowHGriaDHq8UZKf8B9K6nHpbfA/640?wx_fmt=png"></span></p><p><span>如果我们交换参数“ident”和“group.by”的输入，这将 导致显示所有聚类的单元格比例的不同条形图 由每组中的细胞总数归一化。这是 本质上是“按组”合并单元格比例的堆叠版本，但条形图是堆叠的。</span></p><section><ul><li><li><li><li><li><li></ul><pre data-lang="kotlin"><code><span>my.pbmc.pi &lt;- RunProportionPlot(my.pbmc.pi, ## <span>if</span> input <span>is</span> a Pi <span>object</span>, an updated Pi <span>object</span> will be the returned <span>by</span> <span>default</span></span></code><code><span>                                ident = <span>"Groups"</span>,</span></code><code><span>                                group.<span>by</span> = <span>"cluster.annotation"</span>,</span></code><code><span>                                method = <span>"pooled"</span>,</span></code><code><span>                                pooled.prop.<span>by</span> = <span>"cluster"</span>, ## <span>default</span></span></code><code><span>                                axis.text.size = <span>8</span>)</span></code></pre></section><p><page></page></p><p><span><img data-imgfileid="100010255" data-ratio="1.25" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEWENU6cW4mhGiblcQ2EQjQic0Mq9NZaGnlaroxOCGke2V9mBfbeJuqeXgtM027GiahqEDq5R4q4riaZoA/640?wx_fmt=png" data-type="png" data-w="384" src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEWENU6cW4mhGiblcQ2EQjQic0Mq9NZaGnlaroxOCGke2V9mBfbeJuqeXgtM027GiahqEDq5R4q4riaZoA/640?wx_fmt=png"></span></p><p><span>在另一种情况下，用户可以按组绘制单元格比例， 这意味着给定集群中给定组的细胞计数为由每组的细胞总数归一化。</span></p><section><ul><li><li><li><li><li><li></ul><pre data-lang="cs"><code><span>my.pbmc.pi &lt;- RunProportionPlot(my.pbmc.pi,</span></code><code><span>                                ident = <span>"cluster.annotation"</span>,</span></code><code><span>                                <span>group</span>.<span>by</span> = <span>"Groups"</span>,</span></code><code><span>                                method = <span>"pooled"</span>,</span></code><code><span>                                pooled.prop.<span>by</span> = <span>"group"</span>,</span></code><code><span>                                axis.text.size = <span>8</span>)</span></code></pre></section><p><span></span></p><p><span><img data-imgfileid="100010256" data-ratio="0.6" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEWENU6cW4mhGiblcQ2EQjQic0R2SDoBb8jLkPUajz4DNujm5SLLe8yFShHkKkpkA8dROAiaYjczhV6vA/640?wx_fmt=png" data-type="png" data-w="480" src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEWENU6cW4mhGiblcQ2EQjQic0R2SDoBb8jLkPUajz4DNujm5SLLe8yFShHkKkpkA8dROAiaYjczhV6vA/640?wx_fmt=png"></span><page></page></p><p><span>未池化比例图</span></p><p><span>未合并比例图适用于多样本、多组 研究设计，允许比较感兴趣的组，以及评估它们在内部的变异性每个组。</span></p><section><ul><li><li><li><li><li><li><li></ul><pre data-lang="cs"><code><span>my.pbmc.pi &lt;- RunProportionPlot(my.pbmc.pi,</span></code><code><span>                  ident = <span>"cluster.annotation"</span>,</span></code><code><span>                  <span>group</span>.<span>by</span> = <span>"Groups"</span>,</span></code><code><span>                  method = <span>"unpooled"</span>,</span></code><code><span>                  unpool.<span>by</span> = <span>"Names"</span>,</span></code><code><span>                  unpool.ncol = <span>3</span>,</span></code><code><span>                  title.text.size = <span>6</span>)</span></code></pre></section><p><span></span></p><p><span><img data-imgfileid="100010257" data-ratio="1" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEWENU6cW4mhGiblcQ2EQjQic0REeCpfMRLA5ypibYxKsWiaQicxojdpAOR6QOLUNSrsdOTX7VPA5xgNMww/640?wx_fmt=png" data-type="png" data-w="480" src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEWENU6cW4mhGiblcQ2EQjQic0REeCpfMRLA5ypibYxKsWiaQicxojdpAOR6QOLUNSrsdOTX7VPA5xgNMww/640?wx_fmt=png"></span><page></page></p><p><span>#未池化条形图</span></p><section><ul><li><li><li><li><li><li><li><li></ul><pre data-lang="cs"><code><span>my.pbmc.pi &lt;- RunProportionPlot(my.pbmc.pi,</span></code><code><span>                                ident = <span>"cluster.annotation"</span>,</span></code><code><span>                                <span>group</span>.<span>by</span> = <span>"Groups"</span>,</span></code><code><span>                                method = <span>"unpooled"</span>,</span></code><code><span>                                unpool.<span>by</span> = <span>"Names"</span>,</span></code><code><span>                                unpool.plot.type = <span>"barplot"</span>,</span></code><code><span>                                unpool.ncol = <span>3</span>,</span></code><code><span>                                title.text.size = <span>6</span>)</span></code></pre></section><p><span></span></p><p><span><img data-imgfileid="100010258" data-ratio="1" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEWENU6cW4mhGiblcQ2EQjQic07dqiaBFe48ed88zrcjZckaKrbVlFpWicia3Dxdsr7oSOnYl3QpzgYUoJw/640?wx_fmt=png" data-type="png" data-w="480" src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEWENU6cW4mhGiblcQ2EQjQic07dqiaBFe48ed88zrcjZckaKrbVlFpWicia3Dxdsr7oSOnYl3QpzgYUoJw/640?wx_fmt=png"></span></p><p><span>其他可视化</span><page></page></p><p><span>#带注释的点阵图</span></p><section><ul><li><li><li><li><li><li><li><li></ul><pre data-lang="cs"><code><span>my.list &lt;- list(Monocytes = c(<span>"CD14"</span>,<span>"FCGR3A"</span>),</span></code><code><span>                B = <span>"MS4A1"</span>,</span></code><code><span>                T = c(<span>"CCR7"</span>, <span>"CD8A"</span>),</span></code><code><span>                NK = <span>"NKG7"</span>)</span></code><code><span>RunAnnotatedDotPlot(<span>object</span> = my.pbmc.pi,</span></code><code><span>                    annotations = my.list,</span></code><code><span>                    annotation.cols = c( <span>'Monocytes'</span>= <span>'tomato'</span>, <span>'B'</span> = <span>'seagreen'</span>, <span>'T'</span> = <span>'steelblue'</span>, <span>'NK'</span> = <span>'purple'</span>),</span></code><code><span>                    <span>group</span>.<span>by</span> = <span>"cluster.annotation"</span>)</span></code></pre></section><p><span></span></p><p><span><img data-imgfileid="100010259" data-ratio="0.625" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEWENU6cW4mhGiblcQ2EQjQic0lm7ToLFl68caiaNkEp2JFTaa4mkoFq8sYOvXRVC6rzbxo6SQMUMfF1Q/640?wx_fmt=png" data-type="png" data-w="768" src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEWENU6cW4mhGiblcQ2EQjQic0lm7ToLFl68caiaNkEp2JFTaa4mkoFq8sYOvXRVC6rzbxo6SQMUMfF1Q/640?wx_fmt=png"></span></p><p><span>#堆叠小提琴图</span></p><section><ul><li><li><li><li><li></ul><pre data-lang="cs"><code><span>RunStackedVlnPlot(my.pbmc.pi,</span></code><code><span>                  ident = <span>"cluster.annotation"</span>,    </span></code><code><span>                  features = my.list,</span></code><code><span>                  color.<span>by</span> = <span>"mean.exp"</span>,</span></code><code><span>                  column.names.rotation = <span>90</span>)</span></code></pre></section><p><span></span></p><p><span><img data-imgfileid="100010263" data-ratio="1.75" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEWENU6cW4mhGiblcQ2EQjQic0U27a3RjgKNKndjr1N1VcSiao5PKOl1jYD2SyVGvdB0at48xBgqlPFgQ/640?wx_fmt=png" data-type="png" data-w="384" src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEWENU6cW4mhGiblcQ2EQjQic0U27a3RjgKNKndjr1N1VcSiao5PKOl1jYD2SyVGvdB0at48xBgqlPFgQ/640?wx_fmt=png"></span></p><section><ul><li><li><li><li><li><li></ul><pre data-lang="cs"><code><span>RunStackedVlnPlot(my.pbmc.pi,</span></code><code><span>                  ident = <span>"cluster.annotation"</span>,</span></code><code><span>                  features = my.list,</span></code><code><span>                  split.<span>by</span> = <span>"Groups"</span>,</span></code><code><span>                  color.<span>by</span> = <span>"split.var"</span>,</span></code><code><span>                  column.names.rotation = <span>45</span>)</span></code></pre></section><p><page></page></p><p><span><img data-imgfileid="100010262" data-ratio="1.3333333333333333" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEWENU6cW4mhGiblcQ2EQjQic0Vib5T6Ia4r2oED4vtokUbHGZ34gYHUuh297t0jqsSNibcKTPtRs69ZMQ/640?wx_fmt=png" data-type="png" data-w="576" src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEWENU6cW4mhGiblcQ2EQjQic0Vib5T6Ia4r2oED4vtokUbHGZ34gYHUuh297t0jqsSNibcKTPtRs69ZMQ/640?wx_fmt=png"></span></p><section><ul><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="swift"><code><span>##汇总热图</span></code><code><span>features &lt;- list(<span>"T cell"</span> = <span>c</span>(<span>"CD8A"</span>, <span>"CD8B"</span>),</span></code><code><span>                <span>ISG</span> = <span>c</span>(<span>"ISG15"</span>,<span>"ISG20"</span>,<span>"IFI44L"</span>, <span>"IFI6"</span>, <span>"MX1"</span>),</span></code><code><span>                <span>"Housekeeping"</span> = <span>c</span>(<span>"ACTB"</span>))</span></code><code><span><span>RunSummarizedHeatmap</span>(object = my.pbmc.pi,</span></code><code><span>                     ident = <span>"cluster.annotation"</span>,</span></code><code><span>                     features = features,    </span></code><code><span>                     <span>split</span>.by = <span>"Groups"</span>,</span></code><code><span>                     column.names.rotation = <span>30</span>,</span></code><code><span>                     heatmap.width = <span>20</span>)</span></code></pre></section><p><img data-imgfileid="100010260" data-ratio="0.4166666666666667" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEWENU6cW4mhGiblcQ2EQjQic0top2liaJIkdt0mXVOmPk6icSA2fg2FA4MPxmXQUNRTM4Sdic2JfONNUicA/640?wx_fmt=png" data-type="png" data-w="1152" src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEWENU6cW4mhGiblcQ2EQjQic0top2liaJIkdt0mXVOmPk6icSA2fg2FA4MPxmXQUNRTM4Sdic2JfONNUicA/640?wx_fmt=png"><span></span></p><p><span>##表达式图</span></p><section><ul><li><li><li><li><li></ul><pre data-lang="cs"><code><span>RunExpressionPlot(<span>object</span> = my.pbmc.pi,</span></code><code><span>                  feature = <span>"ISG15"</span>,</span></code><code><span>                  ident = <span>"cluster.annotation"</span>,</span></code><code><span>                  <span>group</span>.<span>by</span> = <span>"Names"</span>,</span></code><code><span>                  split.<span>by</span> = <span>"Groups"</span>)</span></code></pre></section><p><span></span></p><p><span><img data-imgfileid="100010261" data-ratio="0.6" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEWENU6cW4mhGiblcQ2EQjQic0dKpYuxe0TzibBGSlWUj2fVPVZTo33UwRFQQIAcgCPg32uLuiaPkA93CQ/640?wx_fmt=png" data-type="png" data-w="480" src="https://mmbiz.qpic.cn/sz_mmbiz_png/oCSOtP61aEWENU6cW4mhGiblcQ2EQjQic0dKpYuxe0TzibBGSlWUj2fVPVZTo33UwRFQQIAcgCPg32uLuiaPkA93CQ/640?wx_fmt=png"></span></p><p><span>数据导出</span></p><p><span>在 Ragas 中，我们提供了 ExportPiData 函数来导出分析结果。</span></p><section><ul><li><li><li><li><li></ul><pre data-lang="kotlin"><code><span>ExportPiData(<span>object</span> = csle.pbmc.pi,</span></code><code><span>             field = <span>'ds'</span>,</span></code><code><span>             key = <span>'DS|subcluster_idents|edgeR|group=Groups;sample=Names;gp1=cSLE;gp2=cHD;contrast=cSLE-cHD'</span>,</span></code><code><span>             file.prefix = <span>'pbmc.ds'</span> ## A mandatory argument. User should provide an informative prefix (e.g., <span>"csle.pbmc"</span>) <span>for</span> exported <span>data</span></span></code><code><span>)</span></code></pre></section><p><span></span></p><p><span>小结</span></p><p><span>小师妹向大家介绍了Ragas，这是一个基于R的框架，用于单细胞子集群数据的集成分析和可视化。Ragas提供了三个主要贡献：（i）定义了一个与Seurat对象兼容的统一数据结构，以协调单个单元数据的分析和可视化。（ii）开发了一种重新投影算法，将来自多个亚簇的细胞包埋组合在一起，该算法可以显示足够的结构粒度，用于稀有或同质亚群。（iii） 作为对现有Seurat可视化功能的补充，Ragas 提供了一个直观的界面来生成新的或增强的单细胞图。到这里师妹要和大家说再见了！</span><span>最后师妹给大家介绍一个云工具！同学们如果觉得自己的代码水平一般，对于很多的参数不知道怎么改，可以体验一下我们的云生信小工具，只需输入数据，即可轻松生成所需图表，字体大小、标题等也可一键更改。感兴趣的小伙伴去云生信（http://www.biocloudservice.com/home.html）体验一下吧！</span></p><section><section powered-by="xiumi.us"><section><section powered-by="xiumi.us"><section><svg viewbox="0 0 1 1"></svg></section></section></section><section><section powered-by="xiumi.us"><p>E</p></section></section><section><section powered-by="xiumi.us"><p>N</p></section></section><section><section powered-by="xiumi.us"><p>D</p></section></section><section><section powered-by="xiumi.us"><section><br></section></section></section></section></section><p><span>如果您的时间和精力有限或者缺乏相关经验，并且对生信分析和期刊推荐有所需要的话，我们非常乐意为您提供如下服务：免费思路评估、付费生信分析和方案设计以及付费选刊等，有意向的小伙伴欢迎咨询师妹哦！</span></p><p><span></span><span></span></p><section powered-by="xiumi.us"><p><img data-cropselx1="18" data-cropselx2="493" data-cropsely1="0" data-cropsely2="227" data-galleryid="" data-imgfileid="100010265" data-ratio="0.4444444444444444" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_jpg/oCSOtP61aEUqAk7f4mcKxwPx45IICz1wNazhXRTUEVx9ia4aYAawCG8smiaXiaJRzR0LiaEF0rUpevlfn6W3nnm4vA/640?wx_fmt=other&amp;from=appmsg&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1&amp;tp=webp" data-type="jpeg" data-w="900" src="https://mmbiz.qpic.cn/sz_mmbiz_jpg/oCSOtP61aEUqAk7f4mcKxwPx45IICz1wNazhXRTUEVx9ia4aYAawCG8smiaXiaJRzR0LiaEF0rUpevlfn6W3nnm4vA/640?wx_fmt=other&amp;from=appmsg&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1&amp;tp=webp"></p><p><br></p><p><br></p></section><section powered-by="xiumi.us"><section><section powered-by="xiumi.us"><section><section powered-by="xiumi.us"><section><svg viewbox="0 0 1 1"></svg></section></section></section></section></section><section><section powered-by="xiumi.us"><p><strong>往期推荐</strong></p></section></section><section><section powered-by="xiumi.us"><section><section powered-by="xiumi.us"><section><svg viewbox="0 0 1 1"></svg></section></section></section></section></section></section><section powered-by="xiumi.us"><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=Mzg5MDk3Mzg4OA==&amp;mid=2247492496&amp;idx=1&amp;sn=b3be5983f3f8f068ebdf1b773b7cb792&amp;chksm=cfd6d83ff8a15129493abd81502ab5c3128247ff20fd576c1ba880303f346110e8bee971478e&amp;scene=21#wechat_redirect" textvalue="别再说纯生信想不出好思路了！北大深圳医院黄晓彦团队基于UKB数据库,孟德尔随机化一套连招拿下13分+！" linktype="text" imgurl="" imgdata="null" data-itemshowtype="11" tab="innerlink" data-linktype="2" hasload="1">别再说纯生信想不出好思路了！北大深圳医院黄晓彦团队基于UKB数据库,孟德尔随机化一套连招拿下13分+！</a></p><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=Mzg5MDk3Mzg4OA==&amp;mid=2247492387&amp;idx=1&amp;sn=fc12277c2ebd379a297d0c7a2c96c970&amp;chksm=cfd6d88cf8a1519a3a6eb4938cb44ebcb2d1b857ffc558c1c033ac88913384cb2f11d03a00f4&amp;scene=21#wechat_redirect" textvalue="师妹手把手带你复现IF5.8《Cancer Cell International》杂志的高分文章中相关性热图和散点图" linktype="text" imgurl="" imgdata="null" data-itemshowtype="11" tab="innerlink" data-linktype="2" hasload="1">师妹手把手带你复现IF5.8《Cancer Cell International》杂志的高分文章中相关性热图和散点图</a><br></p><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=Mzg5MDk3Mzg4OA==&amp;mid=2247492197&amp;idx=1&amp;sn=40540e16ddc99310445236401505c47e&amp;chksm=cfd6d9caf8a150dc8f010750cacc4bd241ed68a8974593d499c1b34edf5f222de567732a5705&amp;scene=21#wechat_redirect" textvalue="是谁还不会用孟德尔随机化？这篇文章教你！高分SCI不是梦！！" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2" hasload="1">是谁还不会用孟德尔随机化？这篇文章教你！高分SCI不是梦！！</a><br></p><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=Mzg5MDk3Mzg4OA==&amp;mid=2247488665&amp;idx=1&amp;sn=6d7b7b1e8b36a92df691b0ab014e7663&amp;chksm=cfd52f36f8a2a620a790e74d99e3d8af879904a131c4e1f653448590453941b7ff356d486531&amp;scene=21#wechat_redirect" textvalue="师妹教你R包EnhancedVolcano超详尽，让你的火山图喷发姿势更帅" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2" hasload="1">师妹教你R包EnhancedVolcano超详尽，让你的火山图喷发姿势更帅</a></p></section><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/XV1q_0at0-m5MG934lrL-w",target="_blank" rel="noopener noreferrer">原文链接</a>
