---
title: "知识积累---整合细胞形态和空间转录组学深入分析肿瘤生态系统"
date: 2024-08-25T12:43:07Z
draft: ["false"]
tags: [
  "fetched",
  "单细胞空间交响乐"
]
categories: ["Acdemic"]
---
知识积累---整合细胞形态和空间转录组学深入分析肿瘤生态系统 by 单细胞空间交响乐
------
<div><h4>作者，Evil Genius</h4><h4>大家科研不要把自己逼得太紧，适当放松是为了更好的工作，比如最近很好的悟空，3个人凑钱买了一份玩一玩，第一关都过不去。</h4><h4>今日参考文献，</h4><p><img data-imgfileid="100008948" data-ratio="0.553" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmmeSvgfeoibNHo26XCyFx3ENPjyHqy9jctSZ5HtrQtt8NFjDcBGictgYydQibicjicCoKLVLXukVS4MMiaA/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="1000" src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmmeSvgfeoibNHo26XCyFx3ENPjyHqy9jctSZ5HtrQtt8NFjDcBGictgYydQibicjicCoKLVLXukVS4MMiaA/640?wx_fmt=other&amp;from=appmsg"></p><h4>虽然作者是中国人，但却是美国的课题组，</h4><p><img data-imgfileid="100008947" data-ratio="0.20566631689401887" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmmeSvgfeoibNHo26XCyFx3ENuxtNLmtDke4BQPWhKcibWNXGMibsM4DUibIavrJlSXykqOaTibxbNibAqbw/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="953" src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmmeSvgfeoibNHo26XCyFx3ENuxtNLmtDke4BQPWhKcibWNXGMibsM4DUibIavrJlSXykqOaTibxbNibAqbw/640?wx_fmt=other&amp;from=appmsg"></p><h4>我早已工作了，分享了很多的分析方法，但是基本不可能有机会参与原创性的工作了，大家基本上常用的软件，比如Seurat、cellchat等等，都是在美国人的中国人开发的，说明了只要有好的环境，国人的才智简直无可匹敌。</h4><h4>分析目标，<span>绘制癌细胞和TME成分，对细胞类型和状态进行分层，并分析细胞共定位。</span></h4><h4>知识背景</h4><ul><li><p>基于下一代测序(NGS)的平台，如Visium、GeoMx、Slide-Seq。基于杂交的方法，如MERFISH、seqFISH和CosMx。基于NGS的ST方法覆盖了整个转录组，但不是单细胞分辨率，而基于原位杂交的方法提供了优越的空间分辨率，但仅限于基因组的一小部分，限制了它们在基于发现的研究中的潜力（不过5000+探针还是可以的）。</p></li><li><p><strong><span>通过结合空间基因表达、组织组织学和癌症和TME细胞的先验知识，系统地分析癌细胞和TME细胞。</span></strong></p></li></ul><h4>METI 分析框架</h4><ul><li><p>重点关注从正常细胞到癌前细胞再到恶性细胞的进展，同时也检查每个组织切片内的淋巴细胞。</p></li><li><p><strong><span>METI的目标是精确识别各种细胞类型及其在TME中的各自状态。</span></strong></p></li></ul><blockquote><p><strong><span>模块1，METI识别正常细胞和癌前细胞.<br>模块2，METI识别肿瘤细胞富集区并表征其细胞状态的异质性<br>模块3, T细胞的空间定位<br>模块4,中识别其他免疫细胞，包括中性粒细胞、B细胞、浆细胞和巨噬细胞。<br>模块5、成纤维细胞的分析</span></strong></p></blockquote><p><img data-imgfileid="100008949" data-ratio="0.8231357552581262" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmmeSvgfeoibNHo26XCyFx3ENLTpXicKx6M0CSt4habBGsVO56VSYos1XCO46IDu23UE8e272SOhibaqQ/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="1046" src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmmeSvgfeoibNHo26XCyFx3ENLTpXicKx6M0CSt4habBGsVO56VSYos1XCO46IDu23UE8e272SOhibaqQ/640?wx_fmt=other&amp;from=appmsg"></p><h4>模块一、绘制正常细胞和癌前细胞</h4><h5>分析思路：<span>病理学家标注 + 正常细胞的形态学形状 + 基因表达数据</span></h5><ul><li><p>这种检测不能通过流行的空间聚类方法单独实现</p><p><br></p><p><img data-imgfileid="100008951" data-ratio="1.228448275862069" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmmeSvgfeoibNHo26XCyFx3ENka9ehOKWTLRYWkJicdgzQtScfnHewS74FsqrzbxWyfHGZwGraKVcPHA/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="1160" src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmmeSvgfeoibNHo26XCyFx3ENka9ehOKWTLRYWkJicdgzQtScfnHewS74FsqrzbxWyfHGZwGraKVcPHA/640?wx_fmt=other&amp;from=appmsg"></p></li></ul><h4>模块二、癌细胞结构域和异质性的鉴定</h4><p>大多数实体瘤起源于上皮细胞，被称为癌，包括胃癌、肺癌、膀胱癌、乳腺癌、前列腺癌和结肠癌，而其他一些实体瘤起源于其他类型的组织，包括肉瘤和黑色素瘤。无论其细胞来源如何，了解恶性细胞的分子特征和细胞异质性对于揭示肿瘤生长、侵袭、转移和治疗反应的机制至关重要。定量生物组织内细胞的空间分布和密度对于各种应用至关重要，特别是在病理学和肿瘤学领域。虽然基因表达提供了一个molecular lens，但相关的H&amp;E图像可以用来测量空间细胞分布和密度。与模块1一样，<span>METI接下来进行肿瘤细胞核分割，生成三维肿瘤细胞密度图，直观地描绘了癌细胞的空间分布和密度</span>。该功能用于传达感兴趣的细胞类型的空间分布、密度和模式。</p><h4>模块三、T细胞定位和表型</h4><p><img data-imgfileid="100008950" data-ratio="1.1123218776194468" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmmeSvgfeoibNHo26XCyFx3ENozBs5Eia4qWYVuTmK8Emq1MEynGTJnVSsfYicZRKzlfJWsfmpvUsbV8g/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="1193" src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmmeSvgfeoibNHo26XCyFx3ENozBs5Eia4qWYVuTmK8Emq1MEynGTJnVSsfYicZRKzlfJWsfmpvUsbV8g/640?wx_fmt=other&amp;from=appmsg"></p><h4>模块4、深入分析其他免疫细胞，这其中就包括了单细胞难以捕获到的中性粒细胞。</h4><p><img data-imgfileid="100008956" data-ratio="1.1841666666666666" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmmeSvgfeoibNHo26XCyFx3ENooBULdgZgkj7hIYbkVmDwtCZ3eiac4I1FzXQNCkbaZzsL9Cf3asLE4A/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="1200" src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmmeSvgfeoibNHo26XCyFx3ENooBULdgZgkj7hIYbkVmDwtCZ3eiac4I1FzXQNCkbaZzsL9Cf3asLE4A/640?wx_fmt=other&amp;from=appmsg"></p><h4>模块5、成纤维细胞的分析</h4><p><img data-imgfileid="100008955" data-ratio="0.9441666666666667" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmmeSvgfeoibNHo26XCyFx3ENnqNQeVKSra7w2KDYQGz3xjibNt2yXrKvUl4sjuYbUJP6YQvUNASSXoA/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="1200" src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmmeSvgfeoibNHo26XCyFx3ENnqNQeVKSra7w2KDYQGz3xjibNt2yXrKvUl4sjuYbUJP6YQvUNASSXoA/640?wx_fmt=other&amp;from=appmsg"></p><h4>示例代码在 https://github.com/Flashiness/METI。</h4><h4>我们简单看一下，注意如果运用到自己的课题还是需要认真思考的。</h4><p><span aria-label="icon: copy"><svg viewbox="64 64 896 896" focusable="false" data-icon="copy" width="1em" height="1em" fill="currentColor" aria-hidden="true"><path d="M832 64H296c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h496v688c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V96c0-17.7-14.3-32-32-32zM704 192H192c-17.7 0-32 14.3-32 32v530.7c0 8.5 3.4 16.6 9.4 22.6l173.3 173.3c2.2 2.2 4.7 4 7.4 5.5v1.9h4.2c3.5 1.3 7.2 2 11 2H704c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32zM350 856.2L263.9 770H350v86.2zM664 888H414V746c0-22.1-17.9-40-40-40H232V264h432v624z"></path></svg></span></p><pre><code>##pip install <span>METIforST</span><br>#<span>Note</span><span>:</span> you need to make sure that the pip <span>is</span> <span>for</span> python3，or you can install <span>METI</span> by<br>##python3 <span>-</span>m pip install <span>METIforST</span><span>==</span><span>0.2</span><br><br><span>import</span> torch<br><span>import</span> csv<span>,</span>re<span>,</span> time<br><span>import</span> pickle<br><span>import</span> random<br><span>import</span> warnings<br>warnings<span>.</span><span>filterwarnings</span><span>(</span><span>'ignore'</span><span>)</span><br><span>import</span> pandas <span>as</span> pd<br><span>import</span> numpy <span>as</span> np<br>from scipy <span>import</span> stats<br>from scipy<span>.</span>sparse <span>import</span> issparse<br><span>import</span> scanpy <span>as</span> sc<br><span>import</span> matplotlib<span>.</span>colors <span>as</span> clr<br><span>import</span> matplotlib<span>.</span>pyplot <span>as</span> plt<br><span>import</span> cv2<br><span>import</span> <span>TESLA</span> <span>as</span> tesla<br>from <span>IPython</span><span>.</span>display <span>import</span> <span>Image</span><br><span>import</span> scipy<span>.</span>sparse<br><span>import</span> scanpy <span>as</span> sc<br><span>import</span> pandas <span>as</span> pd<br><span>import</span> matplotlib<span>.</span>pyplot <span>as</span> plt<br><span>import</span> seaborn <span>as</span> sns<br>from scanpy <span>import</span> read_10x_h5<br><span>import</span> <span>PIL</span><br>from <span>PIL</span> <span>import</span> <span>Image</span> <span>as</span> <span>IMAGE</span><br><span>import</span> os<br><span>import</span> <span>METI</span> <span>as</span> meti<br><span>import</span> tifffile<br>os<span>.</span>environ<span>[</span><span>'KMP_DUPLICATE_LIB_OK'</span><span>]</span><span>=</span><span>'True'</span><br><span aria-hidden="true"><span></span><span></span><span></span><span></span><span></span></span></code></pre><h4>读取数据</h4><p><span aria-label="icon: copy"><svg viewbox="64 64 896 896" focusable="false" data-icon="copy" width="1em" height="1em" fill="currentColor" aria-hidden="true"><path d="M832 64H296c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h496v688c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V96c0-17.7-14.3-32-32-32zM704 192H192c-17.7 0-32 14.3-32 32v530.7c0 8.5 3.4 16.6 9.4 22.6l173.3 173.3c2.2 2.2 4.7 4 7.4 5.5v1.9h4.2c3.5 1.3 7.2 2 11 2H704c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32zM350 856.2L263.9 770H350v86.2zM664 888H414V746c0-22.1-17.9-40-40-40H232V264h432v624z"></path></svg></span></p><pre><code>adata<span>=</span>sc<span>.</span><span>read_visium</span><span>(</span><span>"/tutorial/data/Spaceranger/"</span><span>)</span><br>spatial<span>=</span>pd<span>.</span><span>read_csv</span><span>(</span><span>"/tutorial/data/Spaceranger/tissue_positions_list.csv"</span><span>,</span>sep<span>=</span><span>","</span><span>,</span>header<span>=</span>None<span>,</span>na_filter<span>=</span><span>False</span><span>,</span>index_col<span>=</span><span>0</span><span>)</span><br><br>adata<span>.</span><span>var_names_make_unique</span><span>(</span><span>)</span><br>adata<span>.</span><span>var</span><span>[</span><span>"mt"</span><span>]</span> <span>=</span> adata<span>.</span>var_names<span>.</span>str<span>.</span><span>startswith</span><span>(</span><span>"MT-"</span><span>)</span><br>sc<span>.</span>pp<span>.</span><span>calculate_qc_metrics</span><span>(</span>adata<span>,</span> qc_vars<span>=</span><span>[</span><span>"mt"</span><span>]</span><span>,</span> inplace<span>=</span><span>True</span><span>)</span><br><br>plt<span>.</span>rcParams<span>[</span><span>"figure.figsize"</span><span>]</span> <span>=</span> <span>(</span><span>8</span><span>,</span> <span>8</span><span>)</span><br>sc<span>.</span>pl<span>.</span><span>spatial</span><span>(</span>adata<span>,</span> img_key<span>=</span><span>"hires"</span><span>,</span> color<span>=</span><span>[</span><span>"total_counts"</span><span>,</span> <span>"n_genes_by_counts"</span><span>]</span><span>,</span> size <span>=</span> <span>1.5</span><span>,</span> save <span>=</span> <span>'UMI_count.png'</span><span>)</span><br><span aria-hidden="true"><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span></span></code></pre><p><img data-imgfileid="100008953" data-ratio="0.7727272727272727" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmmeSvgfeoibNHo26XCyFx3ENWDCb6huG3DEUBFiajhtmiaz7XcehqX9H9mcX1pelAPxw4Q1nPkgAbxfw/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="792" src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmmeSvgfeoibNHo26XCyFx3ENWDCb6huG3DEUBFiajhtmiaz7XcehqX9H9mcX1pelAPxw4Q1nPkgAbxfw/640?wx_fmt=other&amp;from=appmsg"></p><h4>转换数据</h4><p><img data-imgfileid="100008952" data-ratio="0.2205128205128205" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmmeSvgfeoibNHo26XCyFx3ENwmrBpqViaCfblUmhpp3ibUACTqKK0NwIOLF4dVJ6VcmQCgBdhA8WMf4Q/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="585" src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmmeSvgfeoibNHo26XCyFx3ENwmrBpqViaCfblUmhpp3ibUACTqKK0NwIOLF4dVJ6VcmQCgBdhA8WMf4Q/640?wx_fmt=other&amp;from=appmsg"></p><p><span aria-label="icon: copy"><svg viewbox="64 64 896 896" focusable="false" data-icon="copy" width="1em" height="1em" fill="currentColor" aria-hidden="true"><path d="M832 64H296c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h496v688c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V96c0-17.7-14.3-32-32-32zM704 192H192c-17.7 0-32 14.3-32 32v530.7c0 8.5 3.4 16.6 9.4 22.6l173.3 173.3c2.2 2.2 4.7 4 7.4 5.5v1.9h4.2c3.5 1.3 7.2 2 11 2H704c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32zM350 856.2L263.9 770H350v86.2zM664 888H414V746c0-22.1-17.9-40-40-40H232V264h432v624z"></path></svg></span></p><pre><code><span>#================== 3. Read in data ==================#</span><br><span>#Read original 10x_h5 data and save it to h5ad</span><br><span>from</span> scanpy <span>import</span> read_10x_h5<br>adata <span>=</span> read_10x_h5<span>(</span><span>"../tutorial/data/filtered_feature_bc_matrix.h5"</span><span>)</span><br>spatial<span>=</span>pd<span>.</span>read_csv<span>(</span><span>"../tutorial/data/tissue_positions_list.csv"</span><span>,</span>sep<span>=</span><span>","</span><span>,</span>header<span>=</span><span>None</span><span>,</span>na_filter<span>=</span><span>False</span><span>,</span>index_col<span>=</span><span>0</span><span>)</span> <br>adata<span>.</span>obs<span>[</span><span>"x1"</span><span>]</span><span>=</span>spatial<span>[</span><span>1</span><span>]</span><br>adata<span>.</span>obs<span>[</span><span>"x2"</span><span>]</span><span>=</span>spatial<span>[</span><span>2</span><span>]</span><br>adata<span>.</span>obs<span>[</span><span>"x3"</span><span>]</span><span>=</span>spatial<span>[</span><span>3</span><span>]</span><br>adata<span>.</span>obs<span>[</span><span>"x4"</span><span>]</span><span>=</span>spatial<span>[</span><span>4</span><span>]</span><br>adata<span>.</span>obs<span>[</span><span>"x5"</span><span>]</span><span>=</span>spatial<span>[</span><span>5</span><span>]</span> <br>adata<span>.</span>obs<span>[</span><span>"array_x"</span><span>]</span><span>=</span>adata<span>.</span>obs<span>[</span><span>"x2"</span><span>]</span><br>adata<span>.</span>obs<span>[</span><span>"array_y"</span><span>]</span><span>=</span>adata<span>.</span>obs<span>[</span><span>"x3"</span><span>]</span><br>adata<span>.</span>obs<span>[</span><span>"pixel_x"</span><span>]</span><span>=</span>adata<span>.</span>obs<span>[</span><span>"x4"</span><span>]</span><br>adata<span>.</span>obs<span>[</span><span>"pixel_y"</span><span>]</span><span>=</span>adata<span>.</span>obs<span>[</span><span>"x5"</span><span>]</span><br><span>#Select captured samples</span><br>adata<span>=</span>adata<span>[</span>adata<span>.</span>obs<span>[</span><span>"x1"</span><span>]</span><span>==</span><span>1</span><span>]</span><br>adata<span>.</span>var_names<span>=</span><span>[</span>i<span>.</span>upper<span>(</span><span>)</span> <span>for</span> i <span>in</span> <span>list</span><span>(</span>adata<span>.</span>var_names<span>)</span><span>]</span><br>adata<span>.</span>var<span>[</span><span>"genename"</span><span>]</span><span>=</span>adata<span>.</span>var<span>.</span>index<span>.</span>astype<span>(</span><span>"str"</span><span>)</span><br>adata<span>.</span>write_h5ad<span>(</span><span>"../tutorial/data/1957495_data.h5ad"</span><span>)</span><br><br><span>#Read in gene expression and spatial location</span><br>counts<span>=</span>sc<span>.</span>read<span>(</span><span>"../tutorial/data/1957495_data.h5ad"</span><span>)</span><br><span>#Read in hitology image</span><br>PIL<span>.</span>Image<span>.</span>MAX_IMAGE_PIXELS <span>=</span> <span>None</span><br>img <span>=</span> IMAGE<span>.</span><span>open</span><span>(</span><span>r"../tutorial/data/histology.tif"</span><span>)</span> <br>img <span>=</span> np<span>.</span>array<span>(</span>img<span>)</span><br><br><span>#if your image has 4 dimensions, only keep first 3 dims</span><br>img<span>=</span>img<span>[</span><span>.</span><span>.</span><span>.</span><span>,</span><span>:</span><span>3</span><span>]</span><br><br>resize_factor<span>=</span><span>1000</span><span>/</span>np<span>.</span><span>min</span><span>(</span>img<span>.</span>shape<span>[</span><span>0</span><span>:</span><span>2</span><span>]</span><span>)</span><br>resize_width<span>=</span><span>int</span><span>(</span>img<span>.</span>shape<span>[</span><span>1</span><span>]</span><span>*</span>resize_factor<span>)</span><br>resize_height<span>=</span><span>int</span><span>(</span>img<span>.</span>shape<span>[</span><span>0</span><span>]</span><span>*</span>resize_factor<span>)</span><br>counts<span>.</span>var<span>.</span>index<span>=</span><span>[</span>i<span>.</span>upper<span>(</span><span>)</span> <span>for</span> i <span>in</span> counts<span>.</span>var<span>.</span>index<span>]</span><br>counts<span>.</span>var_names_make_unique<span>(</span><span>)</span><br>counts<span>.</span>raw<span>=</span>counts<br>sc<span>.</span>pp<span>.</span>log1p<span>(</span>counts<span>)</span> <span># impute on log scale</span><br><span>if</span> issparse<span>(</span>counts<span>.</span>X<span>)</span><span>:</span>counts<span>.</span>X<span>=</span>counts<span>.</span>X<span>.</span>A<br><br><span>###Contour detection</span><br><br><span># Detect contour using cv2</span><br>cnt<span>=</span>tesla<span>.</span>cv2_detect_contour<span>(</span>img<span>,</span> apertureSize<span>=</span><span>5</span><span>,</span>L2gradient <span>=</span> <span>True</span><span>)</span><br><br>binary<span>=</span>np<span>.</span>zeros<span>(</span><span>(</span>img<span>.</span>shape<span>[</span><span>0</span><span>:</span><span>2</span><span>]</span><span>)</span><span>,</span> dtype<span>=</span>np<span>.</span>uint8<span>)</span><br>cv2<span>.</span>drawContours<span>(</span>binary<span>,</span> <span>[</span>cnt<span>]</span><span>,</span> <span>-</span><span>1</span><span>,</span> <span>(</span><span>1</span><span>)</span><span>,</span> thickness<span>=</span><span>-</span><span>1</span><span>)</span><br><span>#Enlarged filter</span><br>cnt_enlarged <span>=</span> tesla<span>.</span>scale_contour<span>(</span>cnt<span>,</span> <span>1.05</span><span>)</span><br>binary_enlarged <span>=</span> np<span>.</span>zeros<span>(</span>img<span>.</span>shape<span>[</span><span>0</span><span>:</span><span>2</span><span>]</span><span>)</span><br>cv2<span>.</span>drawContours<span>(</span>binary_enlarged<span>,</span> <span>[</span>cnt_enlarged<span>]</span><span>,</span> <span>-</span><span>1</span><span>,</span> <span>(</span><span>1</span><span>)</span><span>,</span> thickness<span>=</span><span>-</span><span>1</span><span>)</span><br>img_new <span>=</span> img<span>.</span>copy<span>(</span><span>)</span><br>cv2<span>.</span>drawContours<span>(</span>img_new<span>,</span> <span>[</span>cnt<span>]</span><span>,</span> <span>-</span><span>1</span><span>,</span> <span>(</span><span>255</span><span>)</span><span>,</span> thickness<span>=</span><span>20</span><span>)</span><br>img_new<span>=</span>cv2<span>.</span>resize<span>(</span>img_new<span>,</span> <span>(</span><span>(</span>resize_width<span>,</span> resize_height<span>)</span><span>)</span><span>)</span><br>cv2<span>.</span>imwrite<span>(</span><span>'../tutorial/data/cnt_1957495.jpg'</span><span>,</span> img_new<span>)</span><br>Image<span>(</span>filename<span>=</span><span>'../tutorial/data/cnt_1957495.jpg'</span><span>)</span><br><br><span>####Gene expression enhancement</span><br><span>#Set size of superpixel</span><br>res<span>=</span><span>40</span><br><span># Note, if the numer of superpixels is too large and take too long, you can increase the res to 100</span><br>enhanced_exp_adata<span>=</span>tesla<span>.</span>imputation<span>(</span>img<span>=</span>img<span>,</span> raw<span>=</span>counts<span>,</span> cnt<span>=</span>cnt<span>,</span> genes<span>=</span>counts<span>.</span>var<span>.</span>index<span>.</span>tolist<span>(</span><span>)</span><span>,</span> shape<span>=</span><span>"None"</span><span>,</span> res<span>=</span>res<span>,</span> s<span>=</span><span>1</span><span>,</span> k<span>=</span><span>2</span><span>,</span> num_nbs<span>=</span><span>10</span><span>)</span><br>enhanced_exp_adata<span>.</span>write_h5ad<span>(</span><span>"../tutorial/data/enhanced_exp.h5ad"</span><span>)</span><br><br><span>####Goblet marker gene expression</span><br><span>#================ determine if markers are in ===============#</span><br>enhanced_exp_adata<span>=</span>sc<span>.</span>read<span>(</span><span>"..tutorial/data/enhanced_exp.h5ad"</span><span>)</span><br>markers <span>=</span> <span>[</span><span>"MS4A10"</span><span>,</span> <span>"MGAM"</span><span>,</span> <span>"CYP4F2"</span><span>,</span> <span>"XPNPEP2"</span><span>,</span> <span>"SLC5A9"</span><span>,</span> <span>"SLC13A2"</span><span>,</span> <span>"SLC28A1"</span><span>,</span> <span>"MEP1A"</span><span>,</span> <span>"ABCG2"</span><span>,</span> <span>"ACE2"</span><span>]</span><br><span>for</span> i <span>in</span> <span>range</span><span>(</span><span>len</span><span>(</span>markers<span>)</span><span>)</span><span>:</span><br>    <span>if</span> markers<span>[</span>i<span>]</span> <span>in</span> enhanced_exp_adata<span>.</span>var<span>.</span>index<span>:</span> <span>print</span><span>(</span><span>"yes"</span><span>)</span><br>    <span>else</span><span>:</span> <span>print</span><span>(</span>markers<span>[</span>i<span>]</span><span>)</span><br>save_dir<span>=</span><span>"..tutorial/data/Goblet/"</span><br><span>if</span> <span>not</span> os<span>.</span>path<span>.</span>exists<span>(</span>save_dir<span>)</span><span>:</span>os<span>.</span>mkdir<span>(</span>save_dir<span>)</span><br><span>#================ Plot gene expression image ===============#</span><br>markers <span>=</span> <span>[</span><span>"MS4A10"</span><span>,</span> <span>"MGAM"</span><span>,</span> <span>"CYP4F2"</span><span>,</span> <span>"XPNPEP2"</span><span>,</span> <span>"SLC5A9"</span><span>,</span> <span>"SLC13A2"</span><span>,</span> <span>"SLC28A1"</span><span>,</span> <span>"MEP1A"</span><span>,</span> <span>"ABCG2"</span><span>,</span> <span>"ACE2"</span><span>]</span><br><span>for</span> i <span>in</span> <span>range</span><span>(</span><span>len</span><span>(</span>markers<span>)</span><span>)</span><span>:</span><br>    cnt_color <span>=</span> clr<span>.</span>LinearSegmentedColormap<span>.</span>from_list<span>(</span><span>'magma'</span><span>,</span> <span>[</span><span>"#000003"</span><span>,</span>  <span>"#3b0f6f"</span><span>,</span>  <span>"#8c2980"</span><span>,</span>   <span>"#f66e5b"</span><span>,</span> <span>"#fd9f6c"</span><span>,</span> <span>"#fbfcbf"</span><span>]</span><span>,</span> N<span>=</span><span>256</span><span>)</span><br>    g<span>=</span>markers<span>[</span>i<span>]</span><br>    enhanced_exp_adata<span>.</span>obs<span>[</span>g<span>]</span><span>=</span>enhanced_exp_adata<span>.</span>X<span>[</span><span>:</span><span>,</span>enhanced_exp_adata<span>.</span>var<span>.</span>index<span>==</span>g<span>]</span><br>    fig<span>=</span>sc<span>.</span>pl<span>.</span>scatter<span>(</span>enhanced_exp_adata<span>,</span>alpha<span>=</span><span>1</span><span>,</span>x<span>=</span><span>"y"</span><span>,</span>y<span>=</span><span>"x"</span><span>,</span>color<span>=</span>g<span>,</span>color_map<span>=</span>cnt_color<span>,</span>show<span>=</span><span>False</span><span>,</span>size<span>=</span><span>10</span><span>)</span><br>    fig<span>.</span>set_aspect<span>(</span><span>'equal'</span><span>,</span> <span>'box'</span><span>)</span><br>    fig<span>.</span>invert_yaxis<span>(</span><span>)</span><br>    plt<span>.</span>gcf<span>(</span><span>)</span><span>.</span>set_dpi<span>(</span><span>600</span><span>)</span><br>    fig<span>.</span>figure<span>.</span>show<span>(</span><span>)</span><br><br>    plt<span>.</span>savefig<span>(</span>save_dir <span>+</span> <span>str</span><span>(</span>markers<span>[</span>i<span>]</span><span>)</span> <span>+</span> <span>".png"</span><span>,</span> dpi<span>=</span><span>600</span><span>)</span><br>    plt<span>.</span>close<span>(</span><span>)</span><br><span aria-hidden="true"><span></span><span></span><span></span><span></span><span></span></span></code></pre><p><img data-imgfileid="100008954" data-ratio="0.5925" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmmeSvgfeoibNHo26XCyFx3ENia0A2FHqL9FH7q54AXG8iawn55vP3tCRTNMdVfbJGyGib5NVBWX4BicdIg/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="1200" src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmmeSvgfeoibNHo26XCyFx3ENia0A2FHqL9FH7q54AXG8iawn55vP3tCRTNMdVfbJGyGib5NVBWX4BicdIg/640?wx_fmt=other&amp;from=appmsg"></p><p><span aria-label="icon: copy"><svg viewbox="64 64 896 896" focusable="false" data-icon="copy" width="1em" height="1em" fill="currentColor" aria-hidden="true"><path d="M832 64H296c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h496v688c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V96c0-17.7-14.3-32-32-32zM704 192H192c-17.7 0-32 14.3-32 32v530.7c0 8.5 3.4 16.6 9.4 22.6l173.3 173.3c2.2 2.2 4.7 4 7.4 5.5v1.9h4.2c3.5 1.3 7.2 2 11 2H704c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32zM350 856.2L263.9 770H350v86.2zM664 888H414V746c0-22.1-17.9-40-40-40H232V264h432v624z"></path></svg></span></p><pre><code>#<span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span> Plot meta gene expression image <span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>=</span>#<br>enhanced_exp_adata<span>=</span>sc<span>.</span><span>read</span><span>(</span><span>"/Users/jjiang6/Desktop/UTH/MDA GRA/Spatial transcriptome/Cell Segmentation/With Jian Hu/S1_54078/TESLA/enhanced_exp.h5ad"</span><span>)</span><br>genes <span>=</span>   <span>[</span><span>"MS4A10"</span><span>,</span> <span>"MGAM"</span><span>,</span> <span>"CYP4F2"</span><span>,</span> <span>"XPNPEP2"</span><span>,</span> <span>"SLC5A9"</span><span>,</span> <span>"SLC13A2"</span><span>,</span> <span>"SLC28A1"</span><span>,</span> <span>"MEP1A"</span><span>,</span> <span>"ABCG2"</span><span>,</span> <span>"ACE2"</span><span>]</span><br>    <br>sudo_adata <span>=</span> meti<span>.</span><span>meta_gene_plot</span><span>(</span>img<span>=</span>img<span>,</span> <br>                                binary<span>=</span>binary<span>,</span><br>                                sudo_adata<span>=</span>enhanced_exp_adata<span>,</span> <br>                                genes<span>=</span>genes<span>,</span> <br>                                resize_factor<span>=</span>resize_factor<span>,</span><br>                                target_size<span>=</span><span>"small"</span><span>)</span><br><br>cnt_color <span>=</span> clr<span>.</span>LinearSegmentedColormap<span>.</span><span>from_list</span><span>(</span><span>'magma'</span><span>,</span> <span>[</span><span>"#000003"</span><span>,</span>  <span>"#3b0f6f"</span><span>,</span>  <span>"#8c2980"</span><span>,</span>   <span>"#f66e5b"</span><span>,</span> <span>"#fd9f6c"</span><span>,</span> <span>"#fbfcbf"</span><span>]</span><span>,</span> N<span>=</span><span>256</span><span>)</span><br>fig<span>=</span>sc<span>.</span>pl<span>.</span><span>scatter</span><span>(</span>sudo_adata<span>,</span>alpha<span>=</span><span>1</span><span>,</span>x<span>=</span><span>"y"</span><span>,</span>y<span>=</span><span>"x"</span><span>,</span>color<span>=</span><span>'meta'</span><span>,</span>color_map<span>=</span>cnt_color<span>,</span>show<span>=</span>False<span>,</span>size<span>=</span><span>5</span><span>)</span><br>fig<span>.</span><span>set_aspect</span><span>(</span><span>'equal'</span><span>,</span> <span>'box'</span><span>)</span><br>fig<span>.</span><span>invert_yaxis</span><span>(</span><span>)</span><br>plt<span>.</span><span>gcf</span><span>(</span><span>)</span><span>.</span><span>set_dpi</span><span>(</span><span>600</span><span>)</span><br>fig<span>.</span>figure<span>.</span><span>show</span><span>(</span><span>)</span><br><br>plt<span>.</span><span>savefig</span><span>(</span>save_dir <span>+</span> <span>"Goblet_meta.png"</span><span>,</span> dpi<span>=</span><span>600</span><span>)</span><br>plt<span>.</span><span>close</span><span>(</span><span>)</span><br><span aria-hidden="true"><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span></span></code></pre><p><img data-imgfileid="100008958" data-ratio="0.5925" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmmeSvgfeoibNHo26XCyFx3ENI2KH7Mf8FCX5D4kpxjH6cLf6iaPH1iaicTbcFJbU7JeIW1Vu4u52Bb4uw/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="1200" src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmmeSvgfeoibNHo26XCyFx3ENI2KH7Mf8FCX5D4kpxjH6cLf6iaPH1iaicTbcFJbU7JeIW1Vu4u52Bb4uw/640?wx_fmt=other&amp;from=appmsg"></p><h4>Region annotation</h4><p><span aria-label="icon: copy"><svg viewbox="64 64 896 896" focusable="false" data-icon="copy" width="1em" height="1em" fill="currentColor" aria-hidden="true"><path d="M832 64H296c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h496v688c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V96c0-17.7-14.3-32-32-32zM704 192H192c-17.7 0-32 14.3-32 32v530.7c0 8.5 3.4 16.6 9.4 22.6l173.3 173.3c2.2 2.2 4.7 4 7.4 5.5v1.9h4.2c3.5 1.3 7.2 2 11 2H704c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32zM350 856.2L263.9 770H350v86.2zM664 888H414V746c0-22.1-17.9-40-40-40H232V264h432v624z"></path></svg></span></p><pre><code>genes<span>=</span><span>[</span><span>"MS4A10"</span><span>,</span> <span>"MGAM"</span><span>,</span> <span>"CYP4F2"</span><span>,</span> <span>"XPNPEP2"</span><span>,</span> <span>"SLC5A9"</span><span>,</span> <span>"SLC13A2"</span><span>,</span> <span>"SLC28A1"</span><span>,</span> <span>"MEP1A"</span><span>,</span> <span>"ABCG2"</span><span>,</span> <span>"ACE2"</span><span>]</span><br>genes<span>=</span><span>list</span><span>(</span><span>set</span><span>(</span><span>[</span>i <span>for</span> i <span>in</span> genes <span>if</span> i <span>in</span> enhanced_exp_adata<span>.</span>var<span>.</span>index <span>]</span><span>)</span><span>)</span><br>#target_size can be <span>set</span> <span>to</span> <span>"small"</span> <span>or</span> <span>"large"</span><span>.</span><br>pred_refined<span>,</span> target_clusters<span>,</span> c_m<span>=</span>meti<span>.</span><span>annotation</span><span>(</span>img<span>=</span>img<span>,</span> <br>                                                    binary<span>=</span>binary<span>,</span><br>                                                    sudo_adata<span>=</span>enhanced_exp_adata<span>,</span> <br>                                                    genes<span>=</span>genes<span>,</span> <br>                                                    resize_factor<span>=</span>resize_factor<span>,</span><br>                                                    num_required<span>=</span><span>1</span><span>,</span> <br>                                                    target_size<span>=</span><span>"small"</span><span>)</span><br>#Plot<br>ret_img<span>=</span>tesla<span>.</span><span>visualize_annotation</span><span>(</span>img<span>=</span>img<span>,</span> <br>                              binary<span>=</span>binary<span>,</span> <br>                              resize_factor<span>=</span>resize_factor<span>,</span><br>                              pred_refined<span>=</span>pred_refined<span>,</span> <br>                              target_clusters<span>=</span>target_clusters<span>,</span> <br>                              c_m<span>=</span>c_m<span>)</span><br><br>cv2<span>.</span><span>imwrite</span><span>(</span>save_dir <span>+</span> <span>'IME.jpg'</span><span>,</span> ret_img<span>)</span><br><span>Image</span><span>(</span>filename<span>=</span>save_dir <span>+</span> <span>'IME.jpg'</span><span>)</span><br><span aria-hidden="true"><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span></span></code></pre><p><img data-imgfileid="100008960" data-ratio="0.9900990099009901" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmmeSvgfeoibNHo26XCyFx3EN4iaV7GLfGFO6BFVpquKX4GDicodMpdlqwa3SlWkhmXql7RVcsDdhQgrw/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="1010" src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmmeSvgfeoibNHo26XCyFx3EN4iaV7GLfGFO6BFVpquKX4GDicodMpdlqwa3SlWkhmXql7RVcsDdhQgrw/640?wx_fmt=other&amp;from=appmsg"></p><p><span aria-label="icon: copy"><svg viewbox="64 64 896 896" focusable="false" data-icon="copy" width="1em" height="1em" fill="currentColor" aria-hidden="true"><path d="M832 64H296c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h496v688c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V96c0-17.7-14.3-32-32-32zM704 192H192c-17.7 0-32 14.3-32 32v530.7c0 8.5 3.4 16.6 9.4 22.6l173.3 173.3c2.2 2.2 4.7 4 7.4 5.5v1.9h4.2c3.5 1.3 7.2 2 11 2H704c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32zM350 856.2L263.9 770H350v86.2zM664 888H414V746c0-22.1-17.9-40-40-40H232V264h432v624z"></path></svg></span></p><pre><code>#<span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>=</span>Convert to spot level<span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span>#<br>adata<span>.</span>obs<span>[</span><span>"color"</span><span>]</span><span>=</span><span>extract_color</span><span>(</span>x_pixel<span>=</span><span>(</span>np<span>.</span><span>array</span><span>(</span>adata<span>.</span>obs<span>[</span><span>"pixel_x"</span><span>]</span><span>)</span><span>*</span>resize_factor<span>)</span><span>.</span><span>astype</span><span>(</span>np<span>.</span><span>int64</span><span>)</span><span>,</span> <br>                                 y_pixel<span>=</span><span>(</span>np<span>.</span><span>array</span><span>(</span>adata<span>.</span>obs<span>[</span><span>"pixel_y"</span><span>]</span><span>)</span><span>*</span>resize_factor<span>)</span><span>.</span><span>astype</span><span>(</span>np<span>.</span><span>int64</span><span>)</span><span>,</span> image<span>=</span>ret_img<span>,</span> beta<span>=</span><span>25</span><span>)</span><br><br><span>type</span> <span>=</span> <span>[</span><span>]</span><br><span>for</span> each in adata<span>.</span>obs<span>[</span><span>"color"</span><span>]</span><span>:</span><br>    <span>if</span> each <span>&lt;</span> adata<span>.</span>obs<span>[</span><span>"color"</span><span>]</span><span>.</span><span>quantile</span><span>(</span><span>0.2</span><span>)</span><span>:</span><br>        r <span>=</span> <span>"yes"</span><br>        <span>type</span><span>.</span><span>append</span><span>(</span>r<span>)</span><br>    <span>else</span><span>:</span><br>        r <span>=</span> <span>"no"</span><br>        <span>type</span><span>.</span><span>append</span><span>(</span>r<span>)</span><br><br>adata<span>.</span>obs<span>[</span><span>'Goblet_GE'</span><span>]</span> <span>=</span> <span>type</span><br><br>fig<span>,</span> ax <span>=</span> plt<span>.</span><span>subplots</span><span>(</span>figsize<span>=</span><span>(</span><span>10</span><span>,</span> <span>10</span><span>)</span><span>)</span>  # Adjust the size as needed<br>ax<span>.</span><span>imshow</span><span>(</span>img<span>)</span><br>ax<span>.</span><span>set_axis_off</span><span>(</span><span>)</span><br>sc<span>.</span>pl<span>.</span><span>scatter</span><span>(</span>adata<span>,</span> x<span>=</span><span>'pixel_y'</span><span>,</span> y<span>=</span><span>'pixel_x'</span><span>,</span> color<span>=</span><span>'Goblet_GE'</span><span>,</span> ax<span>=</span>ax<span>,</span> size <span>=</span> <span>150</span><span>,</span> title<span>=</span><span>'Goblet GE Spot Annotations'</span><span>)</span><br># Save the figure<br>plt<span>.</span><span>savefig</span><span>(</span><span>'./sample_results/Goblet_spot_GE.png'</span><span>,</span> dpi<span>=</span><span>300</span><span>,</span> bbox_inches<span>=</span><span>'tight'</span><span>)</span><br><span aria-hidden="true"><span></span><span></span><span></span><span></span><span></span></span></code></pre><p><img data-imgfileid="100008961" data-ratio="1.1985645933014355" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmmeSvgfeoibNHo26XCyFx3EN36DjicBK1bmgib62icbQh1Qmia1P5oJPicvwlic4Y2r8HPV41zo4vroBbDvw/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="836" src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmmeSvgfeoibNHo26XCyFx3EN36DjicBK1bmgib62icbQh1Qmia1P5oJPicvwlic4Y2r8HPV41zo4vroBbDvw/640?wx_fmt=other&amp;from=appmsg"></p><h4>图像分割Segmentation</h4><p><span aria-label="icon: copy"><svg viewbox="64 64 896 896" focusable="false" data-icon="copy" width="1em" height="1em" fill="currentColor" aria-hidden="true"><path d="M832 64H296c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h496v688c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V96c0-17.7-14.3-32-32-32zM704 192H192c-17.7 0-32 14.3-32 32v530.7c0 8.5 3.4 16.6 9.4 22.6l173.3 173.3c2.2 2.2 4.7 4 7.4 5.5v1.9h4.2c3.5 1.3 7.2 2 11 2H704c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32zM350 856.2L263.9 770H350v86.2zM664 888H414V746c0-22.1-17.9-40-40-40H232V264h432v624z"></path></svg></span></p><pre><code>plot_dir<span>=</span><span>"/rsrch4/home/genomic_med/jjiang6/Project1/S1_54078/Segmentation/NC_review_Goblet_seg/"</span><br>save_dir<span>=</span>plot_dir<span>+</span><span>"/seg_results"</span><br>adata<span>=</span> sc<span>.</span><span>read</span><span>(</span><span>"/rsrch4/home/genomic_med/jjiang6/Project1/S1_54078/TESLA/54078_data.h5ad"</span><span>)</span><br><br>img_path <span>=</span> <span>'/rsrch4/home/genomic_med/jjiang6/Project1/S1_54078/1415785-6 Bx2.tif'</span><br>img <span>=</span> tiff<span>.</span><span>imread</span><span>(</span>img_path<span>)</span><br>d0<span>,</span> d1<span>=</span> img<span>.</span>shape<span>[</span><span>0</span><span>]</span><span>,</span> img<span>.</span>shape<span>[</span><span>1</span><span>]</span><br><br><span>#=====================================Split into patched=====================================================</span><br>patch_size<span>=</span><span>400</span><br>patches<span>=</span><span>patch_split_for_ST</span><span>(</span>img<span>=</span>img<span>,</span> patch_size<span>=</span>patch_size<span>,</span> spot_info<span>=</span>adata<span>.</span>obs<span>,</span> x_name<span>=</span><span>"pixel_x"</span><span>,</span> y_name<span>=</span><span>"pixel_y"</span><span>)</span><br>patch_info<span>=</span>adata<span>.</span>obs<br><br><span># save results</span><br>pickle<span>.</span><span>dump</span><span>(</span>patches<span>,</span> <span>open</span><span>(</span>plot_dir <span>+</span> <span>'patches.pkl'</span><span>,</span> <span>'wb'</span><span>)</span><span>)</span><br><span>#=================================Image Segmentation===================================</span><br>meti<span>.</span><span>Segment_Patches</span><span>(</span>patches<span>,</span> save_dir<span>=</span>save_dir<span>,</span>n_clusters<span>=</span><span>10</span><span>)</span><br><br><span>#=================================Get masks=================================#</span><br>pred_file_locs<span>=</span><span>[</span>save_dir<span>+</span><span>"/patch"</span><span>+</span><span>str</span><span>(</span>j<span>)</span><span>+</span><span>"_pred.npy"</span> <span>for</span> j in <span>range</span><span>(</span>patch_info<span>.</span>shape<span>[</span><span>0</span><span>]</span><span>)</span><span>]</span><br>dic_list<span>=</span>meti<span>.</span><span>get_color_dic</span><span>(</span>patches<span>,</span> seg_dir<span>=</span>save_dir<span>)</span><br>masks_index<span>=</span>meti<span>.</span><span>Match_Masks</span><span>(</span>dic_list<span>,</span> num_mask_each<span>=</span><span>5</span><span>,</span> mapping_threshold1<span>=</span><span>30</span><span>,</span> mapping_threshold2<span>=</span><span>60</span><span>)</span><br><br>masks<span>=</span>meti<span>.</span><span>Extract_Masks</span><span>(</span>masks_index<span>,</span> pred_file_locs<span>,</span> patch_size<span>)</span><br><br>combined_masks<span>=</span>meti<span>.</span><span>Combine_Masks</span><span>(</span>masks<span>,</span> patch_info<span>,</span> d0<span>,</span> d1<span>)</span><br><br><span>#=================================Plot masks=================================#</span><br>plot_dir <span>=</span> <span>'../tutorial/data/seg_results/mask'</span><br><br><span>for</span> i in <span>range</span><span>(</span>masks<span>.</span>shape<span>[</span><span>0</span><span>]</span><span>)</span><span>:</span> <span>#Each mask</span><br>    <span>print</span><span>(</span><span>"Plotting mask "</span><span>,</span> <span>str</span><span>(</span>i<span>)</span><span>)</span><br>    ret<span>=</span><span>(</span>combined_masks<span>[</span>i<span>]</span><span>*</span><span>255</span><span>)</span><br>    cv2<span>.</span><span>imwrite</span><span>(</span>plot_dir<span>+</span><span>'/mask'</span><span>+</span><span>str</span><span>(</span>i<span>)</span><span>+</span><span>'.png'</span><span>,</span> ret<span>.</span><span>astype</span><span>(</span>np<span>.</span>uint8<span>)</span><span>)</span><br><br><span>#=================================Choose one mask to detect cells/nucleis=================================#</span><br>channel<span>=</span><span>1</span><br>converted_image <span>=</span> combined_masks<span>[</span><span>1</span><span>]</span><span>.</span><span>astype</span><span>(</span>np<span>.</span>uint8<span>)</span><br>ret<span>,</span> labels <span>=</span> cv2<span>.</span><span>connectedComponents</span><span>(</span>converted_image<span>)</span><br>features<span>=</span>meti<span>.</span><span>Extract_CC_Features_each_CC</span><span>(</span>labels<span>)</span><br><br>num_labels <span>=</span> labels<span>.</span><span>max</span><span>(</span><span>)</span><br>height<span>,</span> width <span>=</span> labels<span>.</span>shape<br><br>colors <span>=</span> np<span>.</span>random<span>.</span><span>randint</span><span>(</span><span>0</span><span>,</span> <span>255</span><span>,</span> size<span>=</span><span>(</span>num_labels <span>+</span> <span>1</span><span>,</span> <span>3</span><span>)</span><span>,</span> dtype<span>=</span>np<span>.</span>uint8<span>)</span><br>colors<span>[</span><span>0</span><span>]</span> <span>=</span> <span>[</span><span>0</span><span>,</span> <span>0</span><span>,</span> <span>0</span><span>]</span><br>colored_mask <span>=</span> np<span>.</span><span>zeros</span><span>(</span><span>(</span>height<span>,</span> width<span>,</span> <span>3</span><span>)</span><span>,</span> dtype<span>=</span>np<span>.</span>uint8<span>)</span><br>colored_mask <span>=</span> colors<span>[</span>labels<span>]</span><br><br><span># save the colored nucleis channel</span><br>cv2<span>.</span><span>imwrite</span><span>(</span><span>'/rsrch4/home/genomic_med/jjiang6/Project1/S1_54078/Segmentation/NC_review_Goblet_seg/seg_results/goblet.png'</span><span>,</span> colored_mask<span>)</span><br><span># save nucleis label</span><br>np<span>.</span><span>save</span><span>(</span><span>'/rsrch4/home/genomic_med/jjiang6/Project1/S1_54078/Segmentation/NC_review_Goblet_seg/seg_results/labels.npy'</span><span>,</span> labels<span>)</span><br><span># save nucleis features, including, area, length, width</span><br>features<span>.</span><span>to_csv</span><span>(</span><span>'/rsrch4/home/genomic_med/jjiang6/Project1/S1_54078/Segmentation/NC_review_Goblet_seg/seg_results/features.csv'</span><span>,</span> index<span>=</span><span>False</span><span>)</span><br><br><span>#=================================filter out goblet cells=================================#</span><br>plot_dir<span>=</span><span>"../tutorial/data/seg_results/mask"</span><br><span>if</span> not os<span>.</span>path<span>.</span><span>exists</span><span>(</span>plot_dir<span>)</span><span>:</span>os<span>.</span><span>mkdir</span><span>(</span>plot_dir<span>)</span><br><br>labels<span>=</span>np<span>.</span><span>load</span><span>(</span>plot_dir<span>+</span><span>"labels.npy"</span><span>)</span><br><br><span>#Filter - different cell type needs to apply different parameter values</span><br>features<span>=</span>pd<span>.</span><span>read_csv</span><span>(</span>plot_dir<span>+</span><span>"features.csv"</span><span>,</span> header<span>=</span><span>0</span><span>,</span> index_col<span>=</span><span>0</span><span>)</span><br>features<span>[</span><span>'mm_ratio'</span><span>]</span> <span>=</span> features<span>[</span><span>'major_axis_length'</span><span>]</span><span>/</span>features<span>[</span><span>'minor_axis_length'</span><span>]</span><br>features_sub<span>=</span>features<span>[</span><span>(</span>features<span>[</span><span>"area"</span><span>]</span><span>&gt;</span><span>120</span><span>)</span> <span>&amp;</span><br>                      <span>(</span>features<span>[</span><span>"area"</span><span>]</span><span>&lt;</span><span>1500</span><span>)</span> <span>&amp;</span><br>                      <span>(</span>features<span>[</span><span>"solidity"</span><span>]</span><span>&gt;</span><span>0.85</span><span>)</span> <span>&amp;</span><br>                      <span>(</span>features<span>[</span><span>"mm_ratio"</span><span>]</span><span>&lt;</span><span>2</span><span>)</span><span>]</span><br>index<span>=</span>features_sub<span>.</span>index<span>.</span><span>tolist</span><span>(</span><span>)</span><br>labels_filtered<span>=</span>labels<span>*</span>np<span>.</span><span>isin</span><span>(</span>labels<span>,</span> index<span>)</span><br><br>np<span>.</span><span>save</span><span>(</span>plot_dir<span>+</span><span>"nuclei_filtered.npy"</span><span>,</span> labels_filtered<span>)</span><br><br>num_labels <span>=</span> labels_filtered<span>.</span><span>max</span><span>(</span><span>)</span><br>height<span>,</span> width <span>=</span> labels_filtered<span>.</span>shape<br><br>colors <span>=</span> np<span>.</span>random<span>.</span><span>randint</span><span>(</span><span>0</span><span>,</span> <span>255</span><span>,</span> size<span>=</span><span>(</span>num_labels <span>+</span> <span>1</span><span>,</span> <span>3</span><span>)</span><span>,</span> dtype<span>=</span>np<span>.</span>uint8<span>)</span><br>colors<span>[</span><span>0</span><span>]</span> <span>=</span> <span>[</span><span>0</span><span>,</span> <span>0</span><span>,</span> <span>0</span><span>]</span><br>colored_mask <span>=</span> np<span>.</span><span>zeros</span><span>(</span><span>(</span>height<span>,</span> width<span>,</span> <span>3</span><span>)</span><span>,</span> dtype<span>=</span>np<span>.</span>uint8<span>)</span><br>colored_mask <span>=</span> colors<span>[</span>labels_filtered<span>]</span><br><br>cv2<span>.</span><span>imwrite</span><span>(</span>plot_dir<span>+</span><span>'/goblet_filtered.png'</span><span>,</span> colored_mask<span>)</span><br><span aria-hidden="true"><span></span><span></span><span></span><span></span><span></span></span></code></pre><p><img data-imgfileid="100008962" data-ratio="0.99" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmmeSvgfeoibNHo26XCyFx3ENPibBtO2wEdZetAmyHWfLN9XaKrWtEo8C5lc9dFONhmkHNyPtsoUNbvA/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="1200" src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmmeSvgfeoibNHo26XCyFx3ENPibBtO2wEdZetAmyHWfLN9XaKrWtEo8C5lc9dFONhmkHNyPtsoUNbvA/640?wx_fmt=other&amp;from=appmsg"></p><p><span>nuclei segmentation</span></p><p><br></p><p><img data-imgfileid="100008959" data-ratio="0.99" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmmeSvgfeoibNHo26XCyFx3ENntHmzkkYC1Tj67Pu8MVNOAU0sZbicBe7BjzMJNAh8m7HPKWynUiaobicA/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="1200" src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmmeSvgfeoibNHo26XCyFx3ENntHmzkkYC1Tj67Pu8MVNOAU0sZbicBe7BjzMJNAh8m7HPKWynUiaobicA/640?wx_fmt=other&amp;from=appmsg"></p><p><span>goblet segmentation</span></p><p><span aria-label="icon: copy"><svg viewbox="64 64 896 896" focusable="false" data-icon="copy" width="1em" height="1em" fill="currentColor" aria-hidden="true"><path d="M832 64H296c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h496v688c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V96c0-17.7-14.3-32-32-32zM704 192H192c-17.7 0-32 14.3-32 32v530.7c0 8.5 3.4 16.6 9.4 22.6l173.3 173.3c2.2 2.2 4.7 4 7.4 5.5v1.9h4.2c3.5 1.3 7.2 2 11 2H704c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32zM350 856.2L263.9 770H350v86.2zM664 888H414V746c0-22.1-17.9-40-40-40H232V264h432v624z"></path></svg></span></p><pre><code>#<span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>=</span>Convert to spot level<span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span><span>==</span>#<br>plot_dir<span>=</span><span>"./tutorial/sample_results/"</span><br>img_seg <span>=</span> np<span>.</span><span>load</span><span>(</span>plot_dir<span>+</span><span>'nuclei_filtered_white.npy'</span><span>)</span><br><br>adata<span>.</span>obs<span>[</span><span>"color"</span><span>]</span><span>=</span>meti<span>.</span><span>extract_color</span><span>(</span>x_pixel<span>=</span><span>(</span>np<span>.</span><span>array</span><span>(</span>adata<span>.</span>obs<span>[</span><span>"pixel_x"</span><span>]</span><span>)</span><span>)</span><span>.</span><span>astype</span><span>(</span>np<span>.</span><span>int64</span><span>)</span><span>,</span> <br>                    y_pixel<span>=</span><span>(</span>np<span>.</span><span>array</span><span>(</span>adata<span>.</span>obs<span>[</span><span>"pixel_y"</span><span>]</span><span>)</span><span>)</span><span>.</span><span>astype</span><span>(</span>np<span>.</span><span>int64</span><span>)</span><span>,</span> image<span>=</span>img_seg<span>,</span> beta<span>=</span><span>49</span><span>)</span><br><br><span>type</span> <span>=</span> <span>[</span><span>]</span><br><span>for</span> each in adata<span>.</span>obs<span>[</span><span>"color"</span><span>]</span><span>:</span><br>    <span>if</span> each <span>&gt;</span> <span>0</span><span>:</span><br>        r <span>=</span> <span>"yes"</span><br>        <span>type</span><span>.</span><span>append</span><span>(</span>r<span>)</span><br>    <span>else</span><span>:</span><br>        r <span>=</span> <span>"no"</span><br>        <span>type</span><span>.</span><span>append</span><span>(</span>r<span>)</span><br><br>adata<span>.</span>obs<span>[</span><span>'Goblet_seg'</span><span>]</span> <span>=</span> <span>type</span><br><br>fig<span>,</span> ax <span>=</span> plt<span>.</span><span>subplots</span><span>(</span>figsize<span>=</span><span>(</span><span>10</span><span>,</span> <span>10</span><span>)</span><span>)</span><br>ax<span>.</span><span>imshow</span><span>(</span>img<span>)</span><br>ax<span>.</span><span>set_axis_off</span><span>(</span><span>)</span><br>sc<span>.</span>pl<span>.</span><span>scatter</span><span>(</span>adata<span>,</span> x<span>=</span><span>'pixel_y'</span><span>,</span> y<span>=</span><span>'pixel_x'</span><span>,</span> color<span>=</span><span>'Goblet_seg'</span><span>,</span> ax<span>=</span>ax<span>,</span> size <span>=</span> <span>150</span><span>,</span> title<span>=</span><span>'Goblet Segmentation Spot Annotations'</span><span>)</span><br># Save the figure<br>plt<span>.</span><span>savefig</span><span>(</span>plot_dir<span>+</span><span>'Goblet_spot_seg.png'</span><span>,</span> format<span>=</span><span>'png'</span><span>,</span> dpi<span>=</span><span>300</span><span>,</span> bbox_inches<span>=</span><span>'tight'</span><span>)</span><br><span aria-hidden="true"><span></span><span></span><span></span><span></span><span></span></span></code></pre><p><img data-imgfileid="100008964" data-ratio="1.1988023952095808" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmmeSvgfeoibNHo26XCyFx3ENjL1Ihiar7IVFUxUGXrOqYZC7HI8ybu3ibYgulw9pbd8zhXgqXqBoblDg/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="835" src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmmeSvgfeoibNHo26XCyFx3ENjL1Ihiar7IVFUxUGXrOqYZC7HI8ybu3ibYgulw9pbd8zhXgqXqBoblDg/640?wx_fmt=other&amp;from=appmsg"></p><h4><strong><span>Integrarion of gene expression result with segmentation result</span></strong></h4><p><span aria-label="icon: copy"><svg viewbox="64 64 896 896" focusable="false" data-icon="copy" width="1em" height="1em" fill="currentColor" aria-hidden="true"><path d="M832 64H296c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h496v688c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V96c0-17.7-14.3-32-32-32zM704 192H192c-17.7 0-32 14.3-32 32v530.7c0 8.5 3.4 16.6 9.4 22.6l173.3 173.3c2.2 2.2 4.7 4 7.4 5.5v1.9h4.2c3.5 1.3 7.2 2 11 2H704c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32zM350 856.2L263.9 770H350v86.2zM664 888H414V746c0-22.1-17.9-40-40-40H232V264h432v624z"></path></svg></span></p><pre><code>adata.obs['Goblet_combined'] = np.where((adata.obs['Goblet_seg'] == 'yes') | (adata.obs['Goblet_GE'] == 'yes'), 'yes', 'no')<br><br>fig, ax = plt.subplots(figsize=(10, 10))<br>ax.imshow(img)<br>ax.set_axis_off()<br>sc.pl.scatter(adata, x='pixel_y', y='pixel_x', color='Goblet_combined', ax=ax, size = 150,title='Goblet Combined Spot Annotations')<br># Save the figure<br>plt.savefig(plot_dir+'Goblet_spot_combined.png', format='png', dpi=300, bbox_inches='tight')<br><span aria-hidden="true"><span></span><span></span><span></span><span></span><span></span><span></span><span></span><span></span></span></code></pre><p><img data-imgfileid="100008963" data-ratio="1.2002398081534773" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmmeSvgfeoibNHo26XCyFx3ENzjhJFE8FjsbW0HYhPqDIG3B2MHNVlzStIJ89GRpEbI1ZLh3CzqkSHQ/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="834" src="https://mmbiz.qpic.cn/mmbiz_jpg/srXAibe95MmmeSvgfeoibNHo26XCyFx3ENzjhJFE8FjsbW0HYhPqDIG3B2MHNVlzStIJ89GRpEbI1ZLh3CzqkSHQ/640?wx_fmt=other&amp;from=appmsg"></p><p><span>goblet combined result spot level</span></p><h4>生活很好，有你更好</h4><p><br></p><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/DgKttEOZH65YmXaeG0qL7g",target="_blank" rel="noopener noreferrer">原文链接</a>
