---
title: "featureCounts和DEXseq做基于外显子定量的可变剪切"
date: 2023-05-22T10:22:07Z
draft: ["false"]
tags: [
  "fetched",
  "生信技能树"
]
categories: ["Acdemic"]
---
featureCounts和DEXseq做基于外显子定量的可变剪切 by 生信技能树
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><blockquote data-tool="mdnice编辑器"><p>前面我们在《生信技能树》已经是多次分享了变剪切相关教程:</p></blockquote><ul data-tool="mdnice编辑器"><li><section>rMATS这款差异可变剪切分析软件的使用体验</section></li><li><section>用LeafCutter探索转录组数据的可变剪切</section></li><li><section>用Expedition来分析单细胞转录组数据的可变剪切</section></li><li><section>使用SGSeq探索可变剪切</section></li><li><section>用DEXSeq分析可变剪切，外显子差异表达</section></li></ul><p data-tool="mdnice编辑器">但是部分教程因为时间关系，有点过时了，我们已经是大量招募小伙伴重新系统性整理梳理我们过去七年做的工作。也就是这个求贤令：<a href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247504128&amp;idx=1&amp;sn=b3a0fa7db41031f53bfbf010e8b2311e&amp;chksm=9b4b91bbac3c18adbdd8ccf1fb10cf7ddc30fea8bc73a86fcc8a1d53521c9bfa2ba21c7aa47e&amp;scene=21#wechat_redirect" data-linktype="2">曾经我给你带来了十万用户，但现在祝你倒闭</a>，走大运结识了几位优秀小伙伴！其中以前的featureCounts和DEXseq做基于外显子定量的可变剪切就需要更新了：</p><h2 data-tool="mdnice编辑器"><span></span>准备工作</h2><p data-tool="mdnice编辑器">包括conda安装，以及rna环境安装，参考基因组fasta序列获取和构建hisat2索引文件，参考gtf文件处理。</p><p data-tool="mdnice编辑器">统一使用gencode最新版：https://www.gencodegenes.org/human/</p><pre data-tool="mdnice编辑器"><span></span><code>mkdir -p <span>$HOME</span>/rna/SUPPA2<br>mkdir -p leanData  gtf  rawData<br><br>mkdir -p <span>$HOME</span>/rna/SUPPA2/gtf<br><span>cd</span>  <span>$HOME</span>/rna/SUPPA2/gtf <br><br>nohup wget ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_37/gencode.v37.annotation.gtf.gz &amp;  <br><span># 1.4G 2月  17 16:50 gencode.v37.annotation.gtf</span><br></code></pre><h3 data-tool="mdnice编辑器"><span></span>使用conda创建rna小环境：<span></span></h3><p data-tool="mdnice编辑器">代码如下：</p><pre data-tool="mdnice编辑器"><span></span><code>conda create -n rna <br>conda activate rna  <br>conda install -y trim-galore <br><span># python-3.7.9         | 57.3 MB,  openjdk-10.0.2       | 189.2 MB</span><br>conda install -y star hisat2 bowtie2 <span># 没有其它依赖 </span><br>conda install -y subread  bedtools deeptools<br>conda install -y salmon=1.4.0<br>conda install  -y -c bioconda suppa<br></code></pre><h3 data-tool="mdnice编辑器"><span></span>下载参考转录组fasta序列文件然后使用hisat2构建index文件<span></span></h3><p data-tool="mdnice编辑器">这里我们直接使用服务器现成的hisat2的index文件 （其实就是hisat2官网下载的，无需自己构建）</p><pre data-tool="mdnice编辑器"><span></span><code>$ ls -lh  /home/data/server/reference/index/hisat/hg38/ |cut -d<span>" "</span> -f 5-  <br><br>974M 12月 31 17:42 genome.1.ht2<br>728M 12月 31 17:41 genome.2.ht2<br> 15K 12月 31 17:42 genome.3.ht2<br>728M 12月 31 17:42 genome.4.ht2<br>1.3G 12月 31 17:42 genome.5.ht2<br>741M 12月 31 17:42 genome.6.ht2<br>   8 12月 31 17:42 genome.7.ht2<br>   8 12月 31 17:42 genome.8.ht2<br>1.3K 12月 31 17:42 make_hg38.sh<br> <br></code></pre><h3 data-tool="mdnice编辑器"><span></span>测序数据获取<span></span></h3><p data-tool="mdnice编辑器">这里就不赘述数据下载细节了，就是SUPPA软件文章里面举例的数据，走：<a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247492889&amp;idx=2&amp;sn=bc2ef17a3b96a257fb692f73338c6b0f&amp;scene=21#wechat_redirect" data-linktype="2">使用ebi数据库直接下载fastq测序数据</a>  , 这个教程来直接下载fastq文件啦。</p><p data-tool="mdnice编辑器">得到文件如下:</p><pre data-tool="mdnice编辑器"><span></span><code>$ ls -lh rawData/*gz|cut -d<span>" "</span> -f 5-<br>1.7G 2月  17 19:56 rawData/SRR1513329_1.fastq.gz<br>1.7G 2月  17 19:56 rawData/SRR1513329_2.fastq.gz<br>1.7G 2月  17 19:57 rawData/SRR1513330_1.fastq.gz<br>1.7G 2月  17 19:57 rawData/SRR1513330_2.fastq.gz<br>1.8G 2月  17 19:57 rawData/SRR1513331_1.fastq.gz<br>1.8G 2月  17 19:57 rawData/SRR1513331_2.fastq.gz<br>1.8G 2月  17 19:56 rawData/SRR1513332_1.fastq.gz<br>1.8G 2月  17 19:56 rawData/SRR1513332_2.fastq.gz<br>1.9G 2月  17 19:57 rawData/SRR1513333_1.fastq.gz<br>2.0G 2月  17 19:57 rawData/SRR1513333_2.fastq.gz<br>1.7G 2月  17 19:57 rawData/SRR1513334_1.fastq.gz<br>1.8G 2月  17 19:57 rawData/SRR1513334_2.fastq.gz<br></code></pre><p data-tool="mdnice编辑器">项目介绍 分组情况如下:</p><pre data-tool="mdnice编辑器"><span></span><code>negative control siRNA ：SRR1513329,SRR1513330,SRR1513331   <br>TRA2A/B siRNA ：SRR1513332,SRR1513333,SRR1513334  <br></code></pre><p data-tool="mdnice编辑器">然后质控，走fastqc软件，查看报告</p><pre data-tool="mdnice编辑器"><span></span><code>nohup fastqc -t 4   *.fq.gz 1&gt; fastq.log 2&gt;&amp;1 &amp; <br></code></pre><h3 data-tool="mdnice编辑器"><span></span>走trim流程<span></span></h3><p data-tool="mdnice编辑器">因为是双端测序，所以:</p><pre data-tool="mdnice编辑器"><span></span><code><span>cd</span> <span>$HOME</span>/rna/SUPPA2/rawData<br>ls *gz|cut -d<span>"_"</span> -f 1|sort -u |<span>while</span> <span>read</span> id;<span>do</span><br>nohup trim_galore -q 25 --phred33 --length 36 --stringency 3 --paired -o ../cleanData   <span>${id}</span>*.gz &amp; <br><span>done</span> <br><span># 如果是单端数据，需要去除 --paired  参数</span><br></code></pre><p data-tool="mdnice编辑器">得到的clean的fq文件如下:</p><pre data-tool="mdnice编辑器"><span></span><code>$ ls -lh *gz|cut -d<span>" "</span> -f 5-<br>1.6G 2月  18 10:00 SRR1513329_1_val_1.fq.gz<br>1.6G 2月  18 10:00 SRR1513329_2_val_2.fq.gz<br>1.6G 2月  18 10:00 SRR1513330_1_val_1.fq.gz<br>1.6G 2月  18 10:00 SRR1513330_2_val_2.fq.gz<br>1.7G 2月  18 10:02 SRR1513331_1_val_1.fq.gz<br>1.8G 2月  18 10:02 SRR1513331_2_val_2.fq.gz<br>1.7G 2月  18 10:02 SRR1513332_1_val_1.fq.gz<br>1.7G 2月  18 10:02 SRR1513332_2_val_2.fq.gz<br>1.9G 2月  18 11:15 SRR1513333_1_val_1.fq.gz<br>1.9G 2月  18 11:15 SRR1513333_2_val_2.fq.gz<br>1.7G 2月  18 18:04 SRR1513334_1_val_1.fq.gz<br>1.7G 2月  18 18:04 SRR1513334_2_val_2.fq.gz<br></code></pre><h3 data-tool="mdnice编辑器"><span></span>走hisat比对流程<span></span></h3><pre data-tool="mdnice编辑器"><span></span><code><span>cd</span> <span>$HOME</span>/rna/SUPPA2/cleanData <br><br>indexPrefix=/home/data/server/reference/index/hisat/hg38/genome <br><span># 单端测序</span><br>ls *gz|cut -d<span>"_"</span> -f 1|sort -u |<span>while</span> <span>read</span> id;<span>do</span><br>nohup  hisat2 -p 2  -x  <span>$indexPrefix</span> -U <span>${id}</span>*.gz -S <span>${id}</span>.hisat.sam &amp; <br><span>done</span> <br><br><span># 双端测序</span><br>ls *gz|cut -d<span>"_"</span> -f 1|sort -u |<span>while</span> <span>read</span> id;<span>do</span><br>nohup  hisat2 -p 2  -x  <span>$indexPrefix</span> -1 <span>${id}</span>*_1_val_1.fq.gz   -2 <span>${id}</span>*_2_val_2.fq.gz  -S <span>${id}</span>.hisat.sam &amp; <br><span>done</span> <br><br><span># sam文件转bam</span><br>ls *.sam|<span>while</span> <span>read</span> id ;<span>do</span> (nohup samtools sort -O bam -@ 2   -o $(basename <span>${id}</span> <span>".sam"</span>).bam   <span>${id}</span>   &amp; );<span>done</span><br><span># 这个过程会输出大量中间文件</span><br>rm *.sam <br><span># 为bam文件建立索引</span><br>ls *.bam |xargs -i samtools index {}<br><br><br></code></pre><p data-tool="mdnice编辑器">上面的区分不同步骤，每次都是并行处理全部的样品，另外一个并行处理的策略 这里先不讲解，因为这个项目 的样品数量太少了。</p><p data-tool="mdnice编辑器">得到全部的bam文件如下:</p><pre data-tool="mdnice编辑器"><span></span><code>$ ls -lh *bam |cut -d<span>" "</span> -f 5-<br>2.6G 2月  18 20:53 SRR1513329.hisat.bam<br>2.6G 2月  18 20:52 SRR1513330.hisat.bam<br>2.8G 2月  18 20:53 SRR1513331.hisat.bam<br>2.8G 2月  18 20:53 SRR1513332.hisat.bam<br>3.0G 2月  18 20:53 SRR1513333.hisat.bam<br>2.6G 2月  18 20:53 SRR1513334.hisat.bam<br></code></pre><h3 data-tool="mdnice编辑器"><span></span>走featureCounts定量流程<span></span></h3><p data-tool="mdnice编辑器">featureCounts命令的参数多种多样，可以设置基于基因的ID或者名字来进行计算表达量，或者基于外显子来。</p><pre data-tool="mdnice编辑器"><span></span><code>gtf=<span>$HOME</span>/rna/SUPPA2/gtf/gencode.v37.annotation.gtf    <br>ls -lh <span>$gtf</span> <span># 1.4G </span><br><br><span># 针对bam文件进行qc </span><br>ls *bam |<span>while</span> <span>read</span> id;<span>do</span><br>nohup qualimap rnaseq --java-mem-size=20G -gtf <span>$gtf</span> -bam <span>$id</span>  -pe -oc   <span>${id%%.*}</span>  &amp; <br><span>done</span> <br><br><span># 走featureCounts定量流程</span><br>nohup featureCounts -T 5 -p -t exon -g gene_id  -a <span>$gtf</span> \<br>-o  all.id.txt  *.bam  1&gt;counts.id.log 2&gt;&amp;1 &amp;<br>nohup featureCounts -T 5 -p -t exon -g gene_name  -a <span>$gtf</span> \<br>-o  all.name.txt  *.bam  1&gt;counts.name.log 2&gt;&amp;1 &amp;<br><br><span># https://github.com/vivekbhr/Subread_to_DEXSeq</span><br>nohup featureCounts -f -O -s 2 -p -T 5 \<br>-F GTF -a  <span>$gtf</span>  \<br>-o  all.exon.txt  *.bam  1&gt;counts.exon.log 2&gt;&amp;1 &amp;<br></code></pre><p data-tool="mdnice编辑器">仔细看了看以前的教程，但是被辣鸡的《简书》平台都封杀了：<a href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247504128&amp;idx=1&amp;sn=b3a0fa7db41031f53bfbf010e8b2311e&amp;chksm=9b4b91bbac3c18adbdd8ccf1fb10cf7ddc30fea8bc73a86fcc8a1d53521c9bfa2ba21c7aa47e&amp;scene=21#wechat_redirect" data-linktype="2">曾经我给你带来了十万用户，但现在祝你倒闭</a>，</p><ul data-tool="mdnice编辑器"><li><section>https://www.jianshu.com/p/6e6c4ff9f595</section></li><li><section>https://www.jianshu.com/p/b8f480a68cae</section></li></ul><p data-tool="mdnice编辑器">原来是把人家写好的轮子看错了，https://github.com/vivekbhr/Subread_to_DEXSeq</p><h3 data-tool="mdnice编辑器"><span></span>修改后的gtf走featureCounts<span></span></h3><p data-tool="mdnice编辑器">首先需要<strong>格式化</strong>我们的  gencode.v37.annotation.gtf 文件，代码如下:</p><pre data-tool="mdnice编辑器"><span></span><code><span>cd</span> <span>$HOME</span>/rna/SUPPA2/ref <br>git <span>clone</span> https://github.com/vivekbhr/Subread_to_DEXSeq<br>pip install HTSeq <br>python Subread_to_DEXSeq/dexseq_prepare_annotation2.py  \<br>-f gencode.v37.for_dexseq.gtf  gencode.v37.annotation.gtf  gencode.v37.for_dexseq.gff<br><br><span># 在 umac 服务器，指定全路径 </span><br>gtf=<span>"/home/yb77613/reference/gtf/gencode/gencode.v32.annotation.gtf"</span><br>gtf=/home/yb77613/reference/gtf/gencode/gencode.v37.annotation.gtf <br>python Subread_to_DEXSeq/dexseq_prepare_annotation2.py  \<br>-f gencode.v37.for_dexseq.gtf   <span>$gtf</span>  gencode.v37.for_dexseq.gff<br><br><span># 这个脚本把1.4G的gencode.v37.annotation.gtf格式化成为了如下的两个文件：</span><br><span># 152M 2月  19 10:45 gencode.v37.for_dexseq.gff</span><br><span># 143M 2月  19 10:45 gencode.v37.for_dexseq.gtf</span><br><br><span>cd</span> ../align/ <br>gtf=<span>"<span>$HOME</span>/rna/SUPPA2/ref/gencode.v37.for_dexseq.gtf"</span>   <br>ls -lh <span>$gtf</span> <br>nohup featureCounts -f -O -s 2 -p -T 5 \<br>-F GTF -a  <span>$gtf</span>  \<br>-o  all.for_DEXSeq.txt  ../align_star/*bam   1&gt;counts.for_DEXSeq.log 2&gt;&amp;1 &amp;<br><br></code></pre><h3 data-tool="mdnice编辑器"><span></span>下游的DEXSeq分析<span></span></h3><p data-tool="mdnice编辑器">项目介绍 分组情况如下:</p><pre data-tool="mdnice编辑器"><span></span><code>negative control siRNA ：SRR1513329,SRR1513330,SRR1513331   <br>TRA2A/B siRNA ：SRR1513332,SRR1513333,SRR1513334  <br></code></pre><p data-tool="mdnice编辑器">还是要借助 https://github.com/vivekbhr/Subread_to_DEXSeq 里面的代码</p><pre data-tool="mdnice编辑器"><span></span><code><span>source</span>(<span>"Subread_to_DEXSeq/load_SubreadOutput.R"</span>)<br>library(tidyverse)<br>samp &lt;- data.frame(row.names = c(<span>'SRR1513329'</span>,<span>'SRR1513330'</span>,<span>'SRR1513331'</span>,<br>                                 <span>'SRR1513332'</span>,<span>'SRR1513333'</span>,<span>'SRR1513334'</span>), <br>                   condition = rep(c(<span>"control"</span>,<span>"trt"</span>),each=3))<br>samp<br>sampleData = samp<br>countfile=<span>"featurecounts/all.for_DEXSeq.txt"</span><br>flattenedfile = <span>"gtf/gencode.v37.for_dexseq.gff"</span><br>dxd &lt;- DEXSeqDataSetFromFeatureCounts(countfile,<br>                                         flattenedfile = flattenedfile,<br>                                         sampleData = samp)<br><br><span># 重难点就是构建 dxd 这个对象。 </span><br>colData(dxd) <br>head( counts(dxd), 5 ) <br>split( seq_len(ncol(dxd)), colData(dxd)<span>$exon</span> ) <br>head( featureCounts(dxd), 5 ) <br>head( rowRanges(dxd), 3 )<br>sampleAnnotation( dxd )<br>dxd = estimateSizeFactors( dxd )<br><span>#  差异分析方法</span><br>dxd = estimateDispersions( dxd )<br>plotDispEsts( dxd )<br>dxd = testForDEU( dxd )<br>dxd = estimateExonFoldChanges( dxd, fitExpToVar=<span>"condition"</span>)<br>dxr1 = DEXSeqResults( dxd )<br>dxr1<br>mcols(dxr1)<span>$description</span><br>table ( dxr1<span>$padj</span> &lt; 0.01 )<br>table ( tapply( dxr1<span>$padj</span> &lt; 0.01, dxr1<span>$groupID</span>, any ) )<br><br><span>## ----MvsA, fig.small=TRUE, fig.cap="MA plot. Mean expression versus $log{2}$ fold change plot. </span><br><span># Significant hits at an $FDR=0.1$ are coloured in red."----</span><br>plotMA( dxr1, cex=0.8 )<br>sampleAnnotation(dxd) <br><br>save(dxr1,file = <span>'output_of_DEXSeq.Rdata'</span>) <br></code></pre><p data-tool="mdnice编辑器">统计结果存储在一个非常复杂的对象里面哦！</p><p><img data-galleryid="" data-ratio="0.6343612334801763" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wwBuaTwWHPPpibERHPJjgibrgptm4dbyo7WM3QHMN3Ogc7ibkOMTmicZYASUdD5zlCcdxosZ8B4w3Zh7A/640?wx_fmt=png" data-type="png" data-w="1362" src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wwBuaTwWHPPpibERHPJjgibrgptm4dbyo7WM3QHMN3Ogc7ibkOMTmicZYASUdD5zlCcdxosZ8B4w3Zh7A/640?wx_fmt=png"></p><figure data-tool="mdnice编辑器"><figcaption>可变剪切差异分析统计结果</figcaption></figure><h3 data-tool="mdnice编辑器"><span></span>可视化差异可变剪切结果<span></span></h3><p data-tool="mdnice编辑器">首先进行简单的统计：</p><pre data-tool="mdnice编辑器"><span></span><code>&gt; mcols(dxr1)<span>$description</span><br> [1] <span>"group/gene identifier"</span>                                       <br> [2] <span>"feature/exon identifier"</span>                                     <br> [3] <span>"mean of the counts across samples in each feature/exon"</span>      <br> [4] <span>"exon dispersion estimate"</span>                                    <br> [5] <span>"LRT statistic: full vs reduced"</span>                              <br> [6] <span>"LRT p-value: full vs reduced"</span>                                <br> [7] <span>"BH adjusted p-values"</span>                                        <br> [8] <span>"exon usage coefficient"</span>                                      <br> [9] <span>"exon usage coefficient"</span>                                      <br>[10] <span>"relative exon usage fold change"</span>                             <br>[11] <span>"GRanges object of the coordinates of the exon/feature"</span>       <br>[12] <span>"matrix of integer counts, of each column containing a sample"</span><br>[13] <span>"list of transcripts overlapping with the exon"</span>               <br>&gt; table ( dxr1<span>$padj</span> &lt; 0.01 )<br><br> FALSE   TRUE <br>227203   2333 <br>&gt; table ( tapply( dxr1<span>$padj</span> &lt; 0.01, dxr1<span>$groupID</span>, any ) )<br><br>FALSE  TRUE <br>  796  1206 <br></code></pre><p data-tool="mdnice编辑器">任何可以选取特定的基因去看可变剪切情况，前提是拿到了全部的统计学显著的基因：</p><pre data-tool="mdnice编辑器"><span></span><code>dat=dxr1@listData<br>kp=dat<span>$padj</span> &lt; 0.01<br>table(kp)<br>kp[is.na(kp)]=<span>'FALSE'</span><br>table(kp)<br>allG=dat<span>$groupID</span><br>sg=allG[as.logical(kp)];head(sg)<br>sg=unique(sg);length(sg)<br>head(sg)<br><br>&gt; head(sg)<br>[1] <span>"ENSG00000001497.18"</span>                    <span>"ENSG00000002586.20_PAR_Y"</span>             <br>[3] <span>"ENSG00000002834.18"</span>                    <span>"ENSG00000003393.16"</span>                   <br>[5] <span>"ENSG00000004455.17"</span>                    <span>"ENSG00000004534.15+ENSG00000003756.17"</span><br></code></pre><p data-tool="mdnice编辑器">我们就可以随机选取一个进行可视化啦！</p><pre data-tool="mdnice编辑器"><span></span><code>plotDEXSeq( dxr1, <span>"ENSG00000003393.16"</span>, <br>            legend=TRUE, cex.axis=1.2, cex=1.3, lwd=2 )<br><br></code></pre><p data-tool="mdnice编辑器">这应该是展现了一个很明显的外显子跳跃吧：</p><p><img data-galleryid="" data-ratio="0.8091822094691535" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wwBuaTwWHPPpibERHPJjgibrgZ1gEZSpGY3rPm1eicZAZaQurlLA9hhvaVQ6QiceOvx4qQ0YicNVMOqCqw/640?wx_fmt=png" data-type="png" data-w="1394" src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wwBuaTwWHPPpibERHPJjgibrgZ1gEZSpGY3rPm1eicZAZaQurlLA9hhvaVQ6QiceOvx4qQ0YicNVMOqCqw/640?wx_fmt=png"></p><figure data-tool="mdnice编辑器"><figcaption> </figcaption></figure><p data-tool="mdnice编辑器">而其它软件，比如 SUPPA通常是会对可变剪切进行分类，如下所示：</p><ul data-tool="mdnice编辑器"><li><section><strong>SE</strong> (exon skipping)</section></li><li><section><strong>RI</strong> (retained intron)</section></li><li><section><strong>MXE</strong> (mutually exclusive exons)</section></li><li><section><strong>A5SS</strong> (alternative 5' splice site)</section></li><li><section><strong>A3SS</strong> (alternative 3' splice site)</section></li><li><section><strong>ALE</strong> (alternative last exon)</section></li><li><section><strong>AFE</strong> (alternative first exon)</section></li></ul></section><h3 data-tool="mdnice编辑器">文末友情推荐</h3><p data-tool="mdnice编辑器">与十万人一起学生信，你值得拥有下面的学习班：</p><ul data-tool="mdnice编辑器"><li><section><a href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247504010&amp;idx=1&amp;sn=3839c684dde6f75684cb116984291986&amp;chksm=9b4b9031ac3c19270220e8846f3cd13a159ff4abbcda4ef2dbb9f16c2f6b045723b21f5cc051&amp;scene=21#wechat_redirect" data-linktype="2">数据挖掘（GEO,TCGA,单细胞）2021第5期</a></section></li><li><section><a href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247504010&amp;idx=1&amp;sn=3839c684dde6f75684cb116984291986&amp;chksm=9b4b9031ac3c19270220e8846f3cd13a159ff4abbcda4ef2dbb9f16c2f6b045723b21f5cc051&amp;scene=21#wechat_redirect" data-linktype="2">生信爆款入门-2021第5期</a></section></li></ul></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/6_WUuWpyt8tRsTy9R3-fRA",target="_blank" rel="noopener noreferrer">原文链接</a>
