---
title: "Seurat空间转录组分析（二）标准分析流程"
date: 2023-05-18T01:41:56Z
draft: ["false"]
tags: [
  "fetched",
  "生信随笔"
]
categories: ["Acdemic"]
---
Seurat空间转录组分析（二）标准分析流程 by 生信随笔
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><p data-tool="mdnice编辑器">上期推文 <a href="https://mp.weixin.qq.com/s?__biz=Mzg5MjcxNzg1NA==&amp;mid=2247486428&amp;idx=1&amp;sn=9be6ba02adeeb86aa7eea5e00aa70070&amp;scene=21#wechat_redirect" data-linktype="2">Seurat空间转录组分析（一）数据读入 </a>介绍了三种方法将空转数据读入R语言中。本期推文是跟着Seurat官网对空转数据进行标准化分析（https://satijalab.org/seurat/articles/spatial_vignette.html）。<strong>值得注意的是，本文基于Seurat v4进行分析。</strong>在撰写本文时，Seurat官网进行了更新，目前最新版本是Seurat v5测试版。Seurat v5相比于v4，有更高的计算性能和更低的硬件要求。在Seurat v5中，作者引入了新方法可以更好的分析大样本的单细胞数据集，此外还极大地完善了单细胞多组学数据处理的需求，特别是对单细胞空间转录组数据的处理上进行了极大的提升。参见：</p><ul data-tool="mdnice编辑器"><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247511497&amp;idx=1&amp;sn=d1e9661d4df9063b5f1225b45ed41e11&amp;scene=21#wechat_redirect" data-linktype="2">Seurat V5 | 构建单细胞多模态数据整合的统一场论</a></section></li></ul><p data-tool="mdnice编辑器"><span>需要注意的是，目前v5仍存在诸多bug，有待开发者去解决，所以建议初学者谨慎尝试。总的来说，尽管v5将至，但是其分析的标准流程仍然和v4差不多，所以</span><span>了解和熟悉Seurat v4的空间转录组标准流程，对后续v5版本的学习具有指导意义。</span></p><h3 data-tool="mdnice编辑器"><span></span>一. 读入数据<span></span></h3><h4 data-tool="mdnice编辑器"><span></span>Step1. R包加载及安装<span></span></h4><pre data-tool="mdnice编辑器"><span></span><code><span>library</span>(Seurat)<br><span>library</span>(SeuratData)<br><span>library</span>(ggplot2)<br><span>library</span>(patchwork)<br><span>library</span>(dplyr)<br></code></pre><h4 data-tool="mdnice编辑器"><span></span>Step2. 加载Seurat官网的示例数据<span></span></h4><p data-tool="mdnice编辑器">示例数据在https://support.10xgenomics.com/spatial-gene-expression/datasets获得，并使用<code>Load10X_Spatial()</code>函数将其加载到<code>Seurat</code>。这将读取<code>spaceranger</code>管道的输出，并返回<code>Seurat</code>对象，该对象包含spot级别的表达数据以及相关的组织切片图像。</p><p data-tool="mdnice编辑器">本文的示例数据是10X平台的，内置于<code>SeuratData</code>R包，加载方式如下：</p><pre data-tool="mdnice编辑器"><span></span><code>InstallData(<span>"stxBrain"</span>)<br>brain &lt;- LoadData(<span>"stxBrain"</span>, type = <span>"anterior1"</span>)<br>brain<br><span># An object of class Seurat </span><br><span># 31053 features across 2696 samples within 1 assay </span><br><span># Active assay: Spatial (31053 features, 0 variable features)</span><br><span>## 检查示例数据</span><br>head(brain@meta.data)<br><span>#                   orig.ident nCount_Spatial nFeature_Spatial slice   region</span><br><span># AAACAAGTATCTCCCA-1  anterior1          13069             4242     1 anterior</span><br><span># AAACACCAATAACTGC-1  anterior1          37448             7860     1 anterior</span><br><span># AAACAGAGCGACTCCT-1  anterior1          28475             6332     1 anterior</span><br><span># AAACAGCTTTCAGAAG-1  anterior1          39718             7957     1 anterior</span><br><span># AAACAGGGTCTATATT-1  anterior1          33392             7791     1 anterior</span><br><span># AAACATGGTGAGAGGA-1  anterior1          20955             6291     1 anterior</span><br>SpatialDimPlot(brain,alpha = <span>0</span>)<br><br><span>#更多参数参考</span><br>?SpatialDimPlot()<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.785958904109589" data-type="other" data-w="584" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxpkLqfUKqweA3hlib8JchW2DNn4X2eu1HQKt9wSwUmTk7D12zl7XdTDg/640?wx_fmt=other" src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxpkLqfUKqweA3hlib8JchW2DNn4X2eu1HQKt9wSwUmTk7D12zl7XdTDg/640?wx_fmt=other"></figure><h3 data-tool="mdnice编辑器"><span></span>二. 数据预处理<span></span></h3><p data-tool="mdnice编辑器">在spot中基因表达数据进行初始的预处理步骤与典型的scRNA-seq相似。我们首先需要对数据进行<strong>规范化</strong>，以考虑数据点之间测序深度的差异。我们注意到，对于空间数据集来说，分子数/点的差异可能是巨大的，特别是在整个组织的细胞密度存在差异的情况下。我们在这里看到大量的异质性，这需要有效的<strong>标准化</strong>。</p><pre data-tool="mdnice编辑器"><span></span><code>plot1 &lt;- VlnPlot(brain, features = "nCount_Spatial", pt.size = 0.1) + NoLegend()<br>plot2 &lt;- SpatialFeaturePlot(brain, features = "nCount_Spatial") + theme(legend.position = "right")<br>p = wrap_plots(plot1, plot2)<br>#ggsave(p,filename = "../Output/test.pdf")<br></code></pre><img data-ratio="0.6083333333333333" data-type="other" data-w="1080" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxKJf0TktVAniaB5nFBcT8WhZCj07QQ1fMgIPiafLKLgAPp0icV85hYb6Sg/640?wx_fmt=other" src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxKJf0TktVAniaB5nFBcT8WhZCj07QQ1fMgIPiafLKLgAPp0icV85hYb6Sg/640?wx_fmt=other"><pre data-tool="mdnice编辑器"><span></span><code>p3 &lt;- VlnPlot(brain,  features = "nFeature_Spatial",pt.size = 0.1,cols = "tomato",group.by = "orig.ident") + NoLegend()<br>p4 &lt;- FeatureScatter(brain, feature1 = "nCount_Spatial",feature2= "nFeature_Spatial",group.by = "orig.ident")+ NoLegend()<br>wrap_plots(p3, p4)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.6171428571428571" data-type="other" data-w="700" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxZP2DUhJHZ9qnlBUeCmX7hQO14dgh3HSes5MQgEhgBMXHHDiaxmibjPnQ/640?wx_fmt=other" src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxZP2DUhJHZ9qnlBUeCmX7hQO14dgh3HSes5MQgEhgBMXHHDiaxmibjPnQ/640?wx_fmt=other"><figcaption>image-20211114140540071</figcaption></figure><p data-tool="mdnice编辑器">这些图表明，分子计数（molecular counts）在点间的差异不仅是技术性的，而且还依赖于组织解剖。例如，组织中神经元消耗的区域(如皮层白质)，可再生地显示出较低的分子计数。因此，标准方法(如LogNormalize函数)可能会有问题，因为它会强制每个数据点在标准化之后具有相同的底层“大小”。</p><p data-tool="mdnice编辑器">作为一种替代方法，我们推荐使用sctransform (Hafemeister和Satija，已出版)，它构建了基因表达的正则化负二项模型，以便在保留生物差异的同时考虑技术因素。有关sctransform的更多信息，请参见 here(https://www.biorxiv.org/content/10.1101/576827v2)的预印和here(https://satijalab.org/seurat/sctransform_vignette.html)的Seurat教程。sctransform将数据归一化，检测高方差特征，并将数据存储在SCT分析中:</p><pre data-tool="mdnice编辑器"><span></span><code>brain &lt;- SCTransform(brain, assay = "Spatial", return.only.var.genes = FALSE, verbose = FALSE)<br><br>#提取表达量变化最高的10个基因<br>top10&lt;- head(VariableFeatures(brain),10)<br>top10<br>##[1] "Ttr"     "Hbb-bs"  "Plp1"    "Mbp"     "Hba-a1"  "Ptgds"   "Penk"    "Hba-a2"  "S100a5"  "Ppp1r1b"<br><br>#类似常规转录组，我们也可以展示高变基因；<br>p5 &lt;- VariableFeaturePlot(brain,cols = c( "gray60", "red"))+NoLegend()<br>p6 &lt;- LabelPoints(plot = p5,points = top10, repel = TRUE,xnudge=0,ynudge=0)<br>p5 | p6<br></code></pre><img data-ratio="0.6171428571428571" data-type="other" data-w="700" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxga6lP6xrw1dfib2rSC2KA0ic3tTpq2ZricusvA8a5Dc5iclR6f85cokRoA/640?wx_fmt=other" src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxga6lP6xrw1dfib2rSC2KA0ic3tTpq2ZricusvA8a5Dc5iclR6f85cokRoA/640?wx_fmt=other"><blockquote data-tool="mdnice编辑器"><p>Q1：空转数据需不需要进行过滤？</p><p>A：参见追风少年【关于10X空间转录组质控分析的讨论】https://www.jianshu.com/p/13fafd6567a4。据我的了解，基于10x Visium测序技术的空转数据，一般不需要对单个spot进行过滤，除非有一些质量极低的spot（例如极低的基因数量）。</p><p>另外，不同于单细胞数据分析，空转的数据强烈建议使用SCT转换进行标准化和归一化。</p></blockquote><h3 data-tool="mdnice编辑器"><span></span>三. 基因表达可视化<span></span></h3><p data-tool="mdnice编辑器">在<code>Seurat</code>，我们有功能来探索和互动空间数据固有的视觉特性。<code>Seurat</code>中的<code>SpatialFeaturePlot()</code>函数扩展了<code>FeaturePlot()</code>，可以将分子数据覆盖在组织组织学上。例如，在这个小鼠大脑的数据集中，<em>Hpca</em>基因是强海马marker，<em>Ttr</em>基因是脉络膜丛marker。</p><pre data-tool="mdnice编辑器"><span></span><code>SpatialFeaturePlot(brain, features = c("Hpca", "Ttr"))<br></code></pre><img data-ratio="0.5920281359906213" data-type="other" data-w="853" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxOibru27Z959pCzNtu4yDE0eNt2woyutQBt1LiaxO08MiadlBtice77ee6Q/640?wx_fmt=other" src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxOibru27Z959pCzNtu4yDE0eNt2woyutQBt1LiaxO08MiadlBtice77ee6Q/640?wx_fmt=other"><p data-tool="mdnice编辑器"><code>Seurat</code>的默认参数强调分子数据的可视化。但是，你也可以调整斑点的大小(和它们的透明度)来改善组织学图像的可视化，通过改变以下参数:</p><ul data-tool="mdnice编辑器"><li><section>pt.size。因素-这将比例大小的斑点。默认为1.6</section></li><li><section>alpha -最小和最大透明度。默认是c(1,1)</section></li><li><section>尝试设置为alpha c(0.1, 1)，以降低表达式较低的点的透明度</section></li><li><section><strong>stroke：</strong>设置spots的描边粗细,默认stroke = 0.25</section></li></ul><pre data-tool="mdnice编辑器"><span></span><code>p7 &lt;- SpatialFeaturePlot(brain, features = "Ttr", pt.size.factor = 1)<br>p8 &lt;- SpatialFeaturePlot(brain, features = "Ttr", alpha = c(0.1, 1))<br>p7 + p8<br></code></pre><img data-ratio="0.600459242250287" data-type="other" data-w="871" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxkHvpBFchldm3qmK6SQWtuuqmPR21AtX73Mib7cKicLibknD8xwsh25LVA/640?wx_fmt=other" src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxkHvpBFchldm3qmK6SQWtuuqmPR21AtX73Mib7cKicLibknD8xwsh25LVA/640?wx_fmt=other"><h3 data-tool="mdnice编辑器"><span></span>四. 降维聚类及可视化<span></span></h3><p data-tool="mdnice编辑器">然后，我们可以继续对RNA表达数据进行降维和聚类，使用与我们用于scRNA-seq分析相同的工作流程：</p><pre data-tool="mdnice编辑器"><span></span><code>brain &lt;- RunPCA(brain, assay = "SCT", verbose = FALSE)<br>brain &lt;- FindNeighbors(brain, reduction = "pca", dims = 1:30)<br>brain &lt;- FindClusters(brain, verbose = FALSE)<br>brain &lt;- RunUMAP(brain, reduction = "pca", dims = 1:30)<br></code></pre><p data-tool="mdnice编辑器">我们可以在UMAP空间，使用<code>DimPlot</code>或使用<code>SpatialDimPlot</code>将聚类的结果显示在图像上：</p><pre data-tool="mdnice编辑器"><span></span><code>p9 &lt;- DimPlot(brain, reduction = "umap", label = TRUE)<br>p10 &lt;- SpatialDimPlot(brain, label = TRUE, label.size = 3)<br>p9 + p10<br></code></pre><img data-ratio="0.5967741935483871" data-type="other" data-w="868" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxb2icXmK0aNUZ4tBuZzx1YPIxwpmHlNRNT7mibgo2DM5k2sJAbhSK3atA/640?wx_fmt=other" src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxb2icXmK0aNUZ4tBuZzx1YPIxwpmHlNRNT7mibgo2DM5k2sJAbhSK3atA/640?wx_fmt=other"><p data-tool="mdnice编辑器">因为有许多颜色，所以可视化哪个颜色属于哪个簇是很有挑战性的。我们有一些策略来帮助解决这个问题。通过设置label参数，可以在每个集群的中间位置放置一个彩色框（参见上面的图）以及do.hover。<code>SpatialDimPlot</code>的悬停参数允许交互式查看当前的spot标识。</p><pre data-tool="mdnice编辑器"><span></span><code>SpatialDimPlot(brain, cells.highlight = CellsByIdentities(object = brain, idents = c(2, 1, 4, 3,<br>    5, 8)), facet.highlight = TRUE, ncol = 3)<br></code></pre><img data-ratio="0.7657393850658858" data-type="other" data-w="683" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCx34bTOW43FCiaq6ctqjdofxoM8B2iaymaOA2szOCcPU9IZYoxVopJyfYQ/640?wx_fmt=other" src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCx34bTOW43FCiaq6ctqjdofxoM8B2iaymaOA2szOCcPU9IZYoxVopJyfYQ/640?wx_fmt=other"><pre data-tool="mdnice编辑器"><span></span><code>## 或者<br>cluster &lt;- CellsByIdentities(object = brain, idents = c(1, 2, 3, 4,5,8))<br>str(cluster)<br>##List of 6<br>##$ 1: chr [1:291] "AAAGGCTCTCGCGCCG-1" "AAATCGTGTACCACAA-1" "AAATGATTCGATCAGC-1" "AAATTAACGGGTAGCT-1" ...<br>##$ 2: chr [1:288] "AAACAAGTATCTCCCA-1" "AAACTCGTGATATAAG-1" "AAATAGCTTAGACTTT-1" "AAATCTAGCCCTGCTA-1" ...<br>##$ 3: chr [1:215] "AAACAGAGCGACTCCT-1" "AAACCGGGTAGGTACC-1" "AAATAACCATACGGGA-1" "AAATTTGCGGGTGTGG-1" ...<br>##$ 4: chr [1:211] "AAACCGTTCGTCCAGG-1" "AAACGGTTGCGAACTG-1" "AAATCGCGGAAGGAGT-1" "AAATTACACGACTCTG-1" ...<br>##$ 5: chr [1:208] "AAACACCAATAACTGC-1" "AAACATGGTGAGAGGA-1" "AAACGAAGATGGAGTA-1" "AAACTTAATTGCACGC-1" ...<br>##$ 8: chr [1:155] "AAACATTTCCCGGATT-1" "AAACGGGCGTACGGGT-1" "AAGACATACGTGGTTT-1" "AAGAGGCCCTTTGGAA-1" ...<br><br>SpatialDimPlot(brain, cells.highlight = cluster, facet.highlight = TRUE,cols.highlight = c( "green", "red"), ncol = 3)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.6171428571428571" data-type="other" data-w="700" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxcG46Xff91Yr1TWLRDDDjKxNNHdLdUZu2icSJicaGmorS643ex3tMYLRg/640?wx_fmt=other" src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxcG46Xff91Yr1TWLRDDDjKxNNHdLdUZu2icSJicaGmorS643ex3tMYLRg/640?wx_fmt=other"><figcaption>image-20211114141649067</figcaption></figure><h3 data-tool="mdnice编辑器"><span></span>五. 交互式绘图可视化<span></span></h3><pre data-tool="mdnice编辑器"><span></span><code>SpatialDimPlot(brain, interactive = TRUE)<br>SpatialFeaturePlot(brain, features = "Ttr", interactive = TRUE)<br>LinkedDimPlot(brain)<br></code></pre><h3 data-tool="mdnice编辑器"><span></span>六. 空间变量特征的识别<span></span></h3><p data-tool="mdnice编辑器"><code>Seurat</code>提供了两个工作流程来识别与组织空间位置相关的分子特征。</p><h4 data-tool="mdnice编辑器"><span></span>方法1：<span></span></h4><p data-tool="mdnice编辑器">第一种是根据组织内预先标注的解剖区域进行差异表达，这种差异表达可以通过非监督聚类或先验知识来确定。这种策略在这种情况下有效，因为上面的clusters显示出明显的空间差异：</p><pre data-tool="mdnice编辑器"><span></span><code>de_markers &lt;- FindMarkers(brain, ident.1 = 5, ident.2 = 6)<br>head(de_markers)<br></code></pre><section data-tool="mdnice编辑器"><table><thead><tr><th><br></th><th>p_val</th><th>avg_log2FC</th><th>pct.1</th><th>pct.2</th><th>p_val_adj</th></tr></thead><tbody><tr><td>Calb2</td><td>6.427214e-69</td><td>3.336874</td><td>1.000</td><td>0.537</td><td>1.135560e-64</td></tr><tr><td>Camk2n1</td><td>1.519204e-68</td><td>-2.450388</td><td>1.000</td><td>1.000</td><td>2.684130e-64</td></tr><tr><td>Nrgn</td><td>1.573095e-68</td><td>-3.229826</td><td>0.971</td><td>1.000</td><td>2.779344e-64</td></tr><tr><td>Stx1a</td><td>2.104119e-68</td><td>-2.238353</td><td>0.784</td><td>1.000</td><td>3.717558e-64</td></tr><tr><td>Nptxr</td><td>9.820482e-68</td><td>-1.918293</td><td>0.942</td><td>1.000</td><td>1.735083e-63</td></tr><tr><td>Lingo1</td><td>2.124777e-67</td><td>-1.913281</td><td>0.880</td><td>1.000</td><td>3.754056e-63</td></tr></tbody></table></section><p data-tool="mdnice编辑器">可视化，绘制前3个差异基因的表达映射图：</p><pre data-tool="mdnice编辑器"><span></span><code>SpatialFeaturePlot(object = brain, features = rownames(de_markers)[1:3], alpha = c(0.1, 1), ncol = 3)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.6171428571428571" data-type="other" data-w="700" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxg6fSibBlqHlOniavV9Jz6p5PqFVBHPjjjR5ONsbdLrEnce7hAyibPllGQ/640?wx_fmt=other" src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxg6fSibBlqHlOniavV9Jz6p5PqFVBHPjjjR5ONsbdLrEnce7hAyibPllGQ/640?wx_fmt=other"><figcaption>image-20211113175931652</figcaption></figure><h4 data-tool="mdnice编辑器"><span></span>方法2：<span></span></h4><p data-tool="mdnice编辑器">在<code>FindSpatiallyVariables</code>中实现的另一种方法是在没有预注释的情况下搜索显示空间模式的特性。默认的方法<code>(method = 'markvariogram')</code>受到<code>Trendsceek</code>的启发，后者将空间转录组学数据建模为标记点过程，并计算一个<code>variogram</code>，它识别其表达水平取决于其空间位置的基因。更具体地说，这个过程计算伽玛（r）值，测量两个点之间一定的“r”距离的相关性。默认情况下，我们在这些分析中使用的r值为‘5’，并且只计算可变基因的这些值（其中的变异是独立于空间位置计算的），以节省时间。</p><pre data-tool="mdnice编辑器"><span></span><code>## 3000个高变基因<br>length(VariableFeatures(brain)) <br><br>## 从前1000个高变基因中查找差异基因<br>variable_top1000 = VariableFeatures(brain)[1:1000]<br>brain &lt;- FindSpatiallyVariableFeatures(brain, assay = "SCT",<br>    features = variable_top1000,<br>    selection.method = "markvariogram")<br><br>## 得到差异基因<br>DEGs&lt;- SpatiallyVariableFeatures(brain, selection.method = "markvariogram")<br>length(DEGs) # 1000<br><br>## 可视化top差异化的高变基因<br>top.features &lt;- head(DEGs, 6)<br>SpatialFeaturePlot(brain, features = top.features, ncol = 3, alpha = c(0.1, 1)))<br></code></pre><img data-ratio="0.7563025210084033" data-type="other" data-w="952" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxtxY5nH2gtsbic7oJ7zvuNUkJd3GYxq94kY7Sl9MqC8U4dkeUDX0yHQQ/640?wx_fmt=other" src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxtxY5nH2gtsbic7oJ7zvuNUkJd3GYxq94kY7Sl9MqC8U4dkeUDX0yHQQ/640?wx_fmt=other"><h3 data-tool="mdnice编辑器"><span></span>七. 可视化解剖区域的子集<span></span></h3><p data-tool="mdnice编辑器">与单细胞对象一样，您可以对该对象进行子集设置，以将重点放在数据的子集上。在这里，我们大致取到了额叶皮质的子集。这个过程也促进了这些数据与下一节的皮层scRNA-seq数据集的整合。首先，我们取clusters的一个子集，然后根据精确的坐标信息进一步“精修”。设置好亚组后，我们可以在完整图像或裁剪图像上看到皮质细胞。</p><h4 data-tool="mdnice编辑器"><span></span>Step1. 粗略的取到皮层得到子集<span></span></h4><pre data-tool="mdnice编辑器"><span></span><code>cortex &lt;- subset(brain, idents = c(1, 2, 3, 4, 6, 7))<br>## 可视化1 2 3 4 6 7 clusters的子集结果<br>p11 &lt;- SpatialDimPlot(cortex, crop = TRUE, label = TRUE, label.size = 3)<br>p11<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.6171428571428571" data-type="other" data-w="700" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxicqqNbDdYMzB6JydJbjArlnF2JPmCjB91DTEr8nWqKxrHB4tao8B2mw/640?wx_fmt=other" src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxicqqNbDdYMzB6JydJbjArlnF2JPmCjB91DTEr8nWqKxrHB4tao8B2mw/640?wx_fmt=other"><figcaption>image-20211113215830718</figcaption></figure><h4 data-tool="mdnice编辑器"><span></span>Step2. 过滤，取皮层细胞<span></span></h4><p data-tool="mdnice编辑器">可以看到需要删除其他的细胞，这可以用<code>SpatialDimPlot</code>可视化，根据切片图像坐标（前缀需和Key保持一致），进一步去除多余的spots数据:</p><pre data-tool="mdnice编辑器"><span></span><code>## 2.1 获取切片图像坐标和Key：<br>Key(object =cortex@images$anterior1)<br>img&lt;- GetTissueCoordinates(cortex)<br>head(img)<br>##						imagerow 	imagecol<br>##AAACAAGTATCTCCCA-1	385.9725	438.9501		<br>##AAACAGAGCGACTCCT-1	163.3735	410.4991		<br>##AAACCGGGTAGGTACC-1	336.5060	175.9208		<br>##AAACCGTTCGTCCAGG-1	398.3649	225.6971		<br>##AAACGGTTGCGAACTG-1	491.1016	286.1102<br><br>## 2.2 分群信息；<br>ident&lt;- Idents(cortex)<br>head(ident)<br>##					 ident<br>##AAACAAGTATCTCCCA-1	2			<br>##AAACAGAGCGACTCCT-1	3			<br>##AAACCGGGTAGGTACC-1	3			<br>##AAACCGTTCGTCCAGG-1	4			<br>##AAACGGTTGCGAACTG-1	4			<br>##AAACTCGTGATATAAG-1	2<br><br>## 2.3 合并切片图像坐标、分群信息；<br>imgId&lt;- data.frame(img,ident)<br>head(imgId)<br>## 						imagerow 	imagecol ident<br>##AAACAAGTATCTCCCA-1	385.9725	438.9501	2	<br>##AAACAGAGCGACTCCT-1	163.3735	410.4991	3	<br>##AAACCGGGTAGGTACC-1	336.5060	175.9208	3	<br>##AAACCGTTCGTCCAGG-1	398.3649	225.6971	4	<br>##AAACGGTTGCGAACTG-1	491.1016	286.1102	4	<br>##AAACTCGTGATATAAG-1	219.0361	478.0379	2<br></code></pre><p data-tool="mdnice编辑器">显微照片的像素坐标原点在左上角，而图表的原点在左下角，因此需要做垂直翻转<code>y=540-imagerow</code>：</p><pre data-tool="mdnice编辑器"><span></span><code>#spots坐标信息的探索：<br>p12 &lt;- ggplot(img, aes(x=imagecol, y=540-imagerow,color = ident)) +<br>geom_point(size = 1.6)+theme_bw() <br>p11 +p12<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.3509259259259259" data-type="other" data-w="1080" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxjAOOJUBS4KnCTlpbtOBvu1KywnRdqB5m8oklZt7eLczmqDWHcNbvJQ/640?wx_fmt=other" src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxjAOOJUBS4KnCTlpbtOBvu1KywnRdqB5m8oklZt7eLczmqDWHcNbvJQ/640?wx_fmt=other"><figcaption>image-20211113221501425</figcaption></figure><p data-tool="mdnice编辑器"><strong>不垂直翻转的话：</strong></p><pre data-tool="mdnice编辑器"><span></span><code>p13 &lt;- ggplot(img, aes(x=imagecol, y=imagerow,color = ident)) +<br>geom_point(size = 1.6)+theme_bw() <br>p11+p_13<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.39444444444444443" data-type="other" data-w="1080" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxed9ftWoOIF78Tn6QZ6y5gd96ammIlGQmbrwKGCJBNQIyZUViczSzItw/640?wx_fmt=other" src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxed9ftWoOIF78Tn6QZ6y5gd96ammIlGQmbrwKGCJBNQIyZUViczSzItw/640?wx_fmt=other"><figcaption>image-20211113223719709</figcaption></figure><p data-tool="mdnice编辑器">根据上图切片图像坐标（前缀需和Key保持一致，即为<code>anterior1</code>）进一步去除多余的spots数据：</p><pre data-tool="mdnice编辑器"><span></span><code>cortex &lt;- subset(cortex, anterior1_imagerow &gt; 400 | anterior1_imagecol &lt; 150, invert = TRUE)<br>SpatialDimPlot(cortex, crop = TRUE, label = TRUE, label.size = 5)<br></code></pre><img data-ratio="0.8052434456928839" data-type="other" data-w="801" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxPt65vkuRF119Cm9MI5HeHIE4sgOkNZJRlYr7eVIa16yNqHhMHInxDA/640?wx_fmt=other" src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxPt65vkuRF119Cm9MI5HeHIE4sgOkNZJRlYr7eVIa16yNqHhMHInxDA/640?wx_fmt=other"><pre data-tool="mdnice编辑器"><span></span><code>cortex &lt;- subset(cortex, anterior1_imagerow &gt; 275 &amp; anterior1_imagecol &gt; 370, invert = TRUE)<br>SpatialDimPlot(cortex, crop = TRUE, label = TRUE, label.size = 5)<br></code></pre><img data-ratio="0.6969990319457889" data-type="other" data-w="1033" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCx1yaR2aDfsyRR2xItenicEwX1rUiaekicoST6053XWsaicBMaEBUib6q8ZUg/640?wx_fmt=other" src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCx1yaR2aDfsyRR2xItenicEwX1rUiaekicoST6053XWsaicBMaEBUib6q8ZUg/640?wx_fmt=other"><pre data-tool="mdnice编辑器"><span></span><code>cortex &lt;- subset(cortex, anterior1_imagerow &gt; 250 &amp; anterior1_imagecol &gt; 440, invert = TRUE)<br>SpatialDimPlot(cortex, crop = TRUE, label = TRUE, label.size = 5)<br></code></pre><img data-ratio="0.8670967741935484" data-type="other" data-w="775" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxGG1ibCerdCVmcxlwLGFchV9LfW9XGkMnk3Pd3A928av2UG3wt597fkg/640?wx_fmt=other" src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxGG1ibCerdCVmcxlwLGFchV9LfW9XGkMnk3Pd3A928av2UG3wt597fkg/640?wx_fmt=other"><p data-tool="mdnice编辑器">最后，对切片特定区域的数据进行局部和整体可视化；</p><pre data-tool="mdnice编辑器"><span></span><code>p14&lt;- SpatialDimPlot(cortex, crop = <span>TRUE</span>, label = <span>TRUE</span>, label.size = <span>3</span>)<br>p15&lt;- SpatialDimPlot(cortex, crop = <span>FALSE</span>, label = <span>TRUE</span>,<br>pt.size.factor = <span>1</span>, label.size = <span>3</span>)<br>p14+p15<br></code></pre><img data-ratio="0.37592592592592594" data-type="other" data-w="1080" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCx69WnXZ7ffibOYGdubF1rU8o1yf3MndacCHVd4BvyB3VQx6yFJswjrXg/640?wx_fmt=other" src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCx69WnXZ7ffibOYGdubF1rU8o1yf3MndacCHVd4BvyB3VQx6yFJswjrXg/640?wx_fmt=other"><p data-tool="mdnice编辑器">如需进一步分析的话，需要再次进行<code>SCTransform</code>转换，详见下一章节：</p><pre data-tool="mdnice编辑器"><span></span><code>## After subsetting, we renormalize cortex<br>#cortex &lt;- SCTransform(cortex, assay = "Spatial", verbose = FALSE) %&gt;%<br>#    RunPCA(verbose = FALSE)<br>## the annotation is stored in the 'subclass' column of object metadata<br>#DimPlot(allen_reference, group.by = "subclass", label = TRUE)<br></code></pre><h3 data-tool="mdnice编辑器"><span></span>八. 整合空间转录组与单细胞数据<span></span></h3><p data-tool="mdnice编辑器">在~50um时，visium检测的斑点将包含多个细胞的表达谱。对于越来越多可用scRNA-seq数据的系统，用户可能有兴趣“反卷积”每个空间体素，以预测单元类型的底层组成。在准备这篇文章的过程中，我们测试了各种各样的脱卵方法和整合方法（decovonlution and integration methods），使用的是来自Allen研究所的参考scRNA-seq数据集（reference scRNA-seq dataset），其中包含了大约14000只成年小鼠的皮层细胞分类，并使用SMART-Seq2 protocol 生成。</p><p data-tool="mdnice编辑器">作者认为，<strong>使用集成方法（ integration methods）（与反褶积方法deconvolution methods相反）可以获得更好的性能</strong>，这可能是因为空间和单细胞数据集的噪声模型本质上是不同的，而集成方法的特殊设计是为了对这些差异具有鲁棒性。因此我们应用“锚”的集成工作流一如最近在 Seurat v3介绍的，使注释的概率从一个reference 数据query 数据集转移。因此,我们遵循这里介绍的标签转移流程,利用sctransform正常化,但预测新方法被开发来完成这项任务。</p><p data-tool="mdnice编辑器">我们首先加载数据（ https://www.dropbox.com/s/cuowvm4vrf65pvq/allen_cortex.rds?dl=1提供下载，990MB大小），预处理scRNA-seq 参考数据集，然后执行标签传输。该过程为每个spot输出每个scRNA-seq派生类的概率分类。我们将这些预测添加到Seurat对象中作为新的试验。</p><h4 data-tool="mdnice编辑器"><span></span>8.1 下载并读入单细胞数据<span></span></h4><p data-tool="mdnice编辑器">数据来源于《Adult mouse cortical cell taxonomy revealed by single cell transcriptomics》</p><pre data-tool="mdnice编辑器"><span></span><code>allen_reference &lt;- readRDS("../Rawdata/allen_cortex.rds")<br>allen_reference<br>##An object of class Seurat <br>##34617 features across 14249 samples within 1 assay <br>##Active assay: RNA (34617 features, 0 variable features)<br></code></pre><h4 data-tool="mdnice编辑器"><span></span>8.2 数据预处理<span></span></h4><p data-tool="mdnice编辑器">首先对单细胞数据进行<code> SCTransform</code>转换。注意，设置<code>ncells=3000</code>将使完整的数据集normalizes，但在3k时学习噪声模型，这大大提高了<code>SCTransform</code>的速度，并且没有性能损失：</p><pre data-tool="mdnice编辑器"><span></span><code># note that setting ncells=3000 normalizes the full dataset but learns noise models on 3k<br># cells this speeds up SCTransform dramatically with no loss in performance<br>start_time &lt;- Sys.time()<br>allen_reference &lt;- SCTransform(allen_reference, ncells = 3000, verbose = FALSE) %&gt;%<br>    RunPCA(verbose = FALSE) %&gt;%<br>    RunUMAP(dims = 1:30)<br>end_time &lt;- Sys.time()<br>end_time - start_time<br><br>allen_reference<br>##An object of class Seurat <br>##69225 features across 14249 samples within 2 assays <br>##Active assay: SCT (34608 features, 3000 variable features)<br>##1 other assay present: RNA<br>##2 dimensional reductions calculated: pca, umap<br></code></pre><p data-tool="mdnice编辑器">可视化，"subclass"是原文的注释结果：</p><pre data-tool="mdnice编辑器"><span></span><code># the annotation is stored in the 'subclass' column of object metadata<br>DimPlot(allen_reference, group.by = "subclass", label = TRUE)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.6171428571428571" data-type="other" data-w="700" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxX1NqPCysOI4TQ2vX4lRuzq0AIB4xduiarfvIyUIkjBytBA9ex8wuX4Q/640?wx_fmt=other" src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxX1NqPCysOI4TQ2vX4lRuzq0AIB4xduiarfvIyUIkjBytBA9ex8wuX4Q/640?wx_fmt=other"><figcaption>image-20211114154755832</figcaption></figure><p data-tool="mdnice编辑器">对于取子集后的空间数据，再做一次<code>SCTransform</code>转换：</p><pre data-tool="mdnice编辑器"><span></span><code># After subsetting, we renormalize cortex<br>start_time &lt;- Sys.time()<br>cortex &lt;- SCTransform(cortex, assay = "Spatial", verbose = FALSE) %&gt;%<br>    RunPCA(verbose = FALSE)<br>end_time &lt;- Sys.time()<br>end_time - start_time<br>##Time difference of 1.195518 mins<br></code></pre><h4 data-tool="mdnice编辑器"><span></span>8.3 整合空间数据和单细胞数据<span></span></h4><pre data-tool="mdnice编辑器"><span></span><code>start_time &lt;- Sys.time()<br>anchors &lt;- FindTransferAnchors(reference = allen_reference, query = cortex, normalization.method = "SCT")<br>end_time &lt;- Sys.time()<br>end_time - start_time<br>##Time difference of 1.0687 mins<br><br>predictions.assay &lt;- TransferData(anchorset = anchors, refdata = allen_reference$subclass, prediction.assay = TRUE, weight.reduction = cortex[["pca"]], dims = 1:30)<br>cortex[["predictions"]] &lt;- predictions.assay<br></code></pre><p data-tool="mdnice编辑器">现在，我们得到了每个class每个spot的预测分数。在额叶皮层区域特别有趣的是层状兴奋性神经元。在这里，我们可以区分这些神经元亚型的不同顺序层，例如:</p><pre data-tool="mdnice编辑器"><span></span><code>#DefaultAssay(cortex) &lt;- "SCT"<br>rownames(cortex) #基因名names<br><br>DefaultAssay(cortex) &lt;- "predictions"<br>rownames(cortex) #预测的单细胞clusters<br><br>SpatialFeaturePlot(cortex, features = c("L2/3 IT", "L4"), pt.size.factor = 1.6, ncol = 2, crop = TRUE)<br></code></pre><img data-ratio="0.6171428571428571" data-type="other" data-w="700" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxRqf7LCPnYVVB51RiaiazUyiaTw6sG1ziaF46dZgF4KpribcwXkRXkvyNnvg/640?wx_fmt=other" src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxRqf7LCPnYVVB51RiaiazUyiaTw6sG1ziaF46dZgF4KpribcwXkRXkvyNnvg/640?wx_fmt=other"><p data-tool="mdnice编辑器">根据这些预测分数，我们还可以预测位置受到空间限制的细胞类型。我们使用基于标记点过程的相同方法来定义空间变量特征，但使用细胞类型预测评分作为“标记”，而不是使用基因表达。</p><pre data-tool="mdnice编辑器"><span></span><code>## 利用预测的细胞cluster，寻找高变特征（类似常规的高变基因 ）<br>cortex &lt;- FindSpatiallyVariableFeatures(cortex, assay = "predictions", selection.method = "markvariogram",<br>    features = rownames(cortex), r.metric = 5, slot = "data")<br><br>DECs = SpatiallyVariableFeatures(cortex)<br>top.clusters &lt;- head(DECs, 4)<br>SpatialPlot(object = cortex, features = top.clusters, ncol = 2)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.34074074074074073" data-type="other" data-w="1080" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxpOE8ItBRuCLe4fvMm4ZxGrAjB9Jmgf756WLg2fKwMlPxXrCHkuVpHg/640?wx_fmt=other" src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxpOE8ItBRuCLe4fvMm4ZxGrAjB9Jmgf756WLg2fKwMlPxXrCHkuVpHg/640?wx_fmt=other"><figcaption>image-20211114160705058</figcaption></figure><p data-tool="mdnice编辑器">最后，作者证明Seurat的整合程序能够恢复已知的神经元和非神经元亚群的空间定位模式，包括层兴奋性亚群、第1层星形胶质细胞和皮层灰质：</p><pre data-tool="mdnice编辑器"><span></span><code>SpatialFeaturePlot(cortex, features = c("Astro", "L2/3 IT", "L4", "L5 PT", "L5 IT", "L6 CT", "L6 IT",<br>    "L6b", "Oligo"), pt.size.factor = 1, ncol = 3, crop = FALSE, alpha = c(0.1, 1))<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="1.2334183673469388" data-type="other" data-w="784" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxG3NyJcL7c3n8PPRJVkHQE7AgBibHliba64y0YdvVKJlC7Pf3Lz23ibt7g/640?wx_fmt=other" src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxG3NyJcL7c3n8PPRJVkHQE7AgBibHliba64y0YdvVKJlC7Pf3Lz23ibt7g/640?wx_fmt=other"><figcaption>image-20211114161058858</figcaption></figure><h3 data-tool="mdnice编辑器"><span></span>九. 处理Seurat中的多个slices<span></span></h3><p data-tool="mdnice编辑器">这个小鼠大脑的数据集包含了另一个与大脑另一半相对应的切片。这里我们读取它并执行相同的初始规范化。</p><pre data-tool="mdnice编辑器"><span></span><code>brain2 &lt;- LoadData("stxBrain", type = "posterior1")<br># An object of class Seurat <br># 31053 features across 3353 samples within 1 assay <br># Active assay: Spatial (31053 features, 0 variable features)<br><br>brain2 &lt;- SCTransform(brain2, assay = "Spatial", verbose = FALSE)<br>brain2<br># An object of class Seurat <br># 48721 features across 3353 samples within 2 assays <br># Active assay: SCT (17668 features, 3000 variable features)<br>#  1 other assay present: Spatial<br><br>## 这里读入的数据已经设置好了，如果没设置的话，手动设置一下<br>brain2@project.name <br>Idents(brain2)<br>brain2$orig.ident<br>##"Posterior1"<br></code></pre><p data-tool="mdnice编辑器">为了在同一个<code>Seurat</code>对象中处理多个切片，官方提供了<code>merge</code>函数：</p><pre data-tool="mdnice编辑器"><span></span><code>start_time &lt;- Sys.time()<br>	brain.merge &lt;- merge(brain, brain2)<br>end_time &lt;- Sys.time()<br>end_time - start_time<br>##Time difference of 54.28318 secs<br><br>brain.merge<br>##An object of class Seurat <br>##49293 features across 6049 samples within 2 assays <br>##Active assay: SCT (18240 features, 0 variable features)<br>##1 other assay present: Spatial<br><br>## 可以保存一下再继续<br>##save(brain.merge,file = "brain.merge_sct.Rdata")<br></code></pre><p data-tool="mdnice编辑器">这样就可以对底层的RNA表达数据进行联合降维和聚类：</p><pre data-tool="mdnice编辑器"><span></span><code>DefaultAssay(brain.merge) &lt;- "SCT"<br>VariableFeatures(brain.merge) &lt;- c(VariableFeatures(brain), VariableFeatures(brain2))<br><br>brain.merge &lt;- RunPCA(brain.merge, verbose = FALSE)<br>brain.merge &lt;- FindNeighbors(brain.merge, dims = 1:30)<br>brain.merge &lt;- FindClusters(brain.merge, verbose = FALSE)<br>brain.merge &lt;- RunUMAP(brain.merge, dims = 1:30)<br>brain.merge<br>##An object of class Seurat <br>##49293 features across 6049 samples within 2 assays <br>##Active assay: SCT (18240 features, 6000 variable features)<br>##1 other assay present: Spatial<br>##2 dimensional reductions calculated: pca, umap<br></code></pre><p data-tool="mdnice编辑器">最后，可以在单个UMAP图中共同可视化数据。<code>SpatialDimPlot</code>和<code>SpatialFeaturePlot</code>将默认将所有片绘制为列，将<code>groupings/features </code>绘制为行。</p><pre data-tool="mdnice编辑器"><span></span><code>DimPlot(brain.merge, reduction = "umap", group.by = c("ident", "orig.ident"))<br></code></pre><img data-ratio="0.5592592592592592" data-type="other" data-w="1080" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxUwBAysiaQctcCVaBWq1iaia81tN9CrXHY9Q74UlBuEDSTpBahMxOMVicgQ/640?wx_fmt=other" src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxUwBAysiaQctcCVaBWq1iaia81tN9CrXHY9Q74UlBuEDSTpBahMxOMVicgQ/640?wx_fmt=other"><p data-tool="mdnice编辑器">接下来可视化比较两个切片的分群情况，默认使用两个slice的名称分组：</p><pre data-tool="mdnice编辑器"><span></span><code>SpatialDimPlot(brain.merge)<br><br>#如果创建对象时没有对slice重命名，也可使用ident进行分组，绘图效果一样：<br>#SpatialDimPlot(brain.merge,group.by = c( "ident"))<br></code></pre><img data-ratio="0.4583333333333333" data-type="other" data-w="1080" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxx7iaibVmfmDmPdTNJtoUjNrwgPpfIS8IX29wgb4lq8vlanKt9Zib8WFcw/640?wx_fmt=other" src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxx7iaibVmfmDmPdTNJtoUjNrwgPpfIS8IX29wgb4lq8vlanKt9Zib8WFcw/640?wx_fmt=other"><pre data-tool="mdnice编辑器"><span></span><code>#比较两个切片的特定基因表达水平<br>SpatialFeaturePlot(brain.merge, features = c("Hpca", "Plp1"))<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="1.0575079872204474" data-type="other" data-w="626" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxsUT75tWzroOUic8KV3guRlWiaKafIBfB11icMGZaw0b0jIYzZyxbuySvg/640?wx_fmt=other" src="https://mmbiz.qpic.cn/mmbiz_jpg/fTW9zRI3eqUicoeczh7CPfUAF6iazC1uCxsUT75tWzroOUic8KV3guRlWiaKafIBfB11icMGZaw0b0jIYzZyxbuySvg/640?wx_fmt=other"><figcaption>image-20211114165905486</figcaption></figure><h3 data-tool="mdnice编辑器"><span></span>十. 结语<span></span></h3><p data-tool="mdnice编辑器">关于整合空间转录组与单细胞数据整合方法汇总，详见2022年Nat Methods上的一篇Benchmarking文章《Benchmarking spatial and single-cell transcriptomics integration methods for transcript distribution prediction and cell type deconvolution》（DOI: 10.1038/s41592-022-01480-9），<strong>推荐大家使用cell2location或RCTD算法对空间转录组数据进行解卷积</strong>。值得注意的是，Seurat v5空转部分的教程使用了RCTD算法作为示例文档。</p><span>- END -</span></section><p><br></p><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/X8eP58iDEx4j7pSyGbze9Q",target="_blank" rel="noopener noreferrer">原文链接</a>
