---
title: "明明PCA区分非常好，但是差异基因数量很少？"
date: 2023-05-19T04:17:00Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Acdemic"]
---
明明PCA区分非常好，但是差异基因数量很少？ by 生信菜鸟团
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><blockquote data-tool="mdnice编辑器"><p>本周我们将给大家带来一篇文章，在这篇文章中PCA的效果明明很好，但差异表达基因数量却很少，富有经验的曾老师看出了问题并认为值得深入分析，所以我们这周就来探讨这个问题。</p></blockquote><h1 data-tool="mdnice编辑器"><span></span>文章背景内容介绍</h1><p data-tool="mdnice编辑器">Multi-level transcriptome sequencingidentifies COL1A1 as a candidate marker in human heart failure progression</p><p data-tool="mdnice编辑器">https://link.springer.com/article/10.1186/s12916-019-1469-4<svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" width="22px" height="22px" viewbox="0 0 32 32" enable-background="new 0 0 32 32"><image width="32" height="32" x="0" y="0"></image></svg></p><p data-tool="mdnice编辑器">GSE135055</p><p data-tool="mdnice编辑器">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE135055</p><figure data-tool="mdnice编辑器"><img data-ratio="0.09611231101511879" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losicicr6Zujd9wicbp8sBuxa0xu9agC17ia2TXdL6dyC57hXIzjtjibf9jUlsh7NUIPISXhekp4LxcR2OTw/640?wx_fmt=png" data-type="png" data-w="926" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losicicr6Zujd9wicbp8sBuxa0xu9agC17ia2TXdL6dyC57hXIzjtjibf9jUlsh7NUIPISXhekp4LxcR2OTw/640?wx_fmt=png"></figure><blockquote data-tool="mdnice编辑器"><p>背景：</p><p>心力衰竭（HF）已被公认为全球流行病，其相关住院率、发病率和死亡率很高。尽管已经取得了许多进展，但其代表性的分子特征在很大程度上仍然未知，尤其是基因在HF进展中的作用。本前瞻性随访研究的目的是揭示与心力衰竭进展相关的潜在生物标志物。</p><p>方法：</p><p>我们收集了21名HF患者和9名健康供体的左心室心脏组织作为研究队列，并生成了多层次转录组数据。通过使用Masson染色来计算每个样本的纤维化百分比，我们应用lasso回归模型来识别与纤维化和进展相关的基因。</p><p>在同一队列中通过免疫组织化学（IHC）染色和使用另一个独立队列（20名HF和9名健康供体）的qRT-PCR进一步验证了这些基因。</p><p>酶联免疫吸附试验（ELISA）用于测量验证队列（87名HF患者）中的血浆水平，以预测HF进展。</p><p>结果：</p><p>基于多级转录组数据，我们检测了研究队列中差异表达的基因[mRNAs、microRNAs和长非编码RNA（lncRNAs）]。后续的功能注释和调节网络揭示了它们在调节细胞外基质中的潜在作用。我们进一步鉴定了几个与纤维化相关的基因。</p><p>通过使用移植前的存活时间，COL1A1被确定为HF进展的潜在生物标志物，其上调通过IHC和qRT-PCR得到证实。此外，发现血浆中COL1A1含量&gt;281.7ng/ml与心力衰竭心脏移植一年内的不良生存率有关[风险比（HR）：27.52，95%置信区间（CI）：7.474至101.3，对数秩p值&lt;1.0×10-4]。</p><p>结论：</p><p>我们的研究结果表明，COL1A1是HF的血浆生物标志物，与HF的进展有关，尤其是用于预测从HF发作到移植的1年生存期。</p></blockquote><figure data-tool="mdnice编辑器"><img data-ratio="0.8567567567567568" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losicicr6Zujd9wicbp8sBuxa0xuB7hpPPyph1LN8JcDjMqN80GpicQVbOTDgv2sBpDYFxia3h6NPuJau51g/640?wx_fmt=png" data-type="png" data-w="740" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losicicr6Zujd9wicbp8sBuxa0xuB7hpPPyph1LN8JcDjMqN80GpicQVbOTDgv2sBpDYFxia3h6NPuJau51g/640?wx_fmt=png"></figure><blockquote data-tool="mdnice编辑器"><p>As shown in Fig. 3a–c, principal component analysis (PCA) was first performed for the expression profiles of mRNAs, lncRNAs, and miRNAs. All three RNA types could largely distinguish HF patients from healthy controls.</p></blockquote><blockquote data-tool="mdnice编辑器"><p>Interestingly, lncRNA expression profiles had the best performance in distinguishing HF from healthy controls, reinforcing the importance to further investigate their underlying mechanism in HF.</p></blockquote><blockquote data-tool="mdnice编辑器"><p>We identified <strong>126 mRNAs, 16 lncRNAs, and 42 miRNAs</strong> that were differentially expressed in HF versus control samples by requiring the absolute log2-transformed fold change (FC) &gt; 1 and adjusted p value &lt; 0.05 by the BH method.</p></blockquote><h1 data-tool="mdnice编辑器"><span></span>我的思路：</h1><ul data-tool="mdnice编辑器"><li><section>上游分析的影响</section></li><li><section>差异表达分析的影响</section></li><li><section>如何评估PCA的效果？PCA效果好，差异表达基因一定多吗？</section></li></ul><hr data-tool="mdnice编辑器"><h2 data-tool="mdnice编辑器"><span></span>1.查看该实验的数据上游分析，看看注释到的转录本数量</h2><figure data-tool="mdnice编辑器"><img data-ratio="0.19722222222222222" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losicicr6Zujd9wicbp8sBuxa0xuibcbu8uyFliapWV8EFNHO1f8u7Eic1Ix9cOV9EDAf70LDaSaq6PO6HGIQ/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losicicr6Zujd9wicbp8sBuxa0xuibcbu8uyFliapWV8EFNHO1f8u7Eic1Ix9cOV9EDAf70LDaSaq6PO6HGIQ/640?wx_fmt=png"></figure><blockquote data-tool="mdnice编辑器"><p>The fragments per kilobase of transcript per million mapped reads (FPKM) value of <strong>186,363 transcripts</strong> was calculated based on StringTie (version 1.3.4) with default parameters [16]. mRNA with low expression were excluded, defined as those with FPKM less than 1 in more than 80% samples. For lncRNAs, we did not remove the low expressed transcripts, while we identified the potential long non-coding transcripts by filtering out short transcripts (default 200 nt) and removing single exon transcripts based on the FEELnc tool (version 0.11-2) [17]. The remaining transcripts were then used to perform the following analysis.</p></blockquote><blockquote data-tool="mdnice编辑器"><p>For the pre-processing of miRNA data, we removed miRNAs with a missing value in &gt; 10% of the samples. According to the miRNA sequence database miRBase [20] (Release 22), there were 2620 mature human miRNAs. <strong>Our analysis resulted in 604 expressed miRNAs in our samples.</strong></p></blockquote><p data-tool="mdnice编辑器">理解这些信息以便我们下面查看GEO提供的表达量矩阵文件</p><h2 data-tool="mdnice编辑器"><span></span>2.看看作者怎么做的PCA和差异表达分析</h2><figure data-tool="mdnice编辑器"><img data-ratio="0.364190012180268" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losicicr6Zujd9wicbp8sBuxa0xuPjXtRXGibN5oN0SGyOzzsltCPQphEicT9jjeWmkBg8zF4eWNyfB04xCg/640?wx_fmt=png" data-type="png" data-w="821" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losicicr6Zujd9wicbp8sBuxa0xuPjXtRXGibN5oN0SGyOzzsltCPQphEicT9jjeWmkBg8zF4eWNyfB04xCg/640?wx_fmt=png"></figure><figure data-tool="mdnice编辑器"><img data-ratio="0.3336734693877551" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losicicr6Zujd9wicbp8sBuxa0xubrlicOV41D62aDz5lhY7ypTsMwQnz4QZkmtQzNYTWEf4bKz5jRcjNNw/640?wx_fmt=png" data-type="png" data-w="980" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losicicr6Zujd9wicbp8sBuxa0xubrlicOV41D62aDz5lhY7ypTsMwQnz4QZkmtQzNYTWEf4bKz5jRcjNNw/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">对于mRNA，作者走的hisat2+stringtie+ballgown流程</p><blockquote data-tool="mdnice编辑器"><p>参考资料：</p><p><a href="https://mp.weixin.qq.com/s?__biz=MzUzMTEwODk0Ng==&amp;mid=2247512469&amp;idx=1&amp;sn=364469f90477204a6319cd57211e8264&amp;chksm=fa4574a8cd32fdbe68badc2b6c922a164ce3be815877c5e4fed59628c72fd590f0a140752831&amp;scene=21&amp;cur_album_id=2545418429466607617#wechat_redirect" data-linktype="2">使用TPM/FPKM/RPKM进行差异分析真的可以消除系统误差吗？</a></p><p>为什么很少有人用hisat2+stringtie+ballgown发转录组文章？https://www.zhihu.com/question/507628729/answer/2306597069</p><p>学徒作业-hisat2+stringtie+ballgown流程http://www.bio-info-trainee.com/7443.html</p><p><a href="https://mp.weixin.qq.com/mp/appmsgalbum?action=getalbum&amp;scene=1&amp;__biz=MzAxMDkxODM1Ng==&amp;album_id=1322384987060142080#wechat_redirect" data-linktype="2">lncRNA入门到精通</a></p></blockquote><figure data-tool="mdnice编辑器"><img data-ratio="0.8567567567567568" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losicicr6Zujd9wicbp8sBuxa0xuB7hpPPyph1LN8JcDjMqN80GpicQVbOTDgv2sBpDYFxia3h6NPuJau51g/640?wx_fmt=png" data-type="png" data-w="740" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losicicr6Zujd9wicbp8sBuxa0xuB7hpPPyph1LN8JcDjMqN80GpicQVbOTDgv2sBpDYFxia3h6NPuJau51g/640?wx_fmt=png"></figure><blockquote data-tool="mdnice编辑器"><p>We identified <strong>126 mRNAs, 16 lncRNAs, and 42 miRNAs</strong> that were differentially expressed in HF versus control samples by requiring the absolute log2-transformed fold change (FC) &gt; 1 and adjusted p value &lt; 0.05 by the BH method.</p></blockquote><p data-tool="mdnice编辑器"><strong>需要注意的是作者这里是对RNA而非基因进行的差异表达分析</strong></p><p data-tool="mdnice编辑器">下载GEO提供的表达矩阵看看</p><figure data-tool="mdnice编辑器"><img data-ratio="0.1675925925925926" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losicicr6Zujd9wicbp8sBuxa0xusMw9b1TFlPibofrNrUazp0G3g0sjqicYO7zUricFic56fABAWZ0RmJtXMA/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losicicr6Zujd9wicbp8sBuxa0xusMw9b1TFlPibofrNrUazp0G3g0sjqicYO7zUricFic56fABAWZ0RmJtXMA/640?wx_fmt=png"></figure><figure data-tool="mdnice编辑器"><img data-ratio="0.37894736842105264" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losicicr6Zujd9wicbp8sBuxa0xuUZuEW9WCS9y8iaqEUmkUbueFkKB0TyMn46XLiaLRPoNwECibcwkgN39XA/640?wx_fmt=png" data-type="png" data-w="95" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losicicr6Zujd9wicbp8sBuxa0xuUZuEW9WCS9y8iaqEUmkUbueFkKB0TyMn46XLiaLRPoNwECibcwkgN39XA/640?wx_fmt=png"></figure><figure data-tool="mdnice编辑器"><img data-ratio="0.6148148148148148" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losicicr6Zujd9wicbp8sBuxa0xuhqwKxZ5Odnn8lBCTsDjicLhSX0U7bW3qzt7f8e05NRU6wBBXVMQawAQ/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losicicr6Zujd9wicbp8sBuxa0xuhqwKxZ5Odnn8lBCTsDjicLhSX0U7bW3qzt7f8e05NRU6wBBXVMQawAQ/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">根据第一部分我们得到的上游分析信息，可以发现，两个矩阵都是原始矩阵，没有进行过滤和筛选</p><blockquote data-tool="mdnice编辑器"><p>The fragments per kilobase of transcript per million mapped reads (FPKM) value of <strong>186,363 transcripts</strong> was calculated based on StringTie (version 1.3.4) with default parameters [16]. <strong>mRNA with low expression were excluded, defined as those with FPKM less than 1 in more than 80% samples. For lncRNAs, we did not remove the low expressed transcripts, while we identified the potential long non-coding transcripts by filtering out short transcripts (default 200 nt) and removing single exon transcripts based on the FEELnc tool (version 0.11-2)</strong> [17]. The remaining transcripts were then used to perform the following analysis.</p></blockquote><blockquote data-tool="mdnice编辑器"><p>For the pre-processing of miRNA data, <strong>we removed miRNAs with a missing value in &gt; 10% of the samples.</strong> According to the miRNA sequence database miRBase [20] (Release 22), there were 2620 mature human miRNAs. Our analysis resulted in 604 expressed miRNAs in our samples.</p></blockquote><p data-tool="mdnice编辑器">关于如何解读stringtie定量得到的结果文件，参考资料：</p><blockquote data-tool="mdnice编辑器"><p>关于stringtie定量基因的时候，最后输出很多MSTRG样式的geneidhttps://www.jianshu.com/p/1697f2d91eb5</p><p>MSTRG Tag, what does it refer to?https://github.com/gpertea/stringtie/issues/95</p><p>How to deal with MSTRG tag without relevant gene name?https://www.biostars.org/p/282817/</p></blockquote><p data-tool="mdnice编辑器">其实如果不找新基因/新转录本没必要用这套流程，可以直接使用我们之前介绍的hisat2+featureCount+DESeq2流程</p><p data-tool="mdnice编辑器">我们这里使用AnnoProbe包根据gene_name得到RNA的分类信息，如果不是找新基因或新转录本就不需要再看没有gene_name的RNA</p><h3 data-tool="mdnice编辑器"><span></span>①对获取的fpkm矩阵使用limma进行差异表达分析：<span></span></h3><pre data-tool="mdnice编辑器"><span></span><code>rm(list=ls())<br>data &lt;- read.table(<span>"GSE135055_FPKM_Expression_Matrix.txt"</span>,sep = <span>'\t'</span>,header = <span>TRUE</span>)<br><span>library</span>(tidyverse)<br>data &lt;- tibble(data)<br><br><span>library</span>(AnnoProbe)<br>ids=annoGene(data$gene_name,<span>'SYMBOL'</span>,<span>'human'</span>)<br>head(ids)<br>as.data.frame(tail(sort(table(ids$biotypes))))<br><span># Var1  Freq</span><br><span># 1             unprocessed_pseudogene   408</span><br><span># 2 transcribed_unprocessed_pseudogene   432</span><br><span># 3                              miRNA  1855</span><br><span># 4                             lncRNA  2239</span><br><span># 5               processed_pseudogene  2687</span><br><span># 6                     protein_coding 18513</span><br>mRNA_gene_name &lt;- ids$SYMBOL[ids$biotypes == <span>"protein_coding"</span>]<br><br>fpkm &lt;- data[,c(<span>11</span>:<span>40</span>)]<br><br><span># 将FPKM值转换为TPM值</span><br>expMatrix &lt;- fpkm<br>fpkmToTpm &lt;- <span>function</span>(fpkm)<br>{<br>  exp(log(fpkm) - log(sum(fpkm)) + log(<span>1e6</span>))<br>}<br>tpms &lt;- apply(expMatrix,<span>2</span>,fpkmToTpm) <span>#每列的基因</span><br>tpms[<span>1</span>:<span>3</span>,]<br>colSums(tpms) <span>#都是每一百万（深度一样）</span><br><br><span># 筛选出非低表达mRNA</span><br><span># mRNA</span><br>index_mRNA &lt;- data$gene_name %<span>in</span>% mRNA_gene_name<br>table(index_mRNA)<br><span># FALSE   TRUE </span><br><span># 64855 121508</span><br><span># 非低表达</span><br><span># mRNA with low expression were excluded, defined as those with FPKM less than 1 in more than 80% samples.</span><br>keep &lt;- rowSums(fpkm&gt;<span>1</span>) &gt;= <span>24</span>  <span># 30 * 0.8</span><br>table(keep)<br><span># FALSE   TRUE </span><br><span># 175091  11272 </span><br><span># 可以发现这一步过滤掉了很多低表达mRNA</span><br>keep_final &lt;- keep &amp; index_mRNA<br>table(keep_final)<br><span># FALSE   TRUE </span><br><span># 177092   9271 </span><br><span># 只剩下了近1w转录本</span><br><br>t_name_mRNA &lt;- data$t_name[keep_final]<br>tpms_mRNA &lt;- tpms[keep_final,]<br>rownames(tpms_mRNA) &lt;- t_name_mRNA<br>head(tpms_mRNA)<br><br><span># 差异分析</span><br>group_list=c(rep(<span>'HF'</span>,<span>21</span>),rep(<span>'Healthy'</span>,<span>9</span>))<br><span>## 强制限定顺序</span><br>group_list &lt;- factor(group_list,levels = c(<span>"HF"</span>,<span>"Healthy"</span>),ordered = <span>F</span>)<br><span>#表达矩阵数据校正</span><br>table(group_list)<br>exprSet &lt;- tpms_mRNA<br>boxplot(exprSet,outline=<span>FALSE</span>, notch=<span>T</span>,col=group_list, las=<span>2</span>)<br><br><span>library</span>(limma) <br>exprSet=normalizeBetweenArrays(exprSet)<br>boxplot(exprSet,outline=<span>FALSE</span>, notch=<span>T</span>,col=group_list, las=<span>2</span>)<br><br><span>#判断数据是否需要转换</span><br>range(exprSet)<br>exprSet &lt;- log2(exprSet+<span>1</span>)<br>range(exprSet)<br><span>#差异分析：</span><br>dat &lt;- exprSet<br>design=model.matrix(~factor( group_list ))<br>fit=lmFit(dat,design)<br>fit=eBayes(fit)<br>options(digits = <span>4</span>)<br><span># 获取差异表达基因</span><br>deg=topTable(fit,coef=<span>2</span>,adjust=<span>'BH'</span>,number = <span>Inf</span>)<br>head(deg) <br><br><span># 筛选</span><br><span># We identified</span><br><span># 126 mRNAs, 16 lncRNAs, and 42 miRNAs that were differentially expressed in HF versus control samples </span><br><span># by requiring the absolute log2-transformed fold change (FC)&gt; 1 and adjusted p value &lt; 0.05 by the BH method.</span><br><br><span>if</span>(<span>T</span>){<br>  logFC_t=<span>1</span><br>  deg$g=ifelse(deg$P.Value&gt;<span>0.05</span>,<span>'stable'</span>,<br>               ifelse( deg$logFC &gt; logFC_t,<span>'UP'</span>,<br>                       ifelse( deg$logFC &lt; -logFC_t,<span>'DOWN'</span>,<span>'stable'</span>) )<br>  )<br>  table(deg$g)<br>  head(deg)<br>}<br><br>table(deg$g)<br><span># DOWN stable     UP </span><br><span># 116   9108     47</span><br><span># 116+47=163</span><br><span># 作者鉴定出126个差异表达mRNA</span><br></code></pre><p data-tool="mdnice编辑器">我们通过limma对fpkm矩阵按照作者的筛选选标准进行差异表达分析得到的差异基因数量和作者大致一致，所以认为差异基因数量很少主要是因为作者使用转录水平mRNA表达量fpkm矩阵，并进行了严格的低表达过滤,而非进行差异表达分析使用的方法</p><p data-tool="mdnice编辑器">可以发现在过滤低表达mRNA中大部分转录本都被过滤掉了</p><pre data-tool="mdnice编辑器"><span></span><code><span># mRNA with low expression were excluded, defined as those with FPKM less than 1 in more than 80% samples.</span><br>keep &lt;- rowSums(fpkm&gt;<span>1</span>) &gt;= <span>24</span>  <span># 30 * 0.8</span><br></code></pre><p data-tool="mdnice编辑器">在转录组专辑中我们之前使用raw counts进行差异分析，过滤低表达基因一般标准是这样的：</p><pre data-tool="mdnice编辑器"><span></span><code>keep &lt;- rowSums(rawcount&gt;<span>0</span>) &gt;= floor(<span>0.5</span>*ncol(rawcount)) <br></code></pre><h3 data-tool="mdnice编辑器"><span></span>②同时，stringtie转录组定量是如何将转录本表达水平映射到基因上的也值得深入探究，因为作者是对转录本进行的差异表达分析而非基因<span></span></h3><blockquote data-tool="mdnice编辑器"><p>参考资料：</p><p><a href="https://mp.weixin.qq.com/s?__biz=MzUzMTEwODk0Ng==&amp;mid=2247512469&amp;idx=1&amp;sn=364469f90477204a6319cd57211e8264&amp;chksm=fa4574a8cd32fdbe68badc2b6c922a164ce3be815877c5e4fed59628c72fd590f0a140752831&amp;scene=21&amp;cur_album_id=2545418429466607617#wechat_redirect" data-linktype="2">使用TPM/FPKM/RPKM进行差异分析真的可以消除系统误差吗？</a></p><p>stringTie:转录本组装和定量工具</p><p>https://blog.csdn.net/weixin_43569478/article/details/108079257</p><p>本质上，stringTie只提供了转录本水平的表达量，定量方式包括TPM和FPKM值两种。为了进行raw count的定量方式，官方提供了 <code>prepED.py</code>脚本，可以计算出raw count的表达量</p><p>https://github.com/gpertea/stringtie/blob/master/prepDE.py3</p></blockquote><p data-tool="mdnice编辑器">下面是我使用谷歌检索“How does stringtie transcriptome quantification map transcript expression levels to genes”chatGPT插件返回的答案：</p><blockquote data-tool="mdnice编辑器"><p>StringTie is a software tool used to assemble and quantify transcriptomes from RNA-seq data. It uses a reference genome to map RNA-seq reads and identify transcripts. The transcript assembly process involves grouping and constructing transcripts using a graph-based approach. StringTie also quantifies the abundance of each transcript using read coverage information.To map transcript expression levels to genes, StringTie first assigns each transcript to a gene based on genomic location and strand orientation. It then calculates the abundance of transcripts for each gene by summing up the abundance of all transcripts assigned to that gene. <strong>The gene expression level is then obtained by summing up the transcript expression levels across all genes.</strong>StringTie also provides the option to merge transcriptomes from multiple samples, which can improve gene annotation and provide more accurate quantification of gene expression levels. Overall, StringTie provides a reliable and efficient method for transcriptome assembly and quantification.</p></blockquote><p data-tool="mdnice编辑器">根据<a href="https://mp.weixin.qq.com/s?__biz=MzUzMTEwODk0Ng==&amp;mid=2247512469&amp;idx=1&amp;sn=364469f90477204a6319cd57211e8264&amp;chksm=fa4574a8cd32fdbe68badc2b6c922a164ce3be815877c5e4fed59628c72fd590f0a140752831&amp;scene=21&amp;cur_album_id=2545418429466607617#wechat_redirect" data-linktype="2">使用TPM/FPKM/RPKM进行差异分析真的可以消除系统误差吗？</a>得到的基因表达矩阵和转录本表达矩阵，我们对上面的对基因对应转录本表达量直接加和的说法进行验证：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.18425925925925926" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losicicr6Zujd9wicbp8sBuxa0xuK0cyicGk0gd8icrBib4Saf6tyyPGeDbxboxwdj1I9yocdKGePBDr0ib2EQ/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losicicr6Zujd9wicbp8sBuxa0xuK0cyicGk0gd8icrBib4Saf6tyyPGeDbxboxwdj1I9yocdKGePBDr0ib2EQ/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">可以发现确实是直接进行加和</p><p data-tool="mdnice编辑器">那么作者是对转录本进行差异表达分析的而非我们推文中常见的基因表达量，这肯定也会给差异表达分析结果数量特征带来影响</p><p data-tool="mdnice编辑器">tips:</p><ul data-tool="mdnice编辑器"><li><section>感兴趣的同学可以走一遍作者的hisat2+stringtie+ballgown流程，获取基因水平的表达矩阵进行差异分析，看看和转录本差异表达分析结果有什么区别</section></li><li><section>不认为不同流程进行差异分析对差异表达基因数量有特别大的影响（我们这里limma差异表达基因数量也和作者用ballgown差不多），感兴趣的同学可以获取sra/fastq自己走一遍hisat2+featureCount+DESeq2的流程</section></li></ul><h4 data-tool="mdnice编辑器"><span></span>那么转录本表达量差异分析和我们常见的基因有什么区别呢？<span></span></h4><p data-tool="mdnice编辑器">Differential gene and transcript expression analysis of RNA-seq experiments with TopHat and Cufflinks</p><p data-tool="mdnice编辑器">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3334321/</p><figure data-tool="mdnice编辑器"><img data-ratio="0.569351230425056" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losicicr6Zujd9wicbp8sBuxa0xu3U8lRRvnCEa8YxJysVicucsc5wL3yJWR3FU7V2bngj6590p4vyDdMEg/640?wx_fmt=png" data-type="png" data-w="894" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losicicr6Zujd9wicbp8sBuxa0xu3U8lRRvnCEa8YxJysVicucsc5wL3yJWR3FU7V2bngj6590p4vyDdMEg/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">这篇文章虽然作者用的不是stringtie但其从同一数据集获得的差异表达结果中转录本确实更少</p><p data-tool="mdnice编辑器">Gene-level differential analysis at transcript-level resolution</p><p data-tool="mdnice编辑器">https://genomebiology.biomedcentral.com/articles/10.1186/s13059-018-1419-z</p><blockquote data-tool="mdnice编辑器"><p>Abstract:</p><p><strong>与RNA测序转录本差异分析相比，基因水平的差异表达分析更稳健，并且在实验上更可行。然而，使用基因计数进行统计分析可以掩盖转录物水平的动态。</strong>我们证明了“分析第一，聚合第二”，即将转录物分析得出的p值聚合以获得基因水平的结果，提高了灵敏度和准确性。我们提出的方法也可以应用于从读数的伪比对中获得的转录物兼容性计数，这避免了量化的需要，并且快速、准确且无模型。该方法推广到生物学的各个层面，我们展示了它在基因本体论中的应用。</p></blockquote><h2 data-tool="mdnice编辑器"><span></span>3.如何评估PCA的效果，PCA效果好差异表达基因一定多吗</h2><blockquote data-tool="mdnice编辑器"><p>As shown in Fig. 3a–c, principal component analysis (PCA) was first performed for the expression profiles of mRNAs, lncRNAs, and miRNAs. All three RNA types could largely distinguish HF patients from healthy controls.</p></blockquote><blockquote data-tool="mdnice编辑器"><p>Interestingly, lncRNA expression profiles had the best performance in distinguishing HF from healthy controls, reinforcing the importance to further investigate their underlying mechanism in HF.</p></blockquote><p data-tool="mdnice编辑器">在这里小编提出疑问：①作者为什么认为三种RNA类型都可以很好区分实验分组，有没有统计量可以评估PCA的效果？，②为什么认为“lncRNA表达谱的性能最佳”，mRNA、lncRNA、miRNA进行PCA时，PC1和PC2可解释的方差并不相同，也可以进行比较吗？</p><figure data-tool="mdnice编辑器"><img data-ratio="0.8567567567567568" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losicicr6Zujd9wicbp8sBuxa0xuB7hpPPyph1LN8JcDjMqN80GpicQVbOTDgv2sBpDYFxia3h6NPuJau51g/640?wx_fmt=png" data-type="png" data-w="740" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losicicr6Zujd9wicbp8sBuxa0xuB7hpPPyph1LN8JcDjMqN80GpicQVbOTDgv2sBpDYFxia3h6NPuJau51g/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">就这个问题，我咨询了曾老师，这是曾老师的回答：</p><blockquote data-tool="mdnice编辑器"><p>Q：还有一个问题曾老师，我以前看PCA效果都是直接大致看个图比如有没有重叠，有什么统计量可以评估PCA效果呢？作者为什么认为“lncRNA表达谱的性能最佳”，mRNA、lncRNA、miRNA进行PCA时，PC1和PC2可解释的方差并不相同，也可以进行比较吗？</p><p>A：从这里看，好像有两个维度可以比较，第一个维度PC1是最重要的，根据标注的解释度来看miRNA分组是最大的，第二个维度就是两个分组的距离，作者在这里可能只说明第二个维度，也就是两个分组的距离，可能是想说分组的PCA图没有交集，两个圈隔得很远。</p><p>不过你说的很对，好像没有什么统计量可以评估PCA，完全就是肉眼在看，（判断分组意义）HC Horizontal clustering 层次聚类可能好一点，或者其他度量方法。左边两个PCA可能确实解释度一般，mRNA两个分组差异体现在斜对角也比较奇怪。</p><p>我们一般来说就是看<strong>Within-group variance of the samples is smaller than the between-group variance in study.</strong></p></blockquote><p data-tool="mdnice编辑器">根据曾老师的提示和交流群小伙伴的帮助下，我得知了一种新的判断分组是否有意义的方法：ANOSIM，放在这里抛砖引玉，希望有想法的小伙伴可以在评论区积极讨论</p><p data-tool="mdnice编辑器">考虑到这里作者对stringtie定量获得的lncRNA还使用了其他软件进行鉴定、分析，我们在这里仅使用mRNA表达矩阵进行演示</p><pre data-tool="mdnice编辑器"><span></span><code><span># PCA</span><br>rm(list = ls())<br><span>library</span>(FactoMineR)<br><span>library</span>(factoextra)<br>load(<span>"mRNA.Rdata"</span>)<br>dat &lt;- tpms_mRNA<br>dat &lt;- as.data.frame(t(dat))<br>head(dat)<br>dat$group_list &lt;- group_list<br>dat_pca &lt;- PCA(dat[,-ncol(dat)], graph = <span>FALSE</span>)<span>#画图仅需数值型数据，去掉最后一列的分组信息</span><br>p &lt;- fviz_pca_ind(dat_pca,<br>                     geom.ind = <span>"point"</span>, <span># 只显示点，不显示文字</span><br>                     col.ind = dat$group_list, <span># 用不同颜色表示分组</span><br>                     palette = c(<span>"#00AFBB"</span>, <span>"#E7B800"</span>),<br>                     addEllipses = <span>T</span>, <span># 是否圈起来，少于4个样圈不起来</span><br>                     legend.title = <span>"Groups"</span>) + theme_bw()<br>p<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7127272727272728" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losicicr6Zujd9wicbp8sBuxa0xupNhE2VanPWcHdsvorR616ibP27ZYqjhMbcMichN305ZkrPX1O6IZulkw/640?wx_fmt=png" data-type="png" data-w="550" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losicicr6Zujd9wicbp8sBuxa0xupNhE2VanPWcHdsvorR616ibP27ZYqjhMbcMichN305ZkrPX1O6IZulkw/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">差异分析——ANOSIM相似性分析</p><p data-tool="mdnice编辑器">https://mp.weixin.qq.com/s/KMtmN5Uq-8G0ym_ZzeA6bA</p><blockquote data-tool="mdnice编辑器"><p>ANOSIM分析（Analysis of Similarities）即相似性分析，主要用于分析高维数据组间相似性，为数据间差异显著性评价提供依据。<strong>在一些高维数据分析中，需要使用PCA、PCoA、NMDS等方法进行降维，但这些方法并不显示组间差异的显著性指标，因此可以使用ANOSIM分析解决此问题。</strong></p><p>ANOSIM相似性分析是一种非参数检验，用来检验组间（两组或多组）差异是否显著大于组内差异，从而判断分组是否有意义。首先利用Bray-Curtis算法计算两两样品间的距离，然后将所有距离从小到大进行排序，并计算R和P值。R值用于不同组间属否存在差异，P值用于说明是否存在显著差异</p><p><strong>ANOSIM was used to compare within- and between-group similarity through a distance measure</strong>, to test the <strong>null hypothesis that the average rank similarity between samples within a group is the same as the average rank similarity between samples belonging to different groups.</strong></p></blockquote><p data-tool="mdnice编辑器">组间差异分析：Anosim</p><p data-tool="mdnice编辑器">https://zhuanlan.zhihu.com/p/111985434</p><blockquote data-tool="mdnice编辑器"><p>Anosim分析（Analysis of similarities）是一种基于置换检验和秩和检验的非参数检验方法，用来检验组间的差异是否显著大于组内差异，从而判断分组是否有意义</p></blockquote><blockquote data-tool="mdnice编辑器"><p>假如R&gt;0，说明组内距离小于组间距离，也即分组是有效的，这与方差分析中比较组内方差与组间方差来判断的原理是类似的。由上面分析结果可以看到R=0.4613，大于零模型99%分位数0.290，因此p值为0.001，结果是显著的。</p></blockquote><pre data-tool="mdnice编辑器"><span></span><code><span># ANOSIM</span><br><span>library</span>(vegan)<br>anosim=anosim(dat[,-ncol(dat)], group_list, permutations=<span>999</span>)<br>summary(anosim)<br><span># Call:</span><br><span>#   anosim(x = dat[, -ncol(dat)], grouping = group_list, permutations = 999) </span><br><span># Dissimilarity: bray </span><br><span># </span><br><span># ANOSIM statistic R: 0.4647 </span><br><span># Significance: 0.001 </span><br><span># </span><br><span># Permutation: free</span><br><span># Number of permutations: 999</span><br><span># </span><br><span># Upper quantiles of permutations (null model):</span><br><span>#   90%   95% 97.5%   99% </span><br><span>#   0.113 0.152 0.183 0.212 </span><br><span># </span><br><span># Dissimilarity ranks between and within classes:</span><br><span>#   0%    25%   50%    75% 100%   N</span><br><span># Between  9 200.00 291.0 369.00  435 189</span><br><span># HF       3  85.25 170.5 285.75  425 210</span><br><span># Healthy  1  32.75  73.5 160.00  370  36</span><br></code></pre><p data-tool="mdnice编辑器">这里的permutations=999指的是进行随机置换permutations次，用于计算统计显著性。在Anosim分析中，会将各样品间的距离进行排列组合得到许多可能性，然后比较实际组间差异和随机排列组合的差异，通过计算R值来判断组间差异是否显著大于随机差异，进而确定该分组是否有意义。permutations的值越大，计算精度越高，但计算时间也会相应增加。通常permutations的值设置为999或者9999可以满足分析需要。</p><h1 data-tool="mdnice编辑器"><span></span>总结</h1><p data-tool="mdnice编辑器">造成该研究出现PCA效果好但差异表达基因数量较少现象的原因：</p><ul data-tool="mdnice编辑器"><li><section>作者严格的低表达量转录本的过滤</section></li><li><section>作者对转录本进行差异分析而非我们常见的基因（stringtie将转录本表达量映射到基因采取的是直接加和）</section></li></ul><p data-tool="mdnice编辑器">同时，我们在这里提出疑问：①作者为什么认为三种RNA类型都可以很好区分实验分组，有没有统计量可以评估PCA的效果？，②为什么认为“lncRNA表达谱的性能最佳”，mRNA、lncRNA、miRNA进行PCA时，PC1和PC2可解释的方差并不相同，也可以进行比较吗？</p><p data-tool="mdnice编辑器">附上了曾老师的见解并提供了一种新的判断分组意义的方法ANOSIM以供参考，希望能起到抛砖引玉的效果，有想法的小伙伴可以在评论区进行探讨</p><p data-tool="mdnice编辑器"><br></p><p data-tool="mdnice编辑器" data-style='margin-bottom: 0px; padding-top: 8px; padding-bottom: 8px; outline: 0px; color: rgb(0, 0, 0); font-family: Optima-Regular, Optima, PingFangSC-light, PingFangTC-light, "PingFang SC", Cambria, Cochin, Georgia, Times, "Times New Roman", serif; font-size: 16px; text-align: left; white-space: normal; letter-spacing: 0.544px; background-color: rgb(255, 255, 255); line-height: 26px; visibility: visible;'><strong>写在文末</strong></p><p data-tool="mdnice编辑器" data-style='margin-bottom: 0px; padding-top: 8px; padding-bottom: 8px; outline: 0px; color: rgb(0, 0, 0); font-family: Optima-Regular, Optima, PingFangSC-light, PingFangTC-light, "PingFang SC", Cambria, Cochin, Georgia, Times, "Times New Roman", serif; font-size: 16px; text-align: left; white-space: normal; letter-spacing: 0.544px; background-color: rgb(255, 255, 255); line-height: 26px; visibility: visible;'>强烈建议你推荐给身边的<strong>博士后以及年轻生物学PI</strong>，多一点数据认知，让他们的科研上一个台阶：</p><ul data-tool="mdnice编辑器" data-style='margin-top: 8px; margin-bottom: 8px; padding-left: 25px; outline: 0px; color: rgb(0, 0, 0); font-family: Optima-Regular, Optima, PingFangSC-light, PingFangTC-light, "PingFang SC", Cambria, Cochin, Georgia, Times, "Times New Roman", serif; font-size: 16px; text-align: left; white-space: normal; letter-spacing: 0.544px; background-color: rgb(255, 255, 255); visibility: visible;'><li><section data-style="margin-top: 5px; margin-bottom: 5px; outline: 0px; line-height: 26px; color: rgb(1, 1, 1); visibility: visible;"><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247521372&amp;idx=1&amp;sn=373dcb6bd096e42961254c96480c0676&amp;chksm=9b4bd4e7ac3c5df17b9d27667a7ce8daebca8d58776c7f7d914b6c4d5c524eaa5450c11fc458&amp;scene=21#wechat_redirect" textvalue="生物信息学马拉松授课（买‍一得五）" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2" hasload="1">生物信息学马拉松授课（买一得五）</a> ，你的生物信息学入门课（5月8号已开课，可以期待一下下个月的~）</section></li><li><section data-style="margin-top: 5px; margin-bottom: 5px; outline: 0px; line-height: 26px; color: rgb(1, 1, 1); visibility: visible;"><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247519765&amp;idx=2&amp;sn=6cb33654c7751f4c3df0f84743f77aaf&amp;chksm=9b4bceaeac3c47b8899afc00077b96357b87a4ed6b75e7c434ba14071fd6c8448e4c218de5e0&amp;scene=21#wechat_redirect" textvalue="144线程640Gb内存服务器共享一年仍然是仅需800" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2" hasload="1">144线程640Gb内存服务器共享一年仍然是仅需800</a></section></li><li><section data-style="margin-top: 5px; margin-bottom: 5px; outline: 0px; line-height: 26px; color: rgb(1, 1, 1); visibility: visible;"><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247519765&amp;idx=1&amp;sn=ce5a8c8182f854c88043059f8c2cb9ff&amp;chksm=9b4bceaeac3c47b88c19941d43dbb1401f3a92206481a0afc41159927868199643f795d62a7e&amp;scene=21#wechat_redirect" textvalue="千呼万唤始出来的独享生物信息学云服务器" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2" hasload="1">千呼万唤始出来的独享生物信息学云服务器</a></section></li></ul><p data-style='margin-bottom: 0px; outline: 0px; font-size: 16px; text-align: left; white-space: normal; letter-spacing: 0.544px; font-family: system-ui, -apple-system, "system-ui", "Helvetica Neue", "PingFang SC", "Hiragino Sans GB", "Microsoft YaHei UI", "Microsoft YaHei", Arial, sans-serif; color: rgb(34, 34, 34); background-color: rgb(255, 255, 255); visibility: visible;'><span>再怎么强调生物信息学数据处理的计算机基础都不为过，<span data-style='outline: 0px; color: rgb(0, 0, 0); font-family: Optima-Regular, Optima, PingFangSC-light, PingFangTC-light, "PingFang SC", Cambria, Cochin, Georgia, Times, "Times New Roman", serif; visibility: visible;'>我把它粗略的分成</span><strong data-style='outline: 0px; color: rgb(0, 0, 0); font-family: Optima-Regular, Optima, PingFangSC-light, PingFangTC-light, "PingFang SC", Cambria, Cochin, Georgia, Times, "Times New Roman", serif; font-size: 16px; visibility: visible;'>基于R语言的统计可视化，以及基于Linux的NGS数据处理</strong><span data-style='outline: 0px; color: rgb(0, 0, 0); font-family: Optima-Regular, Optima, PingFangSC-light, PingFangTC-light, "PingFang SC", Cambria, Cochin, Georgia, Times, "Times New Roman", serif; visibility: visible;'>：</span></span></p><ul data-tool="mdnice编辑器" data-style='margin-top: 8px; margin-bottom: 8px; padding-left: 25px; outline: 0px; color: rgb(0, 0, 0); font-family: Optima-Regular, Optima, PingFangSC-light, PingFangTC-light, "PingFang SC", Cambria, Cochin, Georgia, Times, "Times New Roman", serif; font-size: 16px; text-align: left; white-space: normal; letter-spacing: 0.544px; background-color: rgb(255, 255, 255); visibility: visible;'><li><section data-style="margin-top: 5px; margin-bottom: 5px; outline: 0px; line-height: 26px; color: rgb(1, 1, 1); visibility: visible;">《<a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247491094&amp;idx=1&amp;sn=3a8ececefdf5894f4ef98b0823f204b8&amp;scene=21#wechat_redirect" data-linktype="2">生信分析人员如何系统入门R(2019更新版)</a>》</section></li><li><section data-style="margin-top: 5px; margin-bottom: 5px; outline: 0px; line-height: 26px; color: rgb(1, 1, 1); visibility: visible;">《<a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247491112&amp;idx=1&amp;sn=5c23959b4f36ba83532debdacf3eaa45&amp;scene=21#wechat_redirect" data-linktype="2">生信分析人员如何系统入门Linux(2019更新版)</a>》</section></li></ul></section><p><br></p><p><mp-style-type data-value="10000"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/9DjEipFIFKZspx7zlwTM3w",target="_blank" rel="noopener noreferrer">原文链接</a>
