---
title: "使用R而不是igv软件来展示bw文件"
date: 2023-03-21T22:25:54Z
draft: ["false"]
tags: [
  "fetched",
  "东林的扯淡小屋"
]
categories: ["Acdemic"]
---
使用R而不是igv软件来展示bw文件 by 东林的扯淡小屋
------
<div><p><img data-galleryid="" data-ratio="0.7361111111111112" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBqQLwLT5z5wrcZ5hwOOHet59YX7zD2HMSTLEak5AMBSC0rBGiaOyjiaSg0TVwHGxAMia941ib5NWGBeLg/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBqQLwLT5z5wrcZ5hwOOHet59YX7zD2HMSTLEak5AMBSC0rBGiaOyjiaSg0TVwHGxAMia941ib5NWGBeLg/640?wx_fmt=png"></p><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="php"><code><span><span>###PLotEpiTrackByR</span></span></code><code><span><span>###Author:wlt</span></span></code><code><span><span>###Date:20211025</span></span></code><code><span><span>###Replicate Figure 1b. in nature 2016 The landscape of accessible chromatin in</span></span></code><code><span><span>###mammalian preimplantation embryos</span></span></code><code><span><span>###Using GEO data GSE66581 for demonstration</span></span></code><code><span><span>###Extra: using mESC from GSM1014187</span></span></code><code><span><br></span></code><code><span><br></span></code><code><span><span>###----0. Basic configuration----</span></span></code><code><span>library(ggplot2)</span></code><code><span>library(patchwork)</span></code><code><span>library(cowplot)</span></code><code><span>library(tidyverse)</span></code><code><span>library(GEOquery)</span></code><code><span>library(GenomicRanges)</span></code><code><span>library(GenomicFeatures)</span></code><code><span>library(org.Mm.eg.db)</span></code><code><span>library(TxDb.Mmusculus.UCSC.mm10.knownGene)</span></code><code><span>library(clusterProfiler)</span></code><code><span>library(future.apply)</span></code><code><span>library(rtracklayer)</span></code><code><span>library(GENOVA)</span></code><code><span>library(sigminer)</span></code><code><span>library(RColorBrewer)</span></code><code><span><br></span></code><code><span><br></span></code><code><span>setwd(<span>"/data2/yudonglin/code/PlotEpiTrackByR-main"</span>)</span></code><code><span>source(<span>"myfunction_lib.R"</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span>####define colorbar</span></span></code><code><span>solarExtra &lt;- colorRampPalette(c(<span>"#3361A5"</span>,<span>"#248AF3"</span>,<span>"#14B3FF"</span>,<span>"#88CEEF"</span>,<span>"#C1D5DC"</span>,<span>"#EAD397"</span>,<span>"#FDB31A"</span>,<span>"#E42A2A"</span>,<span>"#A31D1D"</span>))</span></code><code><span>hic.rdbu &lt;- colorRampPalette(c( <span>"#0065b2"</span>,<span>"#2399de"</span>,<span>"#67cdfe"</span>, <span>"#f5f5f5"</span>,<span>"#fcb09d"</span>,</span></code><code><span><span>"#ed6855"</span>,<span>"#be2a21"</span>))</span></code><code><span>hic.sadle.rdbu &lt;- colorRampPalette(colors = c(<span>"#1f67d1"</span>,<span>"#5299f2"</span>,<span>"#69b2fd"</span>,</span></code><code><span><span>"lightgrey"</span>,</span></code><code><span><span>"#ffc3aa"</span>,<span>"#ff876b"</span>,<span>"red1"</span>))</span></code><code><span>hic.contact.hot &lt;- colorRampPalette(colors = GENOVA:::bezier_corrected_hot) </span></code><code><span>hic.red &lt;- colorRampPalette(c(<span>"white"</span>, <span>"red"</span>))</span></code><code><span>hcl.pals(type = <span>NULL</span>)</span></code><code><span>hcl.colors(<span>200</span>,palette = <span>"YlGnBu"</span>)</span></code><code><span>blueteal &lt;- ggthemes::tableau_color_pal(palette = <span>"Blue-Teal"</span>,type = <span>"ordered-sequential"</span>)</span></code><code><span>blueteal &lt;- colorRampPalette(colors = blueteal(<span>20</span>))</span></code><code><span><br></span></code><code><span>scales::show_col(colours = hcl.colors(<span>10</span>,<span>"YlGnBu"</span>))</span></code><code><span><br></span></code><code><span><span>###---1. download data-----------</span></span></code><code><span><span>###don't run it again</span></span></code><code><span><br></span></code><code><span><span>####you can donwload GEO supp data by getGEOSuppFiles</span></span></code><code><span>tmp.id &lt;- <span>"GSE66581"</span></span></code><code><span>tmp.dir &lt;- <span>"data"</span></span></code><code><span>dir.create(path = tmp.dir)</span></code><code><span>getGEOSuppFiles(GEO = tmp.id,makeDirectory = T,baseDir = <span>"data"</span>)</span></code><code><span><span>#### unzip data</span></span></code><code><span>untar(tarfile=<span>"data/GSE66581/GSE66581_RAW.tar"</span>,exdir=<span>"data/GSE66581/byReplicate/bedgraph"</span>)</span></code><code><span><br></span></code><code><span><span>#### or directly donwload url</span></span></code><code><span>tmp.input &lt;- <span>"https://ftp.ncbi.nlm.nih.gov/geo/samples/GSM1014nnn/GSM1014187/suppl/GSM1014187_mm9_wgEncodeUwDnaseEscj7S129ME0SigRep2.bigWig"</span></span></code><code><span>tmp.output &lt;- <span>"data/ENCODE_mESC_DNaseI.bw"</span> </span></code><code><span>download.file(url=tmp.input,destfile=tmp.output)</span></code></pre></section><p>安装软件和下载数据。<br></p><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="ruby"><code><span><span>###----2. Plot gene Track-------</span></span></code><code><span>library(<span>"TxDb.Mmusculus.UCSC.mm9.knownGene"</span>)</span></code><code><span><span>###----2.1 Prepare gene track data------</span></span></code><code><span>mm9 &lt;- TxDb.Mmusculus.UCSC.mm9.knownGene</span></code><code><span>mm9.tx &lt;- myGetGRangesFromsTxDb(txdb = mm9,</span></code><code><span>                                standard.chromosomes = T,</span></code><code><span>                                verbose = T)</span></code><code><span>tmp &lt;- bitr(mm9.tx$gene_id,</span></code><code><span>            fromType = <span>"ENTREZID"</span>,</span></code><code><span>            toType = <span>"SYMBOL"</span>,</span></code><code><span>            OrgDb = org.Mm.eg.db)</span></code><code><span>tmp &lt;- tmp[!duplicated(tmp$SYMBOL),]</span></code><code><span>ttt.mm9.tx &lt;- mm9.tx[mm9.tx$gene_id %<span>in</span>% tmp$ENTREZID]</span></code><code><span>ttt.mm9.tx$gene_id &lt;- plyr::mapvalues(ttt.mm9.tx$gene_id,</span></code><code><span>                                      from = tmp$ENTREZID,</span></code><code><span>                                      to = tmp$SYMBOL)</span></code><code><span>mm9.tx &lt;- ttt.mm9.tx</span></code><code><span><br></span></code><code><span>tmp.region &lt;- <span>"chr10:128222630-128407404"</span></span></code><code><span>p_gene &lt;- myGenePlot(annotation = mm9.tx,</span></code><code><span>                     region = tmp.region,</span></code><code><span>                     arrow_sbreaks = <span>600</span>,</span></code><code><span>                     font_size = <span>18</span>,</span></code><code><span>                     label_size = <span>3</span>)+</span></code><code><span>  ylab(<span>"Gene"</span>)+</span></code><code><span>  theme(plot.margin = unit(c(<span>0</span>,<span>0</span>,<span>0</span>,<span>0</span>), <span>"cm"</span>),</span></code><code><span>        axis.title.y = element_text(angle = <span>0</span>,vjust = <span>0</span>.<span>5</span>,hjust = <span>0</span>.<span>5</span>),</span></code><code><span>        axis.line = element_line(size = <span>1</span>),</span></code><code><span>        axis.ticks = element_line(size = <span>1</span>))</span></code><code><span>p_gene</span></code><code><span><br></span></code></pre></section><p><img data-galleryid="" data-ratio="0.6027777777777777" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBrj8PiajPFxYgf63teMl1zE3RD6GGXWLws1I4LICNWd5bXmZEJ2ydrGIGHBLwVNiaFEAWCZKZuIfwMg/640?wx_fmt=png" data-type="jpeg" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBrj8PiajPFxYgf63teMl1zE3RD6GGXWLws1I4LICNWd5bXmZEJ2ydrGIGHBLwVNiaFEAWCZKZuIfwMg/640?wx_fmt=png">mm10</p><p><img data-galleryid="" data-ratio="0.6027777777777777" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBrj8PiajPFxYgf63teMl1zE3UyQficeD8magV6kW2U2jn7yEnMOc078GTmkCoPGYxibVreyjibibuyO3Gg/640?wx_fmt=png" data-type="jpeg" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBrj8PiajPFxYgf63teMl1zE3UyQficeD8magV6kW2U2jn7yEnMOc078GTmkCoPGYxibVreyjibibuyO3Gg/640?wx_fmt=png">mm9</p><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="properties"><code><span><span>###----2. plot DNase-seq track from bigwig---------</span></span></code><code><span><br></span></code><code><span><span>tmp.files</span> <span>&lt;- "data/ENCODE_mESC_DNaseI.bw"</span></span></code><code><span><span>tmp.colors</span> <span>&lt;- "#446195"</span></span></code><code><span>  <span>tmp.levels</span> <span>&lt;- "DNase-seq"</span></span></code><code><span>  <span>p_DNase</span> <span>&lt;- myBigwigTrack(region = as(tmp.region,"GRanges"),</span></span></code><code><span>                           <span>bigwig</span> = <span>tmp.files,</span></span></code><code><span>                           <span>smooth</span> = <span>100,</span></span></code><code><span>                           <span>lognorm</span> = <span>F,</span></span></code><code><span>                           <span>type</span> = <span>"coverage",</span></span></code><code><span>                           <span>y_label</span> = <span>tmp.levels,</span></span></code><code><span>                           <span>fontsize</span>=<span>18,</span></span></code><code><span>                           <span>track.color</span>=<span>tmp.colors,</span></span></code><code><span>                           <span>tmp.ylimits</span>=<span>c(0,100),</span></span></code><code><span>                           <span>max.downsample</span> = <span>3000,</span></span></code><code><span>                           <span>downsample.rate</span> = <span>0.1,</span></span></code><code><span>                           <span>tmp.seed</span>=<span>42)</span></span></code><code><span>  <span>p_DNase</span></span></code><code><span>  <span>ggsave(filename</span> = <span>"fig/DNase_coverage_plot.png",</span></span></code><code><span>         <span>width</span> = <span>12,height = 8,dpi = 350,bg = "white")</span></span></code><code><span>  </span></code><code><span>  </span></code><code><span>  <span>p_DNase</span> <span>&lt;- myBigwigTrack(region = as(tmp.region,"GRanges"),</span></span></code><code><span>                           <span>bigwig</span> = <span>tmp.files,</span></span></code><code><span>                           <span>smooth</span> = <span>100,</span></span></code><code><span>                           <span>lognorm</span> = <span>F,</span></span></code><code><span>                           <span>type</span> = <span>"bar",</span></span></code><code><span>                           <span>y_label</span> = <span>tmp.levels,</span></span></code><code><span>                           <span>fontsize</span>=<span>18,</span></span></code><code><span>                           <span>track.color</span>=<span>tmp.colors,</span></span></code><code><span>                           <span>tmp.ylimits</span>=<span>c(0,100),</span></span></code><code><span>                           <span>max.downsample</span> = <span>3000,</span></span></code><code><span>                           <span>downsample.rate</span> = <span>0.1,</span></span></code><code><span>                           <span>tmp.seed</span>=<span>42)</span></span></code><code><span>  <span>p_DNase</span></span></code><code><span>  <span>ggsave(filename</span> = <span>"fig/DNase_bar_plot.png",</span></span></code><code><span>         <span>width</span> = <span>12,height = 8,dpi = 350,bg = "white")</span></span></code><code><span>  </span></code></pre></section><p><img data-galleryid="" data-ratio="0.6027777777777777" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBrj8PiajPFxYgf63teMl1zE3zEO4fZM9boicsDB3ozFfOlQdooCIbmickYvLg9HyI2uox8WWaC4aphXg/640?wx_fmt=png" data-type="jpeg" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBrj8PiajPFxYgf63teMl1zE3zEO4fZM9boicsDB3ozFfOlQdooCIbmickYvLg9HyI2uox8WWaC4aphXg/640?wx_fmt=png"></p><p><img data-galleryid="" data-ratio="0.6027777777777777" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBrj8PiajPFxYgf63teMl1zE3XkgIEKQ7Lb9KNsia4ngWz8YSXrvTGia7wAArtaLLWv6m1PMeZGxoxX7A/640?wx_fmt=png" data-type="jpeg" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBrj8PiajPFxYgf63teMl1zE3XkgIEKQ7Lb9KNsia4ngWz8YSXrvTGia7wAArtaLLWv6m1PMeZGxoxX7A/640?wx_fmt=png"></p><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="php"><code><span>  <span>####---------3.2 plot ATAC-seq track--------------</span></span></code><code><span>  </span></code><code><span>  tmp.levels &lt;- c(<span>"2cell"</span>,<span>"4cell"</span>,<span>"8cell"</span>,<span>"icm"</span>)</span></code><code><span>  tmp.colors &lt;- c(<span>"#3d7b42"</span>,<span>"#6a4181"</span>,<span>"#cf8e67"</span>,<span>"#a9477e"</span>)</span></code><code><span>  scales::show_col(tmp.colors)</span></code><code><span>  </span></code><code><span>  tmp.<span>list</span><span>.1</span> &lt;- lapply(seq_along(tmp.levels),<span><span>function</span><span>(ii)</span></span>{</span></code><code><span>    cat(ii,sep = <span>"\n"</span>)</span></code><code><span>    p1 &lt;- myBigwigTrack(region = <span>as</span>(tmp.region,<span>"GRanges"</span>),</span></code><code><span>                        bigwig = paste0(<span>"data/GSE66581/byReplicate/bigwig/"</span>,tmp.levels[ii],<span>"_re1"</span>,<span>".bw"</span>),</span></code><code><span>                        smooth = <span>100</span>,</span></code><code><span>                        lognorm = F,</span></code><code><span>                        type = <span>"bar"</span>,</span></code><code><span>                        y_label = paste0(tmp.levels[ii],<span>" rep1"</span>),</span></code><code><span>                        fontsize=<span>18</span>,</span></code><code><span>                        track.color=tmp.colors[ii],</span></code><code><span>                        tmp.ylimits=c(<span>0</span>,<span>10</span>),</span></code><code><span>                        max.downsample = <span>3000</span>,</span></code><code><span>                        downsample.rate = <span>0.1</span>,</span></code><code><span>                        tmp.seed=<span>42</span>)</span></code><code><span>    p2 &lt;- myBigwigTrack(region = <span>as</span>(tmp.region,<span>"GRanges"</span>),</span></code><code><span>                        bigwig = paste0(<span>"data/GSE66581/byReplicate/bigwig/"</span>,tmp.levels[ii],<span>"_re2"</span>,<span>".bw"</span>),</span></code><code><span>                        smooth = <span>100</span>,</span></code><code><span>                        lognorm = F,</span></code><code><span>                        type = <span>"bar"</span>,</span></code><code><span>                        y_label =paste0(tmp.levels[ii],<span>" rep2"</span>),</span></code><code><span>                        fontsize=<span>18</span>,</span></code><code><span>                        track.color=tmp.colors[ii],</span></code><code><span>                        tmp.ylimits=c(<span>0</span>,<span>10</span>),</span></code><code><span>                        max.downsample = <span>3000</span>,</span></code><code><span>                        downsample.rate = <span>0.1</span>,</span></code><code><span>                        tmp.seed=<span>42</span>)</span></code><code><span>    p &lt;- <span>list</span>(p1,p2)</span></code><code><span>    <span>return</span>(p)</span></code><code><span>  })</span></code><code><span>  tmp.<span>list</span><span>.1</span> &lt;- Reduce(c,tmp.<span>list</span><span>.1</span>)</span></code><code><span>  p_ATAC_track_list &lt;- tmp.<span>list</span><span>.1</span></span></code><code><span>  plot.<span>list</span> &lt;- c(p_ATAC_track_list,<span>list</span>(p_DNase))</span></code><code><span>  </span></code><code><span>  </span></code><code><span>  (wrap_plots(plot.<span>list</span>,ncol = <span>1</span>)  + </span></code><code><span>      p_gene  + plot_layout(ncol = <span>1</span>,heights = c(rep(<span>1</span>,<span>12</span>),<span>3</span>))) + </span></code><code><span>    plot_annotation(title = <span>"chr10:128,222,630–128,407,404"</span>) &amp; </span></code><code><span>    theme(plot.title = element_text(hjust = <span>1</span>),</span></code><code><span>          axis.line.x = element_blank(),</span></code><code><span>          axis.text = element_blank(),</span></code><code><span>          axis.ticks = element_blank(),</span></code><code><span>          axis.title.x = element_blank())</span></code><code><span>  </span></code><code><span>  <span>###create output directory</span></span></code><code><span>  dir.create(path = <span>"fig"</span>)</span></code><code><span>  ggsave(filename = <span>"fig/ATAC-seq_track_by_R.png"</span>,</span></code><code><span>         width = <span>12</span>,height = <span>8</span>,</span></code><code><span>         dpi = <span>350</span>)</span></code><code><span>  </span></code><code><span>  </span></code><code><span>  <span>####tmp.list.2</span></span></code><code><span>  tmp.levels &lt;- c(<span>"GSM1933924_2cell_re1"</span>,<span>"GSM1625847_4cell_re1"</span>,<span>"GSM1933928_8cell_re1"</span>)</span></code><code><span>  tmp.colors &lt;- c(<span>"#76492a"</span>,<span>"#76492a"</span>,<span>"#76492a"</span>)</span></code><code><span>  scales::show_col(tmp.colors)</span></code><code><span>  </span></code><code><span>  tmp.<span>list</span><span>.2</span> &lt;- lapply(seq_along(tmp.levels),<span><span>function</span><span>(ii)</span></span>{</span></code><code><span>    cat(ii,sep = <span>"\n"</span>)</span></code><code><span>    p &lt;- myBigwigTrack(region = <span>as</span>(tmp.region,<span>"GRanges"</span>),</span></code><code><span>                       bigwig = paste0(<span>"data/GSE66581/bigwig/"</span>,tmp.levels[ii],<span>".bw"</span>),</span></code><code><span>                       smooth = <span>100</span>,</span></code><code><span>                       lognorm = F,</span></code><code><span>                       type = <span>"bar"</span>,</span></code><code><span>                       y_label = tmp.levels[ii],</span></code><code><span>                       fontsize=<span>18</span>,</span></code><code><span>                       track.color=tmp.colors[ii],</span></code><code><span>                       tmp.ylimits=c(<span>0</span>,<span>10</span>),</span></code><code><span>                       max.downsample = <span>3000</span>,</span></code><code><span>                       downsample.rate = <span>0.1</span>,</span></code><code><span>                       tmp.seed=<span>42</span>)</span></code><code><span>    <span>return</span>(p)</span></code><code><span>  })</span></code><code><span>  </span></code><code><span>  p_ATAC_track_list &lt;- c(tmp.<span>list</span><span>.1</span>,tmp.<span>list</span><span>.2</span>)</span></code><code><span>  plot.<span>list</span> &lt;- c(p_ATAC_track_list,<span>list</span>(p_DNase))</span></code><code><span>  </span></code><code><span>  </span></code><code><span>  (wrap_plots(plot.<span>list</span>,ncol = <span>1</span>)  + </span></code><code><span>      p_gene  + plot_layout(ncol = <span>1</span>,heights = c(rep(<span>1</span>,<span>12</span>),<span>3</span>))) + </span></code><code><span>    plot_annotation(title = <span>"chr10:128,222,630–128,407,404"</span>) &amp; </span></code><code><span>    theme(plot.title = element_text(hjust = <span>1</span>),</span></code><code><span>          axis.line.x = element_blank(),</span></code><code><span>          axis.text = element_blank(),</span></code><code><span>          axis.ticks = element_blank(),</span></code><code><span>          axis.title.x = element_blank())</span></code><code><span>  </span></code><code><span>  <span>###create output directory</span></span></code><code><span>  dir.create(path = <span>"fig"</span>)</span></code><code><span>  ggsave(filename = <span>"fig/ATAC-seq_track_by_R.png"</span>,</span></code><code><span>         width = <span>12</span>,height = <span>8</span>,</span></code><code><span>         dpi = <span>350</span>)</span></code><code><span>  </span></code></pre></section><p><img data-galleryid="" data-ratio="0.6027777777777777" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBrj8PiajPFxYgf63teMl1zE3CUrDIGDX3jBDvRoLNkFqwVRibCS4I70cXJVOiasZ73ia0GklrVBqFRrdQ/640?wx_fmt=png" data-type="jpeg" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBrj8PiajPFxYgf63teMl1zE3CUrDIGDX3jBDvRoLNkFqwVRibCS4I70cXJVOiasZ73ia0GklrVBqFRrdQ/640?wx_fmt=png"></p><p><img data-galleryid="" data-ratio="0.7527777777777778" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBrj8PiajPFxYgf63teMl1zE3qSW9G5SrD0JiaQbW7K4SKjq2EoNokDbeDkUxL9IKDu9DaLAwMPr8ALQ/640?wx_fmt=png" data-type="jpeg" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBrj8PiajPFxYgf63teMl1zE3qSW9G5SrD0JiaQbW7K4SKjq2EoNokDbeDkUxL9IKDu9DaLAwMPr8ALQ/640?wx_fmt=png"></p><p>参考资料：</p><p>https://github.com/Landau1994/PlotEpiTrackByR</p><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/UB6H3QS0dDpf7n1uVVSR1w",target="_blank" rel="noopener noreferrer">原文链接</a>
