---
title: "如何实现DNA甲基化和基因表达谱数据的联合分析"
date: 2023-03-28T07:31:17Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Acdemic"]
---
如何实现DNA甲基化和基因表达谱数据的联合分析 by 生信菜鸟团
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><p data-tool="mdnice编辑器"><span>上期我们讲到DNA甲基化数据的免疫浸润分析，这一期接着和大家分享DNA甲基化数据处理的R包，看看甲基化和基因表达谱数据碰撞在一起，能擦出多大的火花~</span></p><blockquote data-tool="mdnice编辑器"><p><span>doi: 10.1186/s13059-015-0668-3.</span></p><p><span>Inferring regulatory element landscapes and transcription factor networks from cancer methylomes - PubMed (nih.gov)</span></p><p><span>来自2015 Genome Biology 一区</span></p></blockquote><p data-tool="mdnice编辑器"><span><strong>一句话形容：</strong><strong><code>ELMER</code>(Enhancer Linking by Methylation/Expression Relationships)使用DNA 甲基化来识别增强子，并将增强子状态与附近基因的表达联系起来，以确定转录目标。</strong></span></p><p data-tool="mdnice编辑器"><span><strong><br></strong></span></p><p><img data-ratio="0.6696658097686375" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/iaRJcrq2Los8iaucDT3eSqBboy4mcgPg70d9vnr3BI5hVsjX1G39I977F0M3XdickQOFBG0gvOflqDmsBMmich6Q4A/640?wx_fmt=jpeg" data-type="jpeg" data-w="778" src="https://mmbiz.qpic.cn/mmbiz_jpg/iaRJcrq2Los8iaucDT3eSqBboy4mcgPg70d9vnr3BI5hVsjX1G39I977F0M3XdickQOFBG0gvOflqDmsBMmich6Q4A/640?wx_fmt=jpeg"></p><p data-tool="mdnice编辑器"><span><br></span></p><p data-tool="mdnice编辑器"><span>在认识这个包的函数之前，我们想先简单介绍两个数据结构：</span></p><p data-tool="mdnice编辑器"><span><strong><code>MultiAssayExperiment</code>和它的姊妹<code>SummarizedExperiment</code>，别小看它们，二者是生信分析的关键数据存储对象。</strong></span></p><p data-tool="mdnice编辑器"><span>MultiAssayExperiment的基本结构如下：</span><img data-backh="463" data-backw="502" data-ratio="0.921875" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8iaucDT3eSqBboy4mcgPg70a3Ca9uIshQ4JdUaz3GKgZCIYl8tHL4a9mLGlaeNeoteOvicSDjnN08Q/640?wx_fmt=png" data-type="png" data-w="384" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8iaucDT3eSqBboy4mcgPg70a3Ca9uIshQ4JdUaz3GKgZCIYl8tHL4a9mLGlaeNeoteOvicSDjnN08Q/640?wx_fmt=png"></p><pre data-tool="mdnice编辑器"><code><span><span>library</span>(MultiAssayExperiment)<br><span>library</span>(GenomicRanges)<br>empty &lt;- MultiAssayExperiment()<br>empty<br></span></code></pre><pre data-tool="mdnice编辑器"><code><span><span>## A MultiAssayExperiment object of 0 listed</span><br><span>##  experiments with no user-defined names and respective classes.</span><br><span>##  Containing an ExperimentList class object of length 0:</span><br><span>##  Functionality:</span><br><span>##  experiments() - obtain the ExperimentList instance</span><br><span>##  colData() - the primary/phenotype DataFrame</span><br><span>##  sampleMap() - the sample coordination DataFrame</span><br><span>##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span><br><span>##  *Format() - convert into a long or wide DataFrame</span><br><span>##  assays() - convert ExperimentList to a SimpleList of matrices</span><br><span>##  exportClass() - save data to flat files</span><br></span></code></pre><pre data-tool="mdnice编辑器"><code><span>slotNames(empty)<br><span>## [1] "ExperimentList" "colData"        "sampleMap"      "drops"         </span><br><span>## [5] "metadata"</span><br></span></code></pre><p data-tool="mdnice编辑器"><span>MultiAssayExperiment对象有三个组成部分：</span></p><ol data-tool="mdnice编辑器"><li><section><code><span>ExperimentList</span></code><span>：一个任意类别的检测数据集的列表，每个观察值有一列。</span></section></li><li><section><code><span>colData</span></code><span>：提供关于病人、细胞系或其他生物单位的数据，每个单位有一行，每个变量有一列。</span></section></li><li><p><br></p><section><code><span>sampleMap</span></code><span>：将assay（experiments）中的每一列（观察）与colData中的一行（生物单位）精确联系起来；然而，colData的一行可以映射到每个assay的零、一或多列，允许缺少和重复assay。绿色条纹表示一个experiment中的一个主题与多个观察点（observations）的映射。</span></section><p><br></p></li></ol><p><br></p><p><br></p><p data-tool="mdnice编辑器"><code><span>SummarizedExperiment</span></code><span>结构相对更容易理解，是一个类似矩阵的容器，其中<strong>行</strong>代表感兴趣的特征（如基因、转录本、外显子等），<strong>列</strong>代表<strong>样本</strong>。</span></p><p data-tool="mdnice编辑器"><span>这些对象包含一个或多个assay，每个assay由一个数字或其他模式的矩阵状对象。</span></p><p data-tool="mdnice编辑器"><span>SummarizedExperiment对象的<strong>行</strong>代表感兴趣的features。</span></p><p data-tool="mdnice编辑器"><span>关于这些features的信息被存储在一个DataFrame对象中，可以用函数rowData()访问。</span></p><p data-tool="mdnice编辑器"><span>DataFrame的<strong>每一行</strong>都提供了SummarizedExperiment对象的<strong>相应行</strong>中的features信息。</span></p><p data-tool="mdnice编辑器"><span>DataFrame的<strong>列</strong>代表感兴趣的features的不同属性。（例如，基因或转录本ID等）</span></p><figure data-tool="mdnice编辑器"><img data-backh="429" data-backw="502" data-ratio="0.8522727272727273" data-src="https://mmbiz.qpic.cn/mmbiz_svg/iaRlzG8zy7Btfqe8AMibtvxofhfU76sOIMcyiaURxWf1qbibocTPt6nraCrJzdqiasNnibLn9icRmF4u2iaydmIf6MVuc5ic7cjTfCXY0/640?wx_fmt=svg" data-type="svg" data-w="176" src="https://mmbiz.qpic.cn/mmbiz_svg/iaRlzG8zy7Btfqe8AMibtvxofhfU76sOIMcyiaURxWf1qbibocTPt6nraCrJzdqiasNnibLn9icRmF4u2iaydmIf6MVuc5ic7cjTfCXY0/640?wx_fmt=svg"></figure><pre data-tool="mdnice编辑器"><code><span><span>library</span>(SummarizedExperiment)<br>data(airway, package=<span>"airway"</span>)<br>se &lt;- airway<br>se<br></span></code></pre><pre data-tool="mdnice编辑器"><code><span><span>## class: RangedSummarizedExperiment </span><br><span>## dim: 64102 8 </span><br><span>## metadata(1): ''</span><br><span>## assays(1): counts</span><br><span>## rownames(64102): ENSG00000000003 ENSG00000000005 ... LRG_98 LRG_99</span><br><span>## rowData names(0):</span><br><span>## colnames(8): SRR1039508 SRR1039509 ... SRR1039520 SRR1039521</span><br><span>## colData names(9): SampleName cell ... Sample BioSample</span><br></span></code></pre><p data-tool="mdnice编辑器"><span>可以用</span><code><span>assays()</span></code><span>取出</span><code><span>SummarizedExperiment</span></code><span> 对象</span></p><p data-tool="mdnice编辑器"><br></p><p data-tool="mdnice编辑器"><span>至于什么时候用前者，什么时候用后者呢？</span></p><blockquote data-tool="mdnice编辑器"><p><span>For assays with <strong>different numbers of rows and even columns</strong>, </span><code><span>MultiAssayExperiment</span></code><span> is recommended.</span></p><p><span>For sets of assays with the <strong>same information across all rows</strong> (e.g., genes or genomic ranges), </span><code><span>SummarizedExperiment</span></code><span> is the recommended data structure.</span></p></blockquote><p data-tool="mdnice编辑器"><br></p><p data-tool="mdnice编辑器"><span>接着就是代码实操啦⬇</span></p><h2 data-tool="mdnice编辑器"><span>构建分析对象</span></h2><h3 data-tool="mdnice编辑器"><span>createMAE</span></h3><section data-tool="mdnice编辑器"><table width="NaN"><thead><tr><th><span>createMAE</span></th><th><span>构建用于ELMER分析的对象</span></th></tr></thead></table></section><p data-tool="mdnice编辑器"><span>基本参数要求如下：【很重要哦】</span></p><figure data-tool="mdnice编辑器"><img data-backh="495" data-backw="502" data-ratio="0.9849498327759197" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8iaucDT3eSqBboy4mcgPg70ia2zSHuoiaxUMaolOicuCwcXHibG3J7yibNuA7M85zKytibKVmpYkANVsG3A/640?wx_fmt=png" data-type="png" data-w="598" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8iaucDT3eSqBboy4mcgPg70ia2zSHuoiaxUMaolOicuCwcXHibG3J7yibNuA7M85zKytibKVmpYkANVsG3A/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><code><span><span># TCGA example using TCGAbiolinks</span><br><span># Testing creating MultyAssayExperiment object</span><br><span># Load library</span><br><span>library</span>(TCGAbiolinks)<br><span>library</span>(SummarizedExperiment)<br>samples &lt;- c(<br><span>"TCGA-BA-4074"</span>, <span>"TCGA-BA-4075"</span>, <span>"TCGA-BA-4077"</span>, <span>"TCGA-BA-5149"</span>,<br><span>"TCGA-UF-A7JK"</span>, <span>"TCGA-UF-A7JS"</span>, <span>"TCGA-UF-A7JT"</span>, <span>"TCGA-UF-A7JV"</span><br>)<br></span></code></pre><p data-tool="mdnice编辑器"><span>DNA甲基化和基因表达对象应该有相同的样本名，并作为列名</span></p><pre data-tool="mdnice编辑器"><code><span><span>#1) Get gene expression matrix</span><br><span># Aligned against Hg19</span><br>query.exp.hg19 &lt;- GDCquery(<br>project = <span>"TCGA-HNSC"</span>,<br>data.category = <span>"Gene expression"</span>,<br>data.type = <span>"Gene expression quantification"</span>,<br>platform = <span>"Illumina HiSeq"</span>,<br>file.type = <span>"normalized_results"</span>,<br>experimental.strategy = <span>"RNA-Seq"</span>,<br>barcode = samples,<br>legacy = <span>TRUE</span><br>)<br>GDCdownload(query.exp.hg19)<br>exp.hg19 &lt;- GDCprepare(query.exp.hg19)<br><span># Our object needs to have emsembl gene id as rownames</span><br>rownames(exp.hg19) &lt;- values(exp.hg19)$ensembl_gene_id<br></span></code></pre><pre data-tool="mdnice编辑器"><code><span><span>#2) DNA Methylation</span><br>query.met &lt;- GDCquery(<br>project = <span>"TCGA-HNSC"</span>,<br>legacy = <span>FALSE</span>,<br>data.category = <span>"DNA Methylation"</span>,<br>data.type = <span>"Methylation Beta Value"</span>,<br>barcode = samples,<br>platform = <span>"Illumina Human Methylation 450"</span><br>)<br>GDCdownload(query.met)<br>met &lt;- GDCprepare(query = query.met)<br></span></code></pre><pre data-tool="mdnice编辑器"><code><span><span>#3)</span><br>distal.enhancer &lt;- get.feature.probe(genome = <span>"hg19"</span>,met.platform = <span>"450k"</span>)<br></span></code></pre><pre data-tool="mdnice编辑器"><code><span><span>#4) Consisering it is TCGA and SE</span><br>mae.hg19 &lt;- createMAE(<br>exp = exp.hg19,<br>met = met,<br>TCGA = <span>TRUE</span>,<br>genome = <span>"hg19"</span>,<br>filter.probes = distal.enhancer<br>)<br>values(getExp(mae.hg19))<br></span></code></pre><h3 data-tool="mdnice编辑器"><span>getTCGA</span></h3><blockquote data-tool="mdnice编辑器"><p><span>直接从TCGA下载特定癌症类型的DNA甲基化、RNA表达和临床数据。</span></p><p><span>这一步对网速的要求比较高</span></p></blockquote><pre data-tool="mdnice编辑器"><code><span>getTCGA(<br>disease = <span>"BRCA"</span>,<br>Meth = <span>FALSE</span>,<br>RNA = <span>FALSE</span>,<br>Clinic = <span>TRUE</span>,<br>basedir = tempdir(),<br>genome = <span>"hg19"</span><br>)<br></span></code></pre><p data-tool="mdnice编辑器"><span>下载的数据将被转换为矩阵或数据框，以便进一步分析。</span></p><h2 data-tool="mdnice编辑器"><span>差异甲基化分析</span></h2><h3 data-tool="mdnice编辑器"><span>get.diff.meth</span></h3><blockquote data-tool="mdnice编辑器"><p><span>识别两组之间的低/高甲基化的CpG位点（即正常与肿瘤样本对比）。</span></p></blockquote><p data-tool="mdnice编辑器"><span>基本参数如下：</span></p><section data-tool="mdnice编辑器"><table width="NaN"><thead><tr><th><span>Arguments</span></th><th><br></th></tr></thead><tbody><tr><td><span>data</span></td><td><span>A multiAssayExperiment with DNA methylation and Gene Expression data.</span></td></tr><tr><td><span>diff.dir</span></td><td><span>A character can be "hypo", "hyper" or "both", showing differential methylation direction</span></td></tr><tr><td><span>cores</span></td><td><span>线程数</span></td></tr><tr><td><span>mode</span></td><td><span>Can be "unsupervised" or "supervised".</span></td></tr><tr><td><span>pvalue</span></td><td><span>A number specifies the significant P value (adjusted P value by BH) threshold Limit for selecting significant hypo/hyper-methylated probes.</span></td></tr><tr><td><span>group.col</span></td><td><span>A column defining the groups of the sample. <strong>可以在createMAE那一步用colData创造</strong>.</span></td></tr><tr><td><span>group1</span></td><td><span>A group from group.col.</span></td></tr><tr><td><span>group2</span></td><td><span>A group from group.col.</span></td></tr><tr><td><span>test</span></td><td><span>统计方法</span></td></tr><tr><td><span>sig.dif</span></td><td><span>A number specifies the smallest DNA methylation difference as a cutoff for selecting significant hypo/hyper-methylated probes. Default is 0.3.</span></td></tr><tr><td><span>save</span></td><td><span>When TRUE, two getMethdiff.XX.csv files will be generated</span></td></tr></tbody></table></section><pre data-tool="mdnice编辑器"><code><span>Hypo.probe &lt;- get.diff.meth(data,<br>diff.dir=<span>"hypo"</span>, <span>##这里可以还选择hyper</span><br>group.col = <span>"definition"</span>,<br>group1 = <span>"Primary solid Tumor"</span>,<br>group2 = <span>"Solid Tissue Normal"</span>,<br>sig.dif = <span>0.1</span>) <span># get hypomethylated probes</span><br></span></code></pre><h3 data-tool="mdnice编辑器"><span>scatter.plot</span></h3><blockquote data-tool="mdnice编辑器"><p><span>绘制基因表达和DNA甲基化之间的散点图</span></p></blockquote><pre data-tool="mdnice编辑器"><code><span>scatter.plot(data,<br>             byProbe=list(probe=c(<span>"cg19403323"</span>),numFlankingGenes=<span>20</span>),<br>             category=<span>"definition"</span>,<br>             save=<span>TRUE</span>) <span>## save to pdf</span><br></span></code></pre><pre data-tool="mdnice编辑器"><code><span><span># b. generate one probe-gene pair</span><br>scatter.plot(data,<br>             byPair=list(probe=c(<span>"cg19403323"</span>),gene=c(<span>"ENSG00000143322"</span>)),<br>             category=<span>"definition"</span>,<br>             save=<span>FALSE</span>,<br>             lm_line=<span>TRUE</span>)<br></span></code></pre><p data-tool="mdnice编辑器"><span>还有一个参数：byTF ==list(TF=c(), probe=c())</span></p><h3 data-tool="mdnice编辑器"><span>metBoxPlot</span></h3><blockquote data-tool="mdnice编辑器"><p><span>绘制基因表达和DNA甲基化之间的散点图</span></p></blockquote><pre data-tool="mdnice编辑器"><code><span>metBoxPlot(data,<br>group.col = group.col, <span>## 可以通过colnames(MultiAssayExperiment::colData(data))调用</span><br>group1 = group1,<br>group2 = group2,<br>probe =<span>"cg17898069"</span>,<br>minSubgroupFrac = <span>0.2</span>,<br>diff.dir = <span>"hypo"</span>)<br></span></code></pre><h2 data-tool="mdnice编辑器"><span>motif分析</span></h2><h3 data-tool="mdnice编辑器"><span>get.enriched.motif</span></h3><blockquote data-tool="mdnice编辑器"><p><span>识别一组探针(HM450K)区域中的高代表度的motif</span></p></blockquote><pre data-tool="mdnice编辑器"><code><span><span># If the MAE is set, the background and the probes.motif will be automatically set</span><br>enriched.motif &lt;- get.enriched.motif(data = data,<br>min.motif.quality = <span>"DS"</span>,<br>probes=probes,<br>pvalue = <span>1</span>,<br>min.incidence=<span>2</span>,<br>label=<span>"hypo"</span>)<br></span></code></pre><h2 data-tool="mdnice编辑器"><span>转录因子分析</span></h2><h3 data-tool="mdnice编辑器"><span>get.TFs</span></h3><blockquote data-tool="mdnice编辑器"><p><span>识别调控TFs。</span></p></blockquote><pre data-tool="mdnice编辑器"><code><span>TF &lt;- get.TFs(data,<br>enriched.motif, <span>##上一步的运行结果</span><br>group.col = <span>"definition"</span>,<br>group1 = <span>"Primary solid Tumor"</span>,<br>group2 = <span>"Solid Tissue Normal"</span>,<br>TFs = data.frame(<br>external_gene_name=c(<span>"TP53"</span>,<span>"TP63"</span>,<span>"TP73"</span>),<br>ensembl_gene_id= c(<span>"ENSG00000141510"</span>,<br><span>"ENSG00000073282"</span>,<span>"ENSG00000078900"</span>),<span>##A data.frame containing TF GeneID and Symbol or a path of XX.csv file </span><br>stringsAsFactors = <span>FALSE</span>),<br>label=<span>"hypo"</span>,<br>save = <span>TRUE</span> <span>##If save is true, two files will be saved: getTF.XX.significant.TFs.with.motif.summary.csv and getTF.hypo.TFs.with.motif.pvalue.rda</span><br>)<br></span></code></pre><pre data-tool="mdnice编辑器"><code><span>TF &lt;- get.TFs(data,<br>group.col = <span>"definition"</span>,<br>group1 = <span>"Primary solid Tumor"</span>,<br>group2 = <span>"Solid Tissue Normal"</span>,<br>enriched.motif,<br>label=<span>"hypo"</span>)<br></span></code></pre><h3 data-tool="mdnice编辑器"><span>GetNearGenes</span></h3><blockquote data-tool="mdnice编辑器"><p><span>为一个基因座收集邻近的基因</span></p></blockquote><pre data-tool="mdnice编辑器"><code><span>geneAnnot &lt;- getTSS(genome = <span>"hg38"</span>)<br>probe &lt;- GenomicRanges::GRanges(seqnames = c(<span>"chr1"</span>,<span>"chr2"</span>),<br>range=IRanges::IRanges(start = c(<span>16058489</span>,<span>236417627</span>), end= c(<span>16058489</span>,<span>236417627</span>)),<br>name= c(<span>"cg18108049"</span>,<span>"cg17125141"</span>))<br>names(probe) &lt;- c(<span>"cg18108049"</span>,<span>"cg17125141"</span>)<br><br>GetNearGenes(<br>data = data,<span>##第一步函数创造的multiAssayExperiment对象</span><br>probes = probe,<br>geneAnnot = geneAnnot,<br>TRange = range,<br>numFlankingGenes = <span>20</span><br>)<br></span></code></pre><h3 data-tool="mdnice编辑器"><span>getTFtargets</span></h3><blockquote data-tool="mdnice编辑器"><p><span>获取TF靶基因</span></p></blockquote><pre data-tool="mdnice编辑器"><code><span>getTFtargets(<br>pairs,<span>##Output of get.pairs function: dataframe or file path</span><br>enriched.motif,<span>##The file created by ELMER is getMotif...enriched.motifs.rda</span><br>TF.result,<span>##get.TFs的结果</span><br>dmc.analysis,<br>mae,<span>##A multiAssayExperiment outputed from createMAE function</span><br>save = <span>TRUE</span>,<br>dir.out = <span>"./"</span>,<br>classification = <span>"family"</span>,<br>cores = <span>1</span>,<br>label = <span>NULL</span><br>)<br></span></code></pre><h3 data-tool="mdnice编辑器"><span>getTSS</span></h3><blockquote data-tool="mdnice编辑器"><p><span>从Bioconductor软件包biomaRt中获取GENCODE基因注释（转录水平）。</span></p><p><span>如果在TSS列表中指定上游和下游，GENCODE基因的启动子区域将被生成产生。</span></p></blockquote><pre data-tool="mdnice编辑器"><code><span><span># get GENCODE gene annotation (transcripts level)</span><br>getTSS &lt;- getTSS()<br>getTSS &lt;- getTSS(genome.build = <span>"hg38"</span>, TSS=list(upstream=<span>1000</span>, downstream=<span>1000</span>))<br></span></code></pre><h3 data-tool="mdnice编辑器"><span>TF.rank.plot</span></h3><blockquote data-tool="mdnice编辑器"><p><span>TF.rank.plot绘制分数（-log10(P值)），评估TF表达水平和motif位点的平均DNA甲基化之间的相关性。</span></p></blockquote><pre data-tool="mdnice编辑器"><code><span>TF &lt;- get.TFs(data,<br>enriched.motif,<br>group.col = <span>"definition"</span>,<br>group1 = <span>"Primary solid Tumor"</span>,<br>group2 = <span>"Solid Tissue Normal"</span>,<br>TFs = data.frame(<br>external_gene_name=c(<span>"TP53"</span>,<span>"TP63"</span>,<span>"TP73"</span>),<br>ensembl_gene_id= c(<span>"ENSG00000141510"</span>,<br><span>"ENSG00000073282"</span>,<br><span>"ENSG00000078900"</span>),<br>stringsAsFactors = <span>FALSE</span>),<br>label=<span>"hypo"</span>)<br></span></code></pre><pre data-tool="mdnice编辑器"><code><span>TF.meth.cor &lt;- get(load(<span>"getTF.hypo.TFs.with.motif.pvalue.rda"</span>))<br></span></code></pre><pre data-tool="mdnice编辑器"><code><span>TF.rank.plot(motif.pvalue=TF.meth.cor,<br>motif=<span>"P53_HUMAN.H11MO.0.A"</span>,<br>TF.label=createMotifRelevantTfs(<span>"subfamily"</span>)[<span>"P53_HUMAN.H11MO.0.A"</span>],<br>save=<span>TRUE</span>)<br></span></code></pre><h2 data-tool="mdnice编辑器"><span>增强子-基因分析</span></h2><h3 data-tool="mdnice编辑器"><span>get.pair</span></h3><blockquote data-tool="mdnice编辑器"><p><span>预测增强子-基因的关联性</span></p></blockquote><pre data-tool="mdnice编辑器"><code><span>Hypo.pair &lt;- get.pair(data = data,<span>##第一步函数创造的multiAssayExperiment对象</span><br>nearGenes = nearGenes,<span>##GetNearGenes运行的结果</span><br>permu.size = <span>5</span>,<br>raw.pvalue = <span>0.2</span>,<br>Pe = <span>0.2</span>,<br>dir.out = <span>"./"</span>,<br>diffExp = <span>TRUE</span>,<br>group.col = <span>"definition"</span>,<br>group1 = <span>"Primary solid Tumor"</span>,<br>group2 = <span>"Solid Tissue Normal"</span>,<br>label = <span>"hypo"</span>)<br></span></code></pre><h3 data-tool="mdnice编辑器"><span>schematic.plot</span></h3><blockquote data-tool="mdnice编辑器"><p><span>绘图显示基因和探针的位置</span></p></blockquote><p data-tool="mdnice编辑器"><code><span>pair</span></code><span>  is the ouput of </span><code><span>get.pair</span></code><span> function.</span></p><pre data-tool="mdnice编辑器"><code><span>schematic.plot(data,<br>group.col = <span>"definition"</span>,<br>group1 = <span>"Primary solid Tumor"</span>,<br>group2 = <span>"Solid Tissue Normal"</span>,<br>pair = pair,<br>byProbe = <span>"cg19403323"</span>)<br></span></code></pre><pre data-tool="mdnice编辑器"><code><span>schematic.plot(data,<br>group.col = <span>"definition"</span>,<br>group1 = <span>"Primary solid Tumor"</span>,<br>group2 = <span>"Solid Tissue Normal"</span>,<br>pair = pair,<br>byGeneID = <span>"ENSG00000009790"</span>)<br></span></code></pre><pre data-tool="mdnice编辑器"><code><span>schematic.plot(data,<br>group.col = <span>"definition"</span>,<br>group1 = <span>"Primary solid Tumor"</span>,<br>group2 = <span>"Solid Tissue Normal"</span>,<br>pair = pair,<br>byCoordinate = list(chr=<span>"chr1"</span>, start = <span>209000000</span>, end = <span>209960000</span>))<br></span></code></pre><h3 data-tool="mdnice编辑器"><span>promoterMeth</span></h3><blockquote data-tool="mdnice编辑器"><p><span>计算基因表达与DNA 启动子区域的甲基化</span></p></blockquote><section data-tool="mdnice编辑器"><table width="NaN"><thead><tr><th><span>Arguments</span></th><th><br></th></tr></thead><tbody><tr><td><span>data</span></td><td><span>A Multi Assay Experiment object with DNA methylation and gene expression Summarized Experiment objects</span></td></tr><tr><td><span>sig.pvalue</span></td><td><span>A number specifies significant cutoff for gene silenced by promoter methylation. Default is 0.01. P value is raw P value without adjustment.</span></td></tr><tr><td><span>minSubgroupFrac</span></td><td><span>A number ranging from 0 to 1 specifying the percentage of samples used to create the groups <strong>U (unmethylated) and M (methylated)</strong> used to link probes to genes.</span></td></tr><tr><td><span>upstream</span></td><td><span>Number of bp upstream of TSS to consider as promoter region</span></td></tr><tr><td><span>downstream</span></td><td><span>Number of bp downstream of TSS to consider as promoter region</span></td></tr><tr><td><span>save</span></td><td><span>If it is true, the result will be saved</span></td></tr></tbody></table></section><pre data-tool="mdnice编辑器"><code><span>promoterMeth(data,<br>             sig.pvalue = <span>0.01</span>,<br>             minSubgroupFrac = <span>0.4</span>,<br>             upstream = <span>200</span>,<br>             downstream = <span>2000</span>,<br>             save = <span>TRUE</span>, <br>             cores = <span>1</span>)<br></span></code></pre><h2 data-tool="mdnice编辑器"><span>生存分析</span></h2><h3 data-tool="mdnice编辑器"><span>TFsurvival.plot</span></h3><blockquote data-tool="mdnice编辑器"><p><span>创建基于TF表达的生存图</span></p></blockquote><pre data-tool="mdnice编辑器"><code><span>TFsurvival.plot(data, <br>                TF,<span>##A gene symbol</span><br>                xlim = <span>NULL</span>,<span>##Limit x axis showed in plot</span><br>                percentage = <span>0.3</span>, <br>                save = <span>TRUE</span>)<br></span></code></pre><p data-tool="mdnice编辑器"><span>这里的data要求有 </span><code><span>vital_status</span></code><span>, </span><code><span>days_to_last_follow_up </span></code><span>and </span><code><span>days_to_death</span></code><span>这三列临床信息~</span></p><h2 data-tool="mdnice编辑器"><span>热图</span></h2><h3 data-tool="mdnice编辑器"><span>heatmapGene</span></h3><blockquote data-tool="mdnice编辑器"><p><span>探针DNA甲基化和单一基因表达之间的相关性热图。</span></p><p><span>To use this function you MAE object (input data) will need all probes and not only the distal ones.</span></p><p><span>This plot can be used to evaluate promoter, and intro, exons regions and closer distal probes of a gene to verify if their DNA methylation level is affecting the gene expression.</span></p></blockquote><p data-tool="mdnice编辑器"><span>全部参数：</span></p><pre data-tool="mdnice编辑器"><code><span>heatmapGene(<br>data,<br>group.col, <span>##colData(mae)调用第一步的结果</span><br>group1,<br>group2,<br>pairs,<br>GeneSymbol,<span>##比如"LAMB3"</span><br>scatter.plot = <span>TRUE</span>,<br>correlation.method = <span>"pearson"</span>,<br>correlation.table = <span>FALSE</span>,<br>annotation.col = <span>NULL</span>,<br>met.metadata = <span>NULL</span>,<br>exp.metadata = <span>NULL</span>,<br>dir.out = <span>"."</span>,<br>filter.by.probe.annotation = <span>TRUE</span>,<br>numFlankingGenes = <span>10</span>,<br>width = <span>10</span>,<br>height = <span>10</span>,<br>scatter.plot.width = <span>10</span>,<br>scatter.plot.height = <span>10</span>,<br>filename = <span>NULL</span><br>)<br></span></code></pre><pre data-tool="mdnice编辑器"><code><span>heatmapGene(data = data,<br>group.col = group.col,<br>group1 = group1,<br>group2 = group2,<br>pairs = pairs,<br>GeneSymbol = <span>"LAMB3"</span>,<br>height = <span>5</span>,<br>annotation.col = c(<span>"ethnicity"</span>,<span>"vital_status"</span>),<br>filename = <span>"heatmap.pdf"</span>)<br></span></code></pre><h3 data-tool="mdnice编辑器"><span>heatmapPairs</span></h3><blockquote data-tool="mdnice编辑器"><p><span>成对基因和探针反相关的热图</span></p></blockquote><pre data-tool="mdnice编辑器"><code><span>heatmapPairs(<br>data = data, <br>group.col = group.col,<br>group1 = group1,<br>group2 = group2,<br>annotation.col = c(<span>"ethnicity"</span>,<span>"vital_status"</span>,<span>"age_at_diagnosis"</span>),<br>pairs,<br>filename = <span>"heatmap.pdf"</span>,<br>height = <span>4</span>,<br>width = <span>11</span><br>)<br></span></code></pre><h2 data-tool="mdnice编辑器"><span>小结</span></h2><blockquote data-tool="mdnice编辑器"><p><span>总而言之，这个包目前囊括的功能可以概括为：</span></p><ul><li><section><p><span>差异甲基化分析</span></p></section></li><li><section><p><span>选择增强子探针</span></p></section></li><li><section><p><span>识别具有癌症特异性DNA甲基化变化的增强子探针</span></p></section></li><li><section><p><span>将有甲基化变化的增强子探针与有表达变化的目标基因联系起来</span></p></section></li><li><section><p><span>Motif 分析</span></p></section></li><li><section><p><span>将TF的表达与TF结合Motif 的甲基化联系起来</span></p></section></li><li><section><p><span>生存分析</span></p></section></li></ul></blockquote><p data-tool="mdnice编辑器"><span>今天就先抛砖引玉，大家可以先用TCGA的数据探索看看~</span></p></section><p><mp-style-type data-value="10000"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/FoEi0FiYOubUyVMZp-F0bw",target="_blank" rel="noopener noreferrer">原文链接</a>
