---
title: "转录组推文纠正--上游四套定量流程一网打尽"
date: 2023-03-08T04:57:59Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Acdemic"]
---
转录组推文纠正--上游四套定量流程一网打尽 by 生信菜鸟团
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><h2 data-tool="mdnice编辑器"><span></span><span>缘起</span><span></span></h2><p data-tool="mdnice编辑器">这周推文内容有一个重点需要先强调下，所以这周就先不进行可变剪切图形复现了哈。因为前几天，老师叫一个小伙伴检查了我的转录组系列推文里的代码，发现我前面的推文中存在两个错误。一个是<a href="https://mp.weixin.qq.com/s?__biz=MzUzMTEwODk0Ng==&amp;mid=2247506102&amp;idx=1&amp;sn=e48142d100b2bfd657d85e9841c96bdb&amp;scene=21#wechat_redirect" data-linktype="2">P值窜天高背后的原因探究及探究如何正确利用好富集分析的结果---（重磅干货的感觉）</a>的推文中处理组与对照组我弄反了，这个就先不关注了；另一个是转录组定量的上游流程中的人鼠双端定量流程有问题，存在报错，这是我们今天关注的重点。<strong>「我们将重新整理与书写转录组上游定量的人鼠单双端四套流程」</strong>，非常抱歉之前粗心以及不够严谨带来的报错，同时非常感谢那位小伙伴的检查提醒。<strong>「下面咱们就将这四套流程再次整理纠正一下，争取一网打尽」</strong>。</p><p data-tool="mdnice编辑器"><span>四套流程的分享按照：人单端+人双端+鼠单端+鼠双端的顺序分享</span></p><h2 data-tool="mdnice编辑器"><span></span><span>正式分析</span><span></span></h2><h3 data-tool="mdnice编辑器"><span></span><span><span></span>1.人单端测序</span><span></span></h3><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzUzMTEwODk0Ng==&amp;mid=2247507081&amp;idx=1&amp;sn=16c5a07d2917f28b23d14a3e15ecb99b&amp;scene=21#wechat_redirect" data-linktype="2">转录组上游定量分析其实真不难，4步可定（一）</a>这个流程运行是没有问题的，但是存在一个不影响结果的错误。featureCounts单端数据定量时不需要添加-p参数；-p代表双端测序，但是加了也不影响结果。<span>值得注意的是，曾老师说最近许多同学用如下的featureCounts -p定量命令会产生报错，这是因为版本差异问题。featureCounts的v2.0.1直接用-p是可以使用的，但featureCounts v2.0.3使用-p不添加--countReadPairs 会产生报错，具体差异可以自身用featureCounts  -h查看下其-p参数</span><span>，<strong>「需要加的同学一定要自己加上哈」</strong>。</span></p><pre data-tool="mdnice编辑器"><span></span><code><span>#</span><span> 一.数据准备：直接自己准备就好，此文用的是GSE180123数据集</span><br><span>#</span><span> 数据下载参考前面的转录组上游定量推文即可，rna小环境的搭建见之前的分享即可。</span><br><span><br>#</span><span> 二.质控过滤</span><br>conda activate rna #激活转录组测序数据处理的小环境<br><span>#</span><span> 01先进行fastqc</span><br>nohup fastqc -t 6 -o ./ SRR*.fastq.gz &gt;qc.log &amp;  <br><span>#</span><span> 02对fastqc后的zip数据进行multiqc</span><br>nohup multiqc ./*.zip -o ./ &gt; ./multiqc.log &amp;<br><span>#</span><span> 03trimmgalore</span><br>ls *gz |while read id;do (nohup trim_galore  -q 25 --phred33 --length 36 --stringency 3 -o ./  $id &amp; );done<br><span>#</span><span>质控后数据也需要用fastqc与multiqc看看质控效果</span><br>nohup fastqc -t 12 -o ./ SRR*_trimmed.fq.gz &gt;qc_trimmed.log &amp; <br>nohup multiqc *.zip -o ./ &gt; ./multiqc_t.log &amp;<br><span><br>#</span><span> 三比对</span><br>ls *_trimmed.fq.gz|while read id;do <br> gtf='$HOME/database/GRCh38.106/Homo_sapiens.GRCh38.106.chr.gtf'<br> hisat_index="$HOME/database/GRCh38.106/Hisat2Index/GRCh38.106"<br> nohup sh -c " hisat2 -p 2 -x ${hisat_index} -U $id 2&gt;${id%%_*}.log  | samtools sort -@ 2 -o ${id%%_*}.bam  " &amp; <br>done<br><span><br>#</span><span> 四 定量（单端不需要加-p）</span><br>gtf=$HOME/database/GRCh38.106/Homo_sapiens.GRCh38.106.chr.gtf<br>nohup featureCounts -T 5  -t exon -g gene_id  -a $gtf \<br>-o  all.id.txt  *bam  1&gt;counts.id.log 2&gt;&amp;1 &amp;<br></code></pre><h3 data-tool="mdnice编辑器"><span></span><span><span></span>2. 人双端测序</span><span></span></h3><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzUzMTEwODk0Ng==&amp;mid=2247507194&amp;idx=1&amp;sn=01e7618bb6e581cdc592b7b97a230bdd&amp;chksm=fa4519c7cd3290d1ac89c066d555868b95030a8f4dff7fdb0cb48b33679f9170594d38a66844&amp;cur_album_id=2545418429466607617&amp;scene=21&amp;key=1dc5edeba6113f2cd8f286ba60d06b5778e6336997a5c28ca5ca634128b479fc6330249eefc8ed1347c1027cce8c96cbbb91c3ebcfb58d4f7dc4e842014994d54580eeef71d343b3414a54119055efad3d73378a56afba51764575c1a178c2187720eef66d4f479521946ab56956a910c10f65ffa6fe9a5e67f77f2aaea208af&amp;ascene=1&amp;uin=MjI3OTYzNDcwNg==&amp;devicetype=Windows%2011%20x64&amp;version=6309001c&amp;lang=zh_CN&amp;countrycode=CN&amp;exportkey=n_ChQIAhIQw9SAEmemPjXooySejfI02BLqAQIE97dBBAEAAAAAAIbIAd1XSrkAAAAOpnltbLcz9gKNyK89dVj0m3WnjAmKx45G6EYKS/nmROkcLJJOXbFWq2h7OcMUlgJ59T6M9va7AsohE25SBnvb%20t9WwVvbNpCMYWrNvZdfYjhcEuOjwt/ZiRW/xgp0CIkpRMe7YHH0YWIHQioDF%20CWWOsjAdMbc4/H3Z7iSoQG9DbEbM2JrGDmKEQxW8LfbQzsnY7VgrIfLUe8fO4CufWUszQkxARyLl1NZu94MTj9lLIXWpRVnGylUJl6DEBCDKoiJbtuTTW2UP5Aqh9n/YWQgbHF6A==&amp;acctmode=0&amp;pass_ticket=/UZqoUKHYEci%20Hg9QzEGvpdmvPBFwtzGv/XueljVhtWA%20ycrdklHGXZ2TSmOKZ0vYigBgnq29lUM9PozWnyQdQ==&amp;wx_header=1&amp;fontgear=2#wechat_redirect" data-linktype="2">转录组上游定量分析其实真不难，4步可定（二）</a>中存在的错误是trimmgalore处，没有按照双端的方式去trim，而是按照单端的方式进行处理，导致后续报错。<span><strong>注双端trim后产生的文件是含有_val的，单端获得的文件是trimmed.fq.gz为后缀的文件</strong></span>。<span><strong>「单端与双端的区别，从trimmgalore处开始，Hisat2比对以及featureCounts处均存在区别」</strong></span>。具体流程如下：</p><pre data-tool="mdnice编辑器"><span></span><code><span>#</span><span> 一.数据准备：直接自己准备就好，此文用的是GSE173906数据集</span><br><span>#</span><span> 数据下载参考前面的转录组上游定量推文即可。</span><br><span><br>#</span><span> 二.质控过滤</span><br>conda activate rna #激活转录组测序数据处理的小环境<br><span>#</span><span> 01先进行fastqc</span><br>nohup fastqc -t 6 -o ./ SRR*.fastq.gz &gt;qc.log &amp;  <br><span>#</span><span> 02对fastqc后的zip数据进行multiqc</span><br>nohup multiqc ./*.zip -o ./ &gt; ./multiqc.log &amp;<br><span>#</span><span> 03trimmgalore</span><br>ls *gz |cut -d"_" -f 1 |uniq  &gt; sample.txt<br>cat sample.txt |while read id;do (nohup trim_galore  -q 25 --phred33 --length 36 --stringency 3 --paired  -o ./ ./${id}_1.fastq.gz  ./${id}_2.fastq.gz &amp; );done<br><span>#</span><span> 质控后数据也需要用fastqc与multiqc看看质控效果</span><br>nohup fastqc -t 12 -o ./ *val*.fq.gz &gt;qc_trimmed.log &amp; <br>nohup multiqc *.zip -o ./ &gt; ./multiqc_t.log &amp;<br><span><br>#</span><span> 三.比对</span><br>ls *_val_2.fq.gz |cut -d"_" -f 1 &gt;sample.txt  # SRR12060480_2_val_2.fq.gz<br>cat sample.txt |while read id;do <br> gtf='$HOME/database/GRCh38.106/Homo_sapiens.GRCh38.106.chr.gtf'<br> hisat_index="$HOME/database/GRCh38.106/Hisat2Index/GRCh38.106"<br> nohup sh -c " hisat2 -p 2 -x ${hisat_index}  -1 ${id}_1_val_1.fq.gz -2 ${id}_2_val_2.fq.gz  2&gt;${id%%_*}.log  | samtools sort -@ 2 -o ${id%%_*}.bam  " &amp; <br>done<br><span><br>#</span><span> 四.定量</span><br>gtf=$HOME/database/GRCh38.106/Homo_sapiens.GRCh38.106.chr.gtf<br>nohup featureCounts -T 5 -p -t exon -g gene_id  -a $gtf \<br>-o  all.id.txt  *bam  1&gt;counts.id.log 2&gt;&amp;1 &amp;<br></code></pre><h3 data-tool="mdnice编辑器"><span></span><span><span></span>3.鼠单端测序</span><span></span></h3><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzUzMTEwODk0Ng==&amp;mid=2247507891&amp;idx=1&amp;sn=57a46c4401c4fe425defb2ff493dd435&amp;chksm=fa451a8ecd329398e5d6f6135864a284df758e4c281eaeb93603699f634facbb9b994395858e&amp;cur_album_id=2545418429466607617&amp;scene=21&amp;key=f743c1b0b73aaeba80851cab2152b9c6ef448d5e7a0ca21281bf78d090c4bf596436bc8fa52ce9292137623d2b4f1ad6af0588bd667044c7f58e716e73e726f7a4f5cc2945fabf0c438d8bb3ff2b2265bbf56e49dfcab14e36e94aefc6babc2c16de860c2a9940df7b46753b018bfc7bc3a8e99d65b2e330aa64df031df8f1ca&amp;ascene=1&amp;uin=MjI3OTYzNDcwNg==&amp;devicetype=Windows%2011%20x64&amp;version=6309001c&amp;lang=zh_CN&amp;countrycode=CN&amp;exportkey=n_ChQIAhIQ6%20NCnjv7JG112S/UlWLr/RLqAQIE97dBBAEAAAAAAKomJTjwFJ8AAAAOpnltbLcz9gKNyK89dVj0ND1mRBIxTNAPzkZOwMPyOZ4/S9Wgs7RAvagm6qJrLKoqJOACzk6/MkJuoZ5RziqZSgFy02jk9ltXv8pr1oZJvdwM0q1r%20AiR%20xwkQSC79iWoCs7bizliwXwBcQpbUr63XAG9DJEKcnJJ9mqHb4axXF3QHcsZ4ilhx3Pd0Q8jHB%206skX2zdf4dzaDWrBgLq4qYrsCiodDlzhNR6MOI2GkNG0mjHR/TfCvKpz2tAN7wag160Q3jVuwC6sOAG/WZclKyI7OkQ==&amp;acctmode=0&amp;pass_ticket=/UZqoUKHYEci%20Hg9QzEGvpdmvPBFwtzGv/XueljVhtVg2FO81frYvmO1GbybmQSv6Cgy6bWqN5z9Ju1iOt9ifA==&amp;wx_header=1&amp;fontgear=2#wechat_redirect" data-linktype="2">转录组上游定量其实真不难，4步可定（三）</a>中存在两个问题1个是hisat2_index部分，当时大意了，直接定位的路径，忘记添加索引；2.featureCounts单端测序定量时不需要添加-p参数。</p><pre data-tool="mdnice编辑器"><span></span><code><span>#</span><span> 一.数据准备：直接自己准备就好，此文用的是GSE193168数据集</span><br><span>#</span><span> 数据下载参考前面的转录组上游定量推文即可。</span><br><span><br>#</span><span> 二.质控过滤</span><br>conda activate rna #激活转录组测序数据处理的小环境<br><span>#</span><span> 01先进行fastqc</span><br>nohup fastqc -t 6 -o ./ SRR*.fastq.gz &gt;qc.log &amp;  <br><span>#</span><span> 02对fastqc后的zip数据进行multiqc</span><br>nohup multiqc ./*.zip -o ./ &gt; ./multiqc.log &amp;<br><span>#</span><span> 03trimmgalore</span><br>ls *gz |while read id;do (nohup trim_galore  -q 25 --phred33 --length 36 --stringency 3 -o ./  $id &amp; );done<br><span>#</span><span>质控后数据也需要用fastqc与multiqc看看质控效果</span><br>nohup fastqc -t 12 -o ./ SRR*_trimmed.fq.gz &gt;qc_trimmed.log &amp; <br>nohup multiqc *.zip -o ./ &gt; ./multiqc_t.log &amp;<br><span><br>#</span><span> 三.比对</span><br>ls *_trimmed.fq.gz|while read id;do <br>gtf=$HOME/database/GRCm39.106/Mus_musculus.GRCm39.106.chr.gtf<br>hisat_index=$HOME/database/GRCm39.106/Hisat2Index/GRCm39.dna  <br>nohup sh -c " hisat2 -p 2 -x ${hisat_index} -U $id 2&gt;${id%%_*}.log  | samtools sort -@ 2 -o ${id%%_*}.bam  " &amp; <br>done<br><span><br>#</span><span> 四 定量</span><br>gtf=$HOME/database/GRCm39.106/Mus_musculus.GRCm39.106.chr.gtf<br>nohup featureCounts -T 5 -t exon -g gene_id  -a $gtf \<br>-o  all.id1.txt  *bam  1&gt;counts1.id.log 2&gt;&amp;1 &amp;<br><br></code></pre><h3 data-tool="mdnice编辑器"><span></span><span><span></span>4.鼠双端测序</span><span></span></h3><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzUzMTEwODk0Ng==&amp;mid=2247507973&amp;idx=1&amp;sn=9e394e7c745ded2eea670b85ccc17282&amp;chksm=fa456538cd32ec2e05f8b4c9d7bf8f2be336a0ec503e70228a42b76a096aa199c3e7e829546f&amp;cur_album_id=2545418429466607617&amp;scene=21&amp;key=a1e4a6001c806118f5ca80cf11ead45accedb4d767a878879d43ec706f45d5b2da298615ffcd2484f7bb17e6989a1e663b8ff0d57ee4556b2cb8e08342b974217e1a6c1f691ee9997cb5e1a1951bf1874eca374df3053c0ec3394b160f4f6373d6fab5be2ca007f3faff137219281d1a64cc344d26bbb8a917944bb6c0682825&amp;ascene=1&amp;uin=MjI3OTYzNDcwNg==&amp;devicetype=Windows%2011%20x64&amp;version=6309001c&amp;lang=zh_CN&amp;countrycode=CN&amp;exportkey=n_ChQIAhIQ0kUWmD3ngSpXT/nBsTmhoBLqAQIE97dBBAEAAAAAANSiNcRuRS0AAAAOpnltbLcz9gKNyK89dVj0pIV8HAQ9NpaNYujF0TYOf0E1tCW8Dm7BLDMCAYWNvnFvG5llvkkzSJcQttOBGHKLdDYmp90M7DXL/M4Xt/UkKaBHTmNsEq0MaiWO/tYJB5YxQufLb9wNX8ds0MSZewdsyaUsFh3cUP8fOQFj3ssJsd0QfOqdrpUb4okw4b7SypN0ksq%20yRg6C0282qOl6mgbeRq8FHc2jCYr0eUjlpoAiWEYTaZQpv1r8kFP/GOk/KpFVw8G7C2jTpIC7Eq0E9aHaw5Zxw==&amp;acctmode=0&amp;pass_ticket=/UZqoUKHYEci%20Hg9QzEGvpdmvPBFwtzGv/XueljVhtWUJogt3Rj/0b0i7TrV7NLI4tma2JYrWuwpLZ3RMyOswA==&amp;wx_header=1&amp;fontgear=2#wechat_redirect" data-linktype="2">转录组上游定量其实真不难，4步可定（四）之终结篇</a>存在的错误同人双端测序流程，也是trimmgalore处，没有按照双端的方式去trim，而是按照单端的方式进行处理，导致后续报错。再次强调下<span><strong>「单端与双端的区别，从trimmgalore处开始，Hisat2比对以及featureCounts处均存在区别」</strong>。<strong>「人与鼠流程的区别则是参考基因组索引与注释文件不同。」</strong></span>具体流程如下：</p><pre data-tool="mdnice编辑器"><span></span><code><span>#</span><span> 一.数据准备：直接自己准备就好，此文用的是GSE132902数据集</span><br><span>#</span><span> 数据下载参考前面的转录组上游定量推文即可。</span><br><span><br>#</span><span> 二.质控过滤</span><br>conda activate rna #激活转录组测序数据处理的小环境<br><span>#</span><span> 01先进行fastqc</span><br>nohup fastqc -t 6 -o ./ SRR*.fastq.gz &gt;qc.log &amp;  <br><span>#</span><span> 02对fastqc后的zip数据进行multiqc</span><br>nohup multiqc ./*.zip -o ./ &gt; ./multiqc.log &amp;<br><span>#</span><span> 03trimmgalore</span><br>ls *gz |cut -d"_" -f 1 |uniq  &gt; sample.txt<br>cat sample.txt |while read id;do (nohup trim_galore  -q 25 --phred33 --length 36 --stringency 3 --paired  -o ./ ./${id}_1.fastq.gz  ./${id}_2.fastq.gz &amp; );done<br><span>#</span><span> 质控后数据也需要用fastqc与multiqc看看质控效果</span><br>nohup fastqc -t 12 -o ./ *val*.fq.gz &gt;qc_trimmed.log &amp; <br>nohup multiqc *.zip -o ./ &gt; ./multiqc_t.log &amp;<br><span><br>#</span><span> 三.比对</span><br>ls *fq.gz|cut -d"_" -f 1 |uniq &gt;sample.id.txt<br>cat sample.id.txt |while read id;do <br> gtf='$HOME/database/GRCm39.106/Mus_musculus.GRCm39.106.chr.gtf '<br> hisat_index="$HOME/database/GRCm39.106/Hisat2Index/GRCm39.dna "<br> nohup sh -c " hisat2 -p 2 -x ${hisat_index}  -1 ${id}_1_val_1.fq.gz -2 ${id}_2_val_2.fq.gz  2&gt;${id%%_*}.log  | samtools sort -@ 2 -o ${id%%_*}.bam  " &amp; <br>done<br><span><br>#</span><span> 四.定量</span><br>gtf=$HOME/database/GRCm39.106/Mus_musculus.GRCm39.106.chr.gtf<br>nohup featureCounts -T 5 -p -t exon -g gene_id  -a $gtf \<br>-o  all.id.txt  *bam  1&gt;counts.id.log 2&gt;&amp;1 &amp;<br><br></code></pre><p data-tool="mdnice编辑器">以上就是转录组上游定量全流程了，感兴趣的同学可以去尝试一下。<span>值得注意的是，上面强调的trim的双端处理问题与featureCounts的版本问题一定要注意哈，尤其是你的featureCounts究竟是v2.0.1还是v2.0.3，v2.0.3的话一定要加--countReadPairs。</span><span></span></p></section><p><mp-style-type data-value="10000"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/1Fgc3BaS5aoZxENFLGGpWQ",target="_blank" rel="noopener noreferrer">原文链接</a>
