---
title: "ggplot2可视化拷贝数变异CNV的GISTIC score"
date: 2024-01-04T23:37:07Z
draft: ["false"]
tags: [
  "fetched",
  "医学和生信笔记"
]
categories: ["Acdemic"]
---
ggplot2可视化拷贝数变异CNV的GISTIC score by 医学和生信笔记
------
<div><section><span>关注公众号，发送</span><strong>R语言</strong><span>或</span><strong>python</strong><span>，可获取资料</span><span></span></section><section><mp-common-profile data-id="MzUzOTQzNzU0NA==" data-pluginname="mpprofile" data-headimg="http://mmbiz.qpic.cn/mmbiz_png/tpAC6lR84R9YDc8IDhqWAHTrZsMuhDpFlw4scqOl1ZVWpeY77cdibaSzPeGALfkEhdVpwHzVibHCRSYZg4csB43g/0?wx_fmt=png" data-nickname="医学和生信笔记" data-alias="yxhsxbj" data-signature="外科医生👨‍⚕️的R语言和生信学习🔖" data-from="0" data-is_biz_ban="0"></mp-common-profile></section><section><span></span></section><section data-role="outer" label="edit by 135editor"><section data-role="paragraph"><section data-role="outer" label="edit by 135editor"><section data-role="paragraph"><section data-role="outer" label="edit by 135editor"><section data-role="paragraph"><section data-role="outer"><section data-role="outer" label="edit by 135editor"><section data-tools="135编辑器" data-id="28"><p data-brushtype="text" hm_fix="440:185"><span>💡专注R语言在🩺生物医学中的使用</span></p></section></section></section></section><section data-role="paragraph"><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><p data-tool="mdnice编辑器">本期目录：</p><ul><li><p>数据准备</p></li><li><p>使用maftools画图</p></li><li><p>ggplot2画图</p></li><ul><li><p>基础知识</p></li><li><p>画图</p></li></ul></ul><p data-tool="mdnice编辑器">大家看文献时可能经常遇到各种<code>CNV gistic score</code>的可视化，都很好看，但是不知道怎么画出来的：<img data-ratio="0.36944444444444446" data-type="png" data-w="1080" data-src="https://mmbiz.qpic.cn/mmbiz_png/tpAC6lR84RictPbJGCVL7jyIiciaRBfyu6gxIbtTNBvuY5wzokBaia6Gic0ibr3FeDPxZ6TlfdGdNOaCibHykkV6RRTmQ/640?wx_fmt=png" src="https://mmbiz.qpic.cn/mmbiz_png/tpAC6lR84RictPbJGCVL7jyIiciaRBfyu6gxIbtTNBvuY5wzokBaia6Gic0ibr3FeDPxZ6TlfdGdNOaCibHykkV6RRTmQ/640?wx_fmt=png"></p><figure data-tool="mdnice编辑器"><img data-ratio="0.2611731843575419" data-type="png" data-w="716" data-src="https://mmbiz.qpic.cn/mmbiz_png/tpAC6lR84RictPbJGCVL7jyIiciaRBfyu6gNZ1Ka7OlpduI5yBulF2acnXsDP2ImCjSI7Is55ibGdpGKC16PP71MiaA/640?wx_fmt=png" src="https://mmbiz.qpic.cn/mmbiz_png/tpAC6lR84RictPbJGCVL7jyIiciaRBfyu6gNZ1Ka7OlpduI5yBulF2acnXsDP2ImCjSI7Is55ibGdpGKC16PP71MiaA/640?wx_fmt=png"><figcaption>DOI: 10.1038/ncomms6457</figcaption></figure><figure data-tool="mdnice编辑器"><img data-ratio="0.6993355481727574" data-type="png" data-w="602" data-src="https://mmbiz.qpic.cn/mmbiz_png/tpAC6lR84RictPbJGCVL7jyIiciaRBfyu6gUxpTgbBGdqLZfXp3KLQoDX3mUKxDXwnlsZ7MrSB24TCQniaeXM0L5yA/640?wx_fmt=png" src="https://mmbiz.qpic.cn/mmbiz_png/tpAC6lR84RictPbJGCVL7jyIiciaRBfyu6gUxpTgbBGdqLZfXp3KLQoDX3mUKxDXwnlsZ7MrSB24TCQniaeXM0L5yA/640?wx_fmt=png"><figcaption>doi:10.1038/nature20805</figcaption></figure><p data-tool="mdnice编辑器">GISTIC2会自动出一些结果，但是并不好看，而且扩增和删失是分开的：<img data-ratio="1.299847792998478" data-type="png" data-w="657" data-src="https://mmbiz.qpic.cn/mmbiz_png/tpAC6lR84RictPbJGCVL7jyIiciaRBfyu6gr3TPBfhw8vqcOOF02MJfF9N77qhWlIqv8uTs7brawiaaV0c6oKiaNs0g/640?wx_fmt=png" src="https://mmbiz.qpic.cn/mmbiz_png/tpAC6lR84RictPbJGCVL7jyIiciaRBfyu6gr3TPBfhw8vqcOOF02MJfF9N77qhWlIqv8uTs7brawiaaV0c6oKiaNs0g/640?wx_fmt=png"></p><p data-tool="mdnice编辑器">网络上也没找到怎么画，只能自己操作一下了！</p><h2 data-tool="mdnice编辑器"><span></span><span>数据准备</span><span></span></h2><p data-tool="mdnice编辑器">首先你要获得<code>GISTIC2.0</code>的输出结果，这是一个linux软件，得到的结果如下：<img data-ratio="2.384920634920635" data-type="png" data-w="252" data-src="https://mmbiz.qpic.cn/mmbiz_png/tpAC6lR84RictPbJGCVL7jyIiciaRBfyu6gdcmqbqvjAm5s9RrlCJUVVymltt5R4NKYGNMfys1iaCqthADVjIjPJ2w/640?wx_fmt=png" src="https://mmbiz.qpic.cn/mmbiz_png/tpAC6lR84RictPbJGCVL7jyIiciaRBfyu6gdcmqbqvjAm5s9RrlCJUVVymltt5R4NKYGNMfys1iaCqthADVjIjPJ2w/640?wx_fmt=png"></p><p data-tool="mdnice编辑器">至于这个软件怎么用，大家可以去百度一下~教程非常多，不过对于小白还是蛮复杂的！</p><h2 data-tool="mdnice编辑器"><span></span><span>使用maftools画图</span><span></span></h2><p data-tool="mdnice编辑器"><code>maftools</code>这个包可以做一些拷贝数变异的可视化，比如上面展示的那种图，但是画出来也不好看，也没有什么自定义选项，很明显是达不到各位的审美水平的。</p><pre data-tool="mdnice编辑器"><span></span><code><span>## 使用maftools分析</span><br><span>library</span>(maftools)<br><br>all.lesions &lt;- <span>"./TCGA_COAD_results/all_lesions.conf_90.txt"</span><br>amp.genes &lt;- <span>"./TCGA_COAD_results/amp_genes.conf_90.txt"</span><br>del.genes &lt;- <span>"./TCGA_COAD_results/del_genes.conf_90.txt"</span><br>scores.gis &lt;- <span>"./TCGA_COAD_results/scores.gistic"</span><br><br>coad.gistic = readGistic(gisticAllLesionsFile = all.lesions, <br>                         gisticAmpGenesFile = amp.genes, <br>                         gisticDelGenesFile = del.genes, <br>                         gisticScoresFile = scores.gis, isTCGA = <span>TRUE</span>)<br><span>## -Processing Gistic files..</span><br><span>## --Processing amp_genes.conf_90.txt</span><br><span>## --Processing del_genes.conf_90.txt</span><br><span>## --Processing scores.gistic</span><br><span>## --Summarizing by samples</span><br><br>gisticChromPlot(gistic = coad.gistic<br>                ,markBands = <span>"all"</span><br>                ,ref.build = <span>"hg38"</span><br>                )<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="1" data-type="png" data-w="504" data-src="https://mmbiz.qpic.cn/mmbiz_png/tpAC6lR84RictPbJGCVL7jyIiciaRBfyu6gAhBibJN0ibUIWCLicDC5tFItKoWHdicUjykFibLWSKGg0NooO5B08vicENOw/640?wx_fmt=png" src="https://mmbiz.qpic.cn/mmbiz_png/tpAC6lR84RictPbJGCVL7jyIiciaRBfyu6gAhBibJN0ibUIWCLicDC5tFItKoWHdicUjykFibLWSKGg0NooO5B08vicENOw/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code><span># gisticBubblePlot(gistic = coad.gistic)</span><br></code></pre><p data-tool="mdnice编辑器">这个图和文献里看到的还是差距很大的！下面我们学习下用<code>ggplot2</code>画图！</p><h2 data-tool="mdnice编辑器"><span></span><span>ggplot2画图</span><span></span></h2><h3 data-tool="mdnice编辑器"><span></span><span>基础知识</span><span></span></h3><p data-tool="mdnice编辑器">首先要了解这个图是什么意思，横坐标是染色体（或基因组？），纵坐标是G-Score，红色表示扩增，蓝色表示删失，如果要用<code>ggplot2</code>画这个图，那我们也要有这个数据才行！</p><p data-tool="mdnice编辑器">在GISTIC2.0的输出结果中，有一个<code>scores.gistic</code>的文件，我们可以用VScode打开看看：<img data-ratio="0.5139911634756995" data-type="png" data-w="679" data-src="https://mmbiz.qpic.cn/mmbiz_png/tpAC6lR84RictPbJGCVL7jyIiciaRBfyu6gQ16X8y2KIWqv8up4ch1fYiax6khCUK5j9IwjMa7Z1plzogaqSWAuduA/640?wx_fmt=png" src="https://mmbiz.qpic.cn/mmbiz_png/tpAC6lR84RictPbJGCVL7jyIiciaRBfyu6gQ16X8y2KIWqv8up4ch1fYiax6khCUK5j9IwjMa7Z1plzogaqSWAuduA/640?wx_fmt=png"></p><p data-tool="mdnice编辑器">看看它的列名，真是太巧了，竟然和我们需要的数据非常相似，有gistic score，也有染色体位置，还有扩增或者删失！</p><p data-tool="mdnice编辑器">但是这个文件理解还是需要一些基础知识的，说实话在学习画这个图之前我是不知道这些知识的，因为从来没用到过，所以也不会专门去学。。。</p><p data-tool="mdnice编辑器">首先这个染色体位置，就<code>Start/End</code>这两列，指的是在每一条染色体上的位置，第一条染色体上有3302046-3371973这个位置，那第2条，第3条等等都有这个位置区间，它并不是从0开始，一直过去的！</p><p data-tool="mdnice编辑器">那我们画图是需要从0开始的，所以我们就需要知道每一条染色体长度是多少，然后分别计算从0开始的位置坐标是多少！</p><p data-tool="mdnice编辑器">真的是让人头大，这个又是我的知识盲区了！所以我去了bioconductor找它的一些文档看看，因为我知道里面是有很多基因组的注释包这些东西的。通过半天的学习，我知道了<code>BSgenome</code>这个东西，还知道了人类的全基因组序列包<code>BSgenome.Hsapiens.UCSC.hg38</code>，这里面就有各个基因组的位置和长度信息。</p><p data-tool="mdnice编辑器">然后再继续学习下就知道<code>BSgenome</code>也是一个对象，可以通过特定函数提取信息。</p><p data-tool="mdnice编辑器">OK，下面就开始提取信息了！</p><pre data-tool="mdnice编辑器"><span></span><code>rm(list = ls())<br><span>library</span>(BSgenome.Hsapiens.UCSC.hg38) <span># 加载R包</span><br><span>## Loading required package: BSgenome</span><br><span>## Loading required package: BiocGenerics</span><br><span>## </span><br><span>## Attaching package: 'BiocGenerics'</span><br><span>## The following objects are masked from 'package:stats':</span><br><span>## </span><br><span>##     IQR, mad, sd, var, xtabs</span><br><span>## The following objects are masked from 'package:base':</span><br><span>## </span><br><span>##     anyDuplicated, append, as.data.frame, basename, cbind, colnames,</span><br><span>##     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,</span><br><span>##     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,</span><br><span>##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span><br><span>##     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,</span><br><span>##     union, unique, unsplit, which.max, which.min</span><br><span>## Loading required package: S4Vectors</span><br><span>## Loading required package: stats4</span><br><span>## </span><br><span>## Attaching package: 'S4Vectors'</span><br><span>## The following objects are masked from 'package:base':</span><br><span>## </span><br><span>##     expand.grid, I, unname</span><br><span>## Loading required package: IRanges</span><br><span>## </span><br><span>## Attaching package: 'IRanges'</span><br><span>## The following object is masked from 'package:grDevices':</span><br><span>## </span><br><span>##     windows</span><br><span>## Loading required package: GenomeInfoDb</span><br><span>## Loading required package: GenomicRanges</span><br><span>## Loading required package: Biostrings</span><br><span>## Loading required package: XVector</span><br><span>## </span><br><span>## Attaching package: 'Biostrings'</span><br><span>## The following object is masked from 'package:base':</span><br><span>## </span><br><span>##     strsplit</span><br><span>## Loading required package: rtracklayer</span><br></code></pre><p data-tool="mdnice编辑器">提取染色体名字及长度：</p><pre data-tool="mdnice编辑器"><span></span><code>df &lt;- data.frame(chromName = seqnames(BSgenome.Hsapiens.UCSC.hg38), <span># 染色体名字</span><br>                 chromlength = seqlengths(BSgenome.Hsapiens.UCSC.hg38)<span># 每条染色体长度</span><br>                 )<br>df$chromNum &lt;- <span>1</span>:length(df$chromName) <span># 染色体名字纯数字版，为了和scores.gistic里面的对应</span><br><br><span># 我们用的原始CNV文件是没有性染色体信息的</span><br>df &lt;- df[<span>1</span>:<span>22</span>,] <span># 只要前22条染色体信息</span><br><br>df<br><span>##       chromName chromlength chromNum</span><br><span>## chr1       chr1   248956422        1</span><br><span>## chr2       chr2   242193529        2</span><br><span>## chr3       chr3   198295559        3</span><br><span>## chr4       chr4   190214555        4</span><br><span>## chr5       chr5   181538259        5</span><br><span>## chr6       chr6   170805979        6</span><br><span>## chr7       chr7   159345973        7</span><br><span>## chr8       chr8   145138636        8</span><br><span>## chr9       chr9   138394717        9</span><br><span>## chr10     chr10   133797422       10</span><br><span>## chr11     chr11   135086622       11</span><br><span>## chr12     chr12   133275309       12</span><br><span>## chr13     chr13   114364328       13</span><br><span>## chr14     chr14   107043718       14</span><br><span>## chr15     chr15   101991189       15</span><br><span>## chr16     chr16    90338345       16</span><br><span>## chr17     chr17    83257441       17</span><br><span>## chr18     chr18    80373285       18</span><br><span>## chr19     chr19    58617616       19</span><br><span>## chr20     chr20    64444167       20</span><br><span>## chr21     chr21    46709983       21</span><br><span>## chr22     chr22    50818468       22</span><br><br>str(df)<br><span>## 'data.frame': 22 obs. of  3 variables:</span><br><span>##  $ chromName  : chr  "chr1" "chr2" "chr3" "chr4" ...</span><br><span>##  $ chromlength: int  248956422 242193529 198295559 190214555 181538259 170805979 159345973 145138636 138394717 133797422 ...</span><br><span>##  $ chromNum   : int  1 2 3 4 5 6 7 8 9 10 ...</span><br></code></pre><p data-tool="mdnice编辑器">然后就是计算从0开始的每条染色体位置坐标，就是简单的线段长度加减法，不过对于我这种好久不搞数学的人来说也是很费脑子的！</p><p data-tool="mdnice编辑器">在<code>scores.gistic</code>这个文件里，第一条染色体位置是从0开始的，所以不用怎么改，但是<strong>第2条染色体的<code>Start</code>的坐标，应该是再加上第一条染色体的长度才是我们需要的</strong>，以此类推，不断相加！</p><p data-tool="mdnice编辑器">所以我们先计算下每条染色体从0开始的起始坐标是多少！第一条染色体起始位置就是0，第二条起始位置是第一条长度的位置，第3条是前两条长度的位置，以此类推！</p><pre data-tool="mdnice编辑器"><span></span><code><span># 小发现，在用cumsum前要把int变成numeric</span><br>df$chromlengthCumsum &lt;- cumsum(as.numeric(df$chromlength)) <span># 染色体累加长度</span><br><br><span># 得到每条染色体从0开始的起始坐标</span><br>df$chormStartPosFrom0 &lt;- c(<span>0</span>,df$chromlengthCumsum[-nrow(df)])<br><br><span># 计算每条染色体中间位置坐标，用来最后加文字</span><br>tmp_middle &lt;- diff(c(<span>0</span>,df$chromlengthCumsum)) / <span>2</span><br>df$chromMidelePosFrom0 &lt;- df$chormStartPosFrom0 + tmp_middle<br><br>df<br><span>##       chromName chromlength chromNum chromlengthCumsum chormStartPosFrom0</span><br><span>## chr1       chr1   248956422        1         248956422                  0</span><br><span>## chr2       chr2   242193529        2         491149951          248956422</span><br><span>## chr3       chr3   198295559        3         689445510          491149951</span><br><span>## chr4       chr4   190214555        4         879660065          689445510</span><br><span>## chr5       chr5   181538259        5        1061198324          879660065</span><br><span>## chr6       chr6   170805979        6        1232004303         1061198324</span><br><span>## chr7       chr7   159345973        7        1391350276         1232004303</span><br><span>## chr8       chr8   145138636        8        1536488912         1391350276</span><br><span>## chr9       chr9   138394717        9        1674883629         1536488912</span><br><span>## chr10     chr10   133797422       10        1808681051         1674883629</span><br><span>## chr11     chr11   135086622       11        1943767673         1808681051</span><br><span>## chr12     chr12   133275309       12        2077042982         1943767673</span><br><span>## chr13     chr13   114364328       13        2191407310         2077042982</span><br><span>## chr14     chr14   107043718       14        2298451028         2191407310</span><br><span>## chr15     chr15   101991189       15        2400442217         2298451028</span><br><span>## chr16     chr16    90338345       16        2490780562         2400442217</span><br><span>## chr17     chr17    83257441       17        2574038003         2490780562</span><br><span>## chr18     chr18    80373285       18        2654411288         2574038003</span><br><span>## chr19     chr19    58617616       19        2713028904         2654411288</span><br><span>## chr20     chr20    64444167       20        2777473071         2713028904</span><br><span>## chr21     chr21    46709983       21        2824183054         2777473071</span><br><span>## chr22     chr22    50818468       22        2875001522         2824183054</span><br><span>##       chromMidelePosFrom0</span><br><span>## chr1            124478211</span><br><span>## chr2            370053187</span><br><span>## chr3            590297731</span><br><span>## chr4            784552788</span><br><span>## chr5            970429195</span><br><span>## chr6           1146601314</span><br><span>## chr7           1311677290</span><br><span>## chr8           1463919594</span><br><span>## chr9           1605686271</span><br><span>## chr10          1741782340</span><br><span>## chr11          1876224362</span><br><span>## chr12          2010405328</span><br><span>## chr13          2134225146</span><br><span>## chr14          2244929169</span><br><span>## chr15          2349446623</span><br><span>## chr16          2445611390</span><br><span>## chr17          2532409283</span><br><span>## chr18          2614224646</span><br><span>## chr19          2683720096</span><br><span>## chr20          2745250988</span><br><span>## chr21          2800828063</span><br><span>## chr22          2849592288</span><br></code></pre><p data-tool="mdnice编辑器">这样我们需要的信息基本都有了，接下来就可以读取文件进行操作了！</p><pre data-tool="mdnice编辑器"><span></span><code><span># 如果你不知道用哪个函数读取，多试几次就知道了！</span><br>scores &lt;- read.table(<span>"./TCGA_COAD_results/scores.gistic"</span>,sep=<span>"\t"</span>,header=<span>T</span>,stringsAsFactors = <span>F</span>)<br><br>head(scores)<br><span>##   Type Chromosome   Start     End X.log10.q.value.  G.score average.amplitude</span><br><span>## 1  Amp          1 3302046 3371973                0 0.021837          0.339205</span><br><span>## 2  Amp          1 3375059 3380822                0 0.021099          0.354653</span><br><span>## 3  Amp          1 3381074 3449929                0 0.020520          0.368893</span><br><span>## 4  Amp          1 3451503 3503571                0 0.022561          0.392242</span><br><span>## 5  Amp          1 3505022 4071958                0 0.022036          0.376773</span><br><span>## 6  Amp          1 4072066 4090117                0 0.022599          0.374114</span><br><span>##   frequency</span><br><span>## 1  0.036810</span><br><span>## 2  0.036810</span><br><span>## 3  0.034765</span><br><span>## 4  0.034765</span><br><span>## 5  0.032720</span><br><span>## 6  0.032720</span><br></code></pre><p data-tool="mdnice编辑器">每一个<code>G.score</code>都对应一个坐标，这样才能画图，但其实每一个<code>Amp</code>或者<code>Del</code>是一个区间，为了方便，我们就用起始坐标代替了，当然你也可以用中点、结束点表示。</p><pre data-tool="mdnice编辑器"><span></span><code>chromID &lt;- scores$Chromosome<br><br>scores$StartPos &lt;- scores$Start + df$chormStartPosFrom0[chromID]<br>scores$EndPos &lt;- scores$End + df$chormStartPosFrom0[chromID]<br></code></pre><p data-tool="mdnice编辑器">我们得到的<code>G.score</code>都是正数，需要把<code>Del</code>的<code>G.score</code>变成负数。</p><pre data-tool="mdnice编辑器"><span></span><code>range(scores$G.score)<br><span>## [1] 0.000000 0.564793</span><br><br>scores[scores$Type == <span>"Del"</span>, <span>"G.score"</span>] &lt;- scores[scores$Type == <span>"Del"</span>, <span>"G.score"</span>] * -<span>1</span><br><br>range(scores$G.score)<br><span>## [1] -0.564793  0.280607</span><br></code></pre><p data-tool="mdnice编辑器">真的是很复杂！有没有大佬知道简单点的方法啊，求告知！</p><h3 data-tool="mdnice编辑器"><span></span><span>画图</span><span></span></h3><pre data-tool="mdnice编辑器"><span></span><code><span>library</span>(ggplot2)<br><span>library</span>(ggsci)<br><br>ggplot(scores, aes(StartPos, G.score))+<br>  geom_area(aes(group=Type, fill=factor(Type,levels = c(<span>"Del"</span>,<span>"Amp"</span>))))+<br>  scale_fill_lancet(guide=guide_legend(reverse = <span>T</span>))+<br>  geom_vline(data = df ,mapping=aes(xintercept=chromlengthCumsum),linetype=<span>2</span>)+<br>  geom_text(data = df,aes(x=chromMidelePosFrom0,y=<span>0.2</span>,label=chromName))+<br>  scale_x_continuous(expand = c(<span>0</span>,-<span>1000</span>),limits = c(<span>0</span>,<span>2.9e9</span>))+<br>  ylim(-<span>0.3</span>,<span>0.3</span>)+<br>  theme_minimal()<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.5" data-type="png" data-w="864" data-src="https://mmbiz.qpic.cn/mmbiz_png/tpAC6lR84RictPbJGCVL7jyIiciaRBfyu6g7KT3NnyiagsmgYnFqUfibQsgf24UnItwLg5te1BZosA0WQcoR5IAyUmg/640?wx_fmt=png" src="https://mmbiz.qpic.cn/mmbiz_png/tpAC6lR84RictPbJGCVL7jyIiciaRBfyu6g7KT3NnyiagsmgYnFqUfibQsgf24UnItwLg5te1BZosA0WQcoR5IAyUmg/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">文字有重叠，我们增加一点错落感。</p><pre data-tool="mdnice编辑器"><span></span><code>df$ypos &lt;- rep(c(<span>0.2</span>,<span>0.25</span>),<span>11</span>)<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>ggplot(scores, aes(StartPos, G.score))+<br>  geom_area(aes(group=Type, fill=factor(Type,levels = c(<span>"Del"</span>,<span>"Amp"</span>))))+<br>  scale_fill_lancet(guide=guide_legend(reverse = <span>T</span>),name=<span>"Type"</span>)+<br>  geom_vline(data = df ,mapping=aes(xintercept=chromlengthCumsum),linetype=<span>2</span>)+<br>  geom_text(data = df,aes(x=chromMidelePosFrom0,y=ypos,label=chromName))+<br>  scale_x_continuous(expand = c(<span>0</span>,-<span>1000</span>),limits = c(<span>0</span>,<span>2.9e9</span>),name = <span>NULL</span>,labels = <span>NULL</span>)+<br>  ylim(-<span>0.3</span>,<span>0.3</span>)+<br>  theme_minimal()+<br>  theme(legend.position = <span>"top"</span>,<br>        axis.text.y = element_text(color = <span>"black"</span>,size = <span>14</span>),<br>        axis.title.y = element_text(color = <span>"black"</span>,size = <span>16</span>)<br>        )<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.5" data-type="png" data-w="864" data-src="https://mmbiz.qpic.cn/mmbiz_png/tpAC6lR84RictPbJGCVL7jyIiciaRBfyu6gibYTPa1ibm2ibqF0mHpia2PIfSMyYvicwz0wko1cpibrDoIViao9FubFDRF6A/640?wx_fmt=png" src="https://mmbiz.qpic.cn/mmbiz_png/tpAC6lR84RictPbJGCVL7jyIiciaRBfyu6gibYTPa1ibm2ibqF0mHpia2PIfSMyYvicwz0wko1cpibrDoIViao9FubFDRF6A/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">这就成了！</p><p data-tool="mdnice编辑器">同理可画<code>frequency</code>，这里就不演示了！</p><blockquote data-tool="mdnice编辑器"><span>“</span><p>画图3分钟，准备数据3小时！</p></blockquote><p data-tool="mdnice编辑器">本次示例使用的数据是<code>TCGA-COAD的Masked Copy Number Segment</code>，经过<code>GISTIC2.0</code>软件得到的，大家完全可以自己搞出来。</p><p data-tool="mdnice编辑器">如果你想获得我的数据，底部赞赏5块，截图通过微信发我即可~</p><p data-tool="mdnice编辑器">欢迎加入交流群交流讨论~</p></section></section></section></section><hr><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><p data-tool="mdnice编辑器"><span>获取更多信息，欢迎加入🐧<strong>QQ交流群：613637742</strong></span></p><hr></section><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><blockquote data-tool="mdnice编辑器"><span>“</span><p>医学和生信笔记，专注R语言在临床医学中的使用、R语言数据分析和可视化。主要分享R语言做<strong>医学统计学、meta分析、网络药理学、临床预测模型、机器学习、生物信息学</strong>等。</p></blockquote></section><section powered-by="xmyeditor.com"><section powered-by="xmyeditor.com"><section powered-by="xmyeditor.com"><section powered-by="xmyeditor.com"><br></section></section></section><section powered-by="xmyeditor.com"><p><span><strong>往期回顾</strong></span></p></section><section powered-by="xmyeditor.com"><section powered-by="xmyeditor.com"><section powered-by="xmyeditor.com"><br></section></section><section powered-by="xmyeditor.com"><section powered-by="xmyeditor.com"><br></section></section></section><section data-type="lspecial01" data-tools="小蚂蚁编辑器" powered-by="xmyeditor.com"><section><p><br></p><section><a href="http://mp.weixin.qq.com/s?__biz=MzUzOTQzNzU0NA==&amp;mid=2247492480&amp;idx=1&amp;sn=cf05a155c221bc6097db446b688d2ce2&amp;chksm=facad307cdbd5a1199bb1902f12137ae8772e0464b5304ffbfe42fd4893c4995bf740a4a59a2&amp;scene=21#wechat_redirect" data-linktype="1"><section><section><section data-mid=""><span data-positionback="static"><img data-ratio="0.42568807339449544" data-w="545" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/tpAC6lR84RicKWIDu42W2GCXIuVykLQMMKO9mQ0jENfgG5hCRkroDPicXI0Ys0ia4ibr84ibWVRhH0WL1hFzcoOJxFg/640?wx_fmt=jpeg" src="https://mmbiz.qpic.cn/mmbiz_jpg/tpAC6lR84RicKWIDu42W2GCXIuVykLQMMKO9mQ0jENfgG5hCRkroDPicXI0Ys0ia4ibr84ibWVRhH0WL1hFzcoOJxFg/640?wx_fmt=jpeg"></span></section><section data-mid=""><p>手动下载的TCGA数据也是可以用TCGAbiolinks包整理的</p></section></section></section></a></section><p><br></p><section><a href="http://mp.weixin.qq.com/s?__biz=MzUzOTQzNzU0NA==&amp;mid=2247492462&amp;idx=1&amp;sn=0773b4ad0ecdd7e3cf206cd611249399&amp;chksm=facad3e9cdbd5aff246e43133b012d0eee40ccd71e15dd97e3fa44940f0d838e082518ef937f&amp;scene=21#wechat_redirect" data-linktype="1"><section><section><section data-mid=""><span data-positionback="static"><img data-ratio="0.42444444444444446" data-w="900" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/tpAC6lR84R9teEZqNruhJTI9To9ia5KDiav4GeLMpW1hiaCJ5V3XLcrcNRFVrXyNRxicZddNB6bLb55jT6qyficFUhA/640?wx_fmt=jpeg" src="https://mmbiz.qpic.cn/mmbiz_jpg/tpAC6lR84R9teEZqNruhJTI9To9ia5KDiav4GeLMpW1hiaCJ5V3XLcrcNRFVrXyNRxicZddNB6bLb55jT6qyficFUhA/640?wx_fmt=jpeg"></span></section><section data-mid=""><p>新版TCGAbiolinks包学习：表达矩阵提取（mRNA/lncRNA/counts/tpm/fpkm）</p></section></section></section></a></section><p><br></p><section><a href="http://mp.weixin.qq.com/s?__biz=MzUzOTQzNzU0NA==&amp;mid=2247492481&amp;idx=1&amp;sn=8690d9a2bf38bf93d8e63240a4b30247&amp;chksm=facad306cdbd5a10a483d8dcc9c67a7b47c69c17e5a6600a24d6c9dca010b957aac8424aa233&amp;scene=21#wechat_redirect" data-linktype="1"><section><section><section data-mid=""><span data-positionback="static"><img data-ratio="0.42444444444444446" data-w="900" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/tpAC6lR84RibL75iav1TBII0nibWuWKv9ibTswqYH4zcFecDw7ggenu2JLMyqORNBdWtfHTEmafn6IyBOoQnJ4ceZw/640?wx_fmt=jpeg" src="https://mmbiz.qpic.cn/mmbiz_jpg/tpAC6lR84RibL75iav1TBII0nibWuWKv9ibTswqYH4zcFecDw7ggenu2JLMyqORNBdWtfHTEmafn6IyBOoQnJ4ceZw/640?wx_fmt=jpeg"></span></section><section data-mid=""><p>树状数据/层次数据可视化</p></section></section></section></a></section><p><br></p><section><a href="http://mp.weixin.qq.com/s?__biz=MzUzOTQzNzU0NA==&amp;mid=2247488672&amp;idx=1&amp;sn=8fda840f5dcdc6facd426e18cc74471d&amp;chksm=fac92427cdbead3125edc332128a54b77773ce759a63522b5ba1fcf0bbf7301fdd09bd5c55c7&amp;scene=21#wechat_redirect" data-linktype="1"><section><section><section data-mid=""><span data-positionback="static"><img data-ratio="0.42444444444444446" data-type="jpeg" data-w="900" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/tpAC6lR84RicbTibkNJTDTriaWDl7TbibawxPFp6Gg1icTv5yiaoYzVia8IpxpOwyJsfqc7OUkHjZfN1euglRiczj1A5Hg/640?wx_fmt=jpeg" src="https://mmbiz.qpic.cn/mmbiz_jpg/tpAC6lR84RicbTibkNJTDTriaWDl7TbibawxPFp6Gg1icTv5yiaoYzVia8IpxpOwyJsfqc7OUkHjZfN1euglRiczj1A5Hg/640?wx_fmt=jpeg"></span></section><section data-mid=""><p>使用R语言画<em>森林图</em>和误差线（合辑）</p></section></section></section></a></section><p><br></p><section><a href="http://mp.weixin.qq.com/s?__biz=MzUzOTQzNzU0NA==&amp;mid=2247488361&amp;idx=1&amp;sn=7dff393441921492257487734c792c51&amp;chksm=fac923eecdbeaaf872efa8319a954f112938a243dba77a93a77315225f53d173515602f48b4e&amp;scene=21#wechat_redirect" data-linktype="1"><section><section><section data-mid=""><span data-positionback="static"><img data-ratio="0.4246031746031746" data-type="jpeg" data-w="504" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/tpAC6lR84R97icwvvu58d1lAMog9oSW5HMicMB7rD8b2XEHU79piaVWNaiaRq7ibeUVia9meicyIYwWWgd0cBPCdWzOkQ/640?wx_fmt=jpeg" src="https://mmbiz.qpic.cn/mmbiz_jpg/tpAC6lR84R97icwvvu58d1lAMog9oSW5HMicMB7rD8b2XEHU79piaVWNaiaRq7ibeUVia9meicyIYwWWgd0cBPCdWzOkQ/640?wx_fmt=jpeg"></span></section><section data-mid=""><p><em>corrplot</em>包可视化相关矩阵详解</p></section></section></section></a></section><p><br></p><section><a href="http://mp.weixin.qq.com/s?__biz=MzUzOTQzNzU0NA==&amp;mid=2247492249&amp;idx=1&amp;sn=ef3565c983840ae777535170f147d0ff&amp;chksm=facad21ecdbd5b0805a7cac1fe9fb7be46586b71262b88f61642b57d3352582205c760e3c7d8&amp;scene=21#wechat_redirect" data-linktype="1"><section><section><section data-mid=""><span data-positionback="static"><img data-ratio="0.42444444444444446" data-w="900" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/tpAC6lR84RibSIILytvkjlVydD6ibrLYGJrgYyMic637x9VTt6qXR128tmapEzTHXOyusARfXXEiaD5iciaFFLibfTtmw/640?wx_fmt=jpeg" src="https://mmbiz.qpic.cn/mmbiz_jpg/tpAC6lR84RibSIILytvkjlVydD6ibrLYGJrgYyMic637x9VTt6qXR128tmapEzTHXOyusARfXXEiaD5iciaFFLibfTtmw/640?wx_fmt=jpeg"></span></section><section data-mid=""><p>使用<em>mlr3</em>搞定二分类资料的多个模型评价和比较</p></section></section></section></a></section><p><br></p></section></section></section></section></section></section></section></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/flL_AdFMSKCEDwAcrKIesA",target="_blank" rel="noopener noreferrer">原文链接</a>
