---
title: "seurat v5全流程—harmmony整合+标准分析+细胞注释+批量差异、富集分析(seurat读取多个txt文件)"
date: 2024-01-13T13:12:52Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Acdemic"]
---
seurat v5全流程—harmmony整合+标准分析+细胞注释+批量差异、富集分析(seurat读取多个txt文件) by 生信菜鸟团
------
<div><p><strong><span>大家好</span></strong><strong><span>，本推文</span></strong><strong><span>是为了测试流程的代码，我在Jimmy老师的代码中比较难理解的地方做了注释，富集分析部分做了魔改，欢迎点赞收藏学习。后续还会加上cellchat细胞通讯、monocle拟时序分析、pyscenic转录因子分析。</span></strong></p><section><mp-common-profile data-pluginname="mpprofile" data-id="Mzg2NDcxMzYwNg==" data-headimg="http://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345Sks7JhVxUX46ZKxPSob6ptqIZgnIEnHOn5VMwCX8sN6MQy1Pq4XXFEOJ6grAmsoQugyCDKOZictDBHA/0?wx_fmt=png" data-nickname="生信小博士" data-alias="bioinformatics_Dr" data-signature="【生物信息学】R语言，学习生信，seurat，单细胞测序，空间转录组。 Python，scanpy，cell2location，资料分享" data-from="2" data-is_biz_ban="0"></mp-common-profile></section><p><strong><span>后面会有一场直播：如何学习、使用本代码。可以关注下。</span></strong><br></p><p><img data-galleryid="" data-imgfileid="100036002" data-ratio="0.5159313725490197" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SktL8QHWsQia4vf0kVbzSQXyMq4j3s1Pe329DetibGUE0nbt3GgHtLqy5MehcicKGzrn3vmVM3Yic4YibeQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="816" src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SktL8QHWsQia4vf0kVbzSQXyMq4j3s1Pe329DetibGUE0nbt3GgHtLqy5MehcicKGzrn3vmVM3Yic4YibeQ/640?wx_fmt=png&amp;from=appmsg"></p><p><br></p><p><strong><span>背景：</span></strong><span>升主动脉瘤（ATAA）是由主动脉壁逐渐变弱和扩张引起的，可能导致主动脉夹层、破裂和其他危及生命的并发症。为了提高我们对ATAA发病机制的理解，我们试图全面描述升主动脉壁的细胞组成，并鉴定人类ATAA组织中每个细胞群体的分子变化。</span></p><p><strong><span>方法</span></strong><span>：我们对来自11名研究参与者的升主动脉组织进行了单细胞RNA测序（sc-RNAseq）分析，其中包括8名ATAA患者（4名女性和4名男性）和3名对照组（2名女性和1名男性）。利用sc-RNAseq数据对从主动脉组织中提取的细胞进行分析和分类，以执行聚类鉴定。然后通过比较ATAA和对照组组织中每个细胞类型的比例和基因表达谱来检查与ATAA相关的变化。我们还通过将我们的sc-RNAseq数据与公开的全基因组关联研究（GWAS）数据进行整合分析，确定可能对ATAA至关重要的基因。</span></p><p><strong><span>结果：</span></strong><span>我们鉴定了人类升主动脉组织中的11种主要细胞类型；对这些细胞进行高分辨率的再聚类将其细分为40个亚型。平滑肌细胞、巨噬细胞和T淋巴细胞观察到多个亚型，表明这些细胞在主动脉壁中具有多个功能性群体。一般而言，ATAA组织中非免疫细胞较少，免疫细胞尤其是T淋巴细胞较多，与对照组组织相比。差异基因表达数据提示ATAA组织中存在广泛的线粒体功能障碍。此外，我们的sc-RNAseq数据与公开的GWAS数据和启动子捕获Hi-C数据的整合分析表明ERG（ETS相关基因）在维持正常主动脉壁功能中发挥重要作用。</span></p><p><span><br></span></p><ul><li><p><strong><span>在这篇文章其中，我们使用其中两个样本进行本次全流程代码测试：</span></strong></p></li></ul><pre tabindex="0" aria-label="Console Output" role="document"><strong><span>GSM4704934</span></strong></pre><p><strong><span>GSM4704931</span></strong></p><p><strong><span><br></span></strong></p><hr><pre tabindex="0" aria-label="Console Output" role="document"><strong></strong></pre><ul><li><p><strong><span>根据GEO的分组信息，我们可以看到GSM4704931是control组，那另外一个就是ATAA组</span></strong></p></li></ul><p><img data-galleryid="" data-imgfileid="100035999" data-ratio="0.6661742983751846" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SktL8QHWsQia4vf0kVbzSQXyMg4v8q6ZEfEL1BfqYVBKBdNdIz8OVVCiagfkjNjbCFqTfRbddrNfBmUQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="677" src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SktL8QHWsQia4vf0kVbzSQXyMg4v8q6ZEfEL1BfqYVBKBdNdIz8OVVCiagfkjNjbCFqTfRbddrNfBmUQ/640?wx_fmt=png&amp;from=appmsg"></p><p><br></p><hr><ul><li><p><strong><span>本推文主要分为四个部分：<br></span></strong></p></li></ul><h3></h3><ol><li><h3><strong><span>第一步：导入数据，harmony整合样本 降维聚类分群后，细胞注释</span></strong></h3></li><li><h3><strong><span>第二步：基于cluster或者celltype的批量差异分析，批量绘制火山图</span></strong></h3></li><li><h3><strong><span>第三步：对差异基因进行批量富集分析：GO、KEGG</span></strong></h3></li><li><h3><strong><span>第四步：展示不同组别中的细胞比例及其他基础图形</span></strong></h3></li></ol><h3><span>第一步：导入数据，harmony整合样本 降维聚类分群后，细胞注释</span></h3><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="perl"><code><span>## </span></code><code><span>### ---------------</span></code><code><span>###</span></code><code><span>### Create: Jianming Zeng</span></code><code><span>### Date:  2023-01-16 20:53:22</span></code><code><span>### Email: jmzeng1314@163.com</span></code><code><span>### Blog: http://www.bio-info-trainee.com/</span></code><code><span>### Forum:  http://www.biotrainee.com/thread-1376-1-1.html</span></code><code><span>### CAFS/SUSTC/Eli Lilly/University of Macau</span></code><code><span>### Update Log: 2023-01-16  First version </span></code><code><span>###</span></code><code><span>### ---------------</span></code><code><span><br></span></code><code><span>#########################设置我自己的r包加载路径，通常你不需要运行这段代码-----</span></code><code><span>.libPaths(c(</span></code><code><span>  '/home/rootyll/seurat_v5/',</span></code><code><span>  "/usr/local/lib/R/site-library", "/usr/lib/R/site-library", "/usr/lib/R/library"</span></code><code><span>));print(getwd())</span></code><code><span>###################################################################################333</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span>rm(list=ls())</span></code><code><span>options(stringsAsFactors = F) </span></code><code><span>source('scRNA_scripts/lib.R')</span></code><code><span><br></span></code><code><span>###### step1:导入数据 ######   </span></code><code><span># 付费环节 800 元人民币 </span></code><code><span># https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE155468</span></code><code><span>dir='inputs/' </span></code><code><span>samples=list.files( dir ,pattern = 'gz')</span></code><code><span>samples </span></code><code><span>library(data.table)</span></code><code><span>ctList = lapply(samples,function(pro){ </span></code><code><span>  # pro=samples[1] </span></code><code><span>  print(pro)</span></code><code><span>  ct=fread(file.path( dir ,pro),data.table = F)</span></code><code><span>  ct[1:4,1:4]</span></code><code><span>  rownames(ct)=ct[,1]</span></code><code><span>  colnames(ct) = paste(gsub('.txt.gz','',pro),</span></code><code><span>                       colnames(ct) ,sep = '_')</span></code><code><span>  ct=ct[,-1] </span></code><code><span>  return(ct)</span></code><code><span>})</span></code><code><span><br></span></code><code><span>#1检查行列数目，目的是找交集基因-----</span></code><code><span>lapply(ctList, dim)</span></code><code><span><br></span></code><code><span>tmp =table(unlist(lapply(ctList, rownames)));head(tmp)</span></code><code><span><br></span></code><code><span>cg = names(tmp)[tmp==length(samples)];head(cg)</span></code><code><span><br></span></code><code><span>#2合并矩阵-----</span></code><code><span>bigct = do.call(cbind,</span></code><code><span>                lapply(ctList,function(ct){ </span></code><code><span>                  ct = ct[cg,] </span></code><code><span>                  return(ct)</span></code><code><span>                }))</span></code><code><span>dim(bigct)</span></code><code><span><br></span></code><code><span>sce.all=CreateSeuratObject(counts =  bigct, </span></code><code><span>                           min.cells = 5,</span></code><code><span>                           min.features = 300)</span></code><code><span>sce.all</span></code><code><span>as.data.frame(sce.all@assays$RNA$counts[1:10, 1:2])</span></code><code><span>head(sce.all@meta.data, 10)</span></code><code><span>table(sce.all@meta.data$orig.ident) </span></code><code><span><br></span></code><code><span><br></span></code><code><span>###### step2:QC质控 ######</span></code><code><span>dir.create("./1-QC")</span></code><code><span>setwd("./1-QC")</span></code><code><span># 如果过滤的太狠，就需要去修改这个过滤代码</span></code><code><span>source('../scRNA_scripts/qc.R')</span></code><code><span>sce.all.filt = basic_qc(sce.all)</span></code><code><span>print(dim(sce.all))</span></code><code><span>print(dim(sce.all.filt))</span></code><code><span>setwd('../')</span></code><code><span><br></span></code><code><span><br></span></code><code><span>sp='human'</span></code><code><span><br></span></code><code><span>###### step3: harmony整合多个单细胞样品 ######</span></code><code><span>dir.create("2-harmony")</span></code><code><span>getwd()</span></code><code><span>setwd("2-harmony")</span></code><code><span>source('../scRNA_scripts/harmony.R')</span></code><code><span># 默认 ScaleData 没有添加"nCount_RNA", "nFeature_RNA"</span></code><code><span># 默认的</span></code><code><span>sce.all.int = run_harmony(sce.all.filt)</span></code><code><span>setwd('../')</span></code><code><span><br></span></code><code><span><br></span></code><code><span>###### step4: 降维聚类分群和看标记基因库 ######</span></code><code><span># 原则上分辨率是需要自己肉眼判断，取决于个人经验</span></code><code><span># 为了省力，我们直接看  0.1和0.8即可</span></code><code><span>table(Idents(sce.all.int))</span></code><code><span>table(sce.all.int$seurat_clusters)</span></code><code><span>table(sce.all.int$RNA_snn_res.0.1) </span></code><code><span>table(sce.all.int$RNA_snn_res.0.8) </span></code><code><span><br></span></code><code><span>getwd()</span></code><code><span>dir.create('check-by-0.1')</span></code><code><span>setwd('check-by-0.1')</span></code><code><span>sel.clust = "RNA_snn_res.0.1"</span></code><code><span>sce.all.int &lt;- SetIdent(sce.all.int, value = sel.clust)</span></code><code><span>table(sce.all.int@active.ident) </span></code><code><span><br></span></code><code><span>source('../scRNA_scripts/check-all-markers.R')</span></code><code><span>setwd('../') </span></code><code><span>getwd()</span></code><code><span><br></span></code><code><span>print(getwd())</span></code><code><span>dir.create('check-by-0.8')</span></code><code><span>setwd('check-by-0.8')</span></code><code><span>sel.clust = "RNA_snn_res.0.8"</span></code><code><span>sce.all.int &lt;- SetIdent(sce.all.int, value = sel.clust)</span></code><code><span>table(sce.all.int@active.ident) </span></code><code><span>source('../scRNA_scripts/check-all-markers.R')</span></code><code><span>setwd('../') </span></code><code><span>getwd()</span></code><code><span><br></span></code><code><span>last_markers_to_check</span></code><code><span><br></span></code><code><span><br></span></code><code><span>###### step5: 确定单细胞亚群生物学名字 ######</span></code><code><span># 一般来说，为了节省工作量，我们选择0.1的分辨率进行命名</span></code><code><span># 因为命名这个步骤是纯人工 操作</span></code><code><span># 除非0.1确实分群太粗狂了，我们就选择0.8 </span></code><code><span>source('scRNA_scripts/lib.R')</span></code><code><span>#sce.all.int = readRDS('2-harmony/sce.all_int.rds')</span></code><code><span>colnames(sce.all.int@meta.data) </span></code><code><span>table(sce.all.int$RNA_snn_res.0.8)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span># 付费环节 800 元人民币</span></code><code><span>print(getwd())</span></code><code><span>if(T){</span></code><code><span>  sce.all.int</span></code><code><span>  table(Idents(sce.all.int))</span></code><code><span>  #选择res0.1 分辨率，进行细胞亚群命名-----</span></code><code><span>  Idents(sce.all.int)=sce.all.int$RNA_snn_res.0.1</span></code><code><span>  table(Idents(sce.all.int))</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span>  celltype=data.frame(ClusterID=0:8 ,</span></code><code><span>                      celltype= 0:8 ) </span></code><code><span>  #定义细胞亚群        </span></code><code><span>  celltype[celltype$ClusterID %in% c( 0, 8 ),2]='Tcells'</span></code><code><span>  celltype[celltype$ClusterID %in% c(3 ),2]='NK'   </span></code><code><span><br></span></code><code><span>  celltype[celltype$ClusterID %in% c( 7 ),2]='Bcells' </span></code><code><span>  celltype[celltype$ClusterID %in% c( 2,5 ),2]='myeloids'  </span></code><code><span>  celltype[celltype$ClusterID %in% c(6,1 ),2]='fibro'</span></code><code><span>  celltype[celltype$ClusterID %in% c( 4 ),2]='endo'  </span></code><code><span><br></span></code><code><span><br></span></code><code><span>  head(celltype)</span></code><code><span>  celltype</span></code><code><span>  table(celltype$celltype)</span></code><code><span>  sce.all.int@meta.data$celltype = "NA"</span></code><code><span><br></span></code><code><span>  for(i in 1:nrow(celltype)){</span></code><code><span>    sce.all.int@meta.data[which(sce.all.int@meta.data$RNA_snn_res.0.1 == celltype$ClusterID[i]),'celltype'] &lt;- celltype$celltype[i]}</span></code><code><span>  Idents(sce.all.int)=sce.all.int$celltype</span></code><code><span><br></span></code><code><span>  sel.clust = "celltype"</span></code><code><span>  sce.all.int &lt;- SetIdent(sce.all.int, value = sel.clust)</span></code><code><span>  table(sce.all.int@active.ident) </span></code><code><span><br></span></code><code><span>  dir.create('check-by-celltype')</span></code><code><span>  setwd('check-by-celltype')</span></code><code><span>  source('../scRNA_scripts/check-all-markers.R')</span></code><code><span>  setwd('../') </span></code><code><span>  getwd()</span></code><code><span><br></span></code><code><span>  # p_all_markers=DotPlot(sce.all.int, group.by = 'celltype',</span></code><code><span>  #                       features = gene_list,</span></code><code><span>  #                       scale = T,assay='RNA' )+  </span></code><code><span>  #   theme(axis.text.x=element_text(angle=45,hjust = 1))</span></code><code><span>  # p_all_markers</span></code><code><span>  # ggsave('check_paper_markers_celltype.pdf',</span></code><code><span>  #        height = 8,width = 12)</span></code><code><span>  print(getwd())</span></code><code><span>}</span></code><code><span><br></span></code><code><span>#在meta信息中添加分组信息</span></code><code><span>sce.all.int@meta.data$group = ifelse(</span></code><code><span>  sce.all.int@meta.data$orig.ident %in% c( 'GSM4704931'),'control','ATAA'  )</span></code><code><span>phe=sce.all.int@meta.data;head(phe);print(getwd())</span></code><code><span><br></span></code><code><span>save(phe,file = 'phe.Rdata')</span></code><code><span><br></span></code><code><span>pdf('orig.ident-vs-celltype.pdf',height = 18)</span></code><code><span>gplots::balloonplot(table(sce.all.int$celltype,sce.all.int$orig.ident))</span></code><code><span>dev.off()</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code></pre></section><p><strong><span>会得到如下文件</span></strong><br></p><p><img data-galleryid="" data-imgfileid="100036001" data-ratio="0.2689434364994664" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SktL8QHWsQia4vf0kVbzSQXyMibrp680rIBd1XzgQibCOxiaPiboS04EWWVMtvTZMFPRjAEag6yRxKhjCNw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="937" src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SktL8QHWsQia4vf0kVbzSQXyMibrp680rIBd1XzgQibCOxiaPiboS04EWWVMtvTZMFPRjAEag6yRxKhjCNw/640?wx_fmt=png&amp;from=appmsg"></p><p><img data-galleryid="" data-imgfileid="100036000" data-ratio="0.24130663856691253" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SktL8QHWsQia4vf0kVbzSQXyMmZ942EGLlTIwnQQD4QXrWra8WTic4EvdF1nKMNM4XQz6XS3aGhdJY0A/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="949" src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SktL8QHWsQia4vf0kVbzSQXyMmZ942EGLlTIwnQQD4QXrWra8WTic4EvdF1nKMNM4XQz6XS3aGhdJY0A/640?wx_fmt=png&amp;from=appmsg"></p><p><img data-galleryid="" data-imgfileid="100036003" data-ratio="0.7215320910973085" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SktL8QHWsQia4vf0kVbzSQXyMw06WcqBWwmn96M7MkVDcyvU1FLk1EO8EN1j7jDkazfVq0NTgpMjl5Q/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="966" src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SktL8QHWsQia4vf0kVbzSQXyMw06WcqBWwmn96M7MkVDcyvU1FLk1EO8EN1j7jDkazfVq0NTgpMjl5Q/640?wx_fmt=png&amp;from=appmsg"></p><p><strong><span>画圈圈的主要用于细胞命名，这里主要是注释大类亚群，如果你想要个性化注释亚群：</span></strong></p><ul><li><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=Mzg2NDcxMzYwNg==&amp;mid=2247484687&amp;idx=1&amp;sn=8742f11a45f0b617b1bc1ec6c1e36dab&amp;chksm=ce646be6f913e2f0486cce8a6f702cc8fa2542557c9542ea551461f7502a65a424238eab3a3e&amp;scene=21#wechat_redirect" textvalue="seurat个性化细胞注释并把细分亚群放回总群" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2"><span><strong>seurat个性化细胞注释并把细分亚群放回总群</strong></span></a><br></p></li><li><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=Mzg2NDcxMzYwNg==&amp;mid=2247484687&amp;idx=2&amp;sn=50fa86f0568163ef2a1cb88ff3c86abc&amp;chksm=ce646be6f913e2f0dc4a486f9fbce2eab523a1b324061a3d613e167d1f14688fa21f28067b58&amp;scene=21#wechat_redirect" textvalue="直播四-单细胞个性化注释、细分亚群并把细分亚群放回总群" linktype="text" imgurl="" imgdata="null" data-itemshowtype="5" tab="innerlink" data-linktype="2"><span><strong>直播四-单细胞个性化注释、细分亚群并把细分亚群放回总群</strong></span></a><span><br></span></p></li></ul><p><br></p><h3><span>第二步：基于cluster或者celltype的差异分析，绘制火山图</span></h3><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="perl"><code><span><br></span></code><code><span>#########################设置我自己的r包加载路径，通常你不需要运行这段代码-----</span></code><code><span>.libPaths(c(</span></code><code><span>  '/home/rootyll/seurat_v5/',</span></code><code><span>  "/usr/local/lib/R/site-library", "/usr/lib/R/site-library", "/usr/lib/R/library"</span></code><code><span>));print(getwd())</span></code><code><span>###################################################################################333</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span>library(Seurat)</span></code><code><span><br></span></code><code><span><br></span></code><code><span>#setwd('../')</span></code><code><span><br></span></code><code><span>#1 在主文件夹下-----</span></code><code><span>source('scRNA_scripts/lib.R')</span></code><code><span>print(getwd())</span></code><code><span>print(.libPaths())</span></code><code><span><br></span></code><code><span><br></span></code><code><span>#setwd("../")</span></code><code><span>sce.all.int = readRDS('2-harmony/sce.all_int.rds')</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span>sp='human'</span></code><code><span>colnames(sce.all.int@meta.data) </span></code><code><span>table(sce.all.int@meta.data$RNA_snn_res.0.8)</span></code><code><span>table(sce.all.int@meta.data$RNA_snn_res.0.1)</span></code><code><span><br></span></code><code><span># 分辨率取决于实际情况</span></code><code><span># 简化，就选择 RNA_snn_res.0.1</span></code><code><span># gplots::balloonplot(table(sce.all.int@meta.data$RNA_snn_res.0.8,sce.all.int@meta.data$orig.ident))</span></code><code><span># gplots::balloonplot(table(sce.all.int@meta.data$RNA_snn_res.0.1,sce.all.int@meta.data$orig.ident))</span></code><code><span>#1差异分析时，使用group分组信息，去文献里面看--------</span></code><code><span>table(sce.all.int$orig.ident)</span></code><code><span>sce.all.int@meta.data$group = ifelse(</span></code><code><span>  sce.all.int@meta.data$orig.ident %in% c( 'GSM4704931'),'control','ATAA'  )</span></code><code><span><br></span></code><code><span><br></span></code><code><span>table(sce.all.int@meta.data$group) #差异分析时，使用group分组信息--------</span></code><code><span>getwd()</span></code><code><span><br></span></code><code><span>load('phe.Rdata');head(phe)</span></code><code><span><br></span></code><code><span>sce.all.int@meta.data$celltype = phe$celltype</span></code><code><span>sce.all = sce.all.int</span></code><code><span>getwd()</span></code><code><span>dir.create("./4_deg",recursive = T)</span></code><code><span>setwd("./4_deg/")</span></code><code><span>getwd()</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span>#获取每个细胞类型的barcodes------</span></code><code><span>cell_list = split(colnames(sce.all),sce.all@meta.data$celltype)</span></code><code><span>cell_list</span></code><code><span>names(cell_list)</span></code><code><span><br></span></code><code><span>table(sce.all$group,sce.all$celltype)</span></code><code><span>print(getwd())</span></code><code><span><br></span></code><code><span>dir.create("./by_celltype")</span></code><code><span>setwd("./by_celltype/")</span></code><code><span>getwd()</span></code><code><span>library(ggplot2)</span></code><code><span><br></span></code><code><span>degs_list=list()</span></code><code><span>for ( pro in names(cell_list) ) {</span></code><code><span>  # pro =  names(cell_list)[10]</span></code><code><span>  print(pro)</span></code><code><span><br></span></code><code><span>  sce= sce.all[, colnames(sce.all) %in%  cell_list[[pro]]]</span></code><code><span>  sce</span></code><code><span>  sce &lt;- CreateSeuratObject(counts = sce@assays$RNA$counts, </span></code><code><span>                            meta.data = sce@meta.data, </span></code><code><span>                            min.cells = 3, </span></code><code><span>                            min.features = 200)  </span></code><code><span>  sce &lt;- NormalizeData(sce)  </span></code><code><span>  sce = FindVariableFeatures(sce)</span></code><code><span>  sce = ScaleData(sce, </span></code><code><span>                  vars.to.regress = c("nFeature_RNA",</span></code><code><span>                                      "percent_mito"))</span></code><code><span><br></span></code><code><span>  Idents(sce)=sce$group</span></code><code><span>  table(Idents(sce))</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span>if (    length(  table(Idents(sce))  ) ==1 )  next </span></code><code><span> {</span></code><code><span>    # 如果细胞 种类数量不为1 </span></code><code><span>    print(paste0("注意，你的组别有 ",length(  table(Idents(sce))  ) ,' 个。我们下面进行差异分析'))  #-   </span></code><code><span>    print(paste0('您的组别是：' ,unique(sce$group),"   "))</span></code><code><span>    # 提示用户输入 </span></code><code><span>    # control_name &lt;- readline("请输入您的control组名称(如不知道名称，请按ESC，退出) : ")</span></code><code><span>    # exp_name &lt;- readline("请输入您exp组名称(如不知道名称，请按ESC，退出) :")</span></code><code><span>    # </span></code><code><span>    control_name=unique(sce$group)[1]</span></code><code><span>    exp_name=unique(sce$group)[2]</span></code><code><span>    degs=FindMarkers(sce,ident.1 = control_name,ident.2 = exp_name)</span></code><code><span>    degs$gene=rownames(degs);degs$symbol=rownames(degs)</span></code><code><span>    degs$comparation=paste0(control_name,'_vs_',exp_name)</span></code><code><span>    degs$celltype=pro</span></code><code><span>    degs$cluster=degs$celltype</span></code><code><span>    print(head(degs))</span></code><code><span><br></span></code><code><span>    res=degs</span></code><code><span>       p=EnhancedVolcano::  EnhancedVolcano(res,</span></code><code><span>                       lab = res$symbol,</span></code><code><span>                       x = 'avg_log2FC',</span></code><code><span>                       y = 'p_val',pCutoff = 0.05,</span></code><code><span>                       FCcutoff =1) #######这里默认使用|fc|为1---------</span></code><code><span>       ggsave(file = paste0(pro,'deg_volcano.pdf' ),plot = p,width = 8,height = 6)</span></code><code><span><br></span></code><code><span><br></span></code><code><span>       degs_list[[paste(pro)]]=degs</span></code><code><span><br></span></code><code><span>       library(dplyr) </span></code><code><span>       sce.markers =degs</span></code><code><span><br></span></code><code><span><br></span></code><code><span>       # 筛选出p_val_adj小于0.05的数据</span></code><code><span>       filtered_data &lt;- sce.markers[sce.markers$p_val_adj &lt; 0.05, ] </span></code><code><span><br></span></code><code><span>       # 提取avg_log2FC列的前10行和后10行</span></code><code><span>       top_rows &lt;-  filtered_data %&gt;% group_by(cluster ) %&gt;% slice_max(avg_log2FC,n = 10)  </span></code><code><span>       bottom_rows &lt;-  filtered_data %&gt;% group_by(cluster ) %&gt;% slice_min(avg_log2FC,n = 10)  </span></code><code><span><br></span></code><code><span>       # 合并前10行和后10行</span></code><code><span>       top10 &lt;- rbind(top_rows, bottom_rows)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span>        DoHeatmap(sce,top10$gene,size=3,draw.lines=F)</span></code><code><span>       ggsave(filename=paste0(pro,'_sce.markers_heatmap.pdf'),</span></code><code><span>              height = 3,width = 5)</span></code><code><span><br></span></code><code><span>       p &lt;- DotPlot(sce , features = unique(top10$gene),</span></code><code><span>                    assay='RNA'  )  + coord_flip()</span></code><code><span><br></span></code><code><span>       p</span></code><code><span>       ggsave(plot=p, </span></code><code><span>              filename=paste0("check_top10-degs_marker_by_",</span></code><code><span>                              pro,"_cluster.pdf"), height = 4,width = 6 );print(getwd())</span></code><code><span><br></span></code><code><span>  }</span></code><code><span><br></span></code><code><span>}</span></code><code><span><br></span></code><code><span>degs_allcelltype_df=do.call(rbind,degs_list)</span></code><code><span>degs_allcelltype_df %&gt;%head()</span></code><code><span><br></span></code><code><span><br></span></code><code><span>write.csv(degs_allcelltype_df,file = "./degs_allcelltype_df.csv")</span></code><code><span>saveRDS(degs_allcelltype_df,file = "./allcelltype_degs_sce.markers.Rdata")</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span>head(degs_allcelltype_df);print(getwd())</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span>library(scRNAtoolVis)</span></code><code><span>#library(colourpicker)</span></code><code><span>degs_allcelltype_df$cluster=degs_allcelltype_df$celltype</span></code><code><span>p2=scRNAtoolVis::jjVolcano(diffData = degs_allcelltype_df,</span></code><code><span>                        log2FC.cutoff =  2,</span></code><code><span>                        adjustP.cutoff = 0.05,</span></code><code><span>          legend.position = c(0.93, 0.99), </span></code><code><span>          topGeneN=0,#top genes to be labeled in plot, default 5.</span></code><code><span>        #  cluster.order=seq(0,23,1),</span></code><code><span>          pSize=0.4,</span></code><code><span>          tile.col = c("#EE3B3B", "#FF7F00", "#CD6600", "#8B2323", "#DEB887", "#76EEC6", </span></code><code><span>                       "#F0FFFF", "#008B8B", "#FFB90F", "#F5F5DC", "#1F1F1F", "#66CD00", </span></code><code><span>                       "#0000FF", "#97FFFF", "#528B8B", "#9400D3", "#EE1289", "#00BFFF", </span></code><code><span>                       "#00FF00", "#191970", "#FFFF00", "#4A708B", "#00FF7F", "#8B8B00", </span></code><code><span>                       "#FF1493", "#FFA500", "#8B4513")) </span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span>ggsave(plot = p2,filename = "allcelltypes_degs_volcano.pdf",width = 9,height = 6)</span></code><code><span>print(getwd())</span></code><code><span><br></span></code><code><span>setwd('../../')</span></code><code><span><br></span></code><code><span>#</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code></pre></section><p><img data-galleryid="" data-imgfileid="100036007" data-ratio="0.9115442278860569" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SktL8QHWsQia4vf0kVbzSQXyMjI4SYZ1xcibE7CJZpM7LLMcjwfS23WRuZvsZmfyLaEE13uNBKFdE87Q/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="667" src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SktL8QHWsQia4vf0kVbzSQXyMjI4SYZ1xcibE7CJZpM7LLMcjwfS23WRuZvsZmfyLaEE13uNBKFdE87Q/640?wx_fmt=png&amp;from=appmsg"></p><p><img data-galleryid="" data-imgfileid="100036005" data-ratio="1.0584615384615386" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SktL8QHWsQia4vf0kVbzSQXyMahthcHHAqN9ZIwSw6NL8QlvepDcLwvn4aLcjuyIoO8cJicvs8XUgLjQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="650" src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SktL8QHWsQia4vf0kVbzSQXyMahthcHHAqN9ZIwSw6NL8QlvepDcLwvn4aLcjuyIoO8cJicvs8XUgLjQ/640?wx_fmt=png&amp;from=appmsg"></p><p><strong><span>最后还贴心的对差异基因进行了汇总。同时你可以根据保存的rdata文件进行更个性化的绘图，如下图</span></strong></p><p><img data-galleryid="" data-imgfileid="100036008" data-ratio="0.6574074074074074" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SktL8QHWsQia4vf0kVbzSQXyMAz1ADRJkN9XI0s2QNYTZwqUmIdNNcf40AiaxlEA9qdGzfzumYtXWx7g/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SktL8QHWsQia4vf0kVbzSQXyMAz1ADRJkN9XI0s2QNYTZwqUmIdNNcf40AiaxlEA9qdGzfzumYtXWx7g/640?wx_fmt=png&amp;from=appmsg"></p><h3><span>第三步：对差异基因进行批量富集分析：GO、KEGG</span></h3><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="perl"><code><span><br></span></code><code><span>#########################设置我自己的r包加载路径，通常你不需要运行这段代码-----</span></code><code><span>.libPaths(c(</span></code><code><span>  '/home/rootyll/seurat_v5/',</span></code><code><span>  "/usr/local/lib/R/site-library", "/usr/lib/R/site-library", "/usr/lib/R/library"</span></code><code><span>));print(getwd())</span></code><code><span>###################################################################################333</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span># 1.加载R包 ----</span></code><code><span><br></span></code><code><span>rm(list = ls())</span></code><code><span>options(stringsAsFactors = F)</span></code><code><span>library(clusterProfiler)</span></code><code><span># 这里没有判断物种选择，默认人类</span></code><code><span># 如果是其他，需要修改不少地方</span></code><code><span>library(org.Hs.eg.db) </span></code><code><span>library(dplyr) </span></code><code><span>library(Seurat)</span></code><code><span><br></span></code><code><span># 2.设置路径 ----</span></code><code><span>#setwd("../")</span></code><code><span>getwd() #在主目录-------</span></code><code><span><br></span></code><code><span>dir.create("5_GO_KEGG")</span></code><code><span>setwd("5_GO_KEGG/")</span></code><code><span><br></span></code><code><span># 3.all 读取数据富集分析----</span></code><code><span>## 3.1 kegg by cluster 可视化 ----</span></code><code><span><br></span></code><code><span>f = '../check-by-0.1/qc-_marker_cosg.Rdata'</span></code><code><span><br></span></code><code><span>if(   file.exists(f)   ){</span></code><code><span><br></span></code><code><span>  load(file = f);print(getwd())</span></code><code><span>  lapply(marker_cosg,head)</span></code><code><span>  # sce.markers=sce.markers[sce.markers$avg_log2FC &gt; 0,]</span></code><code><span>  # top100 &lt;- sce.markers %&gt;% group_by(cluster) %&gt;% top_n(1000, avg_log2FC)</span></code><code><span>  top100 &lt;- unique(as.character(   apply(marker_cosg$names,2,head,100 )    ))</span></code><code><span>  top100 &lt;- as.list(marker_cosg$names)</span></code><code><span><br></span></code><code><span>  top100 &lt;-  data.frame( cluster=rep(names(top100),</span></code><code><span>                                     each=unlist(lapply(top100, length))  ),</span></code><code><span>              gene=unlist(top100)      )</span></code><code><span><br></span></code><code><span>  table(top100$cluster) </span></code><code><span>  head(top100) </span></code><code><span>  library(ggplot2)</span></code><code><span>  ids= bitr(top100$gene,'SYMBOL','ENTREZID','org.Hs.eg.db')</span></code><code><span>  top100=merge(top100,ids,by.x='gene',by.y='SYMBOL');print(head(top100))</span></code><code><span><br></span></code><code><span>  gcSample=split(top100$ENTREZID, top100$cluster)</span></code><code><span>  gcSample # entrez id , compareCluster</span></code><code><span>  xx &lt;- compareCluster(gcSample, fun="enrichGO",</span></code><code><span>                       OrgDb='org.Hs.eg.db');go=xx</span></code><code><span><br></span></code><code><span><br></span></code><code><span>  p=dotplot(xx) </span></code><code><span>  p+ theme(axis.text.x = element_text(angle = 45, </span></code><code><span>                                      vjust = 0.5, hjust=0.5))</span></code><code><span><br></span></code><code><span>  ggsave('compareCluster_enrichgo_-top100-cluster.pdf',width = 18,height = 8,plot = p)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span>  xx &lt;- compareCluster(gcSample, fun="enrichKEGG",</span></code><code><span>                       organism="hsa");kegg=xx</span></code><code><span>  p=dotplot(xx) </span></code><code><span>  p+ theme(axis.text.x = element_text(angle = 45, </span></code><code><span>                                      vjust = 0.5, hjust=0.5))</span></code><code><span>  ggsave('compareCluster_enrichKEGG_-top100-cluster.pdf',width = 18,height = 8,plot = p)</span></code><code><span><br></span></code><code><span><br></span></code><code><span>  save(kegg,go,file = "kegg_go_cluster.Rdata")</span></code><code><span><br></span></code><code><span>}</span></code><code><span><br></span></code><code><span>print(getwd())</span></code><code><span>## 3.2 kegg by celltype 可视化 ----</span></code><code><span><br></span></code><code><span>f = '../check-by-celltype/qc-_marker_cosg.Rdata'</span></code><code><span><br></span></code><code><span>if(  file.exists(f)  ){</span></code><code><span>  print(getwd())</span></code><code><span><br></span></code><code><span>  load(file = f )</span></code><code><span>  # sce.markers=sce.markers[sce.markers$avg_log2FC &gt; 0,]</span></code><code><span>  # top100 &lt;- sce.markers %&gt;% group_by(cluster) %&gt;% top_n(1000, avg_log2FC)</span></code><code><span>  top100 &lt;- unique(as.character(apply(marker_cosg$names,2,head,100)))</span></code><code><span>  top100 &lt;- as.list(marker_cosg$names)</span></code><code><span>  top100 &lt;-  data.frame(cluster=rep(names(top100),each=unlist(lapply(top100, length))),</span></code><code><span>                        gene=unlist(top100))</span></code><code><span>  table(top100$cluster) </span></code><code><span>  head(top100) </span></code><code><span>  library(ggplot2)</span></code><code><span>  ids=bitr(top100$gene,'SYMBOL','ENTREZID','org.Hs.eg.db')</span></code><code><span>  top100=merge(top100,ids,by.x='gene',by.y='SYMBOL')</span></code><code><span>  gcSample=split(top100$ENTREZID, top100$cluster)</span></code><code><span>  gcSample # entrez id , compareCluster </span></code><code><span>  xx &lt;- compareCluster(gcSample, fun="enrichKEGG",</span></code><code><span>                       organism="hsa")</span></code><code><span>  kegg=xx</span></code><code><span>  p=dotplot(xx) </span></code><code><span>  p+ theme(axis.text.x = element_text(angle = 90, </span></code><code><span>                                      vjust = 0, hjust=1))</span></code><code><span>  ggsave('compareCluster_enrichKEGG-top100-celltype.pdf',width = 12,height = 12)</span></code><code><span>  ## 3.3 Go by cluster 可视化 ----</span></code><code><span>  gcSample</span></code><code><span>  xx &lt;- compareCluster(gcSample, fun="enrichGO",</span></code><code><span>                       OrgDb='org.Hs.eg.db');go=xx</span></code><code><span>  summary(xx)</span></code><code><span>  p=dotplot(xx) </span></code><code><span>  p+ theme(axis.text.x = element_text(angle = 90, </span></code><code><span>                                      vjust = 0, hjust=1))</span></code><code><span>  ggsave('compareCluster-enrichGO-top100-celltype.pdf',width = 15,height = 12)</span></code><code><span><br></span></code><code><span> save(kegg,go,file = "kegg_go_celltype.Rdata")</span></code><code><span><br></span></code><code><span>}</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span>## 4.2 GO_KEGG富集分析 -------------------------------------------------------------</span></code><code><span>library(clusterProfiler)</span></code><code><span>library(org.Hs.eg.db)</span></code><code><span>library(EnhancedVolcano)</span></code><code><span>library(ggplot2)</span></code><code><span>library(stringr) </span></code><code><span>print(getwd())</span></code><code><span><br></span></code><code><span># rich_list &lt;- list()</span></code><code><span># dir.create("EnhancedVolcano")</span></code><code><span># dir.create("KEGG_enrichment")</span></code><code><span># dir.create("GO_enrichment")</span></code><code><span># dir.create("DEG")</span></code><code><span><br></span></code><code><span>fs = list.files(path = '../4_deg/by_celltype/',</span></code><code><span>                recursive = T,full.names = T,</span></code><code><span>                pattern = 'Rdata')</span></code><code><span>fs</span></code><code><span><br></span></code><code><span># lapply(fs, function(x){</span></code><code><span>#   #x=fs[1]</span></code><code><span>#   load(x)</span></code><code><span>#   table(sce.markers$cluster)</span></code><code><span>#   write.csv(sce.markers,file = gsub('.Rdata','.csv',x))</span></code><code><span># })</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span># 如果是两个分组，我们前面的deg就可以进行火山图等绘制 </span></code><code><span>fs = list.files(path = '../4_deg/',</span></code><code><span>                recursive = T,full.names = T,</span></code><code><span>                pattern = 'sce.markers.Rdata')</span></code><code><span>fs</span></code><code><span><br></span></code><code><span>if (F) {</span></code><code><span><br></span></code><code><span>  lapply(fs, function(x){</span></code><code><span>    # x=fs[1]</span></code><code><span>    load(x)</span></code><code><span>    table(sce.markers$cluster) </span></code><code><span>    need_DEG= sce.markers </span></code><code><span>    need_DEG$avg_log2FC=need_DEG$avg_log2FC* ifelse(as.numeric( sce.markers$cluster )-1,1,-1) </span></code><code><span>    colnames(need_DEG)</span></code><code><span><br></span></code><code><span>    logFC_cutoff=0.25</span></code><code><span>    need_DEG$change = as.factor(ifelse(need_DEG$p_val&lt; 0.05 &amp; abs(need_DEG$avg_log2FC) &gt; logFC_cutoff,</span></code><code><span>                                       ifelse(need_DEG$avg_log2FC &gt; logFC_cutoff ,'UP','DOWN'),'NOT')</span></code><code><span>    )</span></code><code><span>    table(  need_DEG$change )</span></code><code><span>    need_DEG$symbol=row.names(need_DEG)</span></code><code><span><br></span></code><code><span><br></span></code><code><span>    # 差异基因火山图</span></code><code><span>    res=need_DEG</span></code><code><span>    head(res)</span></code><code><span>    EnhancedVolcano(res,</span></code><code><span>                    lab = res$symbol,</span></code><code><span>                    x = 'avg_log2FC',</span></code><code><span>                    y = 'p_val',pCutoff = 0.05,</span></code><code><span>                    FCcutoff =logFC_cutoff )</span></code><code><span>    ggsave(file = gsub('.Rdata','_Volcano.pdf',x))</span></code><code><span><br></span></code><code><span>  })</span></code><code><span><br></span></code><code><span><br></span></code><code><span>}</span></code><code><span>print(getwd())</span></code><code><span>setwd("../")</span></code><code><span>print(getwd())</span></code><code><span><br></span></code></pre></section><p><img data-galleryid="" data-imgfileid="100036004" data-ratio="0.40865384615384615" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SktL8QHWsQia4vf0kVbzSQXyMKFRHY1wic4RqgTXNW5cicwicmj1c9bN7IebNRnFF1iaktP0IAkSEpEceWA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="624" src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SktL8QHWsQia4vf0kVbzSQXyMKFRHY1wic4RqgTXNW5cicwicmj1c9bN7IebNRnFF1iaktP0IAkSEpEceWA/640?wx_fmt=png&amp;from=appmsg"></p><p><img data-galleryid="" data-imgfileid="100036014" data-ratio="0.8010335917312662" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibreOkI1mnsB3ZPictcMEVL5v5ynia1QpbTp6j5ibTJPpY1BfgsT0shP0mjPv816L35ibvy8qRPVJQbAg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="774" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibreOkI1mnsB3ZPictcMEVL5v5ynia1QpbTp6j5ibTJPpY1BfgsT0shP0mjPv816L35ibvy8qRPVJQbAg/640?wx_fmt=png&amp;from=appmsg"></p><p><span><strong><span>这一步的代码也保留了富集分析的所有数据，如果你想更个性化或者挑选富集分析结果自行展示：</span></strong></span><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=Mzg2NDcxMzYwNg==&amp;mid=2247485216&amp;idx=1&amp;sn=ca88c06f0b5b99b495cbe46a85b1f093&amp;chksm=ce6469c9f913e0df7588e1a4c2a845ad898e65ecbe71e73ec4fc803544140aea5241a78d037b&amp;scene=21#wechat_redirect" textvalue="手动选择富集分析结果，真的不合理吗？" linktype="text" imgurl="" imgdata="null" data-itemshowtype="5" tab="innerlink" data-linktype="2"><span>手动选择富集分析结果，真的不合理吗？</span></a><br></p><p><br></p><p><span><strong><span>甚至你还可以选择目的term中的所有基因进行热图展示：</span></strong></span><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=Mzg2NDcxMzYwNg==&amp;mid=2247485510&amp;idx=1&amp;sn=e3a08892119914c79d163177fe0758c5&amp;chksm=ce6466aff913efb9b1ff18732e6b6e5e3a71c7e2f5994039ebd4948a666a1e54f7693bfc8c80&amp;scene=21#wechat_redirect" textvalue="单细胞|提取富集分析通路基因画个性化热图" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2"><span>单细胞|提取富集分析通路基因画个性化热图</span></a></p><h3><span>第四步：展示不同组别中的细胞比例</span></h3><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="perl"><code><span><br></span></code><code><span>#########################设置我自己的r包加载路径，通常你不需要运行这段代码-----</span></code><code><span>.libPaths(c(</span></code><code><span>  '/home/rootyll/seurat_v5/',</span></code><code><span>  "/usr/local/lib/R/site-library", "/usr/lib/R/site-library", "/usr/lib/R/library"</span></code><code><span>))</span></code><code><span>###################################################################################333</span></code><code><span><br></span></code><code><span><br></span></code><code><span># 1.加载R包 ----</span></code><code><span>rm(list=ls()) </span></code><code><span>library(ggplot2) </span></code><code><span>library(cowplot) </span></code><code><span>library(paletteer)  </span></code><code><span>library(gplots)</span></code><code><span>library(ggpubr)    </span></code><code><span>library(ggsci) </span></code><code><span>library(stringr)</span></code><code><span><br></span></code><code><span><br></span></code><code><span>##1 在主目录 ------------</span></code><code><span>#setwd('../')</span></code><code><span>getwd() </span></code><code><span><br></span></code><code><span>dir.create("./4_group")</span></code><code><span>setwd("./4_group/")</span></code><code><span><br></span></code><code><span>getwd() </span></code><code><span><br></span></code><code><span># 3.读取数据 ----</span></code><code><span>load("../phe.Rdata");print(getwd())</span></code><code><span>#sce.all.int = readRDS('../2-harmony/sce.all_int.rds')</span></code><code><span>head(phe)</span></code><code><span>table(phe$orig.ident)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span>#在meta信息中添加分组信息</span></code><code><span># sce.all.int@meta.data$group = ifelse(</span></code><code><span>#   sce.all.int@meta.data$orig.ident %in% c( 'GSM4704931'),'control','ATAA'  )</span></code><code><span># phe=sce.all.int@meta.data;head(phe);print(getwd())</span></code><code><span>phe$group=ifelse(</span></code><code><span>  phe$orig.ident %in% c( 'GSM4704931'),'control','ATAA'  )</span></code><code><span><br></span></code><code><span>table(phe$group) </span></code><code><span><br></span></code><code><span><br></span></code><code><span># 4.可视化 ----</span></code><code><span>## 4.1 每种细胞类型中，分组所占比例 ----</span></code><code><span>library(tidyr)# 使用的gather &amp; spread</span></code><code><span>library(reshape2) # 使用的函数 melt &amp; dcast </span></code><code><span>library(dplyr)</span></code><code><span>library(ggplot2)</span></code><code><span>tb=table(phe$celltype,</span></code><code><span>         phe$group)</span></code><code><span><br></span></code><code><span>tb</span></code><code><span>str(tb)</span></code><code><span>balloonplot(tb)</span></code><code><span>bar_data &lt;- as.data.frame(tb);bar_data</span></code><code><span><br></span></code><code><span>bar_per &lt;- bar_data %&gt;% </span></code><code><span>  group_by(Var1) %&gt;%</span></code><code><span>  mutate(sum(Freq)) %&gt;% #分组求和</span></code><code><span>  mutate(percent = Freq / `sum(Freq)`) #计算每种细胞类型在各种分组里所占比例</span></code><code><span>head(bar_per) ;bar_per</span></code><code><span>write.csv(bar_per,file = "celltype_by_group_percent.csv");print(getwd())</span></code><code><span><br></span></code><code><span><br></span></code><code><span>ggplot(bar_per, aes(x = Var1, y = percent)) +</span></code><code><span>  geom_bar(aes(fill = Var2) , stat = "identity") + coord_flip() +</span></code><code><span>  theme(axis.ticks = element_line(linetype = "blank"),</span></code><code><span>        legend.position = "top",</span></code><code><span>        panel.grid.minor = element_line(colour = NA,linetype = "blank"), </span></code><code><span>        panel.background = element_rect(fill = NA),</span></code><code><span>        plot.background = element_rect(colour = NA)) +</span></code><code><span>  labs(y = "% Relative cell source", fill = NULL)+labs(x = NULL)+</span></code><code><span>  scale_fill_d3()</span></code><code><span>print(getwd())</span></code><code><span>ggsave("celltype_by_group_percent.pdf",</span></code><code><span>       units = "cm",width = 20,height = 12)</span></code><code><span><br></span></code><code><span>## 4.2 每种细胞类型中，各个样本所占比例 ----</span></code><code><span>bar_data &lt;- as.data.frame(table(phe$celltype,phe$orig.ident));bar_data</span></code><code><span><br></span></code><code><span>bar_per &lt;- bar_data %&gt;% </span></code><code><span>  group_by(Var1) %&gt;%</span></code><code><span>  mutate(sum(Freq)) %&gt;%</span></code><code><span>  mutate(percent = Freq / `sum(Freq)`);bar_per</span></code><code><span><br></span></code><code><span>write.csv(bar_per,file = "celltype_by_orig.ident_percent.csv")</span></code><code><span>ggplot(bar_per, aes(x = Var1, y = percent)) +</span></code><code><span>  geom_bar(aes(fill = Var2) , stat = "identity") + coord_flip() +</span></code><code><span>  theme(axis.ticks = element_line(linetype = "blank"),</span></code><code><span>        legend.position = "top",</span></code><code><span>        panel.grid.minor = element_line(colour = NA,linetype = "blank"), </span></code><code><span>        panel.background = element_rect(fill = NA),</span></code><code><span>        plot.background = element_rect(colour = NA)) +</span></code><code><span>  labs(y = "% Relative cell source", fill = NULL)+labs(x = NULL) </span></code><code><span><br></span></code><code><span>ggsave("celltype_by_orig.ident_percent.pdf",units = "cm",</span></code><code><span>       width = 20,height = 12)</span></code><code><span><br></span></code><code><span>## 绘图函数，运行即可</span></code><code><span>plot_percent &lt;- function(x,y){</span></code><code><span>  # x &lt;- "group"</span></code><code><span>  # y &lt;- "celltype"</span></code><code><span>  plot_data &lt;- data.frame(table(phe[, x ],</span></code><code><span>                                phe[, y ]))</span></code><code><span><br></span></code><code><span>  plot_data$Total &lt;- apply(plot_data,1,function(x) sum( plot_data[plot_data$Var1 == x[1],3] ) )</span></code><code><span>  plot_data &lt;- plot_data %&gt;% mutate(Percentage = round(Freq/Total,3) * 100)</span></code><code><span><br></span></code><code><span>  pro &lt;- x</span></code><code><span>  write.table(plot_data,paste0(pro,"_celltype_proportion.txt"),quote = F)</span></code><code><span>  th=theme(axis.text.x = element_text(angle = 45, </span></code><code><span>                                      vjust = 0.5, hjust=0.5)) </span></code><code><span>  library(paletteer) </span></code><code><span>  color &lt;- c(paletteer_d("awtools::bpalette"),paletteer_d("awtools::a_palette"),paletteer_d("awtools::mpalette"))</span></code><code><span>  ratio1 &lt;- ggplot(plot_data,aes(x = Var1,y = Percentage,fill = Var2)) +</span></code><code><span>    geom_bar(stat = "identity",position = "stack") +</span></code><code><span>    scale_fill_manual(values = color)+</span></code><code><span>    theme_classic() + </span></code><code><span>    theme(axis.title.x = element_blank()) + labs(fill = "Cluster") +th </span></code><code><span>  ratio1</span></code><code><span>  f=paste0('ratio_by_',x,'_VS_',y)</span></code><code><span>  h=floor(5+length(unique(plot_data[,1]))/2)</span></code><code><span>  w=floor(3+length(unique(plot_data[,2]))/2)</span></code><code><span><br></span></code><code><span>  ggsave(paste0('bar1_',f,'.pdf'),ratio1,height = h ,width = w ) </span></code><code><span>  pdf(paste0('balloonplot_',f,'.pdf'),height = 12 ,width = 20)</span></code><code><span>  balloonplot(table(phe[, x ],</span></code><code><span>                    phe[, y ]))</span></code><code><span>  dev.off()</span></code><code><span><br></span></code><code><span>  plot_data$Total &lt;- apply(plot_data,1,function(x)sum(plot_data[plot_data$Var1 == x[1],3]))</span></code><code><span>  plot_data&lt;- plot_data %&gt;% mutate(Percentage = round(Freq/Total,3) * 100)</span></code><code><span>  bar_Celltype=ggplot(plot_data,aes(x = Var1,y = Percentage,fill = Var2)) +</span></code><code><span>    geom_bar(stat = "identity",position = "stack") +</span></code><code><span>    theme_classic() + </span></code><code><span>    theme(axis.text.x=element_text(angle=45,hjust = 1)) + labs(fill = "Cluster")+</span></code><code><span>    facet_grid(~Var2,scales = "free")</span></code><code><span>  bar_Celltype</span></code><code><span>  ggsave(paste0('bar2_',f,'.pdf'),bar_Celltype,height = 8 ,width =  40) </span></code><code><span>}</span></code><code><span><br></span></code><code><span>## 4.3 每个分组中，不同细胞类型所占比例 ----</span></code><code><span><br></span></code><code><span>plot_percent("group","celltype")</span></code><code><span><br></span></code><code><span>## 4.4 每个分组中，不同细胞类型所占比例 ----</span></code><code><span>plot_percent("orig.ident","celltype")</span></code><code><span><br></span></code><code><span>## 4.5 每个细胞类型中，不同分组所占比例箱型图 ----</span></code><code><span>colnames(phe)</span></code><code><span><br></span></code><code><span>lapply(c("group"), function(pro){</span></code><code><span><br></span></code><code><span>  tb &lt;- data.frame(table(phe$celltype,phe$orig.ident,</span></code><code><span>                         phe[,pro]))</span></code><code><span><br></span></code><code><span>  tb=tb[which(tb$Freq != 0),]</span></code><code><span>  tb=tb[,c(1,3,4)]</span></code><code><span>  head(tb) </span></code><code><span><br></span></code><code><span>  tb$Total &lt;- apply(tb,1,function(x)sum(tb[tb$Var1 == x[1],3]))</span></code><code><span>  tb&lt;- tb %&gt;% mutate(Percentage = round(Freq/Total,3) * 100)</span></code><code><span>  tb=tb[,c(1,2,5)]</span></code><code><span>  tb$Var1=as.factor(tb$Var1)</span></code><code><span>  tb$Var3=as.factor(tb$Var3)</span></code><code><span><br></span></code><code><span>  tb</span></code><code><span>  write.table(tb,paste0(pro,"_celltype_proportion.txt"),quote = F)</span></code><code><span><br></span></code><code><span>  library(paletteer) </span></code><code><span>  color &lt;- c(paletteer_d("awtools::bpalette"),paletteer_d("awtools::a_palette"),paletteer_d("awtools::mpalette"))</span></code><code><span><br></span></code><code><span>  p &lt;- ggplot(tb,aes(x=Var3,y=Percentage,fill=Var1,color=Var1))+</span></code><code><span>    geom_boxplot(outlier.alpha=0 )+facet_grid(~Var1)+</span></code><code><span>    geom_jitter(aes(x = Var3, y = Percentage ))+</span></code><code><span>    stat_compare_means(aes(group = Var3), </span></code><code><span>                       # label = "p.signif"</span></code><code><span>                       label = "p.format",label.y = 50)+ </span></code><code><span>    scale_fill_igv(alpha = 0.6)+scale_color_igv(alpha = 0.8)+</span></code><code><span>    theme_bw()+</span></code><code><span>    theme(axis.text.x = element_text(face = "bold",angle = 45, hjust=1, vjust=0.5,size = 14), </span></code><code><span>          legend.position= "none",</span></code><code><span>          strip.background = element_rect(color="black",fill = "steelblue3", linetype = "solid"),</span></code><code><span>          strip.text = element_text(face = "bold", color = "black",hjust = 0.5, size = 12,),</span></code><code><span>          plot.title=element_text(face="bold.italic",size="20", color="brown",hjust = 0.5),</span></code><code><span>          axis.text.y = element_text(face = "bold",size = 14) ,</span></code><code><span>          axis.title = element_text(size = 14),</span></code><code><span>          panel.grid.major = element_blank(),</span></code><code><span>          panel.grid.minor = element_blank())+</span></code><code><span>    labs(title = paste0("The proportion of  cell types in ",pro),</span></code><code><span>         x = "")</span></code><code><span>  # scale_color_d3(palette= "category20") #色卡 ，使用?scale_color_d3()查看</span></code><code><span>  p  </span></code><code><span>  ggsave(paste0(pro,"_boxplot.tiff"),width = 24,height = 10)</span></code><code><span>  ggsave(paste0(pro,"_boxplot.pdf"),width = 24,height = 10)</span></code><code><span>})</span></code><code><span><br></span></code><code><span><br></span></code><code><span>## 4.6 每个cluster中，不同分组所占比例箱型图 ----</span></code><code><span>colnames(phe)</span></code><code><span>lapply(c("group"), function(pro){</span></code><code><span><br></span></code><code><span>  # phe$RNA_snn_res.0.8，注意要和分群注释保持一致</span></code><code><span>  tb &lt;- data.frame(table(phe$RNA_snn_res.0.8,phe$orig.ident,</span></code><code><span>                         phe[,pro]))</span></code><code><span><br></span></code><code><span>  tb=tb[which(tb$Freq != 0),]</span></code><code><span>  tb=tb[,c(1,3,4)]</span></code><code><span>  head(tb) </span></code><code><span><br></span></code><code><span>  tb$Total &lt;- apply(tb,1,function(x)sum(tb[tb$Var1 == x[1],3]))</span></code><code><span>  tb&lt;- tb %&gt;% mutate(Percentage = round(Freq/Total,3) * 100)</span></code><code><span>  tb=tb[,c(1,2,5)]</span></code><code><span>  tb$Var1=as.factor(tb$Var1)</span></code><code><span>  tb$Var3=as.factor(tb$Var3)</span></code><code><span><br></span></code><code><span>  tb</span></code><code><span>  write.table(tb,paste0(pro,"_celltype_proportion.txt"),quote = F)</span></code><code><span>  p &lt;- ggplot(tb,aes(x=Var3,y=Percentage,fill=Var1,color=Var1))+</span></code><code><span>    geom_boxplot(outlier.alpha=0 )+facet_grid(~Var1)+</span></code><code><span>    geom_jitter(aes(x = Var3, y = Percentage ))+</span></code><code><span>    stat_compare_means(aes(group = Var3), </span></code><code><span>                       # label = "p.signif"</span></code><code><span>                       label = "p.format",label.y = 50)+ </span></code><code><span>    scale_fill_igv(alpha = 0.6)+scale_color_igv(alpha = 0.8)+</span></code><code><span>    theme_bw()+</span></code><code><span>    theme(axis.text.x = element_text(face = "bold",angle = 45, hjust=1, vjust=0.5,size = 14), </span></code><code><span>          legend.position= "none",</span></code><code><span>          strip.background = element_rect(color="black",fill = "steelblue3", linetype = "solid"),</span></code><code><span>          strip.text = element_text(face = "bold", color = "black",hjust = 0.5, size = 12,),</span></code><code><span>          plot.title=element_text(face="bold.italic",size="20", color="brown",hjust = 0.5),</span></code><code><span>          axis.text.y = element_text(face = "bold",size = 14) ,</span></code><code><span>          axis.title = element_text(size = 14),</span></code><code><span>          panel.grid.major = element_blank(),</span></code><code><span>          panel.grid.minor = element_blank())+</span></code><code><span>    labs(title = paste0("The proportion of  cell types in ",pro),</span></code><code><span>         x = "")</span></code><code><span>  # scale_color_d3(palette= "category20") #色卡 ，使用?scale_color_d3()查看</span></code><code><span>  p  </span></code><code><span>  ggsave(paste0(pro,"_CCA_snn_res.0.8-boxplot.tiff"),width = 24,height = 10)</span></code><code><span>  ggsave(paste0(pro,"_CCA_snn_res.0.8-boxplot.pdf"),width = 24,height = 10)</span></code><code><span>})</span></code><code><span>print(getwd())</span></code><code><span><br></span></code><code><span>## 4.7 每个样本中各个细胞类型所占比例 ----</span></code><code><span>tb &lt;- as.data.frame(table(phe$celltype,</span></code><code><span>                          phe$orig.ident,phe$group));tb</span></code><code><span><br></span></code><code><span>pro = 'group'</span></code><code><span>dat&lt;-tb %&gt;%</span></code><code><span>  group_by(Var2) %&gt;%</span></code><code><span>  mutate(sum(Freq)) %&gt;%</span></code><code><span>  mutate(Percentage = Freq / `sum(Freq)`);dat</span></code><code><span>head(dat)</span></code><code><span>dat &lt;- dat[dat$Percentage != 0,]</span></code><code><span><br></span></code><code><span>tb=dat[,c(1,3,6)]</span></code><code><span>head(tb) </span></code><code><span><br></span></code><code><span>tb$Var1=as.factor(tb$Var1)</span></code><code><span>tb$Var3=as.factor(tb$Var3)</span></code><code><span>colnames(tb)</span></code><code><span>tb</span></code><code><span>write.csv(tb,paste0(pro,"_celltype_proportion.csv"),quote = F)</span></code><code><span><br></span></code><code><span>p &lt;- ggplot(tb,aes(x=Var3,y=Percentage,fill=Var1,color=Var1))+</span></code><code><span>  geom_boxplot(outlier.alpha=0 )+facet_grid(~Var1)+</span></code><code><span>  geom_jitter(aes(x = Var3, y = Percentage ))+</span></code><code><span>  stat_compare_means(aes(group = Var3), </span></code><code><span>                     # label = "p.signif"</span></code><code><span>                     label = "p.format" )+ </span></code><code><span>  scale_fill_igv(alpha = 0.6)+scale_color_igv(alpha = 0.8)+</span></code><code><span>  theme_bw()+</span></code><code><span>  theme(axis.text.x = element_text(face = "bold",angle = 45, hjust=1, vjust=0.5,size = 14), </span></code><code><span>        legend.position= "none",</span></code><code><span>        strip.background = element_rect(color="black",fill = "steelblue3", linetype = "solid"),</span></code><code><span>        strip.text = element_text(face = "bold", color = "black",hjust = 0.5, size = 12,),</span></code><code><span>        plot.title=element_text(face="bold.italic",size="20", color="brown",hjust = 0.5),</span></code><code><span>        axis.text.y = element_text(face = "bold",size = 14) ,</span></code><code><span>        axis.title = element_text(size = 14),</span></code><code><span>        panel.grid.major = element_blank(),</span></code><code><span>        panel.grid.minor = element_blank())+</span></code><code><span>  labs(title = paste0("The proportion of  cell types in ",pro),</span></code><code><span>       x = "")</span></code><code><span># scale_color_d3(palette= "category20") #色卡 ，使用?scale_color_d3()查看</span></code><code><span>p  </span></code><code><span>ggsave("celltype_orig_group.pdf",width = 24,height = 6,plot = p)</span></code><code><span><br></span></code><code><span># 5.返回工作路径 ----</span></code><code><span>setwd("../../")</span></code><code><span><br></span></code></pre></section><p><img data-galleryid="" data-imgfileid="100036006" data-ratio="0.8822652757078987" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SktL8QHWsQia4vf0kVbzSQXyMlorcAoiaE4vVWqe8xs7PnKpCP2YBurwlVl7nSjoDI9BGRlKpqVLcxCg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="671" src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SktL8QHWsQia4vf0kVbzSQXyMlorcAoiaE4vVWqe8xs7PnKpCP2YBurwlVl7nSjoDI9BGRlKpqVLcxCg/640?wx_fmt=png&amp;from=appmsg"></p><p><img data-galleryid="" data-imgfileid="100036009" data-ratio="0.5978494623655914" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SktL8QHWsQia4vf0kVbzSQXyMOFTjYaiaz5LcmibYIetSibURSHEkHzhNicdrunFrh03BcQicd6tR0A8kwfg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="930" src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SktL8QHWsQia4vf0kVbzSQXyMOFTjYaiaz5LcmibYIetSibURSHEkHzhNicdrunFrh03BcQicd6tR0A8kwfg/640?wx_fmt=png&amp;from=appmsg"></p><p><img data-galleryid="" data-imgfileid="100036013" data-ratio="1.0071428571428571" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibreOkI1mnsB3ZPictcMEVL5VVKeUCMcAB0eXHkfk3P5cxShkElMT6wibAxFiaMWVA6BANL4GmhYibDRQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="700" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibreOkI1mnsB3ZPictcMEVL5VVKeUCMcAB0eXHkfk3P5cxShkElMT6wibAxFiaMWVA6BANL4GmhYibDRQ/640?wx_fmt=png&amp;from=appmsg"></p><p><strong><span>注：这里的代码适合刚接触项目的熟手运用，会让你事半功倍。但是对于纯小白而言，更重要的是理解代码的每一步骤，不要盲目去运行这些代码。不然报错你都不知道咋回事。另外，</span></strong><strong><span>目前此套代码的后半部分只适用于人，如果你的数据是其他物种需要自己调整一下富集分析所需要的数据库。</span></strong></p><p><strong><span></span></strong></p><p><strong><span>如有不妥当之处，欢迎批评指正。</span></strong></p><p><strong><span> ——生信小博士</span></strong></p><p><br></p><p><mp-style-type data-value="10000"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/GK5HWonCcM8BpjMA6MJsbQ",target="_blank" rel="noopener noreferrer">原文链接</a>
