---
title: "单细胞轨迹推断之 SCORPIUS"
date: 2024-01-11T07:59:17Z
draft: ["false"]
tags: [
  "fetched",
  "单细胞追踪"
]
categories: ["Acdemic"]
---
单细胞轨迹推断之 SCORPIUS by 单细胞追踪
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><p data-tool="mdnice编辑器">在复现单细胞的时候，发现了一个单细胞轨迹推断工具SCORPIUS，是基于R语言、使用机器学习里面的随机森林方法选择基因进行轨迹推断，官网如下：GitHub - rcannood/SCORPIUS: Linear trajectory inference for single-cell RNA-seq</p><p data-tool="mdnice编辑器">同时在网上搜到了曾老师的教程：简单直接的拟时序分析方法，R包SCORPIUS推荐-腾讯云开发者社区-腾讯云 (tencent.com)</p><p data-tool="mdnice编辑器">这里我们通过pbmc3k数据进行SCORPIUS轨迹推断。</p><p><br></p><figure data-tool="mdnice编辑器"><p><img data-galleryid="" data-imgfileid="100000663" data-ratio="0.3151969981238274" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/y6jd6X97IxJ56FwaiafYyxmBxtLkLx1icgRBR1ictaFfJODlMa85LnopjoGtf3iaibmiavuz01dxZaxNVR0a9nib1IEmg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1066" src="https://mmbiz.qpic.cn/sz_mmbiz_png/y6jd6X97IxJ56FwaiafYyxmBxtLkLx1icgRBR1ictaFfJODlMa85LnopjoGtf3iaibmiavuz01dxZaxNVR0a9nib1IEmg/640?wx_fmt=png&amp;from=appmsg"></p><figcaption><br></figcaption></figure><hr data-tool="mdnice编辑器"><p data-tool="mdnice编辑器">首先按照并加载SCORPIUS包。</p><pre data-tool="mdnice编辑器"><span></span><code>devtools::install_github(<span>"rcannood/SCORPIUS"</span>, build_vignettes = <span>TRUE</span>)<br><span>library</span>(SCORPIUS)<br></code></pre><p data-tool="mdnice编辑器">从官网中得知我们需要矩阵和细胞注释信息，官网中用的是data数据，而曾老师的教程中用的是scale data数据，这里我们都进行尝试，并进行比较有什么区别。</p><pre data-tool="mdnice编辑器"><span></span><code><span>######pbmc3k</span><br><span>library</span>(SeuratData)<br>data(<span>"pbmc3k"</span>)  <br>sce &lt;- pbmc3k.final  <br><span>library</span>(Seurat)<br>table(Idents(sce))<br>sce = UpdateSeuratObject(object = sce)<br>sce$celltype = sce$seurat_annotations<br>sce= sce[,sce$celltype  %<span>in</span>% c(<span>'Naive CD4 T'</span>,<span>'Memory CD4 T'</span>,<span>'CD8 T'</span>,<span>'NK'</span>)]<br>table(Idents(sce))<br>sce<br>pbmc = CreateSeuratObject(<br>  counts = sce@assays$RNA@counts,<br>  meta.data = sce@meta.data<br>)<br>pbmc &lt;- NormalizeData(pbmc, normalization.method = <span>"LogNormalize"</span>, <br>                      scale.factor = <span>10000</span>) <br>pbmc &lt;- FindVariableFeatures(pbmc, selection.method = <span>"vst"</span>, nfeatures = <span>2000</span>) <br>pbmc &lt;- ScaleData(pbmc )  <br><span>####至此，github官网上expression用的是data的数据，而不是scale data，试试看有什么区别</span><br>expression_data &lt;- t(as.matrix(pbmc@assays$RNA@data))<br>expression_scale &lt;- t(as.matrix(pbmc@assays$RNA@scale.data))<br><br>group_name =  as.factor(as.character(pbmc$celltype))<br>table(group_name)<br>dim(expression_data)<br>dim(expression_scale)<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>group_name<br>       CD8 T Memory CD4 T  Naive CD4 T           NK <br>         271          483          697          155 <br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>dim(expression_data)<br>[1]  1606 13714<br>dim(expression_scale)<br>[1] 1606 2000<br></code></pre><p data-tool="mdnice编辑器">我们提取了4个细胞亚群，共1606个细胞，然后data数据中有13714个基因，而scale data中有2000个基因。接下来，进行单细胞轨迹推断：</p><pre data-tool="mdnice编辑器"><span></span><code><span>#######expression_data----</span><br>space &lt;- reduce_dimensionality(expression_data, <span>"spearman"</span>)<br>draw_trajectory_plot(space, group_name, contour = <span>TRUE</span>,)<br>traj &lt;- infer_trajectory(space)<br>draw_trajectory_plot(space, group_name, traj$path, contour = <span>TRUE</span>)<br><span># warning: setting num_permutations to 10 requires a long time (~30min) to run!</span><br><span># set it to 0 and define a manual cutoff for the genes (e.g. top 200) for a much shorter execution time.</span><br>gimp &lt;- gene_importances(<br>  expression_data, <br>  traj$time, <br>  num_permutations = <span>10</span>, <br>  num_threads = <span>8</span>, <br>  ntree = <span>10000</span>,<br>  ntree_perm = <span>1000</span><br>) <br>gimp$qvalue &lt;- p.adjust(gimp$pvalue, <span>"BH"</span>, length(gimp$pvalue))<br>gene_sel &lt;- gimp$gene[gimp$qvalue &lt; <span>.05</span>]<br>expr_sel &lt;- scale_quantile(expression_data[,gene_sel])<br><br><span># Draw a time series heatmap</span><br>time &lt;- traj$time<br>draw_trajectory_heatmap(expr_sel, time)<br><br><span>## Also show the progression groupings</span><br>draw_trajectory_heatmap(expr_sel, time, <br>                        progression_group=group_name)<br></code></pre><figure data-tool="mdnice编辑器"><p><img data-galleryid="" data-imgfileid="100000667" data-ratio="0.7985989492119089" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/y6jd6X97IxJ56FwaiafYyxmBxtLkLx1icgM0hYUQia8CAdI1PgMYsCwwsk53Bmsr1chE4XFGpBCtrKWmCs3zqGeEw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1142" src="https://mmbiz.qpic.cn/sz_mmbiz_png/y6jd6X97IxJ56FwaiafYyxmBxtLkLx1icgM0hYUQia8CAdI1PgMYsCwwsk53Bmsr1chE4XFGpBCtrKWmCs3zqGeEw/640?wx_fmt=png&amp;from=appmsg"></p><p><img data-galleryid="" data-imgfileid="100000665" data-ratio="0.6606367583212736" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/y6jd6X97IxJ56FwaiafYyxmBxtLkLx1icgYEafIF9ib6p6rEDqGMYSOX3h4HgOXovVCOc9usCJWwHoOSANB6vl5rQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1382" src="https://mmbiz.qpic.cn/sz_mmbiz_png/y6jd6X97IxJ56FwaiafYyxmBxtLkLx1icgYEafIF9ib6p6rEDqGMYSOX3h4HgOXovVCOc9usCJWwHoOSANB6vl5rQ/640?wx_fmt=png&amp;from=appmsg"></p><p><br></p><figcaption><br></figcaption></figure><p data-tool="mdnice编辑器">值得注意的是，由于data中的基因数比较多，因此在gene_importances步骤大概花费了7分钟。</p><p data-tool="mdnice编辑器">接下来进行scale data的分析：</p><pre data-tool="mdnice编辑器"><span></span><code><span>######expression_scale----</span><br>space2 &lt;- reduce_dimensionality(expression_scale, <span>"spearman"</span>)<br>draw_trajectory_plot(space2, group_name, contour = <span>TRUE</span>,)<br>traj2 &lt;- infer_trajectory(space2)<br>draw_trajectory_plot(space2, group_name, traj2$path, contour = <span>TRUE</span>)<br><span># warning: setting num_permutations to 10 requires a long time (~30min) to run!</span><br><span># set it to 0 and define a manual cutoff for the genes (e.g. top 200) for a much shorter execution time.</span><br>gimp2 &lt;- gene_importances(<br>  expression_scale, <br>  traj2$time, <br>  num_permutations = <span>10</span>, <br>  num_threads = <span>8</span>, <br>  ntree = <span>10000</span>,<br>  ntree_perm = <span>1000</span><br>) <br>gimp2$qvalue &lt;- p.adjust(gimp2$pvalue, <span>"BH"</span>, length(gimp2$pvalue))<br>gene_sel2 &lt;- gimp2$gene[gimp2$qvalue &lt; <span>.05</span>]<br>expr_sel2 &lt;- scale_quantile(expression_scale[,gene_sel2])<br><br><span># Draw a time series heatmap</span><br>time2 &lt;- traj2$time<br>draw_trajectory_heatmap(expr_sel2, time2)<br><br><span>## Also show the progression groupings</span><br>draw_trajectory_heatmap(expr_sel2, time2, <br>                        progression_group=group_name)<br>draw_trajectory_heatmap(expr_sel2, time2, <br>                        progression_group=group_name,<br>                        show_labels_row=<span>T</span>)<br>modules_2 &lt;- extract_modules(scale_quantile(expr_sel2), traj2$time, verbose = <span>F</span>)<br>draw_trajectory_heatmap(expr_sel2, time2, <br>                        progression_group=group_name,<br>                        show_labels_row=<span>T</span>,<br>                        modules=modules_2)<br></code></pre><figure data-tool="mdnice编辑器"><p><img data-galleryid="" data-imgfileid="100000666" data-ratio="0.751440329218107" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/y6jd6X97IxJ56FwaiafYyxmBxtLkLx1icgyLUNbNawko0eqyiabkK4MGZibb8guKRN6K4qb3JHkoBibyFayTQ87JtZA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1215" src="https://mmbiz.qpic.cn/sz_mmbiz_png/y6jd6X97IxJ56FwaiafYyxmBxtLkLx1icgyLUNbNawko0eqyiabkK4MGZibb8guKRN6K4qb3JHkoBibyFayTQ87JtZA/640?wx_fmt=png&amp;from=appmsg"></p><p><img data-galleryid="" data-imgfileid="100000664" data-ratio="0.6554199569274947" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/y6jd6X97IxJ56FwaiafYyxmBxtLkLx1icgegCltuPKWRmxoSLGoPHIOibaibscGHRCrhcdg7Hvs2qQMBrmsOiaSq04A/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1393" src="https://mmbiz.qpic.cn/sz_mmbiz_png/y6jd6X97IxJ56FwaiafYyxmBxtLkLx1icgegCltuPKWRmxoSLGoPHIOibaibscGHRCrhcdg7Hvs2qQMBrmsOiaSq04A/640?wx_fmt=png&amp;from=appmsg"></p><figcaption><br></figcaption></figure><p data-tool="mdnice编辑器">这次gene_importances大概花费了几秒钟。如上图，我们就得知了4个细胞亚群中差异基因在不同时间上的分布，值得注意的是，这里的time只是个时间顺序的说明，并不代表time 0到1就是分化的起点到终点，因此还是要结合生物学背景进行解释和说明的，另外，SCORPIUS只能展示单一线性的变化，也就是只有1条轨迹推断，适用于少量亚群或者已经有疑似倾向的亚群变化，不太适合太多亚群的轨迹推断。不过由于方便简洁，很适合小范围的探索。</p></section><p><br></p><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/Uk3IMEPc7qUsueL05VPAMA",target="_blank" rel="noopener noreferrer">原文链接</a>
