---
title: "新冠感染死亡患者的肺部单细胞图谱Python全流程"
date: 2024-01-16T05:55:25Z
draft: ["false"]
tags: [
  "fetched",
  "生信技能树"
]
categories: ["Acdemic"]
---
新冠感染死亡患者的肺部单细胞图谱Python全流程 by 生信技能树
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><blockquote data-tool="mdnice编辑器"><p>我们马拉松授课的第四周转录组环节的新叶老师计划把其授课内容细化和延伸，进阶到单细胞转录组，所以钻研了几篇优秀的CNS文章的图表复现，数据分析环节有R和Python，都值得解读和分享，先把文字版分享给大家哈。</p></blockquote></section><p data-tool="mdnice编辑器">这篇文献提供的代码是R与python版本，是个极好的学习单细胞与python的资源。今天先来看看文献背景。</p><figure data-tool="mdnice编辑器"><img data-ratio="1.309077809798271" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjT3jeYVlBXpj30yNW5yEBasU1ooh10I9lEhotwAas03QPhuR9rvGrLSXQBMHoSTLZLYQaJmV0aVgw/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="1388" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjT3jeYVlBXpj30yNW5yEBasU1ooh10I9lEhotwAas03QPhuR9rvGrLSXQBMHoSTLZLYQaJmV0aVgw/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><h2 data-tool="mdnice编辑器"><span>文章信息：</span></h2><blockquote data-tool="mdnice编辑器"><p><span>标题</span>：A molecular single-cell lung atlas of lethal COVID-19<br><span>期刊</span>：Nature<br><span>日期</span>：2021/04/29<br><span>DOI</span>：https://doi.org/10.1038/s41586-021-03569-1<br><span>通讯作者</span>：<span>Benjamin Izar</span>  <span>Jianwen Que</span> 哥伦比亚大学欧文医学中心血液学/肿瘤科医学系<br><span>代码</span>：https://github.com/IzarLab/CUIMC-NYP_COVID_autopsy_lung<br><span>视频</span>：https://www.youtube.com/watch?v=uvyG9yLuNSE<br><span>代码</span>：https://github.com/mousepixels/sanbomics_scripts<br><span>主代码</span>：https://github.com/mousepixels/sanbomics_scripts/blob/main/single_cell_analysis_complete_class.ipynb</p></blockquote><p data-tool="mdnice编辑器">呼吸衰竭是severe SARS-CoV-2感染患者死亡的主要原因，但肺组织水平的宿主反应尚不清楚。本文对19名死于COVID-19的患者和7名对照组患者的肺中约11.6万个细胞核进行了单核RNA测序。<strong>综合分析发现了细胞组成、转录细胞状态和细胞间相互作用的重大变化，从而为了解致命COVID-19的生物学提供了帮助。</strong></p><p data-tool="mdnice编辑器"><span>看这篇文献的时候，心情突然有点那么沉重。</span></p><h2 data-tool="mdnice编辑器"><span>数据情况</span></h2><ul data-tool="mdnice编辑器"><li><section>处理后的数据: https://singlecell.broadinstitute.org/single_cell/study/SCP1219.</section></li><li><section>处理后的数据在GEO中也有 <span>GSE171524</span>(https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE171524).</section></li><li><section>原始数据: https://duos.broadinstitute.org(study ID DUOS-000130).</section></li><li><section>Source data(https://www.nature.com/articles/s41586-021-03569-1#Sec45) are provided with this paper</section></li></ul><p data-tool="mdnice编辑器">总共包含 COVID-19队列（19例样本）与 对照队列Control（7例样本）</p><figure data-tool="mdnice编辑器"><img data-ratio="0.292749658002736" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjT3jeYVlBXpj30yNW5yEBasCVAvvmLLWWKVicPSVy4wVzvTkiarEHyiaJgBY1Kj5Ulbywic3n3fXggMVw/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="1462" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjT3jeYVlBXpj30yNW5yEBasCVAvvmLLWWKVicPSVy4wVzvTkiarEHyiaJgBY1Kj5Ulbywic3n3fXggMVw/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><h2 data-tool="mdnice编辑器"><span>COVID-19的肺细胞landscape</span></h2><p data-tool="mdnice编辑器">使用单细胞核测序，经过质控流程，得到一个肺的图谱：116,314个单细胞，包括COVID-19队列79,636个细胞以及对照队列Control 36,678个细胞。9个主要的细胞类型鉴定如下：</p><ul data-tool="mdnice编辑器"><li><section>epithelial cells (<em>n</em> = 30,070 cells)</section></li><li><section>myeloid cells (<em>n</em> = 29,632)</section></li><li><section>fibroblasts (<em>n</em> = 22,909)</section></li><li><section>endothelial cells (<em>n</em> = 5,386)</section></li><li><section>T and natural killer (NK) lymphocytes (<em>n</em> = 16,751)</section></li><li><section>B lymphocytes and plasma cells (<em>n</em> = 7,236)</section></li><li><section>neuronal cells (<em>n</em> = 2,017)</section></li><li><section>mast cells (<em>n</em> = 1,464)</section></li><li><section>antigen-presenting cells (APCs; primarily dendritic cells) (<em>n</em> = 849)</section></li></ul><figure data-tool="mdnice编辑器"><img data-ratio="0.3224431818181818" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjT3jeYVlBXpj30yNW5yEBasPmw9WvM7t98OVXicU0RMK1vneCwDl51z5cSq3RfE3vtJvWUM8gOpyvg/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="1408" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjT3jeYVlBXpj30yNW5yEBasPmw9WvM7t98OVXicU0RMK1vneCwDl51z5cSq3RfE3vtJvWUM8gOpyvg/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">具体的样本情况以及详细注释：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.9476688867745005" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjT3jeYVlBXpj30yNW5yEBashg2sd0WtcexoxNYjFlEqRicZTicA4LxlumI42F7iaEyy3BqExbqplZyhw/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="1051" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjT3jeYVlBXpj30yNW5yEBashg2sd0WtcexoxNYjFlEqRicZTicA4LxlumI42F7iaEyy3BqExbqplZyhw/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><h2 data-tool="mdnice编辑器"><span>髓系细胞的异常激活</span></h2><p data-tool="mdnice编辑器">髓系细胞是COVID-19肺的主要细胞成分，且比对照组肺更为普遍。我们鉴定了：</p><ul data-tool="mdnice编辑器"><li><section>单核细胞(n = 3,176)</section></li><li><section>单核细胞衍生的巨噬细胞(MDMs;n = 9,534)</section></li><li><section>transitioning MDMs (n = 4,203)</section></li><li><section>常驻肺泡巨噬细胞(resident alveolar macrophages，AMs;n = 12,511)</section></li></ul><p data-tool="mdnice编辑器">它们在diffusion component(DC)分析中被恢复为不同的轨迹，在COVID-19肺部更常见（图2a-c）。来自COVID-19个体的髓系细胞被高度和异常激活。</p><p data-tool="mdnice编辑器">此外：</p><ul data-tool="mdnice编辑器"><li><section>COVID-19肺中的MDMs差异表达激活基因(例如CTSB、CTSD、CTSZ、PSAP)和两个长链非编码rna NEAT1和MALAT1，它们与异常巨噬细胞激活和受损的T细胞免疫相关</section></li><li><section>AMs产生于胎儿单核细胞，可以自我更新，在COVID-19肺中富集并高度激活</section></li></ul><figure data-tool="mdnice编辑器"><img data-ratio="0.5378571428571428" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjT3jeYVlBXpj30yNW5yEBasUvsjzXmbbS4peYSUSezbseV95MQwaYibS3cezibTTQEas3I6b25W6RjA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="1400" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjT3jeYVlBXpj30yNW5yEBasUvsjzXmbbS4peYSUSezbseV95MQwaYibS3cezibTTQEas3I6b25W6RjA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">这些数据表明，髓系细胞是COVID-19中炎症失调的主要来源。</p><h2 data-tool="mdnice编辑器"><span>血浆和T细胞响应</span></h2><p data-tool="mdnice编辑器">使用the variable heavy (<em>IGHV</em>) and light (<em>IGLV</em>) chains鉴定了浆细胞，T/NK细胞群体分为了：CD8<sup>+</sup> T cells (<em>n</em> = 3,561), T regulatory (Treg) cells (<em>n</em> = 649), other CD4<sup>+</sup> T cells (<em>n</em> = 7,586), and NK cells (<em>n</em> = 2,141)。</p><p data-tool="mdnice编辑器">作者发现COVID-19肺部的T细胞丰度没有显著增加，只有细胞因子和T细胞激活和组织驻留相关程序的适度上调（上图G-I）。</p><h2 data-tool="mdnice编辑器"><span>肺泡上皮再生受损</span></h2><p data-tool="mdnice编辑器">上皮细胞分为：</p><ul data-tool="mdnice编辑器"><li><section>肺泡上皮细胞 (AT1 and AT2 cells; <em>n</em> = 20,949)</section></li><li><section>气道上皮细胞 (basal, ciliated, club, goblet, and mucous cells; <em>n</em> = 7,223)</section></li><li><section>一种以表达炎症和细胞周期基因为特征的亚群, 包括 <em>IRF8</em>, <em>B2M</em>, <em>MKI67</em> and <em>TOP2A</em> (‘cycling epithelium’; <em>n</em> = 609)</section></li><li><section>显示细胞外基质(ECM)成分高表达的亚群 <em>COL6A3</em>, <em>COL1A2</em>, and <em>COL3A1</em></section></li></ul><p data-tool="mdnice编辑器">此外：在肺再生过程中，AT2细胞作为AT1细胞的祖细胞，可以分化为AT1；对照组肺中的AT2和AT1细胞形成了不同的簇。这其中有几个比较重要的基因：</p><ul data-tool="mdnice编辑器"><li><section><em>ETV5</em> ：COVID-19 AT2细胞低表达ETV5，维持AT2细胞身份，ETV5表达降低与向AT1细胞分化相关</section></li><li><section><em>CAV1</em>：AT1成熟晚期的标志，在COVID-19肺的AT1细胞中表达水平显著降低</section></li></ul><figure data-tool="mdnice编辑器"><img data-ratio="0.6923611111111111" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjT3jeYVlBXpj30yNW5yEBas213zfeRZCrEiaNKiaMJtCs3LCv9TvadzdUBSdOSKGBzqrpcJyACGjibYA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="1440" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjT3jeYVlBXpj30yNW5yEBas213zfeRZCrEiaNKiaMJtCs3LCv9TvadzdUBSdOSKGBzqrpcJyACGjibYA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><h2 data-tool="mdnice编辑器"><span>COVID-19中的异位簇状细胞</span></h2><p data-tool="mdnice编辑器">在捕获的气道上皮细胞中，我们发现了四种不同的轨迹:</p><ul data-tool="mdnice编辑器"><li><section>KRT5+ TP63+基底细胞basal (n = 534)</section></li><li><section>club细胞(n = 1232)</section></li><li><section>杯状细胞goblet cells (n = 1757)</section></li><li><section>一种细胞较少的轨迹(n = 110)，主要在COVID-19肺部发现，我们将其确定为簇状细胞putative tuft-like cells</section></li></ul><figure data-tool="mdnice编辑器"><img data-ratio="0.4670406732117812" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjT3jeYVlBXpj30yNW5yEBasJc1uRmJGiaOZ3PsEZ9qbWYAviblpb3uwibgxTMG2wlftYXoWMqynFnxog/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="1426" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjT3jeYVlBXpj30yNW5yEBasJc1uRmJGiaOZ3PsEZ9qbWYAviblpb3uwibgxTMG2wlftYXoWMqynFnxog/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">簇状细胞参与气道炎症和肠组织再生，但它们在病毒性肺炎中的作用尚不清楚。本研究中表现如下：</p><ul data-tool="mdnice编辑器"><li><section>新冠肺炎患者上呼吸道的簇状细胞(CHAT+或POU2F3+)数量增加了3倍，且只在新冠肺炎患者肺实质中异位存在，但在对照组肺中不存在（parenchyma为腺细胞组织，实质）</section></li></ul><h2 data-tool="mdnice编辑器"><span>病理性成纤维细胞和肺纤维化</span></h2><p data-tool="mdnice编辑器">COVID-19肺部的成纤维细胞明显多于对照组(图1d)，α-平滑肌肌动蛋白(α-SMA，基因名字ACTA2)免疫组化染色证实了这一发现：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.15618374558303888" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjT3jeYVlBXpj30yNW5yEBasy9DR2a9YTZiatoUs570trMDK1xghbNtvfExsTrXkXTrHOibHWDYdeQ3w/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="1415" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjT3jeYVlBXpj30yNW5yEBasy9DR2a9YTZiatoUs570trMDK1xghbNtvfExsTrXkXTrHOibHWDYdeQ3w/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">纤维化程度(determined by a Sirius red fibrosis score)与疾病持续时间相关(图4a)，表明COVID-19的肺纤维化随着时间的推移而增加：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.42843419788664744" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjT3jeYVlBXpj30yNW5yEBastbuPEESz1ak46NTia1mr0RqOiay4Sw1A7oFqLBdgqbo4zpv6dmF9XVyg/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="1041" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjT3jeYVlBXpj30yNW5yEBastbuPEESz1ak46NTia1mr0RqOiay4Sw1A7oFqLBdgqbo4zpv6dmF9XVyg/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">此外：作者确定了五种成纤维细胞亚型：与对照肺相比，COVID-19肺中病理或中间病理成纤维细胞的频率增加(图4c)</p><ul data-tool="mdnice编辑器"><li><section>肺泡 alveolar (n = 4670)</section></li><li><section>外膜 adventitial (n = 3773)</section></li><li><section>病理 pathological (n = 2322)</section></li><li><section>中间病理 intermediate pathological  (n = 8779)</section></li><li><section>其他(n = 1099) (图4b)</section></li></ul><p data-tool="mdnice编辑器">主要细胞类型之间的配体-受体分析结果显示：</p><ul data-tool="mdnice编辑器"><li><section>在所有细胞中配体-受体相互作用最多的是TGFβ1 - TGFβ受体2和BMP6-ACVR1，它们分别属于TGFβ家族和超家族。TGFβ信号在促进肺纤维化中具有重要作用，并与成纤维细胞介导的ADI27维持有关，而ADI27与DATP细胞状态密切相关。</section></li></ul><figure data-tool="mdnice编辑器"><img data-ratio="0.7698776758409785" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjT3jeYVlBXpj30yNW5yEBasncvJkicEbiaNej5HuJjxuhbTRpJ3SfvvntORicWJ1YSxdN3LClx3mFZtA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="1308" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjT3jeYVlBXpj30yNW5yEBasncvJkicEbiaNej5HuJjxuhbTRpJ3SfvvntORicWJ1YSxdN3LClx3mFZtA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器"><span>这里还有一个有意思的分析</span>：</p><p data-tool="mdnice编辑器">为了研究针对pFBs（pathological fibroblasts）的潜在治疗策略，作者从<span>单核转录组中推断蛋白质活性</span>，然后将pFBs与其他成纤维细胞进行比较。该分析预测pFBs将表现出JunB和JunD活性的增加(图j)，它们通过增强TGFβ和STAT3信号通路诱导小鼠模型的肺纤维化，并与IL-1β的产生增加有关。最后，作者推断出pFBs中的药物作用靶点，并将MMP14和STAT3作为废除pFBs中有害程序的潜在靶点(图j)。</p><p data-tool="mdnice编辑器">有好几个方法可以利用单细胞RNA数据推断蛋白层面的分子活性，下次介绍~</p><p data-tool="mdnice编辑器">数据背景接上一篇：<a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247507955&amp;idx=1&amp;sn=6268e0602b2678db25a2d9587c7b204e&amp;chksm=ea1cd971dd6b5067604198239b9b285f3a015003f41db4ddec1a3e6f98e316c59d89c182f53f&amp;scene=21#wechat_redirect" textvalue="Python图文复现2022|01-文献阅读：致命COVID-19分子单细胞肺图谱" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2" hasload="1">Python图文复现2022|01-文献阅读：致命COVID-19分子单细胞肺图谱</a></p><h2 data-tool="mdnice编辑器"><span>数据获取</span></h2><p data-tool="mdnice编辑器">有三种途径：</p><ul data-tool="mdnice编辑器"><li><section>处理后的数据：single-cell portal: https://singlecell.broadinstitute.org/single_cell/study/SCP1219.</section></li><li><section>处理后的数据还可以在GEO获取：<span>GSE171524</span></section></li><li><section>原始数据：the Broad Data Use and Oversight System: https://duos.broadinstitute.org (study ID DUOS-000130)，但是需要注册，注册要求好像还有丢丢特殊</section></li></ul><p data-tool="mdnice编辑器">这次就从GEO下载吧，下载完后：3个文件，一个处理后的csv表达数据，一个metadata，一个原始count数据压缩包tar</p><figure data-tool="mdnice编辑器"><img data-ratio="0.15759312320916904" data-type="png" data-w="1047" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjQHNmWTFwpMWFK0xicz1XvicaD6Wnv2wJhrkaibDk0my87vIMo1twukVpYIgPeNJL21SG0WYoQnu4CHA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjQHNmWTFwpMWFK0xicz1XvicaD6Wnv2wJhrkaibDk0my87vIMo1twukVpYIgPeNJL21SG0WYoQnu4CHA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">原文代码：https://github.com/IzarLab/CUIMC-NYP_COVID_autopsy_lung</p><p data-tool="mdnice编辑器">但是本次我们使用一个利用这个数据讲python学习的资源，</p><ul data-tool="mdnice编辑器"><li><section>视频如下：https://www.youtube.com/watch?v=uvyG9yLuNSE</section></li></ul><p data-tool="mdnice编辑器">视频相关代码如下：</p><ul data-tool="mdnice编辑器"><li><section>代码：https://github.com/mousepixels/sanbomics_scripts</section></li><li><section>主代码：https://github.com/mousepixels/sanbomics_scripts/blob/main/single_cell_analysis_complete_class.ipynb</section></li></ul><p data-tool="mdnice编辑器">数据下载后，开工！</p><h2 data-tool="mdnice编辑器"><span>环境准备：</span></h2><pre data-tool="mdnice编辑器"><span></span><code><span># 使用conda安装好相关软件</span><br>conda activate scanpy<br><br><span># 导入包，注意dir路径改成自己的</span><br><span>import</span> scanpy <span>as</span> sc<br>dir = <span>'/path/data/GSE171524/GSE171524_RAW/'</span><br></code></pre><h2 data-tool="mdnice编辑器"><span>数据读取</span></h2><p data-tool="mdnice编辑器">GSM5226574_C51样本是个肺对照样本，总共包含6099个细胞，34546个基因。</p><pre data-tool="mdnice编辑器"><span></span><code><span># 读取数据</span><br>adata = sc.read_csv(dir + <span>'GSM5226574_C51ctr_raw_counts.csv'</span>).T<br>adata<br><br><span>#AnnData object with n_obs × n_vars = 6099 × 34546</span><br><br><span># 查看数据维度</span><br>adata.X.shape<br><span>#(6099, 34546)</span><br></code></pre><h2 data-tool="mdnice编辑器"><span>Doublet过滤</span></h2><pre data-tool="mdnice编辑器"><span></span><code><span># pip install scvi-tools</span><br><span>import</span> scvi<br><br><span># 过滤低表达基因以及高变基因选择</span><br>sc.pp.filter_genes(adata, min_cells = <span>10</span>)<br>sc.pp.highly_variable_genes(adata, n_top_genes = <span>2000</span>, subset = <span>True</span>, flavor = <span>'seurat_v3'</span>)<br><br><span># 训练模型</span><br>scvi.model.SCVI.setup_anndata(adata)<br>vae = scvi.model.SCVI(adata)<br>vae.train()<br><span>#GPU available: False, used: False</span><br><span>#TPU available: False, using: 0 TPU cores</span><br><span>#IPU available: False, using: 0 IPUs</span><br><span>#HPU available: False, using: 0 HPUs</span><br><span>#Epoch 400/400: 100%|████████████████████████████████████████████████████████| 400/400 [23:10&lt;00:00,  3.21s/it, loss=320, v_num=1]`Trainer.fit` stopped: `max_epochs=400` reached.</span><br><span>#Epoch 400/400: 100%|████████████████████████████████████████████████████████| 400/400 [23:10&lt;00:00,  3.48s/it, loss=320, v_num=1]</span><br><br>solo = scvi.external.SOLO.from_scvi_model(vae)<br>solo.train()<br><br>df = solo.predict()<br>df[<span>'prediction'</span>] = solo.predict(soft = <span>False</span>)<br>df.index = df.index.map(<span>lambda</span> x: x[:<span>-2</span>])<br>df<br></code></pre><p data-tool="mdnice编辑器">结果：doublet这一列的值越高，表明这个细胞约可能是双包体；</p><figure data-tool="mdnice编辑器"><img data-ratio="0.4362745098039216" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjQHNmWTFwpMWFK0xicz1XvicaUNYoA8HmJ1libNwk8bkEx7OLT7XuL3RrgH5PwnXNQ5P0S9ZuQjDSgXg/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="816" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjQHNmWTFwpMWFK0xicz1XvicaUNYoA8HmJ1libNwk8bkEx7OLT7XuL3RrgH5PwnXNQ5P0S9ZuQjDSgXg/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">预测结果统计：</p><p data-tool="mdnice编辑器">有1245个细胞被预测为双包体，占总细胞的20%左右，这对于10X来说，双包率有点太高了</p><pre data-tool="mdnice编辑器"><span></span><code>df.groupby(<span>'prediction'</span>).count()<br><br>            doublet  singlet<br>prediction                  <br>doublet        <span>1245</span>     <span>1245</span><br>singlet        <span>4854</span>     <span>4854</span><br></code></pre><p data-tool="mdnice编辑器">因此，这里计算一个df值：</p><pre data-tool="mdnice编辑器"><span></span><code>df[<span>'dif'</span>] = df.doublet - df.singlet<br>df<br></code></pre><p data-tool="mdnice编辑器">新增一列df值：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.4161915621436716" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjQHNmWTFwpMWFK0xicz1XvicasBibMEPvciadV67B56qdcerHA5eRbED9UXvQ2yntfowYh7ib8ibh4AMnIg/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="877" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjQHNmWTFwpMWFK0xicz1XvicasBibMEPvciadV67B56qdcerHA5eRbED9UXvQ2yntfowYh7ib8ibh4AMnIg/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">给这个df绘制一个分布图：</p><pre data-tool="mdnice编辑器"><span></span><code><span>import</span> seaborn <span>as</span> sns<br><span>import</span> matplotlib.pyplot <span>as</span> plt<br><br>sns.displot(df[df.prediction == <span>'doublet'</span>], x = <span>'dif'</span>)<br>plt.savefig(dir+<span>"01-df-displot.png"</span>)<br></code></pre><p data-tool="mdnice编辑器">结果图：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.9930915371329879" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjQHNmWTFwpMWFK0xicz1XvicakSmc63ggvxicG0KwlfcbMpeaibb6OSktFpbnWlIcqokLH8EWPMMzf6zQ/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="579" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjQHNmWTFwpMWFK0xicz1XvicakSmc63ggvxicG0KwlfcbMpeaibb6OSktFpbnWlIcqokLH8EWPMMzf6zQ/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">因此，将df大于1的预测为双包体：</p><pre data-tool="mdnice编辑器"><span></span><code>doublets = df[(df.prediction == <span>'doublet'</span>) &amp; (df.dif &gt; <span>1</span>)]<br>doublets<br><br>adata = sc.read_csv(dir+<span>'GSM5226574_C51ctr_raw_counts.csv'</span>).T<br>adata.obs[<span>'doublet'</span>] = adata.obs.index.isin(doublets.index)<br>adata.obs<br></code></pre><p data-tool="mdnice编辑器">预测结果：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.4049493813273341" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjQHNmWTFwpMWFK0xicz1XvicauQUiaXftF0GFtQZYZ46tSXCvHke4oDRRbM5ghiaiaLicusm4qbrNwggsvA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="889" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjQHNmWTFwpMWFK0xicz1XvicauQUiaXftF0GFtQZYZ46tSXCvHke4oDRRbM5ghiaiaLicusm4qbrNwggsvA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">去除：还剩余5618个细胞</p><pre data-tool="mdnice编辑器"><span></span><code>adata = adata[~adata.obs.doublet]<br>adata<br><br>View of AnnData object <span>with</span> n_obs × n_vars = <span>5618</span> × <span>34546</span><br>    obs: <span>'doublet'</span><br></code></pre><h2 data-tool="mdnice编辑器"><span>预处理</span></h2><pre data-tool="mdnice编辑器"><span></span><code><span># 计算线粒体基因比例</span><br>adata.var[<span>'mt'</span>] = adata.var.index.str.startswith(<span>'MT-'</span>)<br>adata.var<br><br><span># 核糖体RNA基因</span><br><span>import</span> pandas <span>as</span> pd<br>ribo_url = <span>"http://software.broadinstitute.org/gsea/msigdb/download_geneset.jsp?geneSetName=KEGG_RIBOSOME&amp;fileType=txt"</span><br>ribo_genes = pd.read_table(ribo_url, skiprows=<span>2</span>, header = <span>None</span>)<br>ribo_genes<br><br><span>#              0</span><br><span># 0          FAU</span><br><span># 1       MRPL13</span><br><span># 2        RPL10</span><br><span># 3       RPL10A</span><br><span># 4       RPL10L</span><br><span># ..         ...</span><br><span># 83        RPS9</span><br><span># 84        RPSA</span><br><span># 85     RSL24D1</span><br><span># 86  RSL24D1P11</span><br><span># 87       UBA52</span><br><br><span># [88 rows x 1 columns]</span><br><br>adata.var[<span>'ribo'</span>] = adata.var_names.isin(ribo_genes[<span>0</span>].values)<br></code></pre><p data-tool="mdnice编辑器">接着计算qc指标</p><pre data-tool="mdnice编辑器"><span></span><code>sc.pp.calculate_qc_metrics(adata, qc_vars=[<span>'mt'</span>, <span>'ribo'</span>], percent_top=<span>None</span>, log1p=<span>False</span>, inplace=<span>True</span>)<br>adata.var.sort_values(<span>'n_cells_by_counts'</span>)<br></code></pre><p data-tool="mdnice编辑器">结果如下：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.31277533039647576" data-type="png" data-w="1135" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjQHNmWTFwpMWFK0xicz1Xvica6bOM1jYtv3UBA2gJMJ9M4SvadO2ZnJj9j5lnsZUD4vy402NF4S3PGQ/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjQHNmWTFwpMWFK0xicz1Xvica6bOM1jYtv3UBA2gJMJ9M4SvadO2ZnJj9j5lnsZUD4vy402NF4S3PGQ/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">低表达过滤：</p><pre data-tool="mdnice编辑器"><span></span><code>sc.pp.filter_genes(adata, min_cells=<span>3</span>)<br>adata.var.sort_values(<span>'n_cells_by_counts'</span>)<br>adata.obs.sort_values(<span>'n_genes_by_counts'</span>)<br><br><span># 绘制小提琴图</span><br>sc.pl.violin(adata, [<span>'n_genes_by_counts'</span>, <span>'total_counts'</span>, <span>'pct_counts_mt'</span>, <span>'pct_counts_ribo'</span>], jitter=<span>0.4</span>, multi_panel=<span>True</span>)<br>plt.savefig(dir+<span>"02-qc_violin.png"</span>)<br></code></pre><p data-tool="mdnice编辑器">结果图：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.24537607891491986" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjQHNmWTFwpMWFK0xicz1Xvica9UwfTt6z20Vyl5AppTzM0ZiaxxAybavVOjMdaOXEItt13qjib9k6Mib6w/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="1622" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjQHNmWTFwpMWFK0xicz1Xvica9UwfTt6z20Vyl5AppTzM0ZiaxxAybavVOjMdaOXEItt13qjib9k6Mib6w/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">按照分位数来过滤细胞：</p><pre data-tool="mdnice编辑器"><span></span><code><span>import</span> numpy <span>as</span> np<br>upper_lim = np.quantile(adata.obs.n_genes_by_counts.values, <span>.98</span>)<br><span>#upper_lim = 3000</span><br>upper_lim<br>adata = adata[adata.obs.n_genes_by_counts &lt; upper_lim]<br>adata.obs<br></code></pre><p data-tool="mdnice编辑器">过滤之后：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.26666666666666666" data-type="png" data-w="1335" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjQHNmWTFwpMWFK0xicz1Xvica3cujmNibXDwVJc8lbeiao0U7w4QHDYawRy8HNTIjL90Bn7ZbxD0z4iaEw/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjQHNmWTFwpMWFK0xicz1Xvica3cujmNibXDwVJc8lbeiao0U7w4QHDYawRy8HNTIjL90Bn7ZbxD0z4iaEw/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">线粒体比例与核糖体比例：</p><pre data-tool="mdnice编辑器"><span></span><code>adata = adata[adata.obs.pct_counts_mt &lt; <span>20</span>]<br>adata = adata[adata.obs.pct_counts_ribo &lt; <span>2</span>]<br>adata<br></code></pre><p data-tool="mdnice编辑器">过滤后：</p><pre data-tool="mdnice编辑器"><span></span><code>View of AnnData object <span>with</span> n_obs × n_vars = <span>5489</span> × <span>24080</span><br>    obs: <span>'doublet'</span>, <span>'n_genes_by_counts'</span>, <span>'total_counts'</span>, <span>'total_counts_mt'</span>, <span>'pct_counts_mt'</span>, <span>'total_counts_ribo'</span>, <span>'pct_counts_ribo'</span><br>    var: <span>'mt'</span>, <span>'ribo'</span>, <span>'n_cells_by_counts'</span>, <span>'mean_counts'</span>, <span>'pct_dropout_by_counts'</span>, <span>'total_counts'</span>, <span>'n_cells'</span><br></code></pre><h2 data-tool="mdnice编辑器"><span>标准化Normalization</span></h2><p data-tool="mdnice编辑器">标准化前后的区别可看：adata.X.sum(axis = 1)值的变化</p><pre data-tool="mdnice编辑器"><span></span><code>adata.X.sum(axis = <span>1</span>)<br><br><span>#normalize every cell to 10,000 UMI</span><br>sc.pp.normalize_total(adata, target_sum=<span>1e4</span>) <br>adata.X.sum(axis = <span>1</span>)<br><br><span>#change to log counts</span><br>sc.pp.log1p(adata) <br>adata.X.sum(axis = <span>1</span>)<br><br>adata.raw = adata<br></code></pre><h2 data-tool="mdnice编辑器"><span>聚类Clustering</span></h2><pre data-tool="mdnice编辑器"><span></span><code><span># 高变基因选择以及可视化</span><br>sc.pp.highly_variable_genes(adata, n_top_genes = <span>2000</span>)<br>sc.pl.highly_variable_genes(adata)<br>plt.savefig(dir+<span>"03-highly_variable_genes.png"</span>)<br></code></pre><p data-tool="mdnice编辑器">结果图：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.3959778085991678" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjQHNmWTFwpMWFK0xicz1XvicaZwYibHP1pc4vwAKEpT9q0wjvKPvKj7lCyiaRP7BCmsMH3AIlFtzcYHpg/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="1442" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjQHNmWTFwpMWFK0xicz1XvicaZwYibHP1pc4vwAKEpT9q0wjvKPvKj7lCyiaRP7BCmsMH3AIlFtzcYHpg/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">选择主成分：</p><pre data-tool="mdnice编辑器"><span></span><code>adata = adata[:, adata.var.highly_variable]<br>sc.pp.regress_out(adata, [<span>'total_counts'</span>, <span>'pct_counts_mt'</span>, <span>'pct_counts_ribo'</span>])<br>sc.pp.scale(adata, max_value=<span>10</span>)<br>sc.tl.pca(adata, svd_solver=<span>'arpack'</span>)<br>sc.pl.pca_variance_ratio(adata, log=<span>True</span>, n_pcs = <span>50</span>)<br>plt.savefig(dir+<span>"04-pca_variance.png"</span>)<br></code></pre><p data-tool="mdnice编辑器">结果图：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.7319316688567674" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjQHNmWTFwpMWFK0xicz1XvicazvicEg0pFQ3knY1OOTTl9ORVZMkxGXkPKlP3ibh8qhXiav0Xy0k0XuicPg/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="761" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjQHNmWTFwpMWFK0xicz1XvicazvicEg0pFQ3knY1OOTTl9ORVZMkxGXkPKlP3ibh8qhXiav0Xy0k0XuicPg/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">这里选择30个PCs，然后聚类：</p><pre data-tool="mdnice编辑器"><span></span><code>sc.pp.neighbors(adata, n_pcs = <span>30</span>)<br>sc.tl.umap(adata)<br>sc.pl.umap(adata)<br>plt.savefig(dir+<span>"05-umap.png"</span>)<br></code></pre><p data-tool="mdnice编辑器">结果图：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.7942857142857143" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjQHNmWTFwpMWFK0xicz1Xvica74MTL3jkNrxY0yP0cFHmBNzVnWVpKsydSYuiaDglOyxQXRElWL3JGWA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="700" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjQHNmWTFwpMWFK0xicz1Xvica74MTL3jkNrxY0yP0cFHmBNzVnWVpKsydSYuiaDglOyxQXRElWL3JGWA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">使用leiden在低维空间可视化：</p><pre data-tool="mdnice编辑器"><span></span><code>sc.tl.leiden(adata, resolution = <span>0.5</span>)<br>sc.pl.umap(adata, color=[<span>'leiden'</span>])<br>plt.savefig(dir+<span>"05-umap_leiden.png"</span>)<br></code></pre><p data-tool="mdnice编辑器">结果图：聚成了11类</p><figure data-tool="mdnice编辑器"><img data-ratio="0.7120418848167539" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjQHNmWTFwpMWFK0xicz1XvicaWrc5vs7fKOwpGELYcMJkp8q8o1g0ickML9lIsiajjkamxLrqJOwrcoog/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="764" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjQHNmWTFwpMWFK0xicz1XvicaWrc5vs7fKOwpGELYcMJkp8q8o1g0ickML9lIsiajjkamxLrqJOwrcoog/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">单个样本的分析演示到这里，下期进行所有样本整合分析~</p><h2 data-tool="mdnice编辑器"><span>读取数据</span></h2><p data-tool="mdnice编辑器">先定义一个函数，批量运行多个样本：这里一定要注意缩进问题</p><pre data-tool="mdnice编辑器"><span></span><code><span><span>def</span> <span>pp</span><span>(csv_path)</span>:</span>    adata = sc.read_csv(csv_path).T    sc.pp.filter_genes(adata, min_cells = <span>10</span>)    sc.pp.highly_variable_genes(adata, n_top_genes = <span>2000</span>, subset = <span>True</span>, flavor = <span>'seurat_v3'</span>)    scvi.model.SCVI.setup_anndata(adata)    vae = scvi.model.SCVI(adata)    vae.train()    solo = scvi.external.SOLO.from_scvi_model(vae)    solo.train()    df = solo.predict()    df[<span>'prediction'</span>] = solo.predict(soft = <span>False</span>)    df.index = df.index.map(<span>lambda</span> x: x[:<span>-2</span>])    df[<span>'dif'</span>] = df.doublet - df.singlet    doublets = df[(df.prediction == <span>'doublet'</span>) &amp; (df.dif &gt; <span>1</span>)]    adata = sc.read_csv(csv_path).T    adata.obs[<span>'Sample'</span>] = csv_path.split(<span>'_'</span>)[<span>2</span>] <span>#'raw_counts/GSM5226574_C51ctr_raw_counts.csv'</span>    adata.obs[<span>'doublet'</span>] = adata.obs.index.isin(doublets.index)    adata = adata[~adata.obs.doublet]    sc.pp.filter_cells(adata, min_genes=<span>200</span>) <span>#get rid of cells with fewer than 200 genes</span>    <span>#sc.pp.filter_genes(adata, min_cells=3) #get rid of genes that are found in fewer than 3 cells</span>    adata.var[<span>'mt'</span>] = adata.var_names.str.startswith(<span>'mt-'</span>)  <span># annotate the group of mitochondrial genes as 'mt'</span>    adata.var[<span>'ribo'</span>] = adata.var_names.isin(ribo_genes[<span>0</span>].values)    sc.pp.calculate_qc_metrics(adata, qc_vars=[<span>'mt'</span>, <span>'ribo'</span>], percent_top=<span>None</span>, log1p=<span>False</span>, inplace=<span>True</span>)    upper_lim = np.quantile(adata.obs.n_genes_by_counts.values, <span>.98</span>)    adata = adata[adata.obs.n_genes_by_counts &lt; upper_lim]    adata = adata[adata.obs.pct_counts_mt &lt; <span>20</span>]    adata = adata[adata.obs.pct_counts_ribo &lt; <span>2</span>]    <span>return</span> adata</code></pre><p data-tool="mdnice编辑器">这个过程比较久，会依次读取GSE171524数据集中的26例样本，并进行上面的函数里面定义的分析。</p><p data-tool="mdnice编辑器">这个过程中就可以跑去听听视频了。</p><pre data-tool="mdnice编辑器"><span></span><code><span>import</span> os<span># 注意dir改成自己的路径</span>dir = <span>'/path/data/GSE171524/'</span>out = []<span>for</span> file <span>in</span> os.listdir(dir+<span>'raw_counts/'</span>):    out.append(pp(dir + <span>'raw_counts/'</span> + file))</code></pre><p data-tool="mdnice编辑器">将多个样本连接合并在一起，合并后共有105264个细胞：</p><pre data-tool="mdnice编辑器"><span></span><code>adata = sc.concat(out)adataAnnData object <span>with</span> n_obs × n_vars = <span>105264</span> × <span>29236</span>    obs: <span>'Sample'</span>, <span>'doublet'</span>, <span>'n_genes'</span>, <span>'n_genes_by_counts'</span>, <span>'total_counts'</span>, <span>'total_counts_mt'</span>, <span>'pct_counts_mt'</span>, <span>'total_counts_ribo'</span>, <span>'pct_counts_ribo'</span>    var: <span>'n_cells'</span></code></pre><p data-tool="mdnice编辑器">过滤并保存：</p><pre data-tool="mdnice编辑器"><span></span><code>sc.pp.filter_genes(adata, min_cells = <span>10</span>)<span>from</span> scipy.sparse <span>import</span> csr_matrixadata.X = csr_matrix(adata.X)adata.Xadata.write_h5ad(dir+<span>'combined.h5ad'</span>)</code></pre><h2 data-tool="mdnice编辑器"><span>预处理</span></h2><p data-tool="mdnice编辑器">先读取上次保存的数据：105264个细胞</p><pre data-tool="mdnice编辑器"><span></span><code><span>import</span> scanpy <span>as</span> sc<span>import</span> scvi<span>import</span> seaborn <span>as</span> sns<span>import</span> numpy <span>as</span> np<span>import</span> pandas <span>as</span> pd<span>import</span> matplotlib.pyplot <span>as</span> plt<span># 注意dir改成自己的路径</span>dir = <span>'/path/data/GSE171524/'</span>adata = sc.read_h5ad(dir+<span>'combined.h5ad'</span>)adataAnnData object <span>with</span> n_obs × n_vars = <span>105264</span> × <span>29236</span>    obs: <span>'Sample'</span>, <span>'doublet'</span>, <span>'n_genes'</span>, <span>'n_genes_by_counts'</span>, <span>'total_counts'</span>, <span>'total_counts_mt'</span>, <span>'pct_counts_mt'</span>, <span>'total_counts_ribo'</span>, <span>'pct_counts_ribo'</span>    var: <span>'n_cells'</span></code></pre><p data-tool="mdnice编辑器">每个样本中的各种指标统计：</p><pre data-tool="mdnice编辑器"><span></span><code>adata.obs.groupby(<span>'Sample'</span>).count()</code></pre><p data-tool="mdnice编辑器">共26个样本：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.6082406801831263" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSsWWLzNh4H4qhLSBbwsudAsdjpQe6wQibI6yCGIeszYKRkibkVhXRUG5bso9FVyR3JIicgIPJM0vZsw/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="1529" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSsWWLzNh4H4qhLSBbwsudAsdjpQe6wQibI6yCGIeszYKRkibkVhXRUG5bso9FVyR3JIicgIPJM0vZsw/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><h2 data-tool="mdnice编辑器"><span>低表达过滤以及数据标准化</span></h2><pre data-tool="mdnice编辑器"><span></span><code>sc.pp.filter_genes(adata, min_cells = <span>100</span>)adata.layers[<span>'counts'</span>] = adata.X.copy()sc.pp.normalize_total(adata, target_sum = <span>1e4</span>)sc.pp.log1p(adata)adata.raw = adata<span># 查看每个细胞的指标</span>adata.obs.head()</code></pre><p data-tool="mdnice编辑器">结果图：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.09708273553323768" data-type="png" data-w="2091" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSsWWLzNh4H4qhLSBbwsudA4FvIz4I9t8MlaRX2tK4gicU0PhFJvbJr3ccMD25y1WUY7nqkh0SzQCw/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSsWWLzNh4H4qhLSBbwsudA4FvIz4I9t8MlaRX2tK4gicU0PhFJvbJr3ccMD25y1WUY7nqkh0SzQCw/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><h2 data-tool="mdnice编辑器"><span>去Sample间的批次以及聚类</span></h2><p data-tool="mdnice编辑器">这个过程运行时间也会稍长一些</p><pre data-tool="mdnice编辑器"><span></span><code>scvi.model.SCVI.setup_anndata(adata, layer = <span>"counts"</span>, categorical_covariate_keys=[<span>"Sample"</span>], continuous_covariate_keys=[<span>'pct_counts_mt'</span>, <span>'total_counts'</span>, <span>'pct_counts_ribo'</span>])model = scvi.model.SCVI(adata)<span>#may take a while without GPU</span>model.train()adata.obsm[<span>'X_scVI'</span>] = model.get_latent_representation()adata.layers[<span>'scvi_normalized'</span>] = model.get_normalized_expression(library_size = <span>1e4</span>)sc.pp.neighbors(adata, use_rep = <span>'X_scVI'</span>)sc.tl.umap(adata)sc.tl.leiden(adata, resolution = <span>0.5</span>)sc.pl.umap(adata, color = [<span>'leiden'</span>, <span>'Sample'</span>], frameon = <span>False</span>)plt.savefig(dir+<span>"01-intergration_umap.png"</span>)</code></pre><p data-tool="mdnice编辑器">结果图：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.3020354563361786" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSsWWLzNh4H4qhLSBbwsudAyOibgH7ubfSBOL2206JKbbDRcWYicKaFr4fxaVjlCicM96rMg838lkzyQ/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="1523" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSsWWLzNh4H4qhLSBbwsudAyOibgH7ubfSBOL2206JKbbDRcWYicKaFr4fxaVjlCicM96rMg838lkzyQ/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">保存数据：</p><pre data-tool="mdnice编辑器"><span></span><code>adata.write_h5ad(dir + <span>'integrated.h5ad'</span>)</code></pre><h2 data-tool="mdnice编辑器"><span>差异表达分析</span></h2><p data-tool="mdnice编辑器">使用没有矫正的数据做差异表达分析</p><pre data-tool="mdnice编辑器"><span></span><code><span># 更改聚类数</span>sc.tl.leiden(adata, resolution = <span>1</span>)sc.tl.rank_genes_groups(adata, <span>'leiden'</span>)<span># 可视化</span>sc.pl.rank_genes_groups(adata, n_genes=<span>20</span>, sharey=<span>False</span>)plt.savefig(dir+<span>"02-intergration_rank_genes_groups.png"</span>, dpi=<span>300</span>)</code></pre><p data-tool="mdnice编辑器">部分cluster的top基因：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.3537981269510926" data-type="png" data-w="1922" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSsWWLzNh4H4qhLSBbwsudAgaW7wsoqf3HsOnILyzI8fgCelBkq510ntOBFAQCtWLNcRnCrFzTUFg/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSsWWLzNh4H4qhLSBbwsudAgaW7wsoqf3HsOnILyzI8fgCelBkq510ntOBFAQCtWLNcRnCrFzTUFg/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><pre data-tool="mdnice编辑器"><span></span><code>markers = sc.get.rank_genes_groups_df(adata, <span>None</span>)markers = markers[(markers.pvals_adj &lt; <span>0.05</span>) &amp; (markers.logfoldchanges &gt; <span>.5</span>)]markers<span># 保存</span>markers.to_csv(dir+<span>'markers.csv'</span>)</code></pre><p data-tool="mdnice编辑器">差异结果：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.39243924392439244" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSsWWLzNh4H4qhLSBbwsudAYsYVS3bxiaFuZp1rVibAxOzLFxnRT1rFWtg8pvVaWYMNkEh6iats4hUCA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="1111" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSsWWLzNh4H4qhLSBbwsudAYsYVS3bxiaFuZp1rVibAxOzLFxnRT1rFWtg8pvVaWYMNkEh6iats4hUCA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">使用模型标准化后的值做差异表达分析：</p><pre data-tool="mdnice编辑器"><span></span><code>markers_scvi = model.differential_expression(groupby = <span>'leiden'</span>)<span># 设定阈值</span>markers_scvi = markers_scvi[(markers_scvi[<span>'is_de_fdr_0.05'</span>]) &amp; (markers_scvi.lfc_mean &gt; <span>.5</span>)]markers_scvi<span># 保存</span>markers_scvi.to_csv(dir+<span>'scvi_markers.csv'</span>)</code></pre><p data-tool="mdnice编辑器">结果：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.21399798590130917" data-type="png" data-w="1986" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSsWWLzNh4H4qhLSBbwsudAaW6svnhBKh7dG0BADlJOZicmCrgok9bib4BIyZN1QbyibxZP46nQHEqDA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSsWWLzNh4H4qhLSBbwsudAaW6svnhBKh7dG0BADlJOZicmCrgok9bib4BIyZN1QbyibxZP46nQHEqDA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">重新聚类后的结果可视化：总共得到34个cluster</p><pre data-tool="mdnice编辑器"><span></span><code>sc.pl.umap(adata, color = [<span>'leiden'</span>], frameon = <span>False</span>, legend_loc = <span>"on data"</span>)plt.savefig(dir+<span>"02-intergration_umap_1.png"</span>)<span>## 有多少cluster</span>adata.obs[<span>'leiden'</span>]<span>#Name: leiden, Length: 105264, dtype: category</span><span>#Categories (34, object): ['0', '1', '2', '3', ..., '30', '31', '32', '33']</span></code></pre><p data-tool="mdnice编辑器">结果图：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.7543604651162791" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSsWWLzNh4H4qhLSBbwsudAribduRf2SmzJquVibHxJVB8Mwzyp1uxTeaO7gOmicV56328kuWuhZ7JNA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="688" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSsWWLzNh4H4qhLSBbwsudAribduRf2SmzJquVibHxJVB8Mwzyp1uxTeaO7gOmicV56328kuWuhZ7JNA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><h2 data-tool="mdnice编辑器"><span>细胞类型注释</span></h2><p data-tool="mdnice编辑器">文章中进行了三次注释，第一次注释大类，主要为9个类：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.5281602002503129" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSsWWLzNh4H4qhLSBbwsudAoEsViape5TUE7IcAdjaDotSUMzDOSQE7L87c2KiafRNw7a07oticmfjfA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="799" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSsWWLzNh4H4qhLSBbwsudAoEsViape5TUE7IcAdjaDotSUMzDOSQE7L87c2KiafRNw7a07oticmfjfA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">这几个类的基因文章中没有提供，就用我们自己收集的基因来进行注释好了。</p><p data-tool="mdnice编辑器">在视频资源中，视频speaker老师给了一张图：是目前免疫细胞分类很详细的一张图了：</p><p data-tool="mdnice编辑器">https://learn.cellsignal.com/hubfs/landing-pages/2019/18-IMM-18284/18-IMM_18284-Human%20Markers%20PWHO-digital.pdf</p><figure data-tool="mdnice编辑器"><img data-ratio="0.6668777707409753" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSsWWLzNh4H4qhLSBbwsudATrqDfwqG8eSv5XZztRddcy9hWzM3FJtVTicVuBC1b4uDPT7mK9wrkFA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="1579" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSsWWLzNh4H4qhLSBbwsudATrqDfwqG8eSv5XZztRddcy9hWzM3FJtVTicVuBC1b4uDPT7mK9wrkFA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">基因可视化：</p><pre data-tool="mdnice编辑器"><span></span><code>markers = [<span>'EPCAM'</span>,<span>'KRT8'</span>,<span>'KRT18'</span>,<span>'KRT19'</span>,  <span># epithelial cells</span>          <span>'CD68'</span>, <span>'CTSS'</span>, <span>'FCN1'</span>,<span>'CD163'</span>,   <span># myeloid cells</span>          <span>'ACTA2'</span>, <span>'DCN'</span>, <span>'ACTB'</span> , <span># fibroblasts</span>          <span>'PECAM1'</span>,<span>'CD34'</span>,<span>'VWF'</span>,   <span># endothelial cells</span>          <span>'PTPRC'</span>,<span>'CD3D'</span>,<span>'CD3E'</span>,<span>'CD3G'</span>,<span>'CD8A'</span>, <span>'CD4'</span>, <span># T and natural killer (NK) lymphocytes</span>          <span>'CD79A'</span>, <span>'MS4A1'</span>,<span>'CD19'</span>,  <span># B lymphocytes and plasma cells</span>          <span>'SNAP25'</span>, <span>'SYT1'</span>,         <span># neuronal cells</span>          <span>'CPA3'</span>,                   <span># mast cells</span>          <span>'CST3'</span>, <span>'LAMP3'</span>, <span>'HLA-DQB2'</span>, <span>'HLA-DPB1'</span>, <span>'BIRC3'</span>, <span># antigen-presenting cells (APCs; primarily dendritic cells) </span>          <span>'MKI67'</span>,<span>'TOP2A'</span>, <span># Cycling</span>           <span>'HBB'</span>,<span>'HBD'</span>     <span># Erythroid</span>          ]<span># 看某一个基因的表达情况</span>markers[markers.names==<span>"HBB"</span>]markers[markers.group==<span>"30"</span>]<span>#, layer = 'scvi_normalized'</span><span># , vmax = 5</span>sc.pl.umap(adata, color = [<span>"HBB"</span>], frameon = <span>False</span>, layer = <span>'scvi_normalized'</span>, vmax =<span>4</span>)plt.savefig(dir+<span>"03-intergration_umap_Erythroid_1.png"</span>)</code></pre><p data-tool="mdnice编辑器">以marker为基础，绘制如下类似图，vmax参数可以进行调整让结果看起来更明显，注视不出来的可以看看每个cluster高表达的基因，查一查功能就立马可以推断出来了：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.36507936507936506" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSsWWLzNh4H4qhLSBbwsudAps6vXbv03iaXu5kp63bCamMyHyPhTouYaN4X8z6fyCu8EdicJVeHG1BA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="1323" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSsWWLzNh4H4qhLSBbwsudAps6vXbv03iaXu5kp63bCamMyHyPhTouYaN4X8z6fyCu8EdicJVeHG1BA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">结合marker表达以及cluster特异性高表达基因：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.5687536571094207" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSsWWLzNh4H4qhLSBbwsudAArqHeEw92aEFkuUiaAQX4wibHWNEH4hvmI1ClFN1iaQReJCDK0oZ4v4PQ/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="1709" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSsWWLzNh4H4qhLSBbwsudAArqHeEw92aEFkuUiaAQX4wibHWNEH4hvmI1ClFN1iaQReJCDK0oZ4v4PQ/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">细胞注释如下：</p><pre data-tool="mdnice编辑器"><span></span><code><span>for</span> x <span>in</span> range(<span>0</span>,<span>34</span>):    print(<span>f'"<span>{x}</span>":"", '</span>)cell_type = {<span>"0"</span>:<span>"T Cell"</span>,<span>"1"</span>:<span>"Epithelial Cell"</span>,<span>"2"</span>:<span>"Myeloid Cell"</span>,<span>"3"</span>:<span>"Fibroblast"</span>,<span>"4"</span>:<span>"Myeloid Cell"</span>,<span>"5"</span>:<span>"T Cell"</span>,<span>"6"</span>:<span>"Myeloid Cell"</span>,<span>"7"</span>:<span>"Epithelial Cell"</span>,<span>"8"</span>:<span>"Endothelial"</span>,<span>"9"</span>:<span>"Fibroblast"</span>,<span>"10"</span>:<span>"Myeloid Cell"</span>,<span>"11"</span>:<span>"pDC"</span>,<span>"12"</span>:<span>"Epithelial Cell"</span>,<span>"13"</span>:<span>"Fibroblast"</span>,<span>"14"</span>:<span>"Fibroblast"</span>,<span>"15"</span>:<span>"Epithelial Cell"</span>,<span>"16"</span>:<span>"Epithelial Cell"</span>,<span>"17"</span>:<span>"Myeloid Cell"</span>,<span>"18"</span>:<span>"Epithelial Cell"</span>,<span>"19"</span>:<span>"Epithelial Cell"</span>,<span>"20"</span>:<span>"Neuronal Cell"</span>,<span>"21"</span>:<span>"Epithelial Cell"</span>,<span>"22"</span>:<span>"B Cell"</span>,<span>"23"</span>:<span>"Mast Cell"</span>,<span>"24"</span>:<span>"T Cell"</span>,<span>"25"</span>:<span>"pDC"</span>,<span>"26"</span>:<span>"Myeloid Cell"</span>,<span>"27"</span>:<span>"Endothelial"</span>,<span>"28"</span>:<span>"Fibroblast"</span>,<span>"29"</span>:<span>"Myeloid Cell"</span>,<span>"30"</span>:<span>"Erythroid"</span>,<span>"31"</span>:<span>"B Cell"</span>,<span>"32"</span>:<span>"Neuronal Cell"</span>,<span>"33"</span>:<span>"Myeloid Cell"</span>}</code></pre><pre data-tool="mdnice编辑器"><span></span><code>adata.obs[<span>'cell type'</span>] = adata.obs.leiden.map(cell_type)sc.pl.umap(adata, color = [<span>'cell type'</span>], frameon = <span>False</span>, legend_loc = <span>"on data"</span>)plt.tight_layout()plt.savefig(dir+<span>"04-intergration_umap_anno.png"</span>, dpi=<span>300</span>)adataadata.uns[<span>'scvi_markers'</span>] = markers_scviadata.uns[<span>'markers'</span>] = markers<span># 保存</span>adata.write_h5ad(dir + <span>'integrated_anno.h5ad'</span>)model.save(dir + <span>'model.model'</span>)</code></pre><p data-tool="mdnice编辑器">细胞注释结果如下：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.4461839530332681" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSsWWLzNh4H4qhLSBbwsudAkY8bqPMKhlczoFyMK15j4nxvBL6TKNb9hm4RkSZ8YICHMc55nXgLjA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="1533" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSsWWLzNh4H4qhLSBbwsudAkY8bqPMKhlczoFyMK15j4nxvBL6TKNb9hm4RkSZ8YICHMc55nXgLjA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">这里注释出来了一串文献中没有的红细胞。</p><p data-tool="mdnice编辑器">文献中的Fig1如下：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.3861111111111111" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSSvpaJznuhelCGY8RMQVg0fcEjqEgNHpLaouImiaaEIuYUhciapD7jJVPtBU3dqFtGwnWMbx6T8CsQ/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSSvpaJznuhelCGY8RMQVg0fcEjqEgNHpLaouImiaaEIuYUhciapD7jJVPtBU3dqFtGwnWMbx6T8CsQ/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"><figcaption>Fig1</figcaption></figure><p data-tool="mdnice编辑器">如果要绘制Fig1，需要对上次笔记(<a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247508067&amp;idx=1&amp;sn=32b6eb2f8b053b4046fc3a83ce15b7b8&amp;chksm=ea1ca6e1dd6b2ff76eccc364f604f71a49e1bb2a7f08799b3c4743d80b9c5189dae814b92c42&amp;scene=21#wechat_redirect" textvalue="Python图文复现2022|03-多样本整合分析" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2" hasload="1">Python图文复现2022|03-多样本整合分析</a>)中的注释再进行进一步的详细注释，上次注释结果：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.4398148148148148" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSSvpaJznuhelCGY8RMQVg0Gt1iaKnxibETWJOU9RVpEneNKNc86S9l6bwz6mVXP2N4O7ibBNwchh4iaA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSSvpaJznuhelCGY8RMQVg0Gt1iaKnxibETWJOU9RVpEneNKNc86S9l6bwz6mVXP2N4O7ibBNwchh4iaA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><h2 data-tool="mdnice编辑器"><span>数据读取</span></h2><pre data-tool="mdnice编辑器"><span></span><code><span>import</span> scanpy <span>as</span> sc<br><span>import</span> scvi<br><span>import</span> seaborn <span>as</span> sns<br><span>import</span> numpy <span>as</span> np<br><span>import</span> pandas <span>as</span> pd<br><span>import</span> matplotlib.pyplot <span>as</span> plt<br><br><span># 注意dir改成自己的路径</span><br>dir = <span>'/dir/data/GSE171524/'</span><br>adata = sc.read_h5ad(dir+<span>'integrated_anno.h5ad'</span>)<br>adata<br><br></code></pre><h2 data-tool="mdnice编辑器"><span>详细注释</span></h2><p data-tool="mdnice编辑器">marker采用来自数据库：PanglaoDB https://panglaodb.se/</p><p data-tool="mdnice编辑器">首先是B，T/NK细胞，有：</p><ul data-tool="mdnice编辑器"><li><section>CD4+ T cells：CD4</section></li><li><section>CD8+ T cells：CD8A, CD8B</section></li><li><section>NK/T cells：NCAM1</section></li><li><section>Plasma cells：MZB1</section></li></ul><p data-tool="mdnice编辑器">上皮群：</p><ul data-tool="mdnice编辑器"><li><section>AT1：AGER</section></li><li><section>AT2：SFTPC</section></li><li><section>Airway epithelial：SEC14L3, CDH1</section></li></ul><p data-tool="mdnice编辑器">髓系：</p><ul data-tool="mdnice编辑器"><li><section>Macrophages：MRC1</section></li><li><section>Dendritic cells：ZBTB46</section></li><li><section>Monocytes：APOBEC3A</section></li></ul><p data-tool="mdnice编辑器">基质细胞：</p><ul data-tool="mdnice编辑器"><li><section>smooth muscle cells：RGS5，ACTA2</section></li></ul><pre data-tool="mdnice编辑器"><span></span><code>markers = adata.uns[<span>'markers'</span>]<br><br><span># 看某一个基因的表达情况</span><br>markers[markers.names==<span>"RGS5"</span>]<br><br><span>#, layer = 'scvi_normalized'</span><br><span># , vmax = 5</span><br>sc.pl.umap(adata, color = [<span>"RGS5"</span>], frameon = <span>False</span>, layer = <span>'scvi_normalized'</span>, vmax =<span>2</span>)<br>plt.savefig(dir+<span>"03-intergration_umap_gene_RGS5.png"</span>)<br></code></pre><p data-tool="mdnice编辑器">先给出每个cluster编号以及需要填充的空格，在后续的操作中这个地方会填上每个注释结果：</p><pre data-tool="mdnice编辑器"><span></span><code><span>for</span> x <span>in</span> range(<span>0</span>,<span>34</span>):<br>    print(<span>f'"<span>{x}</span>":"", '</span>)<br><br>cell_type_sub = {<span>"0"</span>:<span>"CD4+ T cells"</span>,<br><span>"1"</span>:<span>"AT1"</span>,<br><span>"2"</span>:<span>"Macrophages"</span>,<br><span>"3"</span>:<span>"Fibroblast"</span>,<br><span>"4"</span>:<span>"Macrophages"</span>,<br><span>"5"</span>:<span>"CD8+ T cells"</span>,<br><span>"6"</span>:<span>"Macrophages"</span>,<br><span>"7"</span>:<span>"AT2"</span>,<br><span>"8"</span>:<span>"Endothelial"</span>,<br><span>"9"</span>:<span>"Fibroblast"</span>,<br><span>"10"</span>:<span>"Macrophages"</span>,<br><span>"11"</span>:<span>"Plasma cells"</span>,<br><span>"12"</span>:<span>"AT2"</span>,<br><span>"13"</span>:<span>"Fibroblast"</span>,<br><span>"14"</span>:<span>"Fibroblast"</span>,<br><span>"15"</span>:<span>"AT2"</span>,<br><span>"16"</span>:<span>"Airway epithelial"</span>,<br><span>"17"</span>:<span>"Macrophages"</span>,<br><span>"18"</span>:<span>"Airway epithelial"</span>,<br><span>"19"</span>:<span>"AT2"</span>,<br><span>"20"</span>:<span>"Neuronal Cell"</span>,<br><span>"21"</span>:<span>"Airway epithelial"</span>,<br><span>"22"</span>:<span>"B Cell"</span>,<br><span>"23"</span>:<span>"Mast Cell"</span>,<br><span>"24"</span>:<span>"Cycling T/NK"</span>,<br><span>"25"</span>:<span>"Plasma cells"</span>,<br><span>"26"</span>:<span>"DCs"</span>,<br><span>"27"</span>:<span>"Endothelial"</span>,<br><span>"28"</span>:<span>"smooth muscle cells"</span>,<br><span>"29"</span>:<span>"Monocytes"</span>,<br><span>"30"</span>:<span>"Erythroid"</span>,<br><span>"31"</span>:<span>"B Cell"</span>,<br><span>"32"</span>:<span>"Neuronal Cell"</span>,<br><span>"33"</span>:<span>"Macrophages"</span><br>}<br></code></pre><p data-tool="mdnice编辑器">注释后：</p><pre data-tool="mdnice编辑器"><span></span><code>adata.obs[<span>'cell type_sub'</span>] = adata.obs.leiden.map(cell_type_sub)<br>sc.pl.umap(adata, color = [<span>'cell type_sub'</span>], frameon = <span>False</span>, legend_loc = <span>"on data"</span>)<br>plt.tight_layout()<br>plt.savefig(dir+<span>"04-intergration_umap_anno_sub.png"</span>, dpi=<span>300</span>)<br><br><span># 保存</span><br>adata.write_h5ad(dir + <span>'integrated_anno.h5ad'</span>)<br></code></pre><p data-tool="mdnice编辑器">详细注释结果如下：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.762119503945885" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSSvpaJznuhelCGY8RMQVg0yfvOJ2jH64QqRfyibJjUlyNcnKwEDGdpFXq34CAyPtYmibibb2zzjeibOg/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="887" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSSvpaJznuhelCGY8RMQVg0yfvOJ2jH64QqRfyibJjUlyNcnKwEDGdpFXq34CAyPtYmibibb2zzjeibOg/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><h2 data-tool="mdnice编辑器"><span>绘制C、D图</span></h2><p data-tool="mdnice编辑器">首先对细胞进行计数：</p><pre data-tool="mdnice编辑器"><span></span><code><span># adata = sc.read_h5ad('integrated.h5ad')</span><br>adata.obs.Sample.unique().tolist()<br><br><span># 添加一列细胞的样本来源</span><br><span><span>def</span> <span>map_condition</span><span>(x)</span>:</span><br>    <span>if</span> <span>'cov'</span> <span>in</span> x:<br>        <span>return</span> <span>'COVID19'</span><br>    <span>else</span>:<br>        <span>return</span> <span>'Control'</span><br><br>adata.obs[<span>'condition'</span>] = adata.obs.Sample.map(map_condition)<br>adata.obs<br></code></pre><p data-tool="mdnice编辑器">结果图：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.30185185185185187" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSSvpaJznuhelCGY8RMQVg05dKFZW25LiaXcNnZ7StFqwCCvxe5kyANQov4oCB1ibmk1K6tCU67JrHg/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSSvpaJznuhelCGY8RMQVg05dKFZW25LiaXcNnZ7StFqwCCvxe5kyANQov4oCB1ibmk1K6tCU67JrHg/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><pre data-tool="mdnice编辑器"><span></span><code><span># 每个样本中的细胞数统计</span><br>num_tot_cells = adata.obs.groupby([<span>'Sample'</span>]).count()<br>num_tot_cells = dict(zip(num_tot_cells.index, num_tot_cells.doublet))<br>num_tot_cells<br><br><span># {'C51ctr': 10934, 'C52ctr': 3967, 'C53ctr': 6076, 'C54ctr': 3860, 'C55ctr': 5110, 'C56ctr': 3609, 'C57ctr': 4307, 'L01cov': 2737,'L03cov': 3604, 'L04cov': 3056, 'L04covaddon': 4073, 'L05cov': 2435, 'L06cov': 5657, 'L07cov': 4217, 'L08cov': 3473, 'L09cov': 3075, 'L10cov': 2713, 'L11cov': 2531, 'L12cov': 3259, 'L13cov': 4244, 'L15cov': 3516, 'L16cov': 1600, 'L17cov': 3937, 'L18cov': 2392, 'L19cov': 2135, 'L21cov': 2961, 'L22cov': 5786}</span><br><br>cell_type_counts = adata.obs.groupby([<span>'Sample'</span>, <span>'condition'</span>, <span>'cell type'</span>]).count()<br>cell_type_counts = cell_type_counts[cell_type_counts.sum(axis = <span>1</span>) &gt; <span>0</span>].reset_index()<br>cell_type_counts = cell_type_counts[cell_type_counts.columns[<span>0</span>:<span>4</span>]]<br>cell_type_counts<br><br><span>#      Sample condition        cell type  doublet</span><br><span># 0    C51ctr   Control           B Cell       70</span><br><span># 1    C51ctr   Control      Endothelial     1153</span><br><span># 2    C51ctr   Control  Epithelial Cell     3711</span><br><span># 3    C51ctr   Control       Fibroblast     1497</span><br><span># 4    C51ctr   Control        Mast Cell      290</span><br><span># ..      ...       ...              ...      ...</span><br><span># 246  L22cov   COVID19        Mast Cell        7</span><br><span># 247  L22cov   COVID19     Myeloid Cell     1663</span><br><span># 248  L22cov   COVID19    Neuronal Cell       83</span><br><span># 249  L22cov   COVID19           T Cell     1379</span><br><span># 250  L22cov   COVID19              pDC      510</span><br><br><span># [251 rows x 4 columns]</span><br><br><br><span># 计算频率</span><br>cell_type_counts[<span>'total_cells'</span>] = cell_type_counts.Sample.map(num_tot_cells).astype(int)<br>cell_type_counts[<span>'frequency'</span>] = cell_type_counts.doublet / cell_type_counts.total_cells<br>cell_type_counts<br></code></pre><p data-tool="mdnice编辑器">每种细胞类型在COVID19与control两种条件下的频率差异：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.3814814814814815" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSSvpaJznuhelCGY8RMQVg0Q2R8Kjbg5DicuBl6hbiclibSwtbMhWSOmhnNlcTt1Puma1UgINdq3CFrA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSSvpaJznuhelCGY8RMQVg0Q2R8Kjbg5DicuBl6hbiclibSwtbMhWSOmhnNlcTt1Puma1UgINdq3CFrA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">绘图：</p><pre data-tool="mdnice编辑器"><span></span><code>plt.figure(figsize = (<span>10</span>,<span>4</span>))<br>ax = sns.boxplot(data = cell_type_counts, x = <span>'cell type'</span>, y = <span>'frequency'</span>, hue = <span>'condition'</span>)<br>plt.xticks(rotation = <span>35</span>, rotation_mode = <span>'anchor'</span>, ha = <span>'right'</span>)<br>plt.tight_layout()<br>plt.savefig(dir+<span>"05-intergration_Fig1D.png"</span>, dpi=<span>300</span>)<br></code></pre><p data-tool="mdnice编辑器">结果图如下：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.38425925925925924" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSSvpaJznuhelCGY8RMQVg0iaSGOU1iaaWMRPZ23V8GqZUw3HToSibOpHq8xquOXQjMhhTsP88ILjNJQ/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSSvpaJznuhelCGY8RMQVg0iaSGOU1iaaWMRPZ23V8GqZUw3HToSibOpHq8xquOXQjMhhTsP88ILjNJQ/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">C图如下：</p><pre data-tool="mdnice编辑器"><span></span><code>sc.pl.umap(adata, color = [<span>'cell type_sub'</span>,<span>'condition'</span>], frameon = <span>False</span>, legend_loc = <span>"on data"</span>)<br>plt.savefig(dir+<span>"05-intergration_Fig1B_C.png"</span>, dpi=<span>300</span>)<br></code></pre><p data-tool="mdnice编辑器">结果图如下：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.4064814814814815" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSSvpaJznuhelCGY8RMQVg047vniaeJv03YWGaJgzicHFKSThASyQHKnVkrdn7eKMhzPSen77xhGzPg/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSSvpaJznuhelCGY8RMQVg047vniaeJv03YWGaJgzicHFKSThASyQHKnVkrdn7eKMhzPSen77xhGzPg/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">下期见~</p><p data-tool="mdnice编辑器">本次绘制图C：</p><blockquote data-tool="mdnice编辑器"><p>Differential gene expression (log-normalized, scaled; see Methods) of AT1 and AT2 cells from COVID-19 and control lungs. Columns, single cells; rows, expression of top-regulated genes. Left bar, lineage markers for AT1 (purple) and AT2 (pink) cells. Colour-coded top lanes indicate expression strength of signatures (log-normalized; see Methods) and group assignment as indicated on the right. exp., expression</p></blockquote><h2 data-tool="mdnice编辑器"><span>数据读取</span></h2><pre data-tool="mdnice编辑器"><span></span><code><span>import</span> scanpy <span>as</span> sc<br><span>import</span> scvi<br><span>import</span> seaborn <span>as</span> sns<br><span>import</span> numpy <span>as</span> np<br><span>import</span> pandas <span>as</span> pd<br><span>import</span> matplotlib.pyplot <span>as</span> plt<br><br><span># 注意dir改成自己的路径</span><br>dir = <span>'/dir/data/GSE171524/'</span><br>adata = sc.read_h5ad(dir+<span>'integrated_anno.h5ad'</span>)<br>adata<br></code></pre><h2 data-tool="mdnice编辑器"><span>差异分析</span></h2><p data-tool="mdnice编辑器">首先提取注释到的AT1与AT2</p><pre data-tool="mdnice编辑器"><span></span><code>subset = adata[adata.obs[<span>'cell type_sub'</span>].isin([<span>'AT1'</span>, <span>'AT2'</span>])].copy()<br>subset<br></code></pre><p data-tool="mdnice编辑器">总共20741个细胞</p><figure data-tool="mdnice编辑器"><img data-ratio="0.15185185185185185" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjRT4FHZia63uEW5wzBkibCdJJhQJhnjZ473pDvmicOpgvg5U4D94jb1VfcR1icZkL7Qz8RftJpTcZ3COw/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjRT4FHZia63uEW5wzBkibCdJJhQJhnjZ473pDvmicOpgvg5U4D94jb1VfcR1icZkL7Qz8RftJpTcZ3COw/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">差异分析可使用SCVI or diffxpy来做</p><h3 data-tool="mdnice编辑器"><span>diffxpy方法差异分析</span></h3><p data-tool="mdnice编辑器">这里需要安装diffxpy这个包，还比较麻烦。需要安装在scanpy的conda环境中，安装如下：</p><pre data-tool="mdnice编辑器"><span></span><code><span># 先安装依赖 batchglm 需要安装以前的版本：https://pypi.org/project/batchglm/#history 找到0.7.4版本，下载到本地</span><br>tar -zxvf batchglm<span>-0.7</span><span>.4</span>.tar.gz<br>cd batchglm<span>-0.7</span><span>.4</span><br>pip install -e .<br><br><span># 再安装 diffxpy</span><br><span># 下载包到本地：https://github.com/theislab/diffxpy</span><br>unzip diffxpy-master.zip<br>cd diffxpy-master<br>pip install -e .<br></code></pre><p data-tool="mdnice编辑器">这样就安装成功了：</p><pre data-tool="mdnice编辑器"><span></span><code><span># two options: SCVI or diffxpy</span><br><span>import</span> diffxpy.api <span>as</span> de<br><br>subset.X = subset.X.toarray()<br>len(subset.var)<br><span>#20858</span><br><br><span># 低表达过滤</span><br>sc.pp.filter_genes(subset, min_cells=<span>100</span>)<br>len(subset.var)<br><span>#13412</span><br><br><span># 修改一下细胞注释的列名</span><br>subset.obs = subset.obs.rename(columns = {<span>'cell type_sub'</span>:<span>'cell_type_sub'</span>})<br>subset.obs<br></code></pre><p data-tool="mdnice编辑器">差异分析：</p><pre data-tool="mdnice编辑器"><span></span><code><span>#if want to test between covid/non covid</span><br><span># res = de.test.wald(data=subset,</span><br><span>#              formula_loc= '~ 1 + condition',</span><br><span>#              factor_loc_totest='condition'</span><br><span>#                   )</span><br><br>res = de.test.wald(data=subset,<br>             formula_loc= <span>'~ 1 + cell_type_sub'</span>,<br>             factor_loc_totest=<span>'cell_type_sub'</span><br>                  )<br><br>dedf = res.summary().sort_values(<span>'log2fc'</span>, ascending = <span>False</span>).reset_index(drop = <span>True</span>)<br>dedf<br></code></pre><p data-tool="mdnice编辑器">结果如下：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.25925925925925924" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjRT4FHZia63uEW5wzBkibCdJJbU1dyahlaTnkcw4WiblJhAS15HO6s2F0WcpEvZ09VgXRMaKbX3qYtyw/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjRT4FHZia63uEW5wzBkibCdJJbU1dyahlaTnkcw4WiblJhAS15HO6s2F0WcpEvZ09VgXRMaKbX3qYtyw/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">查看谁是实验组，谁是对照组，并调整为 AT1 vs AT2</p><pre data-tool="mdnice编辑器"><span></span><code>subset.obs.cell_type_sub.unique()<br><br><span># ['AT2', 'AT1']</span><br><span># Categories (2, object): ['AT1', 'AT2']</span><br><br>most_up = dedf.iloc[<span>0</span>].gene<br>i = np.where(subset.var_names == most_up)[<span>0</span>][<span>0</span>]<br><br>a = subset[subset.obs.cell_type_sub == <span>'AT1'</span>].X[:, i]<br>b = subset[subset.obs.cell_type_sub == <span>'AT2'</span>].X[:, i]<br>print(<span>f"<span>{most_up}</span> expression:"</span>)<br>print(<span>f"AT1: <span>{a.mean()}</span>"</span>)<br>print(<span>f"AT2: <span>{b.mean()}</span>"</span>)<br><br><span># THBS2 expression:</span><br><span># AT1: 0.0</span><br><span># AT2: 0.04352555051445961</span><br><br><br>dedf[<span>'log2fc'</span>] = dedf[<span>'log2fc'</span>]*<span>-1</span><br>dedf = dedf.sort_values(<span>'log2fc'</span>, ascending = <span>False</span>).reset_index(drop = <span>True</span>)<br>dedf<br></code></pre><p data-tool="mdnice编辑器">结果图：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.25833333333333336" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjRT4FHZia63uEW5wzBkibCdJJK56LDO3DIeicHHc7qicqTJc506eSfvZ2dX87sibJnIsABKLsZIGJCVDeg/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjRT4FHZia63uEW5wzBkibCdJJK56LDO3DIeicHHc7qicqTJc506eSfvZ2dX87sibJnIsABKLsZIGJCVDeg/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">挑选显著差异结果：</p><pre data-tool="mdnice编辑器"><span></span><code>dedf = dedf[(dedf.qval &lt; <span>0.05</span>) &amp; (abs(dedf.log2fc) &gt; <span>.5</span>)]<br>dedf<br></code></pre><p data-tool="mdnice编辑器">结果图如下，有4836个显著差异表达基因：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.2601851851851852" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjRT4FHZia63uEW5wzBkibCdJJ0eFjuicHq12bstP00UJkh5icQwVqHzItKhCJpUp9JX2hFX2ZN8vvWzew/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjRT4FHZia63uEW5wzBkibCdJJ0eFjuicHq12bstP00UJkh5icQwVqHzItKhCJpUp9JX2hFX2ZN8vvWzew/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">卡表达值：</p><pre data-tool="mdnice编辑器"><span></span><code>dedf = dedf[dedf[<span>'mean'</span>] &gt; <span>0.15</span>]<br>dedf<br></code></pre><p data-tool="mdnice编辑器">还有1095个基因：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.2740740740740741" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjRT4FHZia63uEW5wzBkibCdJJ6XcubamBYjicFwyMwMWF2mru0vHbWv5ic5BjiaPgRbkibicHPjOvP0ib5pQQ/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjRT4FHZia63uEW5wzBkibCdJJ6XcubamBYjicFwyMwMWF2mru0vHbWv5ic5BjiaPgRbkibicHPjOvP0ib5pQQ/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">选择top50个基因绘制热图：</p><pre data-tool="mdnice编辑器"><span></span><code><span>#top 25 and bottom 25 from sorted df</span><br>genes_to_show = dedf[<span>-25</span>:].gene.tolist() + dedf[:<span>25</span>].gene.tolist() <br>sc.pl.heatmap(subset, genes_to_show, groupby=<span>'cell_type_sub'</span>, swap_axes=<span>True</span>)<br>plt.savefig(dir+<span>"06-intergration_heatmap.png"</span>)<br></code></pre><p data-tool="mdnice编辑器">结果图：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.8476890756302521" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjRT4FHZia63uEW5wzBkibCdJJCeLVu3qJxzTumUxtJfHsYia8LhEZ1BsZL4vk4RTL7Y9cFqy7h1RannQ/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="952" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjRT4FHZia63uEW5wzBkibCdJJCeLVu3qJxzTumUxtJfHsYia8LhEZ1BsZL4vk4RTL7Y9cFqy7h1RannQ/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><h3 data-tool="mdnice编辑器"><span>scvi方法差异分析</span></h3><pre data-tool="mdnice编辑器"><span></span><code><span># 加载上一次保存的模型</span><br>model  = scvi.model.SCVI.load(dir+ <span>'model.model'</span>, adata)<br>model<br><br><span># 差异分析</span><br>scvi_de = model.differential_expression(<br>    idx1 = [adata.obs[<span>'cell type_sub'</span>] == <span>'AT1'</span>],<br>    idx2 = [adata.obs[<span>'cell type_sub'</span>] == <span>'AT2'</span>]<br>    )<br><br>scvi_de<br><br><span># 显著性结果筛选</span><br>scvi_de = scvi_de[(scvi_de[<span>'is_de_fdr_0.05'</span>]) &amp; (abs(scvi_de.lfc_mean) &gt; <span>.5</span>)]<br>scvi_de = scvi_de.sort_values(<span>'lfc_mean'</span>)<br>scvi_de<br><br><br><span># 表达筛选</span><br>scvi_de = scvi_de[(scvi_de.raw_normalized_mean1 &gt; <span>.5</span>) | (scvi_de.raw_normalized_mean2 &gt; <span>.5</span>)]<br>scvi_de<br></code></pre><p data-tool="mdnice编辑器">最终筛选到952个基因：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.262037037037037" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjRT4FHZia63uEW5wzBkibCdJJicb4o6UFMP3cemlcrt1aQBrV4YgibXwLOXdnHAlpicW0l3CUl7XoFc2iaQ/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjRT4FHZia63uEW5wzBkibCdJJicb4o6UFMP3cemlcrt1aQBrV4YgibXwLOXdnHAlpicW0l3CUl7XoFc2iaQ/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><pre data-tool="mdnice编辑器"><span></span><code><span>#top 25 and bottom 25 from sorted df</span><br>genes_to_show = scvi_de[<span>-25</span>:].index.tolist() + scvi_de[:<span>25</span>].index.tolist() <br>sc.pl.heatmap(subset, genes_to_show, groupby=<span>'cell_type_sub'</span>, swap_axes=<span>True</span>, layer = <span>'scvi_normalized'</span>,log = <span>True</span>)<br>plt.savefig(dir+<span>"06-intergration_heatmap_scvi.png"</span>)<br></code></pre><p data-tool="mdnice编辑器">结果图如下：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.8544819557625145" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjRT4FHZia63uEW5wzBkibCdJJX2FTp71erAeLw4CF0kNL3P56BYhHAdKgkNMZ6Uhtzs5LkUzS5wjdvw/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="859" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjRT4FHZia63uEW5wzBkibCdJJX2FTp71erAeLw4CF0kNL3P56BYhHAdKgkNMZ6Uhtzs5LkUzS5wjdvw/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">结果保存</p><pre data-tool="mdnice编辑器"><span></span><code>dedf.to_csv(dir+<span>'dedf.csv'</span>)<br>scvi_de.to_csv(dir+<span>'scvi_de.csv'</span>)<br></code></pre><p data-tool="mdnice编辑器">下次见~</p><h2 data-tool="mdnice编辑器"><span>数据读取</span></h2><pre data-tool="mdnice编辑器"><span></span><code><span>import</span> scanpy <span>as</span> sc<br><span>import</span> scvi<br><span>import</span> seaborn <span>as</span> sns<br><span>import</span> numpy <span>as</span> np<br><span>import</span> pandas <span>as</span> pd<br><span>import</span> matplotlib.pyplot <span>as</span> plt<br><br><span># 注意dir改成自己的路径</span><br>dir = <span>'/dir/data/GSE171524/'</span><br>adata = sc.read_h5ad(dir+<span>'integrated_anno.h5ad'</span>)<br>adata<br><br><span># 添加一列细胞的样本来源，上次没保存这个</span><br><span><span>def</span> <span>map_condition</span><span>(x)</span>:</span><br>    <span>if</span> <span>'cov'</span> <span>in</span> x:<br>        <span>return</span> <span>'COVID19'</span><br>    <span>else</span>:<br>        <span>return</span> <span>'Control'</span><br><br>adata.obs[<span>'condition'</span>] = adata.obs.Sample.map(map_condition)<br>adata.obs<br><br><span># 提取注释到的AT1与AT2</span><br>subset = adata[adata.obs[<span>'cell type_sub'</span>].isin([<span>'AT1'</span>, <span>'AT2'</span>])].copy()<br>subset<br><br><span># 差异分析结果</span><br>dedf = pd.read_csv(dir + <span>'dedf.csv'</span>, index_col=<span>0</span>)<br>scvi_de = pd.read_csv(dir + <span>'scvi_de.csv'</span>, index_col=<span>0</span>)<br></code></pre><h2 data-tool="mdnice编辑器"><span>富集分析</span></h2><p data-tool="mdnice编辑器">接下来对AT1 vs AT2的差异表达基因进行功能富集分析。</p><p data-tool="mdnice编辑器">python中使用的是gseapy包进行功能富集分析，需要安装在scanpy的conda环境中，并且这个包需要在联网状态下使用：</p><pre data-tool="mdnice编辑器"><span></span><code><span># this method requires internet connection</span><br><span># 安装：conda install -c bioconda gseapy -y</span><br><span>import</span> gseapy <span>as</span> gp <br><br><span># 查看包中都有哪些功能数据库集合</span><br>gp.get_library_name()<br><br>enr = gp.enrichr(gene_list= dedf[dedf.log2fc &gt; <span>0</span>].gene.tolist(),<br>                 gene_sets=[<span>'KEGG_2021_Human'</span>,<span>'GO_Biological_Process_2021'</span>],<br>                 organism=<span>'human'</span>, <span># don't forget to set organism to the one you desired!</span><br>                 outdir=<span>None</span>, <span># don't write to disk,</span><br>                 background = subset.var_names.tolist()<br>                )<br><br>enr.results<br></code></pre><p data-tool="mdnice编辑器">结果如下：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.19825206991720332" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSc2giaAUqAicOhV6eSiajUe8DrCO0pwQNuu9dGdibmianFIvHlee9V29YWRLpPNDZfjokIiayLUQZQg0aw/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="2174" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSc2giaAUqAicOhV6eSiajUe8DrCO0pwQNuu9dGdibmianFIvHlee9V29YWRLpPNDZfjokIiayLUQZQg0aw/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><h2 data-tool="mdnice编辑器"><span>比较：Fig 3-d-e</span></h2><pre data-tool="mdnice编辑器"><span></span><code>subset.obs = subset.obs.rename(columns = {<span>'cell type_sub'</span>:<span>'cell_type_sub'</span>})<br><br>sc.pl.violin(subset[subset.obs.cell_type_sub == <span>'AT2'</span>], <span>'ETV5'</span>, groupby=<span>'condition'</span>)<br>plt.savefig(dir+<span>"07-intergration_condition_ETV5.png"</span>)<br></code></pre><p data-tool="mdnice编辑器">结果图如下：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.5634469696969697" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSc2giaAUqAicOhV6eSiajUe8D6862borkXqs4XPg81t4hzibMXc7sPLxgBAFg6tSPNgGibvfDibksPzDLg/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="1056" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSc2giaAUqAicOhV6eSiajUe8D6862borkXqs4XPg81t4hzibMXc7sPLxgBAFg6tSPNgGibvfDibksPzDLg/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">查看基因ETV5在两组中的表达显著性：使用mannwhitneyu进行检验</p><pre data-tool="mdnice编辑器"><span></span><code><span>from</span> scipy <span>import</span> stats<br>temp = subset[subset.obs.cell_type_sub == <span>'AT2'</span>]<br>i = np.where(temp.var_names == <span>'ETV5'</span>)[<span>0</span>][<span>0</span>]<br><br>a = temp[temp.obs.condition == <span>'COVID19'</span>].X[:,i].todense()<br>b = temp[temp.obs.condition == <span>'Control'</span>].X[:,i].todense()<br><br>stats.mannwhitneyu(a, b)<br></code></pre><p data-tool="mdnice编辑器">结果如下：MannwhitneyuResult(statistic=array([[17353149.]]), pvalue=array([[2.10908943e-171]]))</p><h2 data-tool="mdnice编辑器"><span>对DATP signature打分：Fig 3g</span></h2><p data-tool="mdnice编辑器">首先从文献的补充材料table S4的找到signature集合：our_DATP_sig，总共163个基因</p><figure data-tool="mdnice编辑器"><img data-ratio="0.13532338308457711" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSc2giaAUqAicOhV6eSiajUe8D6rAOhRxQpuswyqT6hw3DV7LPibGXDkJe6fAvTFF4uW7EicxIVGV6Rr9Q/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="2010" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSc2giaAUqAicOhV6eSiajUe8D6rAOhRxQpuswyqT6hw3DV7LPibGXDkJe6fAvTFF4uW7EicxIVGV6Rr9Q/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">另存为：datp_sig.txt，不要表头</p><p data-tool="mdnice编辑器">这个基因集是什么来头呢，文中描述如下：</p><blockquote data-tool="mdnice编辑器"><p>Recent studies have shown that inflammation can induce a cell state that is characterized by failure to fully transition to AT1 cells; this has been termed ‘damage-associated transient progenitors’ (DATPs), ‘alveolar differentiation intermediate’ (ADI), or ‘pre-AT1 transitional cell state’ (PATS)25–27 (hereafter referred to as DATPs)</p></blockquote><p data-tool="mdnice编辑器">现在想看DATPs这个基因集在AT1/AT2中不同分组COVID-19和Control的打分情况，是否支持：incomplete transition of AT2 to AT1 cells in COVID-19 lungs（在肺再生过程中，AT2细胞作为AT1细胞的祖细胞）与 除了病毒感染直接破坏肺泡上皮外，COVID-19患者的肺再生过程也受到损害 。</p><pre data-tool="mdnice编辑器"><span></span><code><span>#gene signature, ie, input list of genes from user</span><br><span>with</span> open(dir + <span>'datp_sig.txt'</span>) <span>as</span> f:<br>    datp_sig = [x.strip() <span>for</span> x <span>in</span> list(f)]<br>    <br>sc.tl.score_genes(subset, datp_sig, score_name = <span>'datp'</span>)<br>subset.obs<br></code></pre><p data-tool="mdnice编辑器">现在可以看到数据中多了一列datp的打分：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.24006810442678775" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSc2giaAUqAicOhV6eSiajUe8DIvM0PTKibicP6qOdfibjdibCGb5vCjJNRMiaMcobF7FyhricbQHYracKgERg/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="1762" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSc2giaAUqAicOhV6eSiajUe8DIvM0PTKibicP6qOdfibjdibCGb5vCjJNRMiaMcobF7FyhricbQHYracKgERg/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">打分可视化：</p><pre data-tool="mdnice编辑器"><span></span><code>sc.pl.violin(subset, <span>'datp'</span>, groupby=<span>'condition'</span>)<br>plt.savefig(dir+<span>"08-intergration_datp_sig.png"</span>)<br></code></pre><p data-tool="mdnice编辑器">结果如下：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.5974973931178311" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSc2giaAUqAicOhV6eSiajUe8D10VbichNmpicjHCqghiaiaRbZIzMlLlJHibO2lgxMDbNyU1WMSykk4U0LLA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="959" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSc2giaAUqAicOhV6eSiajUe8D10VbichNmpicjHCqghiaiaRbZIzMlLlJHibO2lgxMDbNyU1WMSykk4U0LLA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">显著性检验：</p><pre data-tool="mdnice编辑器"><span></span><code>a = subset[subset.obs.condition == <span>'COVID19'</span>].obs.datp.values<br>b = subset[subset.obs.condition == <span>'Control'</span>].obs.datp.values<br>stats.mannwhitneyu(a, b)<br><br><span># 结果</span><br><span># MannwhitneyuResult(statistic=77312171.0, pvalue=0.0)</span><br><br><span># 将打分UMAP可视化</span><br>sc.pl.umap(subset, color = <span>'datp'</span>, vmax = <span>0.5</span>)<br>plt.savefig(dir+<span>"09-intergration_datp_sig_umap.png"</span>)<br></code></pre><p data-tool="mdnice编辑器">结果图如下：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.7019002375296912" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSc2giaAUqAicOhV6eSiajUe8Dq9cWqO3k5ZxdsOMoiaZr9zTGT5BLUpYaBApXF8jnaFiadCd51SkN10gQ/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="842" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjSc2giaAUqAicOhV6eSiajUe8Dq9cWqO3k5ZxdsOMoiaZr9zTGT5BLUpYaBApXF8jnaFiadCd51SkN10gQ/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><p data-tool="mdnice编辑器">本次国庆特刊单细胞python图文复现到此更新完毕，关于结果解读，如为什么选择绘制基因ETV5、CAV1在AT1/AT2不同分组COVID19 vs Control中的表达等等，视频中对文献结果的解读更精彩，可前往观看~</p><p><br></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/PFG0JXHm8BTA1ZBf-r1Lxw",target="_blank" rel="noopener noreferrer">原文链接</a>
