---
title: "SCENIC单细胞转录因子预测|4.精简版流程"
date: 2023-07-04T03:50:46Z
draft: ["false"]
tags: [
  "fetched",
  "Biomamba 生信基地"
]
categories: ["Acdemic"]
---
SCENIC单细胞转录因子预测|4.精简版流程 by Biomamba 生信基地
------
<div><section><mp-common-profile data-pluginname="mpprofile" data-weui-theme="light" data-id="MzAwMzIzOTk5OQ==" data-headimg="http://mmbiz.qpic.cn/mmbiz_png/ImlBFVOwwpzyQpcoEQKDb6ic0Dia8dFR0tfmNm2u0eJfmhJzN3Jjh0jmmTycIeJdu7XMxu6Fvz8jicsKmH2S4MvOQ/0?wx_fmt=png" data-nickname="Biomamba 生信基地" data-alias="" data-signature="本人为在读博士研究生，此公众号旨在分享生信知识及科研经验与体会，欢迎各位同学、老师与专家的批评指正，也欢迎各界人士的合作与交流。" data-from="0" data-is_biz_ban="0"></mp-common-profile></section><section><section data-support="96编辑器" data-style-id="39305"><section data-align="title"><section><section><strong>写在前面</strong></section></section></section></section></section><section data-support="96编辑器" data-style-id="35163"><section><section><section><section>经过几节课的铺垫，SCENIC的运行流程其实非常简单，大部分内容均被SCENIC包高度封装，最核心的内容其实只有四行。本次教程文末快速地和大家一起过一遍流程。当然，输出文件的含义及利用方式还需要通过我们后续的详解视频来解析。学习手册与测试文件不断更新，可在文末解锁或联系客服微信Biomamba领取。</section><section>学习手册：</section><p><img data-galleryid="" data-ratio="0.4148148148148148" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/ImlBFVOwwpyu18ShpajyL1fUdXxZSg5wFP3HiaQpNaQ9NbUA3ialKSzu8hQFRUHSgqENUdpDNSl2WAowteMuqe7A/640?wx_fmt=png" data-type="jpeg" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/ImlBFVOwwpyu18ShpajyL1fUdXxZSg5wFP3HiaQpNaQ9NbUA3ialKSzu8hQFRUHSgqENUdpDNSl2WAowteMuqe7A/640?wx_fmt=png"></p><section><br></section><section>当然更推荐大家去作者的Github学习：</section><section><span>https://github.com/aertslab/SCENIC</span></section><section><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzAwMzIzOTk5OQ==&amp;mid=2247497382&amp;idx=1&amp;sn=95586ea6dc381f8b8941cbd50f914c34&amp;chksm=9b3c8ff6ac4b06e0a44df58181c3c85ea856fbc1b02e819640f7d9963fb598a4c4abb84f000f&amp;scene=21#wechat_redirect" textvalue="2023年订阅须知" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2"></a></section></section></section></section></section><section><br></section><section><section data-support="96编辑器" data-style-id="39305"><section data-align="title"><section><section><strong>视频教程</strong></section></section></section></section></section><section data-support="96编辑器" data-style-id="35163"><section><section><section><p>B站同步播出：</p><p><span>https://www.bilibili.com/video/BV1Mg4y1c7wU?p=4</span></p><section><iframe data-vidtype="2" data-mpvid="wxv_2981834164515930113" data-cover="http%3A%2F%2Fmmbiz.qpic.cn%2Fmmbiz_jpg%2FImlBFVOwwpxudr8bicRHtqrKtVEoWEjfx51Xq97MicnhYhicJ66oLlWkhzFgHKicnmvnAEVZ6qaEABUkUMZe9gTicrQ%2F0%3Fwx_fmt%3Djpeg" allowfullscreen="" frameborder="0" data-ratio="1.7777777777777777" data-w="1920" data-src="https://mp.weixin.qq.com/mp/readtemplate?t=pages/video_player_tmpl&amp;action=mpvideo&amp;auto=0&amp;vid=wxv_2981834164515930113"></iframe></section></section></section></section></section><section><section data-support="96编辑器" data-style-id="39305"><section data-align="title"><section><br></section><section><section data-support="96编辑器" data-style-id="39305"><section data-align="title"><section><section><strong>图文教程</strong></section></section></section></section></section><section data-support="96编辑器" data-style-id="35163"><section><section><section><p><code>SCENIC</code>的分析流程封装的较好，运行完整个流程无非也十几行代码(但是会产生大量的过程及结果文件)。<br>下面这个<strong>示例数据</strong>真的算的很<strong>快</strong>，不过大家跑自己数据时可能会算上<strong>好几天</strong>。</p><h2>3.1. 计算共表达网络</h2><p><br></p><pre><code><span>genesKept <span>&lt;-</span> <span>geneFiltering</span>(exprMat, scenicOptions)</span></code></pre><pre><code>## Maximum value in the expression matrix: 421</code></pre><pre><code>## Ratio of detected vs non-detected: 0.24</code></pre><pre><code>## Number of counts (in the dataset units) per gene:</code></pre><pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. <br>##     0.0    17.0    48.0   146.8   137.0  5868.0</code></pre><pre><code>## Number of cells in which each gene is detected:</code></pre><pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. <br>##    0.00   10.00   25.00   39.01   60.00  180.00</code></pre><pre><code>## <br>## Number of genes left after applying the following filters (sequential):</code></pre><pre><code>##  771 genes with counts per gene &gt; 6</code></pre><pre><code>##  770 genes detected in more than 2 cells</code></pre><pre><code>## Using the column 'features' as feature index for the ranking database.</code></pre><pre><code>##  770 genes available in RcisTarget database</code></pre><pre><code>## Gene list saved in int/1.1_genesKept.Rds</code></pre><p><br></p><pre><code><span>exprMat_filtered <span>&lt;-</span> exprMat[genesKept, ]</span><br><span><span>runCorrelation</span>(exprMat_filtered, scenicOptions)</span><br><span>exprMat_filtered_log <span>&lt;-</span> <span>log2</span>(exprMat_filtered<span>+</span><span>1</span>)</span></code></pre><p><br></p><pre><code><span><span># 基因共表达网络计算</span></span><br><span>mymethod <span>&lt;-</span> <span>'runGenie3'</span> <span># 'grnboost2'</span></span><br><span><span>library</span>(reticulate)</span><br><span><span>if</span>(mymethod<span>==</span><span>'runGenie3'</span>){</span><br><span>  <span>runGenie3</span>(exprMat_filtered_log, scenicOptions)</span><br><span>}<span>else</span>{</span><br><span>  arb.algo <span>=</span> <span>import</span>(<span>'arboreto.algo'</span>)</span><br><span>  tf_names <span>=</span> <span>getDbTfs</span>(scenicOptions)</span><br><span>  tf_names <span>=</span> Seurat<span>::</span><span>CaseMatch</span>(</span><br><span>    <span>search =</span> tf_names,</span><br><span>    <span>match =</span> <span>rownames</span>(exprMat_filtered))</span><br><span>  adj <span>=</span> arb.algo<span>$</span><span>grnboost2</span>(</span><br><span>    <span>as.data.frame</span>(<span>t</span>(<span>as.matrix</span>(exprMat_filtered))),</span><br><span>    <span>tf_names=</span>tf_names, <span>seed=</span>2023L</span><br><span>  )</span><br><span>  <span>colnames</span>(adj) <span>=</span> <span>c</span>(<span>'TF'</span>,<span>'Target'</span>,<span>'weight'</span>)</span><br><span>  <span>saveRDS</span>(adj,<span>file=</span><span>getIntName</span>(scenicOptions,</span><br><span>                              <span>'genie3ll'</span>))</span><br><span>}</span></code></pre><pre><code>## Using 8 TFs as potential regulators...</code></pre><pre><code>## Warning in runGenie3(exprMat_filtered_log, scenicOptions): Only 8 (0%) of the 1721 TFs in the database were found in the dataset. Do they use the same gene IDs?</code></pre><pre><code>## Running GENIE3 part 1</code></pre><pre><code>## Running GENIE3 part 10</code></pre><pre><code>## Running GENIE3 part 2</code></pre><pre><code>## Running GENIE3 part 3</code></pre><pre><code>## Running GENIE3 part 4</code></pre><pre><code>## Running GENIE3 part 5</code></pre><pre><code>## Running GENIE3 part 6</code></pre><pre><code>## Running GENIE3 part 7</code></pre><pre><code>## Running GENIE3 part 8</code></pre><pre><code>## Running GENIE3 part 9</code></pre><pre><code>## Finished running GENIE3.</code></pre><h2>3.2. 构建并计算基因调控网络活性(GRN score)</h2><p><br></p><pre><code><span>exprMat_log <span>&lt;-</span> <span>log2</span>(exprMat<span>+</span><span>1</span>)</span><br><span>scenicOptions<span>@</span>settings<span>$</span>dbs <span>&lt;-</span> scenicOptions<span>@</span>settings<span>$</span>dbs[<span>"10kb"</span>] <span># Toy run settings</span></span><br><span>scenicOptions <span>&lt;-</span> <span>runSCENIC_1_coexNetwork2modules</span>(scenicOptions)</span></code></pre><pre><code>## 02:16    Creating TF modules</code></pre><pre><code>## Number of links between TFs and targets (weight&gt;=0.001): 6139</code></pre><pre><code>##            [,1]<br>## nTFs          8<br>## nTargets    770<br>## nGeneSets    47<br>## nLinks    17299</code></pre><p><br></p><pre><code><span>scenicOptions <span>&lt;-</span> <span>runSCENIC_2_createRegulons</span>(scenicOptions, <span>coexMethod=</span><span>c</span>(<span>"top5perTarget"</span>)) <span># Toy run settings</span></span></code></pre><pre><code>## 02:17    Step 2. Identifying regulons</code></pre><pre><code>## <br>## Attaching package: 'AUCell'</code></pre><pre><code>## The following object is masked from 'package:SCENIC':<br>## <br>##     plotEmb_rgb</code></pre><pre><code>## Using the column 'features' as feature index for the ranking database.</code></pre><pre><code>## tfModulesSummary:</code></pre><pre><code>##               [,1]<br>## top5perTarget    8</code></pre><pre><code>## 02:17    RcisTarget: Calculating AUC</code></pre><pre><code>## Using the column 'features' as feature index for the ranking database.</code></pre><pre><code>## Scoring database:  [Source file: mm9-tss-centered-10kb-7species.mc9nr.feather]</code></pre><pre><code>## 02:17    RcisTarget: Adding motif annotation</code></pre><pre><code>## Using BiocParallel...</code></pre><pre><code>## Number of motifs in the initial enrichment: 867</code></pre><pre><code>## Number of motifs annotated to the matching TF: 126</code></pre><pre><code>## 02:17    RcisTarget: Pruning targets</code></pre><pre><code>## Using the column 'features' as feature index for the ranking database.</code></pre><pre><code>## [1] 22058</code></pre><pre><code>## 02:19    Number of motifs that support the regulons: 126</code></pre><pre><code>##  Preview of motif enrichment saved as: output/Step2_MotifEnrichment_preview.html</code></pre><p><br></p><pre><code><span>scenicOptions <span>&lt;-</span> <span>runSCENIC_3_scoreCells</span>(scenicOptions, exprMat_log)</span></code></pre><pre><code>## 02:19    Step 3. Analyzing the network activity in each individual cell</code></pre><pre><code>##  Number of regulons to evaluate on cells: 14<br>## Biggest (non-extended) regulons: <br>##   Tef (405g)<br>##   Sox9 (150g)<br>##   Irf1 (104g)<br>##   Sox8 (97g)<br>##   Sox10 (88g)<br>##   Dlx5 (35g)<br>##   Stat6 (27g)</code></pre><pre><code>## Quantiles for the number of genes detected by cell: <br>## (Non-detected genes are shuffled at the end of the ranking. Keep it in mind when choosing the threshold for calculating the AUC).</code></pre><pre><code>##    min     1%     5%    10%    50%   100% <br>##  46.00  58.99  77.00  84.80 154.00 342.00</code></pre><pre><code>## Warning in .AUCell_calcAUC(geneSets = geneSets, rankings = rankings, nCores =<br>## nCores, : Using only the first 58.99 genes (aucMaxRank) to calculate the AUC.</code></pre><pre><code>## Using 8 cores with doMC.</code></pre><pre><code>## 02:19    Finished running AUCell.</code></pre><pre><code>## 02:19    Plotting heatmap...</code></pre><pre><code>## 02:19    Plotting t-SNEs...</code></pre><pre><code>## Warning in file.rename(plotsName, file.path(plotsLoc, plotsName)):<br>## cannot rename file 'Step3_RegulonActivity_tSNE_colByActivity' to<br>## 'output/Step3_RegulonActivity_tSNE_colByActivity', reason 'Directory not empty'</code></pre><p><br></p><pre><code><span><span># 通过交互的shiny app选择判定二元矩阵的阈值，这里Rmarkdown不便展示，大家可以自行尝试一下</span></span><br><span><span># aucellApp &lt;- plotTsne_AUCellApp(scenicOptions, exprMat_log)</span></span><br><span><span># savedSelections &lt;- shiny::runApp(aucellApp)</span></span><br><span><span># newThresholds &lt;- savedSelections$thresholds</span></span><br><span><span># scenicOptions@fileNames$int["aucell_thresholds",1] &lt;- "int/newThresholds.Rds"</span></span><br><span><span># saveRDS(newThresholds, file=getIntName(scenicOptions, "aucell_thresholds"))</span></span><br><span></span><br><span>scenicOptions <span>&lt;-</span> <span>runSCENIC_4_aucell_binarize</span>(scenicOptions)<span>#将AUCell矩阵二元化</span></span></code></pre><pre><code>## Binary regulon activity: 6 TF regulons x 197 cells.<br>## (11 regulons including 'extended' versions)<br>## 6 regulons are active in more than 1% (1.97) cells.</code></pre><pre><code>## Warning in AUCell_plotTSNE(tSNE = tSNE, exprMat = exprMat, cellsAUC =<br>## regulonAUC, : Expression plot was requested, but no expression matrix provided.</code></pre><pre><code>## Warning in file.rename(plotsName, file.path(plotsLoc, plotsName)):<br>## cannot rename file 'Step4_BinaryRegulonActivity_tSNE_colByActivity' to<br>## 'output/Step4_BinaryRegulonActivity_tSNE_colByActivity', reason 'Directory not<br>## empty'</code></pre><h2>3.3. 一些下游的可视化与探索</h2><p><br></p><pre><code><span><span>tsneAUC</span>(scenicOptions, <span>aucType=</span><span>"AUC"</span>) <span>#利用AUCell score运行tsne</span></span></code></pre><pre><code>## [1] "int/tSNE_AUC_50pcs_30perpl.Rds"</code></pre><p><br></p><pre><code><span><span># Export:</span></span><br><span><span># saveRDS(cellInfo, file=getDatasetInfo(scenicOptions, "cellInfo")) # Temporary, to add to loom</span></span><br><span><span>if</span>(<span>!</span><span>file.exists</span>(<span>'output/scenic.loom'</span>)){<span>export2loom</span>(scenicOptions, exprMat)}</span><br><span></span><br><span><span>saveRDS</span>(scenicOptions, <span>file=</span><span>"int/scenicOptions.Rds"</span>)<span>#保存结果 </span></span><br><span></span><br><span><span>### Exploring output </span></span><br><span><span># output下存放的scenic.loom文件可以用这个网站做交互式的可视化: http://scope.aertslab.org</span></span><br><span></span><br><span><span># motif富集的相关信息在：</span></span><br><span><span>#output/Step2_MotifEnrichment_preview.html in</span></span><br><span>motifEnrichment_selfMotifs_wGenes <span>&lt;-</span> <span>loadInt</span>(scenicOptions, <span>"motifEnrichment_selfMotifs_wGenes"</span>)</span><br><span>tableSubset <span>&lt;-</span> motifEnrichment_selfMotifs_wGenes[highlightedTFs<span>==</span><span>"Sox8"</span>]<span>#这样查看Sox8的信息</span></span><br><span><span>viewMotifs</span>(tableSubset)</span></code></pre><p><code><span><br></span></code></p><p><img data-galleryid="" data-ratio="0.6312785388127854" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/ImlBFVOwwpxudr8bicRHtqrKtVEoWEjfxwDTjsibcic8O9PnuDNIXiczIw35TLcHFTrXkmT303ocqDI4liabiaIMicQ0g/640?wx_fmt=png" data-type="png" data-w="876" src="https://mmbiz.qpic.cn/mmbiz_png/ImlBFVOwwpxudr8bicRHtqrKtVEoWEjfxwDTjsibcic8O9PnuDNIXiczIw35TLcHFTrXkmT303ocqDI4liabiaIMicQ0g/640?wx_fmt=png"></p><p><span><br></span></p><pre><code><span><span>#查看motif和对应基因：</span></span><br><span><span>#output/Step2_regulonTargetsInfo.tsv</span></span><br><span>regulonTargetsInfo <span>&lt;-</span> <span>loadInt</span>(scenicOptions, <span>"regulonTargetsInfo"</span>)</span><br><span>tableSubset <span>&lt;-</span> regulonTargetsInfo[TF<span>==</span><span>"Stat6"</span> <span>&amp;</span> highConfAnnot<span>==</span><span>TRUE</span>]</span><br><span><span>viewMotifs</span>(tableSubset)</span></code></pre><p><br></p><p><img data-galleryid="" data-ratio="0.6659505907626209" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/ImlBFVOwwpxudr8bicRHtqrKtVEoWEjfxCtz4KGticSqwM6LcHSPPsicqw1mcbrhbicVQ8dUS1NMuvaX3TvxjT5RNQ/640?wx_fmt=png" data-type="png" data-w="931" src="https://mmbiz.qpic.cn/mmbiz_png/ImlBFVOwwpxudr8bicRHtqrKtVEoWEjfxCtz4KGticSqwM6LcHSPPsicqw1mcbrhbicVQ8dUS1NMuvaX3TvxjT5RNQ/640?wx_fmt=png"></p><pre><code><span><span>#计算并展示每种细胞类型特异性的(celltype Cell-type specific regulators (RSS))，类似于拿RGN的活性来计算调控</span></span><br><span>regulonAUC <span>&lt;-</span> <span>loadInt</span>(scenicOptions, <span>"aucell_regulonAUC"</span>)</span><br><span>rss <span>&lt;-</span> <span>calcRSS</span>(<span>AUC=</span><span>getAUC</span>(regulonAUC), <span>cellAnnotation=</span>cellInfo[<span>colnames</span>(regulonAUC), <span>"CellType"</span>], )</span><br><span>rssPlot <span>&lt;-</span> <span>plotRSS</span>(rss)</span></code></pre><pre><code>## Loading required package: grid</code></pre><pre><code>## ========================================<br>## ComplexHeatmap version 2.12.1<br>## Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/<br>## Github page: https://github.com/jokergoo/ComplexHeatmap<br>## Documentation: http://jokergoo.github.io/ComplexHeatmap-reference<br>## <br>## If you use it in published research, please cite either one:<br>## - Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional <br>##     genomic data. Bioinformatics 2016.<br>## - Gu, Z. Complex Heatmap Visualization. iMeta 2022.<br>## <br>## <br>## The new InteractiveComplexHeatmap package can directly export static <br>## complex heatmaps into an interactive Shiny app with zero effort. Have a try!<br>## <br>## This message can be suppressed by:<br>##   suppressPackageStartupMessages(library(ComplexHeatmap))<br>## ========================================</code></pre><pre><code>## Showing regulons and cell types with any RSS &gt; 0.01 (dim: 14x5)</code></pre><pre><code>## Loading required package: ggplot2</code></pre><p><br></p><pre><code><span>plotly<span>::</span><span>ggplotly</span>(rssPlot<span>$</span>plot)</span></code></pre><p><br></p><p><img data-galleryid="" data-ratio="0.6772673733804476" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/ImlBFVOwwpxudr8bicRHtqrKtVEoWEjfxbJW0VvXa22t0MsGeCWzicmP0SC4sbk3FUORrwcphib3Vic1iaVDQDaXW4w/640?wx_fmt=png" data-type="png" data-w="849" src="https://mmbiz.qpic.cn/mmbiz_png/ImlBFVOwwpxudr8bicRHtqrKtVEoWEjfxbJW0VvXa22t0MsGeCWzicmP0SC4sbk3FUORrwcphib3Vic1iaVDQDaXW4w/640?wx_fmt=png"></p><p><span></span></p><h2>3.4 输出文件</h2><p>除了我们上面可视化展示出的结果，<code>SCENIC</code>运行时会自动在本地输出大量文件：<br></p><p>├── scenic.loom<br>├── Step2_MotifEnrichment_preview.html<br>├── Step2_MotifEnrichment.tsv<br>├── Step2_regulonTargetsInfo.tsv<br>├── Step3_RegulonActivity_heatmap<br>├── Step3_RegulonActivity_heatmap.pdf<br>├── Step3_RegulonActivity_tSNE_colByActivity<br>│   ├── Dlx5 (35g)_AUC.png<br>│   ├── Dlx5 (35g)_binaryAUC.png<br>│   ├── Dlx5 (35g)_expression.png<br>│   ├── Dlx5 (35g)_histogram.png<br>│   ├── Dlx5_extended (214g)_AUC.png<br>│   ├── Dlx5_extended (214g)_binaryAUC.png<br>│   ├── Dlx5_extended (214g)_expression.png<br>│   ├── Dlx5_extended (214g)_histogram.png<br>│   ├── Irf1 (104g)_AUC.png<br>│   ├── Irf1 (104g)_binaryAUC.png<br>│   ├── Irf1 (104g)_expression.png<br>│   ├── Irf1 (104g)_histogram.png<br>│   ├── Irf1_extended (108g)_AUC.png<br>│   ├── Irf1_extended (108g)_binaryAUC.png<br>│   ├── Irf1_extended (108g)_expression.png<br>│   ├── Irf1_extended (108g)_histogram.png<br>│   ├── Sox10 (88g)_AUC.png<br>│   ├── Sox10 (88g)_binaryAUC.png<br>│   ├── Sox10 (88g)_expression.png<br>│   ├── Sox10 (88g)_histogram.png<br>│   ├── Sox10_extended (114g)_AUC.png<br>│   ├── Sox10_extended (114g)_binaryAUC.png<br>│   ├── Sox10_extended (114g)_expression.png<br>│   ├── Sox10_extended (114g)_histogram.png<br>│   ├── Sox8 (97g)_AUC.png<br>│   ├── Sox8 (97g)_binaryAUC.png<br>│   ├── Sox8 (97g)_expression.png<br>│   ├── Sox8 (97g)_histogram.png<br>│   ├── Sox8_extended (136g)_AUC.png<br>│   ├── Sox8_extended (136g)_binaryAUC.png<br>│   ├── Sox8_extended (136g)_expression.png<br>│   ├── Sox8_extended (136g)_histogram.png<br>│   ├── Sox9 (150g)_AUC.png<br>│   ├── Sox9 (150g)_binaryAUC.png<br>│   ├── Sox9 (150g)_expression.png<br>│   ├── Sox9 (150g)_histogram.png<br>│   ├── Sox9_extended (183g)_AUC.png<br>│   ├── Sox9_extended (183g)_binaryAUC.png<br>│   ├── Sox9_extended (183g)_expression.png<br>│   ├── Sox9_extended (183g)_histogram.png<br>│   ├── Stat6 (27g)_AUC.png<br>│   ├── Stat6 (27g)_binaryAUC.png<br>│   ├── Stat6 (27g)_expression.png<br>│   ├── Stat6 (27g)_histogram.png<br>│   ├── Stat6_extended (75g)_AUC.png<br>│   ├── Stat6_extended (75g)_binaryAUC.png<br>│   ├── Stat6_extended (75g)_expression.png<br>│   ├── Stat6_extended (75g)_histogram.png<br>│   ├── Tef (405g)_AUC.png<br>│   ├── Tef (405g)_binaryAUC.png<br>│   ├── Tef (405g)_expression.png<br>│   ├── Tef (405g)_histogram.png<br>│   ├── Tef_extended (447g)_AUC.png<br>│   ├── Tef_extended (447g)_binaryAUC.png<br>│   ├── Tef_extended (447g)_expression.png<br>│   └── Tef_extended (447g)_histogram.png<br>├── Step3_RegulonActivity_tSNE_colByActivity.html<br>├── Step3_RegulonActivity_tSNE_colByCellProps.pdf<br>├── Step4_BinaryRegulonActivity_Heatmap_all.pdf<br>├── Step4_BinaryRegulonActivity_Heatmap_corr.pdf<br>├── Step4_BinaryRegulonActivity_Heatmap_onePercent.pdf<br>├── Step4_BinaryRegulonActivity_tSNE_colByActivity<br>│   ├── Dlx5 (35g)_AUC.png<br>│   ├── Dlx5 (35g)_binaryAUC.png<br>│   ├── Dlx5 (35g)_histogram.png<br>│   ├── Dlx5_extended (214g)_AUC.png<br>│   ├── Dlx5_extended (214g)_binaryAUC.png<br>│   ├── Dlx5_extended (214g)_histogram.png<br>│   ├── Irf1 (104g)_AUC.png<br>│   ├── Irf1 (104g)_binaryAUC.png<br>│   ├── Irf1 (104g)_histogram.png<br>│   ├── Irf1_extended (108g)_AUC.png<br>│   ├── Irf1_extended (108g)_binaryAUC.png<br>│   ├── Irf1_extended (108g)_histogram.png<br>│   ├── Sox10 (88g)_AUC.png<br>│   ├── Sox10 (88g)_binaryAUC.png<br>│   ├── Sox10 (88g)_histogram.png<br>│   ├── Sox10_extended (114g)_AUC.png<br>│   ├── Sox10_extended (114g)_binaryAUC.png<br>│   ├── Sox10_extended (114g)_histogram.png<br>│   ├── Sox8 (97g)_AUC.png<br>│   ├── Sox8 (97g)_binaryAUC.png<br>│   ├── Sox8 (97g)_histogram.png<br>│   ├── Sox8_extended (136g)_AUC.png<br>│   ├── Sox8_extended (136g)_binaryAUC.png<br>│   ├── Sox8_extended (136g)_histogram.png<br>│   ├── Sox9 (150g)_AUC.png<br>│   ├── Sox9 (150g)_binaryAUC.png<br>│   ├── Sox9 (150g)_histogram.png<br>│   ├── Sox9_extended (183g)_AUC.png<br>│   ├── Sox9_extended (183g)_binaryAUC.png<br>│   ├── Sox9_extended (183g)_histogram.png<br>│   ├── Stat6 (27g)_AUC.png<br>│   ├── Stat6 (27g)_binaryAUC.png<br>│   ├── Stat6 (27g)_histogram.png<br>│   ├── Stat6_extended (75g)_AUC.png<br>│   ├── Stat6_extended (75g)_binaryAUC.png<br>│   ├── Stat6_extended (75g)_histogram.png<br>│   ├── Tef (405g)_AUC.png<br>│   ├── Tef (405g)_binaryAUC.png<br>│   ├── Tef (405g)_histogram.png<br>│   ├── Tef_extended (447g)_AUC.png<br>│   ├── Tef_extended (447g)_binaryAUC.png<br>│   └── Tef_extended (447g)_histogram.png<br>├── Step4_BinaryRegulonActivity_tSNE_colByActivity.html<br>├── Step4_BinaryRegulonActivity_tSNE_colByAUC.pdf<br>├── Step4_BinaryRegulonActivity_tSNE_colByCellProps.pdf<br>└── Step4_BoxplotActiveCellsRegulon.pdf</p><p>他们收录的信息为:<br></p><h3>3.4.1 我人肉总结归纳的结果</h3><ul><li><p>“int/1.4_GENIE3_linkList.Rds”<br>基因共表达网络预测结果，记录了TF与TG间的weight<br></p></li><li><p>“int/2.1_tfModules_forMotifEnrichmet.Rds” list文件，保存的是经Motif富集检验得到的TF模块<br></p></li><li><p>“int/2.3_motifEnrichment.Rds”<br>经AUCell评分筛选后的motif富集分析结果<br></p></li><li><p>“int/2.4_motifEnrichment_selfMotifs_wGenes.Rds”<br>存放构建regulon包含的target gene信息<br></p></li><li><p>“output/Step2_MotifEnrichment.tsv”<br>2.4_motifEnrichment_selfMotifs_wGenes.Rds的txt版<br></p></li><li><p>“output/Step2_MotifEnrichment_preview.html”<br>regulon的详细信息，html文件可存储motif的logo等信息<br></p></li><li><p>“int/2.5_regulonTargetsInfo.Rds”<br>最终的regulon详细信息<br></p></li><li><p>“output/Step2_regulonTargetsInfo.tsv”<br>int/2.5_regulonTargetsInfo.Rd的txt版<br></p></li><li><p>“int/2.6_regulons_asGeneSet.Rds”<br># regulon信息，包含高/低置信度的regulon的标注<br></p></li><li><p>“int/3.1_regulons_forAUCell.Rds”<br>基因数大于10的regulon，regulon名称后标注了包含基因数量<br></p></li><li><p>“int/3.3_aucellRankings.Rds”<br>AUCell ranking的结果<br></p></li><li><p>“int/3.2_aucellGenesStats.pdf”<br>AUCell ranking的结果可视化图片<br></p></li><li><p>“int/3.4_regulonAUC.Rds”<br>AUCell score计算结果<br></p></li><li><p>“int/3.5_AUCellThresholds.Rds”<br>存放在AUCell判断regulon on/off时的阈值<br></p></li><li><p>“int/3.5_AUCellThresholds_Info.tsv”<br>AUCell 的阈值、细胞、基因数等信息记录<br></p></li><li><p>“output/Step3_RegulonActivity_heatmap”<br>AUCell score 热图<br></p></li><li><p>“output/Step3_RegulonActivity_tSNE_colByCellProps.pdf”<br>利用regulon score的TSNE可视化结果<br></p></li><li><p>“int/4.1_binaryRegulonActivity.Rds”<br>regulon的AUCell二元矩阵<br></p></li><li><p>“output/Step4_BinaryRegulonActivity_Heatmap_notCorr”<br>regulon二元矩阵的相关性热图<br></p></li><li><p>“output/Step4_BinaryRegulonActivity_tSNE_colByCellProps.pdf”<br>二元矩阵TSNE降维可视化结果<br></p></li></ul><h3>3.4.2 ChatGPT的回答</h3><ul><li><p>scenic.loom<br>SCENIC输出结果的主要文件，包含整个结果的基因表达数据、调控网络和调控子模块等信息。</p></li><li><p>Step2_MotifEnrichment_preview.html<br>展示了motif enrichment可视化结果的html文件。</p></li><li><p>Step2_MotifEnrichment.tsv<br>motif enrichment表格文件，包含每个转录因子集合的motif分数。</p></li><li><p>Step2_regulonTargetsInfo.tsv<br>每个regulon涵盖的基因、Gene Ontology、Pathway与OmniPath分析的信息。</p></li><li><p>Step3_RegulonActivity_heatmap<br>各个Regulon在每个单元中的活动水平heatmap文件。</p></li><p><mp-pay-preview-filter data-offset="145"></mp-pay-preview-filter></p></ul></section></section></section></section></section></section></section></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/k32-ZVZr3QhHMtO-TWXGIA",target="_blank" rel="noopener noreferrer">原文链接</a>
