---
title: "TCGA的28篇教程-CNV全攻略"
date: 2023-07-15T23:58:06Z
draft: ["false"]
tags: [
  "fetched",
  "生信技能树"
]
categories: ["Acdemic"]
---
TCGA的28篇教程-CNV全攻略 by 生信技能树
------
<div><section data-tools="新媒体管家" data-label="powered by xmt.cn"></section><section data-style-type="1" data-tools="新媒体排版" data-id="8363"><blockquote><p>长期更新列表：</p><section data-style="color:inherit;font-size:14px;"><p cid="n276" mdtype="paragraph"><span><a spellcheck="false" href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247486492&amp;idx=1&amp;sn=3a7251244377fdd4b2a3aa5c8cd1131a&amp;chksm=9b484ca7ac3fc5b1a21202cf25ff15a8eec434424aa3e48787129fa6f5e66ebe57ffcb631772&amp;scene=21#wechat_redirect">使用R语言的cgdsr包获取TCGA数据</a><span>（cBioPortal）</span></span><span cid="n277" mdtype="line"><a spellcheck="false" href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247486585&amp;idx=1&amp;sn=3035f6420904aad2c8161b362cdeb472&amp;chksm=9b484cc2ac3fc5d479fc5bce3d68d4666b763652a21a55b281aad8c0c4df9b56b4d3b353cc4c&amp;scene=21#wechat_redirect">TCGA的28篇教程- 使用R语言的RTCGA包获取TCGA数据</a><span> （离线打包版本）</span></span><span cid="n277" mdtype="line"><a spellcheck="false" href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247486728&amp;idx=1&amp;sn=3990dff5efccedc060443b7f3af3b6ee&amp;chksm=9b484db3ac3fc4a51ee34ba578280d89ea6159a48d9dec1c7ecd5dbe208c53a1b36c439de75d&amp;scene=21#wechat_redirect">TCGA的28篇教程- 使用R语言的RTCGAToolbox包获取TCGA数据</a><span> （FireBrowse portal）</span></span><span cid="n277" mdtype="line"><a spellcheck="false" href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247486746&amp;idx=1&amp;sn=b7c5ad7eff8cffb3620756f5feaff587&amp;chksm=9b484da1ac3fc4b741a6e3b59ba1bf668a11e21eb610f1a1d4582e33d429c67c14e6659c0771&amp;scene=21#wechat_redirect">TCGA的28篇教程-  批量下载TCGA所有数据</a><span> （ UCSC的 XENA）</span></span><span cid="n277" mdtype="line"><span><a href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247486789&amp;idx=1&amp;sn=bf65ccc84b3d920b284118d146006f54&amp;chksm=9b484dfeac3fc4e86174a536b33edeb60fa064fa10d9528acdc2e2b7c0ee72c4bdedeb0b7180&amp;scene=21#wechat_redirect" target="_blank">TCGA的28篇教程- 数据下载就到此为止吧</a></span></span><span cid="n277" mdtype="line"><span><a href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247486845&amp;idx=1&amp;sn=b735a4690d9f4efdd601bc8ddd9c4362&amp;chksm=9b484dc6ac3fc4d025c20665213efa64d4b83ff3d923633c8f76d7aedf1a21d4b5feb4086e8c&amp;scene=21#wechat_redirect" target="_blank">TCGA的28篇教程- 指定癌症查看感兴趣基因的表达量</a></span><span><a href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247486915&amp;idx=1&amp;sn=19a6e45f82712e7a94d409e22de7a23a&amp;chksm=9b484d78ac3fc46e92f63cc0c9918e9988c72bb9726152002a4d00b58fc9ea3751fa1732b946&amp;scene=21#wechat_redirect" target="_blank">TCGA的28篇教程- 对TCGA数据库的任意癌症中任意基因做生存分析</a></span><span><a href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247487143&amp;idx=1&amp;sn=ac54a5a679af47f1ad5516631830d66d&amp;chksm=9b484e1cac3fc70aead6b03efb5a393910bea818f9b36d8802307253b71b7a0657993fbaa447&amp;scene=21#wechat_redirect" target="_blank">TCGA的28篇教程-整理GDC下载的xml格式的临床资料</a><br></span><span><a href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247487152&amp;idx=1&amp;sn=244a799c035fd081e01964f4c2d39bed&amp;chksm=9b484e0bac3fc71de3de79b4863ad5a118a1f735deac86e1b6b39627387016ccd2a6b07eef7e&amp;scene=21#wechat_redirect" target="_blank">TCGA的28篇教程-风险因子关联图-一个价值1000但是迟到的答案</a><br></span><span><a href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247487183&amp;idx=1&amp;sn=8f5bde28f608e267e7195a83797da300&amp;chksm=9b484e74ac3fc762ff2e209d54d27ceeacfa7a186f2433f30eceeb04966045ae14e3cb3bf8bb&amp;scene=21#wechat_redirect" target="_blank">TCGA的28篇教程-数据挖掘三板斧之ceRNA</a><br></span><span><a href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247487369&amp;idx=1&amp;sn=fed7321eb1dda047ed7ee9a8e01b8595&amp;chksm=9b484f32ac3fc624dcf67f8e56d611e797cc2c477105093ae8593b003c8ba39eb3fff72c1b54&amp;scene=21#wechat_redirect" target="_blank">TCGA的28篇教程-所有癌症的突变全景图</a><br></span><span><a href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247487415&amp;idx=1&amp;sn=715eacae5c4e4373660eddb25b976dca&amp;chksm=9b484f0cac3fc61a3df681d6fde596cf8b5f6c44e6f4f554c3888edcc691d04257a3f88e5108&amp;scene=21#wechat_redirect" target="_blank">TCGA的28篇教程-早期泛癌研究</a><br></span></span></p></section></blockquote></section><blockquote><p>拷贝数变异情况可以由SNP6.0等比较基因组杂交芯片得到，也可以由WES测序得到，当然了，WGS测序会更好，不过很大概率上选择什么样的实用技术，往往受限于资金和设备。TCGA计划也不外乎此。</p></blockquote><h3><span>拷贝数变异的定义</span></h3><p>CNV（copy-numbervariant）是指拷贝数目变异，也称拷贝数目多态性（copy-number polymorphism，CNP），是一个大小介于1kb至3MB的DNA片段的变异，在人类及动植物基因组中广泛分布，其覆盖的核苷酸总数大大超过单核苷酸多态性（SNP）的总数，极大地丰富了基因组遗传变异的多样性。按照CNV是否致病可分为致病性CNV、非致病性CNV和不明临床意义CNV。</p><p>而在TCGA数据库里面我们通常关心的是somatic CNV，也就是那些肿瘤病人的拷贝数变异然后要剔除正常对照里面的CNV多态性信息，只有这些somatic CNV才更可能是跟肿瘤相关的。</p><h3><span>拷贝数的SNP6.0等芯片测量</span></h3><blockquote><p>TCGA里面主要是通过Affymetrix SNP6.0 array这款芯片来测拷贝数变异！</p></blockquote><p>值得注意的是，并不是只有TCGA利用了SNP6.这个芯片数据，著名的CCLE计划也对一千多细胞系处理了SNP6.0芯片，数据也是可以下载的。</p><p>对SNP6.0的拷贝数芯片来说，通常是用PICNIC等软件处理原始数据，就可以得到的<code>segment记录文件</code>，每个样本一个结果，下面是示例结果：</p><pre><code><span>Chromosome</span>    <span>Start</span>   <span>End</span> <span>Num_Probes</span>  <span>Segment_Mean</span><br>1    61735   1510801 226 <span>-0</span><span>.0397</span><br>1    1627918 1672603 17  <span>-0</span><span>.92</span><br>1    1687587 16153497    8176    0<span>.0077</span><br>1    16153536    16153925    5   <span>-2</span><span>.7441</span><br>1    16154201    16155010    4   <span>-0</span><span>.8711</span><br>1    16165661    72768498    34630   0<span>.0048</span><br>1    72768916    72811148    46  <span>-1</span><span>.7394</span><br>1    72811904    95674710    14901   0<span>.0026</span><br>1    95676511    95676518    2   <span>-1</span><span>.6636</span><br></code></pre><p>表明了某条染色体的某个区域内，SNP6.0芯片设计了多少个探针，芯片结果的拷贝数值是多少(这个区域的拷贝数用Segment_Mean)。</p><p>通常二倍体的Segment_Mean值为0，可以用-0.2和0.2来作为该区域是否缺失或者扩增，也有人选择0.4作为阈值。</p><p>具体数据处理流程见NIH的TCGA官网：<br>https://docs.gdc.cancer.gov/Data/Bioinformatics_Pipelines/CNV_Pipeline/</p><p>参考文献：http://mcr.aacrjournals.org/content/12/4/485.long</p><h3><span>拷贝数变异的NGS测量</span></h3><blockquote><p>TCGA计划的几乎每个样本都是有WES测试数据的，而且少量病人的肿瘤组织及正常对照组织均进行了WES测序，理论上是多余SNP6.0芯片的。</p></blockquote><p>首先从测序仪中得到FASTQ文件即原始数据，通过变异查找流程来得到BAM和VCF文件。比较流行的是GATK的best practice，识别的small Variants（SNV，indel）会以VCF文件格式输出，bam格式的比对文件可以放在IGV进行可视化查看细节。而CNV caller虽然目前也成为标准流程，但是还没有一家独大的软件或者流程，我通常使用4个流程组合起来，这样检测到的CNV事件，涉及从单个靶向基因水平到整个染色体水平。比如外显子水平的deletions, duplications,一直到整个的染色体水平的拷贝数变化。值得一提的是NGS测序数据找拷贝数变化信息，除了需要利用BAM文件的coverage数据，还需要VCF文件中的VAF信息。对WES或者panel的数据，还建议运行 LOH（杂合性缺失）算法。我在生信技能树这里分享过教程的就有如下：</p><ul><li><p><a href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247485243&amp;idx=1&amp;sn=c6e5ca8bf2fea7c137dd4a99b5e1e57e&amp;chksm=9b484780ac3fce9681b3aed124de4133dd075367200e908dbab522c5781232152f2ce18023a5&amp;scene=21#wechat_redirect">WES的CNV探究-conifer软件使用</a></p></li><li><p><a href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247485246&amp;idx=1&amp;sn=c512c7d943f8653776345f379ba767a4&amp;chksm=9b484785ac3fce93768da4b171ea367a70e435eede7254024e531200156f2f3e6e7baa3ceb95&amp;scene=21#wechat_redirect">单个样本NGS数据如何做拷贝数变异分析呢</a></p></li><li><p><a href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247485270&amp;idx=1&amp;sn=f8fd1abea515d69c2ca61165f09822ad&amp;chksm=9b4847edac3fcefbd241e2520330989da07a48da5eed03df5fa38316ce5062b50a116ac033e8&amp;scene=21#wechat_redirect">肿瘤配对样本用varscan 做cnv分析</a></p></li><li><p><a href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247485404&amp;idx=1&amp;sn=29a3f279762a7a6a43888161480f39ea&amp;chksm=9b484767ac3fce71b0a2eb25d849d5b75d021cd605945c0de5ee9d87ff26b526558b1ecd4f53&amp;scene=21#wechat_redirect">使用cnvkit来对大批量wes样本找cnv</a></p></li><li><p><a href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247486409&amp;idx=1&amp;sn=54ef373e6ad6db9de26920882db6c84d&amp;chksm=9b484b72ac3fc264d58b67d1f986ed6aaebffba71c8b07a794de1b3fbf17d958db56ffe963ab&amp;scene=21#wechat_redirect">使用sequenza软件判定肿瘤纯度</a></p></li><li><p><a href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247486362&amp;idx=1&amp;sn=0e8ca112641b4d81608b598095255a6f&amp;chksm=9b484b21ac3fc237a1fecf40858a161f33f545f954049bf5142adbc5191615f62006a6c42509&amp;scene=21#wechat_redirect">GATK4的CNV流程-hg38</a></p></li></ul><p>当然，我没有推荐过的工具也有很多很优秀，欢迎大家给我们生信技能树投稿自己的软件使用心得哦。</p><h3><span>TCGA的CNV数据下载</span></h3><blockquote><p>众所周知，TCGA的数据的开放程度分成了4个等级，一般人都是下载level 3 的数据，对CNV数据也是如此。</p></blockquote><p>我比较喜欢去broad institute下载TCGA的数据，所有的文件都以目录的形式存放着：</p><pre><code><span>https:</span>/<span>/gdac.broadinstitute.org/runs</span><span>/stddata__latest/</span><br><span>https:</span>/<span>/gdac.broadinstitute.org/runs</span><span>/analyses__latest/</span> <br></code></pre><p>如果要下载level3的数据，就用<code>stddata__latest</code> 这个url即可，打开可以看到里面列出了所有的癌症种类，假如我们感兴趣的是BRCA，就直接点击进入，用下面的url即可。</p><pre><code><span>https:</span>/<span>/gdac.broadinstitute.org/runs</span><span>/analyses__latest/data</span><span>/BRCA/</span><span>20160128</span>/ <br></code></pre><p>打开url可以看到非常多的文件，这里我们感兴趣的是<strong>snp6芯片的拷贝数结果</strong>，而且一般是基于hg19版本的。</p><pre><code>wget -c -r -np -nH -k -L --cut-dirs <span>6</span> -p  -A <span>"*snp_6*hg19*Level_3*"</span>  <span>http:</span>/<span>/gdac.broadinstitute.org/runs</span><span>/stddata__2016_01_28/data</span><span>/BRCA/</span><span>20160128</span>/<br></code></pre><p>如果要下载其它癌症种类，只需要改变url里面的BRCA即可。<br>如果要下载其它类型的数据，只需要改变-A 后面的匹配规则即可，其实就是打开上面url看到的几十个文件的文件名的规律。</p><pre><code>"<span>*snp_6*</span>hg19<span>*Level_3*</span>" <br></code></pre><p>几分钟就下载完数据啦，然后你就会看到下面两个截然不同的：</p><pre><code>Merge<span>_snp_</span><span>_genome_</span>wide<span>_snp_</span>6<span>__broad_mit_edu__</span>Level<span>_3_</span><span>_segmented_</span>scna<span>_hg19_</span>_seg <br>Merge<span>_snp_</span><span>_genome_</span>wide<span>_snp_</span>6<span>__broad_mit_edu__</span>Level<span>_3_</span><span>_segmented_</span>scna<span>_minus_</span>germline<span>_cnv_</span>hg19__seg<br></code></pre><p>其中minus了germline的CNV的就是我们想要的癌症相关的somatic CNV咯!</p><h3><span>思考题</span></h3><p>既然SNP6.0芯片的拷贝数变异信息可以下载，那么TCGA计划的每个肿瘤病人的WES测序数据最后的CNV结果可以下载吗？如何下载额？</p><h3><span>对CNV区域注释基因</span></h3><p>这里可以参考我前两天分享的使用R语言完成bed格式的区域片段注释。</p><p><a href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247487422&amp;idx=2&amp;sn=1aa0b464dce837011252bae542ae9b5b&amp;chksm=9b484f05ac3fc613104af5b879f0618269a279bb39cba471db7c04293b12a3f8eae217444f3d&amp;scene=21#wechat_redirect" target="_blank">在R里面对坐标进行基因组区域注释</a><br></p><p>或者使用bedtools来注释基因，不过需要自己制作基因的各种元件坐标信息bed文件了，而不像R一样，种类繁多的注释包，按需要加载即可。</p><blockquote><p>前面我们下载的CNV都是基于基因组区域的，比如1号染色体的61735起始坐标到1510801终止坐标。在IGV里面倒是可以看出一些pattern，但是人们感兴趣的往往是这些位置上面到底有哪些基因。接下来就可以对基因进行各种下游分析。</p></blockquote><p>既然是对基因组片段做基因注释，那么首先就需要拿到基因的坐标信息咯，我是在gencode数据库里面下载，然后解析成下面的bed格式的，如下：</p><pre><code>head ~<span>/reference/gtf</span><span>/gencode/protein</span>_coding.hg19.position <br>chr1    <span>69091</span>   <span>70008</span>   OR4F5<br>chr1    <span>367640</span>  <span>368634</span>  OR4F29<br>chr1    <span>621096</span>  <span>622034</span>  OR4F16<br>chr1    <span>859308</span>  <span>879961</span>  SAMD11<br>chr1    <span>879584</span>  <span>894689</span>  NOC2L<br>chr1    <span>895967</span>  <span>901095</span>  KLHL17<br>chr1    <span>901877</span>  <span>911245</span>  PLEKHN1<br>chr1    <span>910584</span>  <span>917473</span>  PERM1<br>chr1    <span>934342</span>  <span>935552</span>  HES4<br>chr1    <span>936518</span>  <span>949921</span>  ISG15<br></code></pre><p>然后要把我们下载的<strong>CNV文本文件</strong>，转为bed格式的，就是把列的顺序调换一下：</p><pre><code><span>head</span> <span>Features</span><span>.bed</span>  <br><span>chr1</span>    3218610 95674710    <span>TCGA-3C-AAAU-10A-01D-A41E-01</span>    53225   0<span>.0055</span><br><span>chr1</span>    95676511    95676518    <span>TCGA-3C-AAAU-10A-01D-A41E-01</span>    2   <span>-1</span><span>.6636</span><br><span>chr1</span>    95680124    167057183   <span>TCGA-3C-AAAU-10A-01D-A41E-01</span>    24886   0<span>.0053</span><br><span>chr1</span>    167057495   167059336   <span>TCGA-3C-AAAU-10A-01D-A41E-01</span>    3   <span>-1</span><span>.0999</span><br><span>chr1</span>    167059760   181602002   <span>TCGA-3C-AAAU-10A-01D-A41E-01</span>    9213    <span>-8e-04</span><br><span>chr1</span>    181603120   181609567   <span>TCGA-3C-AAAU-10A-01D-A41E-01</span>    6   <span>-1</span><span>.2009</span><br><span>chr1</span>    181610685   201473647   <span>TCGA-3C-AAAU-10A-01D-A41E-01</span>    12002   0<span>.0055</span><br><span>chr1</span>    201474400   201474544   <span>TCGA-3C-AAAU-10A-01D-A41E-01</span>    2   <span>-1</span><span>.4235</span><br><span>chr1</span>    201475220   247813706   <span>TCGA-3C-AAAU-10A-01D-A41E-01</span>    29781   <span>-4e-04</span><br></code></pre><p>避免重复造轮子，我就用我擅长的bedtools解决这个需求吧，命令很简单，如下：</p><pre><code>bedtools intersect -a Features.bed  -b  ~<span>/reference/gtf</span><span>/gencode/protein</span>_coding.hg19.position  -wa -wb  \<br><span>| bedtools groupby -i - -g 1-4 -c 10 -o collapse<br></span></code></pre><p>注释结果，我挑了几个可以看的给大家，可以看到，每个CNV片段都注释到了对应的基因，有些特别大的片段，会被注释到非常多的基因。</p><pre><code>chr8    <span>42584924</span>    <span>42783715</span>    TCGA-5<span>T</span>-A9QA-01A-11D-A41E-<span>01</span>    CHRNB3,CHRNA6,THAP1,RNF170,HOOK3<br>chr8    <span>42789728</span>    <span>42793594</span>    TCGA-5<span>T</span>-A9QA-01A-11D-A41E-<span>01</span>    HOOK3<br>chr8    <span>42797957</span>    <span>42933372</span>    TCGA-5<span>T</span>-A9QA-01A-11D-A41E-<span>01</span>    RP11-598P20.5,FNTA,HOOK3<br>chr8    <span>70952673</span>    <span>70964372</span>    TCGA-5<span>T</span>-A9QA-01A-11D-A41E-<span>01</span>    PRDM14<br>chr10    <span>42947970</span>    <span>43833200</span>    TCGA-5<span>T</span>-A9QA-01A-11D-A41E-<span>01</span>    BMS1,RET,RASGEF1A,ZNF33B,CSGALNACT2<br>chr10    <span>106384615</span>   <span>106473355</span>   TCGA-5<span>T</span>-A9QA-01A-11D-A41E-<span>01</span>    SORCS3<br>chr10    <span>106478366</span>   <span>107298256</span>   TCGA-5<span>T</span>-A9QA-01A-11D-A41E-<span>01</span>    SORCS3<br>chr10    <span>117457285</span>   <span>117457859</span>   TCGA-5<span>T</span>-A9QA-01A-11D-A41E-<span>01</span>    ATRNL1<br>chr11    <span>68990173</span>    <span>69277078</span>    TCGA-5<span>T</span>-A9QA-01A-11D-A41E-<span>01</span>    MYEOV<br>chr11    <span>76378708</span>    <span>76926535</span>    TCGA-5<span>T</span>-A9QA-01A-11D-A41E-<span>01</span>    LRRC32,B3GNT6,OMP,TSKU,MYO7A,ACER3,CAPN5<br></code></pre><p>也算是间接回答了前些天在我们生信技能树论坛提问的朋友的疑问。</p><h3><span>使用GISTIC2寻找recurrent CNV regions</span></h3><blockquote><p>仔细看上面IGV的pattern你会发现某些染色体的某些片段经常会扩增或者缺失，这个现象就是人们想研究是recurrent CNV regions，当然不会用肉眼看咯，这时候需要用GISTIC这个软件。<br>找到了recurrent CNV regions同样是需要进行基因注释，才能进行下游分析咯。</p></blockquote><p>基于matlab开发的软件，所以对大部分生物信息学数据处理环境不友好，包括linux和mac，安装起来会比较费劲，而且使用起来也非常费劲。</p><ul><li><p>PICNIC 软件安装见我博客：http://www.bio-info-trainee.com/1299.html</p></li><li><p>GISTIC软件安装见我博客：http://www.bio-info-trainee.com/1648.html</p></li></ul><pre><code>nohup wget ftp://ftp.broadinstitute.org/pub/GISTIC2.0/GISTIC_2_0_23.tar.gz &amp;<br>tar zxvf GISTIC_2_0_23.tar.gz<br></code></pre><p>这里就不再赘述啦。</p><p>软件GISTIC 使用非常简单。</p><pre><code><span>#</span><span>!/bin/sh</span><br><span>#</span><span><span># run example GISTIC analysis</span></span><br>root=/home/jmzeng/Programs/GISTIC<br><span>#</span><span><span># output directory</span></span><br>echo --- creating output directory ---<br>basedir=`pwd`/example_results<br>mkdir -p $basedir<br><span>#</span><span><span>## 下面只是软件自带的测试数据例子。</span></span><br>echo --- running GISTIC ---<br><span>#</span><span><span># input file definitions</span></span><br>segfile=$root/examplefiles/segmentationfile.txt<br>markersfile=$root/examplefiles/markersfile.txt<br>refgenefile=$root/refgenefiles/hg16.mat  ## 软件自带，根据物种自己挑选合适的参考基因组<br>alf=$root/examplefiles/arraylistfile.txt<br>cnvfile=$root/examplefiles/cnvfile.txt<br><span>#</span><span><span># call script that sets MCR environment and calls GISTIC executable</span></span><br>./gistic2 -b $basedir -seg $segfile -mk $markersfile -refgene $refgenefile -alf $alf -cnv $cnvfile -genegistic 1 -smallmem 1 -broad 1 -brlen 0.5 -conf 0.90 -armpeel 1 -savegene 1 -gcm extreme<br><span>#</span><span><span># 通常运行真实数据，需要给的bed格式的cnv信息文件，以及对应的参考基因组的mat文件即可。其余的无所谓。</span></span><br></code></pre><h3><span>GISTIC2结果的下游分析</span></h3><p>这里可以使用maftools来结合每个肿瘤病人对应的somatic SNV信息可视化。</p><h3><span>泛癌研究之拷贝数变异</span></h3><p>发表于 September 2013 Pan-cancer patterns of somatic copy number alteration</p><h4><span>对拷贝数的分类整理</span></h4><p>We identified a total of 202,244 SCNAs, a median of 39 per cancer sample, comprising 6 categories:</p><ul><li><p><strong>focal SCNAs</strong> that were shorter than the chromosome arm (median of 11 amplifications and 12 deletions per sample);</p></li><li><p><strong>arm-level SCNAs</strong> that were chromosome-arm length or longer (median of 3 amplifications and 5 deletions per sample);</p></li><li><p><strong>copy-neutral loss-of-heterozygosity (LOH) events</strong> in which one allele was deleted and the other was amplified coextensively (median of 1 per sample);</p></li><li><p><strong>whole-genome duplications</strong> (WGDs; in 37% of cancers).</p></li></ul><p>By amplifications and deletions, we refer to copy number gains and losses, respectively, of any length and amplitude.</p><h4><span>使用 ABSOLUTE分析肿瘤纯度</span></h4><p>这款软件比较老旧了。</p><h4><span>统计学显著的拷贝数区域</span></h4><p>把经过肿瘤纯度矫正后的拷贝数情况输入到GISTIC软件，参数包括:-</p><ul><li><p><span>a noise threshold of 0.3,</span></p></li><li><p><span>a broad length cutoff of 0.5 chromosome arms,</span></p></li><li><p><span>a confidence level of 95% and a copy-ratio cap of 1.5.</span></p></li></ul><h3><span>拷贝数和表达量的关系</span></h3><p>这是一系列数据挖掘的分析，限于篇幅，不做过多介绍啦，感兴趣的可以私聊，我继续把剩余的介绍发给你咯。</p><section data-style-type="5" data-tools="新媒体排版" data-id="1207070"><section powered-by="xiumi.us"><section><section><section><p><strong>独家福利</strong></p></section><section><section></section></section></section></section></section></section><section data-style-type="5" data-tools="新媒体排版" data-id="1207058"><section data-style-type="5" data-tools="新媒体排版" data-id="1207203"><p><br></p><section data-tools="135编辑器" data-id="85398" data-color="rgb(245, 245, 244)" data-custom="rgb(33, 33, 34)"><section><section></section><section data-style="text-align:justify"><p><br></p><p><span>如果需要组装自己的服务器；</span><a href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247487052&amp;idx=1&amp;sn=9b577069df20032d9cd1d8665c1efcfd&amp;chksm=9b484ef7ac3fc7e1468b54905098a7d7fec17184599663c7090cfe57f94a35cc9dd417cba1a3&amp;scene=21#wechat_redirect" target="_blank"><span>代办生物信息学服务器</span></a></p><p><span>如果需要帮忙下载海外数据(GEO/TCGA/GTEx等等)，</span><a href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247487025&amp;idx=1&amp;sn=18fdfdaddfa9226f0b6107986cec9043&amp;chksm=9b484e8aac3fc79cf1616837a4f5a91efb16559434a3c93fabdecb59da53a97e4a1ab9c6e778&amp;scene=21#wechat_redirect" target="_blank"><span>点我？</span></a></p><p><span>如果需要线下辅导及培训，看</span><a href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247486794&amp;idx=1&amp;sn=60ed4be8c35d3bc584641987b4b8a942&amp;chksm=9b484df1ac3fc4e73ab072129306eca9469cff0a94a9eddc5daaa17faaa9f9c7fb71c5041c94&amp;scene=21#wechat_redirect" target="_blank"><span>招学徒</span></a></p><p><span>如果需要个人电脑：</span><a href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247487098&amp;idx=1&amp;sn=0e28a20911f5a76446bb0363c9bc9d84&amp;chksm=9b484ec1ac3fc7d70ff3e6e0a2589b29e7aa766e0fc0fd9ef106af7ae821e9ca288d316cdd23&amp;scene=21#wechat_redirect" target="_blank"><span>个人计算机推荐</span></a></p><p><span>如果需要置办生物信息学书籍，看：</span><a href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247487033&amp;idx=1&amp;sn=413e69a526f4f97ab62b619a1f289eb6&amp;chksm=9b484e82ac3fc794cabae9f58dde5cc7ea65eeae0c74aed89a4d78723628fd08bb6692c37769&amp;scene=21#wechat_redirect" target="_blank"><span>生信人必备书单</span></a></p><p><span>如果需要实习岗位：</span><a href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247487081&amp;idx=1&amp;sn=60a620fb3dc64814e891f03b214a23b2&amp;chksm=9b484ed2ac3fc7c4facaa45a46878f28672d390ec6c5ee99f5ac39208089386c9b95fee5b1c7&amp;scene=21#wechat_redirect" target="_blank"><span>实习职位发布</span></a></p><p><span>如果需要售后：</span><span><a href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247487213&amp;idx=1&amp;sn=8154b0dddba81c493a649e21a9744fd0&amp;chksm=9b484e56ac3fc740bfe3b5aef3177fa11e312bd1b12624c3970c6d6a78252afff0b1f9d2bc87&amp;scene=21#wechat_redirect" target="_blank">点我</a></span></p></section></section></section></section></section><p><br></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/uOB46wdyEvdPkmrlNCbNLw",target="_blank" rel="noopener noreferrer">原文链接</a>
