---
title: "使用Clipper控制FDR应对高通量数据分析p值失真问题"
date: 2023-07-05T06:09:36Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Acdemic"]
---
使用Clipper控制FDR应对高通量数据分析p值失真问题 by 生信菜鸟团
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><blockquote data-tool="mdnice编辑器"><p>上期推文我们讨论了，在DESeq2的分析结果中，一些p值，特别是adj.p出现NA的情况。关于多重检验的p值，我们经常使用<strong>FDR</strong> （Benjamini and Hochberg（BH））方法进行校正，但其实我们平时常用的校正方法对数据分布和样本量都有要求，而我们通过高通量测序得到的表达矩阵往往是样本量较小、特征维度高的数据，导致我们在使用这些工具方法时容易忽略统计学意义上的严谨性。美国加州大学洛杉矶分校统计系的李婧翌团队开发的Clipper工具使用户能在无需计算p值和假设参数模型的情况下直接控制高通量数据分析中的假阳性率，提高分析结果的可靠度，本期我们将从差异表达分析中的FDR入手，学习这个工具。</p></blockquote><h1 data-tool="mdnice编辑器"><span></span>FDR</h1><p data-tool="mdnice编辑器">我们往期有许多推文介绍了各种进行多重检验矫正的方法，其中就包括了如何计算FDR</p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzUzMTEwODk0Ng==&amp;mid=2247498432&amp;idx=1&amp;sn=a20c7bcb1d449880329091a49c66fc18&amp;scene=21#wechat_redirect" data-linktype="2">一文了解P-value，多重比较，FDR和Q value的差别</a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzUzMTEwODk0Ng==&amp;mid=2247505526&amp;idx=1&amp;sn=18bf163ac9e51024dce03d8b70d3d599&amp;scene=21#wechat_redirect" data-linktype="2">转录组差异分析P值与FDR值区别有多大</a></p><p data-tool="mdnice编辑器"><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247490048&amp;idx=2&amp;sn=6f4da175cbe2ad9feb11857dcfc50b27&amp;scene=21#wechat_redirect" data-linktype="2">p.value和FDR</a></p><p data-tool="mdnice编辑器">首先我们需亚了解差异表达分析的基本假设：</p><ul data-tool="mdnice编辑器"><li><section>H0:差别是由抽样误差所致；</section></li><li><section>H1:差别不是由抽样误差所致，即并不来自同一总体</section></li></ul><pre data-tool="mdnice编辑器"><span></span><code>###这里我用到哈佛大学统计的一个数据集<br>library(devtools)<br>install_github("genomicsclass/GSE5859Subset")<br><br>##加载对应的包和数据<br><br>library(qvalue)<br>library(GSE5859Subset)<br>data(GSE5859Subset)<br><br>### 查看数据<br>dim(geneExpression)<br></code></pre><p data-tool="mdnice编辑器">看看一个单一的基因:</p><pre data-tool="mdnice编辑器"><span></span><code>g&lt;- sampleInfo$group<br><br>e&lt;- geneExpression[25,]<br>e<br><br># t-test, expression should be normal distribution<br># 正态分布假设<br>qqnorm(e[g==1])<br>qqline(e[g==1])<br>qqnorm(e[g==0])<br>qqline(e[g==0])<br></code></pre><p data-tool="mdnice编辑器">对其进行t-test：</p><pre data-tool="mdnice编辑器"><span></span><code>t.test(e[g==1], e[g==0])<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code> Welch Two Sample t-test<br><br>data:  e[g == 1] and e[g == 0]<br>t = 0.28382, df = 21.217, p-value = 0.7793<br>alternative hypothesis: <span>true</span> difference <span>in</span> means is not equal to 0<br>95 percent confidence interval:<br> -0.1431452  0.1884244<br>sample estimates:<br>mean of x mean of y <br> 10.52505  10.50241 <br></code></pre><p data-tool="mdnice编辑器">对所有基因进行 t 检验</p><pre data-tool="mdnice编辑器"><span></span><code>mytest&lt;- function(x) t.test(x[g==1], x[g==0], var.equal=T)$p.value<br><br>pvals&lt;- apply(geneExpression, 1, mytest)<br><br>sum(pvals&lt; 0.05)  # 统计p值小于0.05的基因个数<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>[1] 1383<br></code></pre><p data-tool="mdnice编辑器">看看 p 值分布</p><pre data-tool="mdnice编辑器"><span></span><code>hist(pvals)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.6181262729124236" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O2VbpOaAx8OINico30D47ltgjcRvZWsn8deKYQfOvsg2ZA5eaLWTOWO7A/640?wx_fmt=png" data-type="png" data-w="982" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O2VbpOaAx8OINico30D47ltgjcRvZWsn8deKYQfOvsg2ZA5eaLWTOWO7A/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">用随机数据模拟多重比较</p><pre data-tool="mdnice编辑器"><span></span><code>m&lt;- nrow(geneExpression)<br>n&lt;- ncol(geneExpression)<br><br># generate random numbers<br>randomData&lt;- matrix(rnorm(n*m), m, n)<br>nullpvalues&lt;- apply(randomData, 1, mytest)<br>hist(nullpvalues)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.6181262729124236" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O2K5OMYOB1EjWt2rfNejMicRmQEZHGLKgT6AKrwBrlXT2OHL7wD1BNO9g/640?wx_fmt=png" data-type="png" data-w="982" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O2K5OMYOB1EjWt2rfNejMicRmQEZHGLKgT6AKrwBrlXT2OHL7wD1BNO9g/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">我们随机生成数据，应该没有表达的基因。然而，我们看到了<strong>几乎均匀分布的 p 值。</strong></p><p data-tool="mdnice编辑器">所以，控制多重比较的假阳性是十分必要的</p><p data-tool="mdnice编辑器">常见方法：</p><ul data-tool="mdnice编辑器"><li><section>Bonferroni 校正</section></li></ul><p data-tool="mdnice编辑器">直接用p值除以进行比较的次数就得到校正后p值，但这种方法非常保守，一般用于全基因组关联研究 (GWAS)</p><ul data-tool="mdnice编辑器"><li><section><strong>FDR</strong> （Benjamini and Hochberg（BH））</section></li></ul><p data-tool="mdnice编辑器">错误发现率FDR= 0.05。这意味着在所有发现中，预计有 5% 是假阳性。</p><p data-tool="mdnice编辑器">一种控制 FDR 的方法：让 k 成为最大的 i，使得p(i) &lt;= (i/m) * alpha，（m 是比较次数）然后拒绝 H(i) for i =1, 2, …k</p><p data-tool="mdnice编辑器">通过下面的图，可以更好理解</p><pre data-tool="mdnice编辑器"><span></span><code>## order the pvals computed above and plot it.<br>alpha = 0.05<br>m = length(pvals)<br>#m is the number of 8793 comparisons <br><br>plot(x=seq(1,100), y=pvals[order(pvals)][1:100])<br>abline(a=0, b=alpha/m)  # y = alpha/m * x + 0 (x = 1,2,3...)<br>title("slop is alpha/m")<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.6181262729124236" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O2wTAe7ArcV3yZVn7baLibz30rI3HtCkoezktz7sHGTQ4Kcyfy6Xs8goQ/640?wx_fmt=png" data-type="png" data-w="982" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O2wTAe7ArcV3yZVn7baLibz30rI3HtCkoezktz7sHGTQ4Kcyfy6Xs8goQ/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code># let's zoom in to look at the first 15 p values from small to big<br><br>plot(x=seq(1,100), y=pvals[order(pvals)][1:100], xlim=c(1,15))<br>abline(a=0, b=alpha/m)<br>title("slop is alpha/m")<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.6181262729124236" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O2icQk4plyppCfvocYz6lLTsaaQcMAh8ibLCOd4JibQ1IPQXrkZf2YtOnsQ/640?wx_fmt=png" data-type="png" data-w="982" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O2icQk4plyppCfvocYz6lLTsaaQcMAh8ibLCOd4JibQ1IPQXrkZf2YtOnsQ/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">我们可以看到第 14 个 p 值大于它自己的阈值，由 (0.05/m) * 14 = 7.960878e-05 计算</p><pre data-tool="mdnice编辑器"><span></span><code># current threshold<br>(0.05/m)*14<br><br># current pvalue<br>pvals[order(pvals)][14]  # 比threshold大<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>[1] 7.960878e-05<br> 202556_s_at <br>8.503827e-05 <br></code></pre><p data-tool="mdnice编辑器">我们将使用 p.adjust 函数和方法“fdr”或“BH”来纠正p 值</p><p data-tool="mdnice编辑器">p(i)&lt;= (i/m) * alpha</p><p data-tool="mdnice编辑器">p(i) * m/i &lt;= alpha</p><p data-tool="mdnice编辑器"><strong>p_adjusted(i) = p(i) * (m/i)，其中 p(i) 是原始的第 i 个 p 值，m 是的检验个数</strong></p><pre data-tool="mdnice编辑器"><span></span><code>sum( p.adjust(pvals, method="fdr") &lt; 0.05 )  # 13<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>sum( p.adjust(pvals, method="BH") &lt; 0.05 )  # 13<br></code></pre><p data-tool="mdnice编辑器">从图中看到的一样</p><pre data-tool="mdnice编辑器"><span></span><code>adj.pvals &lt;- p.adjust(pvals, method="fdr")<br>compare_list &lt;- data.frame(org.pvalue = pvals[order(pvals)][1:13],<br>           adj.pvalue = adj.pvals[order(adj.pvals)][1:13])<br><br># p_adjusted(i) = p(i) * (m/i)<br>i &lt;- seq(1:13)<br>man.adj.pvals &lt;- pvals[order(pvals)][1:13] * (m/i)<br>man.adj.pvals<br><br>compare_list$man.adj.pvalue &lt;- man.adj.pvals<br>compare_list<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>org.pvalue<br>&lt;dbl&gt;<br>adj.pvalue<br>&lt;dbl&gt;<br>man.adj.pvalue<br>&lt;dbl&gt;<br>201909_at 8.453679e-23 7.433320e-19 7.433320e-19<br>214218_s_at 5.429425e-20 2.387047e-16 2.387047e-16<br>206700_s_at 1.044354e-19 3.061001e-16 3.061001e-16<br>205000_at 3.691745e-19 8.115378e-16 8.115378e-16<br>204409_s_at 1.224535e-16 2.153468e-13 2.153468e-13<br>206624_at 5.053589e-12 7.406034e-09 7.406034e-09<br>203974_at 9.187018e-09 1.154021e-05 1.154021e-05<br>200933_x_at 1.385484e-08 1.522820e-05 1.522820e-05<br>201018_at 6.702439e-07 6.548283e-04 6.548283e-04<br>203992_s_at 3.209897e-06 2.822463e-03 2.822463e-03<br>204061_at 1.293600e-05 1.034056e-02 1.034056e-02<br>211149_at 3.493280e-05 2.559701e-02 2.559701e-02<br>201029_s_at 3.838790e-05 2.596499e-02 2.596499e-02<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>p.adjust.methods<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>[1] <span>"holm"</span>       <span>"hochberg"</span>   <span>"hommel"</span>     <span>"bonferroni"</span><br>[5] <span>"BH"</span>         <span>"BY"</span>         <span>"fdr"</span>        <span>"none"</span><br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>sum( p.adjust(pvals, method="bonferroni") &lt; 0.05 )  # 10 很保守<br></code></pre><h1 data-tool="mdnice编辑器"><span></span>clipper</h1><p data-tool="mdnice编辑器">参考资料：</p><blockquote data-tool="mdnice编辑器"><p>Clipper:  p -value-free FDR control on high-throughput data from two conditions<svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" width="22px" height="22px" viewbox="0 0 32 32" enable-background="new 0 0 32 32"><image width="32" height="32" x="0" y="0"></image></svg></p><p><a href="https://mp.weixin.qq.com/s?__biz=MzU5ODg0MTAwMw==&amp;mid=2247498603&amp;idx=1&amp;sn=5abe7f433b16393c153caadf6f6469d6&amp;scene=21#wechat_redirect" data-linktype="2">斯隆奖获得者李婧翌：AI+X并非总是有效，生物数据量小、噪音大，可解释性是关键</a></p><p><a href="https://mp.weixin.qq.com/s?__biz=MzA3MzQyNjY1MQ==&amp;mid=2652557373&amp;idx=6&amp;sn=6ff49cc8a6182f52b2ef8cc7a574ffc5&amp;scene=21#wechat_redirect" data-linktype="2">李婧翌团队提出针对高通量数据进行富集或差异分析的统计学方法Clipper，无需计算p值即可实现对假发现率的控制</a></p><p><a href="https://mp.weixin.qq.com/s?__biz=MzA3MzQyNjY1MQ==&amp;mid=2652577881&amp;idx=7&amp;sn=512f82c5a09421309ae833ec917dfdd4&amp;scene=21#wechat_redirect" data-linktype="2">Genome Biology | 李婧翌/李蔚团队合作报道流行的差异表达分析软件在人群数据上有极高的假发现率</a></p><p>Introduction to Clipper: p-value-free FDR control on high-throughput data from two conditions</p><p>https://github.com/JSB-UCLA/Clipper/blob/master/vignettes/</p></blockquote><p data-tool="mdnice编辑器">虽然上述统计学方法被广泛使用，但一个常见的问题是，生物数据往往样本量过小，很多针对新数据类型开发的计算方法无法或者很难给出正确的p值。李婧翌研究团队发现，非参数秩和检验在评估中表现最好，像DESeq2和edgeR这类基于参数模型假设的方法不能很好地控制假发现率的一个原因是实际的群体数据并不符合它们假设的分布。并提出了一种新的计算方法，使用户能在无需计算p值的情况下直接控制高通量数据分析中的假阳性率。</p><p data-tool="mdnice编辑器">Clipper的优势在于无需对数据分布进行参数化的假设，从而适用于样本量小的情况，避免了p值计算的难点，并节省了p值计算的时间</p><figure data-tool="mdnice编辑器"><img data-ratio="0.7194444444444444" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O2yuib1DYOzQiar0wETTsLwyC4hl0ThJhFGdsnWdmVIrzLfQU6LXMmEEcg/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O2yuib1DYOzQiar0wETTsLwyC4hl0ThJhFGdsnWdmVIrzLfQU6LXMmEEcg/640?wx_fmt=png"></figure><figure data-tool="mdnice编辑器"><img data-ratio="0.5638888888888889" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O2med98xYmqJ1lOLZAemKE6ZadgnksZyNxzZRX4OyNeLaSoWbLS46f0w/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O2med98xYmqJ1lOLZAemKE6ZadgnksZyNxzZRX4OyNeLaSoWbLS46f0w/640?wx_fmt=png"></figure><figure data-tool="mdnice编辑器"><img data-ratio="0.5770700636942675" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O2Tw4pkumL5PMGp92Br5jFbVgRhdmGFUiadKcTU45GpicEdrgekd2S30Gg/640?wx_fmt=png" data-type="png" data-w="785" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O2Tw4pkumL5PMGp92Br5jFbVgRhdmGFUiadKcTU45GpicEdrgekd2S30Gg/640?wx_fmt=png"></figure><p>根据文章的描述，Clipper可以应用于多个高通量数据分析场景</p><p><img data-galleryid="" data-ratio="0.5789473684210527" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O2ibvTODA1RfN4UibEOf7ovLk3NGulBDrX6Jyo9rTdFpEIQk7Se9CaWIJg/640?wx_fmt=jpeg" data-type="jpeg" data-w="893" src="https://mmbiz.qpic.cn/mmbiz_jpg/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O2ibvTODA1RfN4UibEOf7ovLk3NGulBDrX6Jyo9rTdFpEIQk7Se9CaWIJg/640?wx_fmt=jpeg"></p><p>这里我们将挑转录组常用“DEG analysis from RNA-seq data”进行介绍</p><h2 data-tool="mdnice编辑器"><span></span>Analyzing high-throughput data with Clipper</h2><p data-tool="mdnice编辑器">https://github.com/JSB-UCLA/Clipper/blob/master/vignettes/Clipper.pdf</p><pre data-tool="mdnice编辑器"><span></span><code>library(devtools)<br>install_github("JSB-UCLA/Clipper")<br><br>library(Clipper)<br></code></pre><h3 data-tool="mdnice编辑器"><span></span>General pipeline<span></span></h3><p data-tool="mdnice编辑器">下面的代码中我们使用的是包内置数据集，无需另外下载</p><h4 data-tool="mdnice编辑器"><span></span>Enrichment analysis<span></span></h4><p data-tool="mdnice编辑器">测试数据集特征：there are 10,000 features and the first 1000 features are interesting.</p><pre data-tool="mdnice编辑器"><span></span><code>exp_e[c(1:3, 1001:1003), ]<br><br>back_e[c(1:3, 1001:1003), ]<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>           [,1]        [,2]      [,3]<br>[1,]  4.9914685  5.26705194  3.945790<br>[2,]  4.8837421  4.65166005  6.889625<br>[3,]  5.0160066  5.00388853  4.830051<br>[4,] -0.3823919  1.70146747 -0.335157<br>[5,] -0.9822577  0.75032253  1.722063<br>[6,] -0.3423613 -0.06262634  1.129383<br>           [,1]        [,2]       [,3]<br>[1,] -0.6264538  0.18364332 -0.8356286<br>[2,]  1.5952808  0.32950777 -0.8204684<br>[3,]  0.4874291  0.73832471  0.5757814<br>[4,]  0.7391149  0.38660873  1.2963972<br>[5,] -0.8035584 -1.60262567  0.9332510<br>[6,]  1.8060893 -0.05650363  1.8859113<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>re1 &lt;- Clipper(score.exp = exp_e, score.back = back_e,<br>               analysis = "enrichment",<br>               FDR = c(0.01, 0.05, 0.1))<br><br>names(re1)<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>[1] <span>"contrast.score"</span>       <span>"contrast.score.value"</span><br>[3] <span>"FDR"</span>                  <span>"contrast.score.thre"</span> <br>[5] <span>"discoveries"</span>          <span>"q"</span>   <br></code></pre><p data-tool="mdnice编辑器">分析结果：Clipper returns a list and its component <code>discovery</code> is a <strong>list</strong> of indices of identified interesting discoveries <strong>corresponding to the FDR threshold(s)</strong></p><pre data-tool="mdnice编辑器"><span></span><code>#indices of identified interesting genes with the second input FDR threshold (0.05)<br>re1$discoveries[[2]][1:5]<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>[1] 1 2 3 4 5<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>lapply(re1$discoveries, length)<br><br>re1$FDR<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>[[1]]<br>[1] 971<br><br>[[2]]<br>[1] 1041<br><br>[[3]]<br>[1] 1104<br><br>[1] 0.01 0.05 0.10<br></code></pre><p data-tool="mdnice编辑器">We can then calculate the resulting false discovery proportion (FDP) and power:</p><pre data-tool="mdnice编辑器"><span></span><code>#FDP (the first 1000 genes are true positives)<br>trueid &lt;- 1:1000<br><br>sum(!re1$discoveries[[1]] %in% trueid)<br>length(re1$discoveries[[1]])<br><br>sum(!re1$discoveries[[1]] %in% trueid)/length(re1$discoveries[[1]])<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>[1] 5<br>[1] 971<br>[1] 0.005149331<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>#power<br>sum(re1$discoveries[[1]] %in% trueid)/length(trueid)<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>[1] 0.966<br></code></pre><h4 data-tool="mdnice编辑器"><span></span>Differential analysis<span></span></h4><p data-tool="mdnice编辑器">测试数据集特征：there are 10,000 features and the fifirst 2000 features are interesting</p><pre data-tool="mdnice编辑器"><span></span><code>exp_d[c(1:3, 1001:1003), ]<br><br>back_d[c(1:3, 1001:1003), ]<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>         replicates1 replicates2 replicates3<br>gene1             40          43          29<br>gene2             27          46          39<br>gene3             33          27          35<br>gene1001           6           9           4<br>gene1002           2           1           2<br>gene1003           1           3           5<br>         replicates1 replicates2 replicates3<br>gene1             19          28          27<br>gene2             29          18          29<br>gene3             17          17          13<br>gene1001          24          28          23<br>gene1002          15          16          18<br>gene1003           8          12           6<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>re2 &lt;- Clipper(score.exp = exp_d, score.back = back_d, analysis = "differential",FDR = c(0.01, 0.05, 0.1))<br><br>names(re2)<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>[1] <span>"contrast.score"</span>       <span>"contrast.score.value"</span><br>[3] <span>"FDR"</span>                  <span>"contrast.score.thre"</span> <br>[5] <span>"discoveries"</span>          <span>"q"</span>  <br></code></pre><p data-tool="mdnice编辑器">分析结果：Clipper still returns a list and its component discovery is a list of indices of identified interesting discoveries corresponding to the FDR threshold(s):</p><pre data-tool="mdnice编辑器"><span></span><code>#indices of identified interesting genes with the second input FDR threshold (0.05)<br>re2$discoveries[[2]][1:5]<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>[1] 1 2 3 5 6<br></code></pre><p data-tool="mdnice编辑器">We can then calculate the resulting false discovery proportion (FDP) and power:</p><pre data-tool="mdnice编辑器"><span></span><code>#FDP (the first 2000 genes are true positives)<br>trueid &lt;- 1:2000<br>sum(!re2$discoveries[[1]] %in% trueid)/length(re2$discoveries[[1]])<br><br>#power<br>sum(re2$discoveries[[1]] %in% trueid)/length(trueid)<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>[1] 0.004234298<br>[1] 0.7055<br></code></pre><h3 data-tool="mdnice编辑器"><span></span>DEG identifification<span></span></h3><p data-tool="mdnice编辑器">Users are recommended to <strong>normalize</strong> and <strong>log-tranform</strong> read count matrices before inputing them to Clipper</p><p data-tool="mdnice编辑器">The following example uses <strong>edgeR</strong> to preprocess raw read count matrices as an example.Users can also use their preferred normalization methods.</p><pre data-tool="mdnice编辑器"><span></span><code>library(edgeR)<br>dataurl = 'http://www.stat.ucla.edu/~jingyi.li/data/Clipper/'<br>count1 = readRDS(url(paste0(dataurl, 'simcount1.rds')))<br>count2 = readRDS(url(paste0(dataurl, 'simcount2.rds')))<br>############ use edgeR to normalize ############<br>r1 = ncol(count1)  # 3<br>r2 = ncol(count2)  # 3<br>cond_idx = rep(2, r1 + r2)<br>cond_idx[1:r1] = 1<br>cond_idx<br><br>dat = cbind(count1, count2)<br>cond_idx = factor(cond_idx)<br>y &lt;- DGEList(counts=dat,group=cond_idx)<br>y &lt;- calcNormFactors(y)<br>count_norm = cpm(y)<br></code></pre><p data-tool="mdnice编辑器">Next we apply Clipper to preprocessed gene expression matrices. <strong>Log-transformation</strong> is recommendedbecause <strong>empirical evidence indicates that log-transformed read counts are more likely to satisfy Clipper’s assumption for FDR control than read counts without log-transformation.</strong></p><pre data-tool="mdnice编辑器"><span></span><code>##### apply clipper to log2-transformed matrices with target FDR = 0.05<br>re_clipper = Clipper(<br>  score.exp = log(base = 2, count_norm[,1:r1] + 1),<br>  score.back = log(base = 2, count_norm[,-(1:r1)] + 1),<br>  FDR = 0.05,<br>  analysis = "differential")<br><br>head(re_clipper$discoveries[[1]]) ###### identified DEG indices at FDR = 0.05<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>[1]  676 2173 3103 3105 3484 3490<br></code></pre><p data-tool="mdnice编辑器">手动查看阈值，和通过阈值的结果对应分数</p><pre data-tool="mdnice编辑器"><span></span><code>re_clipper$contrast.score.thre<br><br>re_clipper$contrast.score.value[re_clipper$discoveries[[1]]]<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>[1] 6.122079<br> [1]  6.226567  6.754237  7.691055  7.557135  7.713961  6.427339<br> [7]  6.146831  6.644286  6.122079  6.131032  8.234947  7.585539<br>[13]  6.404031  6.796154  7.091714  6.388610 10.743190  7.561171<br>[19]  6.695440  6.452022  9.600434<br></code></pre><p data-tool="mdnice编辑器">将结果与edgeR分析结果比较</p><pre data-tool="mdnice编辑器"><span></span><code># edgeR<br>#1 表达矩阵<br>#2 实验设计<br>design &lt;- model.matrix(~0+rev(cond_idx))<br>design<br>rownames(design) &lt;- colnames(dat)<br>colnames(design) &lt;- levels(factor(cond_idx))<br>design<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>#3 计算线性模型的参数<br>DEG &lt;- estimateGLMCommonDisp(y,design)<br>DEG &lt;- estimateGLMTrendedDisp(DEG, design)<br>DEG &lt;- estimateGLMTagwiseDisp(DEG, design)<br>#4 拟合线性模型<br>fit &lt;- glmFit(DEG, design)<br>lrt &lt;- glmLRT(fit, contrast=c(1,-1)) <br>#5 提取过滤差异分析结果<br>DEG_edgeR &lt;- as.data.frame(topTags(lrt, n=nrow(DEG)))<br>head(DEG_edgeR)<br><br># 设定阈值，筛选显著上下调差异基因<br>fc &lt;- 2.0<br>p &lt;- 0.05<br>DEG_edgeR$regulated &lt;- ifelse(DEG_edgeR$logFC&gt;log2(fc)&amp;DEG_edgeR$FDR&lt;p,"up",<br>                              ifelse(DEG_edgeR$logFC&lt;(-log2(fc))&amp;DEG_edgeR$FDR&lt;p,"down","normal"))<br></code></pre><p data-tool="mdnice编辑器">结果比较：</p><p data-tool="mdnice编辑器">韦恩图</p><pre data-tool="mdnice编辑器"><span></span><code># edgeR<br>table(DEG_edgeR$regulated)  # 121 DEGs<br># clipper<br>length(re_clipper$discoveries[[1]])<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>  down normal     up <br>    65  52255     56 <br>[1] 21<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>edgeR_DEGs &lt;- rownames(DEG_edgeR[DEG_edgeR$regulated != "normal",])<br>clipper_DEGs &lt;- rownames(dat[re_clipper$discoveries[[1]],])<br><br>##画韦恩图<br>library(VennDiagram)<br>venn.diagram(list( edgeR=edgeR_DEGs, <br>                   clipper=clipper_DEGs), <br>             fill=c("red","blue"),imagetype = "png",<br>             main.cex = 2,pattern = 'Euler_3set_scaled',<br>             filename="./Venn.png")<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O2IUnJPIp85L6yDQTyg8iaEtBtPvAVpHcSJJWuDKwXlMfQmEogicUMN7yg/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O2IUnJPIp85L6yDQTyg8iaEtBtPvAVpHcSJJWuDKwXlMfQmEogicUMN7yg/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">散点图</p><pre data-tool="mdnice编辑器"><span></span><code># 散点图<br># 相关性<br># 交集基因 根据venn图结果其实就是clipper的discoveries<br>library(tidyverse)<br>overlap_genes_df &lt;- data.frame(genes = clipper_DEGs)<br>overlap_genes_df &lt;- overlap_genes_df %&gt;% <br>  mutate(logfc_rank = rank(-abs(DEG_edgeR$logFC[match(.$genes,rownames(DEG_edgeR))])),  # 绝对值从大到小<br>         edgeR_Pvalue_rank = rank(DEG_edgeR$PValue[match(.$genes,rownames(DEG_edgeR))]),  # 从小到大<br>         clipper_contrast_score_rank = rank(-re_clipper$contrast.score.value[re_clipper$discoveries[[1]]]))  # 从大到小<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>library(ggplot2)<br># basic scatterplot<br>scatter_plot1 &lt;- ggplot(overlap_genes_df, aes(x=edgeR_Pvalue_rank, y=logfc_rank)) + <br>    geom_point()<br><br>scatter_plot1+labs(caption =  str_c("Spearman Cor",cor(overlap_genes_df$logfc_rank,overlap_genes_df$edgeR_Pvalue_rank),sep = "="))<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.6181262729124236" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O24iaaIpxF954LdhcUnnZfQYtib98JDibL19n2wr25DF9ib7AM14IdplXNOA/640?wx_fmt=png" data-type="png" data-w="982" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O24iaaIpxF954LdhcUnnZfQYtib98JDibL19n2wr25DF9ib7AM14IdplXNOA/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code>library(ggplot2)<br># basic scatterplot<br>scatter_plot1 &lt;- ggplot(overlap_genes_df, aes(x=clipper_contrast_score_rank, y=logfc_rank)) + <br>    geom_point()<br><br>scatter_plot1+labs(caption =  str_c("Spearman Cor",cor(overlap_genes_df$logfc_rank,overlap_genes_df$clipper_contrast_score_rank),sep = "="))<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.6181262729124236" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O25LbRbR289DEWvmREdNxKFFibFCxuEQriauRUFtOq5KNEqaHfp0X36d3Q/640?wx_fmt=png" data-type="png" data-w="982" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O25LbRbR289DEWvmREdNxKFFibFCxuEQriauRUFtOq5KNEqaHfp0X36d3Q/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">还是可以看出基本的logFC绝对值和clipper打分的相关系数比edgeR的P值高一点，可能因为我们这里的基因很少，只有21个所以差别很小</p><p data-tool="mdnice编辑器">下面是作者文章的相关结果：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.3347763347763348" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O2ibJHE2OtZf8APWiaD6AcbVzIMSrFpmLGKiahXVtF9XIibnRpqiaCewf4mvw/640?wx_fmt=png" data-type="png" data-w="693" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O2ibJHE2OtZf8APWiaD6AcbVzIMSrFpmLGKiahXVtF9XIibnRpqiaCewf4mvw/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">作者将clipper方法和我们常用的方法在不同方面做了比较，我们在这里主要还是介绍工具，使用的是内置数据集，顺带对结果进行简单比较验证，感兴趣的同学可以拿自己的数据参考作者原文，在自己需要的应用方向上进行比较</p><h1 data-tool="mdnice编辑器"><span></span>Difffferential gene expression anlaysis with Clipper</h1><p data-tool="mdnice编辑器">有的时候比如前面那样需要进行标准化，或者有批次效应需要处理，还是可以基于常用的DESeq2和edgeR先进行分析，再使用Clipper控制FDR</p><p data-tool="mdnice编辑器">在项目仓库中也有这么一个示例文档</p><p data-tool="mdnice编辑器">https://github.com/JSB-UCLA/Clipper/blob/master/vignettes/DEG_extra_covariate.pdf</p><p data-tool="mdnice编辑器">可惜的是文档作者提供的示例文件，现在无法访问</p><p data-tool="mdnice编辑器">http://jsb1data.stat.ucla.edu/repository/xinzhou/example_data.rds</p><p data-tool="mdnice编辑器">但我们这里仍直接截图带大家看看示例代码是如何工作的</p><figure data-tool="mdnice编辑器"><img data-ratio="0.5370370370370371" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O2rMoD7y6Q0Lm1iapVU2icuZsfQuVRhIiazRQmgqFibVEyUWsa9S3iakVXfrQ/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O2rMoD7y6Q0Lm1iapVU2icuZsfQuVRhIiazRQmgqFibVEyUWsa9S3iakVXfrQ/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">首先准备好自己的数据集，以及感兴趣的分组和批次，且因为这里是使用scDesign3产生的示例数据，真正的DEGs我们已知以便后面计算FDR和power</p><figure data-tool="mdnice编辑器"><img data-ratio="0.525" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O2nib6w00R9koQiaZiaSxht7iariatBcLwxDfsqw8bn5FtmyPjBfNP55hDiazw/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O2nib6w00R9koQiaZiaSxht7iariatBcLwxDfsqw8bn5FtmyPjBfNP55hDiazw/640?wx_fmt=png"></figure><figure data-tool="mdnice编辑器"><img data-ratio="0.5638888888888889" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O218qQMnUcK6ib3E4hSBfeA1ot4bKDWoBTqkCexDogL6hXYq29bnugkkQ/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O218qQMnUcK6ib3E4hSBfeA1ot4bKDWoBTqkCexDogL6hXYq29bnugkkQ/640?wx_fmt=png"></figure><figure data-tool="mdnice编辑器"><img data-ratio="0.2777777777777778" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O22g2vO4t9euofa1DtN9XmsKTFyib65HHs2GsBGOGibSGQBWibt9ZJibwv0A/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O22g2vO4t9euofa1DtN9XmsKTFyib65HHs2GsBGOGibSGQBWibt9ZJibwv0A/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">几个包装好的函数：使用DESeq2、edgeR进行差异分析（批次效应并不和我们之前谈到的那样提前去除，而是作为实验设计design的一部分），筛选出DEGs以及计算FDR和power</p><figure data-tool="mdnice编辑器"><img data-ratio="0.21818181818181817" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O2lI3SBiajkgKbunFFnRA5oOibGJ6enlJJLDZy6S7u4o6afuWrowGjnTBA/640?wx_fmt=png" data-type="png" data-w="1045" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O2lI3SBiajkgKbunFFnRA5oOibGJ6enlJJLDZy6S7u4o6afuWrowGjnTBA/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">使用这些函数处理数据集获得相关结果</p><p data-tool="mdnice编辑器">接下来，介绍了两种不同的方法创建null dataset，并在此基础上开始引入Clipper结合original和null的数据集来控制FDR，在使用Clipper前还需考虑其前提假设是否成立：<strong>来自两个数据集（original和null）的输入数据（这里的指，-log转换后的p值）对所有非DEGs是否遵循相同的分布。如果违反了这个假设，还需有一个额外的标准化步骤。</strong></p><blockquote data-tool="mdnice编辑器"><p>a null dataset is a dataset where <strong>the expressions of the genes are from the same distribution as the original dataset</strong>,</p><p>and <strong>the covariate of interest (cell type in this example) does not have any effffect on the expression.</strong></p></blockquote><figure data-tool="mdnice编辑器"><img data-ratio="0.22037037037037038" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O2S9ISG4fWymj5FQN3O5sibcg5M077LO585X8NlxO7AicdYucVwbxsRUKA/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O2S9ISG4fWymj5FQN3O5sibcg5M077LO585X8NlxO7AicdYucVwbxsRUKA/640?wx_fmt=png"></figure><figure data-tool="mdnice编辑器"><img data-ratio="0.5046296296296297" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O2DRicmuFJCFDPeOFejROJlpQ4JzLCZUUbFmN4QiaibC0JnialbbOkiaESj4Q/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O2DRicmuFJCFDPeOFejROJlpQ4JzLCZUUbFmN4QiaibC0JnialbbOkiaESj4Q/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">第一种还是使用之前scDesign3创建好的</p><figure data-tool="mdnice编辑器"><img data-ratio="0.4824074074074074" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O21hqfJQ6kbfjudEhrjtcXAsicibZYAVDiaLuJIBDymsaSKzzJr1Adm1v8w/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O21hqfJQ6kbfjudEhrjtcXAsicibZYAVDiaLuJIBDymsaSKzzJr1Adm1v8w/640?wx_fmt=png"></figure><figure data-tool="mdnice编辑器"><img data-ratio="0.6284916201117319" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O2GPjgEibFFz3aMHuicicicibjTESPpmxSjChccXtjFZ0nRltuPbXSf5ibkLaQ/640?wx_fmt=png" data-type="png" data-w="1074" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O2GPjgEibFFz3aMHuicicicibjTESPpmxSjChccXtjFZ0nRltuPbXSf5ibkLaQ/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">第二种则是自己使用permutation创建null dataset</p><hr data-tool="mdnice编辑器"><p data-tool="mdnice编辑器">除了这个R包外，作者团队也开发了一个在线网页工具，方便大家使用</p><p data-tool="mdnice编辑器">https://app.superbio.ai/apps/108</p><figure data-tool="mdnice编辑器"><img data-ratio="0.4842592592592593" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O2XQeEwdUEUFUXz0DGIszbNIq0lZG30tpRqPXWta9GXabrfibmsbIdm9g/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8wQTEDJeHs5OzDoto0h4O2XQeEwdUEUFUXz0DGIszbNIq0lZG30tpRqPXWta9GXabrfibmsbIdm9g/640?wx_fmt=png"></figure></section><p><br></p><p><mp-style-type data-value="10000"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/Hp6OIv7WJsn28QczXh7mHQ",target="_blank" rel="noopener noreferrer">原文链接</a>
