---
title: "孟德尔中介分析全流程代码（一）"
date: 2023-07-10T09:39:26Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Acdemic"]
---
孟德尔中介分析全流程代码（一） by 生信菜鸟团
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><blockquote data-tool="mdnice编辑器"><p><span>上周的孟德尔随机化专题给大家介绍了中介分析的方法，这周我们趁热打铁，介绍一篇新鲜出炉的中介孟德尔随机化分析</span></p></blockquote><figure data-tool="mdnice编辑器"><img data-ratio="0.4555084745762712" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9f208iaAwk6M1Vdsb5Lgfh9YzPwgcalLibZpibsSiaJxGBn9ia9rc8U6rTcVh3F9GBtKrfZicelD5LRngg/640?wx_fmt=png" data-type="png" data-w="944" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9f208iaAwk6M1Vdsb5Lgfh9YzPwgcalLibZpibsSiaJxGBn9ia9rc8U6rTcVh3F9GBtKrfZicelD5LRngg/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器"><code><span>背景</span></code><span>：据观察，较高的教育程度与较低的阿尔茨海默病风险有关。然而，支撑这种关联的生物学机制仍不清楚。教育程度对阿尔茨海默病的保护作用可能是通过增加大脑储备来介导的。</span></p><p data-tool="mdnice编辑器"><code><span>方法</span></code><span>：我们使用双样本孟德尔随机法来探索教育程度、由核磁共振表型代表的脑结构储备和阿尔茨海默病之间的假定因果关系。我们进行了<span>单变量</span>孟德尔随机化分析，以研究(i)教育程度和阿尔茨海默病；(ii)教育程度和成像表型；以及(iii)成像表型和阿尔茨海默病之间的双向关联。<span>多变量</span>孟德尔随机化被用来评估脑结构表型是否介导了教育对阿尔茨海默病风险的影响。</span></p><p data-tool="mdnice编辑器"><code><span>结果</span></code><span>：我们的结果支持教育程度对阿尔茨海默病风险的保护性因果效应，以及教育程度与大脑宏观和微观结构之间潜在的双向因果关系。然而，我们并没有发现这些结构标志物影响阿尔茨海默病风险的证据。教育程度对阿尔茨海默病的保护作用可能是通过本研究中未包括的其他大脑储备措施，或通过其他机制来介导的。</span></p><figure data-tool="mdnice编辑器"><img data-ratio="0.3087179487179487" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9f208iaAwk6M1Vdsb5Lgfh9BNnc4CXMIjrJP0LNucHUYMJY6tfQ2R3uIC8n3ibPV7n8rVFY03LXlxg/640?wx_fmt=png" data-type="png" data-w="975" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9f208iaAwk6M1Vdsb5Lgfh9BNnc4CXMIjrJP0LNucHUYMJY6tfQ2R3uIC8n3ibPV7n8rVFY03LXlxg/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器"><span>GX代表EA的工具性变量集。参数a代表EA对成像衍生表型1, ..., N的直接因果效应，参数b代表成像衍生表型1, ..., N对阿尔茨海默病的直接因果效应。参数c'代表EA对阿尔茨海默病的直接因果效应。参数a和c'是用变量集GX估计的，参数b是用变量集GM估计的。图中省略了混杂变量。</span></p><p data-tool="mdnice编辑器"><span>大概了解文章思路后，我们来看看有没有我们最关注的东西——</span></p><blockquote data-tool="mdnice编辑器"><p><span>R code for reproducing all MR analyses can be found on https://github.com/as2970/EA_brain_AD_MR.</span></p></blockquote><p data-tool="mdnice编辑器"><span>太棒了！！！感谢文章作者~</span></p><figure data-tool="mdnice编辑器"><img data-ratio="0.4564814814814815" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9f208iaAwk6M1Vdsb5Lgfh9IBKnhQk2ibLuicNHNbKHbnZB54sEK27CRzpwpDTO6jiagvF3iaZCNbhcdQ/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9f208iaAwk6M1Vdsb5Lgfh9IBKnhQk2ibLuicNHNbKHbnZB54sEK27CRzpwpDTO6jiagvF3iaZCNbhcdQ/640?wx_fmt=png"></figure><h2 data-tool="mdnice编辑器"><span>step0 核心代码</span></h2><pre data-tool="mdnice编辑器"><code><span><span>library</span>(tidyverse)<br><span>library</span>(data.table)<br><span># library(ivpack)</span><br><span>library</span>(meta)<br><span>library</span>(devtools)<br><span>library</span>(pacman)<br><span>library</span>(TwoSampleMR)<br><span>library</span>(MRInstruments)<br><span>library</span>(ieugwasr)<br><span># install_github("phenoscanner/phenoscanner")</span><br><span>library</span>(phenoscanner)<br><span>library</span>(LDlinkR)<br><span>library</span>(mr.raps)<br><span>library</span>(MRPRESSO)<br><span>library</span>(extrafont)<br><span>library</span>(anchors)<br></span></code></pre><p data-tool="mdnice编辑器"><span>这里作者写了一个函数，如果上过</span></p><p data-tool="mdnice编辑器"><span><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247522216&amp;idx=1&amp;sn=7e8a61383489f4a2c68172c4c436fa83&amp;chksm=9b4bd713ac3c5e0525da8dc1d038854366c4ad643396d3ee67f1e8a117c3657530043dda1b8d&amp;scene=21#wechat_redirect" textvalue="生信入门&amp;数据挖掘线上直播课6月班" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">生信入门&amp;数据挖掘线上直播课6月班</a><br></span></p><p data-tool="mdnice编辑器"><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247520963&amp;idx=2&amp;sn=24b6c980fa5a5077c9541a1d297f7b08&amp;chksm=9b4bd278ac3c5b6edef814d4af7e92f74a68ec879487872e146265a8d508fc9385df759d6348&amp;scene=21#wechat_redirect" textvalue="马拉松授课的GEO数据挖掘单元" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">马拉松授课的GEO数据挖掘单元</a><br></p><p data-tool="mdnice编辑器"><span>有一定R语言基础的话，不妨尝试理解这个函数，后续的分析会大大简化⬇</span></p><pre data-tool="mdnice编辑器"><code><span><span>## MR_prep &lt;- function(exp, out) {</span><br><br>  <span># Extracting instruments</span><br>  exp_dat &lt;- read_exposure_data(<br>    filename = paste0(deparse(substitute(exp)), <span>"_exposure.txt"</span>, sep = <span>""</span>),<br>    clump = <span>FALSE</span>,<br>    sep = <span>" "</span>,<br>    snp_col = <span>"SNP"</span>,<br>    beta_col = <span>"BETA"</span>,<br>    se_col = <span>"SE"</span>,<br>    eaf_col = <span>"AF1"</span>,<br>    effect_allele_col = <span>"A1"</span>,<br>    other_allele_col = <span>"A2"</span>,<br>    pval_col = <span>"P"</span>,<br>    ncase_col = <span>"N_case"</span>,<br>    ncontrol_col = <span>"N_control"</span>,<br>    samplesize_col = <span>"N"</span>,<br>    min_pval = <span>1e-200</span>, <br>    log_pval = <span>FALSE</span>, <br>    chr_col = <span>"CHR"</span>,<br>    pos_col = <span>"POS"</span><br>    )<br>  <br>  <span># Clumping instruments</span><br>  exp_dat &lt;- exp_dat %&gt;% <br>    rename(<br>      rsid = SNP,<br>      pval = pval.exposure<br>    )<br>  <br>  exp_dat_clumped &lt;- ld_clump(<br>    dat = exp_dat,<br>    clump_kb = <span>10000</span>, <br>    clump_r2 = <span>0.001</span>, <br>    clump_p = <span>5e-8</span>,<br>    plink_bin = genetics.binaRies::get_plink_binary(),<br>    bfile = <span>""</span> <span>#path to LD reference dataset</span><br>  )<br>  <br>  exp_dat_clumped &lt;- exp_dat_clumped %&gt;% <br>    rename(<br>      SNP = rsid,<br>      pval.exposure = pval<br>    )<br><br>  <br>  <span># Printing number of IVs for exposure</span><br>  print(paste0(<span>"Number of IVs: "</span>, as.character(length(exp_dat_clumped$SNP))))<br>  <br> <br></span></code></pre><p data-tool="mdnice编辑器"><span>读取结局数据</span></p><pre data-tool="mdnice编辑器"><code><span> <span># Extracting instruments from outcome GWAS</span><br>  out_dat &lt;- read_outcome_data(<br>    filename = paste0(deparse(substitute(out)), <span>"_outcome.txt"</span>, sep = <span>""</span>), <br>    snps = exp_dat_clumped$SNP, <br>    sep = <span>" "</span>,<br>    snp_col = <span>"SNP"</span>,<br>    beta_col = <span>"BETA"</span>,<br>    se_col = <span>"SE"</span>,<br>    eaf_col = <span>"AF1"</span>,<br>    effect_allele_col = <span>"A1"</span>,<br>    other_allele_col = <span>"A2"</span>,<br>    pval_col = <span>"P"</span>,<br>    ncase_col = <span>"N_case"</span>,<br>    ncontrol_col = <span>"N_control"</span>,<br>    samplesize_col = <span>"N"</span>,<br>    min_pval = <span>1e-200</span>, <br>    log_pval = <span>FALSE</span>, <br>    chr_col = <span>"CHR"</span>,<br>    pos_col = <span>"POS"</span><br>  )<br>  <br>  <span># Identifying &amp; printing exposure instruments missing from outcome GWAS</span><br>  missing_IVs &lt;- exp_dat_clumped$SNP[!(exp_dat_clumped$SNP %<span>in</span>% out_dat$SNP)]<br>  print(paste0(<span>"Number of IVs missing from outcome GWAS: "</span>, as.character(length(missing_IVs))))<br>  print(<span>"List of IVs missing from outcome GWAS:"</span>)<br>  <span>for</span> (i <span>in</span> <span>1</span>:length(missing_IVs)) {<br>    print(paste0(missing_IVs[i]))<br>  }<br>  <br></span></code></pre><p data-tool="mdnice编辑器"><span>寻找代理SNPs</span></p><pre data-tool="mdnice编辑器"><code><span> <span># Replacing missing instruments from outcome GWAS with proxies</span><br>  <span>if</span>(length(missing_IVs) == <span>0</span>) {<br>    <br>    print(<span>"All exposure IVs found in outcome GWAS."</span>)<br>    <br>  } <span>else</span> {<br>    <br>    print(<span>"Some exposure IVs missing from outcome GWAS."</span>)<br>    out_full &lt;- fread(paste0(deparse(substitute(out)), <span>"_outcome.txt"</span>, sep = <span>""</span>))<br>    <br>    <span>for</span> (i <span>in</span> <span>1</span>:length(missing_IVs)) {<br>      <br>      proxies &lt;- LDproxy(snp = missing_IVs[i], pop = <span>"EUR"</span>, r2d = <span>"r2"</span>, token = <span>"6fb632e022ef"</span>, file = <span>FALSE</span>)<br>      proxies &lt;- proxies[proxies$R2 &gt; <span>0.8</span>, ]<br>      proxy_present = <span>FALSE</span><br>      <br>      <span>if</span>(length(proxies$RS_Number) == <span>0</span>){<br>        <br>        print(paste0(<span>"No proxy SNP available for "</span>, missing_IVs[i]))<br>        <br>      } <span>else</span> {<br>        <br>        <span>for</span> (j <span>in</span> <span>1</span>:length(proxies$RS_Number)) {<br>          <br>          proxy_present &lt;- proxies$RS_Number[j] %<span>in</span>% out_full$SNP<br>          <br>          <span>if</span> (proxy_present) {<br>            proxy_SNP = proxies$RS_Number[j]<br>            proxy_SNP_allele_1 = str_sub(proxies$Alleles[j], <span>2</span>, <span>2</span>)<br>            proxy_SNP_allele_2 = str_sub(proxies$Alleles[j], <span>4</span>, <span>4</span>)<br>            original_SNP_allele_1 = str_sub(proxies$Alleles[<span>1</span>], <span>2</span>, <span>2</span>)<br>            original_SNP_allele_2 = str_sub(proxies$Alleles[<span>1</span>], <span>4</span>, <span>4</span>)<br>            <span>break</span><br>          }<br>        }<br>      }<br>      <br></span></code></pre><pre data-tool="mdnice编辑器"><code><span> <span>if</span>(proxy_present == <span>TRUE</span>) {<br>        print(paste0(<span>"Proxy SNP found. "</span>, missing_IVs[i], <span>" replaced with "</span>, proxy_SNP))<br>        proxy_row &lt;- out_dat[<span>1</span>, ]<br>        proxy_row$SNP = missing_IVs[i]<br>        proxy_row$beta.outcome = as.numeric(out_full[out_full$SNP == proxy_SNP, <span>"BETA"</span>])<br>        proxy_row$se.outcome = as.numeric(out_full[out_full$SNP == proxy_SNP, <span>"SE"</span>])<br>        <span>if</span> (out_full[out_full$SNP == proxy_SNP, <span>"A1"</span>] == proxy_SNP_allele_1) proxy_row$effect_allele.outcome = original_SNP_allele_1<br>        <span>if</span> (out_full[out_full$SNP == proxy_SNP, <span>"A1"</span>] == proxy_SNP_allele_2) proxy_row$effect_allele.outcome = original_SNP_allele_2<br>        <span>if</span> (out_full[out_full$SNP == proxy_SNP, <span>"A2"</span>] == proxy_SNP_allele_1) proxy_row$other_allele.outcome = original_SNP_allele_1<br>        <span>if</span> (out_full[out_full$SNP == proxy_SNP, <span>"A2"</span>] == proxy_SNP_allele_2) proxy_row$other_allele.outcome = original_SNP_allele_2<br>        proxy_row$pval.outcome = as.numeric(out_full[out_full$SNP == proxy_SNP, <span>"P"</span>])<br>        proxy_row$samplesize.outcome = as.numeric(out_full[out_full$SNP == proxy_SNP, <span>"N"</span>])<br>        <span>if</span>(<span>"N_case"</span> %<span>in</span>% colnames(out_full)) proxy_row$ncase.outcome = as.numeric(out_full[out_full$SNP == proxy_SNP, <span>"N_case"</span>])<br>        <span>if</span>(<span>"N_control"</span> %<span>in</span>% colnames(out_full))proxy_row$ncontrol.outcome = as.numeric(out_full[out_full$SNP == proxy_SNP, <span>"N_control"</span>])<br>        proxy_row$chr.outcome = as.numeric(exp_dat_clumped[exp_dat_clumped$SNP == missing_IVs[i], <span>"chr.exposure"</span>])<br>        proxy_row$pos.outcome = as.numeric(exp_dat_clumped[exp_dat_clumped$SNP == missing_IVs[i], <span>"pos.exposure"</span>])<br>        <span>if</span>(<span>"AF1"</span> %<span>in</span>% colnames(out_full)) proxy_row$eaf.outcome = as.numeric(out_full[out_full$SNP == proxy_SNP, <span>"AF1"</span>])<br>        out_dat &lt;- rbind(out_dat, proxy_row)<br>      }<br>      <br>      <span>if</span>(proxy_present == <span>FALSE</span>) {<br>        print(paste0(<span>"No proxy SNP available for "</span>, missing_IVs[i], <span>" in outcome GWAS."</span>))<br>      }<br>    }<br>    <br> }<br></span></code></pre><p data-tool="mdnice编辑器"><span>接着harmonise</span></p><pre data-tool="mdnice编辑器"><code><span> <br>  <span># Harmonising exposure and outcome datasets</span><br>  dat &lt;- harmonise_data(<br>    exposure_dat = exp_dat_clumped, <br>    outcome_dat = out_dat, <br>    action = <span>2</span><br>  )<br><br><span>#</span><span>#</span> } <span>##自写函数结束</span><br></span></code></pre><p data-tool="mdnice编辑器"><span>这个函数特别长，刚开始看可能会觉得有些吃力，别着急，有问题善用</span><code><span>??</span></code><span>函数`，也欢迎大家在评论区讨论解决~</span></p><p data-tool="mdnice编辑器"><span>比如</span><code><span>read_exposure_data</span></code><span>函数，我们在准备暴露数据的时候就要考虑到这个函数需要的数据类型，这时候就可以</span><code><span>??read_exposure_data</span></code><span>看一下，主要看</span><code><span>Arguments</span></code><span>，哪些是必须的，哪些数据可以不需要。</span></p><figure data-tool="mdnice编辑器"><img data-ratio="1.2720247295208655" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9f208iaAwk6M1Vdsb5Lgfh9tKdgg0ib66jGJjfPiclyyocGIvlHiazaEiabUp8v7PgeK3LEK7V6aaTElA/640?wx_fmt=png" data-type="png" data-w="647" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9f208iaAwk6M1Vdsb5Lgfh9tKdgg0ib66jGJjfPiclyyocGIvlHiazaEiabUp8v7PgeK3LEK7V6aaTElA/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器"><code><span>??read_outcome_data</span></code><span>函数也是如此，这是在准备数据阶段需要我们提前了解的！！！凡事预而后立（小编的教训……）</span></p><h3 data-tool="mdnice编辑器"><span>小贴士</span></h3><section><span>关于eaf和maf的换算，是一个小细节，但也不能大意哦~</span></section><section><span>这里放一下参考</span></section><section><span>【<span>https://www.bilibili.com/video/BV1De4y1577E/?vd_source=e5e36ce10569a7a5d647d18bdb42e4b5</span>】</span></section><pre data-tool="mdnice编辑器"><code><span><span>for</span> (i <span>in</span> <span>1</span>:length(BRCA1_TN$eaf_col)) {<br>  <span>if</span> (BRCA1_TN$eaf_col[i]&lt;<span>0.5</span>) {BRCA1_TN$eaf_col[i]= <span>1</span> -BRCA1_TN$eaf_col[i]<br>  }<span>else</span>{BRCA1_TN$eaf_col[i] = BRCA1_TN$eaf_col[i]}<br>}<br></span></code></pre><hr data-tool="mdnice编辑器"><p data-tool="mdnice编辑器"><span>后续还有4步代码，咱们一步一步来~</span></p></section><p><mp-style-type data-value="10000"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/cOSGUeJBa48HYwUvmG-Zgg",target="_blank" rel="noopener noreferrer">原文链接</a>
