---
title: "【孟德尔随机化】代码分享：用循环代替大海捞针"
date: 2024-04-10T05:10:22Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Acdemic"]
---
【孟德尔随机化】代码分享：用循环代替大海捞针 by 生信菜鸟团
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><blockquote data-tool="mdnice编辑器"><span><span>前几期几乎都是以文献分享为主，这一期直接一点，跟大家分享一下同时跑多个变量和多个结局的代码，拿来就能用的那种~</span></span></blockquote><h2 data-tool="mdnice编辑器"><span></span><span>第一步，加载包</span><span></span></h2><pre data-tool="mdnice编辑器"><span></span><code><span># if (!require("R.utils", quietly = TRUE))  install.packages("R.utils")</span><br><span># </span><br><span># devtools::install_github("jrs95/hyprcoloc", build_opts = c("--resave-data", "--no-manual"), build_vignettes = TRUE)</span><br><span># browseVignettes("hyprcoloc")</span><br><span># install.packages("gwasvcf")</span><br><span># devtools::install_github("mrcieu/gwasglue") #needed for getting coloc data from opengwas</span><br><span># devtools::install_github("jrs95/gassocplot")</span><br><span># devtools::install_github('bar-woolf/MRSamePopTest')</span><br><span># devtools::install_github("bar-woolf/TwoStepCisMR")</span><br><span># devtools::install_github("slowkow/ggrepel")</span><br><br>rm(list = ls())<br><span># 加载包 ---------------------------------------------------------------------</span><br><span>if</span> (<span>T</span>) {<br>  <span>library</span>(data.table)<br>  <span>library</span>(coloc)<br>  <span>library</span>(TwoSampleMR)<br>  <span>library</span>(ieugwasr)<br>  <span>library</span>(ggplot2)<br>  <span>library</span>(ggrepel)<br>  <span># library(gwasglue)</span><br>  <span># library(gassocplot)</span><br>  <span># library(MRSamePopTest)</span><br>  <span># library(TwoStepCisMR)</span><br>  <span>library</span>(TwoSampleMR)<br>  <span>library</span>(dplyr)<br>  <span>library</span>(stringr)<br>  <span>library</span>(tidyverse)<br>}<br></code></pre><h2 data-tool="mdnice编辑器"><span></span><span>第二步，获取工具变量</span><span></span></h2><p data-tool="mdnice编辑器">这里我用了一篇文章的补充材料提供的暴露作为示例Phenome-wide Mendelian randomisation analysis of 378,142 cases reveals risk factors for eight common cancers | Nature Communications</p><blockquote data-tool="mdnice编辑器"><span>“</span><p>https://static-content.springer.com/esm/art%3A10.1038%2Fs41467-024-46927-z/MediaObjects/41467_2024_46927_MOESM5_ESM.xlsx</p></blockquote><pre data-tool="mdnice编辑器"><span></span><code><span>source</span>(<span>"step1_lib.R"</span>)<br></code></pre><p data-tool="mdnice编辑器">这里包括了将近4000个trait，所以我用了循环</p><pre data-tool="mdnice编辑器"><span></span><code>options(ieugwasr_api = <span>'gwas-api.mrcieu.ac.uk/'</span>)<br><br><span># 如果出现以下报错：</span><br><span># (Error in r$status_code : $ operator is invalid for atomic vectors)策略：</span><br><span># 删除R包</span><br><span># remove.packages("ieugwasr")</span><br><span># 重新安装一下</span><br><span># remotes::install_github("mrcieu/ieugwasr", force = TRUE)</span><br><span>library</span>(ieugwasr)<br><span>library</span>(doParallel)<br><span>library</span>(foreach)<br><br>Phenome &lt;- readxl::read_xlsx(<span>"./risk factors for eight common cancers.xlsx"</span>,sheet = <span>"Ovarian"</span>)<br>exp &lt;- Phenome$`Trait ID`<br><br>result_list &lt;- list()  <span># Create an empty list to store the results</span><br><br><span># 定义每次循环处理的子集大小</span><br>subset_size &lt;- <span>100</span><br><br><br><span># 计算需要的循环次数</span><br>num_iterations &lt;- ceiling(length(exp) / subset_size)<br><br><span># 遍历每个子集</span><br><span>for</span> (i <span>in</span> <span>1</span>:num_iterations) {<br>  <span># i=38</span><br>  start_index &lt;- (i - <span>1</span>) * subset_size + <span>1</span>  <span># 子集的起始索引</span><br>  end_index &lt;- min(i * subset_size, length(exp))  <span># 子集的结束索引</span><br>  <br>  subset &lt;- exp[start_index:end_index]  <span># 提取子集进行处理</span><br>  <br>  <span># 在此处进行子集处理</span><br>  <span>for</span> (j <span>in</span> subset) {<br>    print(j)<br>    exp_gwas &lt;- <span>tryCatch</span>(<br>      TwoSampleMR::extract_instruments(outcomes = j, p1 = <span>5e-08</span>, clump = <span>TRUE</span>,<br>                                       p2 = <span>5e-08</span>, r2 = <span>0.01</span>, kb = <span>10000</span>,<br>                                       access_token = ieugwasr::check_access_token(),<br>                                       force_server = <span>FALSE</span>),<br>      error = <span>function</span>(e) {<br>        message(paste(<span>"Error occurred for GWAS:"</span>, j))<br>        <span>NULL</span> <span># Return NULL when an error occurs</span><br>      }<br>    )<br>    <br>    <span>if</span> (!is.null(exp_gwas)) {<br>      result_list[[j]] &lt;- exp_gwas<br>    }<br>  }<br>  <br>}<br><br><br><span># 打印结果列表</span><br>print(names(result_list))<br>save(result_list,file = <span>"phegwas_result_list.Rdata"</span>)<br></code></pre><p data-tool="mdnice编辑器">这样就获取了多个变量的IVs并保存为一个list，后续也可以用这个数据去和多种结局一一尝试。</p><h2 data-tool="mdnice编辑器"><span></span><span>第三步，整理结局数据</span><span></span></h2><p data-tool="mdnice编辑器">这一步是针对gwas catalog的数据，当然也可以根据自己的数据来修改</p><pre data-tool="mdnice编辑器"><span></span><code><span>source</span>(<span>"step1_lib.R"</span>)<br><br>dir &lt;- <span>"./tmp"</span><br>dat_38 &lt;- list.files(dir,pattern = <span>"38.tsv.gz"</span>) <br><br>newfile &lt;- str_split(dat_38,<span>"_"</span>,simplify = <span>T</span>)[,<span>1</span>]<br><br><span># hg19</span><br><span># [1] "chromosome"              "variant_id"             </span><br><span># [3] "base_pair_location"      "effect_allele"          </span><br><span># [5] "other_allele"            "N"                      </span><br><span># [7] "effect_allele_frequency" "T"                      </span><br><span># [9] "SE_T"                    "P_noSPA"                </span><br><span># [11] "beta"                    "standard_error"         </span><br><span># [13] "p_value"                 "CONVERGE"</span><br><br><span># hg38</span><br><span># [1] "Name"                    "chromosome"             </span><br><span># [3] "base_pair_location"      "other_allele"           </span><br><span># [5] "effect_allele"           "Trait"                  </span><br><span># [7] "Cohort"                  "Model"                  </span><br><span># [9] "odds_ratio"              "ci_lower"               </span><br><span># [11] "ci_upper"                "p_value"                </span><br><span># [13] "effect_allele_frequency" "standard_error" </span><br><br>dir.create(<span>"./tmp/dat_38/"</span>)<br><br><span>for</span> (i <span>in</span> dat_38) {<br>  f= file.path(dir,i)<br>  print(f)<br>  gwas &lt;- fread(f,data.table = <span>F</span>)<br>  gwas[<span>1</span>:<span>4</span>,<span>1</span>:<span>4</span>]<br>  gwas$Effect &lt;- log(gwas$odds_ratio)<br>  setnames(gwas,old = c(<span>"chromosome"</span>,<span>"base_pair_location"</span>,<span>"other_allele"</span>,<span>"effect_allele"</span>,<span>"effect_allele_frequency"</span>),new = c(<span>"CHR"</span>,<span>"BP"</span>,<span>"A1"</span>,<span>"A2"</span>,<span>"EAF"</span>)) <br>  <br>  newfile &lt;- str_split(i,<span>"_"</span>,simplify = <span>T</span>)[,<span>1</span>]<br>  <span># 判断数据框中A1或A2列中是否包含指定字符串</span><br>  <span>if</span> (any(apply(gwas[, c(<span>"A1"</span>, <span>"A2"</span>)], <span>2</span>, <span>function</span>(col) any(col %<span>in</span>% c(<span>"A"</span>, <span>"G"</span>, <span>"C"</span>, <span>"T"</span>))))) {<br>    <span># 如果包含指定字符串，则执行相应的操作</span><br>    reformatted2 &lt;- MungeSumstats::format_sumstats(gwas,<br>                                    ref_genome = <span>"GRCh38"</span>,<br>                                    convert_ref_genome = <span>"GRCh37"</span>,<br>                                    nThread = <span>2</span>,<br>                                    save_path =paste0(<span>"./tmp/dat_38/"</span>,newfile,<span>".tsv.gz"</span>))      <span># 在这里你可以执行其他的操作</span><br>  } <span>else</span> {<br>    <span># 如果不包含指定字符串，则打印"条件不符合"</span><br>    print(paste(<span>"uncorrect A1&amp;A2:"</span>,i))<br>  }<br>  <br>}<br><br></code></pre><p data-tool="mdnice编辑器">这一步的目的是根据CHR+POS+A1|A2获取RSID，如果你的数据已经有了rsid，就跳过这一步。</p><p data-tool="mdnice编辑器">不同的数据对应不用的列名，这里因为我用MungeSumstats包以后，列名发生了变化；如果你也使用了这个包，那么列名下面的应该是对应的。如果是其他数据，在运行下面代码之前需要用colnames(dat)检查一下自己的列名，对应上去就好。</p><pre data-tool="mdnice编辑器"><span></span><code> outcome &lt;- read_outcome_data(<span>"你的文件夹"</span>, <br>                               sep = <span>"\t"</span>, <br>                               snps = exposure$SNP, <br>                               snp_col = <span>"SNP"</span>,<br>                               effect_allele_col = <span>"A2"</span>, <br>                               other_allele_col = <span>"A1"</span>,<br>                               pval_col = <span>"P"</span>,<br>                               beta_col = <span>"Effect"</span>, <br>                               se_col = <span>"SE"</span>, <br>                               eaf_col = <span>"FRQ"</span>, <br>                               chr_col = <span>"CHR"</span>)<br></code></pre><h2 data-tool="mdnice编辑器"><span></span><span>第四步，MR</span><span></span></h2><pre data-tool="mdnice编辑器"><span></span><code><span>source</span>(<span>"step1_lib.R"</span>)<br><br>load(<span>"phegwas_result_list.Rdata"</span>)<br><br>dir.create(<span>"./mr_outputs"</span>)<br><br><span>for</span> (i <span>in</span> <span>1</span>:length(result_list)) {<br>  exposure &lt;- result_list[[i]]<br>  <span>##outcome data</span><br>  dir &lt;- <span>"./tmp/dat_38/"</span><br>  out &lt;- list.files(dir,pattern = <span>"38.tsv.gz"</span>)<br>  <span>for</span> (j <span>in</span> <span>1</span>:length(out)) {<br>    outdat &lt;- out[j]<br>    print(outdat)<br>    <br>    expname &lt;- unique(exposure$id.exposure)<br>    outname &lt;- trimws(substr(outdat,<span>1</span>,<span>12</span>))<br>    <br>    outcome&lt;-<span>tryCatch</span>(<br>      outcome &lt;- read_outcome_data(file.path(dir,outdat), <br>                                   sep = <span>"\t"</span>, <br>                                   snps = exposure$SNP, <br>                                   snp_col = <span>"variant_id"</span>,<br>                                   effect_allele_col = <span>"effect_allele"</span>, <br>                                   other_allele_col = <span>"other_allele"</span>,<br>                                   pval_col = <span>"p_value"</span>,<br>                                   beta_col = <span>"beta"</span>, <br>                                   se_col = <span>"standard_error"</span>, <br>                                   eaf_col = <span>"effect_allele_frequency"</span>, <br>                                   chr_col = <span>"chromosome"</span>),<br>      error = <span>function</span>(e) {<br>        message(paste(<span>"Error occurred for outcome:"</span>, outname))<br>        <span>NULL</span>  <span># Return NULL when an error occurs</span><br>      }<br>    )<br>    <br>    <span>if</span> (!is.null(outcome)) {<br>      dat&lt;-harmonise_data(exposure,outcome)<br>      <br>      mr.methods=c(<span>"mr_wald_ratio"</span>, <span>"mr_ivw"</span>, <span>"mr_ivw_fe"</span>, <span>"mr_weighted_median"</span>, <span>"mr_weighted_mode"</span>, <span>"mr_egger_regression"</span>)<br>      mr.res &lt;- mr(dat, method_list=mr.methods)<br>      <br>      threshold &lt;- <span>0.05</span> <span>##定义p值范围</span><br>      target_column &lt;- <span>"pval"</span><br>      <span>if</span> (any(mr.res[[target_column]] &lt; threshold)) {<br>        <span># 如果存在小于阈值的元素，则执行相应的操作</span><br>        print(paste(<span>"At least one value in column"</span>, target_column, <span>"is less than"</span>, threshold))<br>        <span># 在这里你可以执行其他的操作</span><br>        ors &lt;- generate_odds_ratios(mr.res)<br>        write.csv(ors, paste0(<span>"./mr_outputs/"</span>,expname,<span>"_"</span>,outname,<span>"_mr_results.csv"</span>))<br>      }<br>    }<br>    <br>  }<br>}<br></code></pre><p data-tool="mdnice编辑器">这一步，不仅运行了mr，并且将任意一种mr方法的p值小于0.05的结果写为csv文件。</p><p data-tool="mdnice编辑器">这里用了两个for循环，目的是分析多个暴露和多个结局的相关性。如果是一对多，那就把i对应的循环拿掉；如果是多对一，那就把j对应的循环拿掉即可。遇到问题欢迎后台留言~</p></section><p><span>ps:后续可以根据阳性结果再进行敏感性分析和MVMR等等更深入的分析</span></p><p><mp-style-type data-value="10000"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/AKK-UdZJeXMXEoliMDPMhw",target="_blank" rel="noopener noreferrer">原文链接</a>
