---
title: "ATAC-seq 数据分析实操教程"
date: 2024-04-28T13:05:07Z
draft: ["false"]
tags: [
  "fetched",
  "老俊俊的生信笔记"
]
categories: ["Acdemic"]
---
ATAC-seq 数据分析实操教程 by 老俊俊的生信笔记
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com" data-mpa-powered-by="yiban.io"><section><mp-common-profile data-pluginname="mpprofile" data-id="MzkyMTI1MTYxNA==" data-headimg="http://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4usezgsqIGqjITSMggCTSoViaYeoKe2xoZr1IIvNJoztibQxibYHLDDoiabwAc6Ggws3Tvdo8EPss2nLgaVQ/0?wx_fmt=png" data-nickname="老俊俊的生信笔记" data-alias="JunJunLab" data-signature="老俊俊的生信技能和知识分享,我不是巨人,但你可以站在我的肩膀上更进一步!" data-from="0" data-is_biz_ban="0"></mp-common-profile></section><section><mp-common-clmusic data-pluginname="insertaudio" type="1" music_name="在你的身边" albumurl="http://wx.y.gtimg.cn/music/photo_new/T002R500x500M000004FKCj71O4Hhz_1.jpg" singer="盛哲" duration="262000" username="" music_source="1" is_vip="0" listenid="78221633865669056"></mp-common-clmusic></section><h2 data-tool="mdnice编辑器"><span></span><span>引言</span><span></span></h2><blockquote data-tool="mdnice编辑器"><span>❝</span><p><strong>ATAC-seq( assay for transposase-accessible chromatin using sequencing )</strong> 具体是什么我就不再叙述了,大家可以看看网上的介绍,我们这次来下载文献里的数据来实操熟悉一下具体流程。</p></blockquote><p data-tool="mdnice编辑器"><strong>不同 SEQ 的区别:</strong></p><figure data-tool="mdnice编辑器"><img data-imgfileid="100029422" data-ratio="1.109375" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j31jdODd8tEvIQWu6UDBRicKGeyUbGhwEUZpGhVqwQHcMdlZQauRAfjww/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="640" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j31jdODd8tEvIQWu6UDBRicKGeyUbGhwEUZpGhVqwQHcMdlZQauRAfjww/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器"><strong>ATAC-seq:</strong></p><figure data-tool="mdnice编辑器"><img data-imgfileid="100029423" data-ratio="0.7246861924686192" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3zjt6V6AFskxD6H64gGwJfFqLZPUv5glRYxf7sIHjtzNVfZyfCWEqVQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1195" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3zjt6V6AFskxD6H64gGwJfFqLZPUv5glRYxf7sIHjtzNVfZyfCWEqVQ/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器"><strong>流程分析步骤:</strong></p><figure data-tool="mdnice编辑器"><img data-imgfileid="100029421" data-ratio="0.6536485097636177" data-src="https://mmbiz.qpic.cn/sz_mmbiz_jpg/G5jjcE4useznokKBNJTarGiaq6WgdX9j3xyGoP0NQfFpdFu94brJK4HxExRjhPL9SoOmXvxQzhGlfORKdCJWzBw/640?wx_fmt=jpeg&amp;from=appmsg" data-type="jpeg" data-w="1946" src="https://mmbiz.qpic.cn/sz_mmbiz_jpg/G5jjcE4useznokKBNJTarGiaq6WgdX9j3xyGoP0NQfFpdFu94brJK4HxExRjhPL9SoOmXvxQzhGlfORKdCJWzBw/640?wx_fmt=jpeg&amp;from=appmsg"></figure><h2 data-tool="mdnice编辑器"><span></span><span>数据下载</span><span></span></h2><p data-tool="mdnice编辑器"><strong>GSE242671</strong> 6 个样本,CTL 对照组和 siNCoR 处理组:</p><figure data-tool="mdnice编辑器"><img data-imgfileid="100029419" data-ratio="0.4199655765920826" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3BIEUd1sMianKnPekibjuwpx0ggvUwTYEEqmkD03bkH4zwSqVnsmw1CXA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="581" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3BIEUd1sMianKnPekibjuwpx0ggvUwTYEEqmkD03bkH4zwSqVnsmw1CXA/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器"><strong>看看数据处理的方式,作为参考:</strong></p><figure data-tool="mdnice编辑器"><img data-imgfileid="100029420" data-ratio="0.7702552719200888" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3QBFxUZnjtqzPlF97dMLeicicp1fLBdReWNpKBLuVAU5TZdLxDrxALtcA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="901" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3QBFxUZnjtqzPlF97dMLeicicp1fLBdReWNpKBLuVAU5TZdLxDrxALtcA/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">我们复制 <strong>PRJNA1014001</strong> 去 <strong>SRA Explorer</strong> 下载就行了:</p><pre data-tool="mdnice编辑器"><span></span><code><span>#!/usr/bin/env bash</span><br>ascp -QT -l 300m -P33001 -i $HOME/.aspera/connect/etc/asperaweb_id_dsa.openssh era-fasp@fasp.sra.ebi.ac.uk:vol1/fastq/SRR261/<span>035</span>/SRR26124135/SRR26124135_1.fastq.gz . &amp;&amp; mv SRR26124135_1.fastq.gz SRR26124135_GSM7766625_CTL_1_Homo_sapiens_ATAC-seq_1.fastq.gz<br>ascp -QT -l 300m -P33001 -i $HOME/.aspera/connect/etc/asperaweb_id_dsa.openssh era-fasp@fasp.sra.ebi.ac.uk:vol1/fastq/SRR261/<span>035</span>/SRR26124135/SRR26124135_2.fastq.gz . &amp;&amp; mv SRR26124135_2.fastq.gz SRR26124135_GSM7766625_CTL_1_Homo_sapiens_ATAC-seq_2.fastq.gz<br>ascp -QT -l 300m -P33001 -i $HOME/.aspera/connect/etc/asperaweb_id_dsa.openssh era-fasp@fasp.sra.ebi.ac.uk:vol1/fastq/SRR261/<span>032</span>/SRR26124132/SRR26124132_1.fastq.gz . &amp;&amp; mv SRR26124132_1.fastq.gz SRR26124132_GSM7766628_siNCoR_1_Homo_sapiens_ATAC-seq_1.fastq.gz<br>ascp -QT -l 300m -P33001 -i $HOME/.aspera/connect/etc/asperaweb_id_dsa.openssh era-fasp@fasp.sra.ebi.ac.uk:vol1/fastq/SRR261/<span>032</span>/SRR26124132/SRR26124132_2.fastq.gz . &amp;&amp; mv SRR26124132_2.fastq.gz SRR26124132_GSM7766628_siNCoR_1_Homo_sapiens_ATAC-seq_2.fastq.gz<br>ascp -QT -l 300m -P33001 -i $HOME/.aspera/connect/etc/asperaweb_id_dsa.openssh era-fasp@fasp.sra.ebi.ac.uk:vol1/fastq/SRR261/<span>033</span>/SRR26124133/SRR26124133_1.fastq.gz . &amp;&amp; mv SRR26124133_1.fastq.gz SRR26124133_GSM7766627_CTL_3_Homo_sapiens_ATAC-seq_1.fastq.gz<br>ascp -QT -l 300m -P33001 -i $HOME/.aspera/connect/etc/asperaweb_id_dsa.openssh era-fasp@fasp.sra.ebi.ac.uk:vol1/fastq/SRR261/<span>033</span>/SRR26124133/SRR26124133_2.fastq.gz . &amp;&amp; mv SRR26124133_2.fastq.gz SRR26124133_GSM7766627_CTL_3_Homo_sapiens_ATAC-seq_2.fastq.gz<br>ascp -QT -l 300m -P33001 -i $HOME/.aspera/connect/etc/asperaweb_id_dsa.openssh era-fasp@fasp.sra.ebi.ac.uk:vol1/fastq/SRR261/<span>030</span>/SRR26124130/SRR26124130_1.fastq.gz . &amp;&amp; mv SRR26124130_1.fastq.gz SRR26124130_GSM7766630_siNCoR_3_Homo_sapiens_ATAC-seq_1.fastq.gz<br>ascp -QT -l 300m -P33001 -i $HOME/.aspera/connect/etc/asperaweb_id_dsa.openssh era-fasp@fasp.sra.ebi.ac.uk:vol1/fastq/SRR261/<span>030</span>/SRR26124130/SRR26124130_2.fastq.gz . &amp;&amp; mv SRR26124130_2.fastq.gz SRR26124130_GSM7766630_siNCoR_3_Homo_sapiens_ATAC-seq_2.fastq.gz<br>ascp -QT -l 300m -P33001 -i $HOME/.aspera/connect/etc/asperaweb_id_dsa.openssh era-fasp@fasp.sra.ebi.ac.uk:vol1/fastq/SRR261/<span>031</span>/SRR26124131/SRR26124131_1.fastq.gz . &amp;&amp; mv SRR26124131_1.fastq.gz SRR26124131_GSM7766629_siNCoR_2_Homo_sapiens_ATAC-seq_1.fastq.gz<br>ascp -QT -l 300m -P33001 -i $HOME/.aspera/connect/etc/asperaweb_id_dsa.openssh era-fasp@fasp.sra.ebi.ac.uk:vol1/fastq/SRR261/<span>031</span>/SRR26124131/SRR26124131_2.fastq.gz . &amp;&amp; mv SRR26124131_2.fastq.gz SRR26124131_GSM7766629_siNCoR_2_Homo_sapiens_ATAC-seq_2.fastq.gz<br>ascp -QT -l 300m -P33001 -i $HOME/.aspera/connect/etc/asperaweb_id_dsa.openssh era-fasp@fasp.sra.ebi.ac.uk:vol1/fastq/SRR261/<span>034</span>/SRR26124134/SRR26124134_1.fastq.gz . &amp;&amp; mv SRR26124134_1.fastq.gz SRR26124134_GSM7766626_CTL_2_Homo_sapiens_ATAC-seq_1.fastq.gz<br>ascp -QT -l 300m -P33001 -i $HOME/.aspera/connect/etc/asperaweb_id_dsa.openssh era-fasp@fasp.sra.ebi.ac.uk:vol1/fastq/SRR261/<span>034</span>/SRR26124134/SRR26124134_2.fastq.gz . &amp;&amp; mv SRR26124134_2.fastq.gz SRR26124134_GSM7766626_CTL_2_Homo_sapiens_ATAC-seq_2.fastq.gz<br></code></pre><p data-tool="mdnice编辑器">为了方便,我自己重新命名了一下:</p><figure data-tool="mdnice编辑器"><img data-imgfileid="100029424" data-ratio="0.7339667458432304" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3P9ibqjMaWI6KW3ZTcx118prZywmmEDB797cJtuvGYYCSsTiboLRYzdEA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="421" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3P9ibqjMaWI6KW3ZTcx118prZywmmEDB797cJtuvGYYCSsTiboLRYzdEA/640?wx_fmt=png&amp;from=appmsg"></figure><h2 data-tool="mdnice编辑器"><span></span><span>质控</span><span></span></h2><p data-tool="mdnice编辑器">用 <strong>fastqc</strong> 和 <strong>multiqc</strong> 软件,看一下序列有没有接头:</p><figure data-tool="mdnice编辑器"><img data-imgfileid="100029425" data-ratio="0.5177419354838709" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3eticWblcSgQxtAH7AuEg1cMshP3vWfgxDibhoRbAA5RMRGzyPMlweeZQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1240" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3eticWblcSgQxtAH7AuEg1cMshP3vWfgxDibhoRbAA5RMRGzyPMlweeZQ/640?wx_fmt=png&amp;from=appmsg"></figure><figure data-tool="mdnice编辑器"><img data-imgfileid="100029427" data-ratio="0.5143835616438356" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3NFEmuWPxN899LSsnGiaIaDzUtqhuIq5w3iaNpelM6CAzrWPVFRkRibpuA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1460" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3NFEmuWPxN899LSsnGiaIaDzUtqhuIq5w3iaNpelM6CAzrWPVFRkRibpuA/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">没有就直接进行比对。</p><h2 data-tool="mdnice编辑器"><span></span><span>比对及数据处理</span><span></span></h2><blockquote data-tool="mdnice编辑器"><span>❝</span><p>ATAC-seq 需要对比对后的文件进行过滤,包括 <strong>去除低质量序列</strong>, <strong>去除线粒体序列</strong>, <strong>去除 PCR 重复</strong>, <strong>去除 blacklist 区域的序列</strong>, <strong>对序列位置进行矫正,正链+4,负链-5</strong>。最终得到 <strong>shifted.bam</strong> 文件。</p></blockquote><p data-tool="mdnice编辑器"><strong>blacklist 文件下载:</strong></p><p data-tool="mdnice编辑器"><strong><em>https://github.com/Boyle-Lab/Blacklist/tree/master/lists</em></strong></p><figure data-tool="mdnice编辑器"><img data-imgfileid="100029426" data-ratio="0.8184713375796179" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j333EwfhIib8KUXbJVLXP2yACGIdbZqM585sUWlvTINibiceD8NHRAibkuWQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="942" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j333EwfhIib8KUXbJVLXP2yACGIdbZqM585sUWlvTINibiceD8NHRAibkuWQ/640?wx_fmt=png&amp;from=appmsg"></figure><pre data-tool="mdnice编辑器"><span></span><code><span>$</span><span> wget https://github.com/Boyle-Lab/Blacklist/blob/master/lists/hg19-blacklist.v2.bed.gz</span><br><span>$</span><span> gunzip hg19-blacklist.v2.bed.gz</span><br></code></pre><p data-tool="mdnice编辑器">写一个脚本,一起完成这些事情:</p><pre data-tool="mdnice编辑器"><span></span><code><span>#</span><span>!/bin/bash</span><br><br>for i in CTL_1 CTL_2 CTL_3 siNCoR_1 siNCoR_2 siNCoR_3<br>do<br> bowtie2 -p 15 -x ~/index-data/hg19_bw2/hg19 \<br>  --very-sensitive --no-mixed --no-discordant -X 2000 \<br>  -1 ../0.raw-data/${i}_1.fastq.gz \<br>  -2 ../0.raw-data/${i}_2.fastq.gz \<br>  | samtools view -bS | samtools sort -@ 6 -o ${i}.sorted.bam<br><span><br><br> #</span><span> remove mitochondrial reads and low-quality alignments</span><br> samtools view -h -q 30 ${i}.sorted.bam \<br>  | grep -v chrM \<br>  | samtools sort -o ${i}.rmchrM.bam<br><span><br><br> #</span><span> remove duplicates</span><br> sambamba markdup --remove-duplicates --nthreads 10 ${i}.rmchrM.bam ${i}.rmDup.bam<br><span><br> #</span><span> remove ENCODE blacklist regions</span><br> bedtools intersect -nonamecheck -v -abam ${i}.rmDup.bam \<br>      -b hg19-blacklist.v2.bed &gt; ${i}.tmp.bam<br><span><br> #</span><span> sort and index the bam file</span><br> samtools sort -@ 10  -o ${i}.bf.bam ${i}.tmp.bam<br> samtools index -@ 10 ${i}.bf.bam<br> rm ${i}.tmp.bam<br><span><br> #</span><span> <span>shift</span> <span>read</span> coordinates</span><br> alignmentSieve --numberOfProcessors 15 \<br>         --ATACshift \<br>         --blackListFileName hg19-blacklist.v2.bed \<br>         --bam ${i}.bf.bam -o ${i}.shifted.bam<br><span><br>  #</span><span> sort and index</span><br>  samtools sort -@ 10 -o ${i}.shift.sorted.bam ${i}.shifted.bam<br> samtools index -@ 10 ${i}.shift.sorted.bam<br><br>done<br></code></pre><h2 data-tool="mdnice编辑器"><span></span><span>bam 转 bigwig/bedgraph 可视化</span><span></span></h2><blockquote data-tool="mdnice编辑器"><span>❝</span><p>对上面最终得到的 bam 文件转换为 <strong>bigwig/bedgraph</strong> 可以在 <strong>igv</strong> 里面查看。此外还可以在全基因水平上查看在 <strong>TSS/TES</strong> 出的富集情况。</p></blockquote><pre data-tool="mdnice编辑器"><span></span><code><span>#</span><span>!/bin/bash</span><br><br>for i in CTL_1 CTL_2 CTL_3 siNCoR_1 siNCoR_2 siNCoR_3<br>do<br><span> #</span><span> bam2bigwig</span><br> bamCoverage -p 15 --outFileFormat bigwig \<br>      --binSize 50 --normalizeUsing BPM \<br>      -b ../2.map-data/${i}.shift.sorted.bam \<br>      -o ${i}.bw<br><span><br> #</span><span>bam2bedgraph</span><br> bamCoverage -p 15 --outFileFormat bedgraph \<br>                    --binSize 50 --normalizeUsing BPM \<br>                    -b ../2.map-data/${i}.shift.sorted.bam \<br>                    -o ${i}.bdg<br>done<br></code></pre><blockquote data-tool="mdnice编辑器"><span>❝</span><p>对于多个生物学重复样本,我们可以合并然后出图,这样可以不用展示很多图,尤其是样本数量多有生物学重复的情况,如果不多也可以都画出来。这里我们使用 <strong>macs3 cmbreps</strong> 来用均值合并。</p></blockquote><pre data-tool="mdnice编辑器"><span></span><code><span>#</span><span> merge replicate of bedgraph files</span><br><span>$</span><span> macs3 cmbreps -i CTL_1.bdg CTL_2.bdg CTL_3.bdg -o CTL_cb.bedGraph --method mean</span><br><span>$</span><span> macs3 cmbreps -i siNCoR_1.bdg siNCoR_2.bdg siNCoR_3.bdg -o siNCoR_cb.bedGraph --method mean</span><br><span><br>#</span><span> bedgraph to bigwig</span><br><span>$</span><span> conda install ucsc-bedgraphtobigwig</span><br><span>$</span><span> wget https://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/hg19.chrom.sizes</span><br><span><br>$</span><span> bedGraphToBigWig CTL_cb.bedGraph hg19.chrom.sizes  CTL_cb.bigwig</span><br><span>$</span><span> bedGraphToBigWig siNCoR_cb.bedGraph hg19.chrom.sizes  siNCoR_cb.bigwig</span><br></code></pre><p data-tool="mdnice编辑器"><strong>下载注释文件:</strong></p><pre data-tool="mdnice编辑器"><span></span><code><span>$</span><span> wget https://hgdownload.soe.ucsc.edu/goldenPath/hg19/bigZips/genes/hg19.ncbiRefSeq.gtf.gz</span><br><span>$</span><span> gunzip hg19.ncbiRefSeq.gtf.gz</span><br></code></pre><p data-tool="mdnice编辑器">绘图:</p><pre data-tool="mdnice编辑器"><span></span><code><span>#</span><span> =============================================================================================</span><br><span>#</span><span> heatmap plot</span><br><span>#</span><span> =============================================================================================</span><br>computeMatrix scale-regions -R ~/index-data/hg19.ncbiRefSeq.gtf \<br>                            -b 3000 -a 3000 \<br>                            --regionBodyLength 5000 \<br>                            --missingDataAsZero \<br>                            -S CTL_cb.bigwig siNCoR_cb.bigwig \<br>                            --skipZeros -p 15 \<br>                            --metagene \<br>                            -o matrix_sr.gz<br><br>computeMatrix reference-point --referencePoint TSS \<br>                              -R ~/index-data/hg19.ncbiRefSeq.gtf \<br>                              -b 3000 -a 3000 \<br>                              --missingDataAsZero \<br>                              -S CTL_cb.bigwig siNCoR_cb.bigwig \<br>                              --skipZeros -p 15 \<br>                              --metagene \<br>                              -o matrix_rp.gz<br><br>plotHeatmap -m matrix_sr.gz --colorList 'white,#0066CC' --heatmapHeight 12 --heatmapWidth 6 -o sr.pdf<br>plotHeatmap -m matrix_rp.gz --colorList 'white,#0066CC' --heatmapHeight 12 --heatmapWidth 6 -o rp.pdf<br></code></pre><p data-tool="mdnice编辑器"><strong>TSS-TES 分布:</strong></p><figure data-tool="mdnice编辑器"><img data-imgfileid="100029428" data-ratio="1.2209677419354839" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3NZoumCn0zxHy03OgxM9F6kl1lMODQ2vPiavQ5ib7QGOET9aenRwzIdLA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="620" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3NZoumCn0zxHy03OgxM9F6kl1lMODQ2vPiavQ5ib7QGOET9aenRwzIdLA/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器"><strong>TSS 分布:</strong></p><figure data-tool="mdnice编辑器"><img data-imgfileid="100029432" data-ratio="1.2195512820512822" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3NOv6N5Qag40QlqP5YxqicAFDJzewqibHnaP6pB7hZy9UicQictmG0RibxZg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="624" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3NOv6N5Qag40QlqP5YxqicAFDJzewqibHnaP6pB7hZy9UicQictmG0RibxZg/640?wx_fmt=png&amp;from=appmsg"></figure><h2 data-tool="mdnice编辑器"><span></span><span>ATAC 质控</span><span></span></h2><blockquote data-tool="mdnice编辑器"><span>❝</span><p>片段长度分布是有特点的,一般 100bp 以内位 <strong>NFR</strong> 无核小体区, 一个核小体(200bp),两个核小体(400bp), 三个核小体(600bp)区域等等。我们用 <strong>ATACseqQC</strong> 来质控。</p></blockquote><figure data-tool="mdnice编辑器"><img data-imgfileid="100029429" data-ratio="0.47832369942196534" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3KIY4zqDwh0cicIXd6jbsJKPo4UbUBMibqRxFniaZiaJK3OjPUrf0cKR0Qw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="692" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3KIY4zqDwh0cicIXd6jbsJKPo4UbUBMibqRxFniaZiaJK3OjPUrf0cKR0Qw/640?wx_fmt=png&amp;from=appmsg"></figure><pre data-tool="mdnice编辑器"><span></span><code>setwd(<span>"~/JunJun/8.atacSeq_proj/4.atacQC-data/"</span>)<br><br><span># BiocManager::install("ATACseqQC")</span><br><span>library</span>(ATACseqQC)<br><span>library</span>(ggplot2)<br><span>library</span>(dplyr)<br><br><span># bamfile files</span><br>bamfile &lt;- list.files(<span>"../2.map-data"</span>,pattern = <span>"shift.sorted.bam$"</span>,full.names = <span>T</span>)<br>bamlabel &lt;- sapply(strsplit(bamfile,split = <span>"../2.map-data/|.shift.sorted.bam"</span>),<span>"["</span>,<span>2</span>)<br><br><span># Estimate the library complexity</span><br>estimateLibComplexity(readsDupFreq(bamfile[<span>1</span>]))<br><br><span>################################################################################</span><br><span># generate fragement size distribution</span><br><span>################################################################################</span><br>x = <span>1</span><br>lapply(seq_along(bamfile),<span>function</span>(x){<br>  fragSize &lt;- fragSizeDist(bamFiles = bamfile[x], bamFiles.labels = bamlabel[x])<br><br>  <span># replot</span><br>  df &lt;- data.frame(fragSize)<br>  colnames(df) &lt;- c(<span>"x"</span>,<span>"y"</span>)<br>  df$sample &lt;- bamlabel[x]<br>  df$y &lt;- df$y/sum(df$y)<br><br>  <span>return</span>(df)<br>}) %&gt;% do.call(<span>"rbind"</span>,.) -&gt; fz_data<br><br><span># plot 6x10</span><br>ggplot(data = fz_data,aes(x = as.numeric(x),y = y*<span>10</span>^<span>3</span>)) +<br>  geom_line() +<br>  theme_bw() +<br>  theme(panel.grid = element_blank(),<br>        strip.text = element_text(face = <span>"bold.italic"</span>)) +<br>  facet_wrap(~sample,ncol = <span>3</span>,scales = <span>"free"</span>) +<br>  scale_x_continuous(limits = c(<span>0</span>,<span>1000</span>),breaks = c(<span>0</span>,<span>200</span>,<span>400</span>,<span>600</span>,<span>800</span>,<span>1000</span>)) +<br>  xlab(<span>"Fragment length (bp)"</span>) +<br>  ylab(<span>"Normalized read density(10^-3)"</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100029433" data-ratio="0.5983263598326359" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3leJ92lvf18dNunNx7WH0VYl14zqOCSDyJaJVH8B8Gos0ibyCp0bvdYw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1195" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3leJ92lvf18dNunNx7WH0VYl14zqOCSDyJaJVH8B8Gos0ibyCp0bvdYw/640?wx_fmt=png&amp;from=appmsg"></figure><h2 data-tool="mdnice编辑器"><span></span><span>PeakCalling</span><span></span></h2><blockquote data-tool="mdnice编辑器"><span>❝</span><p>一般用 <strong>macs</strong> 的比较多,但对于具体参数不同文章也不一样, 如果你使用 <strong>nomodel --shift -100 --extsize 200</strong> 参数就是寻找 Tn5 酶的切割位点, 对于双端数据,我们不用就行。在 <strong>macs3.0</strong> 版本中,作者将另外一款专门针对 ATAC-seq 的 callpeak 软件 <strong>HMMRATAC</strong> 也纳入进来了。</p></blockquote><figure data-tool="mdnice编辑器"><img data-imgfileid="100029431" data-ratio="0.611365719523373" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3VybMJO4buSGqdYpkiaIncFadpFRVjWZjPibuUxnr1v1ia5Iib45chxfsZw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1091" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3VybMJO4buSGqdYpkiaIncFadpFRVjWZjPibuUxnr1v1ia5Iib45chxfsZw/640?wx_fmt=png&amp;from=appmsg"></figure><figure data-tool="mdnice编辑器"><img data-imgfileid="100029430" data-ratio="0.39316239316239315" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3jWqV3hs3qzFHdnicwLvCenSo0uzNLJhPjSH9opa7T2znjWncKicd868Q/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="819" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3jWqV3hs3qzFHdnicwLvCenSo0uzNLJhPjSH9opa7T2znjWncKicd868Q/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">我们可以都试一下(macs3 hmmratac 慢很多):</p><pre data-tool="mdnice编辑器"><span></span><code><span>#</span><span>!/bin/bash</span><br><br>for i in CTL_1 CTL_2 CTL_3 siNCoR_1 siNCoR_2 siNCoR_3<br>do<br><span>  #</span><span> using macs3 to call peaks</span><br>  macs3 callpeak -f BAMPE -t ../2.map-data/${i}.shift.sorted.bam \<br>                 -g hs -n ${i} -B -q 0.01 --keep-dup all<br><span><br>  #</span><span> using hmmratac to call peaks</span><br>  macs3 hmmratac -i ../2.map-data/${i}.shift.sorted.bam \<br>                    -n ${i} --format BAMPE \<br>                    --outdir hmmratac_peaks<br>done<br></code></pre><h2 data-tool="mdnice编辑器"><span></span><span>差异 Peaks 分析</span><span></span></h2><blockquote data-tool="mdnice编辑器"><span>❝</span><p>差异 peak 的分析软件有多种, 基于 <strong>peak set-based</strong> 和 <strong>sliding window</strong>, 一种最原始的方法就是对 peak 区域进行计数获得 count,然后使用差异软件进行差异分析。我们使用: <strong>原始计数+DESeq2</strong>, <strong>DiffBind</strong> 和 <strong>DiffChIPL</strong> 来看看结果。</p></blockquote><p data-tool="mdnice编辑器"><strong>一些差异 peak 分析软件:</strong></p><figure data-tool="mdnice编辑器"><img data-imgfileid="100029436" data-ratio="0.3273488928785159" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3vewtSnYYHGVZUgHPiaJice9ibc2Ej9M2WcmqSZpqu23icE11qwnW2D9ACg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1671" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3vewtSnYYHGVZUgHPiaJice9ibc2Ej9M2WcmqSZpqu23icE11qwnW2D9ACg/640?wx_fmt=png&amp;from=appmsg"></figure><hr data-tool="mdnice编辑器"><p data-tool="mdnice编辑器"><strong>DiffChIPL</strong> 是一款 2022 年发表在 <strong>Bioinformatics</strong> 的差异分析软件:</p><figure data-tool="mdnice编辑器"><img data-imgfileid="100029437" data-ratio="0.36167800453514737" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3KUxxvNn9gfiaLwDgd9s2yLiaib4pMnY1WrVhg5szp9wjAXTF0BnbgeCQg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="882" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3KUxxvNn9gfiaLwDgd9s2yLiaib4pMnY1WrVhg5szp9wjAXTF0BnbgeCQg/640?wx_fmt=png&amp;from=appmsg"></figure><h3 data-tool="mdnice编辑器"><span></span><span>原始计数+DESeq2</span><span></span></h3><blockquote data-tool="mdnice编辑器"><span>❝</span><p>首先我们需要对生物学重复间的 peak 取交集,然后合并不同分组的 peaks,然后进行计数获得 count 矩阵,最后用 DESeq2 差异分析。</p></blockquote><pre data-tool="mdnice编辑器"><span></span><code><span>#</span><span> control peaks replicate overlap</span><br>bedtools intersect -a CTL_1_peaks.narrowPeak -b CTL_2_peaks.narrowPeak | bedtools intersect -a stdin -b CTL_3_peaks.narrowPeak | awk '{print $1"\t"$2"\t"$3}' &gt; CTL_overlap.bed<br><span><br>#</span><span> treat peaks replicate overlap</span><br>bedtools intersect -a siNCoR_1_peaks.narrowPeak -b siNCoR_2_peaks.narrowPeak | bedtools intersect -a stdin -b siNCoR_3_peaks.narrowPeak | awk '{print $1"\t"$2"\t"$3}' &gt; siNCoR_overlap.bed<br><span><br>#</span><span> merge control and treat peaks</span><br>cat CTL_overlap.bed siNCoR_overlap.bed | sort -k1,1 -k2,2n &gt; CTL_siNCoR.bed<br>bedtools merge -i CTL_siNCoR.bed &gt; CTL_siNCoR.merge.bed<br></code></pre><p data-tool="mdnice编辑器">计数:</p><pre data-tool="mdnice编辑器"><span></span><code><span>#</span><span> make count matrix</span><br>bedtools multicov -bams ../2.map-data/*.shift.sorted.bam -bed CTL_siNCoR.merge.bed &gt; peak_count_matrix.txt<br></code></pre><p data-tool="mdnice编辑器">差异分析:</p><pre data-tool="mdnice编辑器"><span></span><code>setwd(<span>"~/JunJun/8.atacSeq_proj/5.peak-data/"</span>)<br><br><span>library</span>(DESeq2)<br><span>library</span>(dplyr)<br><br><span># load data</span><br>raw_count &lt;- read.table(<span>'peak_count_matrix.txt'</span>)<br>raw_count$id &lt;- paste(raw_count$V1,raw_count$V2,raw_count$V3,sep = <span>"|"</span>)<br>raw_count &lt;- raw_count %&gt;% dplyr::select(-V1,-V2,-V3) %&gt;%<br>  tibble::column_to_rownames(var = <span>"id"</span>)<br><br>sample &lt;- list.files(<span>"../2.map-data/"</span>,pattern = <span>"*.shift.sorted.bam$"</span>)<br>sample_name &lt;- sapply(strsplit(sample,split = <span>".shift.sorted.bam"</span>),<span>"["</span>,<span>1</span>)<br>colnames(raw_count) &lt;- sample_name<br><br><span># save folder</span><br>dir.create(<span>'1.Diff-data/'</span>)<br><br><span># meta data</span><br>coldata &lt;- data.frame(condition = factor(rep(c(<span>'control'</span>, <span>'treat'</span>), each = <span>3</span>),<br>                                         levels = c(<span>'control'</span>, <span>'treat'</span>)))<br><span># DESeqDataSet</span><br>dds &lt;- DESeqDataSetFromMatrix(countData = raw_count,<br>                              colData = coldata,<br>                              design= ~condition)<br><br><span># diff</span><br>dds &lt;- DESeq(dds)<br><br>norm &lt;- counts(dds,normalized = <span>T</span>)<br><br><span># get results</span><br>res &lt;- results(dds, contrast = c(<span>'condition'</span>, <span>'treat'</span>, <span>'control'</span>))<br>res &lt;- cbind(norm,res)<br><br><span># filter NA</span><br>res_all &lt;- as.data.frame(res) %&gt;% filter(log2FoldChange != <span>'NA'</span>)<br><br><span># add annotation</span><br>res_all$id &lt;- rownames(res_all)<br>res_all &lt;- res_all %&gt;% separate_wider_delim(id,delim = <span>"|"</span>,names = c(<span>"chr"</span>,<span>"start"</span>,<span>"end"</span>))<br><br><span># add sig type</span><br>res_all_anno &lt;- res_all |&gt;<br>  mutate(type = case_when(log2FoldChange &gt;= <span>1</span> &amp; pvalue &lt; <span>0.05</span> ~ <span>"sigUp"</span>,<br>                          log2FoldChange &lt;= -<span>1</span> &amp; pvalue &lt; <span>0.05</span> ~ <span>"sigDown"</span>,<br>                          .default = <span>"nonSig"</span>))<br><br><span># check</span><br>table(res_all_anno$type)<br><span># nonSig sigDown   sigUp</span><br><span># 58446    5962    3449</span><br><br><span># export all diff</span><br>write.csv(res_all_anno,file = paste(<span>'1.Diff-data/peak_diffAll.csv'</span>,sep = <span>''</span>) ,row.names = <span>F</span>)<br><br>colSums(raw_count)<br><span>#   CTL_1    CTL_2    CTL_3 siNCoR_1 siNCoR_2 siNCoR_3</span><br><span># 4115852  3178135  3737711 16961289 23533595 35302961</span><br></code></pre><blockquote data-tool="mdnice编辑器"><span>❝</span><p><strong>思考:</strong> 这里有个问题, 我们对于一个组有明显富集的 peak 和一个比较弱的富集的 peak 进行计数,那么它们总的 reads 数量肯定相差很大,而 DESeq2 需要对每个样本的文库大小进行标准化,那么富集多的那组很多 peak 经过标准化则会变成下调的相反结果,事实上我看了很多确实是这样的,<strong>所以并不推荐这种做法去找差异的 peak</strong>。</p></blockquote><h3 data-tool="mdnice编辑器"><span></span><span>DiffBind 差异分析</span><span></span></h3><pre data-tool="mdnice编辑器"><span></span><code><span># BiocManager::install("DiffBind")</span><br><span>library</span>(DiffBind)<br><br><span># SampleID Tissue Factor Condition Treatment Replicate bamReads ControlID bamControl Peaks PeakCaller</span><br>meta &lt;- data.frame(SampleID = paste0(rep(c(<span>"CTL"</span>,<span>"siNCoR"</span>),each = <span>3</span>),c(<span>1</span>:<span>3</span>)),<br>                   Tissue = <span>NA</span>,<br>                   Factor = <span>"atac"</span>,<br>                   Condition = rep(c(<span>"control"</span>,<span>"treat"</span>),each = <span>3</span>),<br>                   Treatment = rep(c(<span>"control"</span>,<span>"treat"</span>),each = <span>3</span>),<br>                   Replicate = rep(c(<span>1</span>,<span>2</span>,<span>3</span>),<span>2</span>),<br>                   bamReads = list.files(<span>"../2.map-data"</span>,pattern = <span>".shift.sorted.bam$"</span>,full.names = <span>T</span>),<br>                   ControlID = <span>NA</span>,<br>                   bamControl = <span>NA</span>,<br>                   Peaks = list.files(<span>"."</span>,pattern = <span>".narrowPeak"</span>,full.names = <span>T</span>),<br>                   PeakCaller = <span>"narrow"</span>)<br><br>tamoxifen &lt;-dba(sampleSheet = meta)<br><br>plot(tamoxifen)<br><br>tamoxifen &lt;- dba.count(tamoxifen,summits = <span>100</span>)<br><br>dba.plotPCA(tamoxifen,attributes = DBA_CONDITION,label = DBA_ID)<br><span># =======================================================</span><br><span># show libsizes</span><br>info &lt;- dba.show(tamoxifen)<br>libsizes &lt;- cbind(LibReads=info$Reads, FRiP=info$FRiP,<br>                  PeakReads=round(info$Reads * info$FRiP))<br>rownames(libsizes) &lt;- info$ID<br>libsizes<br><span>#         LibReads FRiP PeakReads</span><br><span># CTL1    17653370 0.09   1588803</span><br><span># CTL2    28665764 0.05   1433288</span><br><span># CTL3    28460519 0.05   1423026</span><br><span># siNCoR1 35846252 0.18   6452325</span><br><span># siNCoR2 32999878 0.27   8909967</span><br><span># siNCoR3 42458342 0.32  13586670</span><br><br>tamoxifen &lt;- dba.normalize(tamoxifen)<br>tamoxifen &lt;- dba.contrast(tamoxifen,contrast = c(<span>"Condition"</span>,<span>"treat"</span>,<span>"control"</span>))<br>dba.show(DBA = tamoxifen,bContrasts = <span>T</span>)<br><br><span># test &lt;-dba.contrast(tamoxifen,reorderMeta = list(Condition = "control"))</span><br><span># dba.show(DBA = test,bContrasts = T)</span><br><br>tamoxifen &lt;- dba.analyze(tamoxifen)<br>diffPeaks &lt;- dba.report(tamoxifen,th = <span>1</span>) %&gt;%<br>  data.frame() %&gt;%<br>  mutate(type = case_when(Fold &gt;= <span>1</span> &amp; FDR &lt; <span>0.05</span> ~ <span>"sigUp"</span>,<br>                          Fold &lt;= -<span>1</span> &amp; FDR &lt; <span>0.05</span> ~ <span>"sigDown"</span>,<br>                          .default = <span>"nonSig"</span>))<br><br><span># check</span><br>table(diffPeaks$type)<br><span># nonSig sigDown   sigUp</span><br><span># 12928    3555   75505</span><br><br><span># export all diff</span><br>write.csv(diffPeaks,file = paste(<span>'1.Diff-data/diffbind_peak_diffAll.csv'</span>,sep = <span>''</span>) ,row.names = <span>F</span>)<br></code></pre><p data-tool="mdnice编辑器">可以看到显著上调的 peak 比下调的多了一个数量级,也和我们的 <strong>TSS heatmap profile</strong> 比较像。</p><p data-tool="mdnice编辑器"><strong>样本相关性:</strong></p><figure data-tool="mdnice编辑器"><img data-imgfileid="100029434" data-ratio="0.9947712418300654" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3OP3kRQaXkZ8r3Qwz8jzNtUPJOrQkRbwLCDznUTKDCzIQ94iahcOxTNw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="765" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3OP3kRQaXkZ8r3Qwz8jzNtUPJOrQkRbwLCDznUTKDCzIQ94iahcOxTNw/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器"><strong>PCA:</strong></p><figure data-tool="mdnice编辑器"><img data-imgfileid="100029438" data-ratio="0.6682692307692307" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3tKvicHKKzRJf7M8WlYtcxFGaDJ5LqMXIatiajDQvp7X7ibFkyT3tKnaMA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1040" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3tKvicHKKzRJf7M8WlYtcxFGaDJ5LqMXIatiajDQvp7X7ibFkyT3tKnaMA/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">注释一下差异的 peak:</p><pre data-tool="mdnice编辑器"><span></span><code><span># ==============================================================================</span><br><span># annotate for peaks</span><br><span># ==============================================================================</span><br><span>library</span>(ChIPseeker)<br><span>library</span>(TxDb.Hsapiens.UCSC.hg19.knownGene)<br><span>library</span>(org.Hs.eg.db)<br><br>sigdiff &lt;- diffPeaks %&gt;% dplyr::filter(type == <span>"sigUp"</span>) %&gt;%<br>  GRanges()<br><br>peakAnno &lt;- annotatePeak(sigdiff, tssRegion = c(-<span>3000</span>, <span>3000</span>),<br>                         TxDb = TxDb.Hsapiens.UCSC.hg19.knownGene)<br><br>peakAnno_df &lt;- data.frame(peakAnno)<br><br><span># id transformation</span><br>ids &lt;- mapIds(org.Hs.eg.db, keys = peakAnno_df$geneId, keytype = <span>"ENTREZID"</span>, column = <span>"SYMBOL"</span>)<br>peakAnno_df$GeneName &lt;- ids<br><br><span># output</span><br>write.csv(peakAnno_df, <span>'1.Diff-data/diffbind_sigPeak_anno.csv'</span>, row.names = <span>FALSE</span>)<br><br><span># plot</span><br>plotAnnoBar(peakAnno)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100029435" data-ratio="0.6049792531120332" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3HB8akDGicpdCoT01fwfia6enXYmyE9uQozx7sB4R2Wop0pbQPJMibpnFg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1205" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3HB8akDGicpdCoT01fwfia6enXYmyE9uQozx7sB4R2Wop0pbQPJMibpnFg/640?wx_fmt=png&amp;from=appmsg"></figure><hr data-tool="mdnice编辑器"><p data-tool="mdnice编辑器">输出把上下调和不变的 peak:</p><pre data-tool="mdnice编辑器"><span></span><code><span># ==============================================================================</span><br><span># output peaks with types</span><br><span># ==============================================================================</span><br>diffPeaks &lt;- read.csv(<span>"1.Diff-data/diffbind_peak_diffAll.csv"</span>)<br><br>types &lt;- c(<span>"sigUp"</span>,<span>"sigDown"</span>,<span>"nonSig"</span>)<br><br><span># x = 2</span><br>lapply(seq_along(type),<span>function</span>(x){<br>  tmp &lt;- diffPeaks %&gt;% dplyr::filter(type == types[x]) %&gt;%<br>    dplyr::select(seqnames,start,end)<br><br>  write.table(tmp,file = paste0(types[x],<span>".bed"</span>),<br>              quote = <span>F</span>,col.names = <span>F</span>,row.names = <span>F</span>,sep = <span>"\t"</span>)<br>})<br></code></pre><p data-tool="mdnice编辑器">把上下调和不变的 peak 画个 profile 图看看:</p><pre data-tool="mdnice编辑器"><span></span><code><span>#</span><span> plot diff peaks region signal</span><br>computeMatrix reference-point --referencePoint TSS \<br>                              -R ../5.peak-data/sigUp.bed ../5.peak-data/sigDown.bed ../5.peak-data/nonSig.bed \<br>                              -b 3000 -a 3000 \<br>                              --missingDataAsZero \<br>                              -S CTL_cb.bigwig siNCoR_cb.bigwig \<br>                              --skipZeros -p 15 \<br>                              -o matrix_diff.gz<br><br>plotHeatmap -m matrix_diff.gz --colorList 'white,#0066CC' --heatmapHeight 12 --heatmapWidth 6 -o diffPeaks.pdf<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100029443" data-ratio="1.4283054003724396" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3Dn7xsq6DEImdW6l3eHQwiaq5Ep9RGYviaD1pAVgvpIrFLmia56X6SBNUQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="537" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3Dn7xsq6DEImdW6l3eHQwiaq5Ep9RGYviaD1pAVgvpIrFLmia56X6SBNUQ/640?wx_fmt=png&amp;from=appmsg"></figure><h3 data-tool="mdnice编辑器"><span></span><span>DiffChIPL 差异分析</span><span></span></h3><pre data-tool="mdnice编辑器"><span></span><code>setwd(<span>"~/JunJun/8.atacSeq_proj/5.peak-data/"</span>)<br><br><span># install DiffChIPL</span><br><span># pacman::p_install_gh("yancychy/DiffChIPL")</span><br><span>library</span>(DiffChIPL)<br><br><br><span># SampleID Tissue Factor Condition Treatment Replicate bamReads ControlID bamControl Peaks PeakCaller</span><br>meta &lt;- data.frame(SampleID = paste0(rep(c(<span>"CTL"</span>,<span>"siNCoR"</span>),each = <span>3</span>),c(<span>1</span>:<span>3</span>)),<br>                   Tissue = <span>NA</span>,<br>                   Factor = <span>"atac"</span>,<br>                   Condition = rep(c(<span>"control"</span>,<span>"treat"</span>),each = <span>3</span>),<br>                   Treatment = rep(c(<span>"control"</span>,<span>"treat"</span>),each = <span>3</span>),<br>                   Replicate = rep(c(<span>1</span>,<span>2</span>,<span>3</span>),<span>2</span>),<br>                   bamReads = list.files(<span>"../2.map-data"</span>,pattern = <span>".shift.sorted.bam$"</span>,full.names = <span>T</span>),<br>                   ControlID = <span>NA</span>,<br>                   bamControl = <span>NA</span>,<br>                   Peaks = list.files(<span>"."</span>,pattern = <span>".narrowPeak"</span>,full.names = <span>T</span>),<br>                   PeakCaller = <span>"narrow"</span>)<br><br><span># output</span><br>write.csv(meta,file = <span>"DiffChIPL_meta.csv"</span>,row.names = <span>F</span>)<br><br><span># ==============================================================================</span><br><span># 1_Get read count</span><br><span># ==============================================================================</span><br><br>countL = getReadCount(inputF = <span>"DiffChIPL_meta.csv"</span>,<br>                      overlap = <span>FALSE</span>,<br>                      fragment = <span>0</span>,<br>                      removeBackground = <span>TRUE</span>)<br><br><span># ==============================================================================</span><br><span># 2_Make the design matrix and normalization</span><br><span># ==============================================================================</span><br>group= c(<span>0</span>,<span>0</span>,<span>0</span>,<span>1</span>,<span>1</span>,<span>1</span>)<br>ctrName = <span>"control"</span><br>treatName = <span>"treat"</span><br>groupName = rep(c(ctrName,treatName),each = <span>3</span>)<br>design0 &lt;- cbind(rep(<span>1</span>, <span>6</span>), group)<br>colnames(design0) &lt;- c(ctrName, treatName)<br>design0<br><br><span>#      control treat</span><br><span># [1,]       1     0</span><br><span># [2,]       1     0</span><br><span># [3,]       1     0</span><br><span># [4,]       1     1</span><br><span># [5,]       1     1</span><br><span># [6,]       1     1</span><br><br>countAll &lt;- countL$countAll<br><br><span>for</span>(i <span>in</span> <span>1</span>:ncol(countAll)){<br>  id = which(countAll[,i] &lt; <span>1</span>)<br>  countAll[id,i] = <span>0</span><br>}<br><br>cpmD = cpmNorm(countAll, libsize = countL$fd$lsIP)<br><br><span># write.csv(cpmD,file = "cpmD.csv",row.names = F)</span><br><br><span># ==============================================================================</span><br><span># 3_Do differential analysis with DiffChIPL</span><br><span># ==============================================================================</span><br>resA = DiffChIPL(cpmD, design0, group0 = group)<br><span># fitRlimm3 = resA$fitDiffL</span><br><br>rtRlimm3 &lt;- resA$resDE %&gt;%<br>  mutate(type = case_when(logFC &gt;= <span>1</span> &amp; adj.P.Val &lt; <span>0.05</span> ~ <span>"sigUp"</span>,<br>                          logFC &lt;= -<span>1</span> &amp; adj.P.Val &lt; <span>0.05</span> ~ <span>"sigDown"</span>,<br>                          .default = <span>"nonSig"</span>))<br><br><span># check</span><br>head(rtRlimm3,<span>3</span>)<br><span>#                         logFC   AveExpr         t      P.Value   adj.P.Val            B   type</span><br><span># chr1_752456_752814  0.9113821 0.6859282  1.796055 0.0724863331 0.146245882 -4.907984089 nonSig</span><br><span># chr1_756889_757238 -0.7992602 0.9601723 -1.547512 0.1217404297 0.219235256 -5.313989648 nonSig</span><br><span># chr1_762153_763224  1.1434761 2.5288585  3.644235 0.0002682122 0.001128709  0.004337593  sigUp</span><br><br>table(rtRlimm3$type)<br><span># nonSig sigDown   sigUp</span><br><span>#  93102     753   50758</span><br><br>rtRlimm3$id &lt;- rownames(rtRlimm3)<br><br><span># export all diff</span><br>write.csv(rtRlimm3,file = paste(<span>'1.Diff-data/DiffChIPL_peak_diffAll.csv'</span>,sep = <span>''</span>) ,row.names = <span>F</span>)<br></code></pre><p data-tool="mdnice编辑器">这里我们看看 <strong>DiffChIPL</strong> 和 <strong>DiffBind</strong> 找到的差异 peaks 的重叠性怎么样:</p><pre data-tool="mdnice编辑器"><span></span><code><span># ==============================================================================</span><br><span># overlap with diffbind results</span><br><span># ==============================================================================</span><br>sig_peak &lt;- rtRlimm3 %&gt;% dplyr::filter(type != <span>"nonSig"</span>) %&gt;%<br>  separate_wider_delim(id,delim = <span>"_"</span>,names = c(<span>"chr"</span>,<span>"start"</span>,<span>"end"</span>),too_many = <span>"merge"</span>) %&gt;%<br>  dplyr::select(chr,start,end) %&gt;% head(n = <span>51468</span>) %&gt;% GRanges()<br><br>diffPeaks &lt;- read.csv(<span>"1.Diff-data/diffbind_peak_diffAll.csv"</span>) %&gt;%<br>  dplyr::filter(type != <span>"nonSig"</span>) %&gt;% GRanges()<br><br><span># overlap</span><br>ChIPpeakAnno::makeVennDiagram(Peaks = list(sig_peak,diffPeaks),<br>                              NameOfPeaks = c(<span>"DiffChIPL"</span>,<span>"DiffBind"</span>),<br>                              fill = c(<span>"#D55E00"</span>, <span>"#0072B2"</span>))<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100029439" data-ratio="1" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3j7X4lcZOGfaxvHu6QQuWYV44c8MpMOrzqMcszQQ8tytyN0Hibd35ugw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="615" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3j7X4lcZOGfaxvHu6QQuWYV44c8MpMOrzqMcszQQ8tytyN0Hibd35ugw/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">可以看到大部分是一致的,但是 <strong>DiffBind</strong> 数量更多一些,可能是默认调用的 <strong>DESeq2</strong> 来寻找差异的, <strong>dba.analyze</strong> 的 <strong>method</strong> 可以换成 <strong>edgeR</strong> 都试试看。</p><h2 data-tool="mdnice编辑器"><span></span><span>footprinting 分析</span><span></span></h2><blockquote data-tool="mdnice编辑器"><span>❝</span><p>转录因子结合在 DNA 上会阻碍 Tn5 酶的切割,就会形成下面这样的小区域,成为足迹(footprint),因此我们可以查看转录因子在全基因组上的结合状态。我们使用 <strong>rbt hint</strong> 来进行转录因子分析。</p></blockquote><figure data-tool="mdnice编辑器"><img data-imgfileid="100029442" data-ratio="1.141566265060241" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3bo1s3mIDd1DPspap1iaPFEhibibDDNj2ZOZ3eKUCL2DolsPE1Dnic5eIzA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="996" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3bo1s3mIDd1DPspap1iaPFEhibibDDNj2ZOZ3eKUCL2DolsPE1Dnic5eIzA/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器"><strong>文献:</strong></p><figure data-tool="mdnice编辑器"><img data-imgfileid="100029441" data-ratio="0.4629418472063854" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j34Bn6GPyRuicGDoiaic1IJWXAFdnZRrua6gZibd3uvBeFydXQxGFc0gxDaQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="877" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j34Bn6GPyRuicGDoiaic1IJWXAFdnZRrua6gZibd3uvBeFydXQxGFc0gxDaQ/640?wx_fmt=png&amp;from=appmsg"></figure><hr data-tool="mdnice编辑器"><p data-tool="mdnice编辑器">安装:</p><pre data-tool="mdnice编辑器"><span></span><code><span>#</span><span> installing RGT</span><br><span>$</span><span> pip install cython numpy scipy</span><br><span>$</span><span> pip install RGT</span><br></code></pre><p data-tool="mdnice编辑器">配置基因组文件:</p><pre data-tool="mdnice编辑器"><span></span><code><span>#</span><span> install all the necessary human genome (hg19) data sets</span><br><span>$</span><span> <span>cd</span> ~/rgtdata</span><br><span>$</span><span> python setupGenomicData.py --hg19</span><br></code></pre><p data-tool="mdnice编辑器">我们获得一个总的 peaks 文件和 bam 文件,来处理生物学重复样本:</p><pre data-tool="mdnice编辑器"><span></span><code><span>#</span><span> using macs3 to call peaks</span><br>for i in CTL siNCoR<br>do<br>  macs3 callpeak -f BAMPE \<br>                 -t ../2.map-data/${i}_1.shift.sorted.bam \<br>                 ../2.map-data/${i}_2.shift.sorted.bam \<br>                 ../2.map-data/${i}_3.shift.sorted.bam \<br>                 -g hs -n ${i} -q 0.01 --keep-dup all<br>done<br><span><br>#</span><span> merge replicates bam files</span><br>samtools merge -@ 12 -o CTL_cb.bam \<br>                ../2.map-data/CTL_1.shift.sorted.bam \<br>                ../2.map-data/CTL_2.shift.sorted.bam \<br>                ../2.map-data/CTL_3.shift.sorted.bam<br><br>samtools merge -@ 12 -o siNCoR_cb.bam \<br>                ../2.map-data/siNCoR_1.shift.sorted.bam \<br>                ../2.map-data/siNCoR_2.shift.sorted.bam \<br>                ../2.map-data/siNCoR_3.shift.sorted.bam<br></code></pre><p data-tool="mdnice编辑器"><strong>footprinting</strong> analysis:</p><pre data-tool="mdnice编辑器"><span></span><code><span>#</span><span> call footprints</span><br>rgt-hint footprinting -h<br><br>for i in CTL siNCoR<br>do<br>  rgt-hint footprinting --atac-seq --paired-end \<br>                        --organism=hg19 \<br>                        --output-location=./ --output-prefix=${i} \<br>                        ${i}_cb.bam \<br>                        ${i}_peaks.narrowPeak<br>done<br><span><br>#</span><span> finding motifs overlapping with predicted footprints</span><br>rgt-motifanalysis matching --help<br><br>rgt-motifanalysis matching --organism=hg19 --input-files CTL.bed siNCoR.bed<br></code></pre><p data-tool="mdnice编辑器">然后进行差异转录因子分析:</p><pre data-tool="mdnice编辑器"><span></span><code><span>#</span><span> generate average ATAC-seq profiles around binding sites of particular TF</span><br>rgt-hint differential -h<br><br>rgt-hint differential --organism=hg19 \<br>                      --bc --nc 12 \<br>                      --mpbs-files=./match/CTL_mpbs.bed,./match/siNCoR_mpbs.bed \<br>                      --reads-files=CTL_cb.bam,siNCoR_cb.bam \<br>                      --conditions=CTL,siNCoR \<br>                      --output-location=./CTL_siNCoR<br></code></pre><p data-tool="mdnice编辑器"><strong>差异转录因子散点图:</strong></p><figure data-tool="mdnice编辑器"><img data-imgfileid="100029440" data-ratio="1" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j35UA7JFEuHNlrVSGyUDu0BLe4ib8QV3zVdibsZzME1tbUjbqD0HKzL2Dg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="634" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j35UA7JFEuHNlrVSGyUDu0BLe4ib8QV3zVdibsZzME1tbUjbqD0HKzL2Dg/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器"><strong>我们选择上面两个查看在全基因组的结合情况, 可以看到一个结合增强,一个减弱:</strong></p><figure data-tool="mdnice编辑器"><img data-imgfileid="100029445" data-ratio="0.7270341207349081" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3eqdFVldmYrdfxIG8OzLibT4yvucs7hW5MK2LX93FgBKPPSibBUiaDgoiaQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="762" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j3eqdFVldmYrdfxIG8OzLibT4yvucs7hW5MK2LX93FgBKPPSibBUiaDgoiaQ/640?wx_fmt=png&amp;from=appmsg"></figure><figure data-tool="mdnice编辑器"><img data-imgfileid="100029446" data-ratio="0.7450722733245729" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j35Iiat9uNnprz9PzCoTlLox0pywn8k3N1QK8Ne49LWFAEfu9cfEMka4Q/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="761" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j35Iiat9uNnprz9PzCoTlLox0pywn8k3N1QK8Ne49LWFAEfu9cfEMka4Q/640?wx_fmt=png&amp;from=appmsg"></figure><h2 data-tool="mdnice编辑器"><span></span><span>结尾</span><span></span></h2><blockquote data-tool="mdnice编辑器"><span>❝</span><p><strong>路漫漫其修远兮,吾将上下而求索。</strong></p></blockquote><hr data-tool="mdnice编辑器"><p data-tool="mdnice编辑器">欢迎加入生信交流群。加我微信我也拉你进 <strong>微信群聊</strong> <strong>老俊俊生信交流群</strong> <strong>(微信交流群需收取 20 元入群费用,一旦交费,拒不退还!(防止骗子和便于管理))</strong> 。QQ 群可免费加入, 记得进群按格式修改备注哦。</p><section data-tool="mdnice编辑器"><section><p><strong>老俊俊微信：</strong></p><figure><img data-imgfileid="100029444" data-ratio="1" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j36N2GkZo9Cjawr3vwDro6gViahiau99OicsERH6FVVnhCqoMLuoG30V3CA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="430" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j36N2GkZo9Cjawr3vwDro6gViahiau99OicsERH6FVVnhCqoMLuoG30V3CA/640?wx_fmt=png&amp;from=appmsg"></figure><figure><img data-imgfileid="100029447" data-ratio="1.3668430335097002" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j339h18DYxiaPSyA6orlSbjR7YFYrvpXckspadX0X88iamXgOpATxUibicYw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="567" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4useznokKBNJTarGiaq6WgdX9j339h18DYxiaPSyA6orlSbjR7YFYrvpXckspadX0X88iamXgOpATxUibicYw/640?wx_fmt=png&amp;from=appmsg"></figure></section><section><p><strong>知识星球：</strong></p><figure><img data-imgfileid="100029448" data-ratio="1.5896226415094339" data-src="https://mmbiz.qpic.cn/sz_mmbiz_jpg/G5jjcE4useznokKBNJTarGiaq6WgdX9j3yU6HMo5OzLwdopu81pcd4ZGBaAJfL6YsicGLtXLav8MHfI2NdicFx0BQ/640?wx_fmt=jpeg&amp;from=appmsg" data-type="jpeg" data-w="1060" src="https://mmbiz.qpic.cn/sz_mmbiz_jpg/G5jjcE4useznokKBNJTarGiaq6WgdX9j3yU6HMo5OzLwdopu81pcd4ZGBaAJfL6YsicGLtXLav8MHfI2NdicFx0BQ/640?wx_fmt=jpeg&amp;from=appmsg"></figure></section></section><hr data-tool="mdnice编辑器"><p data-tool="mdnice编辑器"><strong></strong></p><center data-tool="mdnice编辑器"><strong> 往期回顾目录</strong></center><blockquote data-tool="mdnice编辑器"><span>❝</span><p><strong></strong></p><center><strong><a href="https://mp.weixin.qq.com/s?__biz=MzkyMTI1MTYxNA==&amp;mid=2247513054&amp;idx=1&amp;sn=e87d34b1b715ff2a00628e4bb0886ba3&amp;chksm=c18489aff6f300b9ab3cdda646acb3a02914b79c3f28433b9a05722625ac65175318d0f8e7c2&amp;token=1860740043&amp;lang=zh_CN&amp;scene=21#wechat_redirect" data-linktype="2">ATAC-seq 分析方法一览</a></strong></center><strong><center><a href="https://mp.weixin.qq.com/s?__biz=MzkyMTI1MTYxNA==&amp;mid=2247513016&amp;idx=1&amp;sn=902232584345e1435aade90821238a9e&amp;chksm=c18489c9f6f300dfe8e7e436f76c891eb8234ec40a388f361f7838dc6dc61fd17ccb33f96bac&amp;token=1218322320&amp;lang=zh_CN&amp;scene=21#wechat_redirect" data-linktype="2">locuszoomr 可视化 GWAS 结果</a></center></strong><strong><center><a href="https://mp.weixin.qq.com/s?__biz=MzkyMTI1MTYxNA==&amp;mid=2247512947&amp;idx=1&amp;sn=87d8684deca7bcf68ec887e067449ce5&amp;chksm=c1848902f6f30014f25b1e24e98e8ec4edf704bf70a037d4d291d3e5c7aae7e826635926e345&amp;token=538711716&amp;lang=zh_CN&amp;scene=21#wechat_redirect" data-linktype="2">2024 了,听说你还在用 dplyr 处理数据?</a></center></strong><strong><center><a href="https://mp.weixin.qq.com/s?__biz=MzkyMTI1MTYxNA==&amp;mid=2247512936&amp;idx=1&amp;sn=53d5444bfa7e6028c169578cd6fc873d&amp;chksm=c1848919f6f3000f297af5def286cf09caad0898d6598277f610b9563116c7c4000f34d79daf&amp;token=1008140360&amp;lang=zh_CN&amp;scene=21#wechat_redirect" data-linktype="2">计算 motif 到 peak center 的距离</a></center></strong><strong><center><a href="https://mp.weixin.qq.com/s?__biz=MzkyMTI1MTYxNA==&amp;mid=2247512922&amp;idx=1&amp;sn=247004f40306ea4e555fb6e32480a42f&amp;chksm=c184892bf6f3003d132a9779834b2d4df5c3f1b52ec9c3a04041d7398f3ec155560efe6eb578&amp;token=2004442534&amp;lang=zh_CN&amp;scene=21#wechat_redirect" data-linktype="2">写个包 metaplot 绘制 m6A peak 分布</a></center></strong><strong><center><a href="https://mp.weixin.qq.com/s?__biz=MzkyMTI1MTYxNA==&amp;mid=2247512856&amp;idx=1&amp;sn=7f9fd9406d4768535450ef9e2df44890&amp;chksm=c1848969f6f3007fb1e0010b0ce49fd5738fa5fd8f4759ffea56cfbe89ffaaf29e1016ed149e&amp;token=709285857&amp;lang=zh_CN&amp;scene=21#wechat_redirect" data-linktype="2">当 ggSCvis 遇到 ggmagnify</a></center></strong><strong><center><a href="https://mp.weixin.qq.com/s?__biz=MzkyMTI1MTYxNA==&amp;mid=2247512812&amp;idx=1&amp;sn=c90eeab4ddc6aeaae7986ca163eba10e&amp;chksm=c184889df6f3018bc89c07de8b7aea1a22b7b284320dbf4280cd60ce5b4fad1a65486116ec62&amp;token=555204851&amp;lang=zh_CN&amp;scene=21#wechat_redirect" data-linktype="2">在模仿中精进数据可视化_使用ggcirclize绘制circos圈图</a></center></strong><strong><center><a href="https://mp.weixin.qq.com/s?__biz=MzkyMTI1MTYxNA==&amp;mid=2247512783&amp;idx=1&amp;sn=2535eaafd12934a3067e88f00f17d59a&amp;chksm=c18488bef6f301a8fcd1efa4dbe75d2c0d794cdedde2151528d3841e61e902d9c3ebbc2915c4&amp;token=1338300207&amp;lang=zh_CN&amp;scene=21#wechat_redirect" data-linktype="2">单细胞二维密度图一文拿捏!</a></center></strong><strong><center><a href="https://mp.weixin.qq.com/s?__biz=MzkyMTI1MTYxNA==&amp;mid=2247512748&amp;idx=1&amp;sn=6678a6672774f1a629701ea74db526fa&amp;chksm=c18488ddf6f301cb2522153dfab8b690bd26ea46e7db2fc03dcf923daef3f594b0442dadd9a6&amp;token=1867459500&amp;lang=zh_CN&amp;scene=21#wechat_redirect" data-linktype="2">coord_polar2 补充极坐标图形短板</a></center></strong><strong><center><a href="https://mp.weixin.qq.com/s?__biz=MzkyMTI1MTYxNA==&amp;mid=2247512715&amp;idx=1&amp;sn=e599a44ed4013bf5512dc5f926833495&amp;chksm=c18488faf6f301ecdcd5dd4b5324fb6b86a560ce381102c45111e7ac9498ff10648127086b19&amp;token=2102927728&amp;lang=zh_CN&amp;scene=21#wechat_redirect" data-linktype="2">ggplot 绘制热图标注特定名称</a></center></strong><p><br></p></blockquote></section><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/3lgYngY5Ui0EID-320Zasw",target="_blank" rel="noopener noreferrer">原文链接</a>
