---
title: "DoRothEA--单细胞转录因子评分"
date: 2024-04-15T13:20:16Z
draft: ["false"]
tags: [
  "fetched",
  "生信小博士"
]
categories: ["Acdemic"]
---
DoRothEA--单细胞转录因子评分 by 生信小博士
------
<div><h3 data-tool="mdnice编辑器"><span><strong><span>写在开头</span></strong></span></h3><p><span>jimmy老师在他的生信游民系列提出了很多学徒任务，仔细去看每一篇文章的动机，都是科研过程中经常会经历的。</span></p><p><span>今天我们分享的任务是来自：</span><strong><span><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247526714&amp;idx=1&amp;sn=aba39bc658288a9f607b13bfdf0a0346&amp;chksm=9b4b2981ac3ca097398092480e463c9f7ea87f72635d3bcb50e6f02af8cb2488b8a45acccaa3&amp;scene=21#wechat_redirect" textvalue="生物学功能注释三板斧" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">生物学功能注释三板斧</a></span></strong><span>。 本次任务我计划分为四个部分来写，今天写第一部分。</span></p><ul><li><p><span><span>1.DoRothEA--单细胞转录因子评分</span></span></p></li><li><p><span><span>2.PROGENy--单细胞通路活性评分</span></span></p></li><li><p><span><span>3.<span>D</span><span>oRothEA--转录组转录因子评分</span></span></span></p></li><li><p><span><span><span>4.</span><span>PROGENy--转录组通路活性评分</span></span></span></p><hr><p><br></p><p><br></p></li></ul><p><span>正文部分</span><br></p><p>通常我们都是利用基因集合去做单细胞评分，那么是否可以通过<strong>有权重的基因集</strong>来计算细胞的转录活性？<span>这就是DoRothEA开发出来的意义。</span></p><p><span><br></span></p><h2 data-tool="mdnice编辑器">DoRothEA数据集简介</h2><pre data-tool="mdnice编辑器"></pre><p><span><img data-imgfileid="100003062" data-ratio="0.6685185185185185" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SksQ3ejxMoaY6cV0G6D3fx1WoyJ4qBZm66H7ycjQTXmzRSOzlldtgya4mjC4xMLdXEdTG4lF9MHOnQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SksQ3ejxMoaY6cV0G6D3fx1WoyJ4qBZm66H7ycjQTXmzRSOzlldtgya4mjC4xMLdXEdTG4lF9MHOnQ/640?wx_fmt=png&amp;from=appmsg"></span></p><section><span><img data-imgfileid="100003060" data-ratio="0.36018518518518516" data-src="https://mmbiz.qpic.cn/sz_mmbiz_jpg/xVhD7345SksQ3ejxMoaY6cV0G6D3fx1WeGcZCzQd2BHb3TQz0mTDic1FOZicwMWCh8q1fg0wudKoyLLR5GhS9anQ/640?wx_fmt=jpeg&amp;from=appmsg" data-type="other" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_jpg/xVhD7345SksQ3ejxMoaY6cV0G6D3fx1WeGcZCzQd2BHb3TQz0mTDic1FOZicwMWCh8q1fg0wudKoyLLR5GhS9anQ/640?wx_fmt=jpeg&amp;from=appmsg"></span><span>DoRothEA是一种包含转录因子（TF）与其靶标相互作用的基因集合。这些相互作用形成了被定义为调节子（regulons）的TF及其对应靶点的集合。DoRothEA regulons 整合了来自文献、ChIP-seq峰值、TF结合位点基序以及基因表达推断等不同类型的互作证据。TF和靶标之间的互作可信度根据支持的证据数量划分为A-E五个等级，A表示最可信，E表示可信度较低。DoRothEA适用于bulk RNAseq和scRNAseq的数据。该工具可以与多种统计方法结合使用，从而生成一种功能分析工具，以推断基因表达数据中的TF活性。通过考虑TF直接转录靶标的mRNA水平，而不是TF本身的基因表达水平，来计算活性。</span></section><h2 data-tool="mdnice编辑器"><span>CollecTRI数据集简介</span></h2><pre data-tool="mdnice编辑器"></pre><p><img data-imgfileid="100003061" data-ratio="0.75" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345Skt4y6dB68KVcrmGWpkkA5yTMPy6ic6MjbrbibvEskyAs80tGj9b5OFHXfFRQLic8I5mibgcgXHSdqiajuA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345Skt4y6dB68KVcrmGWpkkA5yTMPy6ic6MjbrbibvEskyAs80tGj9b5OFHXfFRQLic8I5mibgcgXHSdqiajuA/640?wx_fmt=png&amp;from=appmsg"></p><p><span>CollecTRI 来源的调控子总结了从 12 种不同 转录因子（TF）-目标基因相互作用的数据库。该集合扩大了转录因子的覆盖范围，并与其他已知的 GRN 进行了比较。结果表明，在使用 knockTF 数据集根据基因表达数据识别受干扰的 TF 方面，该集合具有卓越的性能。</span></p><p><span>简单来说，CollecTRI 是第二代DoRothEA，对DoRothEA做了更新。</span></p><h2 data-tool="mdnice编辑器"><span>加载Regulons数据集、单细胞数据</span></h2><p><br></p><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="sql"><code><span><br></span></code><code><span>load("~/gzh/pbmc3k_final_v4.rds")</span></code><code><span><br></span></code><code><span><br></span></code><code><span>#1 加载数据集合----</span></code><code><span>## We read Dorothea Regulons for Human:</span></code><code><span>dorothea_regulon_human &lt;- get(data("dorothea_hs", package = "dorothea"))</span></code><code><span>##如果是小鼠，就用</span></code><code><span>##dorothea_regulon_mouse &lt;- get(data("dorothea_mm", package = "dorothea"))</span></code><code><span><br></span></code><code><span><br></span></code><code><span>#1 过滤数据集的置信度----</span></code><code><span>## We obtain the regulons based on interactions with confidence level A, B and C</span></code><code><span>regulon &lt;- dorothea_regulon_human %&gt;%</span></code><code><span>  dplyr::filter(confidence %in% c("A","B","C"))</span></code><code><span><br></span></code><code><span>#类似结果，作者现在更推荐collectri的结果！</span></code><code><span>net &lt;- decoupleR::get_collectri(split_complexes = T,organism='human' )</span></code><code><span>head(net)</span></code><code><span>regulon</span></code><code><span><br></span></code></pre></section><p><br></p><p><img data-galleryid="" data-imgfileid="100003059" data-ratio="0.7774798927613941" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SksQ3ejxMoaY6cV0G6D3fx1WLltmXS2gWn7nTps8zTmbQ7d2MwLaL29bshdibo38UU5ZJ0WnW4uOudQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="373" src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SksQ3ejxMoaY6cV0G6D3fx1WLltmXS2gWn7nTps8zTmbQ7d2MwLaL29bshdibo38UU5ZJ0WnW4uOudQ/640?wx_fmt=png&amp;from=appmsg"></p><p><img data-galleryid="" data-imgfileid="100003058" data-ratio="0.7362924281984334" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SksQ3ejxMoaY6cV0G6D3fx1WjYiaHOpkemib372jQLMic4n4R6pAluvrGg7TItrC0eL0Hf76ASTExR8tQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="383" src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SksQ3ejxMoaY6cV0G6D3fx1WjYiaHOpkemib372jQLMic4n4R6pAluvrGg7TItrC0eL0Hf76ASTExR8tQ/640?wx_fmt=png&amp;from=appmsg"></p><p><span>作者现在更推荐使用collectri的数据</span></p><h2 data-tool="mdnice编辑器"><span>计算转录因子活性</span></h2><pre data-tool="mdnice编辑器"></pre><p><span><em> </em></span></p><section><ul><li><li><li><li><li><li><li><li><li></ul><pre data-lang="php"><code><span>#2计算细胞的TF活性----</span></code><code><span>## We compute Viper Scores </span></code><code><span>pbmc &lt;- run_viper(pbmc, regulon,</span></code><code><span>                  options = list(method = "scale", minsize = 4, </span></code><code><span>                                 eset.filter = FALSE, cores = 4, </span></code><code><span>                                 verbose = FALSE))</span></code><code><span><br></span></code><code><span>pbmc@assays$dorothea</span></code><code><span><br></span></code></pre></section><p><br></p><p><img data-galleryid="" data-imgfileid="100003063" data-ratio="0.29529243937232524" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SksQ3ejxMoaY6cV0G6D3fx1WViam49D09ZVBLjmPROKxuuyeDhctrxhgTlzlcwMJJ69cwsHbkJl3MmQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="701" src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SksQ3ejxMoaY6cV0G6D3fx1WViam49D09ZVBLjmPROKxuuyeDhctrxhgTlzlcwMJJ69cwsHbkJl3MmQ/640?wx_fmt=png&amp;from=appmsg"></p><p><span><strong>计算出得分之后，后续就是考验个人的可视化功底了！<br></strong></span></p><h2 data-tool="mdnice编辑器"><span>可视化一</span></h2><pre data-tool="mdnice编辑器"></pre><p><span>可视化方案一</span></p><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="php"><code><span>#3 对dorothea矩阵进行降维聚类分群-----</span></code><code><span>## We compute the Nearest Neighbours to perform cluster</span></code><code><span>DefaultAssay(object = pbmc) &lt;- "dorothea"</span></code><code><span>pbmc &lt;- ScaleData(pbmc)</span></code><code><span>pbmc &lt;- RunPCA(pbmc, features = rownames(pbmc), verbose = FALSE)</span></code><code><span>pbmc &lt;- FindNeighbors(pbmc, dims = 1:10, verbose = FALSE)</span></code><code><span>pbmc &lt;- FindClusters(pbmc, resolution = 0.5, verbose = FALSE)</span></code><code><span><br></span></code><code><span>pbmc &lt;- RunUMAP(pbmc, dims = 1:10, umap.method = "uwot", metric = "cosine")</span></code><code><span><br></span></code><code><span>pbmc.markers &lt;- FindAllMarkers(pbmc, only.pos = TRUE, min.pct = 0.25, </span></code><code><span>                               logfc.threshold = 0.25, verbose = FALSE)</span></code><code><span><br></span></code><code><span>top10 &lt;- pbmc.markers %&gt;% group_by(cluster) %&gt;% top_n(10, avg_log2FC);top10</span></code><code><span>pbmc@assays$dorothea@data[1:4,1:4]</span></code><code><span>top10 =top10[top10$avg_log2FC &gt; 0.3,] </span></code><code><span>DoHeatmap(pbmc ,top10$gene,size=3,slot = 'scale.data')</span></code></pre></section><p><br></p><p><span>从热图可以看出，其实还是有一大部分TF在不同的细胞类型中特异性表达的。</span></p><p><img data-galleryid="" data-imgfileid="100003066" data-ratio="0.7675925925925926" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SksQ3ejxMoaY6cV0G6D3fx1Wv2h2ARz2JgyI4tpY0KQS8YFdByBcrNGT0eiaiabR4Ya1BREJib4NgJhWQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SksQ3ejxMoaY6cV0G6D3fx1Wv2h2ARz2JgyI4tpY0KQS8YFdByBcrNGT0eiaiabR4Ya1BREJib4NgJhWQ/640?wx_fmt=png&amp;from=appmsg"><br></p><p><span>如果想把热图画好看一点，可以参考：</span><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=Mzg2NDcxMzYwNg==&amp;mid=2247485493&amp;idx=1&amp;sn=189d4790fecdb83d17a3d6ff0d50b82b&amp;chksm=ce6466dcf913efca6f914a029fdd8de24f661d0c83b449b02cd893eeb451bb9fe9eb78e1bec0&amp;scene=21#wechat_redirect" textvalue="优雅地展示20w单细胞热图|非Doheatmap" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">优雅地展示20w单细胞热图|非Doheatmap</a></p><p><br></p><h2 data-tool="mdnice编辑器"><span>可视化方案二</span></h2><pre data-tool="mdnice编辑器"></pre><p><br></p><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="perl"><code><span><br></span></code><code><span>  d.all=pbmc</span></code><code><span>  head(d.all@meta.data)</span></code><code><span><br></span></code><code><span><br></span></code><code><span>  Idents(d.all)=d.all$cell.type</span></code><code><span>  a=AverageExpression(d.all,return.seurat = TRUE)</span></code><code><span>  #可以用 a=AggregateExpression(d.all,return.seurat = TRUE)试试看哦</span></code><code><span>  </span></code><code><span>  a$orig.ident=rownames(a@meta.data)</span></code><code><span>  head(a@meta.data);a</span></code><code><span><br></span></code><code><span>  markers=pbmc.markers ; head(markers)</span></code><code><span><br></span></code><code><span>  markers$cluster=factor(markers$cluster,levels = unique(markers$cluster) )</span></code><code><span><br></span></code><code><span>  DoHeatmap(a,draw.lines = FALSE, slot = 'scale.data',assay = 'dorothea',</span></code><code><span>            features = markers %&gt;%group_by(cluster) %&gt;%dplyr::slice_max(avg_log2FC,n = 5) %&gt;% .$gene )</span></code><code><span><br></span></code><code><span> </span></code></pre></section><p><img data-galleryid="" data-imgfileid="100003065" data-ratio="0.8194444444444444" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345Skt4y6dB68KVcrmGWpkkA5yTGnJiaUn6cOjxld3qlvlKNQN9Dkoc4ChMCcAZ2wzWNrzUgufztlZhK4w/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345Skt4y6dB68KVcrmGWpkkA5yTGnJiaUn6cOjxld3qlvlKNQN9Dkoc4ChMCcAZ2wzWNrzUgufztlZhK4w/640?wx_fmt=png&amp;from=appmsg"></p><h2 data-tool="mdnice编辑器"><span>可视化方案三</span></h2><pre data-tool="mdnice编辑器"></pre><p>上面进行可视化的TF都是通过findmarkers找到的，我们还可以根据std值找TF<br></p><p><span>制备画图所有数据<br></span></p><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="php"><code><span>#4获取Viper得分矩阵，评价细胞群的TF活性----</span></code><code><span>## We transform Viper scores, scaled by seurat, into a data frame to better </span></code><code><span>## handling the results</span></code><code><span>viper_scores_df &lt;- GetAssayData(pbmc, slot = "scale.data", </span></code><code><span>                                assay = "dorothea") %&gt;%</span></code><code><span>  data.frame(check.names = F) %&gt;%</span></code><code><span>  t();viper_scores_df[1:4,1:4]</span></code><code><span><br></span></code><code><span><br></span></code><code><span>## We create a data frame containing the cells and their clusters</span></code><code><span>CellsClusters &lt;- data.frame(cell = names(Idents(pbmc)), </span></code><code><span>                            cell_type = as.character(Idents(pbmc)),</span></code><code><span>                            check.names = F) #也可以使用其他的分类信息</span></code><code><span>CellsClusters[1:4, ]</span></code><code><span><br></span></code><code><span>## We create a data frame with the Viper score per cell and its clusters</span></code><code><span>viper_scores_clusters &lt;- viper_scores_df  %&gt;%</span></code><code><span>  data.frame() %&gt;% </span></code><code><span>  rownames_to_column("cell") %&gt;%</span></code><code><span>  gather(tf, activity, -cell) %&gt;%</span></code><code><span>  inner_join(CellsClusters);viper_scores_clusters[1:4,]</span></code><code><span><br></span></code><code><span><br></span></code><code><span>## We summarize the Viper scores by cellpopulation</span></code><code><span>summarized_viper_scores &lt;- viper_scores_clusters %&gt;% </span></code><code><span>  group_by(tf, cell_type) %&gt;%</span></code><code><span>  summarise(avg = mean(activity),</span></code><code><span>            std = sd(activity));summarized_viper_scores</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span>var(1,3  )</span></code><code><span><br></span></code><code><span>5#5细胞群间变化最大的20个TFs进行可视化----</span></code><code><span><br></span></code><code><span>## We select the 20 most variable TFs. (20*9 populations = 180)  9个细胞亚群</span></code><code><span>highly_variable_tfs &lt;- summarized_viper_scores %&gt;%</span></code><code><span>  group_by(tf) %&gt;%</span></code><code><span>  mutate(var = var(avg))  %&gt;% # 计算变异度 calculates the variance of the "avg" column. So, for each transcription factor, it computes the variance of its average score.</span></code><code><span>  ungroup() %&gt;%</span></code><code><span>  top_n(180, var) %&gt;%</span></code><code><span>  distinct(tf);highly_variable_tfs</span></code><code><span><br></span></code><code><span><br></span></code><code><span>## We prepare the data for the plot</span></code><code><span>summarized_viper_scores_df &lt;- summarized_viper_scores %&gt;%</span></code><code><span>  semi_join(highly_variable_tfs, by = "tf") %&gt;%</span></code><code><span>  dplyr::select(-std) %&gt;%   </span></code><code><span>  spread(tf, avg) %&gt;%</span></code><code><span>  data.frame(row.names = 1, check.names = FALSE) ;summarized_viper_scores_df</span></code><code><span><br></span></code><code><span><br></span></code></pre></section><p><br></p><p><img data-galleryid="" data-imgfileid="100003067" data-ratio="0.4624463519313305" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SksQ3ejxMoaY6cV0G6D3fx1WH4osGibYMmYc5yxJ1mwtIIBXPtNSdrSmg5Wxur1q0dibKxTLk7T4YnjQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="932" src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SksQ3ejxMoaY6cV0G6D3fx1WH4osGibYMmYc5yxJ1mwtIIBXPtNSdrSmg5Wxur1q0dibKxTLk7T4YnjQ/640?wx_fmt=png&amp;from=appmsg"></p><p><br></p><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="perl"><code><span>palette_length = 100</span></code><code><span>my_color = colorRampPalette(c("Darkblue", "white","red"))(palette_length)</span></code><code><span><br></span></code><code><span>my_breaks &lt;- c(seq(min(summarized_viper_scores_df), 0, </span></code><code><span>                   length.out=ceiling(palette_length/2) + 1),</span></code><code><span>               seq(max(summarized_viper_scores_df)/palette_length, </span></code><code><span>                   max(summarized_viper_scores_df), </span></code><code><span>                   length.out=floor(palette_length/2)))</span></code><code><span><br></span></code><code><span>viper_hmap &lt;- pheatmap(t(summarized_viper_scores_df),fontsize=14, </span></code><code><span>                       fontsize_row = 10, </span></code><code><span>                       color=my_color, breaks = my_breaks, </span></code><code><span>                       main = "DoRothEA (ABC)", angle_col = 45,</span></code><code><span>                       treeheight_col = 0,  border_color = NA)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code></pre></section><p><img data-galleryid="" data-imgfileid="100003064" data-ratio="0.9288025889967637" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SksQ3ejxMoaY6cV0G6D3fx1WXKgdwlywKJ5XETJGv4DF9vsic5BQ1qCHjEp4YTbHaM9b4pESN5UgtSQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="618" src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SksQ3ejxMoaY6cV0G6D3fx1WXKgdwlywKJ5XETJGv4DF9vsic5BQ1qCHjEp4YTbHaM9b4pESN5UgtSQ/640?wx_fmt=png&amp;from=appmsg"></p><h2 data-tool="mdnice编辑器"><span>可视化方案四</span></h2><pre data-tool="mdnice编辑器"></pre><p>遗憾的是在这种可视化方案中，效果似乎不太好。<br></p><p>这也可以理解，毕竟这是转录因子活性分析，看的是转录因子下游基因的表达情况，而不是转录因子本身。所以对转录因子本身进行<span>可视化</span>似乎不是一个好的选择。</p><section><ul><li><li><li><li><li><li><li><li></ul><pre data-lang="kotlin"><code><span>Idents(pbmc)='RNA'</span></code><code><span>pbmc@meta.data$GATA1= pbmc@assays$dorothea@scale.data['GATA1',] </span></code><code><span><br></span></code><code><span>FeaturePlot(pbmc,features = "GATA1",reduction = "umap" )| DimPlot(pbmc,label=T,group.by = 'cell.type')</span></code><code><span><br></span></code><code><span>FeaturePlot(pbmc,features = "PBX2",reduction = "umap" )| DimPlot(pbmc,label=T,group.by = 'cell.type')</span></code><code><span><br></span></code><code><span><br></span></code></pre></section><p><img data-galleryid="" data-imgfileid="100003068" data-ratio="0.6333333333333333" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345Sktib3gTWF4LEjGd7hrvldy78TtO0ibdFe1KSoq7fJqibJBpOiaPHMQo5FIYWxiaw4Lyyuy9YZSQZSE3zibQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345Sktib3gTWF4LEjGd7hrvldy78TtO0ibdFe1KSoq7fJqibJBpOiaPHMQo5FIYWxiaw4Lyyuy9YZSQZSE3zibQ/640?wx_fmt=png&amp;from=appmsg"></p><section><mp-common-profile data-pluginname="mpprofile" data-id="Mzg2NDcxMzYwNg==" data-headimg="http://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345Sks7JhVxUX46ZKxPSob6ptqIZgnIEnHOn5VMwCX8sN6MQy1Pq4XXFEOJ6grAmsoQugyCDKOZictDBHA/0?wx_fmt=png" data-nickname="生信小博士" data-alias="bioinformatics_Dr" data-signature="【生物信息学】R语言开始，学习生信。Seurat，单细胞测序，空间转录组。 Python，scanpy，cell2location。资料分享" data-from="2" data-is_biz_ban="0"></mp-common-profile></section><section><ul><li><li><li><li><li><li><li></ul><pre data-lang="javascript"><code><span>参考：</span></code><code><span><br></span></code><code><span>https:<span>//www.jianshu.com/p/49617173a8ad</span></span></code><code><span>https:<span>//bioconductor.org/packages/release/data/experiment/vignettes/dorothea/inst/doc/dorothea.html</span></span></code><code><span>https:<span>//github.com/saezlab/CollecTRI</span></span></code><code><span>https:<span>//github.com/saezlab/CollecTRI/blob/main/scripts/casestudy/02.2_pbmc_TFactivity.R</span></span></code><code><span><br></span></code></pre></section><p><br></p><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/XjVqpSfR-NBN3NkI_cEOGw",target="_blank" rel="noopener noreferrer">原文链接</a>
