---
title: "中国人肝癌全基因组项目部分图表重现"
date: 2024-04-21T14:31:59Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Acdemic"]
---
中国人肝癌全基因组项目部分图表重现 by 生信菜鸟团
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><h1 data-tool="mdnice编辑器"><span>项目简介</span></h1><p data-tool="mdnice编辑器">前面推文介绍过文章 Deep whole-genome analysis of 494 hepatocellularcarcinomas，详情见：<a href="https://mp.weixin.qq.com/s?__biz=MzUzMTEwODk0Ng==&amp;mid=2247520746&amp;idx=1&amp;sn=f34714a5b1050fd94e9f24b45437725e&amp;scene=21#wechat_redirect" data-linktype="2">中国人肝癌全基因组项目</a>。</p><p data-tool="mdnice编辑器">该项目包含494个肝癌病人的 WGS分析结果。作者在文章附件上传了部分数据，同时构建了网页数据库供读者使用。因对文章结果感兴趣，因此从文章附件和网页数据库：http://lifeome.net:8080/clca/#/下载了部分数据进行文章图表重现，数据包括：病人的临床信息、体细胞突变结果，突变特征、拷贝数变异、结构变异、ecDNA等。因为方法上的差异，所以重现结果无法做到和原文一致，<strong>如有差异，请以原文分析结果为准</strong>。</p><h1 data-tool="mdnice编辑器"><span>数据处理</span></h1><h2 data-tool="mdnice编辑器"><span>数据下载</span></h2><p data-tool="mdnice编辑器">这次重现数据来自于文章附件和网页数据库，无需注册登录即可直接下载，很方便：<img data-imgfileid="100037858" data-ratio="0.6787037037037037" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7gh12YU2yjjSco0yqgvTvafAn1WYJ5b6QgYQNYdiaZE1piacyC0ToN3FDQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" title="" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7gh12YU2yjjSco0yqgvTvafAn1WYJ5b6QgYQNYdiaZE1piacyC0ToN3FDQ/640?wx_fmt=png&amp;from=appmsg"> <img data-imgfileid="100037861" data-ratio="0.9555555555555556" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7gibOoriaeSEayZg0llWSgnaP5qMx2uBqJd3UvJfLeGrx7N4KnCeVGGB8w/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7gibOoriaeSEayZg0llWSgnaP5qMx2uBqJd3UvJfLeGrx7N4KnCeVGGB8w/640?wx_fmt=png&amp;from=appmsg"></p><h2 data-tool="mdnice编辑器"><span>临床信息</span></h2><p data-tool="mdnice编辑器">从数据库下载到的临床信息，有 494 个患者，相关的信息有：Province、Gender、 BCLC、 Age、 Hepatitis、 Cirrhosis/Fibrosis、 Edmondson、Smoking、 Alcohol、 Multiple、 lesions、 Recurrence、 Death，前 20位患者的临床信息如下表所示：</p><pre data-tool="mdnice编辑器"><code><span># 情况环境并载入R包</span><br>rm(list = ls())<br><span>library</span>(maftools)<br><span>library</span>(stringr)<br><span>library</span>(ggpubr)<br><span>library</span>(tidyr)<br><span>library</span>(data.table)<br><span>library</span>(pheatmap)<br><span>library</span>(ggrepel)<br><span>library</span>(ggsci)<br><span>library</span>(ggplot2)<br><span>library</span>(VennDiagram)<br><span>library</span>(ggVennDiagram)<br><br>clinical = rio::import(<span>"Cases_20240315.xlsx"</span>)<br>head(clinical,n=<span>20</span>)<br></code></pre><section data-tool="mdnice编辑器"><table><thead><tr><th><span>CaseID</span></th><th><span>Province</span></th><th><span>Gender</span></th><th><span>BCLC</span></th><th><span>Age</span></th><th><span>Hepatitis</span></th><th><span>Cirrhosis/Fibrosis</span></th><th><span>Edmondson</span></th><th><span>Smoking</span></th><th><span>Alcohol</span></th><th><span>Multiple lesions</span></th><th><span>Recurrence</span></th><th><span>Death</span></th></tr></thead><tbody><tr><td><span>CLCA_0001</span></td><td><span>Fujian</span></td><td><span>Male</span></td><td><span>A</span></td><td><span>63</span></td><td><span>HBV</span></td><td><span>Cirrhosis</span></td><td><span>Level III</span></td><td><span>No</span></td><td><span>No</span></td><td><span>No</span></td><td><span>No</span></td><td><span>No</span></td></tr><tr><td><span>CLCA_0002</span></td><td><span>Henan</span></td><td><span>Female</span></td><td><span>A</span></td><td><span>76</span></td><td><span>HBV</span></td><td><span>Cirrhosis</span></td><td><span>Level III</span></td><td><span>No</span></td><td><span>No</span></td><td><span>No</span></td><td><span>No</span></td><td><span>No</span></td></tr><tr><td><span>CLCA_0003</span></td><td><span>Jiangsu</span></td><td><span>Male</span></td><td><span>C</span></td><td><span>61</span></td><td><span>HBV</span></td><td><span>Cirrhosis</span></td><td><span>Level III</span></td><td><span>Yes</span></td><td><span>Yes</span></td><td><span>No</span></td><td><span>Yes</span></td><td><span>Not Available</span></td></tr><tr><td><span>CLCA_0004</span></td><td><span>Zhejiang</span></td><td><span>Male</span></td><td><span>A</span></td><td><span>66</span></td><td><span>HBV</span></td><td><span>Cirrhosis</span></td><td><span>Level III</span></td><td><span>Yes</span></td><td><span>Yes</span></td><td><span>No</span></td><td><span>Not Available</span></td><td><span>Not Available</span></td></tr><tr><td><span>CLCA_0005</span></td><td><span>Jiangsu</span></td><td><span>Male</span></td><td><span>B</span></td><td><span>74</span></td><td><span>HBV</span></td><td><span>Fibrosis</span></td><td><span>Level III</span></td><td><span>No</span></td><td><span>No</span></td><td><span>No</span></td><td><span>No</span></td><td><span>No</span></td></tr><tr><td><span>CLCA_0006</span></td><td><span>Jiangxi</span></td><td><span>Male</span></td><td><span>B</span></td><td><span>65</span></td><td><span>HBV</span></td><td><span>Fibrosis</span></td><td><span>Level III</span></td><td><span>No</span></td><td><span>No</span></td><td><span>No</span></td><td><span>No</span></td><td><span>No</span></td></tr><tr><td><span>CLCA_0007</span></td><td><span>Zhejiang</span></td><td><span>Male</span></td><td><span>B</span></td><td><span>68</span></td><td><span>HBV</span></td><td><span>Cirrhosis</span></td><td><span>Level II</span></td><td><span>Yes</span></td><td><span>No</span></td><td><span>No</span></td><td><span>Not Available</span></td><td><span>Not Available</span></td></tr><tr><td><span>CLCA_0008</span></td><td><span>Jiangsu</span></td><td><span>Male</span></td><td><span>C</span></td><td><span>66</span></td><td><span>HBV</span></td><td><span>Cirrhosis</span></td><td><span>Level III</span></td><td><span>Yes</span></td><td><span>Yes</span></td><td><span>Yes</span></td><td><span>Yes</span></td><td><span>Yes</span></td></tr><tr><td><span>CLCA_0009</span></td><td><span>Jiangsu</span></td><td><span>Male</span></td><td><span>B</span></td><td><span>69</span></td><td><span>HBV</span></td><td><span>Cirrhosis</span></td><td><span>Level III</span></td><td><span>Yes</span></td><td><span>Yes</span></td><td><span>Yes</span></td><td><span>No</span></td><td><span>No</span></td></tr><tr><td><span>CLCA_0010</span></td><td><span>Zhejiang</span></td><td><span>Male</span></td><td><span>0</span></td><td><span>65</span></td><td><span>HBV</span></td><td><span>Fibrosis</span></td><td><span>Level III</span></td><td><span>No</span></td><td><span>No</span></td><td><span>No</span></td><td><span>Yes</span></td><td><span>No</span></td></tr><tr><td><span>CLCA_0011</span></td><td><span>Liaoning</span></td><td><span>Male</span></td><td><span>B</span></td><td><span>64</span></td><td><span>HBV</span></td><td><span>Fibrosis</span></td><td><span>Level III</span></td><td><span>Yes</span></td><td><span>No</span></td><td><span>No</span></td><td><span>Not Available</span></td><td><span>Not Available</span></td></tr><tr><td><span>CLCA_0012</span></td><td><span>Anhui</span></td><td><span>Male</span></td><td><span>B</span></td><td><span>74</span></td><td><span>HBV</span></td><td><span>Fibrosis</span></td><td><span>Level III</span></td><td><span>No</span></td><td><span>No</span></td><td><span>No</span></td><td><span>Yes</span></td><td><span>No</span></td></tr><tr><td><span>CLCA_0013</span></td><td><span>Fujian</span></td><td><span>Male</span></td><td><span>C</span></td><td><span>57</span></td><td><span>HBV</span></td><td><span>Fibrosis</span></td><td><span>Level III</span></td><td><span>Yes</span></td><td><span>Yes</span></td><td><span>Yes</span></td><td><span>Not Available</span></td><td><span>Not Available</span></td></tr><tr><td><span>CLCA_0014</span></td><td><span>Jiangsu</span></td><td><span>Male</span></td><td><span>C</span></td><td><span>70</span></td><td><span>HBV</span></td><td><span>Fibrosis</span></td><td><span>Level III</span></td><td><span>Yes</span></td><td><span>No</span></td><td><span>No</span></td><td><span>Yes</span></td><td><span>Yes</span></td></tr><tr><td><span>CLCA_0015</span></td><td><span>Anhui</span></td><td><span>Male</span></td><td><span>C</span></td><td><span>49</span></td><td><span>HBV</span></td><td><span>Cirrhosis</span></td><td><span>Level III</span></td><td><span>Yes</span></td><td><span>No</span></td><td><span>No</span></td><td><span>Not Available</span></td><td><span>Not Available</span></td></tr><tr><td><span>CLCA_0016</span></td><td><span>Jiangsu</span></td><td><span>Male</span></td><td><span>A</span></td><td><span>47</span></td><td><span>HBV</span></td><td><span>Fibrosis</span></td><td><span>Level III</span></td><td><span>No</span></td><td><span>No</span></td><td><span>No</span></td><td><span>No</span></td><td><span>No</span></td></tr><tr><td><span>CLCA_0017</span></td><td><span>Fujian</span></td><td><span>Male</span></td><td><span>C</span></td><td><span>61</span></td><td><span>HBV</span></td><td><span>Fibrosis</span></td><td><span>Level III</span></td><td><span>No</span></td><td><span>No</span></td><td><span>Yes</span></td><td><span>Not Available</span></td><td><span>Not Available</span></td></tr><tr><td><span>CLCA_0018</span></td><td><span>Jiangsu</span></td><td><span>Male</span></td><td><span>B</span></td><td><span>60</span></td><td><span>HBV</span></td><td><span>Cirrhosis</span></td><td><span>Level III</span></td><td><span>Yes</span></td><td><span>Yes</span></td><td><span>Yes</span></td><td><span>Not Available</span></td><td><span>Not Available</span></td></tr><tr><td><span>CLCA_0019</span></td><td><span>Jiangxi</span></td><td><span>Male</span></td><td><span>B</span></td><td><span>79</span></td><td><span>HBV</span></td><td><span>Cirrhosis</span></td><td><span>Level II</span></td><td><span>No</span></td><td><span>No</span></td><td><span>No</span></td><td><span>No</span></td><td><span>No</span></td></tr><tr><td><span>CLCA_0020</span></td><td><span>Zhejiang</span></td><td><span>Male</span></td><td><span>0</span></td><td><span>56</span></td><td><span>HBV</span></td><td><span>Cirrhosis</span></td><td><span>Level III</span></td><td><span>No</span></td><td><span>No</span></td><td><span>No</span></td><td><span>Yes</span></td><td><span>No</span></td></tr></tbody></table></section><h2 data-tool="mdnice编辑器"><span>突变信息</span></h2><p data-tool="mdnice编辑器">虽然文章中提到鉴定出来的突变有 9287828个，但下载得到的突变信息Excel表格（可以简单处理为maf格式），显示的也只有283223个突变位点，这个比例约为3%。因为上传的是注释后的结果，作者WGS得到的 9287828个突变位点，有很多是落在非编码区或者未知的区域的，只有283223个约3%的突变位点可以被注释到。</p><pre data-tool="mdnice编辑器"><code>somatic = rio::import(<span>"Mutations_20240314.xlsx"</span>)<br>head(somatic,n=<span>20</span>)<br></code></pre><section data-tool="mdnice编辑器"><br><table><thead><tr><th><span>CaseID</span></th><th><span>Gene</span></th><th><span>Chr</span></th><th><span>Start</span></th><th><span>End</span></th><th><span>Strand</span></th><th><span>Classification</span></th><th><span>Type</span></th><th><span>Ref</span></th><th><span>Allele</span></th><th><span>RefReads</span></th><th><span>AlleleReads</span></th><th><span>c.HGVS</span></th><th><span>p.HGVS</span></th><th><span>transcript</span></th></tr></thead><tbody><tr><td><span>CLCA_0001</span></td><td><span>RNF223</span></td><td><span>chr1</span></td><td><span>1006750</span></td><td><span>1006750</span></td><td><br></td><td><span>3'UTR</span></td><td><span>SNP</span></td><td><span>A</span></td><td><span>G</span></td><td><span>143</span></td><td><span>18</span></td><td><span>.</span></td><td><span>.</span></td><td><span>.</span></td></tr><tr><td><span>CLCA_0001</span></td><td><span>PRKCZ</span></td><td><span>chr1</span></td><td><span>2062535</span></td><td><span>2062535</span></td><td><br></td><td><span>promoter</span></td><td><span>SNP</span></td><td><span>T</span></td><td><span>C</span></td><td><span>129</span></td><td><span>45</span></td><td><span>.</span></td><td><span>.</span></td><td><span>.</span></td></tr><tr><td><span>CLCA_0001</span></td><td><span>PRKCZ</span></td><td><span>chr1</span></td><td><span>2103770</span></td><td><span>2103770</span></td><td><br></td><td><span>nonsynonymous SNV</span></td><td><span>SNP</span></td><td><span>A</span></td><td><span>T</span></td><td><span>123</span></td><td><span>56</span></td><td><span>c.1228A&gt;T</span></td><td><span>p.T410S</span></td><td><span>NM_002744</span></td></tr><tr><td><span>CLCA_0001</span></td><td><span>LINC00982</span></td><td><span>chr1</span></td><td><span>2978977</span></td><td><span>2978977</span></td><td><br></td><td><span>lncRNA</span></td><td><span>SNP</span></td><td><span>A</span></td><td><span>T</span></td><td><span>194</span></td><td><span>19</span></td><td><span>.</span></td><td><span>.</span></td><td><span>.</span></td></tr><tr><td><span>CLCA_0001</span></td><td><span>PRDM16</span></td><td><span>chr1</span></td><td><span>3352980</span></td><td><span>3352980</span></td><td><br></td><td><span>3'UTR</span></td><td><span>SNP</span></td><td><span>G</span></td><td><span>A</span></td><td><span>161</span></td><td><span>79</span></td><td><span>.</span></td><td><span>.</span></td><td><span>.</span></td></tr><tr><td><span>CLCA_0001</span></td><td><span>LINC01134</span></td><td><span>chr1</span></td><td><span>3831499</span></td><td><span>3831499</span></td><td><br></td><td><span>lncRNA</span></td><td><span>SNP</span></td><td><span>A</span></td><td><span>T</span></td><td><span>129</span></td><td><span>56</span></td><td><span>.</span></td><td><span>.</span></td><td><span>.</span></td></tr><tr><td><span>CLCA_0001</span></td><td><span>AJAP1</span></td><td><span>chr1</span></td><td><span>4849691</span></td><td><span>4849691</span></td><td><br></td><td><span>3'UTR</span></td><td><span>SNP</span></td><td><span>A</span></td><td><span>T</span></td><td><span>178</span></td><td><span>60</span></td><td><span>.</span></td><td><span>.</span></td><td><span>.</span></td></tr><tr><td><span>CLCA_0001</span></td><td><span>CHD5</span></td><td><span>chr1</span></td><td><span>6163698</span></td><td><span>6163698</span></td><td><br></td><td><span>3'UTR</span></td><td><span>SNP</span></td><td><span>A</span></td><td><span>T</span></td><td><span>173</span></td><td><span>17</span></td><td><span>.</span></td><td><span>.</span></td><td><span>.</span></td></tr><tr><td><span>CLCA_0001</span></td><td><span>ICMT</span></td><td><span>chr1</span></td><td><span>6293565</span></td><td><span>6293565</span></td><td><br></td><td><span>nonsynonymous SNV</span></td><td><span>SNP</span></td><td><span>T</span></td><td><span>A</span></td><td><span>136</span></td><td><span>49</span></td><td><span>c.423A&gt;T</span></td><td><span>p.L141F</span></td><td><span>NM_012405</span></td></tr><tr><td><span>CLCA_0001</span></td><td><span>HES2</span></td><td><span>chr1</span></td><td><span>6472873</span></td><td><span>6472873</span></td><td><br></td><td><span>3'UTR</span></td><td><span>SNP</span></td><td><span>T</span></td><td><span>A</span></td><td><span>167</span></td><td><span>8</span></td><td><span>.</span></td><td><span>.</span></td><td><span>.</span></td></tr><tr><td><span>CLCA_0001</span></td><td><span>HES2</span></td><td><span>chr1</span></td><td><span>6476021</span></td><td><span>6476021</span></td><td><br></td><td><span>3'UTR</span></td><td><span>SNP</span></td><td><span>A</span></td><td><span>T</span></td><td><span>149</span></td><td><span>52</span></td><td><span>.</span></td><td><span>.</span></td><td><span>.</span></td></tr><tr><td><span>CLCA_0001</span></td><td><span>ESPN</span></td><td><span>chr1</span></td><td><span>6520683</span></td><td><span>6520683</span></td><td><br></td><td><span>3'UTR</span></td><td><span>SNP</span></td><td><span>C</span></td><td><span>T</span></td><td><span>103</span></td><td><span>39</span></td><td><span>.</span></td><td><span>.</span></td><td><span>.</span></td></tr><tr><td><span>CLCA_0001</span></td><td><span>LOC102725193</span></td><td><span>chr1</span></td><td><span>7449404</span></td><td><span>7449404</span></td><td><br></td><td><span>lncRNA</span></td><td><span>SNP</span></td><td><span>C</span></td><td><span>T</span></td><td><span>195</span></td><td><span>82</span></td><td><span>.</span></td><td><span>.</span></td><td><span>.</span></td></tr><tr><td><span>CLCA_0001</span></td><td><span>RERE</span></td><td><span>chr1</span></td><td><span>8418104</span></td><td><span>8418104</span></td><td><br></td><td><span>promoter</span></td><td><span>SNP</span></td><td><span>C</span></td><td><span>A</span></td><td><span>202</span></td><td><span>10</span></td><td><span>.</span></td><td><span>.</span></td><td><span>.</span></td></tr><tr><td><span>CLCA_0001</span></td><td><span>G000447</span></td><td><span>chr1</span></td><td><span>9217104</span></td><td><span>9217104</span></td><td><br></td><td><span>lncRNA</span></td><td><span>SNP</span></td><td><span>T</span></td><td><span>A</span></td><td><span>199</span></td><td><span>15</span></td><td><span>.</span></td><td><span>.</span></td><td><span>.</span></td></tr><tr><td><span>CLCA_0001</span></td><td><span>H6PD</span></td><td><span>chr1</span></td><td><span>9323798</span></td><td><span>9323798</span></td><td><br></td><td><span>nonsynonymous SNV</span></td><td><span>SNP</span></td><td><span>A</span></td><td><span>T</span></td><td><span>136</span></td><td><span>59</span></td><td><span>c.1246A&gt;T</span></td><td><span>p.R416W</span></td><td><span>NM_004285</span></td></tr><tr><td><span>CLCA_0001</span></td><td><span>H6PD</span></td><td><span>chr1</span></td><td><span>9324512</span></td><td><span>9324512</span></td><td><br></td><td><span>nonsynonymous SNV</span></td><td><span>SNP</span></td><td><span>A</span></td><td><span>T</span></td><td><span>126</span></td><td><span>57</span></td><td><span>c.1960A&gt;T</span></td><td><span>p.M654L</span></td><td><span>NM_004285</span></td></tr><tr><td><span>CLCA_0001</span></td><td><span>NMNAT1</span></td><td><span>chr1</span></td><td><span>10042751</span></td><td><span>10042751</span></td><td><br></td><td><span>stopgain</span></td><td><span>SNP</span></td><td><span>A</span></td><td><span>T</span></td><td><span>148</span></td><td><span>57</span></td><td><span>c.832A&gt;T</span></td><td><span>p.K278*</span></td><td><span>NM_022787</span></td></tr><tr><td><span>CLCA_0001</span></td><td><span>G000514</span></td><td><span>chr1</span></td><td><span>10686025</span></td><td><span>10686025</span></td><td><br></td><td><span>lncRNA</span></td><td><span>SNP</span></td><td><span>A</span></td><td><span>T</span></td><td><span>187</span></td><td><span>20</span></td><td><span>.</span></td><td><span>.</span></td><td><span>.</span></td></tr><tr><td><span>CLCA_0001</span></td><td><span>EXOSC10</span></td><td><span>chr1</span></td><td><span>11151098</span></td><td><span>11151098</span></td><td><br></td><td><span>nonsynonymous SNV</span></td><td><span>SNP</span></td><td><span>G</span></td><td><span>A</span></td><td><span>180</span></td><td><span>11</span></td><td><span>c.616C&gt;T</span></td><td><span>p.P206S</span></td><td><span>NM_001001998</span></td></tr></tbody></table></section><pre data-tool="mdnice编辑器"><code><span># 但是作者文章中方法部分没有提到注释用到的软件，查看其突变注释分类可以看到并非像 VEP 、ANNOVAR 软件注释的</span><br>table(somatic$Classification)<br><br><span>## </span><br><span>##                      3'UTR                      5'UTR </span><br><span>##                      73142                      20544 </span><br><span>##        frameshift deletion       frameshift insertion </span><br><span>##                       1971                        698 </span><br><span>##                     lncRNA                lncrna.prom </span><br><span>##                      48380                      10845 </span><br><span>##     nonframeshift deletion    nonframeshift insertion </span><br><span>##                        435                         66 </span><br><span>## nonframeshift substitution          nonsynonymous SNV </span><br><span>##                        409                      52418 </span><br><span>##                   promoter                   splicing </span><br><span>##                      67674                       2349 </span><br><span>##                  startloss                   stopgain </span><br><span>##                        158                       4001 </span><br><span>##                   stoploss </span><br><span>##                        133</span><br></code></pre><h1 data-tool="mdnice编辑器"><span>图表重现</span></h1><h2 data-tool="mdnice编辑器"><span>突变图谱</span><span></span></h2><p data-tool="mdnice编辑器">文章中的 fig 1b是体细胞突变图谱，展示的是每个患者特定基因的突变情况，患者有添加上临床信息</p><figure data-tool="mdnice编辑器"><img data-imgfileid="100037860" data-ratio="0.775" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7g40GIBRibB0ibbn74iaictvIycGA3MIwHcsdZia8q1mzpCsSre0XnicxCq47A/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7g40GIBRibB0ibbn74iaictvIycGA3MIwHcsdZia8q1mzpCsSre0XnicxCq47A/640?wx_fmt=png&amp;from=appmsg"></figure><pre data-tool="mdnice编辑器"><code><span># 简单将数据处理一下，以方便后续进行 maftools 处理及可视化</span><br>colnames(somatic) = c(<span>"Tumor_Sample_Barcode"</span>,<span>"Hugo_Symbol"</span>,<span>"Chromosome"</span>,<br>                      <span>"Start_Position"</span>,<span>"End_Position"</span>,<span>"Strand"</span>,<span>"Variant_Classification"</span>,<br>                      <span>"Variant_Type"</span>,<span>"Reference_Allele"</span>,<span>"Tumor_Seq_Allele2"</span>,<span>"RefReads"</span>,<span>"AlleleReads"</span>,  <br>                      <span>"c.HGVS"</span>,<span>"p.HGVS"</span>,<span>"transcript"</span>)<br>colnames(clinical)[<span>1</span>] = <span>"Tumor_Sample_Barcode"</span><br><span># 将临床信息和突变信息读入到 maftools中</span><br>maf = read.maf(maf = somatic,vc_nonSyn=unique(somatic$Variant_Classification),clinicalData = clinical)<br><br><span>## -Validating</span><br><span>## --Non MAF specific values in Variant_Classification column:</span><br><span>##   promoter</span><br><span>##   nonsynonymous SNV</span><br><span>##   lncRNA</span><br><span>##   stopgain</span><br><span>##   splicing</span><br><span>##   lncrna.prom</span><br><span>##   nonframeshift substitution</span><br><span>##   frameshift deletion</span><br><span>##   stoploss</span><br><span>##   frameshift insertion</span><br><span>##   startloss</span><br><span>##   nonframeshift deletion</span><br><span>##   nonframeshift insertion</span><br><span>## -Summarizing</span><br><span>## --Possible FLAGS among top ten genes:</span><br><span>##   TTN</span><br><span>## -Processing clinical data</span><br><span>## -Finished in 7.440s elapsed (49.0s cpu)</span><br><br><span># 可以从文章附件中提取到 oncogenes</span><br>onco_genes=read.table(<span>"onco_genes.txt"</span>,header = <span>F</span>)[,<span>1</span>]<br><br><span># 突变图谱可视化，添加上临床信息</span><br>oncoplot(maf,<br>         genes = onco_genes,<br>         keepGeneOrder = <span>T</span>,<br>         annotationFontSize = <span>1.2</span>,<br>         legendFontSize = <span>1.0</span>,<br>         removeNonMutated = <span>FALSE</span>,<br>         anno_height = <span>2</span>,<br>         clinicalFeatures = c(<span>"Gender"</span>,<br>                              <span>"Hepatitis"</span>,<br>                              <span>"BCLC"</span>,<br>                              <span>"Cirrhosis/Fibrosis"</span>,<br>                              <span>"Edmondson"</span>,<br>                              <span>"Multiple_lesions"</span>,<br>                              <span>"Smoking"</span>,<br>                              <span>"Alcohol"</span>,<br>                              <span>"Recurrence"</span>)<br>         )<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100037857" data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7gnc1G8jJuovskbTmhCPBcKZ7cpL5TvIQsNlJ8RFh8vHb8fmfrcX05cg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="960" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7gnc1G8jJuovskbTmhCPBcKZ7cpL5TvIQsNlJ8RFh8vHb8fmfrcX05cg/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">结果显示只有493名患者，少了一位，但这从下载到的数据就是这样，处理过程并没有改变患者数量，缺失的患者ID 是 CLCA_0209，从数据库网页下载到的表格中就缺失这个患者的突变信息</p><pre data-tool="mdnice编辑器"><code>sort(unique(somatic$Tumor_Sample_Barcode))<br><br><span>##   [1] "CLCA_0001" "CLCA_0002" "CLCA_0003" "CLCA_0004" "CLCA_0005" "CLCA_0006"</span><br><span>##   [7] "CLCA_0007" "CLCA_0008" "CLCA_0009" "CLCA_0010" "CLCA_0011" "CLCA_0012"</span><br><span>##  [13] "CLCA_0013" "CLCA_0014" "CLCA_0015" "CLCA_0016" "CLCA_0017" "CLCA_0018"</span><br><span>##  [19] "CLCA_0019" "CLCA_0020" "CLCA_0021" "CLCA_0022" "CLCA_0023" "CLCA_0024"</span><br><span>##  [25] "CLCA_0025" "CLCA_0026" "CLCA_0027" "CLCA_0028" "CLCA_0029" "CLCA_0030"</span><br><span>##  [31] "CLCA_0031" "CLCA_0032" "CLCA_0033" "CLCA_0034" "CLCA_0035" "CLCA_0036"</span><br><span>##  [37] "CLCA_0037" "CLCA_0038" "CLCA_0039" "CLCA_0040" "CLCA_0041" "CLCA_0042"</span><br><span>##  [43] "CLCA_0043" "CLCA_0044" "CLCA_0045" "CLCA_0046" "CLCA_0047" "CLCA_0048"</span><br><span>##  [49] "CLCA_0049" "CLCA_0050" "CLCA_0051" "CLCA_0052" "CLCA_0053" "CLCA_0054"</span><br><span>##  [55] "CLCA_0055" "CLCA_0056" "CLCA_0057" "CLCA_0058" "CLCA_0059" "CLCA_0060"</span><br><span>##  [61] "CLCA_0061" "CLCA_0062" "CLCA_0063" "CLCA_0064" "CLCA_0065" "CLCA_0066"</span><br><span>##  [67] "CLCA_0067" "CLCA_0068" "CLCA_0069" "CLCA_0070" "CLCA_0071" "CLCA_0072"</span><br><span>##  [73] "CLCA_0073" "CLCA_0074" "CLCA_0075" "CLCA_0076" "CLCA_0077" "CLCA_0078"</span><br><span>##  [79] "CLCA_0079" "CLCA_0080" "CLCA_0081" "CLCA_0082" "CLCA_0083" "CLCA_0084"</span><br><span>##  [85] "CLCA_0085" "CLCA_0086" "CLCA_0087" "CLCA_0088" "CLCA_0089" "CLCA_0090"</span><br><span>##  [91] "CLCA_0091" "CLCA_0092" "CLCA_0093" "CLCA_0094" "CLCA_0095" "CLCA_0096"</span><br><span>##  [97] "CLCA_0097" "CLCA_0098" "CLCA_0099" "CLCA_0100" "CLCA_0101" "CLCA_0102"</span><br><span>## [103] "CLCA_0103" "CLCA_0104" "CLCA_0105" "CLCA_0106" "CLCA_0107" "CLCA_0108"</span><br><span>## [109] "CLCA_0109" "CLCA_0110" "CLCA_0111" "CLCA_0112" "CLCA_0113" "CLCA_0114"</span><br><span>## [115] "CLCA_0115" "CLCA_0116" "CLCA_0117" "CLCA_0118" "CLCA_0119" "CLCA_0120"</span><br><span>## [121] "CLCA_0121" "CLCA_0122" "CLCA_0123" "CLCA_0124" "CLCA_0125" "CLCA_0126"</span><br><span>## [127] "CLCA_0127" "CLCA_0128" "CLCA_0129" "CLCA_0130" "CLCA_0131" "CLCA_0132"</span><br><span>## [133] "CLCA_0133" "CLCA_0134" "CLCA_0135" "CLCA_0136" "CLCA_0137" "CLCA_0138"</span><br><span>## [139] "CLCA_0139" "CLCA_0140" "CLCA_0141" "CLCA_0142" "CLCA_0143" "CLCA_0144"</span><br><span>## [145] "CLCA_0145" "CLCA_0146" "CLCA_0147" "CLCA_0148" "CLCA_0149" "CLCA_0150"</span><br><span>## [151] "CLCA_0151" "CLCA_0152" "CLCA_0153" "CLCA_0154" "CLCA_0155" "CLCA_0156"</span><br><span>## [157] "CLCA_0157" "CLCA_0158" "CLCA_0159" "CLCA_0160" "CLCA_0161" "CLCA_0162"</span><br><span>## [163] "CLCA_0163" "CLCA_0164" "CLCA_0165" "CLCA_0166" "CLCA_0167" "CLCA_0168"</span><br><span>## [169] "CLCA_0169" "CLCA_0170" "CLCA_0171" "CLCA_0172" "CLCA_0173" "CLCA_0174"</span><br><span>## [175] "CLCA_0175" "CLCA_0176" "CLCA_0177" "CLCA_0178" "CLCA_0179" "CLCA_0180"</span><br><span>## [181] "CLCA_0181" "CLCA_0182" "CLCA_0183" "CLCA_0184" "CLCA_0185" "CLCA_0186"</span><br><span>## [187] "CLCA_0187" "CLCA_0188" "CLCA_0189" "CLCA_0190" "CLCA_0191" "CLCA_0192"</span><br><span>## [193] "CLCA_0193" "CLCA_0194" "CLCA_0195" "CLCA_0196" "CLCA_0197" "CLCA_0198"</span><br><span>## [199] "CLCA_0199" "CLCA_0200" "CLCA_0201" "CLCA_0202" "CLCA_0203" "CLCA_0204"</span><br><span>## [205] "CLCA_0205" "CLCA_0206" "CLCA_0207" "CLCA_0208" "CLCA_0210" "CLCA_0211"</span><br><span>## [211] "CLCA_0212" "CLCA_0213" "CLCA_0214" "CLCA_0215" "CLCA_0216" "CLCA_0217"</span><br><span>## [217] "CLCA_0218" "CLCA_0219" "CLCA_0220" "CLCA_0221" "CLCA_0222" "CLCA_0223"</span><br><span>## [223] "CLCA_0224" "CLCA_0225" "CLCA_0226" "CLCA_0227" "CLCA_0228" "CLCA_0229"</span><br><span>## [229] "CLCA_0230" "CLCA_0231" "CLCA_0232" "CLCA_0233" "CLCA_0234" "CLCA_0235"</span><br><span>## [235] "CLCA_0236" "CLCA_0237" "CLCA_0238" "CLCA_0239" "CLCA_0240" "CLCA_0241"</span><br><span>## [241] "CLCA_0242" "CLCA_0243" "CLCA_0244" "CLCA_0245" "CLCA_0246" "CLCA_0247"</span><br><span>## [247] "CLCA_0248" "CLCA_0249" "CLCA_0250" "CLCA_0251" "CLCA_0252" "CLCA_0253"</span><br><span>## [253] "CLCA_0254" "CLCA_0255" "CLCA_0256" "CLCA_0257" "CLCA_0258" "CLCA_0259"</span><br><span>## [259] "CLCA_0260" "CLCA_0261" "CLCA_0262" "CLCA_0263" "CLCA_0264" "CLCA_0265"</span><br><span>## [265] "CLCA_0266" "CLCA_0267" "CLCA_0268" "CLCA_0269" "CLCA_0270" "CLCA_0271"</span><br><span>## [271] "CLCA_0272" "CLCA_0273" "CLCA_0274" "CLCA_0275" "CLCA_0276" "CLCA_0277"</span><br><span>## [277] "CLCA_0278" "CLCA_0279" "CLCA_0280" "CLCA_0281" "CLCA_0282" "CLCA_0283"</span><br><span>## [283] "CLCA_0284" "CLCA_0285" "CLCA_0286" "CLCA_0287" "CLCA_0288" "CLCA_0289"</span><br><span>## [289] "CLCA_0290" "CLCA_0291" "CLCA_0292" "CLCA_0293" "CLCA_0294" "CLCA_0295"</span><br><span>## [295] "CLCA_0296" "CLCA_0297" "CLCA_0298" "CLCA_0299" "CLCA_0300" "CLCA_0301"</span><br><span>## [301] "CLCA_0302" "CLCA_0303" "CLCA_0304" "CLCA_0305" "CLCA_0306" "CLCA_0307"</span><br><span>## [307] "CLCA_0308" "CLCA_0309" "CLCA_0310" "CLCA_0311" "CLCA_0312" "CLCA_0313"</span><br><span>## [313] "CLCA_0314" "CLCA_0315" "CLCA_0316" "CLCA_0317" "CLCA_0318" "CLCA_0319"</span><br><span>## [319] "CLCA_0320" "CLCA_0321" "CLCA_0322" "CLCA_0323" "CLCA_0324" "CLCA_0325"</span><br><span>## [325] "CLCA_0326" "CLCA_0327" "CLCA_0328" "CLCA_0329" "CLCA_0330" "CLCA_0331"</span><br><span>## [331] "CLCA_0332" "CLCA_0333" "CLCA_0334" "CLCA_0335" "CLCA_0336" "CLCA_0337"</span><br><span>## [337] "CLCA_0338" "CLCA_0339" "CLCA_0340" "CLCA_0341" "CLCA_0342" "CLCA_0343"</span><br><span>## [343] "CLCA_0344" "CLCA_0345" "CLCA_0346" "CLCA_0347" "CLCA_0348" "CLCA_0349"</span><br><span>## [349] "CLCA_0350" "CLCA_0351" "CLCA_0352" "CLCA_0353" "CLCA_0354" "CLCA_0355"</span><br><span>## [355] "CLCA_0356" "CLCA_0357" "CLCA_0358" "CLCA_0359" "CLCA_0360" "CLCA_0361"</span><br><span>## [361] "CLCA_0362" "CLCA_0363" "CLCA_0364" "CLCA_0365" "CLCA_0366" "CLCA_0367"</span><br><span>## [367] "CLCA_0368" "CLCA_0369" "CLCA_0370" "CLCA_0371" "CLCA_0372" "CLCA_0373"</span><br><span>## [373] "CLCA_0374" "CLCA_0375" "CLCA_0376" "CLCA_0377" "CLCA_0378" "CLCA_0379"</span><br><span>## [379] "CLCA_0380" "CLCA_0381" "CLCA_0382" "CLCA_0383" "CLCA_0384" "CLCA_0385"</span><br><span>## [385] "CLCA_0386" "CLCA_0387" "CLCA_0388" "CLCA_0389" "CLCA_0390" "CLCA_0391"</span><br><span>## [391] "CLCA_0392" "CLCA_0393" "CLCA_0394" "CLCA_0395" "CLCA_0396" "CLCA_0397"</span><br><span>## [397] "CLCA_0398" "CLCA_0399" "CLCA_0400" "CLCA_0401" "CLCA_0402" "CLCA_0403"</span><br><span>## [403] "CLCA_0404" "CLCA_0405" "CLCA_0406" "CLCA_0407" "CLCA_0408" "CLCA_0409"</span><br><span>## [409] "CLCA_0410" "CLCA_0411" "CLCA_0412" "CLCA_0413" "CLCA_0414" "CLCA_0415"</span><br><span>## [415] "CLCA_0416" "CLCA_0417" "CLCA_0418" "CLCA_0419" "CLCA_0420" "CLCA_0421"</span><br><span>## [421] "CLCA_0422" "CLCA_0423" "CLCA_0424" "CLCA_0425" "CLCA_0426" "CLCA_0427"</span><br><span>## [427] "CLCA_0428" "CLCA_0429" "CLCA_0430" "CLCA_0431" "CLCA_0432" "CLCA_0433"</span><br><span>## [433] "CLCA_0434" "CLCA_0435" "CLCA_0436" "CLCA_0437" "CLCA_0438" "CLCA_0439"</span><br><span>## [439] "CLCA_0440" "CLCA_0441" "CLCA_0442" "CLCA_0443" "CLCA_0444" "CLCA_0445"</span><br><span>## [445] "CLCA_0446" "CLCA_0447" "CLCA_0448" "CLCA_0449" "CLCA_0450" "CLCA_0451"</span><br><span>## [451] "CLCA_0452" "CLCA_0453" "CLCA_0454" "CLCA_0455" "CLCA_0456" "CLCA_0457"</span><br><span>## [457] "CLCA_0458" "CLCA_0459" "CLCA_0460" "CLCA_0461" "CLCA_0462" "CLCA_0463"</span><br><span>## [463] "CLCA_0464" "CLCA_0465" "CLCA_0466" "CLCA_0467" "CLCA_0468" "CLCA_0469"</span><br><span>## [469] "CLCA_0470" "CLCA_0471" "CLCA_0472" "CLCA_0473" "CLCA_0474" "CLCA_0475"</span><br><span>## [475] "CLCA_0476" "CLCA_0477" "CLCA_0478" "CLCA_0479" "CLCA_0480" "CLCA_0481"</span><br><span>## [481] "CLCA_0482" "CLCA_0483" "CLCA_0484" "CLCA_0485" "CLCA_0486" "CLCA_0487"</span><br><span>## [487] "CLCA_0488" "CLCA_0489" "CLCA_0490" "CLCA_0491" "CLCA_0492" "CLCA_0493"</span><br><span>## [493] "CLCA_0494"</span><br></code></pre><p data-tool="mdnice编辑器">文章中的突变图谱还对患者进行了分组，Group1 是在oncogene 发生coding突变的患者,Group2则为仅发生 synonymous 突变的患者，Group3为在oncogene上未发生突变的患者（在其他基因有发生突变）。结果显示Group1为418人，Group2为39人，Group3为36人，另外前面提到过突变信息缺少一名患者CLCA_0209。</p><pre data-tool="mdnice编辑器"><code>onco_genes_group1 = onco_genes[<span>1</span>:<span>23</span>]<br>onco_genes_group2 = onco_genes[<span>24</span>:<span>54</span>]<br>coding_mutations = c(<span>"nonsynonymous SNV"</span>,<br>                     <span>"stopgain"</span>,<br>                     <span>"splicing"</span>,<br>                     <span>"nonframeshift substitution"</span>,<br>                     <span>"frameshift deletion"</span>,<br>                     <span>"stoploss"</span>,<br>                     <span>"frameshift insertion"</span>,<br>                     <span>"startloss"</span>,<br>                     <span>"nonframeshift deletion"</span>,<br>                     <span>"nonframeshift insertion"</span><br>                     )<br>noncoding_mutations = c(<span>"3'UTR"</span>,<span>"5'UTR"</span>,<span>"lncRNA"</span>,<span>"lncrna.prom"</span>,<span>"promoter"</span>)<br><br>group1.id = unique(somatic[(somatic$Hugo_Symbol %<span>in</span>% onco_genes_group1) &amp; (somatic$Variant_Classification %<span>in</span>% coding_mutations), <span>1</span>])<br><br>group2.id = setdiff(unique(somatic[(somatic$Hugo_Symbol %<span>in</span>% onco_genes_group2) , <span>1</span>]),group1.id) <br><br>group3.id = setdiff(unique(somatic$Tumor_Sample_Barcode), c(group1.id,group2.id))<br>group.df = data.frame(Tumor_Sample_Barcode = c(group1.id,<br>                                               group2.id,<br>                                               group3.id),<br>                      Group = c(rep(<span>"Group1"</span>,times=length(group1.id)),<br>                                rep(<span>"Group2"</span>,times=length(group2.id)),<br>                                rep(<span>"Group3"</span>,times=length(group3.id)))<br>                      )<br>table(group.df$Group)<br><br><span>## </span><br><span>## Group1 Group2 Group3 </span><br><span>##    418     39     36</span><br></code></pre><p data-tool="mdnice编辑器">重新做突变图谱可视化加上 Group 分组信息：</p><pre data-tool="mdnice编辑器"><code>clinical = merge(clinical,group.df,by=<span>"Tumor_Sample_Barcode"</span>)<br>maf = read.maf(maf = somatic,<br>               vc_nonSyn=unique(somatic$Variant_Classification),<br>               clinicalData = clinical)<br><br><span>## -Validating</span><br><span>## --Non MAF specific values in Variant_Classification column:</span><br><span>##   promoter</span><br><span>##   nonsynonymous SNV</span><br><span>##   lncRNA</span><br><span>##   stopgain</span><br><span>##   splicing</span><br><span>##   lncrna.prom</span><br><span>##   nonframeshift substitution</span><br><span>##   frameshift deletion</span><br><span>##   stoploss</span><br><span>##   frameshift insertion</span><br><span>##   startloss</span><br><span>##   nonframeshift deletion</span><br><span>##   nonframeshift insertion</span><br><span>## -Summarizing</span><br><span>## --Possible FLAGS among top ten genes:</span><br><span>##   TTN</span><br><span>## -Processing clinical data</span><br><span>## -Finished in 8.028s elapsed (52.9s cpu)</span><br><br><span># 添加上临床信息</span><br>oncoplot(maf,<br>         genes = onco_genes,<br>         keepGeneOrder = <span>T</span>,<br>         sortByAnnotation = <span>T</span>,<br>         annotationFontSize = <span>1.2</span>,<br>         legendFontSize = <span>1.0</span>,<br>         removeNonMutated = <span>FALSE</span>,<br>         anno_height = <span>2</span>,<br>         clinicalFeatures = c(<span>"Group"</span>,<br>                              <span>"Gender"</span>,<br>                              <span>"Hepatitis"</span>,<br>                              <span>"BCLC"</span>,<br>                              <span>"Cirrhosis/Fibrosis"</span>,<br>                              <span>"Edmondson"</span>,<br>                              <span>"Multiple_lesions"</span>,<br>                              <span>"Smoking"</span>,<br>                              <span>"Alcohol"</span>,<br>                              <span>"Recurrence"</span>)<br>         )<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100037859" data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7gEpXQSKgBGFTyKKL1vLpkicrLLQaN4FibicWH720XSbYCpAvylsyia4NgLQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="960" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7gEpXQSKgBGFTyKKL1vLpkicrLLQaN4FibicWH720XSbYCpAvylsyia4NgLQ/640?wx_fmt=png&amp;from=appmsg"></figure><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><h2 data-tool="mdnice编辑器"><span>突变特征</span></h2><p data-tool="mdnice编辑器">作者使用的是 mSigHdp 和 SigProfilerExtractor 包进行突变特征分析：</p><blockquote data-tool="mdnice编辑器"><p>We used mSigHdp (v.1.1.2) and SigProfilerExtractor from SigProfiler bioinformatics tool suite (v.1.1.0)6 to extract SBS, DBS and ID signatures.For SigProfiler signature extraction, 1,000 iterations were performed (nmf_replicates = 1000). We report only signatures supported by both mSigHdp and SigProfiler.</p></blockquote><p data-tool="mdnice编辑器">得到的Signature 结果是：</p><blockquote data-tool="mdnice编辑器"><p>We identified 17 single-base substitution (SBS), 3 doublet-base substitution (DBS) and 8 small insertion-and-deletion (ID) signatures.</p></blockquote><p data-tool="mdnice编辑器">除了正文的 fig2 之外，还有 Extended Data fig2<img data-imgfileid="100037872" data-ratio="0.7833333333333333" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7gcUChHjGOcHQBxPWPuu7ZuUGGoWtaaI6tAwWAF4ib1nLcepgHWvLIfkQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7gcUChHjGOcHQBxPWPuu7ZuUGGoWtaaI6tAwWAF4ib1nLcepgHWvLIfkQ/640?wx_fmt=png&amp;from=appmsg"><img data-imgfileid="100037873" data-ratio="1.1407407407407408" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7giaWfjSTda3FHJsvbibn9f9y1gic1IwA7Ncpyg3NEtqJnfr81myKNjVXtg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7giaWfjSTda3FHJsvbibn9f9y1gic1IwA7Ncpyg3NEtqJnfr81myKNjVXtg/640?wx_fmt=png&amp;from=appmsg"></p><p data-tool="mdnice编辑器">考虑到作者用的方法较为复杂，这里改用maftools 里的signature 分析流程和 sigminer 包的分析流程两种方法：</p><pre data-tool="mdnice编辑器"><code><span># 突变特征方法一：maftools ----</span><br><span>library</span>(maftools)<br><span>library</span>(NMF)<br><span>library</span>(pheatmap)<br><span>library</span>(barplot3d)<br><span>library</span>(BSgenome.Hsapiens.UCSC.hg19)<br><span># 先构建三连核苷酸矩阵</span><br>maf.tnm = trinucleotideMatrix(maf = maf, <br>                              <span>#prefix = 'chr', </span><br>                              <span>#add = TRUE, </span><br>                              ref_genome = <span>"BSgenome.Hsapiens.UCSC.hg19"</span>)<br><br><span>## -Extracting 5' and 3' adjacent bases</span><br><span>## -Extracting +/- 20bp around mutated bases for background C&gt;T estimation</span><br><span>## -Estimating APOBEC enrichment scores</span><br><span>## --Performing one-way Fisher's test for APOBEC enrichment</span><br><span>## ---APOBEC related mutations are enriched in  0.408 % of samples (APOBEC enrichment score &gt; 2 ;  2  of  490  samples)</span><br><span>## -Creating mutation matrix</span><br><span>## --matrix of dimension 493x96</span><br><br><span># 运行 NMF非负矩阵分解，并拟合</span><br><span># 如果突变较少，需要设置 pConstant = 0.1</span><br>maf.sign = estimateSignatures(mat = maf.tnm, nTry = <span>12</span>)<br><br><span>## -Running NMF for 12 ranks</span><br><span>## Compute NMF rank= 2  ... + measures ... OK</span><br><span>## Compute NMF rank= 3  ... + measures ... OK</span><br><span>## Compute NMF rank= 4  ... + measures ... OK</span><br><span>## Compute NMF rank= 5  ... + measures ... OK</span><br><span>## Compute NMF rank= 6  ... + measures ... OK</span><br><span>## Compute NMF rank= 7  ... + measures ... OK</span><br><span>## Compute NMF rank= 8  ... + measures ... OK</span><br><span>## Compute NMF rank= 9  ... + measures ... OK</span><br><span>## Compute NMF rank= 10  ... + measures ... OK</span><br><span>## Compute NMF rank= 11  ... + measures ... OK</span><br><span>## Compute NMF rank= 12  ... + measures ... OK</span><br><br><span>## -Finished in 00:07:04 elapsed (00:01:26 cpu)</span><br><br><span># 确定最佳突变特征数量</span><br>plotCophenetic(res = maf.sign)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100037870" data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7gPJA5qwQeNcjFS6eyYVbD7rKq0PsTWSLeOyNBKgHCHDib3gTzib18NUxQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="960" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7gPJA5qwQeNcjFS6eyYVbD7rKq0PsTWSLeOyNBKgHCHDib3gTzib18NUxQ/640?wx_fmt=png&amp;from=appmsg"></figure><pre data-tool="mdnice编辑器"><code><span># 使用非负矩阵分解将矩阵分解为n签名</span><br>maf.sig = extractSignatures(mat = maf.tnm, n = <span>5</span>)<br><span># 与 COSMIC 的突变特征比较，计算余弦相似度</span><br>maf.v3.cosm = compareSignatures(nmfRes = maf.sig, sig_db = <span>"SBS"</span>)<br><span># 热图展示余弦相似度</span><br>pheatmap::pheatmap(mat = maf.v3.cosm$cosine_similarities, cluster_rows = <span>FALSE</span>, main = <span>"cosine similarity against validated signatures"</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100037869" data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7gkGt1xyjQuG6KW81r6gAwRZN5icsXRrcaPCchXl0kdd8AFn0gykrxmEA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="960" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7gkGt1xyjQuG6KW81r6gAwRZN5icsXRrcaPCchXl0kdd8AFn0gykrxmEA/640?wx_fmt=png&amp;from=appmsg"></figure><pre data-tool="mdnice编辑器"><code><span># 可视化突变特征</span><br>maftools::plotSignatures(nmfRes = maf.sig, title_size = <span>1.2</span>, sig_db = <span>"SBS"</span>)<br></code></pre><p data-tool="mdnice编辑器"><img data-imgfileid="100037871" data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7g6vs9EXhKoQOKMj5ze0MYibIgOlRlUHVwqEl7wSsok6lYRsWpibH9ymUw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="960" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7g6vs9EXhKoQOKMj5ze0MYibIgOlRlUHVwqEl7wSsok6lYRsWpibH9ymUw/640?wx_fmt=png&amp;from=appmsg"> 从 maftools 的突变特征分析结果上看，得到的 5 个突变特征分别与 COSMIC 数据库的 SBS30、SBS24、SBS6、SBS5、SBS22 余弦相似度较高。这与原文的结果相差较大，且 maftools 的方法仅分析 SBS 模式的 signature，如果要分析 DBS 或者 INDEL 等 signature，可以使用 sigminer（虽然sigminer 也提供了 SigProfiler的方法，不过用法也相对复杂，这里暂时不考虑。） sigminer 分析的 SBS突变特征有 8个，DBS 有4个，INDEL 有 8个：</p><pre data-tool="mdnice编辑器"><code><span># 突变特征方法二：sigminer ----</span><br><span>library</span>(sigminer)<br><span>## SBS ----</span><br>mt_tally &lt;- sig_tally(<br>  maf,<br>  ref_genome = <span>"BSgenome.Hsapiens.UCSC.hg19"</span>,<br>  useSyn = <span>TRUE</span>,<br>  mode = <span>"SBS"</span><br>)<br>mt_sig2 &lt;- sig_unify_extract(mt_tally$nmf_matrix, <br>                             range = <span>10</span>, <br>                             nrun = <span>10</span>)<br><br><span>## 10000 24224.97 25193.74 315481.7 2.800485e-07 8 8 </span><br><span>## 10000 24616.58 24754.58 303697.1 3.845836e-06 9 9 </span><br><span>## 20000 24616.44 24750.54 303665.8 5.317353e-07 9 9 </span><br><span>## 10000 24616.41 24748.37 303630.9 5.924343e-06 9 9 </span><br><span>## 20000 24614.28 24739.68 303965.6 2.87272e-05 9 9 </span><br><span>## 30000 24612.43 24743.56 304385.7 2.572798e-06 9 9 </span><br><span>## 10000 24314.31 25043.34 315276.9 0.0001300736 8 8 </span><br><span>## 20000 24294.42 25089.31 316380.3 5.047734e-06 8 8 </span><br><span>## 30000 24292.8 25085.5 315789.9 2.433658e-06 8 8 </span><br><span>## 40000 24287.52 25093.22 315099.2 1.110607e-05 8 8 </span><br><span>## 50000 24284.55 25087.06 314354.7 8.901478e-05 8 8 </span><br><span>## 10000 24605.98 24685.06 304196 1.692444e-05 9 9 </span><br><span>## 10000 23889.64 25687.55 328564.6 4.943007e-07 7 7 </span><br><span>## 10000 24226.84 25185.32 316387.9 1.451832e-07 8 8 </span><br><span>## 10000 24604.67 24688.01 303900.7 2.310264e-06 9 9</span><br><br>sim &lt;- get_sig_similarity(mt_sig2, sig_db = <span>"SBS"</span>)<br>pheatmap::pheatmap(sim$similarity)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100037878" data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7gk4XgNWd1al2CLibibTGicUzPuAgIYgSEibPtZT5C7K1P9icFfS6ZsrFLZzw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="960" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7gk4XgNWd1al2CLibibTGicUzPuAgIYgSEibPtZT5C7K1P9icFfS6ZsrFLZzw/640?wx_fmt=png&amp;from=appmsg"></figure><pre data-tool="mdnice编辑器"><code>show_sig_profile(mt_sig2, mode = <span>"SBS"</span>, style = <span>"cosmic"</span>, x_label_angle = <span>90</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100037875" data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7g7riawo9C1BAVbemQYYxV3hvwM6DnpfhpsqTgqeKdyYkt6fra5WTuWdw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="960" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7g7riawo9C1BAVbemQYYxV3hvwM6DnpfhpsqTgqeKdyYkt6fra5WTuWdw/640?wx_fmt=png&amp;from=appmsg"></figure><pre data-tool="mdnice编辑器"><code><span>## DBS ----</span><br>mt_tally_DBS &lt;- sig_tally(<br>  maf,<br>  ref_genome = <span>"BSgenome.Hsapiens.UCSC.hg19"</span>,<br>  useSyn = <span>TRUE</span>,<br>  mode = <span>"DBS"</span><br>)<br>mt_sig2_DBS &lt;- sig_unify_extract(mt_tally_DBS$nmf_matrix, <br>                             range = <span>10</span>, <br>                             nrun = <span>10</span>)<br></code></pre><pre data-tool="mdnice编辑器"><code>sim_DBS &lt;- get_sig_similarity(mt_sig2_DBS, sig_db = <span>"DBS"</span>)<br>pheatmap::pheatmap(sim_DBS$similarity)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100037874" data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7gEZZnVuI42KWwwMmtQG2KsJb3qibYoFaf6px1je5LSX32lEUJhA1hbiaA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="960" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7gEZZnVuI42KWwwMmtQG2KsJb3qibYoFaf6px1je5LSX32lEUJhA1hbiaA/640?wx_fmt=png&amp;from=appmsg"></figure><pre data-tool="mdnice编辑器"><code>show_sig_profile(mt_sig2_DBS, mode = <span>"DBS"</span>, style = <span>"cosmic"</span>, x_label_angle = <span>90</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100037876" data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7gEiaJZWp5VNZNmqspZHHGgXRtMicDkk7ibz3r89hdSHSwibSgCpKyqpToRA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="960" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7gEiaJZWp5VNZNmqspZHHGgXRtMicDkk7ibz3r89hdSHSwibSgCpKyqpToRA/640?wx_fmt=png&amp;from=appmsg"></figure><pre data-tool="mdnice编辑器"><code><span>## INDEL ----</span><br>mt_tally_ID &lt;- sig_tally(<br>  maf,<br>  ref_genome = <span>"BSgenome.Hsapiens.UCSC.hg19"</span>,<br>  useSyn = <span>TRUE</span>,<br>  mode = <span>"ID"</span><br>)<br>mt_sig2_ID &lt;- sig_unify_extract(mt_tally_ID$nmf_matrix, <br>                             range = <span>10</span>, <br>                             nrun = <span>10</span>)<br></code></pre><pre data-tool="mdnice编辑器"><code>sim_ID &lt;- get_sig_similarity(mt_sig2_ID, sig_db = <span>"ID"</span>)<br>pheatmap::pheatmap(sim_ID$similarity)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100037877" data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7g5LOjDuQQX8Tic9XxHMoWLI7DftEGOiaO02m6pSGYuPicxMdJavRzmgTicA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="960" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7g5LOjDuQQX8Tic9XxHMoWLI7DftEGOiaO02m6pSGYuPicxMdJavRzmgTicA/640?wx_fmt=png&amp;from=appmsg"></figure><pre data-tool="mdnice编辑器"><code>show_sig_profile(mt_sig2_ID, mode = <span>"ID"</span>, style = <span>"cosmic"</span>, x_label_angle = <span>90</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100037880" data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7g5Hn1icGsRBeSibbEUJHmGWbcfehwcB52MNJevz2YUKmyFU8xSvjmFlMg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="960" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7g5Hn1icGsRBeSibbEUJHmGWbcfehwcB52MNJevz2YUKmyFU8xSvjmFlMg/640?wx_fmt=png&amp;from=appmsg"></figure><h2 data-tool="mdnice编辑器"><span>ecDNA 分析</span></h2><p data-tool="mdnice编辑器">文章的 fig 3 a是ecDNA分析，以饼图形式展示，类型有 BFB、Circular(ecDNA)、Heavily rearranged、Linear 和 No fSCNA 其中 fig3a 原文注释信息是：</p><blockquote data-tool="mdnice编辑器"><p>The proportion of different amplicons across the CLCA cohort. Circular, breakage–fusion–bridge (BFB), heavily rearranged and linear, and no focal somatic copy-number amplification detected (fSCNA) amplicon categories are shown.</p></blockquote><p data-tool="mdnice编辑器">且正文中也提到了：</p><blockquote data-tool="mdnice编辑器"><p>ecDNA was detected in 27.3% of CLCA tumours</p></blockquote><p data-tool="mdnice编辑器">如果这个对应饼图的 Circular(ecDNA) 部分，那就是说在 27.3% 的肿瘤患者中检测到了 ecDNA 事件。<img data-imgfileid="100037882" data-ratio="0.6259259259259259" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7g0sfdk6faOSib0Y1zVjex8t67mJ4COCFoIaxCZhnojZV8UnxSz1aNU2Q/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7g0sfdk6faOSib0Y1zVjex8t67mJ4COCFoIaxCZhnojZV8UnxSz1aNU2Q/640?wx_fmt=png&amp;from=appmsg"></p><p data-tool="mdnice编辑器">我们可以在文章附件可以找到该图的数据，且数据显示，每一个患者可能发生4种 amp 事件的任意组合 。（注：文章上传的附件Supplementary Table 4：41586_2024_7054_MOESM6_ESM.xlsx 中Table 4g 第一行第三列是 Heavily rearranged rearranged，我的理解应该改为 Heavily rearranged ，以下读入的数据仅手动修改了这一项，其余的没做修改）</p><pre data-tool="mdnice编辑器"><code>amp = readxl::read_xlsx(<span>"41586_2024_7054_MOESM6_ESM.xlsx"</span>,sheet = <span>7</span>,skip = <span>2</span>)<br>amp = as.data.frame(amp)<br><span># 第一列是患者ID，</span><br><span># 第二列是amp的类型，</span><br><span># 第三列是发生某一 amp 类型的 interval counts 数</span><br>head(amp,n=<span>100</span>)<br></code></pre><section data-tool="mdnice编辑器"><br><table><thead><tr><th><span>sample_name</span></th><th><span>class</span></th><th><span>NIntervals</span></th><th><span>Intervals</span></th><th><span>OncogenesAmplified</span></th><th><span>TotalIntervalSize</span></th><th><span>AmplifiedIntervalSize</span></th><th><span>AverageAmplifiedCopyCount</span></th><th><span>Chromosomes</span></th><th><span>SeqenceEdges</span></th><th><span>BreakpointEdges</span></th><th><span>CoverageShifts</span></th><th><span>MeanshiftSegmentsCopyCount&gt;5</span></th><th><span>Foldbacks</span></th><th><span>CoverageShiftsWithBreakpointEdges</span></th></tr></thead><tbody><tr><td><span>CLCA_0001</span></td><td><span>Heavily rearranged rearranged</span></td><td><span>9</span></td><td><span>chr1:179385001-203717000,chr3:129798365-129808957,chr4:9699978-9720571,chr7:5927483-5948075,chr7:6851001-39963000,chr7:62463058-62473650,chr8:18014530-18025122,chr15:40853596-40864189,chr21:33796774-33807367</span></td><td><span>ETV1,CDC73,HOXA13,JAZF1,HOXA11,PTPRC,HNRNPA2B1,HOXA9,TPR,</span></td><td><span>57538154</span></td><td><span>52083461</span></td><td><span>2.765754199</span></td><td><span>7</span></td><td><span>145</span></td><td><span>59</span></td><td><span>0</span></td><td><span>0</span></td><td><span>0</span></td><td><span>0</span></td></tr><tr><td><span>CLCA_0001</span></td><td><span>Heavily rearranged</span></td><td><span>4</span></td><td><span>chr7:149730451-149741044,chr7:152588001-159138663,chr10:98451408-98662000,chr18:19772106-19792698</span></td><td><span>,</span></td><td><span>6792443</span></td><td><span>1210371</span></td><td><span>2.635659582</span></td><td><span>3</span></td><td><span>86</span></td><td><span>34</span></td><td><span>1</span></td><td><span>0</span></td><td><span>0</span></td><td><span>1</span></td></tr><tr><td><span>CLCA_0001</span></td><td><span>Circular</span></td><td><span>10</span></td><td><span>chr1:112603401-112813993,chr3:56765044-56775637,chr6:119458107-119568700,chr7:105224001-148714000,chr9:6484724-6695316,chr9:33786290-33806883,chr10:20048538-20059130,chr10:35290516-35311108,chr12:74004366-74014958,chr16:26592312-26612904</span></td><td><span>CREB3L2,KIAA1549,POT1,SMO,MET,EZH2,BRAF,</span></td><td><span>44115340</span></td><td><span>33806618</span></td><td><span>2.657480486</span></td><td><span>8</span></td><td><span>189</span></td><td><span>77</span></td><td><span>10</span></td><td><span>0</span></td><td><span>2</span></td><td><span>9</span></td></tr><tr><td><span>CLCA_0001</span></td><td><span>Heavily rearranged</span></td><td><span>10</span></td><td><span>chr1:26455415-26466008,chr5:94594394-94604987,chr7:64739205-64749797,chr11:77108686-77119279,chr12:34386996-34413582,chr12:34419001-34560000,chr12:38392853-38603445,chr13:27923699-28134291,chr19:23248743-23259335,chr19:28301468-28342061</span></td><td><span>,</span></td><td><span>682335</span></td><td><span>177647</span></td><td><span>4.184546951</span></td><td><span>7</span></td><td><span>113</span></td><td><span>25</span></td><td><span>0</span></td><td><span>0</span></td><td><span>0</span></td><td><span>0</span></td></tr><tr><td><span>CLCA_0001</span></td><td><span>Heavily rearranged</span></td><td><span>5</span></td><td><span>chr1:17217822-17238415,chr1:144593226-144603819,chr1:146382400-147865001,chr1:148549086-148559679,chr1:149205805-149246397</span></td><td><span>BCL9,</span></td><td><span>1564977</span></td><td><span>1074755</span></td><td><span>2.731614502</span></td><td><span>1</span></td><td><span>36</span></td><td><span>13</span></td><td><span>3</span></td><td><span>0</span></td><td><span>0</span></td><td><span>3</span></td></tr><tr><td><span>CLCA_0006</span></td><td><span>Circular</span></td><td><span>6</span></td><td><span>chr4:8438956-8449575,chr6:16220142-16230760,chr7:140356252-140376871,chr10:44079274-44099893,chr11:68389001-69076000,chr19:50449724-50460342</span></td><td><span>,</span></td><td><span>760098</span></td><td><span>700923</span></td><td><span>6.917366833</span></td><td><span>6</span></td><td><span>35</span></td><td><span>10</span></td><td><span>1</span></td><td><span>0</span></td><td><span>1</span></td><td><span>1</span></td></tr><tr><td><span>CLCA_0008</span></td><td><span>Linear</span></td><td><span>1</span></td><td><span>chr12:1-5343000</span></td><td><span>CCND2,KDM5A,</span></td><td><span>5343000</span></td><td><span>58827</span></td><td><span>3.867129349</span></td><td><span>1</span></td><td><span>18</span></td><td><span>6</span></td><td><span>3</span></td><td><span>0</span></td><td><span>2</span></td><td><span>3</span></td></tr><tr><td><span>CLCA_0008</span></td><td><span>Heavily rearranged</span></td><td><span>4</span></td><td><span>chr2:179287992-179308657,chr17:46222000-78477201,chr19:28933400-33881601,chrX:48988160-49017173</span></td><td><span>CEBPA,CCNE1,CANT1,SRSF2,MSI2,COL1A1,RNF43,DDX5,PRKAR1A,CLTC,BRIP1,HLF,CD79B,H3F3B,</span></td><td><span>37253084</span></td><td><span>32087537</span></td><td><span>2.673472586</span></td><td><span>4</span></td><td><span>101</span></td><td><span>37</span></td><td><span>5</span></td><td><span>0</span></td><td><span>0</span></td><td><span>5</span></td></tr><tr><td><span>CLCA_0008</span></td><td><span>Heavily rearranged</span></td><td><span>5</span></td><td><span>chr1:16865110-16885775,chr1:16987734-17018400,chr1:144593611-144614277,chr1:146382400-147845001,chr1:148539013-148559679</span></td><td><span>BCL9,</span></td><td><span>1555269</span></td><td><span>1143064</span></td><td><span>3.077854662</span></td><td><span>1</span></td><td><span>20</span></td><td><span>5</span></td><td><span>1</span></td><td><span>0</span></td><td><span>0</span></td><td><span>1</span></td></tr><tr><td><span>CLCA_0010</span></td><td><span>Heavily rearranged</span></td><td><span>10</span></td><td><span>chr4:435628-456203,chr5:34438809-34459385,chr5:94594411-94604987,chr6:119458120-119568696,chr7:64873330-64896483,chr8:47908000-146364022,chr10:50452244-50462819,chr12:132926005-132946581,chr18:29064978-29085553,chr19:28319060-28339636</span></td><td><span>NCOA2,RECQL4,CHCHD7,EXT1,RAD21,TCEA1,UBR5,NDRG1,MYC,PLAG1,COX6C,HEY1,</span></td><td><span>98713790</span></td><td><span>98276040</span></td><td><span>4.541944931</span></td><td><span>9</span></td><td><span>281</span></td><td><span>120</span></td><td><span>8</span></td><td><span>1</span></td><td><span>3</span></td><td><span>8</span></td></tr><tr><td><span>CLCA_0011</span></td><td><span>Heavily rearranged</span></td><td><span>5</span></td><td><span>chr1:17217881-17238420,chr1:144589853-144610392,chr1:146382400-147865001,chr1:148549140-148559679,chr1:149191007-149241546</span></td><td><span>BCL9,</span></td><td><span>1584762</span></td><td><span>1013046</span></td><td><span>2.739061578</span></td><td><span>1</span></td><td><span>46</span></td><td><span>15</span></td><td><span>2</span></td><td><span>0</span></td><td><span>0</span></td><td><span>2</span></td></tr><tr><td><span>CLCA_0011</span></td><td><span>Heavily rearranged</span></td><td><span>9</span></td><td><span>chr1:112612943-112823482,chr1:204034001-220189000,chr7:35964539-35985077,chr7:62462925-62473464,chr7:116223104-116243643,chr9:6494254-6704793,chr10:20048538-20059076,chr17:39246276-39266814,chr17:39296183-39316721</span></td><td><span>MDM4,SLC45A3,ELK4,</span></td><td><span>16679316</span></td><td><span>11301097</span></td><td><span>2.574830765</span></td><td><span>5</span></td><td><span>65</span></td><td><span>21</span></td><td><span>6</span></td><td><span>0</span></td><td><span>0</span></td><td><span>6</span></td></tr><tr><td><span>CLCA_0011</span></td><td><span>Heavily rearranged</span></td><td><span>14</span></td><td><span>chr1:4647909-4658448,chr1:156349001-171291000,chr1:174158001-202291000,chr1:226540001-227068000,chr2:117477265-117487804,chr4:437786-455812,chr5:94585136-94605674,chr18:29065444-29085983,chr19:19915204-19965742,chr19:20608823-20639361,chr19:20944620-20955158,chr19:24032644-24043183,chrX:4682464-4693002,chrY:19514391-19524930</span></td><td><span>CDC73,PRCC,FCGR2B,NTRK1,PTPRC,TPR,SDHC,ABL2,PBX1,</span></td><td><span>43806422</span></td><td><span>12521489</span></td><td><span>2.555923692</span></td><td><span>8</span></td><td><span>147</span></td><td><span>54</span></td><td><span>2</span></td><td><span>0</span></td><td><span>0</span></td><td><span>2</span></td></tr><tr><td><span>CLCA_0011</span></td><td><span>Heavily rearranged</span></td><td><span>3</span></td><td><span>chr1:235674001-242041000,chr5:692512-727516,chr5:766156-796694</span></td><td><span>FH,</span></td><td><span>6432544</span></td><td><span>60563</span></td><td><span>3.810204351</span></td><td><span>2</span></td><td><span>25</span></td><td><span>12</span></td><td><span>0</span></td><td><span>0</span></td><td><span>0</span></td><td><span>0</span></td></tr><tr><td><span>CLCA_0011</span></td><td><span>Heavily rearranged</span></td><td><span>2</span></td><td><span>chr1:227774001-233912001,chr7:157264846-157275384</span></td><td><span>,</span></td><td><span>6148540</span></td><td><span>4185741</span></td><td><span>2.542445875</span></td><td><span>2</span></td><td><span>13</span></td><td><span>5</span></td><td><span>0</span></td><td><span>0</span></td><td><span>0</span></td><td><span>0</span></td></tr><tr><td><span>CLCA_0440</span></td><td><span>BFB</span></td><td><span>1</span></td><td><span>chr11:68690001-69655000</span></td><td><span>CCND1,</span></td><td><span>965000</span></td><td><span>807634</span></td><td><span>11.64473457</span></td><td><span>1</span></td><td><span>15</span></td><td><span>4</span></td><td><span>5</span></td><td><span>1</span></td><td><span>1</span></td><td><span>4</span></td></tr><tr><td><span>CLCA_0440</span></td><td><span>Linear</span></td><td><span>1</span></td><td><span>chr7:77263001-78044000</span></td><td><span>,</span></td><td><span>781000</span></td><td><span>761473</span></td><td><span>6.945048095</span></td><td><span>1</span></td><td><span>5</span></td><td><span>1</span></td><td><span>1</span></td><td><span>0</span></td><td><span>1</span></td><td><span>1</span></td></tr><tr><td><span>CLCA_0443</span></td><td><span>Circular</span></td><td><span>7</span></td><td><span>chr6:922938-943526,chr6:15337944-15348533,chr8:64227297-64237886,chr8:102644618-102665206,chr11:78306545-78317133,chr12:9841925-9862513,chr13:74163001-115169878</span></td><td><span>ERCC5,</span></td><td><span>41100414</span></td><td><span>40427161</span></td><td><span>11.58781582</span></td><td><span>5</span></td><td><span>214</span></td><td><span>79</span></td><td><span>52</span></td><td><span>7</span></td><td><span>18</span></td><td><span>48</span></td></tr><tr><td><span>CLCA_0446</span></td><td><span>Heavily rearranged</span></td><td><span>1</span></td><td><span>chr2:195584001-196962000</span></td><td><span>,</span></td><td><span>1378000</span></td><td><span>1333450</span></td><td><span>3.102743007</span></td><td><span>1</span></td><td><span>19</span></td><td><span>8</span></td><td><span>1</span></td><td><span>0</span></td><td><span>0</span></td><td><span>1</span></td></tr><tr><td><span>CLCA_0446</span></td><td><span>Linear</span></td><td><span>1</span></td><td><span>chr2:81784001-83676000</span></td><td><span>,</span></td><td><span>1892000</span></td><td><span>1891198</span></td><td><span>3.586781312</span></td><td><span>1</span></td><td><span>35</span></td><td><span>17</span></td><td><span>0</span></td><td><span>0</span></td><td><span>0</span></td><td><span>0</span></td></tr><tr><td><span>CLCA_0446</span></td><td><span>BFB</span></td><td><span>1</span></td><td><span>chr16:48881001-51378000</span></td><td><span>CYLD,</span></td><td><span>2497000</span></td><td><span>2238057</span></td><td><span>4.281281454</span></td><td><span>1</span></td><td><span>47</span></td><td><span>22</span></td><td><span>3</span></td><td><span>0</span></td><td><span>0</span></td><td><span>2</span></td></tr><tr><td><span>CLCA_0446</span></td><td><span>Linear</span></td><td><span>1</span></td><td><span>chr20:21647001-23960000</span></td><td><span>,</span></td><td><span>2313000</span></td><td><span>2312969</span></td><td><span>3.797447309</span></td><td><span>1</span></td><td><span>31</span></td><td><span>13</span></td><td><span>1</span></td><td><span>0</span></td><td><span>0</span></td><td><span>1</span></td></tr><tr><td><span>CLCA_0446</span></td><td><span>Linear</span></td><td><span>1</span></td><td><span>chr6:1895001-3964000</span></td><td><span>,</span></td><td><span>2069000</span></td><td><span>2065337</span></td><td><span>3.474956733</span></td><td><span>1</span></td><td><span>27</span></td><td><span>12</span></td><td><span>0</span></td><td><span>0</span></td><td><span>0</span></td><td><span>0</span></td></tr><tr><td><span>CLCA_0446</span></td><td><span>Circular</span></td><td><span>2</span></td><td><span>chr20:12306001-15355000,chr20:19155447-19165995</span></td><td><span>,</span></td><td><span>3059549</span></td><td><span>3058791</span></td><td><span>4.059239442</span></td><td><span>1</span></td><td><span>52</span></td><td><span>22</span></td><td><span>2</span></td><td><span>0</span></td><td><span>0</span></td><td><span>1</span></td></tr><tr><td><span>CLCA_0446</span></td><td><span>Heavily rearranged</span></td><td><span>2</span></td><td><span>chr1:170813001-179368000,chr1:222176476-222187024</span></td><td><span>ABL2,</span></td><td><span>8565549</span></td><td><span>8538860</span></td><td><span>3.531468993</span></td><td><span>1</span></td><td><span>87</span></td><td><span>42</span></td><td><span>0</span></td><td><span>0</span></td><td><span>0</span></td><td><span>0</span></td></tr><tr><td><span>CLCA_0446</span></td><td><span>Heavily rearranged</span></td><td><span>5</span></td><td><span>chr15:20470882-20491430,chr17:50675340-50695889,chr19:28933400-33881601,chr21:28757235-28767784,chrX:48996665-49017214</span></td><td><span>CEBPA,CCNE1,</span></td><td><span>5020401</span></td><td><span>4958439</span></td><td><span>3.348515983</span></td><td><span>5</span></td><td><span>59</span></td><td><span>24</span></td><td><span>0</span></td><td><span>0</span></td><td><span>0</span></td><td><span>0</span></td></tr><tr><td><span>CLCA_0446</span></td><td><span>Heavily rearranged</span></td><td><span>6</span></td><td><span>chr1:16865098-16885646,chr1:17214199-17244747,chr1:144591196-144603819,chr1:146382400-147866149,chr1:148549130-148569679,chr1:149203321-149233869</span></td><td><span>BCL9,</span></td><td><span>1598571</span></td><td><span>1412987</span></td><td><span>3.654959179</span></td><td><span>1</span></td><td><span>59</span></td><td><span>21</span></td><td><span>3</span></td><td><span>0</span></td><td><span>0</span></td><td><span>3</span></td></tr><tr><td><span>CLCA_0446</span></td><td><span>Heavily rearranged</span></td><td><span>9</span></td><td><span>chr4:438101-448649,chr5:94594438-94614987,chr7:64739206-64749754,chr7:64873322-64896456,chr12:38490794-38511342,chr17:42089204-42109752,chr18:28342001-29429000,chr19:23258743-23269291,chr19:28293719-28339630</span></td><td><span>,</span></td><td><span>1249342</span></td><td><span>1215391</span></td><td><span>3.195655085</span></td><td><span>7</span></td><td><span>85</span></td><td><span>25</span></td><td><span>0</span></td><td><span>0</span></td><td><span>0</span></td><td><span>0</span></td></tr><tr><td><span>CLCA_0447</span></td><td><span>Circular</span></td><td><span>9</span></td><td><span>chr1:112703401-112713953,chr2:544014-554566,chr3:8487494-8498046,chr3:137211704-137222256,chr5:37606001-40604000,chr6:32439501-32566760,chr9:1-12739000,chr10:20048538-20059090,chr15:54214001-63803000</span></td><td><span>CD274,JAK2,LIFR,TCF12,</span></td><td><span>25506025</span></td><td><span>25387775</span></td><td><span>9.946383591</span></td><td><span>8</span></td><td><span>266</span></td><td><span>120</span></td><td><span>22</span></td><td><span>5</span></td><td><span>15</span></td><td><span>16</span></td></tr></tbody></table></section><pre data-tool="mdnice编辑器"><code><span># 每一列的大致信息</span><br>str(amp)<br><br><span>## 'data.frame':    2081 obs. of  15 variables:</span><br><span>##  $ sample_name                      : chr  "CLCA_0001" "CLCA_0001" "CLCA_0001" "CLCA_0001" ...</span><br><span>##  $ class                            : chr  "Heavily rearranged" "Heavily rearranged" "Circular" "Heavily rearranged" ...</span><br><span>##  $ NIntervals                       : num  9 4 10 10 5 6 1 4 5 10 ...</span><br><span>##  $ Intervals                        : chr  "chr1:179385001-203717000,chr3:129798365-129808957,chr4:9699978-9720571,chr7:5927483-5948075,chr7:6851001-399630"| __truncated__ "chr7:149730451-149741044,chr7:152588001-159138663,chr10:98451408-98662000,chr18:19772106-19792698" "chr1:112603401-112813993,chr3:56765044-56775637,chr6:119458107-119568700,chr7:105224001-148714000,chr9:6484724-"| __truncated__ "chr1:26455415-26466008,chr5:94594394-94604987,chr7:64739205-64749797,chr11:77108686-77119279,chr12:34386996-344"| __truncated__ ...</span><br><span>##  $ OncogenesAmplified               : chr  "ETV1,CDC73,HOXA13,JAZF1,HOXA11,PTPRC,HNRNPA2B1,HOXA9,TPR," "," "CREB3L2,KIAA1549,POT1,SMO,MET,EZH2,BRAF," "," ...</span><br><span>##  $ TotalIntervalSize                : num  57538154 6792443 44115340 682335 1564977 ...</span><br><span>##  $ AmplifiedIntervalSize            : num  52083461 1210371 33806618 177647 1074755 ...</span><br><span>##  $ AverageAmplifiedCopyCount        : num  2.77 2.64 2.66 4.18 2.73 ...</span><br><span>##  $ Chromosomes                      : num  7 3 8 7 1 6 1 4 1 9 ...</span><br><span>##  $ SeqenceEdges                     : num  145 86 189 113 36 35 18 101 20 281 ...</span><br><span>##  $ BreakpointEdges                  : num  59 34 77 25 13 10 6 37 5 120 ...</span><br><span>##  $ CoverageShifts                   : num  0 1 10 0 3 1 3 5 1 8 ...</span><br><span>##  $ MeanshiftSegmentsCopyCount&gt;5     : num  0 0 0 0 0 0 0 0 0 1 ...</span><br><span>##  $ Foldbacks                        : num  0 0 2 0 0 1 2 0 0 3 ...</span><br><span>##  $ CoverageShiftsWithBreakpointEdges: num  0 1 9 0 3 1 3 5 1 8 ...</span><br><br><span># 总共有 2081 列，amp 信息中，一位患者可以有多行记录，class 类型即为上面提到的类型。</span><br>nrow(amp)<br><br><span>## [1] 2081</span><br><br><span># 如果直接对表格的第三列 class 进行可视化，会发现结果缺失了 No fSCNA 类型，且比例也不对</span><br><span>library</span>(ggstatsplot)<br>ggpiestats(<br>  data = amp,<br>  x = class,<br>  palette = <span>"Set1"</span>,<br>  <span>#title = "Amplicon",</span><br>  results.subtitle = <span>F</span><br>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100037879" data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7g5LIzJwGICKDPd2EIjePictQtAqibUchEAhW5VSWp0lbYdEWoQnWkqS1w/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="960" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7g5LIzJwGICKDPd2EIjePictQtAqibUchEAhW5VSWp0lbYdEWoQnWkqS1w/640?wx_fmt=png&amp;from=appmsg"></figure><pre data-tool="mdnice编辑器"><code><span># 这是因为附件的 amp 数据，只包含发生拷贝数变异 Amplicon 的信息，如果患者没有发生，即 No fSCNA 类型，则没有记录在表格中。</span><br>table(amp$sample_name)<br><br><span>## </span><br><span>## CLCA_0001 CLCA_0006 CLCA_0008 CLCA_0010 CLCA_0011 CLCA_0013 CLCA_0015 CLCA_0016 </span><br><span>##         5         1         3         1         5         7         1         1 </span><br><span>## CLCA_0017 CLCA_0018 CLCA_0021 CLCA_0022 CLCA_0023 CLCA_0025 CLCA_0026 CLCA_0029 </span><br><span>##         3         6         2         3         2         2        29         3 </span><br><span>## CLCA_0031 CLCA_0034 CLCA_0038 CLCA_0039 CLCA_0040 CLCA_0044 CLCA_0045 CLCA_0046 </span><br><span>##         6         2         4         4         2         7         3         2 </span><br><span>## CLCA_0052 CLCA_0056 CLCA_0059 CLCA_0060 CLCA_0061 CLCA_0062 CLCA_0065 CLCA_0066 </span><br><span>##         6        24         5         1         1         5         1         4 </span><br><span>## CLCA_0067 CLCA_0068 CLCA_0069 CLCA_0070 CLCA_0071 CLCA_0072 CLCA_0073 CLCA_0074 </span><br><span>##         3         3         1         4         4         3         4         7 </span><br><span>## CLCA_0078 CLCA_0079 CLCA_0080 CLCA_0089 CLCA_0090 CLCA_0091 CLCA_0092 CLCA_0093 </span><br><span>##         2         4         7         5         4         3         3         4 </span><br><span>## CLCA_0095 CLCA_0096 CLCA_0097 CLCA_0098 CLCA_0099 CLCA_0100 CLCA_0101 CLCA_0102 </span><br><span>##         4        13         8         2         1         4         2         1 </span><br><span>## CLCA_0103 CLCA_0104 CLCA_0105 CLCA_0106 CLCA_0107 CLCA_0108 CLCA_0109 CLCA_0110 </span><br><span>##         3         2         4         3         2         2         1         1 </span><br><span>## CLCA_0111 CLCA_0112 CLCA_0113 CLCA_0114 CLCA_0115 CLCA_0116 CLCA_0118 CLCA_0119 </span><br><span>##         1        13         2         9         3         4       149        13 </span><br><span>## CLCA_0120 CLCA_0121 CLCA_0122 CLCA_0123 CLCA_0125 CLCA_0126 CLCA_0128 CLCA_0129 </span><br><span>##       235        20        31       217        39         1       201         2 </span><br><span>## CLCA_0130 CLCA_0132 CLCA_0133 CLCA_0135 CLCA_0137 CLCA_0139 CLCA_0140 CLCA_0141 </span><br><span>##         4         5         2         2         1         1         3         5 </span><br><span>## CLCA_0143 CLCA_0144 CLCA_0145 CLCA_0146 CLCA_0147 CLCA_0148 CLCA_0150 CLCA_0153 </span><br><span>##         7         2         9         2         1         3         8        45 </span><br><span>## CLCA_0154 CLCA_0156 CLCA_0157 CLCA_0158 CLCA_0159 CLCA_0160 CLCA_0165 CLCA_0166 </span><br><span>##         7         4         2         5         8         1         4         3 </span><br><span>## CLCA_0167 CLCA_0168 CLCA_0171 CLCA_0173 CLCA_0174 CLCA_0176 CLCA_0177 CLCA_0178 </span><br><span>##         6         5         3         1         1         7         1         3 </span><br><span>## CLCA_0182 CLCA_0187 CLCA_0188 CLCA_0189 CLCA_0190 CLCA_0191 CLCA_0192 CLCA_0194 </span><br><span>##         5         1         1         1         3         8         3         1 </span><br><span>## CLCA_0197 CLCA_0198 CLCA_0201 CLCA_0202 CLCA_0203 CLCA_0204 CLCA_0205 CLCA_0206 </span><br><span>##         2         2         4         5         5         2         3         1 </span><br><span>## CLCA_0207 CLCA_0208 CLCA_0210 CLCA_0212 CLCA_0215 CLCA_0216 CLCA_0217 CLCA_0218 </span><br><span>##         4         1         3         2         7         2         2         5 </span><br><span>## CLCA_0219 CLCA_0221 CLCA_0222 CLCA_0223 CLCA_0224 CLCA_0227 CLCA_0229 CLCA_0231 </span><br><span>##         4        10         6         8         5         1         2         1 </span><br><span>## CLCA_0232 CLCA_0233 CLCA_0235 CLCA_0236 CLCA_0237 CLCA_0239 CLCA_0243 CLCA_0245 </span><br><span>##         2         1         2         1         1         6         9         1 </span><br><span>## CLCA_0246 CLCA_0248 CLCA_0249 CLCA_0251 CLCA_0254 CLCA_0255 CLCA_0256 CLCA_0257 </span><br><span>##         4         1         1         1         1         1         1         1 </span><br><span>## CLCA_0258 CLCA_0259 CLCA_0261 CLCA_0263 CLCA_0265 CLCA_0268 CLCA_0270 CLCA_0271 </span><br><span>##         2         4         4         3         3         2         2         5 </span><br><span>## CLCA_0277 CLCA_0278 CLCA_0281 CLCA_0282 CLCA_0283 CLCA_0284 CLCA_0285 CLCA_0289 </span><br><span>##         1         2         1         2         3         2         5         4 </span><br><span>## CLCA_0291 CLCA_0293 CLCA_0294 CLCA_0295 CLCA_0296 CLCA_0301 CLCA_0303 CLCA_0305 </span><br><span>##         1         1         3         4         1         3         2         1 </span><br><span>## CLCA_0309 CLCA_0310 CLCA_0311 CLCA_0314 CLCA_0315 CLCA_0316 CLCA_0317 CLCA_0321 </span><br><span>##         1         1         2         2         3         1         3         1 </span><br><span>## CLCA_0323 CLCA_0324 CLCA_0325 CLCA_0327 CLCA_0330 CLCA_0331 CLCA_0332 CLCA_0334 </span><br><span>##         2         4        11         1         4         4         3         2 </span><br><span>## CLCA_0336 CLCA_0337 CLCA_0338 CLCA_0341 CLCA_0342 CLCA_0343 CLCA_0344 CLCA_0345 </span><br><span>##         5         5         3         1         2         3         3         4 </span><br><span>## CLCA_0346 CLCA_0347 CLCA_0348 CLCA_0349 CLCA_0351 CLCA_0352 CLCA_0354 CLCA_0356 </span><br><span>##         2         4         2         3         4         6         1         5 </span><br><span>## CLCA_0357 CLCA_0359 CLCA_0365 CLCA_0366 CLCA_0367 CLCA_0369 CLCA_0372 CLCA_0373 </span><br><span>##        14         2         3        10         3        11         1         5 </span><br><span>## CLCA_0375 CLCA_0376 CLCA_0377 CLCA_0378 CLCA_0379 CLCA_0382 CLCA_0384 CLCA_0385 </span><br><span>##         3         1        12         1         2         6        13         4 </span><br><span>## CLCA_0387 CLCA_0388 CLCA_0389 CLCA_0390 CLCA_0391 CLCA_0392 CLCA_0393 CLCA_0394 </span><br><span>##         1         1         4        20         1        12         4         1 </span><br><span>## CLCA_0395 CLCA_0398 CLCA_0399 CLCA_0400 CLCA_0401 CLCA_0402 CLCA_0403 CLCA_0404 </span><br><span>##         3         9         1         1         4         4         2         4 </span><br><span>## CLCA_0406 CLCA_0407 CLCA_0408 CLCA_0409 CLCA_0410 CLCA_0411 CLCA_0412 CLCA_0413 </span><br><span>##        11         5         7         2         2         2         2         1 </span><br><span>## CLCA_0414 CLCA_0416 CLCA_0418 CLCA_0419 CLCA_0420 CLCA_0421 CLCA_0424 CLCA_0425 </span><br><span>##         1         4         2        10         4         1         4         1 </span><br><span>## CLCA_0426 CLCA_0428 CLCA_0429 CLCA_0433 CLCA_0435 CLCA_0439 CLCA_0440 CLCA_0443 </span><br><span>##         2         1        14         3         7         2         2         1 </span><br><span>## CLCA_0446 CLCA_0447 CLCA_0448 CLCA_0450 CLCA_0451 CLCA_0458 CLCA_0461 CLCA_0462 </span><br><span>##        10        18         3         1         7         2         3         1 </span><br><span>## CLCA_0465 CLCA_0467 CLCA_0470 CLCA_0472 CLCA_0474 CLCA_0475 CLCA_0477 CLCA_0478 </span><br><span>##         5         1         1         3         2         1        16         3 </span><br><span>## CLCA_0479 CLCA_0480 CLCA_0481 CLCA_0482 CLCA_0484 CLCA_0485 CLCA_0486 CLCA_0487 </span><br><span>##         2        20         7        23         2         9         7         4 </span><br><span>## CLCA_0488 CLCA_0492 CLCA_0493 CLCA_0494 </span><br><span>##         1         1         3         2</span><br><br>table(amp$class)<br><br><span>## </span><br><span>##                BFB           Circular Heavily rearranged             Linear </span><br><span>##                830                231                704                316</span><br><br><span># 总共是494名患者，其中amp 表格记录的患者有 300 名</span><br>unique(amp$sample_name) %&gt;% length()<br><br><span>## [1] 300</span><br><br><span># 那么没有 amp 记录的患者就是 194 名，比例为 39% 和原图符合</span><br><span>194</span>/<span>494</span><br><br><span>## [1] 0.3927126</span><br><br><span># 先简单粗暴地获取每一种amp类型的患者ID</span><br>BFB.id = unique(amp[amp$class == <span>"BFB"</span>,<span>1</span>])<br>Circular.id = unique(amp[amp$class == <span>"Circular"</span>,<span>1</span>])<br>Heavily_rearranged.id = unique(amp[amp$class == <span>"Heavily rearranged"</span>,<span>1</span>])<br>Linear.id = unique(amp[amp$class == <span>"Linear"</span>,<span>1</span>])<br>No_fSCNA.id = setdiff(clinical$Tumor_Sample_Barcode, unique(amp$sample_name))<br>length(BFB.id);length(Circular.id);length(Heavily_rearranged.id);length(Linear.id);length(No_fSCNA.id)<br><br><span>## [1] 81</span><br><br><span>## [1] 135</span><br><br><span>## [1] 233</span><br><br><span>## [1] 137</span><br><br><span>## [1] 193</span><br></code></pre><p data-tool="mdnice编辑器">韦恩图进行可视化可以发现，这样获取到的患者ID是有交集的，前面就提到过了，每一个患者可能发生4种 amp 事件的任意组合。所以有交集才是正常的。但这样的话，原文的饼图就无法解释了。</p><pre data-tool="mdnice编辑器"><code><span># 韦恩图进行可视化</span><br>amp.list = list(BFB.id,Circular.id,Heavily_rearranged.id,Linear.id,No_fSCNA.id)<br>names(amp.list) = c(<span>'BFB'</span>,<span>'Circular'</span>,<span>'Heavily_rearranged'</span>,<span>'Linear'</span>,<span>'No_fSCNA'</span>)<br>venn.plot1 &lt;- venn.diagram(<br>  x = amp.list,<br>  col = <span>"transparent"</span>,<br>  euler.d = <span>TRUE</span>,<br>  fill = c(<span>"#E64B35B2"</span>, <span>"#4DBBD5B2"</span>, <span>"#00A087B2"</span>, <span>"#3C5488B2"</span>, <span>"#F39B7FB2"</span>),<br>  alpha = rep(<span>0.6</span>,time = <span>5</span>),<br>  cex = <span>1.2</span>,<br>  cat.cex = <span>1.0</span>,<br>  <span># main = patients[i],</span><br>  main.cex = <span>1.0</span>,<br>  print.mode = c(<span>"raw"</span>, <span>"percent"</span>),<br>  category.names = names(amp.list),<br>  filename = <span>NULL</span><br>  <br>)<br>p = as_ggplot(venn.plot1)<br>print(p)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100037883" data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7gLOdztfIVFmxr8yo0orMNJMNSx4UPAgZtJdPydLLWNql6rqUnzVHwVg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="960" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7gLOdztfIVFmxr8yo0orMNJMNSx4UPAgZtJdPydLLWNql6rqUnzVHwVg/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">尝试探索一下数据以获取和原文中的比例接近的结果。从数据上看，发生 Circular(ecDNA) 患者是 135名， 135/494=27.3% 符合原文饼图比例。但其他amp事件Heavily rearranged、Linear、BFB 就不符合比例了,不满足。除非取差集，也就是对 amp 事件划分优先级，发生 Circular(ecDNA) 事件的患者不再记录其他事件，即 Circular(ecDNA) &gt; BFB &gt; Heavily rearranged &gt;Linear，这样比例符合了，但无法理解这样做的意义何在？</p><pre data-tool="mdnice编辑器"><code><span># Circular(ecDNA) </span><br>length(Circular.id)/<span>494</span><br><br><span>## [1] 0.2732794</span><br><br><span># BFB</span><br>setdiff(BFB.id,Circular.id) %&gt;% length() /<span>494</span><br><br><span>## [1] 0.09311741</span><br><br><span># Heavily rearranged </span><br>setdiff(Heavily_rearranged.id,c(BFB.id,Circular.id)) %&gt;% length() /<span>494</span><br><br><span>## [1] 0.2226721</span><br><br><span># Linear </span><br>setdiff(Linear.id,c(BFB.id,Circular.id,Heavily_rearranged.id)) %&gt;% length() /<span>494</span><br><br><span>## [1] 0.01821862</span><br><br><span># 虽然这样结果和作者的结果吻合，但是这样做的意义何在呢？</span><br>amp2 = data.frame(sample_name = c(Circular.id,<br>                                  setdiff(BFB.id,Circular.id),<br>                                  setdiff(Heavily_rearranged.id,c(BFB.id,Circular.id)),<br>                                  setdiff(Linear.id,c(BFB.id,Circular.id,Heavily_rearranged.id)),<br>                                  No_fSCNA.id<br>                                  ),<br>                  class = c(rep(<span>"Circular"</span>,times = length(Circular.id)),<br>                            rep(<span>"BFB"</span>,times = length(setdiff(BFB.id,Circular.id))),<br>                            rep(<span>"Heavily_rearranged"</span>,times = length(setdiff(Heavily_rearranged.id,<br>                                                                            c(BFB.id,Circular.id)))),<br>                            rep(<span>"Linear"</span>,times = length(setdiff(Linear.id,<br>                                                                   c(BFB.id,<br>                                                                     Circular.id,<br>                                                                     Heavily_rearranged.id)))),<br>                            rep(<span>"No_fSCNA"</span>,times = length(No_fSCNA.id)))<br>                  )<br></code></pre><pre data-tool="mdnice编辑器"><code><span># 饼图</span><br><br>ggpiestats(<br>  data = amp2,<br>  x = class,<br>  palette = <span>"Set1"</span>,<br>  <span>#title = "Amplicon",</span><br>  results.subtitle = <span>F</span><br>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100037881" data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7gdgg3JLOph3Ahnhox5RtKGjHvyCZJTDKUSxibsmAwicyhKFnRxpIbxw0A/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="960" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7gdgg3JLOph3Ahnhox5RtKGjHvyCZJTDKUSxibsmAwicyhKFnRxpIbxw0A/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">fig3b 是 ecDNA 上的基因列表，进行柱状图可视化。但是根据作者上传的附件重现出来的结果和文章的fig.3b 并不止一致，如文章原图中的 EXT1 MYC RAD21 NDRG1柱子高度相接近，但上面可视化出来的结果显示MYC 较高，其他的较低。</p><pre data-tool="mdnice编辑器"><code><span># 获取 ecDNA </span><br>ecDNA_amp = amp[amp$class==<span>"Circular"</span>,]<br>head(ecDNA_amp,n=<span>20</span>)<br></code></pre><section data-tool="mdnice编辑器"><table><thead><tr><th><span>sample_name</span></th><th><span>class</span></th><th><span>NIntervals</span></th><th><span>Intervals</span></th><th><span>OncogenesAmplified</span></th><th><span>TotalIntervalSize</span></th><th><span>AmplifiedIntervalSize</span></th><th><span>AverageAmplifiedCopyCount</span></th><th><span>Chromosomes</span></th><th><span>SeqenceEdges</span></th><th><span>BreakpointEdges</span></th><th><span>CoverageShifts</span></th><th><span>MeanshiftSegmentsCopyCount&gt;5</span></th><th><span>Foldbacks</span></th><th><span>CoverageShiftsWithBreakpointEdges</span></th></tr></thead><tbody><tr><td><span>CLCA_0001</span></td><td><span>Circular</span></td><td><span>10</span></td><td><span>chr1:112603401-112813993,chr3:56765044-56775637,chr6:119458107-119568700,chr7:105224001-148714000,chr9:6484724-6695316,chr9:33786290-33806883,chr10:20048538-20059130,chr10:35290516-35311108,chr12:74004366-74014958,chr16:26592312-26612904</span></td><td><span>CREB3L2,KIAA1549,POT1,SMO,MET,EZH2,BRAF,</span></td><td><span>44115340</span></td><td><span>33806618</span></td><td><span>2.65748</span></td><td><span>8</span></td><td><span>189</span></td><td><span>77</span></td><td><span>10</span></td><td><span>0</span></td><td><span>2</span></td><td><span>9</span></td></tr><tr><td><span>CLCA_0006</span></td><td><span>Circular</span></td><td><span>6</span></td><td><span>chr4:8438956-8449575,chr6:16220142-16230760,chr7:140356252-140376871,chr10:44079274-44099893,chr11:68389001-69076000,chr19:50449724-50460342</span></td><td><span>,</span></td><td><span>760098</span></td><td><span>700923</span></td><td><span>6.917367</span></td><td><span>6</span></td><td><span>35</span></td><td><span>10</span></td><td><span>1</span></td><td><span>0</span></td><td><span>1</span></td><td><span>1</span></td></tr><tr><td><span>CLCA_0443</span></td><td><span>Circular</span></td><td><span>7</span></td><td><span>chr6:922938-943526,chr6:15337944-15348533,chr8:64227297-64237886,chr8:102644618-102665206,chr11:78306545-78317133,chr12:9841925-9862513,chr13:74163001-115169878</span></td><td><span>ERCC5,</span></td><td><span>41100414</span></td><td><span>40427161</span></td><td><span>11.58782</span></td><td><span>5</span></td><td><span>214</span></td><td><span>79</span></td><td><span>52</span></td><td><span>7</span></td><td><span>18</span></td><td><span>48</span></td></tr><tr><td><span>CLCA_0446</span></td><td><span>Circular</span></td><td><span>2</span></td><td><span>chr20:12306001-15355000,chr20:19155447-19165995</span></td><td><span>,</span></td><td><span>3059549</span></td><td><span>3058791</span></td><td><span>4.059239</span></td><td><span>1</span></td><td><span>52</span></td><td><span>22</span></td><td><span>2</span></td><td><span>0</span></td><td><span>0</span></td><td><span>1</span></td></tr><tr><td><span>CLCA_0447</span></td><td><span>Circular</span></td><td><span>9</span></td><td><span>chr1:112703401-112713953,chr2:544014-554566,chr3:8487494-8498046,chr3:137211704-137222256,chr5:37606001-40604000,chr6:32439501-32566760,chr9:1-12739000,chr10:20048538-20059090,chr15:54214001-63803000</span></td><td><span>CD274,JAK2,LIFR,TCF12,</span></td><td><span>25506025</span></td><td><span>25387775</span></td><td><span>9.946384</span></td><td><span>8</span></td><td><span>266</span></td><td><span>120</span></td><td><span>22</span></td><td><span>5</span></td><td><span>15</span></td><td><span>16</span></td></tr><tr><td><span>CLCA_0447</span></td><td><span>Circular</span></td><td><span>1</span></td><td><span>chr3:171543001-172009000</span></td><td><span>,</span></td><td><span>466000</span></td><td><span>446113</span></td><td><span>4.344856</span></td><td><span>1</span></td><td><span>6</span></td><td><span>1</span></td><td><span>2</span></td><td><span>0</span></td><td><span>1</span></td><td><span>2</span></td></tr><tr><td><span>CLCA_0447</span></td><td><span>Circular</span></td><td><span>1</span></td><td><span>chr22:35836001-36188000</span></td><td><span>,</span></td><td><span>352000</span></td><td><span>351993</span></td><td><span>3.970767</span></td><td><span>1</span></td><td><span>7</span></td><td><span>3</span></td><td><span>0</span></td><td><span>0</span></td><td><span>0</span></td><td><span>0</span></td></tr><tr><td><span>CLCA_0447</span></td><td><span>Circular</span></td><td><span>1</span></td><td><span>chr15:65816001-66021000</span></td><td><span>,</span></td><td><span>205000</span></td><td><span>204997</span></td><td><span>4.891946</span></td><td><span>1</span></td><td><span>3</span></td><td><span>1</span></td><td><span>0</span></td><td><span>0</span></td><td><span>0</span></td><td><span>0</span></td></tr><tr><td><span>CLCA_0447</span></td><td><span>Circular</span></td><td><span>1</span></td><td><span>chr6:43482001-44354000</span></td><td><span>,</span></td><td><span>872000</span></td><td><span>852707</span></td><td><span>3.857693</span></td><td><span>1</span></td><td><span>9</span></td><td><span>2</span></td><td><span>1</span></td><td><span>0</span></td><td><span>0</span></td><td><span>1</span></td></tr><tr><td><span>CLCA_0447</span></td><td><span>Circular</span></td><td><span>1</span></td><td><span>chr15:67144001-67445000</span></td><td><span>,</span></td><td><span>301000</span></td><td><span>300105</span></td><td><span>5.693422</span></td><td><span>1</span></td><td><span>6</span></td><td><span>2</span></td><td><span>0</span></td><td><span>0</span></td><td><span>0</span></td><td><span>0</span></td></tr><tr><td><span>CLCA_0451</span></td><td><span>Circular</span></td><td><span>7</span></td><td><span>chr2:70504051-70524611,chr5:112940001-118170000,chr6:32432533-32579209,chr7:40448001-43603000,chr8:109545001-120080000,chr11:113948001-116387000,chr11:130256001-135006516</span></td><td><span>EXT1,RAD21,</span></td><td><span>26276754</span></td><td><span>41610</span></td><td><span>3.206914</span></td><td><span>6</span></td><td><span>209</span></td><td><span>92</span></td><td><span>1</span></td><td><span>0</span></td><td><span>0</span></td><td><span>1</span></td></tr><tr><td><span>CLCA_0461</span></td><td><span>Circular</span></td><td><span>16</span></td><td><span>chr1:4649294-4659888,chr1:150319900-223199001,chr1:225205000-233912001,chr4:437018-457611,chr5:40686831-40697425,chr5:94594161-94604755,chr7:64865863-64896457,chr12:38490794-38511387,chr17:42089772-42100365,chr18:29073967-29084561,chr19:20599211-20629805,chr19:20944362-20955213,chr19:23476178-23486772,chr19:28280720-28349688,chrX:4689289-4699883,chrY:19505241-19515835</span></td><td><span>H3F3A,ARNT,PRCC,FCGR2B,MUC1,CDC73,TPM3,NTRK1,SLC45A3,PTPRC,TPR,ELK4,SDHC,ABL2,MDM4,PBX1,</span></td><td><span>81853062</span></td><td><span>81341590</span></td><td><span>4.435998</span></td><td><span>10</span></td><td><span>366</span></td><td><span>127</span></td><td><span>10</span></td><td><span>0</span></td><td><span>0</span></td><td><span>9</span></td></tr><tr><td><span>CLCA_0465</span></td><td><span>Circular</span></td><td><span>4</span></td><td><span>chr11:2189298-2209845,chr11:59494001-59723000,chr11:60340001-61377000,chr11:68706001-70495000</span></td><td><span>CCND1,</span></td><td><span>3075548</span></td><td><span>2898925</span></td><td><span>4.283449</span></td><td><span>1</span></td><td><span>37</span></td><td><span>10</span></td><td><span>9</span></td><td><span>0</span></td><td><span>2</span></td><td><span>7</span></td></tr><tr><td><span>CLCA_0465</span></td><td><span>Circular</span></td><td><span>8</span></td><td><span>chr5:34438837-34459385,chr6:32432537-32579209,chr6:119458160-119658708,chr8:69214001-146364022,chr10:50452244-50462791,chr12:131860272-131880819,chr12:132926033-132936581,chr20:1380675-1391222</span></td><td><span>NCOA2,RECQL4,EXT1,RAD21,COX6C,NDRG1,MYC,UBR5,HEY1,</span></td><td><span>77569986</span></td><td><span>76966860</span></td><td><span>3.3338</span></td><td><span>6</span></td><td><span>376</span></td><td><span>156</span></td><td><span>9</span></td><td><span>1</span></td><td><span>2</span></td><td><span>8</span></td></tr><tr><td><span>CLCA_0470</span></td><td><span>Circular</span></td><td><span>1</span></td><td><span>chr1:154834001-155367000</span></td><td><span>MUC1,</span></td><td><span>533000</span></td><td><span>532996</span></td><td><span>5.465167</span></td><td><span>1</span></td><td><span>4</span></td><td><span>2</span></td><td><span>0</span></td><td><span>0</span></td><td><span>0</span></td><td><span>0</span></td></tr><tr><td><span>CLCA_0472</span></td><td><span>Circular</span></td><td><span>5</span></td><td><span>chr16:46518719-46529341,chr16:46552064-46649972,chr16:46715036-46735657,chr16:46767016-46787638,chr16:46825001-49898000</span></td><td><span>,</span></td><td><span>3222777</span></td><td><span>3200819</span></td><td><span>7.860472</span></td><td><span>1</span></td><td><span>100</span></td><td><span>43</span></td><td><span>8</span></td><td><span>0</span></td><td><span>0</span></td><td><span>7</span></td></tr><tr><td><span>CLCA_0480</span></td><td><span>Circular</span></td><td><span>3</span></td><td><span>chr6:32434297-32464893,chr6:32478990-32571656,chr20:832001-2745000</span></td><td><span>,</span></td><td><span>2036264</span></td><td><span>32725</span></td><td><span>3.50202</span></td><td><span>2</span></td><td><span>94</span></td><td><span>45</span></td><td><span>0</span></td><td><span>0</span></td><td><span>0</span></td><td><span>0</span></td></tr><tr><td><span>CLCA_0481</span></td><td><span>Circular</span></td><td><span>10</span></td><td><span>chr1:112603401-112814054,chr3:197842684-197853337,chr4:190896293-190916947,chr5:34438730-34459384,chr6:119458046-119668700,chr7:116222989-116233643,chr9:6494140-6604794,chr10:18594001-37659000,chr16:26592312-26612965,chr18:14772319-14782973</span></td><td><span>KIF5B,ABI1,MLLT10,</span></td><td><span>19690892</span></td><td><span>16577851</span></td><td><span>2.984219</span></td><td><span>10</span></td><td><span>151</span></td><td><span>56</span></td><td><span>6</span></td><td><span>0</span></td><td><span>0</span></td><td><span>6</span></td></tr><tr><td><span>CLCA_0482</span></td><td><span>Circular</span></td><td><span>1</span></td><td><span>chr1:196437001-197013000</span></td><td><span>,</span></td><td><span>576000</span></td><td><span>471329</span></td><td><span>3.063173</span></td><td><span>1</span></td><td><span>19</span></td><td><span>7</span></td><td><span>4</span></td><td><span>0</span></td><td><span>0</span></td><td><span>4</span></td></tr><tr><td><span>CLCA_0484</span></td><td><span>Circular</span></td><td><span>4</span></td><td><span>chr10:1746001-4002667,chr11:66975001-67656269,chr11:71568859-71579471,chr14:35431001-38257865</span></td><td><span>FOXA1,NKX2-1,</span></td><td><span>5775414</span></td><td><span>4977134</span></td><td>7.391996</td><td><span>3</span></td><td><span>41</span></td><td><span>14</span></td><td><span>10</span></td><td><span>3</span></td><td><span>3</span></td><td>7</td></tr></tbody></table></section><pre data-tool="mdnice编辑器"><code>    <span># 获取ecDNA 的 top20 基因</span><br>    genes = paste(ecDNA_amp$OncogenesAmplified[<span>1</span>:nrow(ecDNA_amp)],collapse = <span>","</span>) %&gt;% str_split(pattern = <span>","</span>)<br>    genes = genes[[<span>1</span>]]<br>    top20 = rev(head(tail(sort(table(genes)),n=<span>21</span>),n=<span>20</span>))<br>    top20_gene = names(top20)<br>    <span># top20 gene 对应的 amp 类型</span><br>    amp_top20 = data.frame()<br>    <span>for</span> (i <span>in</span> top20_gene) {<br>      amp_gene = amp[grep(pattern = i,ignore.case = <span>F</span>,x = amp$OncogenesAmplified),]<br>      amp_gene$gene = i<br>      amp_top20 = rbind(amp_top20,amp_gene)<br>    }<br>    <br>    <span># 柱状图可视化</span><br>    amp_top20$gene = factor(amp_top20$gene,levels = top20_gene)<br>    amp_top20$class = factor(amp_top20$class,levels = c(<span>"Linear"</span>,<br>                                                        <span>"Heavily rearranged"</span>,<br>                                                        <span>"BFB"</span>,<br>                                                        <span>"Circular"</span>))<br>    <br>    p = ggplot(data = amp_top20) + <br>      geom_bar( aes(x = gene, fill = class),<br>                <span>#width = 0.5,</span><br>                <span>#position =position_dodge2(padding = 0.5, preserve = "single"),</span><br>                stat = <span>"count"</span>) + <br>      <span># facet_grid(. ~ Patient, scales = 'free_x', space = 'free') + </span><br>      theme_classic() + <br>      theme(<br>            panel.border = element_blank()) +<br>      xlab(label = <span>"top20 gene"</span>)+<br>      ylab(label = <span>"Frequency"</span>) +<br>      scale_fill_manual(values = c(<span>"#377EB8"</span>, <span>"#4DAF4A"</span>, <span>"#FF7F00"</span>, <span>"#984EA3"</span>))<br>    p<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100037886" data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7gnjHhOkVT9KrjfaTfGXpP7ehSpWZbS4IWkZgIGBPnvyKibibOiakT4bibMw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="960" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7gnjHhOkVT9KrjfaTfGXpP7ehSpWZbS4IWkZgIGBPnvyKibibOiakT4bibMw/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">还有就是，统计出来的 top20 基因列表和文章的不一致：</p><pre data-tool="mdnice编辑器"><code>    top20_paper = c(<span>"CCND1"</span>,<span>"EXT1"</span>,<span>"MYC"</span>,<span>"RAD21"</span>,<span>"NDRG1"</span>,<br>                    <span>"UBR5"</span>,<span>"COX6C"</span>,<span>"RECQL4"</span>,<span>"MUC1"</span>,<span>"TPM3"</span>,<br>                    <span>"NCOA2"</span>,<span>"NTRK1"</span>,<span>"PBX1"</span>,<span>"PRCC"</span>,<span>"ARNT"</span>,<br>                    <span>"FCGR2B"</span>,<span>"HEY1"</span>,<span>"SDHC"</span>,<span>"CHCHD7"</span>,<span>"MET"</span>)<br>    amp.list = list(top20_gene=top20_gene,top20_paper=top20_paper)<br><br><br>    <span>library</span>(ggvenn)<br>    ggvenn(amp.list, <br>           show_elements = <span>F</span>, <br>           show_percentage = <span>T</span>,<br>           label_sep = <span>"\n"</span>, <br>           fill_color = c(<span>"#E64B35B2"</span>, <span>"#4DBBD5B2"</span>),<br>           auto_scale = <span>T</span><br>           )<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100037888" data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7gMJrnqR6DUibdYbSORVxw7assfoibG1rxPZUbryckETTcv0KLXw0XbObg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="960" title="" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7gMJrnqR6DUibdYbSORVxw7assfoibG1rxPZUbryckETTcv0KLXw0XbObg/640?wx_fmt=png&amp;from=appmsg"></figure><pre data-tool="mdnice编辑器"><code>    ggvenn(amp.list, <br>           show_elements = <span>T</span>, <br>           show_percentage = <span>F</span>,<br>           label_sep = <span>"\n"</span>, <br>           fill_color = c(<span>"#E64B35B2"</span>, <span>"#4DBBD5B2"</span>),<br>           auto_scale = <span>T</span><br>           )<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100037887" data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7gASIh8b2gCvqTaxKuKticeTjicg4CnXSyBDKvAPqFAW7uPmaXVgPfXgOQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="960" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7gASIh8b2gCvqTaxKuKticeTjicg4CnXSyBDKvAPqFAW7uPmaXVgPfXgOQ/640?wx_fmt=png&amp;from=appmsg"></figure><figure data-tool="mdnice编辑器"><br></figure></section><figure data-tool="mdnice编辑器"><br></figure><figure data-tool="mdnice编辑器"><br></figure></section><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><h2 data-tool="mdnice编辑器"><span></span><span><span>基因组重排的 circle plot</span></span></h2><p data-tool="mdnice编辑器">文章中的 fig.4b 是基因组重排的 circle plot，以 CLCA_0119 患者为例，circle plot 纳入了拷贝数变异信息 CN 和结构变异SV信息。<img data-imgfileid="100037912" data-ratio="0.587037037037037" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7gWO3IeOWZzRXZOia5Xpibh7shcEvLXEhFGOqZdibREDiblhPghgGHJeibdLQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7gWO3IeOWZzRXZOia5Xpibh7shcEvLXEhFGOqZdibREDiblhPghgGHJeibdLQ/640?wx_fmt=png&amp;from=appmsg"></p><p data-tool="mdnice编辑器">这部分信息可以从该文章报导的数据库上http://lifeome.net:8080/clca 获取到</p><pre data-tool="mdnice编辑器"><code>    <span># 读入 CN 数据</span><br>    CN_data = readxl::read_xlsx(<span>"Copy_Number_Alteration_20240315.xlsx"</span>)<br>    CN_data = as.data.frame(CN_data)<br>    CN_data$Start = as.numeric(CN_data$Start)<br>    CN_data$End = as.numeric(CN_data$End)<br>    CN_data$CopyNumber = as.numeric(CN_data$CopyNumber)<br></code></pre><pre data-tool="mdnice编辑器"><code>    <span># 读入 SV 数据</span><br>    SV_data = readxl::read_xlsx(<span>"Structure_Variation_20240315.xlsx"</span>)<br>    SV_data = as.data.frame(SV_data)<br>    SV_data$PosA = as.integer(SV_data$PosA)<br>    SV_data$PosB = as.integer(SV_data$PosB)<br>    <span># 这里仔细查看发现 SV 数据的RelatedGeneB(s) 和 GeneB(s).Func 两列的信息应该颠倒过来了</span><br>    head(SV_data,n=<span>20</span>)<br></code></pre><section data-tool="mdnice编辑器"><table><thead><tr><th>CaseID</th><th>ChrA</th><th>PosA</th><th>RelatedGeneA(s)</th><th>GeneA(s).Func</th><th>A.Strand</th><th>ChrB</th><th>PosB</th><th>RelatedGeneB(s)</th><th>GeneB(s).Func</th><th>B.Strand</th><th>Chromoplexy</th><th>Chromothripsis</th></tr></thead><tbody><tr><td>CLCA_0023</td><td>chr1</td><td>8875290</td><td>RERE</td><td>intronic</td><td>+</td><td>chr1</td><td>8877433</td><td>UTR5</td><td>RERE</td><td>-</td><td>.</td><td>.</td></tr><tr><td>CLCA_0023</td><td>chr2</td><td>13604672</td><td>LOC100506474;LINC00276</td><td>intergenic</td><td>-</td><td>chr2</td><td>14858959</td><td>intergenic</td><td>FAM84A;NBAS</td><td>+</td><td>.</td><td>.</td></tr><tr><td>CLCA_0023</td><td>chr3</td><td>68999992</td><td>FAM19A4;EOGT</td><td>intergenic</td><td>+</td><td>chr3</td><td>69000212</td><td>intergenic</td><td>FAM19A4;EOGT</td><td>-</td><td>.</td><td>.</td></tr><tr><td>CLCA_0023</td><td>chr3</td><td>170362147</td><td>LOC101928583</td><td>ncRNA_intronic</td><td>+</td><td>chr3</td><td>170362183</td><td>ncRNA_intronic</td><td>LOC101928583</td><td>-</td><td>.</td><td>.</td></tr><tr><td>CLCA_0023</td><td>chr4</td><td>2495362</td><td>RNF4</td><td>intronic</td><td>+</td><td>chr4</td><td>2495727</td><td>intronic</td><td>RNF4</td><td>-</td><td>.</td><td>.</td></tr><tr><td>CLCA_0023</td><td>chr4</td><td>17266283</td><td>LINC02493;SNORA75B</td><td>intergenic</td><td>+</td><td>chr4</td><td>17266334</td><td>intergenic</td><td>LINC02493;SNORA75B</td><td>-</td><td>.</td><td>.</td></tr><tr><td>CLCA_0023</td><td>chr7</td><td>78027496</td><td>MAGI2</td><td>intronic</td><td>+</td><td>chr7</td><td>78028666</td><td>intronic</td><td>MAGI2</td><td>-</td><td>.</td><td>.</td></tr><tr><td>CLCA_0023</td><td>chr8</td><td>64638461</td><td>LOC102724612;LINC01289</td><td>intergenic</td><td>-</td><td>chr8</td><td>64667679</td><td>intergenic</td><td>LOC102724612;LINC01289</td><td>+</td><td>.</td><td>.</td></tr><tr><td>CLCA_0023</td><td>chr9</td><td>103941193</td><td>PLPPR1</td><td>intronic</td><td>+</td><td>chr9</td><td>103941257</td><td>intronic</td><td>PLPPR1</td><td>-</td><td>.</td><td>.</td></tr><tr><td>CLCA_0023</td><td>chr10</td><td>94663880</td><td>EXOC6</td><td>intronic</td><td>+</td><td>chr10</td><td>94663934</td><td>intronic</td><td>EXOC6</td><td>-</td><td>.</td><td>.</td></tr><tr><td>CLCA_0023</td><td>chr12</td><td>110239299</td><td>TRPV4</td><td>intronic</td><td>-</td><td>chr12</td><td>110243757</td><td>intronic</td><td>TRPV4</td><td>+</td><td>.</td><td>.</td></tr><tr><td>CLCA_0023</td><td>chr13</td><td>77667696</td><td>MYCBP2</td><td>intronic</td><td>+</td><td>chr13</td><td>77683238</td><td>intronic</td><td>MYCBP2</td><td>-</td><td>.</td><td>.</td></tr><tr><td>CLCA_0023</td><td>chr17</td><td>79257653</td><td>SLC38A10</td><td>intronic</td><td>+</td><td>chr17</td><td>79257943</td><td>intronic</td><td>SLC38A10</td><td>-</td><td>.</td><td>.</td></tr><tr><td>CLCA_0023</td><td>chr20</td><td>57261314</td><td>STX16-NPEPL1</td><td>ncRNA_intronic</td><td>+</td><td>chr20</td><td>57263239</td><td>ncRNA_intronic</td><td>STX16-NPEPL1</td><td>-</td><td>.</td><td>.</td></tr><tr><td>CLCA_0023</td><td>chr21</td><td>45289099</td><td>AGPAT3</td><td>intronic</td><td>+</td><td>chr21</td><td>45290317</td><td>intronic</td><td>AGPAT3</td><td>-</td><td>.</td><td>.</td></tr><tr><td>CLCA_0023</td><td>chr22</td><td>24834833</td><td>ADORA2A-AS1;SPECC1L-ADORA2A</td><td>ncRNA_intronic</td><td>+</td><td>chr22</td><td>24911204</td><td>exonic</td><td>UPB1</td><td>-</td><td>.</td><td>.</td></tr><tr><td>CLCA_0023</td><td>chrX</td><td>125812403</td><td>DCAF12L1;PRR32</td><td>intergenic</td><td>+</td><td>chrX</td><td>127402085</td><td>intergenic</td><td>ACTRT1;SMARCA1</td><td>-</td><td>.</td><td>.</td></tr><tr><td>CLCA_0023</td><td>chr6</td><td>46665473</td><td>TDRD6</td><td>intronic</td><td>-</td><td>chr17</td><td>56730435</td><td>intronic</td><td>TEX14</td><td>-</td><td>.</td><td>.</td></tr><tr><td>CLCA_0023</td><td>chr2</td><td>14856430</td><td>FAM84A;NBAS</td><td>intergenic</td><td>-</td><td>chr2</td><td>14917781</td><td>intergenic</td><td>FAM84A;NBAS</td><td>-</td><td>.</td><td>.</td></tr><tr><td>CLCA_0023</td><td>chr2</td><td>14917766</td><td>FAM84A;NBAS</td><td>intergenic</td><td>+</td><td>chr2</td><td>23985393</td><td>intronic</td><td>ATAD2B</td><td>+</td><td>.</td><td>.</td></tr></tbody></table></section><pre data-tool="mdnice编辑器"><code>    <span>## 获取 CLCA_0119 患者的数据</span><br>    CLCA_0119_CN = CN_data[CN_data$CaseID == <span>"CLCA_0119"</span>,<span>2</span>:<span>9</span>]<br>    CLCA_0119_SV = SV_data[SV_data$CaseID == <span>"CLCA_0119"</span>,<span>2</span>:<span>13</span>]<br>    <br>    <span># RCircos plot</span><br>    <span>library</span>(RCircos)<br>    data(UCSC.HG19.Human.CytoBandIdeogram)<br>    RCircos.Set.Core.Components(cyto.info = UCSC.HG19.Human.CytoBandIdeogram,<br>                                chr.exclude=<span>NULL</span>,<br>                                tracks.inside =<span>3</span>, <br>                                tracks.outside = <span>0</span>)  <br>    RCircos.List.Plot.Parameters()<br>    RCircos.Set.Plot.Area()    <br>    RCircos.Chromosome.Ideogram.Plot()<br>    <br>    <span># 添加拷贝数变异信息，散点图</span><br>    RCircos.Scatter.Plot(scatter.data = CLCA_0119_CN, <br>                         data.col=<span>4</span>,<br>                         track.num=<span>1</span>, <br>                         side=<span>"in"</span>, <br>                         by.fold=<span>2</span>);<br>    <span># 添加结构变异曲线</span><br>    <br>    <span>## 添加End，这里只是为了方便可视化，所以 End 是在 start 上加1，没有实际意义的</span><br>    CLCA_0119_SV$EndA = CLCA_0119_SV$PosA+<span>1</span><br>    CLCA_0119_SV$EndB = CLCA_0119_SV$PosB+<span>1</span><br>    <br>    <span>## 添加 Patterns 进行分类，原数据没有，但是文章的RCircos plot 有</span><br>    CLCA_0119_SV$Patterns = <br>      ifelse(CLCA_0119_SV$A.Strand == <span>"+"</span> &amp; CLCA_0119_SV$B.Strand == <span>"+"</span>,<br>             yes = <span>"Head to head(+/+)"</span>,<br>             ifelse(CLCA_0119_SV$A.Strand == <span>"-"</span> &amp; CLCA_0119_SV$B.Strand == <span>"-"</span>,<br>                    yes = <span>"Tail to tail(-/-)"</span>,<br>                    ifelse(CLCA_0119_SV$A.Strand == <span>"+"</span> &amp; CLCA_0119_SV$B.Strand == <span>"-"</span>,<br>                           yes = <span>"Deletion like(+/-)"</span>,<br>                           no = <span>"Duplication like(-/+)"</span>)))<br>    <span>## 添加 PlotColor 设置颜色</span><br>    CLCA_0119_SV$PlotColor = <br>      ifelse(CLCA_0119_SV$A.Strand == <span>"+"</span> &amp; CLCA_0119_SV$B.Strand == <span>"+"</span>,<br>             yes = <span>"black"</span>,<br>             ifelse(CLCA_0119_SV$A.Strand == <span>"-"</span> &amp; CLCA_0119_SV$B.Strand == <span>"-"</span>,<br>                    yes = <span>"#26853A"</span>,<br>                    ifelse(CLCA_0119_SV$A.Strand == <span>"+"</span> &amp; CLCA_0119_SV$B.Strand == <span>"-"</span>,<br>                           yes = <span>"#EE7B1C"</span>,<br>                           no = <span>"#15499D"</span>)))<br>    <span>## 重新进行列排序</span><br>    CLCA_0119_SV_link = CLCA_0119_SV[,c(<span>"ChrA"</span>,<span>"PosA"</span>,<span>"EndA"</span>,<span>"ChrB"</span>,<span>"PosB"</span>,<span>"EndB"</span>,<span>"PlotColor"</span>,<br>                                        <span>"RelatedGeneA(s)"</span>, <span>"GeneA(s).Func"</span>,<span>"A.Strand"</span>,<br>                                        <span>"RelatedGeneB(s)"</span>, <span>"GeneB(s).Func"</span>,<span>"B.Strand"</span>,<br>                                        <span>"Chromoplexy"</span>,<span>"Chromothripsis"</span>,<span>"Patterns"</span> )]<br>                                        <br>    CLCA_0119_SV_link$ChrA = factor(CLCA_0119_SV_link$ChrA,levels = c(paste0(<span>"chr"</span>,c(<span>1</span>:<span>22</span>,<span>"X"</span>,<span>"Y"</span>))))<br>    CLCA_0119_SV_link$ChrB = factor(CLCA_0119_SV_link$ChrB,levels = c(paste0(<span>"chr"</span>,c(<span>1</span>:<span>22</span>,<span>"X"</span>,<span>"Y"</span>))))<br>    <br>    RCircos.Link.Plot(<br>      link.data = CLCA_0119_SV_link,<br>      track.num = <span>2</span>,<br>      <span># by.chromosome = T,</span><br>      <span>#start.pos = 0.8,</span><br>      genomic.columns = <span>3</span>,<br>      is.sorted = <span>T</span><br>      <br>    )<br>    legend(<span>"bottomright"</span>, <br>           <span>#inset=.05, </span><br>           title=<span>"Patterns of SVs"</span>, <br>           legend = c(unique(CLCA_0119_SV_link$Patterns)),<br>           <span>#lty=1, </span><br>           pch=<span>15</span>, bty = <span>"n"</span>,<br>           col=c(<span>"black"</span>, <span>"#26853A"</span>,<span>"#EE7B1C"</span>,<span>"#15499D"</span>))<br><br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100037911" data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7g5mjIfTljI7PdPcvtSWdIpqTyHAvlx9xQbcucdt8JnakXfB3v3wfFIA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="960" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8ia5oDiboBTpqiaRicrmsE7r7g5mjIfTljI7PdPcvtSWdIpqTyHAvlx9xQbcucdt8JnakXfB3v3wfFIA/640?wx_fmt=png&amp;from=appmsg"><span><br></span></figure><pre data-tool="mdnice编辑器"><code>    <span>sessionInfo()</span></code><code>    <br>    <span>## R version 4.3.2 (2023-10-31)</span><br>    <span>## Platform: x86_64-pc-linux-gnu (64-bit)</span><br>    <span>## Running under: Ubuntu 20.04.6 LTS</span><br>    <span>## </span><br>    <span>## Matrix products: default</span><br>    <span>## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so;  LAPACK version 3.9.0</span><br>    <span>## </span><br>    <span>## locale:</span><br>    <span>##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span><br>    <span>##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span><br>    <span>##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span><br>    <span>##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span><br>    <span>##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span><br>    <span>## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span><br>    <span>## </span><br>    <span>## time zone: Asia/Shanghai</span><br>    <span>## tzcode source: system (glibc)</span><br>    <span>## </span><br>    <span>## attached base packages:</span><br>    <span>##  [1] parallel  stats4    grid      stats     graphics  grDevices utils    </span><br>    <span>##  [8] datasets  methods   base     </span><br>    <span>## </span><br>    <span>## other attached packages:</span><br>    <span>##  [1] RCircos_1.2.2                     ggvenn_0.1.10                    </span><br>    <span>##  [3] dplyr_1.1.4                       ggstatsplot_0.12.1               </span><br>    <span>##  [5] purrr_1.0.2                       sigminer_2.3.0                   </span><br>    <span>##  [7] doParallel_1.0.17                 iterators_1.0.14                 </span><br>    <span>##  [9] foreach_1.5.2                     BSgenome.Hsapiens.UCSC.hg19_1.4.3</span><br>    <span>## [11] BSgenome_1.70.2                   rtracklayer_1.62.0               </span><br>    <span>## [13] BiocIO_1.12.0                     Biostrings_2.70.3                </span><br>    <span>## [15] XVector_0.42.0                    GenomicRanges_1.54.1             </span><br>    <span>## [17] GenomeInfoDb_1.38.8               IRanges_2.36.0                   </span><br>    <span>## [19] S4Vectors_0.40.2                  barplot3d_1.0.1                  </span><br>    <span>## [21] NMF_0.26                          synchronicity_1.3.10             </span><br>    <span>## [23] bigmemory_4.6.1                   Biobase_2.62.0                   </span><br>    <span>## [25] BiocGenerics_0.48.1               cluster_2.1.6                    </span><br>    <span>## [27] rngtools_1.5.2                    registry_0.5-1                   </span><br>    <span>## [29] ggVennDiagram_1.4.9               VennDiagram_1.7.3                </span><br>    <span>## [31] futile.logger_1.4.3               ggsci_3.0.0                      </span><br>    <span>## [33] ggrepel_0.9.4                     pheatmap_1.0.12                  </span><br>    <span>## [35] data.table_1.15.4                 tidyr_1.3.0                      </span><br>    <span>## [37] ggpubr_0.6.0                      ggplot2_3.5.0                    </span><br>    <span>## [39] stringr_1.5.1                     maftools_2.18.0                  </span><br>    <span>## </span><br>    <span>## loaded via a namespace (and not attached):</span><br>    <span>##   [1] splines_4.3.2               prismatic_1.1.1            </span><br>    <span>##   [3] bitops_1.0-7                ggplotify_0.1.2            </span><br>    <span>##   [5] tibble_3.2.1                R.oo_1.25.0                </span><br>    <span>##   [7] cellranger_1.1.0            datawizard_0.9.1           </span><br>    <span>##   [9] XML_3.99-0.16.1             lifecycle_1.0.4            </span><br>    <span>##  [11] rstatix_0.7.2               globals_0.16.2             </span><br>    <span>##  [13] lattice_0.22-5              MASS_7.3-60.0.1            </span><br>    <span>##  [15] insight_0.19.7              backports_1.4.1            </span><br>    <span>##  [17] magrittr_2.0.3              rmarkdown_2.25             </span><br>    <span>##  [19] yaml_2.3.8                  cowplot_1.1.2              </span><br>    <span>##  [21] RColorBrewer_1.1-3          multcomp_1.4-25            </span><br>    <span>##  [23] abind_1.4-5                 zlibbioc_1.48.2            </span><br>    <span>##  [25] R.utils_2.12.3              RCurl_1.98-1.14            </span><br>    <span>##  [27] yulab.utils_0.1.4           TH.data_1.1-2              </span><br>    <span>##  [29] sandwich_3.1-0              GenomeInfoDbData_1.2.11    </span><br>    <span>##  [31] correlation_0.8.4           listenv_0.9.0              </span><br>    <span>##  [33] parallelly_1.36.0           codetools_0.2-19           </span><br>    <span>##  [35] DelayedArray_0.28.0         DNAcopy_1.76.0             </span><br>    <span>##  [37] tidyselect_1.2.1            farver_2.1.1               </span><br>    <span>##  [39] matrixStats_1.2.0           GenomicAlignments_1.38.2   </span><br>    <span>##  [41] jsonlite_1.8.8              survival_3.5-7             </span><br>    <span>##  [43] emmeans_1.9.0               tools_4.3.2                </span><br>    <span>##  [45] rio_1.0.1                   Rcpp_1.0.12                </span><br>    <span>##  [47] glue_1.7.0                  SparseArray_1.2.4          </span><br>    <span>##  [49] xfun_0.42                   MatrixGenerics_1.14.0      </span><br>    <span>##  [51] withr_3.0.0                 formatR_1.14               </span><br>    <span>##  [53] BiocManager_1.30.22         fastmap_1.1.1              </span><br>    <span>##  [55] fansi_1.0.6                 digest_0.6.34              </span><br>    <span>##  [57] R6_2.5.1                    gridGraphics_0.5-1         </span><br>    <span>##  [59] estimability_1.4.1          colorspace_2.1-0           </span><br>    <span>##  [61] R.methodsS3_1.8.2           utf8_1.2.4                 </span><br>    <span>##  [63] generics_0.1.3              S4Arrays_1.2.1             </span><br>    <span>##  [65] parameters_0.21.3           pkgconfig_2.0.3            </span><br>    <span>##  [67] gtable_0.3.4                statsExpressions_1.5.2     </span><br>    <span>##  [69] furrr_0.3.1                 htmltools_0.5.7            </span><br>    <span>##  [71] carData_3.0-5               scales_1.3.0               </span><br>    <span>##  [73] bigmemory.sri_0.1.6         knitr_1.45                 </span><br>    <span>##  [75] lambda.r_1.2.4              rstudioapi_0.15.0          </span><br>    <span>##  [77] reshape2_1.4.4              rjson_0.2.21               </span><br>    <span>##  [79] uuid_1.1-1                  coda_0.19-4                </span><br>    <span>##  [81] cachem_1.0.8                zoo_1.8-12                 </span><br>    <span>##  [83] restfulr_0.0.15             pillar_1.9.0               </span><br>    <span>##  [85] vctrs_0.6.5                 car_3.1-2                  </span><br>    <span>##  [87] xtable_1.8-4                paletteer_1.5.0            </span><br>    <span>##  [89] evaluate_0.23               zeallot_0.1.0              </span><br>    <span>##  [91] mvtnorm_1.2-4               cli_3.6.2                  </span><br>    <span>##  [93] compiler_4.3.2              futile.options_1.0.1       </span><br>    <span>##  [95] Rsamtools_2.18.0            rlang_1.1.3                </span><br>    <span>##  [97] crayon_1.5.2                ggsignif_0.6.4             </span><br>    <span>##  [99] labeling_0.4.3              rematch2_2.1.2             </span><br>    <span>## [101] plyr_1.8.9                  forcats_1.0.0              </span><br>    <span>## [103] fs_1.6.3                    stringi_1.8.3              </span><br>    <span>## [105] gridBase_0.4-7              BiocParallel_1.36.0        </span><br>    <span>## [107] munsell_0.5.0               bayestestR_0.13.1          </span><br>    <span>## [109] Matrix_1.6-5                patchwork_1.1.3            </span><br>    <span>## [111] future_1.33.1               SummarizedExperiment_1.32.0</span><br>    <span>## [113] highr_0.10                  broom_1.0.5                </span><br>    <span>## [115] memoise_2.0.1               readxl_1.4.3</span><br></code></pre></section><p><br></p><p><mp-style-type data-value="10000"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/Oq_SUfuoaa6x0zt4y3jEPg",target="_blank" rel="noopener noreferrer">原文链接</a>
