---
title: "单细胞TPM数据遇上Seurat v5"
date: 2024-04-06T23:06:23Z
draft: ["false"]
tags: [
  "fetched",
  "生信星球"
]
categories: ["Acdemic"]
---
单细胞TPM数据遇上Seurat v5 by 生信星球
------
<div><section data-mpa-powered-by="yiban.io"><br></section><section><span>‍</span></section><section><img data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz_png/8oKPbJgbBHrDic8XGmJ0b7oibVJajb0emLBHSvuibGG49ooBgtaAibE3TNJ00iaHviaMtdIKQJfCwtUfuHicDImtSfIxg/640?wx_fmt=png" data-type="png" data-w="64" width="20px" data-imgfileid="100010583" src="https://mmbiz.qpic.cn/mmbiz_png/8oKPbJgbBHrDic8XGmJ0b7oibVJajb0emLBHSvuibGG49ooBgtaAibE3TNJ00iaHviaMtdIKQJfCwtUfuHicDImtSfIxg/640?wx_fmt=png"><img data-imgfileid="100010584" data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz_png/8oKPbJgbBHrDic8XGmJ0b7oibVJajb0emLPukRHCbicy4pNKeEv9qd7aWSfsx7roib2od3xPrRPicw3a0kbn0uQ6JmQ/640?wx_fmt=png" data-type="png" data-w="64" width="20px" src="https://mmbiz.qpic.cn/mmbiz_png/8oKPbJgbBHrDic8XGmJ0b7oibVJajb0emLPukRHCbicy4pNKeEv9qd7aWSfsx7roib2od3xPrRPicw3a0kbn0uQ6JmQ/640?wx_fmt=png"><span> 今天是生信星球陪你的<span><strong>第937天</strong></span></span><img data-imgfileid="100010585" data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz_png/8oKPbJgbBHrDic8XGmJ0b7oibVJajb0emLBHSvuibGG49ooBgtaAibE3TNJ00iaHviaMtdIKQJfCwtUfuHicDImtSfIxg/640?wx_fmt=png" data-type="png" data-w="64" width="20px" src="https://mmbiz.qpic.cn/mmbiz_png/8oKPbJgbBHrDic8XGmJ0b7oibVJajb0emLBHSvuibGG49ooBgtaAibE3TNJ00iaHviaMtdIKQJfCwtUfuHicDImtSfIxg/640?wx_fmt=png"></section><hr><section><span><span>   </span><span>大神一句话，菜鸟跑半年。我不是大神，但我可以缩短你走弯路的半年~</span></span></section><section><span>   就像歌儿唱的那样，如果你不知道该往哪儿走，就留在这学点生信好不好~</span></section><p><span>   这里有豆豆和花花的学习历程，从新手到进阶，生信路上有你有我！</span></p><section><h4><span><span> </span>0.单细胞的TPM数据</span></h4><p>众所周知，正常的单细胞数据应该给我们提供count，但偏偏有一些数据是特殊的，给的是tpm数据。准确来说应该是log2(tpm/10+1)</p><blockquote><p>为啥除以10，这么解释的：<br>TPM values were divided by 10 since we estimate the complexity of our single cell libraries to be on the order of 100,000 transcripts and would like to avoid counting each transcript ~10 times, as would be the case with TPM, which may inflate the difference between the expression level of a gene in cells in which the gene is detected and those in which it is not detected.</p></blockquote><p>查阅seurat的github，对tpm数据的说明是</p><blockquote><p>If you have TPM data, you can simply manually log transform the gene expression matrix in the object@data slot before scaling the data. You have to replace your object@data slot with the desired gene expression matrix as follows: sce.all@data = log(x = norm + 1))</p></blockquote><p>在Seurat v5 版本之前，直接跳过NormalizeData 这个函数即可：</p><blockquote><p>Yes, run CreateSeuratObject() and skip the data normalization step since you’re starting with normalized data</p></blockquote><p>但是呢，当我们使用Seurat v5 ，粗暴的跳过NormalizeData 是会报错的。例如：</p><h4><span><span> </span>1.读取数据</span></h4><pre><code>dat = data.table::fread("GSE72056_melanoma_single_cell_revised_v2.txt.gz",data.table = F)<br>dat[1:4,1:4]<br><span><br>#</span><span>#                                                           Cell</span><br><span>#</span><span># 1                                                        tumor</span><br><span>#</span><span># 2                           malignant(1=no,2=yes,0=unresolved)</span><br><span>#</span><span># 3 non-malignant cell type (1=T,2=B,3=Macro.4=Endo.,5=CAF;6=NK)</span><br><span>#</span><span># 4                                                     C9orf152</span><br><span>#</span><span>#   Cy72_CD45_H02_S758_comb CY58_1_CD45_B02_S974_comb Cy71_CD45_D08_S524_comb</span><br><span>#</span><span># 1                      72                        58                      71</span><br><span>#</span><span># 2                       1                         1                       2</span><br><span>#</span><span># 3                       2                         1                       0</span><br><span>#</span><span># 4                       0                         0                       0</span><br><br>dat = dat[-(1:3),]<br>library(tidyverse)<br>dat = distinct(dat,Cell,.keep_all = T)<br>library(tibble)<br>dat = column_to_rownames(dat,"Cell")<br>dat[1:4,1:4]<br><span><br>#</span><span>#          Cy72_CD45_H02_S758_comb CY58_1_CD45_B02_S974_comb</span><br><span>#</span><span># C9orf152                  0.0000                    0.0000</span><br><span>#</span><span># RPS11                     9.2172                    8.3745</span><br><span>#</span><span># ELMO2                     0.0000                    0.0000</span><br><span>#</span><span># CREB3L1                   0.0000                    0.0000</span><br><span>#</span><span>#          Cy71_CD45_D08_S524_comb Cy81_FNA_CD45_B01_S301_comb</span><br><span>#</span><span># C9orf152                  0.0000                      0.0000</span><br><span>#</span><span># RPS11                     9.3130                      7.8876</span><br><span>#</span><span># ELMO2                     2.1263                      0.0000</span><br><span>#</span><span># CREB3L1                   0.0000                      0.0000</span><br></code></pre><h4><span><span> </span>2.正常创建对象</span></h4><pre><code>library(Seurat)<br>sce.all <span>&lt;<span>-</span> <span>CreateSeuratObject</span>(<span>counts</span> = <span>dat,</span> <br>                           <span>project</span> = <span>"a"</span>, <br>                           <span>min.cells</span> = <span>3,</span> <br>                           <span>min.features</span> = <span>200)</span><br></span></code></pre><h4><span><span> </span>3.质控</span></h4><pre><code>sce.all[[<span>"percent.mt"</span>]] &lt;- PercentageFeatureSet(sce.all, pattern = <span>"^MT-"</span>)<br>head(sce.all@meta.data, <span>3</span>)<br><br><span>##                           orig.ident nCount_RNA nFeature_RNA percent.mt</span><br><span>## Cy72_CD45_H02_S758_comb            a   7143.363         3365          0</span><br><span>## CY58_1_CD45_B02_S974_comb          a   8915.833         3637          0</span><br><span>## Cy71_CD45_D08_S524_comb            a  12578.256         4660          0</span><br><br>VlnPlot(sce.all, <br>        features = c(<span>"nFeature_RNA"</span>,<br>                     <span>"nCount_RNA"</span>, <br>                     <span>"percent.mt"</span>), <br>        ncol = <span>3</span>,pt.size = <span>0.5</span>)<br></code></pre><figure><img data-imgfileid="100010600" data-ratio="0.7138888888888889" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/8oKPbJgbBHrBiciadYeIDMTzRywRO3OOpDGTlVIM90Bc8RANI5PGf75uEwrFAQWjsibrJkSdF71sVCpUEDjdkKo2A/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="1080" title="image.png" src="https://mmbiz.qpic.cn/mmbiz_jpg/8oKPbJgbBHrBiciadYeIDMTzRywRO3OOpDGTlVIM90Bc8RANI5PGf75uEwrFAQWjsibrJkSdF71sVCpUEDjdkKo2A/640?wx_fmt=other&amp;from=appmsg"><figcaption>image.png</figcaption></figure><p>线粒体基因比例为0，另两个图也没有很利群的值，这个数据可以不过滤。</p><h4><span><span> </span>4.企图跳过NormalizeData</span></h4><p>当我们企图跳过NormalizeData就会报错</p><pre><code><span>sce</span>.<span>all</span> = sce.<span>all</span> %&gt;% <br>  <span>#NormalizeData() %&gt;%  </span><br>  <span>FindVariableFeatures</span>() %&gt;%  <br>  <span>ScaleData</span>()<br><br><span>## Error in `ScaleData()`:</span><br><span>## ! No layer matching pattern 'data' found. Please run NormalizeData and retry</span><br></code></pre><p>明明4版本这样做是木有问题的嘞。</p><h4><span><span> </span>5.查呀！</span></h4><p>各种细枝末节都可以在seurat github页面上找到答案：</p><blockquote><p>In the previous versions, “data” was automatically populated with the raw counts prior to running NormalizeData, which is no longer the case in v5. Therefore, if you run ScaleData and the “data” layer does not exist, this will not work.</p><p>If you want to run ScaleData on the raw counts (which I wouldn’t recommend doing in most cases), you could do the following: sce.all_small &lt;- ScaleData(sce.all_small, layer = “counts”) https://github.com/satijalab/seurat/issues/8003</p></blockquote><p>所以解决办法就是加上个参数layer = “counts”。虽然叫counts但我们创建对象的时候放了个什么东西进去自己心里没点数嘛？那其实是tpm😄。</p><pre><code>sce.all = sce.all %&gt;% <br><span>  #</span><span>NormalizeData() %&gt;%  </span><br>  FindVariableFeatures() %&gt;%  <br>  ScaleData(layer = "counts") %&gt;%<br>  RunPCA(pc.genes = pbmc@var.genes)  %&gt;%<br>  FindNeighbors(dims = 1:15) %&gt;% <br>  FindClusters(resolution = 0.5) %&gt;% <br>  RunUMAP(dims = 1:15) <br><span><br>#</span><span># Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck</span><br><span>#</span><span># </span><br><span>#</span><span># Number of nodes: 4645</span><br><span>#</span><span># Number of edges: 150415</span><br><span>#</span><span># </span><br><span>#</span><span># Running Louvain algorithm...</span><br><span>#</span><span># Maximum modularity in 10 random starts: 0.9164</span><br><span>#</span><span># Number of communities: 16</span><br><span>#</span><span># Elapsed time: 0 seconds</span><br></code></pre><p>后面就可以正常进行啦，顺便把这个数据做完</p><h4><span><span> </span>6.还能做个singleR</span></h4><pre><code># 注释<br><span>library</span>(celldex)<br><span>library</span>(SingleR)<br>#ref &lt;- celldex::HumanPrimaryCellAtlasData()<br>#因为下载太慢，使用了提前存好的本地数据<br>ref &lt;- <span>get</span>(load(<span>"ref_Human_all.RData"</span>))<br><span>library</span>(BiocParallel)<br>pred.scRNA &lt;- SingleR(test = sce.all<span>@assays</span>$RNA$counts, <br>                      ref = ref,<br>                      labels = ref$label.main, <br>                      clusters = sce.all<span>@active</span>.ident)<br>pred.scRNA$pruned.labels<br><br>##  [<span>1</span>] <span>"T_cells"</span>           <span>"T_cells"</span>           <span>"B_cell"</span>           <br>##  [<span>4</span>] <span>"Neurons"</span>           <span>"Neurons"</span>           <span>"T_cells"</span>          <br>##  [<span>7</span>] <span>"Monocyte"</span>          <span>"T_cells"</span>           <span>"Tissue_stem_cells"</span><br>## [<span>10</span>] <span>"Neurons"</span>           <span>"Neurons"</span>           <span>"Endothelial_cells"</span><br>## [<span>13</span>] <span>"T_cells"</span>           <span>"Fibroblasts"</span>       <span>"T_cells"</span>          <br>## [<span>16</span>] <span>"Neurons"</span><br><br>plotScoreHeatmap(pred.scRNA, clusters=pred.scRNA<span>@rownames</span>, fontsize.row = <span>9</span>,show_colnames = T)<br></code></pre><figure><img data-imgfileid="100010601" data-ratio="0.7138888888888889" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/8oKPbJgbBHrBiciadYeIDMTzRywRO3OOpDsoicyvEISiaUf9pMBFia0ZmU844OC0YCIfWEnBaXKor9WRJZbe5eXdS8Q/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="1080" title="" src="https://mmbiz.qpic.cn/mmbiz_jpg/8oKPbJgbBHrBiciadYeIDMTzRywRO3OOpDsoicyvEISiaUf9pMBFia0ZmU844OC0YCIfWEnBaXKor9WRJZbe5eXdS8Q/640?wx_fmt=other&amp;from=appmsg"><figcaption></figcaption></figure><pre><code>new.cluster.ids &lt;- pred.scRNA$pruned.labels<br>names(new.cluster.ids) &lt;- levels(sce.all)<br>levels(sce.all)<br><br><span>##  [1] <span>"0"</span>  <span>"1"</span>  <span>"2"</span>  <span>"3"</span>  <span>"4"</span>  <span>"5"</span>  <span>"6"</span>  <span>"7"</span>  <span>"8"</span>  <span>"9"</span>  <span>"10"</span> <span>"11"</span> <span>"12"</span> <span>"13"</span> <span>"14"</span></span><br><span>## [16] <span>"15"</span></span><br><br>sce.all &lt;- RenameIdents(sce.all,new.cluster.ids)<br>levels(sce.all)<br><br><span>## [1] <span>"T_cells"</span>           <span>"B_cell"</span>            <span>"Neurons"</span>          </span><br><span>## [4] <span>"Monocyte"</span>          <span>"Tissue_stem_cells"</span> <span>"Endothelial_cells"</span></span><br><span>## [7] <span>"Fibroblasts"</span></span><br><br>p = UMAPPlot(object = sce.all, pt.size = <span>0.5</span>, label = <span>TRUE</span>)<br>p<br></code></pre><figure><img data-imgfileid="100010599" data-ratio="0.7138888888888889" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/8oKPbJgbBHrBiciadYeIDMTzRywRO3OOpDNASnaeBot3NFr0iauK6nLsQo668rBTBwzfz0aVNUEMr9hicKKxpcZLcg/640?wx_fmt=other&amp;from=appmsg" data-type="other" data-w="1080" title="" src="https://mmbiz.qpic.cn/mmbiz_jpg/8oKPbJgbBHrBiciadYeIDMTzRywRO3OOpDNASnaeBot3NFr0iauK6nLsQo668rBTBwzfz0aVNUEMr9hicKKxpcZLcg/640?wx_fmt=other&amp;from=appmsg"><figcaption></figcaption></figure></section><section><blockquote><section><span>如果因为代码看不懂，而跟不上正文的节奏，可以来找我，相当于来一个新手保护期。我的课程都是循环开课。下一期的时间，点进去咨询微信咯</span><br></section><section><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzU4NjU4ODQ2MQ==&amp;mid=2247494223&amp;idx=1&amp;sn=956c0412cf4235ad73269a1c219cd90c&amp;chksm=fdfba20dca8c2b1b8cfafefb05d6fa736f2f126c8883086f8963e9aafeb2af39791e2e56e254&amp;scene=21#wechat_redirect" textvalue="生信分析‍直播课程" linktype="text" imgurl="" imgdata="null" data-itemshowtype="11" tab="innerlink" data-linktype="2">生信分析直播课程</a></section><section><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzU4NjU4ODQ2MQ==&amp;mid=2247494009&amp;idx=1&amp;sn=035d253f973f65c0d1069e515eec2ce8&amp;chksm=fdfba13bca8c282de5a9222a2ab8bc0caa2ddf60cab543c2181e6d4a90b553f4dcde5f154adf&amp;scene=21#wechat_redirect" textvalue="生信新手保护学习小组‍" linktype="text" imgurl="" imgdata="null" data-itemshowtype="11" tab="innerlink" data-linktype="2">生信新手保护学习小组</a><br></section></blockquote></section><section><section><br></section></section><section><br></section><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/uFLgc62O-ER28vkNHJJWig",target="_blank" rel="noopener noreferrer">原文链接</a>
