---
title: "日光性皮炎和银屑病单细胞数据集复现"
date: 2024-04-04T03:35:35Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Acdemic"]
---
日光性皮炎和银屑病单细胞数据集复现 by 生信菜鸟团
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><blockquote data-tool="mdnice编辑器"><span></span><p>这周推文更新一篇银屑病相关的单细胞转录组数据的文献复现。使用的SeuratV5版本R包及其常规流程代码。</p></blockquote><h4 data-tool="mdnice编辑器"><span></span><span>文章标题：</span><span></span></h4><h4 data-tool="mdnice编辑器"><span></span><span>Single-cell transcriptomics suggest distinct upstream drivers of IL-17A/F in hidradenitis versus psoriasis  （J ALLERGY CLIN IMMUNOL）</span><span></span></h4><p data-tool="mdnice编辑器">单细胞转录组学表明，IL-17A/F 在日光性皮炎和银屑病中具有不同的上游驱动力。</p><figure data-tool="mdnice编辑器"><img data-imgfileid="100037411" data-ratio="0.3212962962962963" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los88qwNHPY4vibic2ayh6Rd1BEiaNKH3U7KOKEoox3E80oUiafmffhgD8eIMm2MjnaMy5lic1jRjfFbic5dg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los88qwNHPY4vibic2ayh6Rd1BEiaNKH3U7KOKEoox3E80oUiafmffhgD8eIMm2MjnaMy5lic1jRjfFbic5dg/640?wx_fmt=png&amp;from=appmsg"></figure><h4 data-tool="mdnice编辑器"><span></span><span>文章简介：</span><span></span></h4><p data-tool="mdnice编辑器"><strong>背景：</strong> 越来越多的证据表明 17T (T17) 细胞和增加的 IL-17 在驱动化脓性汗腺炎 (HS) 病变发展中发挥关键作用。以前在银屑病中使用的生物制剂可阻断 IL-17A 和/或的信号传导 或 IL-17F 同工型已被重新用于治疗 HS。</p><p data-tool="mdnice编辑器"><strong>目的：</strong> 研究旨在比较 HS T17 细胞的转录组与银屑病 T17 细胞的转录组，以及它们与邻近免疫细胞亚群的配体-受体相互作用。</p><p data-tool="mdnice编辑器"><strong>方法：</strong> 将来自 8 个去顶手术 HS 皮肤样本（包括真皮隧道）的 12,300 个皮肤免疫细胞的单细胞数据与银屑病皮肤（来自 11 个样本的 19,525 个细胞）和对照皮肤（来自 10 个样本的 11,920 个细胞）的单细胞数据进行比较。</p><p data-tool="mdnice编辑器"><strong>数据集：</strong> https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE220116</p><p data-tool="mdnice编辑器">所用数据集包括8个HIS，11个Psoriasis，10个control样本。</p><figure data-tool="mdnice编辑器"><img data-imgfileid="100037408" data-ratio="2.1215277777777777" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los88qwNHPY4vibic2ayh6Rd1BERZE2h1IsJ8hq4NfxoahosV79EWibew0TV4SuWYzAvpQDlhJYOShSrvQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="288" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los88qwNHPY4vibic2ayh6Rd1BERZE2h1IsJ8hq4NfxoahosV79EWibew0TV4SuWYzAvpQDlhJYOShSrvQ/640?wx_fmt=png&amp;from=appmsg"></figure><h4 data-tool="mdnice编辑器"><span></span><span>复现的图：<span>文章主图直接展示的是HS组的亚群细分，但是在做亚群细分之前要先做第一层次的细胞分群，所以这周我先复现第一层次的单细胞转录组数据。</span></span></h4><p data-tool="mdnice编辑器"><img data-imgfileid="100037410" data-ratio="1.2064896755162242" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los88qwNHPY4vibic2ayh6Rd1BEx9OPbHpm9n9co1yaFCUVXvrq9IunwzHtllpic2z4w0ibTt1vUXUUibhLA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="678" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los88qwNHPY4vibic2ayh6Rd1BEx9OPbHpm9n9co1yaFCUVXvrq9IunwzHtllpic2z4w0ibTt1vUXUUibhLA/640?wx_fmt=png&amp;from=appmsg"><span>代码</span></p><p data-tool="mdnice编辑器"><strong>导入数据（10X多样本数据）</strong></p><pre data-tool="mdnice编辑器"><span></span><code>rm(list=ls())<br>options(stringsAsFactors = <span>F</span>) <br><span>source</span>(<span>'scRNA_scripts/lib.R'</span>)<br><span>###### step1:导入数据 ######   </span><br><span># https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE220116</span><br>dir=<span>'GSE220116/'</span> <br>fs = list.files(<span>'./GSE220116/'</span>,pattern = <span>'^GSM'</span>)<br><span>#执行这一步需要解压tar -xvf</span><br>samples=str_split(fs,<span>'_'</span>,simplify = <span>T</span>)[,<span>1</span>]<br>samples <br>fs<br>lapply(unique(samples),<span>function</span>(x){<br>  y=fs[grepl(x,fs)]<br>  folder=paste0(<span>"GSE220116/"</span>, str_split(y[<span>1</span>],<span>'_'</span>,simplify = <span>T</span>)[,<span>1</span>])<br>  dir.create(folder,recursive = <span>T</span>)<br>  <span>#为每个样本创建子文件夹</span><br>  file.rename(paste0(<span>"GSE220116/"</span>,y[<span>1</span>]),file.path(folder,<span>"barcodes.tsv.gz"</span>))<br>  <span>#重命名文件，并移动到相应的子文件夹里</span><br>  file.rename(paste0(<span>"GSE220116/"</span>,y[<span>2</span>]),file.path(folder,<span>"features.tsv.gz"</span>))<br>  file.rename(paste0(<span>"GSE220116/"</span>,y[<span>3</span>]),file.path(folder,<span>"matrix.mtx.gz"</span>))<br>})<br>tmp = list.dirs(<span>'GSE220116'</span>)[-<span>1</span>]<br>tmp<br>names(tmp)&lt;-list.dirs(<span>"GSE220116"</span>,recursive = <span>F</span>)#给orig.ident名字<br>ct = Read10X(tmp) <br>ct<br>sce.all=CreateSeuratObject(counts = ct, <br>                           min.cells = <span>5</span>,<br>                           min.features = <span>300</span>,)<br>sce.all<br>as.data.frame(sce.all@assays$RNA$counts[<span>1</span>:<span>10</span>, <span>1</span>:<span>2</span>])<br>head(sce.all@meta.data, <span>10</span>)<br>table(sce.all@meta.data$orig.ident) <br><span>#HIS1-8；Psoriasis：9-26；Control 27-36</span><br>sce.all$group&lt;-ifelse(grepl(<span>"GSM6840117|118|119|120|121|122|123|124"</span>,sce.all$orig.ident),<span>"HIS"</span>,<br>                  ifelse(grepl(<span>"GSM6840143|144|145|146|147|148|149|150|151|152"</span>,sce.all$orig.ident),<span>"Control"</span>,<br>                                <span>"Psoriasis"</span>))<br>table(sce.all$group)<br>getwd()<br></code></pre><p data-tool="mdnice编辑器"><strong>1. 质控</strong></p><pre data-tool="mdnice编辑器"><span></span><code><span>###### step2:QC质控 ######</span><br>dir.create(<span>"./1-QC"</span>)<br>setwd(<span>"./1-QC"</span>)<br><span># 如果过滤的太狠，就需要去修改这个过滤代码</span><br><span>source</span>(<span>'../scRNA_scripts/qc.R'</span>)<br>sce.all.filt = basic_qc(sce.all)<br>print(dim(sce.all))<br>print(dim(sce.all.filt))<br>setwd(<span>'../'</span>)<br>table(sce.all.filt$group)<br></code></pre><p data-tool="mdnice编辑器"><strong>qc.R 代码</strong></p><pre data-tool="mdnice编辑器"><span></span><code>basic_qc &lt;- <span>function</span>(input_sce){<br>  <span>#计算线粒体基因比例</span><br>  mito_genes=rownames(input_sce)[grep(<span>"^MT-"</span>, rownames(input_sce),ignore.case = <span>T</span>)] <br>  print(mito_genes) <span>#可能是13个线粒体基因</span><br>  <span>#input_sce=PercentageFeatureSet(input_sce, "^MT-", col.name = "percent_mito")</span><br>  input_sce=PercentageFeatureSet(input_sce, features = mito_genes, col.name = <span>"percent_mito"</span>)<br>  fivenum(input_sce@meta.data$percent_mito)<br>  <span>#计算核糖体基因比例</span><br>  ribo_genes=rownames(input_sce)[grep(<span>"^Rp[sl]"</span>, rownames(input_sce),ignore.case = <span>T</span>)]<br>  print(ribo_genes)<br>  input_sce=PercentageFeatureSet(input_sce,  features = ribo_genes, col.name = <span>"percent_ribo"</span>)<br>  fivenum(input_sce@meta.data$percent_ribo)<br>  <span>#计算红血细胞基因比例</span><br>  Hb_genes=rownames(input_sce)[grep(<span>"^Hb[^(p)]"</span>, rownames(input_sce),ignore.case = <span>T</span>)]<br>  print(Hb_genes)<br>  input_sce=PercentageFeatureSet(input_sce,  features = Hb_genes,col.name = <span>"percent_hb"</span>)<br>  fivenum(input_sce@meta.data$percent_hb)  <br>  <span>#可视化细胞的上述比例情况</span><br>  feats &lt;- c(<span>"nFeature_RNA"</span>, <span>"nCount_RNA"</span>, <span>"percent_mito"</span>, <span>"percent_ribo"</span>, <span>"percent_hb"</span>)<br>  feats &lt;- c(<span>"nFeature_RNA"</span>, <span>"nCount_RNA"</span>)<br>  p1=VlnPlot(input_sce, group.by = <span>"orig.ident"</span>, features = feats, pt.size = <span>0</span>, ncol = <span>2</span>) + <br>    NoLegend()<br>  p1 <br>  w=length(unique(input_sce$orig.ident))/<span>3</span>+<span>5</span>;w<br>  ggsave(filename=<span>"Vlnplot1.pdf"</span>,plot=p1,width = w,height = <span>5</span>)<br>  <br>  feats &lt;- c(<span>"percent_mito"</span>, <span>"percent_ribo"</span>, <span>"percent_hb"</span>)<br>  p2=VlnPlot(input_sce, group.by = <span>"orig.ident"</span>, features = feats, pt.size = <span>0</span>, ncol = <span>3</span>, same.y.lims=<span>T</span>) + <br>    scale_y_continuous(breaks=seq(<span>0</span>, <span>100</span>, <span>5</span>)) +<br>    NoLegend()<br>  p2 <br>  w=length(unique(input_sce$orig.ident))/<span>2</span>+<span>5</span>;w<br>  ggsave(filename=<span>"Vlnplot2.pdf"</span>,plot=p2,width = w,height = <span>5</span>)<br>  <br>  p3=FeatureScatter(input_sce, <span>"nCount_RNA"</span>, <span>"nFeature_RNA"</span>, group.by = <span>"orig.ident"</span>, pt.size = <span>0.5</span>)<br>  ggsave(filename=<span>"Scatterplot.pdf"</span>,plot=p3)<br>  <br>  <span>#根据上述指标，过滤低质量细胞/基因</span><br>  <span>#过滤指标1:最少表达基因数的细胞&amp;最少表达细胞数的基因</span><br>  <span># 一般来说，在CreateSeuratObject的时候已经是进行了这个过滤操作</span><br>  <span># 如果后期看到了自己的单细胞降维聚类分群结果很诡异，就可以回过头来看质量控制环节</span><br>  <span># 先走默认流程即可</span><br>  <span>if</span>(<span>F</span>){<br>    selected_c &lt;- WhichCells(input_sce, expression = nFeature_RNA &gt; <span>500</span>)<br>    selected_f &lt;- rownames(input_sce)[Matrix::rowSums(input_sce@assays$RNA$counts &gt; <span>0</span> ) &gt; <span>3</span>]<br>    input_sce.filt &lt;- subset(input_sce, features = selected_f, cells = selected_c)<br>    dim(input_sce) <br>    dim(input_sce.filt) <br>  }<br>  <br>  input_sce.filt =  input_sce<br>  <br>  <span># 这里的C 这个矩阵，有一点大，可以考虑随抽样 </span><br>  C=subset(input_sce.filt,downsample=<span>100</span>)@assays$RNA@layers$counts<br>  dim(C)<br>  C=Matrix::t(Matrix::t(C)/Matrix::colSums(C)) * <span>100</span><br><br>  most_expressed &lt;- order(apply(C, <span>1</span>, median), decreasing = <span>T</span>)[<span>50</span>:<span>1</span>]<br>  <br>  pdf(<span>"TOP50_most_expressed_gene.pdf"</span>,width=<span>14</span>)<br>  boxplot(as.matrix(Matrix::t(C[most_expressed, ])),<br>          cex = <span>0.1</span>, las = <span>1</span>, <br>          xlab = <span>"% total count per cell"</span>, <br>          col = (scales::hue_pal())(<span>50</span>)[<span>50</span>:<span>1</span>], <br>          horizontal = <span>TRUE</span>)<br>  dev.off()<br>  rm(C)<br>  <br>  <span>#过滤指标2:线粒体/核糖体基因比例(根据上面的violin图)</span><br>  selected_mito &lt;- WhichCells(input_sce.filt, expression = percent_mito &lt; <span>25</span>)<br>  selected_ribo &lt;- WhichCells(input_sce.filt, expression = percent_ribo &gt; <span>3</span>)<br>  selected_hb &lt;- WhichCells(input_sce.filt, expression = percent_hb &lt; <span>1</span> )<br>  length(selected_hb)<br>  length(selected_ribo)<br>  length(selected_mito)<br>  <br>  input_sce.filt &lt;- subset(input_sce.filt, cells = selected_mito)<br>  input_sce.filt &lt;- subset(input_sce.filt, cells = selected_ribo)<br>  input_sce.filt &lt;- subset(input_sce.filt, cells = selected_hb)<br>  dim(input_sce.filt)<br>  <br>  table(input_sce.filt$orig.ident) <br>  <br>  <span>#可视化过滤后的情况</span><br>  feats &lt;- c(<span>"nFeature_RNA"</span>, <span>"nCount_RNA"</span>)<br>  p1_filtered=VlnPlot(input_sce.filt, group.by = <span>"orig.ident"</span>, features = feats, pt.size = <span>0</span>, ncol = <span>2</span>) + <br>    NoLegend()<br>  w=length(unique(input_sce.filt$orig.ident))/<span>3</span>+<span>5</span>;w <br>  ggsave(filename=<span>"Vlnplot1_filtered.pdf"</span>,plot=p1_filtered,width = w,height = <span>5</span>)<br>  <br>  feats &lt;- c(<span>"percent_mito"</span>, <span>"percent_ribo"</span>, <span>"percent_hb"</span>)<br>  p2_filtered=VlnPlot(input_sce.filt, group.by = <span>"orig.ident"</span>, features = feats, pt.size = <span>0</span>, ncol = <span>3</span>) + <br>    NoLegend()<br>  w=length(unique(input_sce.filt$orig.ident))/<span>2</span>+<span>5</span>;w <br>  ggsave(filename=<span>"Vlnplot2_filtered.pdf"</span>,plot=p2_filtered,width = w,height = <span>5</span>) <br>  <span>return</span>(input_sce.filt) <br>  <br>}<br></code></pre><p data-tool="mdnice编辑器"><strong>2. harmony</strong></p><pre data-tool="mdnice编辑器"><span></span><code><span>#物种是mouse，human要修改sp</span><br>sp=<span>'human'</span><br><span># install.packages("Matrix", type = "source")</span><br><span># install.packages("irlba", type = "source")</span><br><span>###### step3: harmony整合多个单细胞样品 ######</span><br>dir.create(<span>"2-harmony"</span>)<br>getwd()<br>setwd(<span>"2-harmony"</span>)<br><span>source</span>(<span>'../scRNA_scripts/harmony.R'</span>)<br><span># 默认 ScaleData 没有添加"nCount_RNA", "nFeature_RNA"</span><br><span># 默认的</span><br>sce.all.int = run_harmony(sce.all.filt)<br>setwd(<span>'../'</span>)<br></code></pre><p data-tool="mdnice编辑器"><strong>harmony.R的代码</strong></p><pre data-tool="mdnice编辑器"><span></span><code>run_harmony &lt;- <span>function</span>(input_sce){<br>  <br>  print(dim(input_sce))<br>  input_sce &lt;- NormalizeData(input_sce, <br>                             normalization.method = <span>"LogNormalize"</span>,<br>                             scale.factor = <span>1e4</span>) <br>  input_sce &lt;- FindVariableFeatures(input_sce)<br>  input_sce &lt;- ScaleData(input_sce)<br>  input_sce &lt;- RunPCA(input_sce, features = VariableFeatures(object = input_sce))<br>  seuratObj &lt;- RunHarmony(input_sce, <span>"orig.ident"</span>)<br>  names(seuratObj@reductions)<br>  seuratObj &lt;- RunUMAP(seuratObj,  dims = <span>1</span>:<span>15</span>, <br>                       reduction = <span>"harmony"</span>) <br>  input_sce=seuratObj<br>  input_sce &lt;- FindNeighbors(input_sce, reduction = <span>"harmony"</span>,<br>                             dims = <span>1</span>:<span>15</span>) <br>  input_sce.all=input_sce<br>  <br>  <span>#设置不同的分辨率，观察分群效果(选择哪一个？)</span><br>  <span>for</span> (res <span>in</span> c(<span>0.01</span>, <span>0.05</span>, <span>0.1</span>, <span>0.2</span>, <span>0.3</span>, <span>0.5</span>,<span>0.8</span>,<span>1</span>)) {<br>    input_sce.all=FindClusters(input_sce.all, <span>#graph.name = "CCA_snn", </span><br>                               resolution = res, algorithm = <span>1</span>)<br>  }<br>  colnames(input_sce.all@meta.data)<br>  apply(input_sce.all@meta.data[,grep(<span>"RNA_snn"</span>,colnames(input_sce.all@meta.data))],<span>2</span>,table)<br>  <br>  p1_dim=plot_grid(ncol = <span>3</span>, DimPlot(input_sce.all, reduction = <span>"umap"</span>, group.by = <span>"RNA_snn_res.0.01"</span>) + <br>                     ggtitle(<span>"louvain_0.01"</span>), DimPlot(input_sce.all, reduction = <span>"umap"</span>, group.by = <span>"RNA_snn_res.0.1"</span>) + <br>                     ggtitle(<span>"louvain_0.1"</span>), DimPlot(input_sce.all, reduction = <span>"umap"</span>, group.by = <span>"RNA_snn_res.0.2"</span>) + <br>                     ggtitle(<span>"louvain_0.2"</span>))<br>  ggsave(plot=p1_dim, filename=<span>"Dimplot_diff_resolution_low.pdf"</span>,width = <span>14</span>)<br>  <br>  p1_dim=plot_grid(ncol = <span>3</span>, DimPlot(input_sce.all, reduction = <span>"umap"</span>, group.by = <span>"RNA_snn_res.0.8"</span>) + <br>                     ggtitle(<span>"louvain_0.8"</span>), DimPlot(input_sce.all, reduction = <span>"umap"</span>, group.by = <span>"RNA_snn_res.1"</span>) + <br>                     ggtitle(<span>"louvain_1"</span>), DimPlot(input_sce.all, reduction = <span>"umap"</span>, group.by = <span>"RNA_snn_res.0.3"</span>) + <br>                     ggtitle(<span>"louvain_0.3"</span>))<br>  ggsave(plot=p1_dim, filename=<span>"Dimplot_diff_resolution_high.pdf"</span>,width = <span>18</span>)<br>  <br>  p2_tree=clustree(input_sce.all@meta.data, prefix = <span>"RNA_snn_res."</span>)<br>  ggsave(plot=p2_tree, filename=<span>"Tree_diff_resolution.pdf"</span>)<br>  table(input_sce.all@active.ident) <br>  saveRDS(input_sce.all, <span>"sce.all_int.rds"</span>)<br>  <span>return</span>(input_sce.all) <br>}<br></code></pre><p data-tool="mdnice编辑器"><strong>3. 降维分群</strong></p><pre data-tool="mdnice编辑器"><span></span><code><span>###### step4: 降维聚类分群和看标记基因库 ######</span><br><span># 原则上分辨率是需要自己肉眼判断，取决于个人经验</span><br><span># 为了省力，我们直接看  0.1和0.8即可</span><br>table(Idents(sce.all.int))<br>table(sce.all.int$seurat_clusters)<br>table(sce.all.int$RNA_snn_res.0.1) <br>table(sce.all.int$RNA_snn_res.0.8) <br><br>p_umap=DimPlot(sce.all.int, reduction = <span>"umap"</span>, group.by = <span>"group"</span>,label = <span>T</span>,label.box = <span>T</span>);p_umap<br>getwd()<br>dir.create(<span>'check-by-0.8'</span>)<br>setwd(<span>'check-by-0.8'</span>)<br>sel.clust = <span>"RNA_snn_res.0.8"</span><br>sce.all.int &lt;- SetIdent(sce.all.int, value = sel.clust)<br>table(sce.all.int@active.ident) <br><span>source</span>(<span>'../scRNA_scripts/check-all-markers.R'</span>)<br>setwd(<span>'../'</span>) <br>getwd()<br></code></pre><p data-tool="mdnice编辑器"><strong>所用marker gene</strong></p><pre data-tool="mdnice编辑器"><span></span><code>GSE220116gene&lt;-c(<span>"KLRB1,GNLY,CD3D,TRAC,TRBC1,CD8A,GZMK,IL2RA,FOXP3,CTLA4,CD79A,MS4A1,IGLC2,IGKC,TPSB2,KIT,LAMP3,LY75,CIITA,CD40,LA-DQA1,LA-DQB1,LA-DRB1,HLA-DRA,LA-DRB5,LYZ,THBD,IL10,DCT,TYRP1,MLANA,SPRR2G,FABP5,KRT10,KRT2,KRT14,KRT5,DCN,COL1A1"</span>)<br>GSE220116gene2&lt;-str_split(GSE220116gene,<span>","</span>)<br>GSE220116gene3&lt;-unlist(GSE220116gene2)<br>GSE220116gene3<br><br>last_markers = c(<span>'PTPRC'</span>, <span>'CD3D'</span>, <span>'CD3E'</span>, <span>'CD4'</span>,<span>'CD8A'</span>,<br>                   <span>'CD19'</span>, <span>'CD79A'</span>, <span>'MS4A1'</span> ,<br>                   <span>'IGHG1'</span>, <span>'MZB1'</span>, <span>'SDC1'</span>,<br>                   <span>'CD68'</span>, <span>'CD163'</span>, <span>'CD14'</span>, <br>                   <span>'TPSAB1'</span> , <span>'TPSB2'</span>,  <span># mast cells,</span><br>                   <span>'RCVRN'</span>,<span>'FPR1'</span> , <span>'ITGAM'</span> ,<br>                   <span>'C1QA'</span>,  <span>'C1QB'</span>,  <span># mac</span><br>                   <span>'S100A9'</span>, <span>'S100A8'</span>, <span>'MMP19'</span>,<span># monocyte</span><br>                   <span>'FCGR3A'</span>,<br>                   <span>'LAMP3'</span>, <span>'IDO1'</span>,<span>'IDO2'</span>,<span>## DC3 </span><br>                   <span>'CD1E'</span>,<span>'CD1C'</span>, <span># DC2</span><br>                   <span>'KLRB1'</span>,<span>'NCR1'</span>, <span># NK </span><br>                   <span>'FGF7'</span>,<span>'MME'</span>, <span>'ACTA2'</span>, <span>## human  fibo </span><br>                 <span>'GJB2'</span>, <span>'RGS5'</span>,<br>                   <span>'DCN'</span>, <span>'LUM'</span>,  <span>'GSN'</span> , <span>## mouse PDAC fibo </span><br>                   <span>'MKI67'</span> , <span>'TOP2A'</span>, <br>                   <span>'PECAM1'</span>, <span>'VWF'</span>,  <span>## endo </span><br>                 <span>"PLVAP"</span>,<span>'PROX1'</span>,<span>'ACKR1'</span>,<span>'CA4'</span>,<span>'HEY1'</span>,<br>                   <span>'EPCAM'</span> , <span>'KRT19'</span>,<span>'KRT7'</span>, <span># epi </span><br>                   <span>'FYXD2'</span>, <span>'TM4SF4'</span>, <span>'ANXA4'</span>,<span># cholangiocytes</span><br>                   <span>'APOC3'</span>, <span>'FABP1'</span>,  <span>'APOA1'</span>,  <span># hepatocytes</span><br>                   <span>'Serpina1c'</span>,<br>                   <span>'PROM1'</span>, <span>'ALDH1A1'</span> )<br><br>markers = c(<span>"GSE220116gene3"</span>， <br>            <span>'last_markers'</span> )<br><br>p_umap=DimPlot(sce.all.int, reduction = <span>"umap"</span>,label = <span>T</span>,repel = <span>T</span>) <br>p_umap <br><br><span>if</span>(sp==<span>'human'</span>){<br>   lapply(markers, <span>function</span>(x){<br>     <span>#x=markers[1]</span><br>     genes_to_check=str_to_upper(get(x)) <br>     DotPlot(sce.all.int , features = genes_to_check )  + <br>       coord_flip() + <br>       theme(axis.text.x=element_text(angle=<span>45</span>,hjust = <span>1</span>))<br>     <br>     h=length( genes_to_check )/<span>6</span>+<span>3</span>;h<br>     ggsave(paste(<span>'check_for_'</span>,x,<span>'.pdf'</span>),height = h)<br>   })<br>  <br>  last_markers_to_check &lt;- str_to_upper(last_markers ) <br><br> }<span>else</span> <span>if</span>(sp==<span>'mouse'</span>){<br>   lapply(markers, <span>function</span>(x){<br>     <span>#x=markers[1]</span><br>     genes_to_check=str_to_title(get(x)) <br>     DotPlot(sce.all.int , features = genes_to_check )  + <br>       coord_flip() + <br>       theme(axis.text.x=element_text(angle=<span>45</span>,hjust = <span>1</span>))<br>     <br>     h=length( genes_to_check )/<span>6</span>+<span>3</span>;h<br>     ggsave(paste(<span>'check_for_'</span>,x,<span>'.pdf'</span>),height = h)<br>   })<br>   <br>   last_markers_to_check &lt;- str_to_title(last_markers ) <br>}<span>else</span> {<br>  print(<span>'we only accept human or mouse'</span>)<br>} <br></code></pre><p data-tool="mdnice编辑器"><strong>标准亚群命名</strong></p><pre data-tool="mdnice编辑器"><span></span><code>sce.all.int<br>  celltype=data.frame(ClusterID=<span>0</span>:<span>18</span> ,<br>                      celltype= <span>0</span>:<span>18</span>) <br>  <span>#定义细胞亚群        </span><br>  celltype[celltype$ClusterID %<span>in</span>% c( <span>2</span>,<span>4</span>,<span>5</span>,<span>7</span>,<span>8</span>,<span>11</span>,<span>12</span>,<span>15</span>,<span>16</span> ),<span>2</span>]=<span>'Keratinocyte'</span><br>  celltype[celltype$ClusterID %<span>in</span>% c( <span>13</span> ),<span>2</span>]=<span>'Fibroblast'</span>   <br>  celltype[celltype$ClusterID %<span>in</span>% c( <span>9</span> ),<span>2</span>]=<span>'DC-2'</span>  <br>  celltype[celltype$ClusterID %<span>in</span>% c( <span>3</span> ),<span>2</span>]=<span>'DC-1'</span><br>  celltype[celltype$ClusterID %<span>in</span>% c( <span>0</span>,<span>1</span>),<span>2</span>]=<span>'T'</span>   <br>  celltype[celltype$ClusterID %<span>in</span>% c( <span>10</span>),<span>2</span>]=<span>'melanocytes'</span> <br>  celltype[celltype$ClusterID %<span>in</span>% c( <span>17</span>),<span>2</span>]=<span>'cycling'</span> <br>  celltype[celltype$ClusterID %<span>in</span>% c( <span>14</span>),<span>2</span>]=<span>'B'</span>   <br>  celltype[celltype$ClusterID %<span>in</span>% c( <span>6</span> ),<span>2</span>]=<span>'plasma'</span>    <br>  celltype[celltype$ClusterID %<span>in</span>% c( <span>18</span>),<span>2</span>]=<span>'Mast'</span> <br>  head(celltype)<br>  celltype<br>  table(celltype$celltype)<br>  sce.all.int@meta.data$celltype = <span>"NA"</span><br>  <br>  <span>for</span>(i <span>in</span> <span>1</span>:nrow(celltype)){<br>    sce.all.int@meta.data[which(sce.all.int@meta.data$RNA_snn_res.0.8 == celltype$ClusterID[i]),<span>'celltype'</span>] &lt;- celltype$celltype[i]}<br>  Idents(sce.all.int)=sce.all.int$celltype<br>  <br>  table( Idents(sce.all.int))<br><br>  sel.clust = <span>"celltype"</span><br>  sce.all.int &lt;- SetIdent(sce.all.int, value = sel.clust)<br>  table(sce.all.int@active.ident) <br>  p_umap=DimPlot(sce.all.int, reduction = <span>"umap"</span>, group.by = <span>"celltype"</span>,label = <span>T</span>,label.box = <span>T</span>)<br>  p_umap<br><br>  sp=<span>"human"</span><br>  dir.create(<span>'check-by-celltype'</span>)<br>  setwd(<span>'check-by-celltype'</span>)<br>  <span>source</span>(<span>'../scRNA_scripts/check-all-markers.R'</span>)<br>  setwd(<span>'../'</span>) <br>  getwd()<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100037409" data-ratio="1.1143695014662756" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los88qwNHPY4vibic2ayh6Rd1BEib6DibMrkfTHmCbY6DNXS74UcA8jicpzeaeQmZ7HlNAnEBibsYF9tczxZA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="682" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los88qwNHPY4vibic2ayh6Rd1BEib6DibMrkfTHmCbY6DNXS74UcA8jicpzeaeQmZ7HlNAnEBibsYF9tczxZA/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">今天推文代码更新的很完整，大家可以下载数据运行尝试一下。还有一些细节部分没有完全复现，所以下次继续更新这篇文献没有复现的部分。</p><h3 data-tool="mdnice编辑器"><span></span><span></span><span>微信交流群</span><span></span></h3><p data-tool="mdnice编辑器">如果你也想参与到群里的讨论中，给我提出一些推文更新建议的话欢迎入群。同时你也可以得到我前期更新的所有数据及代码，在群里我会把代码分享给大家，而且大家可以在群里「提出自己感兴趣的单细胞文献」<strong>我们尽可能优先选择大家的数据集去复现和实战，也欢迎大家在群里分享更多其它资料，咱们一起进步，</strong>比较高级的单细胞分析 我们也会一起尝试，相信你肯定是会不虚此行哈。</p><p data-tool="mdnice编辑器">附上之前推文的链接 <a href="https://mp.weixin.qq.com/s?__biz=MzUzMTEwODk0Ng==&amp;mid=2247512238&amp;idx=1&amp;sn=e0c6dc2ea0e089aabb78133e50fb5d7f&amp;scene=21#wechat_redirect" data-linktype="2">为爱发电不可取（单细胞周更需要你的支持）</a></p><p data-tool="mdnice编辑器"><strong>「目前群里已有近490人，所以你想要进群的话就需要添加我的微信，私聊给我发99元的红包，我把你拉进群里。」</strong></p><p data-tool="mdnice编辑器">我们这个群是真正的付费交流群，提供数据，代码，学习资料。如果介意99元的门槛且不认可我们知识分享的理念，<strong>请不要加我好友</strong>。因个人时间有限，不能及时回复还请见谅，再说明一下<strong>群里并没有一对一答疑服务</strong>，只是知识分享相互交流。</p><figure data-tool="mdnice编辑器"><img data-imgfileid="100037407" data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los88qwNHPY4vibic2ayh6Rd1BE6ibgEiaUP2ydu5OLaeAiayyxaEYc14GwSibVc10HPhR0vrE7NEaezZxGtg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="172" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los88qwNHPY4vibic2ayh6Rd1BE6ibgEiaUP2ydu5OLaeAiayyxaEYc14GwSibVc10HPhR0vrE7NEaezZxGtg/640?wx_fmt=png&amp;from=appmsg"></figure></section><p><br></p><p><mp-style-type data-value="10000"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/_RazFn6npuj1vBbE3n-mCw",target="_blank" rel="noopener noreferrer">原文链接</a>
