---
title: "irGSEA：基于秩次的单细胞基因集富集分析整合框架"
date: 2023-12-05T04:29:28Z
draft: ["false"]
tags: [
  "fetched",
  "生信技能树"
]
categories: ["Acdemic"]
---
irGSEA：基于秩次的单细胞基因集富集分析整合框架 by 生信技能树
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><blockquote data-tool="mdnice编辑器"><p>距离上一次介绍irGSEA，已经是两年前了，详见：<a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247507596&amp;idx=1&amp;sn=1bae2446918825f2f171f8eb4ea44d0e&amp;chksm=9b4b9e37ac3c17216e111769f6bc56e57d021d554d8b53161235eaace67830bb5b56e30eb3ac&amp;scene=21#wechat_redirect" textvalue="8种方法可视化你的单细胞基因集打分" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">8种方法可视化你的单细胞基因集打分</a>，目前Seurat都已经更新到了V5，<a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247526354&amp;idx=1&amp;sn=a4fb075968d8a78ecda292067ba18204&amp;chksm=9b4b2769ac3cae7fed9721883cebbff801188810f3949e0c1d1eaa52619fba81b1d0ba30946f&amp;scene=21#wechat_redirect" textvalue="假如你不喜欢最新版的Seurat包的单细胞理念" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">假如你不喜欢最新版的Seurat包的单细胞理念</a>，正好<span><strong>这个irGSEA也是与时俱进，不仅仅是更新到了17种基因集合打分算法，而且是配合V4和V5版本的Seurat工作流</strong></span>，非常的方便！如果大家看完了这个万字长文的介绍，仍然是愿意深入了解irGSEA可以看看文末的交流群哈！</p></blockquote></section><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><h3 data-tool="mdnice编辑器"><span></span><span></span><span>背景</span><span></span></h3><p data-tool="mdnice编辑器">许多Functional Class Scoring (FCS)方法，如<code>GSEA</code>, <code>GSVA</code>,<code>PLAGE</code>, <code>addModuleScore</code>, <code>SCSE</code>, <code>Vision</code>, <code>VAM</code>, <code>gficf</code>, <code>pagoda2</code>和<code>Sargent</code>，都会受数据集组成的影响，数据集组成的轻微变化将改变细胞的基因集富集分数。</p><p data-tool="mdnice编辑器">假如将新的单细胞数据集整合到现有数据中，使用这些FCS方法需要重新计算每个细胞的基因集富集分数。<strong>这个步骤可能是繁琐且资源密集的</strong>。</p><p data-tool="mdnice编辑器">相反，基于单个细胞表达等级的FCS，如<code>AUCell</code>、<code>UCell</code>、<code>singscore</code>、<code>ssGSEA</code>、<code>JASMINE</code>和<code>Viper</code>，只需要计算新添加的单细胞数据集的富集分数，而无需重新计算所有细胞的基因集富集分数。原因是这些方法生成的富集分数仅依赖于单个细胞水平上的相对基因表达，与数据集组成无关。因此，这些方法可以节省大量的时间。</p><h3 data-tool="mdnice编辑器"><span></span><span></span><span>审视结果</span><span></span></h3><p data-tool="mdnice编辑器">在这里，我们审视了<strong>17种常见的FCS方法：</strong></p><ol data-tool="mdnice编辑器"><li><section><code>GSEA </code>检测排序基因列表顶部或底部的基因集富集程度，该列表是分组后计算排序基因信噪比或排序基因倍数变化得到的；</section></li><li><section><code>GSVA </code>估计所有细胞之间每个基因的累积密度函数的核。 这个过程中需要考虑所有样本，容易受到样本背景信息的影响；</section></li><li><section><code>PLAGE </code>对跨细胞的基因表达矩阵进行标准化，并提取奇异值分解作为基因集富集分数；</section></li><li><section><code>Zscore </code>聚合了基因集中所有基因的表达，通过细胞间的平均值和标准差缩放表达；</section></li><li><section><code>AddModuleScore</code>需要先计算基因集中所有基因的平均值，再根据平均值把表达矩阵切割成若干份，然后从切割后的每一份中随机抽取对照基因（基因集外的基因）作为背景值。因此，在整合不同样本的情况下，即使使用相同基因集为相同细胞打分，也会产生不同的富集评分；</section></li><li><section><code>SCSE</code> 使用基因集所有基因的归一化的总和来量化基因集富集分数；</section></li><li><section><code>Vision</code> 使用随机签名的预期均值和方差对基因集富集分数进行 z 归一化从而校正基因集富集分数；</section></li><li><section><code>VAM</code> 根据经典Mahalanobis多元距离从单细胞 RNA 测序数据生成基因集富集分数；</section></li><li><section><code>Gficf </code>利用通过非负矩阵分解获得的基因表达值的潜在因子的信息生物信号；</section></li><li><section><code>Pagoda2</code> 拟合每个细胞的误差模型，并使用其第一个加权主成分量化基因集富集分数；</section></li><li><section><code>AUCell</code> 基于单个样本中的基因表达排名,使用曲线下面积来评估输入基因集是否在单个样本的前5%表达基因内富集；</section></li><li><section><code>UCell</code> 基于单个样本的基因表达排名，使用Mann-Whitney U统计量计算单个样本的基因集富集分数；</section></li><li><section><code>Singscore</code> 根据基因表达等级评估距单个细胞中心的距离。 基因集中的基因根据单个细胞中的转录本丰度进行排序。 平均等级相对于理论最小值和最大值单独标准化，以零为中心，然后聚合，所得分数代表基因集的富集分数；</section></li><li><section><code>ssGSEA </code>根据每个细胞的基因表达等级计算内部和外部基因集之间的经验累积分布的差异分数。 使用全局表达谱对差异分数进行标准化。 标准化这一步容易受样本构成的影响。</section></li><li><section><code>JASMINE</code> 根据在单个细胞中表达基因中的基因排名和表达基因中基因集的富集度计算近似平均值。 这两个值均标准化为 0-1 范围，并通过平均进行组合，得出基因集的最终富集分数。</section></li><li><section><code>Viper</code> 通过根据细胞间基因表达的排名执行three-tailed计算来估计基因集的富集分数。</section></li><li><section><code>Sargent </code>将给定细胞的非零表达基因从高表达到低表达进行排序，并将输入的基因逐细胞表达矩阵转换为相应的gene-set-by-cell assignment score matrix。 但<code>Sargent </code>需要计算细胞间的gini-index后，将按gene-set-by-cell assignment score matrix转换为distribution of indexes。</section></li></ol><h3 data-tool="mdnice编辑器"><span></span><span></span><span>工作流程</span><span></span></h3><figure data-tool="mdnice编辑器"><img data-imgfileid="100043153" data-ratio="0.8610354223433242" data-src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wxW0zhpl0fFDmP4Hr9iczb2BkWsp4Z3gc96z6mvjgmvOA1rHkpEbN5IGmpV6EM1UiaLYonjFpw7gj3Q/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="734" src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wxW0zhpl0fFDmP4Hr9iczb2BkWsp4Z3gc96z6mvjgmvOA1rHkpEbN5IGmpV6EM1UiaLYonjFpw7gj3Q/640?wx_fmt=png&amp;from=appmsg"></figure><ul data-tool="mdnice编辑器"><li><section>使用<code>AUCell、UCell、singscore、ssgsea、JASMINE 和 viper</code>分别对各个细胞进行评分，得到不同的富集评分矩阵。</section></li><li><section>通过<code>wilcoxon检验</code>计算不同的富集评分矩阵中每个细胞亚群差异表达的基因集。<code>up或down</code>表示该细胞簇内差异基因集的富集程度高于或低于其他簇。</section></li></ul><p data-tool="mdnice编辑器"><code>单一的基因集富集分析方法不仅只能反映有限的信息，而且也容易带来误差</code>。我们期待从多个角度解释复杂的生物学问题，并找到生物学问题中的共性部分。简单地为多种基因集富集分析方法的结果取共同交集，不仅容易得到少而保守的结果，而且忽略了富集分析方法中很多的其他信息，例如不同基因集的相对富集程度信息。</p><p data-tool="mdnice编辑器"><code>我们希望目标基因集在大部分富集分析方法中都是富集且富集程度没有明显差异</code>。因此，我们通过RobustRankAggreg包中的<code>秩聚合算法(robust rank aggregation, RRA)</code>对差异分析的结果进行评估，筛选出在6种方法中表现出相似的富集程度的差异基因集。</p><h3 data-tool="mdnice编辑器"><span></span><span></span><span>irGSEA安装</span><span></span></h3><h4 data-tool="mdnice编辑器"><span></span>1.irGSEA安装（基础配置）<span></span></h4><p data-tool="mdnice编辑器">仅使用 AUCell, UCell, singscore, ssGSEA, JASMINE和viper</p><pre data-tool="mdnice编辑器"><span></span><code><span># install packages from CRAN</span><br>cran.packages &lt;- c(<span>"aplot"</span>, <span>"BiocManager"</span>, <span>"data.table"</span>, <span>"devtools"</span>, <br>                   <span>"doParallel"</span>, <span>"doRNG"</span>, <span>"dplyr"</span>, <span>"ggfun"</span>, <span>"gghalves"</span>, <br>                   <span>"ggplot2"</span>, <span>"ggplotify"</span>, <span>"ggridges"</span>, <span>"ggsci"</span>, <span>"irlba"</span>,<br>                   <span>"magrittr"</span>, <span>"Matrix"</span>, <span>"msigdbr"</span>, <span>"pagoda2"</span>, <span>"pointr"</span>, <br>                   <span>"purrr"</span>, <span>"RcppML"</span>, <span>"readr"</span>, <span>"reshape2"</span>, <span>"reticulate"</span>, <br>                   <span>"rlang"</span>, <span>"RMTstat"</span>, <span>"RobustRankAggreg"</span>, <span>"roxygen2"</span>, <br>                   <span>"Seurat"</span>, <span>"SeuratObject"</span>, <span>"stringr"</span>, <span>"tibble"</span>, <span>"tidyr"</span>, <br>                   <span>"tidyselect"</span>, <span>"tidytree"</span>, <span>"VAM"</span>)<br><br><span>for</span> (i <span>in</span> cran.packages) {<br>  <span>if</span> (!requireNamespace(i, quietly = TRUE)) {<br>    install.packages(i, ask = F, update = F)<br>  }<br>}<br><br><span># install packages from Bioconductor</span><br>bioconductor.packages &lt;- c(<span>"AUCell"</span>, <span>"BiocParallel"</span>, <span>"ComplexHeatmap"</span>, <br>                           <span>"decoupleR"</span>, <span>"fgsea"</span>, <span>"ggtree"</span>, <span>"GSEABase"</span>, <br>                           <span>"GSVA"</span>, <span>"Nebulosa"</span>, <span>"scde"</span>, <span>"singscore"</span>,<br>                           <span>"SummarizedExperiment"</span>, <span>"UCell"</span>,<br>                           <span>"viper"</span>,<span>"sparseMatrixStats"</span>)<br><br><span>for</span> (i <span>in</span> bioconductor.packages) {<br>  <span>if</span> (!requireNamespace(i, quietly = TRUE)) {<br>    install.packages(i, ask = F, update = F)<br>  }<br>}<br><br><span># install packages from Github</span><br><span>if</span> (!requireNamespace(<span>"irGSEA"</span>, quietly = TRUE)) { <br>    devtools::install_github(<span>"chuiqin/irGSEA"</span>, force =T)<br>}<br></code></pre><h4 data-tool="mdnice编辑器"><span></span>2.irGSEA安装（进阶配置）<span></span></h4><p data-tool="mdnice编辑器">想使用VISION, gficf, Sargent, ssGSEApy, GSVApy等其他方法（这部分是可选安装）</p><pre data-tool="mdnice编辑器"><span></span><code><span># VISION</span><br><span>if</span> (!requireNamespace(<span>"VISION"</span>, quietly = TRUE)) { <br>    devtools::install_github(<span>"YosefLab/VISION"</span>, force =T)<br>}<br><br><span># mdt need ranger</span><br><span>if</span> (!requireNamespace(<span>"ranger"</span>, quietly = TRUE)) { <br>  devtools::install_github(<span>"imbs-hl/ranger"</span>, force =T)<br>}<br><br><span># gficf need RcppML (version &gt; 0.3.7) package</span><br><span>if</span> (!utils::packageVersion(<span>"RcppML"</span>) &gt; <span>"0.3.7"</span>) {<br>  message(<span>"The version of RcppML should greater than 0.3.7 and install RcppML package from Github"</span>)<br>  devtools::install_github(<span>"zdebruine/RcppML"</span>, force =T)<br>}<br><br><span># please first `library(RcppML)` if you want to perform gficf</span><br><span>if</span> (!requireNamespace(<span>"gficf"</span>, quietly = TRUE)) { <br>    devtools::install_github(<span>"gambalab/gficf"</span>, force =T)<br>}<br><br><span># GSVApy and ssGSEApy need SeuratDisk package</span><br><span>if</span> (!requireNamespace(<span>"SeuratDisk"</span>, quietly = TRUE)) { <br>    devtools::install_github(<span>"mojaveazure/seurat-disk"</span>, force =T)<br>}<br><br><span># sargent</span><br><span>if</span> (!requireNamespace(<span>"sargent"</span>, quietly = TRUE)) { <br>    devtools::install_github(<span>"Sanofi-Public/PMCB-Sargent"</span>, force =T)<br>}<br><br><span># pagoda2 need scde package</span><br><span>if</span> (!requireNamespace(<span>"scde"</span>, quietly = TRUE)) { <br>  devtools::install_github(<span>"hms-dbmi/scde"</span>, force =T)<br>}<br><br><span># if error1 (functio 'sexp_as_cholmod_sparse' not provided by package 'Matrix')</span><br><span># or error2 (functio 'as_cholmod_sparse' not provided by package 'Matrix') occurs</span><br><span># when you perform pagoda2, please check the version of irlba and Matrix</span><br><span># It's ok when I test as follow：</span><br><span># R 4.2.2 irlba(v 2.3.5.1) Matrix(1.5-3)</span><br><span># R 4.3.1 irlba(v 2.3.5.1) Matrix(1.6-1.1)</span><br><span># R 4.3.2 irlba(v 2.3.5.1) Matrix(1.6-3)</span><br><br><br><span>#### create conda env</span><br><span># If error (Unable to find conda binary. Is Anaconda installed) occurs, </span><br><span># please perform `reticulate::install_miniconda()`</span><br><span>if</span> (! <span>"irGSEA"</span> %<span>in</span>% reticulate::conda_list()<span>$name</span>) {<br>      reticulate::conda_create(<span>"irGSEA"</span>)<br>}<br><br><span># if python package exist</span><br>python.package &lt;- reticulate::py_list_packages(envname = <span>"irGSEA"</span>)<span>$package</span><br>require.package &lt;- c(<span>"anndata"</span>, <span>"scanpy"</span>, <span>"argparse"</span>, <span>"gseapy"</span>, <span>"decoupler"</span>)<br><span>for</span> (i <span>in</span> seq_along(require.package)) {<br>  <span>if</span> (i %<span>in</span>% python.package) {<br>    reticulate::conda_install(envname = <span>"irGSEA"</span>, packages = i, pip = T)<br>  }<br>}<br><br></code></pre><h4 data-tool="mdnice编辑器"><span></span>3.国内镜像加速安装<span></span></h4><p data-tool="mdnice编辑器">安装github包和pip包有困难的参考这里，我把github包克隆到了gitee上</p><pre data-tool="mdnice编辑器"><span></span><code>options(BioC_mirror=<span>"https://mirrors.tuna.tsinghua.edu.cn/bioconductor/"</span>)<br>options(<span>"repos"</span> = c(CRAN=<span>"http://mirrors.cloud.tencent.com/CRAN/"</span>))<br><br><span># install packages from CRAN</span><br>cran.packages &lt;- c(<span>"aplot"</span>, <span>"BiocManager"</span>, <span>"data.table"</span>, <span>"devtools"</span>, <br>                   <span>"doParallel"</span>, <span>"doRNG"</span>, <span>"dplyr"</span>, <span>"ggfun"</span>, <span>"gghalves"</span>, <br>                   <span>"ggplot2"</span>, <span>"ggplotify"</span>, <span>"ggridges"</span>, <span>"ggsci"</span>, <span>"irlba"</span>,<br>                   <span>"magrittr"</span>, <span>"Matrix"</span>, <span>"msigdbr"</span>, <span>"pagoda2"</span>, <span>"pointr"</span>, <br>                   <span>"purrr"</span>, <span>"RcppML"</span>, <span>"readr"</span>, <span>"reshape2"</span>, <span>"reticulate"</span>, <br>                   <span>"rlang"</span>, <span>"RMTstat"</span>, <span>"RobustRankAggreg"</span>, <span>"roxygen2"</span>, <br>                   <span>"Seurat"</span>, <span>"SeuratObject"</span>, <span>"stringr"</span>, <span>"tibble"</span>, <span>"tidyr"</span>, <br>                   <span>"tidyselect"</span>, <span>"tidytree"</span>, <span>"VAM"</span>)<br><br><span>for</span> (i <span>in</span> cran.packages) {<br>  <span>if</span> (!requireNamespace(i, quietly = TRUE)) {<br>    install.packages(i, ask = F, update = F)<br>  }<br>}<br><br><span># install packages from Bioconductor</span><br>bioconductor.packages &lt;- c(<span>"AUCell"</span>, <span>"BiocParallel"</span>, <span>"ComplexHeatmap"</span>, <br>                           <span>"decoupleR"</span>, <span>"fgsea"</span>, <span>"ggtree"</span>, <span>"GSEABase"</span>, <br>                           <span>"GSVA"</span>, <span>"Nebulosa"</span>, <span>"scde"</span>, <span>"singscore"</span>,<br>                           <span>"SummarizedExperiment"</span>, <span>"UCell"</span>, <span>"viper"</span>)<br><br><span>for</span> (i <span>in</span> bioconductor.packages) {<br>  <span>if</span> (!requireNamespace(i, quietly = TRUE)) {<br>    install.packages(i, ask = F, update = F)<br>  }<br>}<br><br><span># install packages from git</span><br><span>if</span> (!requireNamespace(<span>"irGSEA"</span>, quietly = TRUE)) { <br>    devtools::install_git(<span>"https://gitee.com/fan_chuiqin/irGSEA.git"</span>, force =T)<br>}<br><span># VISION</span><br><span>if</span> (!requireNamespace(<span>"VISION"</span>, quietly = TRUE)) { <br>    devtools::install_git(<span>"https://gitee.com/fan_chuiqin/VISION.git"</span>, force =T)<br>}<br><br><span># mdt need ranger</span><br><span>if</span> (!requireNamespace(<span>"ranger"</span>, quietly = TRUE)) { <br>  devtools::install_git(<span>"https://gitee.com/fan_chuiqin/ranger.git"</span>, force =T)<br>}<br><br><span># gficf need RcppML (version &gt; 0.3.7) package</span><br><span>if</span> (!utils::packageVersion(<span>"RcppML"</span>) &gt; <span>"0.3.7"</span>) {<br>  message(<span>"The version of RcppML should greater than 0.3.7 and install RcppML package from Git"</span>)<br>  devtools::install_git(<span>"https://gitee.com/fan_chuiqin/RcppML.git"</span>, force =T)<br>}<br><br><span># please first `library(RcppML)` if you want to perform gficf</span><br><span>if</span> (!requireNamespace(<span>"gficf"</span>, quietly = TRUE)) { <br>    devtools::install_git(<span>"https://gitee.com/fan_chuiqin/gficf.git"</span>, force =T)<br>}<br><br><span># GSVApy and ssGSEApy need SeuratDisk package</span><br><span>if</span> (!requireNamespace(<span>"SeuratDisk"</span>, quietly = TRUE)) { <br>    devtools::install_git(<span>"https://gitee.com/fan_chuiqin/seurat-disk.git"</span>, <br>                          force =T)}<br><br><span># sargent</span><br><span>if</span> (!requireNamespace(<span>"sargent"</span>, quietly = TRUE)) { <br>    devtools::install_git(<span>"https://gitee.com/fan_chuiqin/PMCB-Sargent.git"</span>, <br>                          force =T)}<br><br><span># pagoda2 need scde package</span><br><span>if</span> (!requireNamespace(<span>"scde"</span>, quietly = TRUE)) { <br>  devtools::install_git(<span>"https://gitee.com/fan_chuiqin/scde.git"</span>, force =T)<br>}<br><br><span>#### create conda env</span><br><span># If error (Unable to find conda binary. Is Anaconda installed) occurs, </span><br><span># please perform `reticulate::install_miniconda()`</span><br><span>if</span> (! <span>"irGSEA"</span> %<span>in</span>% reticulate::conda_list()<span>$name</span>) {<br>      reticulate::conda_create(<span>"irGSEA"</span>)<br>}<br><br><span># if python package exist</span><br>python.package &lt;- reticulate::py_list_packages(envname = <span>"irGSEA"</span>)<span>$package</span><br>require.package &lt;- c(<span>"anndata"</span>, <span>"scanpy"</span>, <span>"argparse"</span>, <span>"gseapy"</span>, <span>"decoupler"</span>)<br><span>for</span> (i <span>in</span> require.package) {<br>  <span>if</span> (! i %<span>in</span>% python.package) {<br>    reticulate::conda_install(envname = <span>"irGSEA"</span>, packages = i, pip = T,<br>     pip_options = <span>"-i https://pypi.tuna.tsinghua.edu.cn/simple"</span>)<br>  }<br>}<br></code></pre><h3 data-tool="mdnice编辑器"><span></span><span></span><span>使用教程</span><span></span></h3><h4 data-tool="mdnice编辑器"><span></span>1.irGSEA支持Seurat 对象（V5或V4），Assay对象（V5或V4）<span></span></h4><pre data-tool="mdnice编辑器"><span></span><code><span># 我们通过SeuratData包加载示例数据集（注释好的PBMC数据集）作为演示</span><br><br><span>#### Seurat V4对象 ####</span><br>library(Seurat)<br>library(SeuratData)<br>library(RcppML)<br>library(irGSEA)<br>data(<span>"pbmc3k.final"</span>)<br>pbmc3k.final &lt;- irGSEA.score(object = pbmc3k.final,assay = <span>"RNA"</span>, <br>                             slot = <span>"data"</span>, seeds = 123, ncores = 1,<br>                             min.cells = 3, min.feature = 0,<br>                             custom = F, geneset = NULL, msigdb = T, <br>                             species = <span>"Homo sapiens"</span>, category = <span>"H"</span>,  <br>                             subcategory = NULL, geneid = <span>"symbol"</span>,<br>                             method = c(<span>"AUCell"</span>,<span>"UCell"</span>,<span>"singscore"</span>,<span>"ssgsea"</span>, <span>"JASMINE"</span>, <span>"viper"</span>),<br>                             aucell.MaxRank = NULL, ucell.MaxRank = NULL, <br>                             kcdf = <span>'Gaussian'</span>)<br> Assays(pbmc3k.final)<br>&gt;[1] <span>"RNA"</span>       <span>"AUCell"</span>    <span>"UCell"</span>     <span>"singscore"</span> <span>"ssgsea"</span>    <span>"JASMINE"</span>   <span>"viper"</span> <br><br><span>#### Seurat V5对象 ####</span><br>data(<span>"pbmc3k.final"</span>)<br>pbmc3k.final &lt;- SeuratObject::UpdateSeuratObject(pbmc3k.final)<br>pbmc3k.final2 &lt;- CreateSeuratObject(counts = CreateAssay5Object(GetAssayData(pbmc3k.final, assay = <span>"RNA"</span>, slot = <span>"counts"</span>)),<br>                                    meta.data = pbmc3k.final[[]])<br>pbmc3k.final2 &lt;- NormalizeData(pbmc3k.final2)<br>pbmc3k.final2 &lt;- irGSEA.score(object = pbmc3k.final2,assay = <span>"RNA"</span>, <br>                             slot = <span>"data"</span>, seeds = 123, ncores = 1,<br>                             min.cells = 3, min.feature = 0,<br>                             custom = F, geneset = NULL, msigdb = T, <br>                             species = <span>"Homo sapiens"</span>, category = <span>"H"</span>,  <br>                             subcategory = NULL, geneid = <span>"symbol"</span>,<br>                             method = c(<span>"AUCell"</span>,<span>"UCell"</span>,<span>"singscore"</span>,<span>"ssgsea"</span>, <span>"JASMINE"</span>, <span>"viper"</span>),<br>                             aucell.MaxRank = NULL, ucell.MaxRank = NULL, <br>                             kcdf = <span>'Gaussian'</span>)<br>Assays(pbmc3k.final2)<br>[1] <span>"RNA"</span>       <span>"AUCell"</span>    <span>"UCell"</span>     <span>"singscore"</span> <span>"ssgsea"</span>    <span>"JASMINE"</span>   <span>"viper"</span> <br><br><span>#### Assay5 对象  ####</span><br>data(<span>"pbmc3k.final"</span>)<br>pbmc3k.final &lt;- SeuratObject::UpdateSeuratObject(pbmc3k.final)<br>pbmc3k.final3 &lt;- CreateAssay5Object(counts = GetAssayData(pbmc3k.final, assay = <span>"RNA"</span>, slot = <span>"counts"</span>))<br>pbmc3k.final3 &lt;- NormalizeData(pbmc3k.final3)<br>pbmc3k.final3 &lt;- irGSEA.score(object = pbmc3k.final3,assay = <span>"RNA"</span>, <br>                              slot = <span>"counts"</span>, seeds = 123, ncores = 1,<br>                              min.cells = 3, min.feature = 0,<br>                              custom = F, geneset = NULL, msigdb = T, <br>                              species = <span>"Homo sapiens"</span>, category = <span>"H"</span>,  <br>                              subcategory = NULL, geneid = <span>"symbol"</span>,<br>                              method = c(<span>"AUCell"</span>,<span>"UCell"</span>,<span>"singscore"</span>,<span>"ssgsea"</span>, <span>"JASMINE"</span>, <span>"viper"</span>),<br>                              aucell.MaxRank = NULL, ucell.MaxRank = NULL, <br>                              kcdf = <span>'Gaussian'</span>)<br>Assays(pbmc3k.final3)<br><span># Assay5对象存放在RNA assay中</span><br></code></pre><p data-tool="mdnice编辑器">结果文件大小统计：</p><pre data-tool="mdnice编辑器"><span></span><code><span># 看起来Seurat V5和Assay 5确实存放数据会比较小</span><br>&gt; cat(object.size(pbmc3k.final) / (1024^2), <span>"M"</span>)<br>339.955 M<br>&gt; cat(object.size(pbmc3k.final2) / (1024^2), <span>"M"</span>)<br>69.33521 M<br>&gt; cat(object.size(pbmc3k.final3) / (1024^2), <span>"M"</span>)<br>69.27851 M<br></code></pre><h4 data-tool="mdnice编辑器"><span></span>2.irGSEA支持的基因集打分方法<span></span></h4><p data-tool="mdnice编辑器">测试了不同数据大小下各种评分方法使用50个Hallmark基因集进行打分所需的时间和内存峰值， 大家根据自己的电脑和时间进行酌情选择；</p><p data-tool="mdnice编辑器"><img data-imgfileid="100043155" data-ratio="0.43703703703703706" data-src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wxW0zhpl0fFDmP4Hr9iczb2BHof4o7YlWOAYAfyQcHXQBUSYXtbFCAkVuxQUtcGaEgRAhiaqoN3RosQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wxW0zhpl0fFDmP4Hr9iczb2BHof4o7YlWOAYAfyQcHXQBUSYXtbFCAkVuxQUtcGaEgRAhiaqoN3RosQ/640?wx_fmt=png&amp;from=appmsg">GSVApy、ssGSEApy 和 viperpy 分别代表 GSVA、ssGSEA 和 viper 的 Python 版本。Python版本的GSVA比R版本的GSVA节约太多时间了。</p><p data-tool="mdnice编辑器"><code>我们对singscore、ssGSEA、JASMINE、viper的内存峰值进行了优化</code>。 对于超过 50000 个细胞的数据集，我们实施了一种策略，将它们划分为5000 个细胞/单元进行评分。 虽然这可以缓解内存峰值问题，但确实会延长处理时间。</p><h4 data-tool="mdnice编辑器"><span></span>3.irGSEA支持的基因集打分方法<span></span></h4><p data-tool="mdnice编辑器">为了方便用户获取MSigDB数据库中预先定义好的基因集，我们内置了<code>msigdbr包</code>进行MSigDB的基因集数据的获取。<code>msigdbr包</code>支持多个物种的基因集获取，以及<code>多种基因格式</code>的表达矩阵的输入。</p><p data-tool="mdnice编辑器"><strong>①irGSEA通过内置的msigdbr包进行打分</strong></p><pre data-tool="mdnice编辑器"><span></span><code>library(Seurat)<br>library(SeuratData)<br>library(RcppML)<br>library(irGSEA)<br>data(<span>"pbmc3k.final"</span>)<br><br><span>#### Hallmark基因集打分 ####</span><br>pbmc3k.final &lt;- irGSEA.score(object = pbmc3k.final,assay = <span>"RNA"</span>, <br>                             slot = <span>"data"</span>, seeds = 123, ncores = 1,<br>                             min.cells = 3, min.feature = 0,<br>                             custom = F, geneset = NULL, msigdb = T, <br>                             species = <span>"Homo sapiens"</span>, category = <span>"H"</span>,  <br>                             subcategory = NULL, geneid = <span>"symbol"</span>,<br>                             method = c(<span>"AUCell"</span>,<span>"UCell"</span>,<span>"singscore"</span>,<span>"ssgsea"</span>, <span>"JASMINE"</span>, <span>"viper"</span>),<br>                             aucell.MaxRank = NULL, ucell.MaxRank = NULL, <br>                             kcdf = <span>'Gaussian'</span>)<br><br><span>#### KEGG基因集打分 ####</span><br>pbmc3k.final &lt;- irGSEA.score(object = pbmc3k.final,assay = <span>"RNA"</span>, <br>                             slot = <span>"data"</span>, seeds = 123, ncores = 1,<br>                             min.cells = 3, min.feature = 0,<br>                             custom = F, geneset = NULL, msigdb = T, <br>                             species = <span>"Homo sapiens"</span>, category = <span>"C2"</span>,  <br>                             subcategory = <span>"CP:KEGG"</span>, geneid = <span>"symbol"</span>,<br>                             method = c(<span>"AUCell"</span>,<span>"UCell"</span>,<span>"singscore"</span>,<span>"ssgsea"</span>, <span>"JASMINE"</span>, <span>"viper"</span>),<br>                             aucell.MaxRank = NULL, ucell.MaxRank = NULL, <br>                             kcdf = <span>'Gaussian'</span>)<br><br><span>#### GO-BP基因集打分 ####</span><br>pbmc3k.final &lt;- irGSEA.score(object = pbmc3k.final,assay = <span>"RNA"</span>, <br>                             slot = <span>"data"</span>, seeds = 123, ncores = 1,<br>                             min.cells = 3, min.feature = 0,<br>                             custom = F, geneset = NULL, msigdb = T, <br>                             species = <span>"Homo sapiens"</span>, category = <span>"C5"</span>,  <br>                             subcategory = <span>"GO:BP"</span>, geneid = <span>"symbol"</span>,<br>                             method = c(<span>"AUCell"</span>,<span>"UCell"</span>,<span>"singscore"</span>,<span>"ssgsea"</span>, <span>"JASMINE"</span>, <span>"viper"</span>),<br>                             aucell.MaxRank = NULL, ucell.MaxRank = NULL, <br>                             kcdf = <span>'Gaussian'</span>)<br></code></pre><p data-tool="mdnice编辑器"><strong>②irGSEA使用最新版的MSigDB基因集进行打分</strong></p><blockquote data-tool="mdnice编辑器"><p>遗憾的是，<code>msigdbr包</code>内置的MSigDB的版本是<code>MSigDB 2022.1</code>。然而，目前MSigDB数据库已经更新到了<code>2023.2</code>，包括<code>2023.2.Hs</code>和<code>2023.2.Mm</code>。我们可以从这个链接(https://data.broadinstitute.org/gsea-msigdb/msigdb/release/)下载<code>2023.2.Hs</code>的<code>gmt文件</code>或者<code>db.zip文件</code>。相比<code>gmt文件</code>，<code>db.zip文件</code>包含了基因集的描述，可以用来筛选XX功能相关基因。下面的例子中，我将介绍如何筛选<code>血管生成相关的基因集</code>。</p></blockquote><pre data-tool="mdnice编辑器"><span></span><code><span>#### work with newest Msigdb ####</span><br><br><span># https://data.broadinstitute.org/gsea-msigdb/msigdb/release/</span><br><span># In this page, you can download human/mouse gmt file or db.zip file</span><br><span># The db.zip file contains metadata information for the gene set</span><br><br><span># load library</span><br>library(clusterProfiler)<br>library(tidyverse)<br>library(DBI)<br>library(RSQLite)<br><br><br><span>### db.zip ###</span><br><br><span># download zip file and unzip zip file</span><br>zip_url &lt;- <span>"https://data.broadinstitute.org/gsea-msigdb/msigdb/release/2023.2.Hs/msigdb_v2023.2.Hs.db.zip"</span><br>local_zip_path &lt;- <span>"./msigdb_v2023.2.Hs.db.zip"</span><br>download.file(zip_url, local_zip_path)<br>unzip(local_zip_path, exdir = <span>"./"</span>)<br><br><span># code modified by https://rdrr.io/github/cashoes/sear/src/data-raw/1_parse_msigdb_sqlite.r</span><br>con &lt;- DBI::dbConnect(RSQLite::SQLite(), dbname = <span>'./msigdb_v2023.2.Hs.db'</span>)<br>DBI::dbListTables(con)<br><br><span># define tables we want to combine</span><br>geneset_db &lt;- dplyr::tbl(con, <span>'gene_set'</span>)                                              <span># standard_name, collection_name</span><br>details_db &lt;- dplyr::tbl(con, <span>'gene_set_details'</span>)                                      <span># description_brief, description_full</span><br>geneset_genesymbol_db &lt;- dplyr::tbl(con, <span>'gene_set_gene_symbol'</span>)                       <span># meat and potatoes</span><br>genesymbol_db &lt;- dplyr::tbl(con, <span>'gene_symbol'</span>)                                        <span># mapping from ids to gene symbols</span><br>collection_db &lt;- dplyr::tbl(con, <span>'collection'</span>) %&gt;% dplyr::select(collection_name, full_name)  <span># collection metadata</span><br><br><span># join tables</span><br>msigdb &lt;- geneset_db %&gt;%<br>  dplyr::left_join(details_db, by = c(<span>'id'</span> = <span>'gene_set_id'</span>)) %&gt;%<br>  dplyr::left_join(collection_db, by = <span>'collection_name'</span>) %&gt;%<br>  dplyr::left_join(geneset_genesymbol_db, by = c(<span>'id'</span> = <span>'gene_set_id'</span>)) %&gt;%<br>  dplyr::left_join(genesymbol_db, by = c(<span>'gene_symbol_id'</span> = <span>'id'</span>)) %&gt;%<br>  dplyr::select(collection = collection_name, subcollection = full_name, geneset = standard_name, description = description_brief, symbol) %&gt;%<br>  dplyr::as_tibble() <br><br><span># clean up</span><br>DBI::dbDisconnect(con)<br><br><br>unique(msigdb<span>$collection</span>)<br><span># [1] "C1"                 "C2:CGP"             "C2:CP:BIOCARTA"    </span><br><span># [4] "C2:CP:KEGG_LEGACY"  "C2:CP:PID"          "C3:MIR:MIRDB"      </span><br><span># [7] "C3:MIR:MIR_LEGACY"  "C3:TFT:GTRD"        "C3:TFT:TFT_LEGACY" </span><br><span># [10] "C4:3CA"             "C4:CGN"             "C4:CM"             </span><br><span># [13] "C6"                 "C7:IMMUNESIGDB"     "C7:VAX"            </span><br><span># [16] "C8"                 "C5:GO:BP"           "C5:GO:CC"          </span><br><span># [19] "C5:GO:MF"           "H"                  "C5:HPO"            </span><br><span># [22] "C2:CP:KEGG_MEDICUS" "C2:CP:REACTOME"     "C2:CP:WIKIPATHWAYS"</span><br><span># [25] "C2:CP" </span><br>unique(msigdb<span>$subcollection</span>)<br><span># [1] "C1"                 "C2:CGP"             "C2:CP:BIOCARTA"    </span><br><span># [4] "C2:CP:KEGG_LEGACY"  "C2:CP:PID"          "C3:MIR:MIRDB"      </span><br><span># [7] "C3:MIR:MIR_LEGACY"  "C3:TFT:GTRD"        "C3:TFT:TFT_LEGACY" </span><br><span># [10] "C4:3CA"             "C4:CGN"             "C4:CM"             </span><br><span># [13] "C6"                 "C7:IMMUNESIGDB"     "C7:VAX"            </span><br><span># [16] "C8"                 "C5:GO:BP"           "C5:GO:CC"          </span><br><span># [19] "C5:GO:MF"           "H"                  "C5:HPO"            </span><br><span># [22] "C2:CP:KEGG_MEDICUS" "C2:CP:REACTOME"     "C2:CP:WIKIPATHWAYS"</span><br><span># [25] "C2:CP" </span><br><br><span># convert to list[Hallmark] required by irGSEA package</span><br>msigdb.h &lt;- msigdb %&gt;% <br>  dplyr::filter(collection==<span>"H"</span>) %&gt;% <br>  dplyr::select(c(<span>"geneset"</span>, <span>"symbol"</span>))<br>msigdb.h<span>$geneset</span> &lt;- factor(msigdb.h<span>$geneset</span>)<br>msigdb.h &lt;- msigdb.h %&gt;% <br>  dplyr::group_split(geneset, .keep = F) %&gt;%<br>  purrr::map( ~.x %&gt;% dplyr::pull(symbol) %&gt;% unique(.)) %&gt;%<br>  purrr::set_names(levels(msigdb.h<span>$geneset</span>))<br><span>#### Hallmark基因集打分 ####</span><br>pbmc3k.final &lt;- irGSEA.score(object = pbmc3k.final, assay = <span>"RNA"</span>, slot = <span>"data"</span>,<br>                             custom = T, geneset = msigdb.h, <br>                             method = c(<span>"AUCell"</span>,<span>"UCell"</span>,<span>"singscore"</span>,<span>"ssgsea"</span>, <span>"JASMINE"</span>, <span>"viper"</span>),<br>                             kcdf = <span>'Gaussian'</span>)<br><br><span># convert to list[go bp] required by irGSEA package</span><br>msigdb.go.bp &lt;- msigdb %&gt;% <br>  dplyr::filter(collection==<span>"C5:GO:BP"</span>) %&gt;% <br>  dplyr::select(c(<span>"geneset"</span>, <span>"symbol"</span>))<br>msigdb.go.bp<span>$geneset</span> &lt;- factor(msigdb.go.bp<span>$geneset</span>)<br>msigdb.go.bp &lt;- msigdb.go.bp %&gt;% <br>  dplyr::group_split(geneset, .keep = F) %&gt;%<br>  purrr::map( ~.x %&gt;% dplyr::pull(symbol) %&gt;% unique(.)) %&gt;%<br>  purrr::set_names(levels(msigdb.go.bp<span>$geneset</span>))<br><span>#### GO-BP基因集打分 ####</span><br>pbmc3k.final &lt;- irGSEA.score(object = pbmc3k.final, assay = <span>"RNA"</span>, slot = <span>"data"</span>,<br>                             custom = T, geneset = msigdb.go.bp, <br>                             method = c(<span>"AUCell"</span>,<span>"UCell"</span>,<span>"singscore"</span>,<span>"ssgsea"</span>, <span>"JASMINE"</span>, <span>"viper"</span>),<br>                             kcdf = <span>'Gaussian'</span>)<br><br><span># convert to list[KEGG] required by irGSEA package</span><br>msigdb.kegg &lt;- msigdb %&gt;% <br>  dplyr::filter(collection==<span>"C2:CP:KEGG_MEDICUS"</span>) %&gt;% <br>  dplyr::select(c(<span>"geneset"</span>, <span>"symbol"</span>))<br>msigdb.kegg<span>$geneset</span> &lt;- factor(msigdb.kegg<span>$geneset</span>)<br>msigdb.kegg &lt;- msigdb.kegg %&gt;% <br>  dplyr::group_split(geneset, .keep = F) %&gt;%<br>  purrr::map( ~.x %&gt;% dplyr::pull(symbol) %&gt;% unique(.)) %&gt;%<br>  purrr::set_names(levels(msigdb.kegg<span>$geneset</span>))<br><span>#### KEGG基因集打分 ####</span><br>pbmc3k.final &lt;- irGSEA.score(object = pbmc3k.final, assay = <span>"RNA"</span>, slot = <span>"data"</span>,<br>                             custom = T, geneset = msigdb.kegg, <br>                             method = c(<span>"AUCell"</span>,<span>"UCell"</span>,<span>"singscore"</span>,<span>"ssgsea"</span>, <span>"JASMINE"</span>, <span>"viper"</span>),<br>                             kcdf = <span>'Gaussian'</span>)<br><br><br><span># Look for the gene sets associated with angiogenesis from gene sets names and </span><br><span># gene sets descriptions</span><br><br>category &lt;- c(<span>"angiogenesis"</span>, <span>"vessel"</span>)<br>msigdb.vessel &lt;- list()<br><span>for</span> (i <span>in</span> category) {<br>  <span># Ignore case matching</span><br>  find.index.description &lt;- stringr::str_detect(msigdb<span>$description</span>, pattern = regex(all_of(i), ignore_case=TRUE))<br>  find.index.name &lt;- stringr::str_detect(msigdb<span>$geneset</span>, pattern = regex(all_of(i), ignore_case=TRUE))<br>  msigdb.vessel[[i]] &lt;- msigdb[find.index.description | find.index.name, ] %&gt;% mutate(category = i)<br>  <br>}<br>msigdb.vessel &lt;- do.call(rbind, msigdb.vessel)<br><br>head(msigdb.vessel)<br><span># # A tibble: 6 × 6</span><br><span># collection subcollection                      geneset            description    symbol category</span><br><span># &lt;chr&gt;      &lt;chr&gt;                              &lt;chr&gt;              &lt;chr&gt;          &lt;chr&gt;  &lt;chr&gt;   </span><br><span>#   1 C2:CGP     Chemical and Genetic Perturbations HU_ANGIOGENESIS_UP Up-regulated … HECW1  angioge…</span><br><span># 2 C2:CGP     Chemical and Genetic Perturbations HU_ANGIOGENESIS_UP Up-regulated … JADE2  angioge…</span><br><span># 3 C2:CGP     Chemical and Genetic Perturbations HU_ANGIOGENESIS_UP Up-regulated … SEMA3C angioge…</span><br><span># 4 C2:CGP     Chemical and Genetic Perturbations HU_ANGIOGENESIS_UP Up-regulated … STUB1  angioge…</span><br><span># 5 C2:CGP     Chemical and Genetic Perturbations HU_ANGIOGENESIS_UP Up-regulated … FAH    angioge…</span><br><span># 6 C2:CGP     Chemical and Genetic Perturbations HU_ANGIOGENESIS_UP Up-regulated … COL7A1 angioge…</span><br><br>length(unique(msigdb.vessel<span>$geneset</span>))<br><span># [1] 112</span><br><br><span># convert gene sets associated with angiogenesis to list </span><br><span># required by irGSEA package</span><br>msigdb.vessel &lt;- msigdb.vessel %&gt;% <br>  dplyr::select(c(<span>"geneset"</span>, <span>"symbol"</span>))<br>msigdb.vessel<span>$geneset</span> &lt;- factor(msigdb.vessel<span>$geneset</span>)<br>msigdb.vessel &lt;- msigdb.vessel %&gt;% <br>  dplyr::group_split(geneset, .keep = F) %&gt;%<br>  purrr::map( ~.x %&gt;% dplyr::pull(symbol) %&gt;% unique(.)) %&gt;%<br>  purrr::set_names(levels(msigdb.vessel<span>$geneset</span>))<br><span>#### 血管生成相关基因集打分 ####</span><br>pbmc3k.final &lt;- irGSEA.score(object = pbmc3k.final, assay = <span>"RNA"</span>, slot = <span>"data"</span>,<br>                             custom = T, geneset = msigdb.vessel, <br>                             method = c(<span>"AUCell"</span>,<span>"UCell"</span>,<span>"singscore"</span>,<span>"ssgsea"</span>, <span>"JASMINE"</span>, <span>"viper"</span>),<br>                             kcdf = <span>'Gaussian'</span>)<br><br><span>### gmt file ###</span><br><br><span># download gmt file</span><br>gmt_url &lt;- <span>"https://data.broadinstitute.org/gsea-msigdb/msigdb/release/2023.2.Hs/msigdb.v2023.2.Hs.symbols.gmt"</span><br>local_gmt &lt;- <span>"./msigdb.v2023.2.Hs.symbols.gmt"</span><br>download.file(gmt_url , local_gmt)<br><br>msigdb &lt;- clusterProfiler::read.gmt(<span>"./msigdb.v2023.2.Hs.symbols.gmt"</span>)<br><br><span># convert to list[hallmarker] required by irGSEA package</span><br>msigdb.h &lt;- msigdb %&gt;% <br>  dplyr::filter(str_detect(term, pattern = regex(<span>"HALLMARK_"</span>, ignore_case=TRUE)))<br>msigdb.h<span>$term</span> &lt;- factor(msigdb.h<span>$term</span>)<br>msigdb.h &lt;- msigdb.h %&gt;% <br>  dplyr::group_split(term, .keep = F) %&gt;%<br>  purrr::map( ~.x %&gt;% dplyr::pull(gene) %&gt;% unique(.)) %&gt;%<br>  purrr::set_names(levels(msigdb.h<span>$term</span>))<br><span>#### Hallmark基因集打分 ####</span><br>pbmc3k.final &lt;- irGSEA.score(object = pbmc3k.final, assay = <span>"RNA"</span>, slot = <span>"data"</span>,<br>                             custom = T, geneset = msigdb.h, <br>                             method = c(<span>"AUCell"</span>,<span>"UCell"</span>,<span>"singscore"</span>,<span>"ssgsea"</span>, <span>"JASMINE"</span>, <span>"viper"</span>),<br>                             kcdf = <span>'Gaussian'</span>)<br><br><span># convert to list[go bp] required by irGSEA package</span><br><br>msigdb.go.bp &lt;- msigdb %&gt;% <br>  dplyr::filter(str_detect(term, pattern = regex(<span>"GOBP_"</span>, ignore_case=TRUE)))<br>msigdb.go.bp<span>$term</span> &lt;- factor(msigdb.go.bp<span>$term</span>)<br>msigdb.go.bp &lt;- msigdb.go.bp %&gt;% <br>  dplyr::group_split(term, .keep = F) %&gt;%<br>  purrr::map( ~.x %&gt;% dplyr::pull(gene) %&gt;% unique(.)) %&gt;%<br>  purrr::set_names(levels(msigdb.go.bp<span>$term</span>))<br><span>#### GO-BP基因集打分 ####</span><br>pbmc3k.final &lt;- irGSEA.score(object = pbmc3k.final, assay = <span>"RNA"</span>, slot = <span>"data"</span>,<br>                             custom = T, geneset = msigdb.go.bp, <br>                             method = c(<span>"AUCell"</span>,<span>"UCell"</span>,<span>"singscore"</span>,<span>"ssgsea"</span>, <span>"JASMINE"</span>, <span>"viper"</span>),<br>                             kcdf = <span>'Gaussian'</span>)<br><br><span># convert to list[KEGG] required by irGSEA package</span><br>msigdb.kegg &lt;- msigdb %&gt;% <br>  dplyr::filter(str_detect(term, pattern = regex(<span>"KEGG_"</span>, ignore_case=TRUE)))<br>msigdb.kegg<span>$term</span> &lt;- factor(msigdb.kegg<span>$term</span>)<br>msigdb.kegg &lt;- msigdb.kegg %&gt;% <br>  dplyr::group_split(term, .keep = F) %&gt;%<br>  purrr::map( ~.x %&gt;% dplyr::pull(gene) %&gt;% unique(.)) %&gt;%<br>  purrr::set_names(levels(msigdb.kegg<span>$term</span>))<br><span>#### KEGG基因集打分 ####</span><br>pbmc3k.final &lt;- irGSEA.score(object = pbmc3k.final, assay = <span>"RNA"</span>, slot = <span>"data"</span>,<br>                             custom = T, geneset = msigdb.kegg, <br>                             method = c(<span>"AUCell"</span>,<span>"UCell"</span>,<span>"singscore"</span>,<span>"ssgsea"</span>, <span>"JASMINE"</span>, <span>"viper"</span>),<br>                             kcdf = <span>'Gaussian'</span>)<br><br></code></pre><p data-tool="mdnice编辑器"><strong>③irGSEA使用clusterProfiler包同款基因集进行打分</strong></p><pre data-tool="mdnice编辑器"><span></span><code><span>#### work with clusterProfiler package ####</span><br><span># load library</span><br>library(clusterProfiler)<br>library(tidyverse)<br><br><span>### kegg ###</span><br><span># download kegg pathway (human) and write as gson file</span><br>kk &lt;- clusterProfiler::gson_KEGG(species = <span>"hsa"</span>)<br>gson::write.gson(kk, file = <span>"./KEGG_20231128.gson"</span>)<br><br><span># read gson file</span><br>kk2 &lt;- gson::read.gson(<span>"./KEGG_20231123.gson"</span>)<br><span># Convert to a data frame</span><br>kegg.list &lt;- dplyr::left_join(kk2@gsid2name,<br>                              kk2@gsid2gene,<br>                              by = <span>"gsid"</span>)<br>head(kegg.list)<br><span># gsid               name      gene</span><br><span># 1 hsa01100 Metabolic pathways        10</span><br><span># 2 hsa01100 Metabolic pathways       100</span><br><span># 3 hsa01100 Metabolic pathways     10005</span><br><span># 4 hsa01100 Metabolic pathways     10007</span><br><span># 5 hsa01100 Metabolic pathways 100137049</span><br><span># 6 hsa01100 Metabolic pathways     10020</span><br><br><span># Convert gene ID to gene symbol</span><br>gene_name &lt;- clusterProfiler::bitr(kegg.list<span>$gene</span>, <br>                                   fromType = <span>"ENTREZID"</span>, <br>                                   toType = <span>"SYMBOL"</span>, <br>                                   OrgDb = <span>"org.Hs.eg.db"</span>)<br>kegg.list &lt;- dplyr::full_join(kegg.list,<br>                              gene_name,<br>                              by = c(<span>"gene"</span>=<span>"ENTREZID"</span>))<br><span># remove NA value if exist</span><br>kegg.list &lt;- kegg.list[complete.cases(kegg.list[, c(<span>"gene"</span>, <span>"SYMBOL"</span>)]), ]<br>head(kegg.list)<br><span># gsid               name      gene  SYMBOL</span><br><span># 1 hsa01100 Metabolic pathways        10    NAT2</span><br><span># 2 hsa01100 Metabolic pathways       100     ADA</span><br><span># 3 hsa01100 Metabolic pathways     10005   ACOT8</span><br><span># 4 hsa01100 Metabolic pathways     10007  GNPDA1</span><br><span># 5 hsa01100 Metabolic pathways 100137049 PLA2G4B</span><br><span># 6 hsa01100 Metabolic pathways     10020     GNE</span><br><br><span># convert to list required by irGSEA package</span><br>kegg.list<span>$name</span> &lt;- factor(kegg.list<span>$name</span>)<br>kegg.list &lt;- kegg.list %&gt;% <br>  dplyr::group_split(name, .keep = F) %&gt;%<br>  purrr::map( ~.x %&gt;% dplyr::pull(SYMBOL) %&gt;% unique(.)) %&gt;%<br>  purrr::set_names(levels(kegg.list<span>$name</span>))<br>head(kegg.list)<br><span>### 整理好之后就可以进行KEGG打分</span><br><span>#### KEGG基因集打分 ####</span><br>pbmc3k.final &lt;- irGSEA.score(object = pbmc3k.final, assay = <span>"RNA"</span>, slot = <span>"data"</span>,<br>                             custom = T, geneset = kegg.list, <br>                             method = c(<span>"AUCell"</span>,<span>"UCell"</span>,<span>"singscore"</span>,<span>"ssgsea"</span>, <span>"JASMINE"</span>, <span>"viper"</span>),<br>                             kcdf = <span>'Gaussian'</span>)<br><br><span>### go bp ###</span><br><span># download go bp (human) and write as gson file</span><br>go &lt;- clusterProfiler::gson_GO(OrgDb = <span>"org.Hs.eg.db"</span>, ont = <span>"BP"</span>)<br>gson::write.gson(go, file = <span>"./go_20231128.gson"</span>)<br><br><span># read gson file</span><br>go2 &lt;- gson::read.gson(<span>"./go_20231128.gson"</span>)<br><br><span># Convert to a data frame</span><br>go.list &lt;- dplyr::left_join(go2@gsid2name,<br>                            go2@gsid2gene,<br>                            by = <span>"gsid"</span>)<br>head(go.list)<br><span># gsid                             name gene</span><br><span># 1 GO:0000001        mitochondrion inheritance &lt;NA&gt;</span><br><span>#   2 GO:0000002 mitochondrial genome maintenance  142</span><br><span># 3 GO:0000002 mitochondrial genome maintenance  291</span><br><span># 4 GO:0000002 mitochondrial genome maintenance 1763</span><br><span># 5 GO:0000002 mitochondrial genome maintenance 1890</span><br><span># 6 GO:0000002 mitochondrial genome maintenance 2021</span><br><br><span># Convert gene ID to gene symbol</span><br>go.list &lt;- dplyr::full_join(go.list,<br>                            go2@gene2name,<br>                            by = c(<span>"gene"</span>=<span>"ENTREZID"</span>))<br><span># remove NA value if exist</span><br>go.list &lt;- go.list[complete.cases(go.list[, c(<span>"gene"</span>, <span>"SYMBOL"</span>)]), ]<br>head(go.list)<br><span># gsid                             name gene  SYMBOL</span><br><span># 2 GO:0000002 mitochondrial genome maintenance  142   PARP1</span><br><span># 3 GO:0000002 mitochondrial genome maintenance  291 SLC25A4</span><br><span># 4 GO:0000002 mitochondrial genome maintenance 1763    DNA2</span><br><span># 5 GO:0000002 mitochondrial genome maintenance 1890    TYMP</span><br><span># 6 GO:0000002 mitochondrial genome maintenance 2021   ENDOG</span><br><span># 7 GO:0000002 mitochondrial genome maintenance 3980    LIG3</span><br><br><span># convert to list required by irGSEA package</span><br>go.list<span>$name</span> &lt;- factor(go.list<span>$name</span>)<br>go.list &lt;- go.list %&gt;% <br>  dplyr::group_split(name, .keep = F) %&gt;%<br>  purrr::map( ~.x %&gt;% dplyr::pull(SYMBOL) %&gt;% unique(.)) %&gt;%<br>  purrr::set_names(levels(go.list<span>$name</span>))<br>head(go.list)<br><span>#### GO-BP基因集打分 ####</span><br>pbmc3k.final &lt;- irGSEA.score(object = pbmc3k.final, assay = <span>"RNA"</span>, slot = <span>"data"</span>,<br>                             custom = T, geneset = go.list, <br>                             method = c(<span>"AUCell"</span>,<span>"UCell"</span>,<span>"singscore"</span>,<span>"ssgsea"</span>, <span>"JASMINE"</span>, <span>"viper"</span>),<br>                             kcdf = <span>'Gaussian'</span>)<br></code></pre><h4 data-tool="mdnice编辑器"><span></span>4.irGSEA的完整流程<span></span></h4><pre data-tool="mdnice编辑器"><span></span><code>library(Seurat)<br>library(SeuratData)<br>library(RcppML)<br>library(irGSEA)<br>data(<span>"pbmc3k.final"</span>)<br><span># 基因集打分</span><br>pbmc3k.final &lt;- irGSEA.score(object = pbmc3k.final,assay = <span>"RNA"</span>, <br>                             slot = <span>"data"</span>, seeds = 123, ncores = 1,<br>                             min.cells = 3, min.feature = 0,<br>                             custom = F, geneset = NULL, msigdb = T, <br>                             species = <span>"Homo sapiens"</span>, category = <span>"H"</span>,  <br>                             subcategory = NULL, geneid = <span>"symbol"</span>,<br>                             method = c(<span>"AUCell"</span>,<span>"UCell"</span>,<span>"singscore"</span>,<span>"ssgsea"</span>, <span>"JASMINE"</span>, <span>"viper"</span>),<br>                             aucell.MaxRank = NULL, ucell.MaxRank = NULL, <br>                             kcdf = <span>'Gaussian'</span>)<br><span># 计算差异基因集，并进行RRA</span><br><span># 如果报错，考虑加句代码options(future.globals.maxSize = 100000 * 1024^5)</span><br>result.dge &lt;- irGSEA.integrate(object = pbmc3k.final,<br>                               group.by = <span>"seurat_annotations"</span>,<br>                               method = c(<span>"AUCell"</span>,<span>"UCell"</span>,<span>"singscore"</span>,<span>"ssgsea"</span>, <span>"JASMINE"</span>, <span>"viper"</span>))<br><span># 查看RRA识别的在多种打分方法中都普遍认可的差异基因集</span><br>geneset.show &lt;- result.dge<span>$RRA</span> %&gt;% <br>  dplyr::filter(pvalue &lt;= 0.05) %&gt;% <br>  dplyr::pull(Name) %&gt;% unique(.)<br></code></pre><h3 data-tool="mdnice编辑器"><span></span><span></span><span>可视化展示</span><span></span></h3><h4 data-tool="mdnice编辑器"><span></span>1)全局展示<span></span></h4><p data-tool="mdnice编辑器"><strong>①热图</strong></p><p data-tool="mdnice编辑器">你还可以把method从'RRA"换成“ssgsea”，展示特定基因集富集分析方法中差异上调或差异下调的基因集；</p><pre data-tool="mdnice编辑器"><span></span><code>irGSEA.heatmap.plot &lt;- irGSEA.heatmap(object = result.dge,<br>                                      method = <span>"RRA"</span>,<br>                                      top = 50,<br>                                      show.geneset = NULL)<br><br>irGSEA.heatmap.plot<br><br></code></pre><p data-tool="mdnice编辑器"><img data-imgfileid="100043156" data-ratio="0.6509259259259259" data-src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wxW0zhpl0fFDmP4Hr9iczb2BDp1b6H0icOa1Qf0v183DvzOH4EEhFIZRmH3dQO4sxrEec9R2FcAgMiaw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wxW0zhpl0fFDmP4Hr9iczb2BDp1b6H0icOa1Qf0v183DvzOH4EEhFIZRmH3dQO4sxrEec9R2FcAgMiaw/640?wx_fmt=png&amp;from=appmsg">默认展示前50，你也可以展示你想展示的基因集。例如，我想展示RRA识别的差异基因集。</p><pre data-tool="mdnice编辑器"><span></span><code>irGSEA.heatmap.plot1 &lt;- irGSEA.heatmap(object = result.dge, <br>                                       method = <span>"RRA"</span>,<br>                                       show.geneset = geneset.show)<br>irGSEA.heatmap.plot1<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100043157" data-ratio="0.6657407407407407" data-src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wxW0zhpl0fFDmP4Hr9iczb2BuDZib6icJ4WFf9Q9lj5RpZkahXd4BjyYQXnCiboCLbGlLr5XlE87Aj1oQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wxW0zhpl0fFDmP4Hr9iczb2BuDZib6icJ4WFf9Q9lj5RpZkahXd4BjyYQXnCiboCLbGlLr5XlE87Aj1oQ/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器"><strong>②气泡图</strong></p><pre data-tool="mdnice编辑器"><span></span><code>irGSEA.bubble.plot &lt;- irGSEA.bubble(object = result.dge,<br>                                    method = <span>"RRA"</span>,<br>                                     show.geneset = geneset.show)<br>irGSEA.bubble.plot<br><br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100043154" data-ratio="0.512962962962963" data-src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wxW0zhpl0fFDmP4Hr9iczb2BRrQbmpECrKe5cB3amIY9Xqf1tMX7fM2JR6ibJribKxzfwESmpiafkj7YQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wxW0zhpl0fFDmP4Hr9iczb2BRrQbmpECrKe5cB3amIY9Xqf1tMX7fM2JR6ibJribKxzfwESmpiafkj7YQ/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器"><strong>③upset plot</strong></p><p data-tool="mdnice编辑器">upset图展示了综合评估中每个细胞亚群具有统计学意义差异的基因集的数目，以及不同细胞亚群之间具有交集的差异基因集数目；</p><pre data-tool="mdnice编辑器"><span></span><code>irGSEA.upset.plot &lt;- irGSEA.upset(object = result.dge, <br>                                  method = <span>"RRA"</span>,<br>                                  mode = <span>"intersect"</span>,<br>                                  upset.width = 20,<br>                                  upset.height = 10,<br>                                  set.degree = 2,<br>                                  pt_size = grid::unit(2, <span>"mm"</span>))<br>irGSEA.upset.plot<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100043158" data-ratio="0.6780487804878049" data-src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wxW0zhpl0fFDmP4Hr9iczb2BqKKGicm45MlfTKCR7vG66tyjyPe7wqMNNB6GRCALehribOdnLTRXIddw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="615" src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wxW0zhpl0fFDmP4Hr9iczb2BqKKGicm45MlfTKCR7vG66tyjyPe7wqMNNB6GRCALehribOdnLTRXIddw/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">左边不同颜色的条形图代表不同的细胞亚群；上方的条形图代表具有交集的差异基因集的数目；中间的气泡图单个点代表单个细胞亚群，多个点连线代表多个细胞亚群取交集（）这里只展示两两取交集；</p><p data-tool="mdnice编辑器"><strong>④堆叠条形图</strong></p><p data-tool="mdnice编辑器">堆叠柱状图具体展示每种基因集富集分析方法中每种细胞亚群中上调、下调和没有统计学差异的基因集数目；</p><pre data-tool="mdnice编辑器"><span></span><code>irGSEA.barplot.plot &lt;- irGSEA.barplot(object = result.dge,<br>                                      method = c(<span>"AUCell"</span>, <span>"UCell"</span>, <span>"singscore"</span>,<br>                                                 <span>"ssgsea"</span>, <span>"JASMINE"</span>, <span>"viper"</span>, <span>"RRA"</span>))<br>irGSEA.barplot.plot<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100043160" data-ratio="0.512962962962963" data-src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wxW0zhpl0fFDmP4Hr9iczb2B7MYfh1wVq8FPsCzW5anZd6mAPqQCaICtqECIanic7QshyQklBnDhTBw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wxW0zhpl0fFDmP4Hr9iczb2B7MYfh1wVq8FPsCzW5anZd6mAPqQCaICtqECIanic7QshyQklBnDhTBw/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">上方的条形代表每个亚群中不同方法中差异的基因数目，红色代表上调的差异基因集，蓝色代表下调的差异基因集；中间的柱形图代表每个亚群中不同方法中上调、下调和没有统计学意义的基因集的比例；</p><h4 data-tool="mdnice编辑器"><span></span>2）局部展示<span></span></h4><p data-tool="mdnice编辑器"><strong>①密度散点图</strong></p><p data-tool="mdnice编辑器">密度散点图将基因集的富集分数和细胞亚群在低维空间的投影结合起来，展示了特定基因集在空间上的表达水平。</p><pre data-tool="mdnice编辑器"><span></span><code>scatterplot &lt;- irGSEA.density.scatterplot(object = pbmc3k.final,<br>                            method = <span>"UCell"</span>,<br>                            show.geneset = <span>"HALLMARK-INFLAMMATORY-RESPONSE"</span>,<br>                            reduction = <span>"umap"</span>)<br><br>scatterplot<br><br></code></pre><p data-tool="mdnice编辑器"><img data-imgfileid="100043159" data-ratio="0.6055555555555555" data-src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wxW0zhpl0fFDmP4Hr9iczb2BGgKWibkP4aRa2OgmsvQwUA9sEMW5mLMAxc1mYzKll5ibtszsUQFPTjrg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wxW0zhpl0fFDmP4Hr9iczb2BGgKWibkP4aRa2OgmsvQwUA9sEMW5mLMAxc1mYzKll5ibtszsUQFPTjrg/640?wx_fmt=png&amp;from=appmsg">其中，颜色越黄，密度分数越高，代表富集分数越高；</p><p data-tool="mdnice编辑器"><strong>②半小提琴图</strong></p><p data-tool="mdnice编辑器">半小提琴图同时以小提琴图（左边）和箱线图（右边）进行展示。不同颜色代表不同的细胞亚群；</p><pre data-tool="mdnice编辑器"><span></span><code>halfvlnplot &lt;- irGSEA.halfvlnplot(object = pbmc3k.final,<br>                                  method = <span>"UCell"</span>,<br>                                  show.geneset = <span>"HALLMARK-INFLAMMATORY-RESPONSE"</span>)<br><br>halfvlnplot<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100043163" data-ratio="0.6055555555555555" data-src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wxW0zhpl0fFDmP4Hr9iczb2Biaza7LgYYSE2molbxeiaWfL0yclkIvExZModoYyKEn6T1XlyMgWMOUzw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wxW0zhpl0fFDmP4Hr9iczb2Biaza7LgYYSE2molbxeiaWfL0yclkIvExZModoYyKEn6T1XlyMgWMOUzw/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">除此之外，还可以</p><pre data-tool="mdnice编辑器"><span></span><code>vlnplot &lt;- irGSEA.vlnplot(object = pbmc3k.final,<br>                          method = c(<span>"AUCell"</span>, <span>"UCell"</span>, <span>"singscore"</span>, <span>"ssgsea"</span>, <br>                                     <span>"JASMINE"</span>, <span>"viper"</span>),<br>                          show.geneset = <span>"HALLMARK-INFLAMMATORY-RESPONSE"</span>)<br>vlnplot<br></code></pre><p data-tool="mdnice编辑器"><img data-imgfileid="100043162" data-ratio="0.6055555555555555" data-src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wxW0zhpl0fFDmP4Hr9iczb2Bek0qOsmtXWiaSfZBcxd9ydG4TDCG12qXJoKE6uFQRCGqYAaLNTSMGXQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wxW0zhpl0fFDmP4Hr9iczb2Bek0qOsmtXWiaSfZBcxd9ydG4TDCG12qXJoKE6uFQRCGqYAaLNTSMGXQ/640?wx_fmt=png&amp;from=appmsg"><strong>③山峦图</strong></p><p data-tool="mdnice编辑器">山峦图中上方的核密度曲线展示了数据的主要分布，下方的条形编码图展示了细胞亚群具体的数量。不同颜色代表不同的细胞亚群，而横坐标代表不同的表达水平；</p><pre data-tool="mdnice编辑器"><span></span><code>ridgeplot &lt;- irGSEA.ridgeplot(object = pbmc3k.final,<br>                              method = <span>"UCell"</span>,<br>                              show.geneset = <span>"HALLMARK-INFLAMMATORY-RESPONSE"</span>)<br>ridgeplot<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100043167" data-ratio="0.6055555555555555" data-src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wxW0zhpl0fFDmP4Hr9iczb2BrklvKFvB80raNCwt82pgs4y6MbEI3wbYVY9ic3I0KJWPdwhYtW7MLhA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wxW0zhpl0fFDmP4Hr9iczb2BrklvKFvB80raNCwt82pgs4y6MbEI3wbYVY9ic3I0KJWPdwhYtW7MLhA/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器"><strong>④密度热图</strong></p><p data-tool="mdnice编辑器">密度热图展示了具体差异基因在不同细胞亚群中的表达和分布水平。颜色越红，代表富集分数越高；</p><pre data-tool="mdnice编辑器"><span></span><code>densityheatmap &lt;- irGSEA.densityheatmap(object = pbmc3k.final,<br>                                        method = <span>"UCell"</span>,<br>                                        show.geneset = <span>"HALLMARK-INFLAMMATORY-RESPONSE"</span>)<br>densityheatmap<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100043168" data-ratio="0.849624060150376" data-src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wxW0zhpl0fFDmP4Hr9iczb2BgolGDibUPzA1UWjXGKCNuvvIY7XPYQMs2vQpy9CPr8WibIBRpPXdVI0Q/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1064" src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wxW0zhpl0fFDmP4Hr9iczb2BgolGDibUPzA1UWjXGKCNuvvIY7XPYQMs2vQpy9CPr8WibIBRpPXdVI0Q/640?wx_fmt=png&amp;from=appmsg"></figure><h3 data-tool="mdnice编辑器"><span>参考资料</span><br></h3><section data-tool="mdnice编辑器"><span><span>[1] </span><p>AUCell: <span>https://doi.org/10.1038/nmeth.4463</span></p></span><span><span>[2] </span><p>UCell: <span>https://doi.org/10.1016/j.csbj.2021.06.043</span></p></span><span><span>[3] </span><p>singscore: <span>https://doi.org/10.1093/nar/gkaa802</span></p></span><span><span>[4] </span><p>ssgsea: <span>https://doi.org/10.1038/nature08460</span></p></span><span><span>[5] </span><p>JASMINE: <span>https://doi.org/10.7554/eLife.71994</span></p></span><span><span>[6] </span><p>VAM: <span>https://doi.org/10.1093/nar/gkaa582</span></p></span><span><span>[7] </span><p>scSE: <span>https://doi.org/10.1093/nar/gkz601</span></p></span><span><span>[8] </span><p>VISION: <span>https://doi.org/10.1038/s41467-019-12235-0</span></p></span><span><span>[9] </span><p>wsum: <span>https://doi.org/10.1093/bioadv/vbac016</span></p></span><span><span>[10] </span><p>wmean: <span>https://doi.org/10.1093/bioadv/vbac016</span></p></span><span><span>[11] </span><p>mdt: <span>https://doi.org/10.1093/bioadv/vbac016</span></p></span><span><span>[12] </span><p>viper: <span>https://doi.org/10.1038/ng.3593</span></p></span><span><span>[13] </span><p>GSVApy: <span>https://doi.org/10.1038/ng.3593</span></p></span><span><span>[14] </span><p>gficf: <span>https://doi.org/10.1093/nargab/lqad024</span></p></span><span><span>[15] </span><p>GSVA: <span>https://doi.org/10.1186/1471-2105-14-7</span></p></span><span><span>[16] </span><p>zscore: <span>https://doi.org/10.1371/journal.pcbi.1000217</span></p></span><span><span>[17] </span><p>plage: <span>https://doi.org/10.1186/1471-2105-6-225</span></p></span><span><span>[18] </span><p>ssGSEApy: <span>https://doi.org/10.1093/bioinformatics/btac757</span></p></span><span><span>[19] </span><p>viperpy: <span>https://doi.org/10.1093/bioinformatics/btac757</span></p></span><span><span>[20] </span><p>AddModuleScore: <span>https://doi.org/10.1126/science.aad0501</span></p></span><span><span>[21] </span><p>pagoda2: <span>https://doi.org/10.1038/nbt.4038</span></p></span><span><span>[22] </span><p>Sargent: <span>https://doi.org/10.1016/j.mex.2023.102196</span></p></span></section></section><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><h3 data-tool="mdnice编辑器"><span></span>文末交流群<span></span></h3></section><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><blockquote data-tool="mdnice编辑器"><p>R包开发是学术性质，也是公益的，所以交流群并不会设置门票也不会有二次收费。但是希望是一些高质量数据分析实战派小伙伴进群交流，可以是关于irGSEA的开发建议和意见，比如其它 Functional Class Scoring (FCS)方法，其它可视化方法，其它统计学算法的植入。如果仅仅是有关于irGSEA的使用方法疑问，可以直接看官方文档即可哈。</p></blockquote></section><p><img data-galleryid="" data-imgfileid="100043245" data-ratio="1.5258215962441315" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wxW0zhpl0fFDmP4Hr9iczb2BaPY15nCZGxVQfQJ241VEgl2xIYibSHRxZHaOMzrUECcian7dkicEU3t0w/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="213" src="https://mmbiz.qpic.cn/mmbiz_png/cZNhZQ6j4wxW0zhpl0fFDmP4Hr9iczb2BaPY15nCZGxVQfQJ241VEgl2xIYibSHRxZHaOMzrUECcian7dkicEU3t0w/640?wx_fmt=png&amp;from=appmsg"></p><p><br></p><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/VI4ISO6y5_rt8Yy_VnIR-w",target="_blank" rel="noopener noreferrer">原文链接</a>
