---
title: "文献美图复现 12: ggplot2 画 单细胞批量/多组对比火山图 & 展示DEG"
date: 2023-12-11T23:59:50Z
draft: ["false"]
tags: [
  "fetched",
  "生物慕课"
]
categories: ["Acdemic"]
---
文献美图复现 12: ggplot2 画 单细胞批量/多组对比火山图 & 展示DEG by 生物慕课
------
<div><p>目标效果</p><p>原图 A Spatiotemporal Organ-Wide Gene Expression and Cell Atlas of the Developing Human Heart, <strong>Cell 2019</strong> fig2D,H</p><ul><li><p><span>https://pubmed.ncbi.nlm.nih.gov/31835037/</span></p></li><li><p><span>DOI: 10.1016/j.cell.2019.11.025</span></p></li></ul><p><img data-imgfileid="100000498" data-ratio="0.5909090909090909" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaF1T4nQ8WAnQZS9CcXg1yqiaPDwTJiaeKQHVEFHmvxglbXweQiaDTwQLPrtcA1dTt36ba2NIfniaibibUY2bL4qQVic5Q/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="902" src="https://mmbiz.qpic.cn/mmbiz_png/iaF1T4nQ8WAnQZS9CcXg1yqiaPDwTJiaeKQHVEFHmvxglbXweQiaDTwQLPrtcA1dTt36ba2NIfniaibibUY2bL4qQVic5Q/640?wx_fmt=png&amp;from=appmsg"></p><p>本文给出2个模仿示例</p><ul><li><p>第一个是逐步绘制，适合加深理解并个性化定制；</p></li><li><p>第二个是自定义函数，适合快速直接使用。</p></li></ul><p><br></p><p>注：使用Seurat FindAllMarkers 求DEG时，建议设置 only.pos = F，否则本绘图可能报错。</p><p><br></p><p><strong><span>1.逐步绘制</span></strong></p><p>效果<br></p><p><img data-imgfileid="100000499" data-ratio="1.1573236889692586" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaF1T4nQ8WAnQZS9CcXg1yqiaPDwTJiaeKQicD2tdB6qLacAMx4UIXX7vXib6icAjposqX9diaR2TdEZXSxLcB6MXPvRw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="553" src="https://mmbiz.qpic.cn/mmbiz_png/iaF1T4nQ8WAnQZS9CcXg1yqiaPDwTJiaeKQicD2tdB6qLacAMx4UIXX7vXib6icAjposqX9diaR2TdEZXSxLcB6MXPvRw/640?wx_fmt=png&amp;from=appmsg"></p><p>代码：<br></p><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="powershell"><code><span>###############</span></code><code><span># input dataset</span></code><code><span>###############</span></code><code><span>DimPlot(pbmc, label = T)</span></code><code><span>pbmc.markers &lt;- FindAllMarkers(pbmc, min.pct = 0.25, logfc.threshold = 0.25)</span></code><code><span>pbmc.markers %&gt;% group_by(cluster) %&gt;% top_n(n = 2, wt = abs(avg_log2FC) )</span></code><code><span><br></span></code><code><span>head(pbmc.markers )</span></code><code><span># &gt; head(pbmc.markers )</span></code><code><span>#p_val avg_log2FC pct.1 pct.2     p_val_adj     cluster  gene</span></code><code><span>#RPS12 1.273332e-143  0.7298951 1.000 0.991 1.746248e-139 Naive CD4 T RPS12</span></code><code><span>#RPS6  6.817653e-143  0.6870694 1.000 0.995 9.349729e-139 Naive CD4 T  RPS6</span></code><code><span><br></span></code><code><span><br></span></code><code><span>###############</span></code><code><span># begin to plot</span></code><code><span>###############</span></code><code><span># get data</span></code><code><span>DEG = pbmc.markers</span></code><code><span><br></span></code><code><span># p.adj&lt;0.01? In origin paper</span></code><code><span>#DEG$label &lt;- ifelse(DEG$p_val_adj&lt;0.01, "adjusted P-value&lt;0.01", "adjusted P-value&gt;=0.01")</span></code><code><span>DEG$label &lt;- ifelse(DEG$p_val_adj&lt;0.0001,</span></code><code><span>                    ifelse(DEG$avg_log2FC&gt;0, "Up", "Down"),</span></code><code><span>                    "None")</span></code><code><span>DEG$label = factor(DEG$label, levels = c("Up", "None", "Down"))</span></code><code><span>table(DEG$label)</span></code><code><span>#Down None   Up</span></code><code><span># 931 2048 1680</span></code><code><span>#adjusted P-value&lt;0.01 adjusted P-value&gt;=0.01</span></code><code><span>#                 3144                   1515</span></code><code><span><br></span></code><code><span><br></span></code><code><span>if(0){</span></code><code><span>  cut_off_pvalue = 0.0001</span></code><code><span>  cut_off_logFC = 2</span></code><code><span>  # only used for single volcano plot. not used later</span></code><code><span>  DEG$Sig = ifelse(DEG$p_val_adj&lt; cut_off_pvalue &amp; abs(DEG$avg_log2FC) &gt;= cut_off_logFC,</span></code><code><span>                   ifelse(DEG$avg_log2FC &gt; cut_off_logFC ,'Up','Down'),</span></code><code><span>                   'None')</span></code><code><span>  table(DEG$Sig)</span></code><code><span>  #Down None   Up</span></code><code><span>  #115 4330  214</span></code><code><span><br></span></code><code><span>  head(DEG)</span></code><code><span>  #              p_val avg_log2FC pct.1 pct.2     p_val_adj     cluster  gene  Sig</span></code><code><span>  #RPS12 1.273332e-143  0.7298951 1.000 0.991 1.746248e-139 Naive CD4 T RPS12 None</span></code><code><span><br></span></code><code><span><br></span></code><code><span>  # only one volcano</span></code><code><span>  ggplot(DEG, aes(x = avg_log2FC, y = -log10(p_val_adj), colour=Sig)) +</span></code><code><span>    geom_point(alpha=0.4, size=1.5) +</span></code><code><span>    scale_color_manual(values=c("#546de5", "#d2dae2","#ff4757"))+</span></code><code><span>    geom_vline(xintercept=c(-1,1),lty=4,col="black",lwd=0.8) +</span></code><code><span>    labs(x="log2(Fold Change)",y="-log10 (P-value)")+</span></code><code><span>    theme_bw()+</span></code><code><span>    ggtitle("Volcano Plot")+</span></code><code><span>    theme(</span></code><code><span>      plot.title = element_text(hjust = 0.5),</span></code><code><span>      legend.position="right",legend.title = element_blank()</span></code><code><span>    )</span></code><code><span>}</span></code><code><span><br></span></code><code><span># expand cluster on x axis;</span></code><code><span># Among p.adj &lt;0.05 genes, select logFC top10 &amp; bottom10, annotate gene symbol</span></code><code><span>DEG.text = NULL;</span></code><code><span>for(cluster.name in unique(DEG$cluster) ){</span></code><code><span>  print(cluster.name)</span></code><code><span>  tmp=DEG %&gt;% filter(cluster==cluster.name &amp; p_val_adj &lt;0.05) %&gt;% top_n(n = 10, wt = avg_log2FC)</span></code><code><span>  DEG.text=rbind(DEG.text, tmp);</span></code><code><span>  tmp=DEG %&gt;% filter(cluster==cluster.name &amp; p_val_adj &lt;0.05 ) %&gt;% top_n(n = 10, wt = -avg_log2FC)</span></code><code><span>  DEG.text=rbind(DEG.text, tmp);</span></code><code><span>}</span></code><code><span>dim(DEG.text)</span></code><code><span>table(DEG.text$label, DEG.text$cluster)</span></code><code><span>head(DEG.text)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span># plot</span></code><code><span>p1 =  ggplot() +</span></code><code><span>  geom_jitter(data = DEG, aes(cluster, y=avg_log2FC, color=label), size = 0.85,width =0.4)+</span></code><code><span>  geom_jitter(data = DEG.text, aes(x = cluster, y = avg_log2FC, color = label), size = 1,width =0.4)</span></code><code><span>p1</span></code><code><span><br></span></code><code><span><br></span></code><code><span># bar background of each cluster</span></code><code><span>len = nlevels(DEG$cluster)</span></code><code><span><br></span></code><code><span>dfbarUp  &lt;- data.frame(x=c(1:len),</span></code><code><span>                       y=DEG %&gt;% group_by(cluster) %&gt;% top_n(1, wt=avg_log2FC) %&gt;% pull(avg_log2FC))</span></code><code><span>dfbarDown&lt;- data.frame(x=c(1:len),</span></code><code><span>                       y=DEG %&gt;% group_by(cluster) %&gt;% top_n(1, wt=-avg_log2FC) %&gt;% pull(avg_log2FC))</span></code><code><span>p1bar &lt;- ggplot()+</span></code><code><span>  geom_col(data = dfbarUp,  mapping = aes(x = x,y = y),fill = "#dcdcdc",alpha = 0.6)+</span></code><code><span>  geom_col(data = dfbarDown,mapping = aes(x = x,y = y),fill = "#dcdcdc",alpha = 0.6)</span></code><code><span>p1bar</span></code><code><span><br></span></code><code><span><br></span></code><code><span>###############</span></code><code><span># Error: when putting background bar at back? ----</span></code><code><span>###############</span></code><code><span>p1bar + geom_jitter(data = DEG, aes(cluster, y=avg_log2FC, color=label), size = 0.85,width =0.4)</span></code><code><span># Error: Discrete value supplied to continuous scale</span></code><code><span><br></span></code><code><span># store the original value, and change it to numeric</span></code><code><span>#DEG$cluster2=DEG$cluster</span></code><code><span>#DEG$cluster=as.numeric(DEG$cluster) # solve the error, but revoke another: figure legend</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span># Another way: put background bar at front of dots.</span></code><code><span>p2=ggplot() +</span></code><code><span>  #2. dot plot</span></code><code><span>  geom_jitter(data = DEG, aes(cluster, y=avg_log2FC, color=label), size = 0.55,width =0.4, show.legend = F)+</span></code><code><span>  geom_jitter(data = DEG.text, aes(x = cluster, y = avg_log2FC, color = label), size = 1,width =0.4, show.legend = F)+</span></code><code><span>  #theme_classic()</span></code><code><span>  #1. bar background</span></code><code><span>  geom_col(data = dfbarUp,  mapping = aes(x = x, y = y),fill = "#dcdcdc",alpha = 0.2)+</span></code><code><span>  geom_col(data = dfbarDown,mapping = aes(x = x, y = y),fill = "#dcdcdc",alpha = 0.2)</span></code><code><span>p2</span></code><code><span><br></span></code><code><span><br></span></code><code><span># add color box on x axis</span></code><code><span>dfcol &lt;- data.frame(x=c(1:len), y=0, label=c(1:len), cluster=levels(DEG$cluster) )</span></code><code><span>dfcol</span></code><code><span>mycol &lt;- scales::hue_pal()(len)  #c("#E64B357F","#00A0877F","#F39B7F7F","#8491B47F","#DC00007F")</span></code><code><span>p3 &lt;- p2 +</span></code><code><span>  geom_tile(data = dfcol, aes(x=x, y=y, fill=cluster), height=0.5, color = NA,</span></code><code><span>            #fill = mycol,</span></code><code><span>            alpha = 1, show.legend = F)</span></code><code><span>p3</span></code><code><span><br></span></code><code><span><br></span></code><code><span># annotate gene symbol on plot</span></code><code><span>library(ggrepel)</span></code><code><span>p4 &lt;- p3+geom_text_repel(data=DEG.text,</span></code><code><span>                         aes(x=cluster,y=avg_log2FC,label=gene),</span></code><code><span>                         force = 1.2,</span></code><code><span>                         size = 2.5, #font size of gene symbols</span></code><code><span>                         arrow = arrow(length = unit(0.008, "npc"), type = "open", ends = "last"))+</span></code><code><span>  scale_color_manual(name=NULL, breaks = c("Up", "None", "Down"),values = c("#c0252d", "grey", "navy")) # dot color</span></code><code><span>p4</span></code><code><span><br></span></code><code><span># add text/number in box on x axis</span></code><code><span>p5 &lt;- p4+</span></code><code><span>  labs(x="Cluster",y="average log2FC")+</span></code><code><span>  geom_text(data=dfcol, aes(x=x, y=y, label=label),</span></code><code><span>            size = 6, #cluster font size</span></code><code><span>            color ="white")+</span></code><code><span>  guides( color = guide_legend(override.aes = list(size=5))) #legend circle size</span></code><code><span>p5</span></code><code><span><br></span></code><code><span># themes</span></code><code><span>p6 &lt;- p5+theme_minimal()+theme(</span></code><code><span>  axis.title = element_text(size = 13, #face = "bold",</span></code><code><span>                            color = "black"),</span></code><code><span>  axis.line.y = element_line(color = "black", size = 1.2),</span></code><code><span>  axis.line.x = element_blank(),</span></code><code><span><br></span></code><code><span>  #axis.text.x = element_blank(), #no x axis text</span></code><code><span>  axis.text.x = element_text(angle=60, hjust=1,size=12,color="black"), #rotate x axis text</span></code><code><span><br></span></code><code><span>  panel.grid = element_blank(),</span></code><code><span>  #legend.position = "top",</span></code><code><span>  legend.direction = "vertical",</span></code><code><span>  legend.justification = c(1,0),</span></code><code><span>  legend.text = element_text(size = 15)</span></code><code><span>); </span></code><code><span>p6</span></code><code><span><br></span></code><code><span># save: 004_3volcano_singleCell_multi</span></code><code><span>ggsave(filename = "01/004_3volcano_singleCell_multi.pdf", width=7.1, height=8.2, useDingbats=F)</span></code></pre></section><p><br></p><p>效果<br></p><p><img data-imgfileid="100000500" data-ratio="0.859375" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaF1T4nQ8WAnQZS9CcXg1yqiaPDwTJiaeKQGRtzlCosJ19Iibwf2oRlA8IvATvAyOia036n7JSUjLxbkepTuBELEoDA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="256" src="https://mmbiz.qpic.cn/mmbiz_png/iaF1T4nQ8WAnQZS9CcXg1yqiaPDwTJiaeKQGRtzlCosJ19Iibwf2oRlA8IvATvAyOia036n7JSUjLxbkepTuBELEoDA/640?wx_fmt=png&amp;from=appmsg"></p><p>代码：</p><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="xml"><code><span>###################</span></code><code><span># Part II: circled number followed by text</span></code><code><span>###################</span></code><code><span># circle filled background followed by text ----</span></code><code><span># len = nlevels(DEG$cluster)</span></code><code><span># dfcol <span>&lt;<span>-</span> <span>data.frame</span>(<span>x</span>=<span>c(1:len),</span> <span>y</span>=<span>0,</span> <span>label</span>=<span>c(1:len),</span> <span>cluster</span>=<span>levels(DEG$cluster)</span> )</span></span></code><code><span><span>dfcol</span></span></code><code><span>#  <span>x</span> <span>y</span> <span>label</span>      <span>cluster</span></span></code><code><span>#<span>1</span> <span>1</span> <span>0</span>     <span>1</span>  <span>Naive</span> <span>CD4</span> <span>T</span></span></code><code><span>#<span>2</span> <span>2</span> <span>0</span>     <span>2</span>   <span>CD14</span>+ <span>Mono</span></span></code><code><span>#<span>3</span> <span>3</span> <span>0</span>     <span>3</span> <span>Memory</span> <span>CD4</span> <span>T</span></span></code><code><span>#<span>4</span> <span>4</span> <span>0</span>     <span>4</span>            <span>B</span></span></code><code><span>#<span>5</span> <span>5</span> <span>0</span>     <span>5</span>        <span>CD8</span> <span>T</span></span></code><code><span>#<span>6</span> <span>6</span> <span>0</span>     <span>6</span> <span>FCGR3A</span>+ <span>Mono</span></span></code><code><span>#<span>7</span> <span>7</span> <span>0</span>     <span>7</span>           <span>NK</span></span></code><code><span>#<span>8</span> <span>8</span> <span>0</span>     <span>8</span>           <span>DC</span></span></code><code><span>#<span>9</span> <span>9</span> <span>0</span>     <span>9</span>     <span>Platelet</span></span></code><code><span><br></span></code><code><span># <span>use</span> <span>2</span> <span>col:</span> <span>label</span> <span>and</span> <span>cluster</span></span></code><code><span><span>pb2</span>=<span>ggplot(</span> <span>dfcol</span>, <span>aes</span>(<span>x</span>=<span>0,</span> <span>y</span>=<span>-label,</span> <span>color</span>=<span>cluster)</span> )+</span></code><code><span>  <span>geom_point</span>(<span>size</span>=<span>8,</span> <span>show.legend</span> = <span>F)+</span></span></code><code><span>  <span>xlim</span>(<span>-0.1</span>, <span>0.5</span>) +</span></code><code><span>  <span>geom_text</span>(<span>data</span>=<span>dfcol,</span> <span>aes</span>(<span>x</span>=<span>0,</span> <span>y</span>=<span>-label,</span> <span>label</span>=<span>label),</span> <span>color</span>=<span>"white"</span>, <span>size</span>=<span>6)+</span></span></code><code><span>  <span>geom_label</span>(<span>data</span>=<span>dfcol,</span> <span>aes</span>(<span>x</span>=<span>0.03,</span> <span>y</span>=<span>-label,</span> <span>label</span>=<span>cluster),</span> <span>color</span>=<span>"black"</span>, <span>size</span>=<span>6,</span></span></code><code><span>             <span>hjust</span>=<span>"left"</span>, <span>label.size</span>=<span>0)+</span></span></code><code><span>  <span>theme_nothing</span>()</span></code><code><span><span>pb2</span></span></code><code><span><span>ggsave</span>(<span>filename</span> = <span>"01/004_3volcano_singleCell_circleAnnotation.pdf"</span>,</span></code><code><span>       <span>plot</span>=<span>pb2,</span></span></code><code><span>       <span>width</span>=<span>2.8,</span> <span>height</span>=<span>2.3,</span> <span>useDingbats</span>=<span>F)</span></span></code><code><span><br></span></code><code><span># <span>with</span> <span>DimPot</span></span></code><code><span><span>pb1</span>=<span>DimPlot(pbmc,</span> <span>label</span>=<span>T)+NoLegend();</span></span></code><code><span><span>pb1</span></span></code><code><span><span>library</span>(<span>patchwork</span>)</span></code><code><span><span>pb1</span> / <span>pb2</span></span></code></pre></section><p><br></p><p><br></p><p><strong>2.定义函数法<br></strong></p><p>效果图<br></p><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="xml"><code><span>DimPlot(pbmc, label = T)</span></code><code><span>pbmc.markers <span>&lt;<span>-</span> <span>FindAllMarkers</span>(<span>pbmc</span>, <span>min.pct</span> = <span>0.25,</span> <span>logfc.threshold</span> = <span>0.25)</span></span></span></code><code><span><span><span>pbmc.markers</span> %&gt;</span>% group_by(cluster) %&gt;% top_n(n = 2, wt = abs(avg_log2FC) )</span></code><code><span><br></span></code><code><span>DEG = pbmc.markers</span></code><code><span>multiVolcanoPlot(DEG, color.pals)</span></code><code><span><br></span></code><code><span># 画图参数</span></code><code><span>#multiVolcanoPlot(DEG, color.pals)</span></code><code><span>#multiVolcanoPlot(scObj.markers)</span></code><code><span>#multiVolcanoPlot(scObj.markers, onlyAnnotateUp = F)</span></code><code><span>#multiVolcanoPlot(scObj.markers, top_marker = 10, onlyAnnotateUp = F)</span></code><code><span># onlyAnnotateUp =T 则只展示上调的基因，默认T</span></code></pre></section><p><img data-imgfileid="100000501" data-ratio="0.720160481444333" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaF1T4nQ8WAnQZS9CcXg1yqiaPDwTJiaeKQbwicGQU6JdvgTlicvDj1SxDJibW6Qgg73m1AxddAraPRr48tyD6RicrBeQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="997" src="https://mmbiz.qpic.cn/mmbiz_png/iaF1T4nQ8WAnQZS9CcXg1yqiaPDwTJiaeKQbwicGQU6JdvgTlicvDj1SxDJibW6Qgg73m1AxddAraPRr48tyD6RicrBeQ/640?wx_fmt=png&amp;from=appmsg"></p><p><br></p><p>函数定义：</p><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="ruby"><code><span><span># &gt; head(DEG)</span></span></code><code><span><span>#              p_val avg_log2FC pct.1 pct.2     p_val_adj     cluster  gene label</span></span></code><code><span><span>#RPS12 1.273332e-143  0.7298951 1.000 0.991 1.746248e-139 Naive CD4 T RPS12    Up</span></span></code><code><span><span>#RPS6  6.817653e-143  0.6870694 1.000 0.995 9.349729e-139 Naive CD4 T  RPS6    Up</span></span></code><code><span><span>#</span></span></code><code><span><br></span></code><code><span><br></span></code><code><span>color.pals = c(<span>"#DC143C"</span>,<span>"#0000FF"</span>,<span>"#20B2AA"</span>,<span>"#FFA500"</span>,<span>"#9370DB"</span>,<span>"#98FB98"</span>,<span>"#F08080"</span>,<span>"#1E90FF"</span>,<span>"#7CFC00"</span>,<span>"#FFFF00"</span>,</span></code><code><span>               <span>"#808000"</span>,<span>"#FF00FF"</span>,<span>"#FA8072"</span>,<span>"#7B68EE"</span>,<span>"#9400D3"</span>,<span>"#800080"</span>,<span>"#A0522D"</span>,<span>"#D2B48C"</span>,<span>"#D2691E"</span>,<span>"#87CEEB"</span>,<span>"#40E0D0"</span>,<span>"#5F9EA0"</span>,</span></code><code><span>               <span>"#FF1493"</span>,<span>"#0000CD"</span>,<span>"#008B8B"</span>,<span>"#FFE4B5"</span>,<span>"#8A2BE2"</span>,<span>"#228B22"</span>,<span>"#E9967A"</span>,<span>"#4682B4"</span>,<span>"#32CD32"</span>,<span>"#F0E68C"</span>,<span>"#FFFFE0"</span>,<span>"#EE82EE"</span>,</span></code><code><span>               <span>"#FF6347"</span>,<span>"#6A5ACD"</span>,<span>"#9932CC"</span>,<span>"#8B008B"</span>,<span>"#8B4513"</span>,<span>"#DEB887"</span>)</span></code><code><span><br></span></code><code><span><span>#</span></span></code><code><span><span>#' multi volcano plot for scRNA-seq</span></span></code><code><span><span>#' <span>@version</span> 0.2 change legend order</span></span></code><code><span><span>#' <span>@version</span> 0.3 add max_overlaps for annotation</span></span></code><code><span><span>#'</span></span></code><code><span><span>#' <span>@param</span> dat Seurat FindAllMarkers returns, must set only.pos = F;</span></span></code><code><span><span>#' <span>@param</span> color.arr color list, default same as Seurat</span></span></code><code><span><span>#' <span>@param</span> onlyAnnotateUp only annote gene symbols for up genes</span></span></code><code><span><span>#' <span>@param</span> log2Foldchang threshold for annotation</span></span></code><code><span><span>#' <span>@param</span> adjp  threshold for annotation</span></span></code><code><span><span>#' <span>@param</span> top_marker gene number for annotation</span></span></code><code><span><span>#' <span>@param</span> max_overlaps annotation label overlapping</span></span></code><code><span><span>#'</span></span></code><code><span><span>#' <span>@return</span> ggplot2 obj</span></span></code><code><span><span>#' <span>@export</span></span></span></code><code><span><span>#'</span></span></code><code><span><span>#' <span>@examples</span></span></span></code><code><span>multiVolcanoPlot = function(dat, color.arr=NULL, onlyAnnotateUp=T,</span></code><code><span>                            log2Foldchang=<span>0</span>.<span>58</span>, adjp=<span>0</span>.<span>05</span>, top_marker=<span>5</span>, </span></code><code><span>                            max_overlaps=<span>10</span>, width=<span>0</span>.<span>9</span>){</span></code><code><span>  library(dplyr)</span></code><code><span>  library(ggrepel)</span></code><code><span>  <span># set default color list</span></span></code><code><span>  <span>if</span>(is.null(color.arr)){</span></code><code><span>    len = length(unique(dat$cluster))</span></code><code><span>    color.arr=scales::hue_pal()(len)</span></code><code><span>  }</span></code><code><span><br></span></code><code><span>  dat.plot &lt;- dat %&gt;% mutate(</span></code><code><span>    <span>"significance"</span>=case_when(p_val_adj &lt; adjp &amp; avg_log2FC &gt;= log2Foldchang  ~ <span>'Up'</span>,</span></code><code><span>                             p_val_adj &lt; adjp &amp; avg_log2FC &lt;= -log2Foldchang  ~ <span>'Down'</span>,</span></code><code><span>                             TRUE ~ <span>'None'</span>))</span></code><code><span>  tbl = table(dat.plot$significance)</span></code><code><span>  print( tbl )</span></code><code><span>  background.dat &lt;- data.frame(</span></code><code><span>    dat.plot %&gt;% group_by(cluster) %&gt;% filter(avg_log2FC&gt;<span>0</span>) %&gt;%</span></code><code><span>      summarise(<span>"y.localup"</span>=max(avg_log2FC)),</span></code><code><span>    dat.plot %&gt;% group_by(cluster) %&gt;% filter(avg_log2FC&lt;=<span>0</span>) %&gt;%</span></code><code><span>      summarise(<span>"y.localdown"</span>=min(avg_log2FC)),</span></code><code><span>    x.local=seq(<span>1</span><span>:length</span>(unique(dat.plot$cluster)))</span></code><code><span>  ) %&gt;% select(-cluster.<span>1</span>)</span></code><code><span>  <span>#names(background.dat)</span></span></code><code><span>  <span>#head(background.dat)</span></span></code><code><span>  <span>#dim(background.dat)</span></span></code><code><span><br></span></code><code><span>  <span>#</span></span></code><code><span>  x.number &lt;- background.dat %&gt;% select(cluster, x.local)</span></code><code><span>  dat.plot &lt;- dat.plot%&gt;% left_join(x.number,by = <span>"cluster"</span>)</span></code><code><span>  <span>#names(dat.plot)</span></span></code><code><span>  <span>#head(dat.plot)</span></span></code><code><span><br></span></code><code><span>  <span>#selecting top-up and top-down proteins</span></span></code><code><span>  dat.marked.up &lt;- dat.plot %&gt;% filter(significance==<span>"Up"</span>) %&gt;%</span></code><code><span>    group_by(cluster) %&gt;% arrange(-avg_log2FC) %&gt;%</span></code><code><span>    top_n(top_marker,abs(avg_log2FC))</span></code><code><span>  dat.marked.down &lt;- dat.plot %&gt;% filter(significance==<span>"Down"</span>) %&gt;%</span></code><code><span>    group_by(cluster) %&gt;% arrange(avg_log2FC) %&gt;%</span></code><code><span>    top_n(top_marker,abs(avg_log2FC))</span></code><code><span>  dat.marked &lt;- dat.marked.up %&gt;% bind_rows(dat.marked.down)</span></code><code><span>  <span>#referring group information data</span></span></code><code><span>  dat.infor &lt;- background.dat %&gt;%</span></code><code><span>    mutate(<span>"y.infor"</span>=rep(<span>0</span>,length(cluster)))</span></code><code><span>  <span>#names(dat.infor)</span></span></code><code><span>  <span>#dim(dat.infor)</span></span></code><code><span>  <span>#head(dat.infor)</span></span></code><code><span><br></span></code><code><span>  <span>##plotting:</span></span></code><code><span>  <span>#setting color by loading local color schemes</span></span></code><code><span>  vol.plot &lt;- ggplot()+</span></code><code><span>    <span># background</span></span></code><code><span>    geom_col(background.dat,mapping=aes(x.local, y.localup),</span></code><code><span>             fill=<span>"grey80"</span>, alpha=<span>0</span>.<span>2</span>, width=<span>0</span>.<span>9</span>, just = <span>0</span>.<span>5</span>)+</span></code><code><span>    geom_col(background.dat,mapping=aes(x.local,y.localdown),</span></code><code><span>             fill=<span>"grey80"</span>, alpha=<span>0</span>.<span>2</span>, width=<span>0</span>.<span>9</span>, just = <span>0</span>.<span>5</span>)+</span></code><code><span>    <span># point plot</span></span></code><code><span>    geom_jitter(dat.plot, mapping=aes(x.local, avg_log2FC, <span>#x= should be number, Not string or factor</span></span></code><code><span>                                      color=significance),</span></code><code><span>                size=<span>0</span>.<span>8</span>, width = <span>0</span>.<span>4</span>, alpha= <span>1</span>)+</span></code><code><span>    scale_color_manual(name=<span>"significance"</span>, </span></code><code><span>                       breaks = c(<span>'Up'</span>, <span>'None'</span>, <span>'Down'</span>),</span></code><code><span>                       values = c(<span>"#d56e5e"</span>,<span>"#cccccc"</span>, <span>"#5390b5"</span>)) + <span>#set color for: Down None   Up</span></span></code><code><span>    geom_tile(dat.infor, mapping=aes(x.local, y.infor), <span>#x axis color box</span></span></code><code><span>              height = log2Foldchang*<span>1.3</span>,</span></code><code><span>              fill = color.arr[<span>1</span><span>:length</span>(unique(dat.plot$cluster))],</span></code><code><span>              alpha = <span>0</span>.<span>5</span>,</span></code><code><span>              width=width) +</span></code><code><span>    labs(x=NULL,y=<span>"log2 Fold change"</span>)+</span></code><code><span>    geom_text(dat.infor, mapping=aes(x.local,y.infor,label=cluster))+</span></code><code><span>    <span># Down is not recommend, not meaningful, hard to explain; so prefer dat.marked.up to dat.marked</span></span></code><code><span>    ggrepel::geom_label_repel(data=<span>if</span>(onlyAnnotateUp) dat.marked.up <span>else</span> dat.marked, <span>#gene symbol, of up group default</span></span></code><code><span>                              mapping=aes(x=x.local, y=avg_log2FC, label=gene),</span></code><code><span>                              force = <span>2</span>, <span>#size=2,</span></span></code><code><span>                              max.overlaps = max_overlaps,</span></code><code><span>                              label.size = <span>0</span>, <span>#no border</span></span></code><code><span>                              fill=<span>"#00000000"</span>, <span>#box fill color</span></span></code><code><span>                              seed = <span>233</span>,</span></code><code><span>                              min.segment.length = <span>0</span>,</span></code><code><span>                              force_pull = <span>2</span>,</span></code><code><span>                              box.padding = <span>0</span>.<span>1</span>,</span></code><code><span>                              segment.linetype = <span>3</span>,</span></code><code><span>                              <span>#segment.color = 'black',</span></span></code><code><span>                              <span>#segment.alpha = 0.5,</span></span></code><code><span>                              <span>#direction = "x", #line direction</span></span></code><code><span>                              hjust = <span>0</span>.<span>5</span>)+</span></code><code><span>    annotate(<span>"text"</span>, x=<span>1.5</span>, y=max(background.dat$y.localup)+<span>1</span>,</span></code><code><span>             label=paste<span>0</span>(<span>"|log2FC|&gt;="</span>, log2Foldchang, <span>" &amp; FDR&lt;"</span>, adjp))+</span></code><code><span>    theme_classic(base_size = <span>12</span>)+</span></code><code><span><br></span></code><code><span>    theme(</span></code><code><span>      axis.title = element_text(size = <span>13</span>, color = <span>"black"</span>),</span></code><code><span>      axis.text = element_text(size = <span>15</span>, color = <span>"black"</span>),</span></code><code><span>      axis.line.y = element_line(color = <span>"black"</span>, size = <span>0</span>.<span>8</span>),</span></code><code><span>      <span>#</span></span></code><code><span>      axis.line.x = element_blank(), <span>#no x axis line</span></span></code><code><span>      axis.ticks.x = element_blank(), <span>#no x axis ticks</span></span></code><code><span>      axis.title.x = element_blank(), <span>#</span></span></code><code><span>      axis.text.x = element_blank(),</span></code><code><span>      <span>#</span></span></code><code><span>      legend.spacing.x = unit(<span>0</span>.<span>1</span>,<span>'cm'</span>),</span></code><code><span>      legend.key.width = unit(<span>0</span>.<span>5</span>,<span>'cm'</span>),</span></code><code><span>      legend.key.height = unit(<span>0</span>.<span>5</span>,<span>'cm'</span>),</span></code><code><span>      legend.background = element_blank(),</span></code><code><span>      legend.box = <span>"horizontal"</span>,</span></code><code><span>      legend.position = c(<span>0</span>.<span>13</span>, <span>0</span>.<span>77</span>),legend.justification = c(<span>1</span>,<span>0</span>)</span></code><code><span>    )+</span></code><code><span>    guides( <span>#color = guide_legend( override.aes = list(size=5) ), #legend circle size</span></span></code><code><span>      color=guide_legend( override.aes = list(size=<span>5</span>), title=<span>"Change"</span>)</span></code><code><span>    )</span></code><code><span>  <span>#guides(fill=guide_legend(title="Change"))+ #change legend title</span></span></code><code><span>  vol.plot</span></code><code><span>}</span></code><code><span><span>#multiVolcanoPlot(DEG, color.pals)</span></span></code><code><span>multiVolcanoPlot(scObj.markers.time)</span></code><code><span>multiVolcanoPlot(scObj.markers.time, onlyAnnotateUp = F)</span></code><code><span><br></span></code><code><span><span># Also at: https://blog.csdn.net/wangjunliang/article/details/134850344</span></span></code></pre></section><p><br></p><p><span>-- End --</span></p><p><img data-galleryid="" data-ratio="0.37777777777777777" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaF1T4nQ8WAnHXEickJsNHxveZ0tddRnUkOriad8jptJ3WWAF2Ra3wWB7ia0OPmAcHxCz6pMuJ62W6GJUvuyglSjWw/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaF1T4nQ8WAnHXEickJsNHxveZ0tddRnUkOriad8jptJ3WWAF2Ra3wWB7ia0OPmAcHxCz6pMuJ62W6GJUvuyglSjWw/640?wx_fmt=png"></p><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/QUjjzfGQ85iEN9QmPVj2dw",target="_blank" rel="noopener noreferrer">原文链接</a>
