---
title: "共表达网络| WGCNA与hdWGCNA实操"
date: 2023-04-08T00:02:31Z
draft: ["false"]
tags: [
  "fetched",
  "朴素的科研打工仔"
]
categories: ["Acdemic"]
---
共表达网络| WGCNA与hdWGCNA实操 by 朴素的科研打工仔
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><p data-tool="mdnice编辑器"><strong>写在前面的话</strong><br></p><section><strong>      加权基因共表达网络分析 (WGCNA, Weighted correlation network analysis)是用来描述不同样品之间基因关联模式的系统生物学方法，利用数千或近万个变化最大的基因或全部基因的信息识别感兴趣的基因集并结合与表型之间的关联鉴定候补生物标记基因或治疗靶点。感受Bulk水平到单细胞水平WGCNA的应用~~~</strong></section><h2 data-tool="mdnice编辑器"><span></span><span>普通转录组</span></h2><p data-tool="mdnice编辑器"><img data-ratio="0.8838782412626832" data-src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26EhsnyQqTI2cxM3kibvJXvcWu09ZC25hmUFYqkJsRfoPCriaYEsvqw9BXA/640?wx_fmt=png" data-type="png" data-w="887" src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26EhsnyQqTI2cxM3kibvJXvcWu09ZC25hmUFYqkJsRfoPCriaYEsvqw9BXA/640?wx_fmt=png">引：Langfelder, P., Horvath, S. WGCNA: an R package for weighted correlation network analysis. BMC Bioinformatics 9, 559 (2008).</p><h3 data-tool="mdnice编辑器"><span></span><span>数据清洗与整理</span><span></span></h3><pre data-tool="mdnice编辑器"><span></span><code># 读取基因表达矩阵数据<br>fpkm &lt;- read.table(<span>"data/fpkm.txt"</span>, header = T, row.names = <span>1</span>, check.names = F)<br>## 推荐用DESeq2处理后的矩阵<br>counts &lt;- read.table(<span>"data/counts.txt"</span>, header = T, row.names = <span>1</span>, check.names = F)<br>condition &lt;- factor(c(rep(<span>"Con"</span>,<span>6</span>)),rep(<span>"Treat"</span>,<span>6</span>))<br>colData &lt;- data.frame(row.names=colnames(counts),condition)<br>dds &lt;- DESeqDataSetFromMatrix(round(counts),colData, design = ~condition))<br>vst &lt;- vst(dds, blind=FALSE)<br>vst &lt;- assay(vst) # 得到标准化后的矩阵<br># 选取基因<br>## <span>1</span>.标准差选择<br>fpkm_sd &lt;-  apply(fpkm,<span>1</span>,sd)<br>fpkm_sd_sorted &lt;-  order(fpkm_sd, decreasing = T)<br>## 选择前<span>5000</span>个标准差较大的基因<br>fpkm_num &lt;- fpkm_sd_sorted[<span>1</span>:<span>5000</span>]<br>fpkm_filter &lt;-  fpkm[fpkm_num,]<br>WGCNA_matrix &lt;- t(fpkm_filter)<br>## <span>2</span>.绝对中位差选择<br>WGCNA_matrix = t(fpkm[order(apply(fpkm,<span>1</span>,mad), decreasing = T)[<span>1</span>:<span>5000</span>],])<br>## 任性点，那么好的电脑当然全部基因<br>WGCNA_matrix = t(fpkm)<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>library(WGCNA)<br>### 通过样本聚类识别离群样本，去除离群样本 ###<br>sampleTree = hclust(dist(WGCNA_matrix), method = <span>"average"</span>);<br>plot(sampleTree, main = <span>"Sample clustering to detect outliers"</span>, sub=<span>""</span>, xlab=<span>""</span>, cex.lab = <span>1.5</span>,cex.axis = <span>1.5</span>, cex.main = <span>2</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.30092592592592593" data-src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26E4iagOC5J8cKIgYla1Xa8icY3Y4Imnibqnxa8Woc5PX73biay3edLBRbR2A/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26E4iagOC5J8cKIgYla1Xa8icY3Y4Imnibqnxa8Woc5PX73biay3edLBRbR2A/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code># 去除离群的聚类样本，高于cutHeight的样本剔除<br>clust &lt;- cutreeStatic(sampleTree, cutHeight = <span>120</span>, minSize = <span>10</span>)<br>table(clust)<br># clust<br># <span>0</span>   <span>1</span>     ##<span>0</span>是要去除的，<span>1</span>是要保存的<br># <span>15</span> <span>162</span><br># 提取clust <span>1</span>聚类中的样本<br>keepSamples &lt;- (clust==<span>1</span>)<br>WGCNA_matrix &lt;- WGCNA_matrix[keepSamples, ]<br># 记录基因和样本数<br>nGenes = ncol(WGCNA_matrix)#基因数<br>nSamples = nrow(WGCNA_matrix)#样本数<br></code></pre><h3 data-tool="mdnice编辑器"><span></span><span>软阈值的确认</span><span></span></h3><pre data-tool="mdnice编辑器"><span></span><code>### 选择软阈值 ###<br>powers &lt;- c(c(<span>1</span>:<span>10</span>), seq(from = <span>12</span>, to=<span>20</span>, by=<span>2</span>))<br># 进行网络拓扑分析<br>sft &lt;- pickSoftThreshold(WGCNA_matrix, powerVector = powers, verbose = <span>5</span>)<br># 无尺度网络阈值的选择<br>plot(sft$fitIndices[,<span>1</span>], <br>     -sign(sft$fitIndices[,<span>3</span>])*sft$fitIndices[,<span>2</span>],<br>     xlab=<span>"Soft Threshold (power)"</span>,#x轴<br>     ylab=<span>"Scale Free Topology Model Fit,signed R^2"</span>,type=<span>"n"</span>,#y轴<br>     main = paste(<span>"Scale independence"</span>));#标题<br>text(sft$fitIndices[,<span>1</span>], <br>     -sign(sft$fitIndices[,<span>3</span>])*sft$fitIndices[,<span>2</span>],<br>     labels=powers,<br>     cex=cex1,<br>     col=<span>"red"</span>);<br><br># 用红线标出R^<span>2</span>的参考值<br>abline(h=<span>0.90</span>,col=<span>"red"</span>)<br># 平均连接度<br>plot(sft$fitIndices[,<span>1</span>], <br>     sft$fitIndices[,<span>5</span>],<br>     xlab=<span>"Soft Threshold (power)"</span>,<br>     ylab=<span>"Mean Connectivity"</span>, <br>     type=<span>"n"</span>,<br>     main = paste(<span>"Mean connectivity"</span>))<br>text(sft$fitIndices[,<span>1</span>], <br>     sft$fitIndices[,<span>5</span>], <br>     labels=powers, <br>     cex=cex1,<br>     col=<span>"red"</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.5370370370370371" data-src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26EvsWxwRn5DZbwsNgtOoF6BgibOphiamG4eDxqLoA0J8Xnc5fwaCnicnROA/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26EvsWxwRn5DZbwsNgtOoF6BgibOphiamG4eDxqLoA0J8Xnc5fwaCnicnROA/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code>softpower &lt;- sft$powerEstimate<br>ADJ &lt;- abs(cor(datExpr,use=<span>"p"</span>)) #相关性取绝对值再幂次<br>k &lt;- as.vector(apply(ADJ,<span>2</span>,sum,na.rm=T))#对ADJ的每一列取和<br>hist(k)<br>scaleFreePlot(k,main=<span>"Check scale free topology"</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.4759259259259259" data-src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26Eyz8Z3SQOWaY0mxjXBguTEOsfuTibZjAyeJxVw7gj3TrKMKASsqx67uQ/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26Eyz8Z3SQOWaY0mxjXBguTEOsfuTibZjAyeJxVw7gj3TrKMKASsqx67uQ/640?wx_fmt=png"></figure><h3 data-tool="mdnice编辑器"><span></span><span>共表达网络构建</span><span></span></h3><pre data-tool="mdnice编辑器"><span></span><code>### 一步构建网络 ###<br>net &lt;- blockwiseModules(WGCNA_matrix, #处理好的表达矩阵<br>                       power = sft$powerEstimate,#选择的软阈值<br>                       TOMType = <span>"unsigned"</span>, #拓扑矩阵类型，none表示邻接矩阵聚类，unsigned最常用，构建无方向<br>                       maxBlockSize =nGenes，#计算机能处理的最大模块的基因数量<br>                       minModuleSize = <span>30</span>,#网络模块包含的最少基因数<br>                       reassignThreshold = <span>0</span>, #模块间基因重分类的阈值<br>                       mergeCutHeight = <span>0.25</span>,#合并相异度低于<span>0.25</span>的模块<br>                       numericLabels = TRUE, #<span>true</span>，返回模块的数字标签 <span>false</span>返回模块的颜色标签<br>                       pamRespectsDendro = FALSE,#调用动态剪切树算法识别网络模块后，进行第二次的模块比较，合并相关性高的模块<br>                       saveTOMs = TRUE,#保存拓扑矩阵<br>                       saveTOMFileBase = <span>"data/TOM"</span>, <br>                       verbose = <span>3</span>,#<span>0</span>，不返回任何信息，＞<span>0</span>返回计算过程<br>                       randomSeed = <span>0407</span> #确保数据可重复性<br>                       )<br></code></pre><h3 data-tool="mdnice编辑器"><span></span><span>可视化</span><span></span></h3><pre data-tool="mdnice编辑器"><span></span><code># 将标签转化为颜色<br>mergedColors = labels2colors(net$colors)<br># 绘制聚类和网络模块对应图<br>plotDendroAndColors(dendro = net$dendrograms[[<span>1</span>]], #hclust函数生成的聚类结果<br>                    colors = mergedColors[net$blockGenes[[<span>1</span>]]],#基因对应的模块颜色<br>                    groupLabels = <span>"Module colors"</span>,#分组标签<br>                    dendroLabels = FALSE, #<span>false</span>,不显示聚类图的每个分支名称<br>                    hang = <span>0.03</span>,#调整聚类图分支所占的高度<br>                    addGuide = TRUE, #为聚类图添加辅助线<br>                    guideHang = <span>0.05</span>,#辅助线所在高度<br>                    main = <span>"Gene dendrogram and module colors"</span>)<br># 加载TOM矩阵<br>load(<span>"data/TOM.RData"</span>)<br># 网络特征向量<br>MEs = moduleEigengenes(WGCNA_matrix, mergedColors)$eigengenes<br># 对特征向量排序<br>MEs = orderMEs(MEs)#这个MEs是可用于寻找hub基因<br># 可视化模块间的相关性<br>plotEigengeneNetworks(MEs, <span>""</span>, marDendro = c(<span>0</span>,<span>4</span>,<span>1</span>,<span>2</span>), <br>                      marHeatmap = c(<span>3</span>,<span>4</span>,<span>1</span>,<span>2</span>), cex.lab = <span>0.8</span>, <br>                      xLabelsAngle = <span>90</span>)<br># <span>TOMplot<br><span>load</span><span>(net$TOMFiles, verbose=T)</span><br>dissTOM </span>= <span>1</span>-TOMsimilarityFromExpr(WGCNA_matrix, power = sft$powerEstimate); #<span>1</span>-相关性=相异性<br>TOMplot(dissTOM, #拓扑矩阵，该矩阵记录了每个节点之间的相关性<br>        net$dendrograms[[<span>1</span>]], #基因的聚类结果<br>        mergedColors[net$blockGenes[[<span>1</span>]]], #基因对应的模块颜色<br>        main = <span>"Network heatmap plot, selected genes"</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7539149888143176" data-src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26EI0209crx3K2PqVbB6WaVUCzIT0zCSic6zMTmicnZ9ibnarqG387nMVOtg/640?wx_fmt=png" data-type="png" data-w="894" src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26EI0209crx3K2PqVbB6WaVUCzIT0zCSic6zMTmicnZ9ibnarqG387nMVOtg/640?wx_fmt=png"></figure><figure data-tool="mdnice编辑器"><img data-ratio="1.5317460317460319" data-src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26Elt5drRicLwOdAx5He5BsgYfm39tiaaH7mK1fhHBxjicNfZdzkBnF3UDkw/640?wx_fmt=png" data-type="png" data-w="504" src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26Elt5drRicLwOdAx5He5BsgYfm39tiaaH7mK1fhHBxjicNfZdzkBnF3UDkw/640?wx_fmt=png"></figure><figure data-tool="mdnice编辑器"><img data-ratio="1.0644391408114557" data-src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26EDQLZsnLqicNKicl6y4g2FlT7nL16Xhmk6fDoicE3YWanaLibXJaFibabQTg/640?wx_fmt=png" data-type="png" data-w="838" src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26EDQLZsnLqicNKicl6y4g2FlT7nL16Xhmk6fDoicE3YWanaLibXJaFibabQTg/640?wx_fmt=png"></figure><h3 data-tool="mdnice编辑器"><span></span><span>模块与特征矩阵的联系</span><span></span></h3><pre data-tool="mdnice编辑器"><span></span><code># 读入特征矩阵<br>## 用于关联分析的性状必须是数值型特征，区域或分类变量，需要转换为<span>0</span>-<span>1</span>矩阵的形式<br>trait &lt;- <span>"TraitsClean.txt"</span><br><br># 将特征信息转换为颜色，白色表示低，红色表示高，灰色表示缺失<br>traitColors = numbers2colors(datTraits, signed = FALSE)<br><br># 样本聚类图与样本性状热图<br>plotDendroAndColors(sampleTree2, <br>                    traitColors,<br>                    groupLabels = names(datTraits), <br>                    main = <span>"Sample dendrogram and trait heatmap"</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.2861111111111111" data-src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26EiamByPmV9TTPtyxHbFIHxjibPgpiaicxInyUHONzotbc1fr6tEzXjrdNMg/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26EiamByPmV9TTPtyxHbFIHxjibPgpiaicxInyUHONzotbc1fr6tEzXjrdNMg/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code># 网络的分析<br>## 基因和所在模块信息<br>gene_module &lt;- data.frame(ID=colnames(WGCNA_matrix), <span>module</span>=mergedColors)<br>gene_module = gene_module[order(gene_module$<span>module</span>),]<br>write.table(gene_module,<span>"gene_module.csv"</span>),row.names=F)<br>## 保存模块代表性信息<br>MEs_col &lt;- MEs<br>MEs_colt &lt;- as.data.frame(t(MEs_col))<br>colnames(MEs_colt) &lt;- rownames(WGCNA_matrix)<br>write.csv(MEs_colt,<span>"module_eipgengene.csv"</span>)<br>## 基因模块与特征信息的关系<br># 计算模块特征矩阵和样本特征矩阵的相关度。<br>moduleTraitCor=cor(MEs, Trait, use=<span>"p"</span>)<br>write.csv(<span>"modPhysiological.cor.csv"</span>,moduleTraitCor,sep=<span>"\t"</span>,quote=F)<br>moduleTraitPvalue=corPvalueStudent(moduleTraitCor, nSamples)<br>write.csv(<span>"modPhysiological.p.csv"</span>,moduleTraitPvalue,sep=<span>"\t"</span>,quote=F)<br>textMatrix=paste(signif(moduleTraitCor,<span>2</span>),<span>"\n("</span>,signif(moduleTraitPvalue,<span>1</span>),<span>")"</span>,sep=<span>""</span>)<br># 基因模块与样本特征信息相关性图<br>labeledHeatmap(Matrix=moduleTraitCor,#模块和表型的相关性矩阵<br>               xLabels=colnames(datTraits),<br>               yLabels=names(MEs),<br>               ySymbols=names(MEs),<br>               colorLabels=FALSE,<br>               colors=blueWhiteRed(<span>50</span>),<br>               textMatrix=textMatrix,<br>               setStdMargins=FALSE,<br>               cex.text=<span>0.5</span>,<br>               cex.lab=<span>0.5</span>,<br>               zlim=c(-<span>1</span>,<span>1</span>),<br>               main=paste(<span>"Module-trait relationships"</span>))<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7617021276595745" data-src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26EW1ImAjhNL76oWTCxh8QSGnTh48kGk2SWmCRm1t42KcgOACfh69yqgw/640?wx_fmt=png" data-type="png" data-w="940" src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26EW1ImAjhNL76oWTCxh8QSGnTh48kGk2SWmCRm1t42KcgOACfh69yqgw/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code>## 最关键的就是得到基因模块与样本特征这张图。找到自己感兴趣的模块，可以找hub基因，功能富集分析等等<br>## 模块中的hub基因<br>## 为每一个模块寻找hub基因<br>HubGenes &lt;- chooseTopHubInEachModule(WGCNA_matrix,#WGCNA分析输入的表达矩阵<br>                                     mergedColors )#模块颜色信息<br># 保存hub基因结果<br>write.csv (HubGenes,<span>"HubGenes_of_each_module.csv"</span>,col.names = F)<br>#### 与某种特征相关的hub基因<br>NS &lt;- networkScreening(trait$age,#年龄<br>                      MEs,<br>                      WGCNA_matrix)<br>## 单一特征与所有模块关联分析<br>M_stage &lt;- as.numeric(cor(trait$M_stage,datExpr, use=<span>"p"</span>))<br>GeneSignificance = abs(GS)<br>plotModuleSignificance(GeneSignificance, moduleColors, ylim=c(<span>0</span>,<span>0.2</span>), main=<span>"Gene significance for M stage across module"</span>)<br>## 单一模块与某一表型相关性<br>M_stage = as.data.frame(trait$M_stage)<br># 分析自己感兴趣的特征信息，此处以M_stage为示例<br>names(M_stage) = <span>"M_stage"</span><br># 模块对应的颜色<br>modNames = substring(names(MEs), <span>3</span>)<br># 模块对应的颜色<br>modNames = substring(names(MEs), <span>3</span>)<br># 计算基因模块特征<br>geneModuleMembership = as.data.frame(cor(WGCNA_matrix, MEs, use = <span>"p"</span>));<br>MMPvalue = as.data.frame(corPvalueStudent(as.matrix(geneModuleMembership), nSamples));<br># 对结果进行命名<br>names(geneModuleMembership) = paste(<span>"MM"</span>, modNames, sep=<span>""</span>);<br>names(MMPvalue) = paste(<span>"p.MM"</span>, modNames, sep=<span>""</span>);<br># 计算M分期基因特征显著性<br>geneTraitSignificance = as.data.frame(cor(WGCNA_matrix, M_stage, use = <span>"p"</span>));<br>GSPvalue = as.data.frame(corPvalueStudent(as.matrix(geneTraitSignificance), nSamples));<br># 对结果进行命名<br>names(geneTraitSignificance) = paste(<span>"GS."</span>, names(M_stage), sep=<span>""</span>)<br>names(GSPvalue) = paste(<span>"p.GS."</span>, names(M_stage), sep=<span>""</span>)<br># 设置需要分析的模块名称，此处为brown模块<br><span>module</span> = <span>"brown"</span><br># 提取brown模块数据<br>column = match(<span>module</span>, modNames);<br>moduleGenes = moduleColors==<span>module</span>;<br># 可视化brown模块与M分期的相关性分析结果<br>verboseScatterplot(abs(geneModuleMembership[moduleGenes, column]),<br>                   abs(geneTraitSignificance[moduleGenes, <span>1</span>]),<br>                   xlab = paste(<span>"Module Membership in"</span>, <span>module</span>, <span>"module"</span>),<br>                   ylab = <span>"Gene significance for M Stage"</span>,<br>                   main = paste(<span>"Module membership vs. gene significance\n"</span>),<br>                   cex.main = <span>1.2</span>, cex.lab = <span>1.2</span>, cex.axis = <span>1.2</span>, col = <span>module</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.5101851851851852" data-src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26EjMNlE0icBk9HHTUpl4ggT8YRwgNkgn7YDIU6y24O2JH8Js5CGvAZvjg/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26EjMNlE0icBk9HHTUpl4ggT8YRwgNkgn7YDIU6y24O2JH8Js5CGvAZvjg/640?wx_fmt=png"></figure><figure data-tool="mdnice编辑器"><img data-ratio="0.9986263736263736" data-src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26Ecg3ia2iaObpAupakoOOviaUxGxg1mvbJ2QSwf13868hr8821RsoicVotWQ/640?wx_fmt=png" data-type="png" data-w="728" src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26Ecg3ia2iaObpAupakoOOviaUxGxg1mvbJ2QSwf13868hr8821RsoicVotWQ/640?wx_fmt=png"></figure><h3 data-tool="mdnice编辑器"><span></span><span>转化cytoscape格式文件</span><span></span></h3><pre data-tool="mdnice编辑器"><span></span><code>### 输出cytoscape可视化<br># 输出全部网络模块<br>cyt = exportNetworkToCytoscape(TOM,<br>                               edgeFile = <span>"CytoscapeInput-edges-all.txt"</span>,#基因间的共表达关系<br>                               nodeFile = <span>"CytoscapeInput-nodes-all.txt"</span>,#<br>                               weighted = TRUE,<br>                               threshold = <span>0.1</span>,<br>                               nodeNames = geneID$SYMBOL,<br>                               altNodeNames = geneID$ENTREZID,<br>                               nodeAttr = moduleColors)<br># 输出感兴趣网络模块<br>modules = c(<span>"brown"</span>, <span>"red"</span>)<br># 选择上面模块中包含的基因<br>inModule = is.finite(match(moduleColors, modules))<br>modGenes = geneID[inModule,]<br># 选择指定模块的TOM矩阵<br>modTOM = TOM[inModule, inModule]<br></code></pre><h2 data-tool="mdnice编辑器"><span></span><span>单细胞转录组</span></h2><p data-tool="mdnice编辑器"><img data-ratio="0.6269430051813472" data-src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26EdLlAqicA6PibIsjU3fFgCgXxicAia7pDLTMRicy2GV45YVrJRcQveeYz5dQ/640?wx_fmt=png" data-type="png" data-w="965" src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26EdLlAqicA6PibIsjU3fFgCgXxicAia7pDLTMRicy2GV45YVrJRcQveeYz5dQ/640?wx_fmt=png">引：Morabito, S., Miyoshi, E., Michael, N. et al. Single-nucleus chromatin accessibility and transcriptomic characterization of Alzheimer’s disease. Nat Genet 53, 1143–1155 (2021).</p><h3 data-tool="mdnice编辑器"><span></span><span>细胞亚群选取与数据清洗</span><span></span></h3><pre data-tool="mdnice编辑器"><span></span><code>library(hdWGCNA)<br>library(Seurat)<br>library(tidyverse)<br>library(WGCNA)<br># 读取单细胞数据集<br>scRNA=readRDS(<span>'scRNA.RDS'</span>)<br># 提T细胞亚群,重新降维聚类<br>scRNA=subset(scRNA,celltype %in% <span>'T_cells'</span>)<br>scRNA &lt;- SCTransform(scRNA)<br>scRNA &lt;- FindNeighbors(scRNA, dims = <span>1</span>:<span>10</span>)<br>scRNA &lt;- FindClusters(scRNA)<br>scRNA &lt;- RunUMAP(scRNA, dims = <span>1</span>:<span>10</span>)<br>DimPlot(scRNA,label=T)<br>#过滤出至少在<span>5</span>%的细胞中表达的基因<br>scRNA &lt;- SetupForWGCNA(<br>  scRNA,<br>  gene_select = <span>"fraction"</span>,<br>  fraction = <span>0.05</span>,<br>  wgcna_name = <span>"T_wgcna"</span><br>)<br></code></pre><h3 data-tool="mdnice编辑器"><span></span><span>构建metacells及软阈值的确定</span><span></span></h3><pre data-tool="mdnice编辑器"><span></span><code># 构建metacells<br>scRNA&lt;- MetacellsByGroups(<br>  seurat_obj = scRNA,k=<span>20</span>,<br>  max_shared = <span>10</span>,<br>  # group.by可以是组织类型、细胞类型或分组<br>  group.by = c(<span>"celltype"</span>,<span>'orig.ident'</span>), <br>  ident.group = <span>'celltype'</span> <br>)<br># 标准化<br>scRNA &lt;- NormalizeMetacells(scRNA)<br>metacell_obj &lt;- GetMetacellObject(scRNA)<br># 转置表达矩阵<br>seurat_obj  &lt;- SetDatExpr(<br>  scRNA,<br>  group_name = <span>"T_cells"</span>, <br>  group.by=<span>'celltype'</span><br>)<br># 选择softpower<br>seurat_obj &lt;- TestSoftPowers(<br>  seurat_obj,<br>  setDatExpr = FALSE<br>)<br># 可视化<br>plot_list &lt;- PlotSoftPowers(seurat_obj)<br>wrap_plots(plot_list, ncol=<span>2</span>)<br># 查看powerTable<br>power_table &lt;- GetPowerTable(seurat_obj)<br>head(power_table)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.6472222222222223" data-src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26EDPOoJU1cK7ySpO3ARVicicZ7n6rbGQicn3BCWgU9onoT1OW8DLjEFDdcw/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26EDPOoJU1cK7ySpO3ARVicicZ7n6rbGQicn3BCWgU9onoT1OW8DLjEFDdcw/640?wx_fmt=png"></figure><h3 data-tool="mdnice编辑器"><span></span><span>共表达网络构建</span><span></span></h3><pre data-tool="mdnice编辑器"><span></span><code># 构建共表达网络<br>softpower=<span>4</span><br>seurat_obj &lt;- ConstructNetwork(<br>  seurat_obj, soft_power=softpower,<br>  group.by=<span>'celltype'</span>, group_name=<span>'T_cells'</span>,setDatExpr = F)<br># 可视化WGCNA网络<br>PlotDendrogram(seurat_obj, main=<span>'hdWGCNA Dendrogram'</span>)<br># 获取TOM矩阵<br>TOM &lt;- GetTOM(seurat_obj)<br># 计算模块协调特征<br>seurat_obj &lt;- Seurat::ScaleData(<br>  seurat_obj,<br>  features = GetWGCNAGenes(seurat_obj)<br>)<br># 计算<span>ME<br><span>library</span><span>(harmony)</span><br>seurat_obj &lt;- <span>ModuleEigengenes</span><span>(<br>  seurat_obj,<br>  group.by.vars=<span>"orig.ident"</span> #harmony对象<br>)</span><br>seurat_obj &lt;- <span>ModuleConnectivity</span><span>(seurat_obj)</span><br># 可视化每个模块根据kME打分排序的基因<br><span>PlotKMEs</span><span>(seurat_obj, ncol=<span>3</span>)</span><br># 获取hub genes<br>hub_df &lt;- <span>GetHubGenes</span><span>(seurat_obj, n_hubs = <span>25</span>)</span><br><br><span>PlotKMEs</span><span>(seurat_obj)</span><br></span></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.6583333333333333" data-src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26EXVJZ5pGF0ic0GqaY6lcwjRZyW6hQTVv0CFP9FVs77KPLvNyv5MqvB0Q/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26EXVJZ5pGF0ic0GqaY6lcwjRZyW6hQTVv0CFP9FVs77KPLvNyv5MqvB0Q/640?wx_fmt=png"></figure><figure data-tool="mdnice编辑器"><img data-ratio="0.6527777777777778" data-src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26EbK4IF1ssVBBHHyO64IfibWNRr8Th7zIkYGROswV6Cr0iaFJpbUtoCXyQ/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26EbK4IF1ssVBBHHyO64IfibWNRr8Th7zIkYGROswV6Cr0iaFJpbUtoCXyQ/640?wx_fmt=png"></figure><h3 data-tool="mdnice编辑器"><span></span><span>模块与细胞亚群的联系</span><span></span></h3><pre data-tool="mdnice编辑器"><span></span><code>## 模块间的相关性<br>library(igraph)<br>library(qgraph)<br># 画模块间相关性图<br>ModuleCorrelogram(seurat_obj, sig.level = <span>0.001</span>, pch.cex=<span>2</span>)<br># hub基因打分<br># <span>1</span>.Seurat<br>seurat_obj &lt;- ModuleExprScore(<br>  seurat_obj,<br>  n_genes = <span>25</span>,<br>  method=<span>'Seurat'</span><br>)<br># <span>2</span>.<span>Ucell<br><span>library</span><span>(UCell)</span><br>seurat_obj &lt;- <span>ModuleExprScore</span><span>(<br>  seurat_obj,<br>  n_genes = <span>25</span>,<br>  method=<span>'UCell'</span><br>)</span><br># 可视化<br>plot_list &lt;- <span>ModuleFeaturePlot</span><span>(<br>  seurat_obj,<br>  features=<span>'hMEs'</span>, <br>  order=TRUE <br>)</span><br><span>wrap_plots</span><span>(plot_list)</span><br></span></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.8399581589958159" data-src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26ERMwhwrBBJAvJuCMjpbgpicDfkRzmeD4rjicw7L0v4ichwS3bEFE2GRu8A/640?wx_fmt=png" data-type="png" data-w="956" src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26ERMwhwrBBJAvJuCMjpbgpicDfkRzmeD4rjicw7L0v4ichwS3bEFE2GRu8A/640?wx_fmt=png"></figure><figure data-tool="mdnice编辑器"><img data-ratio="0.6324074074074074" data-src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26EPPCFbJwoBI1gpaD7SJQ2xkmwqRpzA7nYX0MDic4rib69PZVLBKRsDckg/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26EPPCFbJwoBI1gpaD7SJQ2xkmwqRpzA7nYX0MDic4rib69PZVLBKRsDckg/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code># 模块与类群的联系<br>MEs &lt;- GetMEs(seurat_obj, harmonized=TRUE)<br>mods &lt;- colnames(MEs); mods &lt;- mods[mods != <span>'grey'</span>]<br>seurat_obj<span>@meta</span>.data &lt;- cbind(seurat_obj<span>@meta</span>.data, MEs)<br>p &lt;- DotPlot(seurat_obj, features=mods)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.6472222222222223" data-src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26EQUYBfQSKbH1ibM9ytxGDLyyNjPiatvLgct2ZLsR2o3hYR3X0dDImF8TQ/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676hkdx5G5kGQpLFAsUp26EQUYBfQSKbH1ibM9ytxGDLyyNjPiatvLgct2ZLsR2o3hYR3X0dDImF8TQ/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">总而言之，整理bulk水平和单细胞的WGCNA分析会发现，因单细胞转录组的稀疏性问题，引入构建metacells，其次两者共性都在于共表达网络的构建和模块的选取，MEs的定义是给定模型的第一主成分，代表整个模型的基因表达谱。相较于差异基因而言，WGCNA分析考虑的因素更多，与表型的联系更符合生物学意义！！！</p><h2><span>彩蛋分享</span></h2><p>外接Chatgpt和Bing的小插件，前提是要有账号哦！<br></p><p><img data-galleryid="" data-ratio="0.40925925925925927" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc67vFiciaePTPIYAN1zacmIBiaUNRaGkcSsgsFyv5vxgJsibwbN2ykKmEE5lccrtM6CxFRAmUYd93wEJcw/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc67vFiciaePTPIYAN1zacmIBiaUNRaGkcSsgsFyv5vxgJsibwbN2ykKmEE5lccrtM6CxFRAmUYd93wEJcw/640?wx_fmt=png"></p><p>https://platform.openai.com/account/api-keys</p><p><img data-galleryid="" data-ratio="0.4722222222222222" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc67vFiciaePTPIYAN1zacmIBiaUC5NCvhZk3gejCfazWjLiaIRhFJxicxZj0xv5mnicVUXL2xQLtbDKibQp2Q/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc67vFiciaePTPIYAN1zacmIBiaUC5NCvhZk3gejCfazWjLiaIRhFJxicxZj0xv5mnicVUXL2xQLtbDKibQp2Q/640?wx_fmt=png"></p><p><img data-galleryid="" data-ratio="0.562037037037037" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc67vFiciaePTPIYAN1zacmIBiaUytQFPkcrlN1TFGAe7bS2wBJT4RHsfuJGhldNkg4SIdMhjfRKBAqdOw/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc67vFiciaePTPIYAN1zacmIBiaUytQFPkcrlN1TFGAe7bS2wBJT4RHsfuJGhldNkg4SIdMhjfRKBAqdOw/640?wx_fmt=png"></p><p><img data-galleryid="" data-ratio="0.5722222222222222" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc67vFiciaePTPIYAN1zacmIBiaUxenZX2cIqJeZSicCOSEevf5ceVZf0fOCqTseJsVPKVuib6mzWR4FfrJw/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc67vFiciaePTPIYAN1zacmIBiaUxenZX2cIqJeZSicCOSEevf5ceVZf0fOCqTseJsVPKVuib6mzWR4FfrJw/640?wx_fmt=png"></p><p><img data-galleryid="" data-ratio="0.5666666666666667" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc67vFiciaePTPIYAN1zacmIBiaU4a7WfSrQwB6MZOlrVq44hUnq08B6gujpFtD2icib5XGo5peGuVytP6kw/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc67vFiciaePTPIYAN1zacmIBiaU4a7WfSrQwB6MZOlrVq44hUnq08B6gujpFtD2icib5XGo5peGuVytP6kw/640?wx_fmt=png"></p><p><br></p><p data-tool="mdnice编辑器"><br></p></section><p><br></p><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/J4PKvxmNNrZ5m_rGaEl1jg",target="_blank" rel="noopener noreferrer">原文链接</a>
