---
title: "学习大佬看问题的角度"
date: 2023-04-10T04:36:32Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Acdemic"]
---
学习大佬看问题的角度 by 生信菜鸟团
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><p data-tool="mdnice编辑器">近期在Twitter 上刷到一篇文献，主要内容描述的是关于Seurat 函数管道各个过程中，参数的修改对结果的影响。</p><blockquote data-tool="mdnice编辑器"><p>Schneider I, Cepela J, Shetty M, Wang J, Nelson AC, Winterhoff B, Starr TK. Use of “default” parameter settings when analyzing single cell RNA sequencing data using Seurat: a biologist’s perspective. <em>J Transl Genet Genom</em> 2021;5:37-49. http://dx.doi.org/10.20517/jtgg.2020.48</p></blockquote><p data-tool="mdnice编辑器"><img data-ratio="0.4444444444444444" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/iaRJcrq2Losic9O1kpZt7nicvyPwhfD9AfNUkYicuqWaaWJib0TlUKr95wGaUfBE3ibTtPVK176DibqbA0paAiawJcr9yQ/640?wx_fmt=jpeg" data-type="jpeg" data-w="1080" title="Seurat 流程" src="https://mmbiz.qpic.cn/mmbiz_jpg/iaRJcrq2Losic9O1kpZt7nicvyPwhfD9AfNUkYicuqWaaWJib0TlUKr95wGaUfBE3ibTtPVK176DibqbA0paAiawJcr9yQ/640?wx_fmt=jpeg"><span>Seurat 流程</span></p><p data-tool="mdnice编辑器">在我们使用过程中， Seurat 使用的基本聚类算法是基于共享最近邻(SNN)图的聚类方法。在 Seurat 管道内执行无偏细胞聚类的管道是:</p><p data-tool="mdnice编辑器">(1)基于genes/cell,cells/gene的minimum/maximum  截断值和可选参数(如线粒体基因 UMI 计数占总数的百分比)过滤数据集;</p><p data-tool="mdnice编辑器">(2)normalize 数据；</p><p data-tool="mdnice编辑器">(3)找到可变基因;</p><p data-tool="mdnice编辑器">(4)Scale 数据;</p><p data-tool="mdnice编辑器">(5)识别主要成分;</p><p data-tool="mdnice编辑器">(6)识别细胞“邻居”和细胞簇。</p><p data-tool="mdnice编辑器">每个步骤都需要多个用户定义的输入参数。但分析从新鲜样本分离的单个细胞，使用每个步骤的一系列输入参数时，聚类解决方案是如何相似/不同的？</p><p data-tool="mdnice编辑器">这也就使得，探讨Seurat 各个步骤对结果有什么影响这个问题有了意义。因此作者团队基于其课题组的样本，进行cellranger → seurat 流程，并在各个流程中使用不同的参数，随后去比较随参数改变导致结果出现了什么偏倚。<span>这里不过多介绍。</span><span>简要提及，</span><strong>Fig5形象的展示了不同scale factors 下载umap图上各种 聚类的迁移的情况。</strong></p><p data-tool="mdnice编辑器"><img data-ratio="0.7546296296296297" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic9O1kpZt7nicvyPwhfD9AfN0ZsxgibuUBQsASaSFILvZlKs2TqNqrGpibSWiba3SV82picv725fTDhTUw/640?wx_fmt=png" data-type="png" data-w="1080" title="Figure 5. UMAP plots of clustering solutions using scale factors of: 10,000 (baseline) (A, D); 1000 (B, E); and 100,000 (C, F). The cells are colored by: cluster (A-C); and cell type (D-F). Red circles and arrows indicate cluster gained when scale factor was reduced to 1000 (B, E) and cluster lost when increasing scale factor to 100,000 (C, E). Boxes below (D-F) show enlarged versions of the clusters within the red circles. UMAP: uniform manifold approximation and projectio" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic9O1kpZt7nicvyPwhfD9AfN0ZsxgibuUBQsASaSFILvZlKs2TqNqrGpibSWiba3SV82picv725fTDhTUw/640?wx_fmt=png"><span>Figure 5. UMAP plots of clustering solutions using scale factors of: 10,000 (baseline) (A, D); 1000 (B, E); and 100,000 (C, F). The cells are colored by: cluster (A-C); and cell type (D-F). Red circles and arr</span><span>ows indicate cluster gained when scale factor was reduced to 1000 (B, E) and cluster lost when increasing scale factor to 100,000 (C, E). Boxes below (D-F) show enlarged versions of the clusters within the red circles. UMAP: uniform manifold approximation and projectio</span></p><p data-tool="mdnice编辑器">文章末尾作者讨论部分强调：</p><blockquote data-tool="mdnice编辑器"><p>The Seurat R package, at its core, uses an SNN graph-based algorithm to identify cell clusters based on UMI count data. <span><strong>The most important parameters affecting the clustering solutions include the number of dimensional reductions to use, the k-parameter, a prune parameter, and a resolution parameter</strong><strong>.</strong> </span>Altering these values generated clustering solutions ranging from 6 to 27 clusters [Supplementary Figure 2A].</p><p>As there is no biological rationale for choosing the exact values of these parameters, we recommend running an iterative analysis over a range of values and select the clustering solution that is produced at the highest frequency [Supplementary Figure 2A].</p><p>Altering other parameters will also have subtle effects on the clustering solutions produced. We found that the normalization method [Figure 3A-C] has a stronger effect than changing the scale factor during normalization [Figure 3D-F], with both of them changing cluster solutions [Figures 4 and 5]. Changing the method used to select variable genes had a significant effect on which genes were chosen [Figure 6A] and resulted in somewhat dissimilar cluster placement [Figure 6D]. <span>Surprisingly, however, the cell type assignments were highly similar [Figures 6C and 7]. </span></p><p><span>Increasing the number of variable genes to use, from ~1000 to ~3000, did not affect the clustering solution as much as lowering the number of variable genes from ~1000 to ~125 [Figure 8]. The parameters for loess span, bin widths, or binning method used when finding variable genes had negligible effects on the clustering solutions [Supplementary Figure 3].</span></p><p>When scaling the data, in our dataset, regressing out UMI count had a strong effect, while regressing out mitochondrial gene percentages had a much smaller effect [Figure 9]. Changing the scale maximum when scaling data had essentially no effect on the clustering solutions [Supplementary Figure 4].</p></blockquote><p data-tool="mdnice编辑器">简单来说：<strong>the number of dimensional reductions to use, the k-parameter, a prune parameter, and a resolution parameter</strong> 这些参数对Seurat 的结果很重要，需要多加尝试。</p><p data-tool="mdnice编辑器">原文及作者介绍视频已传百度网盘，需要自取。链接：https://pan.baidu.com/s/1rckC7RnCKlZtmkbo94l_tA?pwd=bwba提取码：bwba</p><p data-tool="mdnice编辑器">看完这篇文献，我只想说，大佬看待问题的角度都是不一样的，向大佬学习！</p><span>- END -</span></section><p><br></p><p><mp-style-type data-value="10000"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/n-9l4csjS48ydlikiSFwGg",target="_blank" rel="noopener noreferrer">原文链接</a>
