---
title: "单细胞测序 空间转录组联合分析-使用cell2location获取细胞丰度"
date: 2023-11-25T09:58:45Z
draft: ["false"]
tags: [
  "fetched",
  "生信小博士"
]
categories: ["Acdemic"]
---
单细胞测序 空间转录组联合分析-使用cell2location获取细胞丰度 by 生信小博士
------
<div><p><span>大家好，欢迎来的</span><span>单细胞空间转录组联合分析.续上上期：</span><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=Mzg2NDcxMzYwNg==&amp;mid=2247484289&amp;idx=1&amp;sn=4a31cb4472cbbeebc2802f0ba8c7f568&amp;chksm=ce646d68f913e47ec80139ad1f10867059e21691cf1f5c8fc32bb25a6de91d154e0b90ea1073&amp;scene=21#wechat_redirect" textvalue="Spatial multi-omic map of human myocardial infarction" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2" hasload="1"><span>Spatial multi-omic map of human </span><span>myocardial infarction</span></a><span>.</span><span></span></p><p><span>使用 cell2location软件获取细胞丰度分为三步，前两步使用带有gpu的高性能计算机来运行。如果你使用cell2location，必须使用gpu哦，如果只有cpu，速度慢的难以想象。</span></p><ol><li><p><span>使用单细胞的数据，获取infer_ref</span></p></li><li><p><span>使用infer_ref去跑c2l，获得sp.h5ad文件</span></p></li><li><p><span>从sp.h5ad文件提取细胞丰度</span></p><hr><p><span>我是按照nature的这篇文章来写的，所以使用cell2location。如果我们的目的只是为了获取细胞丰度，可以使用其他方法，比如贝叶斯。最后都会获得每个spot内不同细胞类型的含量。 </span></p><p></p></li></ol><p><br></p><p><span>详细代码如下<br></span></p><hr><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="python"><code><span>step1 get_infer</span></code><code><span><span>#例：GPU作业脚本 #############################测试脚本-##################</span></span></code><code><span>cat &gt;test_gpuet_ref_model_gpu_silicosis_mapped.py.lsf</span></code><code><span><br></span></code><code><span><span>#BSUB -J case_gpu</span></span></code><code><span><span>#BSUB -q normal_test</span></span></code><code><span><span>#BSUB -n 24</span></span></code><code><span><span>#BSUB -o %J.out</span></span></code><code><span><span>#BSUB -e %J.err</span></span></code><code><span><span>#BSUB -gpu "num=1:mode=exclusive_process:aff=yes"</span></span></code><code><span><span>#BSUB -m "gpu18"</span></span></code><code><span><br></span></code><code><span>cd /seu_share/home/chaojie/<span>230218989</span>/silicosis/spatial</span></code><code><span><span>#conda activate r4.2.2</span></span></code><code><span>module load anaconda3</span></code><code><span><br></span></code><code><span>conda activate cell2loc_env_in_seu_module</span></code><code><span><br></span></code><code><span>python cell2location_get_ref_model_gpu_silicosis_mapped.py</span></code><code><span><br></span></code><code><span><span>#提交作业  ###运行作业之前需要进入conda r4.2.2环境</span></span></code><code><span>bsub &lt;test_gpuet_ref_model_gpu_silicosis_mapped.py.lsf</span></code><code><span><br></span></code><code><span><span>#######################################正式提交作业###############</span></span></code><code><span><br></span></code><code><span>例：提交一个GPU作业</span></code><code><span><br></span></code><code><span>cat &gt;c2l_gpu_cell2location_get_ref_model_gpu_silicosis_mapped_kaiti.py.lsf</span></code><code><span><br></span></code><code><span><span>#BSUB -J cell2location_ref_gpu</span></span></code><code><span><span>#BSUB -q gpu_v100</span></span></code><code><span><span>#BSUB -o %J.out</span></span></code><code><span><span>#BSUB -e %J.err</span></span></code><code><span><span>#BSUB -gpu "num=1:mode=exclusive_process:aff=yes"</span></span></code><code><span><span>#BSUB -m "gpu18"</span></span></code><code><span><br></span></code><code><span>cd /seu_share/home/chaojie/<span>230218989</span>/silicosis/spatial</span></code><code><span><br></span></code><code><span>module load anaconda3</span></code><code><span><br></span></code><code><span>conda activate   cell2loc_env_in_seu_module</span></code><code><span><br></span></code><code><span>python cell2location_get_ref_model_gpu_silicosis_mapped_kaiti.py</span></code><code><span><br></span></code><code><span><span>#提交作业  ###运行作业之前需要进入conda r4.2.2环境</span></span></code><code><span>bsub &lt; c2l_gpu_cell2location_get_ref_model_gpu_silicosis_mapped_kaiti.py.lsf</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span>#######################-------------------------------------------</span></span></code><code><span><br></span></code><code><span>cat &gt; cell2location_get_ref_model_gpu_silicosis_mapped_kaiti.py</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><span>import</span> os</span></code><code><span><span>import</span> scanpy <span>as</span> sc</span></code><code><span><span>import</span> numpy <span>as</span> np</span></code><code><span><span>import</span> matplotlib.pyplot <span>as</span> plt</span></code><code><span><span>import</span> matplotlib <span>as</span> mpl</span></code><code><span><span>import</span> cell2location</span></code><code><span><span>from</span> matplotlib <span>import</span> rcParams</span></code><code><span>rcParams[<span>'pdf.fonttype'</span>] = <span>42</span> <span># enables correct plotting of text for PDFs</span></span></code><code><span><br></span></code><code><span><br></span></code><code><span>results_folder = <span>'./silicosis_mapped_after_kaiti/sc_silicosis_analysis'</span></span></code><code><span><span># create paths and names to results folders for reference regression and cell2location models</span></span></code><code><span>ref_run_name = <span>f'<span>{results_folder}</span>/reference_signatures'</span></span></code><code><span>run_name = <span>f'<span>{results_folder}</span>/cell2location_map'</span></span></code><code><span><br></span></code><code><span>os.makedirs(ref_run_name, exist_ok=<span>True</span>)</span></code><code><span>os.makedirs(run_name, exist_ok=<span>True</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># 在R中把数据导出成scanpy可以读取的格式 geneinfo  cellinfo counts</span></span></code><code><span><span># # geneinfo=data.frame(gene=rownames(All.merge),row.names = rownames(All.merge))</span></span></code><code><span><span># geneinfo</span></span></code><code><span><span># getwd()</span></span></code><code><span><span># dir.create("/home/data/t040413/silicosis/data/tabula_scRNAseq/integration_with_sc_silicosis/silicosis_fibro_AM3_mappedbacked_nohhip_univer_frc")</span></span></code><code><span><span># write.csv(geneinfo,file =  "/home/data/t040413/silicosis/data/tabula_scRNAseq/integration_with_sc_silicosis/silicosis_fibro_AM3_mappedbacked_nohhip_univer_frc/geneinfo.csv")</span></span></code><code><span><span># </span></span></code><code><span><span># head(All.merge@meta.data)</span></span></code><code><span><span># </span></span></code><code><span><span># cell_info=data.frame("fibro_mapped_backt0_silicosis"=Idents(All.merge))</span></span></code><code><span><span># head(cell_info)</span></span></code><code><span><span># write.csv(cell_info,file = "/home/data/t040413/silicosis/data/tabula_scRNAseq/integration_with_sc_silicosis/silicosis_fibro_AM3_mappedbacked_nohhip_univer_frc/cell_info.csv",col.names = TRUE)</span></span></code><code><span><span># </span></span></code><code><span><span># # Convert the counts data to a sparse matrix</span></span></code><code><span><span># counts_sparse &lt;- Matrix::Matrix(as.matrix(GetAssayData(All.merge,slot = "counts")), sparse = TRUE)</span></span></code><code><span><span># # Save the sparse matrix in Matrix Market format (MM)</span></span></code><code><span><span># Matrix::writeMM(counts_sparse, file = "/home/data/t040413/silicosis/data/tabula_scRNAseq/integration_with_sc_silicosis/silicosis_fibro_AM3_mappedbacked_nohhip_univer_frc/counts_sparse.mtx")</span></span></code><code><span><span># </span></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><span>import</span> pandas <span>as</span> pd</span></code><code><span>cellinfo = pd.read_csv(<span>"/seu_share/home/chaojie/230218989/silicosis/scRNA-seq/for_c2l_after_mapped_back_kaiti/cell_info.csv"</span>,index_col=<span>0</span>)</span></code><code><span>geneinfo = pd.read_csv(<span>"/seu_share/home/chaojie/230218989/silicosis/scRNA-seq/for_c2l_after_mapped_back_kaiti/geneinfo.csv"</span>,index_col=<span>0</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span>for</span> col <span>in</span> cellinfo.columns:</span></code><code><span>    cellinfo[col] = cellinfo[col].astype(<span>'category'</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span>adata_ref=sc.read(<span>"/seu_share/home/chaojie/230218989/silicosis/scRNA-seq/for_c2l_after_mapped_back_kaiti/counts_sparse.mtx"</span>,index_col=<span>0</span>,header=<span>None</span>)</span></code><code><span>adata_ref=adata_ref.T  <span>########非常重要</span></span></code><code><span>adata_ref = sc.AnnData(adata_ref.X,obs=cellinfo,var = geneinfo)</span></code><code><span><br></span></code><code><span><span>#adata_ref = adata_ref[~(adata_ref.obs['fibro_mapped_backt0_silicosis'] == "Inflammatory fibroblast")]</span></span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># adata=sc.read_h5ad("/seu_share/home/chaojie/230218989/silicosis/scRNA-seq/silicosis.h5ad")</span></span></code><code><span><span># </span></span></code><code><span><span># adata.obs['fine_cell.type']=adata.obs['new.cluster.idents'].astype("category")</span></span></code><code><span><span># adata.obs['cell.type']=adata.obs['new.cluster.idents'].astype("category")</span></span></code><code><span><span># adata.obs['stim']=adata.obs['stim'].astype("category")</span></span></code><code><span><span># </span></span></code><code><span><span># </span></span></code><code><span><span># adata.obsm['original_harmony_X_umap_fromseurat']=adata.obsm['X_umap']</span></span></code><code><span><span># </span></span></code><code><span><span># sc.pl.umap(adata, color="new.cluster.idents")</span></span></code><code><span><span># sc.pl.umap(adata, color="fine_cell.type")</span></span></code><code><span><span># # 保存热图为图像文件</span></span></code><code><span><span># plt.savefig(f"{ref_run_name}/umap_fine_cell.type.png", bbox_inches='tight')</span></span></code><code><span><span># adata.obs['Sample'] = np.where(adata.obs['stim'] == 1, 'libary1',</span></span></code><code><span><span>#                                np.where(adata.obs['stim'] == 2 , 'libary2',</span></span></code><code><span><span>#                                         np.where(adata.obs['stim'] == 3 , 'libary3' , 'libary4')</span></span></code><code><span><span>#                                )</span></span></code><code><span><span># )</span></span></code><code><span><span># </span></span></code><code><span><span># adata.obs['stim'] = np.where(adata.obs['stim'] == 1, 'libary1',</span></span></code><code><span><span>#                              np.where(adata.obs['stim'] == 2 , 'libary2',</span></span></code><code><span><span>#                                       np.where(adata.obs['stim'] == 3 , 'libary3' , 'libary4')</span></span></code><code><span><span>#                              )</span></span></code><code><span><span># )</span></span></code><code><span><span># </span></span></code><code><span><span># </span></span></code><code><span><span># adata.obs = adata.obs[['stim', 'new.cluster.idents']]</span></span></code><code><span><span># adata.obs['fine_cell.type']=adata.obs['new.cluster.idents'].astype("category")</span></span></code><code><span><span># adata.obs['cell.type']=adata.obs['new.cluster.idents'].astype("category")</span></span></code><code><span><span># adata.obs['stim']=adata.obs['stim'].astype("category")</span></span></code><code><span><span># </span></span></code><code><span><span>#adata_ref=adata</span></span></code><code><span><span>#</span></span></code><code><span>adata_ref.var[<span>'SYMBOL'</span>] = adata_ref.var.index</span></code><code><span><br></span></code><code><span><span># find mitochondria-encoded (MT) genes</span></span></code><code><span>adata_ref.var[<span>'MT_gene'</span>] = [gene.startswith(<span>'MT-'</span>) <span>for</span> gene <span>in</span> adata_ref.var[<span>'SYMBOL'</span>]]</span></code><code><span>adata_ref.var[<span>'mt_gene'</span>] = [gene.startswith(<span>'mt-'</span>) <span>for</span> gene <span>in</span> adata_ref.var[<span>'SYMBOL'</span>]]</span></code><code><span><br></span></code><code><span>adata_ref.var.groupby(<span>'MT_gene'</span>).count()</span></code><code><span>adata_ref.var.groupby(<span>'mt_gene'</span>).count()</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># remove MT genes for spatial mapping (keeping their counts in the object)</span></span></code><code><span>adata_ref.obsm[<span>'mt'</span>] = adata_ref[:, adata_ref.var[<span>'mt_gene'</span>].values].X.toarray()</span></code><code><span>adata_ref = adata_ref[:, ~adata_ref.var[<span>'mt_gene'</span>].values]</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><span>from</span> cell2location.utils.filtering <span>import</span> filter_genes</span></code><code><span>selected = filter_genes(adata_ref, cell_count_cutoff=<span>5</span>, </span></code><code><span>  cell_percentage_cutoff2=<span>0.03</span>, </span></code><code><span>  nonz_mean_cutoff=<span>1.15</span>)<span>#cell_percentage_cutoff2 and nonz_mean_cutoff can be increased to select between 8k-16k genes.</span></span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># filter the object</span></span></code><code><span>adata_ref = adata_ref[:, selected].copy()</span></code><code><span>adata_ref.obs[<span>'Sample'</span>]=adata_ref.obs[<span>'stim'</span>]</span></code><code><span><br></span></code><code><span><br></span></code><code><span>adata_ref.obs[<span>'Subset'</span>]=adata_ref.obs[<span>'fibro_mapped_backt0_silicosis_after_kaiti'</span>]</span></code><code><span>adata_ref.obs[<span>'cell_type'</span>]=adata_ref.obs[<span>'fibro_mapped_backt0_silicosis_after_kaiti'</span>]</span></code><code><span><br></span></code><code><span><span># filter the object</span></span></code><code><span>adata_ref = adata_ref[:, selected].copy()</span></code><code><span>adata_ref.obs[<span>'Sample'</span>]=adata_ref.obs[<span>'stim'</span>]</span></code><code><span><br></span></code><code><span><span>#adata_ref.obs['Sample'] = np.where(adata_ref.obs['stim'] == 1, 'libary1',</span></span></code><code><span> <span>#                             np.where(adata_ref.obs['stim'] == 2 , 'libary2',</span></span></code><code><span>  <span>#                                      np.where(adata_ref.obs['stim'] == 3 , 'libary3' , 'libary4')</span></span></code><code><span>   <span>#                                     )</span></span></code><code><span>    <span>#                          )</span></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># #https://cell2location.readthedocs.io/en/latest/notebooks/cell2location_estimating_signatures.html</span></span></code><code><span><span># adata_snrna_raw=adata_ref</span></span></code><code><span><span># adata_snrna_raw.raw = adata_snrna_raw</span></span></code><code><span><span># adata_snrna_raw.X = adata_snrna_raw.raw.X.copy()</span></span></code><code><span><span># sc.pp.log1p(adata_snrna_raw)</span></span></code><code><span><span># </span></span></code><code><span><span># sc.pp.scale(adata_snrna_raw, max_value=10)</span></span></code><code><span><span># sc.tl.pca(adata_snrna_raw, svd_solver='arpack', n_comps=80, use_highly_variable=False)</span></span></code><code><span><span># </span></span></code><code><span><span># # Plot total counts over PC to check whether PC is indeed associated with total counts</span></span></code><code><span><span># #sc.pl.pca_variance_ratio(adata_snrna_raw, log=True)</span></span></code><code><span><span># #sc.pl.pca(adata_snrna_raw, color=['total_counts'],</span></span></code><code><span><span># #          components=['0,1', '2,3', '4,5', '6,7', '8,9', '10,11', '12,13'],</span></span></code><code><span><span># #          color_map = 'RdPu', ncols = 3, legend_loc='on data',</span></span></code><code><span><span># #          legend_fontsize=10, gene_symbols='SYMBOL')</span></span></code><code><span><span># </span></span></code><code><span><span># # remove the first PC which explains large amount of variance in total UMI count (likely technical variation)</span></span></code><code><span><span># adata_snrna_raw.obsm['X_pca'] = adata_snrna_raw.obsm['X_pca'][:, 1:]</span></span></code><code><span><span># adata_snrna_raw.varm['PCs'] = adata_snrna_raw.varm['PCs'][:, 1:]</span></span></code><code><span><span># #########################</span></span></code><code><span><span># import scanpy.external as sce</span></span></code><code><span><span># # Here BBKNN (https://github.com/Teichlab/bbknn) is used to align batches (10X experiments)</span></span></code><code><span><span># #import bbknn</span></span></code><code><span><span># #bbknn.bbknn(adata_snrna_raw, neighbors_within_batch = 3, batch_key = 'sample', n_pcs = 79)</span></span></code><code><span><span># #sc.tl.umap(adata_snrna_raw, min_dist = 0.8, spread = 1.5)</span></span></code><code><span><span># #adata_snrna_raw.obs['stim']=adata_snrna_raw.obs['stim'].astype("category")</span></span></code><code><span><span># sce.pp.harmony_integrate(adata_snrna_raw, key="stim")</span></span></code><code><span><span># </span></span></code><code><span><span># sc.pp.neighbors(adata_snrna_raw, n_neighbors=10, n_pcs=30,use_rep="X_pca_harmony")</span></span></code><code><span><span># </span></span></code><code><span><span># sc.tl.umap(adata_snrna_raw, min_dist = 0.8, spread = 1.5)</span></span></code><code><span><span># plt.savefig(f"{ref_run_name}/after_harmony_umap.png", dpi=600)  # 保存为PNG格式，设置dpi为300</span></span></code><code><span><span># </span></span></code><code><span><span># adata_ref=adata_snrna_raw</span></span></code><code><span><span># </span></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># prepare anndata for the regression model</span></span></code><code><span>cell2location.models.RegressionModel.setup_anndata(adata=adata_ref,</span></code><code><span>                        <span># 10X reaction / sample / batch</span></span></code><code><span>                        batch_key=<span>'Sample'</span>,</span></code><code><span>                        <span># cell type, covariate used for constructing signatures</span></span></code><code><span>                        labels_key=<span>'Subset'</span></span></code><code><span>                        <span>#,multiplicative technical effects (platform, 3' vs 5', donor effect)</span></span></code><code><span>                      <span>#  categorical_covariate_keys=['Method']</span></span></code><code><span>                       )</span></code><code><span><span># create the regression model</span></span></code><code><span><span>from</span> cell2location.models <span>import</span> RegressionModel</span></code><code><span>mod = RegressionModel(adata_ref)</span></code><code><span><br></span></code><code><span><span># view anndata_setup as a sanity check</span></span></code><code><span>mod.view_anndata_setup()</span></code><code><span><br></span></code><code><span><br></span></code><code><span>mod.train(max_epochs=<span>500</span>, use_gpu=<span>True</span>) <span>##确定服务器一定要有gpu才可以----------</span></span></code><code><span><br></span></code><code><span><span>#mod.train(max_epochs=250, use_gpu=False)</span></span></code><code><span>print(<span>"====================done=====here1====================="</span>)</span></code><code><span><br></span></code><code><span><span>import</span> matplotlib.pyplot <span>as</span> plt</span></code><code><span><span># 绘制图形</span></span></code><code><span>mod.plot_history(<span>20</span>)</span></code><code><span><br></span></code><code><span><span># 保存图形</span></span></code><code><span>plt.savefig(<span>f"<span>{ref_run_name}</span>/my_training_ephoes_plot.png"</span>)  <span># 保存为PNG格式</span></span></code><code><span>mod.plot_history(<span>1</span>)</span></code><code><span><span># 保存图形并设置dpi参数</span></span></code><code><span>plt.savefig(<span>f"<span>{ref_run_name}</span>/my_plot.png"</span>, dpi=<span>600</span>)  <span># 保存为PNG格式，设置dpi为300</span></span></code><code><span><br></span></code><code><span><br></span></code><code><span><span>#这一段不运行看看是否可以正常 </span></span></code><code><span>adata_ref = mod.export_posterior(</span></code><code><span>    adata_ref, use_quantiles=<span>True</span>,</span></code><code><span>   <span>#  choose quantiles</span></span></code><code><span>   add_to_varm=[<span>"q05"</span>,<span>"q50"</span>, <span>"q95"</span>, <span>"q0001"</span>], <span>#https://github.com/BayraktarLab/cell2location/issues/281</span></span></code><code><span>    sample_kwargs={<span>'batch_size'</span>: <span>2500</span>, <span>'use_gpu'</span>: <span>True</span>}</span></code><code><span>)</span></code><code><span><br></span></code><code><span><span># In this section, we export the estimated cell abundance (summary of the posterior distribution).</span></span></code><code><span>adata_ref = mod.export_posterior(</span></code><code><span>    adata_ref, sample_kwargs={<span>'num_samples'</span>: <span>1000</span>, <span>'batch_size'</span>: <span>2500</span>, <span>'use_gpu'</span>: <span>True</span>}</span></code><code><span>)</span></code><code><span><br></span></code><code><span><span># Save model</span></span></code><code><span>mod.save(<span>f"<span>{ref_run_name}</span>"</span>, overwrite=<span>True</span>)</span></code><code><span><span># Save anndata object with results</span></span></code><code><span>adata_file = <span>f"<span>{ref_run_name}</span>/sc.h5ad"</span></span></code><code><span>adata_ref.write(adata_file)</span></code><code><span>print(<span>"====================done=====here2====================="</span>)</span></code><code><span><br></span></code><code><span><span># export estimated expression in each cluster</span></span></code><code><span><span>if</span> <span>'means_per_cluster_mu_fg'</span> <span>in</span> adata_ref.varm.keys():</span></code><code><span>    inf_aver = adata_ref.varm[<span>'means_per_cluster_mu_fg'</span>][[<span>f'means_per_cluster_mu_fg_<span>{i}</span>'</span></span></code><code><span>                                    <span>for</span> i <span>in</span> adata_ref.uns[<span>'mod'</span>][<span>'factor_names'</span>]]].copy()</span></code><code><span><span>else</span>:</span></code><code><span>    inf_aver = adata_ref.var[[<span>f'means_per_cluster_mu_fg_<span>{i}</span>'</span></span></code><code><span>                                    <span>for</span> i <span>in</span> adata_ref.uns[<span>'mod'</span>][<span>'factor_names'</span>]]].copy()</span></code><code><span>inf_aver.columns = adata_ref.uns[<span>'mod'</span>][<span>'factor_names'</span>]</span></code><code><span>inf_aver.iloc[<span>0</span>:<span>5</span>, <span>0</span>:<span>5</span>]</span></code><code><span><br></span></code><code><span>inf_aver.to_csv(<span>f"<span>{ref_run_name}</span>/inf_aver.csv"</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span>mod.plot_history(<span>1</span>)</span></code><code><span><span># 保存图形并设置dpi参数</span></span></code><code><span>plt.savefig(<span>f"<span>{ref_run_name}</span>/epos_1_my_plot.png"</span>, dpi=<span>600</span>)  <span># 保存为PNG格式，设置dpi为300</span></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><span>##QC</span></span></code><code><span>mod.plot_QC()</span></code><code><span><span>#plt.savefig(f"{ref_run_name}/QC1.png", dpi=600)  # 保存为PNG格式，设置dpi为300</span></span></code><code><span>plt.savefig(<span>f"<span>{ref_run_name}</span>/QC1.png"</span>, dpi=<span>600</span>)</span></code><code><span><br></span></code><code><span>print(<span>"====================done=====here3====================="</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span>obs_dtype = adata_ref.obs.dtypes</span></code><code><span>var_dtype = adata_ref.var.dtypes</span></code><code><span><br></span></code><code><span>print(<span>"Data types of obs:"</span>)</span></code><code><span>print(obs_dtype)</span></code><code><span><br></span></code><code><span>print(<span>"Data types of var:"</span>)</span></code><code><span>print(var_dtype)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span>#adata_ref.var.columns = adata_ref.var.columns.str.lstrip('_')</span></span></code><code><span>adata_ref.var.columns = adata_ref.var.columns.map(<span>lambda</span> x: <span>f"new_name<span>{x}</span>"</span> <span>if</span> x.startswith(<span>'_'</span>) <span>else</span> x)</span></code><code><span>adata_ref.obs.columns = adata_ref.obs.columns.map(<span>lambda</span> x: <span>f"new_name<span>{x}</span>"</span> <span>if</span> x.startswith(<span>'_'</span>) <span>else</span> x)</span></code><code><span><br></span></code><code><span>adata_ref.var.columns.map(<span>lambda</span> x: <span>f"new_name<span>{x}</span>.lstrip('_')"</span> <span>if</span> x.startswith(<span>'_'</span>) <span>else</span> x)</span></code><code><span>adata_ref.obs.columns.map(<span>lambda</span> x: <span>f"new_name<span>{x}</span>.lstrip('_')"</span> <span>if</span> x.startswith(<span>'_'</span>) <span>else</span> x)</span></code><code><span><br></span></code><code><span>adata_ref.var.columns=[<span>f"new_name<span>{x.lstrip(<span>'_'</span>)}</span>"</span> <span>if</span> x.startswith(<span>'_'</span>) <span>else</span> x <span>for</span> x <span>in</span> adata_ref.var.columns]</span></code><code><span><br></span></code><code><span><br></span></code><code><span>step2 c2l</span></code><code><span><span>#例：GPU作业脚本 #############################测试脚本-##################</span></span></code><code><span>cat &gt;test_gpu_cell2location_get_spatial_mapping_model_gpu_alpha_20.lsf</span></code><code><span><br></span></code><code><span><span>#BSUB -J test_cell2location_sp_gpu_20231006</span></span></code><code><span><span>#BSUB -q normal_test</span></span></code><code><span><span>#BSUB -n 24</span></span></code><code><span><span>#BSUB -o %J.out</span></span></code><code><span><span>#BSUB -e %J.err</span></span></code><code><span><span>#BSUB -gpu "num=1:mode=exclusive_process:aff=yes"</span></span></code><code><span><br></span></code><code><span>cd /seu_share/home/chaojie/<span>230218989</span>/silicosis/spatial</span></code><code><span><span>#cd /seu_share/home/chaojie/230218989/silicosis/spatial</span></span></code><code><span><span>#conda activate r4.2.2</span></span></code><code><span>conda activate cell2loc_env</span></code><code><span><br></span></code><code><span>python cell2location_get_spatial_mapping_model_gpu_alpha_20.py</span></code><code><span><br></span></code><code><span><span>#提交作业  ###运行作业之前需要进入conda r4.2.2环境</span></span></code><code><span>bsub &lt;test_gpu_cell2location_get_spatial_mapping_model_gpu_alpha_20.lsf</span></code><code><span><br></span></code><code><span><span>###</span></span></code><code><span><span>#</span></span></code><code><span><span>#</span></span></code><code><span><span>#</span></span></code><code><span><span>#</span></span></code><code><span><span>#</span></span></code><code><span><span>#######################################正式提交作业###############</span></span></code><code><span><br></span></code><code><span>例：提交一个GPU作业</span></code><code><span><br></span></code><code><span>cat &gt;cell_mapping_model_gpu_alpha_20_mappedback_after_kaiti.lsf</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span>#BSUB -J cell2location_sp_mapping_gpu</span></span></code><code><span><span>#BSUB -q gpu_v100</span></span></code><code><span><span>#BSUB -o %J.out</span></span></code><code><span><span>#BSUB -e %J.err</span></span></code><code><span><span>#BSUB -gpu "num=1:mode=exclusive_process:aff=yes"</span></span></code><code><span><span>#BSUB -m "gpu18"</span></span></code><code><span><br></span></code><code><span><br></span></code><code><span>cd /seu_share/home/chaojie/<span>230218989</span>/silicosis/spatial</span></code><code><span><span>#conda activate r4.2.2</span></span></code><code><span><span>#https://www.ibm.com/docs/en/spectrum-lsf/10.1.0?topic=bsub-options</span></span></code><code><span><span>#https://www.ibm.com/docs/zh/spectrum-lsf/10.1.0?topic=bsub-options</span></span></code><code><span>module load anaconda3</span></code><code><span><br></span></code><code><span>conda activate cell2loc_env_in_seu_module</span></code><code><span><br></span></code><code><span><br></span></code><code><span>python cell_mapping_model_gpu_alpha_20_mappedback_after_kaiti.py</span></code><code><span><br></span></code><code><span><span>#提交作业  ###运行作业之前需要进入conda r4.2.2环境</span></span></code><code><span>bsub &lt; cell_mapping_model_gpu_alpha_20_mappedback_after_kaiti.lsf</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><span>#######################################</span></span></code><code><span><span>#</span></span></code><code><span><span>#</span></span></code><code><span><span>#</span></span></code><code><span><span>#######################--################正式脚本-----------------------------------------</span></span></code><code><span>cat &gt;cell_mapping_model_gpu_alpha_20_mappedback_after_kaiti.py</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><span>#conda activate cell2loc_env</span></span></code><code><span><span>#cd /seu_share/home/chaojie/230218989/silicosis/spatial</span></span></code><code><span><br></span></code><code><span><span>#https://cell2location.readthedocs.io/en/latest/notebooks/cell2location_short_demo.html</span></span></code><code><span><span>import</span> sys</span></code><code><span><span>import</span> scanpy <span>as</span> sc</span></code><code><span><span>import</span> anndata</span></code><code><span><span>import</span> pandas <span>as</span> pd</span></code><code><span><span>import</span> numpy <span>as</span> np</span></code><code><span><span>import</span> os</span></code><code><span><span>import</span> gc</span></code><code><span><br></span></code><code><span><span># this line forces theano to use the GPU and should go before importing cell2location</span></span></code><code><span>os.environ[<span>"THEANO_FLAGS"</span>] = <span>'device=cuda0,floatX=float32,force_device=True'</span></span></code><code><span><span># if using the CPU uncomment this:</span></span></code><code><span><span>#os.environ["THEANO_FLAGS"] = 'device=cpu,floatX=float32,openmp=True,force_device=True'</span></span></code><code><span><span>import</span> os</span></code><code><span><span>import</span> cell2location</span></code><code><span><br></span></code><code><span><span>import</span> matplotlib <span>as</span> mpl</span></code><code><span><span>from</span> matplotlib <span>import</span> rcParams</span></code><code><span><span>import</span> matplotlib.pyplot <span>as</span> plt</span></code><code><span><span>import</span> seaborn <span>as</span> sns</span></code><code><span><br></span></code><code><span><span># silence scanpy that prints a lot of warnings</span></span></code><code><span><span>import</span> warnings</span></code><code><span>warnings.filterwarnings(<span>'ignore'</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span>results_folder = <span>'./silicosis_mapped_after_kaiti/sc_silicosis_analysis'</span></span></code><code><span><span># create paths and names to results folders for reference regression and cell2location models</span></span></code><code><span>ref_run_name = <span>f'<span>{results_folder}</span>/reference_signatures'</span></span></code><code><span>run_name = <span>f'<span>{results_folder}</span>/cell2location_map'</span></span></code><code><span><br></span></code><code><span>os.makedirs(ref_run_name, exist_ok=<span>True</span>)</span></code><code><span>os.makedirs(run_name, exist_ok=<span>True</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span>ns7=sc.read_visium(path=<span>"./ns7/"</span>,count_file=<span>'./2.3.h5_files/filtered_feature_bc_matrix.h5'</span>,library_id=<span>"NS_7"</span>,load_images=<span>True</span>,source_image_path=<span>"./ns7/spatial/"</span>)</span></code><code><span><br></span></code><code><span>blank_ns7=pd.read_csv(<span>'./blank/ns7d_A72_.csv'</span>,index_col=<span>0</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># 获取不包含 blank 行的索引</span></span></code><code><span>filtered_index = [barcode <span>for</span> barcode <span>in</span> ns7.obs.index <span>if</span> barcode <span>not</span> <span>in</span> blank_ns7.index]</span></code><code><span>filtered_index</span></code><code><span>type(filtered_index)</span></code><code><span><span># 使用索引来创建新的 ns7 对象</span></span></code><code><span>ns7 = ns7[filtered_index, :]</span></code><code><span><br></span></code><code><span><span># remove MT genes for spatial mapping (keeping their counts in the object)</span></span></code><code><span><span>#adata_vis.obsm['MT'] = adata_vis[:, adata_vis.var['MT_gene'].values].X.toarray()</span></span></code><code><span><span>#adata_vis = adata_vis[:, ~adata_vis.var['MT_gene'].values]</span></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span>ns56=sc.read_visium(path=<span>"./ns56/"</span>,count_file=<span>'./2.3.h5_files/filtered_feature_bc_matrix.h5'</span>,library_id=<span>"NS_56"</span>,load_images=<span>True</span>,source_image_path=<span>"./ns56/spatial/"</span>)</span></code><code><span><br></span></code><code><span>blank_ns56=pd.read_csv(<span>"./blank/ns56d_A73_.csv"</span>,index_col=<span>0</span>)</span></code><code><span>blank_ns56</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># 获取不包含 blank 行的索引</span></span></code><code><span>filtered_index = [barcode <span>for</span> barcode <span>in</span> ns56.obs.index <span>if</span> barcode <span>not</span> <span>in</span> blank_ns56.index]</span></code><code><span>filtered_index</span></code><code><span>type(filtered_index)</span></code><code><span>len(filtered_index)</span></code><code><span><span># 使用索引来创建新的 ns56 对象</span></span></code><code><span>ns56 = ns56[filtered_index, :]</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span>sio56=sc.read_visium(path=<span>"./sio56d/"</span>, </span></code><code><span>                     count_file=<span>'./2.3.h5_files/filtered_feature_bc_matrix.h5'</span>,</span></code><code><span>                     library_id=<span>"SiO2_56"</span>,</span></code><code><span>                     load_images=<span>True</span>,source_image_path=<span>"./sio56d/spatial/"</span>)</span></code><code><span><br></span></code><code><span>blank_sio56=pd.read_csv(<span>"./blank/sio56_B11.csv"</span>,index_col=<span>0</span>)</span></code><code><span>sio56</span></code><code><span><span># 获取不包含 blank 行的索引</span></span></code><code><span>filtered_index = [barcode <span>for</span> barcode <span>in</span> sio56.obs.index <span>if</span> barcode <span>not</span> <span>in</span> blank_sio56.index]</span></code><code><span>filtered_index</span></code><code><span>type(filtered_index)</span></code><code><span>len(filtered_index)</span></code><code><span><span># 使用索引来创建新的 sio56 对象</span></span></code><code><span>sio56 = sio56[filtered_index, :]</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span>sio7=sc.read_visium(path=<span>"./sio7d/"</span>,count_file=<span>'./2.3.h5_files/filtered_feature_bc_matrix.h5'</span>,library_id=<span>"SiO2_7"</span>,load_images=<span>True</span>,source_image_path=<span>"./sio7d/spatial/"</span>)</span></code><code><span>blank_sio7=pd.read_csv(<span>"./blank/sio7_B26.csv"</span>,index_col=<span>0</span>)</span></code><code><span>sio7</span></code><code><span><span># 获取不包含 blank 行的索引</span></span></code><code><span>filtered_index = [barcode <span>for</span> barcode <span>in</span> sio7.obs.index <span>if</span> barcode <span>not</span> <span>in</span> blank_sio7.index]</span></code><code><span>filtered_index</span></code><code><span>type(filtered_index)</span></code><code><span>len(filtered_index)</span></code><code><span><span># 使用索引来创建新的 sio7 对象</span></span></code><code><span>sio7 = sio7[filtered_index, :]</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><span><span>def</span> <span>read_and_qc</span><span>(sample_name, adata)</span>:</span></span></code><code><span>    <span>r""" This function reads the data for one 10X spatial experiment into the anndata object.</span></span></code><code><span>    It also calculates QC metrics. Modify this function if required by your workflow.</span></code><code><span>    这里主要就是改一下adata的样本名字，添加qc信息</span></code><code><span>    :param sample_name: Name of the sample</span></code><code><span>    </span></code><code><span>    """</span></code><code><span><br></span></code><code><span>    adata.obs[<span>'sample'</span>] = sample_name</span></code><code><span>    adata.var[<span>'SYMBOL'</span>] = adata.var_names</span></code><code><span>    adata.var.rename(columns={<span>'gene_ids'</span>: <span>'ENSEMBL'</span>}, inplace=<span>True</span>)</span></code><code><span>    adata.var_names = adata.var[<span>'ENSEMBL'</span>]</span></code><code><span>    adata.var.drop(columns=<span>'ENSEMBL'</span>, inplace=<span>True</span>)</span></code><code><span><br></span></code><code><span>    <span># Calculate QC metrics</span></span></code><code><span>    <span>from</span> scipy.sparse <span>import</span> csr_matrix</span></code><code><span>    adata.X = adata.X.toarray()</span></code><code><span>    sc.pp.calculate_qc_metrics(adata, inplace=<span>True</span>)</span></code><code><span>  </span></code><code><span>    adata.X = csr_matrix(adata.X)</span></code><code><span>    adata.var[<span>'mt'</span>] = [gene.startswith(<span>'mt-'</span>) <span>for</span> gene <span>in</span> adata.var[<span>'SYMBOL'</span>]]</span></code><code><span>    adata.obs[<span>'mt_frac'</span>] = adata[:, adata.var[<span>'mt'</span>].tolist()].X.sum(<span>1</span>).A.squeeze()/adata.obs[<span>'total_counts'</span>]</span></code><code><span><br></span></code><code><span>    </span></code><code><span>    adata.var[<span>'rps'</span>] = [gene.startswith(<span>'rps-'</span>) <span>for</span> gene <span>in</span> adata.var[<span>'SYMBOL'</span>]]</span></code><code><span>    adata = adata[:, ~ (adata.var[<span>'mt'</span>].values | </span></code><code><span>                              adata.var[<span>'rps'</span>].values)]</span></code><code><span>    </span></code><code><span>    <span>#adata = adata[(adata.obs['n_genes_by_counts'].values &gt; 300) &amp; </span></span></code><code><span>    <span>#         (adata.obs['total_counts'].values &gt; 500), :] </span></span></code><code><span>    </span></code><code><span>        </span></code><code><span>        </span></code><code><span>    <span># add sample name to obs names</span></span></code><code><span>    adata.obs[<span>"sample"</span>] = [str(i) <span>for</span> i <span>in</span> adata.obs[<span>'sample'</span>]]</span></code><code><span>    adata.obs_names = adata.obs[<span>"sample"</span>] \</span></code><code><span>                          + <span>'_'</span> + adata.obs_names</span></code><code><span>    adata.obs.index.name = <span>'spot_id'</span></span></code><code><span><br></span></code><code><span>    <span>return</span> adata</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span><span>def</span> <span>select_slide</span><span>(adata, s, s_col=<span>'sample'</span>)</span>:</span></span></code><code><span>    <span>r""" This function selects the data for one slide from the spatial anndata object.</span></span></code><code><span><br></span></code><code><span>    :param adata: Anndata object with multiple spatial experiments</span></code><code><span>    :param s: name of selected experiment</span></code><code><span>    :param s_col: column in adata.obs listing experiment name for each location</span></code><code><span>    """</span></code><code><span><br></span></code><code><span>    slide = adata[adata.obs[s_col].isin([s]), :]</span></code><code><span>    s_keys = list(slide.uns[<span>'spatial'</span>].keys())</span></code><code><span>    s_spatial = np.array(s_keys)[[s <span>in</span> k <span>for</span> k <span>in</span> s_keys]][<span>0</span>]</span></code><code><span><br></span></code><code><span>    slide.uns[<span>'spatial'</span>] = {s_spatial: slide.uns[<span>'spatial'</span>][s_spatial]}</span></code><code><span><br></span></code><code><span>    <span>return</span> slide</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span>ns7=read_and_qc(sample_name=<span>"NS_7"</span>,adata=ns7)</span></code><code><span>ns56=read_and_qc(sample_name=<span>"NS_56"</span>,adata=ns56)</span></code><code><span>sio7=read_and_qc(sample_name=<span>"SiO_7"</span>,adata=sio7)</span></code><code><span>sio56=read_and_qc(sample_name=<span>"SiO_56"</span>,adata=sio56)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><span>import</span> scanpy <span>as</span> sc</span></code><code><span><br></span></code><code><span><span>#Create a list of AnnData objects</span></span></code><code><span>adata_list = [ ns7,sio7, ns56,sio56]</span></code><code><span><br></span></code><code><span><span># Define batch categories for each AnnData object</span></span></code><code><span>batch_categories = [ <span>"NS_7"</span>,<span>"SiO2_7"</span>, <span>"NS_56"</span>,<span>"SiO2_56"</span> ]</span></code><code><span><span># Combine anndata objects together</span></span></code><code><span><span># Combine anndata objects together</span></span></code><code><span>adata = adata_list[<span>0</span>].concatenate(</span></code><code><span>    adata_list[<span>1</span>:],</span></code><code><span>    batch_key=<span>"sample"</span>,</span></code><code><span>    uns_merge=<span>"unique"</span>,</span></code><code><span>    batch_categories=batch_categories,</span></code><code><span>    index_unique=<span>None</span></span></code><code><span>)</span></code><code><span><br></span></code><code><span><span># mitochondria-encoded (MT) genes should be removed for spatial mapping</span></span></code><code><span>adata.obsm[<span>'mt'</span>] = adata[:, adata.var[<span>'mt'</span>].values].X.toarray()</span></code><code><span>adata = adata[:, ~adata.var[<span>'mt'</span>].values]</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><span>#results_folder = './results_allcelltypes_AM3_newidents/sc_silicosis_analysis'</span></span></code><code><span><span># create paths and names to results folders for reference regression and cell2location models</span></span></code><code><span>ref_run_name = <span>f'<span>{results_folder}</span>/reference_signatures'</span></span></code><code><span>run_name = <span>f'<span>{results_folder}</span>/cell2location_map'</span></span></code><code><span><br></span></code><code><span><br></span></code><code><span><span>#pwd https://cell2location.readthedocs.io/en/latest/notebooks/cell2location_tutorial.html</span></span></code><code><span><span># First generate the reference data frame</span></span></code><code><span><span>#inf_aver = pd.read_csv("/seu_share/home/chaojie/230218989/silicosis/spatial/results_allcelltypes_AM3_newidents/sc_silicosis_analysis/reference_signatures/inf_aver.csv",index_col = 0)</span></span></code><code><span><br></span></code><code><span>inf_aver=pd.read_csv(<span>f"<span>{ref_run_name}</span>/inf_aver.csv"</span>,index_col = <span>0</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span>adata.var[<span>'ENSEMBL'</span>]=adata.var.index</span></code><code><span>adata.var[<span>'symbol'</span>]=adata.var.SYMBOL</span></code><code><span>adata.var.set_index(<span>'SYMBOL'</span>, drop=<span>True</span>, inplace=<span>True</span>)</span></code><code><span>adata_vis=adata</span></code><code><span><br></span></code><code><span><span># find shared genes and subset both anndata and reference signatures</span></span></code><code><span><span># </span></span></code><code><span>adata_vis.var_names_make_unique()</span></code><code><span>intersect = np.intersect1d(adata_vis.var_names, inf_aver.index)</span></code><code><span><br></span></code><code><span>adata_vis = adata_vis[:, intersect].copy()</span></code><code><span>inf_aver = inf_aver.loc[intersect, :].copy()</span></code><code><span><br></span></code><code><span><span>#module_score.reindex(adata_vis_plt.obs_names)</span></span></code><code><span><span>#adata_vis.var.reindex(inf_aver.index)</span></span></code><code><span><span>#df[df.index.duplicated()]</span></span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># prepare anndata for cell2location model</span></span></code><code><span>cell2location.models.Cell2location.setup_anndata(adata=adata_vis, batch_key=<span>"sample"</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># create and train the model</span></span></code><code><span>mod = cell2location.models.Cell2location(</span></code><code><span>    adata_vis, cell_state_df=inf_aver,</span></code><code><span>    <span># the expected average cell abundance: tissue-dependent</span></span></code><code><span>    <span># hyper-prior which can be estimated from paired histology:</span></span></code><code><span>    N_cells_per_location=<span>8</span>,</span></code><code><span>    <span># hyperparameter controlling normalisation of</span></span></code><code><span>    <span># within-experiment variation in RNA detection:</span></span></code><code><span>    detection_alpha=<span>20</span></span></code><code><span>)</span></code><code><span>mod.view_anndata_setup()</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span>mod.train(max_epochs=<span>30000</span>,</span></code><code><span>          <span># train using full data (batch_size=None)</span></span></code><code><span>          batch_size=<span>None</span>,</span></code><code><span>          <span># use all data points in training because</span></span></code><code><span>          <span># we need to estimate cell abundance at all locations</span></span></code><code><span>          train_size=<span>1</span>,</span></code><code><span>          use_gpu=<span>True</span>,</span></code><code><span>         )</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># In this section, we export the estimated cell abundance (summary of the posterior distribution).</span></span></code><code><span>adata_vis = mod.export_posterior(</span></code><code><span>    adata_vis, sample_kwargs={<span>'num_samples'</span>: <span>1000</span>, <span>'batch_size'</span>: mod.adata.n_obs, <span>'use_gpu'</span>: <span>True</span>}</span></code><code><span>)</span></code><code><span><br></span></code><code><span><span># Save model</span></span></code><code><span>mod.save(<span>f"<span>{run_name}</span>"</span>, overwrite=<span>True</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># mod = cell2location.models.Cell2location.load(f"{run_name}", adata_vis)</span></span></code><code><span><br></span></code><code><span><span># Save anndata object with results</span></span></code><code><span>adata_file = <span>f"<span>{run_name}</span>/sp.h5ad"</span></span></code><code><span>adata_vis.write(adata_file)</span></code><code><span>adata_file</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># plot ELBO loss history during training, removing first 100 epochs from the plot</span></span></code><code><span>mod.plot_history(<span>1000</span>)</span></code><code><span>plt.legend(labels=[<span>'full data training'</span>])</span></code><code><span><span># 保存图形并设置dpi参数</span></span></code><code><span>plt.savefig(<span>"ELBO loss history_my_plot_alpha_20.png"</span>, dpi=<span>600</span>)  <span># 保存为PNG格式，设置dpi为300</span></span></code><code><span><br></span></code><code><span><br></span></code><code><span>fig = mod.plot_spatial_QC_across_batches()</span></code><code><span>plt.savefig(<span>f"<span>{run_name}</span>/qc__spatial_QC_across_batches_alpha_20.png"</span>, dpi=<span>600</span>)  <span># 保存为PNG格式，设置dpi为300</span></span></code><code><span><br></span></code><code><span><span># 保存图形为PNG格式，设置dpi（每英寸点数）为600</span></span></code><code><span>fig.savefig(<span>f"<span>{run_name}</span>/qc__spatial_QC_across_batches_alpha_20__.png"</span>, dpi=<span>600</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># add 5% quantile, representing confident cell abundance, 'at least this amount is present',</span></span></code><code><span><span># to adata.obs with nice names for plotting</span></span></code><code><span>adata_vis.obs[adata_vis.uns[<span>'mod'</span>][<span>'factor_names'</span>]] = adata_vis.obsm[<span>'q05_cell_abundance_w_sf'</span>]</span></code><code><span><br></span></code><code><span><span># select one slide</span></span></code><code><span><span>from</span> cell2location.utils <span>import</span> select_slide</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span>sample_names=adata_vis.obs[<span>'sample'</span>].unique()</span></code><code><span>sample_names</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span>for</span> slide_name <span>in</span> sample_names:</span></code><code><span>  <span>#slide=adata_vis[adata_vis.obs['sample']==slide_name,]</span></span></code><code><span>  <span>#slide.uns['spatial']={slide_name:slide.uns['spatial'][slide_name]} #提取所需样本的键值对 放入相应的slide</span></span></code><code><span><br></span></code><code><span><br></span></code><code><span>  slide = select_slide(adata_vis, slide_name)</span></code><code><span><br></span></code><code><span>  <span># plot in spatial coordinates</span></span></code><code><span>  <span>with</span> mpl.rc_context({<span>'axes.facecolor'</span>:  <span>'black'</span>,</span></code><code><span>                   <span>'figure.figsize'</span>: [<span>4.5</span>, <span>5</span>]}):</span></code><code><span><br></span></code><code><span>    sc.pl.spatial(slide, cmap=<span>'magma'</span>,</span></code><code><span>      <span># show first 8 cell types</span></span></code><code><span>      color=adata_vis.uns[<span>'mod'</span>][<span>'factor_names'</span>],</span></code><code><span>      ncols=<span>4</span>, size=<span>1.3</span>,</span></code><code><span>      img_key=<span>'hires'</span>,</span></code><code><span>      <span># limit color scale at 99.2% quantile of cell abundance</span></span></code><code><span>      vmin=<span>0</span>, vmax=<span>'p99.2'</span>,title=slide_name</span></code><code><span>      )</span></code><code><span><br></span></code><code><span>  plt.savefig(<span>f"<span>{run_name}</span>/all_celltypes_alpha200_<span>{slide_name}</span>.png"</span>, dpi=<span>600</span>)  </span></code><code><span><br></span></code><code><span>adata_vis.uns[<span>'mod'</span>][<span>'factor_names'</span>]</span></code><code><span><br></span></code><code><span><br></span></code><code><span>mod.plot_QC()</span></code><code><span>plt.savefig(<span>f"<span>{run_name}</span>/qc_alpha_20.png"</span>, dpi=<span>600</span>)  <span># 保存为PNG格式，设置dpi为300</span></span></code><code><span><br></span></code><code><span>step3 get cell_abundance</span></code><code><span><br></span></code><code><span><span>#https://cell2location.readthedocs.io/en/latest/notebooks/cell2location_short_demo.html</span></span></code><code><span><span>import</span> sys</span></code><code><span><span>import</span> scanpy <span>as</span> sc</span></code><code><span><span>import</span> anndata</span></code><code><span><span>import</span> pandas <span>as</span> pd</span></code><code><span><span>import</span> numpy <span>as</span> np</span></code><code><span><span>import</span> os</span></code><code><span><span>import</span> gc</span></code><code><span><br></span></code><code><span><span># this line forces theano to use the GPU and should go before importing cell2location</span></span></code><code><span>os.environ[<span>"THEANO_FLAGS"</span>] = <span>'device=cuda0,floatX=float32,force_device=True'</span></span></code><code><span><span># if using the CPU uncomment this:</span></span></code><code><span><span>#os.environ["THEANO_FLAGS"] = 'device=cpu,floatX=float32,openmp=True,force_device=True'</span></span></code><code><span><span>import</span> os</span></code><code><span><span>import</span> cell2location</span></code><code><span><br></span></code><code><span><span>import</span> matplotlib <span>as</span> mpl</span></code><code><span><span>from</span> matplotlib <span>import</span> rcParams</span></code><code><span><span>import</span> matplotlib.pyplot <span>as</span> plt</span></code><code><span><span>import</span> seaborn <span>as</span> sns</span></code><code><span><br></span></code><code><span><span># silence scanpy that prints a lot of warnings</span></span></code><code><span><span>import</span> warnings</span></code><code><span>warnings.filterwarnings(<span>'ignore'</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span>adata_vis=sc.read_h5ad(<span>"./sp.h5ad"</span>)</span></code><code><span><br></span></code><code><span>adata_vis.obsm[<span>'q05_cell_abundance_w_sf'</span>]</span></code><code><span><br></span></code><code><span><br></span></code><code><span>adata_vis.obs_names=adata_vis.obs_names.str.replace(<span>"SiO_56"</span>,<span>"SiO2_56"</span>)</span></code><code><span>adata_vis.obs_names=adata_vis.obs_names.str.replace(<span>"SiO_7"</span>,<span>"SiO2_7"</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span>import</span> pandas <span>as</span> pd</span></code><code><span><span># 假设您的数据存储在名为 `adata_vis` 的 DataFrame 中</span></span></code><code><span>data = adata_vis.obsm[<span>'q05_cell_abundance_w_sf'</span>]</span></code><code><span>data.columns=data.columns.str.replace(<span>"q05cell_abundance_w_sf_"</span>,<span>""</span>)</span></code><code><span><br></span></code><code><span><span># 将数据保存为CSV文件</span></span></code><code><span>data.to_csv(<span>f"./q05_cell_abundance_w_sf.csv"</span>, index=<span>True</span>)  <span># 设置index=True以保存索引</span></span></code><code><span><br></span></code><code><span>data.columns.str.replace(<span>"q05_cell_abundance_w_sf"</span>,<span>""</span>)</span></code><code><span><br></span></code><code><span>data</span></code></pre></section><p>最终的细胞丰度文件：</p><p><img data-galleryid="" data-imgfileid="100000758" data-ratio="0.42027114267269206" data-s="300,640" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SktIpVSo4qLeticdwcv5enJS691l08hXye8LV1MpI4KVnG1MViaDIglQPNbyoj7REcytArG8PgibgsYwA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1549" src="https://mmbiz.qpic.cn/sz_mmbiz_png/xVhD7345SktIpVSo4qLeticdwcv5enJS691l08hXye8LV1MpI4KVnG1MViaDIglQPNbyoj7REcytArG8PgibgsYwA/640?wx_fmt=png&amp;from=appmsg"></p><p><br></p><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/5MekQ0dXxPTI_lqgVBin4g",target="_blank" rel="noopener noreferrer">原文链接</a>
