---
title: "单细胞转录组高级分析二：转录调控网络分析"
date: 2023-11-17T16:44:30Z
draft: ["false"]
tags: [
  "fetched",
  "生信会客厅"
]
categories: ["Acdemic"]
---
单细胞转录组高级分析二：转录调控网络分析 by 生信会客厅
------
<div><section><br></section><section><em><span>上期专题我们介绍了单细胞转录组数据的基础分析，然而那些分析只是揭开了组织异质性的面纱，还有更多的生命奥秘隐藏在数据中等待我们发掘。本专题将介绍一些单细胞转录组的高级分析内容：多样本批次校正、转录因子分析、细胞通讯分析、基因集变异分析<em><span>和更全面的基因集富集分析</span></em>。</span></em><em><span>不足之处请大家批评指正，欢迎添加Kinesin微信交流探讨！（<em>扫描文末二维码）</em></span></em></section><section><br></section><h2><strong><span>单细胞转录组基础分析专题</span></strong></h2><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzIyMzMwNDQ2MA==&amp;mid=2247483780&amp;idx=1&amp;sn=37775b519d608bc5663389c6cbabac59&amp;chksm=e8210fe7df5686f159342978a9727d6a8165d0af8a8cb77a99bdbd6f11a420db16f1f4556b63&amp;scene=21#wechat_redirect" data-itemshowtype="0" tab="innerlink" data-linktype="2">单细胞转录组基础分析一：分析环境搭建</a><br></p><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzIyMzMwNDQ2MA==&amp;mid=2247483781&amp;idx=1&amp;sn=3024055fe2b120ac17c5bb39a0750a61&amp;chksm=e8210fe6df5686f0de4978f173a87c96b8e8cc0b87a1375f5470307ccb97cf93e52e2d077d95&amp;scene=21#wechat_redirect" data-itemshowtype="0" tab="innerlink" data-linktype="2">单细胞转录组基础分析二：数据质控与标准化</a><br></p><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzIyMzMwNDQ2MA==&amp;mid=2247483782&amp;idx=1&amp;sn=962b840f2a362d1dbab306ca4eedf3cd&amp;chksm=e8210fe5df5686f31a96f7cfa37210232371356c0894421f14fea3a3e9c0b4e9f90179a2d974&amp;scene=21#wechat_redirect" data-itemshowtype="0" tab="innerlink" data-linktype="2">单细胞转录组基础分析三：降维与聚类</a><br></p><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzIyMzMwNDQ2MA==&amp;mid=2247483783&amp;idx=1&amp;sn=8836b7ec0a87d70ca889fff317417340&amp;chksm=e8210fe4df5686f2179b4a470418b62143fb97ef63dfe74cc137b6b8950a0cb26ce4e29b0d63&amp;scene=21#wechat_redirect" data-itemshowtype="0" tab="innerlink" data-linktype="2">单细胞转录组基础分析四：细胞类型鉴定</a><br></p><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzIyMzMwNDQ2MA==&amp;mid=2247483784&amp;idx=1&amp;sn=37cf87fe9b26566731a1afe264aad0ce&amp;chksm=e8210febdf5686fda5026e20bc0a949e1916b028fdea6f5caf1da91333f17dd3c7a86a406e0b&amp;scene=21#wechat_redirect" data-itemshowtype="0" tab="innerlink" data-linktype="2">单细胞转录组基础分析五：细胞再聚类</a><br></p><section><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzIyMzMwNDQ2MA==&amp;mid=2247483790&amp;idx=1&amp;sn=b5326820aae9007f1c23f0f14bd29af1&amp;chksm=e8210feddf5686fbc99f09c42b83218efe46faf8ff4903f411675069eb5818a69043cae6f651&amp;scene=21#wechat_redirect" data-itemshowtype="0" tab="innerlink" data-linktype="2">单细胞转录组基础分析六：伪时间分析</a></section><section><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzIyMzMwNDQ2MA==&amp;mid=2247483791&amp;idx=1&amp;sn=c5080db8e0fcd9fd7b657fc16068cffc&amp;chksm=e8210fecdf5686fa5012bc8226b15c6c3f149d7c3f6bb3a58eedcce51e995ca1e3a83ac42f34&amp;scene=21#wechat_redirect" data-itemshowtype="0" tab="innerlink" data-linktype="2">单细胞转录组基础分析七：差异基因富集分析</a></section><section><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzIyMzMwNDQ2MA==&amp;mid=2247483794&amp;idx=1&amp;sn=6425ab85a1f9c728b2685d1d4e7921c5&amp;chksm=e8210ff1df5686e782223e6d5818f604adc7a19b8d9acfc7067c45ee11328aa0fadd36b0bf35&amp;scene=21#wechat_redirect" data-itemshowtype="0" tab="innerlink" data-linktype="2">单细胞转录组基础分析八：可视化工具总结</a></p></section><section><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzIyMzMwNDQ2MA==&amp;mid=2247483784&amp;idx=1&amp;sn=37cf87fe9b26566731a1afe264aad0ce&amp;chksm=e8210febdf5686fda5026e20bc0a949e1916b028fdea6f5caf1da91333f17dd3c7a86a406e0b&amp;scene=21#wechat_redirect" data-itemshowtype="0" tab="innerlink" data-linktype="2"></a><br></section><section><span>SCENIC简介</span></section><section>组织内细胞异质性的基础是细胞转录状态的差异，转录状态的特异性又是由转录因子主导的基因调控网络（GRNs）决定并维持稳定的。因此分析单细胞的GRNs有助于深入挖掘细胞异质性背后的生物学意义，并为疾病的诊断、治疗以及发育分化的研究提供有价值的线索。然而单细胞转录组数据具有背景噪音高、基因检出率低和表达矩阵稀疏性的特点，给传统统计学和生物信息学方法推断高质量的GRNs带来了挑战。Single-cell regulatory network inference and clustering (SCENIC)是一种专为单细胞数据开发的GRNs算法，它的创新之处在于引入了转录因子motif序列验证统计学方法推断的基因共表达网络，从而识别高可靠性的由转录因子主导的GRNs。SCENIC相关的文章2017年首先发表于nature methods，2020年又将流程整理后发表于nature protocls。需要深入了解分析原理和流程的朋友可以参考这两篇文章：</section><p>SCENIC : single-cell regulatory network inference and clustering</p><p>A scalable SCENIC workflow for single-cell gene regulatory network analysis</p><section><br></section><section><strong>SCENIC分析流程</strong><br></section><section>官方介绍的主要分析有四步：</section><ol><li><p>GENIE3/GRNBoost：基于共表达情况鉴定每个TF的潜在靶点；</p></li><li><p>RcisTarget：基于DNA-motif 分析选择潜在的直接结合靶点；</p></li><li><p>AUCell：分析每个细胞的regulons活性；</p></li><li><p>细胞聚类：基于regulons的活性鉴定稳定的细胞状态并对结果进行探索<br></p></li></ol><section><br></section><section>分析流程可以细化为以下步骤：<strong><br></strong></section><p><img data-ratio="1.1482701812191103" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/VTXDic6Eia0AENFs370KozeMyOFOGczbPEqKAvGRfrYqg4fInq9JeGMyaf4SxesK6LiaI0bk4Vc2DhILCFm4439oA/640?wx_fmt=png" data-type="png" data-w="607" src="https://mmbiz.qpic.cn/mmbiz_png/VTXDic6Eia0AENFs370KozeMyOFOGczbPEqKAvGRfrYqg4fInq9JeGMyaf4SxesK6LiaI0bk4Vc2DhILCFm4439oA/640?wx_fmt=png"></p><section><br></section><p><br></p><p><span>SCENIC安装</span></p><p>详细的安装教程参阅官网：</p><p>https://rawcdn.githack.com/aertslab/SCENIC/701cc7cc4ac762b91479b3bd2eaf5ad5661dd8c2/inst/doc/SCENIC_Setup.html</p><p><br></p><p><strong>安装代码如下</strong><br></p><p>在安装SCENIC之前，请按照以下代码安装一些依赖包：</p><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="php"><code><span>if (!requireNamespace("BiocManager", quietly = TRUE)) install.packages("BiocManager")</span></code><code><span>BiocManager::version()</span></code><code><span># If your bioconductor version is previous to 3.9, see the section bellow</span></code><code><span><br></span></code><code><span>## Required</span></code><code><span>BiocManager::install(c("AUCell", "RcisTarget"))</span></code><code><span>BiocManager::install(c("GENIE3")) # Optional. Can be replaced by GRNBoost</span></code><code><span><br></span></code><code><span>## Optional (but highly recommended):</span></code><code><span># To score the network on cells (i.e. run AUCell):</span></code><code><span>BiocManager::install(c("zoo", "mixtools", "rbokeh"))</span></code><code><span># For various visualizations and perform t-SNEs:</span></code><code><span>BiocManager::install(c("DT", "NMF", "pheatmap", "R2HTML", "Rtsne"))</span></code><code><span># To support paralell execution (not available in Windows):</span></code><code><span>BiocManager::install(c("doMC", "doRNG"))</span></code><code><span># To export/visualize in http://scope.aertslab.org</span></code><code><span>if (!requireNamespace("devtools", quietly = TRUE)) install.packages("devtools")</span></code><code><span>devtools::install_github("aertslab/SCopeLoomR", build_vignettes = TRUE)</span></code></pre></section><p>检查一下核心依赖包的版本，并确保版本符合以下要求：</p><p>AUCell &gt;=1.4.1 (minimum 1.2.4)；</p><p>RcisTarget&gt;=1.2.0 (minimum 1.0.2)；</p><p>GENIE3&gt;=1.4.0 (minimum 1.2.1).<br></p><section><ul><li><li><li></ul><pre data-lang="javascript"><code><span>packageVersion("AUCell")</span></code><code><span>packageVersion("RcisTarget")</span></code><code><span>packageVersion("GENIE3")</span></code></pre></section><p>安装SCENIC包<br></p><section><ul><li><li><li></ul><pre data-lang="php"><code><span>if (!requireNamespace("devtools", quietly = TRUE)) install.packages("devtools")</span></code><code><span>devtools::install_github("aertslab/SCENIC") </span></code><code><span>packageVersion("SCENIC")</span></code></pre></section><p>还需要下载一些数据库</p><p>可以在此网址用浏览器下载：https://resources.aertslab.org/cistarget/</p><p>也可以使用以下代码在R环境中直接下载：<br></p><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="nginx"><code><span>##1, For human:</span></code><code><span>dbFiles &lt;- c("https://resources.aertslab.org/cistarget/databases/homo_sapiens/hg19/refseq_r45/mc9nr/gene_based/hg19-500bp-upstream-7species.mc9nr.feather",</span></code><code><span>"https://resources.aertslab.org/cistarget/databases/homo_sapiens/hg19/refseq_r45/mc9nr/gene_based/hg19-tss-centered-10kb-7species.mc9nr.feather")</span></code><code><span># mc9nr: Motif collection version 9: 24k motifs</span></code><code><span><br></span></code><code><span>##2, For mouse:</span></code><code><span>dbFiles &lt;- c("https://resources.aertslab.org/cistarget/databases/mus_musculus/mm9/refseq_r45/mc9nr/gene_based/mm9-500bp-upstream-7species.mc9nr.feather",</span></code><code><span>"https://resources.aertslab.org/cistarget/databases/mus_musculus/mm9/refseq_r45/mc9nr/gene_based/mm9-tss-centered-10kb-7species.mc9nr.feather")</span></code><code><span># mc9nr: Motif collection version 9: 24k motifs</span></code><code><span><br></span></code><code><span>##3, For fly:</span></code><code><span>dbFiles &lt;- c("https://resources.aertslab.org/cistarget/databases/drosophila_melanogaster/dm6/flybase_r6.02/mc8nr/gene_based/dm6-5kb-upstream-full-tx-11species.mc8nr.feather")</span></code><code><span># mc8nr: Motif collection version 8: 20k motifs</span></code><code><span><br></span></code><code><span>##4, download</span></code><code><span>dir.create("cisTarget_databases");   #创建一个文件夹保存数据库</span></code><code><span>setwd("cisTarget_databases")</span></code><code><span>#如果3个参考数据库都想下载，每次设置变量dbFiles后，都要运行以下代码</span></code><code><span>for(featherURL in dbFiles)</span></code><code><span>{</span></code><code><span>  download.file(featherURL, destfile=basename(featherURL)) # saved in current dir</span></code><code><span>}</span></code></pre></section><p><br></p><p><span>SCENIC分析准备</span></p><p>SCENIC分析需要输入单细胞表达矩阵和细胞meta信息，正式分析之前还要设置<span>运行线程数，参考数据库目录、版本，细胞meta信息存放目录等配置信息。</span></p><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="php"><code><span>library(Seurat)</span></code><code><span>library(tidyverse)</span></code><code><span>library(patchwork)</span></code><code><span>library(SCENIC)</span></code><code><span>rm(<span>list</span>=ls())</span></code><code><span><br></span></code><code><span><span>##==分析准备==##</span></span></code><code><span>dir.create(<span>"SCENIC"</span>)</span></code><code><span>dir.create(<span>"SCENIC/int"</span>)</span></code><code><span>scRNA &lt;- readRDS(<span>"scRNA.rds"</span>)</span></code><code><span>setwd(<span>"~/project/10xDemo2/SCENIC"</span>) </span></code><code><span><span>##准备细胞meta信息</span></span></code><code><span>cellInfo &lt;- data.frame(scRNA@meta.data)</span></code><code><span>colnames(cellInfo)[which(colnames(cellInfo)==<span>"orig.ident"</span>)] &lt;- <span>"sample"</span></span></code><code><span>colnames(cellInfo)[which(colnames(cellInfo)==<span>"seurat_clusters"</span>)] &lt;- <span>"cluster"</span></span></code><code><span>colnames(cellInfo)[which(colnames(cellInfo)==<span>"celltype_Monaco"</span>)] &lt;- <span>"celltype"</span></span></code><code><span>cellInfo &lt;- cellInfo[,c(<span>"sample"</span>,<span>"cluster"</span>,<span>"celltype"</span>)]</span></code><code><span>saveRDS(cellInfo, file=<span>"int/cellInfo.Rds"</span>)</span></code><code><span><span>##准备表达矩阵</span></span></code><code><span><span>#为了节省计算资源，随机抽取1000个细胞的数据子集</span></span></code><code><span>subcell &lt;- sample(colnames(scRNA),<span>1000</span>)</span></code><code><span>scRNAsub &lt;- scRNA[,subcell]</span></code><code><span>saveRDS(scRNAsub, <span>"scRNAsub.rds"</span>)</span></code><code><span>exprMat &lt;- <span>as</span>.matrix(scRNAsub@assays$RNA@counts)</span></code><code><span><span>##设置分析环境</span></span></code><code><span>mydbDIR &lt;- <span>"./cisTarget"</span></span></code><code><span>mydbs &lt;- c(<span>"hg38__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr.feather"</span>,</span></code><code><span>           <span>"hg38__refseq-r80__10kb_up_and_down_tss.mc9nr.feather"</span>)</span></code><code><span>names(mydbs) &lt;- c(<span>"500bp"</span>, <span>"10kb"</span>)</span></code><code><span>scenicOptions &lt;- initializeScenic(org=<span>"hgnc"</span>, </span></code><code><span>                                  nCores=<span>8</span>,</span></code><code><span>                                  dbDir=mydbDIR, </span></code><code><span>                                  dbs = mydbs,</span></code><code><span>                                  datasetTitle = <span>"HNSCC"</span>)</span></code><code><span>saveRDS(scenicOptions, <span>"int/scenicOptions.rds"</span>)</span></code></pre></section><p><span>scenicOptions &lt;- initializeScenic()</span>这一步要注意自己的情况设置参数：</p><ul><li><p><span>nCores=8</span>，代表开启8个线程计算，你的参数要根据自己电脑CPU的情况设置；</p></li><li><p><span>mydbDIR &lt;- "./cisTarget"</span>，设置的是我存放数据库的目录，你要填写自己存放数据库的目录；</p></li><li><p><span><span>dbs = mydbs</span>，</span>我在前面的变量设置中，把hg38的数据库文件赋值给了mydbs，你用hg19数据库或小鼠的数据库需要相应调整。<br></p></li></ul><p><span><br></span></p><p><span>共表达网络计算</span></p><p><strong>相关性回归分析</strong></p><p>SCENIC正式分析的第一步是计算转录因子与每个基因的相关性，此步骤消耗的计算资源非常大，作者建议两个策略处理大数据集：<br></p><ul><li><p>采用GENIE3推断共表达模块时随机抽取少量细胞计算，计算regulons的活性时，所有细胞都代入运算；</p></li><li><p>使用python版本的SCENIC，作者强烈推荐。</p></li></ul><p>为了减少初学者的学习负担，本文采用第一种策略。</p><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="properties"><code><span><span>##==转录调控网络推断==##</span></span></code><code><span><span>##基因过滤</span></span></code><code><span><span>#过滤标准是基因表达量之和&gt;细胞数*3%，且在1%的细胞中表达</span></span></code><code><span><span>genesKept</span> <span>&lt;- geneFiltering(exprMat, scenicOptions, </span></span></code><code><span>              <span>minCountsPerGene</span> = <span>3 * 0.01 * ncol(exprMat), </span></span></code><code><span>              <span>minSamples</span> = <span>ncol(exprMat) * 0.01)</span></span></code><code><span><span>exprMat_filtered</span> <span>&lt;- exprMat[genesKept, ]</span></span></code><code><span><span>##计算相关性矩阵</span></span></code><code><span><span>runCorrelation(exprMat_filtered, scenicOptions)</span></span></code><code><span><span>##TF-Targets相关性回归分析</span></span></code><code><span><span>exprMat_filtered_log &lt;- log2(exprMat_filtered+1)</span></span></code><code><span><span>runGenie3(exprMat_filtered_log, scenicOptions, nParts </span>=<span> 20)</span></span></code><code><span><span>#这一步消耗的计算资源非常大，个人电脑需要几个小时的运行时间</span></span></code></pre></section><p><span>runGenie3(exprMat_filtered_log, scenicOptions, nParts = 20)</span>需要注意<span>nParts</span>参数，它的作用是把表达矩阵分成n份分开计算，目的是防止数据量大时内存不够。以上代码运行后，int目录下有不少中间结果产生，简要解释一下：</p><ul><li><p>1.2_corrMat.Rds：基因之间的相关性矩阵</p></li><li><p>1.3_GENIE3_weightMatrix_part_1.Rds等：<span>GENIE3</span>的中间结果</p></li><li><p>1.4_GENIE3_linkList.Rds：<span>GENIE3</span>最终结果，是把“1.3_”开头的文件合并在一起。</p></li></ul><p><span>查看1.4_GENIE3_linkList.Rds</span>文件，<span>head(`1.4_GENIE3_linkList`)</span></p><pre tabindex="0"><span>           TF Target    weight<br>2003093  XBP1   MZB1 0.1616139<br>2003094  XBP1  DERL3 0.1489923<br>2003095  XBP1  PRDX4 0.1393732<br>1003616 RPS4X  RPL34 0.1285798<br>1254129 RPS4X   RPS6 0.1265229<br>1       RPS4X  RPL13 0.1256756</span></pre><p>TF是转录因子名称，Target是潜在靶基因的名字，weight是TF与Target之间的相关性权重。<br></p><p><br></p><p><strong>推断共表达模块</strong><br></p><p>上一步计算了转录因子与每一个基因的相关性，接下来需要过滤低相关性的组合形成共表达基因集（模块）。作者尝试了多种策略（标准）过滤低相关性TF-Target，研究发现没有一种最佳策略，因此他们的建议是6种过滤标准都用。这6种方法分别是：<br></p><ol><li><p>w001：以每个TF为核心保留weight&gt;0.001的基因形成共表达模块；</p></li><li><p>w005：<span>以每个TF为核心保留weight&gt;0.005的基因形成共表达模块；</span></p></li><li><p>top50：<span>以每个TF为核心保留weight值top50的基因形成共表达模块；</span></p></li><li><p>top5perTarget：每个基因保留<span>weight值top5的TF得到精简的TF-<span>Target</span>关联表，然后把基因分配给TF构建<span>共表达模块</span>；</span></p></li><li><p><span>top10perTarget：<span>每个基因保留weight值top10的TF得到精简的TF-Target关联表，然后把基因分配给TF构建共表达模块；</span></span></p></li><li><p><span>top50perTarget：<span>每个基因保留weight值top50的TF得到精简的TF-Target关联表，然后把基因分配给TF构建共表达模块；</span></span><br></p></li></ol><section><ul><li><li></ul><pre data-lang="css"><code><span>##推断共表达模块</span></code><code><span><span>runSCENIC_1_coexNetwork2modules</span>(<span>scenicOptions</span>)</span></code></pre></section><p>主要运行结果是int目录下的<span>1.6_tfModules_asDF.Rds</span></p><pre tabindex="0"><span>     Target    TF method corr<br>1   BCDIN3D ABCF2   w001    1<br>2    STRADB ABCF2   w001    1<br>3 HIST1H2BG ABCF2   w001    1<br>4      GID4 ABCF2   w001    1<br>5    LPCAT1 ABCF2   w001   -1<br>6      ZHX2 ABCF2   w001    1</span></pre><p><span>method是上面提到的6种方法，corr是<span>runCorrelation(exprMat_filtered, scenicOptions)</span>命令得到的，1代表激活，-1代表抑制，0代表中性，SCENIC只会采用corr值为1的数据用于后续分析。</span><br></p><p><span><br></span></p><p><span>Motif验证共表达模块</span></p><p><span>经过上述分析每个转录因子都找到了强相关的靶基因，很多基因调控网络分析到此就结束了。SCENIC的创新之处是对此结果提出了质疑，并通过以下步骤修剪共表达模块形成有生物学意义的调控单元（regulons）：</span></p><ol><li><p><span>对每个共表达模块进行motif富集分析，保留显著富集的motif；此项分析依赖gene-motif评分（排行）数据库，其行为基因、列为motif、值为排名，就是我们下载的<span>cisTarget数据库。</span></span></p></li><li><p><span>使用数据库对motif进行TF注释，注释结果分</span>高、低可信度 <span>。数据库直接注释和同源基因推断的TF是高可信结果，使用motif序列相似性注释<span>的TF是低可信结果</span>。</span></p></li><li><p><span>用保留的motif对共表达模块内的基因进行打分（同样依据<span>cisTarget数据库</span>），识别显著高分的基因（理解为motif离这些基因的TSS很近）；</span></p></li><li><p><span>删除共表达模块内与<span>motif评分不高的基因，剩下的基因集作者称为<strong>调控单元（regulon）</strong>。</span></span></p></li></ol><section><ul><li><li><li><li></ul><pre data-lang="swift"><code><span>##推断转录调控网络（regulon）</span></code><code><span>runSCENIC_2_createRegulons(scenicOptions)</span></code><code><span>#以上代码可增加参数coexMethod=<span>c</span>(<span>"w001"</span>, <span>"w005"</span>, <span>"top50"</span>, <span>"top5perTarget"</span>, <span>"top10perTarget"</span>, <span>"top50perTarget"</span>))</span></code><code><span>#默认<span>6</span>种方法的共表达网络都计算，可以少选几种方法以减少计算量</span></code></pre></section><p>内存不够的电脑在<span>这一步运行时</span>会报错，重要的结果保存在output文件夹：<br></p><p><strong>Step2_MotifEnrichment.tsv</strong></p><p><img data-ratio="0.12734375" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/VTXDic6Eia0AHsjbBuaaQoj9yKS1wHSY0pJ49f5OwIdKKRuh6XNd6cic6zkdoufPWd2YRCg9SwpUqVkejib2CvbLUA/640?wx_fmt=png" data-type="png" data-w="1280" src="https://mmbiz.qpic.cn/mmbiz_png/VTXDic6Eia0AHsjbBuaaQoj9yKS1wHSY0pJ49f5OwIdKKRuh6XNd6cic6zkdoufPWd2YRCg9SwpUqVkejib2CvbLUA/640?wx_fmt=png"></p><p>这是各个共表达模块显著富集motif的注释信息，表头官方解释如下：<br></p><ul><li><p>geneSet: Name of the gene set</p></li><li><p>motif: ID of the motif</p></li><li><p>NES: Normalized enrichment score of the motif in the gene-set</p></li><li><p>AUC: Area Under the Curve (used to calculate the NES)</p></li><li><p>TFinDB: Indicates whether the <em>highlightedTFs</em> are included within the high confidence annotation (two asterisks) or low confidence annotation (one asterisk).</p></li><li><p>TF_highConf: Transcription factors annotated to the motif according to ‘motifAnnot_highConfCat’.</p></li><li><p>TF_lowConf: Transcription factors annotated to the motif according to ‘motifAnnot_lowConfCat’.</p></li><li><p>enrichedGenes: Genes that are highly ranked for the given motif.</p></li><li><p>nErnGenes: Number of genes highly ranked</p></li><li><p>rankAtMax: Ranking at the maximum enrichment, used to determine the number of enriched genes.</p></li></ul><p><strong>Step2_MotifEnrichment_preview.html</strong></p><p><img data-ratio="0.3022421524663677" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/VTXDic6Eia0AHsjbBuaaQoj9yKS1wHSY0p8g5zxjerr3YJa1muatu5AzFIq1uQpMRKFl7ZibUoSRD3MxYbcU44wGw/640?wx_fmt=png" data-type="png" data-w="1115" src="https://mmbiz.qpic.cn/mmbiz_png/VTXDic6Eia0AHsjbBuaaQoj9yKS1wHSY0p8g5zxjerr3YJa1muatu5AzFIq1uQpMRKFl7ZibUoSRD3MxYbcU44wGw/640?wx_fmt=png"></p><p><span>显著富集的motif注释信息，表头含义同上个文件。</span><br></p><p><span><br></span></p><p><strong>Step2_regulonTargetsInfo.tsv</strong></p><p><img data-ratio="0.14913294797687862" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/VTXDic6Eia0AHsjbBuaaQoj9yKS1wHSY0pfmyFr0qaj5VuGm1VaZBJMUic0G4VMrfiax9kmhKBBrQ8JQfUE5rOXH8A/640?wx_fmt=png" data-type="png" data-w="865" src="https://mmbiz.qpic.cn/mmbiz_png/VTXDic6Eia0AHsjbBuaaQoj9yKS1wHSY0pfmyFr0qaj5VuGm1VaZBJMUic0G4VMrfiax9kmhKBBrQ8JQfUE5rOXH8A/640?wx_fmt=png"></p><p>最终的regulon文件，将第一个文件的数据整合在了一起，表头含义如下：</p><ul><li><p><span>TF：转录因子名称</span></p></li><li><p><span>gene：TF靶基因名称<br></span></p></li><li><p><span>nMotif：靶基因在数据库的motif数量<br></span></p></li><li><p><span>bestMotif：最显著富集的motif名称<br></span></p></li><li><p><span>NES：标准富集分数，分值越高越显著<br></span></p></li><li><p><span>highConfAnnot：是不是高可信注释<br></span></p></li><li><p><span>Genie3Weight：TF与靶基因的相关性权重</span></p></li></ul><p><span><strong>请特别注意这个文件，后续分析找到了有价值的regulon，需要回到这个文件找对应的转录因子和靶基因 。</strong></span>SCENIC中<span>regulon的名称有两种，一种是TF名称+<span>extended+靶基因数目，另一种是<span>TF</span><span>名称+</span><span>靶基因数目。第一种是</span></span></span><span>转录因子与</span><span>所有靶基因组成的基因</span><span>调控</span><span>网络，</span><span>第二种是<span>转录因子与高可信</span></span><span>靶基因（<span>即highConfAnnot=TRUE的基因</span>）组成的基因调控网络。</span></p><p><span><br></span></p><p><span></span></p><p><span>Regulon活性评分与可视化</span><br></p><p><span>每个Regulon就是一个转录因子及其直接调控靶基因的基因集，SCENIC接下来的工作就是对每个regulon在各个细胞中的活性评分。评分的基础是基因的表达值，分数越高代表基因集的激活程度越高。我们推断regulons虽然只用了1000个随机抽取的细胞，但是regulon评分的时候可以把所有细胞导进来计算。<br></span></p><section><ul><li><li><li><li><li></ul><pre data-lang="nginx"><code><span><span>##==regulon活性评分与可视化==##</span></span></code><code><span><span>##regulons计算AUC值并进行下游分析</span></span></code><code><span><span>exprMat_all</span> &lt;- as.matrix(scRNA<span>@assays</span><span>$RNA</span><span>@counts</span>)</span></code><code><span>exprMat_all &lt;- log2(exprMat_all+<span>1</span>)</span></code><code><span>runSCENIC_3_scoreCells(scenicOptions, exprMat=exprMat_all)</span></code></pre></section><p><span></span><span><span>runSCENIC_3_scoreCells(</span><span>)</span></span>是一个多种功能打包的函数，它包含的子功能有：</p><ol><li><p>计算regulon在每个细胞中AUC值，最后得到一个以regulon为行细胞为列的矩阵。</p><p>结果目录：int/3.4_regulonAUC.Rds</p></li><li><p>计算每个regulon的AUC阈值，细胞中regulonAUC值&gt;阈值，代表此regulon在细胞中处于激活状态，否则代表沉默状态。</p><p>结果目录：int/3.5_AUCellThresholds.Rds</p></li><li><p>使用regulonAUC矩阵对细胞进行降维聚类</p></li><li><p>用heatmap图展示regulonAUC矩阵，用<span>t-SNE</span>图分别展示每个regulon的活性分布情况。</p><p>结果目录：output/Step3_开头的系列文件</p></li></ol><p>个人觉得分析结果中regulon活性tSNE图更有价值，举例说明如下：</p><p><img data-ratio="0.25218774860779636" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/VTXDic6Eia0AEP8PichqBMO7ianibf0ABuicYw4DKPe5E8jRXS7kUKnAVBbVUcWn3JGia7FELGdvXtcbeLssVSvq3wlrQ/640?wx_fmt=png" data-type="png" data-w="1257" src="https://mmbiz.qpic.cn/mmbiz_png/VTXDic6Eia0AEP8PichqBMO7ianibf0ABuicYw4DKPe5E8jRXS7kUKnAVBbVUcWn3JGia7FELGdvXtcbeLssVSvq3wlrQ/640?wx_fmt=png"></p><p><span><span>左图展示的是某个regulon在所有细胞中AUC</span>值分布情况，红色虚线是自动分配的阈值分界线；左中图蓝色点<span>是regulonAUC值大于阈值的细胞，灰色点是regulon<span>AUC值小于阈值的细胞</span></span>；右<span>中图是regulonAUC原始值的tSNE图；右图是regulon中转录因子的表达量<span>tSNE图</span>。</span></span></p><p><span><br></span></p><p><span><span>Regulon活性二进制转换与可视化</span></span></p><p>对于细胞类型清晰的数据集而言，构建regulons并对其活性打分足够后续分析。但是，在很多情况下将评分转换为二进制的“开/关”，则既方便解释，又能最大化体现细胞类型的差异。将特定的regulon转换为“0/1”有利于探索和解释关键转录因子。将所有的regulons转换为“0/1”后创建二进制的活性矩阵，则可以用于细胞聚类，对消除技术偏倚特别有用。<span>AUCell会自动计算可能的阈值进行“0/1”转换，作者建议在转换之前手工检查并调整这些阈值。</span></p><p><br></p><p><strong>查看调整阈值的代码(可选)</strong></p><section><ul><li><li><li><li><li><li><li><li></ul><pre data-lang="nginx"><code><span><span>#使用shiny互动调整阈值</span></span></code><code><span><span>aucellApp</span> &lt;- plotTsne_AUCellApp(scenicOptions, exprMat_all)</span></code><code><span>savedSelections &lt;- shiny::runApp(aucellApp)</span></code><code><span><span>#保存调整后的阈值</span></span></code><code><span>newThresholds &lt;- savedSelections<span>$thresholds</span></span></code><code><span>scenicOptions<span>@fileNames</span><span>$int</span>[<span>"aucell_thresholds"</span>,<span>1</span>] &lt;- <span>"int/newThresholds.Rds"</span></span></code><code><span>saveRDS(newThresholds, file=getIntName(scenicOptions, <span>"aucell_thresholds"</span>))</span></code><code><span>saveRDS(scenicOptions, file=<span>"int/scenicOptions.Rds"</span>)</span></code></pre></section><p><span><span></span></span><strong>二进制转换及衍生分析</strong><br></p><section><ul><li></ul><pre><code><span>runSCENIC_4_aucell_binarize(scenicOptions, exprMat=exprMat_all)</span></code></pre></section><p><span><span></span></span><span>runSCENIC_4_aucell_binarize()</span>将regulonAUC矩阵转换为二进制矩阵后，会重新降维聚类，输出的结果与<span>runSCENIC_3_scoreCells()</span>类似。</p><p><span><span><br></span></span></p><p><span><span>SCENIC结果可视化<br></span></span></p><p><span>runSCENIC_3_scoreCells(</span><span>)和<span>runSCENIC_4_aucell_binarize()</span></span>运行之后会生成一些可视化的heatmap图与tSNE图，但是他们既不容易与seurat分析的结果联系起来，又不容易调整图形参数和分析内容。我们可以调用SCENIC的分析结果，使用seurat和pheatmap进行可视化。</p><p><img data-ratio="0.3125" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/VTXDic6Eia0AEePeQRte1OwiboHfqh7zpNpQ9UUibFn3HkicHWH5UNibUricltuJuajAtstC4afBrh2DPjia7FQVI6jlWA/640?wx_fmt=png" data-type="png" data-w="1280" src="https://mmbiz.qpic.cn/mmbiz_png/VTXDic6Eia0AEePeQRte1OwiboHfqh7zpNpQ9UUibFn3HkicHWH5UNibUricltuJuajAtstC4afBrh2DPjia7FQVI6jlWA/640?wx_fmt=png"></p><p><span>热图图例显示不全，尺寸不可调；中图和右图是<span>runSCENIC_3与<span>runSCENIC_4</span></span>得到的tSNE图，与seurat的tSNE图很难联系起来。</span></p><p><br></p><p><strong>Seurat可视化SCENIC结果</strong></p><p>把SCENIC结果中最重要的regulonAUC矩阵导入Seurat，这样得到的可视化结果更容易与我们之前的分析联系起来。</p><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="perl"><code><span><span>##导入原始regulonAUC矩阵</span></span></code><code><span>AUCmatrix &lt;- readRDS(<span>"int/3.4_regulonAUC.Rds"</span>)</span></code><code><span>AUCmatrix &lt;- AUCmatrix@assays@data@listData$AUC</span></code><code><span>AUCmatrix &lt;- data.frame(t(AUCmatrix), check.names=F)</span></code><code><span>RegulonName_AUC &lt;- colnames(AUCmatrix)</span></code><code><span>RegulonName_AUC &lt;- gsub(<span>' \\('</span>,<span>'_'</span>,RegulonName_AUC)</span></code><code><span>RegulonName_AUC &lt;- gsub(<span>'\\)'</span>,<span>''</span>,RegulonName_AUC)</span></code><code><span>colnames(AUCmatrix) &lt;- RegulonName_AUC</span></code><code><span>scRNAauc &lt;- AddMetaData(scRNA, AUCmatrix)</span></code><code><span>scRNAauc@assays$integrated &lt;- NULL</span></code><code><span>saveRDS(scRNAauc,<span>'scRNAauc.rds'</span>)</span></code><code><span><br></span></code><code><span><span>##导入二进制regulonAUC矩阵</span></span></code><code><span>BINmatrix &lt;- readRDS(<span>"int/4.1_binaryRegulonActivity.Rds"</span>)</span></code><code><span>BINmatrix &lt;- data.frame(t(BINmatrix), check.names=F)</span></code><code><span>RegulonName_BIN &lt;- colnames(BINmatrix)</span></code><code><span>RegulonName_BIN &lt;- gsub(<span>' \\('</span>,<span>'_'</span>,RegulonName_BIN)</span></code><code><span>RegulonName_BIN &lt;- gsub(<span>'\\)'</span>,<span>''</span>,RegulonName_BIN)</span></code><code><span>colnames(BINmatrix) &lt;- RegulonName_BIN</span></code><code><span>scRNAbin &lt;- AddMetaData(scRNA, BINmatrix)</span></code><code><span>scRNAbin@assays$integrated &lt;- NULL</span></code><code><span>saveRDS(scRNAbin, <span>'scRNAbin.rds'</span>)</span></code><code><span><br></span></code><code><span><span>##利用Seurat可视化AUC</span></span></code><code><span>dir.create(<span>'scenic_seurat'</span>)</span></code><code><span><span>#FeaturePlot</span></span></code><code><span>p1 = FeaturePlot(scRNAauc, features=<span>'CEBPB_extended_2290g'</span>, label=T, reduction = <span>'tsne'</span>)</span></code><code><span>p2 = FeaturePlot(scRNAbin, features=<span>'CEBPB_extended_2290g'</span>, label=T, reduction = <span>'tsne'</span>)</span></code><code><span>p3 = DimPlot(scRNA, reduction = <span>'tsne'</span>, group.by = <span>"celltype_Monaco"</span>, label=T)</span></code><code><span>plotc = p1|p2|p3</span></code><code><span>ggsave(<span>'scenic_seurat/CEBPB_extended_2290g.png'</span>, plotc, width=<span>14</span> ,height=<span>4</span>)</span></code></pre></section><p><img data-ratio="0.28515625" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/VTXDic6Eia0AENFs370KozeMyOFOGczbPEmekicZGqokLKkALuIU0jxhpRmIJl1H6koTasNkgDV3rwH4eibu4hCqOg/640?wx_fmt=png" data-type="png" data-w="1280" src="https://mmbiz.qpic.cn/mmbiz_png/VTXDic6Eia0AENFs370KozeMyOFOGczbPEmekicZGqokLKkALuIU0jxhpRmIJl1H6koTasNkgDV3rwH4eibu4hCqOg/640?wx_fmt=png"></p><p><span>决定单核细胞命运的regulon：转录因子CEBPB主导的调控网络</span><br></p><section><ul><li><li><li><li><li><li><li></ul><pre data-lang="cs"><code><span><span>#RidgePlot&amp;VlnPlot</span></span></code><code><span>p1 = RidgePlot(scRNAauc, features = <span>"CEBPB_extended_2290g"</span>, <span>group</span>.<span>by</span>=<span>"celltype_Monaco"</span>) + </span></code><code><span>               theme(legend.position=<span>'none'</span>)</span></code><code><span>p2 = VlnPlot(scRNAauc, features = <span>"CEBPB_extended_2290g"</span>, pt.size = <span>0</span>, <span>group</span>.<span>by</span>=<span>"celltype_Monaco"</span>) + </span></code><code><span>             theme(legend.position=<span>'none'</span>)</span></code><code><span>plotc = p1 + <span>p2</span></span></code><code><span><span>ggsave</span>(<span><span>'scenic_seurat/Ridge-Vln_CEBPB_extended_2290g.png'</span>, plotc, width=<span>10</span>, height=<span>8</span></span>)</span></code></pre></section><p><img data-ratio="0.8" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/VTXDic6Eia0AENFs370KozeMyOFOGczbPEr4bWJ9gIbRKRsexiaiaYRF2FujdBzZHNIoj5k8EKe8cK9QTX6zsxJ5sA/640?wx_fmt=png" data-type="png" data-w="1280" src="https://mmbiz.qpic.cn/mmbiz_png/VTXDic6Eia0AENFs370KozeMyOFOGczbPEr4bWJ9gIbRKRsexiaiaYRF2FujdBzZHNIoj5k8EKe8cK9QTX6zsxJ5sA/640?wx_fmt=png"></p><p><span>用山脊图和小提琴图展示CEBPB调控网络的AUC值</span><br></p><p><br></p><p><strong>pheatmap可视化<strong>SCENIC结果</strong></strong></p><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="perl"><code><span>library(pheatmap)</span></code><code><span>cellInfo &lt;- readRDS(<span>"int/cellInfo.Rds"</span>)</span></code><code><span>celltype = subset(cellInfo,<span>select</span> = <span>'celltype'</span>)</span></code><code><span>AUCmatrix &lt;- t(AUCmatrix)</span></code><code><span>BINmatrix &lt;- t(BINmatrix)</span></code><code><span><span>#挑选部分感兴趣的regulons</span></span></code><code><span>my.regulons &lt;- c(<span>'ETS1_2372g'</span>,<span>'ETV7_981g'</span>,<span>'IRF7_239g'</span>,<span>'XBP1_854g'</span>,<span>'ATF4_37g'</span>,</span></code><code><span>                 <span>'KLF13_78g'</span>,<span>'ATF6_129g'</span>,<span>'CREB3L2_619g'</span>,<span>'TAGLN2_13g'</span>,</span></code><code><span>                 <span>'STAT1_extended_1808g'</span>,<span>'CEBPB_extended_2290g'</span>,<span>'IRF5_extended_422g'</span>,</span></code><code><span>                 <span>'SPI1_1606g'</span>,<span>'HMGA1_14g'</span>,<span>'SPIB_1866g'</span>,<span>'IRF8_348g'</span>,<span>'BCL11A_136g'</span>,</span></code><code><span>                 <span>'EBF1_40g'</span>,<span>'MAF_45g'</span>,<span>'BATF_131g'</span>,<span>'FOXP3_55g'</span>,<span>'TBX21_388g'</span>,</span></code><code><span>                 <span>'EOMES_extended_101g'</span>,<span>'TCF7_extended_31g'</span>,<span>'LEF1_extended_49g'</span>)</span></code><code><span>myAUCmatrix &lt;- AUCmatrix[rownames(AUCmatrix)%in%my.regulons,]</span></code><code><span>myBINmatrix &lt;- BINmatrix[rownames(BINmatrix)%in%my.regulons,]</span></code><code><span><span>#使用regulon原始AUC值绘制热图</span></span></code><code><span>pheatmap(myAUCmatrix, show_colnames=F, annotation_col=celltype,</span></code><code><span>         filename = <span>'scenic_seurat/myAUCmatrix_heatmap.png'</span>,</span></code><code><span>         width = <span>6</span>, height = <span>5</span>)</span></code><code><span><span>#使用regulon二进制AUC值绘制热图</span></span></code><code><span>pheatmap(myBINmatrix, show_colnames=F, annotation_col=celltype,</span></code><code><span>         filename = <span>'scenic_seurat/myBINmatrix_heatmap.png'</span>,</span></code><code><span>         color = colorRampPalette(colors = c(<span>"white"</span>,<span>"black"</span>))(<span>100</span>),</span></code><code><span>         width = <span>6</span>, height = <span>5</span>)</span></code></pre></section><p><img data-croporisrc="https://mmbiz.qlogo.cn/mmbiz_jpg/VTXDic6Eia0AEePeQRte1OwiboHfqh7zpNpHUKOib7sn05pR811sa5RE0dLd6TK3635Gw4vAenibenjalrqnH2PzMWw/0?wx_fmt=jpeg" data-cropx1="0" data-cropx2="1000" data-cropy1="88.23529411764707" data-cropy2="600.3460207612458" data-ratio="0.512" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/VTXDic6Eia0AEePeQRte1OwiboHfqh7zpNpficIfjzMYY2297LyVibxunBu7Kx0FIZCyf3Sq1m6icytxmgK6oCGKrn6w/640?wx_fmt=jpeg" data-type="jpeg" data-w="1000" src="https://mmbiz.qpic.cn/mmbiz_jpg/VTXDic6Eia0AEePeQRte1OwiboHfqh7zpNpficIfjzMYY2297LyVibxunBu7Kx0FIZCyf3Sq1m6icytxmgK6oCGKrn6w/640?wx_fmt=jpeg"></p><p>有兴趣的朋友还可以做个附加题：把CEBPB主导的基因调控网络做GO和KEGG富集分析。</p><p><br></p><p><span><span><strong>后记</strong></span></span></p><p>写这篇教程我花了一周的业余时间，这几天我都在忙些什么呢？考虑到服务器的配置和使用门槛较高，大家很可能在个人电脑上运行教程的内容，我所有的代码都会在自己的笔记本上验证一遍。此前文章的代码都没有问题，这次运行SCENIC遇到过不去的坎了。SCENIC有几步的计算量非常大，我的电脑运行一次要10几个小时，连续尝试了几个晚上都报错。后来搞清楚根本原因是内存不够，最后用服务器才顺利跑完了流程。如果你也遇到以下警告和错误提示信息，请放弃分析或减少分析的细胞数。</p><section><ul><li><li><li><li><li></ul><pre data-lang="sql"><code><span>Warning message:</span></code><code><span>In mclapply(argsList, FUN, mc.preschedule = preschedule, mc.set.seed = set.seed,  :</span></code><code><span> scheduled cores <span>3</span>, <span>6</span>, <span>8</span> did <span>not</span> deliver results, <span>all</span> <span>values</span> <span>of</span> the jobs will be affected</span></code><code><span><br></span></code><code><span><span>Error</span> <span>in</span> signif(trhAssignment, <span>3</span>)</span></code></pre></section><p><span></span></p><section><br></section><section><span>获取帮助</span></section><p cid="n22" mdtype="heading">本教程的目的在于把单细胞分析流程串起来，适合有一定R语言基础的朋友参考。分析原理和代码我没有详细解释，官网有详细的教程和权威的解释，翻译好的中文教程也有多个版本，有兴趣的可以搜索一下。如果您学习本教程有一定困难，可以分享此篇文章到朋友圈，截图微信发给Kinesin（文末二维码），我会抽时间组群直播讲解单细胞数据分析的全过程。本专题所用的软件、数据、代码脚本和中间结果，我打包放在了百度云上，需要的朋友添加Kinesin微信索取。</p><p><br></p><p><br></p><p><img data-cropselx1="0" data-cropselx2="578" data-cropsely1="0" data-cropsely2="200" data-ratio="0.3157262905162065" data-s="300,640" data-type="png" data-w="833" data-src="https://mmbiz.qpic.cn/mmbiz_png/VTXDic6Eia0AE6434HBqdqv7YmtjwOUcB2dc6dY8hNvAExTBrmNFFOHzG88eUtXwEvVIibzQbibfxfczawALicJfS4w/640?wx_fmt=png" src="https://mmbiz.qpic.cn/mmbiz_png/VTXDic6Eia0AE6434HBqdqv7YmtjwOUcB2dc6dY8hNvAExTBrmNFFOHzG88eUtXwEvVIibzQbibfxfczawALicJfS4w/640?wx_fmt=png"></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/QBBQ2TXQzNrjPILNi4EHdA",target="_blank" rel="noopener noreferrer">原文链接</a>
