---
title: "整合单细胞和空转数据多种方法之CellTrek"
date: 2023-11-05T09:29:52Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Acdemic"]
---
整合单细胞和空转数据多种方法之CellTrek by 生信菜鸟团
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><p data-tool="mdnice编辑器">CellTrek发表于2022年的<strong>Nature Biotechnology</strong>，题为《Spatial charting of single-cell transcriptomes in tissues》。CellTrek可以结合单细胞和空间转录组数据准确地定位组织内单个细胞的位置，并构建空间细胞图谱。gitHub在https://github.com/navinlabcode/CellTrek</p><p data-tool="mdnice编辑器"><strong>本文将介绍：</strong></p><ul data-tool="mdnice编辑器"><li><section>CellTrek算法的工作流程；</section></li><li><section>使用R实现方法。</section></li></ul><h3 data-tool="mdnice编辑器">一. CellTrek算法的工作流程</h3><figure data-tool="mdnice编辑器"><img data-ratio="0.6016597510373444" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/iaRJcrq2Los9fJy5icvttNBiaoVLticicSO6arUVSsh3YKbFiaCniabtrbFtibiaSXLayruHDJrVnrCgKd6xpb5iciazMjz3w/640?wx_fmt=other" data-type="other" data-w="723" src="https://mmbiz.qpic.cn/mmbiz_jpg/iaRJcrq2Los9fJy5icvttNBiaoVLticicSO6arUVSsh3YKbFiaCniabtrbFtibiaSXLayruHDJrVnrCgKd6xpb5iciazMjz3w/640?wx_fmt=other"><figcaption>图1:CellTrek工作流程概述。</figcaption></figure><ul data-tool="mdnice编辑器"><li><section><p>CellTrek首先将ST和scRNA-seq的原始矩阵合并，接着将合并后的矩阵进行共降维。然后提取其中的空间转录组数据构建一个多元随机森林模型(RF)，其中空间坐标是结果，潜在特征是预测因子；</p></section></li><li><section><p>对ST数据进行二维空间插值以增强ST的spots。然后，将训练好的RF模型应用于共嵌数据（ST插值），以生成RF距离矩阵，该矩阵将使用最近邻对（MNN）转换为稀疏图；</p></section></li><li><section><p>最后基于稀疏图，将单细胞转录组数据套入RF模型中，构建Spot-Cell表达相似性矩阵，为每个单细胞结果添加空间坐标信息并实现可视化作图。</p></section></li></ul><p data-tool="mdnice编辑器">众所周知，10X空转数据的一个spots不是单细胞分辨率的，而是包含十余个细胞的混合点。<strong>而CellTrek算法可以充分利用单细胞转录组数据，将单细胞信息映射至空间转录组切片上 。值得注意的是，不同于之前的去卷积方法，CellTrek可以实现空转数据的单细胞分辨率。</strong></p><h3 data-tool="mdnice编辑器">二. CellTrek代码实现</h3><p data-tool="mdnice编辑器">CellTrek安装比较简单：</p><pre data-tool="mdnice编辑器"><span></span><code><span>library</span>(devtools)<br>install_github(<span>"navinlabcode/CellTrek"</span>)<br></code></pre><p data-tool="mdnice编辑器">R包加载：</p><pre data-tool="mdnice编辑器"><span></span><code><span>library</span>(Seurat)<br><span>library</span>(SeuratData)<br><span>library</span>(ggplot2)<br><span>library</span>(patchwork)<br><span>library</span>(dplyr)<br><span>library</span>(readr)<br><span>library</span>(CellTrek)<br><span>library</span>(viridis)<br><span>library</span>(ConsensusClusterPlus)<br><span>library</span>(SeuratObject)<br></code></pre><h4 data-tool="mdnice编辑器">Step1.加载示例数据</h4><pre data-tool="mdnice编辑器"><span></span><code>brain_st_cortex &lt;- read_rds(<span>"./Rawdata/brain_st_cortex.rds"</span>)<br>brain_sc &lt;- read_rds(<span>"./Rawdata/brain_sc.rds"</span>)<br><br><span>## Rename the cells/spots with syntactically valid names</span><br>brain_st_cortex &lt;- RenameCells(brain_st_cortex, new.names=make.names(Cells(brain_st_cortex)))<br>brain_sc &lt;- RenameCells(brain_sc, new.names=make.names(Cells(brain_sc)))<br></code></pre><p data-tool="mdnice编辑器">可视化空转数据：</p><pre data-tool="mdnice编辑器"><span></span><code><span>## Visualize the ST data</span><br>SpatialDimPlot(brain_st_cortex)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.6649395509499136" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/iaRJcrq2Los9fJy5icvttNBiaoVLticicSO6aNyZiaEboBU27asJTs6icQerafvX2NsHhlN3drWYIPOj5nN0Dldd29wKg/640?wx_fmt=other" data-type="other" data-w="579" src="https://mmbiz.qpic.cn/mmbiz_jpg/iaRJcrq2Los9fJy5icvttNBiaoVLticicSO6aNyZiaEboBU27asJTs6icQerafvX2NsHhlN3drWYIPOj5nN0Dldd29wKg/640?wx_fmt=other"><figcaption>image-20231105144510038</figcaption></figure><p data-tool="mdnice编辑器">可视化单细胞数据：</p><pre data-tool="mdnice编辑器"><span></span><code><span>## Visualize the scRNA-seq data</span><br>DimPlot(brain_sc, label = <span>T</span>, label.size = <span>4.5</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7130801687763713" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/iaRJcrq2Los9fJy5icvttNBiaoVLticicSO6au8Sib5IldfWiboyUxyeIdcflft52yBoMNHgL2qaBSMzpDSrICibjWNkbQ/640?wx_fmt=other" data-type="other" data-w="474" src="https://mmbiz.qpic.cn/mmbiz_jpg/iaRJcrq2Los9fJy5icvttNBiaoVLticicSO6au8Sib5IldfWiboyUxyeIdcflft52yBoMNHgL2qaBSMzpDSrICibjWNkbQ/640?wx_fmt=other"><figcaption>image-20231105144526583</figcaption></figure><h4 data-tool="mdnice编辑器">Step2. 使用CellTrek进行细胞映射</h4><pre data-tool="mdnice编辑器"><span></span><code><span>#### 2. 使用 CellTrek 进行细胞映射</span><br><span># We first co-embed ST and scRNA-seq datasets using traint</span><br>brain_traint &lt;- CellTrek::traint(st_data=brain_st_cortex, <br>                                 sc_data=brain_sc, <br>                                 sc_assay=<span>'RNA'</span>, <br>                                 cell_names=<span>'cell_type'</span>)<br><br><span>## 我们可以检查共嵌入的结果，以查看这两种数据模态之间是否存在重叠。</span><br>DimPlot(brain_traint, group.by = <span>"type"</span>) <br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.8868421052631579" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/iaRJcrq2Los9fJy5icvttNBiaoVLticicSO6axdPlYj3waaF4ibFo4zalBsEWsnKXS8BUsdaOHyut54kDmYJj0cVloWg/640?wx_fmt=other" data-type="other" data-w="380" src="https://mmbiz.qpic.cn/mmbiz_jpg/iaRJcrq2Los9fJy5icvttNBiaoVLticicSO6axdPlYj3waaF4ibFo4zalBsEWsnKXS8BUsdaOHyut54kDmYJj0cVloWg/640?wx_fmt=other"><figcaption>image-20231105144622926</figcaption></figure><pre data-tool="mdnice编辑器"><span></span><code><span># 在共嵌入之后，我们可以将单个细胞映射到它们的空间位置。</span><br><span># 在这里，我们使用非线性插值（intp = T，intp_lin=F）方法来增强ST的spots</span><br>brain_celltrek &lt;- CellTrek::celltrek(st_sc_int=brain_traint, int_assay=<span>'traint'</span>, sc_data=brain_sc, sc_assay = <span>'RNA'</span>, <br>                                     reduction=<span>'pca'</span>, intp=<span>T</span>, intp_pnt=<span>5000</span>, intp_lin=<span>F</span>, nPCs=<span>30</span>, ntree=<span>1000</span>, <br>                                     dist_thresh=<span>0.55</span>, top_spot=<span>5</span>, spot_n=<span>5</span>, repel_r=<span>20</span>, repel_iter=<span>20</span>, keep_model=<span>T</span>)$celltrek<br><br><span># 在细胞映射完成后，我们可以使用 celltrek_vis 交互式可视化 CellTrek 的结果</span><br>brain_celltrek$cell_type &lt;- factor(brain_celltrek$cell_type, <br>                                   levels=sort(unique(brain_celltrek$cell_type)))<br><br>CellTrek::celltrek_vis(brain_celltrek@meta.data %&gt;% <br>                         dplyr::select(coord_x, coord_y, cell_type:id_new),<br>                       brain_celltrek@images$anterior1@image, <br>                       brain_celltrek@images$anterior1@scale.factors$lowres)<br></code></pre><h4 data-tool="mdnice编辑器">Step3. CellTrek细胞共定位分析</h4><pre data-tool="mdnice编辑器"><span></span><code><span>#### 3. 细胞共定位分析</span><br><span># 基于 CellTrek 的结果，我们可以使用 SColoc 总结不同细胞类型之间的共定位模式。</span><br><span># 在这里，我们以谷氨酸能神经元细胞类型作为示例（建议删除一些细胞类型，例如，n&lt;20，细胞数量非常少的细胞类型）。</span><br><span># 我们首先从我们的映射结果中子集化谷氨酸能神经元细胞类型。</span><br>glut_cell &lt;- c(<span>'L2/3 IT'</span>, <span>'L4'</span>, <span>'L5 IT'</span>, <span>'L5 PT'</span>, <span>'NP'</span>, <span>'L6 IT'</span>, <span>'L6 CT'</span>,  <span>'L6b'</span>)<br>names(glut_cell) &lt;- make.names(glut_cell)<br>brain_celltrek_glut &lt;- subset(brain_celltrek, subset=cell_type %<span>in</span>% glut_cell)<br><br><span>#然后使用scoloc进行共定位分析：</span><br>brain_celltrek_glut$cell_type &lt;- factor(brain_celltrek_glut$cell_type,<br>                                        levels=glut_cell)<br><br><span>## 我们从图中提取最小生成树（MST）的结果。</span><br>brain_sgraph_KL_mst_cons &lt;- brain_sgraph_KL$mst_cons<br>rownames(brain_sgraph_KL_mst_cons) &lt;- colnames(brain_sgraph_KL_mst_cons) &lt;- glut_cell[colnames(brain_sgraph_KL_mst_cons)]<br><br><span>## 然后，我们提取meta.data数据（包括细胞类型及其频率信息）。</span><br>brain_cell_class &lt;- brain_celltrek@meta.data %&gt;% dplyr::select(id=cell_type) %&gt;% unique<br>brain_celltrek_count &lt;- data.frame(freq = table(brain_celltrek$cell_type))<br>brain_cell_class_new &lt;- merge(brain_cell_class, brain_celltrek_count, by.x =<span>"id"</span>, by.y = <span>"freq.Var1"</span>)<br><br>brain_sgraph_KL &lt;- CellTrek::scoloc(brain_celltrek_glut, <br>                                    col_cell=<span>'cell_type'</span>,<br>                                    use_method=<span>'KL'</span>, eps=<span>1e-50</span>)<br>CellTrek::scoloc_vis(brain_sgraph_KL_mst_cons,<br>                     meta_data=brain_cell_class_new )<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7940414507772021" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/iaRJcrq2Los9fJy5icvttNBiaoVLticicSO6aFycB8ciczJoFYs8C8KCHBjQicZ2ccpNwpmSxBtDUc5B6hhRtFzb1YMew/640?wx_fmt=other" data-type="other" data-w="772" src="https://mmbiz.qpic.cn/mmbiz_jpg/iaRJcrq2Los9fJy5icvttNBiaoVLticicSO6aFycB8ciczJoFYs8C8KCHBjQicZ2ccpNwpmSxBtDUc5B6hhRtFzb1YMew/640?wx_fmt=other"><figcaption>image-20231105145004137</figcaption></figure><h4 data-tool="mdnice编辑器">Step4. 目标细胞类型的空间加权共表达分析</h4><pre data-tool="mdnice编辑器"><span></span><code><span># 基于CellTrek的结果，我们可以使用SCoexp模块进一步研究感兴趣的细胞类型中的共表达模式。</span><br><span># 在这里，我们将使用共识聚类(CC)方法以L5 IT单元为例。首先从图表结果中提取L5 IT单元格。</span><br>brain_celltrek_l5 &lt;- subset(brain_celltrek, subset=cell_type==<span>'L5 IT'</span>)<br>brain_celltrek_l5@assays$RNA@scale.data &lt;- matrix(<span>NA</span>, <span>1</span>, <span>1</span>)<br>brain_celltrek_l5$cluster &lt;- gsub(<span>'L5 IT VISp '</span>, <span>''</span>, brain_celltrek_l5$cluster)<br>DimPlot(brain_celltrek_l5, group.by = <span>'cluster'</span>)<br><span># 我们选择前2000个可变基因(不包括线粒体、核糖体和高零基因)</span><br>brain_celltrek_l5 &lt;- FindVariableFeatures(brain_celltrek_l5)<br>vst_df &lt;- brain_celltrek_l5@assays$RNA@meta.features %&gt;% data.frame %&gt;% mutate(id=rownames(.))<br>nz_test &lt;- apply(as.matrix(brain_celltrek_l5[[<span>'RNA'</span>]]@data), <span>1</span>, <span>function</span>(x) mean(x!=<span>0</span>)*<span>100</span>)<br>hz_gene &lt;- names(nz_test)[nz_test&lt;<span>20</span>]<br>mt_gene &lt;- grep(<span>'^Mt-'</span>, rownames(brain_celltrek_l5), value=<span>T</span>)<br>rp_gene &lt;- grep(<span>'^Rpl|^Rps'</span>, rownames(brain_celltrek_l5), value=<span>T</span>)<br>vst_df &lt;- vst_df %&gt;% dplyr::filter(!(id %<span>in</span>% c(mt_gene, rp_gene, hz_gene))) %&gt;% arrange(., -vst.variance.standardized)<br>feature_temp &lt;- vst_df$id[<span>1</span>:<span>2000</span>]<br><span># 我们使用 scoexp 进行空间加权基因共表达分析。</span><br>brain_celltrek_l5_scoexp_res_cc &lt;- CellTrek::scoexp(celltrek_inp=brain_celltrek_l5, <br>                                                    assay=<span>'RNA'</span>, <br>                                                    approach=<span>'cc'</span>, <br>                                                    gene_select = feature_temp, <br>                                                    sigm=<span>140</span>, <br>                                                    avg_cor_min=<span>.4</span>, <br>                                                    zero_cutoff=<span>3</span>, <br>                                                    min_gen=<span>40</span>, max_gen=<span>400</span>)<br>                 <br><span># 我们可以使用热图可视化共表达式模块。</span><br>brain_celltrek_l5_k &lt;- rbind(data.frame(gene=c(brain_celltrek_l5_scoexp_res_cc$gs[[<span>1</span>]]), G=<span>'K1'</span>), <br>                             data.frame(gene=c(brain_celltrek_l5_scoexp_res_cc$gs[[<span>2</span>]]), G=<span>'K2'</span>)) %&gt;% <br>  magrittr::set_rownames(.$gene) %&gt;% dplyr::select(-<span>1</span>)<br>pheatmap::pheatmap(brain_celltrek_l5_scoexp_res_cc$wcor[rownames(brain_celltrek_l5_k), rownames(brain_celltrek_l5_k)], <br>                   clustering_method=<span>'ward.D2'</span>, annotation_row=brain_celltrek_l5_k, show_rownames=<span>F</span>, show_colnames=<span>F</span>, <br>                   treeheight_row=<span>10</span>, treeheight_col=<span>10</span>, annotation_legend = <span>T</span>, fontsize=<span>8</span>,<br>                   color=viridis(<span>10</span>), main=<span>'L5 IT spatial co-expression'</span>)                 <br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7303370786516854" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/iaRJcrq2Los9fJy5icvttNBiaoVLticicSO6a6PEbsRK0M9FCcWo7l6iaHCMicwg8TOygcE0Aqca5aQYyUFmu9va06OOA/640?wx_fmt=other" data-type="other" data-w="445" src="https://mmbiz.qpic.cn/mmbiz_jpg/iaRJcrq2Los9fJy5icvttNBiaoVLticicSO6a6PEbsRK0M9FCcWo7l6iaHCMicwg8TOygcE0Aqca5aQYyUFmu9va06OOA/640?wx_fmt=other"><figcaption>image-20231105151101908</figcaption></figure><pre data-tool="mdnice编辑器"><span></span><code><span># 我们确定了3个不同的基因模块。基于我们识别的共表达模块，可以计算模块的分数。</span><br>brain_celltrek_l5 &lt;- AddModuleScore(brain_celltrek_l5, <br>                                    features=brain_celltrek_l5_scoexp_res_cc$gs, <br>                                    name=<span>'CC_'</span>, nbin=<span>10</span>, ctrl=<span>50</span>, seed=<span>42</span>)<br><span>## 可视化1</span><br>FeaturePlot(brain_celltrek_l5, <br>            grep(<span>'CC_'</span>, colnames(brain_celltrek_l5@meta.data), <br>                 value=<span>T</span>), ncol = <span>1</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.2888675623800384" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/iaRJcrq2Los9fJy5icvttNBiaoVLticicSO6aHDLpM8nGGIEXskTE0Edibq3TynicylvmKO1g4ATvMGicDUVCKbyuUQ8mw/640?wx_fmt=other" data-type="other" data-w="1042" src="https://mmbiz.qpic.cn/mmbiz_jpg/iaRJcrq2Los9fJy5icvttNBiaoVLticicSO6aHDLpM8nGGIEXskTE0Edibq3TynicylvmKO1g4ATvMGicDUVCKbyuUQ8mw/640?wx_fmt=other"><figcaption>image-20231105151156465</figcaption></figure><pre data-tool="mdnice编辑器"><span></span><code><span>## 可视化2</span><br>SpatialFeaturePlot(brain_celltrek_l5, <br>                   grep(<span>'CC_'</span>, colnames(brain_celltrek_l5@meta.data), value=<span>T</span>))<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.2975557917109458" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/iaRJcrq2Los9fJy5icvttNBiaoVLticicSO6a89oE9mfia4vuBOzcR4GZGuZBBxcWJU6OLkB9PEbDSkqKibZAdQ7u3MSg/640?wx_fmt=other" data-type="other" data-w="941" src="https://mmbiz.qpic.cn/mmbiz_jpg/iaRJcrq2Los9fJy5icvttNBiaoVLticicSO6a89oE9mfia4vuBOzcR4GZGuZBBxcWJU6OLkB9PEbDSkqKibZAdQ7u3MSg/640?wx_fmt=other"><figcaption>image-20231105151243508</figcaption></figure><h3 data-tool="mdnice编辑器">三. 彩蛋</h3><p data-tool="mdnice编辑器">鉴于有些读者抨击我们只会“抄英文教程”，“照搬英文帖”，这里我基于对CellTrek的理解复现下图E<strong>（这是官方教程里没有的）。</strong></p><figure data-tool="mdnice编辑器"><img data-ratio="0.2553495007132668" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/iaRJcrq2Los9fJy5icvttNBiaoVLticicSO6aN7oS6eDd7QdCPNUPrnvcXAico3Hw4WSQEXu9HZ6JdfpTUrKhibSts7cg/640?wx_fmt=other" data-type="other" data-w="701" src="https://mmbiz.qpic.cn/mmbiz_jpg/iaRJcrq2Los9fJy5icvttNBiaoVLticicSO6aN7oS6eDd7QdCPNUPrnvcXAico3Hw4WSQEXu9HZ6JdfpTUrKhibSts7cg/640?wx_fmt=other"><figcaption>image-20231105151822596</figcaption></figure><p data-tool="mdnice编辑器">图E计算了目标细胞类型（L2/3T）与其余细胞类型在空间上的距离，结果表明L4细胞亚群在空间上距离L2/3T最近（见图C-E，图D在上文复现过了）：</p><p data-tool="mdnice编辑器">第一步，获取细胞类型和空间坐标信息：</p><pre data-tool="mdnice编辑器"><span></span><code><span># coord_x和coord_y是空间坐标：</span><br>inp_df &lt;- brain_celltrek_glut@meta.data %&gt;% dplyr::select(cell_names = dplyr::one_of(<span>'cell_type'</span>), <br>                                                          coord_x, coord_y)<br>inp_df$coord_x = <span>270</span>-inp_df$coord_x<br>head(inp_df)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.30980392156862746" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/iaRJcrq2Los9fJy5icvttNBiaoVLticicSO6aYQZ9N1pnuaLV8eFGib9UFz4V25M0tIiaYZSnN1Rh6QF9tT9sxsTwyDMA/640?wx_fmt=other" data-type="other" data-w="510" src="https://mmbiz.qpic.cn/mmbiz_jpg/iaRJcrq2Los9fJy5icvttNBiaoVLticicSO6aYQZ9N1pnuaLV8eFGib9UFz4V25M0tIiaYZSnN1Rh6QF9tT9sxsTwyDMA/640?wx_fmt=other"><figcaption>image-20231105164435672</figcaption></figure><blockquote data-tool="mdnice编辑器"><p>感兴趣的小伙伴可以理解一下为啥需要“270-inp_df$coord_x”。</p></blockquote><p data-tool="mdnice编辑器">第二步，计算目标细胞与其他细胞的聚类：</p><pre data-tool="mdnice编辑器"><span></span><code>output &lt;- kdist(inp_df = inp_df, <br>                ref = <span>"L2/3 IT"</span>, <span>#目标细胞类型</span><br>                ref_type = <span>'all'</span>, <br>                que = glut_cell,  <span>#其余的细胞</span><br>                k = <span>10</span>, <br>                new_name = <span>"L23ITvs.Others"</span>,<br>                keep_nn = <span>F</span>)<br>head(output$kdist_df) <span>#最后的结果</span><br><span>#                         L23ITvs.Others</span><br><span># F2S4_160422_009_G01           59.72424</span><br><span># F2S4_160422_011_G01           62.78655</span><br><span># F2S4_160428_007_D01           58.06266</span><br><span># F2S4_160516_020_D01           53.70867</span><br><span># F2S4_180110_072_C01          376.52573</span><br><span># F2S4_161129_006_F01          757.52189</span><br>res = output$kdist_df<br>res$barcode = row.names(res)<br>inp_df$barcode = row.names(inp_df)<br>res = left_join(res, inp_df)<br>head(res)<br><span>#   L23ITvs.Others             barcode cell_names    coord_x  coord_y</span><br><span># 1           59.72424 F2S4_160422_009_G01    L2/3 IT  -861.4217 5370.832</span><br><span># 2           62.78655 F2S4_160422_011_G01    L2/3 IT  -903.3036 5301.647</span><br><span># 3           58.06266 F2S4_160428_007_D01    L2/3 IT  -837.6159 5328.688</span><br><span># 4           53.70867 F2S4_160516_020_D01    L2/3 IT  -877.2844 5331.978</span><br><span># 5          376.52573 F2S4_180110_072_C01      L5 IT -2085.6755 3122.758</span><br><span># 6          757.52189 F2S4_161129_006_F01    L2/3 IT -4477.5370 2710.916</span><br></code></pre><p data-tool="mdnice编辑器">第三步，可视化：</p><pre data-tool="mdnice编辑器"><span></span><code><span>library</span>(ggpubr)<br>ggboxplot(data = res, <br>          x = <span>"cell_names"</span>,<br>          y = <span>"L23ITvs.Others"</span>, <br>          fill = <span>"cell_names"</span>, <br>          title = <span>"K-distance to L2/3 IT cells"</span>)+ <br>  stat_compare_means(method = <span>"kruskal.test"</span>) +<br>  theme(plot.title = element_text(color=<span>"black"</span>,hjust = <span>0.5</span>),<br>        axis.text.x = element_text(angle = <span>90</span>, hjust = <span>1</span>,vjust = <span>0.5</span>), <span>#,vjust = 0.5</span><br>        legend.position = <span>"none"</span>) + labs(y = <span>"Distance"</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7340241796200345" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/iaRJcrq2Los9fJy5icvttNBiaoVLticicSO6aIVIibmx2bPxFXgmGw97DYFx8CiaC52YwxGTWG9Artlg4LaBXnhFgw0cA/640?wx_fmt=other" data-type="other" data-w="579" src="https://mmbiz.qpic.cn/mmbiz_jpg/iaRJcrq2Los9fJy5icvttNBiaoVLticicSO6aIVIibmx2bPxFXgmGw97DYFx8CiaC52YwxGTWG9Artlg4LaBXnhFgw0cA/640?wx_fmt=other"><figcaption>image-20231105165306137</figcaption></figure><p data-tool="mdnice编辑器">基于和原文一摸一样。</p><p data-tool="mdnice编辑器">我再额外辩解一下：</p><ul data-tool="mdnice编辑器"><li><section>其实“抄英文教程”，本身对于一小部分小伙伴来说就有一些困难，阅读上的困难其次，主要还是代码理解上；</section></li><li><section>第二个小问题是，有些“抄英文教程”有大量的bug，我们生信技能树的小伙伴在写帖子的时候，肯定会测试一般教程代码，有bug肯定会帮忙解决，然后把解决方案写进教程贴里，<strong>要知道有的bug部分小伙伴花几天甚至几周都没办法解决，所以“抄英文教程”本身也不会是一个简单的事情</strong>；</section></li><li><section>第三个就是，绝大部分的教程，我们都会加入自己的理解和知识体系的整理串联，有不对的地方也很正常，我们非常欢迎友好的探讨和质疑，希望受到别人的质疑来提升自己，本身也是我分享笔记的一个初衷；</section></li><li><section>最后，写帖子真的不挣钱...我们基本都是用爱发电的（说实话，我写帖子一般要写大半天，有的甚至一两天，有这功夫，我都能挣不少钱...）。所以还请珍惜我们技能树的小伙伴...</section></li></ul><span>- END -</span></section><p><br></p><p><mp-style-type data-value="10000"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/Ie9h4uEmJolqRPyDndWNZA",target="_blank" rel="noopener noreferrer">原文链接</a>
