---
title: "斑马鱼单细胞数据运行SCENIC的骚操作"
date: 2023-11-19T02:02:27Z
draft: ["false"]
tags: [
  "fetched",
  "东林的扯淡小屋"
]
categories: ["Acdemic"]
---
斑马鱼单细胞数据运行SCENIC的骚操作 by 东林的扯淡小屋
------
<div><p>将斑马鱼的基因直接改为大写，然后再去分析，有16534个基因，比上一次的1570个人类同源基因多，说不定就可以分析出结果。</p><p><img data-galleryid="" data-imgfileid="100024146" data-ratio="0.7018518518518518" data-s="300,640" data-type="png" data-w="1080" data-src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBqoM1zOHszyGsJ1CaOwCNC3zDrNmF3GeDO3xvJU7zRibvqUVsD5BjHQsfnKbrICeHicKLPcuGYib0TYg/640?wx_fmt=png&amp;from=appmsg" src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBqoM1zOHszyGsJ1CaOwCNC3zDrNmF3GeDO3xvJU7zRibvqUVsD5BjHQsfnKbrICeHicKLPcuGYib0TYg/640?wx_fmt=png&amp;from=appmsg"></p><p><img data-galleryid="" data-imgfileid="100024147" data-ratio="0.8546296296296296" data-s="300,640" data-type="png" data-w="1080" data-src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBqoM1zOHszyGsJ1CaOwCNC35q3dE7TLkQW9TnQFFYYu2PJdJjVfdLl3LSwNlYE9h7RXcQ4iakbYDjg/640?wx_fmt=png&amp;from=appmsg" src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBqoM1zOHszyGsJ1CaOwCNC35q3dE7TLkQW9TnQFFYYu2PJdJjVfdLl3LSwNlYE9h7RXcQ4iakbYDjg/640?wx_fmt=png&amp;from=appmsg">http://www.zfish.cn/upload/file/20190815/ZFIN%E6%96%91%E9%A9%AC%E9%B1%BC%E5%91%BD%E5%90%8D%E8%A7%84%E5%88%99-190814.pdf</p><p><img data-galleryid="" data-imgfileid="100024145" data-ratio="0.4230769230769231" data-s="300,640" data-type="png" data-w="1066" data-src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBqoM1zOHszyGsJ1CaOwCNC3mwOAf2as0WBRS7MdvX5lbbJQ1pLAI70gqqniaCBc4t1jJKXAzE8MFDA/640?wx_fmt=png&amp;from=appmsg" src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBqoM1zOHszyGsJ1CaOwCNC3mwOAf2as0WBRS7MdvX5lbbJQ1pLAI70gqqniaCBc4t1jJKXAzE8MFDA/640?wx_fmt=png&amp;from=appmsg"></p><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="perl"><code><span><br></span></code><code><span><span>#骚操作：</span></span></code><code><span>listEnsembl()</span></code><code><span>ensembl &lt;- useEnsembl(biomart = <span>"genes"</span>)</span></code><code><span>datasets &lt;- listDatasets(ensembl)</span></code><code><span>head(datasets)</span></code><code><span><br></span></code><code><span>searchDatasets(mart = ensembl, pattern = <span>"hsapiens"</span>)</span></code><code><span>searchDatasets(mart = ensembl, pattern = <span>"drerio"</span>)</span></code><code><span><br></span></code><code><span>ensembl &lt;- useDataset(dataset = <span>"drerio_gene_ensembl"</span>, mart = ensembl)</span></code><code><span>mart &lt;- useEnsembl(dataset = <span>"drerio_gene_ensembl"</span>, biomart=<span>'ensembl'</span>)</span></code><code><span>head(keytypes(mart), n = <span>300</span>)</span></code><code><span><br></span></code><code><span>keytype = <span>"with_hsapiens_homolog"</span></span></code><code><span>filters = listFilters(ensembl)</span></code><code><span>filters[<span>1</span>:<span>100</span>,]</span></code><code><span>searchAttributes(mart = ensembl,pattern = <span>'hsapiens'</span>)</span></code><code><span>searchFilters(mart = ensembl, pattern = <span>"zfin"</span>)</span></code><code><span><br></span></code><code><span>getBM(attributes = c(<span>"ensembl_gene_id"</span>,<span>"external_synonym"</span>,<span>"external_gene_name"</span>,<span>"hgnc_symbol"</span>),</span></code><code><span>      filters = <span>"zfin_id_symbol"</span>,<span>values</span> = c(<span>"mki67"</span>),</span></code><code><span>      mart = ensembl)</span></code><code><span><br></span></code><code><span>getBM(attributes = c(<span>"ensembl_gene_id"</span>,<span>"external_synonym"</span>,<span>"external_gene_name"</span>,<span>"hgnc_symbol"</span>),</span></code><code><span>      filters = <span>"zfin_id_symbol"</span>,<span>values</span> =c(<span>"p2ry12"</span>,<span>"hexb"</span>,<span>"bzw2"</span>,<span>"g0s2"</span>,<span>"csf1ra"</span>,<span>"ptprc"</span>,<span>"emilin2a"</span>,<span>"emilin2b"</span>,<span>"gda"</span>,<span>"hp"</span>),</span></code><code><span>      mart = ensembl)</span></code><code><span><br></span></code><code><span>getBM(attributes = c(<span>"ensembl_gene_id"</span>,<span>"external_synonym"</span>,<span>"external_gene_name"</span>,<span>"hgnc_symbol"</span>),</span></code><code><span>      filters = <span>"zfin_id_symbol"</span>,<span>values</span> = c(<span>"tp53"</span>),</span></code><code><span>      mart = ensembl)</span></code><code><span><br></span></code><code><span>getBM(attributes = c(<span>"ensembl_gene_id"</span>,<span>"external_synonym"</span>,<span>"external_gene_name"</span>,<span>"hgnc_symbol"</span>),</span></code><code><span>      filters = <span>"zfin_id_symbol"</span>,<span>values</span> =c(<span>"tp53a"</span> , <span>"brca1b"</span>,  <span>"egfra"</span> ),</span></code><code><span>      mart = ensembl)</span></code><code><span><br></span></code><code><span>gene_names &lt;- read.csv(<span>"/data1/yudonglin/zebrafish/genename_zebrafish.csv"</span>)</span></code><code><span><br></span></code><code><span><span># Split and extract the first part for each gene name</span></span></code><code><span>gene_names_first_part &lt;- sapply(strsplit(gene_names$x, <span>"|"</span>, fixed = TRUE), function(<span>x</span>) <span>x</span>[<span>2</span>])</span></code><code><span><br></span></code><code><span><span># Print the result</span></span></code><code><span><span>print</span>(gene_names_first_part)</span></code><code><span><br></span></code><code><span>gene_human_zebrafish&lt;-getBM(attributes = c(<span>"ensembl_gene_id"</span>,<span>"external_synonym"</span>,<span>"external_gene_name"</span>,<span>"hgnc_symbol"</span>),</span></code><code><span>                            filters = <span>"zfin_id_symbol"</span>,<span>values</span> = gene_names_first_part,</span></code><code><span>                            mart = ensembl)</span></code><code><span><br></span></code><code><span>gene_human_zebrafish&lt;-getBM(attributes = c(<span>"ensembl_gene_id"</span>,<span>"external_synonym"</span>,<span>"external_gene_name"</span>,<span>"hgnc_symbol"</span>),</span></code><code><span>                            filters = <span>"zfin_id_symbol"</span>,<span>values</span> = gene_names_first_part,</span></code><code><span>                            mart = human)</span></code><code><span><br></span></code><code><span>head(gene_human_zebrafish)</span></code><code><span>dim(gene_human_zebrafish)</span></code><code><span>gene_human_zebrafish$external_gene_name&lt;-toupper(gene_human_zebrafish$external_gene_name)</span></code><code><span><span>#https://github.com/lcdorman/zebrafish2021/blob/main/6pseudotime_humanfetalcomparison.Rmd</span></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><span>#load in the motifannotation this will load it into your environment but as the name in which is given to the list argument</span></span></code><code><span>data(list=<span>"motifAnnotations_hgnc_v9"</span>, <span>package</span>=<span>"RcisTarget"</span>)</span></code><code><span><br></span></code><code><span><span>#rename the motif annnotion by attributing it to the variable that is in the error</span></span></code><code><span>motifAnnotations_hgnc &lt;- motifAnnotations_hgnc_v9</span></code><code><span><br></span></code><code><span>set.seed(<span>123</span>)</span></code><code><span>library(Seurat)</span></code><code><span>library(tidyverse)</span></code><code><span>library(patchwork)</span></code><code><span>library(SCENIC)</span></code><code><span>library(Seurat)</span></code><code><span>library(tidyverse)</span></code><code><span>library(patchwork)</span></code><code><span>library(SCENIC)</span></code><code><span>library(plyr)</span></code><code><span>library(permute)</span></code><code><span>library(data.table)</span></code><code><span>library(SCopeLoomR)</span></code><code><span>library(<span>"doSNOW"</span>)</span></code><code><span>library(<span>"doParallel"</span>) </span></code><code><span>library(<span>"doMPI"</span>)</span></code><code><span><span>##==分析准备==##</span></span></code><code><span>dir.create(<span>"SCENIC"</span>)</span></code><code><span>dir.create(<span>"SCENIC/int"</span>)</span></code><code><span><span>#scRNA &lt;-sce.all</span></span></code><code><span>YDL &lt;-readRDS(<span>"zebrafish.RDS"</span>)</span></code><code><span><span>#rm(YDL)</span></span></code><code><span>setwd(<span>"./SCENIC"</span>) </span></code><code><span><span>##准备细胞meta信息</span></span></code><code><span>cellInfo &lt;- data.frame(YDL@meta.data)</span></code><code><span>head((YDL@meta.data))</span></code><code><span>colnames(cellInfo)[which(colnames(cellInfo)==<span>"Sample"</span>)] &lt;- <span>"Sample"</span></span></code><code><span>colnames(cellInfo)[which(colnames(cellInfo)==<span>"CellType"</span>)] &lt;- <span>"CellType"</span></span></code><code><span>cellInfo &lt;- cellInfo[,c(<span>"Sample"</span>,<span>"CellType"</span>)]</span></code><code><span>saveRDS(cellInfo, file=<span>"int/cellInfo.Rds"</span>)</span></code><code><span><span>#为了节省计算资源，随机抽取1000个细胞的数据子集</span></span></code><code><span>subcell &lt;- sample(colnames(YDL),<span>1000</span>)</span></code><code><span>YDLsub &lt;- YDL[,subcell]</span></code><code><span>saveRDS(YDLsub, <span>"YDLsub.rds"</span>)</span></code><code><span>exprMat &lt;- as.matrix(YDLsub@assays$RNA@counts)</span></code><code><span><br></span></code><code><span><span># Split and extract the first part for each gene name</span></span></code><code><span>rownames(exprMat) &lt;- sapply(strsplit(rownames(exprMat), <span>"-"</span>, fixed = TRUE), function(<span>x</span>) <span>x</span>[<span>1</span>])</span></code><code><span>exprMat[<span>1</span>:<span>4</span>,<span>1</span>:<span>4</span>]</span></code><code><span>exprMat&lt;-as.data.frame(exprMat)</span></code><code><span>class(exprMat)</span></code><code><span>gene_name&lt;-gene_human_zebrafish</span></code><code><span>dim(gene_name)</span></code><code><span>head(gene_name)</span></code><code><span>gene_name&lt;-gene_name[,c(<span>1</span>,<span>3</span>)]</span></code><code><span>unique_gene_name &lt;- gene_name[!duplicated(gene_name$external_gene_name), ]</span></code><code><span>dim(unique_gene_name)</span></code><code><span><br></span></code><code><span><span># Convert row names to a column in exprMat</span></span></code><code><span>exprMat$ensembl_gene_id &lt;- rownames(exprMat)</span></code><code><span><span># Merge the data frames based on ensembl_gene_id</span></span></code><code><span>merged_data &lt;- merge(exprMat, gene_name, by=<span>"ensembl_gene_id"</span>)</span></code><code><span>merged_data[<span>1</span>:<span>4</span>,<span>1</span>:<span>4</span>]</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># Identify rows in exprMat that have matching ensembl_gene_id in gene_name</span></span></code><code><span>matching_rows &lt;- exprMat[exprMat$ensembl_gene_id %in% unique_gene_name$ensembl_gene_id, ]</span></code><code><span>matching_rows[<span>1</span>:<span>4</span>,<span>1</span>:<span>4</span>]</span></code><code><span><span># selected_data &lt;- matching_rows[1:4, ncol(matching_rows):(ncol(matching_rows)-3)]</span></span></code><code><span><span># # Print the result</span></span></code><code><span><span># print(selected_data)</span></span></code><code><span><span># Set hgnc_symbol as row names</span></span></code><code><span>rownames(matching_rows) &lt;- unique_gene_name$external_gene_name[match(matching_rows$ensembl_gene_id, unique_gene_name$ensembl_gene_id)]</span></code><code><span>matching_rows &lt;- matching_rows[, -ncol(matching_rows)]</span></code><code><span>exprMat&lt;-matching_rows</span></code><code><span>exprMat&lt;-as.matrix(exprMat)</span></code><code><span><span># dim(exprMat)</span></span></code><code><span><span># [1] 32285  1000</span></span></code><code><span><span>#head(cellInfo)</span></span></code><code><span><span>##设置分析环境</span></span></code><code><span>mydbDIR &lt;- <span>"/data/yudonglin/reference/scenic/"</span></span></code><code><span>mydbs &lt;- c(<span>"hg19-500bp-upstream-7species.mc9nr.feather"</span>,</span></code><code><span>           <span>"hg19-tss-centered-10kb-7species.mc9nr.feather"</span>)</span></code><code><span><br></span></code><code><span>names(mydbs) &lt;- c(<span>"500bp"</span>, <span>"10kb"</span>)</span></code><code><span>scenicOptions &lt;- initializeScenic(org=<span>"hgnc"</span>, </span></code><code><span>                                  nCores=<span>60</span>,</span></code><code><span>                                  dbDir=mydbDIR, </span></code><code><span>                                  dbs = mydbs,</span></code><code><span>                                  datasetTitle = <span>"SCENIC"</span>)</span></code><code><span>saveRDS(scenicOptions, <span>"int/scenicOptions.rds"</span>)</span></code><code><span>genesKept &lt;- geneFiltering(exprMat, scenicOptions, </span></code><code><span>                           minCountsPerGene = <span>3</span> * <span>0</span>.<span>01</span> * ncol(exprMat), </span></code><code><span>                           minSamples = ncol(exprMat) * <span>0</span>.<span>01</span>)</span></code><code><span>exprMat_filtered &lt;- exprMat[genesKept, ]</span></code><code><span><span>##计算相关性矩阵</span></span></code><code><span>runCorrelation(exprMat_filtered, scenicOptions)</span></code><code><span><span>##TF-Targets相关性回归分析</span></span></code><code><span>exprMat_filtered_log &lt;- log2(exprMat_filtered+<span>1</span>)</span></code><code><span>runGenie3(exprMat_filtered_log, scenicOptions, nParts = <span>20</span>)</span></code><code><span><span>#这一步消耗的计算资源非常大，个人电脑需要几个小时的运行时间</span></span></code><code><span><span>##推断共表达模块</span></span></code><code><span>runSCENIC_1_coexNetwork2modules(scenicOptions)</span></code><code><span><span># scenicOptions &lt;- initializeScenic(org="hgnc", </span></span></code><code><span><span>#                                   nCores=9,</span></span></code><code><span><span>#                                   dbDir=mydbDIR, </span></span></code><code><span><span>#                                   dbs = mydbs,</span></span></code><code><span><span>#                                   datasetTitle = "erythrogenesis")</span></span></code><code><span><span>##推断转录调控网络（regulon）</span></span></code><code><span>runSCENIC_2_createRegulons(scenicOptions)</span></code><code><span><span>#以上代码可增加参数coexMethod=c("w001", "w005", "top50", "top5perTarget", "top10perTarget", "top50perTarget"))</span></span></code><code><span><span>#默认6种方法的共表达网络都计算，可以少选几种方法以减少计算量</span></span></code><code><span>scenicOptions &lt;- initializeScenic(org=<span>"hgnc"</span>, </span></code><code><span>                                  nCores=<span>1</span>,</span></code><code><span>                                  dbDir=mydbDIR, </span></code><code><span>                                  dbs = mydbs,</span></code><code><span>                                  datasetTitle = <span>"erythrogenesis"</span>)</span></code><code><span><span>##==regulon活性评分与可视化==##</span></span></code><code><span><span>##regulons计算AUC值并进行下游分析</span></span></code><code><span>exprMat_all &lt;- as.matrix(YDL@assays$RNA@counts)</span></code><code><span>exprMat_all &lt;- log2(exprMat_all+<span>1</span>)</span></code><code><span><br></span></code><code><span><span># Split and extract the first part for each gene name</span></span></code><code><span>rownames(exprMat_all) &lt;- sapply(strsplit(rownames(exprMat_all), <span>"-"</span>, fixed = TRUE), function(<span>x</span>) <span>x</span>[<span>1</span>])</span></code><code><span>exprMat_all[<span>1</span>:<span>4</span>,<span>1</span>:<span>4</span>]</span></code><code><span>exprMat_all&lt;-as.data.frame(exprMat_all)</span></code><code><span>class(exprMat_all)</span></code><code><span>gene_name&lt;-gene_human_zebrafish</span></code><code><span>dim(gene_name)</span></code><code><span>head(gene_name)</span></code><code><span>gene_name&lt;-gene_name[,c(<span>1</span>,<span>3</span>)]</span></code><code><span>unique_gene_name &lt;- gene_name[!duplicated(gene_name$external_gene_name), ]</span></code><code><span>dim(unique_gene_name)</span></code><code><span><br></span></code><code><span><span># Convert row names to a column in exprMat_all</span></span></code><code><span>exprMat_all$ensembl_gene_id &lt;- rownames(exprMat_all)</span></code><code><span><span># Merge the data frames based on ensembl_gene_id</span></span></code><code><span>merged_data &lt;- merge(exprMat_all, gene_name, by=<span>"ensembl_gene_id"</span>)</span></code><code><span>merged_data[<span>1</span>:<span>4</span>,<span>1</span>:<span>4</span>]</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># Identify rows in exprMat_all that have matching ensembl_gene_id in gene_name</span></span></code><code><span>matching_rows &lt;- exprMat_all[exprMat_all$ensembl_gene_id %in% unique_gene_name$ensembl_gene_id, ]</span></code><code><span>matching_rows[<span>1</span>:<span>4</span>,<span>1</span>:<span>4</span>]</span></code><code><span><span># selected_data &lt;- matching_rows[1:4, ncol(matching_rows):(ncol(matching_rows)-3)]</span></span></code><code><span><span># # Print the result</span></span></code><code><span><span># print(selected_data)</span></span></code><code><span><span># Set hgnc_symbol as row names</span></span></code><code><span>rownames(matching_rows) &lt;- unique_gene_name$external_gene_name[match(matching_rows$ensembl_gene_id, unique_gene_name$ensembl_gene_id)]</span></code><code><span>matching_rows &lt;- matching_rows[, -ncol(matching_rows)]</span></code><code><span>exprMat_all&lt;-matching_rows</span></code><code><span>exprMat_all&lt;-as.matrix(exprMat_all)</span></code><code><span>saveRDS(exprMat_all,<span>"exprMat_all.rds"</span>)</span></code><code><span><span># rm(list=ls())</span></span></code><code><span><span># scenicOptions&lt;-readRDS("scenicOptions.rds")</span></span></code><code><span><span># exprMat_all&lt;-readRDS("exprMat_all.rds")</span></span></code><code><span>runSCENIC_3_scoreCells(scenicOptions, exprMat=exprMat_all)</span></code><code><span><br></span></code><code><span><span>#runSCENIC_3_scoreCells(scenicOptions, exprMat=log2(as.matrix(YDL@assays$RNA@counts)+1))</span></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># #使用shiny互动调整阈值</span></span></code><code><span><span># aucellApp &lt;- plotTsne_AUCellApp(scenicOptions, exprMat_all)</span></span></code><code><span><span># savedSelections &lt;- shiny::runApp(aucellApp)</span></span></code><code><span><span># #保存调整后的阈值</span></span></code><code><span><span># newThresholds &lt;- savedSelections$thresholds</span></span></code><code><span><span># scenicOptions@fileNames$int["aucell_thresholds",1] &lt;- "int/newThresholds.Rds"</span></span></code><code><span><span># saveRDS(newThresholds, file=getIntName(scenicOptions, "aucell_thresholds"))</span></span></code><code><span><span># saveRDS(scenicOptions, file="int/scenicOptions.Rds")</span></span></code><code><span>runSCENIC_4_aucell_binarize(scenicOptions, exprMat=exprMat_all)</span></code><code><span>savehistory(<span>"scenic_code.txt"</span>)</span></code></pre></section><p><img data-galleryid="" data-imgfileid="100024155" data-ratio="0.6947637292464879" data-s="300,640" data-type="png" data-w="783" data-src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBqoM1zOHszyGsJ1CaOwCNC3uF1cBHia5ib0pkQdgvM4YapjSWQaP0JCzGicJXuiblO3iafrplvXXSdiavVA/640?wx_fmt=png&amp;from=appmsg" src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBqoM1zOHszyGsJ1CaOwCNC3uF1cBHia5ib0pkQdgvM4YapjSWQaP0JCzGicJXuiblO3iafrplvXXSdiavVA/640?wx_fmt=png&amp;from=appmsg"></p><p><img data-galleryid="" data-imgfileid="100024160" data-ratio="0.9653767820773931" data-s="300,640" data-type="png" data-w="982" data-src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBqoM1zOHszyGsJ1CaOwCNC3V9UrFhpZjJ6Mpv8JUkicF4ueDR58Iu49sqTEruYz6BKvvum98FmWg4g/640?wx_fmt=png&amp;from=appmsg" src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBqoM1zOHszyGsJ1CaOwCNC3V9UrFhpZjJ6Mpv8JUkicF4ueDR58Iu49sqTEruYz6BKvvum98FmWg4g/640?wx_fmt=png&amp;from=appmsg"></p><p>有432个转录因子，看起来很有希望。</p><p><img data-galleryid="" data-imgfileid="100024163" data-ratio="0.4753488372093023" data-s="300,640" data-type="png" data-w="1075" data-src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBqoM1zOHszyGsJ1CaOwCNC3jWqBjgcaYAq1GjHxQe76JWWSK2gyuib8RLvibsbdU3ShRthgVOibQ0m2w/640?wx_fmt=png&amp;from=appmsg" src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBqoM1zOHszyGsJ1CaOwCNC3jWqBjgcaYAq1GjHxQe76JWWSK2gyuib8RLvibsbdU3ShRthgVOibQ0m2w/640?wx_fmt=png&amp;from=appmsg"></p><p>231个regulon，这个分析结果应该是可以用的。<br></p><p><img data-galleryid="" data-imgfileid="100024171" data-ratio="0.14177215189873418" data-s="300,640" data-type="png" data-w="790" data-src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBqoM1zOHszyGsJ1CaOwCNC3cmhg0BwHN4RWE28LG3l7ZfTdqGgvg77bXKsK8rbqohLT62HE7nVicqA/640?wx_fmt=png&amp;from=appmsg" src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBqoM1zOHszyGsJ1CaOwCNC3cmhg0BwHN4RWE28LG3l7ZfTdqGgvg77bXKsK8rbqohLT62HE7nVicqA/640?wx_fmt=png&amp;from=appmsg"></p><p><img data-galleryid="" data-imgfileid="100024161" data-ratio="0.6592592592592592" data-s="300,640" data-type="png" data-w="1080" data-src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBqoM1zOHszyGsJ1CaOwCNC3QeHqX9LrQ4jh72DKd7IBiaN9IzicK3UGxDu7ibJleftrNPK5EPWYn1lbg/640?wx_fmt=png&amp;from=appmsg" src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBqoM1zOHszyGsJ1CaOwCNC3QeHqX9LrQ4jh72DKd7IBiaN9IzicK3UGxDu7ibJleftrNPK5EPWYn1lbg/640?wx_fmt=png&amp;from=appmsg"></p><p><img data-galleryid="" data-imgfileid="100024166" data-ratio="1.112950340798442" data-s="300,640" data-type="png" data-w="1027" data-src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBqoM1zOHszyGsJ1CaOwCNC3bw6ibibK6icYA9KGvWMlJXlnTib18RicEBbB9viap8PwI2BMdBSPefYpBFUw/640?wx_fmt=png&amp;from=appmsg" src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBqoM1zOHszyGsJ1CaOwCNC3bw6ibibK6icYA9KGvWMlJXlnTib18RicEBbB9viap8PwI2BMdBSPefYpBFUw/640?wx_fmt=png&amp;from=appmsg"></p><p><img data-galleryid="" data-imgfileid="100024164" data-ratio="1" data-s="300,640" data-type="png" data-w="1003" data-src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBqoM1zOHszyGsJ1CaOwCNC3DTUBg7h3aWQjsTKtyw1XuicuIH0ZAVb8eQQ9ErYgQXqI4cvYianNgSibw/640?wx_fmt=png&amp;from=appmsg" src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBqoM1zOHszyGsJ1CaOwCNC3DTUBg7h3aWQjsTKtyw1XuicuIH0ZAVb8eQQ9ErYgQXqI4cvYianNgSibw/640?wx_fmt=png&amp;from=appmsg"></p><p><img data-galleryid="" data-imgfileid="100024165" data-ratio="0.175" data-s="300,640" data-type="png" data-w="1080" data-src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBqoM1zOHszyGsJ1CaOwCNC3icddDSTgeIiaKYmClbjfH9SpU5ezshgqlQDFibDQVZS7539KKXiaKF1hrw/640?wx_fmt=png&amp;from=appmsg" src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBqoM1zOHszyGsJ1CaOwCNC3icddDSTgeIiaKYmClbjfH9SpU5ezshgqlQDFibDQVZS7539KKXiaKF1hrw/640?wx_fmt=png&amp;from=appmsg"></p><p>ATF3_extended (2458g)GABPA_extended (3027g)GTF2F1_extended (2812g)MAX_extended (3203g)ATF3 (1572g)RBBP5_extended (2517g)UBTF_extended (2176g)NR3C1_extended (2195g)ETS2_extended (1662g)CEBPB_extended (1087g)EGR1 (1315g)BHLHE40_extended (1531g)EGR1_extended (2223g)CTCF_extended (2365g)EZH2_extended (2637g)GABPA (2460g)RCOR1_extended (2603g)GTF2F1 (1896g)TAF7_extended (2487g)HDAC1_extended (54g) </p><p>这些转录因子被认为是普遍有活性的，可能比较重要。</p><p><img data-galleryid="" data-imgfileid="100024167" data-ratio="1.004004004004004" data-s="300,640" data-type="png" data-w="999" data-src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBqoM1zOHszyGsJ1CaOwCNC3mIKJ5vuho2DibickKCcmLicOB9j1dYmB2xcG3de4PpUcdrZrVVgIqiavAw/640?wx_fmt=png&amp;from=appmsg" src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBqoM1zOHszyGsJ1CaOwCNC3mIKJ5vuho2DibickKCcmLicOB9j1dYmB2xcG3de4PpUcdrZrVVgIqiavAw/640?wx_fmt=png&amp;from=appmsg"></p><p><img data-galleryid="" data-imgfileid="100024170" data-ratio="1.004004004004004" data-s="300,640" data-type="png" data-w="999" data-src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBqoM1zOHszyGsJ1CaOwCNC3yVvjqXELdJibfJaXpWBLL0zLgV4anxyib8TkPo4OfoDOYg6NUGSommFg/640?wx_fmt=png&amp;from=appmsg" src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBqoM1zOHszyGsJ1CaOwCNC3yVvjqXELdJibfJaXpWBLL0zLgV4anxyib8TkPo4OfoDOYg6NUGSommFg/640?wx_fmt=png&amp;from=appmsg"></p><p><img data-galleryid="" data-imgfileid="100024168" data-ratio="0.07222222222222222" data-s="300,640" data-type="png" data-w="1080" data-src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBqoM1zOHszyGsJ1CaOwCNC3gG8TX16dg4XibDPewcvO1icjgFyKMVeDxdMvRUOyYlkSwj8Y2qC6JInA/640?wx_fmt=png&amp;from=appmsg" src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBqoM1zOHszyGsJ1CaOwCNC3gG8TX16dg4XibDPewcvO1icjgFyKMVeDxdMvRUOyYlkSwj8Y2qC6JInA/640?wx_fmt=png&amp;from=appmsg"></p><p>经典的红系调控因子LMO2,TAL1,KLF1。<br></p><p> RBBP5_extended (2517g)RBBP5 (1371g)ZBTB33 (1605g)ZBTB33_extended (1810g)NRF1 (1305g)MXI1_extended (1532g)UBTF_extended (2176g)TAF7_extended (2487g)EZH2_extended (2637g)CTCF_extended (2365g)GTF2F1 (1896g)NFYC_extended (867g)NFYC (647g)POLE3_extended (573g)YBX1_extended (802g)GTF2B_extended (536g)EZH2 (27g)TFDP2 (657g)TFDP2_extended (1093g)MYEF2_extended (142g)HDAC1_extended (54g)CTCF (18g)MAX (13g)MAX_extended (3203g)FOXP4_extended (16g)CRX (57g)VSX1_extended (27g)CEBPG_extended (46g)TCF12_extended (53g)TCF12 (14g)CTNNB1 (12g)E2F7_extended (111g)E2F7 (92g)POLE3 (62g)E2F8_extended (167g)E2F8 (91g)RB1_extended (119g)RB1 (42g)POLE4 (17g)E2F3 (121g)E2F3_extended (196g)POLE4_extended (211g)POU3F1 (99g)POU3F1_extended (102g)GBX1_extended (60g)NEUROG1_extended (18g)MXI1 (16g)SOX3_extended (65g)SOX3 (37g)SOX2_extended (28g)SOX13_extended (21g)SOX13 (11g)TAF7 (15g)ELF1_extended (1511g)ELF1 (1261g)ETS2_extended (1662g)CEBPB_extended (1087g)KLF3_extended (378g)JUND_extended (397g)CEBPA_extended (263g)CEBPA (108g)CEBPD (432g)CEBPD_extended (647g)JUND (98g)FOSB_extended (204g)CEBPB (145g)DDIT3_extended (80g)FOSL2_extended (236g)FOSL2 (170g)FOSB (30g)KLF3 (19g)MAFF_extended (46g)MAFF (14g)CREB3L2 (116g)CREB3L2_extended (227g)XBP1_extended (262g)XBP1 (199g)CREB3L1_extended (493g)CREB3L1 (174g)JUN (39g)JUN_extended (203g)NFIL3_extended (29g)PHF8_extended (973g)PHF8 (418g)NFYA_extended (315g)NFYA (253g)USF1_extended (716g)SREBF2 (847g)ATF1_extended (1721g)SREBF2_extended (2316g)CHD2 (1259g)NR3C1 (1007g)GMEB1_extended (718g)ATF6_extended (193g)TP53_extended (52g)NR3C1_extended (2195g)REST_extended (297g)SP1_extended (561g)SP1 (439g)GMEB1 (364g)MLX_extended (144g)MLX (98g)VAX1_extended (37g)VAX1 (15g)GTF3C2_extended (217g)HMGN3_extended (46g)GBX2 (12g)POU6F1 (10g)FOXP2_extended (12g)GATA3_extended (67g)TLX2_extended (18g)PHOX2A_extended (17g)PHOX2A (12g)SHOX2 (10g)REST (15g)HMGA2 (15g)BARX1_extended (13g)MEOX1_extended (25g)CDX4_extended (12g)MTA3_extended (21g)FOXD2_extended (21g)NKX3−1_extended (11g)FOXD1_extended (13g)EMX2_extended (16g)TBX15_extended (44g)GLI1_extended (14g)FOXJ3 (25g)FOXJ3_extended (53g)BHLHE41_extended (513g)BHLHE41 (111g)HSF2_extended (20g)BHLHE40 (87g)HEY2_extended (19g)GOT1 (36g)IRF2_extended (46g)IRF2 (14g)EGR3_extended (21g)ZBTB7B_extended (26g)IRF6_extended (24g)ZDHHC24 (33g)TP53 (15g)EGR1 (1315g)EGR1_extended (2223g)BHLHE40_extended (1531g)TFAP2C (23g)TFAP2C_extended (30g)TP63_extended (19g)FOXA1 (63g)FOXA1_extended (100g)FOXA3_extended (78g)FOXA3 (40g)NR1H3_extended (27g)ETS2 (782g)ETV4_extended (1297g)ETV4 (1021g)ERF_extended (1195g)ERF (597g)TBP_extended (27g)LMO2_extended (37g)TAL1 (23g)TAL1_extended (121g)KLF17_extended (118g)KLF1_extended (36g)LMO2 (20g)KLF13_extended (17g)MAFA_extended (13g)OSR2_extended (26g)OSR2 (22g)NFKB2_extended (25g)NFKB2 (14g)CEBPG (19g)CRX_extended (103g)THRB_extended (14g)MYB (17g)GSX2_extended (11g)RFX2 (10g)MYOD1 (71g)MYOD1_extended (148g)MEF2D (166g)MEF2D_extended (275g)MYOG_extended (107g)MYOG (80g)FOXD3_extended (33g)RFX4_extended (12g)LEF1 (15g)LBX2_extended (11g)PBX4 (26g)TCF7L2 (16g)SOX2 (15g)LHX9_extended (44g)BARHL2 (23g)POU4F1 (18g)POU4F1_extended (23g)LHX5_extended (59g)BARHL2_extended (69g)SOX14_extended (40g)LHX4_extended (72g)NEUROD1_extended (49g)ONECUT1_extended (16g)SOX12_extended (35g)SOX12 (19g)BHLHE23_extended (40g)SHOX2_extended (50g)LHX5 (12g)SP4_extended (37g)GBX2_extended (52g)GSX1_extended (20g)GSX1 (13g)POU6F1_extended (26g)NEUROD1 (16g)USF2_extended (31g)VSX2_extended (23g)TAF1_extended (2426g)TAF1 (2158g)THAP11_extended (542g)THAP11 (230g)NFE2L3_extended (15g)CEBPZ (24g)MYCN_extended (37g)MYCN (24g) </p><p><img data-galleryid="" data-imgfileid="100024172" data-ratio="1.0289855072463767" data-s="300,640" data-type="png" data-w="1035" data-src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBqoM1zOHszyGsJ1CaOwCNC3mnhKLFK2icSXUibd3sMsXeHMDuOlvtPUST58C7ajRE5q6MbYgrdYZOTw/640?wx_fmt=png&amp;from=appmsg" src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBqoM1zOHszyGsJ1CaOwCNC3mnhKLFK2icSXUibd3sMsXeHMDuOlvtPUST58C7ajRE5q6MbYgrdYZOTw/640?wx_fmt=png&amp;from=appmsg"></p><p><img data-galleryid="" data-imgfileid="100024173" data-ratio="0.8444444444444444" data-s="300,640" data-type="png" data-w="1035" data-src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBqoM1zOHszyGsJ1CaOwCNC3rjIg62mquMVArzJKNfacvialj9ib7rwGOwiaLqrzh66WBGtciaNicWZgp2A/640?wx_fmt=png&amp;from=appmsg" src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBqoM1zOHszyGsJ1CaOwCNC3rjIg62mquMVArzJKNfacvialj9ib7rwGOwiaLqrzh66WBGtciaNicWZgp2A/640?wx_fmt=png&amp;from=appmsg"></p><p>有这么多种细胞类型和对应的调控因子，对学术界是否有什么启发作用呢？</p><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/KNyfBLmH5WBD20_CYiIntg",target="_blank" rel="noopener noreferrer">原文链接</a>
