---
title: "计算 Ribosome 距离二肽/三肽基序距离的密度"
date: 2024-09-01T12:04:59Z
draft: ["false"]
tags: [
  "fetched",
  "老俊俊的生信笔记"
]
categories: ["Acdemic"]
---
计算 Ribosome 距离二肽/三肽基序距离的密度 by 老俊俊的生信笔记
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com" data-mpa-powered-by="yiban.io"><section><mp-common-profile data-pluginname="mpprofile" data-id="MzkyMTI1MTYxNA==" data-headimg="http://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4usezgsqIGqjITSMggCTSoViaYeoKe2xoZr1IIvNJoztibQxibYHLDDoiabwAc6Ggws3Tvdo8EPss2nLgaVQ/0?wx_fmt=png" data-nickname="老俊俊的生信笔记" data-alias="JunJunLab" data-signature="老俊俊的生信技能和知识分享,我不是巨人,但你可以站在我的肩膀上更进一步!" data-from="0" data-is_biz_ban="0"></mp-common-profile></section></section><section><mp-common-clmusic data-pluginname="insertaudio" type="1" music_name="%E4%BA%8C%E5%AC%A2" albumurl="http://wx.y.gtimg.cn/music/photo_new/T002R500x500M000000vkaAP4PJqeY_1.jpg" singer="尧十三" duration="391000" username="" music_source="1" is_vip="0" listenid="78221539386991872"></mp-common-clmusic></section><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><h2 data-tool="mdnice编辑器"><span></span><span>引言</span><span></span></h2><blockquote data-tool="mdnice编辑器"><span>❝</span><p>在 <a href="https://mp.weixin.qq.com/s?__biz=MzkyMTI1MTYxNA==&amp;mid=2247514475&amp;idx=1&amp;sn=2de16f76c92493bae019affe1119673d&amp;chksm=c184831af6f30a0c07a80a8e5e6b1885c0151cb465254322391dd1ffe823ad2673df70434589&amp;token=1519752094&amp;lang=zh_CN&amp;scene=21#wechat_redirect" data-linktype="2">Molecular Cell 文章部分结果复现(续)</a> 中我们计算了一些在 <strong>eIF5A</strong> 缺失条件下一些核糖体占据明显增多的三肽基序。</p></blockquote><figure data-tool="mdnice编辑器"><img data-imgfileid="100030864" data-ratio="0.5111111111111111" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4usewOgZpv1hGq8TNmYYU4My7iamAbfuu5bxiarKySgm1nmhAtiapibiabicNmMEOZ0aow7Q6PjEJnf7MBqr7g/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="990" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4usewOgZpv1hGq8TNmYYU4My7iamAbfuu5bxiarKySgm1nmhAtiapibiabicNmMEOZ0aow7Q6PjEJnf7MBqr7g/640?wx_fmt=png&amp;from=appmsg"></figure><blockquote data-tool="mdnice编辑器"><span>❝</span><p>接下来我们计算核糖体的 <strong>P/E</strong> 位点距离这些三肽基序密度的分布情况，下面是图中结果：</p></blockquote><figure data-tool="mdnice编辑器"><img data-imgfileid="100030868" data-ratio="0.4827361563517915" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4usewOgZpv1hGq8TNmYYU4My7iaUTKBbOlFpHacCORQY1Cia7fzpufibLRibax9KffNQPZ05KCoOibEN9Urxw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1535" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4usewOgZpv1hGq8TNmYYU4My7iaUTKBbOlFpHacCORQY1Cia7fzpufibLRibax9KffNQPZ05KCoOibEN9Urxw/640?wx_fmt=png&amp;from=appmsg"></figure><h2 data-tool="mdnice编辑器"><span></span><span>安装</span><span></span></h2><pre data-tool="mdnice编辑器"><span></span><code><span># install.packages("devtools")</span><br>devtools::install_github(<span>"junjunlab/RiboProfiler"</span>)<br><br><span># or</span><br>remotes::install_github(<span>"junjunlab/RiboProfiler"</span>)<br><br><span>library</span>(RiboProfiler)<br></code></pre><h2 data-tool="mdnice编辑器"><span></span><span>测试</span><span></span></h2><p data-tool="mdnice编辑器"><strong>adjust_offset</strong> 函数默认矫正后是 <strong>P</strong> 位点的位置：</p><pre data-tool="mdnice编辑器"><span></span><code><span># test</span><br>offset &lt;- PsiteOffsetCheck(qc_data = qc_df,read_length = c(<span>28</span>,<span>28</span>))<br>offset$plot<br>offset$summary_offset<br><span>#   sample readLengths Offsets bamFiles bamLegends</span><br><span># 1 eIF5Ad          28     -12   eIF5Ad     eIF5Ad</span><br><span># 2     WT          28     -12       WT         WT</span><br><br>offset_df &lt;- adjust_offset(offset_df = offset$summary_offset,<br>                           qc_df = qc_df)<br><br>norm_df &lt;- get_nomalized_counts(longest_trans_file = <span>"longest_info_extend.txt"</span>,<br>                                qc_df = offset_df)<br></code></pre><blockquote data-tool="mdnice编辑器"><span>❝</span><p>计算距离 <strong>PPP</strong> 基序的相对距离的密度分布, <strong>min_counts</strong> 筛选 counts 数大于等于 64 的，保持和文章中一致，<strong>motif</strong> 指定我们的基序，<strong>upstream/downstream</strong> 则是基序上下游的距离 ：</p></blockquote><pre data-tool="mdnice编辑器"><span></span><code>rel_dist_motif(amino_file = <span>"./sac_cds_aa.fa"</span>,<br>               longest_trans_file = <span>"longest_info_extend.txt"</span>,<br>               normed_file = norm_df,<br>               min_counts = <span>64</span>,<br>               motif = <span>"PPP"</span>,<br>               upstream = -<span>50</span>,<br>               downstream = <span>50</span>)<br></code></pre><p data-tool="mdnice编辑器">筛选原理就是匹配指定的氨基酸并返回位置，筛选在指定位置范围里的目标，然后把 <strong>density</strong> 储存起来，下面是代码：</p><pre data-tool="mdnice编辑器"><span></span><code><span>###########################################################################################</span><br><span># 2_load longtest transcript anno file</span><br><span>###########################################################################################</span><br><span>with</span> open(normed_file,<span>'r'</span>) <span>as</span> input:<br>    <span>for</span> line <span>in</span> input:<br>        <span># fileds = line.split()</span><br>        <span># print(line.split())</span><br>        _,_,_,_,_,_,trans_pos,trans_id,counts,_,_,exp = line.split()<br><br>        <span>if</span> trans_id <span>in</span> gene_info:<br>            seq = str(amino[trans_id])<br><br>            <span># search motif patterns for amino acids</span><br>            <span># pattern = re.compile("PPP",flags=re.IGNORECASE)</span><br>            pattern = re.compile(motif, flags=re.IGNORECASE)<br>            pos_res = pattern.finditer(seq)<br><br>            <span># motif codon position</span><br>            codon_motif_pos = [pos.start() + <span>1</span> <span>for</span> pos <span>in</span> pos_res]<br><br>            <span># check whether has target motif</span><br>            <span>if</span> len(codon_motif_pos) &gt; <span>0</span>:<br>                <span># motif codon position to nt position</span><br>                motif_nt_pos = [pos*<span>3</span>- <span>2</span> <span>for</span> pos <span>in</span> codon_motif_pos]<br><br>                <span># calculate relative distance to motif</span><br>                utr5_len = gene_info[trans_id][<span>0</span>]<br>                rel_pos_to_motif = [int(trans_pos) - utr5_len - nt_pos <span>for</span> nt_pos <span>in</span> motif_nt_pos]<br><br>                <span># filter rel_pos in limited range (-50,50)</span><br>                rel_pos_to_motif_in_range = [i <span>for</span> i <span>in</span> rel_pos_to_motif <span>if</span> i &gt;= int(upstream) <span>and</span> i &lt;= int(downstream)]<br><br>                <span># save in dict</span><br>                <span>for</span> j <span>in</span> rel_pos_to_motif_in_range:<br>                    <span>if</span> j <span>in</span> rel_dist:<br>                        rel_dist[j][<span>0</span>] += <span>1</span><br>                        rel_dist[j][<span>1</span>] += float(exp)<br>                        <span># rel_dist[j][1] += int(counts)</span><br><br><span># normalize density | Average ribosome occupancy</span><br>sum_counts = sum([val[<span>1</span>] <span>for</span> key,val <span>in</span> rel_dist.items()])<br>average_exp = sum_counts/len(range(int(upstream),int(downstream + <span>1</span>)))<br>rel_dist_norm = {key:val[<span>1</span>]/average_exp <span>for</span> key,val <span>in</span> rel_dist.items()}<br></code></pre><p data-tool="mdnice编辑器">然后会在 <strong>relative_distance_motif</strong> 文件夹下产生这些文件：</p><figure data-tool="mdnice编辑器"><img data-imgfileid="100030867" data-ratio="0.45228215767634855" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4usewOgZpv1hGq8TNmYYU4My7iaz2T4R2kTp2dPSsCxJHia6ttYHIVIGLPUdF4wWYe9rUTsE8JQCrz8gVw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1205" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4usewOgZpv1hGq8TNmYYU4My7iaz2T4R2kTp2dPSsCxJHia6ttYHIVIGLPUdF4wWYe9rUTsE8JQCrz8gVw/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">然后画图：</p><pre data-tool="mdnice编辑器"><span></span><code><span>library</span>(ggplot2)<br><br><span># plot</span><br>files &lt;- c(<span>"relative_distance_motif/WT_PPP_motif_occupancy.txt"</span>,<br>           <span>"relative_distance_motif/eIF5Ad_PPP_motif_occupancy.txt"</span>)<br><br>samples &lt;- c(<span>"WT"</span>,<span>"eIF5Ad"</span>)<br><br>rel_dist_motif_plot(motif_occupancy_file = files,<br>                    sample_name = samples,<br>                    motif = <span>"PPP"</span>,<br>                    site = <span>"P"</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100030866" data-ratio="0.6765899864682002" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4usewOgZpv1hGq8TNmYYU4My7iaaCMwSyGvGiauIfaSX4NZ2k53QmxcX6RCsMpWfffSJ8tDGH1Sgkwqdkw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="739" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4usewOgZpv1hGq8TNmYYU4My7iaaCMwSyGvGiauIfaSX4NZ2k53QmxcX6RCsMpWfffSJ8tDGH1Sgkwqdkw/640?wx_fmt=png&amp;from=appmsg"></figure><blockquote data-tool="mdnice编辑器"><span>❝</span><p>如果我们要算 <strong>E site</strong> 的话，<strong>adjust_offset</strong> 函数指定一下 <strong>shift</strong> 参数再向左移动 <strong>3nt</strong> 即可，<strong>A site</strong> 则向右移动 <strong>3nt</strong>：</p></blockquote><pre data-tool="mdnice编辑器"><span></span><code><span># E site</span><br>offset_df &lt;- adjust_offset(offset_df = offset<span>$summary_offset</span>,<br>                           qc_df = qc_df,<span>shift</span> = -3)<br><br><span># A site</span><br>offset_df &lt;- adjust_offset(offset_df = offset<span>$summary_offset</span>,<br>                           qc_df = qc_df,<span>shift</span> = 3)<br></code></pre><p data-tool="mdnice编辑器">同样的我们计算 <strong>PP</strong> 二肽基序的看看：</p><pre data-tool="mdnice编辑器"><span></span><code>rel_dist_motif(amino_file = <span>"./sac_cds_aa.fa"</span>,<br>               longest_trans_file = <span>"longest_info_extend.txt"</span>,<br>               normed_file = norm_df,<br>               min_counts = <span>64</span>,<br>               motif = <span>"PP"</span>,<br>               upstream = -<span>50</span>,<br>               downstream = <span>50</span>)<br><br>files &lt;- c(<span>"relative_distance_motif/WT_PP_motif_occupancy.txt"</span>,<br>           <span>"relative_distance_motif/eIF5Ad_PP_motif_occupancy.txt"</span>)<br><br>samples &lt;- c(<span>"WT"</span>,<span>"eIF5Ad"</span>)<br><br>rel_dist_motif_plot(motif_occupancy_file = files,<br>                    sample_name = samples,<br>                    motif = <span>"PP"</span>,<br>                    site = <span>"P"</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100030865" data-ratio="0.518918918918919" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4usewOgZpv1hGq8TNmYYU4My7ia5gNfQ9ZxBlawxicRJzSlyA3ceic4EibpPBecKPn8VciauyRIBuAgsuFx4A/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="740" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4usewOgZpv1hGq8TNmYYU4My7ia5gNfQ9ZxBlawxicRJzSlyA3ceic4EibpPBecKPn8VciauyRIBuAgsuFx4A/640?wx_fmt=png&amp;from=appmsg"></figure><h2 data-tool="mdnice编辑器"><span></span><span>tri-peptide 计算</span><span></span></h2><p data-tool="mdnice编辑器">前面我讲过这个问题：</p><figure data-tool="mdnice编辑器"><img data-imgfileid="100030869" data-ratio="0.367561260210035" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4usewOgZpv1hGq8TNmYYU4My7iaibP8FPibghD7ZEvTGUY5UKvic6Bcm1XCtiaogGJ4mVcrRvXtx1BlNulJcA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="857" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4usewOgZpv1hGq8TNmYYU4My7iaibP8FPibghD7ZEvTGUY5UKvic6Bcm1XCtiaogGJ4mVcrRvXtx1BlNulJcA/640?wx_fmt=png&amp;from=appmsg"></figure><blockquote data-tool="mdnice编辑器"><span>❝</span><p>仔细想想，我们海油一种更综合的算法，就是争对特定的一个氨基酸位置，可分为向两边取，向前取两位和向后取两位三种方式，我们都纳入进来，就可以把氨基酸长度不是 3 的倍数的序列都容纳进来了。</p></blockquote><pre data-tool="mdnice编辑器"><span></span><code><span>with</span> open(codon_exp_file,<span>'r'</span>) <span>as</span> input:<br>      <span>for</span> line <span>in</span> input:<br>          trans_id,codon_pos,exp = line.split()<br><br>          <span># # get tripeptide position</span><br>          <span># if int(codon_pos)%3 == 0:</span><br>          <span>#     pos = int(codon_pos)//3</span><br>          <span># else:</span><br>          <span>#     pos = int(int(codon_pos)//3) + 1</span><br><br>          <span># # check transid in AA fasta file</span><br>          <span># if trans_id in amino.keys():</span><br>          <span>#     tripeptide_motif = amino.fetch(trans_id,(pos*3 - 2,pos*3))</span><br><br>          <span>#     # make sure length is 3</span><br>          <span>#     if len(tripeptide_motif) == 3:</span><br>          <span>#         # get tripeptide and save in dict</span><br>          <span>#         if tripeptide_motif in tripeptide_seq:</span><br>          <span>#             tmp_exp,tmp_count = tripeptide_seq[tripeptide_motif]</span><br>          <span>#             tripeptide_seq[tripeptide_motif] = [float(exp) + tmp_exp,tmp_count + 1]</span><br>          <span>#         else:</span><br>          <span>#             tripeptide_seq[tripeptide_motif] = [float(exp),1]</span><br><br>          <span># check transid in AA fasta file (revised code)!</span><br>          <span>if</span> trans_id <span>in</span> amino.keys():<br>              <span># 3nt window to fetch tri-peptide motif</span><br>              type1 = amino.fetch(trans_id,(int(codon_pos) - <span>1</span>,int(codon_pos) + <span>1</span>))<br>              type2 = amino.fetch(trans_id,(int(codon_pos) - <span>2</span>,int(codon_pos)))<br>              type3 = amino.fetch(trans_id,(int(codon_pos),int(codon_pos) + <span>2</span>))<br><br>              tripeptide_motif_list = [type1, type2, type3]<br><br>              <span># filter motif length is 3</span><br>              tripeptide_motif_list_filterd = [i <span>for</span> i <span>in</span> tripeptide_motif_list <span>if</span> len(i) == <span>3</span>]<br><br>              <span># get tripeptide and save in dict</span><br>              <span>for</span> motif <span>in</span> tripeptide_motif_list_filterd:<br>                  <span>if</span> motif <span>in</span> tripeptide_seq:<br>                      tmp_exp,tmp_count = tripeptide_seq[motif]<br>                      tripeptide_seq[motif] = [float(exp) + tmp_exp,tmp_count + <span>1</span>]<br>                  <span>else</span>:<br>                      tripeptide_seq[motif] = [float(exp),<span>1</span>]<br></code></pre><p data-tool="mdnice编辑器">再分析看看：</p><pre data-tool="mdnice编辑器"><span></span><code><span># calculate</span><br>peptide_motif_score(amino_file = <span>"sac_cds_aa.fa"</span>,<br>                    codon_exp_file = c(<span>"codon_occupancy/WT_codon_pos_exp.txt"</span>,<br>                                       <span>"codon_occupancy/eIF5Ad_codon_pos_exp.txt"</span>),<br>                    sample_name = c(<span>"WT"</span>,<span>"eIF5Ad"</span>),<br>                    occurrence_threshold = <span>100</span>)<br><br><span># plot</span><br>triAmino_scater_plot(occupancy_file = c(<span>"peptide_motif/WT_tripeptide_occupancy.txt"</span>,<br>                                        <span>"peptide_motif/eIF5Ad_tripeptide_occupancy.txt"</span>),<br>                     sampe_name = c(<span>"WT"</span>,<span>"eIF5Ad"</span>),<br>                     mark_motif = c(<span>"DDP"</span>,<span>"DPP"</span>,<span>"PPP"</span>,<span>"PDK"</span>,<span>"PDG"</span>,<br>                                    <span>"GPP"</span>,<span>"PDP"</span>,<span>"DNP"</span>,<span>"DGP"</span>,<span>"SPP"</span>))<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100030871" data-ratio="0.99800796812749" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4usewOgZpv1hGq8TNmYYU4My7iafMcXoV8JBDBreiaoddbIY58hEzgDDBibeyr9eKyk2M3tiakvg63E5mJDg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="502" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4usewOgZpv1hGq8TNmYYU4My7iafMcXoV8JBDBreiaoddbIY58hEzgDDBibeyr9eKyk2M3tiakvg63E5mJDg/640?wx_fmt=png&amp;from=appmsg"></figure><h2 data-tool="mdnice编辑器"><span></span><span>结尾</span><span></span></h2><blockquote data-tool="mdnice编辑器"><span>❝</span><p><strong>路漫漫其修远兮,吾将上下而求索。</strong></p></blockquote><hr data-tool="mdnice编辑器"><p data-tool="mdnice编辑器">欢迎加入生信交流群。加我微信我也拉你进 <strong>微信群聊</strong> <strong>老俊俊生信交流群</strong> <strong>(微信交流群需收取 20 元入群费用,一旦交费,拒不退还!(防止骗子和便于管理))</strong> 。QQ 群可免费加入, 记得进群按格式修改备注哦。</p><section data-tool="mdnice编辑器"><section><p><strong>老俊俊微信：</strong></p><figure><img data-imgfileid="100030870" data-ratio="1" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4usewOgZpv1hGq8TNmYYU4My7iaKFgZqplnwtfNuiaaf6UcJ8laXF66aGibj6F8sWMVOicQ51iaIyqgvC5FMQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="430" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4usewOgZpv1hGq8TNmYYU4My7iaKFgZqplnwtfNuiaaf6UcJ8laXF66aGibj6F8sWMVOicQ51iaIyqgvC5FMQ/640?wx_fmt=png&amp;from=appmsg"></figure><figure><img data-imgfileid="100030872" data-ratio="1.3668430335097002" data-src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4usewOgZpv1hGq8TNmYYU4My7ia5icNm5yHeX63Dwq2BszlqL5rQ2Yk3H8WQMfJCVAvNx3St71tPt99rXw/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="567" src="https://mmbiz.qpic.cn/sz_mmbiz_png/G5jjcE4usewOgZpv1hGq8TNmYYU4My7ia5icNm5yHeX63Dwq2BszlqL5rQ2Yk3H8WQMfJCVAvNx3St71tPt99rXw/640?wx_fmt=png&amp;from=appmsg"></figure></section><section><p><strong>知识星球：</strong></p><figure><img data-imgfileid="100030873" data-ratio="1.5896226415094339" data-src="https://mmbiz.qpic.cn/sz_mmbiz_jpg/G5jjcE4usewOgZpv1hGq8TNmYYU4My7iaqWdpgXnr9LjicgdicbLxYRPjTTAeGwvR1ahibuD1bQv7uQuFFj6WqEhjg/640?wx_fmt=jpeg&amp;from=appmsg" data-type="jpeg" data-w="1060" src="https://mmbiz.qpic.cn/sz_mmbiz_jpg/G5jjcE4usewOgZpv1hGq8TNmYYU4My7iaqWdpgXnr9LjicgdicbLxYRPjTTAeGwvR1ahibuD1bQv7uQuFFj6WqEhjg/640?wx_fmt=jpeg&amp;from=appmsg"></figure></section></section><hr data-tool="mdnice编辑器"><p data-tool="mdnice编辑器"><strong></strong></p><center data-tool="mdnice编辑器"><strong> 往期回顾目录</strong></center><blockquote data-tool="mdnice编辑器"><span>❝</span><p><strong></strong></p><center><strong><a href="https://mp.weixin.qq.com/s?__biz=MzkyMTI1MTYxNA==&amp;mid=2247514511&amp;idx=1&amp;sn=60509453e63d2659e4b728e33795fedc&amp;chksm=c18483fef6f30ae8ebba89897bdcbd5d15de10fe9131eb9d14689635e5c617aa65eba8404017&amp;token=1519752094&amp;lang=zh_CN&amp;scene=21#wechat_redirect" data-linktype="2">怎么给书籍插入参考文献</a></strong></center><strong><center><a href="https://mp.weixin.qq.com/s?__biz=MzkyMTI1MTYxNA==&amp;mid=2247514492&amp;idx=1&amp;sn=a23e705c9ce6412033056c162ba1f513&amp;chksm=c184830df6f30a1be8b6fa9fefa846ff7e19836f2069f24761e440f9ee0320a6a0c8cca49ef3&amp;token=1519752094&amp;lang=zh_CN&amp;scene=21#wechat_redirect" data-linktype="2">Ribo-seq 核糖体去除新方法？</a></center></strong><strong><center><a href="https://mp.weixin.qq.com/s?__biz=MzkyMTI1MTYxNA==&amp;mid=2247514475&amp;idx=1&amp;sn=2de16f76c92493bae019affe1119673d&amp;chksm=c184831af6f30a0c07a80a8e5e6b1885c0151cb465254322391dd1ffe823ad2673df70434589&amp;token=1519752094&amp;lang=zh_CN&amp;scene=21#wechat_redirect" data-linktype="2">Molecular Cell 文章部分结果复现(续)</a></center></strong><strong><center><a href="https://mp.weixin.qq.com/s?__biz=MzkyMTI1MTYxNA==&amp;mid=2247514430&amp;idx=1&amp;sn=a04b166f24c17f225ee887247090e4eb&amp;chksm=c184834ff6f30a591e4cd7d0fa40174e458babe656f689279ae78963dc671f635a8b86a10835&amp;token=1698951491&amp;lang=zh_CN&amp;scene=21#wechat_redirect" data-linktype="2">Molecular Cell 文章部分结果复现</a></center></strong><strong><center><a href="https://mp.weixin.qq.com/s?__biz=MzkyMTI1MTYxNA==&amp;mid=2247514370&amp;idx=1&amp;sn=156f2fa314f498a749aabeeb8604d7d9&amp;chksm=c1848373f6f30a65ca990dfea862b54041fb6f2d0aa5c25dc1b1073d4008918c9a5e8b14f964&amp;token=920737487&amp;lang=zh_CN&amp;scene=21#wechat_redirect" data-linktype="2">提取最长转录本信息+exon 序列+CDS 序列+氨基酸序列</a></center></strong><strong><center><a href="https://mp.weixin.qq.com/s?__biz=MzkyMTI1MTYxNA==&amp;mid=2247514340&amp;idx=1&amp;sn=3ad6754b293f99787d9b009c239f1191&amp;chksm=c1848295f6f30b83a2595f08e428c7bc0021fbc907ca70ed230c84ea58e5c883cce1f057d00d&amp;token=920737487&amp;lang=zh_CN&amp;scene=21#wechat_redirect" data-linktype="2">检测 Ribo-SEQ 中的 offset</a></center></strong><strong><center><a href="https://mp.weixin.qq.com/s?__biz=MzkyMTI1MTYxNA==&amp;mid=2247514243&amp;idx=1&amp;sn=1a5f31be3e9aa3d1491f3f50a9846746&amp;chksm=c18482f2f6f30be4fa1eaf5767656003a2a171684584967764b99fa57785ff3e0eca3bbf4dae&amp;token=1594426429&amp;lang=zh_CN&amp;scene=21#wechat_redirect" data-linktype="2">ribo-seq 计算 polarity score</a></center></strong><strong><center><a href="https://mp.weixin.qq.com/s?__biz=MzkyMTI1MTYxNA==&amp;mid=2247514243&amp;idx=1&amp;sn=1a5f31be3e9aa3d1491f3f50a9846746&amp;chksm=c18482f2f6f30be4fa1eaf5767656003a2a171684584967764b99fa57785ff3e0eca3bbf4dae&amp;token=1594426429&amp;lang=zh_CN&amp;scene=21#wechat_redirect" data-linktype="2">ribo-seq 计算 polarity score</a></center></strong><strong><center><a href="https://mp.weixin.qq.com/s?__biz=MzkyMTI1MTYxNA==&amp;mid=2247514198&amp;idx=1&amp;sn=26b646505c4132cbe2c4fab32fa82a17&amp;chksm=c1848227f6f30b31a43e1ce2211776404739397742ce31f63022123840ae0db66a738a959dd8&amp;token=1594426429&amp;lang=zh_CN&amp;scene=21#wechat_redirect" data-linktype="2">Ribo-SEQ 分析相关软件一览</a></center></strong><strong><center><a href="https://mp.weixin.qq.com/s?__biz=MzkyMTI1MTYxNA==&amp;mid=2247514164&amp;idx=1&amp;sn=0b7278817ff0a773cdfd89d3e373576f&amp;chksm=c1848245f6f30b53b8ed7138363e72ec8a882d34f454a463a7ce79b840153c91712045ef1f97&amp;token=857496904&amp;lang=zh_CN&amp;scene=21#wechat_redirect" data-linktype="2">听说你想把 fgsea 结果用 GseaVis 美化？</a></center></strong><p><br></p></blockquote></section><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/fthHfpMlAWwEooCnsSItyw",target="_blank" rel="noopener noreferrer">原文链接</a>
