---
title: "单细胞和空间转录组联合分析-RCTD去卷积"
date: 2024-03-17T02:41:27Z
draft: ["false"]
tags: [
  "fetched",
  "朴素的科研打工仔"
]
categories: ["Acdemic"]
---
单细胞和空间转录组联合分析-RCTD去卷积 by 朴素的科研打工仔
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><section><span><strong>数据来源</strong>：</span></section><section><span>Spatial proteogenomics reveals distinct andevolutionarily conserved hepatic macrophageniches,Cell,2022</span></section><section><span>Single-Cell, Single-Nucleus, andSpatial RNA Sequencing of the HumanLiver Identifies Cholangiocyte andMesenchymal Heterogeneity,Hepatology communications,2021</span></section><h2 data-tool="mdnice编辑器"><span>数据读取-空转</span></h2><pre data-tool="mdnice编辑器"><code>library(Seurat)<br>library(ggplot2)<br>library(stringr)<br>library(harmony)<br>library(patchwork)<br>library(dplyr)<br>#读取数据 空间转录组部分<br>## 批量读取数据<br>assays &lt;- dir(<span>"rawdata_st/"</span>)<br>dir &lt;- paste0(<span>"rawdata_st/"</span>, assays)<br>samples_name = c(<span>'D1'</span>,<span>'D2'</span>,<span>'H1'</span>,<span>'D3'</span>,<span>'H2'</span>)<br>spatial.list = list()<br><span>for</span>(i in <span>1</span>:length(dir)){<br>  spatial &lt;- Load10X_Spatial(data.dir = dir[i],slice = samples_name[i])<br>  spatial$orig.ident &lt;- samples_name[i]<br>  spatial$group &lt;- stringr::str_extract_all(samples_name[i],<span>"[aA-zZ]+"</span>)<br>  spatial &lt;- RenameCells(spatial, add.cell.id = samples_name[i])<br>  spatial [[<span>"percent.mt"</span>]] &lt;- PercentageFeatureSet(spatial, pattern = <span>"^MT[-]"</span>)<br>  spatial.list[[samples_name[i]]] &lt;- spatial<br>}<br>p_mt.list &lt;- lapply(spatial.list, function(x){<br>  p1 &lt;- VlnPlot(x, features = <span>"percent.mt"</span>) + ggtitle(unique(x$orig.ident)) +<br>    theme(legend.position = <span>"none"</span>, axis.text.x = element_blank())<br>  p2 &lt;- SpatialFeaturePlot(x, features = <span>"percent.mt"</span>) + theme(legend.position = <span>"right"</span>)<br>  p &lt;- p1|p2<br>  p })<br>wrap_plots(p_mt.list, ncol = <span>2</span>)<br># 整合样本<br>stRNA &lt;- merge(spatial.list[[<span>1</span>]], spatial.list[<span>2</span>:length(spatial.list)])<br># 标准化<br>stRNA &lt;- SCTransform(stRNA, assay = <span>"Spatial"</span>, verbose = T)<br>## 降维聚类<br>stRNA &lt;- RunPCA(stRNA, assay = <span>"SCT"</span>, verbose = T)<br>ElbowPlot(stRNA, ndims=<span>30</span>, reduction=<span>"pca"</span>) <br>stRNA = RunHarmony(stRNA, group.by.vars = <span>"orig.ident"</span>,assay.use = <span>"SCT"</span>)<br>stRNA &lt;- RunUMAP(stRNA, reduction = <span>"harmony"</span>, dims = <span>1</span>:<span>20</span>)<br>stRNA &lt;- FindNeighbors(stRNA, reduction = <span>"harmony"</span>, dims = <span>1</span>:<span>20</span>)<br>stRNA &lt;- FindClusters(stRNA, verbose = T,resolution = <span>0.5</span>)<br>## 可视化<br>p1 &lt;- DimPlot(stRNA, reduction = <span>"umap"</span>, label = TRUE,group.by = <span>'SCT_snn_res.0.5'</span>,pt.size = <span>1</span>)<br>p1<br>p2 &lt;- SpatialDimPlot(stRNA,label = T)<br>p2<br>p1/p2<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100001268" data-ratio="0.8194444444444444" data-src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676vUjSTSqhJXOpGdiaatD6ODKQJAYibnRib8kcPGFBRgNbicn6eHhibXaMh9icAghXchIGIPbN5jF169iaQ/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676vUjSTSqhJXOpGdiaatD6ODKQJAYibnRib8kcPGFBRgNbicn6eHhibXaMh9icAghXchIGIPbN5jF169iaQ/640?wx_fmt=png&amp;from=appmsg"></figure><figure data-tool="mdnice编辑器"><img data-imgfileid="100001269" data-ratio="0.18981481481481483" data-src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676vUjSTSqhJXOpGdiaatD6OUtkUpzJHPjM7tQp9IFJQmrpAtsFRKLe72kHpibawoDCokZ7VozFL1oA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676vUjSTSqhJXOpGdiaatD6OUtkUpzJHPjM7tQp9IFJQmrpAtsFRKLe72kHpibawoDCokZ7VozFL1oA/640?wx_fmt=png&amp;from=appmsg"></figure><h2 data-tool="mdnice编辑器"><span>数据读取-单细胞</span></h2><pre data-tool="mdnice编辑器"><code>#读取数据 单细胞部分<br>## 批量读取数据<br>setwd(<span>"./rawdata_sc/"</span>)<br>fs=list.files(pattern = <span>'.h5'</span>)<br>samples_name = c(<span>'Ob1'</span>,<span>'Ob2'</span>,<span>'Ob3'</span>,<span>'Ob4'</span>,<span>'Ob5'</span>,<span>'Ob6'</span>,<span>'Le1'</span>,<span>"Le2"</span>)<br>scRNA.list &lt;- list()<br><span>for</span>(i in <span>1</span>:length(fs)){<br>  counts &lt;- Read10X_h5(fs[i])<br>  sample=str_split(fs[i],<span>'_'</span>,simplify = T)[,<span>1</span>]<br>  scRNA &lt;- CreateSeuratObject(counts, project=samples_name[i],<br>                                       min.cells=<span>3</span>, min.features = <span>200</span>)<br>  scRNA &lt;- RenameCells(scRNA, add.cell.id = samples_name[i])<br>  scRNA$group &lt;- stringr::str_extract_all(samples_name[i],<span>"[aA-zZ]+"</span>)<br>  scRNA.list[[samples_name[i]]] &lt;- scRNA<br>}<br><br>scRNA &lt;- merge(scRNA.list[[<span>1</span>]], scRNA.list[<span>2</span>:length(scRNA.list)])<br># 计算细胞中线粒体基因比例<br>scRNA[[<span>"percent.mt"</span>]] &lt;- PercentageFeatureSet(scRNA, pattern = <span>"^MT-"</span>)<br># 计算细胞中核糖体基因比例<br>scRNA[[<span>"percent.rb"</span>]] &lt;- PercentageFeatureSet(scRNA, pattern = <span>"^RP[LS]"</span>)<br># 质控<br>scRNA &lt;- subset(scRNA, nCount_RNA&lt;<span>8000</span>&amp;nFeature_RNA&gt;<span>300</span>&amp;percent.mt&lt;<span>25</span>)<br># 标准化<br>scRNA &lt;- NormalizeData(scRNA, normalization.method = <span>"LogNormalize"</span>, scale.factor = <span>10000</span>)<br># 鉴定高变基因<br>scRNA &lt;- FindVariableFeatures(scRNA, selection.method = <span>"vst"</span>, nfeatures = <span>3000</span>)<br># 尺度变换<br>scRNA &lt;- ScaleData(scRNA, features = rownames(scRNA))<br># 降维<br>scRNA &lt;- RunPCA(scRNA, features = VariableFeatures(object = scRNA))<br>scRNA &lt;- RunHarmony(scRNA, group.by.vars=<span>"orig.ident"</span>, max.iter.harmony = <span>20</span>)<br>scRNA &lt;- RunUMAP(scRNA,reduction = <span>"harmony"</span>, dims = <span>1</span>:<span>30</span>, verbose = FALSE)<br>## 确定分辨率<br>scRNA &lt;- FindNeighbors(scRNA, reduction = <span>"harmony"</span>,dims = <span>1</span>:<span>30</span>, verbose = FALSE)<br><span>for</span> (<span>res in <span>c</span><span>(<span>0.01</span>, <span>0.05</span>, <span>0.1</span>, <span>0.2</span>, <span>0.3</span>, <span>0.5</span>,<span>0.8</span>,<span>1</span>)</span>) </span>{<br>  print(res)<br>  scRNA &lt;- FindClusters(scRNA, graph.name = <span>"RNA_snn"</span>, <br>                        resolution = res, algorithm = <span>1</span>)<br>}<br>colors &lt;- c(<span>'#e41a1c'</span>,<span>'#377eb8'</span>,<span>'#4daf4a'</span>,<span>'#984ea3'</span>,<span>'#ff7f00'</span>,<span>'#ffff33'</span>,<span>'#a65628'</span>,<span>'#f781bf'</span>)<br>names(colors) &lt;- unique(colnames(scRNA<span>@meta</span>.data)[grep(<span>"RNA_snn_res"</span>, colnames(scRNA<span>@meta</span>.data))]) <br>library(clustree)<br>clustree(scRNA<span>@meta</span>.data, prefix = <span>"RNA_snn_res."</span>)<br>Idents(scRNA) &lt;- <span>"RNA_snn_res.0.3"</span><br>DimPlot(scRNA,label = T)<br># 细胞类型鉴定<br>DefaultAssay(scRNA) &lt;- <span>"RNA"</span><br>library(COSG)<br>marker_cosg &lt;- cosg(scRNA,groups=<span>'all'</span>,assay=<span>'RNA'</span>,slot=<span>'data'</span>,mu=<span>1</span>,n_genes_user=<span>100</span>)<br><span>new</span>.cluster.ids &lt;- c(<span>"Hep1"</span>,<span>"Hep2"</span>, <span>"Hep3"</span>,<span>"Hep4"</span>,<br>                     <span>"LSEC"</span>,<span>"Kup"</span>,<span>"HSC"</span>,<span>"NKT"</span>,<span>"Cho"</span>, <br>                     <span>"UN1"</span>, <span>"CV_EC"</span>, <span>"UN2"</span>,<span>"Plasma"</span>)<br>names(<span>new</span>.cluster.ids) &lt;- levels(scRNA)<br>scRNA &lt;- RenameIdents(scRNA, <span>new</span>.cluster.ids)<br>table(scRNA<span>@active</span>.ident)<br>scRNA$celltype &lt;- scRNA<span>@active</span>.ident<br># 可视化<br>Dimplot（scRNA,label=T）<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100001267" data-ratio="0.9959058341862845" data-src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676vUjSTSqhJXOpGdiaatD6OE5pyHoqAPHicw0Aia0cq96gGv0S1ZMAsHpribhF9M5HHjwWTLx9wwC5kA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="977" src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676vUjSTSqhJXOpGdiaatD6OE5pyHoqAPHicw0Aia0cq96gGv0S1ZMAsHpribhF9M5HHjwWTLx9wwC5kA/640?wx_fmt=png&amp;from=appmsg"></figure><h2 data-tool="mdnice编辑器"><span>去卷积-RCTD</span></h2><pre data-tool="mdnice编辑器"><code># RCTD <br>## 单细胞参考数据库构建<br># 整体<br>library(spacexr)<br>library(tidyverse)<br>sc_counts &lt;- as.matrix(scRNA[[<span>'RNA'</span>]]<span>@counts</span>)<br>sc_nUMI = colSums(sc_counts)<br>cellType=data.frame(barcode=colnames(scRNA), celltype=scRNA$celltype)<br>names(cellType) = c(<span>'barcode'</span>, <span>'cell_type'</span>)<br>cell_types = cellType$cell_type; names(cell_types) &lt;- cellType$barcode <br>cell_types =as.factor(cell_types)<br>cell_types<br>reference = Reference(sc_counts, cell_types, sc_nUMI)<br># 拆分疾病和正常数据库<br>scRNA_Ob=subset(scRNA,group == <span>'Ob'</span>)<br>sc_counts &lt;- as.matrix(scRNA_Ob[[<span>'RNA'</span>]]<span>@counts</span>)<br>sc_nUMI = colSums(sc_counts)<br>cellType=data.frame(barcode=colnames(scRNA_Ob), celltype=scRNA_Ob$celltype)<br>names(cellType) = c(<span>'barcode'</span>, <span>'cell_type'</span>)<br>cell_types = cellType$cell_type; names(cell_types) &lt;- cellType$barcode<br>cell_types =as.factor(cell_types) <br>cell_types<br>reference_D = Reference(sc_counts, cell_types, sc_nUMI)<br><br>scRNA_Le=subset(scRNA,group == <span>'Le'</span>)<br>scRNA_Le=subset(scRNA_Le,celltype != <span>'UN1'</span>)<br>scRNA_Le$celltype=factor(scRNA_Le$celltype)<br>sc_counts &lt;- as.matrix(scRNA_Le[[<span>'RNA'</span>]]<span>@counts</span>)<br>sc_nUMI = colSums(sc_counts)<br>cellType=data.frame(barcode=colnames(scRNA_Le), celltype=scRNA_Le$celltype)<br>names(cellType) = c(<span>'barcode'</span>, <span>'cell_type'</span>)<br>cell_types = cellType$cell_type; names(cell_types) &lt;- cellType$barcode <br>cell_types =as.factor(cell_types)<br>reference_H = Reference(sc_counts, cell_types, sc_nUMI)<br></code></pre><pre data-tool="mdnice编辑器"><code># 拆分空间样本一一去卷积 --对应整体单细胞数据库<br>sto.list &lt;- SplitObject(stRNA, split.by = <span>"orig.ident"</span>)<br>slice &lt;- names(sto.list)<br>result.RCTD=data.frame()<br><span>for</span>(i in slice){<br>  coords &lt;- stRNA<span>@images</span>[[i]]<span>@coordinates</span>[,c(<span>4</span>,<span>5</span>)]<br>  colnames(coords) &lt;- c(<span>"x"</span>,<span>"y"</span>)<br>  query&lt;- SpatialRNA(coords = coords, counts = sto.list[[i]]<span>@assays</span>$Spatial<span>@counts</span>, <br>                          nUMI = structure(sto.list[[i]]$nCount_Spatial, names=colnames(sto.list[[i]])))<br>  RCTD &lt;- create.RCTD(spatialRNA = query, reference = reference, max_cores = <span>10</span>)<br>  RCTD &lt;- run.RCTD(RCTD, doublet_mode = <span>'doublet'</span>)<br>  result_df &lt;- RCTD<span>@results</span>$results_df<br>  result_df &lt;- as.data.frame(result_df)<br>  result.RCTD=result.RCTD%&gt;%rbind(result_df)<br>}<br>stRNA1=stRNA[,rownames(result.RCTD)]<br>stRNA1 &lt;- AddMetaData(stRNA1,metadata = result.RCTD)<br>Idents(stRNA1)=stRNA1$second_type<br>p1&lt;-SpatialDimPlot(stRNA1)<br># 拆分空间样本一一去卷积 --对应疾病和正常<br>result.RCTD_group=data.frame()<br><span>for</span>(i in slice){<br>  <span>if</span> (str_extract_all(i,<span>"[aA-zZ]+"</span>)==<span>'H'</span>){<br>  coords &lt;- stRNA<span>@images</span>[[i]]<span>@coordinates</span>[,c(<span>4</span>,<span>5</span>)]<br>  colnames(coords) &lt;- c(<span>"x"</span>,<span>"y"</span>)<br>  query&lt;- SpatialRNA(coords = coords, counts = sto.list[[i]]<span>@assays</span>$Spatial<span>@counts</span>, <br>                     nUMI = structure(sto.list[[i]]$nCount_Spatial, names=colnames(sto.list[[i]])))<br>  RCTD &lt;- create.RCTD(spatialRNA = query, reference = reference_H, max_cores = <span>10</span>)<br>  RCTD &lt;- run.RCTD(RCTD, doublet_mode = <span>'doublet'</span>)<br>  result_df &lt;- RCTD<span>@results</span>$results_df<br>  result_df &lt;- as.data.frame(result_df)<br>  } <span>else</span> <br>  {<br>  coords &lt;- stRNA<span>@images</span>[[i]]<span>@coordinates</span>[,c(<span>4</span>,<span>5</span>)]<br>  colnames(coords) &lt;- c(<span>"x"</span>,<span>"y"</span>)<br>  query&lt;- SpatialRNA(coords = coords, counts = sto.list[[i]]<span>@assays</span>$Spatial<span>@counts</span>, <br>                     nUMI = structure(sto.list[[i]]$nCount_Spatial, names=colnames(sto.list[[i]])))<br>  RCTD &lt;- create.RCTD(spatialRNA = query, reference = reference_D, max_cores = <span>10</span>)<br>  RCTD &lt;- run.RCTD(RCTD, doublet_mode = <span>'doublet'</span>)<br>  result_df &lt;- RCTD<span>@results</span>$results_df<br>  result_df &lt;- as.data.frame(result_df)<br>  }<br>  result.RCTD_group=result.RCTD_group%&gt;%rbind(result_df)<br>}<br>stRNA2=stRNA[,rownames(result.RCTD_group)]<br>stRNA2 &lt;- AddMetaData(stRNA2,metadata = result.RCTD_group)<br>Idents(stRNA2)=stRNA2$second_type<br>p2&lt;-SpatialDimPlot(stRNA2)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100001266" data-ratio="0.42034943473792397" data-src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676vUjSTSqhJXOpGdiaatD6OmKHqc2icX94Q76ZeOXnuVib1JoshyHwgDGYlMjNicQVQ3doicic9icFoBkcA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="973" src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676vUjSTSqhJXOpGdiaatD6OmKHqc2icX94Q76ZeOXnuVib1JoshyHwgDGYlMjNicQVQ3doicic9icFoBkcA/640?wx_fmt=png&amp;from=appmsg"></figure><figure data-tool="mdnice编辑器"><img data-imgfileid="100001270" data-ratio="0.5694444444444444" data-src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676vUjSTSqhJXOpGdiaatD6Oe5dmVFQCQiblDIsc51Dq2gFMFCqSiaiatgsH83JY7yicJP5QZVlX6zbfGA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676vUjSTSqhJXOpGdiaatD6Oe5dmVFQCQiblDIsc51Dq2gFMFCqSiaiatgsH83JY7yicJP5QZVlX6zbfGA/640?wx_fmt=png&amp;from=appmsg"></figure><figure data-tool="mdnice编辑器"></figure><figure data-tool="mdnice编辑器"><span>区分正常和疾病组的数据库进行去卷积效果会优于整个数据库</span></figure><figure data-tool="mdnice编辑器"><br></figure><h2 data-tool="mdnice编辑器"><span>基因集评分</span></h2><pre data-tool="mdnice编辑器"><code>### 读取基因列表<br>geneset=read.table(<span>'../fatty.txt'</span>,header = T,check.names = F,sep = <span>'\t'</span>)<br>geneset=list(geneset$GT)<br>names(geneset)=<span>'GTs'</span><br>df_forGMT &lt;- data.frame(<br>  geneset=c(<span>"geneset"</span>,geneset$GTs)<br>)<br>write.table(t(df_forGMT),<span>"geneset.gmt"</span>,quote=F,sep=<span>"\t"</span>,row.names=T,col.names=F)<br>## <span>scMetabolsim<br><span>library</span><span>(scMetabolism)</span><br><span>library</span><span>(rsvd)</span><br>scRNA&lt;-sc.metabolism.<span>Seurat</span><span>(obj = scRNA, method = <span>"VISION"</span>, imputation = F, ncores = <span>1</span>, metabolism.type = <span>"KEGG"</span>)</span><br>DimPlot.<span>metabolism</span><span>(obj = scRNA, pathway = <span>"geneset"</span>, dimention.reduction.type = <span>"umap"</span>, dimention.reduction.run = F, size = <span>1</span>)</span><br>vision &lt;- as.data.<span>frame</span><span>(scRNA@assays$METABOLISM$score)</span><br>vision</span>=as.data.frame(t(vision))<br>## <span>AUCell<br><span>library</span><span>(AUCell)</span><br>cells_rankings &lt;- <span>AUCell_buildRankings</span><span>(scRNA@assays$RNA@data,  nCores=<span>1</span>, plotStats=TRUE)</span> <br>cells_AUC &lt;- <span>AUCell_calcAUC</span><span>(geneset, cells_rankings,nCores =<span>1</span>, aucMaxRank=nrow(cells_rankings)</span>*0.1)<br>aucs &lt;- as.<span>numeric</span><span>(getAUC(cells_AUC)</span>['GTs', ])<br>## Ucells &amp; singscore &amp; ssgsea<br><span>library</span><span>(irGSEA)</span><br>scRNA &lt;- irGSEA.<span>score</span><span>(object = scRNA, assay = <span>"RNA"</span>,<br>                      slot = <span>"data"</span>, seeds = <span>123</span>, ncores = <span>1</span>,msigdb = F,<br>                      custom = T, geneset = geneset,<br>                      method = c(<span>'UCell'</span>,<span>'singscore'</span>,<span>'ssgsea'</span>)</span>,<br>                      kcdf </span>= <span>'Gaussian'</span>)<br>uc=as.data.frame(scRNA<span>@assays</span>$UCell<span>@counts</span>)<br>uc=as.data.frame(t(uc))<br><br>singscore=as.data.frame(scRNA<span>@assays</span>$singscore<span>@counts</span>)<br>singscore=as.data.frame(t(singscore))<br><br>ssgsea=as.data.frame(scRNA<span>@assays</span>$ssgsea<span>@counts</span>)<br>ssgsea=as.data.frame(t(ssgsea))<br>## addmodulescore<br>scRNA=AddModuleScore(scRNA,features = geneset,name = <span>'Add'</span>)<br># 组合<br>score=data.frame(AUCell=aucs,UCell=uc$GTs,singscore=singscore$GTs,ssgsea=ssgsea$GTs,Add=scRNA$Add1,vision=vision$geneset)<br># scale 标准化<br>score&lt;- scale(score)<br># <span>0</span>-<span>1</span>标准化<br>normalize=function(x){<br>  <span>return</span>((x-min(x))/(max(x)-min(x)))}<br>score=apply(score, <span>2</span>, normalize)<br>score=as.data.frame(score)<br>score$Scoring=rowSums(score)<br>scRNA$Add1=NULL<br>scRNA<span>@meta</span>.data=cbind(scRNA<span>@meta</span>.data,score)<br># 可视化<br>DotPlot(scRNA,features = colnames(score)) + RotatedAxis() +<br>  theme(axis.text.x = element_text(angle = <span>45</span>, face=<span>"italic"</span>, hjust=<span>1</span>), axis.text.y = element_text(face=<span>"bold"</span>)) + theme(legend.position=<span>"right"</span>)  + labs(title = <span>"cluster markers"</span>, y = <span>""</span>, x=<span>""</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100001275" data-ratio="0.9824074074074074" data-src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676vUjSTSqhJXOpGdiaatD6OZiazDMoud5lcDKrckjgvelCLuLvMRLxB5G2LicgKICGTnCzVticGWic3wg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676vUjSTSqhJXOpGdiaatD6OZiazDMoud5lcDKrckjgvelCLuLvMRLxB5G2LicgKICGTnCzVticGWic3wg/640?wx_fmt=png&amp;from=appmsg"></figure><figure data-tool="mdnice编辑器">多种基因集评分手段比较<br></figure><pre data-tool="mdnice编辑器"><code># 关注细胞类型<br>scRNA_Hep &lt;- subset(scRNA,celltype==<span>"Hep3"</span>)<br>scRNA_Hep$FAT_group=ifelse(scRNA_Hep$Scoring &gt; <span>2.5</span>,<span>'FAT_highHEP'</span>,<span>'FAT_lowHEP'</span>)<br># 进行空间定位<br>Idents(scRNA_Hep) &lt;- <span>"FAT_group"</span><br>sc_counts &lt;- as.matrix(scRNA_Hep[[<span>'RNA'</span>]]<span>@counts</span>)<br>sc_nUMI = colSums(sc_counts)<br>cellType=data.frame(barcode=colnames(scRNA_Hep), celltype=scRNA_Hep$FAT_group)<br>names(cellType) = c(<span>'barcode'</span>, <span>'cell_type'</span>)<br>cell_types = cellType$cell_type; names(cell_types) &lt;- cellType$barcode<br>cell_types =as.factor(cell_types) <br>reference_FAT = Reference(sc_counts, cell_types, sc_nUMI)<br>result.RCTD=data.frame()<br><span>for</span>(i in slice){<br>  coords &lt;- stRNA<span>@images</span>[[i]]<span>@coordinates</span>[,c(<span>4</span>,<span>5</span>)]<br>  colnames(coords) &lt;- c(<span>"x"</span>,<span>"y"</span>)<br>  query&lt;- SpatialRNA(coords = coords, counts = sto.list[[i]]<span>@assays</span>$Spatial<span>@counts</span>, <br>                          nUMI = structure(sto.list[[i]]$nCount_Spatial, names=colnames(sto.list[[i]])))<br>  RCTD &lt;- create.RCTD(spatialRNA = query, reference = reference, max_cores = <span>10</span>)<br>  RCTD &lt;- run.RCTD(RCTD, doublet_mode = <span>'doublet'</span>)<br>  result_df &lt;- RCTD<span>@results</span>$results_df<br>  result_df &lt;- as.data.frame(result_df)<br>  result.RCTD=result.RCTD%&gt;%rbind(result_df)<br>}<br>#stRNA_FAT=stRNA[,rownames(result.RCTD)]<br>#stRNA_FAT &lt;- AddMetaData(stRNA_FAT,metadata = result.RCTD)<br>#Idents(stRNA)=stRNA$second_type<br>#DimPlot(stRNA)<br>#SpatialDimPlot(stRNA)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100001274" data-ratio="0.31574074074074077" data-src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676vUjSTSqhJXOpGdiaatD6OSqfS11Q23R9ic65MC4JEjMsRYFZ7iaCm7fiaBqF3ic2iaFvpYEAHansR9sg/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676vUjSTSqhJXOpGdiaatD6OSqfS11Q23R9ic65MC4JEjMsRYFZ7iaCm7fiaBqF3ic2iaFvpYEAHansR9sg/640?wx_fmt=png&amp;from=appmsg"></figure><pre data-tool="mdnice编辑器"><code><span># 可能是因为样本来源不同的原因，单细胞的发现无法在空间中定位</span></code><code><span>stRNA=AddModuleScore(stRNA,features = geneset,name = </span><span>'Add'</span><span>)</span></code><code>SpatialFeaturePlot(stRNA, features = <span>"Add1"</span>, pt.size.factor = <span>1</span>,<br>                   crop = FALSE, alpha = c(<span>0.1</span>,<span>1</span>), min.cutoff = <span>0</span>, max.cutoff = <span>1</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-imgfileid="100001276" data-ratio="0.27037037037037037" data-src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676vUjSTSqhJXOpGdiaatD6Odv8gGnSjYYtjtNaic7dicVzc4Ckv9r2ztdicuZwInSCrQd4P0TicXibz5jA/640?wx_fmt=png&amp;from=appmsg" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/Ze2f5C6nc676vUjSTSqhJXOpGdiaatD6Odv8gGnSjYYtjtNaic7dicVzc4Ckv9r2ztdicuZwInSCrQd4P0TicXibz5jA/640?wx_fmt=png&amp;from=appmsg"></figure></section><p><br></p><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/jmNw8yxqV3IsMeLAIJpubg",target="_blank" rel="noopener noreferrer">原文链接</a>
