---
title: "孟德尔随机化：下载IEU OPEN GWAS数据本地处理"
date: 2024-03-14T02:01:37Z
draft: ["false"]
tags: [
  "fetched",
  "医学和生信笔记"
]
categories: ["Acdemic"]
---
孟德尔随机化：下载IEU OPEN GWAS数据本地处理 by 医学和生信笔记
------
<div><section><span>关注公众号，发送</span><strong>R语言</strong><span>或</span><strong>python</strong><span>，可获取资料</span><span></span></section><section><mp-common-profile data-pluginname="mpprofile" data-id="MzUzOTQzNzU0NA==" data-headimg="http://mmbiz.qpic.cn/mmbiz_png/tpAC6lR84R9YDc8IDhqWAHTrZsMuhDpFlw4scqOl1ZVWpeY77cdibaSzPeGALfkEhdVpwHzVibHCRSYZg4csB43g/0?wx_fmt=png" data-nickname="医学和生信笔记" data-alias="yxhsxbj" data-signature="外科医生👨‍⚕️的R语言和生信学习🔖" data-from="2" data-weuitheme="light"></mp-common-profile></section><section data-role="outer" label="edit by 135editor"><section data-role="paragraph"><section data-role="outer" label="edit by 135editor"><section data-role="paragraph"><section data-role="outer" label="edit by 135editor"><section data-role="paragraph"><section data-role="outer"><section data-role="outer" label="edit by 135editor"><section data-tools="135编辑器" data-id="28"><p data-brushtype="text" hm_fix="440:185"><span>💡专注R语言在🩺生物医学中的使用</span></p></section></section></section></section></section><hr><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><p data-tool="mdnice编辑器"><span><span>设为“</span><strong><span>星标</span></strong><span>”，精彩不错过</span></span><br></p><hr><p data-tool="mdnice编辑器"><span><strong></strong></span></p><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><p data-tool="mdnice编辑器">之前的推文详细介绍了<code>TwoSampleMR</code>这个R包的用法，但是这个流程是<strong>在线获取数据</strong>的，有些朋友可能会因为网络问题没法获取成功，直接卡死在第一步！哎，感叹下国内的科研环境真是难！</p><p data-tool="mdnice编辑器">所以今天记录一下如何从IEU下载原始数据，然后在本地进行处理，获得和在线数据一样的结果。</p><p data-tool="mdnice编辑器">为了和在线的结果一致，暴露因素还是选择BMI，它的ID是：<code>ieu-a-2</code>，结局还是选择CHD(冠心病)，它的ID是：<code>ieu-a-7</code>。</p><p data-tool="mdnice编辑器">本文目录：</p><ul><li><p>IEU数据下载</p></li><li><p>准备R包</p></li><li><p>读取暴露数据</p></li><ul><li><p>根据P值过滤</p></li><li><p>去除连锁不平衡</p></li></ul><li><p>读取结局数据</p></li><li><p>进行MR分析</p></li></ul><p data-tool="mdnice编辑器"><br></p><h2 data-tool="mdnice编辑器"><span></span><span>IEU数据下载</span><span></span></h2><p data-tool="mdnice编辑器">打开以下网址：https://gwas.mrcieu.ac.uk/ ，点击<code>datasets</code>：</p><figure data-tool="mdnice编辑器"><img data-imgfileid="100017510" data-ratio="0.8009259259259259" data-type="png" data-w="1080" data-src="https://mmbiz.qpic.cn/mmbiz_png/tpAC6lR84RicozkIPmE9IUpYP6sbjNjXhXANnZiakTOwia9JVH4Jf63TVCVAXJYOLyPZXJ5Pv3fyXRZ5D4WzLyUVQ/640?wx_fmt=png&amp;from=appmsg" src="https://mmbiz.qpic.cn/mmbiz_png/tpAC6lR84RicozkIPmE9IUpYP6sbjNjXhXANnZiakTOwia9JVH4Jf63TVCVAXJYOLyPZXJ5Pv3fyXRZ5D4WzLyUVQ/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">先找暴露数据的SNP。</p><p data-tool="mdnice编辑器">在<code>GWAS ID</code>后面的框里输入<code>ieu-a-2</code>，然后点击<code>Filter</code>，在出来的结果中点击<code>ieu-a-2</code>这个结果：</p><figure data-tool="mdnice编辑器"><img data-imgfileid="100017509" data-ratio="0.8185185185185185" data-type="png" data-w="1080" data-src="https://mmbiz.qpic.cn/mmbiz_png/tpAC6lR84RicozkIPmE9IUpYP6sbjNjXhENC3UrokD0TYYicCkyUnqvyxHfA33XVn62Ktc7vvoYIgt93ibbufY1fA/640?wx_fmt=png&amp;from=appmsg" src="https://mmbiz.qpic.cn/mmbiz_png/tpAC6lR84RicozkIPmE9IUpYP6sbjNjXhENC3UrokD0TYYicCkyUnqvyxHfA33XVn62Ktc7vvoYIgt93ibbufY1fA/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">就可以来到下载界面，点击<code>Download VCF</code>即可下载暴露数据的完整的GWAS数据：</p><figure data-tool="mdnice编辑器"><img data-imgfileid="100017508" data-ratio="0.8009259259259259" data-type="png" data-w="1080" data-src="https://mmbiz.qpic.cn/mmbiz_png/tpAC6lR84RicozkIPmE9IUpYP6sbjNjXhWPkribpAjvmhDib8kPWWLP5oGZYgybRhnTWHvqnXkmPeY95zGBs60y2Q/640?wx_fmt=png&amp;from=appmsg" src="https://mmbiz.qpic.cn/mmbiz_png/tpAC6lR84RicozkIPmE9IUpYP6sbjNjXhWPkribpAjvmhDib8kPWWLP5oGZYgybRhnTWHvqnXkmPeY95zGBs60y2Q/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">下载结局数据也是完全一样的流程，就不重复演示了。我这里为了和前面的在线获取进行比较，结局也选择的是<code>ieu-a-7</code>。</p><p data-tool="mdnice编辑器">两个数据都下载后是这样的：</p><figure data-tool="mdnice编辑器"><div><p><img data-galleryid="" data-imgfileid="100017512" data-ratio="0.14153846153846153" data-s="300,640" data-type="png" data-w="325" data-src="https://mmbiz.qpic.cn/mmbiz_png/tpAC6lR84RicozkIPmE9IUpYP6sbjNjXhXAY0fibCJk0lH17cXKsmGwgxFKcWvnDqIYicG4syO3DwMERjibBOYzWiag/640?wx_fmt=png&amp;from=appmsg" src="https://mmbiz.qpic.cn/mmbiz_png/tpAC6lR84RicozkIPmE9IUpYP6sbjNjXhXAY0fibCJk0lH17cXKsmGwgxFKcWvnDqIYicG4syO3DwMERjibBOYzWiag/640?wx_fmt=png&amp;from=appmsg"><span></span></p></div></figure><p data-tool="mdnice编辑器">这样一个暴露的数据，一个结局的数据，都是VCF格式的，我们就下载好了。下面就是读取数据，进行处理，看看能不能得到和在线获取的数据一样的结果。</p><blockquote data-tool="mdnice编辑器"><p>这两个数据还是蛮大的，内存小的电脑可能会直接卡死。</p></blockquote><h2 data-tool="mdnice编辑器"><span></span><span>准备R包</span><span></span></h2><p data-tool="mdnice编辑器">读取数据及后续进行处理需要首先安装以下几个R包：</p><pre data-tool="mdnice编辑器"><code><span># 记得先改镜像</span><br><span>#options(BioC_mirror="https://mirrors.nju.edu.cn/bioconductor/")</span><br>BiocManager::install(<span>"VariantAnnotation"</span>)<br><br><span># 对网络有要求哦</span><br>remotes::install_github(<span>"MRCIEU/TwoSampleMR"</span>)<br>remotes::install_github(<span>"mrcieu/gwasvcf"</span>)<br>remotes::install_github(<span>"mrcieu/gwasglue"</span>)<br></code></pre><p data-tool="mdnice编辑器">加载一下：</p><pre data-tool="mdnice编辑器"><code><span>library</span>(gwasvcf)<br><span>library</span>(gwasglue)<br><span>library</span>(VariantAnnotation)<br><span>library</span>(TwoSampleMR)<br></code></pre><h2 data-tool="mdnice编辑器"><span></span><span>读取暴露数据</span><span></span></h2><p data-tool="mdnice编辑器">使用<code>VariantAnnotation</code>包进行读取，直接读，非常简单，这是官方的教程：</p><pre data-tool="mdnice编辑器"><code><span># 注意路径不要写错</span><br>vcf_exposure &lt;- VariantAnnotation::readVcf(<span>"./ieu/ieu-a-2.vcf.gz"</span>)<br>class(vcf_exposure)<br><span>## [1] "CollapsedVCF"</span><br><span>## attr(,"package")</span><br><span>## [1] "VariantAnnotation"</span><br><br>vcf_exposure<br><span>## class: CollapsedVCF </span><br><span>## dim: 2554285 1 </span><br><span>## rowRanges(vcf):</span><br><span>##   GRanges with 5 metadata columns: paramRangeID, REF, ALT, QUAL, FILTER</span><br><span>## info(vcf):</span><br><span>##   DataFrame with 1 column: AF</span><br><span>## info(header(vcf)):</span><br><span>##       Number Type  Description     </span><br><span>##    AF A      Float Allele Frequency</span><br><span>## geno(vcf):</span><br><span>##   List of length 9: ES, SE, LP, AF, SS, EZ, SI, NC, ID</span><br><span>## geno(header(vcf)):</span><br><span>##       Number Type   Description                                                </span><br><span>##    ES A      Float  Effect size estimate relative to the alternative allele    </span><br><span>##    SE A      Float  Standard error of effect size estimate                     </span><br><span>##    LP A      Float  -log10 p-value for effect estimate                         </span><br><span>##    AF A      Float  Alternate allele frequency in the association study        </span><br><span>##    SS A      Float  Sample size used to estimate genetic effect                </span><br><span>##    EZ A      Float  Z-score provided if it was used to derive the EFFECT and...</span><br><span>##    SI A      Float  Accuracy score of summary data imputation                  </span><br><span>##    NC A      Float  Number of cases used to estimate genetic effect            </span><br><span>##    ID 1      String Study variant identifier</span><br></code></pre><p data-tool="mdnice编辑器">这样就读取好了，这个函数的读取的结果默认其实是一个<code>S4</code>对象，关于这个对象我下次再详细探索（和<code>TCGAbiolinks</code>下载的<code>SummarisedExperiment</code>对象很像）。我们可以对这个对象进行一些探索，但是这个和今天的主题没有关系，所以就不介绍了，直接下一步。</p><h3 data-tool="mdnice编辑器"><span></span><span>根据P值过滤</span><span></span></h3><p data-tool="mdnice编辑器">下面是根据P值进行过滤，你可以先使用<code>query_gwas</code>进行过滤，再通过<code>gwasvcf_to_TwoSampleMR</code>转成<code>TwoSampleMR</code>需要的格式，如下所示：</p><pre data-tool="mdnice编辑器"><code>exposure_pval_filter &lt;- query_gwas(vcf = vcf_exposure, pval = <span>5e-8</span>)<br>exposure_pval_filter &lt;- gwasvcf_to_TwoSampleMR(vcf = exposure_pval_filter)<br>colnames(exposure_pval_filter)<br><span>##  [1] "chr.exposure"           "pos.exposure"           "other_allele.exposure" </span><br><span>##  [4] "effect_allele.exposure" "beta.exposure"          "se.exposure"           </span><br><span>##  [7] "pval.exposure"          "eaf.exposure"           "samplesize.exposure"   </span><br><span>## [10] "ncase.exposure"         "SNP"                    "ncontrol.exposure"     </span><br><span>## [13] "exposure"               "mr_keep.exposure"       "pval_origin.exposure"  </span><br><span>## [16] "id.exposure"</span><br>dim(exposure_pval_filter)<br><span>## [1] 2041   16</span><br><br>head(exposure_pval_filter)<br><span>##   chr.exposure pos.exposure other_allele.exposure effect_allele.exposure</span><br><span>## 1            1     47684677                     T                      G</span><br><span>## 2            1     47690438                     G                      T</span><br><span>## 3            1     49376820                     T                      C</span><br><span>## 4            1     49438005                     G                      A</span><br><span>## 5            1     49589847                     A                      G</span><br><span>## 6            1     49828663                     A                      G</span><br><span>##   beta.exposure se.exposure pval.exposure eaf.exposure samplesize.exposure</span><br><span>## 1       -0.0168      0.0030  2.181976e-08       0.5333              339152</span><br><span>## 2        0.0165      0.0030  3.416017e-08       0.4746              338820</span><br><span>## 3        0.0209      0.0035  2.371974e-09       0.2250              339110</span><br><span>## 4       -0.0225      0.0031  3.773115e-13       0.6333              338453</span><br><span>## 5       -0.0227      0.0031  2.123244e-13       0.5833              318585</span><br><span>## 6       -0.0213      0.0032  2.710816e-11       0.6667              339129</span><br><span>##   ncase.exposure       SNP ncontrol.exposure exposure mr_keep.exposure</span><br><span>## 1             NA  rs977747                NA  ieu-a-2             TRUE</span><br><span>## 2             NA rs2984618                NA  ieu-a-2             TRUE</span><br><span>## 3             NA rs1343424                NA  ieu-a-2             TRUE</span><br><span>## 4             NA rs3127553                NA  ieu-a-2             TRUE</span><br><span>## 5             NA  rs657452                NA  ieu-a-2             TRUE</span><br><span>## 6             NA rs7531656                NA  ieu-a-2             TRUE</span><br><span>##   pval_origin.exposure id.exposure</span><br><span>## 1             reported      HSY8sy</span><br><span>## 2             reported      HSY8sy</span><br><span>## 3             reported      HSY8sy</span><br><span>## 4             reported      HSY8sy</span><br><span>## 5             reported      HSY8sy</span><br><span>## 6             reported      HSY8sy</span><br></code></pre><p data-tool="mdnice编辑器">过滤后只剩下2041个了。</p><p data-tool="mdnice编辑器">或者也可以先转成<code>TwoSampleMR</code>需要的格式，再使用<code>subset</code>等函数自己过滤下。我选择第1种方法。</p><h3 data-tool="mdnice编辑器"><span></span><span>去除连锁不平衡</span><span></span></h3><p data-tool="mdnice编辑器">再下面就是去除连锁不平衡的数据。<code>extract_instruments</code>默认的<code>clump_kb=10000</code>，<code>clump_r2=0.001</code>，所以我这里也选择这个条件。</p><pre data-tool="mdnice编辑器"><code><span># 这一步也是联网进行的</span><br>exposure_clumped &lt;- clump_data(exposure_pval_filter, <br>                               clump_kb=<span>10000</span>, clump_r2=<span>0.001</span>,<br>                               pop = <span>"EUR"</span>)<span># 参考人种选欧洲人，默认值</span><br><span># 保存下</span><br><span>#save(exposure_clumped,file = "./ieu/exposure_clumped.rdata")</span><br>dim(exposure_clumped)<br><span>## [1] 78 16</span><br><br>head(exposure_clumped)<br><span>##     chr.exposure pos.exposure other_allele.exposure effect_allele.exposure</span><br><span>## 1              1     47684677                     T                      G</span><br><span>## 5              1     49589847                     A                      G</span><br><span>## 118            1     72837239                     T                      C</span><br><span>## 238            1     78048331                     T                      C</span><br><span>## 276            1     96924097                     C                      T</span><br><span>## 302            1    110082886                     C                      T</span><br><span>##     beta.exposure se.exposure pval.exposure eaf.exposure samplesize.exposure</span><br><span>## 1         -0.0168      0.0030  2.181976e-08       0.5333              339152</span><br><span>## 5         -0.0227      0.0031  2.123244e-13       0.5833              318585</span><br><span>## 118        0.0331      0.0030  1.880182e-28       0.6083              338123</span><br><span>## 238        0.0201      0.0031  4.567726e-11       0.4250              339065</span><br><span>## 276        0.0221      0.0030  1.433838e-13       0.5750              337797</span><br><span>## 302        0.0659      0.0087  5.059411e-14       0.0339              313621</span><br><span>##     ncase.exposure        SNP ncontrol.exposure exposure mr_keep.exposure</span><br><span>## 1               NA   rs977747                NA  ieu-a-2             TRUE</span><br><span>## 5               NA   rs657452                NA  ieu-a-2             TRUE</span><br><span>## 118             NA  rs7531118                NA  ieu-a-2             TRUE</span><br><span>## 238             NA rs17381664                NA  ieu-a-2             TRUE</span><br><span>## 276             NA rs11165643                NA  ieu-a-2             TRUE</span><br><span>## 302             NA  rs7550711                NA  ieu-a-2             TRUE</span><br><span>##     pval_origin.exposure id.exposure</span><br><span>## 1               reported      HSY8sy</span><br><span>## 5               reported      HSY8sy</span><br><span>## 118             reported      HSY8sy</span><br><span>## 238             reported      HSY8sy</span><br><span>## 276             reported      HSY8sy</span><br><span>## 302             reported      HSY8sy</span><br></code></pre><p data-tool="mdnice编辑器">这个<code>exposure_clumped</code>就是最终的暴露数据了，最后剩下78个SNP，这个数据和在线版基本一样（理论上应该是完全一模一样的，但是不知道为啥这里少了一个，你可以换其他数据试试看），而且通过和各列进行比较，发现共有的数据都是一样的。</p><p data-tool="mdnice编辑器">但是这一步其实也是联网进行的，所以如果你要本地LD的话，可以使用<code>ieugwasr::ld_clump</code>这个函数实现。</p><p data-tool="mdnice编辑器">首先需要下载LD的参考数据集到本地，下载地址是：http://fileserve.mrcieu.ac.uk/ld/1kg.v3.tgz，下载后解压（全部解压后是7G+的大小），得到以下数据：</p><figure data-tool="mdnice编辑器"><img data-imgfileid="100017507" data-ratio="2.925925925925926" data-type="png" data-w="108" data-src="https://mmbiz.qpic.cn/mmbiz_png/tpAC6lR84RicozkIPmE9IUpYP6sbjNjXhX5CIDlYw6VIO5PXJic6v501bOU5q1M5h37ZwWT9ZzZsd6Tx50VtfYUQ/640?wx_fmt=png&amp;from=appmsg" src="https://mmbiz.qpic.cn/mmbiz_png/tpAC6lR84RicozkIPmE9IUpYP6sbjNjXhX5CIDlYw6VIO5PXJic6v501bOU5q1M5h37ZwWT9ZzZsd6Tx50VtfYUQ/640?wx_fmt=png&amp;from=appmsg"></figure><p data-tool="mdnice编辑器">我们这个数据参考人种是欧洲人，所以会用到<code>EUR.bed</code>、<code>EUR.bim</code>、<code>EUR.fam</code>这3个文件。</p><p data-tool="mdnice编辑器">然后再下载一个<code>plink.exe</code>，下载地址是：https://github.com/explodecomputer/genetics.binaRies/raw/master/binaries/Windows/plink.exe</p><p data-tool="mdnice编辑器">然后就可以运行以下代码实现本地化的LD了：</p><pre data-tool="mdnice编辑器"><code>test &lt;- ld_clump(<br>  <span># 只能是这3列，且列名必须是下面这样的</span><br>  dat = dplyr::tibble(rsid=exposure_pval_filter$SNP,<br>                      pval=exposure_pval_filter$pval.exposure,<br>                      id=exposure_pval_filter$exposure),<br>  clump_kb=<span>10000</span>, clump_r2=<span>0.001</span>,clump_p = <span>1</span>,<br>  bfile = <span>"ieu/1kg/EUR"</span>, <span># 参考数据集，路径不要写错，只要写前缀EUR即可</span><br>  plink_bin = <span>"ieu/plink.exe"</span> <span># plink.exe的存放路径不要写错</span><br>  )<br><br>dim(test)<br><span>## [1] 78  3</span><br>head(test)<br><span>## # A tibble: 6 × 3</span><br><span>##   rsid           pval id     </span><br><span>##   &lt;chr&gt;         &lt;dbl&gt; &lt;chr&gt;  </span><br><span>## 1 rs977747   2.18e- 8 ieu-a-2</span><br><span>## 2 rs657452   2.12e-13 ieu-a-2</span><br><span>## 3 rs7531118  1.88e-28 ieu-a-2</span><br><span>## 4 rs17381664 4.57e-11 ieu-a-2</span><br><span>## 5 rs11165643 1.43e-13 ieu-a-2</span><br><span>## 6 rs7550711  5.06e-14 ieu-a-2</span><br></code></pre><p data-tool="mdnice编辑器">结果也是78个，和在线版的是一模一样的，但是结果只有3列，但是只要只选择上面78个SNP就好了：</p><pre data-tool="mdnice编辑器"><code><span>library</span>(dplyr)<br>exposure_local_ld &lt;- exposure_pval_filter %&gt;% <br>  filter(SNP %<span>in</span>% test$rsid)<br>dim(exposure_local_ld)<br><span>## [1] 78 16</span><br>head(exposure_local_ld)<br><span>##   chr.exposure pos.exposure other_allele.exposure effect_allele.exposure</span><br><span>## 1            1     47684677                     T                      G</span><br><span>## 2            1     49589847                     A                      G</span><br><span>## 3            1     72837239                     T                      C</span><br><span>## 4            1     78048331                     T                      C</span><br><span>## 5            1     96924097                     C                      T</span><br><span>## 6            1    110082886                     C                      T</span><br><span>##   beta.exposure se.exposure pval.exposure eaf.exposure samplesize.exposure</span><br><span>## 1       -0.0168      0.0030  2.181976e-08       0.5333              339152</span><br><span>## 2       -0.0227      0.0031  2.123244e-13       0.5833              318585</span><br><span>## 3        0.0331      0.0030  1.880182e-28       0.6083              338123</span><br><span>## 4        0.0201      0.0031  4.567726e-11       0.4250              339065</span><br><span>## 5        0.0221      0.0030  1.433838e-13       0.5750              337797</span><br><span>## 6        0.0659      0.0087  5.059411e-14       0.0339              313621</span><br><span>##   ncase.exposure        SNP ncontrol.exposure exposure mr_keep.exposure</span><br><span>## 1             NA   rs977747                NA  ieu-a-2             TRUE</span><br><span>## 2             NA   rs657452                NA  ieu-a-2             TRUE</span><br><span>## 3             NA  rs7531118                NA  ieu-a-2             TRUE</span><br><span>## 4             NA rs17381664                NA  ieu-a-2             TRUE</span><br><span>## 5             NA rs11165643                NA  ieu-a-2             TRUE</span><br><span>## 6             NA  rs7550711                NA  ieu-a-2             TRUE</span><br><span>##   pval_origin.exposure id.exposure</span><br><span>## 1             reported      HSY8sy</span><br><span>## 2             reported      HSY8sy</span><br><span>## 3             reported      HSY8sy</span><br><span>## 4             reported      HSY8sy</span><br><span>## 5             reported      HSY8sy</span><br><span>## 6             reported      HSY8sy</span><br></code></pre><h2 data-tool="mdnice编辑器"><span></span><span>读取结局数据</span><span></span></h2><p data-tool="mdnice编辑器">和读取暴露数据完全一样的步骤，就不多做说明了。</p><pre data-tool="mdnice编辑器"><code><span># 释放下内存</span><br>rm(list = ls());gc()<br><span>##            used  (Mb) gc trigger   (Mb)  max used   (Mb)</span><br><span>## Ncells 10747449 574.0   57439848 3067.7  43933768 2346.4</span><br><span>## Vcells 19903045 151.9  118845657  906.8 148489984 1132.9</span><br><br>vcf_outcome &lt;- readVcf(<span>"./ieu/ieu-a-7.vcf.gz"</span>)<br>outcome_origin &lt;- gwasvcf_to_TwoSampleMR(vcf = vcf_outcome)<br><br>rm(vcf_outcome);gc()<br><span>##             used   (Mb) gc trigger   (Mb)  max used   (Mb)</span><br><span>## Ncells  19231085 1027.1  130298919 6958.8 162873648 8698.4</span><br><span>## Vcells 167142346 1275.2  788577285 6016.4 870562547 6641.9</span><br></code></pre><p data-tool="mdnice编辑器"><code>extract_outcome_data</code>是从暴露数据的SNP中挑选结局相关的数据，相当于取个交集，所以我们下面也先对暴露数据和结局数据取交集，获取共同的SNP。</p><pre data-tool="mdnice编辑器"><code>load(file = <span>"./ieu/exposure_clumped.rdata"</span>)<br><span># 取交集</span><br>common_snp &lt;- merge(exposure_clumped,outcome_origin,by=<span>"SNP"</span>)$SNP<br>length(common_snp)<br><span>## [1] 78</span><br></code></pre><p data-tool="mdnice编辑器">然后对结局数据使用<code>format_data</code>函数修改格式，提取结局数据：</p><pre data-tool="mdnice编辑器"><code>outcome_data &lt;- format_data(dat = outcome_origin,<br>                      type = <span>"outcome"</span>, <span># 类型是outcome</span><br>                      snps = common_snp, <span># 共有的SNP</span><br>                      snp_col = <span>"SNP"</span>,<br>                      beta_col = <span>"beta.exposure"</span>,<br>                      se_col = <span>"se.exposure"</span>,<br>                      eaf_col = <span>"eaf.exposure"</span>,<br>                      effect_allele_col = <span>"effect_allele.exposure"</span>,<br>                      other_allele_col = <span>"other_allele.exposure"</span>,<br>                      pval_col = <span>"pval.exposure"</span>,<br>                      samplesize_col = <span>"samplesize.exposure"</span>,<br>                      ncase_col = <span>"ncase.exposure"</span>,<br>                      ncontrol_col = <span>"ncontrol.exposure"</span>,<br>                      id_col = <span>"id.exposure"</span><br>                      )<br></code></pre><p data-tool="mdnice编辑器">这个<code>outcome_data</code>就是我们最终的结局数据了，和在线版获取的也是基本一致哦：</p><pre data-tool="mdnice编辑器"><code>dim(outcome_data)<br><span>## [1] 78 14</span><br>head(outcome_data)<br><span>##   other_allele.outcome effect_allele.outcome beta.outcome se.outcome</span><br><span>## 1                    T                     G    -0.013896  0.0096146</span><br><span>## 2                    A                     G    -0.007670  0.0093844</span><br><span>## 3                    T                     C     0.017675  0.0097331</span><br><span>## 4                    T                     C     0.016188  0.0102099</span><br><span>## 5                    C                     T     0.005150  0.0092844</span><br><span>## 6                    C                     T    -0.048354  0.0305635</span><br><span>##   pval.outcome eaf.outcome samplesize.outcome ncase.outcome        SNP</span><br><span>## 1   0.14837509    0.550761             184305            NA   rs977747</span><br><span>## 2   0.41374714    0.566124             184305            NA   rs657452</span><br><span>## 3   0.06937452    0.480633             184305            NA  rs7531118</span><br><span>## 4   0.11284907    0.349349             184305            NA rs17381664</span><br><span>## 5   0.57910458    0.553284             184305            NA rs11165643</span><br><span>## 6   0.11363078    0.027650             184305            NA  rs7550711</span><br><span>##   ncontrol.outcome id.outcome outcome mr_keep.outcome pval_origin.outcome</span><br><span>## 1               NA     FUNiWE outcome            TRUE            reported</span><br><span>## 2               NA     FUNiWE outcome            TRUE            reported</span><br><span>## 3               NA     FUNiWE outcome            TRUE            reported</span><br><span>## 4               NA     FUNiWE outcome            TRUE            reported</span><br><span>## 5               NA     FUNiWE outcome            TRUE            reported</span><br><span>## 6               NA     FUNiWE outcome            TRUE            reported</span><br></code></pre><p data-tool="mdnice编辑器">最后修改下暴露因素和结局的名称：</p><pre data-tool="mdnice编辑器"><code>outcome_data$id.outcome &lt;- <span>"CHD"</span><br>exposure_clumped$id.exposure &lt;- <span>"BMI"</span><br></code></pre><h2 data-tool="mdnice编辑器"><span></span><span>进行MR分析</span><span></span></h2><p data-tool="mdnice编辑器">暴露数据和结局数据都有了，下面就是协调下数据，然后就可以进行MR分析了，后面的步骤就和之前一模一样了！</p><pre data-tool="mdnice编辑器"><code>dat_har &lt;- harmonise_data(exposure_clumped, outcome_data)<br><br>res &lt;- mr(dat_har)<br></code></pre><p data-tool="mdnice编辑器">在往后的分析就不重复了。</p></section><p><span><br></span></p><hr><blockquote><p><span><strong>联系我们，关注我们</strong></span></p><ol><li><section>免费QQ交流群1：613637742</section></li><li><section>免费QQ交流群2：608720452</section></li><li><section>公众号消息界面关于作者获取联系方式</section></li><li><section>知乎、CSDN、简书同名账号</section></li><li><section>哔哩哔哩：阿越就是我</section></li></ol></blockquote></section></section></section></section></section><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/Acw_YTw5kAlnFUgMypOJNA",target="_blank" rel="noopener noreferrer">原文链接</a>
