---
title: "ScType代码实操：快来试试这款全自动的单细胞细胞类型工具吧"
date: 2023-09-27T03:33:58Z
draft: ["false"]
tags: [
  "fetched",
  "生信宝库"
]
categories: ["Acdemic"]
---
ScType代码实操：快来试试这款全自动的单细胞细胞类型工具吧 by 生信宝库
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><h2 data-tool="mdnice编辑器"><span></span><span>前言</span></h2><p data-tool="mdnice编辑器">在分析单细胞数据时，细胞类型注释是最基础也最关键的步骤之一，相比于耗时且对生物学知识要求较高的纯人工细胞注释，使用准确的自动细胞注释工具不失为一种较好的选择。在现实操作中，很多研究者更倾向于先使用自动软件鉴定出大群，然后联合人工注释，这样才不失为一种最好的方式。</p><p data-tool="mdnice编辑器">在前面的推文：<a href="https://mp.weixin.qq.com/s?__biz=MzI4MjY5ODI1Nw==&amp;mid=2247489261&amp;idx=1&amp;sn=1399c0c7c8d58cebeebbe7abacdc8dc0&amp;chksm=eb94a193dce328855047c83a4842b17a6f577f13929935b0aed43ad363e244c414cfb536f649&amp;token=227123296&amp;lang=zh_CN&amp;scene=21#wechat_redirect" data-linktype="2">ScType--超快准狠的全自动scRNA-seq数据细胞类型注释工具</a>中，Immugent介绍了ScType的功能框架及其相对于其它自动注释软件的优势。那么本期推文，Immugent将通过代码实操的方式来介绍ScType的使用技巧。话不多说，让我们一起来学习一下如何使用ScType对 scRNA-seq 数据进行全自动的细胞类型注释吧。</p><p data-tool="mdnice编辑器"><br></p><hr data-tool="mdnice编辑器"><h2 data-tool="mdnice编辑器"><span></span><span>代码流程</span></h2><h4 data-tool="mdnice编辑器"><span></span>1. 数据准备<span></span></h4><p data-tool="mdnice编辑器">首先让我们加载来自seurat官网的 PBMC 3k 示例数据集。</p><p data-tool="mdnice编辑器">数据下载：https://cf.10xgenomics.com/samples/cell/pbmc3k/pbmc3k_filtered_gene_bc_matrices.tar.gz</p><pre data-tool="mdnice编辑器"><code><span># load libraries</span><br>lapply(c(<span>"dplyr"</span>,<span>"Seurat"</span>,<span>"HGNChelper"</span>), library, character.only = T)<br><br><span># Load the PBMC dataset</span><br>pbmc.data &lt;- Read10X(data.dir = <span>"./filtered_gene_bc_matrices/hg19/"</span>)<br><span># Initialize the Seurat object with the raw (non-normalized data).</span><br>pbmc &lt;- CreateSeuratObject(counts = pbmc.data, project = <span>"pbmc3k"</span>, min.cells = 3, min.features = 200)<br><br><span># normalize data</span><br>pbmc[[<span>"percent.mt"</span>]] &lt;- PercentageFeatureSet(pbmc, pattern = <span>"^MT-"</span>)<br><span># pbmc &lt;- subset(pbmc, subset = nFeature_RNA &gt; 200 &amp; nFeature_RNA &lt; 2500 &amp; percent.mt &lt; 5) # make some filtering based on QC metrics visualizations, see Seurat tutorial: https://satijalab.org/seurat/articles/pbmc3k_tutorial.html</span><br>pbmc &lt;- NormalizeData(pbmc, normalization.method = <span>"LogNormalize"</span>, scale.factor = 10000)<br>pbmc &lt;- FindVariableFeatures(pbmc, selection.method = <span>"vst"</span>, nfeatures = 2000)<br><br><span># scale and run PCA</span><br>pbmc &lt;- ScaleData(pbmc, features = rownames(pbmc))<br>pbmc &lt;- RunPCA(pbmc, features = VariableFeatures(object = pbmc))<br><br><span># Check number of PC components (we selected 10 PCs for downstream analysis, based on Elbow plot)</span><br>ElbowPlot(pbmc)<br><br><span># cluster and visualize</span><br>pbmc &lt;- FindNeighbors(pbmc, dims = 1:10)<br>pbmc &lt;- FindClusters(pbmc, resolution = 0.8)<br>pbmc &lt;- RunUMAP(pbmc, dims = 1:10)<br>DimPlot(pbmc, reduction = <span>"umap"</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-backh="350" data-backw="537" data-ratio="0.6522842639593909" data-src="https://mmbiz.qpic.cn/mmbiz_png/GL6g5Y3aR7dzqeJbHdFLFiaF9JuDZSiawO0St9vXUTvlDzkWz3jiaUBF8tibkokqvZfl5icwiaOM303Oiavu0HKtiaicg1g/640?wx_fmt=png" data-type="png" data-w="788" src="https://mmbiz.qpic.cn/mmbiz_png/GL6g5Y3aR7dzqeJbHdFLFiaF9JuDZSiawO0St9vXUTvlDzkWz3jiaUBF8tibkokqvZfl5icwiaOM303Oiavu0HKtiaicg1g/640?wx_fmt=png"></figure><h4 data-tool="mdnice编辑器"><span></span>2. 细胞类型注释<span></span></h4><pre data-tool="mdnice编辑器"><code><span># 首先加载两个相关函数</span><br><span># load gene set preparation function</span><br><span>source</span>(<span>"https://raw.githubusercontent.com/IanevskiAleksandr/sc-type/master/R/gene_sets_prepare.R"</span>)<br><span># load cell type annotation function</span><br><span>source</span>(<span>"https://raw.githubusercontent.com/IanevskiAleksandr/sc-type/master/R/sctype_score_.R"</span>)<br><br><span># 接下来，从输入的细胞标记文件中准备基因集。默认情况使用内置的细胞标记 DB。</span><br><span># 但ScType支持使用自己的数据。只要准备一个格式与ScType数据库文件相同的 XLSX 文件。</span><br><span># DB 文件应该包含四列(组织类型-组织类型，cellName-细胞类型，基因 Symbolmore 1阳性标记基因，基因 Symbolmore 2-阴性标记基因)</span><br><br><span># DB file</span><br>db_ = <span>"https://raw.githubusercontent.com/IanevskiAleksandr/sc-type/master/ScTypeDB_full.xlsx"</span>;<br><span># 提供数据所属的组织类型</span><br>tissue = <span>"Immune system"</span> <span># e.g. Immune system,Pancreas,Liver,Eye,Kidney,Brain,Lung,Adrenal,Heart,Intestine,Muscle,Placenta,Spleen,Stomach,Thymus </span><br><br><span># prepare gene sets</span><br>gs_list = gene_sets_prepare(db_, tissue)<br><br><span># get cell-type by cell matrix</span><br>es.max = sctype_score(scRNAseqData = pbmc[[<span>"RNA"</span>]]@scale.data, scaled = TRUE, <br>                      gs = gs_list<span>$gs_positive</span>, gs2 = gs_list<span>$gs_negative</span>) <br><br><span># <span>NOTE:</span> scRNAseqData parameter should correspond to your input scRNA-seq matrix. </span><br><span># In case Seurat is used, it is either pbmc[["RNA"]]@scale.data (default), pbmc[["SCT"]]@scale.data, in case sctransform is used for normalization,</span><br><span># or pbmc[["integrated"]]@scale.data, in case a joint analysis of multiple single-cell datasets is performed.</span><br><br><span># merge by cluster</span><br>cL_resutls = do.call(<span>"rbind"</span>, lapply(unique(pbmc@meta.data<span>$seurat_clusters</span>), <span>function</span>(cl){<br>    es.max.cl = sort(rowSums(es.max[ ,rownames(pbmc@meta.data[pbmc@meta.data<span>$seurat_clusters</span>==cl, ])]), decreasing = !0)<br>    head(data.frame(cluster = cl, <span>type</span> = names(es.max.cl), scores = es.max.cl, ncells = sum(pbmc@meta.data<span>$seurat_clusters</span>==cl)), 10)<br>}))<br>sctype_scores = cL_resutls %&gt;% group_by(cluster) %&gt;% top_n(n = 1, wt = scores)  <br><br><span># set low-confident (low ScType score) clusters to "unknown"</span><br>sctype_scores<span>$type</span>[as.numeric(as.character(sctype_scores<span>$scores</span>)) &lt; sctype_scores<span>$ncells</span>/4] = <span>"Unknown"</span><br><span>print</span>(sctype_scores[,1:3])<br><span># A tibble: 11 × 3</span><br><span># Groups:   cluster [11]</span><br>   cluster <span>type</span>                    scores<br>   &lt;fct&gt;   &lt;chr&gt;                    &lt;dbl&gt;<br> 1 1       Memory CD4+ T cells      498. <br> 2 2       Naive B cells           1172. <br> 3 7       Non-classical monocytes  532. <br> 4 8       Natural killer  cells    631. <br> 5 3       CD8+ NKT-like cells      573. <br> 6 6       Naive CD8+ T cells        72.8<br> 7 0       Naive CD8+ T cells       444. <br> 8 4       Classical Monocytes      680. <br> 9 5       Classical Monocytes      574. <br>10 9       Myeloid Dendritic cells  181. <br>11 10      Platelets                284. <br><span># sctype_score 函数通过 gs 和 gs2参数接受正标记和负标记。</span><br><span># 如果没有负标记(即标记提供证据证明细胞具有特定的细胞类型) ，只需将 gs2参数设置为 NULL (即gs2 = NULL)。</span><br></code></pre><h4 data-tool="mdnice编辑器"><span></span>3. 可视化<span></span></h4><pre data-tool="mdnice编辑器"><code><span># 绘制UMAP图</span><br>pbmc@meta.data<span>$customclassif</span> = <span>""</span><br><span>for</span>(j <span>in</span> unique(sctype_scores<span>$cluster</span>)){<br>  cl_type = sctype_scores[sctype_scores<span>$cluster</span>==j,]; <br>  pbmc@meta.data<span>$customclassif</span>[pbmc@meta.data<span>$seurat_clusters</span> == j] = as.character(cl_type<span>$type</span>[1])<br>}<br><br>DimPlot(pbmc, reduction = <span>"umap"</span>, label = TRUE, repel = TRUE, group.by = <span>'customclassif'</span>)        <br><br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.6640926640926641" data-src="https://mmbiz.qpic.cn/mmbiz_png/GL6g5Y3aR7dzqeJbHdFLFiaF9JuDZSiawOWY7a2b8qVcJuspkoUxoViaLIEZUKQyMXzRCibcLuD8Rulf5k09qa7bpQ/640?wx_fmt=png" data-type="png" data-w="777" src="https://mmbiz.qpic.cn/mmbiz_png/GL6g5Y3aR7dzqeJbHdFLFiaF9JuDZSiawOWY7a2b8qVcJuspkoUxoViaLIEZUKQyMXzRCibcLuD8Rulf5k09qa7bpQ/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">在这里，还可以通过气泡图显示 ScType 注释的所有细胞类型。</p><pre data-tool="mdnice编辑器"><code><span># load libraries</span><br>lapply(c(<span>"ggraph"</span>,<span>"igraph"</span>,<span>"tidyverse"</span>, <span>"data.tree"</span>), library, character.only = T)<br><br><span># prepare edges</span><br>cL_resutls=cL_resutls[order(cL_resutls<span>$cluster</span>),]; edges = cL_resutls; edges<span>$type</span> = paste0(edges<span>$type</span>,<span>"_"</span>,edges<span>$cluster</span>); edges<span>$cluster</span> = paste0(<span>"cluster "</span>, edges<span>$cluster</span>); edges = edges[,c(<span>"cluster"</span>, <span>"type"</span>)]; colnames(edges) = c(<span>"from"</span>, <span>"to"</span>); rownames(edges) &lt;- NULL<br><br><span># prepare nodes</span><br>nodes_lvl1 = sctype_scores[,c(<span>"cluster"</span>, <span>"ncells"</span>)]; nodes_lvl1<span>$cluster</span> = paste0(<span>"cluster "</span>, nodes_lvl1<span>$cluster</span>); nodes_lvl1<span>$Colour</span> = <span>"#f1f1ef"</span>; nodes_lvl1<span>$ord</span> = 1; nodes_lvl1<span>$realname</span> = nodes_lvl1<span>$cluster</span>; nodes_lvl1 = as.data.frame(nodes_lvl1); nodes_lvl2 = c(); <br>ccolss= c(<span>"#5f75ae"</span>,<span>"#92bbb8"</span>,<span>"#64a841"</span>,<span>"#e5486e"</span>,<span>"#de8e06"</span>,<span>"#eccf5a"</span>,<span>"#b5aa0f"</span>,<span>"#e4b680"</span>,<span>"#7ba39d"</span>,<span>"#b15928"</span>,<span>"#ffff99"</span>, <span>"#6a3d9a"</span>,<span>"#cab2d6"</span>,<span>"#ff7f00"</span>,<span>"#fdbf6f"</span>,<span>"#e31a1c"</span>,<span>"#fb9a99"</span>,<span>"#33a02c"</span>,<span>"#b2df8a"</span>,<span>"#1f78b4"</span>,<span>"#a6cee3"</span>)<br><span>for</span> (i <span>in</span> 1:length(unique(cL_resutls<span>$cluster</span>))){<br>  dt_tmp = cL_resutls[cL_resutls<span>$cluster</span> == unique(cL_resutls<span>$cluster</span>)[i], ]; nodes_lvl2 = rbind(nodes_lvl2, data.frame(cluster = paste0(dt_tmp<span>$type</span>,<span>"_"</span>,dt_tmp<span>$cluster</span>), ncells = dt_tmp<span>$scores</span>, Colour = ccolss[i], ord = 2, realname = dt_tmp<span>$type</span>))<br>}<br>nodes = rbind(nodes_lvl1, nodes_lvl2); nodes<span>$ncells</span>[nodes<span>$ncells</span>&lt;1] = 1;<br>files_db = openxlsx::read.xlsx(db_)[,c(<span>"cellName"</span>,<span>"shortName"</span>)]; files_db = unique(files_db); nodes = merge(nodes, files_db, all.x = T, all.y = F, by.x = <span>"realname"</span>, by.y = <span>"cellName"</span>, sort = F)<br>nodes<span>$shortName</span>[is.na(nodes<span>$shortName</span>)] = nodes<span>$realname</span>[is.na(nodes<span>$shortName</span>)]; nodes = nodes[,c(<span>"cluster"</span>, <span>"ncells"</span>, <span>"Colour"</span>, <span>"ord"</span>, <span>"shortName"</span>, <span>"realname"</span>)]<br><br>mygraph &lt;- graph_from_data_frame(edges, vertices=nodes)<br><br><span># Make the graph</span><br>gggr&lt;- ggraph(mygraph, layout = <span>'circlepack'</span>, weight=I(ncells)) + <br>  geom_node_circle(aes(filter=ord==1,fill=I(<span>"#F5F5F5"</span>), colour=I(<span>"#D3D3D3"</span>)), alpha=0.9) + geom_node_circle(aes(filter=ord==2,fill=I(Colour), colour=I(<span>"#D3D3D3"</span>)), alpha=0.9) +<br>  theme_void() + geom_node_text(aes(filter=ord==2, label=shortName, colour=I(<span>"#ffffff"</span>), fill=<span>"white"</span>, repel = !1, parse = T, size = I(<span>log</span>(ncells,25)*1.5)))+ geom_node_label(aes(filter=ord==1,  label=shortName, colour=I(<span>"#000000"</span>), size = I(3), fill=<span>"white"</span>, parse = T), repel = !0, segment.linetype=<span>"dotted"</span>)<br>library(cowplot)<br>library(gridExtra)<br>library(patchwork)<br>p1 &lt;- DimPlot(pbmc, reduction = <span>"umap"</span>, label = TRUE, repel = TRUE, cols = ccolss)<br>p_combined = p1 + gggr + plot_layout(ncol = 2)<br><span>print</span>(p_combined)<br><br></code></pre><p><img data-galleryid="" data-ratio="0.525" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/GL6g5Y3aR7e46lVwmlox5ibrhrqOf9R7tw5EmYaq4GFEYneOzS1tY5jMjlFMyL54WeWT0G6vUEEVq29fwVriaKeg/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/GL6g5Y3aR7e46lVwmlox5ibrhrqOf9R7tw5EmYaq4GFEYneOzS1tY5jMjlFMyL54WeWT0G6vUEEVq29fwVriaKeg/640?wx_fmt=png"></p><p data-tool="mdnice编辑器">外部(灰色)气泡对应于每个簇(气泡越大，簇中的细胞越多) ，而内部气泡对应于每个簇所考虑的细胞类型，最大的气泡对应于指定的细胞类型。</p><p data-tool="mdnice编辑器"><br></p><hr data-tool="mdnice编辑器"><h2 data-tool="mdnice编辑器"><span></span><span>小结</span></h2><p data-tool="mdnice编辑器">在本期推文中，Immugent介绍了如何使用ScType对scRNA-seq数据进行细胞注释。我们可以看出ScType使用起来十分轻便，对新手十分友好。在具体进行细胞注释的时候，<span>我们可以通过</span>输入相应的组织信息以及阳性标记基因和阴性标记基因，这使得细胞注释的结果是比较准确的。因此，在大家手动注释遇到瓶颈时不妨来试一试这个高效且准确的细胞类型注释工具吧。</p><p data-tool="mdnice编辑器">好啦，本期的分享到这里就结束了，我们下期再会~</p><p data-tool="mdnice编辑器"><br></p><hr data-tool="mdnice编辑器"></section><p><img data-ratio="1" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/GL6g5Y3aR7eUr3zZytdDPl7kPJBscWxTDlwbSxRjmMoYpSCSmlGicFibnLH3pvictSibFwekOaooibv9Ria0zSCrC2icg/640?wx_fmt=jpeg" data-type="jpeg" data-w="344" src="https://mmbiz.qpic.cn/mmbiz_jpg/GL6g5Y3aR7eUr3zZytdDPl7kPJBscWxTDlwbSxRjmMoYpSCSmlGicFibnLH3pvictSibFwekOaooibv9Ria0zSCrC2icg/640?wx_fmt=jpeg"></p><h3 data-tool="mdnice编辑器"><span>关注下方公众号下回更新不迷路</span></h3><section><br></section><section><mp-common-profile data-pluginname="mpprofile" data-weui-theme="light" data-id="MzI4MjY5ODI1Nw==" data-headimg="http://mmbiz.qpic.cn/mmbiz_png/GL6g5Y3aR7f0yanILMQZCnw1duTMROQRvgDqVjlYcrljTRy1E4ZLppLG6zicdd3h0IwjLpxnum1V5KsowibJM1sw/0?wx_fmt=png" data-nickname="生信宝库" data-alias="sxbk2020" data-signature="本公众号只用于生信知识的收集与传播，以及生信人之间互相交流和学习，不会涉及任何商业利益。本公众号各小编平时忙于科研，更新文章较其它同类型公众号较慢，但保持宁缺毋滥的本心，只更新对大家有用的推文。" data-from="2" data-is_biz_ban="0"></mp-common-profile></section><p><br></p><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/7EeNycJg172uVLw5m_8YRQ",target="_blank" rel="noopener noreferrer">原文链接</a>
