---
title: "tsRNA分类分析：seqac包"
date: 2023-09-03T11:37:58Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Acdemic"]
---
tsRNA分类分析：seqac包 by 生信菜鸟团
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><p data-tool="mdnice编辑器"><span>tRNA的二级结构如下：</span><br></p><figure data-tool="mdnice编辑器"><img data-ratio="0.5382775119617225" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9NibCbHgDb9PQKkUNS8ib0w5hFZomLLPQF0NGgGricpY4Grs9sde9PwU6RMlwQ5VcCszeSpWl8ibSLCQ/640?wx_fmt=png" data-type="png" data-w="836" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9NibCbHgDb9PQKkUNS8ib0w5hFZomLLPQF0NGgGricpY4Grs9sde9PwU6RMlwQ5VcCszeSpWl8ibSLCQ/640?wx_fmt=png"><figcaption>image-20230903154931207</figcaption></figure><p data-tool="mdnice编辑器">根据其二级结构，tsRNA分类如下：</p><p data-tool="mdnice编辑器">早期研究报道，tiRNAs通过血管生成素（ANG）切割成熟tRNA产生，而tRFs来源于Dicer或ANG切割成熟tRNA或tRNA前体。根据成熟或前体tRNA转录本中的裂解位点，可以分为以下几类：</p><ul data-tool="mdnice编辑器"><li><section><p><strong>tRF-1</strong>：来源于前体tRNA，在3' trailer部由RNase Z or ELAC2酶裂解产生。</p></section></li><li><section><p><strong>tRF-3</strong>：来自成熟体tRNA的3'端</p></section></li><li><section><p><strong>tRF-5</strong>：来自成熟体tRNA的5'端</p></section></li><li><section><p><strong>tRF-i</strong>：来自成熟体tRNA的中间区域</p></section></li><li><section><p><strong>tiRNAs</strong>：are generated by specific cleavage by angiogenin in the anticodon loops of mature tRNAs</p></section></li></ul><figure data-tool="mdnice编辑器"><img data-ratio="1.0185810810810811" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9NibCbHgDb9PQKkUNS8ib0w5bl9gcumgZzmtTSviaNtUMXkoR9fhZXdibqVEaRXytFKicibrn6ZX6EHzTw/640?wx_fmt=png" data-type="png" data-w="592" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9NibCbHgDb9PQKkUNS8ib0w5bl9gcumgZzmtTSviaNtUMXkoR9fhZXdibqVEaRXytFKicibrn6ZX6EHzTw/640?wx_fmt=png"><figcaption>image-20230619222644080</figcaption></figure><p data-tool="mdnice编辑器">今天主要给大家介绍一个软件，可以根据tsRNA的序列以及tRNA的二级结构文件来鉴定tsRNA的分类，软件为Seqpac。</p><p data-tool="mdnice编辑器">Seqpac主要用于small RNA Sequence数据分析，教程最新版链接：</p><p data-tool="mdnice编辑器">https://www.bioconductor.org/packages/release/workflows/vignettes/seqpac/inst/doc/seqpac_-_A_guide_to_sRNA_analysis_using_sequence-based_counts.html</p><p data-tool="mdnice编辑器">软件分析输入为fq文件，seqpac workflow的核心对象为：</p><p data-tool="mdnice编辑器"><strong>The PAC object</strong>：这个对象有三个部分，Phenotypic information (P), Annotations (A) and Counts (C)。</p><p data-tool="mdnice编辑器">这个PAC对象有两种格式， S3 list或者S4，并且可以使用函数进行互转，as(pac-object, "list")将S4转为S3， as.PAC(pac-object)将S3转为S4。</p><p data-tool="mdnice编辑器"><strong>The Targeting objects</strong>：用于访问PAC对象内的特定数据子集。</p><h3 data-tool="mdnice编辑器"><span></span><span><strong>seqpac</strong>的分析步骤主要为四步：</span><span></span></h3><ul data-tool="mdnice编辑器"><li><section>Generate a PAC object from fastq-files</section></li><li><section>Preprocess the PAC object</section></li><li><section>Annotate the PAC object</section></li><li><section>Analyze the PAC object.</section></li></ul><h3 data-tool="mdnice编辑器"><span></span><span>示例数据</span><span></span></h3><p data-tool="mdnice编辑器">seqpac包中自带了一个示例数据，6 sRNA fastq文件。</p><figure data-tool="mdnice编辑器"><img data-ratio="0.3354978354978355" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9NibCbHgDb9PQKkUNS8ib0w5TudNaELOdPsCulTetxs3znSPvFJwrxODibTzMDxzQR1NoWrLd5xoq8w/640?wx_fmt=png" data-type="png" data-w="462" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9NibCbHgDb9PQKkUNS8ib0w5TudNaELOdPsCulTetxs3znSPvFJwrxODibTzMDxzQR1NoWrLd5xoq8w/640?wx_fmt=png"><figcaption>image-20230823222638876</figcaption></figure><p data-tool="mdnice编辑器">样本组织类型为果蝇胚胎，测序平台为Illumina NextSeq 500，测序读长为75个碱基。这里抽取原始fq的部分read进行演示。</p><p data-tool="mdnice编辑器">安装包：</p><pre data-tool="mdnice编辑器"><code><span>## Installation</span><br>devtools::install_github(<span>"Danis102/seqpac"</span>, upgrade=<span>"never"</span>, <br> build_manual=TRUE, build_vignettes=TRUE)<br></code></pre><p data-tool="mdnice编辑器">首先快速一览整个包的分析步骤：</p><pre data-tool="mdnice编辑器"><code><span># Setup</span><br> library(seqpac)<br> <br> sys_path = system.file(<span>"extdata"</span>, package = <span>"seqpac"</span>, mustWork = TRUE)<br> fastq &lt;- list.files(path = sys_path, pattern = <span>"fastq"</span>, all.files = FALSE,<br>                 full.names = TRUE)<br> ref_tRNA &lt;- system.file(<span>"extdata/trna"</span>, <span>"tRNA.fa"</span>,<br>                         package = <span>"seqpac"</span>, mustWork = TRUE)<br> <br> <span># Trim NEB adapter, generate counts and create PAC-object (pheno, anno, counts)</span><br> count_list  &lt;- make_counts(fastq, plot = FALSE, <br>                             parse=<span>"default_neb"</span>, trimming=<span>"seqpac"</span>)<br> pac &lt;- make_PAC(count_list<span>$counts</span>)<br> pheno(pac)<span>$groups</span> &lt;- c(rep(<span>"gr1"</span>,3), rep(<span>"gr2"</span>,3)) <span>#Add groups to pheno table</span><br> <br> <span># Preprocess PAC-object and creat means</span><br> pac &lt;- PAC_norm(pac)                        <span># Default normalization is cpm</span><br> pac &lt;- PAC_filter(pac, size=c(15,60))       <span># Here, filter on sequence length</span><br> pac &lt;- PAC_summary(pac, norm = <span>"cpm"</span>, <span>type</span> = <span>"means"</span>, <br>                    pheno_target=list(<span>"groups"</span>, unique(pheno(pac)<span>$groups</span>)))<br> <br> <span># Quickly align (annotate) and plot PAC-object</span><br> map_tRNA &lt;- PAC_mapper(pac, ref=ref_tRNA, override=TRUE)<br> plts &lt;- PAC_covplot(pac, map_tRNA, summary_target=list(<span>"cpmMeans_groups"</span>))<br> plts[[13]]<br></code></pre><h3 data-tool="mdnice编辑器"><span></span><span>详细步骤</span><span></span></h3><h3 data-tool="mdnice编辑器"><span></span><span>1.首先加载fq数据</span><span></span></h3><pre data-tool="mdnice编辑器"><code><span># Use the ShortRead-package to randomly sample a fastq file</span><br>sys_path = system.file(<span>"extdata"</span>, package = <span>"seqpac"</span>, mustWork = TRUE)<br>fq &lt;- list.files(path = sys_path, pattern = <span>"fastq"</span>, all.files = FALSE, full.names = TRUE)<br>fq<br>[1] <span>"C:/Users/86187/AppData/Local/R/win-library/4.2/seqpac/extdata/seqpac_fq1.fastq.gz"</span><br>[2] <span>"C:/Users/86187/AppData/Local/R/win-library/4.2/seqpac/extdata/seqpac_fq2.fastq.gz"</span><br>[3] <span>"C:/Users/86187/AppData/Local/R/win-library/4.2/seqpac/extdata/seqpac_fq3.fastq.gz"</span><br>[4] <span>"C:/Users/86187/AppData/Local/R/win-library/4.2/seqpac/extdata/seqpac_fq4.fastq.gz"</span><br>[5] <span>"C:/Users/86187/AppData/Local/R/win-library/4.2/seqpac/extdata/seqpac_fq5.fastq.gz"</span><br>[6] <span>"C:/Users/86187/AppData/Local/R/win-library/4.2/seqpac/extdata/seqpac_fq6.fastq.gz"</span><br></code></pre><h3 data-tool="mdnice编辑器"><span></span><span>2.生成count矩阵</span><span></span></h3><p data-tool="mdnice编辑器">使用make_counts函数</p><pre data-tool="mdnice编辑器"><code>count_list &lt;- make_counts(input=fq, trimming=<span>"seqpac"</span>, threads=16, parse=<span>"default_neb"</span>)<br><br><span># 查看count_list的结构</span><br>str(count_list, max.level = 1)<br><br>List of 3<br> $ counts         :<span>'data.frame'</span>: 8300 obs. of  6 variables:<br> $ progress_report:<span>'data.frame'</span>: 6 obs. of  19 variables:<br> $ evidence_plots :List of 2<br></code></pre><h3 data-tool="mdnice编辑器"><span></span><span>3.构造样本表型矩阵</span><span></span></h3><p data-tool="mdnice编辑器">使用make_pheno函数</p><pre data-tool="mdnice编辑器"><code>Sample_ID &lt;- colnames(count_list<span>$counts</span>)<br>pheno_fl &lt;- data.frame(Sample_ID=Sample_ID,<br>                    Treatment=c(rep(<span>"heat"</span>, <span>times</span>=1), <br>                                rep(<span>"control"</span>, <span>times</span>=2)),<br>                    Batch=rep(c(<span>"1"</span>, <span>"2"</span>, <span>"3"</span>), <span>times</span>=1))<br><br>pheno &lt;- make_pheno(pheno=pheno_fl, <br>                    progress_report=count_list<span>$progress_report</span>, <br>                    counts=count_list<span>$counts</span>)<br></code></pre><p data-tool="mdnice编辑器">每个样本的表型为一个数据框：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.18888888888888888" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9NibCbHgDb9PQKkUNS8ib0w5Y80jhDyZsqdTNuibNicKoQ6ff8Xr7PrvhsSnluFImBxN1kTlx1Huic9iag/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9NibCbHgDb9PQKkUNS8ib0w5Y80jhDyZsqdTNuibNicKoQ6ff8Xr7PrvhsSnluFImBxN1kTlx1Huic9iag/640?wx_fmt=png"><figcaption>image-20230903161349444</figcaption></figure><h3 data-tool="mdnice编辑器"><span></span><span>4.构造PAC对象</span><span></span></h3><pre data-tool="mdnice编辑器"><code><span>###--------------------------------------------------------------------- </span><br><span>## Generate PAC object (default S4)</span><br>pac_master &lt;- make_PAC(pheno=pheno, counts=count_list<span>$counts</span>)<br>pac_master<br><br>PAC object with: <br>   6 samples<br>   8300 sequences<br>   mean total counts: 4410 (min:4346/max:4476)<br>   best sequence: 230 mean counts<br>   worst sequence: 0 mean counts<br></code></pre><p data-tool="mdnice编辑器">至此，数据的分析对象已经构建好了，PAC对象，后续的分析基本上都是对这个PAC进行操作。</p><h3 data-tool="mdnice编辑器"><span></span><span>tRNA class analysis</span><span></span></h3><p data-tool="mdnice编辑器">我们本次主要关注tRNA class analysis，此包的其他分析可以见分析教程。</p><p data-tool="mdnice编辑器">首先加载包中内置的PAC对象</p><pre data-tool="mdnice编辑器"><code>load(system.file(<span>"extdata"</span>, <span>"drosophila_sRNA_pac_filt_anno.Rdata"</span>, <br>                 package = <span>"seqpac"</span>, mustWork = TRUE))<br>                 <br><span># 查看加载的对象</span><br>pac<br></code></pre><p data-tool="mdnice编辑器">比对：</p><p data-tool="mdnice编辑器">这个包内置比对工具为bwa，此处mismatches为0，还讲序列上下增加了3bp，参考序列为tRNA序列，来自GtRNAdb数据库：http://gtrnadb.ucsc.edu/。</p><pre data-tool="mdnice编辑器"><code><span>###---------------------------------------------------------------------</span><br><span>## tRNA analysis in Seqpac </span><br><br><span># First create an annotation blank PAC with group means</span><br>anno(pac) &lt;- anno(pac)[,1, drop=FALSE]<br>pac_trna &lt;- PAC_summary(pac, norm = <span>"cpm"</span>, <span>type</span> = <span>"means"</span>, <br>                        pheno_target=list(<span>"stage"</span>), merge_pac = TRUE)<br><br><span># Then reannotate only tRNA using the PAC_mapper function</span><br>ref &lt;- <span>"/some/path/to/trna/fasta/recerence.fa"</span>  <br><span># or use the provided fasta</span><br>ref &lt;- system.file(<span>"extdata/trna2"</span>, <span>"tRNA2.fa"</span>, <br>                          package = <span>"seqpac"</span>, mustWork = TRUE)          <br><br>map_object &lt;- PAC_mapper(pac_trna, ref=ref, N_up = <span>"NNN"</span>, N_down = <span>"NNN"</span>, <br>                         mismatches=0, threads=6, report_string=TRUE,<br>                         override=TRUE)<br></code></pre><p data-tool="mdnice编辑器">比对完后的map_object对象是一个list对象，如下：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.26305609284332687" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9NibCbHgDb9PQKkUNS8ib0w55zqU6tFUxeibib74pOPDnicl6mxMRYKtlOc2icibBH1KgZVQ8GR52rj3DWQ/640?wx_fmt=png" data-type="png" data-w="1034" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9NibCbHgDb9PQKkUNS8ib0w55zqU6tFUxeibib74pOPDnicl6mxMRYKtlOc2icibBH1KgZVQ8GR52rj3DWQ/640?wx_fmt=png"><figcaption>image-20230903174938936</figcaption></figure><p data-tool="mdnice编辑器">每一个tRNA上比对到的tsRNA如下，每一行为一个tsRNA的序列：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.31203703703703706" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9NibCbHgDb9PQKkUNS8ib0w5OA5Q0vxR3LWZAdicGLOicbYWy002bWJhtNxnjlbzz0TjwFlKpP6Wndvg/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9NibCbHgDb9PQKkUNS8ib0w5OA5Q0vxR3LWZAdicGLOicbYWy002bWJhtNxnjlbzz0TjwFlKpP6Wndvg/640?wx_fmt=png"><figcaption>image-20230903175956563</figcaption></figure><p data-tool="mdnice编辑器">先看看比对结果，PAC_covplot可以绘制每一个tRNA上的短片段tsRNA的比对情况：</p><pre data-tool="mdnice编辑器"><code><span># Hint: By adding N_up ad N_down you can make sure that modified fragments (like</span><br><span># 3' -CAA in mature tRNA are included).</span><br><br><span>## Plot tRNA using xseq=TRUE gives you the reference sequence as X-axis:</span><br><span># (OBS! Long reference will not show well.)                     </span><br>cov_tRNA &lt;- PAC_covplot(pac_trna, map_object, <br>                        summary_target = list(<span>"cpmMeans_stage"</span>), <br>                        xseq=TRUE, style=<span>"line"</span>, <br>                        color=c(<span>"red"</span>, <span>"black"</span>, <span>"blue"</span>))<br></code></pre><p data-tool="mdnice编辑器">选一个tRNA查看，这里对每类stage进行的绘制：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.7981481481481482" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9NibCbHgDb9PQKkUNS8ib0w5hqrItib7gMGIDn64XAiblRhx8MuN0Jic91V7sgFjRNicia1TicsdUGz9GOQQ/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9NibCbHgDb9PQKkUNS8ib0w5hqrItib7gMGIDn64XAiblRhx8MuN0Jic91V7sgFjRNicia1TicsdUGz9GOQQ/640?wx_fmt=png"><figcaption>image-20230903175253710</figcaption></figure><p data-tool="mdnice编辑器">也可以指定某个tRNA绘制查看：</p><pre data-tool="mdnice编辑器"><code><span># Targeting a single tRNA using a summary data.frame </span><br>PAC_covplot(pac_trna, map=map_object, summary_target= list(<span>"cpmMeans_stage"</span>), <br>            map_target=<span>"tRNA-Lys-CTT-1-9"</span>)<br></code></pre><p data-tool="mdnice编辑器">结果如下：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.7555762081784386" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9NibCbHgDb9PQKkUNS8ib0w5uolEeZtDJ8XtFhF6zicWMomU35TQ5shEZ3jNX7TRjc6Z8uOhTwnbQ2Q/640?wx_fmt=png" data-type="png" data-w="1076" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9NibCbHgDb9PQKkUNS8ib0w5uolEeZtDJ8XtFhF6zicWMomU35TQ5shEZ3jNX7TRjc6Z8uOhTwnbQ2Q/640?wx_fmt=png"><figcaption>image-20230903175431933</figcaption></figure><p data-tool="mdnice编辑器">此外，还可以筛选出tRNA上比对read最多的进行展示：</p><pre data-tool="mdnice编辑器"><code><span># Find tRNAs with many fragments</span><br><span># 1st extract number of rows from each alignment </span><br>n_tRFs &lt;- unlist(lapply(map_object, <span>function</span>(x){nrow(x[[2]])}))<br><span># The test data is highly down an filterd sampled, but still some with tRNA have</span><br><span># a few alignment</span><br>table(n_tRFs)<br>names(map_object)[n_tRFs&gt;2]<br><span># Lets select a few of them and plot them</span><br>selct &lt;- names(map_object)[n_tRFs&gt;2][c(1, 16, 27, 37)]<br>cov_plt &lt;- PAC_covplot(pac_trna, map=map_object, <br>                       summary_target= list(<span>"cpmMeans_stage"</span>), <br>                       map_target=selct)<br>                       <br>cowplot::plot_grid(plotlist=cov_plt, nrow=2, ncol=2)<br></code></pre><p data-tool="mdnice编辑器">选取的四个tRNA绘图结果如下：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.625" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9NibCbHgDb9PQKkUNS8ib0w5G1NsnEN1SblUTHXP7WGT8UCZulfJU2ibDiaGSWT0AdFArM4rfu1NZNibQ/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9NibCbHgDb9PQKkUNS8ib0w5G1NsnEN1SblUTHXP7WGT8UCZulfJU2ibDiaGSWT0AdFArM4rfu1NZNibQ/640?wx_fmt=png"><figcaption>image-20230903180248037</figcaption></figure><p data-tool="mdnice编辑器">接下来就是解析每个tsRNA的分类了：</p><p data-tool="mdnice编辑器">解理tsRNA的分类，如5 '和3 'halves等，更复杂。这种分类涉及到tRNA环(A-loop、anticodon loop和T-loop)中可能发生切割的位置。有一些外部软件可以从二级结构模型中预测环路。预测tRNA二级结构最流行的工具之一是tRNAscan-SE (http://lowelab.ucsc.edu/tRNAscan-SE/)。这个软件的输出是一个ss文件，它以以下格式存储循环信息：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.1405579399141631" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9NibCbHgDb9PQKkUNS8ib0w5ib8B875MUaDexFuZMbY28R9rvyDa3o8iateeuH91ywUzoYXbJLrhgnPg/640?wx_fmt=png" data-type="png" data-w="932" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9NibCbHgDb9PQKkUNS8ib0w5ib8B875MUaDexFuZMbY28R9rvyDa3o8iateeuH91ywUzoYXbJLrhgnPg/640?wx_fmt=png"><figcaption>image-20230903180717224</figcaption></figure><p data-tool="mdnice编辑器">Ss-files for most common genomes are readily available for download at http://gtrnadb.ucsc.edu/。</p><p data-tool="mdnice编辑器">根据ss文件，结合比对结果就可以使用tRNA_class函数对tsRNA进行分类注释了：</p><pre data-tool="mdnice编辑器"><code><span>###---------------------------------------------------------------------</span><br><span>## Generate range types using a ss-file</span><br><br><span># Download ss object from GtRNAdb </span><br><span># ("http://gtrnadb.ucsc.edu/")</span><br>ss_file &lt;- <span>"/some/path/to/GtRNAdb/trna.ss"</span><br><span># or for testing use the provided ss-file in secpac</span><br>ss_file &lt;- system.file(<span>"extdata/trna2"</span>, <span>"tRNA2.ss"</span>, <br>                          package = <span>"seqpac"</span>, mustWork = TRUE)<br><br><span># Classify fragments according to loop cleavage (small loops are omitted)       </span><br>map_object_ss &lt;- map_rangetype(map_object, <span>type</span>=<span>"ss"</span>, <br>                               ss=ss_file, min_loop_width=4)            <br><br><span># Remove reference tRNAs with no hits</span><br>map_object_ss &lt;-  map_object_ss[<br>  !unlist(lapply(map_object_ss, <span>function</span>(x){x[[2]][1,1] == <span>"no_hits"</span>}))]<br><br><span># Now we have quite comprehensive tRNA loop annotations</span><br>names(map_object_ss)<br>map_object_ss[[1]][[2]]<br><br><span># Don't forget to check ?map_rangetype to obtain more options </span><br><br><br><span>###---------------------------------------------------------------------</span><br><span>## Function classifying 5'-tRF, 5'halves, i-tRF, 3'-tRF, 3'halves</span><br><br><span># Does all tRNAs have 3 loops?</span><br>table(unlist(lapply(map_object_ss, <span>function</span>(x){unique(x<span>$Alignments</span><span>$n_ssloop</span>)})))<br><br><span># Set tolerance for classification as a terminal tRF</span><br>tolerance &lt;- 5  <span># 2 nucleotides from start or end of full-length tRNA)</span><br><br><span>### Important:</span><br><span># We set N_up and N_down to "NNN" in the PAC_mapper step. To make sure</span><br><span># that we have a tolerance that include the original tRNA sequence</span><br><span># we set terminal= 2+3 (5).  </span><br><span>## tRNA classifying function </span><br><span># Apply the tRNA_class function and make a tRNA type column</span><br>pac_trna &lt;- tRNA_class(pac_trna, map=map_object_ss, terminal=tolerance)<br><span>## </span><br><span>## -- Anno target was specified, will retain: 29 of 9131 seqs.</span><br>anno(pac_trna)<span>$type</span> &lt;- paste0(anno(pac_trna)<span>$decoder</span>, anno(pac_trna)<span>$acceptor</span>)<br>anno(pac_trna)[1:5, 1:4] <br></code></pre><p data-tool="mdnice编辑器">得到的分类结果如下：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.22526041666666666" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9NibCbHgDb9PQKkUNS8ib0w5Jms2o5PABDib4sFdkn0Ym2K31IkcctmjVkyyWsicZNpNuibDImHnLH5bg/640?wx_fmt=png" data-type="png" data-w="768" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9NibCbHgDb9PQKkUNS8ib0w5Jms2o5PABDib4sFdkn0Ym2K31IkcctmjVkyyWsicZNpNuibDImHnLH5bg/640?wx_fmt=png"><figcaption>image-20230903181248338</figcaption></figure><p data-tool="mdnice编辑器">查看每种tsRNA类别的占比：</p><pre data-tool="mdnice编辑器"><code><span>###---------------------------------------------------------------------</span><br><span>## Plot tRNA types</span><br><br><span># Now use PAC_trna to generate some graphs based on grand means</span><br>trna_result &lt;- PAC_trna(pac_trna, norm=<span>"cpm"</span>, filter = NULL,<br>  join = TRUE, top = 15, log2fc = TRUE,<br>  pheno_target = list(<span>"stage"</span>, c(<span>"Stage1"</span>, <span>"Stage3"</span>)), <br>  anno_target_1 = list(<span>"type"</span>),<br>  anno_target_2 = list(<span>"class"</span>))<br>  <br>cowplot::plot_grid(trna_result<span>$plots</span><span>$Expression_Anno_1</span><span>$Grand_means</span>,<br>               trna_result<span>$plots</span><span>$Log2FC_Anno_1</span>,<br>               trna_result<span>$plots</span><span>$Percent_bars</span><span>$Grand_means</span>,<br>               nrow=1, ncol=3)<br></code></pre><p data-tool="mdnice编辑器">结果如下：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.4666666666666667" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9NibCbHgDb9PQKkUNS8ib0w55ppZLOoM6ibEsdJNtfkrvPmgq6hhzw8BtY2icU5kCcleXwl42zzbRgyQ/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9NibCbHgDb9PQKkUNS8ib0w55ppZLOoM6ibEsdJNtfkrvPmgq6hhzw8BtY2icU5kCcleXwl42zzbRgyQ/640?wx_fmt=png"><figcaption>image-20230903181537012</figcaption></figure><p data-tool="mdnice编辑器">每个tsRNA的最终注释结果如下：</p><pre data-tool="mdnice编辑器"><code>anno_result &lt;- anno(pac_trna)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.34444444444444444" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9NibCbHgDb9PQKkUNS8ib0w5ujKTB1GjDjEG727ORYqt7ICGgqBOqwCgTtlanDRkz2n6cR2icKUhGWg/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9NibCbHgDb9PQKkUNS8ib0w5ujKTB1GjDjEG727ORYqt7ICGgqBOqwCgTtlanDRkz2n6cR2icKUhGWg/640?wx_fmt=png"><figcaption>image-20230903182329699</figcaption></figure><p data-tool="mdnice编辑器">常见文献中tsRNA分类结果展示可以如下：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.6157407407407407" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9NibCbHgDb9PQKkUNS8ib0w5UZEetjBhJp3jicFy7T0h1SlJLKQg7q0zypwpxVk7XrtaGLzTr0yzgoQ/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9NibCbHgDb9PQKkUNS8ib0w5UZEetjBhJp3jicFy7T0h1SlJLKQg7q0zypwpxVk7XrtaGLzTr0yzgoQ/640?wx_fmt=png"><figcaption>image-20230903182439352</figcaption></figure><p data-tool="mdnice编辑器">ref：https://doi.org/10.1186/s12958-022-00978-3 <em>Reproductive Biology and Endocrinology (2022) 20:106</em></p><p data-tool="mdnice编辑器">根据这个分类注释做个性化绘制吧~</p></section><p><br></p><p><mp-style-type data-value="10000"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/Fhaj6_gtqGRMEbpUsySnFQ",target="_blank" rel="noopener noreferrer">原文链接</a>
