---
title: "文献复现-单细胞揭示新辅助治疗后NSCLC的免疫微环境变化"
date: 2023-09-17T14:08:58Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Acdemic"]
---
文献复现-单细胞揭示新辅助治疗后NSCLC的免疫微环境变化 by 生信菜鸟团
------
<div><section><blockquote><p>好久没做文献复现了…</p><p>上一次好像还是在去年的3月份左右，详见：<a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzUzMTEwODk0Ng==&amp;mid=2247501381&amp;idx=1&amp;sn=4d66c995b9f7d18cbb89e6b5e8cd0214&amp;chksm=fa450378cd328a6ef2333cea3fe1ff0fe0083150d3a463953014f70a0242a84089e9cc9cb5ab&amp;scene=21#wechat_redirect" textvalue="【生信文献200篇】专栏也落下帷幕，生信菜鸟团的发展需要你" linktype="text" imgurl="" imgdata="null" data-itemshowtype="11" tab="innerlink" data-linktype="2">【生信文献200篇】专栏也落下帷幕，生信菜鸟团的发展需要你</a>  。。。。</p><p>所以距离这个领域我已经脱节一年了….别骂了别骂了在学了在学了😧。</p><p>如有不对，请指出，下次一定更正！！！</p></blockquote><p>文章在这：Tumor microenvironment remodeling after neoadjuvant immunotherapy in non-small cell lung cancer revealed by single-cell RNA sequencing<br><strong>方法：</strong>来自3名治疗前和12名接受新辅助PD-1阻断联合化疗的非小细胞肺癌（NSCLC）患者的~92,000个单细胞的转录组。<span><strong>根据病理反应将12个治疗后样本分为两组：MPR（n = 4）和非MPR（n = 8）。</strong></span></p><blockquote><p>主要病理反应(<strong>MPR)</strong>被定义为治疗后苏木精和伊红（H&amp;E）染色的残余活肿瘤细胞不超过10%</p></blockquote><p><strong>结果和结论：</strong>不同的治疗诱导的癌细胞转录组与临床反应有关。MPR患者中：癌细胞通过MHC-II表现出活化抗原呈递的特征。aged CCL3+中性粒细胞减少，FCRL4+FCRL5+记忆B细胞和CD16+CX3CR1+单核细胞富集，并且是免疫治疗反应的预测因子。NMPR患者中：癌细胞表现出雌激素代谢酶的过表达和血清雌二醇升高。在所有患者中，治疗促进了细胞毒性T细胞和CD16+NK细胞的扩增和活化，减少了免疫抑制性Treg，并将记忆CD8+T细胞激活为效应表型。组织驻留巨噬细胞在治疗后扩增，肿瘤相关巨噬细胞（TAM）被重塑为中性。</p><p><span><strong>接下来是对文中的图进行的复现…</strong></span></p><p><span><strong>纯手工，自己写代码，并不是文章的代码哦。。。。</strong></span></p><h2><span>1.总分群</span></h2><p>文章原图<br></p><figure><img data-ratio="0.9027777777777778" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJqv7C4JcSlNx4ObhOtR62KZNMicfc4L5Axib8CHjRxJa7CM6gQOic1dfaicw/640?wx_fmt=png" data-type="png" data-w="1080" title="image.png" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJqv7C4JcSlNx4ObhOtR62KZNMicfc4L5Axib8CHjRxJa7CM6gQOic1dfaicw/640?wx_fmt=png"></figure><pre><code>rm(list=ls())<br>options(stringsAsFactors = F) <br><span>source</span>(<span>'scrip/lib.R'</span>)<br>library(readxl)<br><br><span>###### step1:导入数据 ######   </span><br>ct=fread( <span>'GSE207422_RAW/GSE207422_NSCLC_scRNAseq_UMI_matrix.txt.gz'</span>,data.table = F)<br>ct[1:4,1:4]<br>ct[nrow(ct),1:4]<br>rownames(ct)=ct[,1]<br>ct=ct[,-1]<br>ct[1:4,1:4]<br>ct1 = ct<br>s &lt;- colnames(ct1)<br>s[1:10]<br>s1 = substr(s, 1, 11)<br>s1[1:10]<br>table(s1)<br>colnames(ct1) = s1<br>meta &lt;- read_excel(<span>"GSE207422_RAW/GSE207422_NSCLC_scRNAseq_metadata.xlsx"</span>)<br><br>sce.all = CreateSeuratObject(counts =  ct1 , <br>                             min.cells = 5,<br>                             min.features = 500)<br>as.data.frame(sce.all@assays<span>$RNA</span>@counts[1:10, 1:2])<br>head(sce.all@meta.data, 10)<br>table(sce.all@meta.data<span>$orig</span>.ident) <br><br>s = colnames(sce.all)<br>s1 = substr(s, 1, 11)<br>s1[1:10]<br>table(s1)<br>s1 = gsub(c(<span>'BD_immune01|BD_immune05|BD_immune08'</span>),<span>"Pre-treatment"</span>,s1)<br>s2 = gsub(<span>"BD_immune.*$"</span>, <span>"Post-treatment"</span>, s1) <span>#第一个空格直达结尾替换成b</span><br>table(s2)<br><br>phe=str_split(  colnames(sce.all) ,<span>'_'</span>,simplify = T)<br>head(phe)<br>table(phe[,1])<br>table(phe[,2])  <br>table(phe[,2],phe[,1])  <br>sce.all<span>$orig</span>.ident = s2<br>table(sce.all<span>$orig</span>.ident )<br>sce.all<span>$group</span>=sce.all<span>$orig</span>.ident<br>sce.all<span>$orig</span>.ident=paste(phe[,2],phe[,1])  <br>head(sce.all@meta.data) <br>table(sce.all@meta.data<span>$group</span>)<br>table(sce.all@meta.data<span>$orig</span>.ident)<br><br><span>###### step2:QC质控 ######</span><br>dir.create(<span>"./1-QC"</span>)<br>setwd(<span>"./1-QC"</span>)<br><span>source</span>(<span>'../scrip/qc.R'</span>)<br>sce.all.filt = basic_qc(sce.all)<br><span>print</span>(dim(sce.all))<br><span>print</span>(dim(sce.all.filt))<br>setwd(<span>'../'</span>)<br>sp=<span>'human'</span><br><br><span>###### step3: harmony整合多个单细胞样品 ######</span><br>dir.create(<span>"2-harmony"</span>)<br>getwd()<br>setwd(<span>"2-harmony"</span>)<br><span>source</span>(<span>'../scrip/harmony.R'</span>)<br><span># 默认 ScaleData 没有添加"nCount_RNA", "nFeature_RNA"</span><br><span># 默认的</span><br>sce.all.int = run_harmony(sce.all.filt)<br>setwd(<span>'../'</span>)<br><br><span>#######step4: 检查常见分群情况</span><br>rm(list=ls())<br>library(ggplot2)<br>dir.create(<span>"3-cell"</span>)<br>setwd(<span>"3-cell"</span>)  <br>getwd()<br>sce.all.int=readRDS(<span>"../2-harmony/sce.all_int.rds"</span>)<br>table(Idents(sce.all.int))<br>table(sce.all.int<span>$seurat_clusters</span>)<br>table(sce.all.int<span>$RNA_snn_res</span>.0.1) <br>table(sce.all.int<span>$RNA_snn_res</span>.0.3) <br><br>sel.clust = <span>"RNA_snn_res.0.3"</span><br>sce.all &lt;- SetIdent(sce.all.int, value = sel.clust)<br>table(sce.all@active.ident) <br><br>meta &lt;- read_excel(<span>"../GSE207422_RAW/GSE207422_NSCLC_scRNAseq_metadata.xlsx"</span>)<br>s = colnames(sce.all)<br>s[1:10]<br>s1 = substr(s, 1, 11)<br>s1[1:10]<br>table(s1)<br>s2 = gsub(<span>"BD_immune"</span>, <span>"P"</span>, s1) <br>table(s2)<br><br>sce.all<span>$patient</span> = s2<br>table(sce.all<span>$patient</span> )<br><br><span>##RECIST</span><br>colnames(meta)<br>meta<span>$RECIST</span><br>s_RECIST = gsub(c(<span>'BD_immune01|BD_immune07|BD_immune08|BD_immune11|BD_immune12|BD_immune15'</span>),<span>"PR"</span>,s1)<br>s_RECIST = gsub(<span>"BD_immune.*$"</span>, <span>"SD"</span>, s_RECIST) <br>table(s_RECIST)<br>sce.all<span>$RECIST</span> = s_RECIST<br><br><span>##病理反应</span><br><span>##后面是按照TN和治疗后手术样本来分类的</span><br>colnames(meta)<br>meta$`Pathologic Response`<br>s_Pathologic = gsub(c(<span>'BD_immune03|BD_immune06|BD_immune11|BD_immune14'</span>),<span>"MPR"</span>,s1)<br>s_Pathologic = gsub(<span>"BD_immune.*$"</span>, <span>"NMPR"</span>, s_Pathologic) <br>table(s_Pathologic)<br>sce.all<span>$Pathologic</span> = s_Pathologic<br>phe = sce.all@meta.data<br><br>phe[phe<span>$orig</span>.ident == <span>'Pre-treatment'</span>, <span>"Pathologic"</span>] = <span>'TN'</span><br>table(phe<span>$Pathologic</span>)<br>sce.all@meta.data = phe<br>table(sce.all<span>$Pathologic</span>)<br><br><span>##药物治疗</span><br>colnames(meta)<br>meta = meta[1:15,]<br>meta<span>$medicine</span> = paste(meta$`PD1 Antibody`,meta<span>$Chemotherapy</span>,sep=<span>"+"</span>)<br>s1<br>table(meta<span>$medicine</span>)<br>match_vec &lt;- match(s1, unique(s1))<br>s_medicine &lt;- meta<span>$medicine</span>[match_vec]<br><span>##检验一下是不是对的...</span><br>s1[23456]<br>s_medicine[23456]<br><br>s1[45678]<br>s_medicine[45678]<br><br>table(s_medicine)<br>sce.all<span>$medicine</span> = s_medicine<br><br>saveRDS(sce.all, <span>"sce.all_meta.rds"</span>)<br><br><span>#marker</span><br>genes_to_check = c(<span>'PTPRC'</span>, <span>'CD3D'</span>, <span>'CD3E'</span>, <span>'CD4'</span>,<span>'CD8A'</span>,  <span>## T </span><br>                   <span>'CD19'</span>, <span>'CD79A'</span>, <span>'MS4A1'</span> ,  <span>## B </span><br>                   <span>'IGHG1'</span>, <span>'MZB1'</span>, <span>'SDC1'</span>, <span>## Plasma </span><br>                   <span>'CD68'</span>, <span>'CD163'</span>, <span>'CD14'</span>, <br>                   <span>'VCAN'</span>, <span>'FCN1'</span>, <br>                   <span>'TPSAB1'</span> , <span>'TPSB2'</span>,  <span># mast cells,</span><br>                   <span>'RCVRN'</span>,<span>'FPR1'</span> , <span>'ITGAM'</span> ,<br>                   <span>'C1QA'</span>,  <span>'C1QB'</span>,  <span># mac</span><br>                   <span>'S100A9'</span>, <span>'S100A8'</span>, <span>'MMP19'</span>,<span># monocyte</span><br>                   <span>'MKI67'</span> , <span>'TOP2A'</span>,<span>'LYZ'</span>, <span>##myloid</span><br>                   <span>'LAMP3'</span>, <span>'IDO1'</span>,<span>'IDO2'</span>,<span>## DC3 </span><br>                   <span>'CD1E'</span>,<span>'CD1C'</span>, <span># DC2</span><br>                   <span>'CLEC10A'</span>, <span>'CLEC12A'</span>,<span>'LILRA4'</span>, <span>##pDC</span><br>                   <span>'FGFBP2'</span>,<span>'NCR1'</span>,<span>'KLRK1'</span>,<span>'KLRC1'</span>,<span>'NCAM1'</span>,<span>'FCGR3A'</span>, <span># NK </span><br>                   <span>'KIT'</span>, <span>##MAST</span><br>                   <span>'CSF3R'</span>,<span>'ITGAM'</span>,<span>'ITGB2'</span>,<span>'FUT4'</span>,<span>'FCGR3B'</span>,<span>'FCGR2A'</span>, <span>##Neutrophil</span><br>                   <span>'FGF7'</span>,<span>'MME'</span>, <span>'ACTA2'</span>,<span>'VWF'</span>,<span>'COL1A1'</span>, <span>## human  fibo </span><br>                   <span>'DCN'</span>, <span>'LUM'</span>,  <span>'GSN'</span> , <span>## mouse PDAC fibo </span><br>                   <span>'PECAM1'</span>, <span>'VWF'</span>,  <span>## endo </span><br>                   <span>'EPCAM'</span> , <span>'KRT19'</span>,<span>'KRT7'</span> <span># epi </span><br>)<br><br><br>library(stringr)  <br>genes_to_check=str_to_upper(genes_to_check)<br>genes_to_check<br>p &lt;- DotPlot(sce.all, features = unique(genes_to_check),<br>             assay=<span>'RNA'</span>  )  + coord_flip()<br><br>p <br><br>mycolors &lt;-c(<span>'#E5D2DD'</span>, <span>'#53A85F'</span>, <span>'#F1BB72'</span>, <span>'#F3B1A0'</span>, <span>'#D6E7A3'</span>, <span>'#57C3F3'</span>,<br>                        <span>'#E95C59'</span>, <span>'#E59CC4'</span>, <span>'#AB3282'</span>,  <span>'#BD956A'</span>, <br>                        <span>'#9FA3A8'</span>, <span>'#E0D4CA'</span>,  <span>'#C5DEBA'</span>, <span>'#F7F398'</span>,<br>                         <span>'#C1E6F3'</span>, <span>'#6778AE'</span>, <span>'#91D0BE'</span>, <span>'#B53E2B'</span>,<br>                        <span>'#712820'</span>, <span>'#DCC1DD'</span>, <span>'#CCE0F5'</span>,  <span>'#CCC9E6'</span>, <span>'#625D9E'</span>, <span>'#68A180'</span>, <span>'#3A6963'</span>,<br>                        <span>'#968175'</span>)<br><br>col =c(<span>"#3176B7"</span>,<span>"#F78000"</span>,<span>"#3FA116"</span>,<span>"#CE2820"</span>,<span>"#9265C1"</span>,<br>                <span>"#885649"</span>,<span>"#DD76C5"</span>,<span>"#7F7F7F"</span>,<span>"#BBBE00"</span>,<span>"#41BED1"</span>)<br><br><br>p_umap=DimPlot(sce.all, reduction = <span>"umap"</span>,cols = mycolors,pt.size = 0.4,<br>               group.by = <span>"RNA_snn_res.0.3"</span>,label = T,label.box = T) <br>p_umap<br>library(patchwork)<br>p+p_umap<br>ggsave(<span>'markers_umap_2.pdf'</span>,width = 13,height = 8)<br><br><span>#####step5: 细胞生物学命名</span><br>rm(list=ls())<br>options(stringsAsFactors = F)<br>library(Seurat)<br>library(ggplot2)<br>library(clustree)<br>library(cowplot)<br>library(dplyr)<br>getwd()<br>sce.all=readRDS( <span>"../3-cell/sce.all_meta.rds"</span>)<br>sce.all<br><br>celltype=data.frame(ClusterID=0:17,<br>                    celltype= 0:17) <br><br><span>#定义细胞亚群 </span><br>celltype[celltype<span>$ClusterID</span> %<span>in</span>% c(0 ,12 ),2]=<span>'Tcells'</span><br>celltype[celltype<span>$ClusterID</span> %<span>in</span>% c( 4 ),2]=<span>'Bcells'</span><br>celltype[celltype<span>$ClusterID</span> %<span>in</span>% c( 5),2]=<span>'plasma'</span><br>celltype[celltype<span>$ClusterID</span> %<span>in</span>% c( 13 ),2]=<span>'mast'</span><br>celltype[celltype<span>$ClusterID</span> %<span>in</span>% c( 14 ),2]=<span>'pDC'</span><br>celltype[celltype<span>$ClusterID</span> %<span>in</span>% c(  11 ),2]=<span>'fibroblast'</span><br>celltype[celltype<span>$ClusterID</span> %<span>in</span>% c( 1,2,15  ),2]=<span>'myeloids'</span><br>celltype[celltype<span>$ClusterID</span> %<span>in</span>% c( 6,8,9,10,17 ),2]=<span>'epithelium'</span><br>celltype[celltype<span>$ClusterID</span> %<span>in</span>% c( 3,16 ),2]=<span>'neutrophil'</span><br>celltype[celltype<span>$ClusterID</span> %<span>in</span>% c( 7 ),2]=<span>'NK'</span><br><br><br>head(celltype)<br>celltype<br>table(celltype<span>$celltype</span>)<br><br>sce.all@meta.data<span>$celltype</span> = <span>"NA"</span><br><span>for</span>(i <span>in</span> 1:nrow(celltype)){<br>  sce.all@meta.data[<span>which</span>(sce.all@meta.data<span>$RNA_snn_res</span>.0.3 == celltype<span>$ClusterID</span>[i]),<span>'celltype'</span>] &lt;- celltype<span>$celltype</span>[i]}<br>table(sce.all@meta.data<span>$celltype</span>)<br><br><br>th=theme(axis.text.x = element_text(angle = 45, <br>                                    vjust = 0.5, hjust=0.5)) <br>library(patchwork)<br>p_umap=DimPlot(sce.all, reduction = <span>"umap"</span>,cols = col,pt.size = 0.4,<br>                               group.by = <span>"celltype"</span>,label = T) <br>p+p_umap<br>ggsave(<span>'markers_umap_by_celltype_2.pdf'</span>,width = 15,height = 7)<br>saveRDS(sce.all, <span>"sce.all_celltype.rds"</span>)<br>phe=sce.all@meta.data<br>save(phe,file = <span>'phe-by-markers.Rdata'</span>)<br><br>colnames(phe)<br>table(sce.all<span>$Pathologic</span>)<br><br>p = DimPlot(sce.all, reduction = <span>"umap"</span>,split.by = <span>'Pathologic'</span>,<br>        group.by = <span>"celltype"</span>,label = T) <br>install.packages(<span>'tidydr'</span>)<br>library(ggplot2)<br>library(tidydr)<br><br>p &lt;- p + theme_dr()<br>p<br><br>p &lt;- p + theme(panel.grid=element_blank())<br>p<br><br><br>load(<span>'sce-by-celltype.Rdata'</span>)<br>features = c(<span>'PTPRC'</span>,<span>'KRT19'</span>,<span>'COL1A1'</span>,<span>'VWF'</span>,<span>'CD3E'</span>,<span>'CD79A'</span>,<span>'MS4A1'</span>,<br>             <span>'IGHG1'</span>,<span>'CSF3R'</span>,<span>'LYZ'</span>,<span>'FGFBP2'</span>,<span>'KIT'</span>,<span>'LILRA4'</span>,<span>'MKI67'</span>)<br>FeaturePlot(sce.all,features = features,ncol = 5 )<br></code></pre><figure><img data-ratio="0.5462962962962963" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJqicqrKt01qLibZe2Y2u3ibOPliciaPAiaNBN6ibZndXRK2Gia3oD8z94LRickBxQ/640?wx_fmt=png" data-type="png" data-w="1080" title="b06eab46ab71781192b811b09881c01.png" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJqicqrKt01qLibZe2Y2u3ibOPliciaPAiaNBN6ibZndXRK2Gia3oD8z94LRickBxQ/640?wx_fmt=png"><figcaption>上面的代码里面的有一些source步骤，是依赖于外部的script文件夹里面的几个R代码文件。。。详见：<a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247522953&amp;idx=1&amp;sn=9e86346a6de21d02bc0c770f3d2734a6&amp;chksm=9b4bda32ac3c5324d83bfc9f1601648626ad90c84eba1df8c782735c7a648351b03fddb69845&amp;scene=21#wechat_redirect" textvalue="小鼠的5个样品的10x技术单细胞转录组上游定量（文末赠送全套代码）" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">小鼠的5个样品的10x技术单细胞转录组上游定量（文末赠送全套代码）</a><br></figcaption><figcaption><br></figcaption></figure><pre><code><span>###### step6: 单细胞亚群比例差异  ######</span><br><span>## 每种细胞类型中，分组所占比例 ----</span><br>rm(list=ls())<br>options(stringsAsFactors = <span>F</span>)<br><span>library</span>(Seurat)<br><span>library</span>(ggplot2)<br><span>library</span>(clustree)<br><span>library</span>(cowplot)<br><span>library</span>(dplyr)<br>getwd()<br>sce.all=readRDS( <span>"3-cell/sce.all_celltype.rds"</span>)<br>sce.all<br>phe=sce.all@meta.data<br><span>library</span>(tidyr)<span># 使用的gather &amp; spread</span><br><span>library</span>(reshape2) <span># 使用的函数 melt &amp; dcast </span><br><br>colnames(phe)<br>tb=table(phe$celltype,<br>         phe$patient)<br>head(tb)<br><span>library</span> (gplots) <br>balloonplot(tb)<br>bar_data &lt;- as.data.frame(tb)<br><br>bar_per &lt;- bar_data %&gt;% <br>  group_by(Var2) %&gt;%<br>  mutate(sum(Freq)) %&gt;%<br>  mutate(percent = Freq / `sum(Freq)`)<br>head(bar_per) <br>write.csv(bar_per,file = <span>"celltype_by_group_percent.csv"</span>)<br>col =c(<span>"#3176B7"</span>,<span>"#F78000"</span>,<span>"#3FA116"</span>,<span>"#CE2820"</span>,<span>"#9265C1"</span>,<br>                <span>"#885649"</span>,<span>"#DD76C5"</span>,<span>"#7F7F7F"</span>,<span>"#BBBE00"</span>,<span>"#41BED1"</span>)<br>ggplot(bar_per, aes(x = percent, y = Var2)) +<br>  geom_bar(aes(fill = Var1) , stat = <span>"identity"</span>) + coord_flip() +<br>  theme(axis.ticks = element_line(linetype = <span>"blank"</span>),<br>        legend.position = <span>"top"</span>,<br>        panel.grid.minor = element_line(colour = <span>NA</span>,linetype = <span>"blank"</span>), <br>        panel.background = element_rect(fill = <span>NA</span>),<br>        plot.background = element_rect(colour = <span>NA</span>)) +<br>  labs(y = <span>"% Relative cell source"</span>, fill = <span>NULL</span>)+labs(x = <span>NULL</span>)+<br>  scale_fill_manual(values=col)<br><br>ggsave(<span>"celltype_by_group_percent.pdf"</span>,<br>       width = <span>8</span>,height = <span>4</span>)<br></code></pre><figure><img data-ratio="0.5509259259259259" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJq7jqOF8hwca16uiafvsB0YkdjWrEG0xkV7ZGjUkhtbQNAs1npFsKK7gA/640?wx_fmt=png" data-type="png" data-w="1080" title="bdf88210b1d876a3906bd9e1c39f793.png" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJq7jqOF8hwca16uiafvsB0YkdjWrEG0xkV7ZGjUkhtbQNAs1npFsKK7gA/640?wx_fmt=png"></figure><br>再就是不同病理结局中细胞亚群的比例<p><br></p><pre><code><span>###Figure1D 分为未治疗组TN，和治疗组（治疗组又分为MPR和NMPR）</span><br>table(phe$celltype)<br>table(phe$patient)<br>phe$patientoutcone =  paste(phe$patient,phe$Pathologic,sep = <span>'_'</span>)<br><br>phe_immune = phe[phe$celltype %in% c(<span>'Bcells'</span>,<span>'myeloids'</span>,<span>'neutrophil'</span>,<span>'NK'</span>,<span>'Tcells'</span>),]<br><br>tb=table(phe_immune$celltype,<br>         phe_immune$patientoutcone)<br><br>head(tb)<br>balloonplot(tb)<br>bar_data &lt;- as.data.frame(tb)<br><br><br>bar_per &lt;- bar_data %&gt;% <br>  group_by(Var2) %&gt;%<br>  mutate(sum(Freq)) %&gt;%<br>  mutate(percent = Freq / <span>`sum(Freq)`</span>)<br>head(bar_per) <br>bar_per$Var2<br>group = ifelse(str_detect(bar_per$Var2,<span>"TN"</span>),<span>"TN"</span>,<br>               ifelse(str_detect(bar_per$Var2,<span>"NMPR"</span>),<span>"NMPR"</span>,<span>"MPR"</span>))<br>bar_per$group = group<br><br>bar_per$group = factor(bar_per$group,levels =c(<span>"TN"</span>,<span>"MPR"</span>,<span>"NMPR"</span>))<br>table(bar_per$Var1)<br>bar_per$Var1 = factor(bar_per$Var1,levels =c(<span>"Tcells"</span>,<span>"Bcells"</span>,<span>"NK"</span>,<span>'myeloids'</span>,<span>'neutrophil'</span>))<br><br><span>###拼接五个箱线图</span><br>library(gridExtra)<br>p &lt;- list()<br><br><span>for</span>(i in <span>1</span>:<span>5</span>){<br>  <span># 选择当前细胞类型的数据</span><br>  cell &lt;- bar_per[bar_per$Var1 == unique(bar_per$Var1)[i],]<br><br>  p[[i]] &lt;- ggplot(cell, aes(<span>x</span> = group, <span>y</span> = percent, fill = group)) +<br>    geom_boxplot(size = <span>0</span>.<span>5</span>, fill = <span>"white"</span>, outlier.fill = <span>"white"</span>, outlier.color = <span>"white"</span>, color = col) +<br>    geom_jitter(aes(fill = group), width = <span>0</span>.<span>2</span>, shape = <span>21</span>, size = <span>2</span>) +<br>    scale_fill_manual(<span>values</span> = col) +<br>    ggtitle(paste<span>0</span>(unique(bar_per$Var1)[i], <span>" "</span>)) +<br>    theme_bw() +<br>    theme(legend.position = <span>"none"</span>,<br>          axis.text.x = element_text(colour = <span>"black"</span>, family = <span>"Times"</span>, size = <span>14</span>),<br>          axis.text.y = element_text(family = <span>"Times"</span>, size = <span>14</span>, face = <span>"plain"</span>),<br>          axis.title.y = element_text(family = <span>"Times"</span>, size = <span>14</span>, face = <span>"plain"</span>),<br>          axis.title.x = element_text(family = <span>"Times"</span>, size = <span>14</span>, face = <span>"plain"</span>),<br>          plot.title = element_text(family = <span>"Times"</span>, size = <span>15</span>, face = <span>"bold"</span>, hjust = <span>0</span>.<span>5</span>),<br>          panel.grid.major = element_blank(),<br>          panel.grid.minor = element_blank()) +<br>    ylab(<span>"Fraction (%) of immune cells"</span>) + xlab(<span>" "</span>)<br>  <span># 移除y轴标题和刻度标签</span><br>  p[[i]] &lt;- p[[i]] + theme(axis.title.y = element_blank())<br>}<br><br>combined_plot &lt;- grid.arrange(grobs = p, ncol = <span>5</span>, left = <span>"Fraction (%) of immune cells"</span>) <span># 设置共同的纵轴标题和拼图列数</span><br><br><span>print</span>(combined_plot)<br></code></pre><figure><img data-ratio="0.3907407407407407" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJqiaKibwy0hicbvD0QicZQICKMbFwGSBs4vg6xV40Vy7sg0oOCzCGTmlOn8w/640?wx_fmt=png" data-type="png" data-w="1080" title="image.png" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJqiaKibwy0hicbvD0QicZQICKMbFwGSBs4vg6xV40Vy7sg0oOCzCGTmlOn8w/640?wx_fmt=png"></figure><br>和文中比例变化大多是一致的，药物治疗后MPR中T细胞和B细胞比例增加（文中还有NK细胞），中性粒细胞和髓系降低。<p><br></p><h2><span>2.上皮细胞</span></h2><figure><img data-ratio="0.8425925925925926" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJq3wTQM5iaO5IDib2OL5aH1jACOqy4ich26rrRxhbhhZhn5GFwXn7l9Erog/640?wx_fmt=png" data-type="png" data-w="1080" title="image.png" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJq3wTQM5iaO5IDib2OL5aH1jACOqy4ich26rrRxhbhhZhn5GFwXn7l9Erog/640?wx_fmt=png"></figure><pre><code><span>####Epithelial cell重聚类####</span><br>rm(list=ls())<br>options(stringsAsFactors = F)<br>library(Seurat)<br>library(ggplot2)<br>library(clustree)<br>library(cowplot)<br>library(dplyr)<br>dir.create(<span>"4-subEpi"</span>)<br>setwd(<span>"4-subEpi"</span>) <br>getwd()<br>sce.all=readRDS( <span>"../3-cell/sce.all_celltype.rds"</span>)<br>sce = sce.all<br>DefaultAssay(sce) &lt;- <span>"RNA"</span><br>table(sce$orig.ident)<br><br>table(sce$celltype)<br>sce2 = sce[, sce$celltype %<span>in</span>% c( <span>'epithelium'</span> )]<br>table(sce2$celltype)<br>sce = sce2<br>table(sce$patient,sce$celltype)<br>table(sce$medicine,sce$celltype)<br>table(sce$orig.ident)<br>Idents(sce) = sce$patient<br>sce=CreateSeuratObject(counts = sce@assays$RNA@counts,<br>                       meta.data = sce@meta.data) <br><br><br>sce &lt;- NormalizeData(sce, normalization.method =  <span>"LogNormalize"</span>,  <br>                     scale.factor = <span>1</span>e4)<br>GetAssay(sce,assay = <span>"RNA"</span>)<br>sce &lt;- FindVariableFeatures(sce, <br>                            selection.method = <span>"vst"</span>, nfeatures = <span>2000</span>)  <br>sce &lt;- ScaleData(sce) <br>sce &lt;- RunPCA(object = sce, pc.genes = VariableFeatures(sce)) <br>DimHeatmap(sce, dims = <span>1</span><span>:</span><span>12</span>, cells = <span>100</span>, balanced = TRUE)<br>ElbowPlot(sce) <br><br>sce &lt;- FindNeighbors(sce, dims = <span>1</span><span>:</span><span>15</span>)<br>sce &lt;- FindClusters(sce, resolution = <span>0</span>.<span>1</span>)<br>table(sce@meta.data$RNA_snn_res.<span>0</span>.<span>1</span>)  <br><br>set.seed(<span>123</span>)<br>sce &lt;- RunTSNE(object = sce, dims = <span>1</span><span>:</span><span>15</span>, <span>do</span>.fast = TRUE)<br>saveRDS(sce,sce.all, <span>"sce_epi.rds"</span>)<br><br><br>sce &lt;- RunUMAP(object = sce, dims = <span>1</span><span>:</span><span>5</span>, <span>do</span>.fast = TRUE)<br>DimPlot(sce,reduction = <span>"umap"</span>,label=T) <br>head(sce@meta.data)<br>DimPlot(sce,reduction = <span>"umap"</span>,label=T,group.by = <span>'orig.ident'</span>)<br><br>mycolors &lt;-c(<span>'#E5D2DD'</span>, <span>'#53A85F'</span>, <span>'#F1BB72'</span>, <span>'#F3B1A0'</span>, <span>'#D6E7A3'</span>, <span>'#57C3F3'</span>,<br>                      <span>'#E95C59'</span>, <span>'#E59CC4'</span>, <span>'#AB3282'</span>,  <span>'#BD956A'</span>, <br>                      <span>'#9FA3A8'</span>, <span>'#E0D4CA'</span>,  <span>'#C5DEBA'</span>, <span>'#F7F398'</span>,<br>                      <span>'#C1E6F3'</span>, <span>'#6778AE'</span>, <span>'#91D0BE'</span>, <span>'#B53E2B'</span>,<br>                      <span>'#712820'</span>, <span>'#DCC1DD'</span>, <span>'#CCE0F5'</span>,  <span>'#CCC9E6'</span>, <span>'#625D9E'</span>, <span>'#68A180'</span>, <span>'#3A6963'</span>,<br>                      <span>'#968175'</span>)<br>library(harmony)<br>table(sce$orig.ident)<br>seuratObj &lt;- RunHarmony(sce, <span>"orig.ident"</span>)<br>names(seuratObj@reductions)<br>seuratObj &lt;- RunUMAP(seuratObj,  dims = <span>1</span><span>:</span><span>8</span>, <br>                     reduction = <span>"harmony"</span>)<br>DimPlot(seuratObj,reduction = <span>"umap"</span>,label=T )  <br>library(patchwork)<br>library(RColorBrewer)<br>library(scales)<br>col4&lt;-colorRampPalette(brewer.pal(<span>8</span>,<span>'Set2'</span>))(<span>12</span>)<br>DimPlot(seuratObj, reduction = <span>'umap'</span>,pt.size = <span>2</span>,cols = mycolors,label = T,group.by = <span>"RNA_snn_res.0.1"</span>,label.size = <span>6</span>,raster=FALSE) <br><br><br>sce=seuratObj<br>sce &lt;- FindNeighbors(sce, reduction = <span>"harmony"</span>,<br>                     dims = <span>1</span><span>:</span><span>5</span>)<br>sce &lt;- FindClusters(sce, resolution = <span>0</span>.<span>5</span>)<br>table(sce@meta.data$seurat_clusters)<br>DimPlot(sce,reduction = <span>"umap"</span>,label=T) <br><span>##居然发现还是原始配色好看点，我哭死</span><br>DimPlot(sce, reduction = <span>'umap'</span>,pt.size = <span>1</span>,label = T,group.by = <span>"RNA_snn_res.0.5"</span>,label.size = <span>6</span>,raster=FALSE) <br>ggsave(filename = <span>'harmony_umap_recluster_by_0.5.pdf'</span>,width = <span>7</span>,height = <span>7</span>) <br><br>col4&lt;-colorRampPalette(brewer.pal(<span>8</span>,<span>'Set2'</span>))(<span>15</span>)<br>sce$orig.ident<br>DimPlot(sce, reduction = <span>'umap'</span>,pt.size = <span>1</span>,cols = col4,label = T,group.by = <span>"patient"</span>,label.size = <span>6</span>,raster=FALSE) <br>ggsave(filename = <span>'harmony_umap_sce_recluster_by_patient.pdf'</span>,width = <span>7</span>,height = <span>7</span>) <br><br><span>#####患者之间的差异</span><br>DimPlot(sce, reduction = <span>'umap'</span>,pt.size = <span>1</span>,cols = col4,label = T,split.by = <span>'orig.ident'</span>,label.size = <span>6</span>,raster=FALSE) <br>ggsave(filename = <span>'umap_sce_recluster_treatment.pdf'</span>,width = <span>10</span>,height = <span>6.5</span>) <br><br><span>###配色什么的，不重要啦，喜欢什么颜色就放什么颜色</span><br>saveRDS(sce,file = <span>"sce_epi.rds"</span>)<br><br><br><br><span>###copykat区分肿瘤细胞和正常细胞</span><br>devtools::install_github(<span>"navinlabcode/copykat"</span>)<br>library(<span>"copykat"</span>) <br><br>library(Seurat)<br>library(copykat)<br>library(tidyverse)<br>scRNA &lt;- sce<br>counts &lt;- as.matrix(scRNA@assays$RNA@counts)<br>table(scRNA$celltype)<br>cnv &lt;- copykat(rawmat=counts, ngene.chr=<span>5</span>, sam.name=<span>"test"</span>, n.cores=<span>8</span>)<br><span># ngene.chr参数是过滤细胞的一个标准，它要求被用来做CNV预测的细胞，一个染色体上至少有5个基因。</span><br><span># sam.name定义样本名称 (sample name)，会给出来的文件加前缀</span><br>saveRDS(cnv, <span>"cnv.rds"</span>)<br>table(rownames(cnv$CNAmat))<br>a = cnv$prediction$copykat.pred<br>table(a)<br>scRNA$CopyKAT = a<br>p1 &lt;- DimPlot(scRNA, group.by = <span>"celltype"</span>, label = T)<br>p2 &lt;- DimPlot(scRNA, group.by = <span>"CopyKAT"</span>) + scale_color_manual(values = c(<span>"#F8766D"</span>,<span>'#02BFC4'</span>, <span>"gray"</span>))<br>pc &lt;- p1 + p2<br>pc<br>ggsave(<span>"pred_mallignant.pdf"</span>, pc, width = <span>12</span>, height = <span>5</span>)<br><span>###虽然这里得到结果了，但是我觉得上面选5数值太大了，也不想重新跑一遍几小时了....</span><br><br><br>cols = c(<span>"gray"</span>, <span>"coral2"</span>)<br>plot &lt;- FeaturePlot(scRNA, features = c(<span>'AGER'</span>,<span>'SFTPA2'</span>,<span>'SCGB1A1'</span>,<span>'TPPP3'</span>,<span>'KRT17'</span>),cols = cols, pt.size = <span>1</span>)+  <br>  theme(panel.border = element_rect(fill=NA,color=<span>"black"</span>, size=<span>1</span>, linetype=<span>"solid"</span>))<span>#加边框 </span><br>plot_grid(p2,plot)<br><br><br><span>#####注释亚群</span><br><span>#marker</span><br><span>#1. DST：参与细胞骨架的形成和维护，细胞附着和迁移的调节，肌肉成熟和神经元发育等过程。</span><br><span>#2. KRT17：编码角质形成细胞中的一种角蛋白，参与细胞骨架的形成和角质形成。</span><br><span>#3. S100P：与钙离子结合的蛋白质，参与细胞增殖、迁移和侵袭等生物学过程，也可能与多种癌症的发生和发展有关。</span><br><span>#4. PCNA：是细胞周期中S期的细胞核蛋白，在DNA复制和细胞增殖过程中起重要作用。</span><br><span>#5. TOP2A：编码一种DNA拓扑异构酶，参与DNA的超螺旋松弛和重连，影响DNA复制和细胞分裂。</span><br><span>#6. SFTPA2：编码一种肺表面活性物质蛋白，参与肺泡的防御和免疫功能。</span><br><span>#7. SPARCL1：编码一种分泌蛋白，参与肿瘤抑制、细胞迁移和基质重塑等过程。</span><br><span>#8. SERPINB9：编码一种丝氨酸蛋白酶抑制剂，参与炎症调节和细胞凋亡等过程。</span><br><span>#9. SCGB3A1：编码一种分泌蛋白，可能与肺部基底细胞的分化和肺免疫功能有关。</span><br><span>#10. TPPP3：编码一种调节微管动力学的蛋白，可能参与细胞骨架的重塑和细胞运动等生物学过程。</span><br>genes_to_check = c(<span>'DST'</span>, <span>'AKR1B10'</span>, <span>'NDUFA4L2'</span>, <br>                   <span>'KRT17'</span> , <span>'CEACAM5'</span>, <span>'LY6G6C'</span>, <span>'KLK10'</span>,<br>                   <span>'S100P'</span>, <span>'S100A9'</span>, <span>'MT2A'</span>,<br>                   <span>'PCNA'</span>, <span>'TYMS'</span>, <span>'HIST1H2BJ'</span>, <span>'RRM2'</span>,<span>'DUT'</span>,<br>                   <span>'TOP2A'</span>, <span>'TPX2'</span>, <span>'CENPF'</span>,<span>'ARL6IP1'</span>,<br>                   <span>'SFTPA2'</span>, <span>'SFTPA1'</span>, <span>'SFTPB'</span>,<br>                   <span>'SPARCL1'</span> ,<span>'IGKC'</span>,<span>'AL365357.1'</span>,<br>                   <span>'SERPINB9'</span>, <span>'MYH11'</span> ,<span>'GJB6'</span>,<br>                   <span>'SCGB3A1'</span>,<span>'MSMB'</span>, <span>'SCGB1A1'</span>, <span>'SLPI'</span>,<br>                   <span>'TPPP3'</span>,<span>'CAPS'</span>, <span>'C9orf24'</span>, <span>'RSPH1'</span>)<br><br>library(stringr)  <br>genes_to_check=str_to_upper(genes_to_check)<br>genes_to_check<br>p &lt;- DotPlot(scRNA, features = unique(genes_to_check),<br>             assay=<span>'RNA'</span>  )  + coord_flip()<br><br>mycolors &lt;-c(<span>'#E5D2DD'</span>, <span>'#53A85F'</span>, <span>'#F1BB72'</span>, <span>'#F3B1A0'</span>, <span>'#D6E7A3'</span>, <span>'#57C3F3'</span>,<br>                      <span>'#E95C59'</span>, <span>'#E59CC4'</span>, <span>'#AB3282'</span>,  <span>'#BD956A'</span>, <br>                      <span>'#9FA3A8'</span>, <span>'#E0D4CA'</span>,  <span>'#C5DEBA'</span>, <span>'#F7F398'</span>,<br>                      <span>'#C1E6F3'</span>, <span>'#6778AE'</span>, <span>'#91D0BE'</span>, <span>'#B53E2B'</span>,<br>                      <span>'#712820'</span>, <span>'#DCC1DD'</span>, <span>'#CCE0F5'</span>,  <span>'#CCC9E6'</span>, <span>'#625D9E'</span>, <span>'#68A180'</span>, <span>'#3A6963'</span>,<br>                      <span>'#968175'</span>)<br><br>p_umap=DimPlot(scRNA, reduction = <span>"umap"</span>,cols = mycolors,pt.size = <span>0</span>.<span>4</span>,<br>               group.by = <span>"RNA_snn_res.0.5"</span>,label = T,label.box = T) <br>p_umap<br>library(patchwork)<br>p+p_umap<br>ggsave(<span>'markers_umap_2.pdf'</span>,width = <span>13</span>,height = <span>8</span>)<br><br><span>#####细胞生物学命名</span><br>celltype=data.frame(ClusterID=<span>0</span><span>:</span><span>13</span>,<br>                    celltype= <span>0</span><span>:</span><span>13</span>) <br><br><span>#定义细胞亚群 </span><br>celltype[celltype$ClusterID %<span>in</span>% c( <span>0</span> ),<span>2</span>]=<span>'E0_DST'</span><br>celltype[celltype$ClusterID %<span>in</span>% c( <span>9</span> ),<span>2</span>]=<span>'E1_KRT17'</span><br>celltype[celltype$ClusterID %<span>in</span>% c(<span>4</span>,<span>6</span>,<span>10</span>,<span>13</span> ),<span>2</span>]=<span>'E2_S100P'</span><br>celltype[celltype$ClusterID %<span>in</span>% c(<span>2</span>  ),<span>2</span>]=<span>'E3_PCNA'</span><br>celltype[celltype$ClusterID %<span>in</span>% c( <span>7</span> ),<span>2</span>]=<span>'E4_TOP2A'</span><br>celltype[celltype$ClusterID %<span>in</span>% c( <span>5</span>,<span>12</span> ),<span>2</span>]=<span>'E5_SFTPA2'</span><br>celltype[celltype$ClusterID %<span>in</span>% c( <span>3</span> ),<span>2</span>]=<span>'E6_SPARCL1'</span><br>celltype[celltype$ClusterID %<span>in</span>% c( <span>8</span> ),<span>2</span>]=<span>'E7_SERPINB9'</span><br>celltype[celltype$ClusterID %<span>in</span>% c(<span>1</span> ),<span>2</span>]=<span>'E8_SCGB3A1'</span><br>celltype[celltype$ClusterID %<span>in</span>% c( <span>11</span>),<span>2</span>]=<span>'E9_TPPP3'</span><br><br><br>head(celltype)<br>celltype<br>table(celltype$celltype)<br><br>scRNA@meta.data$celltype = <span>"NA"</span><br><span>for</span>(i <span>in</span> <span>1</span><span>:nrow</span>(celltype)){<br>  scRNA@meta.data[which(scRNA@meta.data$RNA_snn_res.<span>0</span>.<span>5</span> == celltype$ClusterID[i]),<span>'celltype'</span>] &lt;- celltype$celltype[i]}<br>table(scRNA@meta.data$celltype)<br><br><br>th=theme(axis.text.x = element_text(angle = <span>45</span>, <br>                                    vjust = <span>0</span>.<span>5</span>, hjust=<span>0</span>.<span>5</span>)) <br>library(patchwork)<br>p_umap=DimPlot(scRNA, reduction = <span>"umap"</span>,cols = mycolors,pt.size = <span>0</span>.<span>4</span>,<br>               group.by = <span>"celltype"</span>,label = T) <br>p+p_umap<br>ggsave(<span>'markers_umap_by_celltype_2.pdf'</span>,width = <span>15</span>,height = <span>7</span>)<br>saveRDS(scRNA, <span>"sce.epi_celltype.rds"</span>)<br></code></pre><figure><img data-ratio="0.46111111111111114" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJqxEVMMrKOBsMeANu6jsZSJ2bLfGoATiam1icnqWMgXKibnlXK4wqzmSkAQ/640?wx_fmt=png" data-type="png" data-w="1080" title="image.png" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJqxEVMMrKOBsMeANu6jsZSJ2bLfGoATiam1icnqWMgXKibnlXK4wqzmSkAQ/640?wx_fmt=png"></figure><br>可以看出E2，5，8，9亚群恶性细胞群较少或没有<br><figure><img data-ratio="0.5351851851851852" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJqHfdUcugmqQYRE30zibL8zxRv5prw9Jeg28hjfAC3WsIdQMSNkzhcxKw/640?wx_fmt=png" data-type="png" data-w="1080" title="image.png" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJqHfdUcugmqQYRE30zibL8zxRv5prw9Jeg28hjfAC3WsIdQMSNkzhcxKw/640?wx_fmt=png"></figure><pre><code><span>######基因在不同病理结局的患者中表达情况</span><br>rm(<span>list</span>=ls())<br>scRNA=readRDS( <span>"sce.epi_celltype.rds"</span>)<br>scRNA<br>table(scRNA$CopyKAT)<br><span>##在这里我查看了加入无法分类的细胞和未加入无法分类的细胞，差别还挺大的</span><br><span>#加入：CX3CL1趋势不对，AKR1C3趋势正确</span><br><span>#不加入：CX3CL1趋势正确，AKR1C3趋势不对</span><br><br><br>sce_obj = scRNA[, scRNA$CopyKAT %in% c( <span>'aneuploid'</span>)]<br>sce_obj$patientoutcone =  paste(sce_obj$patient,sce_obj$Pathologic,sep = <span>'_'</span>)<br><br><br><span># 通过Seurat的AverageExpression函数计算每个患者基因的平均表达量</span><br>avg &lt;- AverageExpression(object = sce_obj, assay = <span>"RNA"</span>, group.by = <span>"patientoutcone"</span>)<br>a = avg$RNA<br><br><br>C1 = c(<span>'CX3CL1'</span>,<span>'CD74'</span>,<span>'HLA-DRA'</span>)<br>D1 = c(<span>'AKR1C1'</span>,<span>'AKR1C2'</span>,<span>'AKR1C3'</span>)<br><span>#这六个基因的生物学意义如下：</span><br><span>#1. CX3CL1是一种细胞因子，也被称为神经内皮炎性因子1（N-TAC），它是一种趋化因子，在炎症和免疫反应中起着重要作用。它与免疫细胞的迁移和炎症反应相关，同时也参与神经发育和神经保护。</span><br><span>#2. CD74是一种细胞膜蛋白，也被称为类别II转运分子细胞膜关联蛋白。CD74与抗原呈递、免疫调节和细胞信号传导等免疫功能密切相关。它在抗原处理和递呈的过程中起重要作用，并在免疫细胞识别、炎症和免疫反应中发挥关键作用。</span><br><span>#3. HLA-DRA（Human Leukocyte Antigen-DR α chain）：HLA-DRA是人白细胞抗原-DR（HLA-DR）的α链。HLA-DR是一种MHC类II复合物，负责呈递外源性抗原给T细胞，从而触发免疫反应。HLA-DRA与免疫系统的抗原递呈、T细胞的激活和调节密切相关，在免疫途径中发挥重要作用。</span><br><span>#4. AKR1C1属于醛酮还原酶家族的一员。它是一种脱氢酶，对多种内源性和外源性化合物具有还原活性。AKR1C1参与睾酮、孕酮等激素的代谢转化，同时还参与抗氧化和抗炎过程。</span><br><span>#5. AKR1C2也是醛酮还原酶家族的成员，与AKR1C1具有相似的酶活性。它在多种生物学过程中发挥重要作用，包括激素代谢、抗氧化、抗炎和神经保护。</span><br><span>#6. AKR1C3是醛酮还原酶家族的一员，与AKR1C1和AKR1C2具有相似的酶活性。它参与睾酮、雄激素和孕激素的代谢转化，同时还具有抗氧化和抗炎活性。在某些癌症中，AKR1C3的过度表达与抗药性和肿瘤的恶性进展有关。</span><br>gene_boxplot &lt;- <span><span>function</span><span>(gene)</span></span>{<br>  b = <span>as</span>.data.frame(a[gene,])<br>  colnames(b) = gene<br>  phe = b<br>  group = ifelse(str_detect(rownames(phe),<span>"TN"</span>),<span>"TN"</span>,<br>                 ifelse(str_detect(rownames(phe),<span>"NMPR"</span>),<span>"NMPR"</span>,<span>"MPR"</span>))<br>  phe$group = group<br><br>  phe$group = factor(phe$group,levels =c(<span>"TN"</span>,<span>"MPR"</span>,<span>"NMPR"</span>))<br>  table(phe$group)<br>  colnames(phe)<br><br>  col = c(<span>"#508E42"</span>, <span>"#DF5E38"</span>,<span>"#7B7DC4"</span>)<br>  P_T &lt;- ggplot(phe,aes(x=group,y=!!rlang::sym(gene),fill=group)) +<br>    geom_boxplot(size=<span>0.5</span>,fill=<span>"white"</span>,outlier.fill=<span>"white"</span>,outlier.color=<span>"white"</span>,color=col) +<br>    geom_jitter(aes(fill=group),width =<span>0.2</span>,shape = <span>21</span>,size=<span>2</span>) +<br>    scale_fill_manual(values = col)+<br>    ggtitle(gene) +<br>    theme_bw() +<br>    theme(panel.grid = element_blank(), panel.background = element_blank(),<br>          axis.line = element_line(color = <span>'black'</span>)) +<br>    ylab(<span>"Average signature score"</span>) +xlab(<span>" "</span>)+<br>    theme(legend.position = <span>"none"</span>,<br>          axis.text.x = element_text(colour = <span>"black"</span>, family = <span>"Times"</span>, size = <span>14</span>),<br>          axis.text.y = element_text(family = <span>"Times"</span>, size = <span>14</span>, face = <span>"plain"</span>),<br>          axis.title.y = element_text(family = <span>"Times"</span>, size = <span>14</span>, face = <span>"plain"</span>),<br>          axis.title.x = element_text(family = <span>"Times"</span>, size = <span>14</span>, face = <span>"plain"</span>),<br>          plot.title = element_text(family = <span>"Times"</span>, size = <span>17</span>, face = <span>"bold"</span>, hjust = <span>0.5</span>),<br>          panel.grid.major = element_blank(),<br>          panel.grid.minor = element_blank(),<br>          panel.border = element_blank())<br><br>  P_T<br>  compaired = <span>list</span>(c(<span>"TN"</span>, <span>"MPR"</span>),<br>                   c(<span>"TN"</span>, <span>"NMPR"</span>))<br>  P_T = P_T+geom_signif(comparisons = compaired,step_increase = <span>0.1</span>,map_signif_level = T,test = t.test,tip_length = <span>0</span>)<br>  P_T<br><br>}<br><br>P_C1 = gene_boxplot(C1[<span>1</span>])+gene_boxplot(C1[<span>2</span>])+gene_boxplot(C1[<span>3</span>])<br>P_D1 = gene_boxplot(D1[<span>1</span>])+gene_boxplot(D1[<span>2</span>])+gene_boxplot(D1[<span>3</span>])<br><br>P_C1/P_D1<br><span>###文中主要是看了两个基因的变化，CX3CL1在MPR的患者中上调，AKR1C3在NMPR的患者中上调</span><br>ggsave(<span>'gene_outcome.pdf'</span>,width = <span>13</span>,height = <span>8</span>)<br>saveRDS(sce_obj, <span>"sce.cancer_celltype.rds"</span>)<br></code></pre><figure><img data-ratio="0.5925925925925926" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJqlGS79MRljibcr2VDMpAdDmmicCKq9XemLmVCpic0RZ0ELtPdlogZPLEqA/640?wx_fmt=png" data-type="png" data-w="1080" title="image.png" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJqlGS79MRljibcr2VDMpAdDmmicCKq9XemLmVCpic0RZ0ELtPdlogZPLEqA/640?wx_fmt=png"></figure><pre><code><span>###GSVA分析<br>rm(list=ls())<br>sce.all=readRDS( "sce.cancer_celltype.rds")<br><br>scRNA <span>&lt;<span>-</span> <span>sce.all</span><br><span>library</span>(<span>msigdbr</span>)<br><span>library</span>(<span>GSVA</span>)<br><span>library</span>(<span>Seurat</span>)<br>#选择基因集<br><span>Idents</span>(<span>scRNA</span>) = <span>scRNA$patientoutcone</span><br>#<span>Idents</span>设置按什么来取组的表达均值（计算<span>case</span> <span>control</span>之间的均值也可以）<br><span>expr</span> &lt;<span>-</span> <span>AverageExpression</span>(<span>scRNA</span>, <span>assays</span> = <span>"RNA"</span>, <span>slot</span> = <span>"data"</span>)[[<span>1</span>]]<br><span>expr</span> &lt;<span>-</span> <span>expr</span>[<span>rowSums</span>(<span>expr</span>)&gt;</span>0,]  #选取非零基因<br>expr <span>&lt;<span>-</span> <span>as.matrix</span>(<span>expr</span>)<br><span>library</span>(<span>GSEABase</span>)<br><br><span>gmt</span> &lt;<span>-</span> '<span>msigdb_v2023.1.Hs_files_to_download_locally</span>/<span>msigdb_v2023.1.Hs_files_to_download_locally</span>/<span>msigdb_v2023.1.Hs_GMTs</span>/<span>h.all.v2023.1.Hs.symbols.gmt</span>'<br><span>geneSet</span> &lt;<span>-</span> <span>getGmt</span>(<span>gmt</span>)<br># <span>gsva</span>(若为<span>count</span>矩阵，<span>kcdf</span> = <span>'Poisson'</span>，若为其他标准化后的数据，使用默认参数即可)<br><span>gsva_scores</span> &lt;<span>-</span> <span>gsva</span>(<span>expr</span>,<span>geneSet</span>,<span>method</span> = <span>'gsva'</span>,<span>min.sz</span>=<span>5,max.sz</span>=<span>500,kcdf</span> = <span>'Poisson'</span>)<br><span>head</span>(<span>gsva_scores</span>)<br><span>gsva_scores</span> = <span>as.data.frame(gsva_scores)</span><br><br><span>library</span>(<span>stringr</span>)<br><span>library</span>(<span>ggplot2</span>)<br><span>library</span>(<span>ggsignif</span>)<br><span>b</span> = <span>as.data.frame(t(gsva_scores[</span>'<span>HALLMARK_ESTROGEN_RESPONSE_LATE</span>',]))<br><span>colnames</span>(<span>b</span>) = <span>'Estrogen response late'</span><br><span>phe</span> = <span>b</span><br><span>group</span> = <span>ifelse(str_detect(rownames(phe),</span>"<span>TN</span>"),"<span>TN</span>",<br>               <span>ifelse</span>(<span>str_detect</span>(<span>rownames</span>(<span>phe</span>),"<span>NMPR</span>"),"<span>NMPR</span>","<span>MPR</span>"))<br><span>phe</span>$<span>group</span> = <span>group</span><br><br><span>phe</span>$<span>group</span> = <span>factor(phe$group,levels</span> =<span>c(</span>"<span>TN</span>","<span>MPR</span>","<span>NMPR</span>"))<br><span>table</span>(<span>phe</span>$<span>group</span>)<br><span>colnames</span>(<span>phe</span>)<br><br><span>col</span> = <span>c(</span>"#<span>508E42</span>", "#<span>DF5E38</span>","#<span>7B7DC4</span>")<br><span>pathway</span> = <span>'Estrogen response late'</span><br><span>P_T</span> &lt;<span>-</span> <span>ggplot</span>(<span>phe</span>,<span>aes</span>(<span>x</span>=<span>group,y</span>=<span>!!rlang::sym(pathway),fill</span>=<span>group))</span> +<br>  <span>geom_boxplot</span>(<span>size</span>=<span>0.5,fill</span>=<span>"white"</span>,<span>outlier.fill</span>=<span>"white"</span>,<span>outlier.color</span>=<span>"white"</span>,<span>color</span>=<span>col)</span> +<br>  <span>geom_jitter</span>(<span>aes</span>(<span>fill</span>=<span>group),width</span> =<span>0.2,shape</span> = <span>21,size</span>=<span>2)</span> +<br>  <span>scale_fill_manual</span>(<span>values</span> = <span>col)+</span><br>  <span>ggtitle</span>(<span>pathway</span>) +<br>  <span>theme_bw</span>() +<br>  <span>theme</span>(<span>panel.grid</span> = <span>element_blank(),</span> <span>panel.background</span> = <span>element_blank(),</span><br>        <span>axis.line</span> = <span>element_line(color</span> = <span>'black'</span>)) +<br>  <span>ylab</span>("<span>Average</span> <span>signature</span> <span>score</span>") +<span>xlab</span>(" ")+<br>  <span>theme</span>(<span>legend.position</span> = <span>"none"</span>,<br>        <span>axis.text.x</span> = <span>element_text(colour</span> = <span>"black"</span>, <span>family</span> = <span>"Times"</span>, <span>size</span> = <span>14),</span><br>        <span>axis.text.y</span> = <span>element_text(family</span> = <span>"Times"</span>, <span>size</span> = <span>14,</span> <span>face</span> = <span>"plain"</span>),<br>        <span>axis.title.y</span> = <span>element_text(family</span> = <span>"Times"</span>, <span>size</span> = <span>14,</span> <span>face</span> = <span>"plain"</span>),<br>        <span>axis.title.x</span> = <span>element_text(family</span> = <span>"Times"</span>, <span>size</span> = <span>14,</span> <span>face</span> = <span>"plain"</span>),<br>        <span>plot.title</span> = <span>element_text(family</span> = <span>"Times"</span>, <span>size</span> = <span>17,</span> <span>face</span> = <span>"bold"</span>, <span>hjust</span> = <span>0.5),</span><br>        <span>panel.grid.major</span> = <span>element_blank(),</span><br>        <span>panel.grid.minor</span> = <span>element_blank(),</span><br>        <span>panel.border</span> = <span>element_blank())</span><br><br><span>P_T</span><br><span>compaired</span> = <span>list(c(</span>"<span>TN</span>", "<span>MPR</span>"),<br>                 <span>c</span>("<span>TN</span>", "<span>NMPR</span>"))<br><span>P_T</span> = <span>P_T+geom_signif(comparisons</span> = <span>compaired,step_increase</span> = <span>0.1,map_signif_level</span> = <span>T,test</span> = <span>t.test,tip_length</span> = <span>0)</span><br><span>P_T</span><br><br><br><span>gmt</span> &lt;<span>-</span> '<span>msigdb_v2023.1.Hs_GMTs</span>/<span>c5.go.bp.v2023.1.Hs.symbols.gmt</span>'<br><span>geneSet</span> &lt;<span>-</span> <span>getGmt</span>(<span>gmt</span>)<br># <span>gsva</span>(若为<span>count</span>矩阵，<span>kcdf</span> = <span>'Poisson'</span>，若为其他标准化后的数据，使用默认参数即可)<br><span>gsva_scores</span> &lt;<span>-</span> <span>gsva</span>(<span>expr</span>,<span>geneSet</span>,<span>method</span> = <span>'gsva'</span>,<span>min.sz</span>=<span>5,max.sz</span>=<span>500,kcdf</span> = <span>'Poisson'</span>)<br><span>head</span>(<span>gsva_scores</span>)<br><span>gsva_scores</span> = <span>as.data.frame(gsva_scores)</span><br><br><br><span>library</span>(<span>stringr</span>)<br><span>library</span>(<span>ggplot2</span>)<br><span>library</span>(<span>ggsignif</span>)<br><span>b</span> = <span>as.data.frame(t(gsva_scores[</span>'<span>GOBP_ANTIGEN_PROCESSING_AND_PRESENTATION_OF_PEPTIDE_OR_POLYSACCHARIDE_ANTIGEN_VIA_MHC_CLASS_II</span>',]))<br><span>colnames</span>(<span>b</span>) = <span>'Antigen presentation via MHC-II'</span><br><span>phe</span> = <span>b</span><br><span>group</span> = <span>ifelse(str_detect(rownames(phe),</span>"<span>TN</span>"),"<span>TN</span>",<br>               <span>ifelse</span>(<span>str_detect</span>(<span>rownames</span>(<span>phe</span>),"<span>NMPR</span>"),"<span>NMPR</span>","<span>MPR</span>"))<br><span>phe</span>$<span>group</span> = <span>group</span><br><br><span>phe</span>$<span>group</span> = <span>factor(phe$group,levels</span> =<span>c(</span>"<span>TN</span>","<span>MPR</span>","<span>NMPR</span>"))<br><span>table</span>(<span>phe</span>$<span>group</span>)<br><span>colnames</span>(<span>phe</span>)<br><br><span>col</span> = <span>c(</span>"#<span>508E42</span>", "#<span>DF5E38</span>","#<span>7B7DC4</span>")<br><span>pathway</span> = <span>'Antigen presentation via MHC-II'</span><br><span>P_T</span> &lt;<span>-</span> <span>ggplot</span>(<span>phe</span>,<span>aes</span>(<span>x</span>=<span>group,y</span>=<span>!!rlang::sym(pathway),fill</span>=<span>group))</span> +<br>  <span>geom_boxplot</span>(<span>size</span>=<span>0.5,fill</span>=<span>"white"</span>,<span>outlier.fill</span>=<span>"white"</span>,<span>outlier.color</span>=<span>"white"</span>,<span>color</span>=<span>col)</span> +<br>  <span>geom_jitter</span>(<span>aes</span>(<span>fill</span>=<span>group),width</span> =<span>0.2,shape</span> = <span>21,size</span>=<span>2)</span> +<br>  <span>scale_fill_manual</span>(<span>values</span> = <span>col)+</span><br>  <span>ggtitle</span>(<span>pathway</span>) +<br>  <span>theme_bw</span>() +<br>  <span>theme</span>(<span>panel.grid</span> = <span>element_blank(),</span> <span>panel.background</span> = <span>element_blank(),</span><br>        <span>axis.line</span> = <span>element_line(color</span> = <span>'black'</span>)) +<br>  <span>ylab</span>("<span>Average</span> <span>signature</span> <span>score</span>") +<span>xlab</span>(" ")+<br>  <span>theme</span>(<span>legend.position</span> = <span>"none"</span>,<br>        <span>axis.text.x</span> = <span>element_text(colour</span> = <span>"black"</span>, <span>family</span> = <span>"Times"</span>, <span>size</span> = <span>14),</span><br>        <span>axis.text.y</span> = <span>element_text(family</span> = <span>"Times"</span>, <span>size</span> = <span>14,</span> <span>face</span> = <span>"plain"</span>),<br>        <span>axis.title.y</span> = <span>element_text(family</span> = <span>"Times"</span>, <span>size</span> = <span>14,</span> <span>face</span> = <span>"plain"</span>),<br>        <span>axis.title.x</span> = <span>element_text(family</span> = <span>"Times"</span>, <span>size</span> = <span>14,</span> <span>face</span> = <span>"plain"</span>),<br>        <span>plot.title</span> = <span>element_text(family</span> = <span>"Times"</span>, <span>size</span> = <span>17,</span> <span>face</span> = <span>"bold"</span>, <span>hjust</span> = <span>0.5),</span><br>        <span>panel.grid.major</span> = <span>element_blank(),</span><br>        <span>panel.grid.minor</span> = <span>element_blank(),</span><br>        <span>panel.border</span> = <span>element_blank())</span><br><br><span>P_T</span><br><span>compaired</span> = <span>list(c(</span>"<span>TN</span>", "<span>MPR</span>"),<br>                 <span>c</span>("<span>TN</span>", "<span>NMPR</span>"))<br><span>P_T</span> = <span>P_T+geom_signif(comparisons</span> = <span>compaired,step_increase</span> = <span>0.1,map_signif_level</span> = <span>T,test</span> = <span>t.test,tip_length</span> = <span>0)</span><br><span>P_T</span><br></span></span></code></pre><p>MHC-II确实是在MPR的患者中上调了，但是这个文章立足之根本的雌激素反应对不上，实在是对不上啊。<br><span><strong>（PS: 埋头苦干三个小时了，此时我正在看着楼下的人在小湖边钓鱼，好想去看😶‍🌫️）</strong></span><br></p><figure><img data-ratio="0.47685185185185186" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJqdkDEC74059LxOqeYpVP0zRIAMR1gu8VqTXXqNBknAx0bMyf7ibs0Hfg/640?wx_fmt=png" data-type="png" data-w="1080" title="image.png" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJqdkDEC74059LxOqeYpVP0zRIAMR1gu8VqTXXqNBknAx0bMyf7ibs0Hfg/640?wx_fmt=png"></figure><h2><span>3.NK_T细胞</span></h2><figure><img data-ratio="1.2814814814814814" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJq1R8S7OYXice6ahu6Ip3TK5chILKvqbpFrEoSPiatDyFV2G8U6ILvaBZg/640?wx_fmt=png" data-type="png" data-w="1080" title="image.png" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJq1R8S7OYXice6ahu6Ip3TK5chILKvqbpFrEoSPiatDyFV2G8U6ILvaBZg/640?wx_fmt=png"></figure><br>由于图形和上述都是一致的，就只放结果图了，请看：：：<br><figure><img data-ratio="0.5462962962962963" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJqibNEP8UXiaLamySHKHEicicVuW0ywjEjicjs2ojo91Xl9xHsEicA1duytIsA/640?wx_fmt=png" data-type="png" data-w="1080" title="967ac51f2459bf3a07641b203a53951.png" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJqibNEP8UXiaLamySHKHEicicVuW0ywjEjicjs2ojo91Xl9xHsEicA1duytIsA/640?wx_fmt=png"><figcaption><br></figcaption></figure><pre><code>rm(list=ls())<br>sce.all=readRDS( <span>"sce.NKT_celltype.rds"</span>)<br><br>seuratObj &lt;- sce.all<br>avg &lt;- AverageExpression(object = seuratObj, assay = <span>"RNA"</span>, group.by = <span>"celltype"</span>)<br>a = avg$RNA<br>b = as.data.frame(a)<br>str(b)<br>genes &lt;- c()<br><br><span>for</span> (i <span>in</span> seq_along(genes_to_check)) {<br>  genes &lt;- c(genes, unlist(genes_to_check<span>[[i]]</span>))<br>}<br>gene_heatmap = b[genes,]<br><br>column_ha = HeatmapAnnotation(<span>'Cell type'</span> = ifelse(str_detect(colnames(gene_heatmap),<span>"NK"</span>),<span>"NK"</span>,<br>                                                   ifelse(str_detect(colnames(gene_heatmap),<span>"CD8"</span>),<span>"CD8"</span>,<br>                                                          ifelse(str_detect(colnames(gene_heatmap),<span>"CD4"</span>),<span>"CD4"</span>,<br>                                                                 ifelse(str_detect(colnames(gene_heatmap),<span>"Treg"</span>),<span>"Treg"</span>,<span>"T"</span>)))) )<br><br>row_ha &lt;- rowAnnotation(<br>  <span>'Marker'</span> = row)<br><br>col_fun = colorRamp2(c(<span>0</span>, <span>2</span>, <span>7</span>), c(<span>"#4EC0DB"</span>, <span>"white"</span>, <span>"#E63946"</span>))<br><br>gene_list = genes_to_check<br>row &lt;- character()<br><span>for</span> (i <span>in</span> <span>1</span>:length(gene_list)) {<br>  row &lt;- c(row, <span>rep</span>(names(gene_list)[i], length(gene_list<span>[[i]]</span>)))<br>}<br><br>column = ifelse(str_detect(colnames(gene_heatmap),<span>"NK"</span>),<span>"NK"</span>,<br>                ifelse(str_detect(colnames(gene_heatmap),<span>"CD8"</span>),<span>"CD8"</span>,<br>                       ifelse(str_detect(colnames(gene_heatmap),<span>"CD4"</span>),<span>"CD4"</span>,<br>                              ifelse(str_detect(colnames(gene_heatmap),<span>"Treg"</span>),<span>"Treg"</span>,<span>"T"</span>))))<br><br>Heatmap(gene_heatmap, <br>        column_split = column,<br>        row_split = c(<span>'Marker'</span> = row),<br>        row_title = NULL,<br>        heatmap_legend_param = list(<br>          title = <span>"Score gene expression"</span>,<br>          legend_height = unit(<span>4</span>, <span>"cm"</span>),<br>          title_position = <span>"lefttop-rot"</span><br>        ),<br>        top_annotation = column_ha,<br>        left_annotation = row_ha,<br>        col = col_fun,<br>        cluster_rows = FALSE, cluster_columns = FALSE,column_title = NULL,<br>        show_row_names = T,show_column_names  = T<br>)<br></code></pre><p>总觉得配色很扎眼。<br></p><figure><img data-ratio="0.5444444444444444" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJq3PiciaFNSjsbSP5Sr6Zib2t9icB1fibibn7z6wGS7MlibodzEqOyoTKqo1wGA/640?wx_fmt=png" data-type="png" data-w="1080" title="a83e7beb9ccdbc0fe659350186808fb.png" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJq3PiciaFNSjsbSP5Sr6Zib2t9icB1fibibn7z6wGS7MlibodzEqOyoTKqo1wGA/640?wx_fmt=png"><figcaption><br></figcaption></figure><pre><code><span>###应该是基因的平均表达量</span><br><span>#取出CD8T</span><br><span>####NK和T细胞</span><br>table(seuratObj$celltype)<br>sce_CD8T = seuratObj[, seuratObj$celltype %in% c( <span>'CD8_GZMB'</span>,<span>'CD8_GZMK'</span>,<span>'CD8_HAVCR2'</span>,<span>'CD8_IL7R'</span>,<span>'CD8_STMN1'</span> )]<br>avg &lt;- AverageExpression(object = sce_CD8T, assay = <span>"RNA"</span>, group.by = <span>"patientoutcone"</span>)<br>a = avg$RNA<br>b = <span>as</span>.data.frame(a)<br>str(b)<br><br>gene_boxplot &lt;- <span><span>function</span><span>(gene)</span></span>{<br>  average_expression &lt;- colMeans(submatrix)<br>  average_expression = <span>as</span>.data.frame(average_expression)<br>  colnames(average_expression) = gene<br>  phe = average_expression<br>  group = ifelse(str_detect(rownames(phe),<span>"TN"</span>),<span>"TN"</span>,<br>                 ifelse(str_detect(rownames(phe),<span>"NMPR"</span>),<span>"NMPR"</span>,<span>"MPR"</span>))<br>  phe$group = group<br><br>  phe$group = factor(phe$group,levels =c(<span>"TN"</span>,<span>"MPR"</span>,<span>"NMPR"</span>))<br>  table(phe$group)<br>  colnames(phe)<br><br>  col = c(<span>"#508E42"</span>, <span>"#DF5E38"</span>,<span>"#7B7DC4"</span>)<br>  P_T &lt;- ggplot(phe,aes(x=group,y=!!rlang::sym(gene),fill=group)) +<br>    geom_boxplot(size=<span>0.5</span>,fill=<span>"white"</span>,outlier.fill=<span>"white"</span>,outlier.color=<span>"white"</span>,color=col) +<br>    geom_jitter(aes(fill=group),width =<span>0.2</span>,shape = <span>21</span>,size=<span>2</span>) +<br>    scale_fill_manual(values = col)+<br>    ggtitle(gene) +<br>    theme_bw() +<br>    theme(panel.grid = element_blank(), panel.background = element_blank(),<br>          axis.line = element_line(color = <span>'black'</span>)) +<br>    ylab(<span>"Average signature score"</span>) +xlab(<span>" "</span>)+<br>    theme(legend.position = <span>"none"</span>,<br>          axis.text.x = element_text(colour = <span>"black"</span>, family = <span>"Times"</span>, size = <span>14</span>),<br>          axis.text.y = element_text(family = <span>"Times"</span>, size = <span>14</span>, face = <span>"plain"</span>),<br>          axis.title.y = element_text(family = <span>"Times"</span>, size = <span>14</span>, face = <span>"plain"</span>),<br>          axis.title.x = element_text(family = <span>"Times"</span>, size = <span>14</span>, face = <span>"plain"</span>),<br>          plot.title = element_text(family = <span>"Times"</span>, size = <span>17</span>, face = <span>"bold"</span>, hjust = <span>0.5</span>),<br>          panel.grid.major = element_blank(),<br>          panel.grid.minor = element_blank(),<br>          panel.border = element_blank())<br><br>  P_T<br>  compaired = <span>list</span>(c(<span>"TN"</span>, <span>"MPR"</span>),<br>                   c(<span>"TN"</span>, <span>"NMPR"</span>))<br>  P_T = P_T+geom_signif(comparisons = compaired,step_increase = <span>0.1</span>,map_signif_level = T,test = t.test,tip_length = <span>0</span>)<br>  P_T<br><br>}<br><br><span># Cytotoxic基因名称列表</span><br>Cytotoxic_gene &lt;- c(<span>"GZMA"</span>, <span>"GZMB"</span>, <span>"GZMK"</span> ,<span>"GNLY"</span>, <span>"IFNG"</span> ,<span>"PRF1"</span> ,<span>"NKG7"</span>)<br><br>submatrix &lt;- b[Cytotoxic_gene, ]<br>gene_boxplot(<span>'Cytotoxic signature'</span>)<br><br><span># Exhausted基因名称列表</span><br><span>##看，确实有CD39，果然是高耗竭基因啊</span><br>gene_list<br>Exhausted_gene &lt;- c(<span>"LAG3"</span> ,  <span>"TIGIT"</span> , <span>"PDCD1"</span> , <span>"HAVCR2"</span>, <span>"CTLA4"</span> , <span>"LAYN"</span>  , <span>"ENTPD1"</span>)<br><br>submatrix &lt;- b[Exhausted_gene, ]<br>gene_boxplot(<span>'Exhausted signature'</span>)<br><br><br><span>#####下面是NK</span><br>table(seuratObj$celltype)<br>sce_NK = seuratObj[, seuratObj$celltype %in% c( <span>'NK_FCGR3A'</span>,<span>'NK_KLRD1'</span> )]<br><br>avg &lt;- AverageExpression(object = sce_NK, assay = <span>"RNA"</span>, group.by = <span>"patientoutcone"</span>)<br>a = avg$RNA<br>b = <span>as</span>.data.frame(a)<br>str(b)<br><br><span># Cytotoxic基因名称列表</span><br>submatrix &lt;- b[Cytotoxic_gene, ]<br>gene_boxplot(<span>'Cytotoxic signature'</span>)<br><br><br><span># Exhausted基因名称列表</span><br>submatrix &lt;- b[Exhausted_gene, ]<br>gene_boxplot(<span>'Exhausted signature'</span>)<br><br><br><br><span>#####下面是Treg</span><br>table(seuratObj$celltype)<br>sce_Treg = seuratObj[, seuratObj$celltype %in% c( <span>'Treg_CTLA4'</span>,<span>'Treg_SELL'</span> )]<br>avg &lt;- AverageExpression(object = sce_Treg, assay = <span>"RNA"</span>, group.by = <span>"patientoutcone"</span>)<br>a = avg$RNA<br>b = <span>as</span>.data.frame(a)<br>str(b)<br><br><br><span># Exhausted基因名称列表</span><br>submatrix &lt;- b[Exhausted_gene, ]<br>gene_boxplot(<span>'Exhausted signature'</span>)<br><span>###在未治疗组Treg这么高？，MPR组抑制性Treg下降了</span><br></code></pre><p>这是CD8T细胞的细胞毒性基因和耗竭基因集的评分，当时忘了拼在一起给它们加上CD8T的标签了…<br></p><figure><img data-ratio="0.38055555555555554" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJqM9kAxhdTv94mp6rRcIeg9lgvFjDibt03wvwBrLuwRfvTGQmHlSdnQPQ/640?wx_fmt=png" data-type="png" data-w="1080" title="image.png" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJqM9kAxhdTv94mp6rRcIeg9lgvFjDibt03wvwBrLuwRfvTGQmHlSdnQPQ/640?wx_fmt=png"></figure><br>下面是NK细胞，这个NK细胞确实有点不太对劲，和第一节里的NK相互印证了。<br><figure><img data-ratio="0.37222222222222223" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJqYdN4Ysmz7U9L2nmUtUgw24rKmSzhUKdKdNrrX37gOtpXxPNKegn46Q/640?wx_fmt=png" data-type="png" data-w="1080" title="image.png" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJqYdN4Ysmz7U9L2nmUtUgw24rKmSzhUKdKdNrrX37gOtpXxPNKegn46Q/640?wx_fmt=png"><figcaption><br></figcaption></figure><pre><code><span>#####箱线图显示TN ( n = 3)、MPR ( n = 4)和NMPR ( n = 8)患者各T/NK细胞群的细胞比例。<br>####不同细胞群里亚组比例<br>phe = seuratObj@meta.data<br>table(phe$celltype)<br>table(phe$patient)<br>phe$patientoutcome =  paste(phe$patient,phe$Pathologic,sep = '_')<br><br>tb=table(phe$celltype,<br>         phe$patientoutcome)<br><br>head(tb)<br>library (gplots) <br>balloonplot(tb)<br>bar_data <span>&lt;<span>-</span> <span>as.data.frame</span>(<span>tb</span>)<br><br><span>bar_per</span> &lt;<span>-</span> <span>bar_data</span> %&gt;</span>% <br>  group_by(Var2) %&gt;%<br>  mutate(sum(Freq)) %&gt;%<br>  mutate(percent = Freq / `sum(Freq)`)<br>head(bar_per) <br>#write.csv(bar_per,file = "celltype_by_group_percent.csv")<br>library(ggplot2)<br>library(stringr)<br>bar_per$Var2<br>group = ifelse(str_detect(bar_per$Var2,"TN"),"TN",<br>               ifelse(str_detect(bar_per$Var2,"NMPR"),"NMPR","MPR"))<br>bar_per$group = group<br><br>bar_per$group = factor(bar_per$group,levels =c("TN","MPR","NMPR"))<br>table(bar_per$Var1)<br>bar_per$Var1 = factor(bar_per$Var1,levels =c("B3_IGHD","B0_MS4A1","B1_IGHM",'B2_HSPA1A','B4_FCRL4','B5_CD83','B6_RGS13'))<br><br>col = c("#508E42", "#DF5E38","#7B7DC4")<br>P <span>&lt;<span>-</span> <span>ggplot</span>(<span>bar_per</span>, <span>aes</span>(<span>x</span> = <span>Var1,</span> <span>y</span> = <span>percent,</span> <span>fill</span> = <span>group,</span> <span>color</span> = <span>group))</span> +<br>  <span>geom_boxplot</span>(<span>width</span> = <span>0.6,</span> <span>outlier.shape</span> = <span>NA)</span> +<br>  <span>geom_jitter</span>(<span>shape</span> = <span>21,</span> <span>size</span> = <span>2,</span> <span>position</span> = <span>position_jitterdodge(0.2))</span> +<br>  <span>scale_fill_manual</span>(<span>values</span> = <span>c(</span>"<span>white</span>", "<span>white</span>", "<span>white</span>")) +<br>  <span>scale_color_manual</span>(<span>values</span> = <span>col,</span><br>                     <span>labels</span> = <span>c(</span>"<span>TN</span>(<span>n</span>=<span>3)</span>", "<span>MPR</span>(<span>n</span>=<span>4)</span>", "<span>NMPR</span>(<span>n</span>=<span>8)</span>"),<br>                     <span>name</span> = <span>"Group"</span>) +<br>  <span>theme_bw</span>() +<br>  <span>theme</span>(<span>legend.position</span> = <span>"bottom"</span>,<br>        <span>legend.direction</span> = <span>'horizontal'</span>,<br>        <span>legend.justification</span> = <span>"left"</span>,<br>        <span>legend.box.just</span> = <span>"left"</span>,<br>        <span>legend.title</span> = <span>element_blank(),</span><br>        <span>axis.text.x</span> = <span>element_text(colour</span> = <span>"black"</span>,<span>angle</span>=<span>45,</span>  <span>hjust</span> = <span>1,</span> <span>family</span> = <span>"Times"</span>, <span>size</span> = <span>12),</span><br>        <span>axis.text.y</span> = <span>element_text(family</span> = <span>"Times"</span>, <span>size</span> = <span>14,</span> <span>face</span> = <span>"plain"</span>),<br>        <span>axis.title.y</span> = <span>element_text(family</span> = <span>"Times"</span>, <span>size</span> = <span>14,</span> <span>face</span> = <span>"plain"</span>),<br>        <span>axis.title.x</span> = <span>element_text(family</span> = <span>"Times"</span>, <span>size</span> = <span>14,</span> <span>face</span> = <span>"plain"</span>),<br>        <span>plot.title</span> = <span>element_text(family</span> = <span>"Times"</span>, <span>size</span> = <span>15,</span> <span>face</span> = <span>"bold"</span>, <span>hjust</span> = <span>0.5),</span><br>        <span>panel.grid.major</span> = <span>element_blank(),</span><br>        <span>panel.grid.minor</span> = <span>element_blank())</span> +<br>  <span>labs</span>(<span>y</span> = <span>"Fraction (%) of immune cells"</span>,<br>       <span>x</span> = <span>""</span>)+<br>  <span>guides</span>(<span>fill</span> = <span>FALSE)</span><br><br><span>P</span><br><br><span>p</span> = <span>P+</span><br>  <span>coord_cartesian</span>(<span>clip</span> = <span>'off'</span>,<span>ylim</span> = <span>c(0,1))+</span> #在非图形区域绘图,且要定好<span>y</span>轴范围<br>  <span>theme</span>(<span>plot.margin</span> = <span>margin(2.2,0.5,1.5,0.5,</span>'<span>cm</span>'))+ #自定义图片上左下右的边框宽度<br>  <span>annotate</span>('<span>text</span>',<span>x</span>=<span>2.6,y</span>=<span>1.13,label</span>=<span>'CD4 Tconv'</span>,<span>size</span>=<span>7,family</span>=<span>'serif'</span>,<span>color</span>=<span>'red'</span>)+ <br>  <span>annotate</span>('<span>segment</span>',<span>x</span>=<span>0.55,xend</span>=<span>4.2,y</span>=<span>1.1,yend</span>=<span>1.1,color</span>=<span>'red'</span>,<span>cex</span>=<span>.8)+</span><br>  <span>annotate</span>('<span>text</span>',<span>x</span>=<span>6.8,y</span>=<span>1.13,label</span>=<span>'CD8 T'</span>,<span>size</span>=<span>7,family</span>=<span>'serif'</span>,<span>color</span>=<span>'red'</span>)+ <br>  <span>annotate</span>('<span>segment</span>',<span>x</span>=<span>4.7,xend</span>=<span>8.9,y</span>=<span>1.1,yend</span>=<span>1.1,color</span>=<span>'red'</span>,<span>cex</span>=<span>.8)+</span><br>  <span>annotate</span>('<span>text</span>',<span>x</span>=<span>10.5,y</span>=<span>1.13,label</span>=<span>'NK'</span>,<span>size</span>=<span>7,family</span>=<span>'serif'</span>,<span>color</span>=<span>'red'</span>)+<br>  <span>annotate</span>('<span>segment</span>',<span>x</span>=<span>9.5,xend</span>=<span>11.5,y</span>=<span>1.1,yend</span>=<span>1.1,color</span>=<span>'red'</span>,<span>cex</span>=<span>.8)+</span><br>  <span>annotate</span>('<span>text</span>',<span>x</span>=<span>13.6,y</span>=<span>1.13,label</span>=<span>'CD4 Treg'</span>,<span>size</span>=<span>7,family</span>=<span>'serif'</span>,<span>color</span>=<span>'red'</span>)+<br>  <span>annotate</span>('<span>segment</span>',<span>x</span>=<span>12.8,xend</span>=<span>14.3,y</span>=<span>1.1,yend</span>=<span>1.1,color</span>=<span>'red'</span>,<span>cex</span>=<span>.8)</span><br><br><span>p</span><br><span>ggsave</span>('<span>NK_T_por.pdf</span>',<span>width</span> = <span>14.5,height</span> = <span>7.5)</span><br></span></span><span>####活化的Tregs比naïve Tregs具有更强的免疫抑制功能，且与预后不良相关。<br>#持续激活Tregs（Treg_CTLA4，表达TNFRSF4和TNFRSF9）仅在MPR患者中降低。<br>#相对于TN患者，MPR和NMPR患者中naïve Tregs (Treg_SELL，表达SELL和LEF1)的比例均下降。<br>#MPR患者的Treg耗竭特征始终低于NMPR患者。</span></code></pre><figure><img data-ratio="0.4583333333333333" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJqZmAThHicbb22ADBmzvFOpx9DKlTJey3hmrBPrZTl2ia05icyxBMQcycFA/640?wx_fmt=png" data-type="png" data-w="1080" title="60a6b745dc1841f955fbb379ebe7b9c.png" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJqZmAThHicbb22ADBmzvFOpx9DKlTJey3hmrBPrZTl2ia05icyxBMQcycFA/640?wx_fmt=png"><figcaption><br></figcaption></figure><pre><code><span>#文章中用的monocle2，这里也用2</span><br><span>###### 拟时序分析 ######</span><br>sce_4CD8T = seuratObj[, seuratObj$celltype %<span>in</span>% c( <span>'CD8_GZMB'</span> ,<span>'CD8_GZMK'</span>,<span>'CD8_HAVCR2'</span>,<span>'CD8_IL7R'</span>)]<br><span>library</span>(Seurat)<br><span>library</span>(ggplot2)<br><span>library</span>(clustree)<br><span>library</span>(cowplot)<br><span>library</span>(dplyr)<br><span>library</span>(monocle)<br>packageVersion(<span>"monocle"</span>)<br><span>#monocle构建CDS需要3个矩阵：expr.matrix、pd、fd</span><br><span># 将Seurat中的对象转换为monocle识别的对象</span><br><span>#cds &lt;- importCDS(GetAssayData(seurat.object))</span><br><span>#选择做拟时序的亚群</span><br>Mono_tj&lt;-sce_4CD8T<br><br>Mono_matrix&lt;-as(as.matrix(GetAssayData(Mono_tj,slot = <span>"counts"</span>)), <span>'sparseMatrix'</span>)<br><span>#构建featuredata，一般featuredata需要两个col，一个是gene_id,一个是gene_short_name,row对应counts的rownames</span><br>feature_ann&lt;-data.frame(gene_id=rownames(Mono_matrix),gene_short_name=rownames(Mono_matrix))<br>rownames(feature_ann)&lt;-rownames(Mono_matrix)<br>Mono_fd&lt;-new(<span>"AnnotatedDataFrame"</span>, data = feature_ann)<br><br><span>#Seurat object中的@meta.data一般会存放表型相关的信息如cluster、sample的来源、group等，所以选择将metadata转换为phenodata</span><br>sample_ann&lt;-Mono_tj@meta.data<br><span>#rownames(sample_ann)&lt;-colnames(Mono_matrix)</span><br><br>Mono_pd&lt;-new(<span>"AnnotatedDataFrame"</span>, data =sample_ann)<br><span>#build new cell data set</span><br>Mono.cds&lt;-newCellDataSet(Mono_matrix,phenoData =Mono_pd,featureData =Mono_fd,expressionFamily=negbinomial.size())<br><br><br><span>#预处理</span><br>Mono.cds &lt;- estimateSizeFactors(Mono.cds)<br>Mono.cds &lt;- estimateDispersions(Mono.cds)<br><span>#筛选基因,这里可以根据自己的需要筛选特定的基因</span><br>disp_table &lt;- dispersionTable(Mono.cds)<br><span>#unsup_clustering_genes &lt;- subset(disp_table, mean_expression &gt;= 0.1)</span><br>unsup_clustering_genes &lt;- subset(disp_table, mean_expression &gt;= <span>0.1</span>)<br><br>Mono.cds &lt;- setOrderingFilter(Mono.cds, unsup_clustering_genes$gene_id)<br><span>#用DDRtree 进行降维分析</span><br>Mono.cds &lt;- reduceDimension(<br>  Mono.cds,<br>  max_components = <span>2</span>,<br>  method = <span>'DDRTree'</span>)<br><span>#计算psudotime值</span><br>Mono.cds &lt;- orderCells(Mono.cds)<br>head(pData(Mono.cds))<br><br>plot_cell_trajectory(Mono.cds,color_by=<span>"celltype"</span>, size=<span>1</span>,show_backbone=<span>TRUE</span>)<br>ggsave(<span>"拟时序.pdf"</span>,width = <span>7</span>,height = <span>6</span>)<br>plot_cell_trajectory(Mono.cds,color_by=<span>"patient"</span>, size=<span>1</span>,show_backbone=<span>TRUE</span>)<br>plot_cell_trajectory(Mono.cds,color_by=<span>"Pathologic"</span>, size=<span>1</span>,show_backbone=<span>TRUE</span>)<br><br>p1 &lt;- plot_cell_trajectory(Mono.cds, x = <span>1</span>, y = <span>2</span>, color_by = <span>"celltype"</span>) + <br>  theme(legend.position=<span>'none'</span>,panel.border = element_blank()) + <span>#去掉第一个的legend</span><br>  scale_color_manual(values = mycolors) <br>p2 &lt;- plot_complex_cell_trajectory(Mono.cds, x = <span>1</span>, y = <span>2</span>,<br>                                   color_by = <span>"celltype"</span>)+<br>  scale_color_manual(values = mycolors) +<br>  theme(legend.title = element_blank()) <br>p1|p2<br><br>trace(<span>'project2MST'</span>, edit = <span>T</span>, where = asNamespace(<span>"monocle"</span>))<br></code></pre><p>让我解释的话那只能一部分耗竭了，另一部分耗竭的被逆转了<br>CD8_IL7R和CD8_GZMK → CD8_GZMB → CD8_HAVCR2<br></p><figure><img data-ratio="0.600925925925926" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJqSnSTIscibqLIvMG7JibdPTgcKnrHKBp2rBDjJXkbZmINTrEC8NcD3WYA/640?wx_fmt=png" data-type="png" data-w="1080" title="0022d88cef18ff4146e835b17537391.png" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJqSnSTIscibqLIvMG7JibdPTgcKnrHKBp2rBDjJXkbZmINTrEC8NcD3WYA/640?wx_fmt=png"></figure><h2><span>4.B</span></h2><figure><img data-ratio="0.6268518518518519" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJqMicM55qCUerSoAzUbNMxJImDmLOebONs6sGfI9p5GaOX8KNf2WT7hXg/640?wx_fmt=png" data-type="png" data-w="1080" title="image.png" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJqMicM55qCUerSoAzUbNMxJImDmLOebONs6sGfI9p5GaOX8KNf2WT7hXg/640?wx_fmt=png"></figure><br>流程基本和上面一样，复现图：</section><section><br><figure><img data-ratio="0.5453703703703704" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJqV54o2ha05ydiaWPXp7k3F6yMOFVA0rfvSKibYEMMheaTgzhLrSypgxbg/640?wx_fmt=png" data-type="png" data-w="1080" title="674bc156a7ea9fe0a0574cff976f046.png" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJqV54o2ha05ydiaWPXp7k3F6yMOFVA0rfvSKibYEMMheaTgzhLrSypgxbg/640?wx_fmt=png"></figure><br><figure><img data-ratio="0.5416666666666666" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJq2qLkJZ98zpiarHltGwqXTACb5icloNTiakaKluPI8q1auqkXibuFn52joQ/640?wx_fmt=png" data-type="png" data-w="1080" title="659ec38f381fabcf6f28258c8950ee7.png" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJq2qLkJZ98zpiarHltGwqXTACb5icloNTiakaKluPI8q1auqkXibuFn52joQ/640?wx_fmt=png"></figure><br>趋势和文中是一样的..<br><figure><img data-ratio="0.5814814814814815" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJqAZQ1avqcFSvYDKZ2K29Zw8YpjrGDHGhFLnTzzFHwial156GJA3L0IiaQ/640?wx_fmt=png" data-type="png" data-w="1080" title="f541185bf567efce6483dd8ae84a2e9.png" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJqAZQ1avqcFSvYDKZ2K29Zw8YpjrGDHGhFLnTzzFHwial156GJA3L0IiaQ/640?wx_fmt=png"></figure><h2><span>5.髓系</span></h2><figure><img data-ratio="0.5435185185185185" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJqPlvJlXPpPB8WEYyO8H3161PZXaz8GLrvsspSA7NIPJeNG6icAjRwfHw/640?wx_fmt=png" data-type="png" data-w="1080" title="a7e220e10709fd0228647fef374c047.png" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic8CbFua7EfN6e5ibP2AcGJqPlvJlXPpPB8WEYyO8H3161PZXaz8GLrvsspSA7NIPJeNG6icAjRwfHw/640?wx_fmt=png"></figure><h1><span>番外</span></h1><p>从复现中学到最多的就是眼熟各种细胞亚群吧，还有，<strong>文献中稀奇古怪的图太多了，虽然不知道别人是通过什么方式画出来的，</strong>但是只求形似也想复现出来哈哈哈。话说为什么文中不对药物治疗中的药物进行分析呢？信迪利特瑞普利。如果大家对我纯手工代码不满意，也可以去看其它文章的官方代码，比如</p><blockquote data-tool="mdnice编辑器"><p>前面我们介绍了新鲜出炉的一篇NC单细胞文章图表复现的7个笔记：</p></blockquote><ul data-tool="mdnice编辑器"><li><section><a href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247513533&amp;idx=1&amp;sn=13418c254b58092398874d151489193e&amp;chksm=ea1cb33fdd6b3a29089900ae32a70d72e8d563518c84fe0334b256d19e137e555a9ce023d9fd&amp;scene=21#wechat_redirect" data-linktype="2">手把手带你复现NC图表之Figure6</a></section></li><li><section><a href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247513446&amp;idx=1&amp;sn=88fbdfe38b95cb851a4c36af199a4302&amp;chksm=ea1cb3e4dd6b3af2bdf1458af9bd3f62950202411d4c770aaec5010737434e34f360697c6083&amp;scene=21#wechat_redirect" data-linktype="2">手把手带你复现NC图表之Figure5</a></section></li><li><section><a href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247513381&amp;idx=1&amp;sn=b95f12df59b2eef3f85cbf829cf29cab&amp;chksm=ea1cb3a7dd6b3ab18ca6484d5db6f5222625012966d708a82ff89d135519e4284f60e1c6910c&amp;scene=21#wechat_redirect" data-linktype="2">手把手带你复现NC图表之Figure 4</a></section></li><li><section><a href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247513310&amp;idx=1&amp;sn=5aa8db2bdc7c6ca1b1bdb6d1a9863130&amp;chksm=ea1cb25cdd6b3b4a2edf5a6c816f9e5ac46f6667dabe4adc2995a3db7a98e065c044004487d0&amp;scene=21#wechat_redirect" data-linktype="2">手把手带你复现NC图表之Figure 3</a></section></li><li><section><a href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247513202&amp;idx=1&amp;sn=3f8a4947cce8596f454fb82090f599e0&amp;chksm=ea1cb2f0dd6b3be68237d48393e535a8399b9e0e8a091dcc225b3bce2100c6fc4d31a0fd027b&amp;scene=21#wechat_redirect" data-linktype="2">手把手带你复现NC图表之Figure 2</a></section></li><li><section><a href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247513057&amp;idx=1&amp;sn=6f737f0a5ae7795645a42894897f7b87&amp;chksm=ea1cb563dd6b3c7567db78011efb52823c1f6bc94a760cc11ef8a07d3d5d6c779e4813c4d523&amp;scene=21#wechat_redirect" data-linktype="2">手把手带你复现NC图表之Figure 1</a></section></li></ul><p data-tool="mdnice编辑器">包括但不限于降维聚类分群和生物学命名，单细胞比例差异和表达量差异以及生物学功能注释，成纤维等细胞亚群的细分，单细胞高级分析等等。其它高级分析，比如转录因子分析 ，主要是对计算机资源消耗会比较大，我也多次分享过细节教程：</p><ul data-tool="mdnice编辑器"><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247497441&amp;idx=1&amp;sn=7a53e6aa00c84b57104a61787f2beb03&amp;scene=21#wechat_redirect" data-linktype="2">张泽民团队的单细胞研究把T细胞分的如此清楚</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247497617&amp;idx=2&amp;sn=fe47078075a3b9994079761f96870aa3&amp;scene=21#wechat_redirect" data-linktype="2">细胞通讯分析的背景知识</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247497617&amp;idx=1&amp;sn=0e63223347afbb0795aa94d339d72214&amp;scene=21#wechat_redirect" data-linktype="2">构建单细胞亚群网络（类似于细胞通讯分析）</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247497653&amp;idx=2&amp;sn=320d26cad3d8141bb081d6ff798ae778&amp;scene=21#wechat_redirect" data-linktype="2">细胞通讯分析结果的解读</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247497665&amp;idx=1&amp;sn=74ac0e87b9689d5df7c0208e1c1dc0ac&amp;scene=21#wechat_redirect" data-linktype="2">SCENIC转录因子分析结果的解读</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247497956&amp;idx=1&amp;sn=5d4deb7cf7b7848b3e2273cbd663bb6a&amp;scene=21#wechat_redirect" data-linktype="2">人人都能学会的单细胞聚类分群注释</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247498040&amp;idx=1&amp;sn=3a39699ab10fb71b966adb5e7e640270&amp;scene=21#wechat_redirect" data-linktype="2">对单细胞表达矩阵做gsea分析</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247498091&amp;idx=1&amp;sn=5c157da6e889f2134ef9342cee560197&amp;scene=21#wechat_redirect" data-linktype="2">单细胞转录因子分析之SCENIC流程</a></section></li></ul><p data-tool="mdnice编辑器">单细胞转录组数据分析的标准降维聚类分群，并且进行生物学注释后的结果。可以参考前面的例子：<a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247497956&amp;idx=1&amp;sn=5d4deb7cf7b7848b3e2273cbd663bb6a&amp;scene=21#wechat_redirect" data-linktype="2">人人都能学会的单细胞聚类分群注释</a> ，我们演示了第一层次的分群。</p><p data-tool="mdnice编辑器">如果你对单细胞数据分析还没有基础认知，可以看基础10讲：</p><ul data-tool="mdnice编辑器"><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247486076&amp;idx=1&amp;sn=52bb851d7dc23461233a2cf458736151&amp;scene=21#wechat_redirect" data-linktype="2">01. 上游分析流程</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247486082&amp;idx=1&amp;sn=03cadceffb2c14ba95d97fe5caf38d94&amp;scene=21#wechat_redirect" data-linktype="2">02.课题多少个样品，测序数据量如何</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247486088&amp;idx=1&amp;sn=3a115338ee4937d20caab78627237553&amp;scene=21#wechat_redirect" data-linktype="2">03. 过滤不合格细胞和基因（数据质控很重要）</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247486096&amp;idx=1&amp;sn=1a99c4c5800b7e0287db3e8ef369fab8&amp;scene=21#wechat_redirect" data-linktype="2">04. 过滤线粒体核糖体基因</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247486098&amp;idx=1&amp;sn=bf9a71df848d74fe665ce7d5e283d5ff&amp;scene=21#wechat_redirect" data-linktype="2">05. 去除细胞效应和基因效应</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247486260&amp;idx=1&amp;sn=c6abf658de73594d1d77d8e1ffa7d153&amp;scene=21#wechat_redirect" data-linktype="2">06.单细胞转录组数据的降维聚类分群</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247486271&amp;idx=1&amp;sn=638b434b6deee63206af1c0eeda175ab&amp;scene=21#wechat_redirect" data-linktype="2">07.单细胞转录组数据处理之细胞亚群注释</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247486278&amp;idx=1&amp;sn=91250ef733833ff00371818b215dc124&amp;scene=21#wechat_redirect" data-linktype="2">08.把拿到的亚群进行更细致的分群</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247486287&amp;idx=1&amp;sn=49627c638ff9c04418282c53518aa7c7&amp;scene=21#wechat_redirect" data-linktype="2">09.单细胞转录组数据处理之细胞亚群比例比较</a></section></li></ul></section><p>如果是文章并没有提供表达量矩阵，或者说你对作者的矩阵不满意， 就有可能需要从fastq文件开始啦。</p><blockquote data-tool="mdnice编辑器"><p>正常走cellranger的定量流程即可，代码我已经是多次分享了。参考：</p></blockquote><ul data-tool="mdnice编辑器"><li><section><a href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247512340&amp;idx=3&amp;sn=1b9609a8870a0209dd27ffdcbc3cac87&amp;chksm=9b4bf1afac3c78b90674678fcec66365b9faaa275ff4b0a2255e0a05fa8b905e15222a643bea&amp;scene=21#wechat_redirect" data-linktype="2">10X单细胞转录组原始测序数据的Cell Ranger流程（仅需800元）</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247496813&amp;idx=1&amp;sn=4151bf2265618eff4e0123722c50e569&amp;scene=21#wechat_redirect" data-linktype="2">10X的单细胞转录组原始数据也可以在EBI下载</a></section></li><li><section><a href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247510920&amp;idx=1&amp;sn=c4561d34e984406693c014cdfe236c0f&amp;chksm=9b4beb33ac3c622542d894344c323ff7cca52f69119d02fc7aa4636af0cbe7df4b6c63dd5ba9&amp;scene=21#wechat_redirect" data-linktype="2">一个10x单细胞转录组项目从fastq到细胞亚群</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247513565&amp;idx=1&amp;sn=092e637017d176c43f00a295d3210592&amp;scene=21#wechat_redirect" data-linktype="2">一文打通单细胞上游：从软件部署到上游分析</a></section></li><li><section><a href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247513605&amp;idx=1&amp;sn=e86a329c887745c6d00d3ededa39dcda&amp;chksm=9b4bf6beac3c7fa8523cef4e7189fb20b914460ddb61e6cd1dd520b5928e1b59a8b7827ce783&amp;scene=21#wechat_redirect" data-linktype="2">PRJNA713302这个10x单细胞fastq实战</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247513968&amp;idx=1&amp;sn=f5a44a7bea0bdacd8af1a20c177763e5&amp;scene=21#wechat_redirect" data-linktype="2">一次曲折且昂贵的单细胞公共数据获取与上游处理</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247514146&amp;idx=1&amp;sn=b9721433d49a2d963eeaab1ad47fc91b&amp;scene=21#wechat_redirect" data-linktype="2">只能下载bam文件的10x单细胞转录组项目数据处理</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247511452&amp;idx=2&amp;sn=83ec97cbc3334a6095e6d63e05e9fd6e&amp;scene=21#wechat_redirect" data-linktype="2">不知道10x单细胞转录组样品和fastq文件的对应关系</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247508521&amp;idx=2&amp;sn=2cf3158e74d37b3a741908d8bfc8f02f&amp;scene=21#wechat_redirect" data-linktype="2">10X单细胞转录组测序数据的 SRA转fastq踩坑那些事</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247514395&amp;idx=2&amp;sn=96c505b76ae87dd0efa737c4c44e2270&amp;scene=21#wechat_redirect" data-linktype="2">10x的单细胞转录组fastq文件的R1和R2不能弄混哦</a></section></li></ul><p data-tool="mdnice编辑器">差不多几个小时就可以完成全部的样品的cellranger的定量流程。基础知识非常重要，我们在单细胞天地多次分享过<strong>cellranger</strong>流程的笔记（2019年5月），大家可以自行前往学习，如下：</p><ul data-tool="mdnice编辑器"><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247484146&amp;idx=1&amp;sn=16e09b82d048eed1ff6100b22970abd5&amp;scene=21#wechat_redirect" data-linktype="2">单细胞实战(一)数据下载</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247484179&amp;idx=1&amp;sn=fe84f5243a6021fe6afea128e3ac273a&amp;scene=21#wechat_redirect" data-linktype="2">单细胞实战(二) cell ranger使用前注意事项</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247484206&amp;idx=1&amp;sn=edeebbdd092f79361aee87e9ce086d80&amp;scene=21#wechat_redirect" data-linktype="2">单细胞实战(三) Cell Ranger使用初探</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247484355&amp;idx=1&amp;sn=7860fe0c46073a55d2d3700822c3103b&amp;scene=21#wechat_redirect" data-linktype="2">单细胞实战(四) Cell Ranger流程概览</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247484402&amp;idx=1&amp;sn=95c2be0dc6499e4b1eb9a91d79e584d1&amp;scene=21#wechat_redirect" data-linktype="2">单细胞实战(五) 理解cellranger count的结果</a></section></li></ul><p><br></p><p><mp-style-type data-value="10000"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/D3DnZPYdWfls4-5UXh1gJw",target="_blank" rel="noopener noreferrer">原文链接</a>
