---
title: "转录组差异分析FPKM与count处理差别大吗"
date: 2023-09-04T03:52:16Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Acdemic"]
---
转录组差异分析FPKM与count处理差别大吗 by 生信菜鸟团
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><h2 data-tool="mdnice编辑器"><span></span><span>缘起</span><span></span></h2><p data-tool="mdnice编辑器">这些天来，我们一般都是处理上游定量好的count数据，然后进行下游的转录组分析。但是，我们查看GEO数据集时，会发现有些数据集并没有提供count数据，而仅仅提供了FPKM或者RPKM等格式的数据。那当数据集提供的是FPKM数据集时，我们还能处理吗。前面曾老师分享的推文中描述了FPKM的处理方式，具体见<a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247492196&amp;idx=1&amp;sn=1bcdef5e19cdde2b0d8cc947c759fac6&amp;scene=21#wechat_redirect" data-linktype="2">RNAseq数据，下载GEO中的FPKM文件后该怎么下游分析</a>，评论区中有小伙伴谈到limma包的作者不推荐用limma处理FPKM数据，最好用原始数据进行分析。那用count与用FPKM去处理获得的差异基因具有巨大的差别吗？曾老师前两天提出了这个疑问,于是便有了今天的推文。接下来，我们就探索一下用count与用FPKM去处理获得的差异基因是否具有巨大差别吧？</p><h2 data-tool="mdnice编辑器"><span></span><span>探究</span><span></span></h2><p data-tool="mdnice编辑器">今天，我们使用标题为  LncRNA-directed Antigenicity Loss Suppresses Immunosurveillance  的数据集 GSE113143 进行探究，数据集的介绍链接如下<span>https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc= GSE113143</span> ，感兴趣的小伙伴可以点进去看看作者的总体设计。在此，我们对文章进行简单归纳，作者通过转录组测序探究了小鼠乳腺癌组织和乳腺组织的表达谱。。</p><h2 data-tool="mdnice编辑器"><span></span><span>转录组数据集介绍</span><span></span></h2><p data-tool="mdnice编辑器">GSE113143 数据集的样本分组如下，两个分组三个重复样本：<img data-ratio="0.25705794947994054" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic4Ov1ktjg8xrG7hGCS7qiaAE2z3q1zm97u1mOwsvhBpBnJoIoYQ8lq4ES3cK4aJJXnCibn4Aoz1hiaQ/640?wx_fmt=png" data-type="png" data-w="673" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic4Ov1ktjg8xrG7hGCS7qiaAE2z3q1zm97u1mOwsvhBpBnJoIoYQ8lq4ES3cK4aJJXnCibn4Aoz1hiaQ/640?wx_fmt=png">处理数据的话，作者仅仅提供了<strong>「FPKM矩阵」</strong>。因为我们要比较FPKM与count分别进行差异分析的区别，所以我们需要自身对于上游转录组数据进行定量，去获得<strong>「Ensenmble count矩阵」</strong>。之后，我们就可以直接走Ensenmble矩阵的差异分析流程进行分析了，链接见<span>https://mp.weixin.qq.com/s/BHt_7WFCtKXIZwQYqhO8jA</span>。</p><p data-tool="mdnice编辑器">以下为作者提供的FPKM矩阵下载链接：<span>https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE113143&amp;format=file&amp;file=GSE113143%5FNormal%5FTumor%5FExpression%2Etab%2Egz</span>；感兴趣的小伙伴可以下载试试。</p><h2 data-tool="mdnice编辑器"><span></span><span>正式分析</span><span></span></h2><h3 data-tool="mdnice编辑器"><span></span><span><span></span>1.利用fpkm值进行差异分析</span><span></span></h3><p data-tool="mdnice编辑器">FPKM处理代码援引自泥人吴老师的<a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247492196&amp;idx=1&amp;sn=1bcdef5e19cdde2b0d8cc947c759fac6&amp;scene=21#wechat_redirect" data-linktype="2">RNAseq数据，下载GEO中的FPKM文件后该怎么下游分析</a>，其中也有一些很好的描述，感兴趣的小伙伴可以看看。FPKM处理关键是要将FPKM值转换为TPM值，然后用Limma进行后续的差异分析。</p><pre data-tool="mdnice编辑器"><span></span><code><span># 1.下载GSE113143数据集</span><br><br>a=read.table(<span>'GSE113143_Normal_Tumor_Expression.tab.gz'</span>,sep=<span>'\t'</span>,quote = <span>""</span>,fill = <span>T</span>,<br>             comment.char = <span>"!"</span>,header = <span>T</span>) <span># 提取表达矩阵</span><br>rownames(a)=a[,<span>1</span>]<br>a &lt;- a[,-<span>1</span>]<br><br><span># 2.将FPKM值转换为TPM值</span><br>expMatrix &lt;- a<br><br>fpkmToTpm &lt;- <span>function</span>(fpkm)<br>{<br>  exp(log(fpkm) - log(sum(fpkm)) + log(<span>1e6</span>))<br>}<br><br>tpms &lt;- apply(expMatrix,<span>2</span>,fpkmToTpm) <span>#每列的基因</span><br>tpms[<span>1</span>:<span>3</span>,]<br><span>##                       N1         N2       N3       T1       T2       T3</span><br><span>## 0610005C13Rik  0.2320471  0.1715468  0.00000  0.00000  0.00000  0.00000</span><br><span>## 0610007P14Rik 48.3908046 39.2631560 46.04003 50.03695 59.05140 67.28521</span><br><span>## 0610009B22Rik 47.4905907 58.5953867 54.26887 49.78854 53.12592 57.99844</span><br>colSums(tpms) <span>#都是每一百万（深度一样）</span><br><span>##    N1    N2    N3    T1    T2    T3 </span><br><span>## 1e+06 1e+06 1e+06 1e+06 1e+06 1e+06</span><br><br><span># 3.差异分析</span><br>group_list=c(rep(<span>'Normal'</span>,<span>3</span>),rep(<span>'Tumor'</span>,<span>3</span>))<br><span>## 强制限定顺序</span><br>group_list &lt;- factor(group_list,levels = c(<span>"Normal"</span>,<span>"Tumor"</span>),ordered = <span>F</span>)<br><span>#表达矩阵数据校正</span><br>exprSet &lt;- tpms<br>boxplot(exprSet,outline=<span>FALSE</span>, notch=<span>T</span>,col=group_list, las=<span>2</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7142857142857143" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic4Ov1ktjg8xrG7hGCS7qiaAxK1y5yR18BFY4Z9FN5TgFG9qvsbs8AWbAHF7CKIlmBEQmwX4twvltQ/640?wx_fmt=png" data-type="png" data-w="672" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic4Ov1ktjg8xrG7hGCS7qiaAxK1y5yR18BFY4Z9FN5TgFG9qvsbs8AWbAHF7CKIlmBEQmwX4twvltQ/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code><span>library</span>(limma) <br>exprSet=normalizeBetweenArrays(exprSet)<br>boxplot(exprSet,outline=<span>FALSE</span>, notch=<span>T</span>,col=group_list, las=<span>2</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7142857142857143" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic4Ov1ktjg8xrG7hGCS7qiaA3bA6Ojfia1Dj6VOnvAXHoK0PRLEziaZUVsvNak4LLTVawpeuHXENq4PQ/640?wx_fmt=png" data-type="png" data-w="672" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic4Ov1ktjg8xrG7hGCS7qiaA3bA6Ojfia1Dj6VOnvAXHoK0PRLEziaZUVsvNak4LLTVawpeuHXENq4PQ/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code><span>#判断数据是否需要转换</span><br>exprSet &lt;- log2(exprSet+<span>1</span>)<br><span>#差异分析：</span><br>dat &lt;- exprSet<br>design=model.matrix(~factor( group_list ))<br>fit=lmFit(dat,design)<br>fit=eBayes(fit)<br>options(digits = <span>4</span>)<br>topTable(fit,coef=<span>2</span>,adjust=<span>'BH'</span>)<br><span>##                 logFC AveExpr       t   P.Value adj.P.Val      B</span><br><span>## Gm2897        -1.1291  0.5645 -314.35 1.062e-10 2.593e-06 10.205</span><br><span>## Tceal3        -2.4753  1.2376 -168.28 1.638e-09 2.000e-05  9.958</span><br><span>## Frk            1.9861  3.1418  114.45 8.853e-09 6.269e-05  9.600</span><br><span>## Nek2           2.2325  3.9233  108.20 1.132e-08 6.269e-05  9.527</span><br><span>## Calca         -8.6440  4.3220 -105.14 1.283e-08 6.269e-05  9.487</span><br><span>## Paqr9         -4.1108  2.5527  -90.86 2.431e-08 9.368e-05  9.257</span><br><span>## Rpl39l        -2.0242  1.0121  -88.82 2.685e-08 9.368e-05  9.217</span><br><span>## Fads3         -1.6945  5.7675  -80.69 4.086e-08 1.247e-04  9.033</span><br><span>## 4930578C19Rik -1.4385  0.7193  -77.15 4.974e-08 1.350e-04  8.939</span><br><span>## Smpd1         -0.8319  5.0280  -74.19 5.902e-08 1.441e-04  8.853</span><br>bp=<span>function</span>(g){<br>  <span>library</span>(ggpubr)<br>  df=data.frame(gene=g,stage=group_list)<br>  p &lt;- ggboxplot(df, x = <span>"stage"</span>, y = <span>"gene"</span>,<br>                 color = <span>"stage"</span>, palette = <span>"jco"</span>,<br>                 add = <span>"jitter"</span>)<br>  <span>#  Add p-value</span><br>  p + stat_compare_means()<br>}<br>deg=topTable(fit,coef=<span>2</span>,adjust=<span>'BH'</span>,number = <span>Inf</span>)<br>head(deg) <br><span>##         logFC AveExpr       t   P.Value adj.P.Val      B</span><br><span>## Gm2897 -1.129  0.5645 -314.35 1.062e-10 2.593e-06 10.205</span><br><span>## Tceal3 -2.475  1.2376 -168.28 1.638e-09 2.000e-05  9.958</span><br><span>## Frk     1.986  3.1418  114.45 8.853e-09 6.269e-05  9.600</span><br><span>## Nek2    2.232  3.9233  108.20 1.132e-08 6.269e-05  9.527</span><br><span>## Calca  -8.644  4.3220 -105.14 1.283e-08 6.269e-05  9.487</span><br><span>## Paqr9  -4.111  2.5527  -90.86 2.431e-08 9.368e-05  9.257</span><br><span>#save(deg,file = 'deg.Rdata')</span><br><br><span># 4.FPKM转换为TPM后，再利用limma进行差异分析</span><br><br><span>## 不同的阈值，筛选到的差异基因数量就不一样，后面的超几何分布检验结果就大相径庭。</span><br><span>if</span>(<span>T</span>){<br>  logFC_t=<span>1.5</span><br>  deg$g=ifelse(deg$P.Value&gt;<span>0.05</span>,<span>'stable'</span>,<br>               ifelse( deg$logFC &gt; logFC_t,<span>'UP'</span>,<br>                       ifelse( deg$logFC &lt; -logFC_t,<span>'DOWN'</span>,<span>'stable'</span>) )<br>  )<br>  table(deg$g)<br>  head(deg)<br>  <br>}<br><span>##         logFC AveExpr       t   P.Value adj.P.Val      B      g</span><br><span>## Gm2897 -1.129  0.5645 -314.35 1.062e-10 2.593e-06 10.205 stable</span><br><span>## Tceal3 -2.475  1.2376 -168.28 1.638e-09 2.000e-05  9.958   DOWN</span><br><span>## Frk     1.986  3.1418  114.45 8.853e-09 6.269e-05  9.600     UP</span><br><span>## Nek2    2.232  3.9233  108.20 1.132e-08 6.269e-05  9.527     UP</span><br><span>## Calca  -8.644  4.3220 -105.14 1.283e-08 6.269e-05  9.487   DOWN</span><br><span>## Paqr9  -4.111  2.5527  -90.86 2.431e-08 9.368e-05  9.257   DOWN</span><br><br>fpkm_deg=deg<br></code></pre><h3 data-tool="mdnice编辑器"><span></span><span><span></span>2.利用count值进行差异分析</span><span></span></h3><pre data-tool="mdnice编辑器"><span></span><code><span># 自身走上游定量，获得count-ensenmble矩阵</span><br><span>library</span>(edgeR)<br><span>library</span>(DESeq2)<br><span>library</span>(FactoMineR)<br><span>library</span>(factoextra)<br><span>library</span>(clusterProfiler)<br><span>library</span>(org.Mm.eg.db)<br><span>library</span>(stringr)<br><span>library</span>(stringi)<br><span>library</span>(tidyverse)<br><span>library</span>(ggplot2)<br><span>library</span>(patchwork)<br><span>library</span>(pheatmap)<br><span>library</span>(VennDiagram)<br><span>library</span>(RColorBrewer)<br><span>library</span>(patchwork)<br><span>library</span>(ggplotify)<br><br><span># 1.读取数据，获得基因表达矩阵以及cpm矩阵</span><br>rawcount &lt;- read.table(<span>"./all.id.txt"</span>,header = <span>T</span>,sep=<span>"\t"</span>,row.names = <span>1</span>)<br>colnames(rawcount)<br><span>##  [1] "Chr"                             "Start"                          </span><br><span>##  [3] "End"                             "Strand"                         </span><br><span>##  [5] "Length"                          "...3.hisat2.SRR7003811.sort.bam"</span><br><span>##  [7] "...3.hisat2.SRR7003812.sort.bam" "...3.hisat2.SRR7003813.sort.bam"</span><br><span>##  [9] "...3.hisat2.SRR7003814.sort.bam" "...3.hisat2.SRR7003815.sort.bam"</span><br><span>## [11] "...3.hisat2.SRR7003816.sort.bam"</span><br>rawcount &lt;- rawcount[,<span>6</span>:<span>11</span>]<br><br><span>##转换ID</span><br>id2symbol &lt;- bitr(rownames(rawcount),   <br>                  fromType = <span>"ENSEMBL"</span>, <br>                  toType = <span>"SYMBOL"</span>, <br>                  OrgDb = org.Mm.eg.db)<br>rawcount &lt;- cbind(GeneID=rownames(rawcount),rawcount) <span>#多一列换成基因ID</span><br>rawcount &lt;- merge(id2symbol,rawcount,<br>                  by.x=<span>"ENSEMBL"</span>,by.y=<span>"GeneID"</span>,all.y=<span>T</span>) <br>rawcount=rawcount[!duplicated(rawcount$SYMBOL),] <br>rawcount &lt;- drop_na(rawcount) <span>#去除含有NA值的行</span><br>rawcount &lt;- rawcount[,!colnames(rawcount)%<span>in</span>%c(<span>"ENSEMBL"</span>,<span>"Ensmble"</span>,<span>"median"</span>)]<span>#正式获得基因rawcount矩阵</span><br>keep &lt;- rowSums(rawcount&gt;<span>0</span>) &gt;= floor(<span>0.5</span>*ncol(rawcount))<br>filter_count &lt;- rawcount[keep,] <span>#获得filter_count矩阵</span><br>rownames(filter_count) &lt;- filter_count$SYMBOL<br>filter_count &lt;- filter_count[,!colnames(filter_count)%<span>in</span>%<span>"SYMBOL"</span>]<br>express_cpm &lt;- log2(cpm(filter_count)+ <span>1</span>)<br>express_cpm[<span>1</span>:<span>4</span>,<span>1</span>:<span>4</span>] <span>#获得cpm矩阵</span><br><span>##       ...3.hisat2.SRR7003811.sort.bam ...3.hisat2.SRR7003812.sort.bam</span><br><span>## Gnai3                           7.152                           7.100</span><br><span>## Cdc45                           3.665                           3.190</span><br><span>## H19                             3.732                           7.790</span><br><span>## Scml2                           1.572                           1.546</span><br><span>##       ...3.hisat2.SRR7003813.sort.bam ...3.hisat2.SRR7003814.sort.bam</span><br><span>## Gnai3                           7.235                           7.684</span><br><span>## Cdc45                           3.345                           4.736</span><br><span>## H19                             5.189                           5.999</span><br><span>## Scml2                           1.731                           1.485</span><br><br><br><span># 3.用limma进行差异分析</span><br><br><span>#根据列的信息，提取分组信息</span><br>colnames(rawcount) =c(rep(<span>'Normal'</span>,<span>3</span>),rep(<span>'Tumor'</span>,<span>3</span>))<br>group=rep(c(<span>"WT"</span>,<span>"Tumor"</span>),each=<span>3</span>)<br>group_list=factor(group,levels = c(<span>"WT"</span>,<span>"Tumor"</span>))<br>table(group_list)<span>#检查一下组别数量</span><br><span>## group_list</span><br><span>##    WT Tumor </span><br><span>##     3     3</span><br><br><br><span># 4.limma----voom差异分析</span><br><span>library</span>(limma)<br>exp=filter_count<br>Group=group_list<br>dge &lt;- edgeR::DGEList(counts=exp)<br>dge &lt;- edgeR::calcNormFactors(dge)<br>design &lt;- model.matrix(~Group)<br>v &lt;- voom(dge,design, normalize=<span>"quantile"</span>)<br>fit &lt;- lmFit(v, design)<br>fit= eBayes(fit)<br>DEG_limma = topTable(fit, coef=<span>2</span>, n=<span>Inf</span>)<br>DEG_limma = na.omit(DEG_limma)<br>pvalue_t=<span>0.05</span><br>logFC_t=<span>1.5</span><br>k1 = (DEG_limma$P.Value &lt; pvalue_t)&amp;(DEG_limma$logFC &lt; -logFC_t)<br>k2 = (DEG_limma$P.Value &lt; pvalue_t)&amp;(DEG_limma$logFC &gt; logFC_t)<br>DEG_limma$regulated = ifelse(k1,<span>"down"</span>,ifelse(k2,<span>"up"</span>,<span>"not"</span>))<br>table(DEG_limma$regulated)<br><span>## </span><br><span>##  down   not    up </span><br><span>##  2230 15771  1709</span><br></code></pre><h3 data-tool="mdnice编辑器"><span></span><span><span></span>3.比较分别用count与fpkm进行差异分析的结果</span><span></span></h3><pre data-tool="mdnice编辑器"><span></span><code><span>## 001 绘制整体基因的相关性散点图</span><br>ids=intersect(rownames(fpkm_deg),rownames(DEG_limma))<br><span># 01看一看分别比较的整体相关性（此处就看整体差异基因吧）</span><br><span>library</span>(ggstatsplot)<br><span>library</span>(ggpmisc) <br><span># 02构建可视化所需的矩阵（相关性就两列值）</span><br>df=data.frame(<br>  deg1=fpkm_deg[ids,<span>"logFC"</span>],<br>  deg2=DEG_limma[ids,<span>"logFC"</span>] <span>#就是两列logFC的值（散点图）</span><br>)<br><br><span># 03绘制相关性图形</span><br>p &lt;- ggplot(df, aes(x=df$deg1,<br>                    df$deg2))+    <br>  geom_point()+<br>  labs(x = <span>"all_fpkm"</span>,<br>       y = <span>"all_count"</span>)+<br>  scale_color_manual(values=c(<span>"blue"</span>, <span>"grey"</span>,<span>"red"</span>))+<br>  geom_vline(xintercept = c( -<span>1.5</span>,<span>0</span>,<span>1.5</span>),lty=<span>4</span>,col=<span>"#ec183b"</span>,lwd=<span>0.8</span>) +<br>  geom_hline(yintercept =  c( -<span>1.5</span>,<span>0</span>,<span>1.5</span>),lty=<span>4</span>,col=<span>"#18a5ec"</span>,lwd=<span>0.8</span>) +<br>  <span>#xlim(-5,5)+</span><br>  <span>#ylim(-3,3)+</span><br>  theme_ggstatsplot()+<br>  theme(panel.grid.minor = element_blank(),<br>        panel.grid.major = element_blank())<br>p+stat_poly_eq(aes(label=paste(..eq.label..,..adj.rr.label..,..p.value.label..,sep = <span>"~~~~"</span>)),formula = y~x,parse=<span>T</span>,size=<span>3.0</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7142857142857143" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic4Ov1ktjg8xrG7hGCS7qiaAxUia9Qb7RbLDjXgQ5NccdBgQlOibtIEl60n4Ss3SLoOvE2icABLweo0TA/640?wx_fmt=png" data-type="png" data-w="672" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic4Ov1ktjg8xrG7hGCS7qiaAxUia9Qb7RbLDjXgQ5NccdBgQlOibtIEl60n4Ss3SLoOvE2icABLweo0TA/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code><br><br><span>## 002 再绘制差异基因的相关性散点图看看（整体都没有问题，差异肯定没有问题）</span><br>deg_limma=rownames(DEG_limma[DEG_limma$regulated!=<span>"not"</span>,])<br>deg_fpkm=rownames(fpkm_deg[fpkm_deg$g!=<span>"stable"</span>,])<br><br>ids=intersect(deg_limma,deg_fpkm)<br><span># 01看一看分别比较的整体相关性（此处就看整体差异基因吧）</span><br><span>library</span>(ggstatsplot)<br><span>library</span>(ggpmisc) <br><span># 02构建可视化所需的矩阵（相关性就两列值）</span><br>df=data.frame(<br>  deg1=fpkm_deg[ids,<span>"logFC"</span>],<br>  deg2=DEG_limma[ids,<span>"logFC"</span>] <span>#就是两列logFC的值（散点图）</span><br>)<br><br><span># 03绘制相关性图形</span><br>p &lt;- ggplot(df, aes(x=df$deg1,<br>                    df$deg2))+    <br>  geom_point()+<br>  labs(x = <span>"deg_fpkm"</span>,<br>       y = <span>"deg_count"</span>)+<br>  scale_color_manual(values=c(<span>"blue"</span>, <span>"grey"</span>,<span>"red"</span>))+<br>  geom_vline(xintercept = c( -<span>1.5</span>,<span>0</span>,<span>1.5</span>),lty=<span>4</span>,col=<span>"#ec183b"</span>,lwd=<span>0.8</span>) +<br>  geom_hline(yintercept =  c( -<span>1.5</span>,<span>0</span>,<span>1.5</span>),lty=<span>4</span>,col=<span>"#18a5ec"</span>,lwd=<span>0.8</span>) +<br>  <span>#xlim(-5,5)+</span><br>  <span>#ylim(-3,3)+</span><br>  theme_ggstatsplot()+<br>  theme(panel.grid.minor = element_blank(),<br>        panel.grid.major = element_blank())<br>p+stat_poly_eq(aes(label=paste(..eq.label..,..adj.rr.label..,..p.value.label..,sep = <span>"~~~~"</span>)),formula = y~x,parse=<span>T</span>,size=<span>3.0</span>)<br></code></pre><p data-tool="mdnice编辑器"><img data-ratio="0.7142857142857143" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic4Ov1ktjg8xrG7hGCS7qiaA2Eic0iccs342sD2urM5oFgKWN8zgGJKEiczqPKx1UxnE2PJVbjYTjctHg/640?wx_fmt=png" data-type="png" data-w="672" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic4Ov1ktjg8xrG7hGCS7qiaA2Eic0iccs342sD2urM5oFgKWN8zgGJKEiczqPKx1UxnE2PJVbjYTjctHg/640?wx_fmt=png">以上演示了count与FPKM进行差异分析的方式，并通过散点图展示了两者差异分析后获得的差异基因的区别，两种方式都是用limma包进行差异分析，感兴趣的小伙伴们也可以尝试下哈。</p><p data-tool="mdnice编辑器">两个散点图分别分析了FPKM与count进行差异分析后整体差异基因的相关性与差异基因的相关性。从图中，我们可以看出FPKM的差异分析结果与count的差异分析结果基本一致，可能算法在数据的差异中并非起绝对作用。<span>但为什么limma包不提倡用FPKM以及有的推文说用FPKM错了呢，然而此处针对FPKM与count进行差异分析，为什么结果没有很明显的区别呢？</span>小编觉得是一个很好的问题，希望明白的小伙伴可以补充下。</p><p data-tool="mdnice编辑器">如果有小伙伴或者很厉害的大牛觉得我的观点有问题有不妥，欢迎投稿给曾老师。小编一定给您排好版，将您的知识传递给小伙伴们，让大家更好的交流。小编自身也是个小菜鸟，受曾老师感染，在此处更多想做的就是传递自己的认知，分享自身的代码，写好这部分内容。希望迎来的不是很霸气的没有义务，劝小编退更的话语，谢谢。</p><p data-tool="mdnice编辑器"><br></p></section><p><br></p><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/KAqD1KVIVszNOShoUyRGnA",target="_blank" rel="noopener noreferrer">原文链接</a>
