---
title: "ArrayExpress数据集，又是安捷伦芯片？！"
date: 2022-11-29T05:18:24Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Acdemic"]
---
ArrayExpress数据集，又是安捷伦芯片？！ by 生信菜鸟团
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><blockquote data-tool="mdnice编辑器"><p>写在前面：菜鸟入坑不久，除了GEO和TCGA之外，认得的数据库资源寥寥可数，恰巧接到了一个任务，是分析上传到ArrayExpress数据的一个数据集，那就……试试看？</p></blockquote><h2 data-tool="mdnice编辑器"><span>ArrayExpress简介</span></h2><blockquote data-tool="mdnice编辑器"><p>ArrayExpress &lt; BioStudies &lt; EMBL-EBI</p></blockquote><figure data-tool="mdnice编辑器"><img data-ratio="0.38976592270005445" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicWW11AFIU8QHwdmrmLciaTf3XAXibnypzCplFRuZsO3y3icF2ewatYICh7aI4I0VAib3vRJBqWL1ASaQ/640?wx_fmt=png" data-type="png" data-w="1837" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicWW11AFIU8QHwdmrmLciaTf3XAXibnypzCplFRuZsO3y3icF2ewatYICh7aI4I0VAib3vRJBqWL1ASaQ/640?wx_fmt=png"></figure><blockquote data-tool="mdnice编辑器"><p>功能基因组学数据收集（ArrayExpress），储存了来自高通量功能基因组学实验的数据，并向研究界提供数据供再利用。根据社区指南，一项研究通常包含元数据，如详细的样本注释、协议、处理后的数据和原始数据。来自高通量测序研究的原始序列读数被中介到欧洲核苷酸档案馆（ENA），并提供链接以从ENA下载序列读数。数据可以通过其专用的提交工具Annotare提交给ArrayExpress集合。</p></blockquote><p data-tool="mdnice编辑器">在尝试分析数据集之前，有两篇文献建议大家先读一下，了解这个数据库的基本架构以及用R包下载和处理数据集的protocol.</p><blockquote data-tool="mdnice编辑器"><ul><li><section>ArrayExpress--a public database of microarray experiments and gene expression profiles - PubMed</section></li><li><section>Importing ArrayExpress datasets into R/Bioconductor - PMC</section></li></ul></blockquote><p data-tool="mdnice编辑器">通常，一个数据集提供的数据会包含以下的内容⬇</p><p data-tool="mdnice编辑器"><strong>MAGE-TAB</strong> is a tabular MIAME supportive file format and MAGE-TAB documents consist of <strong>five</strong> different types of files.</p><ul data-tool="mdnice编辑器"><li><section>(i) A ‘raw’ zip archive contains the <strong>raw data files</strong>, i.e. the files produced by the microarray image analysis software, such as CEL files for Affymetrix GeneChips or GPR files from GenePix.</section></li><li><section>(ii) A ‘data matrix’ file contains processed values, as provided by the data submitter, converted into a common <strong>tab-delimited text format representing a matrix of numbers.</strong></section></li><li><section>(iii) The Sample and Data Relationship Format (<strong>SDRF) tab-delimited file contains the relationships between samples and arrays</strong>, as well as sample properties and experimental factors, as provided by the data submitter.</section></li><li><section>(iv) The Array Design Format <strong>(ADF) tab-delimited file describes the design of an array</strong>, i. e. the sequence located at each feature on the array and annotation of the sequences.</section></li><li><section>(v) The Investigation Description Format <strong>(IDF) tab-delimited file contains top-level information about the experiment</strong> including title, description, submitter contact details and protocols.</section></li></ul><p data-tool="mdnice编辑器">其次，如何将以上的信息匹配到数据分析的过程中呢？</p><ul data-tool="mdnice编辑器"><li><section><p>The <strong>ArrayExpress package</strong> uses the zip archive with either the raw or the processed data to <strong>build the assayData component.</strong></p></section></li><li><section><p>The <strong>SDRF ﬁle is used to construct the phenoData table</strong>.</p></section></li><li><section><p>The <strong>ADF ﬁle is used to construct the featureData</strong>.</p></section></li><li><section><p>The <strong>IDF ﬁle to ﬁll in the experimentData components</strong>.</p></section></li></ul><blockquote data-tool="mdnice编辑器"><p>当然，ArrayExpress 官方教程还提供了如何检索以及下载数据的方式</p><p>（https://www.ebi.ac.uk/training/online/courses/array-express-discover-functional-genomics-data-quickly-and-easily/files-and-download/)</p></blockquote><h2 data-tool="mdnice编辑器"><span>数据集</span></h2><p data-tool="mdnice编辑器">今天我们要分析的数据集是这样的：</p><blockquote data-tool="mdnice编辑器"><p>A complete file that provides all of the mRNAs and miRNAs present in the arrays used  in this study, as well as the experimental conditions, is available online at the  ArrayExpress database (http://www.ebi.ac.uk/arrayexpress), Array Express accession <strong>EMTAB-3564 (for mRNA</strong> hybridizations) and <strong>E-MTAB—3563 (for miRNA</strong>  hybridizations).</p></blockquote><p data-tool="mdnice编辑器">以mRNA为例，文章提到“mRNA Expression Profiling with <strong>Agilent</strong> 4 × 44 K Rat Oligoarrays ”</p><p data-tool="mdnice编辑器">Agilent的芯片之前有分享过，可以借鉴</p><figure data-tool="mdnice编辑器"><img data-ratio="0.4774557165861514" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicWW11AFIU8QHwdmrmLciaTfrXWm1bpUOuNjAgtibuXudI9lPRkBoCNDrOuqgGN8AUiaFVAKsgPKyDug/640?wx_fmt=png" data-type="png" data-w="1242" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicWW11AFIU8QHwdmrmLciaTfrXWm1bpUOuNjAgtibuXudI9lPRkBoCNDrOuqgGN8AUiaFVAKsgPKyDug/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">数据集下载地址在这里：</p><blockquote data-tool="mdnice编辑器"><p>[E-MTAB-3564  https://www.ebi.ac.uk/biostudies/arrayexpress/studies/E-MTAB-3564]</p><p><img data-galleryid="" data-ratio="0.8005284015852048" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicWW11AFIU8QHwdmrmLciaTfGt7TpT3Qhd5cNrdJecJeC8miaTW1nl76bibic7kICaCaWaWNLyaWRyGoA/640?wx_fmt=png" data-type="png" data-w="1514" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicWW11AFIU8QHwdmrmLciaTfGt7TpT3Qhd5cNrdJecJeC8miaTW1nl76bibic7kICaCaWaWNLyaWRyGoA/640?wx_fmt=png"></p><p><br></p></blockquote><h2 data-tool="mdnice编辑器"><span>代码实操</span></h2><h3 data-tool="mdnice编辑器">step01 数据下载</h3><p data-tool="mdnice编辑器">有两种方式，可以用函数，但因为我是第一次用这个数据库，所以选择了保守的方式直接下载</p><blockquote data-tool="mdnice编辑器"><p>从ArrayExpress数据库下载一个芯片做注释_https://blog.csdn.net/u013429737/article/details/120196551)，这篇文章作者有详细的介绍。至于手动的方式就不用多说，进入数据集链接下载就行。</p></blockquote><h3 data-tool="mdnice编辑器">step02 原始数据处理</h3><p data-tool="mdnice编辑器">参考小洁老师的帖子：</p><pre data-tool="mdnice编辑器"><code><span># reference:https://www.jianshu.com/p/19ff109b6409</span><br><br><span>library</span>(ArrayExpress)<br><span>library</span>(limma)<br><span>library</span>(utils)<br><span>library</span>(stringr)<br><br>getwd()<br></code></pre><p data-tool="mdnice编辑器">读取数据：</p><pre data-tool="mdnice编辑器"><code>dir &lt;- <span>"../E-MTAB-3564/"</span> <span>##这个文件夹里只有样本的txt文件哦</span></code><p><img data-galleryid="" data-ratio="0.6174496644295302" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicWW11AFIU8QHwdmrmLciaTfRwfhO8HjqdQtguJPBibmcDpPvfGIpJmfTPVicb9G83ficUdrKCeHXMRrQ/640?wx_fmt=png" data-type="png" data-w="894" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicWW11AFIU8QHwdmrmLciaTfRwfhO8HjqdQtguJPBibmcDpPvfGIpJmfTPVicb9G83ficUdrKCeHXMRrQ/640?wx_fmt=png"></p><code><span></span>rawdat &lt;- list.files(dir,pattern = <span>"_\\d.txt"</span>) <span>##仅读取样本信息</span><br>pd &lt;-read.delim(<span>"../E-MTAB-3564.sdrf.txt"</span>) <span>##读取有临床信息的文件</span></code><p><img data-galleryid="" data-ratio="0.1546134663341646" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicWW11AFIU8QHwdmrmLciaTfEQpcwia0vb9wlosaDOmIUmELmgAM54t4Lrrts9YgZQKszSfAibxYuQZQ/640?wx_fmt=png" data-type="png" data-w="802" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicWW11AFIU8QHwdmrmLciaTfEQpcwia0vb9wlosaDOmIUmELmgAM54t4Lrrts9YgZQKszSfAibxYuQZQ/640?wx_fmt=png"></p><code><span></span></code><code><span><br></span></code></pre><p data-tool="mdnice编辑器">获取分组信息：</p><pre data-tool="mdnice编辑器"><code><span>#调整pd与rawdata的顺序一致</span><br>raw_order = gsub(<span>".txt"</span>,<span>""</span>,rawdat)<br>pd$Source.Name<br>x1 &lt;- rep(c(<span>"medulactrl_"</span>,<span>"medulaovx_"</span>,<span>"calvariactrl_"</span>,<span>"calvariaovx_"</span>),each=<span>3</span>)<br>x2 &lt;- rep(<span>1</span>:<span>3</span>,each=<span>1</span>,times=<span>4</span>)<br>pd$group &lt;- paste0(x1,x2)<br>pd$group<br>pd = pd[match(raw_order,pd$group),]<br><br>group_list &lt;- ifelse(str_detect(pd$group,<span>"calvariactrl"</span>),<span>"calvariactrl"</span>,<br>                    ifelse(str_detect(pd$group,<span>"calvariaovx"</span>),<span>"calvariaovx"</span>,<br>                           ifelse(str_detect(pd$group,<span>"medulactrl"</span>),<span>"medulactrl"</span>,<br>                     <span>"medulaovx"</span>)))<br>table(group_list)<br></code></pre><p data-tool="mdnice编辑器">读取原始数据：</p><pre data-tool="mdnice编辑器"><code>raw_dir = <span>"../E-MTAB-3564/"</span><br>raw_datas = paste0(raw_dir,dir(raw_dir))<br>x &lt;- read.maimages(raw_datas,<br>                   <span>source</span>=<span>"agilent"</span>, <br>                   green.only=<span>TRUE</span>,<br>                   other.columns = <span>"gIsWellAboveBG"</span>)<br>dim(x)<br></code></pre><p data-tool="mdnice编辑器">背景校正和标准化：</p><pre data-tool="mdnice编辑器"><code>y &lt;- backgroundCorrect(x, method=<span>"normexp"</span>)<br>y &lt;- normalizeBetweenArrays(y, method=<span>"quantile"</span>)<br>class(y)<br></code></pre><p data-tool="mdnice编辑器">基因过滤：</p><pre data-tool="mdnice编辑器"><code>Control &lt;- y$genes$ControlType==<span>1L</span>;table(Control)<span>##去除对照探针</span><br>NoSymbol &lt;- is.na(y$genes$GeneName);table(NoSymbol)<span>##去除匹配不到genesymbol的探针</span><br>IsExpr &lt;- rowSums(y$other$gIsWellAboveBG &gt; <span>0</span>) &gt;= <span>6</span>;table(IsExpr)<span>##去除不表达的探针，去除的标准是：至少在一半样本中高于背景，通过y(other)gIsWellAboveBG来判断</span><br>Isdup &lt;- duplicated(y$genes$GeneName);table(Isdup)<span>##测到多次的基因，只保留一个探针</span><br>yfilt &lt;- y[!Control &amp; !NoSymbol &amp; IsExpr &amp; !Isdup, ]<br>dim(yfilt)<br></code></pre><p data-tool="mdnice编辑器">现在应该拿到熟悉的矩阵了吧</p><pre data-tool="mdnice编辑器"><code>exp = yfilt@.Data[[<span>1</span>]]<br>colnames(exp) &lt;- str_split(colnames(exp),<span>"/"</span>,simplify = <span>T</span>)[,<span>3</span>]<br>boxplot(exp,las=<span>2</span>)<br>exp[<span>1</span>:<span>2</span>,<span>1</span>:<span>2</span>]<br></code></pre><p data-tool="mdnice编辑器">基因注释：</p><pre data-tool="mdnice编辑器"><code>anno = yfilt$genes<br>nrow(anno);nrow(exp)<br>rownames(exp)=rownames(anno)<br>ids = unique(anno$GeneName)<br>exp = exp[!duplicated(anno$GeneName),]<br>rownames(exp) = ids<br>exp[<span>1</span>:<span>4</span>,<span>1</span>:<span>4</span>]<br><br>range(exp)<br><br><span>##保存</span><br>getwd()<br>save(group_list,exp,file = <span>"step1-output.Rdata"</span>)<br></code></pre><h3 data-tool="mdnice编辑器">step03 差异分析</h3><p data-tool="mdnice编辑器">多组间的数据该如何做差异分析呢？</p><pre data-tool="mdnice编辑器"><code>rm(list = ls())<br><br>load(<span>"step1-output.Rdata"</span>)<br><br><span># 差异分析 --------------------------------------------------------------------</span><br><span>###构建表达矩阵，非常重要！！！</span><br><span>library</span>(limma)<br>design &lt;- model.matrix(~<span>0</span>+factor(group_list))<br>colnames(design)=levels(factor(group_list))<br>rownames(design)=colnames(exp)<br>px &lt;- colnames(design);px<br>contrast.matrix &lt;- makeContrasts(paste0(px[<span>2</span>],<span>"-"</span>,px[<span>1</span>]), <br>                                 paste0(px[<span>4</span>],<span>"-"</span>,px[<span>3</span>]), <br>                                 paste0(px[<span>3</span>],<span>"-"</span>,px[<span>1</span>]), <br>                                 paste0(px[<span>4</span>],<span>"-"</span>,px[<span>2</span>]),<br>                                 levels=design) <span>##实验组在前，对照组在后</span><br>fit &lt;- lmFit(exp, design)<br>fit2 &lt;- contrasts.fit(fit, contrast.matrix)<br>fit2 &lt;- eBayes(fit2)<br><br>deg = lapply(<span>1</span>:ncol(design), <span>function</span>(i){<br>  topTable(fit2,coef = i,number = <span>Inf</span>)<br>})<br>names(deg) = colnames(contrast.matrix)<br><br></code></pre><p data-tool="mdnice编辑器">差异分析阈值[Unpaired T test statistical analysis (P≤0.005) with a fold change  ≥2.0 was used for mRNA]</p><pre data-tool="mdnice编辑器"><code>logFC_t=<span>2.0</span><br>P.Value_t=<span>0.005</span><br><span>library</span>(dplyr)<br><span>library</span>(clusterProfiler)<br><span>library</span>(org.Rn.eg.db) ##注意物种<br><span>for</span>(i <span>in</span> <span>1</span>:length(deg)){<br>  change=ifelse(deg[[i]] $P.Value&gt;P.Value_t,<span>'stable'</span>, <br>                ifelse( deg[[i]]$logFC &gt;logFC_t,<span>'up'</span>, <br>                        ifelse( deg[[i]]$logFC &lt; -logFC_t,<span>'down'</span>,<span>'stable'</span>) )<br>  )<br>  deg[[i]] &lt;- mutate(deg[[i]],change)<br>  print(table(deg[[i]]$change))<br>  deg[[i]] &lt;- mutate(deg[[i]],v = -log10(P.Value))<br><br><span>#加ENTREZID列，后面富集分析要用</span><br>deg[[i]]$symbol &lt;- rownames(deg[[i]])<br>s2e &lt;- bitr(unique(deg[[i]]$symbol), fromType = <span>"SYMBOL"</span>,<br>            toType = c( <span>"ENTREZID"</span>),<br>            OrgDb = org.Rn.eg.db)<br>deg[[i]] &lt;- inner_join(deg[[i]],s2e,by=c(<span>"symbol"</span>=<span>"SYMBOL"</span>))<br>}<br>save(deg,group_list,file  =<span>"DEG.Rdata"</span>)<br></code></pre><h3 data-tool="mdnice编辑器">step04 可视化</h3><p data-tool="mdnice编辑器">PCA、热图和火山图是不可或缺的，想想怎么能把它们放到一起呢？</p><pre data-tool="mdnice编辑器"><code><span>###主成分分析</span><br>dat=as.data.frame(t(exp))<br><span>library</span>(FactoMineR)<span>#画主成分分析图需要加载这两个包</span><br><span>library</span>(factoextra) <br><span># pca的统一操作走起</span><br>dat.pca &lt;- PCA(dat, graph = <span>FALSE</span>)<br>p1 &lt;- fviz_pca_ind(dat.pca,<br>                   geom.ind = <span>"point"</span>, <span># show points only (nbut not "text")</span><br>                   col.ind = group_list, <span># color by groups</span><br>                   <span>#palette = c("#00AFBB", "#E7B800"),</span><br>                   addEllipses = <span>TRUE</span>, <span># Concentration ellipses</span><br>                   legend.title = <span>"Groups"</span><br>)<br></code></pre><pre data-tool="mdnice编辑器"><code><span>#热图 </span><br>cg=names(tail(sort(apply(exp,<span>1</span>,sd)),<span>500</span>))<br><span>if</span>(!<span>require</span>(pheatmap))install.packages(<span>"pheatmap"</span>)<br><span>library</span>(pheatmap)<br>n=exp[cg,]<br>annotation_col=data.frame(group=group_list)<br>rownames(annotation_col)=colnames(n) <br>p2 &lt;- pheatmap(n,<br>               show_colnames =<span>F</span>,<br>               show_rownames = <span>F</span>,<br>               cluster_cols = <span>F</span>,<br>               annotation_col=annotation_col,<br>               scale = <span>"row"</span>)<br><span>require</span>(ggplotify)<br>p2= as.ggplot(p2)<br></code></pre><pre data-tool="mdnice编辑器"><code><span>##火山图</span><br><span>library</span>(tinyarray)<br>vo = lapply(<span>1</span>:length(deg),<span>function</span>(k){<br>  draw_volcano(deg[[k]],<br>               lab = colnames(contrast.matrix)[k],<br>               pkg = <span>4</span>,<br>               logFC_cutoff = logFC_t,<br>               pvalue_cutoff = P.Value_t,<br>               symmetry = <span>T</span>)<br>})<br></code></pre><p data-tool="mdnice编辑器">拼图</p><pre data-tool="mdnice编辑器"><code><span>library</span>(patchwork)<br>plotlist = c(list(p2,p1),vo)<br>layout(matrix(c(<span>1</span>,<span>1</span>,<span>2</span>,<span>2</span>,<span>3</span>,<span>3</span>,<span>3</span>,<span>3</span>), <span>2</span>, <span>4</span>,byrow = <span>TRUE</span>))<br>patchwork &lt;- wrap_plots(plotlist) <br>ggsave(patchwork,filename = <span>"../output/patchwork.pdf"</span>,width = <span>15</span>,height = <span>15</span>)<br><br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="1.0098870056497176" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicWW11AFIU8QHwdmrmLciaTfJbF2mUrPBibs6GQw9HmLJ5oHgSChRekevVDuvGSYsBXH3bOOIdVEibiaQ/640?wx_fmt=png" data-type="png" data-w="708" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicWW11AFIU8QHwdmrmLciaTfJbF2mUrPBibs6GQw9HmLJ5oHgSChRekevVDuvGSYsBXH3bOOIdVEibiaQ/640?wx_fmt=png"><figcaption>一锅出</figcaption></figure><p data-tool="mdnice编辑器">那么，你感兴趣的到底是哪两个组的差异分析结果呢？anyway，我们都搞定了，任君挑选~~~</p></section><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/5wugXfWD4SfN25qGZk4d8Q",target="_blank" rel="noopener noreferrer">原文链接</a>
