---
title: "CNS图表复现07—原来这篇文章有两个单细胞表达矩阵"
date: 2022-09-01T10:50:04Z
draft: ["false"]
tags: [
  "fetched",
  "单细胞天地"
]
categories: ["Acdemic"]
---
CNS图表复现07—原来这篇文章有两个单细胞表达矩阵 by 单细胞天地
------
<div><section data-tools="新媒体管家" data-label="powered by xmt.cn"><br></section><section data-tools="新媒体排版" data-id="2903370" data-style-type="undefined"><section data-style-type="5" data-id="2430070"><section data-style-type="5" data-id="2390348"><section><section powered-by="xiumi.us"><section><section><section powered-by="xiumi.us"><section><section><br><br></section><section><section><section powered-by="xiumi.us"><section><section><p>分享是一种态度</p></section></section></section></section></section></section></section></section><section><section powered-by="xiumi.us"><section><section><img data-ratio="0.8738095" data-src="https://mmbiz.qpic.cn/mmbiz_gif/siaia0BDGJdjSnnb9ibXpFagGPWQ0JjSSTNVBZthROEhicfINuLcxX2Ro3Zb600LJhrzeYIb120tLjSVXEtLn14DwA/640?wx_fmt=gif" data-type="gif" data-w="420" width="100%" src="https://mmbiz.qpic.cn/mmbiz_gif/siaia0BDGJdjSnnb9ibXpFagGPWQ0JjSSTNVBZthROEhicfINuLcxX2Ro3Zb600LJhrzeYIb120tLjSVXEtLn14DwA/640?wx_fmt=gif"><br></section><br></section><br></section><br></section></section></section></section></section></section><section data-width="100%"><section><br><br></section><br></section><p><br><br></p></section><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><h1 data-tool="mdnice编辑器"><span></span><span>回顾</span></h1><p data-tool="mdnice编辑器">我们的CNS图表复现之旅已经开始，前面6讲是；</p><ul data-tool="mdnice编辑器"><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247488711&amp;idx=1&amp;sn=4e494793706b7106ed120144767e64f7&amp;scene=21#wechat_redirect" data-linktype="2">CNS图表复现01—读入csv文件的表达矩阵构建Seurat对象</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247488727&amp;idx=1&amp;sn=97a1fc3e73c6d4cb4075cff09b8db01e&amp;scene=21#wechat_redirect" data-linktype="2">CNS图表复现02—Seurat标准流程之聚类分群</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247488742&amp;idx=1&amp;sn=43512abff6c516930b96316ca5332b23&amp;scene=21#wechat_redirect" data-linktype="2">CNS图表复现03—单细胞区分免疫细胞和肿瘤细胞</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247488815&amp;idx=1&amp;sn=2c49a080cf45a7909e217cdafd357768&amp;scene=21#wechat_redirect" data-linktype="2">CNS图表复现04—单细胞聚类分群的resolution参数问题</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247488835&amp;idx=1&amp;sn=04f1a1606d83f8273ba6387c69399eb9&amp;scene=21#wechat_redirect" data-linktype="2">CNS图表复现05—免疫细胞亚群再分类</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247488856&amp;idx=1&amp;sn=806bc6a15abbefd94b0b717f3244e826&amp;scene=21#wechat_redirect" data-linktype="2">CNS图表复现06-根据CellMarker网站进行人工校验免疫细胞亚群</a></section></li></ul><p data-tool="mdnice编辑器">如果你也想加入交流群，自己去：<a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247495627&amp;idx=1&amp;sn=6cd6982928b04405fe0caac121c8d4d4&amp;scene=21#wechat_redirect" data-linktype="2">你要的rmarkdown文献图表复现全套代码来了（单细胞）</a>找到我们的拉群小助手哈。</p><h1 data-tool="mdnice编辑器"><span></span><span>正文</span></h1><p data-tool="mdnice编辑器">文章的第一次分群按照 ：</p><ul data-tool="mdnice编辑器"><li><section>immune (CD45+,PTPRC),</section></li><li><section>epithelial/cancer (EpCAM+,EPCAM),</section></li><li><section>stromal (CD10+,MME,fibo or CD31+,PECAM1,endo)</section></li></ul><p data-tool="mdnice编辑器">的表达量分布，可以拿到如下所示的：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.6856368563685636" data-src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjQQExQgPESPiavtGIDz68RBJh2na9VicmCxzWgpzmLx9tDlRsI5cJTEZoEoxEOWibZxLlcQTeEvzly1g/640?wx_fmt=png" data-type="png" data-w="738" src="https://mmbiz.qpic.cn/mmbiz_png/siaia0BDGJdjQQExQgPESPiavtGIDz68RBJh2na9VicmCxzWgpzmLx9tDlRsI5cJTEZoEoxEOWibZxLlcQTeEvzly1g/640?wx_fmt=png"><figcaption>第一次分群</figcaption></figure><p data-tool="mdnice编辑器">文章提到的<strong>各大亚群细胞数量</strong>是：(epithelial cells [n = 5,581], immune cells [n = 13,431], stromal cells[n = 4,249]).</p><p data-tool="mdnice编辑器">我发现自己怎么都重复不出来，因为唯一的质量控制步骤，细胞数量就开始不吻合了：</p><pre data-tool="mdnice编辑器"><span></span><code>  <span># </span><br>  <span># 这里没有绝对的过滤标准</span><br>  <span># Filter cells so that remaining cells have nGenes &gt;= 500 and nReads &gt;= 50000</span><br>  <span># 26485 features across 27489 samples</span><br>  main_tiss_filtered &lt;- subset(x=main_tiss, subset = nCount_RNA &gt; <span>50000</span> &amp; nFeature_RNA &gt; <span>500</span>)<br>  main_tiss_filtered<br>  <span># 26485 features across 21444 samples</span><br></code></pre><p data-tool="mdnice编辑器">也就是说27489个细胞，被过滤成为了 21444细胞，而文章的细胞数量是23261，但是我明明是跟文章作者一模一样的处理过程。</p><p data-tool="mdnice编辑器">真的是百思不得其解。</p><p data-tool="mdnice编辑器">后来认真看了看它提供的文件列表：</p><pre data-tool="mdnice编辑器"><span></span><code> 1.4G Aug 30 12:04 S01_datafinal.csv<br>   11M Aug 30 11:34 S01_metacells.csv<br>  3.5K Aug 30 11:35 by_sample_ratios_driver_muts_3.30.20.csv<br>  3.3K Aug 30 11:35 neo-osi_metadata.csv<br>  184M Aug 30 11:35 neo-osi_rawdata.csv<br>   13K Aug 30 11:34 samples_x_genes_3.30.20.csv<br></code></pre><p data-tool="mdnice编辑器">原来是有两个表达矩阵文件，neo-osi_rawdata.csv 和 S01_datafinal.csv 文件，都需要走流程~</p><p data-tool="mdnice编辑器">需要注意的是这个文章作者并没有上传这些文件到GEO上面，仅仅是上传了<strong>原始数据</strong>在：https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA591860，这个原始数据接近4T的文件。</p><p data-tool="mdnice编辑器">这些csv文件呢，作者上传到了谷歌云盘，我们人工搬运到了百度云，在交流群可以拿到。如果你也想加入交流群，自己去：<a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247495627&amp;idx=1&amp;sn=6cd6982928b04405fe0caac121c8d4d4&amp;scene=21#wechat_redirect" data-linktype="2">你要的rmarkdown文献图表复现全套代码来了（单细胞）</a>找到我们的拉群小助手哈。</p><p data-tool="mdnice编辑器">我漏掉的这个 neo-osi_rawdata.csv 走流程拿到的是  2070 个单细胞</p><pre data-tool="mdnice编辑器"><span></span><code><span># 26485 features across 3507 samples within 1 assay </span><br>  osi_object_filtered &lt;- subset(x=osi_object, subset = nCount_RNA &gt; <span>50000</span> &amp; nFeature_RNA &gt; <span>500</span>)<br>  osi_object_filtered <br>  <span># 26485 features across 2070 samples within 1 assay </span><br></code></pre><p data-tool="mdnice编辑器">然后把两个单细胞表达矩阵构建好的Seurat 对象合并起来，代码如下：</p><pre data-tool="mdnice编辑器"><span></span><code>main_tiss_filtered<br>osi_object_filtered<br>main_tiss_filtered1 &lt;- merge(x = main_tiss_filtered, y = osi_object_filtered)<br>main_tiss_filtered1 <br></code></pre><p data-tool="mdnice编辑器">得到的就差不多是文章里面的 23514 个单细胞啦，文章的细胞数量是23261。</p><pre data-tool="mdnice编辑器"><span></span><code>&gt; main_tiss_filtered1<br>An object of class Seurat <br>26485 features across 23514 samples within 1 assay <br>Active assay: RNA (26485 features, 0 variable features)<br></code></pre><h1 data-tool="mdnice编辑器"><span></span><span>全部代码可以复制粘贴的</span></h1><p data-tool="mdnice编辑器">总共是  270 行，基本上涵盖了我们单细胞天地的方方面面知识点。</p><pre data-tool="mdnice编辑器"><span></span><code>rm(list=ls())<br>options(stringsAsFactors = <span>F</span>)<br><span>library</span>(Seurat)<br>pro=<span>'first'</span><br><br><span># 单细胞项目：来自于30个病人的49个组织样品，跨越3个治疗阶段</span><br><br><span># Single-cell RNA sequencing (scRNA-seq) of</span><br><span># metastatic lung cancer was performed using 49 clinical biopsies obtained from 30 patients before and during</span><br><span># targeted therapy. </span><br><br><span># before initiating systemic targeted therapy (TKI naive [TN]), </span><br><span># at the residual disease (RD) state,</span><br><span># acquired drug resistance (progression [PD]).</span><br><br><span># 首先读取第一个单细胞转录组表达矩阵文件：</span><br><span>if</span>(<span>T</span>){<br>  <br>  <span># Load data </span><br>  dir=<span>'./'</span><br>  Sys.time()<br>  raw.data &lt;- read.csv(paste(dir,<span>"Data_input/csv_files/S01_datafinal.csv"</span>, sep=<span>""</span>), header=<span>T</span>, row.names = <span>1</span>)<br>  Sys.time()<br>  dim(raw.data) <span># </span><br>  raw.data[<span>1</span>:<span>4</span>,<span>1</span>:<span>4</span>]<br>  head(colnames(raw.data)) <br>  <span># Load metadata </span><br>  metadata &lt;- read.csv(paste(dir,<span>"Data_input/csv_files/S01_metacells.csv"</span>, sep=<span>""</span>), row.names=<span>1</span>, header=<span>T</span>)<br>  head(metadata) <br>  <br>  <span># Find ERCC's, compute the percent ERCC, and drop them from the raw data.</span><br>  <br>  erccs &lt;- grep(pattern = <span>"^ERCC-"</span>, x = rownames(x = raw.data), value = <span>TRUE</span>)<br>  percent.ercc &lt;- Matrix::colSums(raw.data[erccs, ])/Matrix::colSums(raw.data)<br>  fivenum(percent.ercc)<br>  ercc.index &lt;- grep(pattern = <span>"^ERCC-"</span>, x = rownames(x = raw.data), value = <span>FALSE</span>)<br>  raw.data &lt;- raw.data[-ercc.index,]<br>  dim(raw.data) <br>  <span># Create the Seurat object with all the data (unfiltered)</span><br>  main_tiss &lt;- CreateSeuratObject(counts = raw.data)<br>  <span># add rownames to metadta </span><br>  row.names(metadata) &lt;- metadata$cell_id<br>  <span># add metadata to Seurat object </span><br>  main_tiss &lt;- AddMetaData(object = main_tiss, metadata = metadata)<br>  main_tiss &lt;- AddMetaData(object = main_tiss, percent.ercc, col.name = <span>"percent.ercc"</span>)<br>  <span># Head to check</span><br>  head(main_tiss@meta.data)<br>  <br>  <span># Calculate percent ribosomal genes and add to metadata</span><br>  ribo.genes &lt;- grep(pattern = <span>"^RP[SL][[:digit:]]"</span>, x = rownames(x = main_tiss@assays$RNA@data), value = <span>TRUE</span>)<br>  percent.ribo &lt;- Matrix::colSums(main_tiss@assays$RNA@counts[ribo.genes, ])/Matrix::colSums(main_tiss@assays$RNA@data)<br>  fivenum(percent.ribo)<br>  main_tiss &lt;- AddMetaData(object = main_tiss, metadata = percent.ribo, col.name = <span>"percent.ribo"</span>)<br>  main_tiss<br>  <br>  <span># 唯一的质量控制步骤：</span><br>  <span># 这里没有绝对的过滤标准</span><br>  <span># Filter cells so that remaining cells have nGenes &gt;= 500 and nReads &gt;= 50000</span><br>  <span># 26485 features across 27489 samples</span><br>  main_tiss_filtered &lt;- subset(x=main_tiss, subset = nCount_RNA &gt; <span>50000</span> &amp; nFeature_RNA &gt; <span>500</span>)<br>  main_tiss_filtered<br>  <span># 26485 features across 21444 samples</span><br>}<br>main_tiss_filtered<br><span># 26485 features across 21444 samples</span><br><br><span># 然后读取第二个单细胞转录组表达矩阵文件：</span><br><span>if</span>(<span>T</span>){<br>  <span># load and clean rawdata</span><br>  <span># Load data </span><br>  dir=<span>'./'</span><br>  Sys.time()<br>  osi.raw.data &lt;- read.csv(paste(dir,<span>"Data_input/csv_files/neo-osi_rawdata.csv"</span>, sep=<span>""</span>), row.names = <span>1</span>)<br>  <br>  Sys.time()<br> <br>  colnames(osi.raw.data) &lt;- gsub(<span>"_S.*.homo"</span>, <span>""</span>, colnames(osi.raw.data))<br>  osi.raw.data[<span>1</span>:<span>4</span>,<span>1</span>:<span>4</span>]<br>  tail( osi.raw.data[ <span>1</span>:<span>4</span>])<br>  dim(osi.raw.data)<br>  <br>  <span># drop sequencing details from gene count table</span><br>  <span># 因为是HTseq这个软件跑出来的表达矩阵</span><br>  row.names(osi.raw.data)[grep(<span>"__"</span>, row.names(osi.raw.data))]<br>  osi.raw.data &lt;- osi.raw.data[-grep(<span>"__"</span>, row.names(osi.raw.data)),]<br>  <br>  <span>#Make osi.metadata by cell from osi.metadata by plate</span><br>  <br>  osi.metadata &lt;- read.csv(paste(dir, <span>"Data_input/csv_files/neo-osi_metadata.csv"</span>, sep = <span>""</span>))<br>  osi.meta.cell &lt;- as.data.frame(colnames(osi.raw.data))<br>  osi.meta.cell &lt;- data.frame(do.call(<span>'rbind'</span>, strsplit(as.character(osi.meta.cell$`colnames(osi.raw.data)`),<span>'_'</span>,fixed=<span>TRUE</span>)))<br>  rownames(osi.meta.cell) &lt;- paste(osi.meta.cell$X1, osi.meta.cell$X2, sep = <span>"_"</span>)<br>  colnames(osi.meta.cell) &lt;- c(<span>"well"</span>, <span>"plate"</span>)<br>  osi.meta.cell$cell_id &lt;- rownames(osi.meta.cell)<br>  <span>library</span>(dplyr)<br>  osi.metadata &lt;- left_join(osi.meta.cell, osi.metadata, by = <span>"plate"</span>)<br>  rownames(osi.metadata) &lt;- osi.metadata$cell_id<br>  head(osi.metadata)<br>  <br>  unique(osi.metadata$plate)<br>  <br>  <br>  <span># Find ERCC's, compute the percent ERCC, and drop them from the raw data.</span><br>  <br>  erccs &lt;- grep(pattern = <span>"^ERCC-"</span>, x = rownames(x = osi.raw.data), value = <span>TRUE</span>)<br>  percent.ercc &lt;- Matrix::colSums(osi.raw.data[erccs, ])/Matrix::colSums(osi.raw.data)<br>  ercc.index &lt;- grep(pattern = <span>"^ERCC-"</span>, x = rownames(x = osi.raw.data), value = <span>FALSE</span>)<br>  osi.raw.data &lt;- osi.raw.data[-ercc.index,]<br>  dim(osi.raw.data)<br>  <br>  <br>  <span># Create the Seurat object with all the data (unfiltered)</span><br>  <br>  osi_object &lt;- CreateSeuratObject(counts = osi.raw.data)<br>  osi_object &lt;- AddMetaData(object = osi_object, metadata = osi.metadata)<br>  osi_object &lt;- AddMetaData(object = osi_object, percent.ercc, col.name = <span>"percent.ercc"</span>)<br>  <span># Changing nUMI column name to nReads</span><br>  colnames(osi_object@meta.data)[colnames(osi_object@meta.data) == <span>'nUMI'</span>] &lt;- <span>'nReads'</span><br>  head(osi_object@meta.data)<br>  <br>  <br>  <span># Calculate percent ribosomal genes and add to osi.metadata</span><br>  <br>  ribo.genes &lt;- grep(pattern = <span>"^RP[SL][[:digit:]]"</span>, x = rownames(x = osi_object@assays$RNA@data), value = <span>TRUE</span>)<br>  percent.ribo &lt;- Matrix::colSums(osi_object@assays$RNA@data[ribo.genes, ])/Matrix::colSums(osi_object@assays$RNA@data)<br>  osi_object &lt;- AddMetaData(object = osi_object, metadata = percent.ribo, col.name = <span>"percent.ribo"</span>)<br>  osi_object<br>  <span># 26485 features across 3507 samples within 1 assay </span><br>  osi_object_filtered &lt;- subset(x=osi_object, subset = nCount_RNA &gt; <span>50000</span> &amp; nFeature_RNA &gt; <span>500</span>)<br>  osi_object_filtered <br>  <span># 26485 features across 2070 samples within 1 assay </span><br>  VlnPlot(osi_object_filtered, features = <span>"nFeature_RNA"</span>)<br>  VlnPlot(osi_object_filtered, features = <span>"nCount_RNA"</span>, log = <span>TRUE</span>)<br><br>}<br>main_tiss_filtered<br>osi_object_filtered<br>main_tiss_filtered1 &lt;- merge(x = main_tiss_filtered, y = osi_object_filtered)<br>main_tiss_filtered1 <br>raw_sce &lt;- main_tiss_filtered1<br><span>if</span>(<span>F</span>){<br>  <span># Drop any samples with 10 or less cells</span><br>  <br>  main_tiss_filtered@meta.data$sample_name &lt;- as.character(main_tiss_filtered@meta.data$sample_name)<br>  sample_name &lt;- as.character(main_tiss_filtered@meta.data$sample_name)<br>  <span># Make table </span><br>  tab.1 &lt;- table(main_tiss_filtered@meta.data$sample_name) <br>  <span># Which samples have less than 10 cells </span><br>  samples.keep &lt;- names(which(tab.1 &gt; <span>10</span>))<br>  metadata_keep &lt;- filter(main_tiss_filtered@meta.data, sample_name %<span>in</span>% samples.keep)<br>  <span># Subset Seurat object </span><br>  tiss_subset &lt;- subset(main_tiss_filtered, cells=as.character(metadata_keep$cell_id))<br>  tiss_subset<br>  raw_sce &lt;- tiss_subset<br>}<br><br><span># Gene-expression profiles of 23,261 cells were retained after quality control filtering.</span><br><br><br><br>raw_sce <br>rownames(raw_sce)[grepl(<span>'^mt-'</span>,rownames(raw_sce),ignore.case = <span>T</span>)]<br>rownames(raw_sce)[grepl(<span>'^Rp[sl]'</span>,rownames(raw_sce),ignore.case = <span>T</span>)]<br><br>raw_sce[[<span>"percent.mt"</span>]] &lt;- PercentageFeatureSet(raw_sce, pattern = <span>"^MT-"</span>)<br>fivenum(raw_sce[[<span>"percent.mt"</span>]][,<span>1</span>])<br>rb.genes &lt;- rownames(raw_sce)[grep(<span>"^RP[SL]"</span>,rownames(raw_sce),ignore.case = <span>T</span>)]<br>C&lt;-GetAssayData(object = raw_sce, slot = <span>"counts"</span>)<br>percent.ribo &lt;- Matrix::colSums(C[rb.genes,])/Matrix::colSums(C)*<span>100</span><br>fivenum(percent.ribo)<br>raw_sce &lt;- AddMetaData(raw_sce, percent.ribo, col.name = <span>"percent.ribo"</span>)<br><br>plot1 &lt;- FeatureScatter(raw_sce, feature1 = <span>"nCount_RNA"</span>, feature2 = <span>"percent.ercc"</span>)<br>plot2 &lt;- FeatureScatter(raw_sce, feature1 = <span>"nCount_RNA"</span>, feature2 = <span>"nFeature_RNA"</span>)<br>CombinePlots(plots = list(plot1, plot2))<br><br>VlnPlot(raw_sce, features = c(<span>"percent.ribo"</span>, <span>"percent.ercc"</span>), ncol = <span>2</span>)<br>VlnPlot(raw_sce, features = c(<span>"nFeature_RNA"</span>, <span>"nCount_RNA"</span>), ncol = <span>2</span>)<br>VlnPlot(raw_sce, features = c(<span>"percent.ribo"</span>, <span>"nCount_RNA"</span>), ncol = <span>2</span>)<br>raw_sce<br> <br>sce=raw_sce<br>sce<br><span>#sce &lt;- NormalizeData(sce, normalization.method =  "LogNormalize",  scale.factor = 1e6)</span><br>sce &lt;- NormalizeData(sce, normalization.method =  <span>"LogNormalize"</span>,  scale.factor = <span>1e5</span>)<br>GetAssay(sce,assay = <span>"RNA"</span>)<br>sce &lt;- FindVariableFeatures(sce, <br>                            selection.method = <span>"vst"</span>, nfeatures = <span>2000</span>) <br><span># 步骤 ScaleData 的耗时取决于电脑系统配置（保守估计大于一分钟）</span><br>sce &lt;- ScaleData(sce) <br>sce &lt;- RunPCA(object = sce, pc.genes = VariableFeatures(sce)) <br>DimHeatmap(sce, dims = <span>1</span>:<span>12</span>, cells = <span>100</span>, balanced = <span>TRUE</span>)<br>ElbowPlot(sce)<br><span># 跟文章保持一致，第一次分群，resolution 采取 0.5</span><br>sce &lt;- FindNeighbors(sce, dims = <span>1</span>:<span>15</span>)<br>sce &lt;- FindClusters(sce, resolution = <span>0.2</span>)<br>table(sce@meta.data$RNA_snn_res.0.2) <br>sce &lt;- FindClusters(sce, resolution = <span>0.8</span>)<br>table(sce@meta.data$RNA_snn_res.0.8) <br>sce &lt;- FindClusters(sce, resolution = <span>0.5</span>)<br>table(sce@meta.data$RNA_snn_res.0.5) <br><br><span># 不同的 resolution 分群的结果不一样</span><br><span>library</span>(gplots)<br>tab.1=table(sce@meta.data$RNA_snn_res.0.2,sce@meta.data$RNA_snn_res.0.8) <br>balloonplot(tab.1)<br><br><span># Check clustering stability at given resolution  </span><br><span># Set different resolutions </span><br>res.used &lt;- seq(<span>0.1</span>,<span>1</span>,by=<span>0.2</span>)<br>res.used<br><span># Loop over and perform clustering of different resolutions </span><br><span>for</span>(i <span>in</span> res.used){<br>  sce &lt;- FindClusters(object = sce, verbose = <span>T</span>, resolution = res.used)<br>}<br><span># Make plot </span><br><span>library</span>(clustree)<br>clus.tree.out &lt;- clustree(sce) +<br>  theme(legend.position = <span>"bottom"</span>) + <br>  scale_color_brewer(palette = <span>"Set1"</span>) +<br>  scale_edge_color_continuous(low = <span>"grey80"</span>, high = <span>"red"</span>)<br><br>clus.tree.out <br><br>res.used &lt;- <span>0.5</span><br>sce &lt;- FindClusters(object = sce, verbose = <span>T</span>, resolution = res.used)<br><br><br>set.seed(<span>123</span>)<br>sce &lt;- RunTSNE(object = sce, dims = <span>1</span>:<span>15</span>, do.fast = <span>TRUE</span>)<br>DimPlot(sce,reduction = <span>"tsne"</span>,label=<span>T</span>)<br>phe=data.frame(cell=rownames(sce@meta.data),<br>               cluster =sce@meta.data$seurat_clusters)<br>head(phe)<br>table(phe$cluster)<br>tsne_pos=Embeddings(sce,<span>'tsne'</span>) <br>head(phe)<br>table(phe$cluster)<br>head(tsne_pos) <br>dat=cbind(tsne_pos,phe)<br><br>save(dat,file=paste0(pro,<span>'_for_tSNE.pos.Rdata'</span>)) <br>load(file=paste0(pro,<span>'_for_tSNE.pos.Rdata'</span>))  <br><span>library</span>(ggplot2)<br>p=ggplot(dat,aes(x=tSNE_1,y=tSNE_2,color=cluster))+geom_point(size=<span>0.95</span>)<br>p=p+stat_ellipse(data=dat,aes(x=tSNE_1,y=tSNE_2,fill=cluster,color=cluster),<br>                 geom = <span>"polygon"</span>,alpha=<span>0.2</span>,level=<span>0.9</span>,type=<span>"t"</span>,linetype = <span>2</span>,show.legend = <span>F</span>)+coord_fixed()<br>print(p) <br>theme= theme(panel.grid =element_blank()) +   <span>## 删去网格</span><br>  theme(panel.border = element_blank(),panel.background = element_blank()) +   <span>## 删去外层边框</span><br>  theme(axis.line = element_line(size=<span>1</span>, colour = <span>"black"</span>)) <br>p=p+theme+guides(colour = guide_legend(override.aes = list(size=<span>5</span>)))<br>print(p)<br>ggplot2::ggsave(filename = paste0(pro,<span>'_tsne.pdf'</span>))<br><br>sce &lt;- RunUMAP(object = sce, dims = <span>1</span>:<span>15</span>, do.fast = <span>TRUE</span>)<br>DimPlot(sce,reduction = <span>"umap"</span>,label=<span>T</span>)<br>ggplot2::ggsave(filename = paste0(pro,<span>'_umap.pdf'</span>)) <br> <br>sce.markers &lt;- FindAllMarkers(object = sce, only.pos = <span>TRUE</span>, min.pct = <span>0.25</span>, <br>                              thresh.use = <span>0.25</span>)<br><br>DT::datatable(sce.markers)<br>write.csv(sce.markers,file=paste0(pro,<span>'_sce.markers.csv'</span>))<br><span>library</span>(dplyr) <br>top10 &lt;- sce.markers %&gt;% group_by(cluster) %&gt;% top_n(<span>10</span>, avg_logFC)<br>DoHeatmap(sce,top10$gene,size=<span>3</span>)<br>ggsave(filename=paste0(pro,<span>'_sce.markers_heatmap.pdf'</span>))<br>sce.first=sce<br>save(sce.first,sce.markers,file = <span>'first_sce.Rdata'</span>)<br><br></code></pre><h2 data-tool="mdnice编辑器"><span></span><span>如果你看不懂代码，请自行回顾：单细胞基础10讲</span></h2><ul data-tool="mdnice编辑器"><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247486076&amp;idx=1&amp;sn=52bb851d7dc23461233a2cf458736151&amp;scene=21#wechat_redirect" data-linktype="2">01. 上游分析流程</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247486082&amp;idx=1&amp;sn=03cadceffb2c14ba95d97fe5caf38d94&amp;scene=21#wechat_redirect" data-linktype="2">02.课题多少个样品，测序数据量如何</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247486088&amp;idx=1&amp;sn=3a115338ee4937d20caab78627237553&amp;scene=21#wechat_redirect" data-linktype="2">03. 过滤不合格细胞和基因（数据质控很重要）</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247486096&amp;idx=1&amp;sn=1a99c4c5800b7e0287db3e8ef369fab8&amp;scene=21#wechat_redirect" data-linktype="2">04. 过滤线粒体核糖体基因</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247486098&amp;idx=1&amp;sn=bf9a71df848d74fe665ce7d5e283d5ff&amp;scene=21#wechat_redirect" data-linktype="2">05. 去除细胞效应和基因效应</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247486260&amp;idx=1&amp;sn=c6abf658de73594d1d77d8e1ffa7d153&amp;scene=21#wechat_redirect" data-linktype="2">06.单细胞转录组数据的降维聚类分群</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247486271&amp;idx=1&amp;sn=638b434b6deee63206af1c0eeda175ab&amp;scene=21#wechat_redirect" data-linktype="2">07.单细胞转录组数据处理之细胞亚群注释</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247486278&amp;idx=1&amp;sn=91250ef733833ff00371818b215dc124&amp;scene=21#wechat_redirect" data-linktype="2">08.把拿到的亚群进行更细致的分群</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247486287&amp;idx=1&amp;sn=49627c638ff9c04418282c53518aa7c7&amp;scene=21#wechat_redirect" data-linktype="2">09.单细胞转录组数据处理之细胞亚群比例比较</a></section></li></ul><p data-tool="mdnice编辑器">更多个性化汇总代码见：单细胞初级8讲和高级分析8讲</p><h3 data-tool="mdnice编辑器"><span></span><span>初级分析</span><span></span></h3><ul data-tool="mdnice编辑器"><li><section><a href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247488315&amp;idx=1&amp;sn=a5cb2f39c34f90d8f1c03e2f021f32de&amp;chksm=ea1f15b9dd689cafa0e70cae1e745113ba722055cef940f200ad111c3ed18227ed7f918a6547&amp;scene=21#wechat_redirect" data-linktype="2">单细胞转录组基础分析一：分析环境搭建</a></section></li><li><section><a href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247488315&amp;idx=2&amp;sn=65bb6c6524b6acf8722f507cc23428d7&amp;chksm=ea1f15b9dd689cafba6e7511d9a4f832d60312e07d11358eae2c273076c796d77d343eaa7fb4&amp;scene=21#wechat_redirect" data-linktype="2">单细胞转录组基础分析二：数据质控与标准化</a></section></li><li><section><a href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247488332&amp;idx=1&amp;sn=b6921478db71d5ae9367370965c45539&amp;chksm=ea1f15cedd689cd84478e45ed87e71bceaedc7d7d57d9a7c4736723e4b13c18cb19086a41a59&amp;scene=21#wechat_redirect" data-linktype="2">单细胞转录组基础分析三：降维与聚类</a></section></li><li><section><a href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247488332&amp;idx=2&amp;sn=a61862ffb546a70eecfe9b03827a27fe&amp;chksm=ea1f15cedd689cd8dd372439af01a198f6204c4e63ff6b870a410f9b50a26ffa8aa4ca1cf626&amp;scene=21#wechat_redirect" data-linktype="2">单细胞转录组基础分析四：细胞类型鉴定</a></section></li><li><section><a href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247488352&amp;idx=1&amp;sn=2e75db93d4fc018f568841ea2d087d5c&amp;chksm=ea1f15e2dd689cf47480646d3f7be489258e8b021d337f22b80f25b7204f3ba08baea57ea672&amp;scene=21#wechat_redirect" data-linktype="2">单细胞转录组基础分析五：细胞再聚类</a></section></li><li><section><a href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247488352&amp;idx=2&amp;sn=c08863668cbde5d66fd5d3e592f3aac9&amp;chksm=ea1f15e2dd689cf4c0fb80add7625bc2c60d074475491e8116131ae7368b52fd5f451597d553&amp;scene=21#wechat_redirect" data-linktype="2">单细胞转录组基础分析六：伪时间分析</a></section></li><li><section><a href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247488367&amp;idx=1&amp;sn=52b69246c2b352abc0343ccc95d76229&amp;chksm=ea1f15eddd689cfb64e6e9201c034de5334f11c3635acddf4b1a547248cc81122dd9dc350306&amp;scene=21#wechat_redirect" data-linktype="2">单细胞转录组基础分析七：差异基因富集分析</a></section></li><li><section><a href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247488367&amp;idx=2&amp;sn=18a60dab3cde76139872026febd7d9a0&amp;chksm=ea1f15eddd689cfbb8c10f4f6e925fa00193235d10a4fcea50e283cae5e796091f9d5948ed13&amp;scene=21#wechat_redirect" data-linktype="2">单细胞转录组基础分析八：可视化工具总结</a></section></li></ul><h3 data-tool="mdnice编辑器"><span></span><span>高级分析</span><span></span></h3><ul data-tool="mdnice编辑器"><li><section><a href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247488375&amp;idx=1&amp;sn=a8c73ea647254baab7125babba027071&amp;chksm=ea1f15f5dd689ce3b7c90dd2aeed140b23b83543e7def6094af98c40de58a2d1b08e15e8d2fe&amp;scene=21#wechat_redirect" data-linktype="2">单细胞转录组高级分析一：多样本合并与批次校正</a></section></li><li><section><a href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247488383&amp;idx=1&amp;sn=7b8504ed4449df3a707d1c83ec0b0a7a&amp;chksm=ea1f15fddd689ceb5edf6635d2c74e9271eac0c30c4d1714403c9057cb3fa187a776e5a4f34b&amp;scene=21#wechat_redirect" data-linktype="2">单细胞转录组高级分析二：转录调控网络分析</a></section></li><li><section><a href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247488392&amp;idx=1&amp;sn=e0aa3d50eb0b1f3251f1ae7cf62c9616&amp;chksm=ea1f150add689c1c0c75f6b1e1e6bf4d3e1faaf230b6b2ef4466d8530f08958bd6196849d61d&amp;scene=21#wechat_redirect" data-linktype="2">单细胞转录组高级分析三：细胞通讯分析</a></section></li><li><section><a href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247488400&amp;idx=1&amp;sn=2cec23311fe972353dec8cbc24c6efbc&amp;chksm=ea1f1512dd689c04ab0e822eabc96158cfd0d437e8cc8721dde77acf5834ccd3d7a26660f8f0&amp;scene=21#wechat_redirect" data-linktype="2">单细胞转录组高级分析四：scRNA数据推断CNV</a></section></li><li><section><a href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247488442&amp;idx=1&amp;sn=cfa26b7e4ee68a6e5a7929a0d5b98595&amp;chksm=ea1f1538dd689c2e2bdfff6bf6956531abd9eee0492efd256f20bcfc877f351646e7e5e74934&amp;scene=21#wechat_redirect" data-linktype="2">单细胞转录组高级分析五：GSEA与GSVA分析</a></section></li><li><section><a href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247488450&amp;idx=1&amp;sn=de7beeb1c144dee1197942cbc2cbe9fc&amp;chksm=ea1f1540dd689c569c6d2a93ec7d4dd707c76a7290f2454ca61c7706799c72199bd5e60d892f&amp;scene=21#wechat_redirect" data-linktype="2">单细胞转录组高级分析六：TCGA生存分析</a></section></li><li><section><a href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247488458&amp;idx=1&amp;sn=890de4c0c4f1e286406560e97c3bf356&amp;chksm=ea1f1548dd689c5e112760efc79562b78fae3e0982f4c9a10fdffb507d35b7de86277f21e4f9&amp;scene=21#wechat_redirect" data-linktype="2">单细胞转录组高级分析七：整合scATAC数据</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247488467&amp;idx=1&amp;sn=96407b7817a64b270752792b5e775d34&amp;scene=21#wechat_redirect" data-linktype="2">单细胞转录组高级分析八：整合V(D)J数据</a></section></li></ul></section><section data-tools="新媒体排版" data-id="3870721" data-style-type="undefined"><p><br><br><br></p><hr><section><p>如果你对单细胞转录组研究感兴趣，但又不知道如何入门，也许你可以关注一下下面的课程<span></span></p><p><br><br></p><ul><li><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247487868&amp;idx=1&amp;sn=c86128e2de43878df1360cd2c43aca52&amp;chksm=ea1f17fedd689ee86385dc8e9db6e4750d19a5351bf3b7c62ce84495dc35093d00b7e33b7a89&amp;scene=21#wechat_redirect" data-itemshowtype="0" tab="innerlink" data-linktype="2" hasload="1"><span>两年过去了，你们的单细胞文章终于发出来了</span></a></p></li><li><p><span><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247488744&amp;idx=2&amp;sn=fb60855ae724f134b8c9dedd7fb59e2f&amp;chksm=ea1f126add689b7c07bc0d6064848f2191c7bb9669b4d000902fe2a8016bc1a11347dcc2c9dd&amp;scene=21#wechat_redirect" data-itemshowtype="11" tab="innerlink" data-linktype="2" hasload="1">生信爆款入门-第9期（线上直播4周，马拉松式陪伴，带你入门）</a>你的生物信息入门课</span><br></p></li><li><p><span><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247488744&amp;idx=1&amp;sn=5089f26b13eee19067944cbfd1fd0d89&amp;chksm=ea1f126add689b7cb9a3e5ffb8cf329fd199d49f29aced097a75fa36f5ee9dad25063f8be9a1&amp;scene=21#wechat_redirect" data-itemshowtype="11" tab="innerlink" data-linktype="2" hasload="1">数据挖掘第7期（线上直播3周，马拉松式陪伴，带你入门）</a> 医学生/医生首选技能提高课</span></p></li></ul><p><img data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz_gif/4TKeL1ZejtlKxOib5kmKX6ic6eX0w0WK5jvhtz9yBRsO3OI4yr6S5iaLNM7AbAeuPDHXMvDdur2DRz9wyiax4lEviag/640?wx_fmt=gif" data-type="gif" data-w="240" src="https://mmbiz.qpic.cn/mmbiz_gif/4TKeL1ZejtlKxOib5kmKX6ic6eX0w0WK5jvhtz9yBRsO3OI4yr6S5iaLNM7AbAeuPDHXMvDdur2DRz9wyiax4lEviag/640?wx_fmt=gif"><br><br><br></p><p><img data-ratio="0.05278592375366569" data-src="https://mmbiz.qpic.cn/mmbiz/4TKeL1Zejtlq03ZOSZiaTlic1MxgdKiaxTbOZ7ZSe0Xx1Ca8xF3L6Nyj1FYUajtYrSmRIHyZVSsAve0EAvEicZONpg/640?wx_fmt=jpeg" data-type="jpeg" data-w="341" src="https://mmbiz.qpic.cn/mmbiz/4TKeL1Zejtlq03ZOSZiaTlic1MxgdKiaxTbOZ7ZSe0Xx1Ca8xF3L6Nyj1FYUajtYrSmRIHyZVSsAve0EAvEicZONpg/640?wx_fmt=jpeg"><br><br></p><p><strong><span>看完记得顺手点个</span></strong><span><strong><span>“在看”</span></strong></span><strong><span>哦！</span></strong></p></section><section><section data-id="93668"><section><section data-width="95%"><section><section><section data-width="38%"><section><section data-id="93668"><section><section data-width="95%"><section><section><section data-width="61.8%"><section><section><section><p><br><br><br></p><span><strong data-burshtype="text"><img data-copyright="0" data-cropselx1="0" data-cropselx2="109" data-cropsely1="0" data-cropsely2="109" data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz/siaia0BDGJdjRMGrkqo64BGKecYk4akuHpGHVQs7FeOpY7eWbIPGC1tRw5Tw0oEPmx053mR9FTVerWvhuZchIpZw/640?wx_fmt=jpeg" data-type="jpeg" data-w="430" title="qrcode_for_gh_5a3c482ed99f_1280.jpg" src="https://mmbiz.qpic.cn/mmbiz/siaia0BDGJdjRMGrkqo64BGKecYk4akuHpGHVQs7FeOpY7eWbIPGC1tRw5Tw0oEPmx053mR9FTVerWvhuZchIpZw/640?wx_fmt=jpeg"><strong data-burshtype="text">生物</strong><strong data-burshtype="text"> | 单细胞 | 转录组丨资料</strong></strong></span></section><section><span><strong data-burshtype="text">每天都精彩</strong></span></section></section></section><section><section><section><section><section><section><p><span>长按扫码可关注</span></p></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/V-t3uzaISVdkDt0P7Uig8w",target="_blank" rel="noopener noreferrer">原文链接</a>
