---
title: "EaSIeR R包预测免疫治疗反应"
date: 2022-12-05T06:41:53Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Acdemic"]
---
EaSIeR R包预测免疫治疗反应 by 生信菜鸟团
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><p data-tool="mdnice编辑器"><span><span lang="zh-CN">今天的分享来自应邀知识整理的【启明非】师兄</span><span lang="en-US">~</span></span></p><p data-tool="mdnice编辑器"><span>E</span><span>aSIeR（ Estimate Systems Immune Response ） ：Interpretable systems biomarkers predict response to immune-checkpoint inhibitors.</span></p><blockquote data-tool="mdnice编辑器"><p>Lapuente-Santana Ó, van Genderen M, Hilbers PAJ, Finotello F, Eduati F. Interpretable systems biomarkers predict response to immune-checkpoint inhibitors. <em>Patterns (N Y)</em>. 2021;2(8):100293. Published 2021 Jun 30. doi:10.1016/j.patter.2021.100293</p></blockquote><p data-tool="mdnice编辑器">EaSIeR包发表在<em>Patterns (N Y)</em>期刊上，最初在Pubmed 检索后Scholarscope对其没有显示影响因子（图1），后试着在letpub上进一步查看（图2），偶然发现这个期刊竟然可能是Cell 的子刊，瞬然对这个包的态度发生了一百八十度大转弯，后续点开该文献的DOI链接，果然呈现的就是CellPress经典的大蓝背景（图3），只能说认识的期刊还是太少了。话不多说，言归正传。</p></section><section><img data-backh="111" data-backw="306" data-ratio="0.3614864864864865" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibW6gLia50EwuHfFL82BVl6ibOdiaCgVp3Uicnne65Z7uD4BvXq6kDqR8uicc313cQaCpBQia7Piad3hY4WQ/640?wx_fmt=png" data-type="png" data-w="888" width="305.763916015625" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibW6gLia50EwuHfFL82BVl6ibOdiaCgVp3Uicnne65Z7uD4BvXq6kDqR8uicc313cQaCpBQia7Piad3hY4WQ/640?wx_fmt=png"><span>图1. EaSIeR检索图</span></section><p><img data-ratio="0.5959147424511545" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibW6gLia50EwuHfFL82BVl6ibgYBnZuqPNZ2cZXhIler72X2OU8rX7BicBUHsLaCKx4JALbGeEkKiadLg/640?wx_fmt=png" data-type="png" data-w="1126" title="图2,. letpub 期刊检索图" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibW6gLia50EwuHfFL82BVl6ibgYBnZuqPNZ2cZXhIler72X2OU8rX7BicBUHsLaCKx4JALbGeEkKiadLg/640?wx_fmt=png"><span>图2. letpub 期刊检索图</span></p><p><img data-ratio="0.4641638225255973" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibW6gLia50EwuHfFL82BVl6ibhrfOgth6yXQWR59ZTeFMiaesuJ0CdicGqTgnR7QD2RV9CcUD8ZrxsPNg/640?wx_fmt=png" data-type="png" data-w="1172" title="图3.Cellpress 论文页首" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibW6gLia50EwuHfFL82BVl6ibhrfOgth6yXQWR59ZTeFMiaesuJ0CdicGqTgnR7QD2RV9CcUD8ZrxsPNg/640?wx_fmt=png"><span>图3.Cellpress 论文页首</span></p><h1 data-tool="mdnice编辑器">1. EaSIeR 开发背景简介</h1><h1 data-tool="mdnice编辑器">恶性肿瘤细胞可通过多种方式干扰肿瘤微环境的免疫细胞以减弱对肿瘤细胞的杀伤作用并诱导免疫细胞免疫耐受。而肿瘤微环境固有的复杂性，多细胞的动态特性，对获取免疫反应生物标志物和预测免疫治疗效果提出了巨大挑战。因此，作者期望使用 bulk RNA-seq数据与不同的先验知识来源（肿瘤浸润细胞，细胞内信号传导，TFs活性,细胞间通讯强弱）相结合，以得出基于系统的肿瘤微环境特征，量化免疫细胞组成以及细胞内和细胞间的通讯。作者通过将多任务学习应用于这些特征，预测免疫反应的不同特征，并基于可解释的Biomarker推导出癌症类型特异性模型。并将该模型应用于来自接受 PD-1/PD-L1 抑制剂治疗的癌症患者的独立 RNA-seq 数据，以证明EaSIeR 的方法可以准确预测治疗结果。后续将对这个R包逐步解析。</h1><figure data-tool="mdnice编辑器"><figcaption><img data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/iaRJcrq2LosibW6gLia50EwuHfFL82BVl6ibDicpB9MvgEEiabEbDibUHfKoaUoUTbbkajfpibI2ibPO3HBO4ekKZQFsJrw/640?wx_fmt=jpeg" data-type="jpeg" data-w="996" src="https://mmbiz.qpic.cn/mmbiz_jpg/iaRJcrq2LosibW6gLia50EwuHfFL82BVl6ibDicpB9MvgEEiabEbDibUHfKoaUoUTbbkajfpibI2ibPO3HBO4ekKZQFsJrw/640?wx_fmt=jpeg"><span>图4. EaSIeR 流程示意</span></figcaption></figure><h1 data-tool="mdnice编辑器">2. EaSIeR 下载、加载</h1><p data-tool="mdnice编辑器">EaSIeR 存储在Biocondoctor， 可以使用BiocManager 下载安装。</p><pre data-tool="mdnice编辑器"><code>BiocManager::install(<span>"easier"</span>)<br><span># 安装依赖项</span><br>BiocManager::install(<span>"easier"</span>, dependencies = <span>TRUE</span>)<br><span># 下载开发版本</span><br>BiocManager::install(<span>"olapuentesantana/easier"</span>)<br><span># 加载R包</span><br><span>library</span>(<span>"easier"</span>)<br></code></pre><p data-tool="mdnice编辑器">可能用到的<strong>数据类型</strong>（如有，可以额外提供患者对免疫疗法的反应）：</p><ul data-tool="mdnice编辑器"><li><p>RNA_counts：data.frame， 包含原始Count值（以 HGNC 基因符号作为行名，样本标识符作为列名）。</p></li><li><section>RNA_tpm：data.frame ，包含 TPM 值（以 HGNC 基因符号作为行名，样本标识符作为列名）。</section></li><li><section>real_patient_response：向量，包含临床患者对免疫疗法的反应的字符（无反应者标记为 NR，反应者标记为 R）。</section></li></ul><h1 data-tool="mdnice编辑器">3. EaSIeR 示例——以膀胱癌患者样本为例 (Mariathasan et al. 2018)</h1><p data-tool="mdnice编辑器">本部分的数据<span>集主要来源于：</span><code><span>Mariathasan et al. “TGF-B attenuates tumour response to P</span></code><code><span>D-L1 blockade by contributing to exclusion of T cells”, published in Nature, 2018</span></code><span>doi:10.1038/nature25501</span><span>的文章，可通过</span><span>IMvigor210CoreBiologies</span><span>包获得。</span></p><p data-tool="mdnice编辑器">在这里，关键数据是RNA-seq 数据（Count和 TPM）、肿瘤突变负荷 (TMB) 信息和 192 名患者对 ICB 治疗的反应结果（完全缓解 (CR) 的 25 名：R；疾病进展 (PD) 的 167 名：NR）。</p><h2 data-tool="mdnice编辑器"><span>3.1  从Mariathasan et al. 2018  获取Demo 数据集</span></h2><pre data-tool="mdnice编辑器"><code>dataset_mariathasan &lt;- get_Mariathasan2018_PDL1_treatment()<br><span>#&gt; snapshotDate(): 2022-10-24</span><br><span>#&gt; Retrieving Mariathasan2018_PDL1_treatment dataset...</span><br><span>#&gt; see ?easierData and browseVignettes('easierData') for documentation</span><br><span>#&gt; loading from cache</span><br><span>##  dataset_mariathasan 是一个S4 向量，形式和singlecell 的S4向量类似;</span><br><span>## 因此，理论上来说，可以使用处理单细胞数据的Code,处理该数据。</span><br>&gt; dataset_mariathasan<br><span>#&gt; class: SummarizedExperiment </span><br><span>#&gt; dim: 31087 192 </span><br><span>#&gt; metadata(1): cancertype</span><br><span>#&gt; assays(2): counts tpm</span><br><span>#&gt; rownames(31087): A1BG NAT2 ... CASP8AP2 SCO2</span><br><span>#&gt; rowData names(0):</span><br><span>#&gt; colnames(192): SAM7f0d9cc7f001 SAM4305ab968b90 ... SAMda4d892fddc8 SAMe3d4266775a9</span><br><span>#&gt; colData names(3): pat_id BOR TMB</span><br><br><span>#获取PD-1治疗反应信息</span><br>patient_ICBresponse &lt;- colData(dataset_mariathasan)[[<span>"BOR"</span>]]<br>names(patient_ICBresponse) &lt;- colData(dataset_mariathasan)[[<span>"pat_id"</span>]]<br><br><span># 获取肿瘤突变负荷信息</span><br>TMB &lt;- colData(dataset_mariathasan)[[<span>"TMB"</span>]]<br>names(TMB) &lt;- colData(dataset_mariathasan)[[<span>"pat_id"</span>]]<br><br><span># 获取肿瘤类型</span><br>cancer_type &lt;- metadata(dataset_mariathasan)[[<span>"cancertype"</span>]]<br><br><span># 获取基因表达矩阵（counts 和tpm）</span><br>RNA_counts &lt;- assays(dataset_mariathasan)[[<span>"counts"</span>]]<br>RNA_tpm &lt;- assays(dataset_mariathasan)[[<span>"tpm"</span>]]<br><br><span># 为节约时间，随机抽样30例，设定随机数，以保证结果可重复</span><br>set.seed(<span>1234</span>)<br>pat_subset &lt;- sample(names(patient_ICBresponse), size = <span>30</span>)<br>patient_ICBresponse &lt;- patient_ICBresponse[pat_subset]<br>TMB &lt;- TMB[pat_subset]<br>RNA_counts &lt;- RNA_counts[, pat_subset]<br>RNA_tpm &lt;- RNA_tpm[, pat_subset]<br><br><span># 考虑到部分基因有多个对应关系，需要进一步处理（保留作者给定的gene symbol）</span><br>genes_info &lt;- easier:::reannotate_genes(cur_genes = rownames(RNA_tpm))<br><br><span>## 去除不支持的基因symbol</span><br>non_na &lt;- !is.na(genes_info$new_names)<br>RNA_tpm &lt;- RNA_tpm[non_na, ]<br>genes_info &lt;- genes_info[non_na, ]<br><br><span>## 去除 entries that are withdrawn</span><br>RNA_tpm &lt;- RNA_tpm[-which(genes_info$new_names == <span>"entry withdrawn"</span>), ]<br>genes_info &lt;- genes_info[-which(genes_info$new_names == <span>"entry withdrawn"</span>), ]<br><br><span>## 找出重复基因</span><br>newnames_dup &lt;- unique(genes_info$new_names[duplicated(genes_info$new_names)])<br>newnames_dup_ind &lt;- do.call(c, lapply(newnames_dup, <span>function</span>(X) which(genes_info$new_names == X)))<br>newnames_dup &lt;- genes_info$new_names[newnames_dup_ind]<br><br><span>## 检索重复基因的数据</span><br>tmp &lt;- RNA_tpm[genes_info$old_names[genes_info$new_names %<span>in</span>% newnames_dup],]<br><br><span>## 删除重复基因的数据</span><br>RNA_tpm &lt;- RNA_tpm[-which(rownames(RNA_tpm) %<span>in</span>% rownames(tmp)),]<br><br><span>## 整合重复基因的数据</span><br>dup_genes &lt;- genes_info$new_names[which(genes_info$new_names %<span>in</span>% newnames_dup)]<br>names(dup_genes) &lt;- rownames(tmp)<br><span>if</span> (anyDuplicated(newnames_dup)){<br>  tmp2 &lt;- stats::aggregate(tmp, by = list(dup_genes), FUN = <span>"mean"</span>)<br>  rownames(tmp2) &lt;- tmp2$Group.1<br>  tmp2$Group.1 &lt;- <span>NULL</span><br>}<br><br><span># 整理归纳到一个表达矩阵</span><br>RNA_tpm &lt;- rbind(RNA_tpm, tmp2)<br></code></pre><p data-tool="mdnice编辑器">以上代码主要就是初步获取Demo 数据集和后续的取样、数据初步处理的过程，不做赘述。</p><h2 data-tool="mdnice编辑器"><span>3.2   计算免疫反应的Hallmarks</span></h2><p data-tool="mdnice编辑器">后续，作者利用TPM数据计算下列免疫特<span>征的评分：</span><code><span>溶细胞活性 (CYT) (Rooney et al. 2015)、Roh 免疫评分 (Roh_IS) (Roh et al. 2017)、趋化因子特征 (chemokines) (Messina et al. . 2012) , Davoli 免疫特征 (Davoli_IS) (Davoli et al. 2017) , IFNy 特征 (IFNy) (McClanahan 2017) , 扩展免疫特征 (Ayers_expIS)  (McClanahan 2017) , T 细胞炎症特征 (Tcell_inflamed) (McClanahan 2017 ) )、免疫抵抗程序（RIR：resF_down、resF_up、resF）(Jerby-Arnon et al. 2018)和三级淋巴结构特征 (TLS) (Cabrita et al. 2020)</span></code><span>。也可以通过编辑</span><code><span>selected_scores</span></code><span>选项来自定义此选择。 </span></p><p data-tool="mdnice编辑器"><span>这里作者归纳的免疫特征评分也是相当齐全了，基本上囊括了大部分研究者关注的表型，R包原文献的附件里面列举了各个特征评分具体包括哪些基因类型，感兴趣的可以看一看。也可以去阅读各类特征对应的原文献，理解为什么这些基因代表这个特征。这里的打分算法，根据作者的描述，是根据不同文章的原始Code的算法计算的，在原始文献中的附件有具体的描述。在作者的</span><code><span>compute_scores_immune_response</span></code><span>函数 源码中 是以elseif实现的。</span></p><p data-tool="mdnice编辑器"><span>作者介绍原文(DeepL翻译)：</span></p><p data-tool="mdnice编辑器"><span>Methods :Transcriptomics-based scores of immune response免疫细胞溶解活性代表了两种细胞溶解效应物的水平，即颗粒</span>酶A和穿孔蛋白，它们在CD8+T细胞激活时被过度表达。Ock免疫特征是基于105个与MAGE-A3抗原免疫治疗反应有关的基因的表达。</p><p data-tool="mdnice编辑器">Immunophenoscore是根据与MHC分子、免疫调节剂、效应细胞和抑制细胞有关的基因来计算的。IMPRES是通过免疫检查点基因对的表达的逻辑比较得到的。Roh免疫评分（Roh_IS）是由一组参与与肿瘤排斥有关的免疫激活的基因定义的。趋化因子签名（Chemokine）是基于一组与炎症和免疫有关的基因，它能够预测宿主的免疫反应和肿瘤局部淋巴结构的形成。Davoli免疫特征（Davoli_IS）来自于细胞毒性CD8+T细胞和NK细胞标记物的表达。IFN-γ签名包括能够区分黑色素瘤中的应答者和非应答者的基因。免疫扩展签名（Ayes_expIS）34是通过搜索与IFN-γ签名基因高度相关的基因产生的；这个新的集合包括所有免疫相关的基因。T细胞炎症微环境签名（T cell_inflamed）是基于IFN-γ和T细胞相关炎症基因在预测对PD-1阻断反应方面的联合潜力。TIDE是在免疫逃逸特征的基础上开发的，如T细胞功能紊乱或排斥。MSI状态是通过MSI相关基因对的逻辑比较确定的。TLS特征是由有TLS的肿瘤中差异表达的基因得出。压抑的免疫抵抗是通过结合一套与T细胞排斥、治疗后和功能抵抗相关的基因签名来定义的。</p><pre data-tool="mdnice编辑器"><code>hallmarks_of_immune_response &lt;- c(<span>"CYT"</span>, <span>"Roh_IS"</span>, <span>"chemokines"</span>, <span>"Davoli_IS"</span>, <span>"IFNy"</span>, <span>"Ayers_expIS"</span>, <span>"Tcell_inflamed"</span>, <span>"RIR"</span>, <span>"TLS"</span>)<br>immune_response_scores &lt;- compute_scores_immune_response(RNA_tpm = RNA_tpm, <br>                                                         selected_scores = hallmarks_of_immune_response)<br><span>#&gt; Following scores can be computed: </span><br><span>#&gt; CYT, TLS, IFNy, Ayers_expIS, Tcell_inflamed, Roh_IS, Davoli_IS, chemokines, RIR</span><br><span>#&gt; CYT score:</span><br><span>#&gt; TLS score:</span><br><span>#&gt; IFNy score:</span><br><span>#&gt; Ayers_expIS score:</span><br><span>#&gt; Tcell_inflamed score:</span><br><span>#&gt; Roh_IS score:</span><br><span>#&gt; Davoli_IS score:</span><br><span>#&gt; chemokines score:</span><br><span>#&gt; RIR score:</span><br><span>#&gt; Differenty named or missing signature genes : </span><br><span>#&gt; XAGE1D</span><br><span>#&gt; C17orf76-AS1</span><br><span>#&gt; MKI67IP</span><br><span>#&gt; RPS17L</span><br><span>#&gt; CTSL1</span><br><span>#&gt; LOC100126784</span><br><span>#&gt; LOC100506190</span><br><span>#&gt; NSG1</span><br><span>#&gt; C20orf112</span><br><span>#&gt; FLJ43663</span><br><span>#&gt; TMEM66</span><br><span>#&gt; FLJ39051</span><br><span>#&gt; LOC100506714</span><br><span>#&gt; LOC100507463</span><br><span>#&gt; Perfomed gene re-annotation to match signature genes</span><br>head(immune_response_scores) <span># 行为样本，列为各评分</span><br><span>#&gt;                       CYT        TLS     IFNy Ayers_expIS Tcell_inflamed</span><br><span>#&gt; SAM23095936e611  1.854240   4.167253 3.419660    3.171988     -3.5591198</span><br><span>#&gt; SAM3330c03fdf00 31.782584 265.759171 6.738235    6.979761     -0.2750762</span><br><span>#&gt; SAM8e469834acc1  6.399774  12.206693 4.748116    4.238499     -2.5181318</span><br><span>#&gt; SAMeb29625f76a5  6.977461   8.648444 5.037934    4.113286     -2.6028720</span><br><span>#&gt; SAMd86389d0d768  1.853023   6.504487 3.441488    2.845997     -3.5550647</span><br><span>#&gt; SAMa321770ac31c  7.518455   4.171804 5.708758    4.386140     -2.2920024</span><br><span>#&gt;                    Roh_IS Davoli_IS chemokines    resF_up  resF_down       resF</span><br><span>#&gt; SAM23095936e611  6.186174 0.1822660 -3.7334420  0.7493652 -1.1423416  1.8917067</span><br><span>#&gt; SAM3330c03fdf00 54.319422 0.9359606  3.6555005 -0.1591575  0.4031406 -0.5622982</span><br><span>#&gt; SAM8e469834acc1 12.387979 0.4236453 -1.0975826 -0.5361122 -0.1691564 -0.3669558</span><br><span>#&gt; SAMeb29625f76a5 14.933200 0.4827586 -1.8486582  0.0781382 -0.5099630  0.5881012</span><br><span>#&gt; SAMd86389d0d768  5.534648 0.1083744 -4.1499736  0.4914592 -0.4957330  0.9871922</span><br><span>#&gt; SAMa321770ac31c 15.062774 0.4827586 -0.4492079  0.4909176 -0.1353869  0.6263045</span><br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.5546697038724373" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibW6gLia50EwuHfFL82BVl6ib6fNxJ2tNibQ4hqO4sSianzllc8r5ibKrtq6mxppHc3dhZ40wlGOiaFQIww/640?wx_fmt=png" data-type="png" data-w="878" title="图5. pheatmap 简单查看免疫评分结果" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibW6gLia50EwuHfFL82BVl6ib6fNxJ2tNibQ4hqO4sSianzllc8r5ibKrtq6mxppHc3dhZ40wlGOiaFQIww/640?wx_fmt=png"><figcaption><span>图5. pheatmap 简单查看免疫评分结果</span></figcaption></figure><h2 data-tool="mdnice编辑器"><span>3.3 对TME的特征进行定量</span></h2><pre data-tool="mdnice编辑器"><p>cell_fractions &lt;- compute_cell_fractions(RNA_tpm = RNA_tpm)<br><br><span>#&gt; </span><br><span>#&gt; Running quanTIseq deconvolution module</span><br><span>#&gt; Gene expression normalization and re-annotation (arrays: FALSE)</span><br><span>#&gt; Removing 17 noisy genes</span><br><span>#&gt; Removing 15 genes with high expression in tumors</span><br><span>#&gt; Signature genes found in data set: 137/138 (99.28%)</span><br><span>#&gt; Mixture deconvolution (method: lsei)</span><br><span>#&gt; Deconvolution successful!</span><br><span>#&gt; Cell fractions computed!</span></p></pre><p><span>quanTIseq</span><span>包是这个部分的主要函数，主要也是通过反卷积算法量化Bulk RNAseq数据中的10种免疫细胞的比例</span><span>（</span><code>B cells；Classically-activated (M1) macrophages；Alternatively-activated (M2) macrophages；Monocytes；Neutrophils；Natural killer (NK) cells；Non-regulatory (helper) CD4+ T cells；Cytotoxic CD8+ T cells；Regulatory CD4+ T (Treg) cells；Myeloid dendritic cells；Other uncharacterized cells</code><span>)。</span><span></span></p><p><span>后续通路算法通过应</span><span>用</span><span>PROGENy</span><span>（Holland、</span><span>Szalai 和 Saez-Rodriguez 2020；Schubert 等人 2018）方法对来自 RNA-seq 的数据进行计数，可以推断出 14 条信号通路的活性，如下图所示。PROGENy推断通路活性的应用已扩展至单细胞数据集，在Github中作者给出了使用方法，感兴趣可以学习。</span></p><pre data-tool="mdnice编辑器"><code><span># 考虑到部分通路相关基因可能被纳入计算免疫反应评分，</span><br><span># 因此参数remove_sig_genes_immune_response  设置为True， 去除这部分重复基因进行计算，</span><br><span># 初次使用可以尝试使用效果。</span><br>pathway_activities &lt;- compute_pathway_activity(RNA_counts = RNA_counts,<br>                                               remove_sig_genes_immune_response = <span>TRUE</span>)<br><span>#&gt; Removing signature genes of hallmarks of immune response</span><br><span>#&gt; Gene counts normalization with DESeq2:</span><br><span>#&gt; gene-wise dispersion estimates</span><br><span>#&gt; mean-dispersion relationship</span><br><span>#&gt; final dispersion estimates</span><br><span>#&gt; Number of genes used per pathway to compute progeny scores:</span><br><span>#&gt; Androgen: 95 (95%)</span><br><span>#&gt; EGFR: 94 (94%)</span><br><span>#&gt; Estrogen: 97 (97%)</span><br><span>#&gt; Hypoxia: 97 (97%)</span><br><span>#&gt; JAK-STAT: 83 (83%)</span><br><span>#&gt; MAPK: 92 (92%)</span><br><span>#&gt; NFkB: 88 (88%)</span><br><span>#&gt; p53: 95 (95%)</span><br><span>#&gt; PI3K: 89 (89%)</span><br><span>#&gt; TGFb: 96 (96%)</span><br><span>#&gt; TNFa: 90 (90%)</span><br><span>#&gt; Trail: 93 (93%)</span><br><span>#&gt; VEGF: 89 (89%)</span><br><span>#&gt; WNT: 97 (97%)</span><br><span>#&gt; Pathway signature genes found in data set: 1207/1301 (92.8%)</span><br><span>#&gt; </span><br><span>#&gt; Pathway activity scores computed!</span><br>head(pathway_activities)<br><span>#&gt;                 Androgen      EGFR Estrogen  Hypoxia JAK-STAT     MAPK     NFkB</span><br><span>#&gt; SAM23095936e611 3413.109 -138.9127 2382.863 7298.016 5687.325 1277.133 5394.316</span><br><span>#&gt; SAM3330c03fdf00 3491.213 -241.4348 2344.289 6411.686 7166.311 1182.673 5928.623</span><br><span>#&gt; SAM8e469834acc1 3489.019 -193.1898 2438.082 7016.232 6369.837 1160.688 5637.852</span><br><span>#&gt; SAMeb29625f76a5 3432.605 -183.9006 2466.230 7419.429 6175.766 1237.699 5419.877</span><br><span>#&gt; SAMd86389d0d768 3542.806 -265.9723 2259.767 7225.572 5848.535 1041.126 5113.226</span><br><span>#&gt; SAMa321770ac31c 3616.811 -159.3274 2326.090 7141.462 6631.886 1185.325 5724.962</span><br><span>#&gt;                      p53      PI3K     TGFb     TNFa     Trail      VEGF</span><br><span>#&gt; SAM23095936e611 4844.378 -2930.195 2842.307 5874.151  87.59961 -310.9487</span><br><span>#&gt; SAM3330c03fdf00 4348.616 -3011.188 2697.512 5998.852 178.44521 -523.8521</span><br><span>#&gt; SAM8e469834acc1 4839.137 -2969.504 3139.592 6004.593 121.14881 -359.5669</span><br><span>#&gt; SAMeb29625f76a5 4729.233 -2974.656 2868.690 5904.540  97.02514 -414.0791</span><br><span>#&gt; SAMd86389d0d768 5144.473 -2957.193 2717.031 5324.415  88.08302 -317.4356</span><br><span>#&gt; SAMa321770ac31c 4662.701 -2950.958 2735.689 6044.925 101.74421 -375.2041</span><br><span>#&gt;                      WNT</span><br><span>#&gt; SAM23095936e611 705.8786</span><br><span>#&gt; SAM3330c03fdf00 690.0206</span><br><span>#&gt; SAM8e469834acc1 697.3891</span><br><span>#&gt; SAMeb29625f76a5 682.4761</span><br><span>#&gt; SAMd86389d0d768 696.3715</span><br><span>#&gt; SAMa321770ac31c 636.9830</span></code></pre><p><code>pathway_activities</code><span> </span> 是一个<code>data.frame</code><span> </span> ，包含每个样本的14条通路的评分。</p><p>后续再使用DoRothEA进行转录因子分析，输入参数是TPM ,转录因子总共包含118个。</p><pre data-tool="mdnice编辑器"><code>tf_activities &lt;- compute_TF_activity(RNA_tpm = RNA_tpm)<br> <br><span>#&gt; Regulated transcripts found in data set: 3084/3254 (94.8%)</span><br><span>#&gt; TF activity computed!</span><br>head(tf_activities[,<span>1</span>:<span>5</span>])<br>                        AR       ARNTL      ATF1      ATF2      ATF4<br><span>#&gt; SAM23095936e611  1.3508113 -0.96612301 0.3565387 1.0213505 -1.591800</span><br><span>#&gt; SAM3330c03fdf00 -1.1199881 -0.89506361 0.3164353 0.3319397 -1.496857</span><br><span>#&gt; SAM8e469834acc1  0.3899521  0.04868563 0.0157571 0.9813438 -1.491290</span><br><span>#&gt; SAMeb29625f76a5  1.8058972 -0.48482979 0.5556480 1.1762553 -1.125674</span><br><span>#&gt; SAMd86389d0d768  1.3191714 -0.56352878 0.1340379 0.3679560 -1.419815</span><br><span>#&gt; SAMa321770ac31c  0.4756138 -0.33771612 0.2982223 0.1423290 -1.285246</span><br></code></pre><p data-tool="mdnice编辑器">使用源于肿瘤特异性细胞间通讯网络，在接下来的算法中量化了867个配受体对的关联强度。作者在原文献中仔细描述了这个过程。核心的代码见下（这里涉及复杂的数学逻辑，没看懂代码）。这个部分参考数据是 存储在easierData的intercell_networks和group_lrpairs。intercell_networks里面包含约18种癌症类型的细胞间网络数据和group_lrpairs含56个类型含多个配受体对的list。原始数据来源在作者的Methods中进行了具体描述。</p><pre data-tool="mdnice编辑器"><code><span>#&gt; lrpair_weights 是个data.frame</span><br>lrpair_weights &lt;- compute_LR_pairs(RNA_tpm = RNA_tpm,<br>                                  cancer_type = <span>"pancan"</span>)<br><span>#&gt; LR signature genes found in data set: 629/644 (97.7%)</span><br><span>#&gt; Ligand-Receptor pair weights computed </span><br>head(lrpair_weights)[,<span>1</span>:<span>5</span>]<br><span>#&gt;                 A2M_APP_CALR_LRPAP1_PSAP_SERPING1_LRP1 ADAM10_AXL</span><br><span>#&gt; SAM23095936e611                               7.746610   4.402992</span><br><span>#&gt; SAM3330c03fdf00                               6.153378   4.377154</span><br><span>#&gt; SAM8e469834acc1                               7.993705   5.621298</span><br><span>#&gt; SAMeb29625f76a5                               7.134311   4.299523</span><br><span>#&gt; SAMd86389d0d768                               6.579407   3.260435</span><br><span>#&gt; SAMa321770ac31c                               6.740991   3.213153</span><br><span>#&gt;                 ADAM10_EFNA1_EPHA3 ADAM12_ITGA9 ADAM12_ITGB1_SDC4</span><br><span>#&gt; SAM23095936e611           2.612012     3.229963          3.229963</span><br><span>#&gt; SAM3330c03fdf00           4.385768     3.990142          3.990142</span><br><span>#&gt; SAM8e469834acc1           4.721780     4.323715          4.323715</span><br><span>#&gt; SAMeb29625f76a5           2.081069     2.946964          2.946964</span><br><span>#&gt; SAMd86389d0d768           2.252113     2.605315          2.605315</span><br><span>#&gt; SAMa321770ac31c           2.108481     3.442606          4.327418</span><br><span>#节选compute_LR_pairs 源  CODE</span><br>  <span>if</span> (is.null(RNA_tpm)) <br>    <span>stop</span>(<span>"TPM gene expression data not found"</span>)<br>  intercell_networks &lt;- suppressMessages(easierData::get_intercell_networks())<br>  group_lrpairs &lt;- suppressMessages(easierData::get_group_lrpairs())<br><span>#&gt;……#&gt;  </span><br><span>if</span> (any(grep(<span>"ENSG00000"</span>, genes))) {<br>    <span>stop</span>(<span>"Hgnc gene symbols are required"</span>, call. = <span>FALSE</span>)<br>  }<br>  gene_expr &lt;- as.data.frame(gene_expr)<br>  intercell_network &lt;- intercell_networks[[cancer_type]]<br>  LR_pairs &lt;- unique(paste0(intercell_network$ligands, <span>"_"</span>, <br>                            intercell_network$receptors))<br>  all_lrpairs_genes &lt;- unique(c(intercell_network$ligands, <br>                                intercell_network$receptors))<br>  genes_kept &lt;- intersect(rownames(gene_expr), all_lrpairs_genes)<br>  genes_left &lt;- setdiff(all_lrpairs_genes, rownames(gene_expr))<br>  message(<span>"LR signature genes found in data set: "</span>, length(genes_kept), <br>          <span>"/"</span>, length(all_lrpairs_genes), <span>" ("</span>, round(length(genes_kept)/length(all_lrpairs_genes), <br>                                                      <span>3</span>) * <span>100</span>, <span>"%)"</span>)<br>  LR_pairs_computed &lt;- do.call(rbind, lapply(seq_len(length(LR_pairs)), <br>                                             <span>function</span>(x) {<br>                                               ligand &lt;- vapply(strsplit(LR_pairs[x], split = <span>"_"</span>, <br>                                                                         fixed = <span>TRUE</span>), utils::head, <span>1</span>, FUN.VALUE = character(<span>1</span>))<br>                                               receptor &lt;- vapply(strsplit(LR_pairs[x], split = <span>"_"</span>, <br>                                                                           fixed = <span>TRUE</span>), utils::tail, <span>1</span>, FUN.VALUE = character(<span>1</span>))<br>                                               pos_lr &lt;- match(c(ligand, receptor), rownames(gene_expr))<br>                                               by_patient &lt;- t(as.data.frame(apply(gene_expr[pos_lr, <br>                                               ], <span>2</span>, min)))<br>                                               rownames(by_patient) &lt;- LR_pairs[x]<br>                                               <span>return</span>(by_patient)<br>                                             }))<br>  LR_pairs_computed &lt;- t(LR_pairs_computed)<br>  <span>for</span> (X <span>in</span> seq_len(length(group_lrpairs))) {<br>    keep &lt;- unique(group_lrpairs[[X]]$main)<br>    remove &lt;- unique(group_lrpairs[[X]]$involved_pairs)<br>    combo_name &lt;- unique(group_lrpairs[[X]]$combo_name)<br>    pos_remove &lt;- match(remove, colnames(LR_pairs_computed))<br>    pos_keep &lt;- match(keep, colnames(LR_pairs_computed))<br>    colnames(LR_pairs_computed)[pos_keep] &lt;- combo_name<br>    LR_pairs_computed &lt;- LR_pairs_computed[, -pos_remove]<br>  }<br>  <span>if</span> (verbose) <br>    message(<span>"Ligand-Receptor pair weights computed \n"</span>)<br>  <span>return</span>(as.data.frame(LR_pairs_computed))<br></code></pre><p data-tool="mdnice编辑器">后续则根据上一步的配受体对分析的结果，量化细胞间的关联强度（这里涉及复杂的数学逻辑，看不懂代码）。</p><pre data-tool="mdnice编辑器"><code> ccpair_scores &lt;- compute_CC_pairs(lrpairs = lrpair_weights, <br>                                   cancer_type = <span>"pancan"</span>)<br>head(ccpair_scores) <br><span>#&gt;                Adipocytes_Adipocytes Adipocytes_DendriticCells Adipocytes_Endothelialcells Adipocytes_Fibroblasts Adipocytes_Macrophages</span><br><span>#&gt; SAM23095936e611              8.509797                  7.363179                    8.426370               8.336850               7.917801</span><br><span>#&gt; SAM3330c03fdf00              8.766845                  8.314185                    8.424617               8.364922               8.261575</span><br><span>#&gt; SAM8e469834acc1              9.070111                  7.865462                    9.115597               8.747848               8.357799</span><br><span>#&gt; SAMeb29625f76a5              8.589783                  7.867214                    8.588126               8.396068               8.057757</span><br><span>#&gt; SAMd86389d0d768              7.997064                  7.161020                    7.936900               7.705764               7.155412</span><br></code></pre><p data-tool="mdnice编辑器">到这里，TME特征基本<span>上分析结束， </span><code><span>cancer_type</span></code><span>的设置取决于</span><code><span>intercell_networks</span></code><span>中有没有这一肿瘤类型。下面是可能</span><code><span>cancer_type</span></code><span>在这个部分可以设定的参数，具体每一种类型需要各自领域的同学去尝试。</span></p><pre data-tool="mdnice编辑器"><code>names(intercell_networks)<br> [<span>1</span>] <span>"BLCA"</span>   <span>"BRCA"</span>   <span>"CESC"</span>   <span>"CRC"</span>    <span>"GBM"</span>    <span>"HNSC"</span>   <span>"KIRC"</span>   <span>"LIHC"</span>   <span>"LUAD"</span>   <span>"LUSC"</span>   <span>"OV"</span>     <span>"PAAD"</span>   <span>"pancan"</span> <span>"PRAD"</span>   <span>"SKCM"</span>  <br>[<span>16</span>] <span>"STAD"</span>   <span>"THCA"</span>   <span>"UCEC"</span> <br></code></pre><h2 data-tool="mdnice编辑器"><span>3.4 预测患者的免疫反应</span></h2><p data-tool="mdnice编辑器">在前述的定量分析的基<span>础上,</span><code><span>predict_immune_response</span></code><span>函数根据前</span>期的计算值，初步计算出预测值。</p><pre data-tool="mdnice编辑器"><code><span># 这个部分 cancer_type需要特别指定，需要注意的是，前面cancer_type 默认值都是 'pancan'  == ‘泛癌’</span><br><span># 但在这个部分 'pancan' 却不能作为其中一个参数，而需要特别指定'BRCA'（膀胱癌样本）</span><br>predictions &lt;- predict_immune_response(pathways = pathway_activities,<br>                                       immunecells = cell_fractions,<br>                                       tfs = tf_activities,<br>                                       lrpairs = lrpair_weights,<br>                                       ccpairs = ccpair_scores,<br>                                       cancer_type = cancer_type, <br>                                       verbose = <span>TRUE</span>)<br>predict_immune_response<br><span>#&gt;function (pathways = NULL, immunecells = NULL, tfs = NULL, lrpairs = NULL, </span><br><span>#&gt;    ccpairs = NULL, cancer_type, verbose = TRUE) </span><br><span>#&gt; {……</span><br>  ……<br>available_cancer_types &lt;- c(<span>"BLCA"</span>, <span>"BRCA"</span>, <span>"CESC"</span>, <span>"CRC"</span>, <br>        <span>"GBM"</span>, <span>"HNSC"</span>, <span>"KIRC"</span>, <span>"KIRP"</span>, <span>"LIHC"</span>, <span>"LUAD"</span>, <span>"LUSC"</span>, <br>        <span>"NSCLC"</span>, <span>"OV"</span>, <span>"PAAD"</span>, <span>"PRAD"</span>, <span>"SKCM"</span>, <span>"STAD"</span>, <span>"THCA"</span>, <br>        <span>"UCEC"</span>)<br>……<br></code></pre><p data-tool="mdnice编辑器">具体如何评估<span>这个</span><code><span>predictions</span></code><span> 的准确性，作者在后续的说明文档中进一步给出评估：如果已知病人的免疫治疗效果，那么可以判断</span><code><span>predictions</span></code><span> 的准确性，而如果未知免疫治疗效果，则预测的结果无从判断。</span></p><h2 data-tool="mdnice编辑器"><span>3.5 使用有免疫治疗反应数据评估EaSIeR的预测</span></h2><p data-tool="mdnice编辑器"><code><span>assess_immune_response</span></code><span> 这部分的函数 主要就是ROC曲线评估免疫治疗反应，这部分的</span><code><span>Gold standard</span></code><span> 使用的是</span><code><span>compute_scores_immune_response</span></code><span>函数计算的结果。TMB 状态和免疫治疗反应关联强度还是要高于其他评分的，部分量化指标比如TFs和LRpairs的AUC都&gt;0.7 ，总体上还是不错的。</span></p><pre data-tool="mdnice编辑器"><code>output_eval_with_resp &lt;- assess_immune_response(predictions_immune_response = predictions,<br>                                                patient_response = patient_ICBresponse,<br>                                                RNA_tpm = RNA_tpm,<br>                                                TMB_values = TMB,<br>                                                easier_with_TMB = <span>"weighted_average"</span>,<br>                                                weight_penalty = <span>0.5</span>)<br><span>#&gt;  |===================================================================================================================================| 100%</span><br><span>#&gt; Following scores can be computed: </span><br><span>#&gt; CYT, TLS, IFNy, Ayers_expIS, Tcell_inflamed, Roh_IS, Davoli_IS, chemokines, RIR</span><br><span>#&gt; CYT score: </span><br><span>#&gt; TLS score: </span><br><span>#&gt; IFNy score: </span><br><span>#&gt; Ayers_expIS score: </span><br><span>#&gt; Tcell_inflamed score: </span><br><span>#&gt; Roh_IS score: </span><br><span>#&gt; Davoli_IS score: </span><br><span>#&gt; chemokines score: </span><br><span>#&gt; RIR score: </span><br><span>#&gt; Differenty named or missing signature genes : </span><br><span>#&gt; XAGE1D</span><br><span>#&gt; C17orf76-AS1</span><br><span>#&gt; MKI67IP</span><br><span>#&gt; RPS17L</span><br><span>#&gt; TSL1</span><br><span>#&gt; LOC100126784</span><br><span>#&gt; LOC100506190</span><br><span>#&gt; NSG1</span><br><span>#&gt; C20orf112</span><br><span>#&gt; FLJ43663</span><br><span>#&gt; TMEM66</span><br><span>#&gt; FLJ39051</span><br><span>#&gt; LOC100506714</span><br><span>#&gt; LOC100507463</span><br><span>#&gt;   |===================================================================================================================================| 100%</span><br><span>#&gt; Perfomed gene re-annotation to match signature genes  </span><br><span>#&gt; Scores of immune response computed!</span><br><span>#&gt; Warning message:</span><br><span>#&gt; In assess_immune_response(predictions_immune_response = predictions,  :</span><br><span>#&gt;   TMB data contains NA values, excluding those patients..</span><br><span># 可以通过下列方式调出图片并保存</span><br><span># inspect output</span><br>output_eval_with_resp[[<span>1</span>]]<br>output_eval_with_resp[[<span>2</span>]]<br>output_eval_with_resp[[<span>3</span>]]<br></code></pre><p data-tool="mdnice编辑器"><img data-ratio="0.5956719817767654" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibW6gLia50EwuHfFL82BVl6ib6S9NpFuZnOekZYUibia8ibTolsRLfpsbRcvmWp9uiaMzQe9dJwnnWrReRA/640?wx_fmt=png" data-type="png" data-w="878" title="图6. AUC值对比" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibW6gLia50EwuHfFL82BVl6ib6S9NpFuZnOekZYUibia8ibTolsRLfpsbRcvmWp9uiaMzQe9dJwnnWrReRA/640?wx_fmt=png"><span>图6. AUC值对比</span><img data-ratio="0.5956719817767654" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibW6gLia50EwuHfFL82BVl6ib07X9qP1aRXpGibKGRXzdvyHIx2iaeJFn5BDU7MrjmwPHGDkfu9JhdDhw/640?wx_fmt=png" data-type="png" data-w="878" title="图7. ROC曲线对比" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibW6gLia50EwuHfFL82BVl6ib07X9qP1aRXpGibKGRXzdvyHIx2iaeJFn5BDU7MrjmwPHGDkfu9JhdDhw/640?wx_fmt=png"><span>图7. ROC曲线对比</span><img data-ratio="0.5956719817767654" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibW6gLia50EwuHfFL82BVl6ibIgLXtlNJ6TA1VJeFwRmSUiaaH4qvPTOH1YxjqLeEmRlE06moGYkUw4w/640?wx_fmt=png" data-type="png" data-w="878" title="图8. 曲线下面积值" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibW6gLia50EwuHfFL82BVl6ibIgLXtlNJ6TA1VJeFwRmSUiaaH4qvPTOH1YxjqLeEmRlE06moGYkUw4w/640?wx_fmt=png"><span>图8. 曲线下面积值</span></p><h2 data-tool="mdnice编辑器"><span>3.6 无免疫治疗数据评估免疫反应</span></h2><p data-tool="mdnice编辑器">在无免疫治疗数据情况下，可以省略函数 assess_immune_response 中的参数 patient_response 来为每个患者分配免疫反应可能性的分数。</p><pre data-tool="mdnice编辑器"><code>output_eval_no_resp &lt;- assess_immune_response(predictions_immune_response = predictions,<br>                                              TMB_values = TMB,<br>                                              easier_with_TMB = <span>"weighted_average"</span>,<br>                                              weight_penalty = <span>0.5</span>)<br><span>#&gt; Scoring patients' as real response is not provided</span><br><br><span>#&gt; Using patient as id variables</span><br><span>#&gt; Warning messages:</span><br><span>#&gt; 1: In assess_immune_response(predictions_immune_response = predictions,  :</span><br><span>#&gt;   Gene expression (TPM) data is missing, skipping computation of scores of immune response..</span><br><span>#&gt; 2: In assess_immune_response(predictions_immune_response = predictions,  :</span><br><span>#&gt;   TMB data contains NA values, excluding those patients..</span><br><span># 图片输出</span><br>output_eval_no_resp[[<span>1</span>]]<br>output_eval_no_resp[[<span>2</span>]]<br></code></pre><p><img data-ratio="0.5956719817767654" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibW6gLia50EwuHfFL82BVl6ibNjpkjA7smpsw6Uia1f4SNbwkgk5t3AOupqbZGX4YXLoWJ4iaTOaQEAbQ/640?wx_fmt=png" data-type="png" data-w="878" title="图9. 预测评分在10次任务中的得分分布" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibW6gLia50EwuHfFL82BVl6ibNjpkjA7smpsw6Uia1f4SNbwkgk5t3AOupqbZGX4YXLoWJ4iaTOaQEAbQ/640?wx_fmt=png"><span>图9. 预测评分在10次任务中的得分分布</span></p><p><img data-ratio="0.5956719817767654" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibW6gLia50EwuHfFL82BVl6ibibpcKsqhRSmia5rC72ZBCrbHjj3icSoc1hoHNAb4ZwTvJ9ReQjfCGlj4w/640?wx_fmt=png" data-type="png" data-w="878" title="图10. 预测评分与肿瘤突变负荷相结合时患者预测的散点图在 10 次任务中的分布" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibW6gLia50EwuHfFL82BVl6ibibpcKsqhRSmia5rC72ZBCrbHjj3icSoc1hoHNAb4ZwTvJ9ReQjfCGlj4w/640?wx_fmt=png"><span>图10. 预测评分与肿瘤突变负荷相结合时患者预测的散点图在 10 次任务中的分布</span></p><p><ne-h3 data-lake-id="Ox7wj"><ne-heading-ext></ne-heading-ext></ne-h3></p><h3 data-tool="mdnice编辑器">3.6.1 获得免疫反应评分</h3><p data-tool="mdnice编辑器">也可<span>以通过</span><code><span>retrieve_easier_score</span></code><span>函数 综合</span><code><span>predictions</span></code><span>和 TMB的数据进行整合，获得评分。</span></p><pre data-tool="mdnice编辑器"><code>easier_derived_scores &lt;- retrieve_easier_score(predictions_immune_response = predictions,<br>                                               TMB_values = TMB,<br>                                               easier_with_TMB = c(<span>"weighted_average"</span>, <br>                                                                   <span>"penalized_score"</span>),<br>                                               weight_penalty = <span>0.5</span>)<br><span>#&gt; head(asier_derived_scores)</span><br><span>#&gt;                 easier_score w_avg_score  pen_score</span><br><span>#&gt; SAM23095936e611 -0.516402256  0.01776351 -1.0164023</span><br><span>#&gt; SAM3330c03fdf00  1.643380092  0.25000000  1.1433801</span><br><span>#&gt; SAM8e469834acc1 -0.146726044  0.18251396 -0.1467260</span><br><span>#&gt; SAMeb29625f76a5 -0.249175716  0.04649778 -0.7491757</span><br><span>#&gt; SAMa321770ac31c -0.153691602  0.30676497  0.3463084</span><br><br></code></pre><h2 data-tool="mdnice编辑器"><span>3.7 通过Biomarker解释对免疫疗法的反应</span></h2><p data-tool="mdnice编辑器">这部分的函数要求提供对免疫治疗反应的结局的数据。</p><pre data-tool="mdnice编辑器"><code>output_biomarkers &lt;- explore_biomarkers(pathways = pathway_activities,<br>                                        immunecells = cell_fractions,<br>                                        tfs = tf_activities,<br>                                        lrpairs = lrpair_weights,<br>                                        ccpairs = ccpair_scores,<br>                                        cancer_type = <span>'BLCA'</span>,<br>                                        patient_label = patient_ICBresponse)<br><span># 输出图片</span><br>output_biomarkers[[<span>1</span>]]<br>output_biomarkers[[<span>2</span>]]<br>output_biomarkers[[<span>3</span>]]<br>output_biomarkers[[<span>4</span>]]<br>output_biomarkers[[<span>5</span>]]<br>output_biomarkers[[<span>6</span>]]<br></code></pre><p><img data-ratio="0.5956719817767654" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibW6gLia50EwuHfFL82BVl6ibEcqZ1NoJAn550k50pbiaIQwOFpIJK5lbg8HexK4uhtM5dOpicJzSBK3Q/640?wx_fmt=png" data-type="png" data-w="878" title="图11. output_biomarkers[[1]]  信号通路与ICB治疗反应" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibW6gLia50EwuHfFL82BVl6ibEcqZ1NoJAn550k50pbiaIQwOFpIJK5lbg8HexK4uhtM5dOpicJzSBK3Q/640?wx_fmt=png"><span>图11. output_biomarkers[[1]]  信号通路与ICB治疗反应</span></p><p><img data-ratio="0.5956719817767654" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibW6gLia50EwuHfFL82BVl6ibwkgFuod8L0mTx5eRjjaR11BugPrbJ3ASQDC200rq3IpJ2y4swgCjsA/640?wx_fmt=png" data-type="png" data-w="878" title="图12. output_biomarkers[[2]]  免疫细胞与ICB治疗反应" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibW6gLia50EwuHfFL82BVl6ibwkgFuod8L0mTx5eRjjaR11BugPrbJ3ASQDC200rq3IpJ2y4swgCjsA/640?wx_fmt=png"><span>图12. output_biomarkers[[2]]  免疫细胞与ICB治疗反应</span></p><p><img data-ratio="0.5956719817767654" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibW6gLia50EwuHfFL82BVl6ibyPUlYRJAHqygic9gfAMHGCU1aQEL0o7LzyE1jA5tI2vMictzabObsAGg/640?wx_fmt=png" data-type="png" data-w="878" title="图13. output_biomarkers[[3]]  TFs与ICB治疗反应" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibW6gLia50EwuHfFL82BVl6ibyPUlYRJAHqygic9gfAMHGCU1aQEL0o7LzyE1jA5tI2vMictzabObsAGg/640?wx_fmt=png"><span>图13. output_biomarkers[[3]]  TFs与ICB治疗反应</span></p><p><img data-ratio="0.5956719817767654" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibW6gLia50EwuHfFL82BVl6ib8iblv1njXkTUqzJByHAawdTnZzvL1AHmKI8G1O0ZOAJnDFicu88glULw/640?wx_fmt=png" data-type="png" data-w="878" title="图14. output_biomarkers[[4]] 配受体与ICB治疗反应" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibW6gLia50EwuHfFL82BVl6ib8iblv1njXkTUqzJByHAawdTnZzvL1AHmKI8G1O0ZOAJnDFicu88glULw/640?wx_fmt=png"><span>图14. output_biomarkers[[4]] 配受体与ICB治疗反应</span></p><ne-p data-lake-id="u69e97a18"><ne-card data-card-name="image" data-card-type="inline" data-event-boundary="card"><p><img data-backh="331" data-backw="556" data-ratio="0.5956719817767654" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibW6gLia50EwuHfFL82BVl6ibzPY46Qf9VqkibicRuF09gNeMeptj5xEricXydDUdMDRfRk73Dj9LR6tpQ/640?wx_fmt=png" data-type="png" data-w="878" title="图15. output_biomarkers[[5]] 细胞间通讯与ICB治疗反应" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibW6gLia50EwuHfFL82BVl6ibzPY46Qf9VqkibicRuF09gNeMeptj5xEricXydDUdMDRfRk73Dj9LR6tpQ/640?wx_fmt=png"><span>图15. output_biomarkers[[5]] 细胞间通讯与ICB治疗反应</span></p></ne-card></ne-p><p data-tool="mdnice编辑器"><img data-ratio="0.5956719817767654" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibW6gLia50EwuHfFL82BVl6ibFwC42FgQhG5P9PuCq5G6nxQhsoictUH7vW85IV4xnib7G6icB5dCbkyLQ/640?wx_fmt=png" data-type="png" data-w="878" title="图16. output_biomarkers[[6]]  火山图的综合展示（TFs+LR）" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibW6gLia50EwuHfFL82BVl6ibFwC42FgQhG5P9PuCq5G6nxQhsoictUH7vW85IV4xnib7G6icB5dCbkyLQ/640?wx_fmt=png"><span>图16. output_biomarkers[[6]]  火山图的综合展示（TFs+LR）</span></p><p><ne-h1 data-lake-id="d96yd"><ne-heading-ext></ne-heading-ext></ne-h1></p><h1 data-tool="mdnice编辑器">4. 总结</h1><p><span>总体而言，这篇学习文档提供的脚本可重复性是很强的，尽管有些地方确实是没看懂具体怎么个具体数学原理。在同期的</span><em>Pattern </em><span>期刊中，Justin </span><span>Guinney 为这篇文章写了一篇'Preview'。目前在Int J Gen Med杂志（Q1）中已有人引用该算法预测TCGA数据中不同分组的免疫反应（Identification of Three Core Secretome Genes Associated with Immune Infiltration in High Tumor Mutation Burden Across 14 Major Solid Tumors）下图是该文献的图。换句话说，这个R包的认可度，以及可应用性都是比较强的。值得学习，强推。</span><br></p><p data-tool="mdnice编辑器"><img data-ratio="1.2473684210526317" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibW6gLia50EwuHfFL82BVl6ib9Fae5eNGak0iaBdXY5zADIhO7zMl7UqUMM2BeLFg0ctDmr51004L1ZA/640?wx_fmt=png" data-type="png" data-w="380" title="图17. EaSIeR 在文献中的应用" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibW6gLia50EwuHfFL82BVl6ib9Fae5eNGak0iaBdXY5zADIhO7zMl7UqUMM2BeLFg0ctDmr51004L1ZA/640?wx_fmt=png"></p><p><span>图17. EaSIeR 在文献中的应用</span></p><p><span>- END -</span></p><p><br></p><p><br></p><p><br></p><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/_Z5mWDNi8hzAc_mLivcjsQ",target="_blank" rel="noopener noreferrer">原文链接</a>
