---
title: "Moonlight：一种识别生物标志物在不同肿瘤类型和分期中作为癌基因或肿瘤抑制因子的多种作用的方法"
date: 2022-12-12T05:35:20Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Acdemic"]
---
Moonlight：一种识别生物标志物在不同肿瘤类型和分期中作为癌基因或肿瘤抑制因子的多种作用的方法 by 生信菜鸟团
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><section><span>Moonlight: an approach to identify multiple role of biomarkers as oncogene or tumorsuppressor in different tumor types and stages<span> [1,2]</span>.</span></section><blockquote data-tool="mdnice编辑器"><p>2020第一次发表（MoonlightR）Colaprico A, Olsen C, Bailey MH, et al. Interpreting pathways to discover cancer driver genes with Moonlight. <em>Nat Commun</em>. 2020;11(1):69. Published 2020 Jan 3. doi:10.1038/s41467-019-13803-0(2022-11-20 更新，包更新名为 Moonlight2R)</p><p>Astrid Saksager, Mona Nourbakhsh, Nikola Tom,et.al.An Automatized Workflow to Study Mechanistic Indicators for Driver Gene Prediction with Moonlight. bioRxiv 2022.11.18.517066; doi: https://doi.org/10.1101/2022.11.18.517066</p></blockquote><h1 data-tool="mdnice编辑器">1. 引入</h1><p data-tool="mdnice编辑器">Moonlight 包 是2020发表在Nature communication(2022 更新版目前发表在BioRxiv，包名为Moonlight2R)。简单从包的名字上看，会让人摸不着头脑——好端端的生物分析的R包和 “”月光“有什么关系。实际上而言，就“Moonlight”单词检索Pubmed，会发现这个单词广泛应用于“生物学现象”的描述。稍稍检索发现：</p><span><figure data-tool="mdnice编辑器"></figure></span><p><img data-ratio="0.6830985915492958" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9UYVqPH6mggc2ichBSu5MTqpdqiaKcZ1ljfWglNHuw5KAgcxpEHOxf0yZ6XSed3nSq55fribH5Qb4rw/640?wx_fmt=png" data-type="png" data-w="710" width="558.763916015625" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9UYVqPH6mggc2ichBSu5MTqpdqiaKcZ1ljfWglNHuw5KAgcxpEHOxf0yZ6XSed3nSq55fribH5Qb4rw/640?wx_fmt=png"></p><p><span>Cite from 微信公众号“英语” 《老外常说的 "moonlight" 是什么意思？可不止是“月光”哦》</span></p><p data-tool="mdnice编辑器"><br></p><p data-tool="mdnice编辑器">根据作者对Moonlight 一词的阐述：</p><blockquote data-tool="mdnice编辑器"><p>The name refers to (i) the concept of protein moonlighting (or gene sharing) is a phenomenon by which a protein can perform more than one function, and (ii) casting genes in a new light can lead to improved treatment regimens and prognostic indicators.</p></blockquote><p data-tool="mdnice编辑器">就纯粹的望文生义的说，如果说月光是太阳光的反射，那么Moonlight 效应就可以类似于理解为旁路效应或者是说非主要效应。引申义，“兼职”的翻译或许也还能说的过去。</p><p data-tool="mdnice编辑器">言归正传，标题已经大部分阐释了Moonlight R包的应用场景：回答生物分子是否为促癌基因或者抑癌基因的一个方法。那么，这个包解决问题的逻辑是怎样的呢？具体如下：</p><ul data-tool="mdnice编辑器"><li><section>获取数据：函数getDataTCGA和getDataGEO，是作者封装的能够从TCGA中及筛选匹配的GEO数据集中下载数据，进行后续分析；</section></li><li><section>差异分析(DPA)：DPA函数是作者封装的能够进行差异分析的函数，TCGA数据依托于TCGAbiolink R 包的差异分析函数，GEO数据依托于edgR函数，分组包括正常和肿瘤、或正常和 I 期、正常和分子亚型等。</section></li><li><section>功能富集分析(FEA)：FEA 函数是作者封装用于富集分析的函数，作者从文献以及,Msigdb数据库整理获得一个“Diseaselist” 基因集，使用作者构建的方法进行富集分析</section></li><li><section>基因调控网络构建(GRN)：GRN 函数是通过Parmigene 包的K邻近聚类分析获取基因调控网络的工具，最终会obtaining a set of regulated genes for each DEG。</section></li><li><section>上游调节分析（URA）：这里可以说是这个R包的核心之一，其应用 z-score，计算参与特定功能的所有基因的所有预测效应的总和与所有基因数量的平方根之间的比率。</section></li><li><section>模式识别分析（PRA）：PRA函数完成的是根据预定义的生物学过程或者机器学习的方式识别候选 TSG（tumor suppressor genes,增值Down，凋亡UP）和 OCG（oncogenes , 增值Up，凋亡Down）。</section></li><li><section>驱动基因突变分析（DMA）:更新后Moonlight2R新增的功能，DMA函数是对PRA输出（OCG）进一步处理，主要内容是通过Cscape-somatic算法[3]分析MAF(Mutation Annotation Format)文件，对OCG基因是否为突变驱动基因或者为“passenger” 基因进行进一步评估。</section></li></ul><blockquote data-tool="mdnice编辑器"><p>Rogers MF, Gaunt TR, Campbell C. CScape-somatic: distinguishing driver and passenger point mutations in the cancer genome [published correction appears in Bioinformatics. 2021 Oct 25;:]. <em>Bioinformatics</em>. 2020;36(12):3637-3644. doi:10.1093/bioinformatics/btaa242</p></blockquote><p data-tool="mdnice编辑器">以上是整个分析的主要函数及流程，而R包中的moonlight 函数则是对上述函数的进一步封装。下图是其简要的流程图示，以及部分绘图函数的结果：</p><p data-tool="mdnice编辑器"><img data-ratio="0.3188202247191011" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9UYVqPH6mggc2ichBSu5MTqvoT5dSZbmdiaK0oZicKjzPLSmvJ5mwtenqkiboMmOx159vc7ibS0aKVYPg/640?wx_fmt=png" data-type="png" data-w="3560" title="图1. Moonlight pipeline 图示" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9UYVqPH6mggc2ichBSu5MTqvoT5dSZbmdiaK0oZicKjzPLSmvJ5mwtenqkiboMmOx159vc7ibS0aKVYPg/640?wx_fmt=png"><span>图1. Moonlight pipeline 图示</span></p><p data-tool="mdnice编辑器"><br></p><h1 data-tool="mdnice编辑器">2. 安装及加载</h1><p data-tool="mdnice编辑器">MoonlightR 可以通过Bioconductor的方式进行安装，</p><pre data-tool="mdnice编辑器"><code><span>if</span> (!requireNamespace(<span>"BiocManager"</span>, quietly=<span>TRUE</span>))<br>    install.packages(<span>"BiocManager"</span>)<br>BiocManager::install(<span>"MoonlightR"</span>)<br><span>library</span>(edgeR)<br><span>library</span>(EDASeq)<br><span>library</span>(MoonlightR)<br></code></pre><p data-tool="mdnice编辑器">如果安装失败，可以尝试更换Rstudio默认的镜像，自测清华的镜像安装不太行。Biocondutor 的R包一般安装不会有什么太大的问题。这个R包需要”edgeR"包和"EDAseq"包，所以在运行前需要预先加载。<br>更新后的Moonlight2R函数提供的是github安装方式，如下</p><pre data-tool="mdnice编辑器"><code>devtools::install_github(repo = <span>"ELELAB/Moonlight2R"</span>)<br></code></pre><p data-tool="mdnice编辑器">如果想要看demo 文档，需要安装BiocStyle包</p><pre data-tool="mdnice编辑器"><code><span>if</span> (!<span>require</span>(<span>"BiocManager"</span>, quietly = <span>TRUE</span>))<br>    install.packages(<span>"BiocManager"</span>)<br><br>BiocManager::install(<span>"BiocStyle"</span>)<br>devtools::install_github(repo = <span>"ELELAB/Moonlight2R"</span>, build_vignettes = <span>TRUE</span>)<br>vignette( <span>"Moonlight2R"</span>, package=<span>"Moonlight2R"</span>)<br></code></pre><p data-tool="mdnice编辑器"><br></p><h1 data-tool="mdnice编辑器">3. 主要函数解析</h1><h2 data-tool="mdnice编辑器"><span>3.1  数据下载（getDataTCGA and getDataGEO）</span></h2><p data-tool="mdnice编辑器">Moonlight 函数内嵌了 TCGAbiolink 包和GEOquery R包，所以在MoonlightR的应用环境中，可以直接使用函数获取TCGA的18种癌症数据库以及可以通过GEOquery 获取的GEO数据。在作者给定的脚本文件中给出了使用方法，具体如下：</p><h3 data-tool="mdnice编辑器">3.1.1 TCGA数据下载</h3><p data-tool="mdnice编辑器">方式一：按癌症类型和数据类型搜索 [Gene expression]</p><pre data-tool="mdnice编辑器"><code>dataFilt &lt;- getDataTCGA(cancerType = <span>"LUAD"</span>, <br>                          dataType = <span>"Gene expression"</span>,<br>                          directory = <span>"data"</span>,<br>                          nSample = <span>4</span>)<br> class(dataFilt)<br><span># [1] "matrix" "array" </span><br>head(dataFilt)<br><span>##          TCGA-44-6146-01B-04R-A277-07 TCGA-55-A48X-01A-11R-A24H-07 </span><br><span>## 100134869                          251                           19                  </span><br><span>## 155060                             851                          512                   </span><br><span>## 8225                               216                         1111                      </span><br><span>## A1BG                               109                          116   </span><br><span>## A2LD1                              184                          133</span><br></code></pre><p data-tool="mdnice编辑器">方式二：按癌症类型和数据类型搜索 [Methylation]</p><pre data-tool="mdnice编辑器"><code>dataFilt &lt;- getDataTCGA(cancerType = <span>"BRCA"</span>, <br>                          dataType = <span>"Methylation"</span>, <br>                          directory = <span>"data"</span>,nSample = <span>4</span>)<br></code></pre><p data-tool="mdnice编辑器">深度解析<code>getDataTCGA</code>函数的代码，可以发现，其中主要内嵌的就是TCGAbiolink的下载流程和相关参数，而在这里作者大概是重点关注基因表达及甲基化相关的内容，所以只覆盖这两类数据的简单快速下载</p><pre data-tool="mdnice编辑器"><code>getDataTCGA()<br><span>function</span> (cancerType, dataType, directory, cor.cut = <span>0.6</span>, qnt.cut = <span>0.25</span>, <br>          nSample, stage = <span>"ALL"</span>, subtype = <span>0</span>, samples = <span>NULL</span>) <br>{<br>  DiseaseList &lt;- get(<span>"DiseaseList"</span>)<br>  GDCprojects &lt;- get(<span>"GDCprojects"</span>)<br>  geneInfo &lt;- get(<span>"geneInfo"</span>)<br>  CancerProject &lt;- paste0(<span>"TCGA-"</span>, cancerType)<br>  DataDirectory &lt;- paste0(directory, <span>"GDC_"</span>, gsub(<span>"-"</span>, <span>"_"</span>, <br>                                                  CancerProject))<br>  <span>if</span> (dataType == <span>"Gene expression"</span>) {<br>    ……<br>                                     }<br>  <span>else</span> <span>if</span> (dataType == <span>"Methylation"</span>) {<br>   ……<br>  }<br>}<br>&lt;bytecode: <span>0x55b600b5cbc8</span>&gt;<br>  &lt;environment: namespace:MoonlightR&gt;<br></code></pre><p data-tool="mdnice编辑器">方式二：按癌症类型和数据类型搜索 [Mutation]<br>Moonlight2R 说明文档中新增MAF文件下载方式，简单来说就是依托TCGAbiolink R包下载TCGA突变数据。</p><pre data-tool="mdnice编辑器"><code><span>library</span>(TCGAbiolinks)<br>query &lt;- GDCquery(<br>  project = <span>"TCGA-LUAD"</span>, <br>  data.category = <span>"Simple Nucleotide Variation"</span>,<br>  data.type = <span>"Masked Somatic Mutation"</span>,<br>  access = <span>"open"</span>, <br>  legacy = <span>F</span>,<br>  workflow.type = <span>"Aliquot Ensemble Somatic Variant Merging and Masking"</span>)<br><br>GDCdownload(query, directory = <span>"data"</span>)<br>dataMAF &lt;- GDCprepare(query, <br>                        directory = <span>"data"</span>)<br><br>dataMAF &lt;- dataMAF %&gt;% sample_n(size = <span>3000</span>, replace = <span>FALSE</span>)<br></code></pre><p data-tool="mdnice编辑器">TCGAbiolink R包提供的MAF的文件下载说明文档链接：https://bioconductor.org/packages/release/bioc/vignettes/TCGAbiolinks/inst/doc/mutation.html</p><h3 data-tool="mdnice编辑器">3.1.2 GEO数据下载</h3><p data-tool="mdnice编辑器">GEO 函数命令如下，根据作者的描述，这里的GSE数据集仅限于作者<code>GEO_TCGAtab</code>表格中提供的GSE 数据集</p><pre data-tool="mdnice编辑器"><code>dataFilt &lt;- getDataGEO(GEOobject = <span>"GSE20347"</span>,platform = <span>"GPL571"</span>)<br><span># Found 1 file(s)</span><br><span># GSE20347_series_matrix.txt.gz</span><br><span>###  或者  ####</span><br>dataFilt &lt;- getDataGEO(TCGAtumor = <span>"ESCA"</span>)<br></code></pre><p data-tool="mdnice编辑器"><code>getDataGEO</code> 实际上就是个封装好<code>GEOquery </code>在内的一个获取数据集的函数，而对于现在的大多数的GSE数据集来说，GEOquery 的可用性一般。</p><pre data-tool="mdnice编辑器"><code>getDataGEO<br><span>function</span> (GEOobject = <span>"GSE39004"</span>, platform = <span>"GPL6244"</span>, TCGAtumor = <span>NULL</span>) <br>{<br>    GEO_TCGAtab &lt;- get(<span>"GEO_TCGAtab"</span>)<br>    <span>if</span> (length(TCGAtumor) != <span>0</span>) {<br>        GEOobject &lt;- GEO_TCGAtab[GEO_TCGAtab$Cancer == TCGAtumor, <br>            <span>"Dataset"</span>]<br>        platform &lt;- GEO_TCGAtab[GEO_TCGAtab$Cancer == TCGAtumor, <br>            <span>"Platform"</span>]<br>    }<br>    gset &lt;- getGEO(GEOobject, GSEMatrix = <span>TRUE</span>, AnnotGPL = <span>TRUE</span>)<br>    <span>if</span> (length(gset) &gt; <span>1</span>) <br>        idx &lt;- grep(platform, attr(gset, <span>"names"</span>))<br>    <span>else</span> idx &lt;- <span>1</span><br>    gset &lt;- gset[[idx]]<br>    fvarLabels(gset) &lt;- make.names(fvarLabels(gset))<br>    <span>return</span>(gset)<br>}<br>&lt;bytecode: <span>0x557cab26d530</span>&gt;<br>&lt;environment: namespace:MoonlightR&gt;<br></code></pre><p data-tool="mdnice编辑器">GEO_TCGAtab 是作者整理好的18个TCGA数据集及其相对应的GSE序列号，以及一些差异基因信息</p><pre data-tool="mdnice编辑器"><code>knitr::kable(GEO_TCGAtab, digits = <span>2</span>, <br>+             caption = <span>"Table with GEO data set matched to one <br>+             of the 18 given TCGA cancer types "</span>,<br>+             row.names = <span>TRUE</span>)<br><span># Table: Table with GEO data set matched to one </span><br><span>#              of the 18 given TCGA cancer types </span><br><span>#</span><br><span># |   |Cancer |TP   |NT  |DEG. |Dataset  |TP.1 |NT.1 |Platform |DEG.. |Common |GEO_Normal      |GEO_Tumor               |</span><br><span># |:--|:------|:----|:---|:----|:--------|:----|:----|:--------|:-----|:------|:---------------|:-----------------------|</span><br><span># |1  |BLCA   |408  |19  |2937 |GSE13507 |165  |10   |GPL65000 |2099  |896    |control         |cancer                  |</span><br><span># |2  |BRCA   |1097 |114 |3390 |GSE39004 |61   |47   |GPL6244  |2449  |1248   |normal          |Tumor                   |</span><br><span># |3  |CHOL   |36   |9   |5015 |GSE26566 |104  |59   |GPL6104  |3983  |2587   |Surrounding     |Tumor</span></code></pre><h2 data-tool="mdnice编辑器"><span>3.2  DPA 差异基因分析</span></h2><pre data-tool="mdnice编辑器"><code><span>##对于TCGA数据， 应用TCGAanalyze_DEA（基因表达数据） 或TCGAanalyze_DMC函数（甲基化数据）</span><br><span>##对于 GEO数据，则应用limma 的差异基因分析流程</span><br>dataFilt &lt;- getDataTCGA(cancerType = <span>"LUAD"</span>, <br>                          dataType = <span>"Gene expression"</span>,<br>                          directory = <span>"data"</span>,<br>                          nSample = <span>4</span>)<br>dataDEGs &lt;- DPA(dataFilt = dataFilt,<br>                dataType = <span>"Gene expression"</span>)  <br><span>#如 为“Methylation” 则相应修改；DPA函数内根据dataType参数设置了条件语句</span><br><span># 函数来源于TCGAbiolink</span><br>head(dataDEGs)<br><span>#            logFC   logCPM       LR       PValue          FDR</span><br><span># ABCA12  8.475687 2.588540 80.86120 2.421419e-19 1.803715e-15</span><br><span># ABCA8  -2.756115 5.109457 14.23398 1.614294e-04 2.530602e-03</span><br><span># ABCC2   6.763848 4.276506 17.73730 2.536046e-05 5.978166e-04</span><br><span># ABCC3   3.258925 7.397783 19.43158 1.042686e-05 2.987296e-04</span><br><span># ABP1    7.000924 6.369022 30.85150 2.785464e-08 2.901947e-06</span><br><span># ACADL  -2.058819 4.093631 11.41242 7.295472e-04 8.178175e-03</span><br><br>DPA<br><span>function</span> (dataType, dataFilt, dataConsortium = <span>"TCGA"</span>, fdr.cut = <span>0.01</span>, <br>          logFC.cut = <span>1</span>, diffmean.cut = <span>0.25</span>, samplesType, colDescription, <br>          gset, gsetFile = <span>"gsetFile.RData"</span>) <br>{<br>  <span>if</span> (dataConsortium == <span>"GEO"</span>) {<br>    ……<br>    fit &lt;- lmFit(gset, design)<br>    xContrast &lt;- c(<span>"G1-G0"</span>)<br>    cont.matrix &lt;- makeContrasts(xContrast, levels = design)<br>    fit2 &lt;- contrasts.fit(fit, cont.matrix)<br>    fit2 &lt;- eBayes(fit2, <span>0.01</span>)<br>    tT &lt;- topTable(fit2, adjust.method = <span>"fdr"</span>, sort.by = <span>"B"</span>, <br>    ……<br>  }<br>  <span>else</span> <span>if</span> (dataConsortium == <span>"TCGA"</span>) {<br>    <span>if</span> (dataType == <span>"Gene expression"</span>) {<br>    ……<br>      dataDEGs &lt;- TCGAanalyze_DEA(mat1 = dataFilt[, dataSmNT], <br>                                  mat2 = dataFilt[, dataSmTP], Cond1type = <span>"Normal"</span>, <br>                                  Cond2type = <span>"Tumor"</span>, fdr.cut = fdr.cut, logFC.cut = logFC.cut, <br>                                  method = <span>"glmLRT"</span>)<br>    ……<br>    }<br>    <span>else</span> <span>if</span> (dataType == <span>"Methylation"</span>) {<br>     ……<br>      system.time(cancer.met &lt;- TCGAanalyze_DMC(dataFilt, <br>                                                groupCol = <span>"shortLetterCode"</span>, group1 = <span>"TP"</span>, <br>                                                group2 = <span>"NT"</span>, p.cut = fdr.cut, diffmean.cut = <span>0.25</span>, <br>                                                legend = <span>"State"</span>, plot.filename = <span>"test.png"</span>))<br>   ……<br>    }<br>  }<br>}<br>&lt;bytecode: <span>0x55b615cbdde8</span>&gt;<br>  &lt;environment: namespace:MoonlightR&gt;<br>差异分析的结果可以用TCGAbiolink 函数的绘制火山图<br><span>library</span>(TCGAbiolinks)<br>TCGAVisualize_volcano(dataDEGs$logFC, dataDEGs$FDR,<br>                      filename = <span>"DEGs_volcano.png"</span>,<br>                      x.cut = <span>1</span>,<br>                      y.cut = <span>0.05</span>,<br>                      names = rownames(DEGsmatrix),<br>                      color = c(<span>"black"</span>,<span>"red"</span>,<span>"dodgerblue3"</span>),<br>                      names.size = <span>2</span>,<br>                      show.names = <span>"highlighted"</span>,<br>                      highlight = c(<span>"gene1"</span>,<span>"gene2"</span>),<br>                      xlab = <span>" Gene expression fold change (Log2)"</span>,<br>                      legend = <span>"State"</span>,<br>                      title = <span>"Volcano plot (Normal NT vs Tumor TP)"</span>,<br>                      width = <span>10</span>)<br></code></pre><p data-tool="mdnice编辑器"><img data-ratio="0.49703791469194314" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9UYVqPH6mggc2ichBSu5MTqiavkmP2Q4QkTriaqGVicJgNYpRDmQtOoahFe5GumJpHV0VshKbWsvFTQw/640?wx_fmt=png" data-type="png" data-w="1688" title="图 2. TCGA volcano" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9UYVqPH6mggc2ichBSu5MTqiavkmP2Q4QkTriaqGVicJgNYpRDmQtOoahFe5GumJpHV0VshKbWsvFTQw/640?wx_fmt=png"><span>图 2. TCGA volcano</span></p><h2 data-tool="mdnice编辑器"><span>3.3  FEA：功能富集分析</span></h2><pre data-tool="mdnice编辑器"><code>data(DEGsmatrix)  <span># DEGsmatrix 即 差异分析的结果</span><br>dataFEA &lt;- FEA(DEGsmatrix = DEGsmatrix)<br></code></pre><p data-tool="mdnice编辑器"><img data-ratio="0.25176470588235295" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9UYVqPH6mggc2ichBSu5MTqTDQ3SibXx9mmBKZ6zqrAaVzB8BS5xcrUTsibrRTYcKDu8WicK05ta8ZZQ/640?wx_fmt=png" data-type="png" data-w="850" title="图3. dataFEA" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9UYVqPH6mggc2ichBSu5MTqTDQ3SibXx9mmBKZ6zqrAaVzB8BS5xcrUTsibrRTYcKDu8WicK05ta8ZZQ/640?wx_fmt=png"><span>图3. dataFEA</span></p><p data-tool="mdnice编辑器">这里如果按照作者给的绘图代码 肯定是画不出来的，因为他写的源码有点小问题，但影响不大，稍修改一下图还是能画出来的。（更新后的Moonlight2R并没有解决这个问题）</p><pre data-tool="mdnice编辑器"><code><span>#绘图</span><br>plotFEA(dataFEA = dataFEA, additionalFilename = <span>"_exampleVignette"</span>, height = <span>20</span>, width = <span>10</span>)<br><br>plotFEA<br><span>function</span> (dataFEA, topBP = <span>10</span>, additionalFilename = <span>NULL</span>, height, <br>          width, offsetValue = <span>5</span>, angle = <span>90</span>, xleg = <span>35</span>, yleg = <span>5</span>, <br>          titleMain, minY = -<span>5</span>, maxY = <span>10</span>, mycols = c(<span>"#8DD3C7"</span>, <span>"#FFFFB3"</span>, <br>                                                      <span>"#BEBADA"</span>)) <br>{<br>  titleMain &lt;- <span>"TCGA BRCA DEGs"</span>  <span>## 这里写的也是有问题的，基本函数自带的titleMain几乎无用</span><br>  <span>if</span> (!is.null(additionalFilename)) {<br>    pdf(additionalFilename, width, height)  <br>    <span>### 这里pdf()不完整，因此图画不出来，需要修改成：</span><br>    <span>### pdf(file = paste0("plotFEA", additionalFilename, ".pdf"))</span><br>  }<br>  ……<br>  <span>if</span> (!is.null(additionalFilename)) {<br>    dev.off()<br>  }<br>}<br><span># 或者代码修改成，这样才能输出pdf文件。</span><br>plotFEA(dataFEA = dataFEA, additionalFilename = <span>"_exampleVignette.pdf"</span>, height = <span>20</span>, width = <span>10</span>)<br><br></code></pre><p data-tool="mdnice编辑器"><img data-ratio="0.9490254872563718" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9UYVqPH6mggc2ichBSu5MTqbJY5ddcscNuicfbnoiaXEia92I14ViaIDR0sXULuJbzlmPremlPFOapIlw/640?wx_fmt=png" data-type="png" data-w="667" title="图4. FEAplot" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9UYVqPH6mggc2ichBSu5MTqbJY5ddcscNuicfbnoiaXEia92I14ViaIDR0sXULuJbzlmPremlPFOapIlw/640?wx_fmt=png"><span>图4. FEAplot</span></p><h2 data-tool="mdnice编辑器"><span>3.4 GRN：基因调节网络分析</span></h2><p data-tool="mdnice编辑器">该步骤的核心是  ‘<code>parmigene</code>’   包的<code> knnmi.cross</code>  函数，本质是K-邻近聚类的算法</p><pre data-tool="mdnice编辑器"><code>dataGRN &lt;- GRN(TFs = rownames(DEGsmatrix)[<span>1</span>:<span>100</span>], normCounts = dataFilt,<br>                   nGenesPerm = <span>10</span>,kNearest = <span>3</span>,nBoot = <span>10</span>)<br><span># TFs  基因名，normCounts 表达矩阵</span><br></code></pre><p data-tool="mdnice编辑器"><img data-ratio="0.15474794841735054" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9UYVqPH6mggc2ichBSu5MTqXcQFvDAcXvw3zvrHaZT2US11ITOROoWBun66EpwibyLUuBCZS2BbQyQ/640?wx_fmt=png" data-type="png" data-w="853" title="图5. dataGRN" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9UYVqPH6mggc2ichBSu5MTqXcQFvDAcXvw3zvrHaZT2US11ITOROoWBun66EpwibyLUuBCZS2BbQyQ/640?wx_fmt=png"><span>图5. dataGRN</span></p><p data-tool="mdnice编辑器">在作者的说明文档里并没有提出该怎么解读这个<code>dataGRN</code>的结果，这里按下不表。对于GRN数据结果，作者与<code>Cosmic</code>数据库相关联，绘制巢状网络图，<code>knownDriverGenes</code>即来自于<code>Cosmic </code>数据库</p><pre data-tool="mdnice编辑器"><code>data(knownDriverGenes)<br>head(knownDriverGenes)<br><span># $TSG</span><br><span># [1] "ABI1"     "ALDH2"    "AMER1"    "APC"      "ARHGAP26" "ARHGEF12" "ARID1A"   "ARID1B"   "ARID2"    "ASXL1"    "ATM"      "BAP1"     "BCOR"    </span><br><span># [14] "BRCA1"    "BRCA2"    "CASP8"    "CBLB"     "CDC73"    "CDKN1B"   "CDKN2A"   "CDKN2C"   "CHEK2"    "CTCF"     "CYLD"     "DICER1"   "DNMT3A"  </span><br><span># [27] "ERCC2"    "FAT1"     "FAT4"     "IKZF1"    "LZTR1"    "MEN1"     "NCOR2"    "PALB2"    "PBRM1"    "PTEN"     "PTPN13"   "RB1"      "RBM10"   </span><br><span># [40] "SDHA"     "SETD2"    "SH2B3"    "SMAD2"    "SMAD3"    "SMAD4"    "SOCS1"    "TET2"     "TGFBR2"   "TP53"     "TSC1"     "TSC2"     "VHL"     </span><br><span># [53] "XPA"      "XPC"      "ZFHX3"   </span><br><br><span># $OCG</span><br><span># [1] "ABL1"    "ABL2"    "ACKR3"   "ACVR1"   "AFF1"    "AFF3"    "AFF4"    "AKAP9"   "AKT1"    "AKT2"    "ALK"     "AR"      "ASPSCR1" "ATP1A1" </span><br><span># [15] "BCL2"    "BCL3"    "BCL5"    "BCL6"    "BCL7A"   "BCL9"    "BCR"     "BRAF"    "CALR"    "CCND1"   "CDK4"    "CDK6"    "CHD4"    "CTNNB1" </span><br><span># [29] "CXCR4"   "ERBB2"   "ERBB3"   "EZR"     "FLT3"    "FOXP1"   "FUS"     "GATA2"   "HIF1A"   "HLF"     "HMGA2"   "HOXA11"  "HOXA13"  "HOXA9"  </span><br><span># [43] "HOXC11"  "HOXC13"  "HOXD11"  "HOXD13"  "HRAS"    "IDH1"    "IDH2"    "JAK2"    "KIT"     "KLF4"    "KMT2A"   "KRAS"    "MDM2"    "MDM4"   </span><br><span># [57] "MECOM"   "MET"     "MLLT1"   "MLLT3"   "MPL"     "MTOR"    "MYC"     "MYCN"    "MYD88"   "NAB2"    "NFKB2"   "NRAS"    "PDGFB"   "PDGFRA" </span><br><span># [71] "PIK3CA"  "PML"     "PRKACA"  "PTPN11"  "RARA"    "REL"     "RET"     "ROS1"    "RUNX1"   "RUNX1T1" "STAT3"   "SYK"     "TRRAP"   "USP8"   </span><br><br>data(dataGRN)<br>plotNetworkHive(dataGRN, knownDriverGenes, <span>0.55</span>)<br></code></pre><p data-tool="mdnice编辑器"><img data-ratio="0.46026986506746626" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9UYVqPH6mggc2ichBSu5MTqBtYl8pUsamh6cdJPpwiaKUyCnKOtLw4iagJj2ggia8vY0ccl2fiar414ag/640?wx_fmt=png" data-type="png" data-w="667" title="图6. plotNetworkHive" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9UYVqPH6mggc2ichBSu5MTqBtYl8pUsamh6cdJPpwiaKUyCnKOtLw4iagJj2ggia8vY0ccl2fiar414ag/640?wx_fmt=png"><span>图6. plotNetworkHive</span></p><h2 data-tool="mdnice编辑器"><span>3.5 URA:上游调节子分析</span></h2><p data-tool="mdnice编辑器">这个步骤需要同时用到<code>GRN</code>和<code>FEA</code>的分析结果，</p><pre data-tool="mdnice编辑器"><code>data(dataGRN)<br>data(DEGsmatrix)<br><br>dataFEA &lt;- FEA(DEGsmatrix = DEGsmatrix)<br><br>BPselected &lt;- dataFEA$Diseases.or.Functions.Annotation[<span>1</span>:<span>5</span>]<br>dataURA &lt;- URA(dataGRN = dataGRN,<br>               DEGsmatrix = DEGsmatrix,<br>               BPname = BPselected,  <span>##可以是指定的Biological process</span><br>               nCores=<span>1</span>)<br>head(dataURA)<br><span>#       abdnominal adenocarcinoma adenocarcinoma alzheimers disease amyloidosis apoptosis of tumore cell lines</span><br><span># A1BG                   0.0000000      0.0000000                  0   0.0000000                      0.6666667</span><br><span># A2M                    0.0000000      0.0000000                  0   0.0000000                     -0.2294157</span><br><span># AASS                   0.0000000      0.0000000                  0   0.0000000                     -0.3779645</span><br><span># ABAT                   0.0000000      0.0000000                  0   0.0000000                      0.0000000</span><br><span># ABCA10                -0.2041241     -0.1740777                  0   0.0000000                     -0.9486833</span><br><span># ABCA12                 0.0000000      0.0000000                  0  -0.5303301                     -0.8340577</span></code></pre><h2 data-tool="mdnice编辑器"><span>3.6  PRA: 模式识别子分析</span></h2><pre data-tool="mdnice编辑器"><code>data(dataURA)<br>dataDual &lt;- PRA(dataURA = dataURA,<br>                          BPname = c(<span>"apoptosis"</span>,<span>"proliferation of cells"</span>),<br>                          thres.role = <span>0</span>)<br>head(dataDual)<br><span>#$TSG</span><br><span>#    ACN9    ADCK5  AFAP1L1 </span><br><span>#1.123390 1.126988 1.112131 </span><br><br><span>#$OCG</span><br><span>#    ABCA3     ABCG1     ABCG2     ABHD6     ABTB1     ACADL      ACAN </span><br><span>#3.3825395 1.1088252 3.3062860 1.6521305 2.7699272 2.8953050 1.6175689 </span><br><span>#     ACE2      ACP5      ACP6      ACRC     ACSS1     ACSS2     ACTG2 </span><br><span>#1.1016235 1.4461591 1.7583342 1.2523922 2.8238690 0.6931391 2.9399313 </span><br><span>#   ADAM19     ADAM6  ADAMTS12  ADAMTS16    ADHFE1     ADRB1    ADSSL1 </span><br><span>#2.0317063 2.9133937 1.5595271 1.0147188 3.7288269 2.3278862 0.7482893 </span><br><span>#   AGPAT4 </span><br><span>#3.2790185</span><br></code></pre><p data-tool="mdnice编辑器">PRA嵌套了画图函数，具体如下</p><pre data-tool="mdnice编辑器"><code>plotURA(dataURA = dataURA[c(names(dataDual$TSG),<br>                            names(dataDual$OCG)),, drop = <span>FALSE</span>], <br>        additionalFilename = <span>"_exampleVignette"</span>)<br></code></pre><p data-tool="mdnice编辑器"><img data-ratio="1.026143790849673" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9UYVqPH6mggc2ichBSu5MTqiaxz6BXGaRrt8dyQk2SqwYo6nnvTpnq0duficLVoYibUkxvaV5SnibtFwg/640?wx_fmt=png" data-type="png" data-w="612" title="图7. plotURA" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9UYVqPH6mggc2ichBSu5MTqiaxz6BXGaRrt8dyQk2SqwYo6nnvTpnq0duficLVoYibUkxvaV5SnibtFwg/640?wx_fmt=png"><span>图7. plotURA</span></p><p data-tool="mdnice编辑器">这里绘图得到的数值和作者提供在示例文档中的略有区别，但整体的趋势是一致的。</p><h2 data-tool="mdnice编辑器"><span>3.7 DMA：Driver Mutation Analysis（驱动突变分析）</span></h2><p data-tool="mdnice编辑器">正如前述，DMA分析嵌套了CScape-somatic 算法，这个部分的输入数据包括三个数据， MAF矩阵，差异分析结果（DEGmatrix）,以及3.6步骤中PRA的输出结果（dataPRA，含OCG）,另外以及CScape的类似于索引文件输入（这个部分为什么不作为R包的内置嵌入，而要人工输入？）。这里提供的文件索引在R包中没有找到，后续在CScape-somatic的网页中找到了这个数据，链接如下http://cscape-somatic.biocompute.org.uk/#download；</p><pre data-tool="mdnice编辑器"><code>data(dataPRA)<br>data(dataMAF)<br>dataDMA &lt;- DMA(dataMAF = dataMAF,<br>               dataDEGs = DEGsmatrix, <br>               dataPRA = dataPRA,<br>               results_folder = <span>"DMAresults"</span>,<br>               coding_file = <span>"/data/databases/CScape/CScape-20210624/css_coding.vcf.gz"</span>,<br>               noncoding_file = <span>"/data/databases/CScape/CScape-20210624/css_noncoding.vcf.gz"</span>)<br></code></pre><p data-tool="mdnice编辑器">稍稍看了下，这两个文件确实有点大，实在是包装不了。<br><img data-ratio="0.3147153598281418" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9UYVqPH6mggc2ichBSu5MTqFiaQes0iamR1OKBbyewsxhoeWibBFWnQmFYNiaTNia2K3vPBuCzcuGRic7Gg/640?wx_fmt=png" data-type="png" data-w="931" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9UYVqPH6mggc2ichBSu5MTqFiaQes0iamR1OKBbyewsxhoeWibBFWnQmFYNiaTNia2K3vPBuCzcuGRic7Gg/640?wx_fmt=png"><br>这个部分主要输出是<code>DEG_Mutations_Annotations </code> <code>Oncogenic_mediators_mutation_summary</code>，Moonligh2R包装了画图函数<code>plotMoonlight</code>，具体使用如下</p><pre data-tool="mdnice编辑器"><code>data(DEG_Mutations_Annotations)<br>data(Oncogenic_mediators_mutation_summary)<br>plotMoonlight(DEG_Mutations_Annotations,<br>        Oncogenic_mediators_mutation_summary, <br>        dataURA, gene_type = <span>"drivers"</span>, n = <span>50</span>)<br></code></pre><p data-tool="mdnice编辑器"><img data-ratio="0.42694928084784256" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9UYVqPH6mggc2ichBSu5MTq40IpHicIUiauEWkQHsBrWmWhrPXKXrCJk2EPqHnBxzBIco6AzRly4icxg/640?wx_fmt=png" data-type="png" data-w="1321" title="图8. plotMoonlight" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9UYVqPH6mggc2ichBSu5MTq40IpHicIUiauEWkQHsBrWmWhrPXKXrCJk2EPqHnBxzBIco6AzRly4icxg/640?wx_fmt=png"><span>图8. plotMoonlight</span></p><p data-tool="mdnice编辑器">以及plotDMA</p><pre data-tool="mdnice编辑器"><code>data(dataDMA)<br>data(DEG_Mutations_Annotations)<br>data(Oncogenic_mediators_mutation_summary)<br>plotDMA(DEG_Mutations_Annotations, <br>        Oncogenic_mediators_mutation_summary, <br>        type = <span>'complete'</span>, additionalFilename = <span>"./DMAresults/"</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.4307228915662651" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9UYVqPH6mggc2ichBSu5MTqjbChMdWH4VTUwwhVu9AePaTiaibMnRr0fYG8C0mbD0EWKUL8nx1JsG8Q/640?wx_fmt=png" data-type="png" data-w="1328" title="图9. plotDMA" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9UYVqPH6mggc2ichBSu5MTqjbChMdWH4VTUwwhVu9AePaTiaibMnRr0fYG8C0mbD0EWKUL8nx1JsG8Q/640?wx_fmt=png"><figcaption><span>图9. plotDMA</span></figcaption></figure><p data-tool="mdnice编辑器"><br></p><p data-tool="mdnice编辑器">以上是Moonlight R语言包的主要函数。</p><hr data-tool="mdnice编辑器"><p data-tool="mdnice编辑器"><br></p><h1 data-tool="mdnice编辑器">4. 使用示例</h1><p data-tool="mdnice编辑器">作者在说明文档里提供了几个案例研究，主要包括以下三类</p><ol data-tool="mdnice编辑器"><li><section>Downstream analysis LUAD using RNA expression data</section></li><li><section>Expression pipeline Pan Cancer with five cancer types</section></li><li><section>Expression pipeline with stages I II III IV (BRCA) </section></li></ol><h2 data-tool="mdnice编辑器"><span>4.1 案例一 :Downstream analysis LUAD using RNA expression data</span></h2><pre data-tool="mdnice编辑器"><code>dataFilt &lt;- getDataTCGA(cancerType = <span>"LUAD"</span>, <br>                          dataType = <span>"Gene expression"</span>,<br>                          directory = <span>"data"</span>,<br>                          nSample = <span>4</span>)<br>dataDEGs &lt;- DPA(dataFilt = dataFilt,<br>                dataType = <span>"Gene expression"</span>)<br><br>dataFEA &lt;- FEA(DEGsmatrix = dataDEGs)<br><br>dataGRN &lt;- GRN(TFs = rownames(dataDEGs)[<span>1</span>:<span>100</span>], <br>               DEGsmatrix = dataDEGs,<br>               DiffGenes = <span>TRUE</span>,<br>               normCounts = dataFilt)<br><br>dataURA &lt;- URA(dataGRN = dataGRN, <br>              DEGsmatrix = dataDEGs, <br>              BPname = c(<span>"apoptosis"</span>,<br>                         <span>"proliferation of cells"</span>))<br><br>dataDual &lt;- PRA(dataURA = dataURA, <br>               BPname = c(<span>"apoptosis"</span>,<br>                          <span>"proliferation of cells"</span>),<br>               thres.role = <span>0</span>)<br><br>CancerGenes &lt;- list(<span>"TSG"</span>=names(dataDual$TSG), <span>"OCG"</span>=names(dataDual$OCG))<br>head(CancerGenes)<br><span># $TSG</span><br><span># [1] "ACAT1"    "ACP5"     "ADAMTS14" "ADCY8"    "ADH1A"    "ADH1B"    "AGTR1"    "ALAS2"    "ALDH1A2"  "ANGPT1"   "ANKRD58"  "ANXA8"    "AP1S2"   </span><br><span># [14] "APLNR"    "APLN"     "APOC1"    "ARC"      "ARHGAP32" "ARHGAP42" "ASPA"     "ATP1B2"   "ATP8A2"   "AVPR2"   </span><br><br><span># $OCG</span><br><span># [1] "ABCC2"    "ADAM12"   "ADAM28"   "ADAMTSL5" "ADH4"     "AMN"      "ANKFN1"   "ANKRD22"  "ANKRD43"  "ANLN"     "ANO3"     "APC2"     "APOB"    </span><br><span># [14] "AQP11"    "ATHL1"    "AURKA"</span><br></code></pre><p data-tool="mdnice编辑器"><code>CancerGenes </code>即<code>MoonlightR </code>所获得的 <code>TSG</code>（肿瘤抑制基因），<code>OCG</code>(促癌基因）。</p><h2 data-tool="mdnice编辑器"><span>4.2 案例二：Expression pipeline Pan Cancer 5 cancer types</span></h2><p data-tool="mdnice编辑器">moonlight 函数是对所有函数的进行一个统一包装，但是，这里没有对dataMAF进行统一处理，新版本的moonlight2R只是简单的把dataMAF简单的写在函数的input里面，也没有对这个函数做获取数据的嵌入，导致他的示例数据无法使用。MoonlightR 版本是可以运行示例数据的，但Moonlight2R这里运行不了，就是因为dataMAF没有很好的嵌入。按照更新版本的moonlight， 理论上来说，pan cancer 的分析代码中应该嵌入5个函数MAF文件的获取（可以参考源code的for 循环），但这里显然是没有的，感兴趣的可以看一下原始函数。</p><pre data-tool="mdnice编辑器"><code><span># MoonlightR version 1</span><br>cancerList &lt;- c(<span>"BLCA"</span>,<span>"COAD"</span>,<span>"ESCA"</span>,<span>"HNSC"</span>,<span>"STAD"</span>)<br><span>## moonlight 嵌套了上述MoonlightR包的主要函数</span><br>listMoonlight &lt;- moonlight(cancerType = cancerList, <br>                      dataType = <span>"Gene expression"</span>,<br>                      directory = <span>"data"</span>,<br>                      nSample = <span>10</span>,<br>                      nTF = <span>100</span>,<br>                      DiffGenes = <span>TRUE</span>,<br>                      BPname = c(<span>"apoptosis"</span>,<span>"proliferation of cells"</span>))<br>save(listMoonlight, file = paste0(<span>"listMoonlight_ncancer4.Rdata"</span>))<br><span>## 绘图</span><br>plotCircos(listMoonlight = listMoonlight, additionalFilename = <span>"_ncancer5"</span>)<br><span># Moonlight2R</span><br>moonlight<br><span>function</span> (cancerType = <span>"panCancer"</span>, dataType = <span>"Gene expression"</span>, <br>          directory = <span>"GDCdata"</span>, BPname = <span>NULL</span>, cor.cut = <span>0.6</span>, qnt.cut = <span>0.25</span>, <br>          Genelist = <span>NULL</span>, fdr.cut = <span>0.01</span>, logFC.cut = <span>1</span>, corThreshold = <span>0.6</span>, <br>          kNearest = <span>3</span>, nGenesPerm = <span>10</span>, DiffGenes = <span>FALSE</span>, nBoot = <span>100</span>, <br>          nTF = <span>NULL</span>, nSample = <span>NULL</span>, thres.role = <span>0</span>, stage = <span>NULL</span>, <br>          subtype = <span>0</span>, samples = <span>NULL</span>, dataMAF, path_cscape_coding,  <span># 仅简单增加dataMAF及相关输入文件</span><br>          path_cscape_noncoding) <br>{……<br>  driverGenes &lt;- DMA(dataMAF = dataMAF, dataDEGs = dataDEGs, <br>                       dataPRA = listCandidates, runCscape = <span>TRUE</span>, coding_file = path_cscape_coding, <br>                       noncoding_file = path_cscape_noncoding, results_folder = <span>"./DMAresults"</span>)<br> ……<br> }<br><br></code></pre><p data-tool="mdnice编辑器"><img data-ratio="0.36666666666666664" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9UYVqPH6mggc2ichBSu5MTqX6etajcibzRt13e60KrB3oEpL9VibKzTMPSBoTXdgg1PJPTH8h7MDsxQ/640?wx_fmt=png" data-type="png" data-w="480" title="图10. listMoonlight 是多个流程结果的List" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9UYVqPH6mggc2ichBSu5MTqX6etajcibzRt13e60KrB3oEpL9VibKzTMPSBoTXdgg1PJPTH8h7MDsxQ/640?wx_fmt=png"><span>图10. listMoonlight 是多个流程结果的List</span></p><p data-tool="mdnice编辑器"><code>plotCircos </code>是对多个结果的 <code>TSGs </code>和<code>OCGs </code>的综合展示。</p><p data-tool="mdnice编辑器"><img data-ratio="0.9683794466403162" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9UYVqPH6mggc2ichBSu5MTqdhBb0qdVxMWCEyIGE8Se2mgCib8osdaIJicxTg0yNBbOyqWQPWxZ13qA/640?wx_fmt=png" data-type="png" data-w="759" title="图11. plotCircos" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9UYVqPH6mggc2ichBSu5MTqdhBb0qdVxMWCEyIGE8Se2mgCib8osdaIJicxTg0yNBbOyqWQPWxZ13qA/640?wx_fmt=png"><span>图11. plotCircos</span></p><h2 data-tool="mdnice编辑器"><span>4.3 案例三：Downstream analysis BRCA with stages</span></h2><p data-tool="mdnice编辑器">在MoonlightR初始版本中，出现了以下问题</p><pre data-tool="mdnice编辑器"><code>rm(list=ls())<br><br>listMoonlight &lt;- <span>NULL</span><br><span>for</span> (i <span>in</span> <span>1</span>:<span>4</span>){<br>  dataDual &lt;- moonlight(cancerType = <span>"BRCA"</span>, <br>                        dataType = <span>"Gene expression"</span>,<br>                        directory = <span>"data"</span>,<br>                       nSample = <span>10</span>,<br>                        nTF = <span>5</span>,<br>                        DiffGenes = <span>TRUE</span>,<br>                        BPname = c(<span>"apoptosis"</span>,<span>"proliferation of cells"</span>),<br>                        stage = i)<br>  listMoonlight &lt;- c(listMoonlight, list(dataDual))<br>  save(dataDual, file = paste0(<span>"dataDual_stage"</span>,as.roman(i), <span>".Rdata"</span>))<br>}<br><span># [1] "-----------------------------------------"</span><br><span># [1] "Get TCGA data"</span><br><span># [1] "-----------------------------------------"</span><br><span># [1] "cancer type: BRCA"</span><br><span># --------------------------------------</span><br><span># o GDCquery: Searching in GDC database</span><br><span># --------------------------------------</span><br><span># Genome of reference: hg19</span><br><span># --------------------------------------------</span><br><span># oo Accessing GDC. This might take a while...</span><br><span># --------------------------------------------</span><br><span># ooo Project: TCGA-BRCA</span><br><span># Connected to your session in progress, last started 2022-Oct-25 10:13:22 UTC (14 minutes ago)</span><br><span># --------------------</span><br><span># oo Filtering results</span><br><span># --------------------</span><br><span># ooo By platform</span><br><span># ooo By data.type</span><br><span># ooo By file.type</span><br>----------------<br><span># oo Checking data</span><br><span># ----------------</span><br><span># ooo Checking if there are duplicated cases</span><br><span># Warning: There are more than one file for the same case. Please verify query results. You can use the command View(getResults(query)) in rstudio</span><br><span># ooo Checking if there are results for the query</span><br><span># -------------------</span><br><span># o Preparing output</span><br><span># -------------------</span><br><span>#Error in `$&lt;-.data.frame`(`*tmp*`, "tumor_stage", value = character(0)) : </span><br><span>#  replacement has 0 rows, data has 1098</span><br></code></pre><p data-tool="mdnice编辑器">按作者提供的代码，出现报错，复盘→ Debug  ，</p><pre data-tool="mdnice编辑器"><code><br>moonlight<br><span>function</span> (cancerType = <span>"panCancer"</span>, dataType = <span>"Gene expression"</span>, <br>    directory = <span>"GDCdata"</span>, BPname = <span>NULL</span>, cor.cut = <span>0.6</span>, <br>    qnt.cut = <span>0.25</span>, Genelist = <span>NULL</span>, fdr.cut = <span>0.01</span>, logFC.cut = <span>1</span>, <br>    corThreshold = <span>0.6</span>, kNearest = <span>3</span>, nGenesPerm = <span>10</span>, DiffGenes = <span>FALSE</span>, <br>    nBoot = <span>100</span>, nTF = <span>NULL</span>, nSample = <span>NULL</span>, thres.role = <span>0</span>, <br>    stage = <span>NULL</span>, subtype = <span>0</span>, samples = <span>NULL</span>) <br>{<br>    GDCprojects &lt;- get(<span>"GDCprojects"</span>)<br>    <span>if</span> (length(cancerType) == <span>1</span> &amp;&amp; cancerType == <span>"panCancer"</span>) {<br>        cancerType &lt;- sort(sapply(strsplit(grep(<span>"TCGA"</span>, <br>            GDCprojects, value = <span>TRUE</span>), <span>"TCGA-"</span>), <span>"["</span>, <br>            <span>2</span>))<br>    }<br>    res &lt;- <span>NULL</span><br>    <span>for</span> (cancer.i <span>in</span> cancerType) {<br>        print(<span>"-----------------------------------------"</span>)<br>        print(<span>"Get TCGA data"</span>)<br>        print(<span>"-----------------------------------------"</span>)<br>        print(paste(<span>"cancer type:"</span>, cancer.i))<br>        dataFilt &lt;- getDataTCGA(cancerType = cancer.i, dataType = dataType, <br>            directory = directory, cor.cut = cor.cut, qnt.cut = qnt.cut, <br>            nSample = nSample, stage = stage, subtype = subtype, <br>            samples = samples)<br>        print(<span>"-----------------------------------------"</span>)<br>        print(<span>"Differential phenotype analysis"</span>)<br>        print(<span>"-----------------------------------------"</span>)<br>        dataDEGs &lt;- DPA(dataType = dataType, dataFilt = dataFilt, <br>            fdr.cut = fdr.cut, logFC.cut = logFC.cut)<br>        print(<span>"-----------------------------------------"</span>)<br>        print(<span>"Gene regulatory network"</span>)<br>        print(<span>"-----------------------------------------"</span>)<br>        <span>if</span> (is.null(nTF)) {<br>            nTF &lt;- nrow(dataDEGs)<br>        }<br>        <span>if</span> (is.null(Genelist)) {<br>            Genelist &lt;- rownames(dataDEGs)[<span>1</span>:nTF]<br>        }<br>        dataGRN &lt;- GRN(TFs = Genelist, normCounts = dataFilt, <br>            DEGsmatrix = dataDEGs, DiffGenes = <span>FALSE</span>, nGenesPerm = nGenesPerm, <br>            kNearest = kNearest, nBoot = nBoot)<br>        print(<span>"-----------------------------------------"</span>)<br>        print(<span>"Upstream regulator analysis"</span>)<br>        print(<span>"-----------------------------------------"</span>)<br>        dataURA &lt;- URA(dataGRN = dataGRN, DEGsmatrix = dataDEGs, <br>            BPname = BPname)<br>        print(<span>"-----------------------------------------"</span>)<br>        print(<span>"Get candidates"</span>)<br>        print(<span>"-----------------------------------------"</span>)<br>        listCandidates &lt;- PRA(dataURA = dataURA, BPname = BPname, <br>            thres.role = thres.role)<br>        res.i &lt;- list(dataDEGs = dataDEGs, dataURA = dataURA, <br>            listCandidates = listCandidates)<br>        res &lt;- c(res, list(res.i))<br>    }<br>    names(res) &lt;- cancerType<br>    <span>return</span>(res)<br>}<br>&lt;bytecode: <span>0x000002312986ce50</span>&gt;<br>&lt;environment: namespace:MoonlightR&gt;<br></code></pre><p data-tool="mdnice编辑器">仔细浏览及逐一运行<code>moonlight</code>函数的代码，发现<code> # o Preparing output</code>  是<code> getDataTCGA</code> 函数内部<code>GDCquery</code>函数的输出提示文本，而下一个提示文本<code>Differential phenotype analysis</code> 还没有出现表明，报错出现在<code>getDataTCGA</code>。继续在<code>getDataTCGA</code>中debug</p><pre data-tool="mdnice编辑器"><code><span>library</span>(TCGAbiolinks)<br>getDataTCGA<br><span>function</span> (cancerType, dataType, directory, cor.cut = <span>0.6</span>, qnt.cut = <span>0.25</span>, <br>          nSample, stage = <span>"ALL"</span>, subtype = <span>0</span>, samples = <span>NULL</span>) <br>{<br>  DiseaseList &lt;- get(<span>"DiseaseList"</span>)<br>  GDCprojects &lt;- get(<span>"GDCprojects"</span>)<br>  geneInfo &lt;- get(<span>"geneInfo"</span>)<br>  CancerProject &lt;- paste0(<span>"TCGA-"</span>, cancerType)<br>  DataDirectory &lt;- paste0(directory, <span>"GDC_"</span>, gsub(<span>"-"</span>, <br>                                                  <span>"_"</span>, CancerProject))<br>  <span>if</span> (dataType == <span>"Gene expression"</span>) {<br>    FileNameData &lt;- paste0(DataDirectory, <span>"_stage_"</span>, <br>                           stage, <span>"_subtype_"</span>, subtype, <span>"_"</span>, <span>"IlluminaHiSeq"</span>, <br>                           <span>".rda"</span>)<br>    <span>if</span> (!file.exists(FileNameData)) {<br>      query &lt;- GDCquery(project = CancerProject, data.category = <span>"Gene expression"</span>, <br>                        data.type = <span>"Gene expression quantification"</span>, <br>                        platform = <span>"Illumina HiSeq"</span>, file.type = <span>"results"</span>, <br>                        legacy = <span>TRUE</span>)<br>      samplesDown &lt;- query$results[[<span>1</span>]]$cases<br>      dataSmTP &lt;- TCGAquery_SampleTypes(barcode = samplesDown, <br>                                        typesample = <span>"TP"</span>)<br>      dataSmNT &lt;- TCGAquery_SampleTypes(barcode = samplesDown, <br>                                        typesample = <span>"NT"</span>)<br><br><span>if</span> (is.numeric(stage) == <span>TRUE</span>) {<br>+         dataClin &lt;- GDCquery_clinic(project = CancerProject, <br>+                                     type = <span>"clinical_patient"</span>)<br>+         curStage &lt;- paste0(<span>"Stage "</span>, as.roman(stage))<br>+         dataClin$tumor_stage &lt;- toupper(dataClin$tumor_stage)<br>+         dataClin$tumor_stage &lt;- gsub(<span>"[ABCDEFGH]"</span>, <br>+                                      <span>""</span>, dataClin$tumor_stage)<br>+         dataClin$tumor_stage &lt;- gsub(<span>"ST"</span>, <span>"Stage"</span>, <br>+                                      dataClin$tumor_stage)<br>+         dataStg &lt;- dataClin[dataClin$tumor_stage %<span>in</span>% <br>+                               curStage, ]<br>+         message(paste(curStage, <span>"with"</span>, nrow(dataStg), <br>+                       <span>"samples"</span>))<br>+         dataStgC &lt;- dataSmTP[substr(dataSmTP, <span>1</span>, <span>12</span>) %<span>in</span>% <br>+                                dataStg$bcr_patient_barcode]<br>+         dataSmTP &lt;- dataStgC<br>+       }<br><span>##  类似的报错在这里出现了 !!!</span><br><span>## Error in `$&lt;-.data.frame`(`*tmp*`, tumor_stage, value = character(0)) :  。</span><br><span>## 替换数据里有0行，但数据有1098</span><br><span>#   继续逐一运行，发现报错主要来自这一行</span><br>dataClin$tumor_stage &lt;- toupper(dataClin$tumor_stage)<br><span>###   Error in `$&lt;-.data.frame`(`*tmp*`, tumor_stage, value = character(0)) : </span><br><span>###   替换数据里有0行，但数据有1098</span><br>dataClin$tumor_stage<br><span>###  NULL</span><br></code></pre><p data-tool="mdnice编辑器"><img data-ratio="0.1690427698574338" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9UYVqPH6mggc2ichBSu5MTqrqAzh6VEqcFwHqYHfmOzntVTXt0Ho4hicuicZyf1P2T9rJJspQiaNsDrg/640?wx_fmt=png" data-type="png" data-w="982" title="图12. dataClin" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9UYVqPH6mggc2ichBSu5MTqrqAzh6VEqcFwHqYHfmOzntVTXt0Ho4hicuicZyf1P2T9rJJspQiaNsDrg/640?wx_fmt=png"><span>图12. dataClin</span></p><p data-tool="mdnice编辑器"><span><img data-ratio="0.29004329004329005" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9UYVqPH6mggc2ichBSu5MTqoicvRfG6oYJ62C5NcnNWod92qdWYCvPX5u6SvzqHAnf3lYsa32cqiagw/640?wx_fmt=png" data-type="png" data-w="693" title="图13. dataClin" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9UYVqPH6mggc2ichBSu5MTqoicvRfG6oYJ62C5NcnNWod92qdWYCvPX5u6SvzqHAnf3lYsa32cqiagw/640?wx_fmt=png"><span>图13. dataClin</span></span></p><p data-tool="mdnice编辑器"><span>由此，Bug 主要来自于</span><code>BRCA</code><span>的数据 </span><code>dataClin</code><span> 没有 </span><code>tumor_stage</code><span> 这一列，而是</span><code>ajcc_pathologic_stage</code><span> 列。而后逐一查看“</span><code>COAD</code><span>”及‘</span><code>BLCA</code><span>’的临床数据集均出现问题。所以 这部分的代码大概率是不太行了。</span><br><span>在Moonlight2R的更新版本中，moonlight函数并没有提供获取MAF矩阵的函数，因此这部分报错是预料之中的，但是让我意外的是，更新后的Moonlight 还是出现了和之前差不多的报错，这……作者上传之前都不检查一下自己写的代码能不能运行的嘛？？</span></p><pre data-tool="mdnice编辑器"><code>&gt; <span>for</span> (i <span>in</span> <span>1</span>:<span>4</span>){<br>+   dataDual &lt;- moonlight(cancerType = <span>"BRCA"</span>, <br>+                         dataType = <span>"Gene expression"</span>,<br>+                         directory = <span>"data"</span>,<br>+                         nSample = <span>10</span>,<br>+                         nTF = <span>5</span>,<br>+                         DiffGenes = <span>TRUE</span>,<br>+                         BPname = c(<span>"apoptosis"</span>,<span>"proliferation of cells"</span>),<br>+                         stage = i)<br>+   listMoonlight &lt;- c(listMoonlight, list(dataDual))<br>+   save(dataDual, file = paste0(<span>"dataDual_stage"</span>,as.roman(i), <span>".Rdata"</span>))<br>+ }<br>[<span>1</span>] <span>"-----------------------------------------"</span><br>[<span>1</span>] <span>"Get TCGA data"</span><br>[<span>1</span>] <span>"-----------------------------------------"</span><br>[<span>1</span>] <span>"cancer type: BRCA"</span><br>--------------------------------------<br>o GDCquery: Searching <span>in</span> GDC database<br>--------------------------------------<br>Genome of reference: hg19<br>--------------------------------------------<br>oo Accessing GDC. This might take a while...<br>--------------------------------------------<br>ooo Project: TCGA-BRCA<br>--------------------<br>oo Filtering results<br>--------------------<br>ooo By platform<br>ooo By data.type<br>ooo By file.type<br>----------------<br>oo Checking data<br>----------------<br>ooo Checking <span>if</span> there are duplicated cases<br>Warning: There are more than one file <span>for</span> the same case. Please verify query results. You can use the command View(getResults(query)) <span>in</span> rstudio<br>ooo Checking <span>if</span> there are results <span>for</span> the query<br>-------------------<br>o Preparing output<br>-------------------<br>Error <span>in</span> `$&lt;-.data.frame`(`*tmp*`, <span>"tumor_stage"</span>, value = character(<span>0</span>)) : <br>  replacement has <span>0</span> rows, data has <span>1098</span><br>In addition: Warning message:<br>In readBin(<span>3L</span>, raw(<span>0</span>), <span>32768L</span>) :<br>  URL <span>'https://api.gdc.cancer.gov/legacy/files/?pretty=true&amp;expand=cases,cases.samples.portions.analytes.aliquots,cases.project,center,analysis,cases.samples&amp;size=15492&amp;filters=%7B%22op%22:%22and%22,%22content%22:[%7B%22op%22:%22in%22,%22content%22:%7B%22field%22:%22cases.project.project_id%22,%22value%22:[%22TCGA-BRCA%22]%7D%7D,%7B%22op%22:%22in%22,%22content%22:%7B%22field%22:%22files.data_category%22,%22value%22:[%22Gene%20expression%22]%7D%7D,%7B%22op%22:%22in%22,%22content%22:%7B%22field%22:%22files.data_type%22,%22value%22:[%22Gene%20expression%20quantification%22]%7D%7D,%7B%22op%22:%22in%22,%22content%22:%7B%22field%22:%22files.platform%22,%22value%22:[%22Illumina%20HiSeq%22]%7D%7D,%7B%22op%22:%22in%22,%22content%22:%7B%22field%22:%22files.tags%22,%22value%22:[%22unnormalized%22]%7D%7D]%7D&amp;format=JSON'</span>: Timeout of <span>60</span> seconds was reached<br></code></pre><p data-tool="mdnice编辑器"><br></p><h1 data-tool="mdnice编辑器">5. 总结</h1><p data-tool="mdnice编辑器">总体而言，MoonlightR 确实是提供了一个推断肿瘤抑制基因（TSG）或者促癌基因（OCG）的一个R包，后续新近更新版本的Moonlight2R本以为会修复一些Bug，但没想作者只是简单的将新函数加入，而没有对原先的Code 进行检查，并且新函数加入的也很粗糙，这就导致总体的体验是有点差的，或许作者也没有想到自己写的R包也会有人学习？实事求是的说，这篇示例文档写的并不太友好，更新版本的Moonlight2R尽管功能更加强大，用也是能用的，但用的就是很不爽，总体观感就不是太好。抛开代码及文档的因素不谈，这个代码流程确实很有意义，有值得学习的地方，也能够用于课题相关肿瘤驱动基因的挖掘中，但是对使用者的能力有更高的要求，能够成功Debug作者的不尽之处，给出正确的解决办法，也是对使用者的一个小考验。</p><span>- END -</span></section><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/YiFreK3B5lQ5L8Y_aMOzlA",target="_blank" rel="noopener noreferrer">原文链接</a>
