---
title: "Mouse Gene 2.0，小鼠芯片升级版"
date: 2022-12-06T06:17:05Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Acdemic"]
---
Mouse Gene 2.0，小鼠芯片升级版 by 生信菜鸟团
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><span></span><p data-tool="mdnice编辑器">曾老师很久以前讲过人类HTA2.0芯片的分析，分析的时候需要区分外显子和非编码转录本。那小鼠的呢？</p><blockquote data-tool="mdnice编辑器"><p><strong>Series GSE197794</strong></p><p>FGF-2 signaling in nasopharyngeal carcinoma modulates pericyte-macrophage crosstalk and metastasis. <span>JCI Insight</span> 2022 May 23;7(10).</p><p>PMID: 35439170</p></blockquote><p data-tool="mdnice编辑器">文章很新，正好和我的博士研究方向息息相关，正好趁分析的时候来读读文章~</p><figure data-tool="mdnice编辑器"><img data-ratio="0.7147058823529412" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8eSBcQs6ib53eN9fICp7Nj72lAKicOG8FDo2LuxJMEbUAGXhTnuYjFM0AKt5jibWqic5Y1MqHbXNWgRQ/640?wx_fmt=png" data-type="png" data-w="1020" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8eSBcQs6ib53eN9fICp7Nj72lAKicOG8FDo2LuxJMEbUAGXhTnuYjFM0AKt5jibWqic5Y1MqHbXNWgRQ/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">感兴趣的同学可以看看作者是如何建立自己的逻辑体系，从而通过一系列假设推理出以上的机制图来的。</p><p data-tool="mdnice编辑器">这部分的芯片分析只是文章的一小部分，是为了论证炎症细胞因子在FGF-2和对照组中的表达差异，因此样本选得也不多（n=6）.</p><blockquote data-tool="mdnice编辑器"><p>[原文描述]：To identify possible factors that mediate FGF-2–induced macrophage activation, we performed an inflammatory cytokine/chemokine profiling  based on genome-wide expression microarray analysis in freshly isolated pericytes in vector- and FGF-2– overexpressing tumors.</p></blockquote><p data-tool="mdnice编辑器">接下来就是代码run run run的部分了⬇</p><h2 data-tool="mdnice编辑器"><span></span><span>step01 原始数据处理</span><span></span></h2><pre data-tool="mdnice编辑器"><code><span>##加载R包</span><br><span>library</span>(GEOquery)<br><span>library</span>(Biobase)<br><span>library</span>(limma)<br><span>library</span>(oligo)<br><br><span>##获取原始数据------------</span><br><span>##数据解压到新建的文件夹中</span><br>dir.create(<span>"sampFile"</span>)<span>##创建新的文件夹</span><br>list.files()<span>##显示文件夹中的文件，及原始文件顺序</span><br>untar(<span>"../rawdata/GSE197794_RAW.tar"</span>, exdir = samPath)<span>##解压原始文件到sampath文件夹中</span><br>setwd(samPath)<br>list.files()<span>##显示文件夹中的文件，及原始文件顺序</span><br><br><br><span># 匹配临床信息 ----------------------------------------</span><br>gse &lt;- <span>"GSE197794"</span><br>gset &lt;- getGEO(gse,getGPL = <span>FALSE</span>)<br>list.files()<span>##显示文件夹中的文件，及原始文件顺序</span><br><span>###匹配临床信息</span><br>celFiles &lt;- list.celfiles(samPath) <span>##listGzipped参数：Logical. List .CEL.gz files? </span><br>celFiles<br>dat &lt;- read.celfiles(filenames = celFiles, phenoData = phenoData(gset[[<span>1</span>]]), <br>                     sampleNames = rownames(pData(gset[[<span>1</span>]])))<br>class(dat)<br><br></code></pre><p data-tool="mdnice编辑器">接下来的一步比较关键，要根据芯片设计来选参数</p><pre data-tool="mdnice编辑器"><code><span># 读取非编码转录本 ------------------------------------</span><br>Eset &lt;- rma(dat, target = <span>"core"</span>)  <br>featureData(Eset) &lt;- getNetAffx(Eset, <span>"transcript"</span>)<br>save(dat,Eset,file = paste0(<span>"../Main_ GSE197794/"</span>,gse,<span>"_transcript.Rdata"</span>))<br></code></pre><h2 data-tool="mdnice编辑器"><span></span><span>step02 芯片注释</span><span></span></h2><pre data-tool="mdnice编辑器"><code><span># 2.1 安装和加载包 ------------------------------------</span><br><span># BiocManager::install("genefilter")</span><br><span># BiocManager::install("pd.mogene.2.0.st")</span><br><span># BiocManager::install("mogene20sttranscriptcluster.db")</span><br><span>library</span>(mogene20sttranscriptcluster.db)<br><span>library</span>(pd.mogene.2.0.st)<br><span>library</span>(genefilter)<br><br><span>##加载数据</span><br>gse &lt;- <span>"GSE197794"</span><br>load(paste0(<span>"../Main_ GSE197794/"</span>,gse,<span>"_transcript.Rdata"</span>))<br>dim(Eset)<br><br><br><span># 2.2 表达量计算并获取表达矩阵 --------------------------</span><br>medrs &lt;- apply(exprs(Eset), <span>1</span>, median)<br>prob_set &lt;- findLargest(featureNames(Eset), testStat = medrs, data = <span>"mogene20sttranscriptcluster.db"</span>)<br>Eset2 &lt;- Eset[prob_set ,]<br>dim(Eset2)<br><br>ExprSet &lt;- exprs(Eset2)<br>ExprSet[<span>1</span>:<span>5</span>,<span>1</span>:<span>5</span>] <span>##查看行名ID</span><br>range(ExprSet)  <span>##根据值的范围初步判断是否已被log</span><br>boxplot(ExprSet,las=<span>2</span>)<br><br><span># 1.3 获取分组信息 ------------------------------------</span><br><span>library</span>(GEOquery)<br>pd &lt;- pData(Eset2)<br>phe &lt;- pd<br>table(phe$title)<br>Group &lt;- ifelse(grepl(<span>"Control "</span>,phe$title),<span>"control"</span>,<span>"case"</span>)<br>table(Group)<br>group_list &lt;- Group<br><br><span># 1.4 探针注释 ---------------------------------------</span><br><span># https://www.bioinfo-scrounger.com/archives/398/</span><br><span># https://blog.csdn.net/shy_321/article/details/117968423</span><br><span>##也可以下载并提取GPL平台数据</span><br>options(timeout = <span>"1000000000"</span>)<br>gpl &lt;- getGEO(<span>"GPL16570"</span>, destdir=<span>"."</span>) <span>##根据GPL号下载的是芯片设计的信息</span><br>GPL=gpl@dataTable@table<br><br><span>library</span>(dplyr)<br><span>library</span>(stringr)<br>colnames(GPL)<br><span>##得到探针-基因ID-genesymbol的注释文件</span><br>annotation &lt;- as.data.frame(GPL) %&gt;%<br>  dplyr::select(<span>"ID"</span>,<span>"gene_assignment"</span>)<span>##跟进GPL的情况选择</span><br>annotation$gene_symbol=trimws(str_split(annotation$gene_assignment,<span>'//'</span>,simplify = <span>T</span>)[,<span>2</span>]) <br><br><span>##探针基因注释</span><br>ids=annotation[,c(<span>1</span>,<span>3</span>)]<br>colnames(ids)=c(<span>"probe_id"</span>,<span>"symbol"</span>)<br><br><span># 基因过滤</span><br>ids=ids[ids$symbol != <span>''</span>,]  <span>##去除空值</span><br>ids=ids[ids$symbol != <span>'---'</span>,] <span>##去除特殊字符串</span><br>dim(ids);dim(ExprSet)<br><br>exp &lt;- ExprSet<br><span>#过滤表达矩阵</span><br>exp &lt;- as.data.frame(exp)<br>exp &lt;- exp[rownames(exp) %<span>in</span>% ids$probe_id,]<br>dim(exp)<br>exp[<span>1</span>:<span>5</span>,<span>1</span>:<span>5</span>]<br><br><span>#ids过滤探针</span><br>ids &lt;- ids[match(rownames(exp),ids$probe_id),]<br>dim(ids)<br>ids[<span>1</span>:<span>2</span>,]<br><br>ids$median=apply(exp,<span>1</span>,median) <span>#ids新建median这一列，列名为median，同时对矩阵按行操作，取每一行的中位数，将结果给到median这一列的每一行</span><br>ids=ids[order(ids$symbol,ids$median,decreasing = <span>T</span>),]<span>#对ids$symbol按照ids$median中位数从大到小排列的顺序排序，将对应的行赋值为一个新的ids</span><br>ids=ids[!duplicated(ids$symbol),]<span>#将symbol这一列取取出重复项，'!'为否，即取出不重复的项，去除重复的gene ，保留每个基因最大表达量结果s</span><br>exp=exp[ids$probe_id,] <span>#新的ids取出基因这一列，将dat按照取出的这一列中的每一行组成一个新的dat</span><br>rownames(exp)=ids$symbol<span>#把ids的symbol这一列中的每一行给矩阵作为矩阵的行名</span><br></code></pre><pre data-tool="mdnice编辑器"><code>&gt; exp[1:4,1:4]<br>       GSM5930496 GSM5930497 GSM5930498 GSM5930499<br>Zzz3     7.487992   7.424313   7.546915   7.665754<br>Zzef1    7.810067   7.819584   7.823778   7.816618<br>Zyx      8.764237   8.662044   8.763678   8.466913<br>Zyg11b   7.570928   7.646608   7.245897   7.598484<br>&gt; exp[<span>'Actb'</span>,] <span>##注意物种基因的写法</span><br>     GSM5930496 GSM5930497 GSM5930498 GSM5930499 GSM5930500 GSM5930501<br>Actb   10.71734   10.68009   10.68059   10.53113   10.55996   10.55179<br>&gt; exp[<span>'Gapdh'</span>,]<br>      GSM5930496 GSM5930497 GSM5930498 GSM5930499 GSM5930500 GSM5930501<br>Gapdh   8.339866   8.355468   8.585903   8.320542    8.03732   8.362418<br></code></pre><p data-tool="mdnice编辑器">保存数据</p><pre data-tool="mdnice编辑器"><code>dat &lt;- exp<br>getwd()<br>save(gse,dat,group_list,ids,file = paste0(<span>"../Main_ GSE197794/"</span>,<span>"step2_"</span>,gse, <span>"_Out.Rdata"</span>))<br></code></pre><h2 data-tool="mdnice编辑器"><span></span><span>step03 差异分析及可视化</span><span></span></h2><pre data-tool="mdnice编辑器"><code>rm(list = ls())<br>gse &lt;- <span>"GSE197794"</span><br><br>load( paste0(<span>"../Main_ GSE197794/"</span>,<span>"step2_"</span>,gse, <span>"_Out.Rdata"</span>))<br>exp &lt;- dat<br><span># 差异分析 --------------------------------------------------------------------</span><br><span>library</span>(limma)<br>design &lt;- model.matrix(~<span>0</span>+factor(group_list))<br>colnames(design)=levels(factor(group_list))<br>rownames(design)=colnames(exp)<br>head(design)<br>contrast.matrix&lt;-makeContrasts(<span>"case -control"</span>,<br>                               levels = design)<br>contrast.matrix <span>##这个矩阵声明，我们要把 case 组跟 control 进行差异分析比较</span><br><br>deg = <span>function</span>(exprSet,design,contrast.matrix){<br>  <span>##step1</span><br>  fit &lt;- lmFit(exprSet,design)<br>  <span>##step2</span><br>  fit2 &lt;- contrasts.fit(fit, contrast.matrix)<br>  <span>##这一步很重要，大家可以自行看看效果</span><br>  <br>  fit2 &lt;- eBayes(fit2)  <span>## default no trend !!!</span><br>  <span>##eBayes() with trend=TRUE</span><br>  <span>##step3</span><br>  tempOutput = topTable(fit2, coef=<span>1</span>, n=<span>Inf</span>)<br>  nrDEG = na.omit(tempOutput)<br>  <span>#write.csv(nrDEG2,"limma_notrend.results.csv",quote = F)</span><br>  head(nrDEG)<br>  <span>return</span>(nrDEG)<br>}<br>deg = deg(exp,design,contrast.matrix)<br>head(deg) <br></code></pre><p data-tool="mdnice编辑器">差异分析结束，接下来是作图</p><pre data-tool="mdnice编辑器"><code><span># 可视化拼图 -------------------------------------------------------------------</span><br><span>###主成分分析</span><br>dat=as.data.frame(t(exp))<br><span>library</span>(FactoMineR)<span>#画主成分分析图需要加载这两个包</span><br><span>library</span>(factoextra) <br><span># pca的统一操作走起</span><br>dat.pca &lt;- PCA(dat, graph = <span>FALSE</span>)<br>p1 &lt;- fviz_pca_ind(dat.pca,<br>                   geom.ind = <span>"point"</span>, <span># show points only (nbut not "text")</span><br>                   col.ind = group_list, <span># color by groups</span><br>                   <span>#palette = c("#00AFBB", "#E7B800"),</span><br>                   addEllipses = <span>TRUE</span>, <span># Concentration ellipses</span><br>                   legend.title = <span>"Groups"</span>,<br>                   title = <span>"GSE197794_PCA"</span><br>)<br>p1<br><br><span>#热图 </span><br>cg=names(tail(sort(apply(exp,<span>1</span>,sd)),<span>500</span>))<br><span>if</span>(!<span>require</span>(pheatmap))install.packages(<span>"pheatmap"</span>)<br><span>library</span>(pheatmap)<br>n=exp[cg,]<br>annotation_col=data.frame(group=group_list)<br>rownames(annotation_col)=colnames(n) <br>p21 &lt;- pheatmap(n,<br>               show_colnames =<span>T</span>,<br>               show_rownames = <span>F</span>,<br>               cluster_cols = <span>F</span>,<br>               annotation_col=annotation_col,<br>               scale = <span>"row"</span>)<br><span>require</span>(ggplotify)<br>p21= as.ggplot(p21)<br>p21<br><br><span>####根据文献复现一下</span><br>infammatory_cytokine &lt;- c(<span>"Ccl11"</span>,<span>"Cxcl14"</span>,<span>"Ccl24"</span>,<span>"Il18"</span>,<span>"Tnfsf18"</span>,<span>"Tnfaip6"</span>,<span>"Il16"</span>,<span>"Il15"</span>,<span>"Tnfsf4"</span>,<span>"Ccl8"</span>,<span>"Xcl1"</span>,<span>"Ifna14"</span>,<br>                          <span>"Il19"</span>,<span>"Cxcl12"</span>,<span>"Ifna2"</span>,<span>"Tnfaip2"</span>,<span>"Ccl25"</span>,<span>"Ccl6"</span>,<span>"Il17f"</span>,<span>"Csf2"</span>)<br>n=exp[infammatory_cytokine,]<br>annotation_col=data.frame(group=group_list)<br>rownames(annotation_col)=colnames(n) <br>p22 &lt;- pheatmap(n,<br>               show_colnames =<span>T</span>,<br>               show_rownames = <span>T</span>,<br>               cluster_cols = <span>F</span>,<br>               fontsize_row = <span>12</span>,<br>               fontsize_col = <span>15</span>,<br>               annotation_col=annotation_col,<br>               scale = <span>"row"</span>)<br>p22= as.ggplot(p22)<br>p22<br><br><br><span>##火山图</span><br><span>if</span>(<span>T</span>){<br>  nrDEG=deg<br>  head(nrDEG)<br>  <span>attach</span>(nrDEG)<br>  plot(logFC,-log10(P.Value))<br>  <span>library</span>(ggpubr)<br>  df=nrDEG<br>  df$v= -log10(P.Value) <span>#df新增加一列'v',值为-log10(P.Value)</span><br>  <br>  ggscatter(df, x = <span>"logFC"</span>, y = <span>"v"</span>,size=<span>0.5</span>)<br>  <br>  df$g=ifelse(df$P.Value&gt;<span>0.05</span>,<span>'stable'</span>, <span>#if 判断：如果这一基因的P.Value&gt;0.01，则为stable基因</span><br>              ifelse( df$logFC &gt;<span>0.5</span>,<span>'up'</span>, <span>#接上句else 否则：接下来开始判断那些P.Value&lt;0.01的基因，再if 判断：如果logFC &gt;1.5,则为up（上调）基因</span><br>                      ifelse( df$logFC &lt; -<span>0.5</span>,<span>'down'</span>,<span>'stable'</span>) )<span>#接上句else 否则：接下来开始判断那些logFC &lt;1.5 的基因，再if 判断：如果logFC &lt;1.5，则为down（下调）基因，否则为stable基因</span><br>  )<br>  table(df$g)<br>  df$name=rownames(df)<br>  head(df)<br>  ggscatter(df, x = <span>"logFC"</span>, y = <span>"v"</span>,size=<span>0.5</span>,color = <span>'g'</span>)<br> p3 &lt;- ggscatter(df, x = <span>"logFC"</span>, y = <span>"v"</span>, color = <span>"g"</span>,size = <span>0.5</span>,<br>            label = <span>"name"</span>, repel = <span>T</span>,<br>            <span>#label.select = rownames(df)[df$g != 'stable'] ,</span><br>            label.select = c(<span>"Ccl11"</span>,<span>"Cxcl14"</span>), <span>#挑选一些基因在图中显示出来</span><br>            font.label = c(<span>12</span>,<span>'black'</span>),<br>            palette = c(<span>"#00AFBB"</span>, <span>"#E7B800"</span>, <span>"#FC4E07"</span>) )<br>p3<br>  <br>  ggscatter(df, x = <span>"AveExpr"</span>, y = <span>"logFC"</span>,size = <span>0.2</span>)<br>  df$p_c = ifelse(df$P.Value&lt;<span>0.001</span>,<span>'p&lt;0.001'</span>,<br>                  ifelse(df$P.Value&lt;<span>0.01</span>,<span>'0.001&lt;p&lt;0.01'</span>,<span>'p&gt;0.01'</span>))<br>  table(df$p_c )<br>p4 &lt;-   ggscatter(df,x = <span>"AveExpr"</span>, y = <span>"logFC"</span>, color = <span>"p_c"</span>,size=<span>0.2</span>, <br>                        palette = c(<span>"green"</span>, <span>"red"</span>, <span>"black"</span>) )<br>p4  <br>  }<br></code></pre><p data-tool="mdnice编辑器">最后来个拼盘，便于我们看图说话~</p><pre data-tool="mdnice编辑器"><code><span>##拼图</span><br><span>library</span>(patchwork)<br>patchwork &lt;- p22 | (p1 / p3)<br>patchwork+plot_annotation(tag_levels = <span>"A"</span>)<br>ggsave(patchwork,filename = <span>"../output//patchwork.png"</span>,width = <span>15</span>,height = <span>15</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7132043734230445" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8eSBcQs6ib53eN9fICp7Nj7AUNIT3sYJ2I7lghjzwl89S63VH72oaWQxCU7Kw1EOmRttzqhOc5Uhg/640?wx_fmt=png" data-type="png" data-w="1189" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los8eSBcQs6ib53eN9fICp7Nj7AUNIT3sYJ2I7lghjzwl89S63VH72oaWQxCU7Kw1EOmRttzqhOc5Uhg/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">好嘞，这个芯片的小插曲就结束了，和文章的结果一致，Cxcl14 gene在FGF-2过表达组中上调。</p><blockquote data-tool="mdnice编辑器"><p>这篇文章真的值得一看，推荐给大家！</p></blockquote><span></span></section><p><br></p><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/QqatakCCRxE69uOTtVEZpg",target="_blank" rel="noopener noreferrer">原文链接</a>
