---
title: "PRIMUS，一个专门针对肿瘤单细胞转录组数据整合的算法"
date: 2023-01-29T14:03:25Z
draft: ["false"]
tags: [
  "fetched",
  "单细胞天地"
]
categories: ["Acdemic"]
---
PRIMUS，一个专门针对肿瘤单细胞转录组数据整合的算法 by 单细胞天地
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><p data-tool="mdnice编辑器">Poisson scRNA Integration of Mixed Unknown Signals（PRIMUS）：混杂未知信号的泊松scRNA数据整合</p><h1 data-tool="mdnice编辑器">背景介绍</h1><p data-tool="mdnice编辑器">PRIMUS 来源于一篇Sci Adv的文献Longitudinal single-cell RNA-seq analysis reveals stress-promoted chemoresistance in metastatic ovarian cancer[1]。作者期望通过研究化疗前后卵巢癌样本的单细胞转录组层面的变化，然而由于卵巢癌肿瘤特有的异质性，不同肿瘤之间细胞成分差异很大，多样本之间整合存在困难。而基于以往的Seurat v3, Harmony, LIGER, mnnCorrect, and fastMNN 以及三种Bulk RNAseq的整合方法（ComBat、ComBat-seq，and limma）整合仍然不能很好的解决这个问题。基于此，作者团队开发了PRIMUS。</p><p data-tool="mdnice编辑器">PRIMUS是一种整体聚类方法，能够从 scRNA-seq 数据中识别表型细胞群，同时考虑日期来源(例如患者，样本，数据集)特定的组分以及技术噪声。PRIMUS 采用双线性泊松回归模型，将表达数据同时分解为明确的干扰因素（defined nuisance factors）、未定义的细胞表型及其相应的转录组学特征。</p><blockquote data-tool="mdnice编辑器"><p>[1]  Zhang K.et al.Longitudinal single-cell RNA-seq analysis reveals stress-promoted chemoresistance in metastatic ovarian cancer. Sci Adv. 2022 Feb 25;8(8):eabm1831. doi: 10.1126/sciadv.abm1831. Epub 2022 Feb 23. PMID: 35196078</p></blockquote><figure data-tool="mdnice编辑器"><img data-ratio="0.7454931071049841" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicLWxG3A6KyWDUKMATWIIicL6PPibcy0x9q5f7anBQXSjPesPGSUwcnLwPeMQuCxsx0reVChqDoFkxQ/640?wx_fmt=png" data-type="png" data-w="943" title="图1 文献附图S2 A-F  描述PRIMUS 的整合效果" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicLWxG3A6KyWDUKMATWIIicL6PPibcy0x9q5f7anBQXSjPesPGSUwcnLwPeMQuCxsx0reVChqDoFkxQ/640?wx_fmt=png"><figcaption><span>图1 文献附图S2 A-F  描述PRIMUS 的整合效果</span></figcaption></figure><h1 data-tool="mdnice编辑器">PRIMUS 安装</h1><p data-tool="mdnice编辑器">PRIMUS 存储在Github，因此，可以使用devtools 安装。</p><pre data-tool="mdnice编辑器"><code>devtools::install_github(<span>"KaiyangZ/PRIMUS"</span>)<br></code></pre><p data-tool="mdnice编辑器">Vignette 文档链接：https://htmlpreview.github.io/?https://github.com/KaiyangZ/PRIMUS/blob/master/vignettes/quickstart.html</p><h1 data-tool="mdnice编辑器">PRIMUS 流程使用</h1><h2 data-tool="mdnice编辑器"><span>数据准备</span><span> </span></h2><p data-tool="mdnice编辑器">在上期我们说过，一般情况下作者都会提供包的使用方法及示例数据，但事情总有例外。PRIMUS  R包的示例数据就如此。作者提供的Counts 文件和Meta 文件都不明原因丢失了，这个时候就需要自己下载其他的示例文件。在这里，我选择作者文章中用到的胰腺癌数据（https://github.com/JinmiaoChenLab/Batch-effect-removal-benchmarking/tree/master/Data/dataset4）作为示例数据进行演示。</p><p data-tool="mdnice编辑器"><img data-ratio="0.45353982300884954" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicLWxG3A6KyWDUKMATWIIicLeZuE38iaoqTERZROSgPiafjWURs3JnTLibXJ1r4fFjTuJn3CHHlBiciagibQ/640?wx_fmt=png" data-type="png" data-w="452" title="图2   Methods 截图" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicLWxG3A6KyWDUKMATWIIicLeZuE38iaoqTERZROSgPiafjWURs3JnTLibXJ1r4fFjTuJn3CHHlBiciagibQ/640?wx_fmt=png"><span>图2   Methods 截图</span></p><p data-tool="mdnice编辑器">而Github 页面显示，该数据存放在Docker，</p><p data-tool="mdnice编辑器"><img data-ratio="0.35853131749460043" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicLWxG3A6KyWDUKMATWIIicL5iaZBELcCMMolwpBebFsibHFA6LMGkuKSFEQplxaSIfYgwprE8vlxphQ/640?wx_fmt=png" data-type="png" data-w="926" title="图3 Github 页面截图" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicLWxG3A6KyWDUKMATWIIicL5iaZBELcCMMolwpBebFsibHFA6LMGkuKSFEQplxaSIfYgwprE8vlxphQ/640?wx_fmt=png"><span>图3 Github 页面截图</span></p><p data-tool="mdnice编辑器">点开链接，发现这里用到了sudo 命令，一般情况下，我们使用的linux 服务器很难有sudo 权限。在我用的服务器中，singularity 替代了docker 的用法。</p><ul data-tool="mdnice编辑器"><li><section>因此，这里可以使用singularity 获取胰腺癌数据。singularity 学习地址（http://www.xtaohub.com/Container-Tech/Singularity-in-nutshell.html）</section></li></ul><pre data-tool="mdnice编辑器"><code>singularity build --sandbox ./batcheffect  docker://jinmiaochenlab/batch-effect-removal-benchmarking<br>cp  ./batcheffect/batch_effect/dataset4 ./dataset4<br></code></pre><h2 data-tool="mdnice编辑器"><span>scRNA 前期处理</span><span> </span></h2><h3 data-tool="mdnice编辑器">数据读入</h3><p data-tool="mdnice编辑器">该胰腺癌数据来源于A benchmark of batch-effect correction methods for single-cell RNA sequencing data" (https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1850-9，感兴趣可以阅读原文献。在这里，我们先读入数据，并进行进行前期的简单处理，</p><pre data-tool="mdnice编辑器"><code><span>library</span>(SingleCellExperiment)<br><span>library</span>(scuttle)<br><span>library</span>(data.table)<br><span>library</span>(dplyr)<br>rm(list = ls())<br><span># 读入数据</span><br>counts = fread(<span>'./dataset4/myData_pancreatic_5batches.txt.gz'</span>) %&gt;% as.data.frame()<br>rownames(counts) &lt;- counts$V1; counts &lt;- counts[,<span>2</span>:ncol(counts)]<br>dim(counts)<br><span>#&gt; [1] 15558 14767</span><br>head(counts)[<span>1</span>:<span>5</span>,<span>1</span>:<span>2</span>]<br><span>#&gt;        human1_lib1.final_cell_0001 human1_lib1.final_cell_0002</span><br><span>#&gt;  A1BG                             0                           0</span><br><span>#&gt;  A1CF                             4                           0</span><br><span>#&gt;  A2M                              0                           0</span><br><span>#&gt;  A2ML1                            0                           0</span><br><span>#&gt;  A4GALT                           0                           0</span><br>meta = fread(<span>'./dataset4/mySample_pancreatic_5batches.txt.gz'</span>,) %&gt;% as.data.frame()<br>rownames(meta) &lt;- meta$V1; meta &lt;- meta[,<span>2</span>:ncol(meta)]<br>dim(meta)<br><span>#&gt; [1] 14767     5</span><br>colnames(meta)<br><span>#&gt;  [1] "batch"         "batchlb"       "celltype_orig" "cellname"      "celltype"</span><br>head(meta)[<span>1</span>:<span>5</span>,<span>1</span>:<span>3</span>]<br><span>#&gt;                            batch  batchlb celltype_orig</span><br><span>#&gt; human1_lib1.final_cell_0001     1 Baron_b1        acinar</span><br><span>#&gt; human1_lib1.final_cell_0002     1 Baron_b1        acinar</span><br><span>#&gt; human1_lib1.final_cell_0003     1 Baron_b1        acinar</span><br><span>#&gt; human1_lib1.final_cell_0004     1 Baron_b1        acinar</span><br><span>#&gt; human1_lib1.final_cell_0005     1 Baron_b1        acinar</span><br></code></pre><h3 data-tool="mdnice编辑器">细胞抽样</h3><p data-tool="mdnice编辑器">读入数据后简单查看了下数据，不难看出，该胰腺癌数据含有14767个细胞，5个样本，为了精简流程，减少运行时间，后续对样本进行随机抽样，以减少数据量。</p><pre data-tool="mdnice编辑器"><code><span>### 抽样前查看各细胞分布 ###</span><br>table(meta$celltype, meta$batch) <br><span>#&gt;                  1    2    3    4    5</span><br><span>#&gt;   acinar        958  219  185    6    0</span><br><span>#&gt;   alpha        2326  812  886  190  886</span><br><span>#&gt;   beta         2525  448  270  111  472</span><br><span>#&gt;   delta         601  193  114    9   49</span><br><span>#&gt;   ductal       1077  245  386   96    0</span><br><span>#&gt;   endothelial   252   21   16    0    0</span><br><span>#&gt;   epsilon        18    3    7    0    0</span><br><span>#&gt;   gamma         255  101  197   18   85</span><br><span>#&gt;   macrophage     55    0    0    0    0</span><br><span>#&gt;   mast           25    0    7    0    0</span><br><span>#&gt;   mesenchymal     0   80    0   27    0</span><br><span>#&gt;   MHC class II    0    0    5    0    0</span><br><span>#&gt;   schwann        13    0    0    0    0</span><br><span>#&gt;   stellate      457    0   54    0    0</span><br><span>#&gt;   t_cell          7    0    0    0    0</span><br><br><span>### 由于部分细胞数稀少，因此这里考虑按每样本抽300细胞作为示例文件 ###，</span><br>allCells=rownames(meta)<br>allbatch = levels(factor(meta$batch))<br>choose_Cells = unlist(lapply(allbatch, <span>function</span>(x){<br>  cgCells = allCells[meta$batch == x]<br>  cg=sample(cgCells,<span>300</span>,replace = <span>F</span>)  <br>  cg<br>}))<br>Ct &lt;- counts[,choose_Cells]<br>mt &lt;- meta[choose_Cells,]<br><span>### 查看抽样结果 ##</span><br>table(mt$celltype, mt$batch) <br><span>#&gt;                  1   2   3   4   5</span><br><span>#&gt;   acinar        21  28  32   5   0</span><br><span>#&gt;   alpha         93 116 120 122 178</span><br><span>#&gt;   beta          96  61  42  78  98</span><br><span>#&gt;   delta         18  25  16   5   7</span><br><span>#&gt;   ductal        34  38  49  62   0</span><br><span>#&gt;   endothelial   13   4   3   0   0</span><br><span>#&gt;   epsilon        2   0   1   0   0</span><br><span>#&gt;   gamma          7  17  29  10  17</span><br><span>#&gt;   macrophage     3   0   0   0   0</span><br><span>#&gt;   mast           1   0   2   0   0</span><br><span>#&gt;   mesenchymal    0  11   0  18   0</span><br><span>#&gt;   MHC class II   0   0   1   0   0</span><br><span>#&gt;   schwann        1   0   0   0   0</span><br><span>#&gt;   stellate      11   0   5   0   0</span><br></code></pre><p data-tool="mdnice编辑器">T 细胞因为数量实在是太少了，一个都没有抽到，不过不影响我们后续演示。</p><h3 data-tool="mdnice编辑器">创建SingleCellExperiment 对象</h3><p data-tool="mdnice编辑器">PRIMUS 全程使用的是 SingleCellExperiment 对象，这里也使用该对象。</p><pre data-tool="mdnice编辑器"><code>simData = logNormCounts(SingleCellExperiment(assays = list(counts = Ct )))<br>simData<br><span>#&gt;  class: SingleCellExperiment </span><br><span>#&gt;  dim: 15558 1500 </span><br><span>#&gt;  metadata(0):</span><br><span>#&gt;  assays(2): counts logcounts</span><br><span>#&gt;  rownames(15558): A1BG A1CF ... ZZEF1 ZZZ3</span><br><span>#&gt;  rowData names(0):</span><br><span>#&gt;  colnames(1500): human3_lib3.final_cell_0151 human3_lib4.final_cell_0497 ... Sample_396 Sample_698</span><br><span>#&gt;  colData names(1): sizeFactor</span><br><span>#&gt;  reducedDimNames(0):</span><br><span>#&gt;  mainExpName: NULL</span><br><span>#&gt;  altExpNames(0):</span><br><span># 这里将sizeFactor 存放到mt数据中，后续需要该数据</span><br>mt$sizeFactor  &lt;- colData(simData)$sizeFactor<br></code></pre><p data-tool="mdnice编辑器">作者在原文中提及的是，胰腺癌数据使用PRISM 包（https://bitbucket.org/anthakki/prism/src/master/）计算 <code>scaling factor</code> ,模拟数据使用 <code>logNormCounts</code> 计算的结果。这里另外计算实在是有点烦，我直接用了<code>logNormCounts</code>的结果。</p><h3 data-tool="mdnice编辑器">umap 可视化</h3><p data-tool="mdnice编辑器">展示未进行 PRIMUS 的umap结果。</p><pre data-tool="mdnice编辑器"><code><span>library</span>(uwot)<br><span>library</span>(ggplot2)<br><span>library</span>(cowplot)<br><br>umap.coor.raw &lt;- uwot::umap(t(simData@assays@data$logcounts),<br>                            n_components = <span>2</span>, metric = <span>"cosine"</span>,<br>                            min_dist = <span>0.3</span>, n_neighbors = <span>30</span>)<br><br>mt$UMAP_raw_1 = umap.coor.raw[, <span>1</span>]<br>mt$UMAP_raw_2 = umap.coor.raw[, <span>2</span>]<br><br>umap.raw.g &lt;- ggplot(mt, aes(x= UMAP_raw_1, y=UMAP_raw_2, color = celltype) ) +<br>  geom_point(size = <span>0.5</span>) +<br>  theme_classic() +<br>  xlab(<span>"UMAP_1"</span>) + <br>  ylab(<span>"UMAP_2"</span>) +<br>  ggtitle(<span>"colored by phenotypic cell group"</span>) +<br>  guides(colour = guide_legend(override.aes = list(size=<span>3</span>)), shape = guide_legend(override.aes = list(size=<span>3</span>)))<br><br>umap.raw.s &lt;- ggplot(mt, aes(x= UMAP_raw_1, y=UMAP_raw_2, color = batch) ) +<br>  geom_point(size = <span>0.5</span>) +<br>  theme_classic() +<br>  xlab(<span>"UMAP_1"</span>) + <br>  ylab(<span>"UMAP_2"</span>) +<br>  ggtitle(<span>"colored by sample"</span>) +<br>  guides(colour = guide_legend(override.aes = list(size=<span>3</span>)), shape = guide_legend(override.aes = list(size=<span>3</span>)))<br><br>cowplot::plot_grid(umap.raw.s, umap.raw.g, nrow = <span>1</span>)<br></code></pre><p><span>图4 初始 umap 展示（sample+celltype）</span></p><figure data-tool="mdnice编辑器"><p><br></p><figcaption><br></figcaption></figure><h2 data-tool="mdnice编辑器"><span>PRIMUS 运算</span><span> </span></h2><p data-tool="mdnice编辑器">runPrimus函数从scRNA-seq数据中识别表型细胞群，同时考虑到干扰因素（这里是指样本的batch effect）。作为输入，runPrimus需要原始计数矩阵（matrix）、指定样本标签的邻接矩阵和每个细胞的大小系数。runPrimus返回一个包含识别的细胞Cluster、每个群的去噪轮廓中心点（the denoised profile centroids）、样本特定效应和模型拟合成本的列表（List）。我们可以通过多次随机重启运行PRIMUS，并选择最佳拟合（成本最低的那个）。</p><h3 data-tool="mdnice编辑器">运算</h3><pre data-tool="mdnice编辑器"><code><span>library</span>(PRIMUS)<br><br><span># 将样本标签转换为邻近矩阵</span><br>labels2weights &lt;- <span>function</span>(L, levels = sort(unique(L)))<br>  <span>return</span> (outer(L, levels, `==`) + <span>0L</span>)<br><br>D = t(labels2weights(mt$batch))<br><br><span># 随机运行PRIMUS 5次</span><br><span># k是聚类的数量。用户可以设置不同的值，并根据BIC选择最佳的k。这里先设置k =5 </span><br>fits &lt;- list()<br><br>Ct &lt;- as.matrix(Ct)<br><span>for</span> (i <span>in</span> seq.int(<span>5</span>) ){<br>  set.seed(i * <span>1234</span>)<br>    fits[[i]] = runPrimus(Y = Ct, D = D, g = mt$sizeFactor, k = <span>5</span>, max.iter = <span>100</span>)<br>}<br><br>fit = fits[[which.min(sapply(fits, <span>function</span>(x) x$cost))]]<br><br><span>library</span>(mclust)<br><br>adjustedRandIndex(fit$L, mt$celltype)<br><span>#&gt;  [1] 0.8846919    ## 模拟数据的结果是1，0.8 的结果 也很高了   </span><br>Z = sapply(seq.int(ncol(Ct)), <span>function</span>(i) primus_centroid(A = fit$X %*% D[, i, drop = <span>F</span>],<br>                                                          B = Ct[, i, drop = <span>F</span>], g = mt$sizeFactor[i]))<br><br><span># take square root for variance-stabilizing</span><br>Z.sqrt = sqrt(Z)<br></code></pre><h3 data-tool="mdnice编辑器">可视化</h3><pre data-tool="mdnice编辑器"><code><span>#可视化</span><br><br><br>umap.coor.Z &lt;- uwot::umap(t(Z.sqrt), n_components = <span>2</span>,<br>                          metric = <span>"cosine"</span>,  min_dist = <span>0.3</span>,<br>                          n_neighbors = <span>30</span>, <br>                          y = as.factor(fit$L),<br>                          target_weight = <span>0.05</span>)<br><br>mt$Z_UMAP_1 = umap.coor.Z[, <span>1</span>]<br>mt$Z_UMAP_2 = umap.coor.Z[, <span>2</span>]<br>mt$cluster = as.factor(fit$L)<br>mt$celltype<br>umap.Z.g &lt;- ggplot(mt, aes(x= Z_UMAP_1, y=Z_UMAP_2, color = celltype) ) +<br>  geom_point(size = <span>0.5</span>) +<br>  theme_classic() +<br>  ggtitle(<span>"colored by phenotypic cell group"</span>) + <br>  guides(colour = guide_legend(override.aes = list(size=<span>3</span>)), shape = guide_legend(override.aes = list(size=<span>3</span>)))<br><br>umap.Z.s &lt;- ggplot(mt, aes(x= Z_UMAP_1, y=Z_UMAP_2, color = batch) ) +<br>  geom_point(size = <span>0.5</span>) +<br>  theme_classic() +<br>  ggtitle(<span>"colored by sample"</span>) + <br>  guides(colour = guide_legend(override.aes = list(size=<span>3</span>)), shape = guide_legend(override.aes = list(size=<span>3</span>)))<br><br>umap.Z.c &lt;- ggplot(mt, aes(x= Z_UMAP_1, y=Z_UMAP_2, color = cluster) ) +<br>  geom_point(size = <span>0.5</span>) +<br>  theme_classic() +<br>  ggtitle(<span>"colored by PRIMUS identified clusters"</span>) +<br>  guides(colour = guide_legend(override.aes = list(size=<span>3</span>)), shape = guide_legend(override.aes = list(size=<span>3</span>)) )<br><br>cowplot::plot_grid(umap.Z.s, umap.Z.g, umap.Z.c, nrow = <span>1</span>)<br><br></code></pre><p data-tool="mdnice编辑器"><img data-ratio="0.26917613636363635" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicLWxG3A6KyWDUKMATWIIicLgxtjiaxB8EhsgUjP46whd5ZPuIlIsjbMZLcw1ISSYpYnPGM6gf0m7Ww/640?wx_fmt=png" data-type="png" data-w="1408" title="图5  k=5 时 umap图" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicLWxG3A6KyWDUKMATWIIicLgxtjiaxB8EhsgUjP46whd5ZPuIlIsjbMZLcw1ISSYpYnPGM6gf0m7Ww/640?wx_fmt=png"><span>图5  k=5 时 umap图</span></p><p data-tool="mdnice编辑器">k =5 时，几个大类的细胞能够很好的区分开，而少部分数量很少的细胞则成散点样在大细胞群周围分布。若设置k =8 则：</p><pre data-tool="mdnice编辑器"><code>adjustedRandIndex(fit$L, mt$celltype)<br><span>#&gt;  [1] 0.6460315</span><br></code></pre><p data-tool="mdnice编辑器">随后可视化（k=8）</p><p data-tool="mdnice编辑器"><img data-ratio="0.27586206896551724" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicLWxG3A6KyWDUKMATWIIicLbTTnhuz1N5tYx9LoBTW86Alre51xs5FEZ5vyC0pnibMfZMQfh15FVJw/640?wx_fmt=png" data-type="png" data-w="1276" title="图6  k=8 时umap 图" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicLWxG3A6KyWDUKMATWIIicLbTTnhuz1N5tYx9LoBTW86Alre51xs5FEZ5vyC0pnibMfZMQfh15FVJw/640?wx_fmt=png"><span>图6  k=8 时umap 图</span></p><p data-tool="mdnice编辑器">k=8 时，明显聚类的效果参差，但显然，大细胞群还是很好的区分开了，但小部分细胞群仍有聚集。</p><h1 data-tool="mdnice编辑器">结束语</h1><p data-tool="mdnice编辑器">本篇简要介绍了文献：Longitudinal single-cell RNA-seq analysis reveals stress-promoted chemoresistance in metastatic ovarian cancer中开发的新多样本整合R包PRIMUS。从出发点来说，该包解决了多样本异质性的问题，某些细胞群只在单个样本中有，而在另一个样本中无，这种情况下的样本整合。根据原文献补充数据上的描述，Seurat v3 以及Harmony在这种情况下处理效果一般，由此说明PRIMUS 整合的优势。</p><p data-tool="mdnice编辑器"><img data-ratio="0.20309278350515464" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicLWxG3A6KyWDUKMATWIIicLgjkOic5YmVyrJTMOGiaTJrvurB3Kw56YSOAuBNjmCSvibWUbugg9uzZ0Q/640?wx_fmt=png" data-type="png" data-w="970" title="图7 文献附图S2 I" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicLWxG3A6KyWDUKMATWIIicLgjkOic5YmVyrJTMOGiaTJrvurB3Kw56YSOAuBNjmCSvibWUbugg9uzZ0Q/640?wx_fmt=png"><span>图7 文献附图S2 I</span></p><p data-tool="mdnice编辑器">K值的选择在PRIMUS的流程中至关重要，作者提及了BIC方法选择K值，但具体如何操作，目前还未能找到踪迹，后续学习补充之。</p><blockquote data-tool="mdnice编辑器"><p>To select the optimal k, we fitted PRIMUS for k = 1,2, …,25 with 10 different random initial parameter sets for each k, and k = 12 was selected on the basis of BIC (fig. S2A).<strong>BIC ：</strong> Bayesian information criterion</p></blockquote><figure data-tool="mdnice编辑器"><img data-ratio="0.8688946015424165" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicLWxG3A6KyWDUKMATWIIicLkg7YwicVseMtviaqWYOLdXIhh546x9LRak63oGrTJzJtA03hR2xRicvLw/640?wx_fmt=png" data-type="png" data-w="389" title="图8 文献附图S3 A" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicLWxG3A6KyWDUKMATWIIicLkg7YwicVseMtviaqWYOLdXIhh546x9LRak63oGrTJzJtA03hR2xRicvLw/640?wx_fmt=png"><figcaption><span>图8 文献附图S3 A</span><br></figcaption></figure><blockquote data-tool="mdnice编辑器"><p><strong>S3 A Legend：</strong> Selection of the number of cancer cell clusters based on Bayesian information criterion (BIC). The boxplot showing the BIC of models fitted for 𝑘𝑘 = 1, 2, . . . , 25 with 10 different random initial parameter sets for each 𝑘. The red line indicates the selected number of clusters (𝑘 = 12).</p></blockquote></section><section><section data-style-type="5" data-tools="新媒体排版" data-id="2440476"><section><section><section><section><img data-ratio="0.9495798319327731" data-src="https://mmbiz.qpic.cn/mmbiz_gif/09gp6SvPE04j3m2v7Hr889icHUyibTOHs8YuUibicl7ibRD0ZwG5pDTjBluRreZvuib1o3BibvLkicYhnA4YW7dQsjn0cA/640?wx_fmt=gif" data-type="gif" data-w="119" data-width="100%" src="https://mmbiz.qpic.cn/mmbiz_gif/09gp6SvPE04j3m2v7Hr889icHUyibTOHs8YuUibicl7ibRD0ZwG5pDTjBluRreZvuib1o3BibvLkicYhnA4YW7dQsjn0cA/640?wx_fmt=gif"></section><section data-brushtype="text">往期回顾</section><section><br></section></section></section></section><section><section data-autoskip="1"><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247510149&amp;idx=1&amp;sn=4c7e8f840785500fd751d8d74c16df2c&amp;chksm=ea1cae07dd6b2711f95f88545df466ba400756ed5cdf782413492bee69d0e241c545ea0fcea0&amp;scene=21#wechat_redirect" textvalue="CellphoneDB及可视化" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2"><span>CellphoneDB及可视化</span></a></p><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247510096&amp;idx=1&amp;sn=8bbcc0affa4d8d87dd89d766fff3f84d&amp;chksm=ea1caed2dd6b27c4d607083572b21c6d7bd7296e18ac5eba317322ff942e8c47d3c781fe7887&amp;scene=21#wechat_redirect" textvalue="单细胞测序揭示 MDA5+ 皮肌炎的特有适应性免疫特征" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2"><span>单细胞测序揭示 MDA5+ 皮肌炎的特有适应性免疫特征</span></a></p><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247510077&amp;idx=1&amp;sn=96fd3e828b409d6d5b8de302031acad5&amp;chksm=ea1caebfdd6b27a96a57293dcc17d3d666700ae7f92b04216c1050c8583bcd85a56f73efa9bf&amp;scene=21#wechat_redirect" textvalue="单细胞多组学ATAC-Signac包-概述" linktype="text" imgurl="" imgdata="null" data-itemshowtype="11" tab="innerlink" data-linktype="2"><span>单细胞多组学ATAC-Signac包-概述</span></a></p><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247510012&amp;idx=1&amp;sn=3c62d3b03151dc455bb7d7e1ac388c45&amp;chksm=ea1ca17edd6b28687cc3089de48d89ee89c58e85935c76459860b9e2c52795ffad10a6b46eff&amp;scene=21#wechat_redirect" textvalue="金虎辞岁、玉兔迎春|2022单细胞天地推文汇总" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2"><span>金虎辞岁、玉兔迎春|2022单细胞天地推文汇总</span></a><br><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247510149&amp;idx=1&amp;sn=4c7e8f840785500fd751d8d74c16df2c&amp;chksm=ea1cae07dd6b2711f95f88545df466ba400756ed5cdf782413492bee69d0e241c545ea0fcea0&amp;scene=21#wechat_redirect" textvalue="" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2"></a><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247509977&amp;idx=1&amp;sn=d198d47fb0408c81fe7e2fb3e81b3b00&amp;chksm=ea1ca15bdd6b284dac03fdb0a96793d10a6ed61e933d2cceee167d80492d1ad3b958ff04e753&amp;scene=21#wechat_redirect" textvalue="单细胞分析工具--Palantir轨迹分析" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2"><span>单细胞分析工具--Palantir轨迹分析</span></a><br></p></section></section><hr><p><br></p></section><section data-style-type="5" data-tools="新媒体排版" data-id="2440475"><hr><p><br></p><hr><section><p>如果你对单细胞转录组研究感兴趣，但又不知道如何入门，也许你可以关注一下下面的课程<span></span></p><p><br></p><ul><li><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247510150&amp;idx=1&amp;sn=0783070ae829e63a4da73e85b04d36eb&amp;chksm=ea1cae04dd6b2712de2f846ae79bdad175684a3e1b693490d687994fd27ef3bacf7b66c5fcd3&amp;scene=21#wechat_redirect" textvalue="新年新气象|生信入门/数据挖掘马拉松线上直播课正月初九要开课啦" linktype="text" imgurl="" imgdata="null" data-itemshowtype="11" tab="innerlink" data-linktype="2"><span>新年新气象|生信入门/数据挖掘马拉松线上直播课正月初九要开课啦</span></a></p></li><li><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247518195&amp;idx=1&amp;sn=d6d82ceda8531ebdc294dab0bf9d5519&amp;chksm=9b4bc748ac3c4e5e5c68748241e194638ed0ab33daac9dd6b6ec3fdb647ff782e6dd182b7c69&amp;scene=21#wechat_redirect" textvalue="在全新服务器配置转录组测序数据处理环境" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2" hasload="1">在全新服务器配置转录组测序数据处理环境</a></p></li></ul><p><img data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz_gif/4TKeL1ZejtlKxOib5kmKX6ic6eX0w0WK5jvhtz9yBRsO3OI4yr6S5iaLNM7AbAeuPDHXMvDdur2DRz9wyiax4lEviag/640?wx_fmt=gif" data-type="gif" data-w="240" src="https://mmbiz.qpic.cn/mmbiz_gif/4TKeL1ZejtlKxOib5kmKX6ic6eX0w0WK5jvhtz9yBRsO3OI4yr6S5iaLNM7AbAeuPDHXMvDdur2DRz9wyiax4lEviag/640?wx_fmt=gif"><br></p><p><img data-ratio="0.05278592375366569" data-src="https://mmbiz.qpic.cn/mmbiz/4TKeL1Zejtlq03ZOSZiaTlic1MxgdKiaxTbOZ7ZSe0Xx1Ca8xF3L6Nyj1FYUajtYrSmRIHyZVSsAve0EAvEicZONpg/640?wx_fmt=jpeg" data-type="other" data-w="341" src="https://mmbiz.qpic.cn/mmbiz/4TKeL1Zejtlq03ZOSZiaTlic1MxgdKiaxTbOZ7ZSe0Xx1Ca8xF3L6Nyj1FYUajtYrSmRIHyZVSsAve0EAvEicZONpg/640?wx_fmt=jpeg"></p><p><strong><span>看完记得顺手点个</span></strong><span><strong><span>“在看”</span></strong></span><strong><span>哦！</span></strong></p></section><section><section data-id="93668"><section><section data-width="95%"><section><section><section data-width="38%"><section><section data-tools="135编辑器" data-id="93668"><section><section data-width="95%"><section><section><section data-width="61.8%"><section><section><section><p><br></p><span><strong data-burshtype="text"><img data-copyright="0" data-cropselx1="0" data-cropselx2="109" data-cropsely1="0" data-cropsely2="109" data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz/siaia0BDGJdjRMGrkqo64BGKecYk4akuHpGHVQs7FeOpY7eWbIPGC1tRw5Tw0oEPmx053mR9FTVerWvhuZchIpZw/640?wx_fmt=jpeg" data-type="other" data-w="430" title="qrcode_for_gh_5a3c482ed99f_1280.jpg" src="https://mmbiz.qpic.cn/mmbiz/siaia0BDGJdjRMGrkqo64BGKecYk4akuHpGHVQs7FeOpY7eWbIPGC1tRw5Tw0oEPmx053mR9FTVerWvhuZchIpZw/640?wx_fmt=jpeg"><strong data-burshtype="text">生物</strong><strong data-burshtype="text"> | 单细胞 | 转录组丨资料</strong></strong></span></section><section><span><strong data-burshtype="text">每天都精彩</strong></span></section></section></section><section><section><section><section><section><section><p><span>长按扫码可关注</span></p></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section><p><mp-style-type data-value="10000"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/fBDXTl3pk3VPDohrENZ-SQ",target="_blank" rel="noopener noreferrer">原文链接</a>
