---
title: "Python版SCENIC转录因子分析（四）一文就够了"
date: 2023-01-31T12:12:55Z
draft: ["false"]
tags: [
  "fetched",
  "生信随笔"
]
categories: ["Acdemic"]
---
Python版SCENIC转录因子分析（四）一文就够了 by 生信随笔
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><blockquote data-tool="mdnice编辑器"><p>我前期写了一些关于pySCENIC的笔记，包括：</p><ul><li><section><a href="https://mp.weixin.qq.com/s?__biz=Mzg5MjcxNzg1NA==&amp;mid=2247485056&amp;idx=2&amp;sn=199b38c0e3da84d7e304e6f6f5704a64&amp;scene=21#wechat_redirect" data-linktype="2">Python版SCENIC转录因子分析（一）</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=Mzg5MjcxNzg1NA==&amp;mid=2247485056&amp;idx=1&amp;sn=88fcd17528f0451ba226b15c7edf93b6&amp;scene=21#wechat_redirect" data-linktype="2">Python版SCENIC转录因子分析（二）</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=Mzg5MjcxNzg1NA==&amp;mid=2247485090&amp;idx=1&amp;sn=849fc5f99323687a5cdd1a7306794f8b&amp;scene=21#wechat_redirect" data-linktype="2">Python版SCENIC转录因子分析（三）debug篇</a></section></li></ul><p>在升级了pySCENIC后，发现转录因子数据库更新了。因此本文基于更新后的转录因子数据库，再次记录了从软件部署到pySCENIC的运行，最后进行可视化的详细笔记，希望对大家有所帮助，少走弯路。</p></blockquote><p data-tool="mdnice编辑器">转录因子 (transcription factors, TFs) 是直接作用于基因组，与特定DNA序列结合 (TFBS/motif) ，调控DNA转录过程的一类蛋白质。转录因子可以调节基因组DNA开放性、募集RNA聚合酶进行转录过程、募集辅助因子调节特定的转录阶段，调控诸多生命进程，诸如免疫反应、发育模式等。因此，分析转录因子表达及其调控活性对于解析复杂生命活动具有重要意义。</p><p data-tool="mdnice编辑器">SCENIC（single-cell regulatory network inference and clustering）是一个基于共表达和motif分析，计算单细胞转录组数据基因调控网络重建以及细胞状态鉴定的方法。在输入单细胞基因表达量矩阵后，SCENIC经过以下三个步骤完成转录因子分析：</p><ul data-tool="mdnice编辑器"><li><section>第一步，GENIE3（随机森林)/GRNBoost (Gradient Boosting) 推断转录因子与候选靶基因之间的共表达模块，每个模块包含一个转录因子及其靶基因，纯粹基于共表达；</section></li><li><section>第二步，RcisTatget分析每个共表达模块中的基因，以鉴定enriched motifs，仅保留TF motif富集的模块和targets，构建TF-targets网络，每个TF及其潜在的直接targets gene被称作一个调节因子（Regulons）；</section></li><li><section>第三步，AUCelll计算调节因子（Regulons）的活性，这将确定Regulon在哪些细胞中处于“打开”状态。</section></li></ul><p data-tool="mdnice编辑器">Python版本的优势是速度快，难点是安装比较麻烦，笔者建议用Conda进行按照。</p><h3 data-tool="mdnice编辑器">Step0.软件安装：</h3><p data-tool="mdnice编辑器">安装主要有三种方法docker，conda和Singularity（这里我使用conda）：</p><h4 data-tool="mdnice编辑器">（2）conda安装</h4><p data-tool="mdnice编辑器">注意python版本python=3.7</p><pre data-tool="mdnice编辑器"><span></span><code><span>#</span><span> 需要一些依赖</span><br>conda create -n pyscenic python=3.7 <br>conda activate pyscenic <br><br>mamba install -y numpy scanpy<br>mamba install -y -c anaconda cytoolz<br><br>pip install pyscenic <br></code></pre><h4 data-tool="mdnice编辑器">（3）数据库配置</h4><p data-tool="mdnice编辑器">转录因子分析还需要自行下载最新更新的数据库（https://resources.aertslab.org/cistarget/databases/）：</p><p data-tool="mdnice编辑器">数据库储存在/data/index_genome/cisTarget_databases/</p><pre data-tool="mdnice编辑器"><span></span><code>mkdir  /data/index_genome/cisTarget_databases/<br><span>cd</span> /data/index_genome/cisTarget_databases/<br></code></pre><p data-tool="mdnice编辑器"><strong>下载数据库：</strong></p><p data-tool="mdnice编辑器"><strong>人的：</strong></p><pre data-tool="mdnice编辑器"><span></span><code>cat &gt;download.bash<br><span>##1, For human:</span><br>wget -c https://resources.aertslab.org/cistarget/databases/homo_sapiens/hg19/refseq_r45/mc9nr/gene_based/hg19-500bp-upstream-10species.mc9nr.genes_vs_motifs.rankings.feather<br>wget -c https://resources.aertslab.org/cistarget/databases/homo_sapiens/hg19/refseq_r45/mc9nr/gene_based/hg19-tss-centered-10kb-10species.mc9nr.genes_vs_motifs.rankings.feather<br><br><span>## 下载</span><br>nohup bash download.bash </code></pre><figure data-tool="mdnice编辑器"><figcaption></figcaption></figure><p data-tool="mdnice编辑器"><strong>小鼠的：</strong></p><pre data-tool="mdnice编辑器"><span></span><code>cat &gt;download.bash<br><span>##2, For mouse:</span><br>wget -c https://resources.aertslab.org/cistarget/databases/mus_musculus/mm9/refseq_r45/mc9nr/gene_based/mm9-500bp-upstream-10species.mc9nr.genes_vs_motifs.rankings.feather<br>wget -c https://resources.aertslab.org/cistarget/databases/mus_musculus/mm9/refseq_r45/mc9nr/gene_based/mm9-tss-centered-10kb-10species.mc9nr.genes_vs_motifs.rankings.feather<br><br><span>## 下载</span><br>nohup bash download.bash &amp;<br></code></pre><p data-tool="mdnice编辑器"><strong>还需：</strong></p><pre data-tool="mdnice编辑器"><span></span><code>wget https://github.com/aertslab/pySCENIC/blob/master/resources/hs_hgnc_tfs.txt<br>wget https://resources.aertslab.org/cistarget/motif2tf/motifs-v9-nr.hgnc-m0.001-o0.0.tbl<br></code></pre><h4 data-tool="mdnice编辑器">（5）R包安装</h4><pre data-tool="mdnice编辑器"><span></span><code><span>## SCENIC需要一些依赖包，先安装好</span><br>install.package(<span>"BiocManager"</span>)<br>BiocManager::install(c(<span>"AUCell"</span>, <span>"RcisTarget"</span>,<span>"GENIE3"</span>,<span>"zoo"</span>, <span>"mixtools"</span>, <span>"rbokeh"</span>,<span>"DT"</span>, <span>"NMF"</span>, <span>"pheatmap"</span>, <span>"R2HTML"</span>, <span>"Rtsne"</span>,<span>"doMC"</span>, <span>"doRNG"</span>,<span>"scRNAseq"</span>))<br>devtools::install_github(<span>"aertslab/SCopeLoomR"</span>, build_vignettes = <span>TRUE</span>)<br>devtools::install_github(<span>"aertslab/SCENIC"</span>)<br>BiocManager::install(<span>"Seurat"</span>)<br><span>#check</span><br><span>library</span>(SCENIC)<br>packageVersion(<span>"SCENIC"</span>)<br></code></pre><p data-tool="mdnice编辑器">其余常规R包如ggplot2等需要自行安装。</p><h3 data-tool="mdnice编辑器">Step1.提取单细胞表达量矩阵</h3><p data-tool="mdnice编辑器">这里以PMB3K测试数据为例。</p><p data-tool="mdnice编辑器">在本地新建了一个测试文件夹：</p><pre data-tool="mdnice编辑器"><span></span><code>mkdir test_scenic<br><span>cd</span> test_scenic<br></code></pre><p data-tool="mdnice编辑器">然后在R语言下运行测试数据：</p><pre data-tool="mdnice编辑器"><span></span><code><span>library</span>(Seurat)<br><span>#remotes::install_github("satijalab/seurat-data")</span><br><span>library</span>(SeuratData)<br>AvailableData()<br><span># InstallData("pbmc3k")</span><br>data(<span>"pbmc3k"</span>)<br>pbmc3k<br><br><span>#或者读入测序数据</span><br><span>#pbmc3k = readRDS("./pbmc3k.test.seurat.Rds")</span><br><span>#pbmc3k</span><br><span>#注意矩阵一定要转置，不然会报错</span><br>write.csv(t(as.matrix(pbmc3k@assays$RNA@counts)),file = <span>"for.scenic.data.csv"</span>)<br></code></pre><h3 data-tool="mdnice编辑器">Step2.pySCENIC常规运行（Python+Linux环境）</h3><p data-tool="mdnice编辑器">自己制作一个python 脚本，可以命名为 <code>change.py</code> ， 内容如下所示：</p><pre data-tool="mdnice编辑器"><span></span><code>conda activate pyscenic <br>cat &gt;change.py<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code><span>import</span> os,sys<br>os.getcwd()<br>os.listdir(os.getcwd()) <br><br><span>import</span> loompy <span>as</span> lp;<br><span>import</span> numpy <span>as</span> np;<br><span>import</span> scanpy <span>as</span> sc;<br>x=sc.read_csv(<span>"for.scenic.data.csv"</span>);<br>row_attrs = {<span>"Gene"</span>: np.array(x.var_names),};<br>col_attrs = {<span>"CellID"</span>: np.array(x.obs_names)};<br>lp.create(<span>"sample.loom"</span>,x.X.transpose(),row_attrs,col_attrs);<br></code></pre><p data-tool="mdnice编辑器">标准化运行：</p><pre data-tool="mdnice编辑器"><span></span><code>cat &gt;scenic.bash<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code><span>#运行change.py</span><br>python change.py<br><br><span>##运行pySCENIC</span><br><span># 不同物种的数据库不一样，这里是人类是human </span><br>dir=/data/index_genome/cisTarget_databases/ <span>#改成自己的目录</span><br>tfs=<span>$dir</span>/hs_hgnc_tfs.txt<br>feather=<span>$dir</span>/hg19-tss-centered-10kb-10species.mc9nr.genes_vs_motifs.rankings.feather<br>tbl=<span>$dir</span>/motifs-v9-nr.hgnc-m0.001-o0.0.tbl <br><span># 一定要保证上面的数据库文件完整无误哦 </span><br>input_loom=./sample.loom<br>ls <span>$tfs</span>  <span>$feather</span>  <span>$tbl</span>  <br><br><span>#2.1 grn</span><br>pyscenic grn \<br>--num_workers 10 \<br>--output adj.sample.tsv \<br>--method grnboost2 \<br>sample.loom \<br><span>$tfs</span> <span>#转录因子文件，1839个基因的名字列表</span><br><br><span>#2.2 cistarget</span><br>pyscenic ctx \<br>adj.sample.tsv <span>$feather</span> \<br>--annotations_fname <span>$tbl</span> \<br>--expression_mtx_fname <span>$input_loom</span>  \<br>--mode <span>"dask_multiprocessing"</span> \<br>--output reg.csv \<br>--num_workers 20  \<br>--mask_dropouts<br><br><span>#2.3 AUCell</span><br>pyscenic aucell \<br><span>$input_loom</span> \<br>reg.csv \<br>--output out_SCENIC.loom \<br>--num_workers 10<br></code></pre><p data-tool="mdnice编辑器">运行：</p><pre data-tool="mdnice编辑器"><span></span><code>nohup bash scenic.bash  <span>1</span>&gt;pySCENIC.log <span>2</span>&gt;&amp;<span>1</span> &amp;<br></code></pre><p data-tool="mdnice编辑器">这一步会得到11M的out_SCENIC.loom文件。最重要的三个文件如下:</p><figure data-tool="mdnice编辑器"><img data-ratio="0.35324015247776364" data-src="https://mmbiz.qpic.cn/mmbiz/fTW9zRI3eqUgYo2zXwz6GNAyvHJM3B0KGlDKWCpgXZyfbfmrJgcGmmQibudQVSJtiblNEicCapnUEFtSKQ3kUibm7A/640?wx_fmt=other" data-type="other" data-w="787" src="https://mmbiz.qpic.cn/mmbiz/fTW9zRI3eqUgYo2zXwz6GNAyvHJM3B0KGlDKWCpgXZyfbfmrJgcGmmQibudQVSJtiblNEicCapnUEFtSKQ3kUibm7A/640?wx_fmt=other"><figcaption>image-20230131191733555</figcaption></figure><p data-tool="mdnice编辑器">在Linux跑完scSCENIC的流程后，接下来基于R语言，将loom数据粗处理，然后导入Seurat单细胞数据进行可视化。</p><h3 data-tool="mdnice编辑器">Step3.sample_SCENIC.loom导入R里面进行可视化</h3><p data-tool="mdnice编辑器">加载R包：</p><pre data-tool="mdnice编辑器"><span></span><code><span>##可视化</span><br>rm(list=ls())<br><span>library</span>(Seurat)<br><span>library</span>(SCopeLoomR)<br><span>library</span>(AUCell)<br><span>library</span>(SCENIC)<br><span>library</span>(dplyr)<br><span>library</span>(KernSmooth)<br><span>library</span>(RColorBrewer)<br><span>library</span>(plotly)<br><span>library</span>(BiocParallel)<br><span>library</span>(grid)<br><span>library</span>(ComplexHeatmap)<br><span>library</span>(data.table)<br><span>library</span>(scRNAseq)<br><span>library</span>(patchwork)<br><span>library</span>(ggplot2) <br><span>library</span>(stringr)<br><span>library</span>(circlize)<br></code></pre><h4 data-tool="mdnice编辑器">3.1 提取 out_SCENIC.loom 信息:</h4><pre data-tool="mdnice编辑器"><span></span><code><span>#### 1.提取 out_SCENIC.loom 信息</span><br>loom &lt;- open_loom(<span>'out_SCENIC.loom'</span>) <br><br>regulons_incidMat &lt;- get_regulons(loom, column.attr.name=<span>"Regulons"</span>)<br>regulons_incidMat[<span>1</span>:<span>4</span>,<span>1</span>:<span>4</span>] <br>regulons &lt;- regulonsToGeneLists(regulons_incidMat)<br>regulonAUC &lt;- get_regulons_AUC(loom,column.attr.name=<span>'RegulonsAUC'</span>)<br>regulonAucThresholds &lt;- get_regulon_thresholds(loom)<br>tail(regulonAucThresholds[order(as.numeric(names(regulonAucThresholds)))])<br><br>embeddings &lt;- get_embeddings(loom)  <br>close_loom(loom)<br><br>rownames(regulonAUC)<br>names(regulons)<br></code></pre><h4 data-tool="mdnice编辑器">3.2 加载SeuratData</h4><pre data-tool="mdnice编辑器"><span></span><code><span>#### 2.加载SeuratData</span><br><span>library</span>(SeuratData) <span>#加载seurat数据集  </span><br>data(<span>"pbmc3k"</span>)  <br>seurat.data = pbmc3k<br><span>#seurat.data = readRDS("./pbmc3k.test.seurat.Rds")</span><br>seurat.data &lt;- seurat.data %&gt;% NormalizeData(verbose = <span>F</span>) %&gt;%<br>  FindVariableFeatures(selection.method = <span>"vst"</span>, nfeatures = <span>2000</span>, verbose = <span>F</span>) %&gt;% <br>  ScaleData(verbose = <span>F</span>) %&gt;%<br>  RunPCA(npcs = <span>30</span>, verbose = <span>F</span>)<br><br>n.pcs = <span>30</span><br>seurat.data &lt;- seurat.data %&gt;% <br>  RunUMAP(reduction = <span>"pca"</span>, dims = <span>1</span>:n.pcs, verbose = <span>F</span>) %&gt;% <br>  FindNeighbors(reduction = <span>"pca"</span>, k.param = <span>10</span>, dims = <span>1</span>:n.pcs)<br><br><span># 这里有自带的注释</span><br>seurat.data$seurat_annotations[is.na(seurat.data$seurat_annotations)] = <span>"B"</span><br>Idents(seurat.data) &lt;- <span>"seurat_annotations"</span><br>DimPlot(seurat.data,reduction = <span>"umap"</span>,label=<span>T</span> ) <br></code></pre><h4 data-tool="mdnice编辑器">3.3 可视化</h4><pre data-tool="mdnice编辑器"><span></span><code>sub_regulonAUC &lt;- regulonAUC[,match(colnames(seurat.data),colnames(regulonAUC))]<br>dim(sub_regulonAUC)<br>seurat.data<br><span>#确认是否一致</span><br>identical(colnames(sub_regulonAUC), colnames(seurat.data))<br><br>cellClusters &lt;- data.frame(row.names = colnames(seurat.data), <br>                           seurat_clusters = as.character(seurat.data$seurat_annotations))<br>cellTypes &lt;- data.frame(row.names = colnames(seurat.data), <br>                        celltype = seurat.data$seurat_annotations)<br>head(cellTypes)<br>head(cellClusters)<br>sub_regulonAUC[<span>1</span>:<span>4</span>,<span>1</span>:<span>4</span>] <br><br><span>#保存一下</span><br>save(sub_regulonAUC,cellTypes,cellClusters,seurat.data,<br>     file = <span>'for_rss_and_visual.Rdata'</span>)<br></code></pre><p data-tool="mdnice编辑器">B细胞有两个非常出名的转录因子，TCF4(+) 以及NR2C1(+)，接下来就可以对这两个进行简单的可视化。首先我们需要把这两个转录因子活性信息 添加到降维聚类分群后的的seurat对象里面。</p><pre data-tool="mdnice编辑器"><span></span><code><span>#B细胞有两个非常出名的转录因子，TCF4(+) 以及NR2C1(+)</span><br>regulonsToPlot = c(<span>'TCF4(+)'</span>,<span>'NR2C1(+)'</span>)<br>regulonsToPlot %<span>in</span>% row.names(sub_regulonAUC)<br>seurat.data@meta.data = cbind(seurat.data@meta.data ,t(assay(sub_regulonAUC[regulonsToPlot,])))<br><br><span># Vis</span><br>p1 = DotPlot(seurat.data, features = unique(regulonsToPlot)) + RotatedAxis()<br>p2 = RidgePlot(seurat.data, features = regulonsToPlot , ncol = <span>2</span>) <br>p3 = VlnPlot(seurat.data, features = regulonsToPlot,pt.size = <span>0</span>)<br>p4 = FeaturePlot(seurat.data,features = regulonsToPlot)<br><br>wrap_plots(p1,p2,p3,p4)<br><span>#可选</span><br><span>#scenic_res = assay(sub_regulonAUC) %&gt;% as.matrix()</span><br><span>#seurat.data[["scenic"]] &lt;- SeuratObject::CreateAssayObject(counts = scenic_res)</span><br><span>#seurat.data &lt;- SeuratObject::SetAssayData(seurat.data, slot = "scale.data",</span><br><span>#                                  new.data = scenic_res, assay = "scenic")</span><br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.5153019023986766" data-src="https://mmbiz.qpic.cn/mmbiz/fTW9zRI3eqUgYo2zXwz6GNAyvHJM3B0K0bzas2fQk1u8DDTAFWX6ah854mMFPOVibpYibVaYqnbnHdcyxsWEtJ6w/640?wx_fmt=other" data-type="other" data-w="1209" src="https://mmbiz.qpic.cn/mmbiz/fTW9zRI3eqUgYo2zXwz6GNAyvHJM3B0K0bzas2fQk1u8DDTAFWX6ah854mMFPOVibpYibVaYqnbnHdcyxsWEtJ6w/640?wx_fmt=other"><figcaption>image-20230131191751021</figcaption></figure><h3 data-tool="mdnice编辑器">Step4.亚群特异性转录因子分析及可视化</h3><h4 data-tool="mdnice编辑器">4.1 TF活性均值</h4><p data-tool="mdnice编辑器">看看不同单细胞亚群的转录因子活性平均值</p><pre data-tool="mdnice编辑器"><span></span><code><span>### 4.1. TF活性均值</span><br><span># 看看不同单细胞亚群的转录因子活性平均值</span><br><span># Split the cells by cluster:</span><br>selectedResolution &lt;- <span>"celltype"</span> <span># select resolution</span><br>cellsPerGroup &lt;- split(rownames(cellTypes), <br>                       cellTypes[,selectedResolution])<br><br><span># 去除extened regulons</span><br>sub_regulonAUC &lt;- sub_regulonAUC[onlyNonDuplicatedExtended(rownames(sub_regulonAUC)),] <br>dim(sub_regulonAUC)<br><br><span># Calculate average expression:</span><br>regulonActivity_byGroup &lt;- sapply(cellsPerGroup,<br>                                  <span>function</span>(cells) <br>                                    rowMeans(getAUC(sub_regulonAUC)[,cells]))<br><br><span># Scale expression. </span><br><span># Scale函数是对列进行归一化，所以要把regulonActivity_byGroup转置成细胞为行，基因为列</span><br><span># 参考：https://www.jianshu.com/p/115d07af3029</span><br>regulonActivity_byGroup_Scaled &lt;- t(scale(t(regulonActivity_byGroup),<br>                                          center = <span>T</span>, scale=<span>T</span>)) <br><span># 同一个regulon在不同cluster的scale处理</span><br>dim(regulonActivity_byGroup_Scaled)<br><span>#[1] 209   9</span><br>regulonActivity_byGroup_Scaled=na.omit(regulonActivity_byGroup_Scaled)<br></code></pre><p data-tool="mdnice编辑器">热图查看TF分布：</p><pre data-tool="mdnice编辑器"><span></span><code>Heatmap(<br>  regulonActivity_byGroup_Scaled,<br>  name                         = <span>"z-score"</span>,<br>  col                          = colorRamp2(seq(from=-<span>2</span>,to=<span>2</span>,length=<span>11</span>),rev(brewer.pal(<span>11</span>, <span>"Spectral"</span>))),<br>  show_row_names               = <span>TRUE</span>,<br>  show_column_names            = <span>TRUE</span>,<br>  row_names_gp                 = gpar(fontsize = <span>6</span>),<br>  clustering_method_rows = <span>"ward.D2"</span>,<br>  clustering_method_columns = <span>"ward.D2"</span>,<br>  row_title_rot                = <span>0</span>,<br>  cluster_rows                 = <span>TRUE</span>,<br>  cluster_row_slices           = <span>FALSE</span>,<br>  cluster_columns              = <span>FALSE</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.9231678486997635" data-src="https://mmbiz.qpic.cn/mmbiz/fTW9zRI3eqUgYo2zXwz6GNAyvHJM3B0KHHZjps2DWtQSuG0aMibjAicpO9iamk8BXNyYXlxWyHhw05v7lj1ibz59hg/640?wx_fmt=other" data-type="other" data-w="846" src="https://mmbiz.qpic.cn/mmbiz/fTW9zRI3eqUgYo2zXwz6GNAyvHJM3B0KHHZjps2DWtQSuG0aMibjAicpO9iamk8BXNyYXlxWyHhw05v7lj1ibz59hg/640?wx_fmt=other"><figcaption>image-20230131191806586</figcaption></figure><p data-tool="mdnice编辑器">可以看到，确实每个单细胞亚群都是有 自己的特异性的激活的转录因子。</p><h4 data-tool="mdnice编辑器">4.2 rss查看特异TF</h4><p data-tool="mdnice编辑器">不过，SCENIC包自己提供了一个 calcRSS函数，帮助我们来挑选各个单细胞亚群特异性的转录因子，全称是：Calculates the regulon specificity score</p><p data-tool="mdnice编辑器">参考文章：The RSS was first used by Suo et al. in: Revealing the Critical Regulators of Cell Identity in the Mouse Cell Atlas. Cell Reports (2018). doi: 10.1016/j.celrep.2018.10.045 运行超级简单。</p><pre data-tool="mdnice编辑器"><span></span><code><span>### 4.2. rss查看特异TF</span><br>rss &lt;- calcRSS(AUC=getAUC(sub_regulonAUC), <br>               cellAnnotation=cellTypes[colnames(sub_regulonAUC), selectedResolution]) <br>rss=na.omit(rss) <br>rssPlot &lt;- plotRSS(rss)<br>plotly::ggplotly(rssPlot$plot)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="1.180940892641737" data-src="https://mmbiz.qpic.cn/mmbiz/fTW9zRI3eqUgYo2zXwz6GNAyvHJM3B0KxdkKFJ2pWSHdgmmtO98icDp4o11PUaDCpU3ViapFVztG7T178SqKEkOQ/640?wx_fmt=other" data-type="other" data-w="829" src="https://mmbiz.qpic.cn/mmbiz/fTW9zRI3eqUgYo2zXwz6GNAyvHJM3B0KxdkKFJ2pWSHdgmmtO98icDp4o11PUaDCpU3ViapFVztG7T178SqKEkOQ/640?wx_fmt=other"><figcaption>image-20220731224415686</figcaption></figure><h4 data-tool="mdnice编辑器">4.3 其他查看TF方式</h4><pre data-tool="mdnice编辑器"><span></span><code>rss=regulonActivity_byGroup_Scaled<br>head(rss)<br>df = do.call(rbind,<br>             lapply(<span>1</span>:ncol(rss), <span>function</span>(i){<br>               dat= data.frame(<br>                 path  = rownames(rss),<br>                 cluster =   colnames(rss)[i],<br>                 sd.1 = rss[,i],<br>                 sd.2 = apply(rss[,-i], <span>1</span>, median)  <br>               )<br>             }))<br>df$fc = df$sd.1 - df$sd.2<br>top5 &lt;- df %&gt;% group_by(cluster) %&gt;% top_n(<span>5</span>, fc)<br>rowcn = data.frame(path = top5$cluster) <br>n = rss[top5$path,] <br><span>#rownames(rowcn) = rownames(n)</span><br>pheatmap(n,<br>         annotation_row = rowcn,<br>         show_rownames = <span>T</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.8105625717566016" data-src="https://mmbiz.qpic.cn/mmbiz/fTW9zRI3eqUgYo2zXwz6GNAyvHJM3B0KBoRcicINU76hEsSAEKnsQw0v6U41n7ktypq4KdenAZQhEM3FqHGabGA/640?wx_fmt=other" data-type="other" data-w="871" src="https://mmbiz.qpic.cn/mmbiz/fTW9zRI3eqUgYo2zXwz6GNAyvHJM3B0KBoRcicINU76hEsSAEKnsQw0v6U41n7ktypq4KdenAZQhEM3FqHGabGA/640?wx_fmt=other"><figcaption>image-20230131191851169</figcaption></figure><span>- END -</span></section><p><br></p><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/gMvsdu-fdw2bJKMeS9O3gA",target="_blank" rel="noopener noreferrer">原文链接</a>
