---
title: "Refgenie |  管理参考基因组注释文件及索引文件的利器"
date: 2022-12-01T00:40:41Z
draft: ["false"]
tags: [
  "fetched",
  "猪猪生信"
]
categories: ["Acdemic"]
---
Refgenie |  管理参考基因组注释文件及索引文件的利器 by 猪猪生信
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><h2 data-tool="mdnice编辑器"><span></span>Refgenie是什么</h2><ul data-tool="mdnice编辑器"><li><section>Refgenie是管理参考基因组注释文件以及索引文件的利器，对于常用的物种人和小鼠的，只需使用Refgenie拉取远程的文件到本地就可以很方便。对于其他物种，下面会展示如何<span>自定义的</span>构建猪的参考基因组及注释文件的资源库。很推荐使用这个软件来管理参考基因组文件，给服务器配置上这个软件吧</section></li></ul><h2 data-tool="mdnice编辑器"><span></span>安装</h2><p data-tool="mdnice编辑器">安装Refgenie，没有root权限使用conda安装</p><pre data-tool="mdnice编辑器"><span></span><code><span># 使用root为所有用户安装</span><br>su root <span>#进入root用户</span><br>pip intall refgenie<br><br><span># 安装在自己的创建的conda环境中, 推荐使用conda安装</span><br>conda create -n refgenie <br>conda activate refgenie<br>pip install refgenie<br><br><span># 安装在当前的环境</span><br>pip install refgenie<br></code></pre><h2 data-tool="mdnice编辑器"><span></span>初始化配置</h2><pre data-tool="mdnice编辑器"><span></span><code><span># 在家目录下生成refgenie_genome文件夹</span><br>refgenie init -c /disk_a/refgenie_genome/genome_config.yaml<br><br><span># 配置文件加入环境变量，我使用的root的安装的，所以配置文件变量写入了/etc/environment中，如果conda安装写入家目录的.bashrc</span><br>sudo vim /etc/environment<br><span>export</span> REFGENIE=/disk_a/refgenie_genome/genome_config.yaml<br><br><span># 查看下目录结构</span><br>tree /disk_a/refgenie_genome/<br><span># 如果想让服务器所有用户使用，需要为其他用户添加执行权限</span><br>chmod -R o+x /disk_a/refgenie_genome/<br></code></pre><h2 data-tool="mdnice编辑器"><span></span>Refgenie基本操作</h2><ul data-tool="mdnice编辑器"><li><section><p>查看本地基因组文件</p><pre><span></span><code>refgenie list<br></code></pre><figure><img data-ratio="0.22753346080305928" data-src="https://mmbiz.qpic.cn/mmbiz_png/hlasRQiaooRxiaRiblyaWvm2g5B3OxueSticQQT1icN0JuJUrGRMUjFkfnX7K1Rk6v5CVMAcsFzhkJhNwzXuoTD6Rnw/640?wx_fmt=png" data-type="png" data-w="523" src="https://mmbiz.qpic.cn/mmbiz_png/hlasRQiaooRxiaRiblyaWvm2g5B3OxueSticQQT1icN0JuJUrGRMUjFkfnX7K1Rk6v5CVMAcsFzhkJhNwzXuoTD6Rnw/640?wx_fmt=png"><figcaption>image-20220930153205932</figcaption></figure></section></li><li><section><p>查看远程基因组文件夹，refgenie服务器上存放着一些已经构建好的参考基因组，可以直接拉取到本地</p><pre><span></span><code>refgenie listr<br></code></pre><figure><img data-ratio="0.446656050955414" data-src="https://mmbiz.qpic.cn/mmbiz_png/hlasRQiaooRxiaRiblyaWvm2g5B3OxueSticowYTAJzspABibiaGxmcyiboFYN14FImOGtaZiaAv7covJ7Wmhv1FTT5qVg/640?wx_fmt=png" data-type="png" data-w="1256" src="https://mmbiz.qpic.cn/mmbiz_png/hlasRQiaooRxiaRiblyaWvm2g5B3OxueSticowYTAJzspABibiaGxmcyiboFYN14FImOGtaZiaAv7covJ7Wmhv1FTT5qVg/640?wx_fmt=png"><figcaption>image-20220930154534747</figcaption></figure></section></li><li><section><p>试下拉取一个参考基因组的索引文件</p><p>使用<code>pull</code>从远程拉取<code>refgenie listr</code>中显示存在的基因组索引。远程只有人和鼠的，也可以使用<code>build</code>构建自己的基因组，先随便获取一个索引文件测试下</p><pre><span></span><code>refgenie pull rCRSd/bowtie2_index<br></code></pre><p>查看下拉取到本地的参考基因组</p><pre><span></span><code>refgenie list<br></code></pre><figure><img data-ratio="0.259047619047619" data-src="https://mmbiz.qpic.cn/mmbiz_png/hlasRQiaooRxiaRiblyaWvm2g5B3OxueSticcF8iaNtpIpPfK4PM587bKLD5iboJQT8qvbv5usJfpBgqJEX52dS29yqw/640?wx_fmt=png" data-type="png" data-w="525" src="https://mmbiz.qpic.cn/mmbiz_png/hlasRQiaooRxiaRiblyaWvm2g5B3OxueSticcF8iaNtpIpPfK4PM587bKLD5iboJQT8qvbv5usJfpBgqJEX52dS29yqw/640?wx_fmt=png"><figcaption>image-20220930154853653</figcaption></figure></section></li><li><section><p>获取本地assets中的文件路径</p><pre><span></span><code>refgenie seek rCRSd/bowtie2_index<br></code></pre><figure><img data-ratio="0.06990291262135923" data-src="https://mmbiz.qpic.cn/mmbiz_png/hlasRQiaooRxiaRiblyaWvm2g5B3OxueSticqRT92ulJ2M7cQydKfFI8oGHKhxHkXPUEXMCibTUD8875L9dFvrFjkSg/640?wx_fmt=png" data-type="png" data-w="515" src="https://mmbiz.qpic.cn/mmbiz_png/hlasRQiaooRxiaRiblyaWvm2g5B3OxueSticqRT92ulJ2M7cQydKfFI8oGHKhxHkXPUEXMCibTUD8875L9dFvrFjkSg/640?wx_fmt=png"><figcaption>image-20220930155007594</figcaption></figure></section></li><li><section><p>删除本地的索引文件</p><pre><span></span><code><span># 删除bowtie2索引文件</span><br>refgenie remove -f -g rCRSd bowtie2_index<br><span># -f 强制删除，不弹出确认删除的信息</span><br><span># -g 指定genome</span><br><br><span># 查看是否删除</span><br>refgenie list<br></code></pre><figure><img data-ratio="0.35549525101763907" data-src="https://mmbiz.qpic.cn/mmbiz_png/hlasRQiaooRxiaRiblyaWvm2g5B3OxueSticog6TSfGCRLF7tpvlmxFQnJPJxYFqdrSOjn8fx53Q4Zic51icykfU6oGA/640?wx_fmt=png" data-type="png" data-w="737" src="https://mmbiz.qpic.cn/mmbiz_png/hlasRQiaooRxiaRiblyaWvm2g5B3OxueSticog6TSfGCRLF7tpvlmxFQnJPJxYFqdrSOjn8fx53Q4Zic51icykfU6oGA/640?wx_fmt=png"><figcaption>image-20220930155310162</figcaption></figure></section></li></ul><h2 data-tool="mdnice编辑器"><span></span>拉取人和鼠的参考基因组及索引文件资源</h2><p data-tool="mdnice编辑器">Refgenie服务器上参考基因组资源</p><ul data-tool="mdnice编辑器"><li><section>在这个网站上查看基因组的相关信息：http://refgenomes.databio.org/</section></li></ul><pre data-tool="mdnice编辑器"><span></span><code><span># 查看Refgenie服务器上的资源</span><br>refgenie listr<br><span># 拉取hg38的所有参考基因组资源</span><br>refgenie pull -g hg38 fasta gencode_gtf refgene_anno dbsnp ensembl_gtf ensembl_rb suffixerator_index bwa_index bowtie2_index dbnsfp star_index fasta_txome hisat2_index cellranger_reference bismark_bt2_index salmon_partial_sa_index tgMap salmon_sa_index<br><span># 拉取mm10的所有参考基因组资源</span><br>mkdir mm10_log<br><span>for</span> i <span>in</span> fasta gencode_gtf refgene_anno ensembl_gtf ensembl_rb suffixerator_index bwa_index bowtie2_index hisat2_index star_index fasta_txome cellranger_reference bismark_bt2_index salmon_partial_sa_index tgMap salmon_sa_index<br><span>do</span><br><span>echo</span> <span>"refgenie pull -g mm10  <span>$i</span> --pull-large  --force-overwrite 1&gt; <span>${i}</span>.log"</span> &gt;&gt; mm10.sh<br><span>done</span><br><br></code></pre><h2 data-tool="mdnice编辑器"><span></span>自己下载参考基因组序列来构建本地的参考基因资源</h2><p data-tool="mdnice编辑器">因为Refgenie服务器上只有人和鼠的参考基因组资源，对于其他物种的可以自己构建，下面以猪的参考基因组为示例</p><p data-tool="mdnice编辑器">参考基因组序列和注释文件选择Ensembl来源的</p><pre data-tool="mdnice编辑器"><span></span><code><span># Sscrofa11.1 (GCA_000003025.6) release-107</span><br><br><span># 创建一个文件夹</span><br>mkdir /disk_a/Sscrofa11-1_release-107 &amp;&amp; <span>cd</span> /disk_a/Sscrofa11-1_release-107<br><br><span># fasta</span><br>wget -c http://ftp.ensembl.org/pub/release-107/fasta/sus_scrofa/dna/Sus_scrofa.Sscrofa11.1.dna.toplevel.fa.gz<br><span># gtf</span><br>wget -c http://ftp.ensembl.org/pub/release-107/gtf/sus_scrofa/Sus_scrofa.Sscrofa11.1.107.gtf.gz<br><span># abinitio.gtf</span><br>wget -c http://ftp.ensembl.org/pub/release-107/gtf/sus_scrofa/Sus_scrofa.Sscrofa11.1.107.abinitio.gtf.gz<br><span># gff</span><br>wget -c http://ftp.ensembl.org/pub/release-107/gff3/sus_scrofa/Sus_scrofa.Sscrofa11.1.107.gff3.gz<br></code></pre><ul data-tool="mdnice编辑器"><li><section><p>官网教程：http://refgenie.databio.org/en/latest/available_assets/</p></section></li><li><section><p>创建Sscrofa11.1，添加fasta</p><p>在下面命令<code>Sscrofa11-1/fasta:ensembl-release107</code>中，冒号后面是添加的标签信息，标注版本，也可以不添加这个信息，具体参考下面的标签设置教程</p><pre><span></span><code>refgenie build Sscrofa11-1/fasta:ensembl-release107 --files fasta=/disk_a/Sscrofa11-1_release-107/Sus_scrofa.Sscrofa11.1.dna.toplevel.fa.gz<br>refgenie seek Sscrofa11-1/fasta<br><br><span># 设置别名</span><br>refgenie <span>alias</span> get<br>refgenie <span>alias</span> <span>set</span> --aliases pig --digest 9dcf099b12bc4abd33f99dfc3696831b0e912e7d3ec0c475<br>refgenie seek pig/fasta<br></code></pre></section></li><li><section><p>添加ensembl_gtf和gff</p><pre><span></span><code><span># gtf</span><br>refgenie build Sscrofa11-1/ensembl_gtf:release107 --files ensembl_gtf=/disk_a/Sscrofa11-1_release-107/Sus_scrofa.Sscrofa11.1.107.gtf.gz<br><br><span># gff</span><br>refgenie add  Sscrofa11-1/ensembl_gff:release107 -p /disk_a/Sscrofa11-1_release-107/Sus_scrofa.Sscrofa11.1.107.gtf.gz<br></code></pre></section></li><li><section><p>添加ensembl_rb</p><pre><span></span><code>refgenie build Sscrofa11-1/ensembl_rb:release107 --files gff=/disk_a/Sscrofa11-1_release-107/Sus_scrofa.Sscrofa11.1.107.gff3.gz<br></code></pre></section></li><li><section><p>添加feat_annotation，包括以下基因组特征注释：增强子、启动子、启动子侧翼区域、5' UTR、3' UTR、外显子和内含子的信息</p><pre><span></span><code>refgenie build Sscrofa11-1/feat_annotation:release107<br></code></pre></section></li><li><section><p>构建索引：bowtie2_index</p><pre><span></span><code><span># 要保证当前环境下要有相应的软件</span><br>bowtie2 --version<br>refgenie build Sscrofa11-1/bowtie2_index:2.3.5.1<br></code></pre></section></li><li><section><p>其他软件的索引</p><pre><span></span><code><span># hisat2_index</span><br>hisat2 --version<br>refgenie build Sscrofa11-1/hisat2_index:2.2.1<br><br><span># star_index</span><br>conda install -c bioconda star<br>STAR --version<br>refgenie build Sscrofa11-1/star_index:2.7.10a<br><br><span># salmon_index</span><br>conda install -c bioconda salmon<br>salmon -v<br>refgenie build Sscrofa11-1/salmon_index:0.14.1<br><br><span># rsem_index_bowtie2</span><br>conda install -c bioconda rsem<br><br>fa=/disk_a/refgenie_genome/<span>alias</span>/pig/fasta/ensembl-release107/pig.fa<br>gtf=/disk_a/Sscrofa11-1_release-107/Sus_scrofa.Sscrofa11.1.107.gtf<br>bowtie2=/home/xuchao/biosoft/miniconda3/envs/atac/bin/bowtie2<br>out_dir=/disk_a/Sscrofa11-1_release-107/resm_index_bowtie2<br>rsem-prepare-reference --gtf <span>$gtf</span>  --bowtie2 --bowtie2-path <span>$bowtie2</span> <span>$fa</span> <span>$out_dir</span><br>refgenie add  Sscrofa11-1/rsem_index -p /disk_a/Sscrofa11-1_release-107/rsem_index_bowtie2<br><br><br><span># 下面是其他软件的构建索引的方法：</span><br><span># bismark_bt1_index and bismark_bt2_index</span><br>refgenie build <span>test</span>/bismark_bt1_index<br>refgenie build <span>test</span>/bismark_bt2_index<br><span># bwa_index</span><br>refgenie build <span>test</span>/bwa_index<br><span># epilog_index</span><br>refgenie build <span>test</span>/epilog_index --params context=CG<br><span># kallisto_index</span><br>refgenie build <span>test</span>/kallisto_index<br><span># salmon_index</span><br>refgenie build <span>test</span>/salmon_index<br><span># star_index</span><br>refgenie build <span>test</span>/star_index<br><span># suffixerator_index</span><br>refgenie build <span>test</span>/suffixerator_index --params memlimit=8GB<br><span># tallymer_index</span><br>refgenie build <span>test</span>/tallymer_index --params mersize=30 minocc=2<br></code></pre></section></li></ul><h2 data-tool="mdnice编辑器"><span></span>使用Genome aliases</h2><p data-tool="mdnice编辑器">aliases可以设置参考基因组的别名</p><pre data-tool="mdnice编辑器"><span></span><code><span># 查询帮助</span><br>refgenie <span>alias</span> --<span>help</span><br><span># 获取aliases, 查看digest，下面命令会用到digest</span><br>refgenie <span>alias</span> get<br><span># 设置aliases</span><br>refgenie <span>alias</span> <span>set</span> --aliases mito --digest 94e0d21feb576e6af61cd2a798ad30682ef2428bb7eabbb4<br><span># 删出aliases</span><br>refgenie <span>alias</span> remove -a mito -d 94e0d21feb576e6af61cd2a798ad30682ef2428bb7eabbb4<br></code></pre><h2 data-tool="mdnice编辑器"><span></span>添加tags</h2><p data-tool="mdnice编辑器">tags可以用来给文件添加标签，用来注释版本信息</p><pre data-tool="mdnice编辑器"><span></span><code><span># 查看帮助</span><br>refgenie tag -h<br><span># 将default更改为most_recent</span><br>refgenie tag rCRSd/bowtie2_index:default --tag most_recent<br><span># 重新创建一个test的tags的索引</span><br>refgenie build rCRSd/bowtie2_index:<span>test</span><br><span># 查看bowtie2索引，可以发现返回的是most_recent标签的索引，因为most_recent是创建的比较早</span><br>refgenie seek rCRSd/bowtie2_index<br><span># 设置test标签的索引为默认返回的</span><br>refgenie tag rCRSd/bowtie2_index:<span>test</span> -d<br><span># 再次查看，这次显示返回的test标签的索引</span><br>refgenie seek rCRSd/bowtie2_index<br><span># 删除test标签的索引</span><br>refgenie remove -g rCRSd bowtie2_index:<span>test</span><br><span># 重新设置most_recent标签为默认的索引</span><br>refgenie tag rCRSd/bowtie2_index:most_recent -d<br></code></pre><h2 data-tool="mdnice编辑器"><span></span>populate命令使用</h2><p data-tool="mdnice编辑器"><code>populate</code>可以在命令行终端使用管道符填充索引路径，示例如下：</p><pre data-tool="mdnice编辑器"><span></span><code><span># populate</span><br><span>echo</span> <span>'bowtie2 -x refgenie://rCRSd/bowtie2_index -U r1.fq -S eg1.sam'</span> | refgenie populate<br><span># 返回结果：bowtie2 -x /disk_a/refgenie_genome/alias/rCRSd/bowtie2_index/most_recent/rCRSd -U r1.fq -S eg1.sam</span><br><br><span># populater </span><br><span>echo</span> <span>'bowtie2 -x refgenie://rCRSd/bowtie2_index -U r1.fq -S eg1.sam'</span> | refgenie populater<br><span># 返回结果：bowtie2 -x http://awspds.refgenie.databio.org/refgenomes.databio.org/94e0d21feb576e6af61cd2a798ad30682ef2428bb7eabbb4/bowtie2_index__default/94e0d21feb576e6af61cd2a798ad30682ef2428bb7eabbb4 -U r1.fq -S eg1.sam</span><br></code></pre></section><p>参考：<br></p><p>http://refgenie.databio.org/en/latest/</p><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/r6VuSfMsMkgvFfoDwqgwag",target="_blank" rel="noopener noreferrer">原文链接</a>
