---
title: "scATAC-seq| motif 分析"
date: 2022-12-31T01:42:16Z
draft: ["false"]
tags: [
  "fetched",
  "生信技能树"
]
categories: ["Acdemic"]
---
scATAC-seq| motif 分析 by 生信技能树
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><blockquote data-tool="mdnice编辑器"><p>在单细胞转录组冒尖的时候，单细胞其它组学却一直不温不火，比如单细胞ATAC以及单细胞免疫组库。所以虽然很早之前我们就推荐了github代码：https://github.com/GreenleafLab/10x-scATAC-2019</p></blockquote><ul data-tool="mdnice编辑器"><li><section>01_Filter_Cells_v2.R</section></li><li><section>02_Get_Peak_Set_hg19_v2.R</section></li><li><section>03_Run_chromVAR_v2.R</section></li><li><section>04_Run_Cicero_v2.R</section></li><li><section>05_Cluster_Unique_Peaks_v2.R</section></li><li><section>06_Analyze_UMAP_Trajectory.R</section></li><li><section>07_ChromVAR_For_GWAS_w_CoAccessbility_v2.R</section></li><li><section>08_Run_scCNV_v2.R</section></li></ul><p data-tool="mdnice编辑器">但是一直没有深入解读它，在2021的时候《单细胞天地》的一个小伙伴写了个开头：</p><ul data-tool="mdnice编辑器"><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247508480&amp;idx=1&amp;sn=15360a6b494d7a536fb8c790dc54a585&amp;scene=21#wechat_redirect" data-linktype="2">scATAC-seq1：由转录组到表观组</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247508480&amp;idx=2&amp;sn=4548d7aa6ba7185426fd0040cda25f73&amp;scene=21#wechat_redirect" data-linktype="2">scATAC-seq2: scATAC-seq技术原理</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247508480&amp;idx=3&amp;sn=51825eb6c789d97682d558fb34b46044&amp;scene=21#wechat_redirect" data-linktype="2">scATAC-seq3:常用工具—SnapATAC简介</a></section></li></ul><p data-tool="mdnice编辑器">但是没有能坚持下来，其实文章给的配套github代码非常齐全了，就是需要花时间钻研和解读。</p><p data-tool="mdnice编辑器">恰好2022我们又看到了一个带有全套github代码以及配套数据的文章，见：<a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247516871&amp;idx=1&amp;sn=7eed3a5b5b6575458ec44ee9ddd0b6de&amp;scene=21#wechat_redirect" data-linktype="2">10x的单细胞ATAC上游流程之cellranger-atac</a>，在等待的期间，交流群的学徒就先拿 atac_v1_pbmc_10k 这样的示例数据开始练习了。所以转发学徒的系列教程给大家。</p><ul data-tool="mdnice编辑器"><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247516873&amp;idx=1&amp;sn=856dbe03b49560bf31beb037e7251a29&amp;scene=21#wechat_redirect" data-linktype="2">scATAC-seq| 数据导入</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247516873&amp;idx=2&amp;sn=7cea3909b042272c1e87dff1470b58fc&amp;scene=21#wechat_redirect" data-linktype="2">scATAC-seq| 数据质控</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247516953&amp;idx=2&amp;sn=3fd443170c123d734945727d510582f7&amp;scene=21#wechat_redirect" data-linktype="2">scATAC-seq| 降维聚类</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247516958&amp;idx=1&amp;sn=d96fa5a32ae4484cb5503c0424ef3c73&amp;scene=21#wechat_redirect" data-linktype="2">整合scRNA和scATAC数据</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247516953&amp;idx=3&amp;sn=2b19785c81c2016d2d3fbb291b8f643f&amp;scene=21#wechat_redirect" data-linktype="2">scATAC-seq| 细胞注释(结合scRNA)</a></section></li></ul><p data-tool="mdnice编辑器">现在继续：</p></section><p data-tool="mdnice编辑器">上回的scRNA文件pbmc已经分好群了，接下来对motif进行分析</p><p data-tool="mdnice编辑器">对motif 简单的介绍：<a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzkyNTIwMTY2OA==&amp;mid=2247484543&amp;idx=1&amp;sn=4dca0620ea6e636a27775391ef573b2d&amp;chksm=c1cb6014f6bce902c56cd71ffd536b75769e239c670f7061ac80bfd3ae450f9ea41679d3ff49&amp;scene=21#wechat_redirect" textvalue="单细胞还能拉上转录因子？？？" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">单细胞还能拉上转录因子？？？</a></p><h3 data-tool="mdnice编辑器"><span></span>流程<span></span></h3><ul><li><p data-tool="mdnice编辑器">载入数据</p></li><li><p data-tool="mdnice编辑器">数据库 JASPAR2020 获取motif矩阵：<code>getMatrixSet</code></p></li><li><p data-tool="mdnice编辑器">添加motif矩阵：<code>AddMotifs</code></p></li><li><p data-tool="mdnice编辑器">计算motif活性分数:<code>RunChromVAR</code></p></li><li><p data-tool="mdnice编辑器">寻找显著差异的peaks：<code>FindMarkers</code></p></li><li><p data-tool="mdnice编辑器">peaks对应的motif :<code>FindMotifs</code></p></li><li><p data-tool="mdnice编辑器">可视化：<code>MotifPlot</code> and <code>FeaturePlot</code></p></li></ul><h2 data-tool="mdnice编辑器"><span></span>1. 载入数据</h2><p data-tool="mdnice编辑器">注意：<code>DefaultAssay(pbmc)&lt;- 'peaks'</code>要改成peaks，已经不是分群时候用到的RNA了</p><pre data-tool="mdnice编辑器"><span></span><code>library(Signac)<br>library(Seurat)<br><span>#BiocManager::install('TFBSTools')</span><br>library(JASPAR2020)<br>library(TFBSTools)<br>library(motifmatchr)<br>library(BSgenome.Hsapiens.UCSC.hg19)<br>library(chromVAR)<br>library(ggseqlogo)<br>set.seed(1234)<br><br><span># 载入数据</span><br>pbmc &lt;- readRDS(<span>"/home/data/yulan/scATAC/data/1.4scATAC_fina.rds"</span>)<br>pbmc<br><span># An object of class Seurat </span><br><span># 107571 features across 7060 samples within 2 assays </span><br><span># Active assay: RNA (20010 features, 0 variable features)</span><br><span># 1 other assay present: peaks</span><br><span># 2 dimensional reductions calculated: lsi, umap</span><br><br><span># 选择peaks</span><br>pbmc@assays<br><span># $peaks</span><br><span># ChromatinAssay data with 87561 features for 7060 cells</span><br><span># Variable features: 43781 </span><br><span># Genome: hg19 </span><br><span># Annotation present: TRUE </span><br><span># Motifs present: FALSE </span><br><span># Fragment files: 1 </span><br><span># </span><br><span># $RNA</span><br><span># Assay data with 20010 features for 7060 cells</span><br><span># First 10 features:</span><br><span>#   PLCXD1, GTPBP6, PPP2R3B, SHOX, CRLF2, CSF2RA, IL3RA, SLC25A6, ASMTL, P2RY8</span><br>DefaultAssay(pbmc) &lt;- <span>'peaks'</span><br></code></pre><h2 data-tool="mdnice编辑器"><span></span>2. 获取motif矩阵</h2><ul><li><p data-tool="mdnice编辑器"><span>从数据库 JASPAR2020 获取motif矩阵</span>：<code>getMatrixSet</code></p></li></ul><pre data-tool="mdnice编辑器"><span></span><code><span># 从公共数据库中获取motif矩阵</span><br><span># Get a list of motif position frequency matrices from the JASPAR database</span><br><span># http://www.bioconductor.org/packages/release/data/annotation/vignettes/JASPAR2020/inst/doc/JASPAR2020.html</span><br><span># https://jaspar.genereg.net/</span><br>pfm &lt;- getMatrixSet(<br>  x = JASPAR2020,<br>  opts = list(collection = <span>"CORE"</span>, <br>              tax_group = <span>'vertebrates'</span>, <span># 脊椎动物</span><br>              all_versions = FALSE)<br>)<br>pfm<br><span># PFMatrixList of length 746</span><br><span># names(746): MA0004.1 MA0006.1 MA0019.1 MA0029.1 MA0030.1 ... MA0526.3 MA0748.2 MA0528.2 MA0609.2</span><br></code></pre><blockquote data-type="2" data-url="" data-author-name="" data-content-utf8-length="192" data-source-title=""><section><section>JASPAR是一个免费公开的转录因子数据库，在该数据库中收录了转录因子(TF)的motif信息，可以用来预测转录因子与序列的结合区域。存储六个分类组中多个物种的TF的位置频率矩阵（PFM）。在JASPAR的第8版中，CORE系列已扩展为245个新的PFM（脊椎动物169个，植物42个，线虫17个，昆虫10个，真菌7个），更新了156个PFM（脊椎动物125个，植物28个，昆虫3个）</section></section></blockquote><p data-tool="mdnice编辑器"><img data-ratio="0.48802083333333335" data-src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc7GUosibw94yaa0ouv98MabfKnicLfdCwDicNu246jx78f0Fkabfa2ShuyTzzr3zvMXQdQUwQeXk6RicQ/640?wx_fmt=png" data-type="png" data-w="1920" src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc7GUosibw94yaa0ouv98MabfKnicLfdCwDicNu246jx78f0Fkabfa2ShuyTzzr3zvMXQdQUwQeXk6RicQ/640?wx_fmt=png"></p><p data-tool="mdnice编辑器"><span><br></span></p><ul><li><p data-tool="mdnice编辑器"><span>添加motif矩阵</span>：<code>AddMotifs</code></p></li></ul><pre data-tool="mdnice编辑器"><span></span><code><span># add motif information</span><br>pbmc &lt;- AddMotifs(<br>  object = pbmc,<br>  genome = BSgenome.Hsapiens.UCSC.hg19,<br>  pfm = pfm<br>)<br><span># Building motif matrix</span><br><span># Finding motif positions</span><br><span># Creating Motif object</span><br><br><span># 得到motifs矩阵</span><br>GetAssayData(object = pbmc, slot = <span>"motifs"</span>)<br><span># A Motif object containing 746 motifs in 87561 regions</span><br></code></pre><ul><li><p data-tool="mdnice编辑器"><span>计算motif活性分数</span>:<code>RunChromVAR</code></p></li></ul><pre data-tool="mdnice编辑器"><span></span><code><span># 计算 motif deviation score</span><br>pbmc &lt;- RunChromVAR(<br>  object = pbmc,<br>  genome = BSgenome.Hsapiens.UCSC.hg19<br>)<br><span># Computing GC bias per region</span><br><span># Selecting background regions</span><br><span># Computing deviations from background</span><br><span># Constructing chromVAR assay</span><br><br><span># 新增了一个assay：chromvar</span><br>pbmc<br><span># An object of class Seurat </span><br><span># 108317 features across 7060 samples within 3 assays </span><br><span># Active assay: peaks (87561 features, 43781 variable features)</span><br><span># 2 other assays present: RNA, chromvar</span><br><span># 2 dimensional reductions calculated: lsi, umap</span><br></code></pre><ul><li><p data-tool="mdnice编辑器"><span>寻找显著差异的 peaks</span>：<code>FindMarkers</code></p></li></ul><pre data-tool="mdnice编辑器"><span></span><code><span># 显著差异的peaks</span><br>DimPlot(pbmc) + DimPlot(pbmc,group.by = <span>'predicted.id'</span>)<br>table(pbmc<span>$predicted</span>.id)<br>table(Idents(pbmc))<br>Idents(pbmc) &lt;- pbmc<span>$predicted</span>.id<br><br><span># 对其中两个细胞亚群差异分析</span><br>da_peaks &lt;- Seurat::FindMarkers(<br>  object = pbmc,<br>  ident.1 = <span>'CD4 Naive'</span>,<br>  ident.2 = <span>'CD8 Naive'</span>,<br>  only.pos = TRUE,<br>  test.use = <span>'LR'</span>,<span># 推荐用logistic regression</span><br>  min.pct = 0.05,<span># 该基因表达数目占该类细胞总数的比例</span><br>  latent.vars = <span>'nCount_peaks'</span><br>)<br><span># get top differentially accessible peaks</span><br>top.da.peak &lt;- rownames(da_peaks[da_peaks<span>$p_val</span> &lt; 0.005, ])<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>&gt; head(da_peaks)<br>                                p_val avg_log2FC pct.1 pct.2    p_val_adj<br>chr1-234544345-234545516 8.454563e-16  0.5295198 0.191 0.022 7.402900e-11<br>chr6-167362837-167366950 1.648728e-13  0.2943871 0.285 0.085 1.443643e-08<br>chr12-6895592-6896474    3.370162e-13  0.4385442 0.123 0.006 2.950948e-08<br>chr12-6898094-6899126    8.623449e-13  0.2885348 0.106 0.003 7.550778e-08<br>chr9-90340551-90341713   1.854913e-12  0.2595704 0.142 0.016 1.624180e-07<br>chr12-6899696-6902624    1.489885e-11  0.2842456 0.206 0.041 1.304558e-06<br></code></pre><ul><li><p data-tool="mdnice编辑器"><span>求peaks对应的 motifs:</span> <code>FindMotifs</code></p></li></ul><pre data-tool="mdnice编辑器"><span></span><code><span># 求peaks对应的motifs</span><br><span># test enrichment</span><br>enriched.motifs &lt;- FindMotifs(<br>  object = pbmc,<br>  features = top.da.peak<br>)<br>head(enriched.motifs)<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>&gt; head(enriched.motifs)<br>            motif observed background percent.observed percent.background fold.enrichment<br>MA0769.2 MA0769.2       19       5146         41.30435            12.8650        3.210598<br>MA0523.1 MA0523.1       20       6377         43.47826            15.9425        2.727192<br>MA0162.4 MA0162.4       19       5961         41.30435            14.9025        2.771639<br>MA0733.1 MA0733.1       14       3953         30.43478             9.8825        3.079664<br>MA1650.1 MA1650.1       13       3442         28.26087             8.6050        3.284238<br>MA0732.1 MA0732.1       13       4031         28.26087            10.0775        2.804353<br>               pvalue motif.name    p.adjust<br>MA0769.2 1.475866e-06       TCF7 0.001100996<br>MA0523.1 8.828251e-06     TCF7L2 0.003292938<br>MA0162.4 1.336985e-05       EGR1 0.003324637<br>MA0733.1 9.317482e-05       EGR4 0.013940194<br>MA1650.1 9.343293e-05     ZBTB14 0.013940194<br>MA0732.1 4.502527e-04       EGR3 0.048616281<br></code></pre><ul><li><p data-tool="mdnice编辑器"><span>motif 可视化</span>：<code>MotifPlot</code> and <code>FeaturePlot</code></p></li></ul><pre data-tool="mdnice编辑器"><span></span><code>head(rownames(enriched.motifs)<br><span># [1] "MA0769.2" "MA0523.1" "MA0162.4" "MA0733.1" "MA1650.1" "MA0732.1"</span><br>MotifPlot(<br>  object = pbmc,<br>  motifs = head(rownames(enriched.motifs)),<br>  assay = <span>'peaks'</span><br>)<br><span># 默认以转录因子为作图标题</span><br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.4209905660377358" data-src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc7GUosibw94yaa0ouv98MabfLazaDmqicwnqFn9DYyUuJsnwvGEIRkrQQpVr9tTCVKm6ibR5JeknmcLQ/640?wx_fmt=png" data-type="png" data-w="848" src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc7GUosibw94yaa0ouv98MabfLazaDmqicwnqFn9DYyUuJsnwvGEIRkrQQpVr9tTCVKm6ibR5JeknmcLQ/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code>FeaturePlot(<br>  object = pbmc,<br>  features = head(rownames(enriched.motifs)),<br>  min.cutoff = <span>'q10'</span>,<br>  max.cutoff = <span>'q90'</span>,<br>  pt.size = 0.1<br>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7833333333333333" data-src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc7GUosibw94yaa0ouv98Mabfg85GvOjfibIlghqQg8Oh4A8ic2jK7Ia0CKCDxCWLwBCLWbkt16hRLACA/640?wx_fmt=png" data-type="png" data-w="900" src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc7GUosibw94yaa0ouv98Mabfg85GvOjfibIlghqQg8Oh4A8ic2jK7Ia0CKCDxCWLwBCLWbkt16hRLACA/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">同样的方法，<strong>寻找显著差异的 motifs</strong></p><pre data-tool="mdnice编辑器"><span></span><code><span># 显著差异的 motifs</span><br>DefaultAssay(pbmc) &lt;- <span>'chromvar'</span><br>da_motifs &lt;- Seurat::FindMarkers(<br>  object = pbmc,<br>  ident.1 = <span>'CD4 Naive'</span>,<br>  ident.2 = <span>'CD8 Naive'</span>,<br>  only.pos = TRUE,<br>  test.use = <span>'LR'</span>, <span># 推荐用logistic regression</span><br>  min.pct = 0.05, <span># 该基因表达数目占该类细胞总数的比例</span><br>  latent.vars = <span>'nCount_peaks'</span><br>)<br>head(da_motifs)<br><span>#                 p_val avg_log2FC pct.1 pct.2  p_val_adj</span><br><span># MA0050.2 2.459215e-05  0.2994300 0.186 0.110 0.01834574</span><br><span># MA0038.2 2.275283e-04  0.4947875 0.428 0.312 0.16973610</span><br><span># MA0836.2 3.334793e-04  2.0343057 0.062 0.025 0.24877553</span><br><span># MA1418.1 3.555421e-04  0.3582367 0.367 0.274 0.26523439</span><br><span># MA0826.1 3.669158e-04  0.5638397 0.484 0.394 0.27371917</span><br><span># MA0466.2 4.664331e-04  1.6827524 0.054 0.032 0.34795909</span><br></code></pre><p data-tool="mdnice编辑器"><br></p><p data-tool="mdnice编辑器">参考:https://satijalab.org/signac/articles/motif_vignette.html</p></section><h4 data-tool="mdnice编辑器"><span>文末友情宣传</span></h4><p data-tool="mdnice编辑器">强烈建议你推荐给身边的<strong>博士后以及年轻生物学PI</strong>，多一点数据认知，让他们的科研上一个台阶：</p><ul data-tool="mdnice编辑器"><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247519083&amp;idx=1&amp;sn=61f07b390469617ca2e9924f5b3929ab&amp;scene=21#wechat_redirect" data-linktype="2">生物信息学马拉松授课（买一得五）</a> ，你的生物信息学入门课</section></li></ul><p><br></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/iawtsDsw2U1Cfywj2e0cHw",target="_blank" rel="noopener noreferrer">原文链接</a>
