---
title: "scATAC-seq| 可视化DARs"
date: 2022-12-31T01:43:15Z
draft: ["false"]
tags: [
  "fetched",
  "生信技能树"
]
categories: ["Acdemic"]
---
scATAC-seq| 可视化DARs by 生信技能树
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><blockquote data-tool="mdnice编辑器"><p>在单细胞转录组冒尖的时候，单细胞其它组学却一直不温不火，比如单细胞ATAC以及单细胞免疫组库。所以虽然很早之前我们就推荐了github代码：https://github.com/GreenleafLab/10x-scATAC-2019</p></blockquote><ul data-tool="mdnice编辑器"><li><section>01_Filter_Cells_v2.R</section></li><li><section>02_Get_Peak_Set_hg19_v2.R</section></li><li><section>03_Run_chromVAR_v2.R</section></li><li><section>04_Run_Cicero_v2.R</section></li><li><section>05_Cluster_Unique_Peaks_v2.R</section></li><li><section>06_Analyze_UMAP_Trajectory.R</section></li><li><section>07_ChromVAR_For_GWAS_w_CoAccessbility_v2.R</section></li><li><section>08_Run_scCNV_v2.R</section></li></ul><p data-tool="mdnice编辑器">但是一直没有深入解读它，在2021的时候《单细胞天地》的一个小伙伴写了个开头：</p><ul data-tool="mdnice编辑器"><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247508480&amp;idx=1&amp;sn=15360a6b494d7a536fb8c790dc54a585&amp;scene=21#wechat_redirect" data-linktype="2">scATAC-seq1：由转录组到表观组</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247508480&amp;idx=2&amp;sn=4548d7aa6ba7185426fd0040cda25f73&amp;scene=21#wechat_redirect" data-linktype="2">scATAC-seq2: scATAC-seq技术原理</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247508480&amp;idx=3&amp;sn=51825eb6c789d97682d558fb34b46044&amp;scene=21#wechat_redirect" data-linktype="2">scATAC-seq3:常用工具—SnapATAC简介</a></section></li></ul><p data-tool="mdnice编辑器">但是没有能坚持下来，其实文章给的配套github代码非常齐全了，就是需要花时间钻研和解读。</p><p data-tool="mdnice编辑器">恰好2022我们又看到了一个带有全套github代码以及配套数据的文章，见：<a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247516871&amp;idx=1&amp;sn=7eed3a5b5b6575458ec44ee9ddd0b6de&amp;scene=21#wechat_redirect" data-linktype="2">10x的单细胞ATAC上游流程之cellranger-atac</a>，在等待的期间，交流群的学徒就先拿 atac_v1_pbmc_10k 这样的示例数据开始练习了。所以转发学徒的系列教程给大家。</p><ul data-tool="mdnice编辑器"><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247516873&amp;idx=1&amp;sn=856dbe03b49560bf31beb037e7251a29&amp;scene=21#wechat_redirect" data-linktype="2">scATAC-seq| 数据导入</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247516873&amp;idx=2&amp;sn=7cea3909b042272c1e87dff1470b58fc&amp;scene=21#wechat_redirect" data-linktype="2">scATAC-seq| 数据质控</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247516953&amp;idx=2&amp;sn=3fd443170c123d734945727d510582f7&amp;scene=21#wechat_redirect" data-linktype="2">scATAC-seq| 降维聚类</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247516958&amp;idx=1&amp;sn=d96fa5a32ae4484cb5503c0424ef3c73&amp;scene=21#wechat_redirect" data-linktype="2">整合scRNA和scATAC数据</a></section></li><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247516953&amp;idx=3&amp;sn=2b19785c81c2016d2d3fbb291b8f643f&amp;scene=21#wechat_redirect" data-linktype="2">scATAC-seq| 细胞注释(结合scRNA)</a></section></li></ul><p data-tool="mdnice编辑器">现在继续：</p></section><p data-tool="mdnice编辑器">上回求出差异的peaks之后，还能挖掘更多的数据信息：</p><p data-tool="mdnice编辑器">例如用热图来展示结果，并且求peak区域附近的基因，看其分布位置</p><p data-tool="mdnice编辑器">↓<span>↓<span>↓</span></span></p><p data-tool="mdnice编辑器">To investigate the differences in chromatin accessibility among all cell types, we first identified a total of 212,326 peaks in our scATAC-seq data by using MACS223 and found that ~10.6% (22,682 unique peaks) of them exhibited significant differences among cell types; these sites were defined as differentially accessible chromatin regions (DARs) (adjusted P &lt; 0.05 and log2(fold change (FC)) &gt; 0.25)(Fig. 2a)</p><p data-tool="mdnice编辑器">The majority of DARs were located in the promoter and intronic regions of the genome, and the distribution of DARs was relatively conserved across cell types (Fig. 2b and Supplementary Fig. S2b).</p><p data-tool="mdnice编辑器">文章用MACS223求出scATAC-seq中的peak，随后用过设置阈值求出有显著差异的<span>染色质开放区域（differentially accessible chromatin regions (DARs)即有显著差异的peak</span>）及其基因的分布位置</p><figure data-tool="mdnice编辑器"><img data-ratio="0.5671277461350691" data-src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc43xAHiaICic5zHFOUAnb8k6XwwgCh1s8p3dxMO3O3Mmk6aOkH8ibZXntEJELtY9LjD3m9HqGUd5zMgA/640?wx_fmt=png" data-type="png" data-w="1229" src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc43xAHiaICic5zHFOUAnb8k6XwwgCh1s8p3dxMO3O3Mmk6aOkH8ibZXntEJELtY9LjD3m9HqGUd5zMgA/640?wx_fmt=png"></figure><blockquote data-type="2" data-url="" data-author-name="" data-content-utf8-length="97" data-source-title=""><section><section>文章：Single-cell multiomics analysis reveals regulatory programs in clear cell renal cell carcinoma</section></section></blockquote><p><br></p><h2 data-tool="mdnice编辑器"><span></span>流程：</h2><ul><li><p data-tool="mdnice编辑器">寻找差异的peaks：<code>FindAllMarkers</code> or <code>FindMarkers</code></p></li><li><p data-tool="mdnice编辑器">peaks附近的基因：<code>ClosestFeature</code></p></li><li><p data-tool="mdnice编辑器">设置阈值：求出DAR</p></li><li><p data-tool="mdnice编辑器">求平均表达量：<code>AverageExpression</code></p></li><li><p data-tool="mdnice编辑器">可视化：热图，饼图，直方图，条形图等</p></li></ul><h2 data-tool="mdnice编辑器">1.<span> 显著差异的peaks</span></h2><pre data-tool="mdnice编辑器"><span></span><code><span># 确保已经分好细胞群，不再是序号</span><br>table(Idents(pbmc))<br><span># CD14+ Monocytes                 NK dim             CD4 Memory              CD4 Naive </span><br><span># 3122                             360                    820                    698 </span><br><span># Double negative T cell           CD8 effector              CD8 Naive        CD16+ Monocytes </span><br><span># 391                                563                    317                    217 </span><br><span># pre-B cell      B cell progenitor         Dendritic cell                    pDC </span><br><span># 280                    170                     68                            49 </span><br><span># NK bright </span><br><span># 5 </span><br>levels(pbmc)<br><span># [1] "CD14+ Monocytes"        "NK dim"                 "CD4 Memory"            </span><br><span># [4] "CD4 Naive"              "Double negative T cell" "CD8 effector"          </span><br><span># [7] "CD8 Naive"              "CD16+ Monocytes"        "pre-B cell"            </span><br><span># [10] "B cell progenitor"      "Dendritic cell"         "pDC"                   </span><br><span># [13] "NK bright"</span><br><br><span>############################# identify differentially accessible chromatin regions between celltypes</span><br>DefaultAssay(pbmc) &lt;- <span>"peaks"</span><br><span># 寻找显著差异的peaks</span><br><span># cellType.DARs &lt;- FindAllMarkers(scATAC.data, </span><br><span>#                                 test.use = 'LR',</span><br><span>#                                 logfc.threshold=0, </span><br><span>#                                 min.pct = 0.05, # often necessary to lower the min.pct threshold</span><br><span>#                                 latent.vars = "peak_region_fragments")</span><br><br><span># 运行比较慢，这里只比较两个细胞群的差异peaks</span><br>cellType.DARs &lt;- Seurat::FindMarkers(<br>  object = pbmc,<br>  ident.1 = <span>'CD4 Naive'</span>,<br>  ident.2 = <span>'CD8 Naive'</span>,<br>  logfc.threshold=0,  <span># 默认是0.25</span><br>  only.pos = TRUE,<br>  test.use = <span>'LR'</span>, <span># 推荐用logistic regression</span><br>  min.pct = 0.05, <span># 该基因表达数目占该类细胞总数的比例</span><br>  latent.vars = <span>'nCount_peaks'</span><br>)<br>head(cellType.DARs)<br><span>#                          p_val         avg_log2FC pct.1 pct.2    p_val_adj</span><br><span># chr1-234544345-234545516 8.454563e-16  0.5295198 0.191 0.022 7.402900e-11</span><br><span># chr16-79632101-79636717  1.431972e-13  0.2235968 0.268 0.073 1.253849e-08</span><br><span># chr6-167362837-167366950 1.648728e-13  0.2943871 0.285 0.085 1.443643e-08</span><br><span># chr12-6895592-6896474    3.370162e-13  0.4385442 0.123 0.006 2.950948e-08</span><br><span># chr12-6898094-6899126    8.623449e-13  0.2885348 0.106 0.003 7.550778e-08</span><br><span># chr9-90340551-90341713   1.854913e-12  0.2595704 0.142 0.016 1.624180e-07</span><br>dim(cellType.DARs) <span>#12642     5</span><br></code></pre><p data-tool="mdnice编辑器">有些教程在<code>FindMarkers</code>用的变量是<code>latent.vars = "peak_region_fragments"</code>，其实出来的结果差不多，因其相关性很高</p><pre data-tool="mdnice编辑器"><span></span><code>cor(pbmc<span>$peak_region_fragments</span>, pbmc<span>$nCount_peaks</span>) <span># 0.999884</span><br></code></pre><h2 data-tool="mdnice编辑器">2. <span>peaks附近的基因</span></h2><p data-tool="mdnice编辑器">peaks附近的基因: <code>ClosestFeature</code></p><pre data-tool="mdnice编辑器"><span></span><code><span># 寻找附近的基因</span><br>cf &lt;- ClosestFeature(pbmc, regions = rownames(cellType.DARs))<span># Find the closest feature to a given set of genomic regions</span><br>head(cf)<br><br>table(rownames(cellType.DARs) %<span>in</span>% cf<span>$query_region</span>) <span># 对应peaks</span><br><span># TRUE </span><br><span># 12642</span><br>table(rownames(cellType.DARs) %<span>in</span>% cf<span>$closest_region</span>) <span># 对应基因的region</span><br><span># FALSE </span><br><span># 12642</span><br></code></pre><p data-tool="mdnice编辑器">这里就把求出的<code>regions = rownames(cellType.DARs)</code>当成所有细胞群差异的peak，如果着重研究某两个细胞群的差异，<code>regions</code>就需要改变一下</p><pre data-tool="mdnice编辑器"><span></span><code>open_CD4_Naive &lt;- rownames(cellType.DARs[cellType.DARs<span>$avg_log2FC</span> &gt; 0.5, ])<br>open_CD8_Naive &lt;- rownames(cellType.DARs[cellType.DARs<span>$avg_log2FC</span> &lt; -0.5, ])<br><br>closest_genes_CD4_Naive &lt;- ClosestFeature(pbmc, regions = open_CD4_Naive)<br>closest_genes_CD8_Naive &lt;- ClosestFeature(pbmc, regions = open_CD8_Naive)<br></code></pre><p data-tool="mdnice编辑器">结果：</p><ul><li><p data-tool="mdnice编辑器"><code>cf$query_region：</code>输入的peaks</p></li><li><p data-tool="mdnice编辑器"><code>cf$closest_region：</code>基因的region</p></li></ul><pre data-tool="mdnice编辑器"><span></span><code>&gt; head(cf)<br>                          tx_id gene_name         gene_id   gene_biotype <span>type</span><br>ENST00000481183 ENST00000481183    TARBP1 ENSG00000059588 protein_coding  gap<br>ENST00000569649 ENST00000569649       MAF ENSG00000178573 protein_coding  cds<br>ENST00000508775 ENST00000508775   RNASET2 ENSG00000026297 protein_coding  cds<br>ENST00000535707 ENST00000535707       CD4 ENSG00000010610 protein_coding  gap<br>ENST00000541982 ENST00000541982       CD4 ENSG00000010610 protein_coding  utr<br>ENST00000343150 ENST00000343150      CTSL ENSG00000135047 protein_coding  utr<br>                          closest_region             query_region distance<br>ENST00000481183 chr1-234541846-234546190 chr1-234544345-234545516        0<br>ENST00000569649  chr16-79632682-79633799  chr16-79632101-79636717        0<br>ENST00000508775 chr6-167365976-167366036 chr6-167362837-167366950        0<br>ENST00000535707    chr12-6896172-6898654    chr12-6895592-6896474        0<br>ENST00000541982    chr12-6898702-6898828    chr12-6898094-6899126        0<br>ENST00000343150   chr9-90340434-90341313   chr9-90340551-90341713        0<br>&gt; table(cf<span>$gene_biotype</span>)<br><br>             lincRNA processed_transcript       protein_coding                 rRNA <br>                1017                  114                11490                   21 <br>&gt; table(cf<span>$type</span>)<br><br> cds exon  gap  utr <br>5465 1781 2724 2672 <br></code></pre><p data-tool="mdnice编辑器">这里出来的基因类型和位置都可以用饼图来展示</p><h2 data-tool="mdnice编辑器">3. <span>求平均表达量</span></h2><ul><li><p data-tool="mdnice编辑器"><code>AverageExpression</code></p></li></ul><pre data-tool="mdnice编辑器"><span></span><code><span># 求表达量，画热图</span><br>cellType.DARs &lt;- cbind(cellType.DARs, <br>                       gene=cf<span>$gene_name</span>, <br>                       query_region = cf<span>$query_region</span>, <br>                       <span>type</span> = cf<span>$type</span>, <br>                       distance=cf<span>$distance</span>)<br><span># 设置阈值</span><br><span>#require logfc.threshold &gt;= 0.25 &amp; p_val_adj &lt; 0.05</span><br>cellType.sig.pos.DARs &lt;- cellType.DARs %&gt;% dplyr::filter(avg_log2FC &gt;=0.25 &amp; p_val_adj &lt; 0.05) %&gt;% arrange(desc(avg_log2FC)) <br>dim(cellType.sig.pos.DARs) <span>#19  9</span><br><span>#plot--- differentially accessible chromatin regions heatmap</span><br>sig.region &lt;- cellType.sig.pos.DARs %&gt;% dplyr::select(query_region) %&gt;% distinct() <br><br><span># 求平均表达量</span><br><span>#average fragment of each peak in each cell type</span><br>sig.region.mean &lt;- AverageExpression(pbmc, <br>                                     features = sig.region<span>$query_region</span>, <br>                                     assays = <span>"peaks"</span>)<br><br>sig.region.mean.scale &lt;- scale(t(sig.region.mean<span>$peaks</span>))<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>&gt; sig.region.mean.scale[1:3,]<br>                chr1-234544345-234545516 chr12-6895592-6896474 chr1-213843536-213844175<br>CD14+ Monocytes               -0.4226576            -0.2815046               -0.3905448<br>NK dim                        -0.4918933            -0.5234233               -0.4341829<br>CD4 Memory                     0.9332080             0.6218469                0.4443438<br>                chr2-112938926-112941640 chr14-59103807-59105663 chr21-46890735-46891216<br>CD14+ Monocytes               -0.5949016              -0.5212354              -0.2926606<br>NK dim                        -0.5968046              -0.5656007              -0.3216966<br>CD4 Memory                     1.1153124              -0.6531111               0.7722765<br>                chr15-57883649-57884489 chr3-118937835-118938422 chr3-32279336-32281981<br>CD14+ Monocytes              -0.4343374              -0.66028682             -0.4855104<br>NK dim                       -0.9293215              -0.15753047             -0.4648244<br>CD4 Memory                    0.1321672               0.09150125              1.4333515<br>                chr21-37660946-37662184 chr8-18870780-18872100 chr6-11728007-11729822<br>CD14+ Monocytes              -0.7490132              0.3656120             0.35311667<br>NK dim                       -0.7044903             -0.6868159            -0.72428267<br>CD4 Memory                    1.5498568             -0.3555641             0.05747466<br>                chr10-35300735-35301082 chr6-167362837-167366950 chr12-6898094-6899126<br>CD14+ Monocytes             -0.23194611               -0.8103845             1.1220201<br>NK dim                      -0.32961628               -0.7481198            -0.7828576<br>CD4 Memory                   0.09374038                0.5397166             0.8174112<br>                chr19-52191949-52194762 chr12-6899696-6902624 chr10-62491054-62493599 chr9-90340551-90341713<br>CD14+ Monocytes              -0.3312373             1.2080742              -1.2251931              1.9848987<br>NK dim                       -0.7100521            -0.8918025              -0.6249917             -0.5224442<br>CD4 Memory                    1.4545402             0.3754083               0.7368930             -0.2492964<br></code></pre><h2 data-tool="mdnice编辑器">4. <span>画热图</span></h2><pre data-tool="mdnice编辑器"><span></span><code><span># ?Heatmap</span><br>Heatmap(sig.region.mean.scale, <br>        name = <span>"z-score"</span>, <br>        show_column_dend = F, <br>        show_row_dend = F, <br>        row_names_side = <span>"left"</span>,<br>        show_column_names = F, <br>        row_names_gp = gpar(fontsize = 10), <br>        width = unit(10, <span>"cm"</span>), <br>        height = unit(8, <span>"cm"</span>),<br>        <span># heatmap_legend_param = list(direction = "horizontal")</span><br>        show_heatmap_legend=F) <br>library(circlize)<br><span># 添加注释</span><br>col_fun = colorRamp2(c(-4,0,4), c(<span>"blue"</span>, <span>"white"</span>, <span>"red"</span>))<br>lgd = Legend(col_fun = col_fun, <br>             title = <span>"z-score"</span>,<br>             direction = <span>"horizontal"</span>, <span># 横向注释</span><br>             title_position=<span>'lefttop'</span>)<br>lgd<br>draw(lgd, x = unit(20.7, <span>"cm"</span>), y = unit(7.4, <span>"cm"</span>)) <span># 调整注释位置</span><br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.6920353982300885" data-src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc43xAHiaICic5zHFOUAnb8k6X91zamRowibZjVicdg8jpEen5GMBLyIdicxEo5l4yEsRmPIrO2lialzDw3Q/640?wx_fmt=png" data-type="png" data-w="565" src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc43xAHiaICic5zHFOUAnb8k6X91zamRowibZjVicdg8jpEen5GMBLyIdicxEo5l4yEsRmPIrO2lialzDw3Q/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">参考：https://satijalab.org/signac/reference/closestfeature</p></section><h4 data-tool="mdnice编辑器"><span>文末友情宣传</span></h4><p data-tool="mdnice编辑器">强烈建议你推荐给身边的<strong>博士后以及年轻生物学PI</strong>，多一点数据认知，让他们的科研上一个台阶：</p><ul data-tool="mdnice编辑器"><li><section><a href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247519083&amp;idx=1&amp;sn=61f07b390469617ca2e9924f5b3929ab&amp;scene=21#wechat_redirect" data-linktype="2">生物信息学马拉松授课（买一得五）</a> ，你的生物信息学入门课</section></li></ul></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/u2I12GvR3YJJ5h6EJktXcQ",target="_blank" rel="noopener noreferrer">原文链接</a>
