---
title: "simspec：基于细胞簇谱系相似性整合单细胞数据"
date: 2022-09-15T14:58:12Z
draft: ["false"]
tags: [
  "fetched",
  "单细胞天地"
]
categories: ["Acdemic"]
---
simspec：基于细胞簇谱系相似性整合单细胞数据 by 单细胞天地
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><h2 data-tool="mdnice编辑器"><span></span></h2><p data-mpa-powered-by="yiban.io"><img data-ratio="0.11424100156494522" data-src="https://mmbiz.qpic.cn/mmbiz_gif/Ljib4So7yuWiaMArGTHZ6TNc7h6HTsh9nNxWsZHiapK6rO8puD1LjwzGcA80hM15GWBakhnUpTwYRrAHPwJ29ZAlw/640?wx_fmt=gif&amp;wxfrom=5&amp;wx_lazy=1" data-type="gif" data-w="639" src="https://mmbiz.qpic.cn/mmbiz_gif/Ljib4So7yuWiaMArGTHZ6TNc7h6HTsh9nNxWsZHiapK6rO8puD1LjwzGcA80hM15GWBakhnUpTwYRrAHPwJ29ZAlw/640?wx_fmt=gif&amp;wxfrom=5&amp;wx_lazy=1"><br></p><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><h2 data-tool="mdnice编辑器"><span>简介</span></h2></section><p data-tool="mdnice编辑器">在单细胞数据分析中，如何对跨不同实验、条件、批次、时间点或其他技术因素的单细胞测序数据进行多样本整合仍然是一个重大的挑战。需要新的计算方法来整合多个样本矫正批次效应，同时保留不同样本之间内在的生物学信息。在这里，我们提出了一种无监督参考的数据表示方法，即细胞簇谱系相似性(CSS)，其中每个细胞由其与不同样本中独立识别的细胞簇的相似性来进行表示。</p><p data-tool="mdnice编辑器">CSS的概念是用不同样本中细胞簇的相似性来表示每个细胞。由于技术差异可能以不同的方式影响不同的样本，因此不同样本聚类之间的相似性很可能是不可比较的。因此，我们需要对每个样本中不同细胞簇间的相似性分别进行归一化处理，然后将每个细胞对不同样本的归一化相似性合并起来。<img data-ratio="1.4879032258064515" data-src="https://mmbiz.qpic.cn/mmbiz_png/zN67RpvO3ka3yKZByMPiaCKgwjibb0jeznic7UX16UST1kxceZCc6T41QoU2sPndyrxIGdGcngKDQj28TVTVQ0o8A/640?wx_fmt=png" data-type="png" data-w="1240" src="https://mmbiz.qpic.cn/mmbiz_png/zN67RpvO3ka3yKZByMPiaCKgwjibb0jeznic7UX16UST1kxceZCc6T41QoU2sPndyrxIGdGcngKDQj28TVTVQ0o8A/640?wx_fmt=png"></p><p data-tool="mdnice编辑器">原则上，我们可以使用不同的方法计算细胞和细胞簇之间的相似性。同样，也可以使用不同的归一化策略。在<code>simspec</code>包中，我们基于在给定的基因列表(默认是高度变化的基因)中使用<strong>Spearman相关性</strong>(默认)或<strong>Pearson相关性</strong>作为相似性的度量。同时，提供了两种可选择的归一化方法:（1）Z-transformation（2）Kernel probability</p><p data-tool="mdnice编辑器">默认情况下，<code>simspec</code>包使用<strong>Spearman相关性</strong>和<strong>z-transformation</strong>归一化方法进行CSS计算。这是对于具有连续分化轨迹和中间状态的发育相关scRNA-seq数据集的首要选择。另一方面，<strong>Pearson相关性</strong>结合<strong>Kernel probability</strong>归一化方法更适合用于细胞类型构成较为离散的scRNA-seq数据集。</p><p data-tool="mdnice编辑器">参考文献：CSS: cluster similarity spectrum integration of single-cell genomics data.</p><figure data-tool="mdnice编辑器"><img data-ratio="0.35" data-src="https://mmbiz.qpic.cn/mmbiz_png/zN67RpvO3ka3yKZByMPiaCKgwjibb0jeznLfNs4QiaCkfx8EEkEQJNocibLibk4ic70qf8DraOkS4LqRtmI5nCr6KRJQ/640?wx_fmt=png" data-type="png" data-w="1240" src="https://mmbiz.qpic.cn/mmbiz_png/zN67RpvO3ka3yKZByMPiaCKgwjibb0jeznLfNs4QiaCkfx8EEkEQJNocibLibk4ic70qf8DraOkS4LqRtmI5nCr6KRJQ/640?wx_fmt=png"><figcaption><br></figcaption></figure><h2 data-tool="mdnice编辑器"><span></span><span>R包安装</span></h2><pre data-tool="mdnice编辑器"><span></span><code><span># install.packages(devtools)</span><br>devtools::install_github(<span>"quadbiolab/simspec"</span>)<br></code></pre><h2 data-tool="mdnice编辑器"><span></span><span>实例演示</span></h2><p data-tool="mdnice编辑器">接下来，我们将使用不同细胞系的人脑类器官单细胞数据集进行实例演示CSS方法整合单细胞数据。 </p><p data-tool="mdnice编辑器">数据集：The 10x scRNA-seq data of 20 two-month-old cerebral organoids with four batches (Kanton et al. 2019) </p><p data-tool="mdnice编辑器">下载链接：To retrieve the cerebral organoid data, please refer to the data archive in Mendeley Data.</p><h3 data-tool="mdnice编辑器"><span></span><span></span><span>加载所需R包和示例数据集</span><span></span></h3><pre data-tool="mdnice编辑器"><span></span><code>library(Seurat)<br>library(simspec)<br><br><span># counts &lt;- readMM("cerebral_organoids_Kanton_2019_org2m/matrix.mtx.gz")</span><br><span># meta &lt;- read.table("cerebral_organoids_Kanton_2019_org2m/cells.tsv.gz", sep="\t")</span><br><span># features &lt;- read.table("cerebral_organoids_Kanton_2019_org2m/features.tsv.gz", sep="\t", stringsAsFactors = F)</span><br><span># rownames(counts) &lt;- make.unique(features[,2])</span><br><span># colnames(counts) &lt;- rownames(meta)</span><br><span># org2m &lt;- CreateSeuratObject(counts = counts, meta.data = meta)</span><br><br>org2m &lt;- readRDS(<span>"cerebral_org2m.rds"</span>)<br>org2m<br><span># An object of class Seurat</span><br><span># 18147 features across 49153 samples within 1 assay</span><br><span># Active assay: RNA (18147 features, 0 variable features)</span><br><br>head(org2m@meta.data)<br><span>#          orig.ident nCount_RNA nFeature_RNA percent.mito batch</span><br><span>#HC00001 SeuratProject      16752         3287   0.01098442     1</span><br><span>#HC00003 SeuratProject      13533         3399   0.01950787     1</span><br><span>#HC00005 SeuratProject       3098         1558   0.02453988     1</span><br> <span>#              organoid   line            barcode region_RSS</span><br><span>#HC00001 h409B2_60d_org1 h409B2 AAACCTGAGAATGTTG-1         MH</span><br><span>#HC00003 h409B2_60d_org1 h409B2 AAACCTGAGAGCCTAG-1         MH</span><br><span>#HC00005 h409B2_60d_org1 h409B2 AAACCTGAGTAATCCC-1         MH</span><br><span>#                                    celltype_RSS region_pred neuron_vs_NPC</span><br><span>#HC00001                   Non-telencephalon NPCs    nontelen    -0.5854159</span><br><span>#HC00003                   Non-telencephalon NPCs    nontelen    -0.7554498</span><br><span>#HC00005 Mesen/rhombencephalon excitatory neurons    nontelen     0.5646920</span><br><span>#          celltype_pred</span><br><span>#HC00001    nontelen_NPC</span><br><span>#HC00003    nontelen_NPC</span><br><span>#HC00005 nontelen_neuron</span><br></code></pre><h3 data-tool="mdnice编辑器"><span></span><span></span><span>数据预处理</span><span></span></h3><p data-tool="mdnice编辑器">首先，我们需要对数据进行预处理：数据标准化、筛选高变异基因、PCA降维</p><pre data-tool="mdnice编辑器"><span></span><code>org2m &lt;- NormalizeData(org2m)<br><span>#Performing log-normalization</span><br><span>#0%   10   20   30   40   50   60   70   80   90   100%</span><br><span>#[----|----|----|----|----|----|----|----|----|----|</span><br><span>#**************************************************|</span><br><br>org2m &lt;- FindVariableFeatures(org2m, nfeatures = 5000)<br><span>#Calculating gene variances</span><br><span>#0%   10   20   30   40   50   60   70   80   90   100%</span><br><span>#[----|----|----|----|----|----|----|----|----|----|</span><br><span>#**************************************************|</span><br><span>#Calculating feature variances of standardized and clipped values</span><br><span>#0%   10   20   30   40   50   60   70   80   90   100%</span><br><span>#[----|----|----|----|----|----|----|----|----|----|</span><br><span>#**************************************************|</span><br><br>org2m &lt;- ScaleData(org2m)<br><span>#Centering and scaling data matrix</span><br><span>#  |======================================================================| 100%</span><br><br>org2m &lt;- RunPCA(org2m, npcs = 20)<br><span>#PC_ 1</span><br><span>#Positive:  STMN2, RTN1, HMP19, MAPT, MLLT11, SOX4, SYT1, GAP43, DCX, GPM6A</span><br><span>#           RAB3A, TMSB10, RBFOX2, NSG1, VAMP2, CD24, SCG3, CRMP1, SEZ6L2, PCSK1N</span><br><span>#           CXADR, MAP1B, GRIA2, STMN4, CNTNAP2, TTC9B, NEUROD2, BEX2, APLP1, RUNX1T1</span><br><span>#Negative:  VIM, HMGB2, SOX2, GNG5, SMC4, ZFP36L1, TUBA1B, SFRP1, HES1, KIAA0101</span><br><span>#           NUSAP1, PHGDH, TTYH1, TYMS, HSPB1, HMGN2, CKS1B, RPS6, FABP7, GSTP1</span><br><span>#           RCN1, CDK1, TOP2A, GAPDH, CKS2, PTTG1, SOX3, HMGN3, BIRC5, MKI67</span><br></code></pre><p data-tool="mdnice编辑器">UMAP降维可视化</p><pre data-tool="mdnice编辑器"><span></span><code>org2m &lt;- RunUMAP(org2m, dims = 1:20)<br>DimPlot(org2m, group.by = <span>"batch"</span>) + DimPlot(org2m, group.by = <span>"celltype_RSS"</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.4" data-src="https://mmbiz.qpic.cn/mmbiz_png/zN67RpvO3ka3yKZByMPiaCKgwjibb0jeznicj91X9TMWCeL2zib1dtiaNTjVYYhYorAKOaX4LVyEbgddicZllIeXLO8Q/640?wx_fmt=png" data-type="png" data-w="1000" src="https://mmbiz.qpic.cn/mmbiz_png/zN67RpvO3ka3yKZByMPiaCKgwjibb0jeznicj91X9TMWCeL2zib1dtiaNTjVYYhYorAKOaX4LVyEbgddicZllIeXLO8Q/640?wx_fmt=png"><figcaption><br></figcaption></figure><p data-tool="mdnice编辑器">可以看到，在未进行批次矫正时，不同样本之间相同的细胞类型分散在不同的区域中，存在较大的批次效应。</p><h3 data-tool="mdnice编辑器"><span></span><span></span><span>计算CSS细胞簇谱系相似性</span><span></span></h3><p data-tool="mdnice编辑器">接下来，我们将使用<code>simspec</code>包中的<code>cluster_sim_spectrum</code>函数计算CSS相似性，并进行批次矫正整合不同样本的数据。</p><p data-tool="mdnice编辑器"><code>label_tag</code><span>：</span><span> is the name of a column in the meta.data table that marks the sample/batch information for integration.</span></p><pre data-tool="mdnice编辑器"><span></span><code>org2m &lt;- cluster_sim_spectrum(object = org2m, label_tag = <span>"organoid"</span>, <br>                              spectrum_type = <span>"corr_ztransform"</span>,corr_method = <span>"spearman"</span>)<br></code></pre><p data-tool="mdnice编辑器">By default, it returns a new Seurat object with an additional reduction object named "css". It can be then used as the reduction methods for following analysis including creating UMAP embedding and clustering. It is worth to mention that all dimensions in the raw CSS output should be used in any following analysis.</p><pre data-tool="mdnice编辑器"><span></span><code>org2m &lt;- RunUMAP(org2m, reduction = <span>"css"</span>, dims = 1:ncol(Embeddings(org2m, <span>"css"</span>)))<br>org2m &lt;- FindNeighbors(org2m, reduction = <span>"css"</span>, dims = 1:ncol(Embeddings(org2m, <span>"css"</span>)))<br>org2m &lt;- FindClusters(org2m, resolution = 1)<br>DimPlot(org2m, group.by = <span>"batch"</span>) + DimPlot(org2m, group.by = <span>"celltype_RSS"</span>) + DimPlot(org2m)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.25" data-src="https://mmbiz.qpic.cn/mmbiz_png/zN67RpvO3ka3yKZByMPiaCKgwjibb0jezn2gq2yxxMAMWsC5NXJ4enNtLiaXcI8L6919QKBwNLtI0kQMSwVx3zhaw/640?wx_fmt=png" data-type="png" data-w="1200" src="https://mmbiz.qpic.cn/mmbiz_png/zN67RpvO3ka3yKZByMPiaCKgwjibb0jezn2gq2yxxMAMWsC5NXJ4enNtLiaXcI8L6919QKBwNLtI0kQMSwVx3zhaw/640?wx_fmt=png"><figcaption><br></figcaption></figure><p data-tool="mdnice编辑器">可以看到，使用CSS方法进行批次矫正整合后，不同样本之间相同的细胞类型很好的聚集在一起。</p><h3 data-tool="mdnice编辑器"><span></span><span></span><span>比较不同的数据整合方法</span><span></span></h3><p data-tool="mdnice编辑器">接下来，我们将比较CSS方法和常用的一些单细胞数据整合方法：harmony，liger，scanorama，MNN和Seurat CCA等，并使用lisi评估不同数据整合方法的整合效果。</p><pre data-tool="mdnice编辑器"><span></span><code><span># 加载所需的R包</span><br>library(reticulate)                                                       <br>scanorama &lt;- import(<span>"scanorama"</span>)<br>library(Matrix)<br>library(Seurat)<br>library(SeuratWrappers)<br>library(dplyr)<br>library(simspec)<br>library(harmony)<br>library(liger)<br><br>library(RANN)<br>library(lisi)<br>library(doParallel)<br></code></pre><p data-tool="mdnice编辑器">使用不同的数据整合方法进行批次矫正</p><pre data-tool="mdnice编辑器"><span></span><code><span># integration</span><br><span>## RSS</span><br>ref_brainspan &lt;- readRDS(<span>"ext/brainspan_fetal.rds"</span>)<br><span># 使用RSS方法基于参考数据集进行整合</span><br>seurat &lt;- ref_sim_spectrum(seurat, ref_brainspan, reduction.name = <span>"RSS"</span>, reduction.key = <span>"RSS_"</span>)<br>seurat &lt;- RunUMAP(seurat, reduction=<span>"RSS"</span>, reduction.name=<span>"umap_RSS"</span>, reduction.key=<span>"UMAPRSS_"</span>, dims = 1:ncol(Embeddings(seurat, <span>"RSS"</span>)))<br><br><span>## CSS</span><br>seurat &lt;- cluster_sim_spectrum(seurat, label_tag=<span>"organoid"</span>, cluster_resolution = 0.6)<br>seurat &lt;- RunUMAP(seurat, reduction=<span>"css"</span>, dims = 1:ncol(Embeddings(seurat,<span>"css"</span>)), reduction.name=<span>"umap_css"</span>, reduction.key=<span>"UMACSS_"</span>)<br><br><span>## MNN</span><br>seurat_mnn &lt;- RunFastMNN(object.list = SplitObject(seurat, split.by = <span>"organoid"</span>))<br>seurat_mnn &lt;- RunUMAP(seurat_mnn, reduction=<span>"mnn"</span>, dims = 1:30)<br>seurat[[<span>'mnn'</span>]] &lt;- CreateDimReducObject(Embeddings(seurat_mnn,<span>"mnn"</span>)[colnames(seurat),], key=<span>"MNN_"</span>)<br>seurat[[<span>'umap_mnn'</span>]] &lt;- CreateDimReducObject(Embeddings(seurat_mnn,<span>"umap"</span>)[colnames(seurat),], key=<span>"UMAPMNN_"</span>)<br><br><span>## scanorama</span><br>mat &lt;- setNames(lapply(SplitObject(seurat, split.by = <span>"organoid"</span>), <span>function</span>(x) as.matrix(t(x@assays<span>$RNA</span>@data[VariableFeatures(seurat),]))), NULL)<br>gene_list &lt;- lapply(unique(seurat<span>$organoid</span>), <span>function</span>(x) VariableFeatures(seurat))<br>mat_integrated &lt;- scanorama<span>$integrate</span>(mat, gene_list)<br>dr_scanorama &lt;- do.call(rbind, mat_integrated[[1]])<br>rownames(dr_scanorama) &lt;- do.call(c, lapply(mat, rownames))<br>dr_scanorama &lt;- dr_scanorama[colnames(seurat),]<br>seurat[[<span>'scanorama'</span>]] &lt;- CreateDimReducObject(dr_scanorama, key=<span>"SCANORAMA_"</span>)<br>seurat &lt;- RunUMAP(seurat, reduction=<span>"scanorama"</span>, dims = 1:ncol(Embeddings(seurat,<span>"scanorama"</span>)), reduction.name=<span>"umap_scanorama"</span>, reduction.key=<span>"UMAPSCANORAMA_"</span>)<br><br><span>## Harmony</span><br>seurat &lt;- RunHarmony(seurat, <span>"organoid"</span>)<br>seurat &lt;- RunUMAP(seurat, reduction=<span>"harmony"</span>, reduction.name=<span>"umap_harmony"</span>, reduction.key=<span>"UMAPHARMONY_"</span>, dims = 1:50)<br><br><span>## Seurat integration</span><br>seurat_samples &lt;- SplitObject(seurat, split.by=<span>"organoid"</span>)<br>seurat_samples &lt;- lapply(seurat_samples, <span>function</span>(obj) FindVariableFeatures(obj, nfeatures = 5000))<br>seurat_anchors &lt;- FindIntegrationAnchors(object.list = seurat_samples, dims = 1:30, anchor.features = 5000)<br>seurat_integrated &lt;- IntegrateData(anchorset = seurat_anchors, dims = 1:30)<br>seurat_integrated &lt;- ScaleData(seurat_integrated) %&gt;% RunPCA(npcs = 50, verbose = F) %&gt;% RunUMAP(dims = 1:20)<br>seurat[[<span>'umap_Seurat'</span>]] &lt;- CreateDimReducObject(Embeddings(seurat_integrated, <span>"umap_integrated"</span>)[colnames(seurat),], key=<span>"UMAPSEURAT_"</span>)<br>seurat[[<span>'pca_Seurat'</span>]] &lt;- CreateDimReducObject(Embeddings(seurat_integrated, <span>"pca"</span>)[colnames(seurat),], key = <span>"PCASEURAT_"</span>)<br><br><span>## LIGER</span><br>seurat_samples &lt;- SplitObject(seurat, split.by=<span>"organoid"</span>)<br>data_samples &lt;- lapply(seurat_samples, <span>function</span>(x) x@assays<span>$RNA</span>@data)<br>data_samples &lt;- lapply(data_samples, <span>function</span>(x) exp(x)-1)<br>liger_samples &lt;- createLiger(data_samples)<br>liger_samples &lt;- normalize(liger_samples)<br>liger_samples &lt;- selectGenes(liger_samples, var.thresh=0.3, do.plot = T)<br>liger_samples &lt;- scaleNotCenter(liger_samples)<br>liger_samples &lt;- optimizeALS(liger_samples, k=20, thresh = 5e-5, nrep = 3)<br>liger_samples &lt;- runUMAP(liger_samples, use.raw=T)<br>liger_samples &lt;- quantileAlignSNF(liger_samples, resolution = 0.4, small.clust.thresh = 20)<br>liger_samples &lt;- runUMAP(liger_samples)<br>coord &lt;- liger_samples@tsne.coords<br>coord &lt;- coord[colnames(seurat),]<br>seurat[[<span>'umap_LIGER'</span>]] &lt;- CreateDimReducObject(coord, key=<span>"UMAPLIGER_"</span>)<br>seurat[[<span>'LIGER'</span>]] &lt;- CreateDimReducObject(liger_samples@H.norm[colnames(seurat),], key=<span>"LIGER_"</span>)<br></code></pre><p data-tool="mdnice编辑器">细胞类型注释</p><pre data-tool="mdnice编辑器"><span></span><code><span># cell-level annotation</span><br><span>### dorsal-ventral-nontelen</span><br>library(glmnet)<br>library(doMC)<br>registerDoMC(10)<br>seurat &lt;- FindNeighbors(seurat, reduction=<span>"pca"</span>, dims = 1:20) %&gt;%<br>  FindClusters(resolution = 0.6)<br><br>cl_dorsal &lt;- c(2,18,0,12,3,11)<br>cl_ventral &lt;- c(4,16,7,13,8)<br>cl_nontelen &lt;- c(19,10,22,20,15)<br>idx &lt;- c(sample(<span>which</span>(seurat@active.ident %<span>in</span>% cl_dorsal), 1000),<br>         sample(<span>which</span>(seurat@active.ident %<span>in</span>% cl_ventral), 1000),<br>         sample(<span>which</span>(seurat@active.ident %<span>in</span>% cl_nontelen), 1000))<br><br>x &lt;- as.matrix(seurat@assays<span>$RNA</span>@data[VariableFeatures(seurat),idx])<br>y &lt;- factor(setNames(rep(c(<span>"dorsal"</span>,<span>"ventral"</span>,<span>"nontelen"</span>), c(6,5,5)), c(cl_dorsal,cl_ventral,cl_nontelen))[as.character(seurat@active.ident[idx])])<br>m &lt;- cv.glmnet(x = t(x), y = y, family = <span>"multinomial"</span>, parallel = T)<br>pred_fates &lt;- predict(m, as.matrix(t(seurat@assays<span>$RNA</span>@data[VariableFeatures(seurat),])), <span>type</span> = <span>"response"</span>)[,,1]<br>lab_pred_fates &lt;- factor(colnames(pred_fates)[apply(pred_fates, 1, which.max)])<br><br><span>### NPC-neuron</span><br>DE_neuron_NPC &lt;- read.table(<span>"ext/DE_NPC_neurons.tsv"</span>, sep=<span>"\t"</span>, stringsAsFactors=F)<br>high_neuron &lt;- intersect(rownames(seurat), DE_neuron_NPC<span>$feature</span>[<span>which</span>(DE_neuron_NPC<span>$group</span> == <span>"neuron"</span> &amp; DE_neuron_NPC<span>$padj</span> &lt; 0.01 &amp; DE_neuron_NPC<span>$logFC</span> &gt; <span>log</span>(1.2) &amp; DE_neuron_NPC<span>$pct_in</span> &gt; 50 &amp; DE_neuron_NPC<span>$pct_out</span> &lt; 20 &amp; DE_neuron_NPC<span>$auc</span> &gt; 0.6)])<br>high_NPC &lt;- intersect(rownames(seurat), DE_neuron_NPC<span>$feature</span>[<span>which</span>(DE_neuron_NPC<span>$group</span> == <span>"NPC"</span> &amp; DE_neuron_NPC<span>$padj</span> &lt; 0.01 &amp; DE_neuron_NPC<span>$logFC</span> &gt; <span>log</span>(1.2) &amp; DE_neuron_NPC<span>$pct_in</span> &gt; 50 &amp; DE_neuron_NPC<span>$pct_out</span> &lt; 20 &amp; DE_neuron_NPC<span>$auc</span> &gt; 0.6)])<br>seurat<span>$neuron_vs_NPC</span> &lt;- colMeans(seurat@assays<span>$RNA</span>@data[high_neuron,]) - colMeans(seurat@assays<span>$RNA</span>@data[high_NPC,])<br>lab_NPC_neuron &lt;- factor(ifelse(seurat<span>$neuron_vs_NPC</span> &gt; 0, <span>"neuron"</span>, <span>"NPC"</span>))<br><br>lab_annot &lt;- paste0(lab_pred_fates, <span>"_"</span>, lab_NPC_neuron)<br>seurat<span>$semi_branch</span> &lt;- lab_pred_fates<br>seurat<span>$semi_celltype</span> &lt;- lab_annot<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.8473557692307693" data-src="https://mmbiz.qpic.cn/mmbiz_png/zN67RpvO3ka3yKZByMPiaCKgwjibb0jeznVgzPwCm2bSElD1kyru4RytZ2Iib0wqZa747IAf8woBaFicWFich504Y8Q/640?wx_fmt=png" data-type="png" data-w="832" src="https://mmbiz.qpic.cn/mmbiz_png/zN67RpvO3ka3yKZByMPiaCKgwjibb0jeznVgzPwCm2bSElD1kyru4RytZ2Iib0wqZa747IAf8woBaFicWFich504Y8Q/640?wx_fmt=png"><figcaption><br></figcaption></figure><p data-tool="mdnice编辑器">使用lisi评估不同整合方法的整合效果</p><pre data-tool="mdnice编辑器"><span></span><code><span># LISI-based benchmark</span><br>registerDoParallel(8)<br>lisi_batch_org2m &lt;- foreach(dr = c(<span>"pca"</span>,<span>"RSS"</span>,<span>"CSS"</span>,<span>"mnn"</span>,<span>"scanorama"</span>,<span>"harmony"</span>,<span>"pca_Seurat"</span>,<span>"LIGER"</span>), .combine = list, .multicombine = T) %dopar%{<br>  compute_lisi(Embeddings(seurat, dr), seurat@meta.data, <span>"organoid"</span>)<br>}<br>lisi_ct_org2m &lt;- foreach(dr = c(<span>"pca"</span>,<span>"RSS"</span>,<span>"CSS"</span>,<span>"mnn"</span>,<span>"scanorama"</span>,<span>"harmony"</span>,<span>"pca_Seurat"</span>,<span>"LIGER"</span>), .combine = list, .multicombine = T) %dopar%{<br>  compute_lisi(Embeddings(seurat, dr), seurat@meta.data, <span>"celltype_reannot"</span>)<br>}<br>stopImplicitCluster()<br><br>layout(matrix(1:2, nrow=1))<br>boxplot(do.call(cbind, lisi_batch_org2m), frame=F, las=2, names = c(<span>"pca"</span>,<span>"RSS"</span>,<span>"CSS"</span>,<span>"mnn"</span>,<span>"scanorama"</span>,<span>"harmony"</span>,<span>"pca_Seurat"</span>,<span>"LIGER"</span>), outline=F)<br>boxplot(length(unique(seurat<span>$celltype_reannot</span>))+1-do.call(cbind, lisi_ct_org2m), frame=F, las=2, names = c(<span>"pca"</span>,<span>"RSS"</span>,<span>"CSS"</span>,<span>"mnn"</span>,<span>"scanorama"</span>,<span>"harmony"</span>,<span>"pca_Seurat"</span>,<span>"LIGER"</span>), outline=F)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="1.2791762013729977" data-src="https://mmbiz.qpic.cn/mmbiz_png/zN67RpvO3ka3yKZByMPiaCKgwjibb0jeznh0pIyHV3vInYFE1Bjwzt0bAqNweMR66ReicTcnwfMGmBEmhTsibZwvkQ/640?wx_fmt=png" data-type="png" data-w="437" src="https://mmbiz.qpic.cn/mmbiz_png/zN67RpvO3ka3yKZByMPiaCKgwjibb0jeznh0pIyHV3vInYFE1Bjwzt0bAqNweMR66ReicTcnwfMGmBEmhTsibZwvkQ/640?wx_fmt=png"><figcaption><span></span><br></figcaption></figure></section><section data-support="96编辑器" data-style-id="25142"><section><section><section><p><br></p><section data-style-type="5" data-tools="新媒体排版" data-id="2440476"><section><section><section><section><img data-ratio="0.9495798319327731" data-src="https://mmbiz.qpic.cn/mmbiz_gif/09gp6SvPE04j3m2v7Hr889icHUyibTOHs8YuUibicl7ibRD0ZwG5pDTjBluRreZvuib1o3BibvLkicYhnA4YW7dQsjn0cA/640?wx_fmt=gif" data-type="gif" data-w="119" data-width="100%" src="https://mmbiz.qpic.cn/mmbiz_gif/09gp6SvPE04j3m2v7Hr889icHUyibTOHs8YuUibicl7ibRD0ZwG5pDTjBluRreZvuib1o3BibvLkicYhnA4YW7dQsjn0cA/640?wx_fmt=gif"></section><section data-brushtype="text">往期回顾</section><section><br></section></section></section></section><section><section data-autoskip="1"><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247507513&amp;idx=1&amp;sn=151cc23875b4e8be27026311db027822&amp;chksm=ea1cd8bbdd6b51ad1a48e053d48e5209a130c1906c09d755a7fbe2ad3f32867f6b28a8dc23b6&amp;scene=21#wechat_redirect" textvalue="单细胞解析新表位CAR记忆样NK细胞对NPM1突变的急性髓系白血病治疗机制" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2"><span>单细胞解析新表位CAR记忆样NK细胞对NPM1突变的急性髓系白血病治疗机制</span></a><br></p><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247507490&amp;idx=1&amp;sn=f232fc8ed2dcadaa6490de889bcaf5d5&amp;chksm=ea1cd8a0dd6b51b6d35e95aaf981cc0220ea2bcc17542ef7aab59b27394ab062aef04efc33f4&amp;scene=21#wechat_redirect" textvalue="肝细胞癌胚内皮细胞重编程促进免疫抑制性巨噬细胞的产生" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2"><span>肝细胞癌胚内皮细胞重编程促进免疫抑制性巨噬细胞的产生</span></a><br></p><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247507414&amp;idx=1&amp;sn=a2dc5adb69c217b5484f3626355f5cac&amp;chksm=ea1cdb54dd6b52424fc816917edc00a3b35a7eb4edf4c54df4c118884810ed2df94dde40cd9b&amp;scene=21#wechat_redirect" textvalue="scanpy官方教程2022|05-空间转录组数据分析" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2"><span>scanpy官方教程2022|05-空间转录组数据分析</span></a><br></p><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247507327&amp;idx=1&amp;sn=35f652e4f8d55f00c0fe89d16eb7d244&amp;chksm=ea1cdbfddd6b52ebff33da7e2eacd2003d32c11fd94ff6b92348d8e16e89532ed716234d7aaf&amp;scene=21#wechat_redirect" textvalue="对t-SNE的一些不成熟思考" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2"><span>对t-SNE的一些不成熟思考</span></a><br></p><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247506971&amp;idx=1&amp;sn=f0242285e2c827d922f938d9858d4ffe&amp;chksm=ea1cda99dd6b538fd0ed7f870f2a5a1dd55724dce280bdce024be41caaa60f668b550765b6bb&amp;scene=21#wechat_redirect" textvalue="髓系免疫细胞细分亚群" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2"><span>髓系免疫细胞细分亚群</span></a><br></p></section></section><hr><p><br></p></section><section data-style-type="5" data-tools="新媒体排版" data-id="2440475"><hr><p><br></p><hr><section><p>如果你对单细胞转录组研究感兴趣，但又不知道如何入门，也许你可以关注一下下面的课程<span></span></p><p><br></p><ul><li><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247505898&amp;idx=1&amp;sn=3caaa355db327f412fe27c969f20533c&amp;chksm=ea1cd168dd6b587e7c955c80a87b78248f4fe7cf88514caf7fdca580e98877d4e2d96e40dcb7&amp;scene=21#wechat_redirect" textvalue="提前锁定年后马拉松授课名额，赠送价值1600的单细胞数据分析一次" linktype="text" imgurl="" imgdata="null" data-itemshowtype="11" tab="innerlink" data-linktype="2" hasload="1">提前锁定年后马拉松授课名额，赠送价值1600的单细胞数据分析一次</a><br></p></li><li><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247501306&amp;idx=1&amp;sn=a38c5d2c3665ec0949649f22a8c7cd72&amp;chksm=ea1cc378dd6b4a6e1cdb72babea1fb4b232810c7656ac81963175e735a8d10a7cfae73921a16&amp;scene=21#wechat_redirect" textvalue="一个10x单细胞样品费用拿下你的专属64线程200G内存服务器" linktype="text" imgurl="" imgdata="null" data-itemshowtype="11" tab="innerlink" data-linktype="2" wah-hotarea="click" hasload="1">一个10x单细胞样品费用拿下你的专属64线程200G内存服务器</a></p></li><li><p><a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzI1Njk4ODE0MQ==&amp;mid=2247501306&amp;idx=3&amp;sn=64ad5b7771462b946c74733d6e184f5e&amp;chksm=ea1cc378dd6b4a6ef6af8f0751e29c7180a7c69a96c54a94cb6163bc1d4b11daee2893102d53&amp;scene=21#wechat_redirect" textvalue="96线程384G内存服务器共享一年仅700（每月一批，循环开通账号）" linktype="text" imgurl="" imgdata="null" data-itemshowtype="11" tab="innerlink" data-linktype="2" wah-hotarea="click" hasload="1">96线程384G内存服务器共享一年仅700（每月一批，循环开通账号）</a></p></li></ul><p><img data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz_gif/4TKeL1ZejtlKxOib5kmKX6ic6eX0w0WK5jvhtz9yBRsO3OI4yr6S5iaLNM7AbAeuPDHXMvDdur2DRz9wyiax4lEviag/640?wx_fmt=gif" data-type="gif" data-w="240" src="https://mmbiz.qpic.cn/mmbiz_gif/4TKeL1ZejtlKxOib5kmKX6ic6eX0w0WK5jvhtz9yBRsO3OI4yr6S5iaLNM7AbAeuPDHXMvDdur2DRz9wyiax4lEviag/640?wx_fmt=gif"><br></p><p><img data-ratio="0.05278592375366569" data-src="https://mmbiz.qpic.cn/mmbiz/4TKeL1Zejtlq03ZOSZiaTlic1MxgdKiaxTbOZ7ZSe0Xx1Ca8xF3L6Nyj1FYUajtYrSmRIHyZVSsAve0EAvEicZONpg/640?wx_fmt=jpeg" data-type="other" data-w="341" src="https://mmbiz.qpic.cn/mmbiz/4TKeL1Zejtlq03ZOSZiaTlic1MxgdKiaxTbOZ7ZSe0Xx1Ca8xF3L6Nyj1FYUajtYrSmRIHyZVSsAve0EAvEicZONpg/640?wx_fmt=jpeg"></p><p><strong><span>看完记得顺手点个</span></strong><span><strong><span>“在看”</span></strong></span><strong><span>哦！</span></strong></p></section><section><section data-id="93668"><section><section data-width="95%"><section><section><section data-width="38%"><section><section data-tools="135编辑器" data-id="93668"><section><section data-width="95%"><section><section><section data-width="61.8%"><section><section><section><p><br></p><span><strong data-burshtype="text"><img data-copyright="0" data-cropselx1="0" data-cropselx2="109" data-cropsely1="0" data-cropsely2="109" data-ratio="1" data-src="https://mmbiz.qpic.cn/mmbiz/siaia0BDGJdjRMGrkqo64BGKecYk4akuHpGHVQs7FeOpY7eWbIPGC1tRw5Tw0oEPmx053mR9FTVerWvhuZchIpZw/640?wx_fmt=jpeg" data-type="other" data-w="430" title="qrcode_for_gh_5a3c482ed99f_1280.jpg" src="https://mmbiz.qpic.cn/mmbiz/siaia0BDGJdjRMGrkqo64BGKecYk4akuHpGHVQs7FeOpY7eWbIPGC1tRw5Tw0oEPmx053mR9FTVerWvhuZchIpZw/640?wx_fmt=jpeg"><strong data-burshtype="text">生物</strong><strong data-burshtype="text"> | 单细胞 | 转录组丨资料</strong></strong></span></section><section><span><strong data-burshtype="text">每天都精彩</strong></span></section></section></section><section><section><section><section><section><section><p><span>长按扫码可关注</span></p></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></section></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/ThmItwx5kpjYCfBbHWi1DQ",target="_blank" rel="noopener noreferrer">原文链接</a>
