---
title: "单细胞转录组分析之CNV推断"
date: 2023-01-04T01:27:04Z
draft: ["false"]
tags: [
  "fetched",
  "医学僧的科研日记"
]
categories: ["Acdemic"]
---
单细胞转录组分析之CNV推断 by 医学僧的科研日记
------
<div><section><span>单细胞转录组公共数据库已经很多了，但单细胞多组学的数据大多数还没公开，如何利用已有工具让我们的研究更出彩成为了大家亟需解决的问题。在这里，继上篇单细胞转录组分析之细胞通讯，小王子给大家带来新的模块-CNV推断。</span><span></span></section><section><iframe data-vidtype="2" data-mpvid="wxv_2521382666508746753" data-cover="http%3A%2F%2Fmmbiz.qpic.cn%2Fmmbiz_jpg%2FeBfvI7Ryx8dGcCicV0GoZLCE7VcyibIRZh1X1gy3heS5iarNJV7bpAZAXZqcKtG46ZzTVUkrLOlWU24lrkKRa1sgQ%2F0%3Fwx_fmt%3Djpeg" allowfullscreen="" frameborder="0" data-ratio="1.7777777777777777" data-w="1920" data-src="https://mp.weixin.qq.com/mp/readtemplate?t=pages/video_player_tmpl&amp;action=mpvideo&amp;auto=0&amp;vid=wxv_2521382666508746753"></iframe></section><section><section powered-by="xiumi.us"><section><section powered-by="xiumi.us"><section><section powered-by="xiumi.us"><section><section><section powered-by="xiumi.us"><section><section powered-by="xiumi.us"><section><section><svg viewbox="0 0 1 1"></svg></section></section></section></section><section><section powered-by="xiumi.us"><section><section><svg viewbox="0 0 1 1"></svg></section></section></section></section><section><section powered-by="xiumi.us"><section><section><svg viewbox="0 0 1 1"></svg></section></section></section></section><section><section powered-by="xiumi.us"><section><section><svg viewbox="0 0 1 1"></svg></section></section></section></section></section><section powered-by="xiumi.us"><section><section powered-by="xiumi.us"><section><section><svg viewbox="0 0 1 1"></svg></section></section></section></section><section><section powered-by="xiumi.us"><section><section><svg viewbox="0 0 1 1"></svg></section></section></section></section></section></section></section></section></section><section><section powered-by="xiumi.us"><section><p><strong>1.什么是CNV？</strong></p></section></section></section></section><section powered-by="xiumi.us"><section><svg viewbox="0 0 1 1"></svg></section></section></section></section><section powered-by="xiumi.us"><section>拷贝数变异（Copy number variation, CNV）：是由基因组发生重排而导致的,一般指长度1KB以上的基因组大片段的拷贝数增加或者减少, 主要表现为亚显微水平的缺失和重复。</section><section>CNV的发生机制就是非等位基因重组，第一次是在减数第一次分裂前期，一对同源染色体染色体上的非姐妹染色单体交叉互换，第二次是在减数第一次分裂后期，同源色体分离，非同源染色体自由组合。基因组上非等位的两个高度同源的DNA序列在减数分裂或者有丝分裂的过程中发生错误的配对，并发生序列交换，从而导致缺失、重复的出现。</section><p><span><em>PS：对背景知识不理解的小伙伴建议去知乎搜索专栏：焦老师讲遗传系列之12：拷贝数变异CNV</em></span></p></section><section powered-by="xiumi.us"><section><section powered-by="xiumi.us"><section><section powered-by="xiumi.us"><section><section><section powered-by="xiumi.us"><section><section powered-by="xiumi.us"><section><section><svg viewbox="0 0 1 1"></svg></section></section></section></section><section><section powered-by="xiumi.us"><section><section><svg viewbox="0 0 1 1"></svg></section></section></section></section><section><section powered-by="xiumi.us"><section><section><svg viewbox="0 0 1 1"></svg></section></section></section></section><section><section powered-by="xiumi.us"><section><section><svg viewbox="0 0 1 1"></svg></section></section></section></section></section><section powered-by="xiumi.us"><section><section powered-by="xiumi.us"><section><section><svg viewbox="0 0 1 1"></svg></section></section></section></section><section><section powered-by="xiumi.us"><section><section><svg viewbox="0 0 1 1"></svg></section></section></section></section></section></section></section></section></section><section><section powered-by="xiumi.us"><section><p><strong>2.文献中怎么用CNV推断？</strong></p></section></section></section></section><section powered-by="xiumi.us"><section><svg viewbox="0 0 1 1"></svg></section></section></section></section><p powered-by="xiumi.us"><strong>一篇21年的揭示转移性胰腺导管腺癌瘤内异质性的文章：</strong></p><section powered-by="xiumi.us"><section><img data-backh="173" data-backw="578" data-ratio="0.2990741" data-src="https://mmbiz.qpic.cn/mmbiz_png/eBfvI7Ryx8dGcCicV0GoZLCE7VcyibIRZhELhpz8gbyaHuFX8ZCU1BukW714EUXj8Ov6ZeJPKH7ufrg05tl6XHuw/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/eBfvI7Ryx8dGcCicV0GoZLCE7VcyibIRZhELhpz8gbyaHuFX8ZCU1BukW714EUXj8Ov6ZeJPKH7ufrg05tl6XHuw/640?wx_fmt=png"></section></section><section powered-by="xiumi.us"><section><img data-backh="253" data-backw="578" data-ratio="0.437963" data-src="https://mmbiz.qpic.cn/mmbiz_png/eBfvI7Ryx8dGcCicV0GoZLCE7VcyibIRZhnjRWYEcX8AcSRoYmqzPIBm2QogcGB3CtibZZZdRWbF7QZbnda1bqkoA/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/eBfvI7Ryx8dGcCicV0GoZLCE7VcyibIRZhnjRWYEcX8AcSRoYmqzPIBm2QogcGB3CtibZZZdRWbF7QZbnda1bqkoA/640?wx_fmt=png"></section></section><section><span>在进行细胞注释后，作者用inferCNV在所测的5个样本中依次进行了不同胰腺导管细胞亚群的CNV推断，发现不同样本具有显著的异质性。</span></section><section powered-by="xiumi.us"><section><section powered-by="xiumi.us"><section><section powered-by="xiumi.us"><section><section><section powered-by="xiumi.us"><section><section powered-by="xiumi.us"><section><section><svg viewbox="0 0 1 1"></svg></section></section></section></section><section><section powered-by="xiumi.us"><section><section><svg viewbox="0 0 1 1"></svg></section></section></section></section><section><section powered-by="xiumi.us"><section><section><svg viewbox="0 0 1 1"></svg></section></section></section></section><section><section powered-by="xiumi.us"><section><section><svg viewbox="0 0 1 1"></svg></section></section></section></section></section><section powered-by="xiumi.us"><section><section powered-by="xiumi.us"><section><section><svg viewbox="0 0 1 1"></svg></section></section></section></section><section><section powered-by="xiumi.us"><section><section><svg viewbox="0 0 1 1"></svg></section></section></section></section></section></section></section></section></section><section><section powered-by="xiumi.us"><section><p><strong>3.该如何进行CNV推断？</strong></p></section></section></section></section></section></section><p>由于篇幅有限，这里着重介绍下常用的inferCNV和copyKAT。</p><section powered-by="xiumi.us"><section><section powered-by="xiumi.us"><section><section powered-by="xiumi.us"><section><section powered-by="xiumi.us"><section><section><section powered-by="xiumi.us"><section><section><section powered-by="xiumi.us"><section><svg viewbox="0 0 1 1"></svg></section></section><section powered-by="xiumi.us"><section><svg viewbox="0 0 1 1"></svg></section></section></section></section></section></section></section></section></section></section></section><section><section powered-by="xiumi.us"><section><p><strong>1.inferCNV</strong></p></section></section></section></section></section></section><section>inferCNV前后一共有3篇CNS文章进行开发和改进，在这里，小王子分别给大家简述其演进过程以方便大家理解inferCNV进行CNV推断的原理：</section><p powered-by="xiumi.us"><strong>(1)提出：</strong>Single-cell RNA-seq highlights intratumoral heterogeneity in primary glioblastoma. Science. 2014 Jun. PMID: 24925914a</p><section powered-by="xiumi.us"><section><img data-backh="190" data-backw="578" data-ratio="0.3287037" data-src="https://mmbiz.qpic.cn/mmbiz_png/eBfvI7Ryx8dGcCicV0GoZLCE7VcyibIRZhwguwibF2e9gyiaQEOsFZv8X598ibBqGzL33KnnPEBaOmXNmHlJrltvO6Q/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/eBfvI7Ryx8dGcCicV0GoZLCE7VcyibIRZhwguwibF2e9gyiaQEOsFZv8X598ibBqGzL33KnnPEBaOmXNmHlJrltvO6Q/640?wx_fmt=png"></section></section><section>第一版整体来讲的原理就是：使用101个基因的平均表达值表示中间基因的染色体CNV值，流程如下：</section><section powered-by="xiumi.us"><section><img data-backh="525" data-backw="578" data-ratio="0.9083799" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/eBfvI7Ryx8dGcCicV0GoZLCE7VcyibIRZhexicKf607GEPd9rlrekhyiaiahwYG76aFO3rgkPviaSAJ4O7HUkb82YjKA/640?wx_fmt=jpeg" data-type="jpeg" data-w="895" src="https://mmbiz.qpic.cn/mmbiz_jpg/eBfvI7Ryx8dGcCicV0GoZLCE7VcyibIRZhexicKf607GEPd9rlrekhyiaiahwYG76aFO3rgkPviaSAJ4O7HUkb82YjKA/640?wx_fmt=jpeg"></section></section><section powered-by="xiumi.us"><section><strong>图1：</strong>原始数据标准化，去除测序深度不同造成的差异；</section><section><strong>图2：</strong>根据公式把基因表达值转换为CNV值；</section><section><strong>图3：</strong>对每个细胞的CNV值进行中心化处理，使细胞之间相同染色体区域的CNV值具有可比性；</section><section><strong>图4：</strong>用肿瘤细胞的CNV值减去对应区域正常细胞的CNV；</section><section><strong>蓝框左图：</strong>如果infercnv::run函数中的参数denoise=TRUE，则使用算法进一步去除背景噪音凸显CNV区域；</section><section><strong>蓝框右图：</strong>如果infercnv::run函数中的参数HMM=TRUE，则使用隐马尔可夫模型（Hidden Markov Model, HMM）预测CNV区域，并用贝叶斯潜在混合模型（Bayesian Network Latent Mixture Model）对结果进行校正。</section></section><p powered-by="xiumi.us"><strong>(2)改进：</strong>Dissectingthe multicellular ecosystem of metastatic melanoma by single-cell RNA-seq.Science. 2016 Apr 8.PMID: 27124452</p><section powered-by="xiumi.us"><section><img data-backh="337" data-backw="578" data-ratio="0.5833333" data-src="https://mmbiz.qpic.cn/mmbiz_png/eBfvI7Ryx8dGcCicV0GoZLCE7VcyibIRZh0y0OfYsxd2FjtO67DeLH5Fo255ZQ4YSZXPa18uIsz97QshuE2jyKeg/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/eBfvI7Ryx8dGcCicV0GoZLCE7VcyibIRZh0y0OfYsxd2FjtO67DeLH5Fo255ZQ4YSZXPa18uIsz97QshuE2jyKeg/640?wx_fmt=png"></section></section><section powered-by="xiumi.us"><section>第二版与第一版相比，有了以下两处改进:A. <span>为避免极端值（-3，-3）；B.</span><span>5种CNV稳定的正常细胞作为基线值进行CNV矫正，用整体CNV和与同一肿瘤top10%细胞的平均CNV的相关性估计细胞的良恶性。</span></section><section powered-by="xiumi.us"><section><span></span></section></section></section><section powered-by="xiumi.us"><section><img data-backh="208" data-backw="578" data-ratio="0.3601852" data-src="https://mmbiz.qpic.cn/mmbiz_png/eBfvI7Ryx8dGcCicV0GoZLCE7VcyibIRZh7d3Q8Lma5TuwfxMZ9wMsvDtUHDwffBQxUHGIHuhhNy2YHPibtg76yqA/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/eBfvI7Ryx8dGcCicV0GoZLCE7VcyibIRZh7d3Q8Lma5TuwfxMZ9wMsvDtUHDwffBQxUHGIHuhhNy2YHPibtg76yqA/640?wx_fmt=png"></section></section><section>到第二版，inferCNV已经能够判断良恶性细胞了</section><p powered-by="xiumi.us"><strong>(3)细化+升华：</strong>Single-CellTranscriptomic Analysis of Primary and Metastatic Head and Neck Cancer. Cell.2017 Dec 14.PMID: 29198524</p><section powered-by="xiumi.us"><section><img data-backh="236" data-backw="578" data-ratio="0.4083333" data-src="https://mmbiz.qpic.cn/mmbiz_png/eBfvI7Ryx8dGcCicV0GoZLCE7VcyibIRZhJPUhH45cFe4vekib9tQK9Z2AzIZdUXgwTOWric445redRNkcibUAt3GDA/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/eBfvI7Ryx8dGcCicV0GoZLCE7VcyibIRZhJPUhH45cFe4vekib9tQK9Z2AzIZdUXgwTOWric445redRNkcibUAt3GDA/640?wx_fmt=png"></section></section><section powered-by="xiumi.us"><section>与第二版使用同一肿瘤top10%细胞的平均CNV的相关性相比，第三版有了如下改进：</section><section>A.用整体CNV和与同一肿瘤所有细胞的平均CNV的相关性估计细胞的良恶性。<br></section><section>B.给定了以下判断良恶性的相关性参考阈值：</section><section>恶性细胞：整体CNV&gt;0.05&amp;CNV相关性&gt;0.5</section><section>非恶性细胞：整体CNV&lt;0.05&amp;CNV相关性&lt;0.5</section><section>未定义细胞：整体CNV&gt;0.05&amp;CNV相关性&lt;0.5或者整体CNV&gt;0.05&amp;CNV相关性&lt;0.5</section><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="sql"><code><span><span>#代码实战：比较简单，只需准备好count矩阵表达文件、</span></span></code><code><span><span>#单个细胞类群的注释文件和基因在染色体上的位置文件即可，</span></span></code><code><span><span>#运行速度看电脑配置和细胞数量，我示例是3317个细胞跑了20分钟。</span></span></code><code><span>rm(list = ls())</span></code><code><span><span>#安装infercnv包</span></span></code><code><span>BiocManager::<span>install</span>(<span>"infercnv"</span>)</span></code><code><span><span>#使用infercnv自带数据-----------------------------------------------------------------------</span></span></code><code><span><span>library</span>(infercnv)</span></code><code><span><span>library</span>(tidyverse)</span></code><code><span>dir.create(<span>"inferCNV"</span>)</span></code><code><span>dir.create(<span>"inferCNV/test"</span>)</span></code><code><span><span>##读取示例数据目录</span></span></code><code><span>exprMatrix = system.file(<span>"extdata"</span>, <span>"oligodendroglioma_expression_downsampled.counts.matrix.gz"</span>, <span>package</span>=<span>"infercnv"</span>)</span></code><code><span>cellAnnota = system.file(<span>"extdata"</span>, <span>"oligodendroglioma_annotations_downsampled.txt"</span>, <span>package</span>=<span>"infercnv"</span>)</span></code><code><span>geneLocate = system.file(<span>"extdata"</span>, <span>"gencode_downsampled.EXAMPLE_ONLY_DONT_REUSE.txt"</span>, <span>package</span>=<span>"infercnv"</span>)</span></code><code><span><span>#创建inferCNV对象，直接给相应的文件路径即可</span></span></code><code><span>infercnv_obj = CreateInfercnvObject(delim = <span>'\t'</span>,</span></code><code><span>                                    raw_counts_matrix = exprMatrix,</span></code><code><span>                                    annotations_file = cellAnnota,</span></code><code><span>                                    gene_order_file = geneLocate,</span></code><code><span>                                    ref_group_names = c(<span>"Microglia/Macrophage"</span>,<span>"Oligodendrocytes (non-malignant)"</span>))</span></code><code><span><br></span></code><code><span><span>##分析细胞CNV</span></span></code><code><span><span>#cutoff阈值，官网建议Smart-seq2数据选“1”, 10x Genomics数据选“0.1”</span></span></code><code><span>infercnv_obj = infercnv::run(infercnv_obj,</span></code><code><span>                             cutoff=<span>1</span>, </span></code><code><span>                             out_dir=<span>'inferCNV/test'</span>, </span></code><code><span>                             cluster_by_groups=<span>TRUE</span>,</span></code><code><span>                             <span>#analysis_mode="subclusters", #默认是"samples"</span></span></code><code><span>                             denoise=<span>TRUE</span>,</span></code><code><span>                             HMM=<span>TRUE</span>)</span></code><code><span><span>library</span>(RColorBrewer)</span></code><code><span><span>library</span>(ggsci)</span></code><code><span>infercnv::plot_cnv(infercnv_obj,</span></code><code><span>                   plot_chr_scale = T,</span></code><code><span>                   output_filename = <span>"inferCNV/test/better_plot"</span>,output_format = <span>"pdf"</span>,</span></code><code><span>                   custom_color_pal =  color.palette(c(pal_nejm()(<span>8</span>)[<span>1</span>],<span>"white"</span>,pal_nejm()(<span>8</span>)[<span>2</span>]), c(<span>2</span>,<span>2</span>)))</span></code><code><span><span>#使用TISCH中的Seurat(LIHC的GSE125449)数据-----------------------------------------------------------------------</span></span></code><code><span>rm(<span>list</span> = ls())</span></code><code><span><span>library</span>(infercnv)</span></code><code><span><span>library</span>(tidyverse)</span></code><code><span><span>library</span>(Seurat)</span></code><code><span><span>library</span>(tidyverse)</span></code><code><span>dir.create(<span>"inferCNV/valudation"</span>)</span></code><code><span><span>#获得Count矩阵</span></span></code><code><span><span>data</span> &lt;- Read10X_h5(<span>'LIHC_GSE125449_aPDL1aCTLA4_expression.h5'</span>)</span></code><code><span><span>data</span> &lt;- CreateSeuratObject(<span>data</span>, <span>project</span> = <span>""</span>, min.cells = <span>3</span>, min.features = <span>200</span>)</span></code><code><span><span>data</span> <span>#3835个细胞，17356个基因</span></span></code><code><span>meta &lt;- read.delim(<span>"LIHC_GSE125449_aPDL1aCTLA4_CellMetainfo_table.tsv"</span>, header=F)</span></code><code><span>colnames(meta) &lt;- meta[<span>1</span>,]</span></code><code><span>meta &lt;- meta[<span>-1</span>,]</span></code><code><span>rownames(<span>data</span>)</span></code><code><span>colnames(meta)</span></code><code><span>colnames(meta) &lt;- c(<span>"Cell"</span>,<span>"UMAP_1"</span>,<span>"UMAP_2"</span>,<span>"Celltype_malignancy"</span>,  </span></code><code><span>                    <span>"Celltype_major_lineage"</span>,<span>"Celltype_minor_lineage"</span>,</span></code><code><span>                    <span>"Cluster"</span>,<span>"Celltype"</span>,<span>"Treatment"</span>,<span>"Patient"</span>,<span>"Sample"</span>,</span></code><code><span>                    <span>"Source"</span>,<span>"Age"</span>,<span>"Gender"</span>,<span>"Stage"</span>)</span></code><code><span><span>data</span>@assays$RNA@meta.features &lt;- as.data.frame(rownames(<span>data</span>))</span></code><code><span>identical(meta$Cell,colnames(<span>data</span>))</span></code><code><span>meta &lt;- meta[meta$Cell%<span>in</span>%colnames(<span>data</span>),]</span></code><code><span>a &lt;- <span>data</span>@meta.data</span></code><code><span>a &lt;- cbind(a,meta)</span></code><code><span><span>data</span>@meta.data &lt;- a</span></code><code><span><span>table</span>(<span>data</span>@meta.data$Celltype_malignancy)</span></code><code><span><span>#数据质控+去除批次效应 ---------------------------------------------------------</span></span></code><code><span><span>data</span>[[<span>"percent.mt"</span>]] &lt;- PercentageFeatureSet(<span>data</span>, pattern = <span>"^MT-"</span>) <span># 注意基因名称变化</span></span></code><code><span>VlnPlot(<span>data</span>, group.by = <span>"Patient"</span>, features = c(<span>"nFeature_RNA"</span>, <span>"nCount_RNA"</span>, <span>"percent.mt"</span>), pt.size = <span>0</span>, ncol = <span>3</span>) + </span></code><code><span>  NoLegend()</span></code><code><span><span>##筛选合格细胞</span></span></code><code><span>cat(<span>"Before filter:"</span>,nrow(<span>data</span>@meta.data),<span>"cells\n"</span>)</span></code><code><span><span>data</span> &lt;- subset(<span>data</span>, subset = </span></code><code><span>                 nFeature_RNA &gt; <span>200</span> &amp; </span></code><code><span>                 nFeature_RNA &lt; <span>3000</span> &amp; </span></code><code><span>                 nCount_RNA &gt; <span>500</span> &amp; </span></code><code><span>                 nCount_RNA &lt; <span>4000</span> &amp;</span></code><code><span>                 percent.mt &lt; <span>5</span>)</span></code><code><span>cat(<span>"After filter :"</span>,nrow(<span>data</span>@meta.data),<span>"cells\n"</span>)</span></code><code><span><span>data</span> <span>#保留17356个基因，3317个细胞。</span></span></code><code><span><span>save</span>(<span>data</span>,<span>file</span> = <span>"GSE125449_count.rda"</span>)</span></code><code><span><br></span></code><code><span>dd &lt;- <span>data</span>[[<span>"RNA"</span>]]@counts</span></code><code><span>dd2 &lt;- <span>data</span>@meta.data  </span></code><code><span>dd &lt;- as.data.frame(dd)</span></code><code><span>dd2 &lt;- dd2[,c(<span>4</span>,<span>7</span>)]</span></code><code><span><span>table</span>(dd2$Celltype_malignancy)</span></code><code><span>dd2$Celltype_malignancy &lt;- factor(dd2$Celltype_malignancy,<span>levels</span> = c(<span>"Immune cells"</span>,<span>"Stromal cells"</span>,<span>"Malignant cells"</span>))</span></code><code><span><br></span></code><code><span><span>load</span>(<span>'GRCh38_v39.gtf.rda'</span>) <span>#参考基因组，可自行下载</span></span></code><code><span>dd3 &lt;- gtf_data[,c(<span>12</span>,<span>1</span>:<span>3</span>)]</span></code><code><span>dd3 &lt;- <span>distinct</span>(dd3,gene_name,.keep_all = T)</span></code><code><span><br></span></code><code><span>gene &lt;- <span>intersect</span>(rownames(dd),dd3$gene_name)</span></code><code><span>dd &lt;- dd[rownames(dd)%<span>in</span>%gene,]</span></code><code><span>dd3 &lt;- dd3[dd3$gene_name%<span>in</span>%gene,]</span></code><code><span><br></span></code><code><span>write.table(as.matrix.data.frame(dd), <span>file</span> =<span>"inferCNV/valudation/counts.matrix"</span>,quote = F,sep = <span>"\t"</span>,row.names = T,col.names = T)</span></code><code><span>write.table(as.data.frame(dd3),<span>file</span> = <span>"inferCNV/valudation/gencode.txt"</span>, quote = F,sep = <span>"\t"</span>,row.names = F,col.names = F)</span></code><code><span>write.table(as.data.frame(dd2),<span>file</span> = <span>"inferCNV/valudation/annotation.txt"</span>, quote = F,sep = <span>"\t"</span>,row.names = F,col.names = F)</span></code><code><span><span>#运行infercnv-----------------------------------------------------------------------</span></span></code><code><span>rm(<span>list</span> = ls())</span></code><code><span><span>library</span>(tidyverse)</span></code><code><span><span>library</span>(infercnv)</span></code><code><span>memory.limit(<span>10000000</span>)</span></code><code><span><span>#使用inferCNV之前，最好过滤掉低质量的细胞</span></span></code><code><span>infercnv_obj = CreateInfercnvObject(raw_counts_matrix=<span>"inferCNV/valudation/counts.matrix"</span>,</span></code><code><span>                                    annotations_file=<span>"inferCNV/valudation/annotation.txt"</span>,</span></code><code><span>                                    delim=<span>"\t"</span>,</span></code><code><span>                                    gene_order_file=<span>"inferCNV/valudation/gencode.txt"</span>,</span></code><code><span>                                    ref_group_names=c(<span>"Immune cells"</span>) </span></code><code><span>)</span></code><code><span><span>#ref_group_names参数根据细胞注释文件填写，在本数据，我们假设用“Immune cells”作为参照，看“Stromal cells”和“Malignant cells”的恶性；</span></span></code><code><span><span>#ref_group_names=NULL，则会选用所有细胞的平均表达作为参照，这种模式下，最好能确保细胞内在有足够的差异</span></span></code><code><span>infercnv_obj = infercnv::run(infercnv_obj,</span></code><code><span>                             cutoff=<span>1</span>, </span></code><code><span>                             out_dir=<span>"try2"</span>,</span></code><code><span>                             cluster_by_groups=<span>TRUE</span>, </span></code><code><span>                             <span>#analysis_mode="subclusters", #默认是"samples"</span></span></code><code><span>                             denoise=<span>TRUE</span>,</span></code><code><span>                             HMM=<span>TRUE</span>,</span></code><code><span>                             num_threads=<span>4</span></span></code><code><span>)</span></code><code><span><span>#去噪，HMM预测CNV这两项我一般都选上</span></span></code></pre></section></section><section powered-by="xiumi.us"><section><section powered-by="xiumi.us"><section><section powered-by="xiumi.us"><section><section powered-by="xiumi.us"><section><section><section powered-by="xiumi.us"><section><section><section powered-by="xiumi.us"><section><svg viewbox="0 0 1 1"></svg></section></section><section powered-by="xiumi.us"><section><svg viewbox="0 0 1 1"></svg></section></section></section></section></section></section></section></section></section></section></section><section><section powered-by="xiumi.us"><section><p><strong>2.copyKAT</strong></p></section></section></section></section></section></section><section>与inferCNV相比，copyKAT是专门为单细胞CNV开发的，不需要给定正常参考细胞，总体效果好于inferCNV。</section><section powered-by="xiumi.us"><section><img data-ratio="0.6340136" data-src="https://mmbiz.qpic.cn/mmbiz_png/eBfvI7Ryx8dGcCicV0GoZLCE7VcyibIRZhr9nutlbVsHVo96uqfkkecoYot2zGLpwrvLTMoLibymCU9fwTF4DEPAA/640?wx_fmt=png" data-type="png" data-w="735" src="https://mmbiz.qpic.cn/mmbiz_png/eBfvI7Ryx8dGcCicV0GoZLCE7VcyibIRZhr9nutlbVsHVo96uqfkkecoYot2zGLpwrvLTMoLibymCU9fwTF4DEPAA/640?wx_fmt=png"></section></section><section powered-by="xiumi.us"><section>copyKAT主要原理：贝叶斯方法+层次聚类</section><section>a:首先输入ScRNA-seq的UMI表达矩阵，通过基因组坐标对它们进行排序，之后用Freeman-Tukey变换和多项式动态线性建模矫正矩阵中的异常值；</section><section>b:之后，将所有单细胞集中到几个小的亚群分类中，并使用高斯混合模型估算每个分类的方差，具有最小估计方差的聚类被定义为“标准的二倍体细胞”。</section><section>c:为了检测染色体断点，他们整合泊松-伽玛模型和马尔可夫链蒙特卡罗迭代生成每个基因窗口的后验均值，然后应用Kolmogorov-Smirnov检验对均值无显著差异的相邻窗口进行合并，然后计算每个窗口的最终拷贝数值，以此作为跨越每个细胞中相邻染色体断点的的所有基因的后验平均值。</section><section><ul><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="sql"><code><span><span>#copyKAT-----------------------------------------------------------------------</span></span></code><code><span>rm(list = ls())</span></code><code><span><br></span></code><code><span><span>#install_github("navinlabcode/copykat")</span></span></code><code><span>library(copykat)</span></code><code><span><span>load</span>(<span>"GSE125449_count.rda"</span>)</span></code><code><span>counts &lt;- as.matrix(<span>data</span>@assays$RNA@counts)</span></code><code><span>cnv &lt;- copykat(rawmat = counts,ngene.chr = <span>5</span>,</span></code><code><span>               sam.name = <span>"GSE125449_copykat"</span>,n.cores = <span>1</span>)</span></code><code><span><span>save</span>(cnv,<span>file</span>=<span>"GSE125449_cnv_copyKAT.rda"</span></span></code></pre></section><section><strong>示例结果展示：</strong>可以直接识别出输入的细胞是非整倍体（恶性）、整倍体（良性）或者未识别</section><p><img data-backh="378" data-backw="578" data-ratio="0.6539440203562341" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/eBfvI7Ryx8dGcCicV0GoZLCE7VcyibIRZh73KnVHvcoffS5tWOzWl8QpGbpeUkgOXs4p3Il7erwN9INffE93v5Yg/640?wx_fmt=png" data-type="png" data-w="786" src="https://mmbiz.qpic.cn/mmbiz_png/eBfvI7Ryx8dGcCicV0GoZLCE7VcyibIRZh73KnVHvcoffS5tWOzWl8QpGbpeUkgOXs4p3Il7erwN9INffE93v5Yg/640?wx_fmt=png"></p></section><section><span><em>需要指出的是：单细胞转录组推断CNV用的是Count数据。不过很多数据库中的数据都已经经过log化，有文献提示利用log后的数据进行inferCNV和copyKAT分析对结果并没影响，不过还是建议用Count数据，因为它会自动进行log标准化。</em></span></section><section powered-by="xiumi.us"><section><section powered-by="xiumi.us"><section><section powered-by="xiumi.us"><section><section><section powered-by="xiumi.us"><section><section powered-by="xiumi.us"><section><section><svg viewbox="0 0 1 1"></svg></section></section></section></section><section><section powered-by="xiumi.us"><section><section><svg viewbox="0 0 1 1"></svg></section></section></section></section><section><section powered-by="xiumi.us"><section><section><svg viewbox="0 0 1 1"></svg></section></section></section></section><section><section powered-by="xiumi.us"><section><section><svg viewbox="0 0 1 1"></svg></section></section></section></section></section><section powered-by="xiumi.us"><section><section powered-by="xiumi.us"><section><section><svg viewbox="0 0 1 1"></svg></section></section></section></section><section><section powered-by="xiumi.us"><section><section><svg viewbox="0 0 1 1"></svg></section></section></section></section></section></section></section></section></section></section><section powered-by="xiumi.us"><section><svg viewbox="0 0 1 1"></svg></section></section></section><section><section powered-by="xiumi.us"><section><p><strong>小结</strong></p></section></section></section></section><section>总的来说，单细胞转录组的CNV推断就是利用转录组的count表达矩阵，根据前后一些基因的表达推断该基因的CNV情况，基本原理就是CNV是以片段为单位的扩增和缺失，一旦该基因发生改变，周围的相邻基因很可能“同步”发生改变，论文中经常使用CNV推断判断细胞的良恶性，感兴趣的同学可以现学现用，立刻就能增加到你的研究中来！</section><section><section><strong>本文主要参考：<br></strong></section><section><span>1.知乎-焦老师讲遗传系列之12：拷贝数变异CNV</span></section><section><span>2.腾讯云-</span><span>单细胞转录组高级分析四：scRNA数据推断CNV</span><span></span></section><section><span>3.简书-单细胞转录组鉴定肿瘤细胞：CopyKAT和InferCNV</span></section><section><span></span></section><section><span>4.<span>简书-</span></span><span>单细胞CNV分析-copyKAT</span></section></section><section><section><span></span></section></section><section><section><br></section><section><strong></strong></section><section><strong>PS:我们在B站的同名账号“医学僧的科研日记”同步有公众号的讲解视频~目前已有10w余粉丝，感兴趣的小伙伴可以多多支持幺。</strong></section></section><p><img data-backh="578" data-backw="578" data-ratio="1" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/eBfvI7Ryx8dGcCicV0GoZLCE7VcyibIRZhOBsX1T7ic9Lp1Nvhce3Tx8Q8qI9x1t8OjcBvX92ZSkOITKHLyEQT93Q/640?wx_fmt=jpeg" data-type="jpeg" data-w="960" src="https://mmbiz.qpic.cn/mmbiz_jpg/eBfvI7Ryx8dGcCicV0GoZLCE7VcyibIRZhOBsX1T7ic9Lp1Nvhce3Tx8Q8qI9x1t8OjcBvX92ZSkOITKHLyEQT93Q/640?wx_fmt=jpeg"></p><section><span>医学僧的心</span><span>愿是为医学僧的支持者和科研</span><span>爱好者营造一个勤学好问、互帮互助、共同进步的优良环境。</span><span>但是人多了难免鱼龙混杂，良莠不齐，群里不乏有真心实意、脚踏实地的科研人，我们很尊重并欢迎他们，因此不希望被少数来发广告的、划水的、看热闹的、占个坑的所打扰，最终我们的心意付之东流，大家的热情消磨殆尽。</span><span>为了解决这个问题，我们决定提高入群的门槛——</span><strong>收费进群，</strong><span>这样，愿意进群的同志相信也是个圈内活跃用户，圈内人相互交流，岂不快哉！</span></section><section powered-by="xiumi.us"><section>因此，为了提升大家的效率，调动大家的积极性，吸收真正的科研爱好者，形成一个良好的学习交流氛围，我们决定增加群的数量，并细化群的功能，暂时增加以下几个主题群：</section><ul><li><p>文献下载-医学僧</p></li><li><p>机器学习（分类）-医学僧</p></li><li><p>ggplot可视化-医学僧</p></li><li><p>热图可视化-医学僧</p></li><li><p>TCGA挖掘-医学僧</p></li><li><p>GEO挖掘-医学僧</p></li><li><p>肿瘤预后模型-医学僧</p></li><li><p>肿瘤分子亚型-医学僧</p></li></ul><p>欢迎小伙伴们进群👏👏👏，每个群收费<strong>15元</strong>。请联系小编微信yixuesengkyrj或扫描下面二维码：</p></section><p><img data-backh="486" data-backw="578" data-ratio="0.8407407407407408" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/eBfvI7Ryx8eToe4Nt4XyydFdvqEM0WohEpzTSD45q9Xib8zpEq7kGgVBYh9AMdPvXJNOZAVz3MgrRJAKKfzxIWQ/640?wx_fmt=jpegwxfrom=5wx_lazy=1wx_co=1" data-type="jpegwxfrom=5wx_lazy=1wx_co=1" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_jpg/eBfvI7Ryx8eToe4Nt4XyydFdvqEM0WohEpzTSD45q9Xib8zpEq7kGgVBYh9AMdPvXJNOZAVz3MgrRJAKKfzxIWQ/640?wx_fmt=jpegwxfrom=5wx_lazy=1wx_co=1"></p></section></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/O5xL-QQXROSlWfkVeioqRw",target="_blank" rel="noopener noreferrer">原文链接</a>
