---
title: "如何预测上游转录因子？"
date: 2022-08-26T10:06:37Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Academic"]
---
如何预测上游转录因子？ by 生信菜鸟团
------
<div><p cid="n3" mdtype="paragraph"><span md-inline="plain">科研过程中，寻找转录因子的下游并不是一件太过困难的事情，无论是ChIP-seq 或者公共数据库都可以给我们想要的结果，但是，如果有靶基因，怎么去寻找可能的上游转录因子呢？</span></p><h2 cid="n4" mdtype="heading"><span>1、通过motif预测可能转录因子</span></h2><p cid="n5" mdtype="paragraph"><span md-inline="plain">从理论上讲，差异基因中可能有许多基因受到某几个转录因子的强烈调控，因此，我们使用Homer对对差异基因的promoter区域进行富集，就可能得到上游转录因子。</span></p><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="php"><code><span><span>#====================================加载R包====================================</span></span></code><code><span><br></span></code><code><span>library(magrittr)</span></code><code><span>library(tidyverse)</span></code><code><span>library(TFBSTools)</span></code><code><span>library(data.table)</span></code><code><span>library(JASPAR2016)</span></code><code><span>library(Biostrings)</span></code><code><span>library(rtracklayer)</span></code><code><span><br></span></code><code><span><span>#=================================构建模拟数据=================================</span></span></code><code><span>gtf &lt;- import(<span>"~/reference/gtf/hg38.gtf"</span>) %&gt;% </span></code><code><span>  <span>as</span>.data.frame %&gt;% </span></code><code><span>  mutate(gene_id = str_sub(gene_id, <span>1</span>, <span>15</span>)) %&gt;% </span></code><code><span>  filter(type == <span>"gene"</span>,</span></code><code><span>         gene_type == <span>"protein_coding"</span>)</span></code><code><span><br></span></code><code><span><span># 随机抽取1000个基因作为差异基因</span></span></code><code><span>genes &lt;- sample(gtf$gene_name, <span>1000</span>)</span></code><code><span>target &lt;- gtf %&gt;% </span></code><code><span>  filter(gene_name %in% all_of(genes)) %&gt;% </span></code><code><span>  dplyr::select(seqnames, start, end, gene_name, gene_name, strand) %T&gt;%</span></code><code><span>  write.table(<span>"target.bed"</span>, row.names = <span>FALSE</span>, col.names = <span>FALSE</span>,</span></code><code><span>              sep = <span>"\t"</span>, quote = <span>FALSE</span>)</span></code></pre></section><section><ul><li><li><li><li></ul><pre data-lang="nginx"><code><span><span># 上游2000bp认为是promoter区域（这个值每人看法不同）</span></span></code><code><span><br></span></code><code><span><span>bedtools</span> flank -s -l <span>2000</span> -r <span>0</span> -i target.bed -g ~/reference/gtf/genome_hg38.txt &gt; target.pro.bed</span></code><code><span>findMotifsGenome.pl target.pro.bed hg38 target -mask -len <span>8</span>,<span>10</span>,<span>12</span> -p <span>20</span></span></code></pre></section><p><img data-ratio="0.3447098976109215" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9LNUD8wl1skEMNz75axx5WDSYVZSsNnXu3R8gNzIgibibrPk5dposH5XSB7NQqRjiafTezuSJQA2JMw/640?wx_fmt=png" data-type="png" data-w="879" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9LNUD8wl1skEMNz75axx5WDSYVZSsNnXu3R8gNzIgibibrPk5dposH5XSB7NQqRjiafTezuSJQA2JMw/640?wx_fmt=png"></p><p cid="n9" mdtype="paragraph"><span md-inline="plain">因为是模拟数据，这些基因之间可能没有什么共同的调控关系，故而表现不佳。</span></p><h2 cid="n10" mdtype="heading"><span>2、通过TFBSTools预测可能的转录因子结合位点</span></h2><p cid="n11" mdtype="paragraph"><span md-inline="plain">值得注意的是，在这里首先要有想预测的转录因子，此处以FOXP3为例</span></p><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="javascript"><code><span>target.gtf &lt;- <span>import</span>(<span>"~/reference/gtf/hg38.gtf"</span>) %&gt;% </span></code><code><span>  <span>as</span>.data.frame %&gt;% </span></code><code><span>  mutate(gene_id = str_sub(gene_id, <span>1</span>, <span>15</span>)) %&gt;% </span></code><code><span>  filter(gene_name %<span>in</span>% all_of(sample(genes, <span>50</span>))) %T&gt;% </span></code><code><span>  <span>export</span>(<span>"target.gtf"</span>, <span>"gtf"</span>)</span></code><code><span> ## shell</span></code><code><span> gffread target.gtf -g ~<span>/reference/g</span>enome/hg38.fa -w target.fa</span></code><code><span><br></span></code><code><span>fa.list &lt;- readDNAStringSet(<span>"target.fa"</span>)</span></code><code><span><br></span></code><code><span>opts &lt;- list()</span></code><code><span>opts[[<span>"species"</span>]] &lt;- <span>9606</span></span></code><code><span>opts[[<span>"name"</span>]] &lt;- <span>"FOXP3"</span></span></code><code><span>opts[[<span>"all_versions"</span>]] &lt;- TRUE</span></code><code><span><br></span></code><code><span>siteset &lt;- getMatrixSet(JASPAR2016, opts) %&gt;% toPWM %&gt;% </span></code><code><span>  searchSeq(fa.list, min.score=<span>"95%"</span>, strand=<span>"+"</span>) %&gt;% </span></code><code><span>  writeGFF3 %T&gt;% </span></code><code><span>  write.csv(<span>"FOXP3-promoter.csv"</span>, row.names = FALSE)</span></code></pre></section><p><img data-galleryid="" data-ratio="0.2336578581363004" data-s="300,640" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9LNUD8wl1skEMNz75axx5W15RnYO2cITZlrSBGVqj23sojpkicQH2F79tcicBdFibjbDP9C4FRwzk7g/640?wx_fmt=png" data-type="png" data-w="719" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9LNUD8wl1skEMNz75axx5W15RnYO2cITZlrSBGVqj23sojpkicQH2F79tcicBdFibjbDP9C4FRwzk7g/640?wx_fmt=png"></p><p cid="n11" mdtype="paragraph"><span md-inline="plain"><span>TFBSTools还有许多其他用法，在这里只是介绍了最简单的一种，其他用法请大家自行探索。</span></span></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/bS-zF34NUXJKrfKFxz-TQw",target="_blank" rel="noopener noreferrer">原文链接</a>
