---
title: "批量读取转录组数据，绘制感兴趣的GSEA通路"
date: 2023-01-12T09:58:40Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Acdemic"]
---
批量读取转录组数据，绘制感兴趣的GSEA通路 by 生信菜鸟团
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><h2 data-tool="mdnice编辑器"><span></span><span>缘起</span><span></span></h2><p data-tool="mdnice编辑器">前两天，曾老师给了我一个3组每组4个样本的转录组数据，原文主要对于该数据进行了GSEA通路分析。老师想让我利用原文提供的FPKM矩阵，验证一下原文的通路变化。本打算对3组之间进行相互之间差异分析的，但是由于作者绘制的GSEA通路分析仅针对WT与KO两组，所以此处只针对这两组进行差异分析，绘制相应的GSEA通路。存在一个小插曲的是，文中绘制的5条GSEA通路中，小编只看见两条，这两条一条是通过gseKEGG获得，另一条是通过自定义通路进行GSEA分析，其余三条在文中没有找到，感兴趣的小伙伴可以尝试下。<span><strong>此外</strong></span>，本文挺有意思的点是，作者提供的基因表达矩阵是按照每个样本提供的，每组样本间的基因数量是不同的，所以需要分别批量读取三个组别，最终将三个组别的基因名取交集，获得相应的基因表达矩阵。</p><h2 data-tool="mdnice编辑器"><span></span><span>探究</span><span></span></h2><p data-tool="mdnice编辑器">今天，我们使用的转录组数据集是2020年发表在CANCER RESEARCH|GENOME AND EPIGENOME杂志上，文献名称为Loss of Apc Rapidly Impairs DNA Methylation Programs  and Cell Fate Decisions in Lgr5þ Intestinal Stem Cells。该文主要对三个组别进行了转录组数据分析，具体介绍如下。</p><p data-tool="mdnice编辑器">注意哈，咱们今天要绘制的通路主要是如下原文结果中框起来的两条通路：<img data-ratio="1.107890499194847" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9SAXwpBbgkuNicklAuexNj27qJz4EAyCqkHlDXnxA6TjWSBzcj610tVYXJVdlREQ0hxHmCAyODhbg/640?wx_fmt=png" data-type="png" data-w="621" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9SAXwpBbgkuNicklAuexNj27qJz4EAyCqkHlDXnxA6TjWSBzcj610tVYXJVdlREQ0hxHmCAyODhbg/640?wx_fmt=png"></p><h2 data-tool="mdnice编辑器"><span></span><span>转录组数据集介绍</span><span></span></h2><p data-tool="mdnice编辑器">该数据集提交在GEO官网，其GSE号是GSE123006。该数据集转录组数据有3个组别，分别是WT组、HT组与KO组，每个组别含有4个样本。感兴趣的小伙伴可以尝试下，具体数据下载链接见 <span>https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE123006&amp;format=file</span>。注意该下载链接中除了转录组数据，还含有RRBS数据，分析转录组数据时，需要将RRBS数据剔除。<img data-ratio="0.46440677966101696" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9SAXwpBbgkuNicklAuexNj2icwhAxKIEgdnfhx9K3eibUsWHRrM77QX44DXuAP6f1m6xKIiblWvt0TBw/640?wx_fmt=png" data-type="png" data-w="590" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9SAXwpBbgkuNicklAuexNj2icwhAxKIEgdnfhx9K3eibUsWHRrM77QX44DXuAP6f1m6xKIiblWvt0TBw/640?wx_fmt=png"></p><h2 data-tool="mdnice编辑器"><span></span><span>正式分析</span><span></span></h2><h3 data-tool="mdnice编辑器"><span></span><span><span></span>1.清空环境，加载包，加载数据</span><span></span></h3><pre data-tool="mdnice编辑器"><span></span><code>rm(list=ls())<br><span>library</span>(data.table)<br>samples=list.files(<span>"GSE123005_RAW/"</span>)<br>dir=<span>"GSE123005_RAW/"</span><br><span>#获取三组样本</span><br>wt=samples[grep(samples,pattern = <span>"wt"</span>)]<br>ht=samples[grep(samples,pattern = <span>"ht"</span>)]<br>ko=samples[grep(samples,pattern = <span>"ko"</span>)]<br><br><span>## 定义一个函数，直接获得每组的基因表达矩阵</span><br>group_exp=<span>function</span>(x){<br>  samples=x <span>#这里锁定组别，应该就可以了</span><br>  df=list()<br>  <span>for</span>(i <span>in</span> <span>1</span>:<span>4</span>){<br>    df[[i]]=fread(paste0(dir,samples[i]))[,c(<span>1</span>,<span>5</span>)]<br>  }<br>  <br>  exp=do.call(cbind,df) <span>#；将list转换为数据框，do.call只能取一列</span><br>  exp=as.data.frame(exp)<br>  exp=exp[,c(<span>1</span>,<span>2</span>,<span>4</span>,<span>6</span>,<span>8</span>)]<br>  table(duplicated(exp$gene_short_name))<br>  <span>## 重复数目有些多，按照最大值去重</span><br>  exp$meadian=apply(exp[,c(<span>2</span>,<span>5</span>)],<span>1</span>,median) <span>#1.先求中位数</span><br>  exp=exp[order(exp$gene_short_name,abs(exp$meadian),decreasing = <span>T</span>),] <span>#2.按order排降序</span><br>  exp=exp[!duplicated(exp$gene_short_name),] <span>#3.去重</span><br>  rownames(exp)=exp$gene_short_name <span>#4.命基因名</span><br>  exp=exp[,<span>2</span>:<span>5</span>]<br>  <span>return</span>(exp) <span>#最终获取exp的结果</span><br>}<br><br>wt_exp=group_exp(wt)<br>ht_exp=group_exp(ht)<br>ko_exp=group_exp(ko)<br><br><span>#将每组的基因表达矩阵整合成为一个大的基因表达矩阵</span><br>intersect_gene=intersect(intersect(rownames(wt_exp),rownames(ht_exp)),rownames(ko_exp))<br>wt_exp=wt_exp[intersect_gene,]<br>ht_exp=ht_exp[intersect_gene,]<br>ko_exp=ko_exp[intersect_gene,]<br>all_exp=cbind(wt_exp,ht_exp,ko_exp)<br></code></pre><h3 data-tool="mdnice编辑器"><span></span><span><span></span>2.对WT组与KO组进行差异分析，获得GSEA排序所需要的logFC</span><span></span></h3><pre data-tool="mdnice编辑器"><span></span><code>expMatrix=all_exp[,c(<span>1</span>:<span>4</span>,<span>9</span>:<span>12</span>)] <span>#分析WT组与KO组就可以了</span><br><span># 2.将FPKM值转换为TPM值</span><br>fpkmToTpm &lt;- <span>function</span>(fpkm)<br>{<br>  exp(log(fpkm) - log(sum(fpkm)) + log(<span>1e6</span>))<br>}<br>class(expMatrix)<br><span>## [1] "data.frame"</span><br><span>## [1] "data.frame"</span><br>expMatrix=na.omit(expMatrix)<br>numer=<span>function</span>(x){as.numeric(x)}<br>expMatrix=apply(expMatrix, <span>1</span>, numer) <br><span>##必须将矩阵中的字符类型由向量型改为字符型</span><br>expMatrix=t(expMatrix)<br>tpms &lt;- apply(expMatrix,<span>2</span>,fpkmToTpm) <br>tpms[<span>1</span>:<span>3</span>,]<br><span>##            [,1]     [,2]     [,3]      [,4]      [,5]      [,6]      [,7]</span><br><span>## Zzz3   42.12465 23.35684 34.31037  28.13509  54.89502  41.10471  47.76531</span><br><span>## Zzef1  40.00925 28.08166 43.80476  42.77359  18.23898  12.34853  15.04639</span><br><span>## Zyx   103.04799 83.06527 73.74318 103.07322 320.04760 190.79061 225.77413</span><br><span>##             [,8]</span><br><span>## Zzz3   25.362704</span><br><span>## Zzef1   9.756136</span><br><span>## Zyx   192.170262</span><br>colSums(tpms)<br><span>## [1] 1e+06 1e+06 1e+06 1e+06 1e+06 1e+06 1e+06 1e+06</span><br>colnames(tpms)=c(<span>"WT1"</span>,<span>"WT2"</span>,<span>"WT3"</span>,<span>"WT4"</span>,<span>"KO1"</span>,<span>"KO2"</span>,<span>"KO3"</span>,<span>"KO4"</span>)<br><span>## 组别设置</span><br>group_list=c(rep(<span>'WT'</span>,<span>4</span>),rep(<span>'KO'</span>,<span>4</span>))<br>group_list &lt;- factor(group_list,levels = c(<span>"WT"</span>,<span>"KO"</span>),ordered = <span>F</span>)<br><span>## 表达矩阵矫正</span><br>exprSet &lt;- tpms<br>boxplot(exprSet,outline=<span>FALSE</span>, notch=<span>T</span>,col=group_list, las=<span>2</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7142857142857143" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9SAXwpBbgkuNicklAuexNj2HBO8ZO6RLIuWZomm2QkmJKq9JBj45CRdaExn1YwkeNM5eiaZZu1HrHA/640?wx_fmt=png" data-type="png" data-w="672" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9SAXwpBbgkuNicklAuexNj2HBO8ZO6RLIuWZomm2QkmJKq9JBj45CRdaExn1YwkeNM5eiaZZu1HrHA/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code><span>library</span>(limma) <br>head(exprSet)<br><span>##               WT1       WT2       WT3        WT4        KO1        KO2</span><br><span>## Zzz3    42.124648 23.356837 34.310374  28.135094  54.895016  41.104711</span><br><span>## Zzef1   40.009245 28.081663 43.804760  42.773594  18.238978  12.348534</span><br><span>## Zyx    103.047985 83.065266 73.743177 103.073219 320.047602 190.790613</span><br><span>## Zyg11b  19.859384 12.025697 17.759368  14.845641  22.107926  15.621464</span><br><span>## Zxdc    20.371582 14.089519 17.435818  14.917163  19.211321  12.538796</span><br><span>## Zxdb     5.170714  5.482996  4.413419   3.496625   7.223751   3.526581</span><br><span>##               KO3        KO4</span><br><span>## Zzz3    47.765309  25.362704</span><br><span>## Zzef1   15.046395   9.756136</span><br><span>## Zyx    225.774133 192.170262</span><br><span>## Zyg11b  20.294522   8.836657</span><br><span>## Zxdc    16.861710   9.471667</span><br><span>## Zxdb     7.879218   2.424491</span><br>exprSet=normalizeBetweenArrays(exprSet)<br>boxplot(exprSet,outline=<span>FALSE</span>, notch=<span>T</span>,col=group_list, las=<span>2</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7142857142857143" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9SAXwpBbgkuNicklAuexNj2PHic2d3Ua86gxrBAxCwwT63CI3ceMhoSRp96LiaxibzyLu4rVqUxSBvPg/640?wx_fmt=png" data-type="png" data-w="672" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9SAXwpBbgkuNicklAuexNj2PHic2d3Ua86gxrBAxCwwT63CI3ceMhoSRp96LiaxibzyLu4rVqUxSBvPg/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code>range(exprSet)<br><span>## [1]     0.0 16468.1</span><br><span>## 判断数据是否需要转换；处理同芯片，20以内与否</span><br>exprSet &lt;- log2(exprSet+<span>1</span>)<br>range(exprSet)<br><span>## [1]  0.00000 14.00747</span><br><span>## 差异分析：</span><br>dat &lt;- exprSet<br>design=model.matrix(~factor( group_list ))<br>fit=lmFit(dat,design)<br>fit=eBayes(fit)<br>options(digits = <span>4</span>)<br>topTable(fit,coef=<span>2</span>,adjust=<span>'BH'</span>)<br><span>##           logFC AveExpr      t   P.Value adj.P.Val     B</span><br><span>## Slc1a1  -11.170   5.682 -52.12 8.303e-12 1.198e-07 16.36</span><br><span>## Sox17     7.471   4.029  44.73 2.971e-11 2.144e-07 15.60</span><br><span>## Msx1      6.912   4.405  40.46 6.846e-11 3.293e-07 15.04</span><br><span>## Notum     8.651   5.182  35.94 1.836e-10 5.677e-07 14.32</span><br><span>## Asb4      6.560   3.802  35.14 2.212e-10 5.677e-07 14.18</span><br><span>## Bpifb5    5.543   2.817  34.87 2.360e-10 5.677e-07 14.13</span><br><span>## Sbspon    5.064   3.056  33.05 3.686e-10 7.598e-07 13.78</span><br><span>## Wif1      5.249   3.437  31.96 4.868e-10 8.151e-07 13.56</span><br><span>## Slc38a4   5.391   4.767  31.58 5.380e-10 8.151e-07 13.47</span><br><span>## Prr18     4.653   3.414  31.39 5.648e-10 8.151e-07 13.43</span><br>deg=topTable(fit,coef=<span>2</span>,adjust=<span>'BH'</span>,number = <span>Inf</span>)<br>head(deg)<br><span>##          logFC AveExpr      t   P.Value adj.P.Val     B</span><br><span>## Slc1a1 -11.170   5.682 -52.12 8.303e-12 1.198e-07 16.36</span><br><span>## Sox17    7.471   4.029  44.73 2.971e-11 2.144e-07 15.60</span><br><span>## Msx1     6.912   4.405  40.46 6.846e-11 3.293e-07 15.04</span><br><span>## Notum    8.651   5.182  35.94 1.836e-10 5.677e-07 14.32</span><br><span>## Asb4     6.560   3.802  35.14 2.212e-10 5.677e-07 14.18</span><br><span>## Bpifb5   5.543   2.817  34.87 2.360e-10 5.677e-07 14.13</span><br>fpkm_deg=deg<br></code></pre><h3 data-tool="mdnice编辑器"><span></span><span><span></span>3.进行gseKEGG分析，绘制Wnt信号通路</span><span></span></h3><pre data-tool="mdnice编辑器"><span></span><code><span>## 按logFC排序，获得genelist,进行gseKEGG分析</span><br>DEG_DESeq2=fpkm_deg<br>head(DEG_DESeq2)<br><span>##          logFC AveExpr      t   P.Value adj.P.Val     B</span><br><span>## Slc1a1 -11.170   5.682 -52.12 8.303e-12 1.198e-07 16.36</span><br><span>## Sox17    7.471   4.029  44.73 2.971e-11 2.144e-07 15.60</span><br><span>## Msx1     6.912   4.405  40.46 6.846e-11 3.293e-07 15.04</span><br><span>## Notum    8.651   5.182  35.94 1.836e-10 5.677e-07 14.32</span><br><span>## Asb4     6.560   3.802  35.14 2.212e-10 5.677e-07 14.18</span><br><span>## Bpifb5   5.543   2.817  34.87 2.360e-10 5.677e-07 14.13</span><br>nrDEG=DEG_DESeq2[,c(<span>1</span>,<span>4</span>)]<br>colnames(nrDEG)=c(<span>'log2FoldChange'</span>,<span>'pvalue'</span>)<br>head(nrDEG) <br><span>##        log2FoldChange    pvalue</span><br><span>## Slc1a1        -11.170 8.303e-12</span><br><span>## Sox17           7.471 2.971e-11</span><br><span>## Msx1            6.912 6.846e-11</span><br><span>## Notum           8.651 1.836e-10</span><br><span>## Asb4            6.560 2.212e-10</span><br><span>## Bpifb5          5.543 2.360e-10</span><br><span>library</span>(org.Hs.eg.db)<br><span>library</span>(org.Mm.eg.db)<br><span>library</span>(clusterProfiler)<br>R.utils::setOption(<span>"clusterProfiler.download.method"</span>,<span>'auto'</span>)<br>gene &lt;- bitr(rownames(nrDEG), fromType = <span>"SYMBOL"</span>,<br>             toType =  <span>"ENTREZID"</span>,<br>             OrgDb = org.Mm.eg.db)<br>gene$logfc &lt;- nrDEG$log2FoldChange[match(gene$SYMBOL,rownames(nrDEG))]<br>geneList=gene$logfc<br>names(geneList)=gene$ENTREZID <br>geneList=sort(geneList,decreasing = <span>T</span>)<br><span>## 进行gseKEGG分析</span><br>kk_gse &lt;- gseKEGG(geneList     = geneList,<br>                  organism     = <span>'mmu'</span>,<br>                  nPerm        = <span>1000</span>,<br>                  minGSSize    = <span>10</span>,<br>                  pvalueCutoff = <span>0.9</span>,<br>                  verbose      = <span>FALSE</span>)<br><span>library</span>(enrichplot)<br><br><span>## 5个通路中只找到Wnt信号通路---对应的通路ID号为mmu04310</span><br>paths=<span>"mmu04310"</span><br>gseaplot2(kk_gse, paths, color = colorspace::rainbow_hcl(<span>4</span>),<br>          pvalue_table = <span>TRUE</span>,title=(<span>"Wnt signaling pathway"</span>))<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7142857142857143" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9SAXwpBbgkuNicklAuexNj2XtZqebibON4XJdoh4RAJVKALzOqNia0JqdYv1sBG2QCiaDUTn1g9Ic25g/640?wx_fmt=png" data-type="png" data-w="672" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9SAXwpBbgkuNicklAuexNj2XtZqebibON4XJdoh4RAJVKALzOqNia0JqdYv1sBG2QCiaDUTn1g9Ic25g/640?wx_fmt=png"></figure><h3 data-tool="mdnice编辑器"><span></span><span><span></span>4. 自定义通路，进行GSEA分析</span><span></span></h3><p data-tool="mdnice编辑器">文中作者根据以前发表的文献，自定义了一条肠干细胞通路；为文章的附表4，感兴趣的同学可以下载,尝试绘制下。</p><pre data-tool="mdnice编辑器"><span></span><code>custome=readxl::read_xls(<span>"custome_gene_signature.xls"</span>)<br>head(custome)<br><span>## # A tibble: 6 x 4</span><br><span>##   `Table S4 : custom gene set signatures`                     ...2   ...3  ...4 </span><br><span>##   &lt;chr&gt;                                                       &lt;chr&gt;  &lt;chr&gt; &lt;chr&gt;</span><br><span>## 1 GSEA INTESTINAL STEM CELL SIGNATURE (Mu&lt;U+00F1&gt;oz, J. et al. 2012) &lt;NA&gt;   &lt;NA&gt;  &lt;NA&gt; </span><br><span>## 2 PROBE                                                       DESCR~ GENE~ GENE~</span><br><span>## 3 Sycn                                                        NM_02~ SYCN  sync~</span><br><span>## 4 Fras1                                                       NM_17~ FRAS1 Fras~</span><br><span>## 5 Vdr                                                         NM_00~ VDR   vita~</span><br><span>## 6 Sfrp5                                                       NM_01~ SFRP5 secr~</span><br>geneset=custome$`Table S4 : custom gene set signatures`<br>geneset=geneset[<span>2</span>:length(geneset)]<br><span># 建立一个符合gmt文件转换的list</span><br>ct_lis=list()<br>ct_lis[[<span>1</span>]]=geneset<br>names(ct_lis)=<span>"Interstinal stem cells"</span><br><span># 获取gmt文件</span><br>file=<span>"gmt.txt"</span><br>gs=ct_lis<br>write.gmt &lt;- <span>function</span>(gs,file){<br>  sink(file)<br>  lapply(names(ct_lis), <span>function</span>(i){<br>    cat( paste(c(i,<span>'tmp'</span>,gs[[i]]),collapse=<span>'\t'</span>) )<br>    cat(<span>'\n'</span>)<br>  })<br>  sink()<br>}<br>write.gmt(gs,file)<br><br>names(geneList)=gene$SYMBOL <br>geneset &lt;- read.gmt(<span>'gmt.txt'</span>) <span>#本质只有两列</span><br>egmt &lt;- GSEA(geneList, TERM2GENE=geneset, <br>             verbose=<span>FALSE</span>)<br><span>## 绘制gsea分析的结果</span><br>gseaplot2(egmt, color = colorspace::rainbow_hcl(<span>4</span>),<br>          geneSetID = rownames(egmt[<span>1</span>,]),title=(<span>"Interstinal stem cells"</span>), pvalue_table = <span>TRUE</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7142857142857143" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9SAXwpBbgkuNicklAuexNj2ebT0GcGjFTptDXe1eqaickoHwIcBzpI4tkkK5G4BxiacptrZLyUgCrwg/640?wx_fmt=png" data-type="png" data-w="672" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9SAXwpBbgkuNicklAuexNj2ebT0GcGjFTptDXe1eqaickoHwIcBzpI4tkkK5G4BxiacptrZLyUgCrwg/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器"><span>以上演示了批量读取单个样本的转录组数据，将其整合，进行差异分析以及绘制GSEA通路图</span>。文中GSEA通路图有5副，但是由于小编在原文中只看见了两幅，并且两幅一定程度上代表了GSEA分析，gseKEGG与自定义通路，所以小编在此展现两幅。当然，可以导入其它的通路以及其它包进行分析，如Msigdb等。感兴趣的小伙伴可以阅读原文尝试下，若看见了其余三条通路，也可以一起沟通交流下。 </p><p data-tool="mdnice编辑器"><span>小编绘制的两幅通路图的结果与作者的结果是一致的，相较于WT组，两个通路在KO组均显著富集。</span></p></section><p><br></p><p><mp-style-type data-value="10000"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/OOXvUVbQsGhYlrfma0i0Eg",target="_blank" rel="noopener noreferrer">原文链接</a>
