---
title: "敏感且精确的全基因组比对工具：anchorwave"
date: 2022-08-28T07:02:49Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Acdemic"]
---
敏感且精确的全基因组比对工具：anchorwave by 生信菜鸟团
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><p data-tool="mdnice编辑器">随着测序技术的不断发展，越来越多的基因组被测序组装，敏感且精细的基因组比对复杂基因组能从中获取很多有用的信息，2021年12月在PNAS发表一个敏感且精确比对的植物基因组软件：anchorwave. 它使用保守序列如外显子或编码蛋白序列作为anchor（锚），使用当前最快的WFA算法进行碱基水平上的全基因组比对。anchorwave依据植物基因组的特点进行设计，能够快速地识别大的和小的变异，且操作简单，容易上手，本文以玉米与高粱的基因组为例对存在染色体融合，易位，全基因组复制事件等变异的基因组间进行全基因组比对。</p><p data-tool="mdnice编辑器"><strong>「anchorwave流程图：」</strong><img data-ratio="0.3439524838012959" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9ghVDRMNWNGxcxticibDa4VFVsMLg7ueKrbFjZsql82GjnZ50WVsqzPjnMvNQZZiaXdeVeYCf5ic2S1A/640?wx_fmt=png" data-type="png" data-w="1852" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9ghVDRMNWNGxcxticibDa4VFVsMLg7ueKrbFjZsql82GjnZ50WVsqzPjnMvNQZZiaXdeVeYCf5ic2S1A/640?wx_fmt=png"></p><p data-tool="mdnice编辑器"><strong>「anchorwave下载安装」</strong></p><pre data-tool="mdnice编辑器"><span></span><code><span>#源代码安装</span><br>git <span>clone</span> https://github.com/baoxingsong/anchorwave.git<br><span>cd</span> anchorwave<br>cmake ./<br>make<br><span>#conda安装anchorwave</span><br>conda install -c bioconda -c conda-forge anchorwave<br><span>#docker安装anchorwave</span><br>docker build -f docker/Dockerfile -t anchorwave ./<br>docker run -it anchorwave anchorwave<br><br></code></pre><p data-tool="mdnice编辑器"><strong>「1.	下载两个基因组的序列文件和参考基因组的注释文件」</strong></p><pre data-tool="mdnice编辑器"><span></span><code><span>#Download and uncompress genome and GFF file of reference genome</span><br><br>wget ftp://ftp.ensemblgenomes.org/pub/plants/release-34/fasta/zea_mays/dna/Zea_mays.AGPv4.dna.toplevel.fa.gz<br>wget ftp://ftp.ensemblgenomes.org/pub/plants/release-34/gff3/zea_mays/Zea_mays.AGPv4.34.gff3.gz<br>wget http://ftp.ensemblgenomes.org/pub/plants/release-54/fasta/sorghum_bicolor/dna/Sorghum_bicolor.Sorghum_bicolor_NCBIv3.dna.toplevel.fa.gz<br><br>gunzip Zea_mays.AGPv4.dna.toplevel.fa.gz<br>gunzip Zea_mays.AGPv4.34.gff3.gz<br>gunzip Sorghum_bicolor.Sorghum_bicolor_NCBIv3.dna.toplevel.fa.gz<br><br></code></pre><p data-tool="mdnice编辑器"><strong>「2.提取参考基因组的中的保守序列并将保守序列分别匹配到两个基因组上」</strong></p><p data-tool="mdnice编辑器">查看基因组的内容，需要删除大豆中不是染色体上的序列。</p><pre data-tool="mdnice编辑器"><span></span><code><span>#Access the level of chromosome and collineraity</span><br>samtools faidx Zea_mays.AGPv4.dna.toplevel.fa<br>samtools faidx Sorghum_bicolor.Sorghum_bicolor_NCBIv3.dna.toplevel.fa<br><span>#Acquire the first contig name "super_16" </span><br>less Sorghum_bicolor.Sorghum_bicolor_NCBIv3.dna.toplevel.fa.fai <br>grep -n <span>"super_16"</span> Sorghum_bicolor.Sorghum_bicolor_NCBIv3.dna.toplevel.fa<br>head -11394099 Sorghum_bicolor.Sorghum_bicolor_NCBIv3.dna.toplevel.fa &gt; Sorghum.fa<br><br></code></pre><p data-tool="mdnice编辑器">这一步需要使用必须使用anchorwave提取保守序列，这样可以过滤一些不必要的anchors，最小化的减少下一步中minimap2的限制。-r代表要输入的参考基因组，-i代表他的注释文件，-o代表他输出的所有的保守序列的文件。</p><pre data-tool="mdnice编辑器"><span></span><code>anchorwave gff2seq -i Zea_mays.AGPv4.34.gff3 -r Zea_mays.AGPv4.dna.toplevel.fa -o cds.fa<br><br></code></pre><p data-tool="mdnice编辑器">使用minimap2将保守序列匹配到两个基因组上，-x splice 代表将保守序列映射到基因组，-t 代表线程数，-k 代表kmer， -a代表输出sam格式的文件，-p代表如果它们的比对得分大于初次比对得分的0.4，-N代表最小长度不小于20，还要输入他们的基因组和保守序列。</p><pre data-tool="mdnice编辑器"><span></span><code>minimap2 -x splice -a -t 10 -p 0.4 -N 20 Zea_mays.AGPv4.dna.toplevel.fa cds.fa &gt; ref.sam<br>minimap2 -x splice -a -t 10 -p 0.4 -N 20 Sorghum.fa cds.fa &gt; cds.sam<br></code></pre><p data-tool="mdnice编辑器"><strong>「如果已经知道两个基因组间的共线性关系可以直接选择genoAli或proali进行比对，不需要进行共线性区域的识别与可视化，可以直接进行第三步单碱基分辨率的全基因组比对。」</strong></p><p data-tool="mdnice编辑器">假设不知道两基因组的共线性关系，可采用如下步骤：格式转化：</p><pre data-tool="mdnice编辑器"><span></span><code>perl alignmentToDotplot.pl Zea_mays.AGPv4.34.gff3 cds.sam &gt; cds.tab<br></code></pre><p data-tool="mdnice编辑器">使用R进行绘图</p><pre data-tool="mdnice编辑器"><span></span><code>library(ggplot2)<br>#Transform Coordinates using a function.<br>changetoM &lt;- function ( position ){<br>position=position/1000000;<br>paste(position, "M", sep="")<br>}<br>#Read gene position, belong to which chromosome and so on<br>data =read.table("cds.tab")<br>#Select all euchromosomes as factor<br>data = data[which(data$V1 %in% c("1", "2", "3", "4", "5", "6", "7", "8", "9", "10")),]<br>data = data[which(data$V3 %in% c("1", "2", "3", "4", "5", "6", "7", "8", "9", "10")),]<br>data$V1 = factor(data$V1, levels=c("1", "2", "3", "4", "5", "6", "7", "8", "9", "10"))<br>data$V3 = factor(data$V3, levels=c("1", "2", "3", "4", "5", "6", "7", "8", "9", "10"))<br>#Using ggplot2 to plot a dotplot and beautify it.<br>ggplot(data=data, aes(x=V4, y=V2)) +geom_point(size=0.5, aes(color=V5)) +<br>facet_grid(V1 ~ V3, scales="free",space="free") +labs(x="sorghum", y="maize")+<br>scale_x_continuous(labels=changetoM) + scale_y_continuous(labels=changetoM) +<br>theme(axis.line = element_blank(),<br>panel.background = element_blank(),<br>panel.border = element_rect(fill=NA,color="black", size=0.5, linetype="solid"),<br>axis.text.y = element_text( colour = "black"),<br>legend.position='none',<br>axis.text.x = element_text(angle=300, hjust=0, vjust=1, colour = "black"))<br><br></code></pre><p data-tool="mdnice编辑器">可得到如下图片：<img data-ratio="0.7314560439560439" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/iaRJcrq2Los9ghVDRMNWNGxcxticibDa4VFBphybWanRoeFRYMVSfXg15HmYcpeOyeyuRATVrHwvcgJJspS1R5S2Q/640?wx_fmt=jpeg" data-type="jpeg" data-w="1456" src="https://mmbiz.qpic.cn/mmbiz_jpg/iaRJcrq2Los9ghVDRMNWNGxcxticibDa4VFBphybWanRoeFRYMVSfXg15HmYcpeOyeyuRATVrHwvcgJJspS1R5S2Q/640?wx_fmt=jpeg">我们可以看出两基因组间经历了全基因组复制事件，选用proali功能。</p><p data-tool="mdnice编辑器">查看两基因组的共线性区</p><pre data-tool="mdnice编辑器"><span></span><code>anchorwave proali -r Zea_mays.AGPv4.dna.toplevel.fa -i Zea_mays.AGPv4.34.gff3 \<br>-a cds.sam -as cds.fa -ar ref.sam -s Sorghum.fa -n align1.anchors -R 1 -Q 2 -ns<br><br></code></pre><p data-tool="mdnice编辑器">使用R进行可视化</p><pre data-tool="mdnice编辑器"><span></span><code>library(ggplot2)<br>#Read gene position and belong to which chromosome and so on<br>data =read.table("align1.anchors", header=TRUE)<br>#Select all euchromosomes as factor<br>data = data[which(data$refChr %in% c("1", "2", "3", "4",<br>"5", "6", "7", "8", "9", "10")),]<br>data = data[which(data$queryChr %in% c("1", "2", "3", "4",<br>"5", "6", "7", "8", "9", "10")),]<br>data$refChr = factor(data$refChr, levels=c("1", "2", "3", "4",<br>"5", "6", "7", "8", "9", "10"))<br>data$queryCh = factor(data$queryChr, levels=c("1", "2", "3", "4",<br>"5", "6", "7", "8", "9", "10"))<br>#Using ggplot2 to plot a dotplot and beautify it.<br>ggplot(data=data, aes(x=queryStart, y=referenceStart))+<br>geom_point(size=0.5, aes(color=strand)) +<br>facet_grid(refChr~queryChr, scales="free", space="free") +<br>labs(x="sorghum", y="maize")+scale_x_continuous(labels=changetoM) +<br>scale_y_continuous(labels=changetoM) +<br>  theme(axis.line = element_blank(),<br>        panel.background = element_blank(),<br>        panel.border = element_rect(fill =NA,color="black", size=0.5, linetype="solid"),<br>        axis.text.y = element_text( colour = "black"),<br>        legend.position='none',<br>        axis.text.x = element_text(angle=300,hjust=0, vjust=0.5, colour = "black") )<br><br></code></pre><p data-tool="mdnice编辑器"><img data-ratio="0.7377501725327812" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9ghVDRMNWNGxcxticibDa4VFW56xh7fCAnx5g9hlfQlibr8Rrv3jiaSt3FWEYV2ghxT3JTt5JNtXaLRQ/640?wx_fmt=png" data-type="png" data-w="1449" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9ghVDRMNWNGxcxticibDa4VFW56xh7fCAnx5g9hlfQlibr8Rrv3jiaSt3FWEYV2ghxT3JTt5JNtXaLRQ/640?wx_fmt=png">可以清晰地看到他们的共线性区域。</p><p data-tool="mdnice编辑器"><strong>「3.使用anchorwave进行全基因组比对」</strong>proali代表对存在染色融合，易位全基因复制等大的变异间的全基因组比对，genoAli代表对有倒位或不存在染色体变异的基因组间进行比对。-i代表输入参考基因组的注释文件，-r 和-s 参考和查询基因组文件，-a 和-ar代表anchors映射到两个基因组的结果，不写-ns代表产生新的anchor。-t 代表线程为10，请根据内存大小进行设置，以免中止，-o和-f均为输出的文件。生成的maf文件可以转换成vcf文件进行后续下游分析。</p><pre data-tool="mdnice编辑器"><span></span><code>anchorwave proali -i Zea_mays.AGPv4.34.gff3 -r Zea_mays.AGPv4.dna.toplevel.fa\<br>-a cds.sam -as cds.fa -ar ref.sam -s Sorghum.fa -n align1.anchors -o align1.maf\<br>-t 10 -R 1 -Q 2 -f align1.f.maf &gt; interlog1 2&gt;&amp;1<br><br></code></pre><p data-tool="mdnice编辑器">参考文献相关论文如下：<span>https://www.pnas.org/doi/10.1073/pnas.2113075119</span></p><section>其他代码见此链接：<span>h</span><span>ttp</span><span>s://github.com/baoxingsong/anchorwave</span></section><p><span><br></span></p></section><p><br></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/NOyGVc4P8DCSGJ1s_SMraw",target="_blank" rel="noopener noreferrer">原文链接</a>
