---
title: "有技巧的读入maf数据"
date: 2022-11-13T06:51:08Z
draft: ["false"]
tags: [
  "fetched",
  "YuYuFiSH"
]
categories: ["Acdemic"]
---
有技巧的读入maf数据 by YuYuFiSH
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><h1 data-tool="mdnice编辑器"><span></span>做个maftools作业</h1><p data-tool="mdnice编辑器">最近在跟着小郭老师的外显子课程学习，讲解的十分详细，语雀上竟然还有上万字资料总结给我们，真不愧是生信技能树的首席生信工程师！</p><p data-tool="mdnice编辑器">外显子下游的基本分析离不开强大的maftools R包，找了老师的学徒作业来做，根据作者给出的csv文件绘制<span><strong>突变全景图（瀑布图）</strong></span></p><p data-tool="mdnice编辑器"><a target="_blank" href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247517177&amp;idx=1&amp;sn=a9bde232c545ef7976f9765d85c0d98a&amp;scene=21#wechat_redirect" textvalue="小队列的肿瘤外显子临床预后意义" linktype="text" imgurl="" imgdata="null" tab="innerlink" data-linktype="2">小队列的肿瘤外显子临床预后意义</a></p><figure data-tool="mdnice编辑器"><img data-ratio="0.6555555555555556" data-src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc7S7E87Jzy6Uady20IRhbftYbtiamlQY4iaFTQ3JPYwSOn0XfiaibIWALvoVG8NUo5hNiaHVPjwwdcMMUw/640?wx_fmt=png" data-type="png" data-w="1080" src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc7S7E87Jzy6Uady20IRhbftYbtiamlQY4iaFTQ3JPYwSOn0XfiaibIWALvoVG8NUo5hNiaHVPjwwdcMMUw/640?wx_fmt=png"></figure><h2 data-tool="mdnice编辑器"><span></span>获得上游分析的VCF文件</h2><p data-tool="mdnice编辑器">以往我们，需要将上游分析获得的VCF文件，注释、整理成标准的Maf格式。注释工具很多，如：SnpEff、ANNOVAR、GATK Funcotator、VEP、CADD等，不同软件依赖不同的运行环境和注释数据库</p><p data-tool="mdnice编辑器">把数据读取成maf对象，然后就可以画图或者分析</p><figure data-tool="mdnice编辑器"><img data-ratio="0.562429696287964" data-src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc7S7E87Jzy6Uady20IRhbftafOjJ8C38o3ncMvFRSqC7AV3YkFFHnnIaeR5paZP9h3iciaxernarlTw/640?wx_fmt=png" data-type="png" data-w="1778" src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc7S7E87Jzy6Uady20IRhbftafOjJ8C38o3ncMvFRSqC7AV3YkFFHnnIaeR5paZP9h3iciaxernarlTw/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">官方文档：https://bioconductor.org/packages/release/bioc/vignettes/maftools/inst/doc/maftools.html</p><ul><li><p data-tool="mdnice编辑器">读取：<code>read.maf</code></p></li></ul><pre data-tool="mdnice编辑器"><span></span><code><span># 先读取 vep 注释的 maf 文件 ----</span><br>vep_maf = data.table::fread(<span>"vep_merge.maf"</span>)<br>vep_maf = as.data.frame(vep_maf)<br>dim(vep_maf) <span># [1] 5386  134</span><br><br><span># Add VAF field</span><br>vep_maf<span>$VAF</span> = vep_maf<span>$t_alt_count</span>/vep_maf<span>$t_depth</span><br><span># 过滤掉非编码区或者同义突变，即保留 Variant Classifications 为</span><br><span># "Frame_Shift_Del", "Frame_Shift_Ins", "Splice_Site", </span><br><span># "Translation_Start_Site","Nonsense_Mutation", "Nonstop_Mutation", </span><br><span># "In_Frame_Del","In_Frame_Ins", "Missense_Mutation"</span><br>Variants = c(<span>"Frame_Shift_Del"</span>, <span>"Frame_Shift_Ins"</span>, <span>"Splice_Site"</span>, <br>  <span>"Translation_Start_Site"</span>,<span>"Nonsense_Mutation"</span>, <span>"Nonstop_Mutation"</span>, <br>  <span>"In_Frame_Del"</span>,<span>"In_Frame_Ins"</span>, <span>"Missense_Mutation"</span>)<br><br>vep_maf = vep_maf[vep_maf<span>$Variant_Classification</span> %<span>in</span>% Variants, ]<br><br><span># 可以先把数据保存一下，后续其他分析用到了可以直接  load</span><br>save(vep_maf, clinical.data, file = <span>"maf_clinical.Rdata"</span>)<br><br><span># 用 maftools 的 read.maf 读取，默认只保留 non-synonymous </span><br>maf = read.maf(vep_maf, clinicalData = clinical.data)<br><span># -Validating</span><br><span># -Summarizing</span><br><span># --Possible FLAGS among top ten genes:</span><br><span>#   TTN</span><br><span># NEB</span><br><span># DNAH17</span><br><span># MUC16</span><br><span># FLG</span><br><span># DNAH11</span><br><span># -Processing clinical data</span><br><span># -Finished in 0.827s elapsed (1.681s cpu) </span><br><br><span># 查看一下 maf 的大致信息 ----</span><br>getSampleSummary(maf)<br>getGeneSummary(maf)<br></code></pre><p data-tool="mdnice编辑器">但这次作业，作者只提供了csv表格，并且内容也不是传统意义上的maf文件</p><p data-tool="mdnice编辑器">Additional file 2: Table S2. Somatic mutations detected in patients with PTCL.</p><figure data-tool="mdnice编辑器"><img data-ratio="0.22764227642276422" data-src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc7S7E87Jzy6Uady20IRhbftdETMKRUCLGQia3pPN7ZdfnxddEL1aUC7KgbYXS3WicT3BOTiaOkQeia2Eg/640?wx_fmt=png" data-type="png" data-w="1722" src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc7S7E87Jzy6Uady20IRhbftdETMKRUCLGQia3pPN7ZdfnxddEL1aUC7KgbYXS3WicT3BOTiaOkQeia2Eg/640?wx_fmt=png"></figure><h2 data-tool="mdnice编辑器"><span>有技巧的读入数据</span></h2><p data-tool="mdnice编辑器">MAF需要的列如果看官方说明文档：https://docs.gdc.cancer.gov/Data/File_Formats/MAF_Format/，会发现有上百列的信息，从最基本的基因组坐标，到更复杂的cosmic变异注释。</p><ul><li><p data-tool="mdnice编辑器">必须包含的列：Hugo_Symbol, Chromosome, Start_Position, End_Position, Reference_Allele, Tumor_Seq_Allele2, Variant_Classification, Variant_Type and Tumor_Sample_Barcode</p></li><li><p data-tool="mdnice编辑器">推荐添加的其他列：VAF (Variant Allele Frequecy) , amino acid change information.</p></li></ul><p data-tool="mdnice编辑器">必须的列无外乎是一些样本基因名，染色体位置，突变类型等，一些csv表没有的数据直接写成NA值</p><p data-tool="mdnice编辑器"><a target="_blank" href="https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&amp;mid=2247506242&amp;idx=2&amp;sn=3578e27e0087472d72b852749e759ee3&amp;scene=21#wechat_redirect" textvalue="不是maf格式的somatic突变数据就没办法读入到maftools了么" linktype="text" imgurl="" imgdata="null" tab="innerlink" data-linktype="2">不是maf格式的somatic突变数据就没办法读入到maftools了么</a> </p><pre data-tool="mdnice编辑器"><span></span><code>rm(list = ls())<br>require(maftools)<br>options(stringsAsFactors = F) <br>library(data.table)<br>setwd(<span>'/home/yulan/wes/homework/'</span>)<br><br>library(readxl)<br>res &lt;- read_xlsx(<span>'40164_2021_200_MOESM2_ESM.xlsx'</span>,col_names = T,skip = 1)<br>head(res)<br>colnames(res)<br><span># 用 maftools 的 read.maf 读取</span><br><span># 必填字段: Hugo_Symbol, Chromosome, Start_Position,End_Position,Reference_Allele,Tumor_Seq_Allele2,Variant_Classification, Variant_Type and Tumor_Sample_Barcode.</span><br>maf_res &lt;- data.frame(Hugo_Symbol=res$`Gene Symbol`, <br>           Chromosome=NA, <br>           Start_Position=NA,<br>           End_Position=NA,<br>           Reference_Allele=NA,<br>           Tumor_Seq_Allele2=NA,<br>           Variant_Classification=res<span>$Function</span>, <br>           Variant_Type=res$`Vary Type`,<br>           Tumor_Sample_Barcode=res<span>$PatientID</span>,<br>           vaf=res<span>$caseAF</span><br>           )<br><br>table(maf_res<span>$Variant_Classification</span> )<br>table(maf_res<span>$Variant_Type</span>)<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>&gt; head(res)<br><span># A tibble: 6 × 11</span><br>  Sampl…¹ Patie…² Gene …³ cHGVS pHGVS Funct…⁴ Trans…⁵ ExIn_ID Cosmi…⁶ Vary …⁷ caseAF<br>  &lt;chr&gt;     &lt;dbl&gt; &lt;chr&gt;   &lt;chr&gt; &lt;chr&gt; &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt; <br>1 189000…       1 AXL     c.13… p.W4… missen… NM_021… EX11    5749610 snv     0.432…<br>2 189000…       1 BARD1   c.15… p.V5… missen… NM_000… EX6     .       indel   0.9829<br>3 189000…       1 BRIP1   c.58… p.N1… missen… NM_032… EX6     249456  snv     0.467…<br>4 189000…       1 CHD3    c.22… p.K7… cds-del NM_001… EX13    437716  indel   3.202…<br>5 189000…       1 CHD8    c.33… p.R1… missen… NM_001… EX16    .       snv     0.467…<br>6 189000…       1 CPS1    c.10… p.T3… missen… NM_001… EX10    .       indel   0.981…<br><span># … with abbreviated variable names ¹SampleID, ²PatientID, ³`Gene Symbol`,</span><br><span>#   ⁴Function, ⁵Transcript, ⁶`Cosmic ID`, ⁷`Vary Type`</span><br><br>&gt; table(maf_res<span>$Variant_Classification</span> )<br><br>      cds-del       cds-ins    frameshift      missense         ncRNA      nonsense <br>           24             5            43           712             2            45 <br>         span      splice-3      splice-5     stop-loss stop-retained <br>            4             6            13             1             1 <br>&gt; table(maf_res<span>$Variant_Type</span>)<br><br>indel   snv <br>  106   750 <br></code></pre><h2 data-tool="mdnice编辑器"><span></span>遇到的几个坑</h2><h3 data-tool="mdnice编辑器"><span>1. 参数 vc_nonSyn</span></h3><p data-tool="mdnice编辑器">我以为设置差不多的格式，那直接<code>read.maf </code>不就行了吗？谁知道又会报错（知道肯定不会那么顺利的...）</p><pre data-tool="mdnice编辑器"><span></span><code>&gt; maf = read.maf(maf_res)<br><br>-Validating<br>--Removed 803 duplicated variants<br>--Non MAF specific values <span>in</span> Variant_Classification column:<br>  missense<br>  cds-del<br>  splice-3<br>  frameshift<br>  nonsense<br>  ncRNA<br>--Non MAF specific values <span>in</span> Variant_Type column:<br>  snv<br>  indel<br>-Silent variants: 53 <br>Error <span>in</span> read.maf(maf_res) : No non-synonymous mutations found<br>Check `vc_nonSyn`` argumet <span>in</span> `read.maf` <span>for</span> details<br></code></pre><p data-tool="mdnice编辑器">看到报错的信息似乎是参数<code>vc_nonSyn</code>引起的</p><pre data-tool="mdnice编辑器"><span></span><code>vc_nonSyn <br>NULL. Provide manual list of variant classifications to be considered as non-synonymous. Rest will be considered as silent variants. Default uses Variant Classifications with High/Moderate variant consequences. https://m.ensembl.org/info/genome/variation/prediction/predicted_data.html: <span>"Frame_Shift_Del"</span>, <span>"Frame_Shift_Ins"</span>, <span>"Splice_Site"</span>, <span>"Translation_Start_Site"</span>,<span>"Nonsense_Mutation"</span>, <span>"Nonstop_Mutation"</span>, <span>"In_Frame_Del"</span>,<span>"In_Frame_Ins"</span>, <span>"Missense_Mutation"</span><br><br>verbose <br>TRUE logical. Default to be talkative and prints summary.<br></code></pre><p data-tool="mdnice编辑器">这个参数默认的是九种非同义突变：</p><ul><li><p data-tool="mdnice编辑器">missense_mutation:错义突变</p></li><li><p data-tool="mdnice编辑器">frame_shift_del：移码缺失突变</p></li><li><p data-tool="mdnice编辑器">nonsense_mutation：无义突变</p></li><li><p data-tool="mdnice编辑器">frame_shift_ins：移码插入突变</p></li><li><p data-tool="mdnice编辑器">splice_site：剪接位点</p></li><li><p data-tool="mdnice编辑器">in_frame_ins：框内插入</p></li><li><p data-tool="mdnice编辑器">in_frame_del：框内缺失</p></li><li><p data-tool="mdnice编辑器">translation_start_site:转录起始位点</p></li><li><p data-tool="mdnice编辑器">nonstop_mutation：终止密码子突变</p></li></ul><p data-tool="mdnice编辑器">但我读入的文件有稍许不一样，所以需要人工来设置</p><pre data-tool="mdnice编辑器"><span></span><code>&gt; table(maf_res<span>$Variant_Classification</span> )<br><br>      cds-del       cds-ins    frameshift      missense         ncRNA      nonsense <br>           24             5            43           712             2            45 <br>         span      splice-3      splice-5     stop-loss stop-retained <br>            4             6            13             1             1<br>      <br></code></pre><h3 data-tool="mdnice编辑器"><span>2. 重复位点 removeDuplicatedVariants</span></h3><p data-tool="mdnice编辑器">这次能顺利读入了</p><pre data-tool="mdnice编辑器"><span></span><code>dim(maf_res) <span># [1] 856  10</span><br><br>maf = read.maf(maf_res,vc_nonSyn=c(<span>"cds-del"</span>,<span>"cds-ins"</span>,<span>"frameshift"</span>,<span>"missense"</span>,<br>+                            <span>"ncRNA"</span>,<span>"nonsense"</span> ,<span>"span"</span>,<span>"splice-3"</span> ,<br>+                            <span>"splice-5"</span>,<span>"stop-loss"</span>,<span>"stop-retained"</span>))<br>-Validating<br>--Removed 803 duplicated variants<br>--Non MAF specific values <span>in</span> Variant_Classification column:<br>  missense<br>  cds-del<br>  splice-3<br>  frameshift<br>  nonsense<br>  ncRNA<br>--Non MAF specific values <span>in</span> Variant_Type column:<br>  snv<br>  indel<br>-Summarizing<br>-Processing clinical data<br>--Missing clinical data<br>-Finished <span>in</span> 0.084s elapsed (0.444s cpu)<br><br>&gt; maf@variants.per.sample[1:3]<br>   Tumor_Sample_Barcode Variants<br>1:                    1        1<br>2:                    2        1<br>3:                    3        1<br></code></pre><p data-tool="mdnice编辑器">但图很奇怪，没可能这样吧，突变都没剩几个,过程第二行显示过滤了98%的位点<code>--Removed 803 duplicated variants</code>。检查下样本突变数目<code>maf@variants.per.sample</code>,一个样本只有一个突变类型，那肯定是错的了！</p><figure data-tool="mdnice编辑器"><img data-ratio="0.4364123159303882" data-src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc7S7E87Jzy6Uady20IRhbft6AypX1rpj35icf25XqXialrFcjPNsNcibrcwfQDaTuCREC8icPAntk949A/640?wx_fmt=png" data-type="png" data-w="1494" src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc7S7E87Jzy6Uady20IRhbft6AypX1rpj35icf25XqXialrFcjPNsNcibrcwfQDaTuCREC8icPAntk949A/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">想到肯定有参数调整 <code>?read.maf</code>,发现有个参数会默认删除同一个基因的多个转录本</p><pre data-tool="mdnice编辑器"><span></span><code>removeDuplicatedVariants <br>removes repeated variants <span>in</span> a particuar sample, mapped to multiple transcripts of same Gene. See Description. Default TRUE.<br></code></pre><p data-tool="mdnice编辑器">其实把<code>removeDuplicatedVariants=F</code>就可以了</p><p data-tool="mdnice编辑器">那究竟按什么规律删除呢，我又看了下github的原始代码，<code>read.maf</code>里面嵌套了<code>validateMaf</code></p><pre data-tool="mdnice编辑器"><span></span><code>validateMaf = <span>function</span>(maf, rdup = TRUE, isTCGA = isTCGA, chatty = TRUE){<br><br>  <span>#necessary fields.</span><br>  required.fields = c(<span>'Hugo_Symbol'</span>, <span>'Chromosome'</span>, <span>'Start_Position'</span>, <span>'End_Position'</span>, <span>'Reference_Allele'</span>, <span>'Tumor_Seq_Allele2'</span>,<br>                      <span>'Variant_Classification'</span>, <span>'Variant_Type'</span>, <span>'Tumor_Sample_Barcode'</span>)<br><br>  <span>#Change column names to standard names; i.e, camel case</span><br>  <span>for</span>(i <span>in</span> 1:length(required.fields)){<br>    colId = suppressWarnings(grep(pattern = paste(<span>"^"</span>,required.fields[i],<span>"$"</span>,sep=<span>""</span>), x = colnames(maf), ignore.case = TRUE))<br>    <span>if</span>(length(colId) &gt; 0){<br>      colnames(maf)[colId] = required.fields[i]<br>    }<br>  }<br><br>  missing.fileds = required.fields[!required.fields %<span>in</span>% colnames(maf)] <span>#check if any of them are missing</span><br><br>  <span>if</span>(length(missing.fileds) &gt; 0){<br>    missing.fileds = paste(missing.fileds[1], sep = <span>','</span>, collapse = <span>', '</span>)<br>    stop(paste(<span>'missing required fields from MAF:'</span>, missing.fileds)) <span>#stop if any of required.fields are missing</span><br>  }<br><br>  <span>#convert "-" to "." in "Tumor_Sample_Barcode" to avoid complexity in naming</span><br>  <span>#maf$Tumor_Sample_Barcode = gsub(pattern = '-', replacement = '.', x = as.character(maf$Tumor_Sample_Barcode))</span><br><br>  <span>if</span>(rdup){<br>    maf = maf[, variantId := paste(Chromosome, Start_Position, Tumor_Sample_Barcode, Reference_Allele, Tumor_Seq_Allele2, sep = <span>':'</span>)]<br>    <span>if</span>(nrow(maf[duplicated(variantId)]) &gt; 0){<br>      <span>if</span>(chatty){<br>        cat(<span>"--Removed"</span>,  nrow(maf[duplicated(variantId)]) ,<span>"duplicated variants\n"</span>)<br>      }<br>      maf = maf[!duplicated(variantId)]<br>    }<br>    maf[,variantId := NULL]<br>  }<br>  ......<br></code></pre><p data-tool="mdnice编辑器">可以看到后面的<code>maf = maf[, variantId := paste(Chromosome, Start_Position, Tumor_Sample_Barcode, Reference_Allele, Tumor_Seq_Allele2, sep = ':')]</code>，其实，是根据这几个变量筛选多个转录本，而我的表格这些信息都没有，都赋值了NA，所有把很多位点都删除掉了。但是按照正常流程上游后跑maftools，这个参数默认就可以，因为该有的全部信息都在</p><h2 data-tool="mdnice编辑器"><span></span>最终结果</h2><pre data-tool="mdnice编辑器"><span></span><code>maf = read.maf(maf_res,removeDuplicatedVariants=F,<br>               vc_nonSyn=c(<span>"cds-del"</span>,<span>"cds-ins"</span>,<span>"frameshift"</span>,<span>"missense"</span>,<br>                                       <span>"ncRNA"</span>,<span>"nonsense"</span> ,<span>"span"</span>,<span>"splice-3"</span> ,<br>                                       <span>"splice-5"</span>,<span>"stop-loss"</span>,<span>"stop-retained"</span>))<br>dim(maf_res)<br><span># 样本突变总数</span><br>maf@variants.per.sample[1:3]<br><span>#       Tumor_Sample_Barcode Variants</span><br><span># 1:                   36       60</span><br><span># 2:                    4       35</span><br><span># 3:                    6       32</span><br><br><br><span># https://github.com/PoisonAlien/maftools/blob/master/R/oncoplot.R</span><br><span># https://bioconductor.org/packages/release/bioc/vignettes/maftools/inst/doc/oncoplots.html</span><br><span>#Selected AML driver genes</span><br>aml_genes = c(<span>'CHD3'</span>,<span>'APC'</span>,<br>              <span>'TP53'</span>,<span>'PALB2'</span>,<span>'FANCA'</span>,<br>              <span>'TET2'</span>,<span>'DNMT3A'</span>,<span>'IDH2'</span>,<span>'ARID1A'</span>,<span>'ARID1B'</span>,<span>'MLL3'</span>,<br>              <span>'TYK2'</span>,<span>'STAT3'</span>,<br>              <span>'LRRK2'</span>,<span>'MAP3K1'</span>,<br>              <span>'PCLO'</span>,<span>'PIEZO1'</span>,<span>'FAT3'</span>,<span>'CSMD1'</span>,<span>'NSD1'</span>,<span>'MKI67'</span>,<br>              <span>'WDR90'</span>,<span>'MGA'</span>,<span>'CPS1'</span>,<span>'SPEN'</span>,<span>'ATP10B'</span>,<span>'ANKRD11'</span>,<br>              <span>'RELN'</span>,<span>'PLCG1'</span>,<br>              <span>'ALK'</span>,<span>'FLT4'</span>,<br>              <span>'RHOA'</span>,<span>'NOTCH1'</span>,<span>'NOTCH4'</span>)<br>              <br>pdf(<span>'oncoplot.pdf'</span>)<br><span>#突变全景图</span><br>oncoplot(maf,drawColBar=F,<br>         genes=aml_genes,<br>         borderCol = <span>'white'</span>,<br>         bgCol = <span>'gray90'</span>,<br>         altered = TRUE)<br>dev.off()         <br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.9006060606060606" data-src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc7S7E87Jzy6Uady20IRhbftWL8hcPpzFmygMs8o0kqPdibGKYNHllENll4AwBmr0vBVJ8MnR7GzQHw/640?wx_fmt=png" data-type="png" data-w="825" src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc7S7E87Jzy6Uady20IRhbftWL8hcPpzFmygMs8o0kqPdibGKYNHllENll4AwBmr0vBVJ8MnR7GzQHw/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">https://github.com/PoisonAlien/maftools/blob/e2ff232d71207cbef54001be69ffcf1f679820a5/R/read_maf_dt.Rhttps://github.com/PoisonAlien/maftools/blob/a62aa15ee561363615bf17d8d1fff3c9a7833d1e/R/validateMaf.R</p><p data-tool="mdnice编辑器"><span><em>解决办法的效率有待提高啊！</em></span></p><p data-tool="mdnice编辑器"><br></p><p data-tool="mdnice编辑器"><br></p><p data-tool="mdnice编辑器">参考：</p><p data-tool="mdnice编辑器">和豆豆一起跟着官网学maftoolshttps://mp.weixin.qq.com/s/KYrS1MW16Jb6dEz7J7U1Ow</p><p data-tool="mdnice编辑器">https://github.com/PoisonAlien/maftools/issues/149</p></section><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/EQHnbQBsl0ihVl4jAWqEeg",target="_blank" rel="noopener noreferrer">原文链接</a>
