---
title: "看着14个肿瘤相关通路活性就够了"
date: 2022-12-31T01:52:51Z
draft: ["false"]
tags: [
  "fetched",
  "生信技能树"
]
categories: ["Acdemic"]
---
看着14个肿瘤相关通路活性就够了 by 生信技能树
------
<div><p data-tool="mdnice编辑器">Title：PROGENy: Pathway RespOnsive GENes for activity inference; 通路反应基因活性推断工具</p><h1 data-tool="mdnice编辑器">1.引入</h1><p data-tool="mdnice编辑器">在上篇推送介绍的R包中，我们用到了PROGENy工具分析14个通路的活性，简单提了一下PROGENy的github 地址，但并没有做过多探索，在后续的学习中，发现PROGENy的学习文档写的也很……就不是很能懂。由此，本篇就PROGENy 进行简要探索。</p><h1 data-tool="mdnice编辑器">2. 介绍</h1><p data-tool="mdnice编辑器">PROGENy （Pathway RespOnsive GENes for activity inference）是2018年发表在Nature Communication的R包[1]。一般认为，基因表达与通路的信号活性相关，在以往的通路富集分析中，基本上都以此为依据，认为在某个通路中高表达的基因越多，则通路激活的可能性越大，然而，却忽略了翻译后修饰的影响，基于此，作者团队开发了PROGENy, 它可以使用公开可用的信号扰动实验<code>(perturbation experiments)</code>数据来推断出人类的样本通路响应过程中的共同核心基因。这些可与任何统计方法相结合，可用于从bulk RNAseq学中推断通路活性。在作者的Abstract中，其将PROGENy的功能主要总结为以下几点：</p><blockquote data-tool="mdnice编辑器"><p><strong>(i) Recover the effect of known driver mutations</strong></p><p><strong>(ii) Provide or improve strong markers for drug indications</strong></p><p><strong>(iii) Distinguish between oncogenic and tumor suppressor pathways for patient survival</strong></p></blockquote><p data-tool="mdnice编辑器">PROGENy计算的14种通路简要介绍如下：</p><blockquote data-tool="mdnice编辑器"><p><strong>Androgen</strong>: involved in the growth and development of the male reproductive organs.</p><p><strong>EGFR</strong>: regulates growth, survival, migration, apoptosis, proliferation, and differentiation in mammalian cells</p><p><strong>Estrogen</strong>: promotes the growth and development of the female reproductive organs.</p><p><strong>Hypoxia</strong>: promotes angiogenesis and metabolic reprogramming when O2 levels are low.</p><p><strong>JAK-STAT</strong>: involved in immunity, cell division, cell death, and tumor formation.</p><p><strong>MAPK</strong>: integrates external signals and promotes cell growth and proliferation.</p><p><strong>NFkB</strong>: regulates immune response, cytokine production and cell survival.</p><p><strong>p53</strong>: regulates cell cycle, apoptosis, DNA repair and tumor suppression.</p><p><strong>PI3K</strong>: promotes growth and proliferation.</p><p><strong>TGFb</strong>: involved in development, homeostasis, and repair of most tissues.</p><p><strong>TNFa</strong>: mediates haematopoiesis, immune surveillance, tumour regression and protection from infection.</p><p><strong>Trail</strong><span>: in</span><span>duces apoptosis.</span></p><p><strong>VEGF</strong>: mediates angiogenesis, vascular permeability, and cell migration.</p><p><strong>WNT</strong>: regulates organ morphogenesis during development and tissue repair.</p></blockquote><p data-tool="mdnice编辑器">在后续几年内，Christian H. Holland 等人不断拓展PROGENy的使用范围，2019年将PROGENy拓展至小鼠的研究[2]。2020年PROGENy可支持单细胞转录数据的分析[3]。目前PROGENy的原始文献引用量已达154。这些信息表明，PROGENy还是较为被研究者们可以的。</p><p data-tool="mdnice编辑器"><img data-ratio="0.18309859154929578" data-type="png" data-w="1065" title="图1  Web of science 检索页面" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibiczT3BTIILXTyR6VsNkndlL5AcwINGCibUD6GcscWdn10xeA3n9AUwmd205YxKS1UMLQcYVhk3J6g/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibiczT3BTIILXTyR6VsNkndlL5AcwINGCibUD6GcscWdn10xeA3n9AUwmd205YxKS1UMLQcYVhk3J6g/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"><span>图1  Web of science 检索页面</span></p><blockquote data-tool="mdnice编辑器"><p>[1] Schubert M, Klinger B, Klünemann M, Sieber A, Uhlitz F, Sauer S, Garnett MJ, Blüthgen N, Saez-Rodriguez J. 2018. Perturbation-response genes reveal signaling footprints in cancer gene expression. <em>Nature Communications</em>: 10.1038/s41467-017-02391-6</p><p>[2] Holland CH, Szalai B, Saez-Rodriguez J. Transfer of regulatory knowledge from human to mouse for functional genomics analysis. Biochim Biophys Acta Gene Regul Mech. 2020;1863(6):194431. doi:10.1016/j.bbagrm.2019.194431</p><p>[3] Holland, C.H., Tanevski, J., Perales-Patón, J. et al. Robustness and applicability of transcription factor and pathway analysis tools on single-cell RNA-seq data. Genome Biol 21, 36 (2020). https://doi.org/10.1186/s13059-020-1949-z</p></blockquote><h1 data-tool="mdnice编辑器">3. PROGENy安装及使用</h1><h2 data-tool="mdnice编辑器"><span></span><span>3.1 安装</span><span></span><span></span></h2><p data-tool="mdnice编辑器">截止到本文发布前，PROGENy存储在Bioconductor的版本已经是最新的v1.20版本，和github的版本是一致的。因此，可以通过Bioconductor进行安装，也可以通过github 安装。</p><pre data-tool="mdnice编辑器"><span></span><code><span>##通过Bioconductor安装</span><br><span>if</span> (!requireNamespace(<span>"BiocManager"</span>, quietly = <span>TRUE</span>))<br>    install.packages(<span>"BiocManager"</span>)<br><br>BiocManager::install(<span>"progeny"</span>)<br><br><span>## 通过github安装</span><br>devtools::install_github(<span>"saezlab/progeny"</span>)<br></code></pre><h2 data-tool="mdnice编辑器"><span></span><span></span><span>3.2 使用</span><span></span><span></span></h2><p data-tool="mdnice编辑器">一般而言，作者开发R包都希望其他研究者来使用他的软件，因此都会配套一个说明文档或者说使用文档，告诉我们这个新东西应该怎么去使用。在github 的描述页面会给说明文档的链接，或者就直接写在描述页面。如果存储在bioconductor ，网页中也会有相关的帮助文档链接地址。PROGENy的更新后的操作文档写的好像不是能够很好的理解。<strong>以下是Bioconductor（https://www.bioconductor.org/packages/release/bioc/vignettes/progeny/inst/doc/progeny.html）及github（https://github.com/saezlab/progeny/blob/master/vignettes/progeny.Rmd）上帮助文档提供的操作代码</strong></p><pre data-tool="mdnice编辑器"><span></span><code><span>library</span>(progeny)<br><span>library</span>(ggplot2)<br><span>library</span>(dplyr)<br>model &lt;- progeny::model_human_full<br>head(model)<br><span>#&gt;       gene pathway      weight     p.value</span><br><span>#&gt; 1     RFC2    EGFR  1.47064662 0.001655275</span><br><span>#&gt; 2    ESRRA    EGFR  0.17858956 0.211837604</span><br><span>#&gt; 3   HNRNPK    EGFR  0.30669860 0.084560061</span><br><span>#&gt; 4     CBX6    EGFR -0.67550734 0.017641119</span><br><span>#&gt; 5   ASRGL1    EGFR -0.25232814 0.295429969</span><br><span>#&gt; 6 FLJ30679    MAPK -0.06047373 0.628461747</span><br><span># Get top 100 significant genes per pathway</span><br>model_100 &lt;- model %&gt;%<br>  group_by(pathway) %&gt;%<br>  slice_min(order_by = p.value, n = <span>200</span>)<br><br><span># Plot</span><br>ggplot(data=model_100, aes(x=weight, color=pathway, fill=pathway)) +<br>  geom_density() +<br>  theme(text = element_text(size=<span>12</span>)) +<br>  facet_wrap(~ pathway, scales=<span>'free'</span>) +<br>  xlab(<span>'scores'</span>) +<br>  ylab(<span>'densities'</span>) +<br>  theme_bw() +<br>  theme(legend.position = <span>"none"</span>)<br></code></pre><p data-tool="mdnice编辑器"><img data-ratio="0.5556844547563805" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibiczT3BTIILXTyR6VsNkndlSgkHml4T2rLbIgV3bfeqvGw4a4FZXkibRY1rwkpoicqE07mR0rzBacjA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="862" title="图2 帮助文档结果展示图" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibiczT3BTIILXTyR6VsNkndlSgkHml4T2rLbIgV3bfeqvGw4a4FZXkibRY1rwkpoicqE07mR0rzBacjA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"><span>图2 帮助文档结果展示图</span></p><p data-tool="mdnice编辑器">看这个文档，实际上看不大懂这个PROGENy到底应该怎么用的。<strong>Google检索发现以前也有人写过相关的操作流程，例如PROGENy 推断通路活性（https://www.jianshu.com/p/6e14680036ad）、单细胞之富集分析-6: PROGENY（https://www.jianshu.com/p/89491b688c80）。前者提供的操作链接已经不可用了， 后者的原始文档在目前的bioconductor/github 上也消失了。</strong></p><p data-tool="mdnice编辑器"><br></p><p data-tool="mdnice编辑器">而检索较早版本的<code>bioconductor</code> progeny发现，在v1.4版本的说明文档链接还能看到后者的部分操作，但单细胞流程的部分直接丢失了网页，至少现在找不到了，PROGENy pathway signatures（https://www.bioconductor.org/packages/3.8/bioc/vignettes/progeny/inst/doc/progeny.html），而使用v1.4版本的说明文档又发现biomaRt的ID转换因为网络问题，也不大好用。而<code>browseVignettes</code>函数也不能调用<code>PROGENy</code>的demo 文档。基于此，本篇主要基于v1.20 附带的函数帮助文档结合前期的一些操作流程进行介绍。</p><pre data-tool="mdnice编辑器"><span></span><code>browseVignettes(<span>"progeny"</span>)<br><span>#&gt; No vignettes found by browseVignettes("progeny")</span><br></code></pre><h2 data-tool="mdnice编辑器"><span></span><span></span><span>3.3  Bulk RNAseq数据分析</span><span></span><span></span></h2><h3 data-tool="mdnice编辑器">3.3.1 信号通路活性分析</h3><p data-tool="mdnice编辑器">相较于最初的版本，v1.20 已经附加了demo 的演示数据，因此，在本篇的演示文档中，我们只需要加载PROGENy包中附带的数据即可。如下：</p><pre data-tool="mdnice编辑器"><span></span><code><span>library</span>(progeny)<br><span>### 获取包内自带数据</span><br>gene_expression &lt;- as.matrix(read.csv(system.file(<span>"extdata"</span>,<br>                                                  <span>"human_input.csv"</span>, package = <span>"progeny"</span>), row.names = <span>1</span>))<br>head(gene_expression)<br><span>#&gt;         SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516 SRR1039517 SRR1039520 SRR1039521</span><br><span>#&gt;  ANKRD37    7.108502   7.318073   6.501028   6.603712   6.550726   6.767772   6.720333   6.862744</span><br><span>#&gt;  ANKZF1     9.397687   9.637948   9.134391   9.405355   9.345567   9.548367   9.185797   9.405191</span><br><span>#&gt;  ATP6V1G2   5.302948   5.302948   5.302948   5.302948   5.302948   5.302948   5.302948   5.302948</span><br><span>#&gt;  BMP4      11.543845  12.103400  11.630489  11.380581  10.901069  11.171775  11.205271  11.235885</span><br><span>#&gt;  BMPR2     11.541010  11.537694  11.695743  11.653518  11.769146  11.893831  11.738505  11.764539</span><br><span>#&gt;  BNIP3L    12.847510  12.889814  12.882546  13.116758  12.923894  13.563225  12.734816  13.108777</span><br></code></pre><p data-tool="mdnice编辑器">Bulk RNAseq 的数据只需提供归一化之后的表达矩阵，以行为基因名，列为样本名即可，</p><pre data-tool="mdnice编辑器"><span></span><code><span># 计算通路活性</span><br>pathways &lt;- progeny(gene_expression, scale=<span>TRUE</span>,<br>                    organism=<span>"Human"</span>, <span>#如为小鼠，填 "Mouse"</span><br>                    top = <span>100</span>, perm = <span>1</span>)<br>head(pathways)[<span>1</span>:<span>5</span>,<span>1</span>:<span>5</span>]<br><span>#&gt;               Androgen       EGFR   Estrogen    Hypoxia    JAK-STAT</span><br><span>#&gt;  SRR1039508 -1.2111870 -0.2213137  0.9492413  0.4971532  1.70999798</span><br><span>#&gt;  SRR1039509  0.5193128 -1.1128063 -1.0750417 -0.0351343 -1.37369841</span><br><span>#&gt;  SRR1039512 -0.7562443 -0.1584680  0.7905036 -0.7592593  0.78242868</span><br><span>#&gt;  SRR1039513  0.9281783 -0.7618999 -1.0702726 -0.9091296  0.54820602</span><br><span>#&gt;  SRR1039516 -0.7653795  0.0702440  1.1708203 -0.6315846 -0.02282687</span><br></code></pre><p data-tool="mdnice编辑器">由此，得到了每个样本的各通路活性值，后续参考zip文件附带code，进一步处理。</p><pre data-tool="mdnice编辑器"><span></span><code>nrow(pathways) <br><span>#&gt; 8</span><br>controls =rep(c(<span>TRUE</span>,<span>FALSE</span>),<span>4</span>)<br>ctl_mean = apply(pathways[controls,], <span>2</span>, mean)<br>ctl_sd = apply(pathways[controls,], <span>2</span>, sd)<br>pathways = t(apply(pathways, <span>1</span>, <span>function</span>(x) x - ctl_mean))<br>pathways = apply(pathways, <span>1</span>, <span>function</span>(x) x / ctl_sd)<br><span>library</span>(pheatmap)<br>myColor = colorRampPalette(c(<span>"Darkblue"</span>, <span>"white"</span>,<span>"red"</span>))(<span>100</span>)<br>pheatmap(t(pathways),fontsize=<span>14</span>, show_rownames = <span>F</span>,<br>         color=myColor, main = <span>"PROGENy"</span>, angle_col = <span>45</span>, treeheight_col = <span>0</span>,  <br>         border_color = <span>NA</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.5162412993039444" data-type="png" data-w="862" title="图3. Bulk RNAseq分组通路活性热图" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibiczT3BTIILXTyR6VsNkndloSLt2CP64wtRibycd14wKmvbzxyf1C2AT8ewv7AbqpfJvyicAgk7Y1Zw/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibiczT3BTIILXTyR6VsNkndloSLt2CP64wtRibycd14wKmvbzxyf1C2AT8ewv7AbqpfJvyicAgk7Y1Zw/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><figure data-tool="mdnice编辑器"><span>图3. Bulk RNAseq分组通路活性热图</span></figure><h3 data-tool="mdnice编辑器">3.3.2  两组间通路活性差异分析</h3><p data-tool="mdnice编辑器">在最初的版本的说明中，Bulk RNAseq的文档里面还提供了计算两组间通路活性差异的代码。<br>简单来说，按上述的demo文件共有8个样本，分成两组，一组为对照（<code>TRUE</code>），一组为实验组（<code>FALSE</code>），假如我要探究某个实验条件对样本的通路活性影响，我可以先计算各样本的通路活性，再利用下面的代码，找到差异最为明显的通路。</p><pre data-tool="mdnice编辑器"><span></span><code><span>library</span>(dplyr)<br>result = apply(pathways, <span>1</span>, <span>function</span>(x) {<br>    broom::tidy(lm(x ~ !controls)) %&gt;%<br>        filter(term == <span>"!controlsTRUE"</span>) %&gt;%<br>        dplyr::select(-term)<br>})<br>res_lm&lt;- mutate(bind_rows(result), pathway=names(result))<br>res_lm<br><span>#&gt;  # A tibble: 14 × 5</span><br><span>#&gt;     estimate std.error statistic    p.value pathway </span><br><span>#&gt;        &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;      &lt;dbl&gt; &lt;chr&gt;   </span><br><span>#&gt;   1   8.45       0.962    8.79   0.000120   Androgen</span><br><span>#&gt;   2  -0.210      2.58    -0.0814 0.938      EGFR    </span><br><span>#&gt;   3 -10.0        0.719  -13.9    0.00000854 Estrogen</span><br><span>#&gt;   4   1.18       1.25     0.944  0.382      Hypoxia </span><br><span>#&gt;   5  -1.45       0.700   -2.07   0.0844     JAK-STAT</span><br><span>#&gt;   6   1.28       1.68     0.760  0.476      MAPK    </span><br><span>#&gt;   7   0.221      0.689    0.321  0.759      NFkB    </span><br><span>#&gt;   8  -5.43       0.960   -5.66   0.00130    p53     </span><br><span>#&gt;   9  -2.58       3.22    -0.798  0.455      PI3K    </span><br><span>#&gt;  10   2.80       0.878    3.19   0.0188     TGFb    </span><br><span>#&gt;  11   0.336      0.665    0.506  0.631      TNFa    </span><br><span>#&gt;  12   0.840      0.810    1.04   0.340      Trail   </span><br><span>#&gt;  13  -0.0104     0.648   -0.0160 0.988      VEGF    </span><br><span>#&gt;  14  -0.964      0.609   -1.58   0.164      WNT</span><br></code></pre><p data-tool="mdnice编辑器">结果显示，在实验组中 p53 在处理后确实不如处理前活跃。源文档没有提供可视化的方法，但我们可以自行用ggplot2画图.</p><pre data-tool="mdnice编辑器"><span></span><code><span>library</span>(ggplot2)<br>ggplot(as.data.frame(res_lm),aes(x=pathway,y=statistic))+<br>  geom_point(aes(size=-log10(p.value),color=pathway))+<br>  theme_bw()+ylim(c(-<span>15</span>,<span>10</span>))+<br>  scale_color_manual(values = paletteer::paletteer_d(<span>'ggsci::category20_d3'</span>,n=<span>14</span>))+<br>  ggplot2::coord_flip()+theme(legend.position = <span>'none'</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.9270833333333334" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibiczT3BTIILXTyR6VsNkndl8BnBrGibMqb6CWnUjamW8v2ibXxjKzGapnDhujaicYaTxBGuJHNfSpqKg/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="480" title="图4. Bulk RNA-seq  通路活性分组差异分析后结果" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibiczT3BTIILXTyR6VsNkndl8BnBrGibMqb6CWnUjamW8v2ibXxjKzGapnDhujaicYaTxBGuJHNfSpqKg/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"><figcaption><span>图4. Bulk RNA-seq  通路活性分组差异分析后结果</span></figcaption></figure><h3 data-tool="mdnice编辑器">3.3.3 与药物作用相关性分析</h3><p data-tool="mdnice编辑器">在PROGENy以往的说明文档中，还提供了与药物作用的相关性分析，Code很简单，但需要一些数据自行下载。<br>本部分用到的数据存储在 the Genomics of Drug Sensitivity in Cancer（GDSC）数据库中，简单来说，通过分析细胞系基因表达及药物处理细胞后的表达谱变化，来推测药物的敏感性。这几个文件长什么样可以自己下载探索一下。</p><pre data-tool="mdnice编辑器"><span></span><code><span># set up a file cache so we download only once</span><br><span>library</span>(BiocFileCache)<br>bfc = BiocFileCache(<span>"."</span>)<br><span># gene expression and drug response</span><br>base = <span>"http://www.cancerrxgene.org/gdsc1000/GDSC1000_WebResources/Data/"</span><br>paths = bfcrpath(bfc, paste0(base, c(<span>"suppData/TableS4A.xlsx"</span>,<br>            <span>"preprocessed/Cell_line_RMA_proc_basalExp.txt.zip"</span>)))<br><span># load the downloaded files</span><br>drug_table &lt;- readxl::read_excel(paths[<span>1</span>], skip=<span>5</span>, na=<span>"NA"</span>)<br>drug_table &lt;- replace(drug_table, is.na(drug_table), <span>0</span>)<br>gene_table &lt;- readr::read_tsv(paths[<span>2</span>])<br><br><span># we need drug response with COSMIC IDs</span><br>drug_response &lt;- data.matrix(drug_table[,<span>3</span>:ncol(drug_table)])<br>rownames(drug_response) &lt;- drug_table[[<span>1</span>]]<br><br><span># we need genes in rows and samples in columns</span><br>gene_expr &lt;- data.matrix(gene_table[,<span>3</span>:ncol(gene_table)])<br>colnames(gene_expr) &lt;- sub(<span>"DATA."</span>, <span>""</span>, colnames(gene_expr), fixed=<span>TRUE</span>)<br>rownames(gene_expr) &lt;- gene_table$GENE_SYMBOLS<br></code></pre><p data-tool="mdnice编辑器">手动下载：<br>Processed gene expression matrix（http://www.cancerrxgene.org/gdsc1000/GDSC1000_WebResources/Data/preprocessed/Cell_line_RMA_proc_basalExp.txt.zip）<br>Drug sensitivities（http://www.cancerrxgene.org/gdsc1000/GDSC1000_WebResources/Data/suppData/TableS4A.xlsx）<br>随后进行PROGENy 运算：</p><pre data-tool="mdnice编辑器"><span></span><code><span>library</span>(progeny)<br>pathways &lt;- progeny(gene_expr,scale = <span>TRUE</span>, organism = <span>"Human"</span>, top = <span>100</span>,<br>    perm = <span>1</span>, verbose = <span>FALSE</span>)<br></code></pre><p data-tool="mdnice编辑器">可视化结果：</p><pre data-tool="mdnice编辑器"><span></span><code><span>library</span>(pheatmap)<br>myColor = colorRampPalette(c(<span>"Darkblue"</span>, <span>"white"</span>,<span>"red"</span>))(<span>100</span>)<br>pheatmap(pathways,fontsize=<span>14</span>, show_rownames = <span>FALSE</span>,<br>    color=myColor, main = <span>"PROGENy"</span>, angle_col = <span>45</span>, treeheight_col = <span>0</span>,  <br>    border_color = <span>NA</span>)<br></code></pre><p data-tool="mdnice编辑器"><img data-ratio="0.8694404591104734" data-type="png" data-w="697" title="图5  细胞系PROGENy 分析结果示意" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibiczT3BTIILXTyR6VsNkndljQ6UF4MYNrJTsF3g6I6bicVfKPulzf2LcicVDKuP7Y9YicyxX5PmTw7Ow/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibiczT3BTIILXTyR6VsNkndljQ6UF4MYNrJTsF3g6I6bicVfKPulzf2LcicVDKuP7Y9YicyxX5PmTw7Ow/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"><span>图5  细胞系PROGENy 分析结果示意</span></p><p data-tool="mdnice编辑器">假设这里我们要讨论<code>Trametinib</code>与<code>MAPK</code> 通路活性的相关性，我们可以先找到匹配的使用<code>Trametinib</code>处理的细胞及其对照组，再找到<code>MAPK</code>通路，随后进行一个相关性分析。</p><pre data-tool="mdnice编辑器"><span></span><code>cell_lines = intersect(rownames(pathways), rownames(drug_response))<br>trametinib = drug_response[cell_lines, <span>"Trametinib"</span>]<br><br>mapk = pathways[cell_lines, <span>"MAPK"</span>]<br><br>associations = lm(trametinib ~ mapk)<br>summary(associations)<br><span>#&gt; Call:</span><br><span>#&gt; lm(formula = trametinib ~ mapk)</span><br><br><span>#&gt; Residuals:</span><br><span>#&gt;    Min     1Q Median     3Q    Max </span><br><span>#&gt; -6.548 -1.255  0.338  1.338  6.819 </span><br><span>#&gt; </span><br><span>#&gt; Coefficients:</span><br><span>#&gt;             Estimate Std. Error t value Pr(&gt;|t|)    </span><br><span>#&gt; (Intercept) -0.94703    0.06442  -14.70   &lt;2e-16 ***</span><br><span>#&gt; mapk        -1.35978    0.06449  -21.09   &lt;2e-16 ***</span><br><span>#&gt; ---</span><br><span>#&gt; Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1</span><br><br><span>#&gt; Residual standard error: 1.998 on 960 degrees of freedom</span><br><span>#&gt; Multiple R-squared:  0.3165,	Adjusted R-squared:  0.3158 </span><br><span>#&gt; F-statistic: 444.6 on 1 and 960 DF,  p-value: &lt; 2.2e-16</span><br></code></pre><p data-tool="mdnice编辑器">根据结果来看，我们发现 MAPK 活性与对 Trametinib 的敏感性密切相关：Pr(&gt;|t|)远小于阈值0.05。</p><h3 data-tool="mdnice编辑器">4. Bulk RNAseq 小结</h3><p data-tool="mdnice编辑器">总体上而言，PROGENy 的操作方式很简单，只需要提供一个基因表达矩阵即可，其功能还是很强大的，提供的药物作用分析也可以用在数据挖掘相关的文章中，用以挖掘药物可能作用的靶点。在本篇简要介绍中，并没有过多描述他的算法原理，感兴趣的可以去看原文。对于当前的PROGENy bioconductor版本，初学者找到演示文档是比较困难的，可能由于其更新了新的文档之后将前次的文档覆盖，进而导致之前版本的文档。具体原因不做深究。下一篇推文继续简要介绍PROGENy在单细胞数据中的应用。</p><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><p data-tool="mdnice编辑器">接上篇，<a target="_blank" href="https://mp.weixin.qq.com/s?__biz=MzUzMTEwODk0Ng==&amp;mid=2247509319&amp;idx=1&amp;sn=a422c15104dc95505d143e307fca8bca&amp;scene=21#wechat_redirect" textvalue="PROGENy: Pathway RespOnsive GENes for activity inference（一）" linktype="text" imgurl="" imgdata="null" tab="innerlink" data-linktype="2">PROGENy: Pathway RespOnsive GENes for activity inference（一）</a></p><p data-tool="mdnice编辑器">Title：PROGENy: Pathway RespOnsive GENes for activity inference; 通路反应基因活性推断工具</p><p data-tool="mdnice编辑器">在上篇中，主要介绍了PROGENy在Bulk RNA seq数据中的应用，以及PROGENy如何联系药物与信号通路。这篇继续探讨PROGENy在单细胞数据中的应用。</p><h3 data-tool="mdnice编辑器">4.0  PROGENy在单细胞转录组中的应用</h3><p data-tool="mdnice编辑器">要运行PROGENy的单细胞版本，首先要获得演示数据。10X genomics 存储的pbmc3k的数据可以直接获得。</p><p data-tool="mdnice编辑器"><span>因此，我们先对该数据进行前期</span><code>Seurat</code><span> 处理 10x genomics pbmc3k存储的链接是</span>https://s3-us-west-2.amazonaws.com/10x.files/samples/cell/pbmc3k/pbmc3k_filtered_gene_bc_matrices.tar.gz<span>如果在服务器上使用，可以直接用wget下载：</span></p><pre data-tool="mdnice编辑器"><span></span><code>wget -b -c https://s3-us-west-2.amazonaws.com/10x.files/samples/cell/pbmc3k/pbmc3k_filtered_gene_bc_matrices.tar.gz<br></code></pre><p data-tool="mdnice编辑器">而后进行Seurat流程：</p><pre data-tool="mdnice编辑器"><span></span><code><span>library</span>(Seurat)<br><span>library</span>(ggplot2)<br><span>library</span>(tidyr)<br><span>library</span>(readr)<br><span>library</span>(pheatmap)<br><span>library</span>(tibble)<br>pbmc.data &lt;- <br>  Read10X(data.dir = <span>"./filtered_gene_bc_matrices/hg19/"</span>)<br><span>## Initialize the Seurat object with the raw (non-normalized data).</span><br>pbmc &lt;- <br>  CreateSeuratObject(counts = pbmc.data, project = <span>"pbmc3k"</span>, min.cells = <span>3</span>, <br>                     min.features = <span>200</span>)<br><span>## Identification of mithocondrial genes</span><br>pbmc[[<span>"percent.mt"</span>]] &lt;- PercentageFeatureSet(pbmc, pattern = <span>"^MT-"</span>)<br><br><span>## Filtering cells following standard QC criteria.</span><br>pbmc &lt;- subset(pbmc, subset = nFeature_RNA &gt; <span>200</span> &amp; nFeature_RNA &lt; <span>2500</span> &amp; <br>    percent.mt &lt; <span>5</span>)<br><br><span>## Normalizing the data</span><br>pbmc &lt;- NormalizeData(pbmc, normalization.method = <span>"LogNormalize"</span>, <br>    scale.factor = <span>10000</span>)<br><br>pbmc &lt;- NormalizeData(pbmc)<br><br><span>## Identify the 2000 most highly variable genes</span><br>pbmc &lt;- FindVariableFeatures(pbmc, selection.method = <span>"vst"</span>, nfeatures = <span>2000</span>)<br><br><span>## In addition we scale the data</span><br>all.genes &lt;- rownames(pbmc)<br>pbmc &lt;- ScaleData(pbmc, features = all.genes)<br>pbmc &lt;- RunPCA(pbmc, features = VariableFeatures(object = pbmc), verbose = <span>FALSE</span>)<br>pbmc &lt;- FindNeighbors(pbmc, dims = <span>1</span>:<span>10</span>, verbose = <span>FALSE</span>)<br>pbmc &lt;- FindClusters(pbmc, resolution = <span>0.5</span>, verbose = <span>FALSE</span>)<br>pbmc &lt;- RunUMAP(pbmc, dims = <span>1</span>:<span>10</span>,  umap.method = <span>'uwot'</span>, metric=<span>'cosine'</span>)<br>DimPlot(pbmc, reduction = <span>"umap"</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7234468937875751" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjobhUokvJVbrf2mGf5eibaaChESHcicQh8JLxE5eQubI9BM4mqlMVak8t4A/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="499" title="图1  pbmc3k umap" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjobhUokvJVbrf2mGf5eibaaChESHcicQh8JLxE5eQubI9BM4mqlMVak8t4A/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"><figcaption><span>图1  pbmc3k umap</span><br></figcaption><figcaption><span><br></span></figcaption></figure><pre data-tool="mdnice编辑器"><code><span>## Finding differentially expressed features (cluster biomarkers)</span><br>pbmc.markers &lt;- FindAllMarkers(pbmc, only.pos = <span>TRUE</span>, min.pct = <span>0.25</span>, <br>    logfc.threshold = <span>0.25</span>, verbose = <span>FALSE</span>)<br><br><span>## Assigning cell type identity to clusters</span><br>new.cluster.ids &lt;- c(<span>"Naive CD4 T"</span>, <span>"Memory CD4 T"</span>, <span>"CD14+ Mono"</span>, <span>"B"</span>, <span>"CD8 T"</span>, <br>    <span>"FCGR3A+ Mono"</span>, <span>"NK"</span>, <span>"DC"</span>, <span>"Platelet"</span>)<br>names(new.cluster.ids) &lt;- levels(pbmc)<br>pbmc &lt;- RenameIdents(pbmc, new.cluster.ids)<br><br><span>## We create a data frame with the specification of the cells that belong to </span><br><span>## each cluster to match with the Progeny scores.</span><br>CellsClusters &lt;- data.frame(Cell = names(Idents(pbmc)), <br>    CellType = as.character(Idents(pbmc)),<br>    stringsAsFactors = <span>FALSE</span>)<br>DimPlot(pbmc, reduction = <span>"umap"</span>, label = <span>TRUE</span>, pt.size = <span>0.5</span>) + NoLegend()<br></code></pre><p data-tool="mdnice编辑器"><img data-ratio="0.7874251497005988" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjobzjY5Z1ibkRTL5cm6ic4M1T5DIMh0GUj7libyYiaL3YwC7iavj8lwpysDUrQ/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="668" title="图2  pbmc3k  注释后 umap" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjobzjY5Z1ibkRTL5cm6ic4M1T5DIMh0GUj7libyYiaL3YwC7iavj8lwpysDUrQ/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"><span>图2  pbmc3k  注释后 umap</span></p></section><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><p data-tool="mdnice编辑器">在完成了细胞注释之后，可对pbmc3k 的样本进行PROGENy 分析。</p><pre data-tool="mdnice编辑器"><code><span>## We compute the Progeny activity scores and add them to our Seurat object</span><br><span>## as a new assay called Progeny. </span><br>pbmc &lt;- progeny(pbmc, scale=<span>FALSE</span>, organism=<span>"Human"</span>, top=<span>500</span>, perm=<span>1</span>, <br>    return_assay = <span>TRUE</span>)<br><br><span>## We can now directly apply Seurat functions in our Progeny scores. </span><br><span>## For instance, we scale the pathway activity scores. </span><br>pbmc &lt;- Seurat::ScaleData(pbmc, assay = <span>"progeny"</span>) <br><br><span>## We transform Progeny scores into a data frame to better handling the results</span><br>progeny_scores_df &lt;- <br>    as.data.frame(t(GetAssayData(pbmc, slot = <span>"scale.data"</span>, <br>        assay = <span>"progeny"</span>))) %&gt;%<br>    rownames_to_column(<span>"Cell"</span>) %&gt;%<br>    gather(Pathway, Activity, -Cell) <br><br><span>## We match Progeny scores with the cell clusters.</span><br>progeny_scores_df &lt;- inner_join(progeny_scores_df, CellsClusters)<br><br><span>## We summarize the Progeny scores by cellpopulation</span><br>summarized_progeny_scores &lt;- progeny_scores_df %&gt;% <br>    group_by(Pathway, CellType) %&gt;%<br>    summarise(avg = mean(Activity), std = sd(Activity))<br><br><span>## We prepare the data for the plot</span><br>summarized_progeny_scores_df &lt;- summarized_progeny_scores %&gt;%<br>    dplyr::select(-std) %&gt;%   <br>    spread(Pathway, avg) %&gt;%<br>    data.frame(row.names = <span>1</span>, check.names = <span>FALSE</span>, stringsAsFactors = <span>FALSE</span>) <br><br>paletteLength = <span>100</span><br>myColor = colorRampPalette(c(<span>"Darkblue"</span>, <span>"white"</span>,<span>"red"</span>))(paletteLength)<br><br>progenyBreaks = c(seq(min(summarized_progeny_scores_df), <span>0</span>, <br>                      length.out=ceiling(paletteLength/<span>2</span>) + <span>1</span>),<br>                  seq(max(summarized_progeny_scores_df)/paletteLength, <br>                      max(summarized_progeny_scores_df), <br>                      length.out=floor(paletteLength/<span>2</span>)))<br>progeny_hmap = pheatmap(t(summarized_progeny_scores_df[,-<span>1</span>]),fontsize=<span>14</span>, <br>                        fontsize_row = <span>10</span>, <br>                        color=myColor, breaks = progenyBreaks, <br>                        main = <span>"PROGENy (500)"</span>, angle_col = <span>45</span>,<br>                        treeheight_col = <span>0</span>,  border_color = <span>NA</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7874251497005988" data-type="png" data-w="668" title="图3  PROGENy 单细胞结果展示" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjobeQySdAdLF1PmZxUpHrZ2X9AShicAPOfNBsZbsrO2fzk9cdk6k47mpPQ/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjobeQySdAdLF1PmZxUpHrZ2X9AShicAPOfNBsZbsrO2fzk9cdk6k47mpPQ/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></figure><ne-p data-lake-id="u313b87b0" ne-alignment="center"><ne-card data-card-name="image" data-card-type="inline" data-event-boundary="card"><p><span>图3  PROGENy 单细胞结果展示</span></p><p><span><br></span></p><p><span><br></span></p></ne-card></ne-p><figure data-tool="mdnice编辑器"><figcaption><span>  5.  一些函数的使用  </span><br></figcaption></figure><p data-tool="mdnice编辑器">在v1.20版本里面，除progeny 函数之外，PROGENy包也存在其他函数，如下：<img data-ratio="0.3263157894736842" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjob93rn860gA1ObcQ926naVfeIcdPogE9W86j8FUicyTTAluANuXCgZR4Q/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="570" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjob93rn860gA1ObcQ926naVfeIcdPogE9W86j8FUicyTTAluANuXCgZR4Q/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></p><p data-tool="mdnice编辑器">但我们在使用过程中并没有太多的用到这些函数，progeny函数是这个包的核心，那么其他的函数是用来干嘛的呢？</p><p data-tool="mdnice编辑器">查看progeny 的源码，我们发现，getModel函数是从作者提供的Human 及Mouse 中取出目标通路的基因集,</p><pre data-tool="mdnice编辑器"><code>progeny = <span>function</span>(expr, scale=<span>TRUE</span>, organism=<span>"Human"</span>, top = <span>100</span>, perm = <span>1</span>, <br>    verbose = <span>FALSE</span>, z_scores = <span>FALSE</span>, get_nulldist = <span>FALSE</span>, assay_name = <span>"RNA"</span>, <br>    return_assay = <span>FALSE</span>, <span>...</span>) {<br>    UseMethod(<span>"progeny"</span>)<br>}<br>methods(<span>"progeny"</span>)<br><span>#&gt; [1] progeny.default*              progeny.ExpressionSet*        progeny.matrix*               progeny.Seurat*               progeny.SingleCellExperiment*</span><br><span>#&gt; see '?methods' for accessing help and source code</span><br>getAnywhere(progeny.matrix)<br><span>#&gt; A single object matching ‘progeny.matrix’ was found</span><br><span>#&gt; It was found in the following places</span><br><span>#&gt;   registered S3 method for progeny from namespace progeny</span><br><span>#&gt;   namespace:progeny </span><br></code></pre><p data-tool="mdnice编辑器">其他函数可自行在帮助文档查阅各自功能。</p><h3 data-tool="mdnice编辑器">附：当我们找不到说明文档怎么办？</h3><p data-tool="mdnice编辑器">在本软件学习过程中，遇到最明显的bug 就是作者在更新PROGENy之后，以往的帮助文档网页丢失了，这给学习使用R包的我们带来麻烦，这里简要分享我如何寻找这篇文章的帮助文档。</p><h4 data-tool="mdnice编辑器">Bioconductor</h4><p data-tool="mdnice编辑器">一般情况下，Bioconductor R包在这个部分会提供说明文档的链接，点开HTML/R_Script 或者查看PDF都可以看到帮助文档，在这里，HTML（https://www.bioconductor.org/packages/release/bioc/vignettes/progeny/inst/doc/progeny.html）并没有太多使用信息，只是介绍了getModel 是怎么用的，而PDF（https://www.bioconductor.org/packages/release/bioc/manuals/progeny/man/progeny.pdf）也只是提供了各个函数的介绍。也就是说，从这里并没有获取到很有价值的信息。<img data-ratio="0.3566333808844508" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjobPCujJc4nE7PnsCuK52ywibmh9tUQH2dD0joc8P9AFknO2X7DBr8uZDQ/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="701" title="图4. PROGENy Bioconductor 界面" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjobPCujJc4nE7PnsCuK52ywibmh9tUQH2dD0joc8P9AFknO2X7DBr8uZDQ/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></p><ne-p data-lake-id="ud84dd430" ne-alignment="center"><ne-card data-card-name="image" data-card-type="inline" data-event-boundary="card"><p><span>图4. PROGENy Bioconductor 界面</span></p></ne-card></ne-p><p data-tool="mdnice编辑器">参照Bioconductor的获取帮助文档的方式也失败了。</p><pre data-tool="mdnice编辑器"><code>browseVignettes(<span>"progeny"</span>)<br><span>#&gt; No vignettes found by browseVignettes("progeny")</span><br></code></pre><p data-tool="mdnice编辑器">那其实也可以检索一下旧版本的progeny 有没有配套的使用说明。<img data-ratio="0.5825734549138805" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjobEWEzARX1nra9wuFLibAExcibY5RJNxeF5XVQia2TrXnf0dyCXGcGWeIIQ/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="987" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjobEWEzARX1nra9wuFLibAExcibY5RJNxeF5XVQia2TrXnf0dyCXGcGWeIIQ/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"><span>但如果你也进行了这一步，你会发现，这个链接点开啥也没有，而在其他版本的HTML文件中，使用信息也不是那么全。</span></p><h4 data-tool="mdnice编辑器">Github</h4><p data-tool="mdnice编辑器">常见的说明文档也会提供在Github （https://github.com/saezlab/progeny）的首页，但在这里并没有。<img data-ratio="0.83" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjobibwsl5bQTdpE03L6u9JYpibPwwicMp8kuKOIxANS2tpBla5ZrMtqp9eIw/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="900" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjobibwsl5bQTdpE03L6u9JYpibPwwicMp8kuKOIxANS2tpBla5ZrMtqp9eIw/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></p><p data-tool="mdnice编辑器">因此，下一步查看他的各个文件夹中里面有没有帮助文档。一般来说vignettes 文件夹是储存帮助文档的，感兴趣的可以试着点开看一下<img data-ratio="0.6305525460455038" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjobAhBkFia2icEKsGjibjibicEto3TKKho6VwrslHSSZMpdKAmeUnrpicdX1aPw/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" data-type="png" data-w="923" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjobAhBkFia2icEKsGjibjibicEto3TKKho6VwrslHSSZMpdKAmeUnrpicdX1aPw/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></p><p data-tool="mdnice编辑器">我这里直接说结果，在docs 的下级目录中找到了这两个文件。如果你在github中打开，要是像我一样小白的话，可能看不懂它网页的源代码格式的数据，因此，就直接把他download下来，用本地浏览器打开，你会发现这就是一份操作说明。<img data-ratio="0.5484276729559748" data-type="png" data-w="795" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjobQmzwaEuQ4mvibY5whTVpia7PQyfwicvBChOiaHic4bNY9yicGELM2GUpdsaA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjobQmzwaEuQ4mvibY5whTVpia7PQyfwicvBChOiaHic4bNY9yicGELM2GUpdsaA/640?wx_fmt=png&amp;wxfrom=5&amp;wx_lazy=1&amp;wx_co=1"></p><h1 data-tool="mdnice编辑器">结语</h1><p data-tool="mdnice编辑器">总体而言，PROGENy 是个很优秀的R包，但也有其不足的地方，比如，它只涵盖了14条通路。上篇写完才发现，原来已经有其他公众号上已经推送过PROGENy，还引用了文献的例子来介绍PROGENy的使用，读者可以互相参考借鉴，由于来源一致，大部分内容可能会比较趋同。如果对R包原理感兴趣，还是要移步作者原文仔细详读才行哦。</p><pre spellcheck="false" lang="" cid="n96" mdtype="fences"><span role="presentation">[1] Schubert M, Klinger B, Klünemann M, Sieber A, Uhlitz F, Sauer S, Garnett MJ, Blüthgen N, Saez-Rodriguez J. 2018. Perturbation-response genes reveal signaling footprints in cancer gene expression. *Nature Communications*: [10.1038/s41467-017-02391-6](https://doi.org/10.1038/s41467-017-02391-6)</span><br><span role="presentation"></span><br><span role="presentation">[2] Holland CH, Szalai B, Saez-Rodriguez J. Transfer of regulatory knowledge from human to mouse for functional genomics analysis. Biochim Biophys Acta Gene Regul Mech. 2020;1863(6):194431. doi:10.1016/j.bbagrm.2019.194431 </span><br><span role="presentation"></span><br><span role="presentation">[3] Holland, C.H., Tanevski, J., Perales-Patón, J. et al. Robustness and applicability of transcription factor and pathway analysis tools on single-cell RNA-seq data. Genome Biol 21, 36 (2020). https://doi.org/10.1186/s13059-020-1949-z</span></pre></section><p><br></p><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/FgdeJcC2i1-nnNTxF11TfQ",target="_blank" rel="noopener noreferrer">原文链接</a>
