---
title: "肝癌患者 snRNA-seq 和 scRNA-seq 测序数据是否有区别？"
date: 2022-10-13T04:47:00Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Acdemic"]
---
肝癌患者 snRNA-seq 和 scRNA-seq 测序数据是否有区别？ by 生信菜鸟团
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><blockquote data-tool="mdnice编辑器"><p>这周的推文是对GSE210679数据集进行复现，发现这个数据是由一个snRNA-seq和一个scRNA-seq测序数据组成。</p></blockquote><p data-tool="mdnice编辑器"><strong>文章：</strong>Comparison of single‑nucleus and single‑cell transcriptomes in hepatocellular carcinoma tissue.</p><figure data-tool="mdnice编辑器"><img data-ratio="0.33935018050541516" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los95RQLs9BnTia2kP40LpCDQJhdLnQYx8TRd1DkrNuoBEpzY06GZGmoNFOEJlBrhcJXMEb8DE1bEiamg/640?wx_fmt=png" data-type="png" data-w="1385" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los95RQLs9BnTia2kP40LpCDQJhdLnQYx8TRd1DkrNuoBEpzY06GZGmoNFOEJlBrhcJXMEb8DE1bEiamg/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器"><strong>文章摘要：</strong>单核 RNA 测序 (snRNA ‑seq) 是 一种用于分析细胞中基因表达的方法 分离是复杂的，例如在肝细胞癌中 (HCC) 组织。它构成了单细胞 RNA 的替代品测序（scRNA-seq）通过分析细胞核而不是整个细胞；但是，是否可以完全替代 HCC 中的 scRNA-seq 仍有待阐明。</p><p data-tool="mdnice编辑器">在目前的研究中， 在肿瘤组织中将 scRNA ‑seq 与 snRNA ‑seq 进行比较 使用 10X Genomics 从 HCC 患者中获得。Seurat 也被用来处理数据 并比较两个测序之间的差异识别不同细胞类型的方法。在目前的研究中， 14,349 个单核和 9,504 个单细胞的转录组 从上述HCC组织中获得。一共 21 个离散细胞簇，包括肝细胞、内皮细胞 细胞、成纤维细胞、B细胞、T细胞、自然杀伤细胞和巨噬细胞被识别。值得注意的是，<strong>使用 snRNA ‑seq 检测到大量的肝细胞</strong>，在<strong>scRNA‑seq 检测到增加的免疫细胞</strong>。</p><p data-tool="mdnice编辑器">本研究的结果提供了一个 以单细胞分辨率显示人类 HCC 的综合图像。此外，本研究的结果进一步表明， 在某些情况下，snRNA ‑seq 可能足以替代 scRNA ‑seq ，snRNA ‑seq 在肝细胞中的表现水平有所提高测序。结合使用两种测序方法 可能有助于细胞间相互作用的研究。</p><p data-tool="mdnice编辑器"><strong>数据集：</strong>GSE210679</p><figure data-tool="mdnice编辑器"><img data-ratio="0.11614173228346457" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los95RQLs9BnTia2kP40LpCDQJvfThug9oUY64YP6BYHPJVlryvkAEDjJzHBrQAsQ3LRB5Pc3oE2A21w/640?wx_fmt=png" data-type="png" data-w="1016" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los95RQLs9BnTia2kP40LpCDQJvfThug9oUY64YP6BYHPJVlryvkAEDjJzHBrQAsQ3LRB5Pc3oE2A21w/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">要复现的图：</p><figure data-tool="mdnice编辑器"><img data-ratio="0.4390735146022155" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los95RQLs9BnTia2kP40LpCDQJicBSE4hjCy0u9bIx41UlsvMYZzx0gT5bC2fBiaG6lGTQriadibXe3sH8QQ/640?wx_fmt=png" data-type="png" data-w="993" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los95RQLs9BnTia2kP40LpCDQJicBSE4hjCy0u9bIx41UlsvMYZzx0gT5bC2fBiaG6lGTQriadibXe3sH8QQ/640?wx_fmt=png"></figure><figure data-tool="mdnice编辑器"><img data-ratio="0.4294755877034358" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los95RQLs9BnTia2kP40LpCDQJ5rNskMibcHxmNld4qNMPckQXhyibG5BePRjhHJo6xZgoPF3eyoGvkIMw/640?wx_fmt=png" data-type="png" data-w="1106" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los95RQLs9BnTia2kP40LpCDQJ5rNskMibcHxmNld4qNMPckQXhyibG5BePRjhHJo6xZgoPF3eyoGvkIMw/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器"><strong>step1:导入QC后的数据以及降维分群</strong></p><p data-tool="mdnice编辑器">由于此次数据涉及到scRNA-seq和snRNA-seq测序数据，需要scRNA-seq进行QC后再与snRNA-seq数据进行整合。之前并没有了解过snRNA-seq测序数据，通过处理这组数据去了解了一下，发现该数据不用过滤线粒体核糖体。</p><pre data-tool="mdnice编辑器"><span></span><code>dir.create(<span>"2-harmony"</span>)<br>getwd()<br>setwd(<span>"2-harmony"</span>)<br><br>sce.all=readRDS(<span>"../1-QC/sce.all_qc.rds"</span>)<br>sce=sce.all.filt <br>sce<br>sce &lt;- NormalizeData(sce, <br>                         normalization.method = <span>"LogNormalize"</span>,<br>                         scale.factor = <span>1e4</span>) <br>sce &lt;- FindVariableFeatures(sce)<br>sce &lt;- ScaleData(sce)<br>sce &lt;- RunPCA(sce, features = VariableFeatures(object = sce))<br><br><span>library</span>(harmony)<br>seuratObj &lt;- RunHarmony(sce, <span>"orig.ident"</span>)<br>names(seuratObj@reductions)<br>seuratObj &lt;- RunUMAP(seuratObj,  dims = <span>1</span>:<span>15</span>, <br>                     reduction = <span>"harmony"</span>)<br>DimPlot(seuratObj,reduction = <span>"umap"</span>,label=<span>T</span> ) <br><br>sce=seuratObj<br>sce &lt;- FindNeighbors(sce, reduction = <span>"harmony"</span>,<br>                     dims = <span>1</span>:<span>15</span>) <br>sce.all=sce<br><br>sce.all=FindClusters(sce.all, <span>#graph.name = "CCA_snn", </span><br>                       resolution = <span>0.5</span>, algorithm = <span>1</span>)<br><span>#接下来分析，按照分辨率为0.5进行 </span><br>sel.clust = <span>"RNA_snn_res.0.5"</span><br>sce.all &lt;- SetIdent(sce.all, value = sel.clust)<br>table(sce.all@active.ident) <br>saveRDS(sce.all, <span>"sce.all_int.rds"</span>)<br>setwd(<span>'../'</span>)<br></code></pre><p data-tool="mdnice编辑器"><strong>step2: marker gene</strong></p><pre data-tool="mdnice编辑器"><span></span><code><span>library</span>(ggplot2) <br>genes_to_check = c(<span>'PTPRC'</span>, <span>'CD3D'</span>, <span>'CD3E'</span>, <span>'CD4'</span>,<span>'CD8A'</span>,<span>'CD19'</span>, <span>'CD79A'</span>, <span>'MS4A1'</span> ,<br>                   <span>'IGHG1'</span>, <span>'MZB1'</span>, <span>'SDC1'</span>,<br>                   <span>'CD68'</span>, <span>'CD163'</span>, <span>'CD14'</span>, <br>                   <span>'TPSAB1'</span> , <span>'TPSB2'</span>,  <span># mast cells,</span><br>                   <span>'MKI67'</span>,<span>'TOP2A'</span>,<span>'KLRC1'</span>,<br>                   <span>'RCVRN'</span>,<span>'FPR1'</span> , <span>'ITGAM'</span> ,<br>                   <span>'FGF7'</span>,<span>'MME'</span>, <span>'ACTA2'</span>,<br>                   <span>'PECAM1'</span>, <span>'VWF'</span>,    <br>                   <span>'KLRB1'</span>,<span>'NCR1'</span>, <span># NK </span><br>                   <span>'EPCAM'</span> , <span>'KRT19'</span>, <span>'PROM1'</span>, <span>'ALDH1A1'</span>,<br>                   <span>'MKI67'</span> ,<span>'TOP2A'</span> )<br><span>library</span>(stringr)  <br>genes_to_check=str_to_title(unique(genes_to_check))<br>genes_to_check<br>p_all_markers &lt;- DotPlot(sce.all, features = genes_to_check,<br>                         assay=<span>'RNA'</span>  )  + coord_flip()<br><br>p_all_markers<br>ggsave(plot=p_all_markers, filename=<span>"check_all_marker_by_seurat_cluster.pdf"</span>)<br><br>p_all_markers<br>p_umap=DimPlot(sce.all, reduction = <span>"umap"</span>, label = <span>T</span>,label.box = <span>T</span>) <br><span>library</span>(patchwork)<br>p_all_markers+p_umap<br>ggsave(<span>'markers_umap.pdf'</span>,width = <span>15</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.5003261578604045" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los95RQLs9BnTia2kP40LpCDQJUGoZCnyKbzCJAg28y3AVtgmZfg2b2I6rmv2VMzZpibFnlAMj56SgQWw/640?wx_fmt=png" data-type="png" data-w="1533" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los95RQLs9BnTia2kP40LpCDQJUGoZCnyKbzCJAg28y3AVtgmZfg2b2I6rmv2VMzZpibFnlAMj56SgQWw/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器"><strong>step3:细胞亚群生物学命名</strong></p><pre data-tool="mdnice编辑器"><span></span><code> table(sce.all@meta.data$RNA_snn_res.0.5)<br>  <span>#细胞分群命名</span><br>  celltype=data.frame(ClusterID=<span>0</span>:<span>18</span>,<br>                      celltype= <span>0</span>:<span>18</span>) <br>  <span>#定义细胞亚群  </span><br>  celltype[celltype$ClusterID %<span>in</span>% c(<span>0</span>,<span>6</span>,<span>8</span>),<span>2</span>]=<span>'T cells'</span><br>  celltype[celltype$ClusterID %<span>in</span>% c(<span>15</span>),<span>2</span>]=<span>'CD20+ B cells'</span><br>  celltype[celltype$ClusterID %<span>in</span>% c(<span>16</span>),<span>2</span>]=<span>'plasma B cells'</span><br>  <br>  celltype[celltype$ClusterID %<span>in</span>% c(<span>7</span>,<span>13</span>),<span>2</span>]=<span>'myeloid'</span><br>  celltype[celltype$ClusterID %<span>in</span>% c(<span>1</span>,<span>2</span>,<span>3</span>,<span>5</span>,<span>14</span>,<span>18</span>),<span>2</span>]=<span>'Hepatocyte'</span><br>  <br>  celltype[celltype$ClusterID %<span>in</span>% c(<span>4</span>,<span>17</span>),<span>2</span>]=<span>'Endo'</span><br>  celltype[celltype$ClusterID %<span>in</span>% c(<span>12</span>),<span>2</span>]=<span>'NK'</span><br>  <br>  celltype[celltype$ClusterID %<span>in</span>% c(<span>9</span>,<span>10</span>),<span>2</span>]=<span>'fibo'</span><br>  celltype[celltype$ClusterID %<span>in</span>% c(<span>11</span>),<span>2</span>]=<span>'cycling'</span><br>  <br>  head(celltype)<br>  celltype<br>  table(celltype$celltype)<br>  sce.all@meta.data$celltype = <span>"NA"</span><br>  <span>for</span>(i <span>in</span> <span>1</span>:nrow(celltype)){<br>    sce.all@meta.data[which(sce.all@meta.data$RNA_snn_res.0.5 == celltype$ClusterID[i]),<span>'celltype'</span>] &lt;- celltype$celltype[i]}<br>  table(sce.all@meta.data$celltype)<br>  saveRDS(sce.all, <span>"sce.all_int_celltype.rds"</span>)<br><br><span>#papermarker</span><br>paper&lt;-<span>"ALB,SERPINA1,HNF4A,EPCAM,CD3D,CD3E,NKG7,CD68,CD14,CD163,<br>CD1C,CLEC4C,KIT, IGHG1,JCHAIN,CD79A,VWF,PECAM1,<br>FCGR2B,ACTA2,COL1A1,COL1A2"</span><br>papermarker&lt;-str_to_upper(trimws(strsplit(paper,<span>','</span>)[[<span>1</span>]]))<br>p &lt;- DotPlot(sce.all, features = unique(papermarker),<br>             assay=<span>'RNA'</span>  )  + coord_flip()<br><br>p<br>ggsave(plot=p, filename=<span>"check_paper_marker_by_seurat_cluster0.5.pdf"</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.567910447761194" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los95RQLs9BnTia2kP40LpCDQJ3wibJMteSqBRgGCmf3J3uIto8mSxNHSTOCV5dPoicITyuI76iaOiacxIyA/640?wx_fmt=png" data-type="png" data-w="1340" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los95RQLs9BnTia2kP40LpCDQJ3wibJMteSqBRgGCmf3J3uIto8mSxNHSTOCV5dPoicITyuI76iaOiacxIyA/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code>  DimPlot(sce.all, reduction = <span>"umap"</span>,split.by = <span>'group'</span>,<br>          group.by = <span>"celltype"</span>,label = <span>T</span><span>#,label.box = T</span><br>          ) <br>  ggsave(<span>'group_umap_celltype_group.pdf'</span>,width = <span>14</span>,height = <span>7</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.4905511811023622" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los95RQLs9BnTia2kP40LpCDQJAov538qRTCnz4rpvWYkNoZVHSTCDCriaa2wUrRTy5k6C6VyxLbOibOfw/640?wx_fmt=png" data-type="png" data-w="1270" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los95RQLs9BnTia2kP40LpCDQJAov538qRTCnz4rpvWYkNoZVHSTCDCriaa2wUrRTy5k6C6VyxLbOibOfw/640?wx_fmt=png"></figure><blockquote data-tool="mdnice编辑器"><p>比较符合总结的scRNA_seq和snRNA_seq的知识点。</p><p>参考：单细胞核转录组测序 - 简书 (jianshu.com)</p><p>虽然snRNA-seq能够获得更加全面完整的细胞类型，但是对于某些细胞类型的获得比例不如scRNA-seq，主要表现为免疫细胞。</p><p>从两分组细胞分布对比来看snRNA-seq测试数据中 T，B，NK的细胞数量都减少了；而Fibo, Hepatcyte,Endo细胞数量增多。</p></blockquote><p data-tool="mdnice编辑器"><strong>step4: 细胞亚群比例</strong></p><pre data-tool="mdnice编辑器"><span></span><code><span># 1.加载R包 ----</span><br>rm(list=ls()) <br><span>library</span>(ggplot2) <br><span>library</span>(cowplot) <br><span>library</span>(paletteer)  <br><span>library</span>(gplots)<br><span>library</span>(ggpubr)    <br><span>library</span>(ggsci) <br><span>library</span>(stringr)<br><span># 2.设置工作路径 ----</span><br><span>#在meta信息中添加分组信息</span><br>getwd() <br>dir.create(<span>"../4_group"</span>)<br>setwd(<span>"../4_group/"</span>)<br><span># 3.读取数据 ----</span><br>load(file = <span>'../3-cell/phe-by-markers.Rdata'</span>)<br>phe[<span>1</span>:<span>4</span>,<span>1</span>:<span>4</span>]<br>table(phe$orig.ident)<br>table(phe$group) <br>sce=readRDS(<span>"../3-cell/sce.all_int_celltype.rds"</span>)<br>table(sce@meta.data$group)<br><span># 4.可视化 ----</span><br><span>## 4.1 每种细胞类型中，分组所占比例 ----</span><br><span>library</span>(tidyr)<span># 使用的gather &amp; spread</span><br><span>library</span>(reshape2) <span># 使用的函数 melt &amp; dcast </span><br><span>library</span>(dplyr)<br><span>library</span>(ggplot2)<br>tb=table(phe$celltype,<br>         phe$group)<br>balloonplot(tb)<br>head(tb)<br>bar_data &lt;- as.data.frame(tb)<br>head(bar_data)<br><br>bar_per &lt;- bar_data %&gt;% <br>  group_by(Var1) %&gt;%<br>  mutate(sum(Freq)) %&gt;%<br>  mutate(percent = Freq / `sum(Freq)`)<br>head(bar_per) <br>write.csv(bar_per,file = <span>"celltype_by_group_percent.csv"</span>)<br><br>bar_per$group=bar_per$Var2<br><span>#可视化</span><br><span>#celltypebygroup</span><br><span>library</span>(ggalluvial)<br>ggplot(bar_per,<br>       aes(x = Var1, y=percent,stratum = group, alluvium = percent,<br>           fill = group, label = group)) +<br>  scale_fill_brewer(type = <span>"qual"</span>, palette = <span>"Set2"</span>) +<br>  geom_flow(stat = <span>"alluvium"</span>, lode.guidance = <span>"frontback"</span>,<br>            color = <span>"darkgray"</span>) +<br>  geom_stratum() +<br>  theme(legend.position = <span>"bottom"</span>) +<br>  <span>#图例位置</span><br>  ggtitle(<span>"by celltype"</span>)<br><br>ggsave(<span>"celltypebygroup_percent.pdf"</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7222739981360671" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los95RQLs9BnTia2kP40LpCDQJSAMCx26q4FIsC4ibKSY2esgd3uOeA8PiaRpWu6XDqHRLUY65mtTQ6MAQ/640?wx_fmt=png" data-type="png" data-w="1073" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los95RQLs9BnTia2kP40LpCDQJSAMCx26q4FIsC4ibKSY2esgd3uOeA8PiaRpWu6XDqHRLUY65mtTQ6MAQ/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code><span>## 每种降维分群中，各个分组所占比例 ----</span><br>bar_data &lt;- as.data.frame(table(phe$celltype,phe$orig.ident))<br><br>bar_per &lt;- bar_data %&gt;% <br>  group_by(Var1) %&gt;%<br>  mutate(sum(Freq)) %&gt;%<br>  mutate(percent = Freq / `sum(Freq)`)<br>head(bar_per)<br>write.csv(bar_per,file = <span>"celltype_by_orig.ident_percent.csv"</span>)<br><br>bar_per$group=bar_per$Var2<br><br><span>#在scales包中找了几个颜色</span><br><span>library</span>(<span>"scales"</span>)<br>pal_npg(<span>"nrc"</span>)(<span>10</span>)<br><span>#  [1] "#E64B35FF" "#4DBBD5FF" "#00A087FF" "#3C5488FF" "#F39B7FFF" "#8491B4FF"</span><br><span>#  [7] "#91D1C2FF" "#DC0000FF" "#7E6148FF" "#B09C85FF"</span><br> show_col(pal_npg(<span>"nrc"</span>)(<span>10</span>))<br><br>pal_npg(<span>"nrc"</span>,alpha = <span>0.6</span>)(<span>10</span>)<br> <span># [1] "#E64B3599" "#4DBBD599" "#00A08799" "#3C548899" "#F39B7F99" "#8491B499"</span><br> <span># [7] "#91D1C299" "#DC000099" "#7E614899" "#B09C8599"</span><br><br>bar_data &lt;- as.data.frame(table(phe$seurat_clusters,phe$group))<br>bar_per &lt;- bar_data %&gt;% <br>  group_by(Var1) %&gt;%<br>  mutate(sum(Freq)) %&gt;%<br>  mutate(percent = Freq / `sum(Freq)`)<br>head(bar_per)<br><br>bar_per$group=bar_per$Var2<br><span>#换了一个画法</span><br>mycolor = c(<span>"#E64B3599"</span>,<br>            <span>"#4DBBD5FF"</span>)<br>ggplot(bar_per, aes(x =Var1, y= percent, fill = group,<br>                    stratum=group, alluvium=group)) + <br>  geom_col(width = <span>0.5</span>, color=<span>'black'</span>)+<br>  geom_flow(width=<span>0.5</span>,alpha=<span>0.4</span>, knot.pos=<span>0.5</span>)+<br>  theme_classic() +<br>  labs(x=<span>'sample'</span>,y =<span>'Ratio'</span>,title = <span>"by cluster"</span><br>  )+<br>  coord_flip()+<br>  scale_fill_manual(values = mycolor)<span>#+  </span><br> <span># stat_compare_means(aes(group=group),</span><br>   <span>#                  symnum.args=list(cutpoints = c(0, 0.001, 0.01, 0.05, 1),</span><br>    <span>#                  symbols = c("***", "**", "*", "NS")),label = "p.signif",</span><br>     <span>#                label.y = 1.2,size = 5)</span><br><br>ggsave(<span>"clusterbygroup_percent.pdf"</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7960199004975125" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los95RQLs9BnTia2kP40LpCDQJDRGBUaicQWBqUk0Hhua56cq7Gibibcw8jVrzCvoq4mWicNrZLE8DAhjM9w/640?wx_fmt=png" data-type="png" data-w="1005" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los95RQLs9BnTia2kP40LpCDQJDRGBUaicQWBqUk0Hhua56cq7Gibibcw8jVrzCvoq4mWicNrZLE8DAhjM9w/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code><span>#如果对其进行统计检验</span><br>+stat_compare_means(aes(group=group),<br>                     symnum.args=list(cutpoints = c(<span>0</span>, <span>0.001</span>, <span>0.01</span>, <span>0.05</span>, <span>1</span>),<br>                     symbols = c(<span>"***"</span>, <span>"**"</span>, <span>"*"</span>, <span>"NS"</span>)),label = <span>"p.signif"</span>,<br>                    label.y = <span>1.2</span>,size = <span>5</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7859960552268245" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los95RQLs9BnTia2kP40LpCDQJ8cvBNv6m74GibhE7O69uEQqeXgfFU9W2otyoxT0TgoqrbwkyBhAHXrw/640?wx_fmt=png" data-type="png" data-w="1014" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los95RQLs9BnTia2kP40LpCDQJ8cvBNv6m74GibhE7O69uEQqeXgfFU9W2otyoxT0TgoqrbwkyBhAHXrw/640?wx_fmt=png"></figure><blockquote data-tool="mdnice编辑器"><p>基于上次有人提到对于单细胞细胞比例差异用不用做统计检验，在这里对本次复现的数据进行统计检验。</p><p>首先两分组数据需要判断是否符合正态分布和方差齐性检验，符合的话为参数检验要用t-test，不符合的话用Wilcoxon检验。</p><p>但是做出来的结果看起来并没有统计学意义。</p></blockquote><p data-tool="mdnice编辑器">参考：<a href="https://mp.weixin.qq.com/s?__biz=MzIyMzA2MTcwMg==&amp;mid=2650734365&amp;idx=1&amp;sn=f18633e789ad18efd3c82fe156e60af6&amp;scene=21#wechat_redirect" data-linktype="2">墙裂推荐！统计方法如何选以及全代码作图实现</a></p></section><p><br></p><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/lFEhw5oEOIp3jgpcfrNg_A",target="_blank" rel="noopener noreferrer">原文链接</a>
