---
title: "表观遗传分析3（ChIP-seq）"
date: 2022-10-22T11:36:13Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Acdemic"]
---
表观遗传分析3（ChIP-seq） by 生信菜鸟团
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><figure data-tool="mdnice编辑器"><img data-ratio="0.6806640625" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicUXogzOC9bq7ZwWnl4JPIADcQTtuNTuZl3LdcIxxp7WsfnOYb9iashpmAKs6Tza83pxvicyfjLFvcw/640?wx_fmt=png" data-type="png" data-w="1024" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicUXogzOC9bq7ZwWnl4JPIADcQTtuNTuZl3LdcIxxp7WsfnOYb9iashpmAKs6Tza83pxvicyfjLFvcw/640?wx_fmt=png"></figure><h1 data-tool="mdnice编辑器"><span></span><strong>表观遗传分析3（ChIP-seq）</strong></h1><h2 data-tool="mdnice编辑器"><span></span>染色质免疫共沉淀（ChIP）简介</h2><p data-tool="mdnice编辑器">我们知道真核生物的基因组DNA以染色质的形式存在。真核生物DNA上基因表达是一系列与DNA互作的蛋白质介导的，研究蛋白质与DNA在染色质环境下的相互作用是阐明核生物DNA上基因表达机制的基本途径。染色质免疫共沉淀（chromatin immunoprecipitation assay, ChIP）是目前研究体内DNA与蛋白质相互作用的一种常用的实验方法。它的基本原理是首先固定在活细胞状态下蛋白质－DNA复合物，并将其随机切断为一定长度范围内的染色质小片段，然后通过免疫学方法沉淀此复合体，特异性地富集目的蛋白结合的DNA片段，通过对目的片断的纯化与检测，从而获得蛋白质与DNA相互作用的信息。该技术主要应用于：组蛋白的修饰研究；染色质表观遗传修饰研究；转录调控分析；药物开发研究；有丝分裂研究；DNA损伤与凋亡分析等。</p><p data-tool="mdnice编辑器">ChIP常常与其他技术结合使用：</p><ul data-tool="mdnice编辑器"><li><section>ChIP-seq：和测序技术结合使用，从而获得全基因组范围内与组蛋白、转录因子等相互作用的DNA区段信息。</section></li><li><section>ChIP-chip：和芯片技术结合使用，实现高通量的筛选，在DNA与蛋白质分离后，以所获得DNA片段为探针通过芯片去筛选靶基因。</section></li></ul><h2 data-tool="mdnice编辑器"><span></span>染色质免疫共沉淀测序（ChIP-seq）简介</h2><p data-tool="mdnice编辑器">染色质免疫共沉淀测序（ChIP-seq）:将ChIP与第二代测序技术相结合的ChIP-Seq技术，能够高效地在全基因组范围内检测与组蛋白、转录因子等互作的DNA区段。ChIP-Seq的原理是：首先通过染色质免疫共沉淀技术（ChIP）特异性地富集目的蛋白结合的DNA片段，并对其进行纯化与文库构建；然后对富集得到的DNA片段进行高通量测序。通过将获得的数百万条序列标签精确定位到基因组上，从而获得全基因组范围内与组蛋白、转录因子等互作的DNA区段信息。</p><p data-tool="mdnice编辑器">实验流程:</p><ol data-tool="mdnice编辑器"><li><section>甲醛交联整个细胞系（组织），即将目标蛋白与染色质连结起来；</section></li><li><section>分离基因组DNA，并用超声波将其打断成一定长度的小片段；</section></li><li><section>添加与目标蛋白质特异的抗体，该抗体与目标蛋白形成免疫沉淀免疫结合复合体；</section></li><li><section>去交联，纯化DNA即得到染色质免疫沉淀的DNA样本，准备测序；</section></li><li><section>将准备好的样本进行深度测序。</section></li></ol><p data-tool="mdnice编辑器">生物信息分析流程</p><ol data-tool="mdnice编辑器"><li><section>将测序得到的短序列片段匹配到参考基因组序列上。</section></li><li><section>有一部分短序列不能匹配到参考基因组上，有可能是未知的基因组序列；另一部分是能够匹配到基因组上的短序列，通常要对这些短序列进行覆盖度计算。</section></li><li><section>从匹配到基因组上的短序列中进行富集区域的扫描。通常扫描到的富集区即被认为是蛋白质与DNA相互结合的区域（也有假阳性位点等的影响）。</section></li><li><section>对扫描到的富集区做深度分析，包括基因，注释，利用基因浏览器进行可视化浏览，研究与基因结构的关系等。</section></li></ol><section data-tool="mdnice编辑器"><section><figure><img data-ratio="1.1970684039087949" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicUXogzOC9bq7ZwWnl4JPIAqSsPTcsia9GbEqHx6m3aBtP6Ts2icdw2jRx2nOWB6a7lzK4EDV86AGEQ/640?wx_fmt=png" data-type="png" data-w="614" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicUXogzOC9bq7ZwWnl4JPIAqSsPTcsia9GbEqHx6m3aBtP6Ts2icdw2jRx2nOWB6a7lzK4EDV86AGEQ/640?wx_fmt=png"></figure></section><section><figure><img data-ratio="0.7313432835820896" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicUXogzOC9bq7ZwWnl4JPIAYK9deevOTAtAyibic9nP5kmLXBfsib93sY225ibHj3VibmfngCnohdBFB2g/640?wx_fmt=png" data-type="png" data-w="670" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicUXogzOC9bq7ZwWnl4JPIAYK9deevOTAtAyibic9nP5kmLXBfsib93sY225ibHj3VibmfngCnohdBFB2g/640?wx_fmt=png"></figure></section></section><p data-tool="mdnice编辑器">::::</p><h2 data-tool="mdnice编辑器"><span></span>ChIP-seq应用</h2><ol data-tool="mdnice编辑器"><li><section>研究组蛋白的修饰情况；</section></li><li><section>研究转录因子结合位点，解析该转录因子作用的通路信息；</section></li><li><section>核小体的定位图谱；</section></li><li><section>研究DNA的甲基化情况...</section></li></ol><h2 data-tool="mdnice编辑器"><span></span>ChIP-seq，分析流程</h2><ol data-tool="mdnice编辑器"><li><section>原始测序reads的质量控制;</section></li><li><section>序列比对;</section></li><li><section>peak calling;</section></li><li><section>富集峰注释;</section></li><li><section>motif分析等</section></li></ol><h2 data-tool="mdnice编辑器"><span></span>ChIP-seq数据分析实战</h2><h3 data-tool="mdnice编辑器"><span></span>数据来源的文章：<span></span></h3><p data-tool="mdnice编辑器">Polycomb associates genome-wide with a specific RNA polymerase II variant, and regulates metabolic genes in ES cells PMID: 22305566 DOI: 10.1016/j.stem.2011.12.017</p><h3 data-tool="mdnice编辑器"><span></span>0 安装必备软件<span></span></h3><pre data-tool="mdnice编辑器"><span></span><code>conda  create -n chipseq  python=2<br>conda info --envs<br><span>source</span> activate chipseq<br><br>conda install -y bwa -c bioconda <span>#比对工具</span><br>conda install -y trim-galore  samtools  -c bioconda <span># 质控和SAM，BAM文件工具</span><br>conda install -y deeptools homer  meme -c bioconda <span>#查看reads分布特征工具</span><br>conda install -y macs2 bowtie bowtie2 -c bioconda <span>#call peak工具</span><br><br>R<br>options(BioC_mirror=<span>"https://mirrors.tuna.tsinghua.edu.cn/bioconductor/"</span>)<br>options(<span>"repos"</span>=c(CRAN=<span>"https://mirrors.tuna.tsinghua.edu.cn/CRAN/"</span>))<br>install.packages(<span>"devtools"</span>)<br>library(devtools) <br>BiocManager::install<br>BiocManager::install(c(<span>'airway'</span>,<span>'DESeq2'</span>,<span>'edgeR'</span>,<span>'limma'</span>))<br>BiocManager::install(c(<span>'ChIPpeakAnno'</span>,<span>'ChIPseeker'</span>))<br>BiocManager::install(<span>'TxDb.Hsapiens.UCSC.hg19.knownGene'</span>,<br>                        ask=F,suppressUpdates=T)<br>BiocManager::install(<span>'TxDb.Hsapiens.UCSC.hg38.knownGene'</span>,<br>                        ask=F,suppressUpdates=T)<br>BiocManager::install(<span>'TxDb.Mmusculus.UCSC.mm10.knownGene'</span>,<br>                        ask=F,suppressUpdates=T)                             <br>library(TxDb.Hsapiens.UCSC.hg19.knownGene) <br>library(TxDb.Mmusculus.UCSC.mm10.knownGene) <br>library(TxDb.Hsapiens.UCSC.hg38.knownGene) <br>library(ChIPpeakAnno) <br>library(ChIPseeker) <br><br><br><span># 安装数据下载工具</span><br>mkdit SRAToolkit<br><span>cd</span> SRAToolkit<br>wget https://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/current/sratoolkit.current-centos_linux64.tar.gz<br>tar -zxvf sratoolkit.current-centos_linux64.tar.gz<br><span>echo</span> <span>'export export PATH=$PATH:/home/SRAToolkit/sratoolkit.2.10.9-centos_linux64/bin'</span> &gt;&gt; ~/.bashrc<br><span>source</span> ~/.bashrc<br></code></pre><h3 data-tool="mdnice编辑器"><span></span>1 下载数据<span></span></h3><p data-tool="mdnice编辑器">数据描述：</p><ul data-tool="mdnice编辑器"><li><section>样本数量：9个样本（每个样本有多个sra文件，多个fastq文件）</section></li><li><section>下载地址：https://www.ncbi.nlm.nih.gov/Traces/study/?acc=SRP009883</section></li><li><section>物种：鼠</section></li><li><section>测序仪：Illumina Genome Analyzer II</section></li><li><section>单端测序</section></li></ul><pre data-tool="mdnice编辑器"><span></span><code><span>## 注意组织好项目文件夹</span><br>mkdir -p  ~/project/chip-seq/<br><span>cd</span> ~/project/chip-seq/<br>mkdir {sra,raw,clean,align,peaks,motif,qc}<br><span>cd</span> sra <br><span>## 根据SraRunTable创建 sra.list 文件, 保存数据ID信息</span><br>cat&gt;config.sra<br>RNAPII_S5P_1    SRR391032<br>RNAPII_S5P_2    SRR391033<br>RNAPII_S2P_1    SRR391034<br>RNAPII_S7P_1    SRR391035<br>RNAPII_8WG16_1    SRR391036<br>RNAPII_8WG16_2    SRR391037<br>RNAPII_S2P_2    SRR391038<br>RNAPII_S2P_3    SRR391039<br>RNAPII_S7P_2    SRR391040<br>H2Aub1_1    SRR391041<br>H2Aub1_2    SRR391042<br>H3K36me3_1    SRR391043<br>H3K36me3_2    SRR391044<br>Control_1    SRR391045<br>Control_2    SRR391046<br>Ring1B_1    SRR391047<br>Ring1B_2    SRR391048<br>Ring1B_3    SRR391049<br>RNAPII_S5PRepeat_1    SRR391050<br><br><span>#创建sra.list文件</span><br>cat&gt;sra.list<br>SRR391032<br>SRR391033<br>SRR391034<br>SRR391035<br>SRR391036<br>SRR391037<br>SRR391038<br>SRR391039<br>SRR391040<br>SRR391041<br>SRR391042<br>SRR391043<br>SRR391044<br>SRR391045<br>SRR391046<br>SRR391047<br>SRR391048<br>SRR391049<br>SRR391050<br><br><span>#批量下载sra</span><br>conda activate chipseq<br>cat sra.list |<span>while</span> <span>read</span> id;<span>do</span> ( nohup  prefetch <span>$id</span> &amp; );<span>done</span><br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.6589147286821705" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicUXogzOC9bq7ZwWnl4JPIAohB2d4Yu5qW8Kzvb1OmibiaI8IgVadBzhqFicBmDCadjLf9jeibD4WO8hQ/640?wx_fmt=png" data-type="png" data-w="645" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicUXogzOC9bq7ZwWnl4JPIAohB2d4Yu5qW8Kzvb1OmibiaI8IgVadBzhqFicBmDCadjLf9jeibD4WO8hQ/640?wx_fmt=png"></figure><h3 data-tool="mdnice编辑器"><span></span>2 将sra文件转换成fastq文件<span></span></h3><pre data-tool="mdnice编辑器"><span></span><code><span>## 下面需要用循环</span><br><span>cd</span> ~/project/chip-seq<br>conda activate chip-seq<br><br><span>## 下面用到的 config.sra 文件，就是上面自行制作的。</span><br><br><span># 创建01_fastq-dump.sh脚本</span><br>vi 01_fastq-dump.sh<br><br>cat sra/config.sra |<span>while</span> <span>read</span> id;<br><span>do</span> <br>arr=(<span>$id</span>) <br>srr=<span>${arr[1]}</span> <br>sample=<span>${arr[0]}</span> <br>nohup fastq-dump sra/<span>$srr</span>.sra  --gzip --split-3  -A  <span>$sample</span> -O raw &amp;<br><span>done</span><br><span># 运行脚本</span><br>bash 01_fastq-dump.sh<br></code></pre><p data-tool="mdnice编辑器">fastq文件</p><figure data-tool="mdnice编辑器"><img data-ratio="0.8275862068965517" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicUXogzOC9bq7ZwWnl4JPIA78WEz0ib7icBsuhbb2vZmgK2Cd6whmcKHIlwzx1UxUIhMsmdxpYAA9bQ/640?wx_fmt=png" data-type="png" data-w="522" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicUXogzOC9bq7ZwWnl4JPIA78WEz0ib7icBsuhbb2vZmgK2Cd6whmcKHIlwzx1UxUIhMsmdxpYAA9bQ/640?wx_fmt=png"></figure><h3 data-tool="mdnice编辑器"><span></span>3 trim_galore测序数据的过滤<span></span></h3><p data-tool="mdnice编辑器">创建02_trim_galore.sh脚本</p><pre data-tool="mdnice编辑器"><span></span><code>vi 02_trim_galore.sh<br><span>cd</span> ~/project/chip-seq/<br>conda activate chipseq<br>ls ../raw/*gz | <span>while</span> <span>read</span> fq1;<br><span>do</span> <br>nohup trim_galore -q 25 --phred33 --length 25 -e 0.1 --stringency 4 -o ~/project/chip-seq/clean  <span>$fq1</span> &amp; <br><span>done</span> <br>sh 02_trim_galore.sh<br></code></pre><p data-tool="mdnice编辑器">过滤后fq文件<img data-ratio="0.9273504273504274" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicUXogzOC9bq7ZwWnl4JPIAUpTh35y186sicEcpY5mvNp6BY5KicUA0jQLAolGJhGicLFfnfTHKqVNVQ/640?wx_fmt=png" data-type="png" data-w="468" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicUXogzOC9bq7ZwWnl4JPIAUpTh35y186sicEcpY5mvNp6BY5KicUA0jQLAolGJhGicLFfnfTHKqVNVQ/640?wx_fmt=png"></p><h3 data-tool="mdnice编辑器"><span></span>4 数据质量的检测<span></span></h3><p data-tool="mdnice编辑器">创建03_fastqc_multiqc.sh脚本</p><pre data-tool="mdnice编辑器"><span></span><code>vi 03_fastqc_multiqc.sh<br><span>cd</span> qc<br>mkdir -p clean<br>fastqc -t 5  ../clean/*gz -o clean/<br>mkdir -p raw<br>fastqc -t 5  ../raw/*gz -o raw/<br>multiqc raw/*zip<br>multiqc clean/*zip<br>sh 03_fastqc_multiqc.sh<br></code></pre><section data-tool="mdnice编辑器"><section><p><strong>过滤前</strong></p><figure><img data-ratio="1.1088435374149659" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicUXogzOC9bq7ZwWnl4JPIAd7PLBKP1FkCQibanOVNtybnsO1DQDQLIE2u4NznApWLwzTjfzSW4YIg/640?wx_fmt=png" data-type="png" data-w="588" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicUXogzOC9bq7ZwWnl4JPIAd7PLBKP1FkCQibanOVNtybnsO1DQDQLIE2u4NznApWLwzTjfzSW4YIg/640?wx_fmt=png"></figure></section><section><p><strong>过滤后</strong></p><figure><img data-ratio="1.1495176848874598" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicUXogzOC9bq7ZwWnl4JPIAP04S16d1qTXe6W5zPFlBYLOCx0X84ichvnmtevMPObY2iaQryRakgjdw/640?wx_fmt=png" data-type="png" data-w="622" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicUXogzOC9bq7ZwWnl4JPIAP04S16d1qTXe6W5zPFlBYLOCx0X84ichvnmtevMPObY2iaQryRakgjdw/640?wx_fmt=png"></figure></section></section><h3 data-tool="mdnice编辑器"><span></span>5 比对<span></span></h3><p data-tool="mdnice编辑器">下载索引文件</p><pre data-tool="mdnice编辑器"><span></span><code><span># 小鼠bowtie2索引大小为3.2GB，可以直接下载bowtie2索引文件，代码如下：</span><br>mkdir referece &amp;&amp; <span>cd</span> reference<br>wget -c https://genome-idx.s3.amazonaws.com/bt/GRCm38.zip<br>unzip GRCm38.zip<br></code></pre><p data-tool="mdnice编辑器">创建04_align_bowtie2.sh脚本</p><pre data-tool="mdnice编辑器"><span></span><code>vi 04_align_bowtie2.sh<br><span>cd</span> ~/project/chip-seq/align<br>bowtie2_index=/reference/GRCm38/Mus_musculus.GRCm38 <span>## 索引文件</span><br>ls ../clean/*gz |<span>while</span> <span>read</span> id;<br><span>do</span> <br>file=$(basename <span>$id</span> )<br>sample=<span>${file%%.*}</span><br><span>echo</span> <span>$file</span> <span>$sample</span><br>bowtie2  -p 5  -x  <span>$bowtie2_index</span> -U  <span>$id</span> | samtools sort  -O bam  -@ 5 -o - &gt; <span>${sample}</span>.bam <br><span>done</span> <br>sh 04_align_bowtie2.sh<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7638640429338104" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicUXogzOC9bq7ZwWnl4JPIAV1GvCAic7GH5PEGlvPGjS6CXcsCSjbNQAnfUYR0WjUgJEWqt5gEvRVw/640?wx_fmt=png" data-type="png" data-w="559" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicUXogzOC9bq7ZwWnl4JPIAV1GvCAic7GH5PEGlvPGjS6CXcsCSjbNQAnfUYR0WjUgJEWqt5gEvRVw/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">对bam文件进行QC</p><pre data-tool="mdnice编辑器"><span></span><code><span>cd</span> ~/project/chip-seq/align<br>ls  *.bam  |xargs -i samtools index {} <br>ls  *.bam  | <span>while</span> <span>read</span> id ;<span>do</span> (nohup samtools flagstat <span>$id</span> &gt; $(basename <span>$id</span> <span>".bam"</span>).<span>stat</span> &amp; );<span>done</span><br></code></pre><p data-tool="mdnice编辑器">合并bam文件</p><p data-tool="mdnice编辑器">该数据一个样品分成了多个lane进行测序，所以在进行peaks calling前，需要把bam进行合并。</p><pre data-tool="mdnice编辑器"><span></span><code><span>cd</span> ~/project/chip-seq/ <br>mkdir mergeBam<br><span>source</span> activate chipseq<br><span>cd</span> ~/project/chip-seq/align<br>ls *.bam|sed <span>'s/_[0-9]_trimmed.bam//g'</span> |sort -u |<span>while</span> <span>read</span> id;<span>do</span> samtools merge ../mergeBam/<span>$id</span>.merge.bam <span>$id</span>*.bam ;<span>done</span><br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.36915077989601386" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicUXogzOC9bq7ZwWnl4JPIAiaKAkcLfYTnenCQIlWpSbhK60rH0HF83AeBm2mYG19BiboGQNf3lzTfA/640?wx_fmt=png" data-type="png" data-w="577" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicUXogzOC9bq7ZwWnl4JPIAiaKAkcLfYTnenCQIlWpSbhK60rH0HF83AeBm2mYG19BiboGQNf3lzTfA/640?wx_fmt=png"></figure><h3 data-tool="mdnice编辑器"><span></span>6 peak calling<span></span></h3><pre data-tool="mdnice编辑器"><span></span><code><span>cd</span>  ~/project/chip-seq/mergeBam <br>ls  *merge.bam |cut -d<span>"."</span> -f 1 |<span>while</span> <span>read</span> id;<br><span>do</span> <br>    <span>if</span> [ ! -s <span>${id}</span>_summits.bed ];<br>    <span>then</span> <br><span>echo</span> <span>$id</span> <br>nohup macs2 callpeak -c  Control.merge.bam -t <span>$id</span>.merge.bam -f BAM -B -g mm -n <span>$id</span> --outdir ../peaks  2&gt; <span>$id</span>.<span>log</span> &amp;  <br>    <span>fi</span> <br><span>done</span>  <br></code></pre><h3 data-tool="mdnice编辑器"><span></span>7 使用deeptool是进行可视化<span></span></h3><p data-tool="mdnice编辑器">首先把bam文件转为bw文件</p><pre data-tool="mdnice编辑器"><span></span><code><span>cd</span>  ~/project/chip-seq/mergeBam <br>ls  *.bam  |xargs -i samtools index {} <br>ls *.bam |<span>while</span> <span>read</span> id;<span>do</span><br>nohup bamCoverage --normalizeUsing CPM -b <span>$id</span> -o <span>${id%%.*}</span>.bw &amp; <br><span>done</span> <br></code></pre><p data-tool="mdnice编辑器">查看单一样本TSS附近信号强度：</p><pre data-tool="mdnice编辑器"><span></span><code>mkdir -p  ~/project/chip-seq/tss <br><span>cd</span>  ~/project/chip-seq/tss <br>computeMatrix reference-point  --referencePoint TSS  -p 15  \<br>-b 1000 -a 1000    \<br>-R /home/Ensemble/Annotation/Mouse/ucsc.refseq.bed  \<br>-S ~/project/chip-seq/mergeBam/H2Aub1.bw  \<br>--skipZeros  -o matrix1_test_TSS.gz  \<br>--outFileSortedRegions regions1_test_genes.bed<br><br><span>##     both plotHeatmap and plotProfile will use the output from   computeMatrix</span><br>plotHeatmap -m matrix1_test_TSS.gz  -out test_Heatmap.png<br>plotHeatmap -m matrix1_test_TSS.gz  -out test_Heatmap.pdf --plotFileFormat pdf  --dpi 720  <br>plotProfile -m matrix1_test_TSS.gz  -out test_Profile.png<br>plotProfile -m matrix1_test_TSS.gz  -out test_Profile.pdf --plotFileFormat pdf --perGroup --dpi 720 <br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.6289156626506024" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicUXogzOC9bq7ZwWnl4JPIARicgRfib6XLA8Rs9Mib1hHCUPChBwRaGAibZr9G3UIHxoVXfIqic353Qk1A/640?wx_fmt=png" data-type="png" data-w="830" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicUXogzOC9bq7ZwWnl4JPIARicgRfib6XLA8Rs9Mib1hHCUPChBwRaGAibZr9G3UIHxoVXfIqic353Qk1A/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">10K附近的信号强度</p><p data-tool="mdnice编辑器">创建 10k.sh</p><pre data-tool="mdnice编辑器"><span></span><code>bed=/home/Ensemble/Annotation/Mouse/ucsc.refseq.bed<br><span>for</span> id <span>in</span>  ~/project/chip-seq/mergeBam/*bw ;<br><span>do</span> <br><span>echo</span> <span>$id</span><br>file=$(basename <span>$id</span> )<br>sample=<span>${file%%.*}</span> <br><span>echo</span> <span>$sample</span>  <br><br>computeMatrix reference-point  --referencePoint TSS  -p 15  \<br>-b 10000 -a 10000    \<br>-R  <span>$bed</span> \<br>-S <span>$id</span>  \<br>--skipZeros  -o matrix1_<span>${sample}</span>_TSS_10K.gz  \<br>--outFileSortedRegions regions1_<span>${sample}</span>_TSS_10K.bed<br><span># 输出的gz为文件用于plotHeatmap, plotProfile</span><br><br><span>##     both plotHeatmap and plotProfile will use the output from   computeMatrix</span><br>plotHeatmap -m matrix1_<span>${sample}</span>_TSS_10K.gz  -out <span>${sample}</span>_Heatmap_10K.png<br>plotHeatmap -m matrix1_<span>${sample}</span>_TSS_10K.gz  -out <span>${sample}</span>_Heatmap_10K.pdf --plotFileFormat pdf  --dpi 720  <br>plotProfile -m matrix1_<span>${sample}</span>_TSS_10K.gz  -out <span>${sample}</span>_Profile_10K.png<br>plotProfile -m matrix1_<span>${sample}</span>_TSS_10K.gz  -out <span>${sample}</span>_Profile_10K.pdf --plotFileFormat pdf --perGroup --dpi 720 <br><span>done</span>  <br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>nohup bash 10k.sh 1&gt;10k.log &amp;<br></code></pre><p data-tool="mdnice编辑器">2K附近的信号强度</p><p data-tool="mdnice编辑器">创建 2k.sh</p><pre data-tool="mdnice编辑器"><span></span><code>bed=/home/Ensemble/Annotation/Mouse/ucsc.refseq.bed<br><span>for</span> id <span>in</span>  ~/project/chip-seq/mergeBam/*bw ;<br><span>do</span> <br><span>echo</span> <span>$id</span><br>file=$(basename <span>$id</span> )<br>sample=<span>${file%%.*}</span> <br><span>echo</span> <span>$sample</span>  <br><br>computeMatrix reference-point  --referencePoint TSS  -p 15  \<br>-b 2000 -a 2000    \<br>-R  <span>$bed</span> \<br>-S <span>$id</span>  \<br>--skipZeros  -o matrix1_<span>${sample}</span>_TSS_2K.gz  \<br>--outFileSortedRegions regions1_<span>${sample}</span>_TSS_2K.bed<br><br><span>##     both plotHeatmap and plotProfile will use the output from   computeMatrix</span><br>plotHeatmap -m matrix1_<span>${sample}</span>_TSS_2K.gz  -out <span>${sample}</span>_Heatmap_2K.png<br>plotHeatmap -m matrix1_<span>${sample}</span>_TSS_2K.gz  -out <span>${sample}</span>_Heatmap_2K.pdf --plotFileFormat pdf  --dpi 720  <br>plotProfile -m matrix1_<span>${sample}</span>_TSS_2K.gz  -out <span>${sample}</span>_Profile_2K.png<br>plotProfile -m matrix1_<span>${sample}</span>_TSS_2K.gz  -out <span>${sample}</span>_Profile_2K.pdf --plotFileFormat pdf --perGroup --dpi 720 <br><br><span>done</span>  <br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>nohup bash 2k.sh 1&gt;2k.log &amp;<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="1.1632270168855534" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicUXogzOC9bq7ZwWnl4JPIADG7VRPBbibtmgQBIzz9NUWWBvzBFK7IanIeYS93WUZvgDRV5Ax7b1iaw/640?wx_fmt=png" data-type="png" data-w="533" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicUXogzOC9bq7ZwWnl4JPIADG7VRPBbibtmgQBIzz9NUWWBvzBFK7IanIeYS93WUZvgDRV5Ax7b1iaw/640?wx_fmt=png"></figure><h3 data-tool="mdnice编辑器"><span></span>8 peaks注释<span></span></h3><p data-tool="mdnice编辑器">ChIPpeakAnno注释peaks区间分布</p><pre data-tool="mdnice编辑器"><span></span><code>R<br>bedPeaksFile  = <span>'RNAPII_8WG16_summits.bed'</span>; <br>require(ChIPseeker)<br>require(TxDb.Mmusculus.UCSC.mm10.knownGene)<br>txdb &lt;- TxDb.Mmusculus.UCSC.mm10.knownGene<br>require(clusterProfiler) <br>peak &lt;- readPeakFile( bedPeaksFile )  <br><span>#BiocManager::install('diffloop')</span><br>peak &lt;- diffloop::addchr(peak)<br>peakAnno &lt;- annotatePeak(peak, tssRegion=c(-3000, 3000), TxDb=txdb, annoDb=<span>"org.Mm.eg.db"</span>) <br>peakAnno_df &lt;- as.data.frame(peakAnno)<br>head(peakAnno_df)<br>write.csv(peakAnno_df, <span>"peakAnno_df.csv"</span>)<br>library(Cairo)<br>CairoPNG(file=<span>"AnnoPie.png"</span>,width=500,height=300)<br>plotAnnoPie(peakAnno)<br>dev.off()<br>CairoPNG(file=<span>"AnnoPie.png"</span>,width=500,height=300)<br>plotAnnoPie(peakAnno)<br>dev.off()<br><br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.6621621621621622" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicUXogzOC9bq7ZwWnl4JPIA4glwOOJrOIIOj3QH9wMp05Xlhx76lUXH0Nia4B1VPHu22BXeOM1fdlQ/640?wx_fmt=png" data-type="png" data-w="962" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicUXogzOC9bq7ZwWnl4JPIA4glwOOJrOIIOj3QH9wMp05Xlhx76lUXH0Nia4B1VPHu22BXeOM1fdlQ/640?wx_fmt=png"></figure><section data-tool="mdnice编辑器"><section><figure><img data-ratio="0.5213675213675214" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicUXogzOC9bq7ZwWnl4JPIAibicZ8HA4ibjaXmdUAIiae5N3Ttm9kcDibhrBXRqYWBbzibDkY5ItTgGQiczw/640?wx_fmt=png" data-type="png" data-w="468" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicUXogzOC9bq7ZwWnl4JPIAibicZ8HA4ibjaXmdUAIiae5N3Ttm9kcDibhrBXRqYWBbzibDkY5ItTgGQiczw/640?wx_fmt=png"></figure></section><section><figure><img data-ratio="0.6012269938650306" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicUXogzOC9bq7ZwWnl4JPIAPspnD89Onxgovukl0DCSxyc7dxBdcVfFS7cl3dNGwFVwrS5ibpicS5HQ/640?wx_fmt=png" data-type="png" data-w="489" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosicUXogzOC9bq7ZwWnl4JPIAPspnD89Onxgovukl0DCSxyc7dxBdcVfFS7cl3dNGwFVwrS5ibpicS5HQ/640?wx_fmt=png"></figure></section></section><h3 data-tool="mdnice编辑器"><span></span>9 寻找motif及注释<span></span></h3><p data-tool="mdnice编辑器">用homer软件来寻找motif</p><pre data-tool="mdnice编辑器"><span></span><code><span># 下载数据库</span><br>perl ~/miniconda3/envs/chipseq/share/homer-4.9.1-5/configureHomer.pl  -install mm10 <br></code></pre><p data-tool="mdnice编辑器">创建 runMotif.sh 脚本</p><pre data-tool="mdnice编辑器"><span></span><code><span>cd</span>  ~/project/chip-seq/motif<br><span>for</span> id <span>in</span> ~/project/chip-seq/peaks/*.bed;<br><span>do</span><br><span>echo</span> <span>$id</span><br>file=$(basename <span>$id</span> )<br>sample=<span>${file%%.*}</span> <br><span>echo</span> <span>$sample</span>  <br>awk <span>'{print $4"\t"$1"\t"$2"\t"$3"\t+"}'</span> <span>$id</span> &gt;homer_peaks.tmp  <br>findMotifsGenome.pl homer_peaks.tmp mm10 <span>${sample}</span>_motifDir -len 8,10,12<br>annotatePeaks.pl    homer_peaks.tmp mm10  1&gt;<span>${sample}</span>.peakAnn.xls 2&gt;<span>${sample}</span>.annLog.txt <br><span>done</span><br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>nohup bash runMotif.sh 1&gt;motif.log &amp;<br></code></pre><p data-tool="mdnice编辑器">参考</p><ol data-tool="mdnice编辑器"><li><section>Polycomb associates genome-wide with a specific RNA polymerase II variant, and regulates metabolic genes in ES cells PMID: 22305566 DOI: 10.1016/j.stem.2011.12.017</section></li><li><section>https://mp.weixin.qq.com/s/fRoRNwDzHNSttuRWqydzwg</section></li><li><section>https://baike.baidu.com/item/CHIP-seq</section></li></ol></section><p><br></p><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/UXVdCQQnEhWCd6X7mr2NoQ",target="_blank" rel="noopener noreferrer">原文链接</a>
