---
title: "跟着Nature学作图：R语言ggplot2散点栅格化能够减小输出pdf的文件大小"
date: 2023-01-11T12:42:04Z
draft: ["false"]
tags: [
  "fetched",
  "小明的数据分析笔记本"
]
categories: ["Acdemic"]
---
跟着Nature学作图：R语言ggplot2散点栅格化能够减小输出pdf的文件大小 by 小明的数据分析笔记本
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com" data-mpa-powered-by="yiban.io"><h2 data-tool="mdnice编辑器"><span></span><span>论文</span><span></span><span> </span></h2><blockquote data-tool="mdnice编辑器"><p>A saturated map of common genetic variants associated with human height</p></blockquote><p data-tool="mdnice编辑器">https://www.nature.com/articles/s41586-022-05275-y</p><p data-tool="mdnice编辑器"><code>s41586-022-05275-y.pdf</code></p><p data-tool="mdnice编辑器">代码没有公开，但是作图数据基本都公开了，争取把每个图都重复一遍</p><p data-tool="mdnice编辑器">今天的推文重复论文中的Figure1</p><h2 data-tool="mdnice编辑器"><span></span><span>代码</span><span></span><span> </span></h2><pre data-tool="mdnice编辑器"><span></span><code>setwd(<span>"data/20221014"</span>)<br>library(readxl)<br>fig1&lt;-read_excel(<span>"Figure1.xlsx"</span>)<br>colnames(fig1)<br><br>library(tidyverse)<br>library(stringr)<br><br>str_replace_all(<span>"[0,5e-100]"</span>,<span>"\\(|5e-|\\]|\\["</span>,<span>""</span>) %&gt;% <br>  str_split_fixed(<span>","</span>,n=2) %&gt;% <br>  as.data.frame() %&gt;% <br>  pull(V1) %&gt;% as.numeric()<br>str_replace_all(<span>"[0,5e-100]"</span>,<span>"\\(|5e-|\\]|\\["</span>,<span>""</span>) %&gt;% <br>  str_split_fixed(<span>","</span>,n=2) %&gt;% <br>  as.data.frame() %&gt;% <br>  pull(V2) %&gt;% as.numeric()<br>fig1 %&gt;% <br>  mutate(max_value=str_replace_all(`P-value Caregory`,<span>"\\(|5e-|\\]|\\["</span>,<span>""</span>) %&gt;% <br>           str_split_fixed(<span>","</span>,n=2) %&gt;% <br>           as.data.frame() %&gt;% <br>           pull(V1) %&gt;% as.numeric(),<br>         min_value=str_replace_all(`P-value Caregory`,<span>"\\(|5e-|\\]|\\["</span>,<span>""</span>) %&gt;% <br>           str_split_fixed(<span>","</span>,n=2) %&gt;% <br>           as.data.frame() %&gt;% <br>           pull(V2) %&gt;% as.numeric()) %&gt;% <br>  mutate(group=case_when(<br>    min_value == 100 &amp; max_value == 0  ~ <span>"group01"</span>,<br>    min_value == 50 &amp; max_value == 100 ~ <span>"group02"</span>,<br>    min_value == 20 &amp; max_value == 50 ~ <span>"group03"</span>,<br>    min_value == 10 &amp; max_value == 20 ~ <span>"group04"</span>,<br>    min_value == 8 &amp; max_value == 10 ~ <span>"group05"</span>,<br>  )) -&gt; new.fig1<br><br>table(new.fig1<span>$group</span>)<br><br>library(ggplot2)<br>library(ggh4x)<br>library(cowplot)<br><br>ggplot(data=new.fig1,<br>       aes(x=`Minor Allele Frequency`,<br>           y=`Join Effect of Minor Allele`,<br>           color=group))+<br>  geom_point( key_glyph = rectangle_key_glyph(color=color,<br>                                              fill=color,<br>                                              padding = margin(3, 3, 3, 3)))+<br>  scale_color_manual(values = c(<span>"group01"</span>=<span>"#ee82ee"</span>,<br>                                <span>"group02"</span>=<span>"#2e8b57"</span>,<br>                                <span>"group03"</span>=<span>"#1e90ff"</span>,<br>                                <span>"group04"</span>=<span>"#daa520"</span>,<br>                                <span>"group05"</span>=<span>"#cdc673"</span>),<br>                     name=<span>""</span>,<br>                     labels=c(<span>"group01"</span>=<span>"P &lt; 5 × 10–100 (672 SNPs)"</span>,<br>                              <span>"group02"</span>=<span>"5 × 10–50 &gt; P &gt; 5 × 10–100 (1,110 SNPs)"</span>,<br>                              <span>"group03"</span>=<span>"5 × 10–20 &gt; P &gt; 5 × 10–50 (3,513 SNPs)"</span>,<br>                              <span>"group04"</span>=<span>"5 × 10–10 &gt; P &gt; 5 × 10–20 (5,192 SNPs)"</span>,<br>                              <span>"group05"</span>=<span>"5 × 10–8 &gt; P &gt; 5 × 10–10 (1,624 SNPs)"</span>))+<br>  theme_bw()+<br>  theme(panel.grid = element_blank(),<br>        panel.border = element_blank(),<br>        axis.line = element_line(),<br>        legend.position = c(0.7,0.8))+<br>  scale_x_continuous(breaks = c(0.01,0.05,0.1,0.2,0.3,0.4,0.5),<br>                     labels = c(1,5,10,20,30,40,50))+<br>  scale_y_continuous(breaks = c(-0.3,-0.2,-0.1,0,0.1,0.2,0.3),<br>                     limits = c(-0.3,0.3))+<br>  guides(x=guide_axis_truncated(trunc_lower = 0.01,<br>                            trunc_upper = 0.5),<br>         y=guide_axis_truncated(trunc_lower = -0.3,<br>                                trunc_upper = 0.3))+<br>  labs(x=<span>"MAF (%) in cross-ancestry meta-analysis"</span>,<br>       y=<span>"Joint effect sizes (s.d.) of minor alleles\nin cross-ancestry meta-analysis"</span>)+<br>  geom_hline(yintercept = 0,color=<span>"gray"</span>)+<br>  geom_smooth(data = new.fig1 %&gt;% <br>                filter(group==<span>"group01"</span>) %&gt;%<br>                filter(`Join Effect of Minor Allele`&lt;0),<br>              aes(x=`Minor Allele Frequency`,<br>                  y=`Join Effect of Minor Allele`),<br>              method = <span>'loess'</span>,<br>              formula = <span>'y~x'</span>,<br>              se=FALSE,color=<span>"gray"</span>,<br>              show.legend = FALSE)+<br>  geom_smooth(data = new.fig1 %&gt;% <br>                filter(group==<span>"group01"</span>) %&gt;%<br>                filter(`Join Effect of Minor Allele`&gt;0),<br>              aes(x=`Minor Allele Frequency`,<br>                  y=`Join Effect of Minor Allele`),<br>              method = <span>'loess'</span>,<br>              formula = <span>'y~x'</span>,<br>              se=FALSE,color=<span>"gray"</span>,<br>              show.legend = FALSE)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7475806451612903" data-src="https://mmbiz.qpic.cn/mmbiz_png/t1wZDoUyFk7giaaPTVXYNdgibdGriaTAReysXV2HaJgJ5JQaMlgS8ichAGltGZBPWdsmcC6DU0CrZPv9EFpqdQSkMw/640?wx_fmt=png" data-type="png" data-w="1240" src="https://mmbiz.qpic.cn/mmbiz_png/t1wZDoUyFk7giaaPTVXYNdgibdGriaTAReysXV2HaJgJ5JQaMlgS8ichAGltGZBPWdsmcC6DU0CrZPv9EFpqdQSkMw/640?wx_fmt=png"><figcaption>image.png</figcaption></figure><p data-tool="mdnice编辑器">关于曲线不太清楚是用什么数据做的，这里直接自动添加拟合曲线</p><p data-tool="mdnice编辑器">图例里的文本上下标 出图后再编辑吧</p><p data-tool="mdnice编辑器">关于散点图今天还新学到一个知识点是：散点图的点如果非常多，如果输出pdf文件的话，pdf文件会非常大，比如GWAS里常用的曼哈顿图，这个pdf文件如果非常大后续如果想要编辑这个pdf文件会比较麻烦。</p><p data-tool="mdnice编辑器">关于如何解决这个问题，看到一个讨论群里有人讨论，他们提到一个办法是可以把散点栅格化 （栅格化是什么意思暂时不太明白）可以借助R包ggrastr</p><p data-tool="mdnice编辑器">对应的github主页是</p><p data-tool="mdnice编辑器">https://github.com/VPetukhov/ggrastr</p><p data-tool="mdnice编辑器">正好我们今天的推文内容是数据量比较多的散点图，我们可以按照这个做法试试，这里参考微信公众号推文 https://mp.weixin.qq.com/s/ou0cjD8dLMNaDLk588KSwQ</p><h2 data-tool="mdnice编辑器"><span></span><span>安装ggrastr这个R包</span><span></span><span> </span></h2><pre data-tool="mdnice编辑器"><span></span><code>install.packages(<span>'ggrastr'</span>)<br></code></pre><p data-tool="mdnice编辑器">如果要把点栅格化，只需要把对应的散点图函数<code>geom_point()</code>换成<code>geom_point_rast()</code></p><pre data-tool="mdnice编辑器"><span></span><code>library(ggrastr)<br>p2&lt;-ggplot(data=new.fig1,<br>           aes(x=`Minor Allele Frequency`,<br>               y=`Join Effect of Minor Allele`,<br>               color=group))+<br>  geom_point_rast( key_glyph = rectangle_key_glyph(color=color,<br>                                              fill=color,<br>                                              padding = margin(3, 3, 3, 3)),<br>                   size=0.1,<br>                   raster.dpi = getOption(<span>"ggrastr.default.dpi"</span>, 300))+<br>  scale_color_manual(values = c(<span>"group01"</span>=<span>"#ee82ee"</span>,<br>                                <span>"group02"</span>=<span>"#2e8b57"</span>,<br>                                <span>"group03"</span>=<span>"#1e90ff"</span>,<br>                                <span>"group04"</span>=<span>"#daa520"</span>,<br>                                <span>"group05"</span>=<span>"#cdc673"</span>),<br>                     name=<span>""</span>,<br>                     labels=c(<span>"group01"</span>=<span>"P &lt; 5 × 10–100 (672 SNPs)$)"</span>,<br>                              <span>"group02"</span>=<span>"5 × 10–50 &gt; P &gt; 5 × 10–100 (1,110 SNPs)"</span>,<br>                              <span>"group03"</span>=<span>"5 × 10–20 &gt; P &gt; 5 × 10–50 (3,513 SNPs)"</span>,<br>                              <span>"group04"</span>=<span>"5 × 10–10 &gt; P &gt; 5 × 10–20 (5,192 SNPs)"</span>,<br>                              <span>"group05"</span>=<span>"5 × 10–8 &gt; P &gt; 5 × 10–10 (1,624 SNPs)"</span>))+<br>  theme_bw()+<br>  theme(panel.grid = element_blank(),<br>        panel.border = element_blank(),<br>        axis.line = element_line(),<br>        legend.position = c(0.7,0.8))+<br>  scale_x_continuous(breaks = c(0.01,0.05,0.1,0.2,0.3,0.4,0.5),<br>                     labels = c(1,5,10,20,30,40,50))+<br>  scale_y_continuous(breaks = c(-0.3,-0.2,-0.1,0,0.1,0.2,0.3),<br>                     limits = c(-0.3,0.3))+<br>  guides(x=guide_axis_truncated(trunc_lower = 0.01,<br>                                trunc_upper = 0.5),<br>         y=guide_axis_truncated(trunc_lower = -0.3,<br>                                trunc_upper = 0.3))+<br>  labs(x=<span>"MAF (%) in cross-ancestry meta-analysis"</span>,<br>       y=<span>"Joint effect sizes (s.d.) of minor alleles\nin cross-ancestry meta-analysis"</span>)+<br>  geom_hline(yintercept = 0,color=<span>"gray"</span>)+<br>  geom_smooth(data = new.fig1 %&gt;% <br>                filter(group==<span>"group01"</span>) %&gt;%<br>                filter(`Join Effect of Minor Allele`&lt;0),<br>              aes(x=`Minor Allele Frequency`,<br>                  y=`Join Effect of Minor Allele`),<br>              method = <span>'loess'</span>,<br>              formula = <span>'y~x'</span>,<br>              se=FALSE,color=<span>"gray"</span>,<br>              show.legend = FALSE)+<br>  geom_smooth(data = new.fig1 %&gt;% <br>                filter(group==<span>"group01"</span>) %&gt;%<br>                filter(`Join Effect of Minor Allele`&gt;0),<br>              aes(x=`Minor Allele Frequency`,<br>                  y=`Join Effect of Minor Allele`),<br>              method = <span>'loess'</span>,<br>              formula = <span>'y~x'</span>,<br>              se=FALSE,color=<span>"gray"</span>,<br>              show.legend = FALSE)<br><br>pdf(<span>"p1.pdf"</span>,width = 6,height = 6)<br>p1<br>dev.off()<br><br><br>pdf(<span>"p2.pdf"</span>,width = 6,height = 6)<br>p2<br>dev.off()<br></code></pre><p data-tool="mdnice编辑器">输出的p2如果放大 点是会变模糊的</p><figure data-tool="mdnice编辑器"><img data-src="https://mmbiz.qpic.cn/mmbiz_png/t1wZDoUyFk7giaaPTVXYNdgibdGriaTARey2umwZwFXnmzlHWKYj5wgIDYXaJ1Dv3aDWpzYcVaWEnEZUk9j3PibqQA/640?wx_fmt=png" data-type="png" src="https://mmbiz.qpic.cn/mmbiz_png/t1wZDoUyFk7giaaPTVXYNdgibdGriaTARey2umwZwFXnmzlHWKYj5wgIDYXaJ1Dv3aDWpzYcVaWEnEZUk9j3PibqQA/640?wx_fmt=png"><figcaption>image.png</figcaption></figure><p data-tool="mdnice编辑器">两个文件的大小也不一样，栅格化之前是700k，栅格化之后只有200k</p><figure data-tool="mdnice编辑器"><img data-src="https://mmbiz.qpic.cn/mmbiz_png/t1wZDoUyFk7giaaPTVXYNdgibdGriaTAReyobA2W4Uo3F4cwwqvicuR4KrrMvWZibrGt8fIqQ8teACKrILO9eCHfeKA/640?wx_fmt=png" data-type="png" src="https://mmbiz.qpic.cn/mmbiz_png/t1wZDoUyFk7giaaPTVXYNdgibdGriaTAReyobA2W4Uo3F4cwwqvicuR4KrrMvWZibrGt8fIqQ8teACKrILO9eCHfeKA/640?wx_fmt=png"><figcaption>image.png</figcaption></figure><blockquote data-tool="mdnice编辑器"><p>示例数据和代码可以给公众号推文点赞，点击在看，最后留言获取</p></blockquote><p data-tool="mdnice编辑器">欢迎大家关注我的公众号</p><p data-tool="mdnice编辑器"><strong>小明的数据分析笔记本</strong></p><section><mp-common-profile data-pluginname="mpprofile" data-weui-theme="light" data-id="MzI3NzQ3MTcxMg==" data-headimg="http://mmbiz.qpic.cn/mmbiz_png/t1wZDoUyFk5t1sOnM0iabvBhnfIj5YpyqrMib0E1MGCd9ibcYxaOPZd0GWhQBDvK2BPEwsicQxd6y5MHLfphnwHnow/0?wx_fmt=png" data-nickname="小明的数据分析笔记本" data-alias="" data-signature="数据分析和数据可视化有意思的简单小例子~石榴研究生的笔记本" data-from="0" data-is_biz_ban="0"></mp-common-profile></section><p data-tool="mdnice编辑器"></p><blockquote data-tool="mdnice编辑器"><p>小明的数据分析笔记本 公众号 主要分享：1、R语言和python做数据分析和数据可视化的简单小例子；2、园艺植物相关转录组学、基因组学、群体遗传学文献阅读笔记；3、生物信息学入门学习资料及自己的学习笔记！</p></blockquote></section><p><br></p><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/F6-o9iVj-jk1LPttIswslA",target="_blank" rel="noopener noreferrer">原文链接</a>
