---
title: "跨越7年，同一个数据集的差异分析怎么不一样了"
date: 2022-11-08T08:42:13Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Acdemic"]
---
跨越7年，同一个数据集的差异分析怎么不一样了 by 生信菜鸟团
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><p data-tool="mdnice编辑器">今天我们来看看同一个数据集在2011年和2018年的处理，沧海桑田，我们还能得到同样的结果吗？</p><p data-tool="mdnice编辑器">先来看看2018年的这篇⬇</p><blockquote data-tool="mdnice编辑器"><ul><li><section>TITLE: Molecular mechanism of triple-negative breast cancer-associated BRCA1 and the identification of signaling pathways</section></li><li><section>DOI: 10.3892/ol.2019.9884</section></li><li><section>Objective: identify the effects of BRAC1 and to find the  signaling pathway(s) involved in the pathogenic mechanism of  triple-negative breast cancer</section></li><li><section>Overall design: differentially expressed genes (DEGs) from  <strong>GSE27447</strong> were distinguished by Significant Analysis of  Microarray. On the basis of the  <strong>SAM</strong> analysis, a total of <strong>132 upregulated and 198 downregulated DEGs</strong> were identified</section></li></ul></blockquote><blockquote data-tool="mdnice编辑器"><p>差异分析方法</p><ul><li><section>利用**微阵列显著性分析(SAM)**确定DEGs。</section></li><li><section>⬆一个并不常见的分析方法，但菜鸟团早有准备。[用samr包对芯片数据做差异分析 | 生信菜鸟团  http://www.bio-info-trainee.com/1608.html]</section></li><li><section>阈值设置：The criteria for  screening DEGs were <strong>|logFC|&gt;1.5</strong> and the false discovery  rate, <strong>(FDR) &lt;0.05</strong>.</section></li></ul></blockquote><p data-tool="mdnice编辑器">用samr包进行分析</p><pre data-tool="mdnice编辑器"><code>rm(list = ls())<br>gse &lt;- <span>"GSE27447"</span><br>set.seed(<span>5</span>)<br>getwd()<br>load(file = paste0(<span>"step1_"</span>,gse, <span>"_output.Rdata"</span>))<br><span># 2.1 samr包 -----------------------------------------# install.packages("samr")</span><br><span>library</span>(samr)<br>dim(dat)<br><span>##dat &lt;- as.matrix(dat)</span><br>data=list(x=dat,y=as.numeric(as.factor(group_list)), <br>          geneid=as.character(<span>1</span>:nrow(dat)),<br>          genenames=rownames(dat), <br>          logged2=<span>TRUE</span><br>)<br><br></code></pre><p data-tool="mdnice编辑器">如果在这一步报错:</p><pre data-tool="mdnice编辑器"><code>Error <span>in</span> (xdif^<span>2</span>) %*% rep(<span>1</span>/(n - <span>1</span>), n) : <br>  requires numeric/complex matrix/vector arguments<br></code></pre><p data-tool="mdnice编辑器">那么回头再看看官方文档对输入数据的格式要求，将dat的格式改为matrix即可。</p><pre data-tool="mdnice编辑器"><code><span>## output normalized sequencing data</span><br>x.norm &lt;- samr.norm.data(data$x)<br>par(mfrow=c(<span>1</span>,<span>2</span>))<br>boxplot(dat, col = rainbow(dat),main=<span>"before normalization"</span>,las=<span>2</span>)<br>boxplot(x.norm,  col = rainbow(dat),main=<span>"after normalization"</span>,las=<span>2</span>)<br><br><span>## Significance analysis of microarrays</span><br>samr.obj&lt;-samr(data, resp.type=<span>"Two class unpaired"</span>, nperms=<span>100</span>,assay.type =<span>"array"</span> )<br><br><br><span>## Assess the sample size for a SAM analysis</span><br>samr.assess.samplesize.obj&lt;- samr.assess.samplesize(samr.obj, data, log2(<span>1.5</span>))<br>samr.assess.samplesize.plot(samr.assess.samplesize.obj)<br><br><span># ## Compute delta table for SAM analysis</span><br>delta.table&lt;- samr.compute.delta.table(samr.obj,min.foldchange=<span>0.1</span>)<br>delta.table &lt;- as.data.frame(delta.table)<br></code></pre><p data-tool="mdnice编辑器">这个时候要怎么去挑选delta值呢？</p><blockquote data-tool="mdnice编辑器"><p>[使用SAMR对蛋白组数据表达量进行差异分析 | https://www.plob.org/article/10934.html)</p></blockquote><blockquote data-tool="mdnice编辑器"><p>在 Delta Table 查看 <strong>Median FDR</strong> 值，从上往下找到该值变为0.05时的delta值，再在左侧 Delta 一栏选择 manually Enter Delta, 然后再在 Delta value 一栏填入该值，再进行计算。</p></blockquote><pre data-tool="mdnice编辑器"><code><span># ##根据Median FDR值来确定delta值</span><br>Delta.table &lt;- delta.table[order(delta.table$`median FDR`),]<br><br>del &lt;-<span>0.6619838622</span> <span>##Median FDR = 0.06186104,最接近0.05</span><br><br><span>## Make Q-Q plot for SAM analysis</span><br>samr.plot(samr.obj, del, min.foldchange=<span>0.1</span>)<br><br><span># ## Compute significant genes table</span><br>siggenes.table&lt;-samr.compute.siggenes.table(samr.obj, del, data, delta.table,<br>                            min.foldchange=<span>0.1</span>, all.genes=<span>TRUE</span>,compute.localfdr=<span>TRUE</span>)<br>up_gene &lt;- siggenes.table[[<span>"genes.up"</span>]]<br>down_gene &lt;- siggenes.table[[<span>"genes.lo"</span>]]<br>siggenes.table$ngenes.up<br>siggenes.table$ngenes.lo<br><br>all_Gene &lt;- rbind(up_gene,down_gene)<br><span>###差异基因阈值设置</span><br>all_Gene &lt;- as.data.frame(all_Gene)<br>all_Gene$`Fold Change` &lt;- as.numeric(all_Gene$`Fold Change`)<br>all_Gene$logFC &lt;- log2(all_Gene$`Fold Change`)<br>df &lt;- all_Gene<br>df$g=ifelse(all_Gene$`localfdr(%)`&gt;<span>5</span>,<span>'stable'</span>, <br>            ifelse( all_Gene$logFC &gt;<span>1.5</span>,<span>'up'</span>, <br>                    ifelse( all_Gene$logFC &lt; -<span>1.5</span>,<span>'down'</span>,<span>'stable'</span>) )<br>)<br>table(df$g)<br><span># down stable   up </span><br><span># 169  18552    116</span><br>up_2018 &lt;- df$`Gene ID`[grep(<span>"up"</span>,df$g)]<br></code></pre><blockquote data-tool="mdnice编辑器"><p>文章应该是下载cel格式的原始数据进行分析的，原文描述“The primary data were formatted into  expression measurements and normalized by the robust  multi-array average <strong>(RMA) algorithm</strong>”。</p><p>CEL格式的数据从何开始？可以参考之前的推文【<a target="_blank" href="http://mp.weixin.qq.com/s?__biz=MzUzMTEwODk0Ng==&amp;mid=2247506157&amp;idx=1&amp;sn=42f34fa1d7b09e939353b186687b6c28&amp;chksm=fa451dd0cd3294c6c153da2032a53cd4e832600048ce6728b93217b32b14420143d7fbe5a9a3&amp;scene=21#wechat_redirect" textvalue="表达量矩阵的值上天入地，竟然从-13385.07 到了14698.99？" linktype="text" imgurl="" imgdata="null" data-itemshowtype="0" tab="innerlink" data-linktype="2">表达量矩阵的值上天入地，竟然从-13385.07 到了14698.99？</a>】</p></blockquote><hr data-tool="mdnice编辑器"><p data-tool="mdnice编辑器">再来看看产生这个数据集的原文献</p><blockquote data-tool="mdnice编辑器"><ul><li><section>TITLE：FZD7 has a critical role in cell proliferation in triple negative breast cancer</section></li><li><section>DOI: 10.1038/onc.2011.145</section></li><li><section>Objective：identify the cell proliferation regulator for TNBC tumor cells.</section></li><li><section>Results: <strong>Two hundred and six genes</strong>, including 169 annotated genes (Supplementary Figure 1) were identiﬁed as being differentially expressed.</section></li></ul></blockquote><p data-tool="mdnice编辑器">之前的操作跟一般数据集一样……</p><blockquote data-tool="mdnice编辑器"><p>[TNBC数据分析-GSE27447-GPL6244 - https://cloud.tencent.com/developer/article/1867936)</p></blockquote><blockquote data-tool="mdnice编辑器"><p>差异分析方法(未提及，那就用limma试试)</p><p>差异分析的阈值设置：Identiﬁcation of differentially expressed genes was ﬁrst carried out under the criteria of 1.5-fold <strong>upregulation</strong> in TNBC with a &gt;1.5 fold, Po0.01.即<strong>（&gt;1.5 fold, P&lt;0.01）</strong></p><p>这意味着作者只挑选了实验组相对于对照组表达量上调≥1.5倍的差异基因……</p></blockquote><p data-tool="mdnice编辑器">只是需要看看文献的阈值设置改一下参数：</p><pre data-tool="mdnice编辑器"><code><span>#根据文献要求设定上下调基因，将logFC还原为FC</span><br>df$FC= <span>2</span>^df$logFC<br>df$g1=ifelse(df$P.Value&gt;<span>0.01</span>,<span>'stable'</span>, <br>            ifelse( df$FC &gt;<span>1.5</span>,<span>'up'</span>, <br>                    <span>"others"</span>) )<br><span>#统计上下调基因数量</span><br>table(df$g1)     <br>others stable     up <br>   <span>258</span>  <span>18388</span>    <span>191</span><br>up_2011 &lt;- rownames(df)[grep(<span>"up"</span>,df$g1)]<br></code></pre><hr data-tool="mdnice编辑器"><p data-tool="mdnice编辑器">既然是用两种差异分析的方法来做的，那我们也来看看两种差异分析有多大的相关性？</p><p data-tool="mdnice编辑器">需要注意的是，两篇文章的原始数据获取方式不一样，我在分析的时候分别用了：<strong>getGEO获取表达量矩阵；直接下载cel格式的raw data</strong>，后期探针注释到基因的数量也有差别。所以在做差异分析和相关性分析的时候，要选取同一种方法注释到的dat矩阵，分别做两种差异分析，再运行下面的代码看相关性，否则代码会报错的哦！</p><pre data-tool="mdnice编辑器"><code><span>## 首先提取samr做差异分析检验的p值</span><br>pv_sam=samr.pvalues.from.perms(samr.obj$tt, samr.obj$ttstar)<br><span>## 然后提取limma包做差异分析检验的p值</span><br>pv_limma=deg$P.Value<br>names(pv_limma)=rownames(deg)<br>head(pv_sam[sort(names(pv_sam))])<br>length(pv_limma)<br>length(pv_sam)<br>cor(pv_sam[sort(names(pv_sam))],pv_limma[sort(names(pv_limma))])<br><span>##[1] 0.9993</span><br></code></pre><p data-tool="mdnice编辑器">两种不同的处理方式，得到的差异基因会有重叠吗？画个韦恩图试试看？</p><pre data-tool="mdnice编辑器"><code><span># install.packages("ggvenn")</span><br><span>library</span>(ggvenn)<br>deg&lt;-list(`up_2018`=up_2018,<br>          `up_2011`=up_2011)<br>          <br>p1 &lt;- ggvenn(deg,<br>             show_percentage = <span>F</span>,<br>             stroke_color = <span>"white"</span>,<br>             fill_color = c(<span>"#b2d4ec"</span>,<span>"#ffb2b2"</span>),<br>             set_name_color = c(<span>"#ff0000"</span>,<span>"#4a9b83"</span>))<br>p1<br><br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7718550106609808" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9ibYrHCm5GeOepNHo4oATQq1iaQuTVnbMnW35oF5IWtPY0ZtYOX13LlvWn0ngngLQE9FfUf8m2XcMA/640?wx_fmt=png" data-type="png" data-w="469" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9ibYrHCm5GeOepNHo4oATQq1iaQuTVnbMnW35oF5IWtPY0ZtYOX13LlvWn0ngngLQE9FfUf8m2XcMA/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">同一个数据集，同一片大海，只是用了不一样的渔网，捞上来了不一样的渔获，后面的故事也有了各自的情节，很神奇吧！</p><p data-tool="mdnice编辑器">可能生物信息学就是一门“捕捞学”，如何在茫茫大海中筛选自己想要的东西，小人鱼还是金鱼，还是金斧头、银斧头……，总之，科学捕捞！！！</p></section><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/Tb8OEg70vmxb39-jTFrukA",target="_blank" rel="noopener noreferrer">原文链接</a>
