---
title: "转录组+甲基化来点新花样：ELMER包"
date: 2023-01-01T05:30:34Z
draft: ["false"]
tags: [
  "fetched",
  "YuYuFiSH"
]
categories: ["Acdemic"]
---
转录组+甲基化来点新花样：ELMER包 by YuYuFiSH
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><h1 data-tool="mdnice编辑器"><span></span>ELMER 多组学R包</h1><blockquote data-type="2" data-url="" data-author-name="" data-content-utf8-length="40" data-source-title=""><section><section>如果你手上有甲基化和转录组两种数据，有太多分析的角度能辅助你完成一个漂亮的故事了</section></section></blockquote><figure data-tool="mdnice编辑器"><img data-ratio="0.41450313339301703" data-src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc5IgrnANu8GpqP6TXajaybAHxPaj3Ko5OP0iajxzupfyeBxicdXicEOVgOKoabSlicVDs56QwicJzbwjJQ/640?wx_fmt=png" data-type="png" data-w="1117" src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc5IgrnANu8GpqP6TXajaybAHxPaj3Ko5OP0iajxzupfyeBxicdXicEOVgOKoabSlicVDs56QwicJzbwjJQ/640?wx_fmt=png"></figure><h2 data-tool="mdnice编辑器"><span></span>自我介绍</h2><p data-tool="mdnice编辑器"><span>ELMER（Enhancer Linking by Methylation/Expression Relationships）</span>，增强子相关甲基化和表达相关网络</p><p data-tool="mdnice编辑器">关键是以甲基化位点作为网络的中心枢纽，在下游筛选出差异甲基化位点对应的碱基序列与对应的靶基因，而上游找到富集的motif（转录因子结合基序即有规律有特殊功能的一串碱基序列），再反推出与这些motif相互作用的转录因子，从而构建出<span><strong>TF—甲基化位点——靶基因网络</strong></span></p><h2 data-tool="mdnice编辑器"><span></span>主要流程</h2><p data-tool="mdnice编辑器">1.识别远端探针（distal probe）：HM450k或者EPIC甲基化芯片</p><p data-tool="mdnice编辑器">2.识别组间甲基化水平差异</p><p data-tool="mdnice编辑器">3.识别差异甲基化探针的靶基因</p><p data-tool="mdnice编辑器">4.确定差异甲基化差异探针以及靶基因相关探针富集的motif</p><p data-tool="mdnice编辑器">5.确认与motif上甲基化相关的调控转录因子</p><blockquote data-type="2" data-url="" data-author-name="" data-content-utf8-length="173" data-source-title=""><section><section><p data-tool="mdnice编辑器">R包第一步可以理解为识别关键位置的甲基化探针，这里指增强子（enhancer）或者启动子区域（promoter）的探针。普遍认为在顺式调控元件部位（包括启动子、增强子等部位）高度甲基化，将影响DNA的结构，从而阻遏该部位基因的转录</p><p data-tool="mdnice编辑器">distal probe：距离转录起始位点位点上游大于2kb的甲基化探针，默认这一区域的甲基化探针区域为增强子区域。</p></section></section></blockquote><h2 data-tool="mdnice编辑器"><span></span>最新2019年版本</h2><p data-tool="mdnice编辑器">新版本再次升级</p><figure data-tool="mdnice编辑器"><img data-ratio="0.6323076923076923" data-src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc5IgrnANu8GpqP6TXajaybAbkz8q6BxB80t35zHZn4Svz0bo8jLiaWQHaTzVd46wgKO1EYhv5VxsTQ/640?wx_fmt=png" data-type="png" data-w="1300" src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc5IgrnANu8GpqP6TXajaybAbkz8q6BxB80t35zHZn4Svz0bo8jLiaWQHaTzVd46wgKO1EYhv5VxsTQ/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">🍥 帮助文档：https://academic.oup.com/bioinformatics/article/35/11/1974/5145133?login=false</p><p data-tool="mdnice编辑器">http://bioconductor.org/packages/release/bioc/vignettes/ELMER/inst/doc/analysis_data_input.html</p><p data-tool="mdnice编辑器">http://bioconductor.org/packages/release/bioc/manuals/ELMER/man/ELMER.pdf</p><p data-tool="mdnice编辑器"><span></span><span>🍥 </span>工作流程workshop：</p><p data-tool="mdnice编辑器">https://benbermanlab.com/assets/code/ELMER%20workshop.html</p><p data-tool="mdnice编辑器">https://rpubs.com/tiagochst/elmer-data-workshop-2019</p><p data-tool="mdnice编辑器"><span></span><span>🍥 </span>文章TCGA-BRCA完整报告：</p><p data-tool="mdnice编辑器">https://media.githubusercontent.com/media/tiagochst/ELMER_supplemental/master/supplemental_files.zip</p><h2 data-tool="mdnice编辑器"><span>目录</span></h2><ul data-tool="mdnice编辑器"><li><section>获得远端探针：<code>get.feature.probe</code></section></li><li><section>载入甲基化，转录组，临床数据</section></li><li><section>创建MAE对象:<code>createMAE</code></section></li><li><section>差异分析：<code>get.diff.meth</code></section></li><li><section>寻找探针上下游基因：<code>GetNearGenes</code></section></li><li><section>识别probe-gene对：<code>get.pair</code></section></li><li><section>筛选motif：<code>get.enriched.motif</code></section></li><li><section>确定转录因子TF：<code>get.TFs</code></section></li><li><section>各种可视化：</section></li><ul><li><section>散点图：<code>scatter.plot</code></section></li><li><section>染色体示意图：<code>schematic.plot</code></section></li><li><section>motif富集森林图：<code>motif.enrichment.plot</code></section></li><li><section>转录因子排序图：<code>TF.rank.plot</code></section></li><li><section>热图：<code>heatmapPairs</code></section></li></ul><li><section>直接下载TCGA数据：<code>TCGA.pipe</code></section></li></ul><h2 data-tool="mdnice编辑器"><span></span>领域展开----</h2><h3 data-tool="mdnice编辑器"><span>1.R包安装</span></h3><p data-tool="mdnice编辑器">两种方法：从github中安装 or Bioconductor安装</p><pre data-tool="mdnice编辑器"><span></span><code>devtools::install_github(repo = <span>"tiagochst/ELMER.data"</span>)<br>devtools::install_github(repo = <span>"tiagochst/ELMER"</span>)<br><br><span>if</span> (!requireNamespace(<span>"BiocManager"</span>, quietly=TRUE))<br>    install.packages(<span>"BiocManager"</span>)<br>BiocManager::install(<span>"ELMER"</span>)<br>BiocManager::install(<span>"sesameData"</span>)<br><br>library(ELMER)<br>library(sesameData)<br>library(DT)<br></code></pre><h3 data-tool="mdnice编辑器"><span>2. 获得远端探针</span></h3><p data-tool="mdnice编辑器">这里以TCGA项目中肺鳞癌（LUSC）项目为例，只选择1号染色体的甲基化探针和基因作为例子进行分析。</p><p data-tool="mdnice编辑器">首先，探针要提前做出筛选，怎么获得增强子和启动子位置上的呢？r包这里已经根据数据库GENCODE和UCSC分好类，不过如果需要启动子上的探针，需要用参数<code>TSS.range</code>提前定义好启动子区域，默认是TSS上2000bp内，不过也可以自定义，毕竟没有硬性标准</p><ul data-tool="mdnice编辑器"><li><section>获得远端探针：<code>get.feature.probe</code></section></li></ul><pre data-tool="mdnice编辑器"><span></span><code><span># Selection of probes within biofeatures</span><br><span># get distal probes that are 2kb away from TSS on chromosome 1</span><br>distal.probes &lt;- get.feature.probe(<br>  genome = <span>"hg19"</span>, <span># hg38 (default) or hg19</span><br>  met.platform = <span>"450K"</span>, <span>#  EPIC or 450K (default)</span><br>  TSS.range = list(upstream = 2000, downstream = 2000), <span># define promoter regions定义启动子区域</span><br>  promoter = FALSE, <span># 是否选择启动子</span><br>  rm.chr = paste0(<span>"chr"</span>,c(2:22,<span>"X"</span>,<span>"Y"</span>)) <span># 移除的染色体</span><br><br>length(distal.probes)<span># [1] 15561</span><br>head(distal.probes)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.47194719471947194" data-src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc5IgrnANu8GpqP6TXajaybAcfCrbDOLzrjAOYic3drIIYp08JFWEgXFIbOMa0yXgJ6A7pZTmSUbRXg/640?wx_fmt=png" data-type="png" data-w="1212" src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc5IgrnANu8GpqP6TXajaybAcfCrbDOLzrjAOYic3drIIYp08JFWEgXFIbOMa0yXgJ6A7pZTmSUbRXg/640?wx_fmt=png"></figure><h3 data-tool="mdnice编辑器"><span>3. 输入数据(TCGA格式)</span></h3><p data-tool="mdnice编辑器">得到1号染色体的distal probe后，我们就需要</p><ul data-tool="mdnice编辑器"><li><section>载入甲基化，转录组数据</section></li><li><section>创建MAE对象: <code>createMAE</code></section></li></ul><pre data-tool="mdnice编辑器"><span></span><code><span># Selection of probes within biofeatures</span><br>library(MultiAssayExperiment)<br>library(ELMER.data)<br><span># 转录组数据</span><br>data(LUSC_RNA_refined,package = <span>"ELMER.data"</span>)<br><span># 甲基化数据</span><br>data(LUSC_meth_refined,package = <span>"ELMER.data"</span>)<br><br>dim(GeneExp) <span>#3842  234</span><br>dim(Meth) <span>#1728  268</span><br></code></pre><p data-tool="mdnice编辑器">可以看到，两种类型的数据样本数量不完全一致</p><pre data-tool="mdnice编辑器"><span></span><code>&gt; GeneExp[1:3,1:3]<br>                TCGA-22-5472-01A-01R-1635-07 TCGA-22-5489-01A-01R-1635-07<br>ENSG00000188984                    0.0000000                      0.00000<br>ENSG00000204518                    0.4303923                      0.00000<br>ENSG00000108270                   10.0817831                     10.71767<br>                TCGA-22-5491-11A-01R-1858-07<br>ENSG00000188984                      0.00000<br>ENSG00000204518                      0.00000<br>ENSG00000108270                     10.18086<br>&gt; Meth[1:3,1:3]<br>           TCGA-43-3394-11A-01D-1551-05 TCGA-43-3920-11B-01D-1551-05<br>cg00045114                    0.8190894                    0.8073763<br>cg00050294                    0.8423084                    0.8241138<br>cg00066722                    0.9101127                    0.9162212<br>           TCGA-56-8305-01A-11D-2294-05<br>cg00045114                    0.8907009<br>cg00050294                    0.5597787<br>cg00066722                    0.7228874<br></code></pre><p><br></p><ul data-tool="mdnice编辑器"><li><section>创建MAE对象:<code>createMAE</code></section></li></ul><pre data-tool="mdnice编辑器"><span></span><code>mae &lt;- createMAE(<br>  exp = GeneExp, <br>  met = Meth,<br>  save = TRUE,<br>  linearize.exp = TRUE, <span>#  log2(exp + 1) </span><br>  save.filename = <span>"mae.rda"</span>, <span># 保存的文件</span><br>  filter.probes = distal.probes, <span># promoter regions or distal feature regions</span><br>  met.platform = <span>"450K"</span>,<br>  genome = <span>"hg19"</span>,<br>  TCGA = TRUE<br>)<br><span># =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span><br><span>#   Creating a SummarizedExperiment from gene expression input</span><br><span># =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span><br><span>#   Creating a SummarizedExperiment from DNA methylation input</span><br><span># Checking if samples have both DNA methylation and Gene expression and if they are in the same order...</span><br><span># Starting to add information to samples</span><br><span># =&gt; Add clinical information to samples</span><br><span># =&gt; Adding TCGA molecular information from marker papers</span><br><span># =&gt; Information will have prefix 'paper_' </span><br><span># lusc subtype information from:doi:10.1038/nature11404<svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" width="32px" height="32px" viewbox="0 0 32 32" enable-background="new 0 0 32 32"><image width="32" height="32" x="0" y="0"></image></svg></span><br><span># Creating MultiAssayExperiment</span><br><span># MAE saved as: mae.rda</span><br></code></pre><p data-tool="mdnice编辑器">MAE对象将数据进行整合，提取同时拥有两种数据的234个样本。可以根据下面提示的函数探索下MAE对象</p><pre data-tool="mdnice编辑器"><span></span><code>&gt; mae<br>A MultiAssayExperiment object of 2 listed<br> experiments with user-defined names and respective classes.<br> Containing an ExperimentList class object of length 2:<br> [1] DNA methylation: RangedSummarizedExperiment with 1642 rows and 234 columns<br> [2] Gene expression: RangedSummarizedExperiment with 3808 rows and 234 columns<br>Functionality:<br> experiments() - obtain the ExperimentList instance<br> colData() - the primary/phenotype DataFrame<br> sampleMap() - the sample coordination DataFrame<br> `$`, `[`, `[[` - extract colData columns, subset, or experiment<br> *Format() - convert into a long or wide DataFrame<br> assays() - convert ExperimentList to a SimpleList of matrices<br> exportClass() - save data to flat files<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.5152603231597845" data-src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc5IgrnANu8GpqP6TXajaybAy2PluJWwkeQJPsnXejAK2JkoxwuRvy16EmGH0eqJ4XucdhOWWLD3aw/640?wx_fmt=png" data-type="png" data-w="1114" src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc5IgrnANu8GpqP6TXajaybAy2PluJWwkeQJPsnXejAK2JkoxwuRvy16EmGH0eqJ4XucdhOWWLD3aw/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code><span># 解析mae对象</span><br><span>#获取甲基化表达谱数据</span><br>getMet(mae) %&gt;% assay()<br><span>#获取MEA对象转录组数据</span><br>getExp(mae) %&gt;% assay()<br><span>#获取MEA对象的metadata信息</span><br>colData(mae) %&gt;% as.data.frame()<br></code></pre><h3 data-tool="mdnice编辑器"><span>3.2 输入数据（非TCGA格式）</span></h3><p data-tool="mdnice编辑器">当然，如果你有自己的测序数据，也可以构建属于自己的MAE对象咯</p><p data-tool="mdnice编辑器">需要准备：</p><p data-tool="mdnice编辑器">1.以探针为行，以样本为列的β值甲基化矩阵</p><p data-tool="mdnice编辑器">2.以基因为行，以样本为列并且标准化处理后的转录组矩阵（FPKM等）</p><p data-tool="mdnice编辑器">3.甲基化样本名和转录组样本名对应的转化矩阵</p><p data-tool="mdnice编辑器">4.样本本身的临床信息（metadata）</p><pre data-tool="mdnice编辑器"><span></span><code>&gt; head(exp)<br>                Sample1 Sample2 Sample3 Sample4 Sample5<br>ENSG00000073282       0       0       0       0       0<br>ENSG00000078900       0       0       0       0       0<br>ENSG00000141510       0       0       0       0       0<br>&gt; head(met)<br>           Sample1 Sample2 Sample3 Sample4 Sample5<br>cg26928153       0       0       0       0       0<br>cg16269199       0       0       0       0       0<br>cg13869341       0       0       0       0       0<br>&gt; head(colData) <span># 后续可以增加多几列临床信息</span><br>         sample<br>Sample1 Sample1<br>Sample2 Sample2<br>Sample3 Sample3<br>Sample4 Sample4<br>Sample5 Sample5<br>&gt; sampleMap <span># 指定样本类型，防止有些样本名不一致，此数据集三个列名固定不变</span><br>             assay primary colname<br>1  DNA methylation Sample1 Sample1<br>2  DNA methylation Sample2 Sample2<br>3  DNA methylation Sample3 Sample3<br>4  DNA methylation Sample4 Sample4<br>5  DNA methylation Sample5 Sample5<br>6  Gene expression Sample1 Sample1<br>7  Gene expression Sample2 Sample2<br>8  Gene expression Sample3 Sample3<br>9  Gene expression Sample4 Sample4<br>10 Gene expression Sample5 Sample5<br></code></pre><p data-tool="mdnice编辑器">这种情况下，你就照着葫芦画瓢，模仿文档给出的数据格式构造即可</p><pre data-tool="mdnice编辑器"><span></span><code>distal.probes &lt;- get.feature.probe(<br>  genome = <span>"hg19"</span>, <br>  met.platform = <span>"EPIC"</span><br>)<br><br>mae &lt;- createMAE(<br>  exp = exp, <br>  met = met,<br>  save = TRUE,<br>  filter.probes = distal.probes,<br>  colData = colData,<br>  sampleMap = sampleMap,<br>  linearize.exp = TRUE,<br>  save.filename = <span>"mae.rda"</span>,<br>  met.platform = <span>"EPIC"</span>,<br>  genome = <span>"hg19"</span>,<br>  TCGA = FALSE<br>)<br></code></pre><h3 data-tool="mdnice编辑器"><span>4. 筛选差异甲基化探针</span></h3><ul data-tool="mdnice编辑器"><li><section>差异分析：<code>get.diff.meth</code></section></li></ul><p data-tool="mdnice编辑器">ELMER提供了两种方式进行筛选差异探针</p><p data-tool="mdnice编辑器">（1）supervised，即使用未配对的单侧t检验，将组1中所有样本的的DNA甲基化水平与组2中的所有样本进行比较</p><p data-tool="mdnice编辑器">（2）unsupervised，即根据每个探针的甲基化水平分别对两个组的样本先进行排序，每组中选择最低20%甲基化水平的样本用于识别此探针是否低甲基化，高甲基化探针的鉴定正好相反。(原因是：疾病（尤其是肿瘤）具有明确的异质性，通过提取20%的子集能够进一步找到疾病中特定的分子亚型)</p><p data-tool="mdnice编辑器">因此，在无监督模式下，分析每个探针的样本可能基于总样本的不同子集，可以表示来自多个分子亚型的探针集的结果。而在监督模式下，所有测试都基于所有样本。</p><figure data-tool="mdnice编辑器"><img data-ratio="1.1711864406779662" data-src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc5IgrnANu8GpqP6TXajaybAOykrvoNcYZIBOmmR74ia8wIskoBtsk6AN9GDyVT0y4ZEj7AFkmMmh4g/640?wx_fmt=png" data-type="png" data-w="1180" src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc5IgrnANu8GpqP6TXajaybAOykrvoNcYZIBOmmR74ia8wIskoBtsk6AN9GDyVT0y4ZEj7AFkmMmh4g/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">这里以unsupervised方式为例，通过比较不同组间样本中所有远端探针（distal probe）的甲基化水平，筛选出adj.P value&lt;0.01，Δβ＞0.3的甲基化位点，并且是在肿瘤组织低甲基化水平的位点</p><pre data-tool="mdnice编辑器"><span></span><code><span>### 差异分析----</span><br>table(colData(mae)[,<span>"definition"</span>])<br><span># Primary solid Tumor Solid Tissue Normal </span><br><span># 226                   8 </span><br>sig.diff &lt;- get.diff.meth(data = mae, <br>                          group.col = <span>"definition"</span>, <span># 组别类型对应的列名</span><br>                          group1 =  <span>"Primary solid Tumor"</span>, <span>#  if direction is hyper, get probes hypermethylated in group 1 compared to group 2</span><br>                          group2 = <span>"Solid Tissue Normal"</span>,<br>                          minSubgroupFrac = 0.2,  <span># 默认指定排序后的20%甲基化水平的样本进行比较</span><br>                          sig.dif = 0.3, <span># 组间最小beta值差异</span><br>                          diff.dir = <span>"hypo"</span>, <span># "hypo", "hyper" or "both"</span><br>                          cores = 1, <span># 并行核数</span><br>                          dir.out =<span>"result"</span>,<br>                          mode = <span>"unsupervised"</span>,<br>                          pvalue = 0.01)<br><br>head(sig.diff)<br>dir(path = <span>"result"</span>, pattern = <span>"getMethdiff"</span>) <br></code></pre><p data-tool="mdnice编辑器">这里要注意参数<code>diff.dir = "hypo"</code>，意味着在group1为低甲基化水平的探针</p><p data-tool="mdnice编辑器">"hypo" which is only selecting hypomethylated probes (one tailed test); "hyper" which is only selecting hypermethylated probes (one tailed test); or "both" which are probes differenly methylated (two tailed test).</p><p data-tool="mdnice编辑器">输出：所有和显著差异位点的csv结果，以及一张火山图</p><pre data-tool="mdnice编辑器"><span></span><code>&gt; dim(sig.diff) <span>#710   4</span><br>&gt; head(sig.diff)<br>       probe       pvalue Primary.solid.Tumor_Minus_Solid.Tissue.Normal<br>1 cg25340711 3.898829e-05                                    -0.3193081<br>2 cg24852892 5.418625e-24                                    -0.4664680<br>3 cg06358191 2.617048e-09                                    -0.3832813<br>4 cg06646682 8.147636e-18                                    -0.4730666<br>5 cg15844005 2.104992e-30                                    -0.3818866<br>6 cg06205922 2.302866e-22                                    -0.3646617<br>      adjust.p<br>1 5.509361e-05<br>2 4.835534e-23<br>3 6.310122e-09<br>4 4.041818e-17<br>5 4.019066e-29<br>6 1.742537e-21<br><br>&gt; dir(path = <span>"result"</span>, pattern = <span>"getMethdiff"</span>)  <br>[1] <span>"getMethdiff.hypo.probes.csv"</span>            <br>[2] <span>"getMethdiff.hypo.probes.significant.csv"</span><br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.48698884758364314" data-src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc5IgrnANu8GpqP6TXajaybAbZGFUSZfoK1xOjOUHx9Ww1ZBPTEYyO8Tv7QC1Ax1rRY85NRMrVWjRg/640?wx_fmt=png" data-type="png" data-w="1345" src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc5IgrnANu8GpqP6TXajaybAbZGFUSZfoK1xOjOUHx9Ww1ZBPTEYyO8Tv7QC1Ax1rRY85NRMrVWjRg/640?wx_fmt=png"></figure><h3 data-tool="mdnice编辑器"><span>5. 识别甲基化-基因对</span></h3><p data-tool="mdnice编辑器">这一步，我们将把远端探针的甲基化水平和靶基因的表达水平进行相关性分析，从而构建出甲基化-基因对（probe-gene）。这个包具体的做法是分别找到差异甲基化远端探针的上游和下游最近的10个基因，分析其余对应探针甲基化水平是否存在负相关，从而筛选出来的潜在probe-gene对。</p><p data-tool="mdnice编辑器">同样提供了两类方式：</p><p data-tool="mdnice编辑器">（1）supervised，即通过上游分析疾病组vs.正常组差异远端探针，在所有样本中进行非参数检，找出同时在疾病组高表达的基因</p><p data-tool="mdnice编辑器">（2）unsupervised，根据基因的distal probe甲基化水平进行排序，提取出甲基化水平分别在前20%和后20%的样本，分别为M和U组，通过U检验方式比较两组间基因表达水平,如果在M组的表达水平显著低于U组，并通过迭代的方式计算出所有远端探针和靶基因的相关性P value,从而筛选出具有意义的关系对。</p><figure data-tool="mdnice编辑器"><img data-ratio="0.5716993051168667" data-src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc5IgrnANu8GpqP6TXajaybADAN8ElYQ7sSz911LMuEob6dsqPfgwtIFEffQzobHJyGSR0qes5iakMg/640?wx_fmt=png" data-type="png" data-w="3166" src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc5IgrnANu8GpqP6TXajaybADAN8ElYQ7sSz911LMuEob6dsqPfgwtIFEffQzobHJyGSR0qes5iakMg/640?wx_fmt=png"></figure><p><br></p><ul data-tool="mdnice编辑器"><li><section>寻找探针上下游基因：<code>GetNearGenes</code></section></li></ul><pre data-tool="mdnice编辑器"><span></span><code><span>### 识别probe-gene对-----</span><br><span># 加载我们之前的数据</span><br>mae &lt;- get(load(<span>"mae.rda"</span>))<br><span>#加载我们筛选得到的差异甲基化远端探针（distal probe）</span><br>sig.diff &lt;- read.csv(<span>"result/getMethdiff.hypo.probes.significant.csv"</span>)<br><span>#找到差异甲基化远端探针（distal probe）上下游10个基因</span><br>nearGenes &lt;- GetNearGenes(data = mae, <br>                          probes = sig.diff<span>$probe</span>, <br>                          numFlankingGenes = 20) <span># 分别是上游10个和下游10基因</span><br><span>## Searching for the 20 near genes</span><br><span>## Identifying gene position for each probe</span><br>head(nearGenes)<br></code></pre><p><br></p><ul data-tool="mdnice编辑器"><li><section>识别probe-gene对：<code>get.pair</code></section></li></ul><pre data-tool="mdnice编辑器"><span></span><code><span>#因为前面使用unsupervised，我们这而也用unsupervised</span><br>hypo.pair &lt;- get.pair(data = mae,<br>                      group.col = <span>"definition"</span>,<br>                      group1 =  <span>"Primary solid Tumor"</span>,<br>                      group2 = <span>"Solid Tissue Normal"</span>,<br>                      nearGenes = nearGenes,<br>                      minSubgroupFrac = 0.4,<span># 默认创建M和U组样本的比例</span><br>                      mode = <span>"unsupervised"</span>,<br>                      permu.dir = <span>"result/permu"</span>,<br>                      permu.size = 100, <span># 一般要设置迭代次数为10000，处于演示目的减少次数</span><br>                      raw.pvalue = 0.05,   <br>                      Pe = 0.01, <span># P值，默认0.001</span><br>                      filter.probes = TRUE, <br>                      filter.percentage = 0.05, <span># 最小考虑的样本比例</span><br>                      filter.portion = 0.3, <span># 甲基化水平截断点</span><br>                      dir.out = <span>"result"</span>,<span>#结果同样输出到result目录</span><br>                      cores = 1,<br>                      label = <span>"hypo"</span>)<br><span># Selecting U (unmethylated) and M (methylated) groups. Each groups has 20% of samples</span><br><span># -------------------</span><br><span>#   * Filtering probes</span><br><span># -------------------</span><br><span>#   For more information see function preAssociationProbeFiltering</span><br><span># Making sure we have at least 5% of beta values lesser than 0.3 and 5% of beta values greater 0.3.</span><br><span># Removing 592 probes out of 1642</span><br><span># Calculating Pp (probe - gene) for all nearby genes</span><br><span># |============================================|100%                      Completed after 5 s </span><br><span># File created: result/getPair.hypo.all.pairs.statistic.csv             474.73MB/s</span><br><span># Calculating Pr (random probe - gene). Permutating  100 probes for 747 genes</span><br><span># Calculate empirical P value.              </span><br>head(hypo.pair)<br>dim(hypo.pair) <span># [1] 167  7</span><br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>&gt; head(nearGenes)<br><span># A tibble: 6 × 5</span><br><span># Groups:   ID [1]</span><br>  ID         GeneID          Symbol  Distance Side <br>  &lt;chr&gt;      &lt;chr&gt;           &lt;chr&gt;      &lt;dbl&gt; &lt;chr&gt;<br>1 cg00045114 ENSG00000076258 FMO4    -1362935 L10  <br>2 cg00045114 ENSG00000117523 PRRC2C  -1111508 L9   <br>3 cg00045114 ENSG00000034971 MYOC    -1052335 L8   <br>4 cg00045114 ENSG00000117533 VAMP4    -962771 L7   <br>5 cg00045114 ENSG00000010165 METTL13  -890995 L6   <br>6 cg00045114 ENSG00000197959 DNM3     -286552 L5 <br><br>&gt; head(hypo.pair)<br>                                Probe          GeneID Symbol Distance<br>cg20701183.ENSG00000143013 cg20701183 ENSG00000143013   LMO4        0<br>cg19403323.ENSG00000143469 cg19403323 ENSG00000143469  SYT14    87476<br>cg12213388.ENSG00000143674 cg12213388 ENSG00000143674   MLK4  -984182<br>cg26607897.ENSG00000143199 cg26607897 ENSG00000143199 ADCY10   267784<br>cg10574861.ENSG00000143013 cg10574861 ENSG00000143013   LMO4        0<br>cg26607897.ENSG00000143147 cg26607897 ENSG00000143147 GPR161   543156<br>                           Sides        Raw.p         Pe<br>cg20701183.ENSG00000143013    L1 7.453984e-14 0.00990099<br>cg19403323.ENSG00000143469    R1 1.671937e-12 0.00990099<br>cg12213388.ENSG00000143674    L4 2.527644e-12 0.00990099<br>cg26607897.ENSG00000143199    R3 4.593610e-12 0.00990099<br>cg10574861.ENSG00000143013    L1 4.770162e-12 0.00990099<br>cg26607897.ENSG00000143147    R6 8.048248e-12 0.00990099<br><br>&gt; dir(path = <span>"result"</span>, pattern = <span>"getPair"</span>) <br>[1] <span>"getPair.hypo.all.pairs.statistic.csv"</span>                  <br>[2] <span>"getPair.hypo.pairs.significant.csv"</span>                    <br>[3] <span>"getPair.hypo.pairs.statistic.with.empirical.pvalue.csv"</span><br></code></pre><h3 data-tool="mdnice编辑器"><span>6. 筛选motif</span></h3><p data-tool="mdnice编辑器">提取探针上下游250bp区域的碱基序列，从而找到富集的motif。接下来再通过转录因子（TF）结合motif数据库（HOMER数据库），从而预测出motif结合的转录因子</p><ul data-tool="mdnice编辑器"><li><section>筛选motif：<code>get.enriched.motif</code></section></li></ul><pre data-tool="mdnice编辑器"><span></span><code><span>### 识别motif-----</span><br><br>enriched.motif &lt;- get.enriched.motif(data = mae,<br>                                     probes = hypo.pair<span>$Probe</span>, <br>                                     dir.out = <span>"result"</span>, <br>                                     label = <span>"hypo"</span>,<br>                                     min.incidence = 1, <span>#  the minimum incidence of motif</span><br>                                     lower.OR = 1.1) <span># motif最低95%置信区间OR阈值</span><br><span>## Loading object: Probes.motif.hg19.450K</span><br><br>names(enriched.motif)<br>enriched.motif[[1]]<br></code></pre><pre data-tool="mdnice编辑器"><span></span><code><span># 识别出六种motif</span><br>&gt; names(enriched.motif)<br>[1] <span>"HXA9_HUMAN.H11MO.0.B"</span>  <span>"FOXH1_HUMAN.H11MO.0.A"</span> <span>"PO5F1_HUMAN.H11MO.1.A"</span><br>[4] <span>"HXB13_HUMAN.H11MO.0.A"</span> <span>"IRF9_HUMAN.H11MO.0.C"</span>  <span>"SOX10_HUMAN.H11MO.1.A"</span><br><br><span># 展示第一个motif包含的甲基化探针</span><br>&gt; enriched.motif[[1]]<br> [1] <span>"cg13067635"</span> <span>"cg24873093"</span> <span>"cg25210796"</span> <span>"cg18345456"</span> <span>"cg14094320"</span><br> [6] <span>"cg17901924"</span> <span>"cg22736624"</span> <span>"cg12213388"</span> <span>"cg13305336"</span> <span>"cg26607897"</span><br>[11] <span>"cg11718886"</span> <span>"cg24593832"</span> <span>"cg25729466"</span> <span>"cg24446417"</span><br><br><span># 输出的结果文件</span><br>&gt; dir(path = <span>"result"</span>, pattern = <span>"getMotif"</span>) <br>[1] <span>"getMotif.hypo.enriched.motifs.rda"</span>  <span>"getMotif.hypo.motif.enrichment.csv"</span><br><br><span># 输出的图片</span><br>&gt; dir(path = <span>"result"</span>, pattern = <span>"motif.enrichment.pdf"</span>) <br>[1] <span>"hypo.quality.A-DS_with_summary.motif.enrichment.pdf"</span><br>[2] <span>"hypo.quality.A-DS.motif.enrichment.pdf"</span><br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="1.190800681431005" data-src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc5IgrnANu8GpqP6TXajaybA6tp1m6iauC7c8eicaThPnxgSLfxz5vgN6zBvnT7n8CDJ1admKy5aib9gQ/640?wx_fmt=png" data-type="png" data-w="587" src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc5IgrnANu8GpqP6TXajaybA6tp1m6iauC7c8eicaThPnxgSLfxz5vgN6zBvnT7n8CDJ1admKy5aib9gQ/640?wx_fmt=png"></figure><h3 data-tool="mdnice编辑器"><span>7. 确定转录因子TF</span></h3><p data-tool="mdnice编辑器">这一步，ELMER包将上一步我们通过富集分析找到富集的motif以及对应motif的转录因子进行分析，筛选出对应转录水平发生改变的转录因子。</p><p data-tool="mdnice编辑器">同样地，提供了两种方式：</p><p data-tool="mdnice编辑器">（1）unsupervised，即将存在相同motif的远端探针（distal probe）根据甲基化水平分为高甲基化组（一般取前20%）和低甲基化组（后20%），比较其两组间对应TF的表达值是否存在差异。</p><p data-tool="mdnice编辑器">（2）Supervised，即对疾病组vs.对照组相同motif对应TF表达值进行分析。进而筛选出前5% P value最小的TFs，并视为潜在的上游调控转录因子。</p><ul data-tool="mdnice编辑器"><li><section>确定转录因子TF：<code>get.TFs</code></section></li></ul><pre data-tool="mdnice编辑器"><span></span><code><span>## 确定转录因子TF-----</span><br>TF &lt;- get.TFs(data = mae, <br>              group.col = <span>"definition"</span>,<br>              group1 =  <span>"Primary solid Tumor"</span>,<br>              group2 = <span>"Solid Tissue Normal"</span>,<br>              mode = <span>"unsupervised"</span>,<br>              enriched.motif = enriched.motif,<br>              dir.out = <span>"result"</span>, <br>              cores = 1, <br>              label = <span>"hypo"</span>)<br>head(TF)<br>dir(path = <span>"result"</span>, pattern = <span>"getTF"</span>) <br></code></pre><pre data-tool="mdnice编辑器"><span></span><code>&gt; dir(path = <span>"result"</span>, pattern = <span>"getTF"</span>) <br>[1] <span>"getTF.hypo.significant.TFs.with.motif.summary.csv"</span><br>[2] <span>"getTF.hypo.TFs.with.motif.pvalue.rda"</span> <br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.19132029339853301" data-src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc5IgrnANu8GpqP6TXajaybAWzVbcegRL5ibrdIBcRQuUPlRbJUx7Z345u5icNKKFG3WNibNrslKXvRXg/640?wx_fmt=png" data-type="png" data-w="1636" src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc5IgrnANu8GpqP6TXajaybAWzVbcegRL5ibrdIBcRQuUPlRbJUx7Z345u5icNKKFG3WNibNrslKXvRXg/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器"><br></p><p data-tool="mdnice编辑器">总结一下，我们通过同一批样本的甲基化数据和转录组数据，首先找到具有差异的远端探针（distal probe），并根据远端探针和对应基因的表达水平构建出probe-gene对。并将probe-gene对中的probe进一步进行motif富集分析，找出潜在富集motif，最后根据富集的motif推测出上游的调控TFs</p><h3 data-tool="mdnice编辑器"><span>8. 各种可视化</span></h3><ul data-tool="mdnice编辑器"><li><section>散点图：<code>scatter.plot</code></section></li><li><section>染色体示意图：<code>schematic.plot</code></section></li><li><section>motif富集森林图：<code>motif.enrichment.plot</code></section></li><li><section>转录因子排序图：<code>TF.rank.plot</code></section></li><li><section>热图：<code>heatmapPairs</code></section></li></ul><pre data-tool="mdnice编辑器"><span></span><code><span>### 各种可视化----</span><br><span>##### 1.散点图----</span><br><span>#加载数据</span><br>mae &lt;- get(load(<span>"mae.rda"</span>))<br>load(<span>"result/getMotif.hypo.enriched.motifs.rda"</span>)<br><span>#探针和上下游分别10个基因的散点图</span><br>scatter.plot(data = mae,<br>             byProbe = list(probe = c(<span>"cg19403323"</span>), <br>                            numFlankingGenes = 20), <br>             category = <span>"definition"</span>, <br>             lm = TRUE, <span># 画出线条</span><br>             save = FALSE)<br><span>## Searching for the 20 near genes</span><br><span>## Identifying gene position for each probe</span><br><br><span>#只展现一对probe-gene散点图</span><br>scatter.plot(data = mae,<br>             byPair = list(probe = c(<span>"cg19403323"</span>), gene = c(<span>"ENSG00000143469"</span>)), <br>             category = <span>"definition"</span>, save = TRUE, lm_line = TRUE)<br><br><br><span>#特定转录因子表达水平以及特定motif对应探针甲基化水平散点图</span><br>scatter.plot(data = mae,<br>             byTF = list(TF = c(<span>"SOX2"</span>),<br>                         probe = enriched.motif[[names(enriched.motif)[1]]]), <br>             category = <span>"definition"</span>,<br>             save = F, <span>#保存为PDF版本</span><br>             lm_line = TRUE)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.6507936507936508" data-src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc5IgrnANu8GpqP6TXajaybAbK5C64CgiauH2C6CtxGCS5Ceom0qun5c8wVQEYvRAuN2tJk19NK8k5Q/640?wx_fmt=png" data-type="png" data-w="1008" src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc5IgrnANu8GpqP6TXajaybAbK5C64CgiauH2C6CtxGCS5Ceom0qun5c8wVQEYvRAuN2tJk19NK8k5Q/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code><span>##### 2.染色体示意图----</span><br><span># 使用genome browser的方式展示远端探针（distal probe）甲基化水平和附近20个基因在染色体上的位置。</span><br>library(GenomicInteractions)<br><span>#加载probe-gene对</span><br>pair &lt;- read.csv(<span>"result/getPair.hypo.pairs.significant.csv"</span>)<br><span>#展现探针附近的gene</span><br>schematic.plot(pair = pair, <br>               data = mae,<br>               group.col = <span>"definition"</span>,<br>               byProbe = pair<span>$Probe</span>[1],<br>               save = FALSE)<br><span>## Searching for the 20 near genes</span><br><span>## Identifying gene position for each probe</span><br><br><span>#展示基因附近的probe</span><br>schematic.plot(pair = pair, <br>               data = mae,   <br>               group.col = <span>"definition"</span>, <br>               byGene = pair<span>$GeneID</span>[1],<br>               save = FALSE)<br><br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.547395388556789" data-src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc5IgrnANu8GpqP6TXajaybAJKqicMSTiaSzdPnOrmlLpTAQJib9C0sHjrrjmZeP9Vuq2JNMXicuVbfw0A/640?wx_fmt=png" data-type="png" data-w="1171" src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc5IgrnANu8GpqP6TXajaybAJKqicMSTiaSzdPnOrmlLpTAQJib9C0sHjrrjmZeP9Vuq2JNMXicuVbfw0A/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code><span>##### 3.motif富集森林图-----</span><br><span>#初步展示</span><br>motif.enrichment.plot(motif.enrichment = <span>"result/getMotif.hypo.motif.enrichment.csv"</span>, <br>                      significant = list(OR = 1.5,lowerOR = 1.3), <br>                      label = <span>"hypo"</span>, <br>                      save = FALSE)<br><br><span>#展现OR值和探针数</span><br>motif.enrichment.plot(motif.enrichment = <span>"result/getMotif.hypo.motif.enrichment.csv"</span>, <br>                      significant = list(OR = 1.5,lowerOR = 1.3), <br>                      label = <span>"hypo"</span>, <br>                      summary = TRUE,<br>                      save = FALSE)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.5552742616033756" data-src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc5IgrnANu8GpqP6TXajaybAiaa2kId4WwEs1L2zcrkrxAlCZicbdicYVVC2ZiaVQeCyric917sfIIHFZ3g/640?wx_fmt=png" data-type="png" data-w="1185" src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc5IgrnANu8GpqP6TXajaybAiaa2kId4WwEs1L2zcrkrxAlCZicbdicYVVC2ZiaVQeCyric917sfIIHFZ3g/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code><span>##### 4.转录因子排序图-----</span><br>load(<span>"result/getTF.hypo.TFs.with.motif.pvalue.rda"</span>)<br>motif &lt;- colnames(TF.meth.cor)[1]<br>TF.rank.plot(motif.pvalue = TF.meth.cor, <br>             motif = motif,<br>             save = FALSE)<br><span>## Retrieving TFClass family classification from ELMER.data.</span><br><span>## Retrieving TFClass subfamily classification from ELMER.data.</span><br><span>## |                                                    |  0%                      |====================================================|100% ~0 s remaining       |====================================================|100%                      Completed after 0 s</span><br><span>## $HXA9_HUMAN.H11MO.0.B</span><br><span>## Warning: ggrepel: 69 unlabeled data points (too many overlaps). Consider</span><br><span>## increasing max.overlaps</span><br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7960264900662252" data-src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc5IgrnANu8GpqP6TXajaybAq63yMoc8N3ncD41Q0hqPicV4w2DZnST0zpCiaIeKvhh3J74eSXiaCicABQ/640?wx_fmt=png" data-type="png" data-w="755" src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc5IgrnANu8GpqP6TXajaybAq63yMoc8N3ncD41Q0hqPicV4w2DZnST0zpCiaIeKvhh3J74eSXiaCicABQ/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code><span>##### 5. 热图-----</span><br>heatmapPairs(data = mae, <br>             group.col = <span>"definition"</span>,<br>             group1 = <span>"Primary solid Tumor"</span>, <br>             annotation.col = c(<span>"years_smoked"</span>,<span>"gender"</span>),<br>             group2 = <span>"Solid Tissue Normal"</span>,<br>             pairs = pair,<br>             filename =  NULL)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.8526077097505669" data-src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc5IgrnANu8GpqP6TXajaybANrvHSID6yEMUvZAWVpUIShZgickIy53t6ibmSN9jMWcdUOmKVG88acAA/640?wx_fmt=png" data-type="png" data-w="882" src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc5IgrnANu8GpqP6TXajaybANrvHSID6yEMUvZAWVpUIShZgickIy53t6ibmSN9jMWcdUOmKVG88acAA/640?wx_fmt=png"></figure><h3 data-tool="mdnice编辑器"><span></span>9. 直接下载TCGA数据<span></span></h3><p data-tool="mdnice编辑器">当然，可以直接使用<code>TCGA.pipe</code>，联合r包TCGAbiolinks直接下载TCGA的数据分析</p><p data-tool="mdnice编辑器">The following command will do distal DNA methylation analysis and predict putative target genes, motif analysis and identify regulatory transcription factors.</p><pre data-tool="mdnice编辑器"><span></span><code>TCGA.pipe(<span>"LUSC"</span>,<br>          wd = <span>"./ELMER.example"</span>,<br>          cores = parallel::detectCores()/2,<br>          mode = <span>"unsupervised"</span>,<br>          permu.size = 300,<br>          Pe = 0.01,<br>          analysis = c(<span>"distal.probes"</span>,<span>"diffMeth"</span>,<span>"pair"</span>,<span>"motif"</span>,<span>"TF.search"</span>),<br>          diff.dir = <span>"hypo"</span>,<br>          rm.chr = paste0(<span>"chr"</span>,c(<span>"X"</span>,<span>"Y"</span>)))<br></code></pre><ul data-tool="mdnice编辑器"><li><section>还可以直接在线分析</section></li></ul><pre data-tool="mdnice编辑器"><span></span><code><span>### 在线版本----</span><br>BiocManager::install(<span>"TCGAbiolinksGUI"</span>)<br>library(TCGAbiolinksGUI)<br>TCGAbiolinksGUI()<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.5686433793663688" data-src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc5IgrnANu8GpqP6TXajaybAAVLYyADUfZa8If58ep4iaW39LeTIHdUvcDibpjb9XlgibPOMyL58hicLrw/640?wx_fmt=png" data-type="png" data-w="1231" src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc5IgrnANu8GpqP6TXajaybAAVLYyADUfZa8If58ep4iaW39LeTIHdUvcDibpjb9XlgibPOMyL58hicLrw/640?wx_fmt=png"></figure><h3 data-tool="mdnice编辑器"><span>10. N个好奇心</span></h3><h4 data-tool="mdnice编辑器"><span></span>10.1 Supervised versus Unsupervised mode？<span></span></h4><p data-tool="mdnice编辑器">可以看到前面很多功能，作者都提供两种模式，监督 or 非监督模式，那在实际工作中采用哪个比较多呢？</p><p data-tool="mdnice编辑器">在第一版ELMER v. 1时，作者推荐当两组样本均为同质性(即处理过的vs未处理的，A分子亚型vs B分子亚型)时，应采用监督模式;当至少有一组样本为异质性(即肿瘤样本)时，应采用无监督模式</p><p data-tool="mdnice编辑器">而最新版，作者使用ELMER v. 2.4.3分析TCGA-BRCA数据，探索增强子-基因对（enhancer-gene）时，发现监督模式有更强的统计效能，并且能识别更多的增强子-基因对</p><p data-tool="mdnice编辑器">This comparison suggests that while Unsupervised mode can serve as a useful exploratory tool when sample subtype is unknown a priori, Supervised mode offers greater statistical power when sample subtype is pre-defined.</p><figure data-tool="mdnice编辑器"><img data-ratio="0.6625824693685203" data-src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc5IgrnANu8GpqP6TXajaybAZpwfCYknCTkP6CZOrNHjwKHmenFv6AGeyKHkecp0yV0E7l6TQHqicLQ/640?wx_fmt=png" data-type="png" data-w="1061" src="https://mmbiz.qpic.cn/mmbiz_png/PsnauFhHibc5IgrnANu8GpqP6TXajaybAZpwfCYknCTkP6CZOrNHjwKHmenFv6AGeyKHkecp0yV0E7l6TQHqicLQ/640?wx_fmt=png"></figure><p data-tool="mdnice编辑器">而对于产生的假阳性率情况，作者建议两种方法都可以尝试下</p><p data-tool="mdnice编辑器">For heterogeneous patient samples composed of multiple subtypes, it thus appears that Unsupervised and Supervised analyses can offer complementary merits, with Unsupervised analysis displaying a higher false negative rate, but a lower false positive rate. It is recommended to run both Supervised and Unsupervised analyses, as we demonstrated here, to gain maximum insight.</p><h4 data-tool="mdnice编辑器"><span></span>10.2 能分析非肿瘤数据吗？<span></span></h4><p data-tool="mdnice编辑器">虽然文章中测评用的都是肿瘤数据，并且能搭配TCGABiolinks一起使用。我觉得还是可以用的，很多数据挖掘类型R包都可以通用，可能会有局限，可以搜索下有没有类似文章</p><p data-tool="mdnice编辑器">we previously developed an R/Bioconductor package ELMER (Enhancer Linking by Methylation/Expression Relationships) to infer regulatory element landscapes and GRNs from cancer methylomes</p></section><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/COMXQ1vlW_CjK5mCWSgJ3A",target="_blank" rel="noopener noreferrer">原文链接</a>
