---
title: "这个bulk RNA-seq反卷积工具，你可能还不知道"
date: 2023-01-16T05:20:41Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Acdemic"]
---
这个bulk RNA-seq反卷积工具，你可能还不知道 by 生信菜鸟团
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><h1 data-tool="mdnice编辑器">引入</h1><p data-tool="mdnice编辑器">今天分享的是一个bulk RNA-seq的反卷积工具 BayesPrism。</p><blockquote data-tool="mdnice编辑器"><p>Chu, T., Wang, Z., Pe’er, D. <em>et al.</em> Cell type and gene expression deconvolution with BayesPrism enables Bayesian integrative analysis across bulk and single-cell RNA sequencing in oncology. <em>Nat Cancer</em> <strong>3</strong>, 505–517 (2022). https://doi.org/10.1038/s43018-022-00356-3</p></blockquote><p data-tool="mdnice编辑器">最初看到这个工具是在BioArt 公众号上（https://mp.weixin.qq.com/s/bQe41r0a20DI1_cFf_hn9A），当时并没有过多深入了解，对其印象局限于是一个借助于单细胞分群来推断Bulk RNAseq 细胞分群的工具。近期，健明老师突然提到这个工具，也促使我对这个R包感兴趣起来。就反卷积算法而言，正像之前那篇推文中提到的，以往算法存在局限性：</p><blockquote data-tool="mdnice编辑器"><p>第一，由于单细胞和bulk 的RNA-seq通常基于不同的实验方法和测序平台，两者之间存在较大的技术性批次效应（technical batch effects）；第二，由于肿瘤异质性，不同样本间的基因表达差异较大，单细胞的参考矩阵与实际bulk RNA-seq中的基因表达往往存在生物学差异（biological variation）。</p></blockquote><p data-tool="mdnice编辑器">而本篇将要介绍的BayesPrism（贝叶斯棱镜）原理在该推文中概括如下：</p><blockquote data-tool="mdnice编辑器"><p>贝叶斯棱镜仅把参考矩阵作为先验信息（prior information），在给定bulk RNA-seq数据后对该bulk RNA-seq中实际的参考矩阵进行建模。贝叶斯棱镜求得</p><p><span>1）每一例bulk样本中的每种细胞的百分比和 2）每种细胞的每个基因的reads数的联合后验概率分布。通过吉布斯采样（Gibbs sampling），可对1和2求得各自的边际（marginal）分布</span><strong>（图1）</strong><span>。由于对bulk RNA-seq中实际的参考矩阵进行建模并将其边际化，贝叶斯棱镜能对抗单细胞RNA-seq提取出的参考矩阵与bulk RNA-seq中的实际的参考矩阵的差别（robustness特性）。</span></p></blockquote><p data-tool="mdnice编辑器"><img data-ratio="0.4097560975609756" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibxgbLQib3sW8AIFqIFiaBbXWhYUvzYIFiagPGibzvPVEIcQPSFhb3pMu9vuNg9X7SxO7ebAzlzOkrXmw/640?wx_fmt=png" data-type="png" data-w="1230" title="图1.  流程图" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibxgbLQib3sW8AIFqIFiaBbXWhYUvzYIFiagPGibzvPVEIcQPSFhb3pMu9vuNg9X7SxO7ebAzlzOkrXmw/640?wx_fmt=png"><span>图1.  流程图</span></p><p data-tool="mdnice编辑器">原理大致如此，那么算法实现该当何如？</p><h1 data-tool="mdnice编辑器"><span></span>BayesPrism 安装与使用</h1><h2 data-tool="mdnice编辑器"><span> </span><span></span><span>安装</span><span></span><span> </span></h2><p data-tool="mdnice编辑器">在作者的论文末尾，已经提供了BayesPrism github 的地址  https://github.com/Danko-Lab/BayesPrism.git.。正如以往github 包的安装方法,其也是通过<code>devtools</code>安装</p><pre data-tool="mdnice编辑器"><span></span><code>library("devtools");<br>install_github("Danko-Lab/BayesPrism/BayesPrism")<br></code></pre><p data-tool="mdnice编辑器">如果说你在服务器上安装也遇到了我这样的安装问题（图2）</p><p data-tool="mdnice编辑器"><img data-ratio="0.3448275862068966" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibxgbLQib3sW8AIFqIFiaBbXWDjr6jOymdicr9U9wUTSSU4HMW6J3iawdB3frib13JDzYDxqTw53Q9Nbsg/640?wx_fmt=png" data-type="png" data-w="754" title="图2. devtools 安装报错" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibxgbLQib3sW8AIFqIFiaBbXWDjr6jOymdicr9U9wUTSSU4HMW6J3iawdB3frib13JDzYDxqTw53Q9Nbsg/640?wx_fmt=png"><span>图2. devtools 安装报错</span></p><p data-tool="mdnice编辑器">推荐解决方式是github 页面依次点击：Code → Download Zip → 上传服务器 → 解压 zip → 找到下级目录<code>[BayesPrism](https://github.com/Danko-Lab/BayesPrism/tree/main/BayesPrism)</code>→ 对其进行 Zip 压缩 → 以下代码：</p><pre data-tool="mdnice编辑器"><span></span><code>devtools::install_local('./BayesPrism-main/BayesPrism.zip')<br></code></pre><p data-tool="mdnice编辑器">一般情况下也不会遇到这种问题，当然如果说你要是连github也进不去，那……我也没办法。</p><h2 data-tool="mdnice编辑器"><span> </span><span></span><span>使用</span><span></span><span> </span></h2><p data-tool="mdnice编辑器">在BayesPrism的github 文档中提供了演示数据，如果是<code>devtools</code> 安装的，则需要自行下载演示数据。如果是下载zip 文件后的，在你的zip 文件夹解压后也能找到该演示数据。以下我将使用该演示数据进行探索BayesPrism</p><h3 data-tool="mdnice编辑器"><span></span>导入演示数据<span></span></h3><p data-tool="mdnice编辑器">作者这里给定的Bulk RNAseq 数据来源于 TCGA-GBM,  scRNA-seq 来源于Yuan的scRNA-seq数据集（https://www.yuque.com/u21872605/nils1z/dbsrgqum7g00db5o）。</p><pre data-tool="mdnice编辑器"><span></span><code>suppressWarnings(library(BayesPrism))<br><span>#</span><span>&gt; Loading required package: snowfall</span><br><span>#</span><span>&gt; Loading required package: snow</span><br><span>#</span><span>&gt; Loading required package: NMF</span><br><span>#</span><span>&gt; Loading required package: registry</span><br><span>#</span><span>&gt; Loading required package: rngtools</span><br><span>#</span><span>&gt; Loading required package: cluster</span><br><span>#</span><span>&gt; NMF - BioConductor layer [OK] | Shared memory capabilities [NO: bigmemory] | Cores 127/128</span><br><span>#</span><span>&gt;   To <span>enable</span> shared memory capabilities, try: install.extras(<span>'</span></span><br><span>#</span><span><span>&gt; NMF</span></span><br><span>#</span><span><span>&gt; '</span>)</span><br>load("BayesPrism-main/tutorial.dat/tutorial.gbm.rdata")<br>ls()<br><span>#</span><span>&gt; [1] <span>"bk.dat"</span>            <span>"cell.state.labels"</span> <span>"cell.type.labels"</span>  <span>"sc.dat"</span> </span><br></code></pre><p data-tool="mdnice编辑器">rdata 文件包含运行 BayesPrism 所需的四个对象。</p><blockquote data-tool="mdnice编辑器"><ul><li><section><code>bk.dat：</code>批量 RNA-seq 表达的逐个基因原始计数矩阵。<code>rownames</code>是批量样本 ID，而 <code>colnames</code>是基因名称/ID。</section></li></ul></blockquote><pre data-tool="mdnice编辑器"><span></span><code>dim(bk.dat)<br><span>#</span><span>&gt; [1]   169 60483</span><br>head(rownames(bk.dat))<br><span>#</span><span>&gt; [1] <span>"TCGA-06-2563-01A-01R-1849-01"</span> <span>"TCGA-06-0749-01A-01R-1849-01"</span> <span>"TCGA-06-5418-01A-01R-1849-01"</span> <span>"TCGA-06-0211-01B-01R-1849-01"</span></span><br><span>#</span><span>&gt; [5] <span>"TCGA-19-2625-01A-01R-1850-01"</span> <span>"TCGA-19-4065-02A-11R-2005-01"</span></span><br>head(colnames(bk.dat))<br><span>#</span><span>&gt; [1] <span>"ENSG00000000003.13"</span> <span>"ENSG00000000005.5"</span>  <span>"ENSG00000000419.11"</span> <span>"ENSG00000000457.12"</span> <span>"ENSG00000000460.15"</span> <span>"ENSG00000000938.11"</span></span><br></code></pre><blockquote data-tool="mdnice编辑器"><ul><li><section><code>sc.dat</code>: 大量 RNA-seq 表达的逐个基因原始计数矩阵。<code>rownames</code>是批量细胞 ID，而 <code>colnames</code>是基因名称/ID。</section></li></ul></blockquote><pre data-tool="mdnice编辑器"><span></span><code>dim(sc.dat)<br><span>#</span><span>&gt; [1] 23793 60294</span><br>head(rownames(sc.dat))<br><span>#</span><span>&gt; [1] <span>"PJ016.V3"</span> <span>"PJ016.V4"</span> <span>"PJ016.V5"</span> <span>"PJ016.V6"</span> <span>"PJ016.V7"</span> <span>"PJ016.V8"</span></span><br>head(colnames(sc.dat))<br><span>#</span><span>&gt; [1] <span>"ENSG00000130876.10"</span> <span>"ENSG00000134438.9"</span>  <span>"ENSG00000269696.1"</span>   <span>"ENSG00000230393.1"</span>  <span>"ENSG00000266744.1"</span>  <span>"ENSG00000202281.1"</span> </span><br></code></pre><blockquote data-tool="mdnice编辑器"><ul><li><section><code>cell.type.labels</code>是一个字符向量，其长度与<code>nrow(sc.dat)</code>表示参考中每个单元格的单元格类型相同。</section></li></ul></blockquote><pre data-tool="mdnice编辑器"><span></span><code>sort(table(cell.type.labels))<br><span>#</span><span>&gt; cell.type.labels</span><br><span>#</span><span>&gt;       tcell       oligo    pericyte endothelial     myeloid       tumor </span><br><span>#</span><span>&gt;          67         160         489         492        2505       20080</span><br></code></pre><blockquote data-tool="mdnice编辑器"><ul><li><section><code>cell.state.labels</code>是与表示参考中每个细胞的细胞状态相同长度的特征向量。</section></li></ul></blockquote><pre data-tool="mdnice编辑器"><span></span><code>sort(table(cell.state.labels))<br><span>#</span><span>&gt; cell.state.labels</span><br><span>#</span><span>&gt; PJ017-tumor-6 PJ032-tumor-5     myeloid_8 PJ032-tumor-4 PJ032-tumor-3 </span><br><span>#</span><span>&gt;            22            41            49            57            62 </span><br><span>#</span><span>&gt; PJ017-tumor-5         tcell PJ032-tumor-2 PJ030-tumor-5     myeloid_7 </span><br><span>#</span><span>&gt;            64            67            72            73            75 </span><br><span>#</span><span>&gt; PJ035-tumor-8 PJ017-tumor-4 PJ017-tumor-3 PJ017-tumor-2 PJ017-tumor-0 </span><br><span>#</span><span>&gt;            81            83            89           101           107 </span><br><span>#</span><span>&gt; PJ017-tumor-1 PJ018-tumor-5     myeloid_6     myeloid_5 PJ035-tumor-7 </span><br><span>#</span><span>&gt;           107           113           130           141           150 </span><br><span>#</span><span>&gt;         oligo PJ048-tumor-8 PJ032-tumor-1 PJ032-tumor-0 PJ048-tumor-7 </span><br><span>#</span><span>&gt;           160           169           171           195           228 </span><br><span>#</span><span>&gt; PJ025-tumor-9 PJ035-tumor-6 PJ048-tumor-6 PJ016-tumor-6 PJ018-tumor-4 </span><br><span>#</span><span>&gt;           236           241           244           261           262 </span><br><span>#</span><span>&gt;     myeloid_4 PJ048-tumor-5 PJ048-tumor-4 PJ016-tumor-5 PJ025-tumor-8 </span><br><span>#</span><span>&gt;           266           277           303           308           319 </span><br><span>#</span><span>&gt; PJ048-tumor-3 PJ035-tumor-5 PJ030-tumor-4 PJ018-tumor-3 PJ016-tumor-4 </span><br><span>#</span><span>&gt;           333           334           348           361           375 </span><br><span>#</span><span>&gt; PJ025-tumor-7     myeloid_3 PJ035-tumor-4     myeloid_2 PJ025-tumor-6 </span><br><span>#</span><span>&gt;           381           382           385           386           397 </span><br><span>#</span><span>&gt; PJ018-tumor-2 PJ030-tumor-3 PJ016-tumor-3 PJ025-tumor-5 PJ030-tumor-2 </span><br><span>#</span><span>&gt;           403           419           420           421           425 </span><br><span>#</span><span>&gt; PJ018-tumor-1 PJ035-tumor-3 PJ048-tumor-2 PJ018-tumor-0 PJ048-tumor-1 </span><br><span>#</span><span>&gt;           429           435           437           444           463 </span><br><span>#</span><span>&gt; PJ035-tumor-2 PJ035-tumor-1 PJ016-tumor-2 PJ030-tumor-1      pericyte </span><br><span>#</span><span>&gt;           471           474           481           482           489 </span><br><span>#</span><span>&gt;   endothelial PJ035-tumor-0 PJ025-tumor-4     myeloid_1 PJ048-tumor-0 </span><br><span>#</span><span>&gt;           492           512           523           526           545 </span><br><span>#</span><span>&gt;     myeloid_0 PJ030-tumor-0 PJ025-tumor-3 PJ016-tumor-1 PJ016-tumor-0 </span><br><span>#</span><span>&gt;           550           563           601           619           621 </span><br><span>#</span><span>&gt; PJ025-tumor-2 PJ025-tumor-1 PJ025-tumor-0 </span><br><span>#</span><span>&gt;           630           941           971</span><br></code></pre><p data-tool="mdnice编辑器"> table 看一下<code>cell.type.labels</code>和<code>cell.state.labels</code>的关系。</p><pre data-tool="mdnice编辑器"><span></span><code>table(cbind.data.frame(cell.state.labels, cell.type.labels))<br><span>#</span><span>&gt;                  cell.type.labels</span><br><span>#</span><span>&gt; cell.state.labels endothelial myeloid oligo pericyte tcell tumor</span><br><span>#</span><span>&gt;     endothelial           492       0     0        0     0     0</span><br><span>#</span><span>&gt;     myeloid_0               0     550     0        0     0     0</span><br><span>#</span><span>&gt;     myeloid_1               0     526     0        0     0     0</span><br><span>#</span><span>&gt;     myeloid_2               0     386     0        0     0     0</span><br><span>#</span><span>&gt;     myeloid_3               0     382     0        0     0     0</span><br><span>#</span><span>&gt;     myeloid_4               0     266     0        0     0     0</span><br><span>#</span><span>&gt;     myeloid_5               0     141     0        0     0     0</span><br><span>#</span><span>&gt;     myeloid_6               0     130     0        0     0     0</span><br><span>#</span><span>&gt;     myeloid_7               0      75     0        0     0     0</span><br><span>#</span><span>&gt;     myeloid_8               0      49     0        0     0     0</span><br><span>#</span><span>&gt;     oligo                   0       0   160        0     0     0</span><br><span>#</span><span>&gt;     pericyte                0       0     0      489     0     0</span><br><span>#</span><span>&gt;     PJ016-tumor-0           0       0     0        0     0   621</span><br><span>#</span><span>&gt;     PJ016-tumor-1           0       0     0        0     0   619</span><br><span>#</span><span>&gt;     PJ016-tumor-2           0       0     0        0     0   481</span><br><span>#</span><span>&gt;     PJ016-tumor-3           0       0     0        0     0   420</span><br><span>#</span><span>&gt;     PJ016-tumor-4           0       0     0        0     0   375</span><br><span>#</span><span>&gt;     PJ016-tumor-5           0       0     0        0     0   308</span><br><span>#</span><span>&gt;     PJ016-tumor-6           0       0     0        0     0   261</span><br><span>#</span><span>&gt;     PJ017-tumor-0           0       0     0        0     0   107</span><br><span>#</span><span>&gt;     PJ017-tumor-1           0       0     0        0     0   107</span><br><span>#</span><span>&gt;     PJ017-tumor-2           0       0     0        0     0   101</span><br><span>#</span><span>&gt;     PJ017-tumor-3           0       0     0        0     0    89</span><br><span>#</span><span>&gt;     PJ017-tumor-4           0       0     0        0     0    83</span><br><span>#</span><span>&gt;     PJ017-tumor-5           0       0     0        0     0    64</span><br><span>#</span><span>&gt;     PJ017-tumor-6           0       0     0        0     0    22</span><br><span>#</span><span>&gt;     PJ018-tumor-0           0       0     0        0     0   444</span><br><span>#</span><span>&gt;     PJ018-tumor-1           0       0     0        0     0   429</span><br><span>#</span><span>&gt;     PJ018-tumor-2           0       0     0        0     0   403</span><br><span>#</span><span>&gt;     PJ018-tumor-3           0       0     0        0     0   361</span><br><span>#</span><span>&gt;     PJ018-tumor-4           0       0     0        0     0   262</span><br><span>#</span><span>&gt;     PJ018-tumor-5           0       0     0        0     0   113</span><br><span>#</span><span>&gt;     PJ025-tumor-0           0       0     0        0     0   971</span><br><span>#</span><span>&gt;     PJ025-tumor-1           0       0     0        0     0   941</span><br><span>#</span><span>&gt;     PJ025-tumor-2           0       0     0        0     0   630</span><br><span>#</span><span>&gt;     PJ025-tumor-3           0       0     0        0     0   601</span><br><span>#</span><span>&gt;     PJ025-tumor-4           0       0     0        0     0   523</span><br><span>#</span><span>&gt;     PJ025-tumor-5           0       0     0        0     0   421</span><br><span>#</span><span>&gt;     PJ025-tumor-6           0       0     0        0     0   397</span><br><span>#</span><span>&gt;     PJ025-tumor-7           0       0     0        0     0   381</span><br><span>#</span><span>&gt;     PJ025-tumor-8           0       0     0        0     0   319</span><br><span>#</span><span>&gt;     PJ025-tumor-9           0       0     0        0     0   236</span><br><span>#</span><span>&gt;     PJ030-tumor-0           0       0     0        0     0   563</span><br><span>#</span><span>&gt;     PJ030-tumor-1           0       0     0        0     0   482</span><br><span>#</span><span>&gt;     PJ030-tumor-2           0       0     0        0     0   425</span><br><span>#</span><span>&gt;     PJ030-tumor-3           0       0     0        0     0   419</span><br><span>#</span><span>&gt;     PJ030-tumor-4           0       0     0        0     0   348</span><br><span>#</span><span>&gt;     PJ030-tumor-5           0       0     0        0     0    73</span><br><span>#</span><span>&gt;     PJ032-tumor-0           0       0     0        0     0   195</span><br><span>#</span><span>&gt;     PJ032-tumor-1           0       0     0        0     0   171</span><br><span>#</span><span>&gt;     PJ032-tumor-2           0       0     0        0     0    72</span><br><span>#</span><span>&gt;     PJ032-tumor-3           0       0     0        0     0    62</span><br><span>#</span><span>&gt;     PJ032-tumor-4           0       0     0        0     0    57</span><br><span>#</span><span>&gt;     PJ032-tumor-5           0       0     0        0     0    41</span><br><span>#</span><span>&gt;     PJ035-tumor-0           0       0     0        0     0   512</span><br><span>#</span><span>&gt;     PJ035-tumor-1           0       0     0        0     0   474</span><br><span>#</span><span>&gt;     PJ035-tumor-2           0       0     0        0     0   471</span><br><span>#</span><span>&gt;     PJ035-tumor-3           0       0     0        0     0   435</span><br><span>#</span><span>&gt;     PJ035-tumor-4           0       0     0        0     0   385</span><br><span>#</span><span>&gt;     PJ035-tumor-5           0       0     0        0     0   334</span><br><span>#</span><span>&gt;     PJ035-tumor-6           0       0     0        0     0   241</span><br><span>#</span><span>&gt;     PJ035-tumor-7           0       0     0        0     0   150</span><br><span>#</span><span>&gt;     PJ035-tumor-8           0       0     0        0     0    81</span><br><span>#</span><span>&gt;     PJ048-tumor-0           0       0     0        0     0   545</span><br><span>#</span><span>&gt;     PJ048-tumor-1           0       0     0        0     0   463</span><br><span>#</span><span>&gt;     PJ048-tumor-2           0       0     0        0     0   437</span><br><span>#</span><span>&gt;     PJ048-tumor-3           0       0     0        0     0   333</span><br><span>#</span><span>&gt;     PJ048-tumor-4           0       0     0        0     0   303</span><br><span>#</span><span>&gt;     PJ048-tumor-5           0       0     0        0     0   277</span><br><span>#</span><span>&gt;     PJ048-tumor-6           0       0     0        0     0   244</span><br><span>#</span><span>&gt;     PJ048-tumor-7           0       0     0        0     0   228</span><br><span>#</span><span>&gt;     PJ048-tumor-8           0       0     0        0     0   169</span><br><span>#</span><span>&gt;     tcell                   0       0     0        0    67     0</span><br></code></pre><p data-tool="mdnice编辑器">注：</p><blockquote data-tool="mdnice编辑器"><ol><li><section>BayesPrism 将对<code>mixture</code>(代指bulk RNA-seq)和 <code>reference</code> （代指scRNA-seq）之间共享的基因执行反卷积，即通过交叉 <code>colnames(mixture) </code>和<code> colnames(reference)</code>。因此，<strong>请确保使用一致的基因注释</strong>（TCGA RNA-seq 使用 GENCODE v22）。</section></li><li><section><strong>建议使用未规范化和未转换（ unnormalized and untransformed）的count data。</strong>当原始计数不可用时，线性归一化（例如 TPM、RPM、RPKM、FPKM）也是可以接受的，因为 BayesPrism 对<code>reference</code>和<code>mixture</code>之间的线性乘法差异具有鲁棒性（换个单词 即Robust）。理想情况下，如果使用规范化数据，最好提供参考数据和通过相同方法转换的批量数据。应避免log转换。</section></li><li><section>所有细胞群应具有一定的细胞数量，例如 &gt;20 或 &gt;50。</section></li></ol></blockquote><p data-tool="mdnice编辑器">在这里，关于<code>cell.type.labels</code>和<code>cell.state.labels</code>我个人理解是后者可以是前者的亚群，简单来说，<code>cell.type.labels</code>是第一步分群的结果，<code>cell.state.labels</code>是在第一步分群的基础上，进一步根据群内异质性继续细分亚群的结果。作者对<code>cell.type.labels</code>和<code>cell.state.labels</code>的描述（翻译 By 彩云小译）：</p><blockquote data-tool="mdnice编辑器"><ul><li><section>细胞类型和细胞状态的定义可能有些随意（类似于为 scRNA-seq 分配细胞类型的问题）并且取决于感兴趣的问题。它们的定义取决于我们关注的细胞（the granularity we aim at ）和 scRNA-seq 数据中<code>cell.type.labels</code> 的置信度。通常，一个好的经验法则如下。</section></li><li><section><strong>1)将细胞类型定义为比其他细胞类型具有足够数量的显著差异表达基因的细胞群，</strong>例如大于50甚至100个。对于转录过于相似的簇，我们建议将它们视为细胞状态，在最终的吉布斯取样之前将对它们进行总结。因此，细胞状态通常适合于在表型流形上形成连续统的细胞，而不是不同的簇。</section></li><li><section><strong>2)为显著异质性的细胞类型(如恶性细胞)定义多种细胞状态，并对其转录进行解卷积</strong>。</section></li></ul></blockquote><h3 data-tool="mdnice编辑器"><span></span>细胞类型和状态标签的 QC<span></span></h3><p data-tool="mdnice编辑器">作者这一块说的很复杂，其实简单来说，<code>cell.type.labels</code>和<code>cell.state.labels</code>要有足够的区分度，如果区分度不够或者过了，那么各个群在聚类图中，可能会聚集在一起。如果出现大部分cluster 具有明显的相似性，可以考虑merge，或者以更低的resolution进行降维聚类（或者说higher granularity）。如果重新聚类或者merge 不合适的话，也可考虑删除，因此，作者建议，首先使用<code>plot.cor.phi</code>函数查看<code>cell.type.labels</code>和<code>cell.state.labels</code>的特征</p><pre data-tool="mdnice编辑器"><span></span><code>plot.cor.phi (input=sc.dat,<br>              input.labels=cell.state.labels,<br>              title="cell state correlation",<br>              #specify pdf.prefix if need to output to pdf<br>              #pdf.prefix="gbm.cor.cs", <br>              cexRow=0.2, cexCol=0.2,<br>              margins=c(2,2))<br></code></pre><figure data-tool="mdnice编辑器"><ne-p data-lake-id="ufbb3d89d"><ne-card data-card-name="image" data-card-type="inline" data-event-boundary="card"><p><br></p></ne-card></ne-p><p><ne-hole data-lake-id="u29180897"><ne-card data-card-name="codeblock" data-card-type="block" data-event-boundary="card"></ne-card></ne-hole></p><figcaption><img data-ratio="0.7015781922525107" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibxgbLQib3sW8AIFqIFiaBbXWC8x8mTYFPVBCibQhpzZFYoup2feC5yyus8VkFaleqOBMc7ZeHMEicpOg/640?wx_fmt=png" data-type="png" data-w="1394" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibxgbLQib3sW8AIFqIFiaBbXWC8x8mTYFPVBCibQhpzZFYoup2feC5yyus8VkFaleqOBMc7ZeHMEicpOg/640?wx_fmt=png"></figcaption><figcaption><span>图3.  cell.state.labels  plot</span></figcaption><figcaption><span><br></span></figcaption></figure><pre data-tool="mdnice编辑器"><span></span><code>plot.cor.phi (input=sc.dat, <br>              input.labels=cell.type.labels, <br>              title="cell type correlation",<br>              #specify pdf.prefix if need to output to pdf<br>              #pdf.prefix="gbm.cor.ct",<br>              cexRow=0.5, cexCol=0.5,<br>)</code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7459016393442623" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibxgbLQib3sW8AIFqIFiaBbXWZvZTECkTRpp9XIFjs8tOGXg5llmzFo32HVORGyicDGOr2bp2XeIYmXg/640?wx_fmt=png" data-type="png" data-w="1098" title="图4. cell.type.labels  plot" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibxgbLQib3sW8AIFqIFiaBbXWZvZTECkTRpp9XIFjs8tOGXg5llmzFo32HVORGyicDGOr2bp2XeIYmXg/640?wx_fmt=png"><figcaption><span>图4. cell.type.labels  plot</span></figcaption></figure><h3 data-tool="mdnice编辑器"><span></span>过滤离群基因<span></span></h3><p data-tool="mdnice编辑器">在单细胞测序中，大量的核糖体基因或者线粒体基因不参与指导分析细胞功能，但其可能主导分布并使推断产生偏差。这些基因在区分细胞类型方面通常不能提供信息，并且可能是大的假变异的来源。因此，它们对反卷积是不利的。因此，有必要移除这些基因。</p><ul data-tool="mdnice编辑器"><li><section>用户可以从 scRNA-seq 参考可视化异常基因的分布。计算所有细胞类型的平均表达和每个基因的表达，以及它们的细胞类型特异性评分。</section></li><li><section>从 scRNA-seq 数据中可视化并确定异常基因</section></li></ul><pre data-tool="mdnice编辑器"><span></span><code><span>#</span><span>查看单细胞数据的离群基因</span><br>sc.stat &lt;- plot.scRNA.outlier(<br>  input=sc.dat, #make sure the colnames are gene symbol or ENSMEBL ID <br>  cell.type.labels=cell.type.labels,<br>  species="hs", #currently only human(hs) and mouse(mm) annotations are supported<br>  return.raw=TRUE #return the data used for plotting. <br><span>  #</span><span>pdf.prefix=<span>"gbm.sc.stat"</span> specify pdf.prefix <span>if</span> need to output to pdf</span><br>)<br>head(sc.stat)  <br><span>#</span><span>&gt;                    exp.mean.log  max.spec other_Rb  chrM  chrX  chrY    Rb   Mrp   act    hb MALAT1</span><br><span>#</span><span>&gt; ENSG00000130876.10    -13.59828 0.8473029    FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE  FALSE</span><br><span>#</span><span>&gt; ENSG00000134438.9     -16.08255 0.8914740    FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE  FALSE</span><br><span>#</span><span>&gt; ENSG00000269696.1     -13.12002 0.7509754    FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE  FALSE</span><br><span>#</span><span>&gt; ENSG00000230393.1     -13.97772 0.5654292    FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE  FALSE</span><br><span>#</span><span>&gt; ENSG00000266744.1     -17.96174 0.4733715    FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE  FALSE</span><br><span>#</span><span>&gt; ENSG00000202281.1     -18.42068 0.1666667    FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE  FALSE</span><br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.6557971014492754" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibxgbLQib3sW8AIFqIFiaBbXWhBEpuYNC1PcY6ljElvibBj3MZ1rtwWy4tYUp5lk7ktIWLsdqXke42MQ/640?wx_fmt=png" data-type="png" data-w="1104" title="图5.  查看单细胞数据的离群基因" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibxgbLQib3sW8AIFqIFiaBbXWhBEpuYNC1PcY6ljElvibBj3MZ1rtwWy4tYUp5lk7ktIWLsdqXke42MQ/640?wx_fmt=png"><figcaption><span>图5.  查看单细胞数据的离群基因</span></figcaption></figure><blockquote data-tool="mdnice编辑器"><p>注：1. 核糖体蛋白质基因通常表现出高平均表达和低细胞类型特异性得分(局限在右下角)。</p><ol><li><section><code>sc.stat </code>显示每个基因的归一化平均表达(x 轴)和最大特异性(y 轴)的对数，以及每个基因是否属于一个潜在的异常值类别。Other _ Rb 是由核糖体假基因组成的一组精选基因。</section></li></ol></blockquote><pre data-tool="mdnice编辑器"><span></span><code>bk.stat &lt;- plot.bulk.outlier(<br>  bulk.input=bk.dat,#make sure the colnames are gene symbol or ENSMEBL ID <br>  sc.input=sc.dat, #make sure the colnames are gene symbol or ENSMEBL ID <br>  cell.type.labels=cell.type.labels,<br>  species="hs", #currently only human(hs) and mouse(mm) annotations are supported<br>  return.raw=TRUE<br><span>  #</span><span>pdf.prefix=<span>"gbm.bk.stat"</span> specify pdf.prefix <span>if</span> need to output to pdf</span><br>)<br>head(bk.stat)<br><span>#</span><span>&gt;                    exp.mean.log  max.spec other_Rb  chrM  chrX  chrY    Rb   Mrp   act    hb MALAT1</span><br><span>#</span><span>&gt; ENSG00000000003.13    -9.263877 0.4825322    FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE  FALSE</span><br><span>#</span><span>&gt; ENSG00000000005.5    -13.336689 0.9736083    FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE  FALSE</span><br><span>#</span><span>&gt; ENSG00000000419.11   -10.455993 0.2275788    FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE  FALSE</span><br><span>#</span><span>&gt; ENSG00000000457.12   -11.298589 0.2169448    FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE  FALSE</span><br><span>#</span><span>&gt; ENSG00000000460.15   -11.648474 0.2920256    FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE  FALSE</span><br><span>#</span><span>&gt; ENSG00000000938.11   -11.152720 0.7347018    FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE  FALSE</span><br></code></pre><p data-tool="mdnice编辑器"><img data-ratio="0.7459016393442623" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibxgbLQib3sW8AIFqIFiaBbXWRPicmicykko0w4VX2LHb3fCvAFnZgx9oqoj0jcQtF5E8Of9d4A1ThdYA/640?wx_fmt=png" data-type="png" data-w="1098" title="图6. 查看Bulk RNAseq的离群基因" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibxgbLQib3sW8AIFqIFiaBbXWRPicmicykko0w4VX2LHb3fCvAFnZgx9oqoj0jcQtF5E8Of9d4A1ThdYA/640?wx_fmt=png"><span>图6. 查看Bulk RNAseq的离群基因</span></p><p data-tool="mdnice编辑器">查看完之后，将从选定的群体中移除基因。注意，当参考基因和混合基因的性别不同时，建议排除 chrX 和 chrY 基因。也删除低转录基因，因为测量这些基因的转录倾向于噪声。去除这些基因也可以加快计算速度。</p><pre data-tool="mdnice编辑器"><span></span><code>sc.dat.filtered &lt;- cleanup.genes (input=sc.dat,<br>                                   input.type="count.matrix",<br>                                   species="hs", <br>                                   gene.group=c( "Rb","Mrp","other_Rb","chrM","MALAT1","chrX","chrY") ,<br>                                   exp.cells=5)<br><span>#</span><span>&gt; EMSEMBLE IDs detected.</span><br><span>#</span><span>&gt; number of genes filtered <span>in</span> each category: </span><br><span>#</span><span>&gt;       Rb      Mrp other_Rb     chrM   MALAT1     chrX     chrY </span><br><span>#</span><span>&gt;       89       78     1011       37        1     2464      594 </span><br><span>#</span><span>&gt; A total of  4214  genes from Rb Mrp other_Rb chrM MALAT1 chrX chrY  have been excluded </span><br><span>#</span><span>&gt; A total of  24343  gene expressed <span>in</span> fewer than  5  cells have been excluded </span><br>dim(sc.dat.filtered)<br><span>#</span><span>&gt; [1] 23793 31737</span><br></code></pre><p data-tool="mdnice编辑器">检查不同类型基因的基因表达的一致性。由于bulk RNA-seq 和scRNAseq data 通常由不同的实验方案收集，它们可能对不同类型的基因具有不同的敏感性。</p><pre data-tool="mdnice编辑器"><span></span><code>plot.bulk.vs.sc (sc.input = sc.dat.filtered,<br>                 bulk.input = bk.dat<br>                 #pdf.prefix="gbm.bk.vs.sc" specify pdf.prefix if need to output to pdf<br>)<br><span>#</span><span>&gt; EMSEMBLE IDs detected.</span><br></code></pre><p data-tool="mdnice编辑器"><img data-croporisrc="https://mmbiz.qlogo.cn/mmbiz_png/iaRJcrq2LosibxgbLQib3sW8AIFqIFiaBbXWt0bzFan8ibaDpgzBZCicy1d1kcOh3w0dc293tuH0rmG7tpicOdOL7DxFw/0?wx_fmt=png" data-cropx1="0" data-cropx2="1098" data-cropy1="253.8387096774194" data-cropy2="716.2580645161291" data-ratio="0.4207650273224044" data-src="https://mmbiz.qpic.cn/mmbiz_jpg/iaRJcrq2LosibxgbLQib3sW8AIFqIFiaBbXWiaptXj8oxJibibwJJ8oKnXicDnqVPjfhQgtWP4oT2IxWfGjk30vQtnvHibg/640?wx_fmt=jpeg" data-type="jpeg" data-w="1098" src="https://mmbiz.qpic.cn/mmbiz_jpg/iaRJcrq2LosibxgbLQib3sW8AIFqIFiaBbXWiaptXj8oxJibibwJJ8oKnXicDnqVPjfhQgtWP4oT2IxWfGjk30vQtnvHibg/640?wx_fmt=jpeg"></p><p data-tool="mdnice编辑器"><span>图</span><span>7.  log2 表达bulk 和scRNA相关性</span></p><p data-tool="mdnice编辑器">根据上图，可以发现：蛋白质编码基因是两个测定之间最一致的组。为了减少批量效应和加快计算速度，后续对蛋白质编码基因进行了反卷积处理。后续代码在所有基因上运行贝叶斯棱镜。二者结果是相似的。</p><ol data-tool="mdnice编辑器"><li><section>根据 “蛋白编码基因”确定子集。</section></li></ol><pre data-tool="mdnice编辑器"><span></span><code>sc.dat.filtered.pc &lt;-  select.gene.type (sc.dat.filtered,<br>                                          gene.type = "protein_coding")<br><span>#</span><span>&gt; EMSEMBLE IDs detected.</span><br><span>#</span><span>&gt; number of genes retained <span>in</span> each category: </span><br><span>#</span><span>&gt; </span><br><span>#</span><span>&gt; protein_coding </span><br><span>#</span><span>&gt;          16148</span><br></code></pre><ol data-tool="mdnice编辑器"><li><section>根据‘signature gene’确定子集</section></li></ol><p data-tool="mdnice编辑器">或者，在细胞类型被定义为其中一些显示非常相似的转录或存在严重的批次效应的情况下，例如，reference 和mixture 来自不同的测定手段，选择signature gene是可行的。这是因为特征基因的选择可以丰富用于反卷积的基因信息，同时减少技术批量效应造成的噪声影响。</p><pre data-tool="mdnice编辑器"><span></span><code><span>#</span><span> Select marker genes (Optional)</span><br><span>#</span><span> performing pair-wise t <span>test</span> <span>for</span> cell states from different cell types</span><br><br>diff.exp.stat &lt;- get.exp.stat(sc.dat=sc.dat[,colSums(sc.dat&gt;0)&gt;3],# filter genes to reduce memory use<br>                              cell.type.labels=cell.type.labels,<br>                              cell.state.labels=cell.state.labels,<br>                              psuedo.count=0.1, #a numeric value used for log2 transformation. =0.1 for 10x data, =10 for smart-seq. Default=0.1.<br>                              cell.count.cutoff=50, # a numeric value to exclude cell state with number of cells fewer than this value for t test. Default=50.<br>                              n.cores=1 #number of threads<br>)<br></code></pre><p data-tool="mdnice编辑器">理想情况下，每种细胞类型最好选择足够数量的基因(&gt; 50)。否则，可能会降低截止值。根据signature gene提取Count matrix subset ，可以运行以下代码</p><pre data-tool="mdnice编辑器"><span></span><code>sc.dat.filtered.pc.sig &lt;- select.marker (sc.dat=sc.dat.filtered.pc,<br>                                                  stat=diff.exp.stat,<br>                                                  pval.max=0.01,<br>                                                  lfc.min=0.1)<br><span>#</span><span>&gt; number of markers selected <span>for</span> each cell <span>type</span>: </span><br><span>#</span><span>&gt; tumor :  8686 </span><br><span>#</span><span>&gt; myeloid :  575 </span><br><span>#</span><span>&gt; pericyte :  114 </span><br><span>#</span><span>&gt; endothelial :  244 </span><br><span>#</span><span>&gt; tcell :  123 </span><br><span>#</span><span>&gt; oligo :  86</span><br>dim(sc.dat.filtered.pc.sig)<br><span>#</span><span>&gt; [1] 23793  7874</span><br></code></pre><p data-tool="mdnice编辑器">注：如果使用signature gene 运行BayesPrism ，则在调用 <code>new.prism </code>时使用  <code>reference = sc.dat.filtered.pc.sig</code> (见下文)。</p><h3 data-tool="mdnice编辑器"><span></span>构建Prism 对象<span></span></h3><p data-tool="mdnice编辑器">Prism对象包含运行 BayesPrism 所需的所有数据，即 scRNA-seq reference 矩阵、每行参考的 cell type and cell state labels 以及mixture（Bulk RNA-seq）</p><p data-tool="mdnice编辑器"><img data-ratio="0.9277620396600567" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibxgbLQib3sW8AIFqIFiaBbXWkWUyd5Ikp5qwCr7cmnnQeVAn2LVcXfuZQRVP4OXYjHZI2711SpUfOg/640?wx_fmt=png" data-type="png" data-w="706" title="图8. Prism 对象示意图" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibxgbLQib3sW8AIFqIFiaBbXWkWUyd5Ikp5qwCr7cmnnQeVAn2LVcXfuZQRVP4OXYjHZI2711SpUfOg/640?wx_fmt=png"><span>图8. Prism 对象示意图</span></p><p data-tool="mdnice编辑器">当使用 scRNA-seq  Count 矩阵作为input (推荐)时，用户需要指定 input.type = “ count. Matrix”。Type 的另一个选项是“ GEP”(gene expression profile) ，它是基因矩阵的细胞状态。当使用Bulk RNAseq的数据作为reference 时， 应设置input.type ='GEP'。参数<code>key</code>是 cell.type.label 中对应于恶性细胞类型的字符向量。如无恶性细胞，即设置为 NULL，此时，所有类型的细胞将被 treated equally。</p><pre data-tool="mdnice编辑器"><span></span><code>myPrism &lt;- new.prism(<br>   reference=sc.dat.filtered.pc, <br>   mixture=bk.dat,<br>   input.type="count.matrix", <br>   cell.type.labels = cell.type.labels, <br>   cell.state.labels = cell.state.labels,<br>   key="tumor",# <br>   outlier.cut=0.01,<br>   outlier.fraction=0.1,<br> )<br><span>#</span><span>&gt; number of cells <span>in</span> each cell state </span><br><span>#</span><span>&gt; cell.state.labels</span><br><span>#</span><span>&gt; PJ017-tumor-6 PJ032-tumor-5     myeloid_8 PJ032-tumor-4 PJ032-tumor-3 PJ017-tumor-5         tcell PJ032-tumor-2 PJ030-tumor-5     myeloid_7 </span><br><span>#</span><span>&gt;            22            41            49            57            62            64            67            72            73            75 </span><br><span>#</span><span>&gt; PJ035-tumor-8 PJ017-tumor-4 PJ017-tumor-3 PJ017-tumor-2 PJ017-tumor-0 PJ017-tumor-1 PJ018-tumor-5     myeloid_6     myeloid_5 PJ035-tumor-7 </span><br><span>#</span><span>&gt;            81            83            89           101           107           107           113           130           141           150 </span><br><span>#</span><span>&gt;         oligo PJ048-tumor-8 PJ032-tumor-1 PJ032-tumor-0 PJ048-tumor-7 PJ025-tumor-9 PJ035-tumor-6 PJ048-tumor-6 PJ016-tumor-6 PJ018-tumor-4 </span><br><span>#</span><span>&gt;           160           169           171           195           228           236           241           244           261           262 </span><br><span>#</span><span>&gt;     myeloid_4 PJ048-tumor-5 PJ048-tumor-4 PJ016-tumor-5 PJ025-tumor-8 PJ048-tumor-3 PJ035-tumor-5 PJ030-tumor-4 PJ018-tumor-3 PJ016-tumor-4 </span><br><span>#</span><span>&gt;           266           277           303           308           319           333           334           348           361           375 </span><br><span>#</span><span>&gt; PJ025-tumor-7     myeloid_3 PJ035-tumor-4     myeloid_2 PJ025-tumor-6 PJ018-tumor-2 PJ030-tumor-3 PJ016-tumor-3 PJ025-tumor-5 PJ030-tumor-2 </span><br><span>#</span><span>&gt;           381           382           385           386           397           403           419           420           421           425 </span><br><span>#</span><span>&gt; PJ018-tumor-1 PJ035-tumor-3 PJ048-tumor-2 PJ018-tumor-0 PJ048-tumor-1 PJ035-tumor-2 PJ035-tumor-1 PJ016-tumor-2 PJ030-tumor-1      pericyte </span><br><span>#</span><span>&gt;           429           435           437           444           463           471           474           481           482           489 </span><br><span>#</span><span>&gt;   endothelial PJ035-tumor-0 PJ025-tumor-4     myeloid_1 PJ048-tumor-0     myeloid_0 PJ030-tumor-0 PJ025-tumor-3 PJ016-tumor-1 PJ016-tumor-0 </span><br><span>#</span><span>&gt;           492           512           523           526           545           550           563           601           619           621 </span><br><span>#</span><span>&gt; PJ025-tumor-2 PJ025-tumor-1 PJ025-tumor-0 </span><br><span>#</span><span>&gt;           630           941           971 </span><br><span>#</span><span>&gt; Number of outlier genes filtered from mixture = 6 </span><br><span>#</span><span>&gt; Aligning reference and mixture... </span><br><span>#</span><span>&gt; Nornalizing reference...</span><br></code></pre><p data-tool="mdnice编辑器">注：</p><blockquote data-tool="mdnice编辑器"><ul><li><section>Note that outlier.cut and outlier.fraction=0.1 filter genes in X whose expression fraction is greater than outlier.cut (Default=0.01) in more than outlier.fraction (Default=0.1) of bulk data.</section></li><li><section>Typically for dataset with reasonable quality control, very few genes will be filtered.</section></li><li><section>Removal of outlier genes will ensure that the inference will not be dominated by outliers, which sometimes may be resulted from poor QC in mapping.</section></li></ul></blockquote><h3 data-tool="mdnice编辑器"><span></span>运行 BayesPrism<span></span></h3><p data-tool="mdnice编辑器"><img data-ratio="0.82" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibxgbLQib3sW8AIFqIFiaBbXWI20iaudwibvBeNHI9ibW5lHmIiaViapcbicZLOqotZh06CbkGSHIUdf1WyYg/640?wx_fmt=png" data-type="png" data-w="950" title="图9. BayesPrism 流程示意" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2LosibxgbLQib3sW8AIFqIFiaBbXWI20iaudwibvBeNHI9ibW5lHmIiaViapcbicZLOqotZh06CbkGSHIUdf1WyYg/640?wx_fmt=png"><span>图9. BayesPrism 流程示意</span></p><p data-tool="mdnice编辑器">后续即运行<code>BayesPrism</code>控制 Gibbs 采样和优化的参数可以使用 <code>Gibbs.control</code> 和 <code>opt.control</code> 指定。<code>?run.prism</code>获取details。这里建议使用默认参数。（代码就一句）</p><pre data-tool="mdnice编辑器"><span></span><code>bp.res &lt;- run.prism(prism = myPrism, n.cores=50)<br><span>#</span><span>&gt; Run Gibbs sampling... </span><br><span>#</span><span>&gt; Current time:  2023-01-08 22:46:55 </span><br><span>#</span><span>&gt; Estimated time to complete:  38mins </span><br><span>#</span><span>&gt; Estimated finishing time:  2023-01-08 23:24:44 </span><br><span>#</span><span>&gt; Start run... </span><br><span>#</span><span>&gt; R Version:  R version 4.2.0 (2022-04-22) </span><br><span><br>#</span><span>&gt; snowfall 1.84-6.2 initialized (using snow 0.4-4): parallel execution on 50 CPUs.</span><br><span><br><br>#</span><span>&gt; Stopping cluster</span><br><span><br>#</span><span>&gt; Update the reference matrix ... </span><br><span>#</span><span>&gt; snowfall 1.84-6.2 initialized (using snow 0.4-4): parallel execution on 50 CPUs.</span><br><span><br><br>#</span><span>&gt; Stopping cluster</span><br><span><br>#</span><span>&gt; Run Gibbs sampling using updated reference ... </span><br><span>#</span><span>&gt; Current time:  2023-01-08 23:08:48 </span><br><span>#</span><span>&gt; Estimated time to complete:  17mins </span><br><span>#</span><span>&gt; Estimated finishing time:  2023-01-08 23:24:50 </span><br><span>#</span><span>&gt; Start run... </span><br><span>#</span><span>&gt; snowfall 1.84-6.2 initialized (using snow 0.4-4): parallel execution on 50 CPUs.</span><br><span><br><br>#</span><span>&gt; Stopping cluster</span><br></code></pre><p data-tool="mdnice编辑器"><code>bp.res</code>即是BayesPrism的结果。后续探讨bp.res的结果及可视化的问题</p><pre data-tool="mdnice编辑器"><span></span><code>bp.res<br><span>#</span><span>&gt; Input prism info: </span><br><span>#</span><span>&gt; Cell states <span>in</span> each cell <span>type</span>: </span><br><span>#</span><span>&gt; <span>$tumor</span></span><br><span>#</span><span>&gt;  [1] <span>"PJ016-tumor-0"</span> <span>"PJ016-tumor-3"</span> <span>"PJ016-tumor-2"</span> <span>"PJ016-tumor-6"</span> <span>"PJ016-tumor-4"</span> <span>"PJ016-tumor-1"</span> <span>"PJ016-tumor-5"</span> <span>"PJ017-tumor-3"</span></span><br><span>#</span><span>&gt;  [9] <span>"PJ017-tumor-2"</span> <span>"PJ017-tumor-1"</span> <span>"PJ017-tumor-0"</span> <span>"PJ017-tumor-4"</span> <span>"PJ017-tumor-5"</span> <span>"PJ017-tumor-6"</span> <span>"PJ018-tumor-4"</span> <span>"PJ018-tumor-2"</span></span><br><span>#</span><span>&gt; [17] <span>"PJ018-tumor-3"</span> <span>"PJ018-tumor-1"</span> <span>"PJ018-tumor-0"</span> <span>"PJ018-tumor-5"</span> <span>"PJ025-tumor-9"</span> <span>"PJ025-tumor-2"</span> <span>"PJ025-tumor-4"</span> <span>"PJ025-tumor-3"</span></span><br><span>#</span><span>&gt; [25] <span>"PJ025-tumor-0"</span> <span>"PJ025-tumor-1"</span> <span>"PJ025-tumor-8"</span> <span>"PJ025-tumor-7"</span> <span>"PJ025-tumor-5"</span> <span>"PJ025-tumor-6"</span> <span>"PJ030-tumor-4"</span> <span>"PJ030-tumor-0"</span></span><br><span>#</span><span>&gt; [33] <span>"PJ030-tumor-1"</span> <span>"PJ030-tumor-5"</span> <span>"PJ030-tumor-3"</span> <span>"PJ030-tumor-2"</span> <span>"PJ032-tumor-0"</span> <span>"PJ032-tumor-1"</span> <span>"PJ032-tumor-5"</span> <span>"PJ032-tumor-2"</span></span><br><span>#</span><span>&gt; [41] <span>"PJ032-tumor-4"</span> <span>"PJ032-tumor-3"</span> <span>"PJ035-tumor-2"</span> <span>"PJ035-tumor-3"</span> <span>"PJ035-tumor-6"</span> <span>"PJ035-tumor-1"</span> <span>"PJ035-tumor-0"</span> <span>"PJ035-tumor-4"</span></span><br><span>#</span><span>&gt; [49] <span>"PJ035-tumor-5"</span> <span>"PJ035-tumor-8"</span> <span>"PJ035-tumor-7"</span> <span>"PJ048-tumor-0"</span> <span>"PJ048-tumor-3"</span> <span>"PJ048-tumor-4"</span> <span>"PJ048-tumor-2"</span> <span>"PJ048-tumor-6"</span></span><br><span>#</span><span>&gt; [57] <span>"PJ048-tumor-1"</span> <span>"PJ048-tumor-8"</span> <span>"PJ048-tumor-7"</span> <span>"PJ048-tumor-5"</span></span><br><span><br>#</span><span>&gt; <span>$myeloid</span></span><br><span>#</span><span>&gt; [1] <span>"myeloid_2"</span> <span>"myeloid_5"</span> <span>"myeloid_3"</span> <span>"myeloid_7"</span> <span>"myeloid_0"</span> <span>"myeloid_6"</span> <span>"myeloid_4"</span> <span>"myeloid_1"</span> <span>"myeloid_8"</span></span><br><span> <br>#</span><span>&gt; <span>$pericyte</span></span><br><span>#</span><span>&gt; [1] <span>"pericyte"</span></span><br><span><br>#</span><span>&gt; <span>$endothelial</span></span><br><span>#</span><span>&gt; [1] <span>"endothelial"</span></span><br><span><br>#</span><span>&gt; <span>$tcell</span></span><br><span>#</span><span>&gt; [1] <span>"tcell"</span></span><br><span><br>#</span><span>&gt; <span>$oligo</span></span><br><span>#</span><span>&gt; [1] <span>"oligo"</span></span><br><span><br><br>#</span><span>&gt; Identifier of the malignant cell <span>type</span>:  tumor </span><br><span>#</span><span>&gt; Number of cell states:  73 </span><br><span>#</span><span>&gt; Number of cell types:  6 </span><br><span>#</span><span>&gt; Number of mixtures:  169 </span><br><span>#</span><span>&gt; Number of genes:  16145 </span><br><span><br>#</span><span>&gt; Initial cell <span>type</span> fractions: </span><br><span>#</span><span>&gt;         tumor myeloid pericyte endothelial tcell oligo</span><br><span>#</span><span>&gt; Min.    0.199   0.004    0.000       0.015  0.00 0.000</span><br><span>#</span><span>&gt; 1st Qu. 0.702   0.051    0.014       0.036  0.00 0.010</span><br><span>#</span><span>&gt; Median  0.775   0.102    0.025       0.046  0.00 0.025</span><br><span>#</span><span>&gt; Mean    0.758   0.109    0.043       0.048  0.00 0.041</span><br><span>#</span><span>&gt; 3rd Qu. 0.848   0.145    0.048       0.060  0.00 0.050</span><br><span>#</span><span>&gt; Max.    0.952   0.578    0.503       0.133  0.01 0.329</span><br><span>#</span><span>&gt; Updated cell <span>type</span> fractions: </span><br><span>#</span><span>&gt;         tumor myeloid pericyte endothelial tcell oligo</span><br><span>#</span><span>&gt; Min.    0.272   0.000    0.000       0.008 0.000 0.000</span><br><span>#</span><span>&gt; 1st Qu. 0.751   0.046    0.006       0.027 0.000 0.002</span><br><span>#</span><span>&gt; Median  0.815   0.090    0.014       0.039 0.000 0.011</span><br><span>#</span><span>&gt; Mean    0.801   0.098    0.030       0.041 0.001 0.029</span><br><span>#</span><span>&gt; 3rd Qu. 0.878   0.130    0.033       0.053 0.001 0.036</span><br><span>#</span><span>&gt; Max.    0.968   0.565    0.385       0.134 0.010 0.307</span><br></code></pre><p data-tool="mdnice编辑器">或者，如果想从1-3单独运行步骤4，可以执行以下操作:</p><pre data-tool="mdnice编辑器"><span></span><code><span>#</span><span> not run</span><br>bp.res.initial &lt;- run.prism(prism = myPrism, n.cores=50, update.gibbs=FALSE)<br>    <br>bp.res.update &lt;- update.theta (bp = bp.res.initial)<br></code></pre><h1 data-tool="mdnice编辑器"><span></span>结语</h1><p data-tool="mdnice编辑器">以上是<code>BayesPrism</code>的全部实操内容，<code>bp.res</code>即是<code>BayesPrism</code> 的运算结果。至于如何理解<code>BayesPrism</code>的仔细原理，我非数学及计算机专业的就不强行解释了，感兴趣的可以深入系统学习数学理论。总的来说，在我这里，我只需要理解它要求的输入参数是经过一定处理的<code>单细胞矩阵+细胞表型+bulk RNAseq矩阵</code>，在<code>BayesPrism </code>运算后能得到Bulk RNA-seq反卷积计算的数据即可。在本篇中，仅讨论了BayesPrism的数据准备及运算过程，尚未提及<code>bp.res</code>的后续处理，留待下期探讨。</p><span>- END -</span></section><p><br></p><p><mp-style-type data-value="10000"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/3dZQxDdY6M1WwMMcus5Gkg",target="_blank" rel="noopener noreferrer">原文链接</a>
