---
title: "通路反应基因活性推断工具：PROGENy（二）"
date: 2022-12-26T04:35:30Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Acdemic"]
---
通路反应基因活性推断工具：PROGENy（二） by 生信菜鸟团
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><p data-tool="mdnice编辑器">接上篇，<a target="_blank" href="https://mp.weixin.qq.com/s?__biz=MzUzMTEwODk0Ng==&amp;mid=2247509319&amp;idx=1&amp;sn=a422c15104dc95505d143e307fca8bca&amp;scene=21#wechat_redirect" textvalue="PROGENy: Pathway RespOnsive GENes for activity inference（一）" linktype="text" imgurl="" imgdata="null" tab="innerlink" data-linktype="2">PROGENy: Pathway RespOnsive GENes for activity inference（一）</a></p><p data-tool="mdnice编辑器">Title：PROGENy: Pathway RespOnsive GENes for activity inference; 通路反应基因活性推断工具</p><p data-tool="mdnice编辑器">在上篇中，主要介绍了PROGENy在Bulk RNA seq数据中的应用，以及PROGENy如何联系药物与信号通路。这篇继续探讨PROGENy在单细胞数据中的应用。</p><h3 data-tool="mdnice编辑器"><span></span>4.0  PROGENy在单细胞转录组中的应用<span></span></h3><p data-tool="mdnice编辑器">要运行PROGENy的单细胞版本，首先要获得演示数据。10X genomics 存储的pbmc3k的数据可以直接获得。</p><p data-tool="mdnice编辑器"><span>因此，我们先对该数据进行前期</span><code>Seurat</code><span> 处理10x genomics pbmc3k存储的链接是</span>https://s3-us-west-2.amazonaws.com/10x.files/samples/cell/pbmc3k/pbmc3k_filtered_gene_bc_matrices.tar.gz<span>如果在服务器上使用，可以直接用wget下载：</span></p><pre data-tool="mdnice编辑器"><span></span><code>wget -b -c https://s3-us-west-2.amazonaws.com/10x.files/samples/cell/pbmc3k/pbmc3k_filtered_gene_bc_matrices.tar.gz<br></code></pre><p data-tool="mdnice编辑器">而后进行Seurat流程：</p><pre data-tool="mdnice编辑器"><span></span><code><span>library</span>(Seurat)<br><span>library</span>(ggplot2)<br><span>library</span>(tidyr)<br><span>library</span>(readr)<br><span>library</span>(pheatmap)<br><span>library</span>(tibble)<br>pbmc.data &lt;- <br>  Read10X(data.dir = <span>"./filtered_gene_bc_matrices/hg19/"</span>)<br><span>## Initialize the Seurat object with the raw (non-normalized data).</span><br>pbmc &lt;- <br>  CreateSeuratObject(counts = pbmc.data, project = <span>"pbmc3k"</span>, min.cells = <span>3</span>, <br>                     min.features = <span>200</span>)<br><span>## Identification of mithocondrial genes</span><br>pbmc[[<span>"percent.mt"</span>]] &lt;- PercentageFeatureSet(pbmc, pattern = <span>"^MT-"</span>)<br><br><span>## Filtering cells following standard QC criteria.</span><br>pbmc &lt;- subset(pbmc, subset = nFeature_RNA &gt; <span>200</span> &amp; nFeature_RNA &lt; <span>2500</span> &amp; <br>    percent.mt &lt; <span>5</span>)<br><br><span>## Normalizing the data</span><br>pbmc &lt;- NormalizeData(pbmc, normalization.method = <span>"LogNormalize"</span>, <br>    scale.factor = <span>10000</span>)<br><br>pbmc &lt;- NormalizeData(pbmc)<br><br><span>## Identify the 2000 most highly variable genes</span><br>pbmc &lt;- FindVariableFeatures(pbmc, selection.method = <span>"vst"</span>, nfeatures = <span>2000</span>)<br><br><span>## In addition we scale the data</span><br>all.genes &lt;- rownames(pbmc)<br>pbmc &lt;- ScaleData(pbmc, features = all.genes)<br>pbmc &lt;- RunPCA(pbmc, features = VariableFeatures(object = pbmc), verbose = <span>FALSE</span>)<br>pbmc &lt;- FindNeighbors(pbmc, dims = <span>1</span>:<span>10</span>, verbose = <span>FALSE</span>)<br>pbmc &lt;- FindClusters(pbmc, resolution = <span>0.5</span>, verbose = <span>FALSE</span>)<br>pbmc &lt;- RunUMAP(pbmc, dims = <span>1</span>:<span>10</span>,  umap.method = <span>'uwot'</span>, metric=<span>'cosine'</span>)<br>DimPlot(pbmc, reduction = <span>"umap"</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7234468937875751" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjobhUokvJVbrf2mGf5eibaaChESHcicQh8JLxE5eQubI9BM4mqlMVak8t4A/640?wx_fmt=png" data-type="png" data-w="499" title="图1  pbmc3k umap" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjobhUokvJVbrf2mGf5eibaaChESHcicQh8JLxE5eQubI9BM4mqlMVak8t4A/640?wx_fmt=png"><figcaption><span>图1  pbmc3k umap</span><br></figcaption><figcaption><span><br></span></figcaption></figure><pre data-tool="mdnice编辑器"><code><span>## Finding differentially expressed features (cluster biomarkers)</span><br>pbmc.markers &lt;- FindAllMarkers(pbmc, only.pos = <span>TRUE</span>, min.pct = <span>0.25</span>, <br>    logfc.threshold = <span>0.25</span>, verbose = <span>FALSE</span>)<br><br><span>## Assigning cell type identity to clusters</span><br>new.cluster.ids &lt;- c(<span>"Naive CD4 T"</span>, <span>"Memory CD4 T"</span>, <span>"CD14+ Mono"</span>, <span>"B"</span>, <span>"CD8 T"</span>, <br>    <span>"FCGR3A+ Mono"</span>, <span>"NK"</span>, <span>"DC"</span>, <span>"Platelet"</span>)<br>names(new.cluster.ids) &lt;- levels(pbmc)<br>pbmc &lt;- RenameIdents(pbmc, new.cluster.ids)<br><br><span>## We create a data frame with the specification of the cells that belong to </span><br><span>## each cluster to match with the Progeny scores.</span><br>CellsClusters &lt;- data.frame(Cell = names(Idents(pbmc)), <br>    CellType = as.character(Idents(pbmc)),<br>    stringsAsFactors = <span>FALSE</span>)<br>DimPlot(pbmc, reduction = <span>"umap"</span>, label = <span>TRUE</span>, pt.size = <span>0.5</span>) + NoLegend()<br></code></pre><p data-tool="mdnice编辑器"><img data-ratio="0.7874251497005988" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjobzjY5Z1ibkRTL5cm6ic4M1T5DIMh0GUj7libyYiaL3YwC7iavj8lwpysDUrQ/640?wx_fmt=png" data-type="png" data-w="668" title="图2  pbmc3k  注释后 umap" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjobzjY5Z1ibkRTL5cm6ic4M1T5DIMh0GUj7libyYiaL3YwC7iavj8lwpysDUrQ/640?wx_fmt=png"><span>图2  pbmc3k  注释后 umap</span></p></section><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><p data-tool="mdnice编辑器">在完成了细胞注释之后，可对pbmc3k 的样本进行PROGENy 分析。</p><pre data-tool="mdnice编辑器"><code><span>## We compute the Progeny activity scores and add them to our Seurat object</span><br><span>## as a new assay called Progeny. </span><br>pbmc &lt;- progeny(pbmc, scale=<span>FALSE</span>, organism=<span>"Human"</span>, top=<span>500</span>, perm=<span>1</span>, <br>    return_assay = <span>TRUE</span>)<br><br><span>## We can now directly apply Seurat functions in our Progeny scores. </span><br><span>## For instance, we scale the pathway activity scores. </span><br>pbmc &lt;- Seurat::ScaleData(pbmc, assay = <span>"progeny"</span>) <br><br><span>## We transform Progeny scores into a data frame to better handling the results</span><br>progeny_scores_df &lt;- <br>    as.data.frame(t(GetAssayData(pbmc, slot = <span>"scale.data"</span>, <br>        assay = <span>"progeny"</span>))) %&gt;%<br>    rownames_to_column(<span>"Cell"</span>) %&gt;%<br>    gather(Pathway, Activity, -Cell) <br><br><span>## We match Progeny scores with the cell clusters.</span><br>progeny_scores_df &lt;- inner_join(progeny_scores_df, CellsClusters)<br><br><span>## We summarize the Progeny scores by cellpopulation</span><br>summarized_progeny_scores &lt;- progeny_scores_df %&gt;% <br>    group_by(Pathway, CellType) %&gt;%<br>    summarise(avg = mean(Activity), std = sd(Activity))<br><br><span>## We prepare the data for the plot</span><br>summarized_progeny_scores_df &lt;- summarized_progeny_scores %&gt;%<br>    dplyr::select(-std) %&gt;%   <br>    spread(Pathway, avg) %&gt;%<br>    data.frame(row.names = <span>1</span>, check.names = <span>FALSE</span>, stringsAsFactors = <span>FALSE</span>) <br><br>paletteLength = <span>100</span><br>myColor = colorRampPalette(c(<span>"Darkblue"</span>, <span>"white"</span>,<span>"red"</span>))(paletteLength)<br><br>progenyBreaks = c(seq(min(summarized_progeny_scores_df), <span>0</span>, <br>                      length.out=ceiling(paletteLength/<span>2</span>) + <span>1</span>),<br>                  seq(max(summarized_progeny_scores_df)/paletteLength, <br>                      max(summarized_progeny_scores_df), <br>                      length.out=floor(paletteLength/<span>2</span>)))<br>progeny_hmap = pheatmap(t(summarized_progeny_scores_df[,-<span>1</span>]),fontsize=<span>14</span>, <br>                        fontsize_row = <span>10</span>, <br>                        color=myColor, breaks = progenyBreaks, <br>                        main = <span>"PROGENy (500)"</span>, angle_col = <span>45</span>,<br>                        treeheight_col = <span>0</span>,  border_color = <span>NA</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7874251497005988" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjobeQySdAdLF1PmZxUpHrZ2X9AShicAPOfNBsZbsrO2fzk9cdk6k47mpPQ/640?wx_fmt=png" data-type="png" data-w="668" title="图3  PROGENy 单细胞结果展示" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjobeQySdAdLF1PmZxUpHrZ2X9AShicAPOfNBsZbsrO2fzk9cdk6k47mpPQ/640?wx_fmt=png"></figure><ne-p data-lake-id="u313b87b0" ne-alignment="center"><ne-card data-card-name="image" data-card-type="inline" data-event-boundary="card"><p><span>图3  PROGENy 单细胞结果展示</span></p><p><span><br></span></p><p><span><br></span></p></ne-card></ne-p><figure data-tool="mdnice编辑器"><figcaption><span>  5.  一些函数的使用  </span><br></figcaption></figure><p data-tool="mdnice编辑器">在v1.20版本里面，除progeny 函数之外，PROGENy包也存在其他函数，如下：<img data-ratio="0.3263157894736842" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjob93rn860gA1ObcQ926naVfeIcdPogE9W86j8FUicyTTAluANuXCgZR4Q/640?wx_fmt=png" data-type="png" data-w="570" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjob93rn860gA1ObcQ926naVfeIcdPogE9W86j8FUicyTTAluANuXCgZR4Q/640?wx_fmt=png"></p><p data-tool="mdnice编辑器">但我们在使用过程中并没有太多的用到这些函数，progeny函数是这个包的核心，那么其他的函数是用来干嘛的呢？</p><p data-tool="mdnice编辑器">查看progeny 的源码，我们发现，getModel函数是从作者提供的Human 及Mouse 中取出目标通路的基因集,</p><pre data-tool="mdnice编辑器"><code>progeny = <span>function</span>(expr, scale=<span>TRUE</span>, organism=<span>"Human"</span>, top = <span>100</span>, perm = <span>1</span>, <br>    verbose = <span>FALSE</span>, z_scores = <span>FALSE</span>, get_nulldist = <span>FALSE</span>, assay_name = <span>"RNA"</span>, <br>    return_assay = <span>FALSE</span>, <span>...</span>) {<br>    UseMethod(<span>"progeny"</span>)<br>}<br>methods(<span>"progeny"</span>)<br><span>#&gt; [1] progeny.default*              progeny.ExpressionSet*        progeny.matrix*               progeny.Seurat*               progeny.SingleCellExperiment*</span><br><span>#&gt; see '?methods' for accessing help and source code</span><br>getAnywhere(progeny.matrix)<br><span>#&gt; A single object matching ‘progeny.matrix’ was found</span><br><span>#&gt; It was found in the following places</span><br><span>#&gt;   registered S3 method for progeny from namespace progeny</span><br><span>#&gt;   namespace:progeny</span><br><span>#&gt; with value</span><br><span>#&gt; </span><br><span>#&gt; function (expr, scale = TRUE, organism = "Human", top = 100, </span><br><span>#&gt;     perm = 1, verbose = FALSE, z_scores = FALSE, get_nulldist = FALSE, </span><br><span>#&gt;     ...) </span><br><span>#&gt; {</span><br><span>#&gt;     if (!is.logical(scale)) {</span><br><span>#&gt;         stop("scale should be a logical value")</span><br><span>#&gt;    }</span><br><span>#&gt;     if (!(is.numeric(perm)) || perm &lt; 1) {</span><br><span>#&gt;         stop("perm should be an integer value")</span><br><span>#&gt;     }</span><br><span>#&gt;     if (!is.logical(verbose)) {</span><br><span>#&gt;         stop("verbose should be a logical value")</span><br><span>#&gt;     }</span><br><span>#&gt;     if (!is.logical(z_scores)) {</span><br><span>#&gt;         stop("z_scores should be a logical value")</span><br><span>#&gt;     }</span><br><span>#&gt;     if (!is.logical(get_nulldist)) {</span><br><span>#&gt;         stop("get_nulldist should be a logical value")</span><br><span>#&gt;     }</span><br><span>#&gt;     if (perm == 1 &amp;&amp; (z_scores || get_nulldist)) {</span><br><span>#&gt;         if (verbose) {</span><br><span>#&gt;             message("z_scores and get_nulldist are only applicable when the\n                number of permutations is larger than 1.")</span><br><span>#&gt;         }</span><br><span>#&gt;     }</span><br><span>#&gt;     model &lt;- getModel(organism, top = top)</span><br><span>#&gt;     common_genes &lt;- intersect(rownames(expr), rownames(model))</span><br><span>#&gt;     if (verbose) {</span><br><span>#&gt;         number_genes &lt;- apply(model, 2, function(x) {</span><br><span>#&gt;             sum(rownames(model)[which(x != 0)] %in% unique(rownames(expr)))</span><br><span>#&gt;         })</span><br><span>#&gt;         message("Number of genes used per pathway to compute progeny scores:")</span><br><span>#&gt;         message(paste(names(number_genes), ": ", number_genes, </span><br><span>#&gt;             " (", (number_genes/top) * 100, "%)", sep = "", "\n"))</span><br><span>#&gt;     }</span><br><span>#&gt;     if (perm == 1) {</span><br><span>#&gt;         result &lt;- t(expr[common_genes, , drop = FALSE]) %*% as.matrix(model[common_genes, </span><br><span>#&gt;             , drop = FALSE])</span><br><span>#&gt;         if (scale &amp;&amp; nrow(result) &gt; 1) {</span><br><span>#&gt;             rn &lt;- rownames(result)</span><br><span>#&gt;             result &lt;- apply(result, 2, scale)</span><br><span>#&gt;             rownames(result) &lt;- rn</span><br><span>#&gt;         }</span><br><span>#&gt;     }</span><br><span>#&gt;     else if (perm &gt; 1) {</span><br><span>#&gt;         expr &lt;- data.frame(names = row.names(expr), row.names = NULL, </span><br><span>#&gt;             expr)</span><br><span>#&gt;         model &lt;- data.frame(names = row.names(model), row.names = NULL, </span><br><span>#&gt;             model)</span><br><span>#&gt;         result &lt;- progenyPerm(expr, model, k = perm, z_scores = z_scores, </span><br><span>#&gt;             get_nulldist = get_nulldist)</span><br><span>#&gt;     }</span><br><span>#&gt;     return(result)</span><br><span>#&gt; }</span><br><span>#&gt; &lt;bytecode: 0x5648eb3c2ba0&gt;</span><br><span>#&gt; &lt;environment: namespace:progeny&gt;</span><br></code></pre><p data-tool="mdnice编辑器">其他函数可自行在帮助文档查阅各自功能。</p><h3 data-tool="mdnice编辑器">附：当我们找不到说明文档怎么办？</h3><p data-tool="mdnice编辑器">在本软件学习过程中，遇到最明显的bug 就是作者在更新PROGENy之后，以往的帮助文档网页丢失了，这给学习使用R包的我们带来麻烦，这里简要分享我如何寻找这篇文章的帮助文档。</p><h4 data-tool="mdnice编辑器">Bioconductor</h4><p data-tool="mdnice编辑器">一般情况下，Bioconductor R包在这个部分会提供说明文档的链接，点开HTML/R_Script 或者查看PDF都可以看到帮助文档，在这里，HTML（https://www.bioconductor.org/packages/release/bioc/vignettes/progeny/inst/doc/progeny.html）并没有太多使用信息，只是介绍了getModel 是怎么用的，而PDF（https://www.bioconductor.org/packages/release/bioc/manuals/progeny/man/progeny.pdf）也只是提供了各个函数的介绍。也就是说，从这里并没有获取到很有价值的信息。<img data-ratio="0.3566333808844508" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjobPCujJc4nE7PnsCuK52ywibmh9tUQH2dD0joc8P9AFknO2X7DBr8uZDQ/640?wx_fmt=png" data-type="png" data-w="701" title="图4. PROGENy Bioconductor 界面" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjobPCujJc4nE7PnsCuK52ywibmh9tUQH2dD0joc8P9AFknO2X7DBr8uZDQ/640?wx_fmt=png"></p><ne-p data-lake-id="ud84dd430" ne-alignment="center"><ne-card data-card-name="image" data-card-type="inline" data-event-boundary="card"><p><span>图4. PROGENy Bioconductor 界面</span></p></ne-card></ne-p><p data-tool="mdnice编辑器">参照Bioconductor的获取帮助文档的方式也失败了。</p><pre data-tool="mdnice编辑器"><code>browseVignettes(<span>"progeny"</span>)<br><span>#&gt; No vignettes found by browseVignettes("progeny")</span><br></code></pre><p data-tool="mdnice编辑器">那其实也可以检索一下旧版本的progeny 有没有配套的使用说明。<img data-ratio="0.5825734549138805" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjobEWEzARX1nra9wuFLibAExcibY5RJNxeF5XVQia2TrXnf0dyCXGcGWeIIQ/640?wx_fmt=png" data-type="png" data-w="987" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjobEWEzARX1nra9wuFLibAExcibY5RJNxeF5XVQia2TrXnf0dyCXGcGWeIIQ/640?wx_fmt=png"><span>但如果你也进行了这一步，你会发现，这个链接点开啥也没有，而在其他版本的HTML文件中，使用信息也不是那么全。</span></p><h4 data-tool="mdnice编辑器">Github</h4><p data-tool="mdnice编辑器">常见的说明文档也会提供在Github （https://github.com/saezlab/progeny）的首页，但在这里并没有。<img data-ratio="0.83" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjobibwsl5bQTdpE03L6u9JYpibPwwicMp8kuKOIxANS2tpBla5ZrMtqp9eIw/640?wx_fmt=png" data-type="png" data-w="900" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjobibwsl5bQTdpE03L6u9JYpibPwwicMp8kuKOIxANS2tpBla5ZrMtqp9eIw/640?wx_fmt=png"></p><p data-tool="mdnice编辑器">因此，下一步查看他的各个文件夹中里面有没有帮助文档。一般来说vignettes 文件夹是储存帮助文档的，感兴趣的可以试着点开看一下<img data-ratio="0.6305525460455038" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjobAhBkFia2icEKsGjibjibicEto3TKKho6VwrslHSSZMpdKAmeUnrpicdX1aPw/640?wx_fmt=png" data-type="png" data-w="923" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjobAhBkFia2icEKsGjibjibicEto3TKKho6VwrslHSSZMpdKAmeUnrpicdX1aPw/640?wx_fmt=png"></p><p data-tool="mdnice编辑器">我这里直接说结果，在docs 的下级目录中找到了这两个文件。如果你在github中打开，要是像我一样小白的话，可能看不懂它网页的源代码格式的数据，因此，就直接把他download下来，用本地浏览器打开，你会发现这就是一份操作说明。<img data-ratio="0.5484276729559748" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjobQmzwaEuQ4mvibY5whTVpia7PQyfwicvBChOiaHic4bNY9yicGELM2GUpdsaA/640?wx_fmt=png" data-type="png" data-w="795" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Los9RXkibd9G3FvR6McxGicMjobQmzwaEuQ4mvibY5whTVpia7PQyfwicvBChOiaHic4bNY9yicGELM2GUpdsaA/640?wx_fmt=png"></p><h1 data-tool="mdnice编辑器">结语</h1><p data-tool="mdnice编辑器">总体而言，PROGENy 是个很优秀的R包，但也有其不足的地方，比如，它只涵盖了14条通路。上篇写完才发现，原来已经有其他公众号上已经推送过PROGENy，还引用了文献的例子来介绍PROGENy的使用，读者可以互相参考借鉴，由于来源一致，大部分内容可能会比较趋同。如果对R包原理感兴趣，还是要移步作者原文仔细详读才行哦。</p><pre spellcheck="false" lang="" cid="n96" mdtype="fences"><span role="presentation">[1] Schubert M, Klinger B, Klünemann M, Sieber A, Uhlitz F, Sauer S, Garnett MJ, Blüthgen N, Saez-Rodriguez J. 2018. Perturbation-response genes reveal signaling footprints in cancer gene expression. *Nature Communications*: [10.1038/s41467-017-02391-6](https://doi.org/10.1038/s41467-017-02391-6)</span><br><span role="presentation"><span cm-text=""></span></span><br><span role="presentation">[2] Holland CH, Szalai B, Saez-Rodriguez J. Transfer of regulatory knowledge from human to mouse for functional genomics analysis. Biochim Biophys Acta Gene Regul Mech. 2020;1863(6):194431. doi:10.1016/j.bbagrm.2019.194431 </span><br><span role="presentation"><span cm-text=""></span></span><br><span role="presentation">[3] Holland, C.H., Tanevski, J., Perales-Patón, J. et al. Robustness and applicability of transcription factor and pathway analysis tools on single-cell RNA-seq data. Genome Biol 21, 36 (2020). https://doi.org/10.1186/s13059-020-1949-z</span></pre><p cid="n94" mdtype="paragraph"><span md-inline="plain"><br></span></p><span>- END -</span></section><p><br></p><p><mp-style-type data-value="10000"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/5QSOaHvz__xoMBRuljKuSg",target="_blank" rel="noopener noreferrer">原文链接</a>
