---
title: "都是FPKM进行差异分析，为啥差异感觉这么大呢？"
date: 2022-12-21T05:06:22Z
draft: ["false"]
tags: [
  "fetched",
  "生信菜鸟团"
]
categories: ["Acdemic"]
---
都是FPKM进行差异分析，为啥差异感觉这么大呢？ by 生信菜鸟团
------
<div><section data-tool="mdnice编辑器" data-website="https://www.mdnice.com"><h2 data-tool="mdnice编辑器"><span></span><span>缘起</span><span></span></h2><p data-tool="mdnice编辑器">上周我们比较了FPKM与count分别进行差异分析的区别，发现两者差异分析的结果基本一致。恰巧曾老师叫我好好看看一篇发表在frontiers in Molecular Neuroscience的文章，他的转录组数据处理使用的就是FPKM矩阵。那这周的推文的话，我们就来看看我们的流程分析结果与作者有何不同。当然，强调一下哈，有count还是尽量用count为好。</p><h2 data-tool="mdnice编辑器"><span></span><span>探究</span><span></span></h2><p data-tool="mdnice编辑器">今天，我们使用2020年发表在frontiers in Molecular Neuroscience杂志上，文献名称为Differential Gene Expression Patterns Between Apical and Basal Inner Hair Cells Revealed by RNA-Seq的转录组FPKM数据（附表4）进行差异分析，感兴趣的小伙伴也可以自身下载一下下面的上游数据定量，同上一篇推文一样，比较两种数据类型进行差异分析的异同。</p><h2 data-tool="mdnice编辑器"><span></span><span>转录组数据集介绍</span><span></span></h2><p data-tool="mdnice编辑器">该数据集提交在ENA官网，项目号是PRJNA593359，如有小伙伴想看上游，链接见<span>https://www.ebi.ac.uk/ena/browser/view/PRJNA593359</span> 。该数据集组别上分为两组，一组是Basal组，一组是Apical组，每组均有三个样本。<img data-ratio="0.8315412186379928" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic4EATWeRRU3tRw62zicM21icJShyxUprQYLJ5EePicEUKuuQtHdqlrE2M5ETCOUu4EaWQ3WFVsqXichQ/640?wx_fmt=png" data-type="png" data-w="558" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic4EATWeRRU3tRw62zicM21icJShyxUprQYLJ5EePicEUKuuQtHdqlrE2M5ETCOUu4EaWQ3WFVsqXichQ/640?wx_fmt=png"></p><h2 data-tool="mdnice编辑器"><span></span><span>正式分析</span><span></span></h2><h3 data-tool="mdnice编辑器"><span></span><span><span></span>1.利用fpkm值进行差异分析</span><span></span></h3><p data-tool="mdnice编辑器">进行差异分析时，阈值按照文章的选法，FC为2，p&lt;0.001</p><pre data-tool="mdnice编辑器"><span></span><code><span># 1.下载文中的补充数据集table 4.xlsx</span><br>rm(list=ls())<br><span>library</span>(openxlsx)<br><span>library</span>(stringr)<br>a=read.xlsx(<span>'table 4.xlsx'</span>,colNames = <span>T</span>) <span># 提取表达矩阵</span><br>class(a)<br><span>## [1] "data.frame"</span><br>colnames(a)=a[<span>1</span>,]<br>a=a[-<span>1</span>,]<br>table(duplicated(a$`Feature ID`))<br><span>## </span><br><span>## FALSE  TRUE </span><br><span>## 15307     1</span><br>a=a[!duplicated(a$`Feature ID`),]<br>rownames(a)=a$`Feature ID`<br>a=a[,-<span>1</span>]<br>keep &lt;- rowSums(a&gt;<span>0</span>) &gt;= floor(<span>0.5</span>*ncol(a))<br>expMatrix &lt;- a[keep,] <span>#获得filter_count矩阵</span><br><span>#(防止0值影响FPKM转换为TPM值)</span><br><br><span># 2.将FPKM值转换为TPM值</span><br>fpkmToTpm &lt;- <span>function</span>(fpkm)<br>{<br>  exp(log(fpkm) - log(sum(fpkm)) + log(<span>1e6</span>))<br>}<br><br>class(expMatrix)<br><span>## [1] "data.frame"</span><br>expMatrix=na.omit(expMatrix)<br>numer=<span>function</span>(x){as.numeric(x)}<br>expMatrix=apply(expMatrix, <span>1</span>, numer) <br><span>#必须将矩阵中的字符类型由向量型改为字符型</span><br>expMatrix=t(expMatrix)<br>tpms &lt;- apply(expMatrix,<span>2</span>,fpkmToTpm) <br>tpms[<span>1</span>:<span>3</span>,]<br><span>##                    [,1]       [,2]     [,3]      [,4]     [,5]       [,6]</span><br><span>## 0610007P14Rik  47.67969 145.695029 22.27847   0.00000 117.7912  80.804891</span><br><span>## 0610009B22Rik 156.09341 300.652882 96.91044 104.00990 252.0514 224.045029</span><br><span>## 0610009L18Rik  30.88343   8.785541 39.00099  30.98678  10.0835   5.584516</span><br>colSums(tpms)<br><span>## [1] 1e+06 1e+06 1e+06 1e+06 1e+06 1e+06</span><br>colnames(tpms)=c(<span>"Apical1"</span>,<span>"Apical2"</span>,<span>"Apical3"</span>,<span>"Basal1"</span>,<span>"Basal2"</span>,<span>"Basal3"</span>)<br><br><span># 3.差异分析</span><br>group_list=c(rep(<span>'Apical'</span>,<span>3</span>),rep(<span>'Basal'</span>,<span>3</span>))<br><span>## 强制限定顺序</span><br>group_list &lt;- factor(group_list,levels = c(<span>"Apical"</span>,<span>"Basal"</span>),ordered = <span>F</span>)<br><span>#表达矩阵数据校正</span><br>exprSet &lt;- tpms<br>boxplot(exprSet,outline=<span>FALSE</span>, notch=<span>T</span>,col=group_list, las=<span>2</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7142857142857143" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic4EATWeRRU3tRw62zicM21ic1d3JQsUOFdcjm1tEVw8D3lCpp7GArSIVExxPz8iaKvxZ8XM7GliawBow/640?wx_fmt=png" data-type="png" data-w="672" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic4EATWeRRU3tRw62zicM21ic1d3JQsUOFdcjm1tEVw8D3lCpp7GArSIVExxPz8iaKvxZ8XM7GliawBow/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code><span>library</span>(limma) <br>head(exprSet)<br><span>##                 Apical1    Apical2  Apical3       Basal1    Basal2     Basal3</span><br><span>## 0610007P14Rik  47.67969 145.695029 22.27847   0.00000000 117.79116  80.804891</span><br><span>## 0610009B22Rik 156.09341 300.652882 96.91044 104.00989902 252.05143 224.045029</span><br><span>## 0610009L18Rik  30.88343   8.785541 39.00099  30.98678285  10.08350   5.584516</span><br><span>## 0610009O20Rik  70.01261  83.931085 46.90684   0.00000000  35.58989  11.294728</span><br><span>## 0610010B08Rik  10.48073  14.035653 27.61583   0.06863075  10.89523   1.544271</span><br><span>## 0610010F05Rik  13.90093  11.348727 14.68230   0.00000000  90.48093   0.000000</span><br>exprSet=normalizeBetweenArrays(exprSet)<br>boxplot(exprSet,outline=<span>FALSE</span>, notch=<span>T</span>,col=group_list, las=<span>2</span>)<br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7142857142857143" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic4EATWeRRU3tRw62zicM21ic2ZiaxrR5o3t8egwTY3EDtxq1vIVic8DDJqtRiclbPb72Qf2kPB4yARc0w/640?wx_fmt=png" data-type="png" data-w="672" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic4EATWeRRU3tRw62zicM21ic2ZiaxrR5o3t8egwTY3EDtxq1vIVic8DDJqtRiclbPb72Qf2kPB4yARc0w/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code><span>#判断数据是否需要转换</span><br>exprSet &lt;- log2(exprSet+<span>1</span>)<br>range(exprSet)<br><span>## [1]  0.00000 14.31486</span><br><span>#差异分析：</span><br>dat &lt;- exprSet<br>design=model.matrix(~factor( group_list ))<br>fit=lmFit(dat,design)<br>fit=eBayes(fit)<br>options(digits = <span>4</span>)<br>topTable(fit,coef=<span>2</span>,adjust=<span>'BH'</span>)<br><span>##          logFC AveExpr       t   P.Value adj.P.Val     B</span><br><span>## Adam8    5.364   2.682  13.180 4.176e-06   0.03458 4.131</span><br><span>## Ocm     -5.603   8.401 -12.813 5.033e-06   0.03458 4.018</span><br><span>## Nwd1     4.124   2.062   9.869 2.756e-05   0.06021 2.856</span><br><span>## Gm4869   3.901   1.951   9.800 2.882e-05   0.06021 2.823</span><br><span>## Rubcnl   3.858   1.929   9.769 2.942e-05   0.06021 2.807</span><br><span>## Igsf10   3.885   1.943   9.293 4.048e-05   0.06021 2.563</span><br><span>## Igfals   3.617   1.811   9.198 4.324e-05   0.06021 2.512</span><br><span>## Gm13547  3.582   1.791   9.123 4.556e-05   0.06021 2.471</span><br><span>## Bbox1    4.626   2.313   9.080 4.693e-05   0.06021 2.448</span><br><span>## Mmp23   -4.935   4.383  -9.050 4.794e-05   0.06021 2.431</span><br>bp=<span>function</span>(g){<br>  <span>library</span>(ggpubr)<br>  df=data.frame(gene=g,stage=group_list)<br>  p &lt;- ggboxplot(df, x = <span>"stage"</span>, y = <span>"gene"</span>,<br>                 color = <span>"stage"</span>, palette = <span>"jco"</span>,<br>                 add = <span>"jitter"</span>)<br>  <span>#  Add p-value</span><br>  p + stat_compare_means()<br>}<br>deg=topTable(fit,coef=<span>2</span>,adjust=<span>'BH'</span>,number = <span>Inf</span>)<br>head(deg) <br><span>##         logFC AveExpr       t   P.Value adj.P.Val     B</span><br><span>## Adam8   5.364   2.682  13.180 4.176e-06   0.03458 4.131</span><br><span>## Ocm    -5.603   8.401 -12.813 5.033e-06   0.03458 4.018</span><br><span>## Nwd1    4.124   2.062   9.869 2.756e-05   0.06021 2.856</span><br><span>## Gm4869  3.901   1.951   9.800 2.882e-05   0.06021 2.823</span><br><span>## Rubcnl  3.858   1.929   9.769 2.942e-05   0.06021 2.807</span><br><span>## Igsf10  3.885   1.943   9.293 4.048e-05   0.06021 2.563</span><br><span>#save(deg,file = 'deg.Rdata')</span><br><br><span># 4.FPKM转换为TPM后，再利用limma进行差异分析</span><br><br><span>## 不同的阈值，筛选到的差异基因数量就不一样，后面的超几何分布检验结果就大相径庭。</span><br><span>if</span>(<span>T</span>){<br>  logFC_t=<span>1</span><br>  deg$g=ifelse(deg$P.Value&gt;<span>0.01</span>,<span>'stable'</span>,<br>               ifelse( deg$logFC &gt; logFC_t,<span>'UP'</span>,<br>                       ifelse( deg$logFC &lt; -logFC_t,<span>'DOWN'</span>,<span>'stable'</span>) )<br>  )<br>  table(deg$g)<br>  head(deg)<br>  <br>}<br><span>##         logFC AveExpr       t   P.Value adj.P.Val     B    g</span><br><span>## Adam8   5.364   2.682  13.180 4.176e-06   0.03458 4.131   UP</span><br><span>## Ocm    -5.603   8.401 -12.813 5.033e-06   0.03458 4.018 DOWN</span><br><span>## Nwd1    4.124   2.062   9.869 2.756e-05   0.06021 2.856   UP</span><br><span>## Gm4869  3.901   1.951   9.800 2.882e-05   0.06021 2.823   UP</span><br><span>## Rubcnl  3.858   1.929   9.769 2.942e-05   0.06021 2.807   UP</span><br><span>## Igsf10  3.885   1.943   9.293 4.048e-05   0.06021 2.563   UP</span><br>fpkm_deg=deg<br></code></pre><h3 data-tool="mdnice编辑器"><span></span><span><span></span>2.绘制文章的两张图</span><span></span></h3><pre data-tool="mdnice编辑器"><span></span><code><span>#  01复现第一张图，fpkm&gt;1的Veen图，可以看看组别独有的基因</span><br><span># 绘制思路：分别获得针对两个组别fpkm&gt;1的基因，取个交集即可</span><br><br>apical=exprSet[,<span>1</span>:<span>3</span>]<br>apical=as.data.frame(apical)<br>apical$mean=apply(apical,<span>1</span>,mean)<br><br>basal=exprSet[,<span>4</span>:<span>6</span>]<br>basal=as.data.frame(basal)<br>basal$mean=apply(basal,<span>1</span>,mean)<br><br><span># 韦恩图取个交集</span><br>Apical &lt;- rownames(apical[apical$mean&gt;<span>1</span>,])<br>Basal &lt;- rownames(basal[basal$mean&gt;<span>1</span>,])<br><span>library</span>(ggvenn)<br>deg&lt;-list(`Apical`=Apical,<br>          `Basal`=Basal)<br>p1 &lt;- ggvenn(deg,<br>             show_percentage = <span>F</span>,<br>             stroke_color = <span>"white"</span>,<br>             fill_color = c(<span>"#b2d4ec"</span>,<span>"#d3c0e2"</span>),<br>             set_name_color = c(<span>"#ff0000"</span>,<span>"#4a9b83"</span>))<br>p1 <span>#30个FDR分析的差异基因全在利用P值分析的差异基因里</span><br></code></pre><figure data-tool="mdnice编辑器"><img data-ratio="0.7142857142857143" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic4EATWeRRU3tRw62zicM21ic4bd2hvV3GoopDsNJsoIVLdl2eSXZpiczDYxvSE5yTB6yWJcu0icnBn9g/640?wx_fmt=png" data-type="png" data-w="672" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic4EATWeRRU3tRw62zicM21ic4bd2hvV3GoopDsNJsoIVLdl2eSXZpiczDYxvSE5yTB6yWJcu0icnBn9g/640?wx_fmt=png"></figure><pre data-tool="mdnice编辑器"><span></span><code><br><span>## 02差异基因箱线图</span><br><span># 绘制箱线图</span><br>table(fpkm_deg$g==<span>"UP"</span>)<br><span>## </span><br><span>## FALSE  TRUE </span><br><span>## 13454   287</span><br>table(fpkm_deg$g==<span>"DOWN"</span>)<br><span>## </span><br><span>## FALSE  TRUE </span><br><span>## 13560   181</span><br>data=data.frame(c(<span>287</span>,<span>181</span>),row.names = c(<span>"up"</span>,<span>"down"</span>))<br>colnames(data)=c(<span>"deg_number"</span>)<br>p2 &lt;- ggplot(data,aes(x=rownames(data),y=deg_number,fill=rownames(data)))+<br>  geom_boxplot(width=<span>0.6</span>,alpha=<span>0.8</span>)<br>p3 &lt;- p2+scale_y_continuous(breaks = seq(<span>0</span>,<span>800</span>,<span>200</span>),<br>                            limits=c(<span>0</span>,<span>800</span>))+theme_classic()<br>p3<br></code></pre><figure data-tool="mdnice编辑器"><img data-cropselx1="0" data-cropselx2="558" data-cropsely1="0" data-cropsely2="399" data-ratio="0.7142857142857143" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic4EATWeRRU3tRw62zicM21icnz1Qsw67fw0u9LoC4c12TsXEFyVhticdcibdXow9udWaq9oHQhVeOyeA/640?wx_fmt=png" data-type="png" data-w="672" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic4EATWeRRU3tRw62zicM21icnz1Qsw67fw0u9LoC4c12TsXEFyVhticdcibdXow9udWaq9oHQhVeOyeA/640?wx_fmt=png"></figure><h3 data-tool="mdnice编辑器"><span></span><span><span></span>3.用作者提出来的上调与下调基因在我的结果中验证</span><span></span></h3><p data-tool="mdnice编辑器">图就不修了哈，火山图的绘制很简单前面已经画了很多遍了，由于作者没有提供差异基因列表，我们就简单地针对其在文中提到的几个基因进行验证吧</p><pre data-tool="mdnice编辑器"><span></span><code>cg=c(<span>"Prkd1"</span>,<span>"Sncg"</span>,<span>"Nme2"</span>,<span>"Fbxo32"</span>)<br>deg_fc=fpkm_deg[cg,]<br>head(deg_fc)<br><span>##          logFC AveExpr      t  P.Value adj.P.Val      B      g</span><br><span>## Prkd1  -1.1511   5.959 -2.853 0.025267    0.4069 -3.381 stable</span><br><span>## Sncg   -2.0205   5.932 -4.412 0.003308    0.1865 -1.377   DOWN</span><br><span>## Nme2    0.8174   7.355  1.333 0.225381    0.7532 -5.430 stable</span><br><span>## Fbxo32  1.3724   6.863  2.908 0.023378    0.4002 -3.305 stable</span><br><br><span># 此处重点看logFC,再与文章</span><br></code></pre><p data-tool="mdnice编辑器">以上我们对该篇文章的FPKM数据集进行了处理，还绘制了<span>文章的</span>两幅结果图以及运用了文中的几个基因进行验证。韦恩图中可能是由于我设置了过滤参数导致我们的结果有一定区别。但是与原文相比，我们的上调差异基因有287个，下调差异基因有181个。与文章的644个上调，45个下调在数量上存在显著不同，这点就很值得咀嚼。<img data-ratio="0.38574660633484165" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic4EATWeRRU3tRw62zicM21ic5BMWneqeEXKwx7KLbdeMOroVKyStIB4etY35U18Ykh2YnIX16tXhmQ/640?wx_fmt=png" data-type="png" data-w="884" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic4EATWeRRU3tRw62zicM21ic5BMWneqeEXKwx7KLbdeMOroVKyStIB4etY35U18Ykh2YnIX16tXhmQ/640?wx_fmt=png"></p><p data-tool="mdnice编辑器">检查的几个差异基因源于文中的描述，如下。与3中我们的差异分析结果相比较，logFC大致是对应了，但是p值两者存在很大不同。<span><strong>为啥同一个数据，一样的阈值两者差异基因的数量相差如此显著</strong></span>，难道是统计方法的不同导致的差异基因在作者的分析结果与我的分析结果中存在不同吗？<span><strong>感兴趣的小伙伴们可以尝试尝试，欢迎点评哈</strong></span>。<img data-ratio="0.2685624012638231" data-src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic4EATWeRRU3tRw62zicM21icibLl3IsMEOYrCAUJyZPhz2nVicq4dI4EKjONL4Bgx2cXT1w1xpeJdmPQ/640?wx_fmt=png" data-type="png" data-w="633" src="https://mmbiz.qpic.cn/mmbiz_png/iaRJcrq2Losic4EATWeRRU3tRw62zicM21icibLl3IsMEOYrCAUJyZPhz2nVicq4dI4EKjONL4Bgx2cXT1w1xpeJdmPQ/640?wx_fmt=png"></p></section><p><br></p><p><mp-style-type data-value="10000"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/DlmGDd31fAO5vs0ZDBS2MQ",target="_blank" rel="noopener noreferrer">原文链接</a>
