---
title: "单细胞转录组细胞注释的一个包CellFunTopic不错耶！"
date: 2024-02-20T17:49:10Z
draft: ["false"]
tags: [
  "fetched",
  "东林的扯淡小屋"
]
categories: ["Acdemic"]
---
单细胞转录组细胞注释的一个包CellFunTopic不错耶！ by 东林的扯淡小屋
------
<div><p><img data-galleryid="" data-ratio="0.7657407407407407" data-s="300,640" data-type="png" data-w="1080" data-src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBovN8xqnl6WU7m9KiaJG0ydiazsVQpPLrDufbqMCQMnbpH7kvYh0ica8dC7PSt1nrzHdvmvgBID0aAyw/640?wx_fmt=png" src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBovN8xqnl6WU7m9KiaJG0ydiazsVQpPLrDufbqMCQMnbpH7kvYh0ica8dC7PSt1nrzHdvmvgBID0aAyw/640?wx_fmt=png"></p><p>https://github.com/compbioNJU/CellFunTopic</p><p><br></p><p>先安装：<br></p><section><ul><li><li></ul><pre data-lang="php"><code><span><span>if</span> (!requireNamespace(<span>"devtools"</span>, quietly = <span>TRUE</span>)) install.packages(<span>"devtools"</span>)</span></code><code><span>devtools::install_github(<span>"compbioNJU/CellFunTopic"</span>)</span></code></pre></section><p><img data-galleryid="" data-ratio="0.4603174603174603" data-s="300,640" data-type="png" data-w="945" data-src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBowc2Vyu1eCWwymS7IiaastA0ibia6vhbaOY9c9ag9NRSE3SNLTb2WGD0OWrAYB4Sbcvu3t4MvtFDwWA/640?wx_fmt=png" src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBowc2Vyu1eCWwymS7IiaastA0ibia6vhbaOY9c9ag9NRSE3SNLTb2WGD0OWrAYB4Sbcvu3t4MvtFDwWA/640?wx_fmt=png"></p><p>进入官方文档学习：</p><p><img data-galleryid="" data-ratio="0.5638888888888889" data-s="300,640" data-type="png" data-w="1080" data-src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBovN8xqnl6WU7m9KiaJG0ydiar2l3J0zziamEQvKowBYrU6sdDiaplthOq1pRPfLZdrKpAJR2DgQNErog/640?wx_fmt=png" src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBovN8xqnl6WU7m9KiaJG0ydiar2l3J0zziamEQvKowBYrU6sdDiaplthOq1pRPfLZdrKpAJR2DgQNErog/640?wx_fmt=png"></p><p><img data-galleryid="" data-ratio="0.5638888888888889" data-s="300,640" data-type="png" data-w="1080" data-src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBowc2Vyu1eCWwymS7IiaastARCNNibYan0UKricicYOAibAlSZGVZicszZQfCfT4iaGgeOxVlgk1QAMVOZYA/640?wx_fmt=png" src="https://mmbiz.qpic.cn/mmbiz_png/kZ1wdgAscBowc2Vyu1eCWwymS7IiaastARCNNibYan0UKricicYOAibAlSZGVZicszZQfCfT4iaGgeOxVlgk1QAMVOZYA/640?wx_fmt=png"></p><p>https://compbionju.github.io/CellFunTopic/articles/CellFunTopic.html</p><section><ul><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li><li></ul><pre data-lang="php"><code><span><span>#The CellFunTopic package provides a convenient workflow for data pre-processing (quality control, normalization, dimension reduction, clustering, differential expression analysis, etc.) by integrating methods of Seurat package.</span></span></code><code><span><br></span></code><code><span>library(CellFunTopic)</span></code><code><span><span># Load in the data</span></span></code><code><span><span># CellFunTopic allows various types of input, including CellRanger outputs, raw gene expression matrix, and popular R/python objects used for single cell analysis such as SingleCellExperiment, Seurat, CellDataSet, AnnData. CellFunTopic will transform different input data into a Seurat object.</span></span></code><code><span><span># </span></span></code><code><span><span># Let us use pbmc3k.final as a toy example.</span></span></code><code><span><br></span></code><code><span>data(<span>'pbmc3k.final'</span>, package = <span>"pbmc3k.SeuratData"</span>)</span></code><code><span>dim(pbmc3k.<span>final</span>)</span></code><code><span><br></span></code><code><span>SeuratObj &lt;- readData(data = pbmc3k.<span>final</span>, type = <span>'Seurat'</span>, species = <span>"Homo sapiens"</span>)</span></code><code><span>rm(pbmc3k.<span>final</span>); gc()</span></code><code><span>Standard pre-processing workflow</span></code><code><span>Then we can perform pre-processing conveniently. <span>If</span> users provide a pre-processed data, this step can be skipped.</span></code><code><span><br></span></code><code><span>SeuratObj &lt;- CalMTpercent(SeuratObj, by = <span>"use_internal_data"</span>)</span></code><code><span>SeuratObj &lt;- QCfun(SeuratObj, plot = F) <span># set plot = T if you want to get the QC reports.</span></span></code><code><span>SeuratObj &lt;- RunSeurat(SeuratObj, nPCs = <span>10</span>, resolution = <span>1</span>, plot = <span>FALSE</span>)</span></code><code><span>unique(Seurat::Idents(SeuratObj)) <span># see how many clusters we got.</span></span></code><code><span><span># Gene Set Enrichment Analysis (GSEA)</span></span></code><code><span><span># Then we perform Gene Set Enrichment Analysis (GSEA) on the Seurat object. CellFunTopic provides a encapsulated function for GSEA with different databases such as GO (Gene Ontology), KEGG, Reactome, MSigDb, WikiPathways, DO, NCG, DGN.</span></span></code><code><span><br></span></code><code><span>SeuratObj &lt;- RunGSEA(SeuratObj, by = <span>'GO'</span>)</span></code><code><span><br></span></code><code><span><span># save object for later use.</span></span></code><code><span>save(SeuratObj, file = <span>"SeuratObj.RData"</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span>library(CellFunTopic)</span></code><code><span><span># First, please make sure that you have previously performed the pre-processing and GSEA steps, see Pre-processing.</span></span></code><code><span><span># </span></span></code><code><span><span># CellFunTopic provides a variety of meaningful visualization methods of GSEA Result, facilitating functional annotation of cell clusters in single cell data. What’s more, the visualization can be explored interactively in the built-in shiny app, see Visualize in Built-in Shiny APP.</span></span></code><code><span><br></span></code><code><span><span># Heatmap of GSEA result</span></span></code><code><span>gseaHeatmap(SeuratObj, by = <span>"GO"</span>, toshow = <span>"-logFDR"</span>, topPath = <span>5</span>, colour = <span>"Greens"</span>, fontsize_row = <span>7</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># Heatmap showing unique and shared pathways of clusters.</span></span></code><code><span>pathway_unique_shared(SeuratObj, by = <span>"GO"</span>, toshow = <span>"-logFDR"</span>, topPath = <span>5</span>, colour = <span>"Greens"</span>, fontsize_row = <span>7</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># Helps to infer the relationship between clusters. Link width shows Pearson correlation or Jaccard coefficient between clusters, calculated with GSEA result. Node size indicates cell number of each cluster.</span></span></code><code><span>clustercorplot(SeuratObj, by = <span>"GO"</span>, link_threshold=<span>0.3</span>, vertex.label.cex=<span>1</span>, weight.scale=T)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># clustercorplot_jaccard(SeuratObj, by = "GO", link_threshold=0.3, vertex.label.cex=1, weight.scale=T)</span></span></code><code><span><span># show relationship between clusters and pathways</span></span></code><code><span>hierarchyplot_tree(SeuratObj, by = <span>"GO"</span>, topaths = <span>5</span>, cluster_cutree_k = <span>6</span>, pathway_cutree_k = <span>10</span>,</span></code><code><span>                   vertex.label.cex=<span>0.5</span>, edge.max.width=<span>1</span>, vertex.size.cex=<span>0.7</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># Scatter plot showing pathway enrichment score on cell map</span></span></code><code><span>pathwayScatterplot(SeuratObj, by = <span>"GO"</span>, reduction = <span>"umap"</span>, pathwayID = <span>"GO:0030217"</span>, pointsize = <span>0.5</span>, label = F)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># Link width shows number of intersection of pathways between clusters</span></span></code><code><span>circleplot(SeuratObj, by = <span>"GO"</span>, pvaluecutoff = <span>0.001</span>, link_threshold = <span>10</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># show embedded histogram or pie chart on UMAP/TSNE plot</span></span></code><code><span>embeddedplot(SeuratObj, type = <span>"pie"</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span>embeddedplot(SeuratObj, topaths = <span>1</span>, reduction = <span>"umap"</span>, type = <span>"hist"</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># Network of cosine similarity between terms</span></span></code><code><span>cosine_network_term(SeuratObj, cosine_cal_by = <span>"GSEA result"</span>, pie_by = <span>"GSEA result"</span>, GSEA_by = <span>"GO"</span>,</span></code><code><span>                    topn = <span>10</span>, layout = <span>"fr"</span>, cos_sim_thresh = <span>0.9</span>, radius = <span>0.1</span>, text_size = <span>2</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># Network showing cosine similarity between clusters according to GSEA result</span></span></code><code><span>Cosine_networkByGSEA(SeuratObj, layout=igraph::layout_with_fr, cos_sim_thresh=<span>0.8</span>, p.adjust_thresh=<span>0.05</span>, SEED=<span>123</span>, node.cex=<span>6</span>, width_range=c(<span>0.8</span>, <span>4</span>), vertex.label.dist=<span>0.5</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># Hierarchical edge bundling plots helps visualizing correlation or similarity between clusters</span></span></code><code><span>SeuratObj$group &lt;- <span>'majorcelltype'</span> <span># edge_bundling_GSEA need major group information</span></span></code><code><span>edge_bundling_GSEA(SeuratObj, link_threshold=<span>0.7</span>, p.adjust_thresh=<span>0.05</span>, link_width=<span>0.9</span>, method=<span>'cosine similarity'</span>, node.by=<span>'seurat_clusters'</span>, group.by=<span>"group"</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># show pathways and genes in chord diagram</span></span></code><code><span>GOcircleplot(SeuratObj, genes=<span>NULL</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># boxplot showing enrichment score of child or parent GOs of specific GO</span></span></code><code><span>GOboxplot(SeuratObj, goid = <span>"GO:0030217"</span>, type = <span>"child"</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span>library(CellFunTopic)</span></code><code><span>library(magrittr)</span></code><code><span><span># First, please make sure that you have previously performed the pre-processing and GSEA steps, see Pre-processing.</span></span></code><code><span><span># </span></span></code><code><span><span># Topic modeling</span></span></code><code><span><span># Latent Dirichlet Allocation (LDA) is a Bayesian mixture model, initially developed for the analysis of text documents. CellFunTopic provides a method to fit an LDA model to the clusters-by-pathways matrix derived from GSEA, where each cluster is considered as a document, pathway enrichment score(log-transformed) as a token in the document, and each pathway is considered as a word. CellFunTopic infers the topic probability distribution by decomposing the clusters-by-pathways matrix into clusters-by-topics (θ) and topics-by-pathways (ϕ) matrices, that is, (1) the probability distributions over the topics for each cluster in the dataset (θ) and (2) the probability distributions over the pathways for each topic (ϕ). These distributions indicate, respectively, how important a regulatory topic is for a cluster (θ), and how important pathways are for the regulatory topic (ϕ). We assume that each cluster can be represented as a mixture of latent topics, which reveal cellular programs shared across cell types or exclusive to a particular cell type.</span></span></code><code><span><span># </span></span></code><code><span><span># Actually, CellFunTopic provides two metrics, log-likelihood and perplexity, to determine the optimal number of topics. But we recommend a customized number of topics such as the number of cell types in scRNA-seq data, in order to improve biological interpretability.</span></span></code><code><span><br></span></code><code><span>k &lt;- length(unique(Seurat::Idents(SeuratObj)))</span></code><code><span><span># if k=NULL, the optimal number of topics will be calculated automatically according to log-likelihood and perplexity.</span></span></code><code><span>SeuratObj &lt;- runLDA(SeuratObj, by = <span>"GO"</span>, k = k, method = <span>"VEM"</span>, SEED = <span>1234</span>, plot = F)</span></code><code><span><span># save object for later use.</span></span></code><code><span>save(SeuratObj, file = <span>"SeuratObj.RData"</span>)</span></code><code><span>Visualization of topic modelling results</span></code><code><span>CellFunTopic provides a few methods to visualize topic modelling results.</span></code><code><span><br></span></code><code><span>ldaOut &lt;- SeuratObj@misc$ldaOut</span></code><code><span>betaDF &lt;- tidytext::tidy(ldaOut, matrix = <span>"beta"</span>)</span></code><code><span>pws &lt;- ID2Description(SeuratObj, by = <span>"GO"</span>)</span></code><code><span>betaDF %&lt;&gt;% dplyr::mutate(descrip=unname(pws[term]))</span></code><code><span>gammaDF &lt;- tidytext::tidy(ldaOut, matrix = <span>"gamma"</span>)</span></code><code><span><span># Show topic distribution on UMAP/TSNE cell map</span></span></code><code><span>topicProb(SeuratObj, reduction=<span>"umap"</span>, topic=<span>3</span>, pointSize=<span>0.1</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># echart4r network showing top terms of topics with tooltips showing description of pathway ID</span></span></code><code><span>topicNW3(betaDF, topn=<span>10</span>, pws = pws)</span></code><code><span><span># Heatmap showing probability between topics and clusters</span></span></code><code><span>cluster_topic_hmp(ldaOut)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># Sankey diagram showing the best assigned topic of each cluster</span></span></code><code><span>plot_sankey(gammaDF, topn=<span>1</span>)</span></code><code><span><br></span></code><code><span><span># word cloud of top terms of topic</span></span></code><code><span>wordcloud_topic(betaDF, pws, topic=<span>3</span>, topn=<span>20</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># 3D word cloud of top terms of topic</span></span></code><code><span>wordcloud_topic_3D(betaDF, pws, topic=<span>3</span>, topn=<span>20</span>)</span></code><code><span><span># Heatmap showing cosine similarity between topics</span></span></code><code><span>cosineheatmap(ldaOut)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># Heatmap showing cosine similarity between clusters</span></span></code><code><span>cosinehmp_cluster(ldaOut)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># Network showing cosine similarity between clusters</span></span></code><code><span>cosine_network_cluster(ldaOut, layout=<span>"fr"</span>, cos_sim_thresh=<span>0.02</span>, SEED=<span>123</span>, radius=<span>0.12</span>, width_range=c(<span>0.1</span>, <span>0.8</span>))</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># Network of cosine similarity between terms</span></span></code><code><span>cosine_network_term(SeuratObj, cosine_cal_by = <span>"Topic modeling"</span>, pie_by = <span>"Topic modeling"</span>, GSEA_by = <span>"GO"</span>,</span></code><code><span>                    topn = <span>10</span>, layout = <span>"fr"</span>, cos_sim_thresh = <span>0.8</span>, radius = <span>0.1</span>, text_size = <span>2</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># UMAP on cluster-topic probability matrix</span></span></code><code><span>umap_cluster(ldaOut)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># Show top terms of each topic</span></span></code><code><span>Topterms_Topic(betaDF, Topic = <span>3</span>, topn = <span>10</span>, text.size = <span>3</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># barplot to show top terms of every topic</span></span></code><code><span>hist_topic_term(betaDF, topics=c(<span>1</span>,<span>2</span>,<span>3</span>,<span>4</span>), topn=<span>20</span>, axis.text.y.size=<span>3</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># barplot to show probability of assigned topics in clusters</span></span></code><code><span>hist_cluster_topic(gammaDF, clusters=head(Seurat::Idents(SeuratObj)))</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span>library(CellFunTopic)</span></code><code><span>library(magrittr)</span></code><code><span><span># Reference topic models</span></span></code><code><span><span># CellFunTopic provides built-in reference topic models for users to take advantage of large landscape datasets such as the human cell landscape (HCL) and the mouse cell atlas (MCA).</span></span></code><code><span><span># </span></span></code><code><span><span># ?HCL_ldaOut</span></span></code><code><span><span># ?MCA_ldaOut</span></span></code><code><span><span># CellFunTopic also provides methods for users to make their own reference topic models.</span></span></code><code><span><span># </span></span></code><code><span><span># Make a reference topic model and transfer</span></span></code><code><span><span># Let us take pbmc3k.final as example to show how to make a reference topic model and transfer. We take the original one as the reference data, and re-cluster it as the query data.</span></span></code><code><span><span># </span></span></code><code><span><span># Make a reference topic model out of pbmc3k.final with the ground-truth cell type.</span></span></code><code><span><br></span></code><code><span>data(<span>'pbmc3k.final'</span>, package = <span>"pbmc3k.SeuratData"</span>)</span></code><code><span>reference_SeuratObj &lt;- readData(data = pbmc3k.<span>final</span>, type = <span>'Seurat'</span>, species = <span>"Homo sapiens"</span>)</span></code><code><span>reference_SeuratObj &lt;- DetectGeneIDtype(reference_SeuratObj)</span></code><code><span>SeuratObj.markers &lt;- Seurat::FindAllMarkers(reference_SeuratObj, only.pos = <span>TRUE</span>, min.pct = <span>0.0001</span>, logfc.threshold = <span>0.0001</span>, <span>return</span>.thresh=<span>0.9</span>)</span></code><code><span>slot(object = reference_SeuratObj, name = <span>'misc'</span>)[[<span>"Allmarkers"</span>]] &lt;- SeuratObj.markers</span></code><code><span>reference_SeuratObj &lt;- RunGSEA(reference_SeuratObj, by = <span>'GO'</span>)</span></code><code><span>k &lt;- length(unique(Seurat::Idents(reference_SeuratObj)))</span></code><code><span>reference_SeuratObj &lt;- runLDA(reference_SeuratObj, by = <span>"GO"</span>, k = k, method = <span>"VEM"</span>, SEED = <span>1234</span>, plot = F)</span></code><code><span>table(Seurat::Idents(reference_SeuratObj))</span></code><code><span><span>## </span></span></code><code><span><span>##  Naive CD4 T Memory CD4 T   CD14+ Mono            B        CD8 T FCGR3A+ Mono </span></span></code><code><span><span>##          697          483          480          344          271          162 </span></span></code><code><span><span>##           NK           DC     Platelet </span></span></code><code><span><span>##          155           32           14</span></span></code><code><span>Re-analyze <span>and</span> re-cluster pbmc3k.<span>final</span> <span>as</span> the query data, each cell owns a <span>new</span> cluster identity.</span></code><code><span><br></span></code><code><span>SeuratObj &lt;- readData(data = pbmc3k.<span>final</span>, type = <span>'Seurat'</span>, species = <span>"Homo sapiens"</span>)</span></code><code><span>SeuratObj &lt;- CalMTpercent(SeuratObj, by = <span>"use_internal_data"</span>)</span></code><code><span>SeuratObj &lt;- QCfun(SeuratObj, plot = F)</span></code><code><span>SeuratObj &lt;- RunSeurat(SeuratObj, nPCs = <span>10</span>, resolution = <span>1</span>, plot = <span>FALSE</span>)</span></code><code><span>SeuratObj &lt;- RunGSEA(SeuratObj, by = <span>'GO'</span>)</span></code><code><span><span># check the new clustering</span></span></code><code><span>table(Seurat::Idents(SeuratObj))</span></code><code><span><span>## </span></span></code><code><span><span>##   0   1   2   3   4   5   6   7   8   9 </span></span></code><code><span><span>## 537 460 325 252 220 159 143 141 140 134</span></span></code><code><span>Transfer topic model from reference data to query data.</span></code><code><span><br></span></code><code><span>ref_ldaOUt &lt;- reference_SeuratObj@misc$ldaOut</span></code><code><span>SeuratObj &lt;- transfer_LDA(query_SeuratObj = SeuratObj, ref_ldaOUt = ref_ldaOUt, by = <span>"GO"</span>)</span></code><code><span>Visualize transferred model</span></code><code><span>dist &lt;- SeuratObj@misc$dist</span></code><code><span>gammaDF &lt;- reshape2::melt(dist$topics) %&gt;% magrittr::set_colnames(c(<span>"document"</span>, <span>"topic"</span>, <span>"gamma"</span>))</span></code><code><span>plot_sankey(gammaDF, topn=<span>1</span>)</span></code><code><span><br></span></code><code><span>mm &lt;- dist$topics</span></code><code><span>colnames(mm) &lt;- paste0(<span>'Topic '</span>, colnames(mm))</span></code><code><span>pheatmap::pheatmap(mm, cluster_rows = T, cluster_cols = T, angle_col = <span>"0"</span>, main = <span>"probability between topic and cluster"</span>,</span></code><code><span>                   color = colorRampPalette(c(<span>'white'</span>, RColorBrewer::brewer.pal(n=<span>9</span>,name=<span>"OrRd"</span>)))(<span>100</span>),</span></code><code><span>                   fontsize_row = <span>8</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span>Visualize correlation <span>or</span> comparison</span></code><code><span>CellFunTopic provides a few methods to visualize correlation <span>or</span> comparison between reference data <span>and</span> query data. <span>For</span> better comparison, we identify the cell types of query data based on ground-truth identity of each cell.</span></code><code><span><br></span></code><code><span>mm &lt;- table(SeuratObj$seurat_annotations, SeuratObj$seurat_clusters)</span></code><code><span>nn &lt;- setNames(rownames(mm)[apply(mm, <span>2</span>, which.max)], colnames(mm))</span></code><code><span>rownames(dist$topics) &lt;- nn[rownames(dist$topics)]</span></code><code><span><br></span></code><code><span>sankey_comparison(ldaOut=ref_ldaOUt, dist=dist, a_prefix=<span>"reference_"</span>, b_prefix=<span>"query_"</span>, height=<span>800</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span>cosine_network_cluster2(ldaOut=ref_ldaOUt, dist=dist, a_prefix=<span>"reference_"</span>, b_prefix=<span>"query_"</span>,</span></code><code><span>                        layout=<span>"fr"</span>, cos_sim_thresh=<span>0.5</span>, radius=<span>0.1</span>, text_size = <span>3</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span>cosine_heatmap2(ldaOut=ref_ldaOUt, dist=dist, a_prefix=<span>"reference_"</span>, b_prefix=<span>"query_"</span>, method=<span>"cosine"</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span>cosine_hive(ldaOut=ref_ldaOUt, dist=dist, a_prefix=<span>"reference_"</span>, b_prefix=<span>"query_"</span>)</span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># </span></span></code><code><span><span># </span></span></code><code><span><span># CellFunTopic provides methods for rapid, automated cell-type annotation across datasets without dependence on marker genes, reducing labor intensive and time-consuming work of manual markers collection and manual cell-type annotation.</span></span></code><code><span><span># </span></span></code><code><span><span># Predict cell type of query data based on reference data</span></span></code><code><span>library(CellFunTopic)</span></code><code><span><span>#Here, we take pbmc3k.final as the reference data:</span></span></code><code><span>  </span></code><code><span>data(<span>'pbmc3k.final'</span>, package = <span>"pbmc3k.SeuratData"</span>)</span></code><code><span>unique(Seurat::Idents(pbmc3k.<span>final</span>))</span></code><code><span><span>## [1] Memory CD4 T B            CD14+ Mono   NK           CD8 T       </span></span></code><code><span><span>## [6] Naive CD4 T  FCGR3A+ Mono DC           Platelet    </span></span></code><code><span><span>## 9 Levels: Naive CD4 T Memory CD4 T CD14+ Mono B CD8 T FCGR3A+ Mono NK ... Platelet</span></span></code><code><span>To make things easier, we re-analyze <span>and</span> re-cluster pbmc3k.<span>final</span> <span>as</span> the query data, each cell owns a <span>new</span> cluster identity.</span></code><code><span><br></span></code><code><span>SeuratObj &lt;- readData(data = pbmc3k.<span>final</span>, type = <span>'Seurat'</span>, species = <span>"Homo sapiens"</span>)</span></code><code><span>SeuratObj &lt;- CalMTpercent(SeuratObj, by = <span>"use_internal_data"</span>)</span></code><code><span>SeuratObj &lt;- QCfun(SeuratObj, plot = F)</span></code><code><span>SeuratObj &lt;- RunSeurat(SeuratObj, nPCs = <span>10</span>, resolution = <span>1</span>, plot = <span>FALSE</span>)</span></code><code><span>SeuratObj &lt;- RunGSEA(SeuratObj, by = <span>'GO'</span>)</span></code><code><span><span># check the new clustering</span></span></code><code><span>table(Seurat::Idents(SeuratObj))</span></code><code><span><span>## </span></span></code><code><span><span>##   0   1   2   3   4   5   6   7   8   9 </span></span></code><code><span><span>## 537 460 325 252 220 159 143 141 140 134</span></span></code><code><span><span>#It’s convenient to transfer annotation from reference to query data:</span></span></code><code><span>  </span></code><code><span>  df &lt;- predictFun(query_SeuratObj = SeuratObj, reference_SeuratObj = pbmc3k.<span>final</span>,</span></code><code><span>                   group_by = <span>'seurat_annotations'</span>, cluster_by = <span>'seurat_clusters'</span>,</span></code><code><span>                   species = <span>"Homo sapiens"</span>, by = <span>'GO'</span>, k = <span>NULL</span>, LDAmethod = <span>"VEM"</span>)</span></code><code><span><span>## Calculating differentially expressed genes for clusters in reference data......</span></span></code><code><span><span>## Calculating cluster 0</span></span></code><code><span><span>## Calculating cluster 1</span></span></code><code><span><span>## Calculating cluster 2</span></span></code><code><span><span>## Calculating cluster 3</span></span></code><code><span><span>## Calculating cluster 4</span></span></code><code><span><span>## Calculating cluster 5</span></span></code><code><span><span>## Calculating cluster 6</span></span></code><code><span><span>## Calculating cluster 7</span></span></code><code><span><span>## Calculating cluster 8</span></span></code><code><span><span>## Performing GSEA on reference data......</span></span></code><code><span><span>## 'select()' returned 1:many mapping between keys and columns</span></span></code><code><span><span>## Performing topic modelling on reference data......</span></span></code><code><span><span>## training a svm classifier......</span></span></code><code><span><span>## 载入需要的程辑包：lattice</span></span></code><code><span><span>## 载入需要的程辑包：ggplot2</span></span></code><code><span><span>## predicting cell types of the query data......</span></span></code><code><span>df  <span># prediction result</span></span></code><code><span><span>##    query   prediction</span></span></code><code><span><span>## 1      0  Naive CD4 T</span></span></code><code><span><span>## 2      1  Naive CD4 T</span></span></code><code><span><span>## 3      2            B</span></span></code><code><span><span>## 4      3   CD14+ Mono</span></span></code><code><span><span>## 5      4   CD14+ Mono</span></span></code><code><span><span>## 6      5        CD8 T</span></span></code><code><span><span>## 7      6  Naive CD4 T</span></span></code><code><span><span>## 8      7           NK</span></span></code><code><span><span>## 9      8 FCGR3A+ Mono</span></span></code><code><span><span>## 10     9        CD8 T</span></span></code><code><span>To obtain the accuracy of prediction in this <span>case</span>, we identify the cell types of each cluster in query data based on ground-truth identity of each cell.</span></code><code><span><br></span></code><code><span>mm &lt;- table(SeuratObj$seurat_annotations, SeuratObj$seurat_clusters)</span></code><code><span>nn &lt;- setNames(rownames(mm)[apply(mm, <span>2</span>, which.max)], colnames(mm))</span></code><code><span>nn[df$query] <span># cell types of query data based on identity of each cell</span></span></code><code><span><span>##              0              1              2              3              4 </span></span></code><code><span><span>## "Memory CD4 T"  "Naive CD4 T"            "B"   "CD14+ Mono"   "CD14+ Mono" </span></span></code><code><span><span>##              5              6              7              8              9 </span></span></code><code><span><span>##        "CD8 T"  "Naive CD4 T"           "NK" "FCGR3A+ Mono"        "CD8 T"</span></span></code><code><span><span># check the accuracy of prediction</span></span></code><code><span>caret::confusionMatrix(data = <span>as</span>.factor(df$prediction), reference = <span>as</span>.factor(nn[df$query]))$overall[[<span>"Accuracy"</span>]]</span></code><code><span><span>## [1] 0.9</span></span></code><code><span>To improve accuracy of prediction, users can oversample the reference data. In this <span>case</span>, we skip this step because it can be time-consuming.</span></code><code><span><br></span></code><code><span>reference_SeuratObj &lt;- oversample_ref(reference_SeuratObj, number_clusters = <span>5</span>,</span></code><code><span>                                      group_by = <span>'cellType'</span>, cluster_by = <span>'seurat_clusters'</span>,</span></code><code><span>                                      species = <span>"Homo sapiens"</span>, by = <span>'GO'</span>, k = <span>NULL</span>, method = <span>"VEM"</span>)</span></code><code><span>predictFun(query_SeuratObj, reference_SeuratObj, group_by = <span>'seurat_annotations'</span>, cluster_by = <span>'seurat_clusters'</span>,</span></code><code><span>           species = <span>"Homo sapiens"</span>, by = <span>'GO'</span>, k = <span>NULL</span>, LDAmethod = <span>"VEM"</span>)</span></code><code><span><span># if true labels of query data are provided, you can use caret::confusionMatrix to inspect the confusion matrix and accuracy</span></span></code><code><span>caret::confusionMatrix(data = prediction , reference = true_label)</span></code><code><span>caret::confusionMatrix(data = prediction , reference = true_label)$overall[[<span>"Accuracy"</span>]]</span></code><code><span>Contents</span></code><code><span>Predict cell type of query data based on reference data</span></code><code><span>Developed by Shanni Cao.</span></code><code><span><br></span></code><code><span><br></span></code><code><span><span># First, please make sure that you have previously performed the pre-processing, GSEA and topic-modelling steps, see Pre-processing and topic-modelling.</span></span></code><code><span><span># </span></span></code><code><span><span># The GSEA and topic modelling results can be explored interactively in the built-in Shiny web application.</span></span></code><code><span><br></span></code><code><span>library(CellFunTopic)</span></code><code><span>visualize_in_shiny(SeuratObj)</span></code><code><span><span># It should be noted that an object named "SeuratObj" is needed in your global environment.</span></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code><code><span><br></span></code></pre></section><p><br></p><p><mp-style-type data-value="3"></mp-style-type></p></div>  
<hr>
<a href="https://mp.weixin.qq.com/s/DfaHcAOm9UtfonbxXRHqiQ",target="_blank" rel="noopener noreferrer">原文链接</a>
